diff --git a/device.c b/device.c
index 04bbef1..e6d63aa 100644
--- a/device.c
+++ b/device.c
@@ -357,7 +357,7 @@ bool cScDevices::Initialize(void)
   *vdr_nci=0; *vdr_ud=vdr_save_ud;
 
   int i, found=0;
-  for(i=0; i<MAXDVBDEVICES; i++) {
+  for(i=0; i<MAXDEVICES; i++) {
     if(UseDevice(NextCardIndex())) {
       char name[128];
       cScDevices::DvbName(DEV_DVB_FRONTEND,i,0,name,sizeof(name));
@@ -382,7 +382,7 @@ bool cScDevices::Initialize(void)
       }
     else NextCardIndex(1);
     }
-  NextCardIndex(MAXDVBDEVICES-i);
+  NextCardIndex(MAXDEVICES-i);
   if(found>0) PRINTF(L_GEN_INFO,"captured %d video device%s",found,found>1 ? "s" : "");
   else PRINTF(L_GEN_INFO,"no DVB device captured");
   return found>0;
@@ -411,7 +411,7 @@ void cScDevices::Shutdown(void)
 
 void cScDevices::SetForceBudget(int n)
 {
-   if(n>=0 && n<MAXDVBDEVICES) budget|=(1<<n);
+   if(n>=0 && n<MAXDEVICES) budget|=(1<<n);
 }
 
 bool cScDevices::ForceBudget(int n)
