diff -Naur umtp-responder-e140e559edc914976e10bcab13d37a56c4c505b2/conf/umtprd.conf umtp-responder-e140e559edc914976e10bcab13d37a56c4c505b2-2/conf/umtprd.conf
--- umtp-responder-e140e559edc914976e10bcab13d37a56c4c505b2/conf/umtprd.conf	2021-11-19 07:59:05.543099366 +0100
+++ umtp-responder-e140e559edc914976e10bcab13d37a56c4c505b2-2/conf/umtprd.conf	2021-11-19 08:02:13.298186811 +0100
@@ -6,14 +6,12 @@
 # Loop / daemon mode
 # Set to 1 to don't shutdown uMTPrd when the link is disconnected.
 
-loop_on_disconnect 0
+loop_on_disconnect 1
 
 # storage command : Create add a storage entry point. Up to 16 entry points supported
 # Syntax : storage "PATH" "NAME"
 
-storage "/"      "root folder" "rw"
-storage "/home"  "home folder" "ro"
-storage "/www"   "www folder" "ro,notmounted"
+storage "/flash"  "SDCard Root" "rw"
 
 #
 # Uncomment the following line if you want to 
@@ -25,15 +23,15 @@
 
 # Set the USB manufacturer string
 
-manufacturer "Viveris Technologies"
+manufacturer "Nintendo"
 
 # Set the USB Product string
 
-product "Viveris Technologies"
+product "Switch(@DISTRO@)"
 
 # Set the USB Serial number string
 
-serial "01234567"
+serial "1"
 
 # Set the MTP firmware version
 
@@ -84,10 +82,10 @@
 
 usb_functionfs_mode 0x1
 
-usb_dev_path   "/dev/ffs-mtp/ep0"
-usb_epin_path  "/dev/ffs-mtp/ep1"
-usb_epout_path "/dev/ffs-mtp/ep2"
-usb_epint_path "/dev/ffs-mtp/ep3"
+usb_dev_path   "/dev/ffs-umtp/ep0"
+usb_epin_path  "/dev/ffs-umtp/ep1"
+usb_epout_path "/dev/ffs-umtp/ep2"
+usb_epint_path "/dev/ffs-umtp/ep3"
 
 usb_max_packet_size 0x200
 
