diff -Naur uxlaunch-0.46/xserver.c uxlaunch-0.46.patch/xserver.c
--- uxlaunch-0.46/xserver.c	2010-03-23 09:47:54.621981367 +0100
+++ uxlaunch-0.46.patch/xserver.c	2010-03-23 09:53:39.703855113 +0100
@@ -231,13 +231,15 @@
 		execl(xserver, xserver,  displayname,
 		      "-nolisten", "tcp", "-dpi", "120", "-noreset",
 		      "-auth", xauth_cookie_file,
-		      vt, "-config", xconfig, NULL);
+		      vt, "-config", xconfig, "-nocursor", "-nohwaccess",
+		      NULL);
 	} else {
 		execl(xserver, xserver,  displayname,
 		      "-nolisten", "tcp", "-dpi", "120", "-noreset",
 		      "-auth", user_xauth_path,
 		      "-logfile", xorg_log,
-		      vt, "-config", xconfig, NULL);
+		      vt, "-config", xconfig, "-nocursor", "-nohwaccess",
+		      NULL);
 	}
 	exit(0);
 }
