diff -Naur flac-1.2.1-old/configure.in flac-1.2.1-new/configure.in
--- flac-1.2.1-old/configure.in	2007-09-13 08:48:42.000000000 -0700
+++ flac-1.2.1-new/configure.in	2010-07-04 17:16:02.000000000 -0700
@@ -297,11 +297,13 @@
 if test "x$debug" = xtrue; then
 	CPPFLAGS="-DDEBUG $CPPFLAGS"
 	CFLAGS="-g $CFLAGS"
+	CXXFLAGS="-g $CXXFLAGS"
 else
 	CPPFLAGS="-DNDEBUG $CPPFLAGS"
 	if test "x$GCC" = xyes; then
 		CPPFLAGS="-DFLaC__INLINE=__inline__ $CPPFLAGS"
 		CFLAGS="-O3 -funroll-loops -finline-functions -Wall -W -Winline $CFLAGS"
+		CXXFLAGS="-O3 -funroll-loops -finline-functions -Wall -W -Winline $CXXFLAGS"
 	fi
 fi
 
