diff -Naur flac-1.2.1-old/examples/cpp/encode/file/main.cpp flac-1.2.1-new/examples/cpp/encode/file/main.cpp
--- flac-1.2.1-old/examples/cpp/encode/file/main.cpp	2007-09-13 08:58:03.000000000 -0700
+++ flac-1.2.1-new/examples/cpp/encode/file/main.cpp	2008-01-08 17:25:20.000000000 -0800
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
 
