diff -Naur flac-1.2.1-old/configure.in flac-1.2.1-new/configure.in
--- flac-1.2.1-old/configure.in	2007-09-13 08:48:42.000000000 -0700
+++ flac-1.2.1-new/configure.in	2010-08-17 20:51:01.000000000 -0700
@@ -21,6 +21,8 @@
 AC_INIT(src/flac/main.c)
 AM_INIT_AUTOMAKE(flac, 1.2.1)
 
+AC_CONFIG_MACRO_DIR([m4])
+
 # Don't automagically regenerate autoconf/automake generated files unless
 # explicitly requested.  Eases autobuilding -mdz
 AM_MAINTAINER_MODE
diff -Naur flac-1.2.1-old/Makefile.am flac-1.2.1-new/Makefile.am
--- flac-1.2.1-old/Makefile.am	2007-09-16 12:29:29.000000000 -0700
+++ flac-1.2.1-new/Makefile.am	2010-08-17 20:51:27.000000000 -0700
@@ -30,6 +30,8 @@
 
 AUTOMAKE_OPTIONS = foreign 1.7
 
+ACLOCAL_AMFLAGS = -I m4
+
 SUBDIRS = doc include m4 man src examples test build obj
 
 DISTCLEANFILES = libtool-disable-static
