From 3af2830856f4356acb9bf1655adf632759782aac Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 24 Sep 2018 13:38:46 -0700
Subject: [PATCH 09/14] libavcodec: v4l2m2m: fix decoder capabilities

---
 libavcodec/v4l2_m2m_dec.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index 8834b3d6fc..0087de35d0 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -267,8 +267,7 @@ static const AVCodecHWConfigInternal *v4l2_m2m_hw_configs[] = {
                                                          AV_PIX_FMT_NONE}, \
         .bsfs           = bsf_name, \
         .hw_configs     = v4l2_m2m_hw_configs, \
-        .capabilities   = AV_CODEC_CAP_HARDWARE | AV_CODEC_CAP_DELAY, \
-	                      AV_CODEC_CAP_AVOID_PROBING, \
+        .capabilities   = AV_CODEC_CAP_HARDWARE | AV_CODEC_CAP_DELAY | AV_CODEC_CAP_AVOID_PROBING, \
         .wrapper_name   = "v4l2m2m", \
     };
 
-- 
2.24.1

