diff -Naur linux-AMLm3-2.6.34-7fe1265/drivers/amlogic/ump/Makefile.common linux-AMLm3-2.6.34-7fe1265.patch/drivers/amlogic/ump/Makefile.common
--- linux-AMLm3-2.6.34-7fe1265/drivers/amlogic/ump/Makefile.common	2013-05-20 16:41:01.000000000 +0200
+++ linux-AMLm3-2.6.34-7fe1265.patch/drivers/amlogic/ump/Makefile.common	2013-05-20 17:56:59.152997553 +0200
@@ -14,7 +14,7 @@
 	$(UMP_FILE_PREFIX)common/ump_kernel_ref_drv.c
 
 # Get subversion revision number, fall back to 0000 if no svn info is available
-SVN_REV:=$(shell ((svnversion | grep -qv exported && echo -n 'Revision: ' && svnversion) || git svn info | sed -e 's/$$$$/M/' | grep '^Revision: ' || echo ${MALI_RELEASE_NAME}) 2>/dev/null | sed -e 's/^Revision: //')
+SVN_REV:=${PKG_VERSION}
 
 EXTRA_CFLAGS += -DSVN_REV=$(SVN_REV)
 EXTRA_CFLAGS += -DSVN_REV_STRING=\"$(SVN_REV)\"
