diff -Naur linux-AMLm1-2.6.34-1405682/drivers/amlogic/mali/Makefile.common linux-AMLm1-2.6.34-1405682.patch/drivers/amlogic/mali/Makefile.common
--- linux-AMLm1-2.6.34-1405682/drivers/amlogic/mali/Makefile.common	2013-05-26 01:19:02.000000000 +0200
+++ linux-AMLm1-2.6.34-1405682.patch/drivers/amlogic/mali/Makefile.common	2013-05-26 04:27:11.982074443 +0200
@@ -40,7 +40,7 @@
 endif
 
 # Get subversion revision number, fall back to only ${MALI_RELEASE_NAME} if no svn info is available
-SVN_REV := $(shell (cd $(DRIVER_DIR); (svnversion | grep -Eqv "exported|Unversioned directory" && svnversion) || git svn info | grep '^Revision: '| sed -e 's/^Revision: //' ) 2>/dev/null )
+SVN_REV := ${PKG_VERSION}
 ifeq ($(SVN_REV),)
 SVN_REV := $(MALI_RELEASE_NAME)
 else
diff -Naur linux-AMLm1-2.6.34-1405682/drivers/amlogic/ump/Makefile.common linux-AMLm1-2.6.34-1405682.patch/drivers/amlogic/ump/Makefile.common
--- linux-AMLm1-2.6.34-1405682/drivers/amlogic/ump/Makefile.common	2013-05-26 01:18:57.000000000 +0200
+++ linux-AMLm1-2.6.34-1405682.patch/drivers/amlogic/ump/Makefile.common	2013-05-26 04:28:31.050824923 +0200
@@ -14,7 +14,7 @@
 	$(UMP_FILE_PREFIX)common/ump_kernel_ref_drv.c
 
 # Get subversion revision number, fall back to 0000 if no svn info is available
-SVN_REV:=$(shell ((svnversion | grep -qv exported && echo -n 'Revision: ' && svnversion) || git svn info | sed -e 's/$$$$/M/' | grep '^Revision: ' || echo ${MALI_RELEASE_NAME}) 2>/dev/null | sed -e 's/^Revision: //')
+SVN_REV:=${PKG_VERSION}
 
 EXTRA_CFLAGS += -DSVN_REV=$(SVN_REV)
 EXTRA_CFLAGS += -DSVN_REV_STRING=\"$(SVN_REV)\"
