diff -Naur linux-3.2.44/tools/perf/builtin-sched.c linux-3.2.44.patch/tools/perf/builtin-sched.c
--- linux-3.2.44/tools/perf/builtin-sched.c	2013-04-25 21:25:51.000000000 +0200
+++ linux-3.2.44.patch/tools/perf/builtin-sched.c	2013-05-06 03:29:10.327408347 +0200
@@ -14,6 +14,7 @@
 #include "util/debug.h"
 
 #include <sys/prctl.h>
+#include <sys/resource.h>
 
 #include <semaphore.h>
 #include <pthread.h>
