diff -Naur linux-AMLm3-2.6.34-7fe1265/init/main.c linux-AMLm3-2.6.34-7fe1265.patch/init/main.c
--- linux-AMLm3-2.6.34-7fe1265/init/main.c	2013-05-20 16:40:52.000000000 +0200
+++ linux-AMLm3-2.6.34-7fe1265.patch/init/main.c	2013-05-24 19:23:24.658367744 +0200
@@ -896,15 +896,14 @@
 	do_basic_setup();
 
 	/* Open the /dev/console on the rootfs, this should never fail */
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		{
-		       build_console();
-                if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-                                {
-                                printk(KERN_WARNING "Warning: unable to open an initial console.\n");
-                                }
-		//printk(KERN_WARNING "Warning: unable to open an initial console.\n");
-		}
+	char *console = "/dev_console";
+
+	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0) {
+		sys_mknod(console, S_IFCHR|0600, (TTYAUX_MAJOR<<8)|1);
+		if (sys_open(console, O_RDWR, 0) < 0)
+			printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+		sys_unlink(console);
+	}
 
 	(void) sys_dup(0);
 	(void) sys_dup(0);
