--- zlib-1.2.2.orig/configure	Wed Nov 17 15:56:53 2004
+++ zlib-1.2.2/configure	Wed Nov 17 15:57:37 2004
@@ -75,7 +75,7 @@
   CC="$cc"
   SFLAGS=${CFLAGS-"-fPIC -O3"}
   CFLAGS="$cflags"
-  case `(uname -s || echo unknown) 2>/dev/null` in
+  case `echo Linux` in
   Linux | linux | GNU | GNU/*) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"};;
   CYGWIN* | Cygwin* | cygwin* | OS/2* )
              EXE='.exe';;
@@ -95,7 +95,7 @@
 else
   # find system name and corresponding cc options
   CC=${CC-cc}
-  case `(uname -sr || echo unknown) 2>/dev/null` in
+  case `echo Linux` in
   HP-UX*)    SFLAGS=${CFLAGS-"-O +z"}
              CFLAGS=${CFLAGS-"-O"}
 #            LDSHARED=${LDSHARED-"ld -b +vnocompatwarnings"}
