diff -Naur systemd-217/src/libudev/libudev-hwdb.c systemd-217.patch/src/libudev/libudev-hwdb.c
--- systemd-217/src/libudev/libudev-hwdb.c	2014-10-28 14:31:25.105131602 +0100
+++ systemd-217.patch/src/libudev/libudev-hwdb.c	2014-10-31 22:42:05.250433377 +0100
@@ -257,7 +257,7 @@
 }
 
 static const char hwdb_bin_paths[] =
-    "/etc/udev/hwdb.bin\0"
+    "/run/hwdb.bin\0"
     UDEVLIBEXECDIR "/hwdb.bin\0";
 
 
diff -Naur systemd-217/src/udev/udevadm-hwdb.c systemd-217.patch/src/udev/udevadm-hwdb.c
--- systemd-217/src/udev/udevadm-hwdb.c	2014-10-28 14:31:25.105131602 +0100
+++ systemd-217.patch/src/udev/udevadm-hwdb.c	2014-10-31 22:42:38.656510374 +0100
@@ -557,7 +557,7 @@
         };
         const char *test = NULL;
         const char *root = "";
-        const char *hwdb_bin_dir = "/etc/udev";
+        const char *hwdb_bin_dir = "/run";
         bool update = false;
         struct trie *trie = NULL;
         int err, c;
