From d49c0c7bd4a901f0f5b93abffb75a5fd4a1aac26 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 11 Dec 2014 21:00:47 +0200
Subject: [PATCH 7/8] remove nonexistant dependency

---
 units/graphical.target |    1 -
 1 file changed, 1 deletion(-)

diff --git a/units/graphical.target b/units/graphical.target
index a2c4532..6e581c6 100644
--- a/units/graphical.target
+++ b/units/graphical.target
@@ -11,5 +11,4 @@ Documentation=man:systemd.special(7)
 Requires=multi-user.target
 After=multi-user.target
 Conflicts=rescue.target
-Wants=display-manager.service
 AllowIsolate=yes
-- 
1.7.10.4

