diff -Naur lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp
--- lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp	2025-05-25 21:47:43.409242301 +0200
+++ lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp	2025-05-25 21:49:06.865812700 +0200
@@ -36,10 +36,10 @@
     
     switch (video_format) {
         case VIDEO_FORMAT_H264:
-            m_decoder = avcodec_find_decoder(AV_CODEC_ID_H264);
+            const m_decoder = avcodec_find_decoder(AV_CODEC_ID_H264);
             break;
         case VIDEO_FORMAT_H265:
-            m_decoder = avcodec_find_decoder(AV_CODEC_ID_HEVC);
+            const m_decoder = avcodec_find_decoder(AV_CODEC_ID_HEVC);
             break;
     }
     
