diff -Naur lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/libgamestream/mkcert.c lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/libgamestream/mkcert.c
--- lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/libgamestream/mkcert.c	2025-05-25 21:03:39.000000000 +0200
+++ lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/libgamestream/mkcert.c	2025-05-25 21:29:55.333144181 +0200
@@ -41,7 +41,7 @@
     EVP_PKEY *pkey = NULL;
     PKCS12 *p12 = NULL;
 
-    CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
+//    CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
     bio_err = BIO_new_fp(stderr, BIO_NOCLOSE);
 
     OpenSSL_add_all_algorithms();
