diff -Naur lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/build_opus_lakka_switch.sh lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/build_opus_lakka_switch.sh
--- lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/build_opus_lakka_switch.sh	2025-05-25 21:03:39.000000000 +0200
+++ lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/build_opus_lakka_switch.sh	2025-05-25 21:16:20.935034194 +0200
@@ -6,13 +6,13 @@
 ./autogen.sh
 
 # Fix TOOLCHAIN path!
-TOOLCHAIN=/home/rock88/Documents/Lakka-LibreELEC/build.Lakka-L4T.aarch64-2.2-devel/toolchain
+#TOOLCHAIN=/home/rock88/Documents/Lakka-LibreELEC/build.Lakka-L4T.aarch64-2.2-devel/toolchain
 CFLAGS='-mcpu=cortex-a57+crypto+crc+fp+simd -mabi=lp64 -Wno-psabi -mtune=cortex-a57 -march=armv8-a+crypto+crc+fp+simd -fomit-frame-pointer -Wall -pipe -O2 -mcpu=cortex-a57 -fPIC' \
 ASFLAGS='-mcpu=cortex-a57+crypto+crc+fp+simd -mabi=lp64 -Wno-psabi -mtune=cortex-a57 -march=armv8-a+crypto+crc+fp+simd -fomit-frame-pointer -Wall -pipe -O2 -mcpu=cortex-a57 -fPIC' \
-CC=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnueabi-gcc \
-LD=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnueabi-ld \
-AR=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnueabi-ar \
-./configure --host=aarch64-libreelec-linux-gnueabi --prefix=$INSTALL_DIR
+CC=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnu-gcc \
+LD=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnu-ld \
+AR=$TOOLCHAIN/bin/aarch64-libreelec-linux-gnu-ar \
+./configure --host=aarch64-libreelec-linux-gnu --prefix=$INSTALL_DIR
 make install
 cd ..
 rm -fr opus_tmp
diff -Naur lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/Makefile lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/Makefile
--- lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0/Makefile	2025-05-25 21:12:31.991723452 +0200
+++ lr_moonlight-20d2d77d219613a2809fa9afe9606f6cf3be2aa0-2/Makefile	2025-05-25 21:17:13.291034299 +0200
@@ -55,8 +55,8 @@
 # Lakka Switch
 else ifeq ($(platform), lakka-switch)
    TOOLCHAIN := /home/rock88/Documents/Lakka-LibreELEC/build.Lakka-L4T.aarch64-2.2-devel/toolchain
-   CC = $(TOOLCHAIN)/bin/aarch64-libreelec-linux-gnueabi-gcc
-   CXX = $(TOOLCHAIN)/bin/aarch64-libreelec-linux-gnueabi-c++
+   CC = $(TOOLCHAIN)/bin/aarch64-libreelec-linux-gnu-gcc
+   CXX = $(TOOLCHAIN)/bin/aarch64-libreelec-linux-gnu-c++
    TARGET := $(TARGET_NAME)_libretro.so
    DEFINES += -mcpu=cortex-a57+crypto+crc+fp+simd -mabi=lp64 -Wno-psabi -mtune=cortex-a57 \
 	 -march=armv8-a+crypto+crc+fp+simd -fomit-frame-pointer -Wall -pipe -fPIC -pthread \
