diff -Naur enna-r1629.orig/data/theme/default/icons.edc enna-r1629/data/theme/default/icons.edc
--- enna-r1629.orig/data/theme/default/icons.edc	2009-07-03 21:16:00.000000000 +0200
+++ enna-r1629/data/theme/default/icons.edc	2009-07-03 21:16:22.000000000 +0200
@@ -116,7 +116,7 @@
 STOCK_ICON("weather_sun_storm.png",         "weather/sun_storm");
 STOCK_ICON("weather_windy.png",             "weather/windy");
 
-//STOCK_ICON("default_backdrop.png",          "backdrop/default");
+STOCK_ICON("default_backdrop.png",          "backdrop/default");
 
 STOCK_ICON("rating0.png",                   "rating/0");
 STOCK_ICON("rating1.png",                   "rating/1");
diff -Naur enna-r1629.orig/data/theme/default/video.edc enna-r1629/data/theme/default/video.edc
--- enna-r1629.orig/data/theme/default/video.edc	2009-07-03 21:16:00.000000000 +0200
+++ enna-r1629/data/theme/default/video.edc	2009-07-03 21:16:25.000000000 +0200
@@ -114,7 +114,7 @@
           description {
              state: "default" 0.0;
              rel1.relative: 0 0;
-             rel2.relative: 1.0 1.0;
+             rel2.relative: 1.0 0.75;
           }
        }
 
@@ -126,20 +126,24 @@
           description {
              state: "default" 0.0;
              visible: 0;
-             color: 0 0 0 0;
              rel1.relative: 0.0 0.75;
              rel2.relative: 1.0 1.0;
              image
              {
-                 normal: "panel_bottom.png";
-                 border: 0 0 24 0;
+                 normal: "filebrowser_tile.png";
+                 border: 0 0 0 0;
              }
           }
           description {
              state: "show" 0.0;
-             inherit: "default" 0.0;
              visible: 1;
-             color: 255 255 255 255;
+             rel1.relative: 0.0 0.75;
+             rel2.relative: 1.0 1.0;
+             image
+             {
+                 normal: "filebrowser_tile.png";
+                 border: 0 0 0 0;
+             }
           }
        }
 
