diff -Naur libressl-2.1.1/configure.ac libressl-2.1.1.patch/configure.ac
--- libressl-2.1.1/configure.ac	2014-10-16 04:36:04.000000000 +0200
+++ libressl-2.1.1.patch/configure.ac	2014-11-01 11:42:32.797637394 +0100
@@ -1,4 +1,4 @@
-+AC_INIT([libressl], m4_esyscmd([tr -d '\n' < VERSION]))
+AC_INIT([libressl], m4_esyscmd([tr -d '\n' < VERSION]))
 AC_CANONICAL_HOST
 AM_INIT_AUTOMAKE([subdir-objects])
 AC_CONFIG_MACRO_DIR([m4])
diff -Naur libressl-2.1.1/VERSION libressl-2.1.1.patch/VERSION
--- libressl-2.1.1/VERSION	1970-01-01 01:00:00.000000000 +0100
+++ libressl-2.1.1.patch/VERSION	2014-11-01 11:42:23.969645930 +0100
@@ -0,0 +1 @@
+2.1.1
\ Kein Zeilenumbruch am Dateiende.
