diff --git a/usr/Makefile b/usr/Makefile
index fd14a10..41b7140 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -66,6 +66,6 @@ clean:
 	rm -f *.o $(PROGRAMS) .depend $(LIBSYS)
 
 depend:
-	gcc $(CFLAGS) -M `ls *.c` > .depend
+	$(CC) $(CFLAGS) -M `ls *.c` > .depend
 
 -include .depend
diff --git a/utils/Makefile b/utils/Makefile
index 2c7e891..440e24e 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -12,6 +12,6 @@ clean:
 	rm -f *.o $(PROGRAMS) .depend
 
 depend:
-	gcc $(CFLAGS) -M `ls *.c` > .depend
+	$(CC) $(CFLAGS) -M `ls *.c` > .depend
 
 -include .depend
diff --git a/utils/fwparam_ibft/Makefile b/utils/fwparam_ibft/Makefile
index c72bb7f..3d2c2a5 100644
--- a/utils/fwparam_ibft/Makefile
+++ b/utils/fwparam_ibft/Makefile
@@ -38,6 +38,6 @@ clean:
 $(OBJS): prom_parse.tab.h prom_parse.h fwparam_ibft.h
 
 depend:
-	gcc $(CFLAGS) -M `ls *.c` > .depend
+	$(CC) $(CFLAGS) -M `ls *.c` > .depend
 
 -include .depend
diff --git a/utils/open-isns/Makefile.in b/utils/open-isns/Makefile.in
index a27199c..a76649f 100644
--- a/utils/open-isns/Makefile.in
+++ b/utils/open-isns/Makefile.in
@@ -73,9 +73,9 @@ distclean::
 	rm -rf autom4te.cache
 
 $(LIB): $(LIBOBJS)
-	ar cr $@ $(LIBOBJS)
+	$(AR) cr $@ $(LIBOBJS)
 
 depend:
-	gcc $(CFLAGS) -M `ls *.c` > .depend
+	$(CC) $(CFLAGS) -M `ls *.c` > .depend
 
 -include .depend
diff --git a/utils/sysdeps/Makefile b/utils/sysdeps/Makefile
index 53c10e5..effe013 100644
--- a/utils/sysdeps/Makefile
+++ b/utils/sysdeps/Makefile
@@ -10,6 +10,6 @@ clean:
 	rm -f *.o .depend
 
 depend:
-	gcc $(CFLAGS) -M `ls *.c` > .depend
+	$(CC) $(CFLAGS) -M `ls *.c` > .depend
 
 -include .depend
