diff -Naur open-iscsi-bf39941/Makefile open-iscsi-bf39941.patch/Makefile
--- open-iscsi-bf39941/Makefile	2013-08-08 17:06:46.000000000 +0200
+++ open-iscsi-bf39941.patch/Makefile	2013-08-08 18:02:13.009068581 +0200
@@ -26,20 +26,18 @@
 
 all: user
 
-user: utils/open-isns/Makefile iscsiuio/Makefile
+user: utils/open-isns/Makefile
 	$(MAKE) -C utils/open-isns
 	$(MAKE) -C utils/sysdeps
 	$(MAKE) -C utils/fwparam_ibft
 	$(MAKE) -C usr
 	$(MAKE) -C utils
-	$(MAKE) -C iscsiuio
 	@echo
 	@echo "Compilation complete                 Output file"
 	@echo "-----------------------------------  ----------------"
 	@echo "Built iSCSI daemon:                  usr/iscsid"
 	@echo "Built management application:        usr/iscsiadm"
 	@echo "Built boot tool:                     usr/iscsistart"
-	@echo "Built iscsiuio daemon:               iscsiuio/src/unix/iscsiuio"
 	@echo
 	@echo "Read README file for detailed information."
 
