diff -Naur evas-43006/configure evas.patch/configure
--- evas-43006/configure	2009-10-11 01:41:13.000000000 +0200
+++ evas.patch/configure	2009-10-11 03:30:26.949656300 +0200
@@ -26077,7 +26077,7 @@
       x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
       x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11"
    fi
-   evas_engine_gl_x11_cflags="-I/usr/include ${x_cflags}"
+   evas_engine_gl_x11_cflags="${x_cflags}"
    evas_engine_gl_x11_libs="${x_libs} -lGL -lpthread"
    evas_engine_gl_common_libs="-lGL -lpthread"
 else
