--- evas-r42497/configure.orig	2009-09-17 14:07:09.000000000 +0200
+++ evas-r42497/configure	2009-09-17 14:07:47.000000000 +0200
@@ -26038,7 +26038,7 @@
       x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
       x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11"
    fi
-   evas_engine_gl_x11_cflags="-I/usr/include ${x_cflags}"
+   evas_engine_gl_x11_cflags="${x_cflags}"
    evas_engine_gl_x11_libs="${x_libs} -lGL -lGLU -lpthread"
 fi
 
