--- bnes/nall/intrinsics.hpp	2014-01-31 20:23:07.481407503 +0100
+++ bnes.patch/nall/intrinsics.hpp	2014-02-05 23:26:40.525326208 +0100
@@ -43,7 +43,7 @@
 
 /* Endian detection */
 
-#if defined(__i386__) || defined(__amd64__) || defined(_M_IX86) || defined(_M_AMD64)
+#if defined(__i386__) || defined(__amd64__) || defined(_M_IX86) || defined(_M_AMD64) || defined(__arm__)
   #define ENDIAN_LSB
   #define ARCH_LSB
   Intrinsics::Endian Intrinsics::endian() { return Intrinsics::Endian::LSB; }
