--- bnes/Makefile	2014-01-31 20:23:07.477407415 +0100
+++ bnes.patch/Makefile	2014-02-05 23:21:11.293998643 +0100
@@ -16,9 +16,9 @@
    TARGET := libretro.so
 endif
 
-ifneq ($(compiler),)
-   CXX = $(compiler)
-endif
+#ifneq ($(compiler),)
+#   CXX = $(compiler)
+#endif
 
 nes := nes
 
@@ -45,8 +45,8 @@
 obj/libnes.o: libretro/libretro.cpp $(call rwildcard,libretro/)
 obj/libco.o: libco/libco.c
 
-c := $(CC) -std=gnu99
-cpp := $(CXX) -std=gnu++0x
+c := $(CC) -std=gnu99 $(CFLAGS)
+cpp := $(CXX) -std=gnu++0x $(CXXFLAGS)
 flags := -O3 -fomit-frame-pointer -fno-tree-vectorize -I. $(fpic)
 
 compile = \
