diff -Naur fuse-2.8.6/util/fusermount.c fuse-2.8.6.patch/util/fusermount.c
--- fuse-2.8.6/util/fusermount.c	2011-09-13 09:23:14.000000000 +0200
+++ fuse-2.8.6.patch/util/fusermount.c	2011-09-18 00:19:46.120163595 +0200
@@ -7,6 +7,9 @@
 */
 /* This program does the mounting and unmounting of FUSE filesystems */
 
+/* for CLONE_NEWNS in sched.h */
+#define _GNU_SOURCE
+
 #include <config.h>
 
 #include "mount_util.h"
