From 459c84a3e5fda1ef7f7060d44903b31215857e70 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 9 Aug 2010 12:10:40 +0400
Subject: [PATCH 2/3] More parentheses

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 lib/fuse.c          |    8 +++-----
 lib/fuse_lowlevel.c |    2 +-
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/lib/fuse.c b/lib/fuse.c
index 95cf50b..76c2681 100644
--- a/lib/fuse.c
+++ b/lib/fuse.c
@@ -961,17 +961,15 @@ static int fuse_compat_open(struct fuse_fs *fs, const char *path,
 {
 	int err;
 	if (!fs->compat || fs->compat >= 25)
-		err = fs->op.open(path, fi);
+		err = (fs->op.open)(path, fi);
 	else if (fs->compat == 22) {
 		struct fuse_file_info_compat tmp;
 		memcpy(&tmp, fi, sizeof(tmp));
-		err = ((struct fuse_operations_compat22 *) &fs->op)->open(path,
-									  &tmp);
+		err = (((struct fuse_operations_compat22 *) &fs->op)->open)(path, &tmp);
 		memcpy(fi, &tmp, sizeof(tmp));
 		fi->fh = tmp.fh;
 	} else
-		err = ((struct fuse_operations_compat2 *) &fs->op)
-			->open(path, fi->flags);
+		err = (((struct fuse_operations_compat2 *) &fs->op)->open)(path, fi->flags);
 	return err;
 }
 
diff --git a/lib/fuse_lowlevel.c b/lib/fuse_lowlevel.c
index c519bfb..d6275b5 100644
--- a/lib/fuse_lowlevel.c
+++ b/lib/fuse_lowlevel.c
@@ -716,7 +716,7 @@ static void do_open(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
 	fi.flags = arg->flags;
 
 	if (req->f->op.open)
-		req->f->op.open(req, nodeid, &fi);
+		(req->f->op.open)(req, nodeid, &fi);
 	else
 		fuse_reply_open(req, &fi);
 }
-- 
1.7.3.1

