diff -Naur fuse-2.9.2/configure.in fuse-2.9.2.patch/configure.in
--- fuse-2.9.2/configure.in	2012-10-01 17:58:00.000000000 +0200
+++ fuse-2.9.2.patch/configure.in	2013-01-12 16:56:37.469681997 +0100
@@ -4,7 +4,7 @@
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES(yes)])
-AM_CONFIG_HEADER(include/config.h)
+AC_CONFIG_HEADERS(include/config.h)
 
 AC_PROG_LIBTOOL
 AC_PROG_CC
