From 7688dae788f3a3f8f5f0478f096904cc885a1a90 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 16 Aug 2014 14:57:02 +0300
Subject: [PATCH] fix pkgconf

---
 builds/unix/freetype2.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/builds/unix/freetype2.in b/builds/unix/freetype2.in
index 6e7fb10..653732d 100644
--- a/builds/unix/freetype2.in
+++ b/builds/unix/freetype2.in
@@ -1,7 +1,7 @@
 prefix="%prefix%"
-exec_prefix="%exec_prefix%"
-libdir="%libdir%"
-includedir="%includedir%/freetype2"
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include/freetype2
 
 Name: FreeType 2
 URL: http://freetype.org
-- 
1.7.2.5

