diff -Naur freetype-2.6.1/builds/unix/freetype2.in freetype-2.6.1.patch/builds/unix/freetype2.in
--- freetype-2.6.1/builds/unix/freetype2.in	2015-09-25 07:58:57.000000000 +0200
+++ freetype-2.6.1.patch/builds/unix/freetype2.in	2015-10-12 21:58:50.736816464 +0200
@@ -1,7 +1,7 @@
 prefix=%prefix%
-exec_prefix=%exec_prefix%
-libdir=%libdir%
-includedir=%includedir%
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
 
 Name: FreeType 2
 URL: http://freetype.org
