--- a/kernel/nv-linux.h	2022-02-20 00:41:53.585928825 +0000
+++ b/kernel/nv-linux.h	2022-02-20 00:43:04.285797471 +0000
@@ -2051,11 +2051,15 @@
         __entry;                                               \
     })
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
 #if defined(NV_PDE_DATA_PRESENT)
 # define NV_PDE_DATA(inode) PDE_DATA(inode)
 #else
 # define NV_PDE_DATA(inode) PDE(inode)->data
 #endif
+#else
+# define NV_PDE_DATA(inode) inode->i_private
+#endif
 
 #if defined(NV_PROC_REMOVE_PRESENT)
 # define NV_REMOVE_PROC_ENTRY(entry)                           \
