--- a/kernel/nv-procfs.c       2015-10-18 04:32:24.214891836 +0100
+++ b/kernel/nv-procfs.c       2015-10-18 04:33:01.611081468 +0100
@@ -607,7 +607,12 @@
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,3,0)
     return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+#else
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
+#endif
 }
 
 static ssize_t
@@ -674,7 +679,12 @@
     void *v
 )
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,3,0)
     return seq_puts(s, s->private);
+#else
+    seq_puts(s, s->private);
+    return 0;
+#endif
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
