diff -Naur xf86-video-nvidia-legacy-304.125/kernel/nv.c xf86-video-nvidia-legacy-304.125.patch/kernel/nv.c
--- xf86-video-nvidia-legacy-304.125/kernel/nv.c	2014-12-02 04:55:59.000000000 +0100
+++ xf86-video-nvidia-legacy-304.125.patch/kernel/nv.c	2015-02-11 19:07:09.844431593 +0100
@@ -17,7 +17,7 @@
 #include "rmil.h"
 
 #if defined(MODULE_LICENSE)
-MODULE_LICENSE("NVIDIA");
+MODULE_LICENSE("GPL\0NVIDIA");
 #endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
@@ -2026,7 +2026,7 @@
     unsigned long i_arg
 )
 {
-    return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
 }
 
 long nv_kern_compat_ioctl(
@@ -2035,7 +2035,7 @@
     unsigned long i_arg
 )
 {
-    return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
 }
 
 /*
