diff -up libtheora-1.0/lib/Makefile.am.orig libtheora-1.0/lib/Makefile.am
--- libtheora-1.0/lib/Makefile.am.orig	2008-10-24 06:07:31.000000000 +0200
+++ libtheora-1.0/lib/Makefile.am	2008-12-16 21:06:12.000000000 +0100
@@ -144,6 +144,7 @@ libtheoraenc_la_SOURCES = \
 libtheoraenc_la_LDFLAGS = \
   -version-info @THENC_LIB_CURRENT@:@THENC_LIB_REVISION@:@THENC_LIB_AGE@ \
   @THEORAENC_LDFLAGS@ $(OGG_LIBS)
+libtheoraenc_la_LIBADD = libtheoradec.la
 
 libtheora_la_SOURCES = \
 	$(decoder_arch_sources) \
diff -up libtheora-1.0/lib/Makefile.in.orig libtheora-1.0/lib/Makefile.in
--- libtheora-1.0/lib/Makefile.in.orig	2008-10-29 11:36:47.000000000 +0100
+++ libtheora-1.0/lib/Makefile.in	2008-12-16 21:06:08.000000000 +0100
@@ -116,7 +116,8 @@ am__libtheoradec_la_SOURCES_DIST = dec/x
 	Version_script-dec
 am_libtheoradec_la_OBJECTS = $(am__objects_2) $(am__objects_3)
 libtheoradec_la_OBJECTS = $(am_libtheoradec_la_OBJECTS)
-libtheoraenc_la_LIBADD =
+libtheoraenc_la_LIBADD = libtheoradec.la
+libtheoraenc_la_DEPENDENCIES = libtheoradec.la
 am__libtheoraenc_la_SOURCES_DIST = $(enc_arch_dir)/dct_decode_mmx.c \
 	$(enc_arch_dir)/dsp_mmx.c $(enc_arch_dir)/dsp_mmxext.c \
 	$(enc_arch_dir)/recon_mmx.c $(enc_arch_dir)/idct_mmx.c \
