From f94717816b9a39869219ede859fe74af3f2ecd19 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Wed, 24 Mar 2021 14:27:43 -0700
Subject: [PATCH 48/49] drm: imx: mhdp: don't depend on DRM_IMX

---
 drivers/gpu/drm/imx/mhdp/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/mhdp/Kconfig b/drivers/gpu/drm/imx/mhdp/Kconfig
index 86950badb947..cf7dfacdd434 100644
--- a/drivers/gpu/drm/imx/mhdp/Kconfig
+++ b/drivers/gpu/drm/imx/mhdp/Kconfig
@@ -6,6 +6,5 @@ config DRM_IMX_CDNS_MHDP
 	select DRM_CDNS_DP
 	select DRM_CDNS_HDMI
 	select DRM_CDNS_AUDIO
-	depends on DRM_IMX
 	help
 	  Choose this if you want to use HDMI on i.MX8.
-- 
2.29.2

