--- vsxu-0.5.1/cmake/modules/FindGLFW.cmake.orig	2015-05-12 14:22:23.123621176 +0200
+++ vsxu-0.5.1/cmake/modules/FindGLFW.cmake	2015-05-12 14:23:14.863620161 +0200
@@ -19,6 +19,14 @@
 # TODO:
 # Allow the user to select to link to a shared library or to a static library.
 
+find_package(PkgConfig)
+
+pkg_check_modules(GLFW libglfw)
+set(GLFW_INCLUDE_PATH ${GLFW_INCLUDE_DIRS})
+set(GLFW_LIBRARY ${GLFW_LIBRARIES} -lXrandr)
+
+if(NOT GLFW_FOUND)
+
 #Search for the include file...
 FIND_PATH(GLFW_INCLUDE_PATH GL/glfw.h DOC "Path to GLFW include directory."
   HINTS
@@ -45,5 +53,7 @@
   ${GLFW_ROOT_DIR}/lib-msvc100/release # added by ptr
 )
 
+endif()
+
 INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(GLFW DEFAULT_MSG GLFW_LIBRARY GLFW_INCLUDE_PATH)
\ No newline at end of file
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(GLFW DEFAULT_MSG GLFW_LIBRARY GLFW_INCLUDE_PATH)
