--- vsxu-0.5.1/plugins/src/texture/CMakeLists.txt.orig	2015-05-12 14:04:42.659641970 +0200
+++ vsxu-0.5.1/plugins/src/texture/CMakeLists.txt	2015-05-12 14:05:43.315640780 +0200
@@ -8,7 +8,7 @@
 
 message("configuring vsxu module            " ${module_id})
 
-include_directories(/usr/include/OpenEXR)
+#include_directories(/usr/include/OpenEXR)
 
 # set defines for MOD_CM, MOD_DM, MOD_NM
 CREATE_STATIC_GLUE(${module_id})
