--- a/net-snmp-create-v3-user.in	2014-12-08 14:23:22.000000000 -0600
+++ b/net-snmp-create-v3-user.in	2017-03-30 13:20:56.336794994 -0500
@@ -28,10 +28,8 @@
    NSC_SRCDIR="NET-SNMP-SOURCE-DIR"
 fi
 
-if @PSCMD@ | egrep ' snmpd *$' > /dev/null 2>&1 ; then
-    echo "Apparently at least one snmpd demon is already running."
-    echo "You must stop them in order to use this command."
-    exit 1
+if @PSCMD@ | egrep 'snmpd'> /dev/null 2>&1 ; then
+    systemctl stop service.net-snmp.service
 fi
 
 Aalgorithm="MD5"
