commit 4f51d0b2ad01b0dce8e93732e70d338e24509d88
Author: mglae <mglmail@arcor.de>
Date:   Tue Jul 30 18:14:28 2019 +0200

    build libtimidity.so

diff --git a/lib/timidity/CMakeLists.txt b/lib/timidity/CMakeLists.txt
index fc5e446..2f930ed 100644
--- a/lib/timidity/CMakeLists.txt
+++ b/lib/timidity/CMakeLists.txt
@@ -55,4 +55,4 @@ endif()
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} timidity libarc utils)
 
-add_library(timidity STATIC ${SOURCES})
+add_library(timidity SHARED ${SOURCES})
