diff --git a/source3/modules/vfs_default.c b/source3/modules/vfs_default.c
index e0ebc7bd..9d9dcebe 100644
--- a/source3/modules/vfs_default.c
+++ b/source3/modules/vfs_default.c
@@ -2348,9 +2348,9 @@ static NTSTATUS vfswrap_offload_copy_file_range(struct tevent_req *req)
 
 	while (state->remaining > 0) {
 		nwritten = copy_file_range(fsp_get_io_fd(state->src_fsp),
-					   &state->src_off,
+					   (__off64_t *)&state->src_off,
 					   fsp_get_io_fd(state->dst_fsp),
-					   &state->dst_off,
+					   (__off64_t *)&state->dst_off,
 					   state->remaining,
 					   0);
 		if (nwritten == -1) {
