diff -Naur dvblast-3.0/Makefile dvblast-3.0.patch/Makefile
--- dvblast-3.0/Makefile	2015-10-05 17:51:14.000000000 +0200
+++ dvblast-3.0.patch/Makefile	2016-03-26 22:54:59.953303040 +0100
@@ -55,11 +55,11 @@
 
 dvblast: $(OBJ_DVBLAST)
 	@echo "LINK    $@"
-	$(Q)$(CROSS)$(CC) -o $@ $(OBJ_DVBLAST) $(LDLIBS_DVBLAST) $(LDLIBS)
+	$(Q)$(CROSS)$(CC) -o $@ $(OBJ_DVBLAST) $(LDFLAGS) $(LDLIBS_DVBLAST) $(LDLIBS)
 
 dvblastctl: $(OBJ_DVBLASTCTL)
 	@echo "LINK    $@"
-	$(Q)$(CROSS)$(CC) -o $@ $(OBJ_DVBLASTCTL) $(LDLIBS)
+	$(Q)$(CROSS)$(CC) -o $@ $(OBJ_DVBLASTCTL) $(LDFLAGS) $(LDLIBS)
 
 clean:
 	@echo "CLEAN   $(CLEAN_OBJS)"
