diff --git a/dvb.c b/dvb.c
index d1c8409..52ff3c2 100644
--- a/dvb.c
+++ b/dvb.c
@@ -86,6 +86,11 @@ static void FrontendRead(struct ev_loop *loop, struct ev_io *w, int revents);
 static void FrontendLockCb(struct ev_loop *loop, struct ev_timer *w, int revents);
 static void FrontendSet( bool b_reset );
 
+// aml 3.14 is meh ?
+#ifndef fe_caps_t
+typedef enum fe_caps fe_caps_t;
+#endif
+
 /*****************************************************************************
  * dvb_Open
  *****************************************************************************/
