diff -Naur a/Makefile b/Makefile
--- a/Makefile	2017-09-11 11:10:40.000000000 -0700
+++ b/Makefile	2017-10-03 13:41:57.992106628 -0700
@@ -1379,7 +1379,7 @@
 	$(call filechk,timestamp.h)
 
 checkbinman: tools
-	@if ! ( echo 'import libfdt' | ( PYTHONPATH=tools $(PYTHON) )); then \
+	@if ! ( echo 'from pylibfdt import libfdt' | ( python )); then \
 		echo >&2; \
 		echo >&2 '*** binman needs the Python libfdt library.'; \
 		echo >&2 '*** Either install it on your system, or try:'; \
diff -Naur a/tools/dtoc/fdt.py b/tools/dtoc/fdt.py
--- a/tools/dtoc/fdt.py	2017-09-11 11:10:40.000000000 -0700
+++ b/tools/dtoc/fdt.py	2017-10-03 13:42:18.398306028 -0700
@@ -10,7 +10,7 @@
 import sys
 
 import fdt_util
-import libfdt
+from pylibfdt import libfdt
 
 # This deals with a device tree, presenting it as an assortment of Node and
 # Prop objects, representing nodes and properties, respectively. This file
diff -Naur a/tools/Makefile b/tools/Makefile
--- a/tools/Makefile	2017-09-11 11:10:40.000000000 -0700
+++ b/tools/Makefile	2017-10-03 13:42:33.286451519 -0700
@@ -232,10 +232,6 @@
 
 always := $(hostprogs-y)
 
-# Build a libfdt Python module if swig is available
-# Use 'sudo apt-get install swig libpython-dev' to enable this
-always += $(if $(shell which swig 2> /dev/null),_libfdt.so)
-
 # Generated LCD/video logo
 LOGO_H = $(objtree)/include/bmp_logo.h
 LOGO_DATA_H = $(objtree)/include/bmp_logo_data.h
