From 06c160f28cceb32be4c8fcead3f33691a2469464 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 21 Nov 2013 23:29:07 +0200
Subject: [PATCH] crosscompile

---
 Makefile |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 86ab501..355fb3d 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,6 @@ TARGET = getscancodes
 SOURCES = getscancodes.c
 
 CFLAGS=-O2 -Wall
-CC=gcc
 
 OBJECTS=$(SOURCES:.c=.o)
 
-- 
1.7.2.5

