diff -Naur libusb-compat-0.1.4-old/configure.ac libusb-compat-0.1.4-new/configure.ac
--- libusb-compat-0.1.4-old/configure.ac	2012-04-23 22:45:46.000000000 -0700
+++ libusb-compat-0.1.4-new/configure.ac	2012-12-29 08:44:33.000000000 -0800
@@ -1,7 +1,7 @@
 AC_INIT([libusb-compat], [0.1.4])
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([libusb/core.c])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_PREREQ([2.50])
