From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 18 Nov 2020 23:31:47 -0600
Subject: [PATCH] sunxi: binman: Support FIT images for 32-bit boards

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm/dts/sunxi-u-boot.dtsi | 58 ++++++++++++++++++++++++++++++++++
 1 file changed, 58 insertions(+)

--- a/arch/arm/dts/sunxi-u-boot.dtsi
+++ b/arch/arm/dts/sunxi-u-boot.dtsi
@@ -1,5 +1,6 @@
 #include <config.h>
 
+#ifdef CONFIG_ARM64
 #ifdef CONFIG_MACH_SUN50I_H6
 #define BL31_ADDR 0x104000
 #define  SCP_ADDR 0x114000
@@ -7,6 +8,9 @@
 #define BL31_ADDR  0x44000
 #define  SCP_ADDR  0x50000
 #endif
+#else
+#define  SCP_ADDR  0x48000
+#endif
 
 / {
 	aliases {
@@ -27,6 +31,7 @@
 			filename = "spl/sunxi-spl.bin";
 		};
 
+#ifdef CONFIG_SPL_FIT
 #ifdef CONFIG_ARM64
 		fit {
 			description = "Configuration to load ATF before U-Boot";
@@ -94,6 +99,59 @@
 			};
 		};
 #else
+		fit {
+			description = "Configuration to load SCP firmware with U-Boot";
+			offset = <CONFIG_SPL_PAD_TO>;
+			#address-cells = <1>;
+			fit,fdt-list = "of-list";
+
+			images {
+				uboot {
+					description = "U-Boot (32-bit)";
+					type = "standalone";
+					os = "u-boot";
+					arch = "arm";
+					compression = "none";
+					load = <CONFIG_SYS_TEXT_BASE>;
+
+					u-boot-nodtb {
+					};
+				};
+
+				scp {
+					description = "SCP firmware";
+					type = "firmware";
+					arch = "or1k";
+					compression = "none";
+					load = <SCP_ADDR>;
+
+					scp {
+						filename = "scp.bin";
+						missing-msg = "scp-sunxi";
+					};
+				};
+
+				@fdt-SEQ {
+					description = "NAME";
+					type = "flat_dt";
+					compression = "none";
+
+				};
+			};
+
+			configurations {
+				default = "@config-DEFAULT-SEQ";
+
+				@config-SEQ {
+					description = "NAME";
+					firmware = "uboot";
+					loadables = "scp";
+					fdt = "fdt-SEQ";
+				};
+			};
+		};
+#endif
+#else
 		u-boot-img {
 			offset = <CONFIG_SPL_PAD_TO>;
 		};
