diff -Naur bcm_sta-5.100.82.112/x86-32/src/wl/sys/wl_cfg80211.c bcm_sta-5.100.82.112.patch/x86-32/src/wl/sys/wl_cfg80211.c
--- bcm_sta-5.100.82.112/x86-32/src/wl/sys/wl_cfg80211.c	2011-10-22 18:55:54.000000000 +0200
+++ bcm_sta-5.100.82.112.patch/x86-32/src/wl/sys/wl_cfg80211.c	2013-05-06 07:03:19.146764181 +0200
@@ -2077,7 +2077,11 @@
 		ie = bss->information_elements;
 		ie_len = bss->len_information_elements;
 		beacon_interval = bss->beacon_interval;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 9, 0)
 		cfg80211_put_bss(bss);
+#else
+		cfg80211_put_bss(wl_to_wiphy(wl), bss);
+#endif
 	}
 
 	tim = bcm_parse_tlvs(ie, ie_len, WLAN_EID_TIM);
diff -Naur bcm_sta-5.100.82.112/x86-64/src/wl/sys/wl_cfg80211.c bcm_sta-5.100.82.112.patch/x86-64/src/wl/sys/wl_cfg80211.c
--- bcm_sta-5.100.82.112/x86-64/src/wl/sys/wl_cfg80211.c	2011-10-22 18:56:55.000000000 +0200
+++ bcm_sta-5.100.82.112.patch/x86-64/src/wl/sys/wl_cfg80211.c	2013-05-06 07:03:55.443652962 +0200
@@ -2077,7 +2077,11 @@
 		ie = bss->information_elements;
 		ie_len = bss->len_information_elements;
 		beacon_interval = bss->beacon_interval;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 9, 0)
 		cfg80211_put_bss(bss);
+#else
+		cfg80211_put_bss(wl_to_wiphy(wl), bss);
+#endif
 	}
 
 	tim = bcm_parse_tlvs(ie, ie_len, WLAN_EID_TIM);
