--- a/x86-32/src/wl/sys/wl_linux.c	2010-12-15 02:58:07.000000000 +0100
+++ b/x86-32/src/wl/sys/wl_linux.c	2010-12-25 18:15:43.356676382 +0100
@@ -240,6 +240,7 @@
 
 static int nompc = 0;
 module_param(nompc, int, 0);
+MODULE_LICENSE("MIXED/Proprietary");
 
 #ifdef quote_str
 #undef quote_str
--- a/x86-64/src/wl/sys/wl_linux.c      2010-12-15 02:58:07.000000000 +0100
+++ b/x86-64/src/wl/sys/wl_linux.c      2010-12-25 18:15:43.356676382 +0100
@@ -240,6 +240,7 @@
 
 static int nompc = 0;
 module_param(nompc, int, 0);
+MODULE_LICENSE("MIXED/Proprietary");
 
 #ifdef quote_str
 #undef quote_str

