diff --git a/Source/Core/Core/Config/GraphicsSettings.cpp b/Source/Core/Core/Config/GraphicsSettings.cpp
index 608e3e2d1..47f7f772c 100644
--- a/Source/Core/Core/Config/GraphicsSettings.cpp
+++ b/Source/Core/Core/Config/GraphicsSettings.cpp
@@ -56,7 +56,7 @@ const ConfigInfo<int> GFX_BITRATE_KBPS{{System::GFX, "Settings", "BitrateKbps"},
 const ConfigInfo<bool> GFX_INTERNAL_RESOLUTION_FRAME_DUMPS{
     {System::GFX, "Settings", "InternalResolutionFrameDumps"}, false};
 const ConfigInfo<bool> GFX_ENABLE_GPU_TEXTURE_DECODING{
-    {System::GFX, "Settings", "EnableGPUTextureDecoding"}, false};
+    {System::GFX, "Settings", "EnableGPUTextureDecoding"}, true};
 const ConfigInfo<bool> GFX_ENABLE_COMPUTE_TEXTURE_ENCODING{ { System::GFX, "Settings", "EnableComputeTextureEncoding" }, false };
 const ConfigInfo<bool> GFX_ENABLE_PIXEL_LIGHTING{{System::GFX, "Settings", "EnablePixelLighting"},
                                                  false};
