diff -Naur ratpoison-1.4.5-old/configure.in ratpoison-1.4.5-new/configure.in
--- ratpoison-1.4.5-old/configure.in	2009-07-12 14:01:04.000000000 -0700
+++ ratpoison-1.4.5-new/configure.in	2009-07-14 19:56:24.000000000 -0700
@@ -146,7 +146,7 @@
 dnl Checks for typedefs, structures, and compiler characteristics.
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS(getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
+AC_CHECK_FUNCS(getline getopt getopt_long setsid setpgid setpgrp putenv vsnprintf usleep)
 
 AC_TYPE_SIGNAL
 
