diff -Naur ratpoison-1.4.5-old/src/actions.c ratpoison-1.4.5-new/src/actions.c
--- ratpoison-1.4.5-old/src/actions.c	2009-07-14 19:56:16.000000000 -0700
+++ ratpoison-1.4.5-new/src/actions.c	2009-07-14 19:56:56.000000000 -0700
@@ -1882,6 +1882,8 @@
 {
   int ret = -1;
 
+  if (!strcasecmp (data, "static") || !strcasecmp (data, "static") || !strcmp (data, "10"))
+    ret = StaticGravity;
   if (!strcasecmp (data, "northwest") || !strcasecmp (data, "nw") || !strcmp (data, "7"))
     ret = NorthWestGravity;
   if (!strcasecmp (data, "north") || !strcasecmp (data, "n") || !strcmp (data, "8"))
diff -Naur ratpoison-1.4.5-old/src/manage.c ratpoison-1.4.5-new/src/manage.c
--- ratpoison-1.4.5-old/src/manage.c	2009-07-14 19:56:16.000000000 -0700
+++ ratpoison-1.4.5-new/src/manage.c	2009-07-14 19:56:56.000000000 -0700
@@ -541,6 +541,9 @@
     case SouthEastGravity:
       win->x = frame->x + frame->width - win->width - win->border;
       break;
+    case StaticGravity:
+      win->x = frame->x + win->x;
+      break;
     }
 
   /* Y coord. */
@@ -561,6 +564,9 @@
     case SouthWestGravity:
       win->y = frame->y + frame->height - win->height - win->border;
       break;
+    case StaticGravity:
+      win->y = frame->y + win->y;
+      break;
     }
 }
 
