diff -Naur systemd-209/src/journal/journald-server.c systemd-209.patch/src/journal/journald-server.c
--- systemd-209/src/journal/journald-server.c	2014-02-20 00:03:09.000000000 +0100
+++ systemd-209.patch/src/journal/journald-server.c	2014-02-21 16:41:49.152296403 +0100
@@ -1430,6 +1430,14 @@
 
         r = sd_event_add_io(s->event, &s->hostname_event_source, s->hostname_fd, 0, dispatch_hostname_change, s);
         if (r < 0) {
+                /* kernels prior to 3.2 don't support polling this file.
+                 * quietly ignore the failure. */
+                if (r == -EPERM) {
+                        close_nointr_nofail(s->hostname_fd);
+                        s->hostname_fd = -1;
+                        return 0;
+                }
+
                 log_error("Failed to register hostname fd in event loop: %s", strerror(-r));
                 return r;
         }
