From 0c50b8332092178378257c2de800fb6b6b4a8706 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 3 Jul 2017 08:45:04 -0700
Subject: [PATCH] parse-util: Do not include unneeded xlocale.h

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Backport [ partial https://github.com/systemd/systemd/commit/284d1cd0a12cad96a5ea61d1afb0dd677dbd147e]

 src/basic/parse-util.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/basic/parse-util.c b/src/basic/parse-util.c
index c98815b9b..a0eb45805 100644
--- a/src/basic/parse-util.c
 b/src/basic/parse-util.c
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <xlocale.h>
 
 #include "alloc-util.h"
 #include "extract-word.h"
-- 
2.13.2

