commit 7723509e9fbc1a215bb72810b1de8366673f3ad0
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sun Sep 15 19:25:27 2013 +0300

    openelec user udev rules

diff --git a/src/udev/udev-rules.c b/src/udev/udev-rules.c
index f14158b..eba2807 100644
--- a/src/udev/udev-rules.c
+++ b/src/udev/udev-rules.c
@@ -1619,6 +1619,7 @@ struct udev_rules *udev_rules_new(struct udev *udev, int resolve_names)
         rules->dirs = strv_new("/etc/udev/rules.d",
                                "/run/udev/rules.d",
                                UDEVLIBEXECDIR "/rules.d",
+                               "/storage/.config/udev.rules.d",
                                NULL);
         if (!rules->dirs) {
                 log_error("failed to build config directory array");
