From dd5204794a1f2ca1924a64d527b39a254a469200 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 28 Jul 2014 21:22:10 +0300
Subject: [PATCH 5/8] autoreconf: remove gcrypt dependency

---
 configure.ac |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index c9d3305..0682845 100644
--- a/configure.ac
+++ b/configure.ac
@@ -716,13 +716,7 @@ AC_ARG_ENABLE([gcrypt],
                 [have_gcrypt=auto])
 
 if test "x${have_gcrypt}" != xno ; then
-        AM_PATH_LIBGCRYPT(
-                [1.4.5],
-                [have_gcrypt=yes],
-                [if test "x$have_gcrypt" = xyes ; then
-                        AC_MSG_ERROR([*** GCRYPT headers not found.])
-                fi])
-
+        have_gcrypt=no
         if test "x$have_gcrypt" = xyes ; then
                 GCRYPT_LIBS="$LIBGCRYPT_LIBS"
                 GCRYPT_CFLAGS="$LIBGCRYPT_CFLAGS"
-- 
1.7.10.4

