From 0c50b670dc1466ce697a72b34cbb7a1a41d1b24f Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Wed, 18 Nov 2015 20:15:43 +0200
Subject: [PATCH 8/8] unsupported btrfs bull****

---
 src/basic/btrfs-util.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/basic/btrfs-util.c b/src/basic/btrfs-util.c
index be40dc5..e5efe4f 100644
--- a/src/basic/btrfs-util.c
+++ b/src/basic/btrfs-util.c
@@ -1103,9 +1103,6 @@ int btrfs_quota_scan_start(int fd) {
 int btrfs_quota_scan_wait(int fd) {
         assert(fd >= 0);
 
-        if (ioctl(fd, BTRFS_IOC_QUOTA_RESCAN_WAIT) < 0)
-                return -errno;
-
         return 0;
 }
 
-- 
2.1.4
