From e52504dca98923302ecbdd018aae8e86376e166a Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Sun, 4 Feb 2018 17:46:59 +0000
Subject: [PATCH] kernel 3.10 compat: RTAX_QUICKACK not added until 3.11.0

---
 src/basic/missing.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/basic/missing.h b/src/basic/missing.h
index 1280e6c..9d4d08e 100644
--- a/src/basic/missing.h
+++ b/src/basic/missing.h
@@ -1047,6 +1047,10 @@ struct input_mask {
 #define RTA_PREF 20
 #endif
 
+#ifndef RTAX_QUICKACK
+#define RTAX_QUICKACK 15
+#endif
+
 #ifndef IPV6_UNICAST_IF
 #define IPV6_UNICAST_IF 76
 #endif
-- 
2.14.1

