From 086daa1279f896a5f3d7e27938dbb3772fbf9692 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 21 Oct 2013 21:50:45 +0300
Subject: [PATCH] user hwdb.d

---
 src/udev/udevadm-hwdb.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/udev/udevadm-hwdb.c b/udev/udevadm-hwdb.c
index d9dc73b..e4b4a27 100644
--- a/src/udev/udevadm-hwdb.c
+++ b/src/udev/udevadm-hwdb.c
@@ -38,6 +38,7 @@
 static const char * const conf_file_dirs[] = {
         "/etc/udev/hwdb.d",
         UDEVLIBEXECDIR "/hwdb.d",
+        "/storage/.config/hwdb.d",
         NULL
 };
 
-- 
1.7.2.5

