diff -Naur systemd-209/src/udev/udev-rules.c systemd-209.patch/src/udev/udev-rules.c
--- systemd-209/src/udev/udev-rules.c	2014-02-18 21:07:20.000000000 +0100
+++ systemd-209.patch/src/udev/udev-rules.c	2014-02-20 13:46:47.885370802 +0100
@@ -50,6 +50,7 @@
         "/etc/udev/rules.d",
         "/run/udev/rules.d",
         UDEVLIBEXECDIR "/rules.d",
+        "/storage/.config/udev.rules.d",
         NULL};
 
 struct udev_rules {
