commit 43aba5ffffc6d35ed97db035c5818c76652b06de
Author: Matthias Reichl <hias@horus.com>
Date:   Mon Sep 28 23:55:13 2020 +0200

    move /etc/systemd/system to /storage/.config/system.d

diff --git a/meson.build b/meson.build
index dbbddb68e2..4592cd1094 100644
--- a/meson.build
+++ b/meson.build
@@ -266,7 +266,7 @@
 conf.set_quoted('SYSTEMD_UPDATE_HELPER_PATH',                 rootlibexecdir / 'systemd-update-helper')
 conf.set_quoted('SYSTEMD_USERWORK_PATH',                      rootlibexecdir / 'systemd-userwork')
 conf.set_quoted('SYSTEMD_VERITYSETUP_PATH',                   rootlibexecdir / 'systemd-veritysetup')
-conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     pkgsysconfdir / 'system')
+conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     '/storage/.config/system.d')
 conf.set_quoted('SYSTEM_DATA_UNIT_DIR',                       systemunitdir)
 conf.set_quoted('SYSTEM_ENV_GENERATOR_DIR',                   systemenvgeneratordir)
 conf.set_quoted('SYSTEM_GENERATOR_DIR',                       systemgeneratordir)
