From 0fe627a4c4a10f5ed7aee8372cee7956aa1ccb80 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 19 Dec 2013 11:21:13 +0200
Subject: [PATCH 3/8] move /etc/systemd/system to /storage/.config/system.d

---
 Makefile.am |    4 ++--
 Makefile.in |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index ac7924e..8288850 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -164,7 +164,7 @@ udevlibexec_PROGRAMS =
 AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-DPKGSYSCONFDIR=\"$(pkgsysconfdir)\" \
-	-DSYSTEM_CONFIG_UNIT_PATH=\"$(pkgsysconfdir)/system\" \
+	-DSYSTEM_CONFIG_UNIT_PATH=\"/storage/.config/system.d\" \
 	-DSYSTEM_DATA_UNIT_PATH=\"$(systemunitdir)\" \
 	-DSYSTEM_SYSVINIT_PATH=\"$(SYSTEM_SYSVINIT_PATH)\" \
 	-DSYSTEM_SYSVRCND_PATH=\"$(SYSTEM_SYSVRCND_PATH)\" \
@@ -5838,7 +5838,7 @@ substitutions = \
        '|SYSTEMCTL=$(rootbindir)/systemctl|' \
        '|SYSTEMD_NOTIFY=$(rootbindir)/systemd-notify|' \
        '|pkgsysconfdir=$(pkgsysconfdir)|' \
-       '|SYSTEM_CONFIG_UNIT_PATH=$(pkgsysconfdir)/system|' \
+       '|SYSTEM_CONFIG_UNIT_PATH=/storage/.config/system.d|' \
        '|USER_CONFIG_UNIT_PATH=$(pkgsysconfdir)/user|' \
        '|pkgdatadir=$(pkgdatadir)|' \
        '|systemunitdir=$(systemunitdir)|' \
diff --git a/Makefile.in b/Makefile.in
index 3b35030..f0ad455 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -5435,7 +5435,7 @@ manual_tests = test-ns test-loopback test-hostname test-daemon \
 AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-DPKGSYSCONFDIR=\"$(pkgsysconfdir)\" \
-	-DSYSTEM_CONFIG_UNIT_PATH=\"$(pkgsysconfdir)/system\" \
+	-DSYSTEM_CONFIG_UNIT_PATH=\"/storage/.config/system.d\" \
 	-DSYSTEM_DATA_UNIT_PATH=\"$(systemunitdir)\" \
 	-DSYSTEM_SYSVINIT_PATH=\"$(SYSTEM_SYSVINIT_PATH)\" \
 	-DSYSTEM_SYSVRCND_PATH=\"$(SYSTEM_SYSVRCND_PATH)\" \
@@ -8954,7 +8954,7 @@ substitutions = \
        '|SYSTEMCTL=$(rootbindir)/systemctl|' \
        '|SYSTEMD_NOTIFY=$(rootbindir)/systemd-notify|' \
        '|pkgsysconfdir=$(pkgsysconfdir)|' \
-       '|SYSTEM_CONFIG_UNIT_PATH=$(pkgsysconfdir)/system|' \
+       '|SYSTEM_CONFIG_UNIT_PATH=/storage/.config/system.d|' \
        '|USER_CONFIG_UNIT_PATH=$(pkgsysconfdir)/user|' \
        '|pkgdatadir=$(pkgdatadir)|' \
        '|systemunitdir=$(systemunitdir)|' \
-- 
1.7.10.4

