From: Allin Cottrell <cottrell@wfu.edu>
Date: Thu, 23 Aug 2012 23:46:38 +0000
Subject: journald: add missing includes

---
diff --git a/src/journal/journald-console.c b/src/journal/journald-console.c
index 6cd2397..2596d44 100644
--- a/src/journal/journald-console.c
+++ b/src/journal/journald-console.c
@@ -21,6 +21,7 @@
 
 #include <fcntl.h>
 #include <unistd.h>
+#include <sys/socket.h>
 
 #include "journald.h"
 #include "journald-console.h"
diff --git a/src/journal/journald-gperf.gperf b/src/journal/journald-gperf.gperf
index 2f83cbd..32474df 100644
--- a/src/journal/journald-gperf.gperf
+++ b/src/journal/journald-gperf.gperf
@@ -1,5 +1,6 @@
 %{
 #include <stddef.h>
+#include <sys/socket.h>
 #include "conf-parser.h"
 #include "journald.h"
 %}
diff --git a/src/journal/journald-kmsg.c b/src/journal/journald-kmsg.c
index aebca5d..b259480 100644
--- a/src/journal/journald-kmsg.c
+++ b/src/journal/journald-kmsg.c
@@ -23,6 +23,7 @@
 #include <sys/epoll.h>
 #include <fcntl.h>
 #include <sys/mman.h>
+#include <sys/socket.h>
 
 #include <systemd/sd-messages.h>
 #include <libudev.h>
diff --git a/src/journal/journald-native.c b/src/journal/journald-native.c
index 7aa99a3..4e44c3a 100644
--- a/src/journal/journald-native.c
+++ b/src/journal/journald-native.c
@@ -20,6 +20,7 @@
 ***/
 
 #include <unistd.h>
+#include <stddef.h>
 #include <sys/epoll.h>
 
 #include "socket-util.h"
diff --git a/src/journal/journald-stream.c b/src/journal/journald-stream.c
index 3e4022a..113c421 100644
--- a/src/journal/journald-stream.c
+++ b/src/journal/journald-stream.c
@@ -21,6 +21,7 @@
 
 #include <fcntl.h>
 #include <unistd.h>
+#include <stddef.h>
 #include <sys/epoll.h>
 
 #ifdef HAVE_SELINUX
diff --git a/src/journal/journald-syslog.c b/src/journal/journald-syslog.c
index daed095..9db9198 100644
--- a/src/journal/journald-syslog.c
+++ b/src/journal/journald-syslog.c
@@ -20,6 +20,7 @@
 ***/
 
 #include <unistd.h>
+#include <stddef.h>
 #include <sys/epoll.h>
 
 #include "socket-util.h"
--
cgit v0.9.0.2-2-gbebe
