From 3bda4f376063971ad068f9c26edbec9029cfa7e6 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 19 Dec 2013 11:21:13 +0200
Subject: [PATCH] move /etc/systemd/system to /storage/.config/system.d

---
 Makefile.am |    4 ++--
 Makefile.in |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 8d9c587..93e9c6d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -144,7 +144,7 @@ udevlibexec_PROGRAMS =
 AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-DPKGSYSCONFDIR=\"$(pkgsysconfdir)\" \
-	-DSYSTEM_CONFIG_UNIT_PATH=\"$(pkgsysconfdir)/system\" \
+	-DSYSTEM_CONFIG_UNIT_PATH=\"/storage/.config/system.d\" \
 	-DSYSTEM_DATA_UNIT_PATH=\"$(systemunitdir)\" \
 	-DSYSTEM_SYSVINIT_PATH=\"$(SYSTEM_SYSVINIT_PATH)\" \
 	-DSYSTEM_SYSVRCND_PATH=\"$(SYSTEM_SYSVRCND_PATH)\" \
@@ -4209,7 +4209,7 @@ substitutions = \
        '|SYSTEMCTL=$(rootbindir)/systemctl|' \
        '|SYSTEMD_NOTIFY=$(rootbindir)/systemd-notify|' \
        '|pkgsysconfdir=$(pkgsysconfdir)|' \
-       '|SYSTEM_CONFIG_UNIT_PATH=$(pkgsysconfdir)/system|' \
+       '|SYSTEM_CONFIG_UNIT_PATH=/storage/.config/system.d|' \
        '|USER_CONFIG_UNIT_PATH=$(pkgsysconfdir)/user|' \
        '|pkgdatadir=$(pkgdatadir)|' \
        '|systemunitdir=$(systemunitdir)|' \
diff --git a/Makefile.in b/Makefile.in
index 6f3d475..66c4c22 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3744,7 +3744,7 @@ manual_tests = test-engine test-ns test-loopback test-hostname \
 AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-DPKGSYSCONFDIR=\"$(pkgsysconfdir)\" \
-	-DSYSTEM_CONFIG_UNIT_PATH=\"$(pkgsysconfdir)/system\" \
+	-DSYSTEM_CONFIG_UNIT_PATH=\"/storage/.config/system.d\" \
 	-DSYSTEM_DATA_UNIT_PATH=\"$(systemunitdir)\" \
 	-DSYSTEM_SYSVINIT_PATH=\"$(SYSTEM_SYSVINIT_PATH)\" \
 	-DSYSTEM_SYSVRCND_PATH=\"$(SYSTEM_SYSVRCND_PATH)\" \
@@ -6269,7 +6269,7 @@ substitutions = \
        '|SYSTEMCTL=$(rootbindir)/systemctl|' \
        '|SYSTEMD_NOTIFY=$(rootbindir)/systemd-notify|' \
        '|pkgsysconfdir=$(pkgsysconfdir)|' \
-       '|SYSTEM_CONFIG_UNIT_PATH=$(pkgsysconfdir)/system|' \
+       '|SYSTEM_CONFIG_UNIT_PATH=/storage/.config/system.d|' \
        '|USER_CONFIG_UNIT_PATH=$(pkgsysconfdir)/user|' \
        '|pkgdatadir=$(pkgdatadir)|' \
        '|systemunitdir=$(systemunitdir)|' \
-- 
1.7.2.5

