diff -Naur a/src/core/mount-setup.c b/src/core/mount-setup.c
--- a/src/core/mount-setup.c	2016-07-17 21:54:48.947510924 +0100
+++ b/src/core/mount-setup.c	2016-07-17 21:56:02.708552744 +0100
@@ -121,7 +121,10 @@
         /* Container bind mounts */
         "/proc/sys\0"
         "/dev/console\0"
-        "/proc/kmsg\0";
+        "/proc/kmsg\0"
+        /* LibreELEC: always busy */
+        "/flash\0"
+        "/storage\0";
 
 bool mount_point_is_api(const char *path) {
         unsigned i;
