diff -Naur xbmc-29428/configure.in xbmc-29428.patch/configure.in
--- xbmc-29428/configure.in	2010-04-22 00:56:44.000000000 +0200
+++ xbmc-29428.patch/configure.in	2010-04-22 19:06:31.906738563 +0200
@@ -478,7 +478,7 @@
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
-  if [ -z "$LIBCURL_BASENAME" ]; then
+  if [[ -z "$LIBCURL_BASENAME" ]]; then
     echo "Unable to determine basename of libcurl library"
   fi
 
