diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 17:12:32.000000000 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 19:39:55.720627540 +0200
@@ -10,6 +10,10 @@
 AM_INIT_AUTOMAKE([foreign])
 AC_CANONICAL_HOST
 
+tolower(){
+    echo "$@" | tr ABCDEFGHIJKLMNOPQRSTUVWXYZ abcdefghijklmnopqrstuvwxyz
+}
+
 # General message strings
 configure_debug="ERROR: this is a configure debug statement"
 missing_library="Could not find a required library. Please see the README for your platform."
@@ -88,6 +92,18 @@
             [use_debug=$enableval],
             [use_debug=yes])
 
+AC_ARG_WITH([arch],
+            [AS_HELP_STRING([--with-arch],
+               [build with given arch passing to internal ffmpeg (default is no, needed for crosscompiling)])],
+            [use_arch=$withval],
+            [use_arch=no])
+
+AC_ARG_WITH([cpu],
+            [AS_HELP_STRING([--with-cpu],
+               [build with given cpu passing to ffmpeg (default is no)])],
+            [use_cpu=$withval],
+            [use_cpu=no])
+
 AC_ARG_ENABLE([optimizations],
             [AS_HELP_STRING([--enable-optimizations],
                [enable optimization (default is yes)])],
@@ -791,6 +807,15 @@
   fi
 fi
 
+if test "$cross_compiling" = "yes"; then
+  final_message="$final_message\n  Crosscomp.:\tYes"
+else
+  final_message="$final_message\n  Crosscomp.:\tNo"
+fi
+
+final_message="$final_message\n  target ARCH:\t$use_arch"
+final_message="$final_message\n  target CPU:\t$use_cpu"
+
 if test "$use_optimizations" = "yes"; then
   final_message="$final_message\n  Optimization:\tYes"
   CXXFLAGS="$CXXFLAGS -O2"
