diff -Naur xbmc-26230/configure.in xbmc-26230.patch/configure.in
--- xbmc-26230/configure.in	2009-12-31 14:25:23.451197507 +0100
+++ xbmc-26230.patch/configure.in	2009-12-31 16:16:48.359195211 +0100
@@ -1120,6 +1120,8 @@
 XB_CONFIG_MODULE([xbmc/lib/libPython/Python], [
   if test "$host_vendor" = "apple" ; then
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1142,6 +1144,8 @@
     cp pyconfig.h Include	
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1159,6 +1163,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \    
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 -C \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-asm-optimizations \
       --disable-xmms-plugin \
@@ -1173,7 +1179,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/libvorbis], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1182,7 +1190,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/ogg], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1195,6 +1205,7 @@
       --disable-amd3dnow \
       --disable-static \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-devices \
@@ -1221,6 +1232,7 @@
       --disable-static \
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-decoder=mpeg_xvmc \
@@ -1233,6 +1245,7 @@
       --enable-postproc \
       --enable-gpl \
       `if test "$use_vdpau" = "yes"; then echo --enable-vdpau; else echo --disable-vdpau; fi` \
+      --disable-vaapi \
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
@@ -1246,12 +1259,16 @@
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     LDFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
@@ -1264,11 +1281,15 @@
   if test "$host_vendor" = "apple" ; then
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
@@ -1279,6 +1300,8 @@
   if test "$host_vendor" = "apple" ; then
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -D__DARWIN__ -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1288,6 +1311,8 @@
   else
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1303,6 +1328,8 @@
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1311,6 +1338,8 @@
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1322,6 +1351,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-O3 -fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --with-mp4v2
   else
@@ -1333,6 +1364,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1342,6 +1375,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1357,6 +1392,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1365,6 +1402,8 @@
       --without-x
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1378,6 +1417,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic
   else
     true
@@ -1388,17 +1429,23 @@
   if test "$host_vendor" = "apple" ; then
     export MACOSX_DEPLOYMENT_TARGET=10.4
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
 XB_CONFIG_MODULE([xbmc/lib/libass], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias
   else
     CFLAGS="$CFLAGS -fPIC" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [$USE_EXTERNAL_LIBASS])
 
@@ -1407,12 +1454,16 @@
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     CPPFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
   else
     CFLAGS=-O3 \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
@@ -1430,6 +1481,8 @@
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
     ./configure  \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-shared \
       --enable-static \
       --with-pic
@@ -1444,6 +1497,8 @@
       CXXFLAGS="-fPIC -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
       LDFLAGS="-L/opt/local/lib" \
       LIBS="-lz" \
+      --host=$host_alias \
+      --build=$build_alias \
       --x-includes=/usr/X11/include --x-libraries=/usr/X11/lib \
       --without-xscreensaver \
       --disable-sound \
@@ -1459,6 +1514,8 @@
     ./configure \
       CFLAGS="-fPIC" \ 
       CXXFLAGS="-fPIC" \
+      --host=$host_alias \
+      --build=$build_alias \
       --without-xscreensaver \
       --disable-sound \
       --disable-cyclone \
@@ -1474,9 +1531,13 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
-  else
-	./configure
+      ./configure --with-ssl-dir=/opt/local/include \
+      --host=$host_alias \
+      --build=$build_alias
+  else
+      ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
diff -Naur xbmc-26230/configure.in.orig xbmc-26230.patch/configure.in.orig
