diff -Naur xbmc-23757/configure.in xbmc-23757.patch/configure.in
--- xbmc-23757/configure.in	2009-10-17 01:37:17.452517140 +0200
+++ xbmc-23757.patch/configure.in	2009-10-17 01:31:52.602517899 +0200
@@ -263,6 +263,25 @@
 PASSED_CFLAGS=$CFLAGS # Hack to override autoconf default values
 AC_PROG_CC
 CFLAGS=$PASSED_CFLAGS
+
+if test x"$CC_FOR_BUILD" = x; then
+  if test x"$cross_compiling" = xyes; then
+    AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+  else
+    CC_FOR_BUILD="$CC"
+  fi
+fi
+AC_SUBST([CC_FOR_BUILD])
+
+if test x"$CXX_FOR_BUILD" = x; then
+  if test x"$cross_compiling" = xyes; then
+    AC_CHECK_PROGS(CXX_FOR_BUILD, g++ cpp)
+  else
+    CXX_FOR_BUILD="$CXX"
+  fi
+fi
+AC_SUBST([CXX_FOR_BUILD])
+
 AC_PROG_AWK
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
diff -Naur xbmc-23757/Makefile.include.in xbmc-23757.patch/Makefile.include.in
--- xbmc-23757/Makefile.include.in	2009-10-16 18:45:26.000000000 +0200
+++ xbmc-23757.patch/Makefile.include.in	2009-10-17 03:52:35.461519121 +0200
@@ -5,8 +5,10 @@
 ARCH=@ARCH@
 abs_top_srcdir=@abs_top_srcdir@
 prefix=@prefix@
-CXX=@CXX@
-CC=@CC@
+CXX?=@CXX@
+CC?=@CC@
+CXX_FOR_BUILD?=@CXX_FOR_BUILD@
+CC_FOR_BUILD?=@CC_FOR_BUILD@
 CXXFLAGS+=@CXXFLAGS@
 CXXFLAGS+=@DEBUG_FLAGS@
 CFLAGS+=@CFLAGS@
diff -Naur xbmc-23757/tools/TexturePacker/Makefile.in xbmc-23757.patch/tools/TexturePacker/Makefile.in
--- xbmc-23757/tools/TexturePacker/Makefile.in	2009-10-16 18:45:41.000000000 +0200
+++ xbmc-23757.patch/tools/TexturePacker/Makefile.in	2009-10-17 03:54:09.299518353 +0200
@@ -7,6 +7,9 @@
 LIBS = ../../xbmc/lib/libsquish/libsquish-@ARCH@.a -lSDL_image -lSDL -llzo2
 endif
 
+CC=$(CC_FOR_BUILD)
+CXX=$(CXX_FOR_BUILD)
+
 OBJS = \
 	SDL_anigif.o \
 	XBTFWriter.o \
@@ -21,4 +24,4 @@
 include ../../Makefile.include
 
 $(TARGET): $(OBJS)
-	$(CXX) $(OBJS) $(LDFLAGS) $(LIBS) -o $(TARGET)
+	$(CXX_FOR_BUILD) $(OBJS) $(LDFLAGS) $(LIBS) -o $(TARGET)
