diff -Naur xbmc-25850/configure.in xbmc-25850.patch/configure.in
--- xbmc-25850/configure.in	2009-12-19 18:47:15.600276781 +0100
+++ xbmc-25850.patch/configure.in	2009-12-19 18:45:16.404157176 +0100
@@ -43,6 +43,8 @@
 avahi_disabled="== Avahi support disabled. =="
 vdpau_not_found="== Could not find libvdpau. VDPAU support disabled. =="
 vdpau_disabled="== VDPAU support manually disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 
 # External library message strings
 external_libraries_enabled="== Use of all supported external libraries enabled. =="
@@ -141,6 +143,12 @@
             [use_goom=$enableval],
             [use_goom=no])
 
+AC_ARG_ENABLE([xbmcprojectm],
+            [AS_HELP_STRING([--enable-xbmcprojectm],
+               [enable XBMCProjectM visualisation (default is yes)])],
+            [use_xbmcprojectm=$enableval],
+            [use_xbmcprojectm=yes])
+
 AC_PATH_PROG(CCACHE,ccache,none)
 AC_ARG_ENABLE([ccache],
             [AS_HELP_STRING([--enable-ccache],
@@ -504,6 +512,15 @@
   fi
 fi
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "yes"; then
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  BUILD_XBMCPROJECTM=0
+else
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  BUILD_XBMCPROJECTM=1
+fi
+
 ### External libraries checks
 if test "$use_external_libraries" = "yes"; then
   AC_MSG_NOTICE($external_libraries_enabled)
@@ -833,6 +850,12 @@
   final_message="$final_message\n  GOOM:\t\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_pcre" = "yes"; then
   final_message="$final_message\n  PCRE Support:\tYes"
 else
@@ -1049,6 +1072,7 @@
 AC_SUBST(SDL_DEFINES)
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
+AC_SUBST(BUILD_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_LIBRARIES)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
@@ -1449,7 +1473,7 @@
   else
     rm -f CMakeCache.txt && CC="" CXX=""  cmake -D CMAKE_BUILD_TYPE:STRING=RelWithDebInfo -D USE_FTGL:BOOL=OFF .
   fi
-], [0])
+], [$BUILD_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
diff -Naur xbmc-25850/Makefile.in xbmc-25850.patch/Makefile.in
--- xbmc-25850/Makefile.in	2009-12-19 13:34:46.000000000 +0100
+++ xbmc-25850.patch/Makefile.in	2009-12-19 18:45:16.406251838 +0100
@@ -100,8 +100,10 @@
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
+VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 ifneq (@BUILD_GOOM@,1)
 VIS_DIRS+=xbmc/visualizations/Goom
 endif
@@ -262,7 +264,9 @@
 visualizations: exports
 	$(MAKE) -C xbmc/visualizations/OpenGLSpectrum
 	$(MAKE) -C xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
 	$(MAKE) -C xbmc/visualizations/XBMCProjectM
+endif
 ifneq (@BUILD_GOOM@,1)
 ifeq ($(or $(findstring powerpc-linux,$(ARCH)),$(findstring powerpc64-linux,$(ARCH))),)
 	$(MAKE) -C xbmc/visualizations/Goom
