diff -Naur xbmc-23782/configure.in xbmc-23782.patch/configure.in
--- xbmc-23782/configure.in	2009-10-17 23:10:17.764644091 +0200
+++ xbmc-23782.patch/configure.in	2009-10-17 23:11:34.893517566 +0200
@@ -45,6 +45,8 @@
 vdpau_disabled="== VDPAU support manually disabled. =="
 xbmcprojectm_enabled="== XBMCProjectM enabled. =="
 xbmxprojectm_disabled="== XBMCProjectM disabled. =="
+rsxs_enabled="== RSXS enabled. =="
+rsxs_disabled="== RSXS disabled. =="
 
 # External library message strings
 external_libraries_enabled="== Use of all supported external libraries enabled. =="
@@ -149,6 +151,12 @@
             [use_xbmcprojectm=$enableval],
             [use_xbmcprojectm=yes])
 
+AC_ARG_ENABLE([rsxs],
+            [AS_HELP_STRING([--enable-rsxs],
+               [enable RSXS Screensaver (default is yes)])],
+            [use_rsxs=$enableval],
+            [use_rsxs=yes])
+
 AC_PATH_PROG(CCACHE,ccache,none)
 AC_ARG_ENABLE([ccache],
             [AS_HELP_STRING([--enable-ccache],
@@ -513,6 +521,15 @@
   BUILD_XBMCPROJECTM=1
 fi
 
+# RSXS
+if test "$use_rsxs" = "yes"; then
+  AC_MSG_NOTICE($rsxs_enabled)
+  BUILD_RSXS=0
+else
+  AC_MSG_NOTICE($rsxs_disabled)
+  BUILD_RSXS=1
+fi
+
 ### External libraries checks
 if test "$use_external_libraries" = "yes"; then
   AC_MSG_NOTICE($external_libraries_enabled)
@@ -843,6 +860,12 @@
   final_message="$final_message\n  XBMCProjectM:\tNo"
 fi
 
+if test "$use_rsxs" = "yes"; then
+  final_message="$final_message\n  RSXS:\t\tYes"
+else
+  final_message="$final_message\n  RSXS:\t\tNo"
+fi
+
 if test "$use_pcre" = "yes"; then
   final_message="$final_message\n  PCRE Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DHAS_PCRE"
@@ -1031,6 +1054,7 @@
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
 AC_SUBST(BUILD_XBMCPROJECTM)
+AC_SUBST(BUILD_RSXS)
 AC_SUBST(USE_EXTERNAL_LIBRARIES)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
@@ -1427,7 +1451,7 @@
       --disable-lattice \
       --disable-skyrocket
   fi
-], [0])
+], [$BUILD_RSXS])
 
 AC_OUTPUT
 
diff -Naur xbmc-23782/Makefile.in xbmc-23782.patch/Makefile.in
--- xbmc-23782/Makefile.in	2009-10-17 23:10:17.769644442 +0200
+++ xbmc-23782.patch/Makefile.in	2009-10-17 23:10:30.315519233 +0200
@@ -90,8 +90,10 @@
 	xbmc/lib/libid3tag \
 	xbmc/lib/libPython/linux
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
+ifneq (@BUILD_RSXS@,1)
+SS_DIRS=+xbmc/screensavers/rsxs-0.9/xbmc
+endif
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
@@ -266,7 +268,9 @@
 endif
 endif
 screensavers: exports
+ifneq (@BUILD_RSXS@,1)
 	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
+endif
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
