diff -Naur xbmc-26230/configure.in xbmc-26230.patch/configure.in
--- xbmc-26230/configure.in	2009-12-31 16:21:14.918195394 +0100
+++ xbmc-26230.patch/configure.in	2009-12-31 16:22:02.708322440 +0100
@@ -919,12 +919,6 @@
   final_message="$final_message\n  MID Support:\tNo"
 fi
 
-if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
-else
-  INCLUDES="$INCLUDES"
-fi
-
 if test "$use_ccache" = "yes"; then
   if test "$ac_cv_path_CCACHE" = "none"; then
      echo "ccache not found. Falling back to default CC"
diff -Naur xbmc-26230/configure.in.orig xbmc-26230.patch/configure.in.orig
diff -Naur xbmc-26230/tools/TexturePacker/XBMCTex.cpp xbmc-26230.patch/tools/TexturePacker/XBMCTex.cpp
--- xbmc-26230/tools/TexturePacker/XBMCTex.cpp	2009-12-31 14:28:41.435233446 +0100
+++ xbmc-26230.patch/tools/TexturePacker/XBMCTex.cpp	2009-12-31 16:22:02.711322585 +0100
@@ -37,7 +37,7 @@
 #endif
 
 #ifdef _LINUX
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 #else
 #include "../../xbmc/lib/liblzo/LZO1X.H"
 #endif
diff -Naur xbmc-26230/tools/TexturePacker/XBMCTex.cpp.orig xbmc-26230.patch/tools/TexturePacker/XBMCTex.cpp.orig
