diff -Naur xbmc-23782/guilib/Makefile xbmc-23782.patch/guilib/Makefile
--- xbmc-23782/guilib/Makefile	2009-10-17 07:13:16.000000000 +0200
+++ xbmc-23782.patch/guilib/Makefile	2009-10-17 22:26:41.936517267 +0200
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 
 SRCS=AnimatedGif.cpp \
      AudioContext.cpp \
diff -Naur xbmc-23782/xbmc/FileSystem/Makefile xbmc-23782.patch/xbmc/FileSystem/Makefile
--- xbmc-23782/xbmc/FileSystem/Makefile	2009-10-17 07:09:13.000000000 +0200
+++ xbmc-23782.patch/xbmc/FileSystem/Makefile	2009-10-17 22:26:38.683529959 +0200
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils
 INCLUDES+=-I../lib/libcdio/libcdio/include
 
 CXXFLAGS+=-D__STDC_FORMAT_MACROS \
diff -Naur xbmc-23782/xbmc/Makefile xbmc-23782.patch/xbmc/Makefile
--- xbmc-23782/xbmc/Makefile	2009-10-17 07:09:12.000000000 +0200
+++ xbmc-23782.patch/xbmc/Makefile	2009-10-17 22:25:18.953643441 +0200
@@ -1,4 +1,4 @@
-INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/ffmpeg
+INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer -Icores/ffmpeg
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
