diff -Naur xbmc-24229/guilib/Makefile xbmc-24229.patch/guilib/Makefile
--- xbmc-24229/guilib/Makefile	2009-11-03 10:52:46.493397566 +0100
+++ xbmc-24229.patch/guilib/Makefile	2009-11-03 12:18:27.292521966 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 
 SRCS=AnimatedGif.cpp \
      AudioContext.cpp \
diff -Naur xbmc-24229/xbmc/FileSystem/Makefile xbmc-24229.patch/xbmc/FileSystem/Makefile
--- xbmc-24229/xbmc/FileSystem/Makefile	2009-11-03 10:49:58.819396264 +0100
+++ xbmc-24229.patch/xbmc/FileSystem/Makefile	2009-11-03 12:18:27.293521205 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 
 CXXFLAGS+=-D__STDC_FORMAT_MACROS \
diff -Naur xbmc-24229/xbmc/Makefile xbmc-24229.patch/xbmc/Makefile
--- xbmc-24229/xbmc/Makefile	2009-11-03 10:49:58.642519675 +0100
+++ xbmc-24229.patch/xbmc/Makefile	2009-11-03 12:18:27.295522056 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/ffmpeg
+INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer -Icores/ffmpeg
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
