diff -Naur xbmc-26925/guilib/Makefile xbmc-26925.patch/guilib/Makefile
--- xbmc-26925/guilib/Makefile	2010-01-17 00:46:52.136847980 +0100
+++ xbmc-26925.patch/guilib/Makefile	2010-01-17 03:49:32.002148640 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores/dvdplayer/Codecs/ffmpeg -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores/dvdplayer/Codecs/ffmpeg -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 
 SRCS=AnimatedGif.cpp \
      AudioContext.cpp \
diff -Naur xbmc-26925/xbmc/FileSystem/Makefile.in xbmc-26925.patch/xbmc/FileSystem/Makefile.in
--- xbmc-26925/xbmc/FileSystem/Makefile.in	2010-01-17 00:48:14.996722616 +0100
+++ xbmc-26925.patch/xbmc/FileSystem/Makefile.in	2010-01-17 03:49:57.202148904 +0100
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 ifeq ($(findstring osx,$(ARCH)), osx)
 INCLUDES+=-I../../lib/libcurl-OSX/include
diff -Naur xbmc-26925/xbmc/Makefile xbmc-26925.patch/xbmc/Makefile
--- xbmc-26925/xbmc/Makefile	2010-01-17 00:47:31.173847479 +0100
+++ xbmc-26925.patch/xbmc/Makefile	2010-01-17 03:50:20.319148465 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/dvdplayer/Codecs/ffmpeg
+INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer -Icores/dvdplayer/Codecs/ffmpeg
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
