diff -Naur xbmc-26230/configure.in xbmc-26230.patch/configure.in
--- xbmc-26230/configure.in	2009-12-31 16:31:24.629197420 +0100
+++ xbmc-26230.patch/configure.in	2009-12-31 16:33:49.712198006 +0100
@@ -1578,6 +1578,7 @@
       --build=$build_alias
   else
       ./configure \
+      `if test "$cross_compiling" = "yes"; then echo ac_cv_openssl_linker_option=no; fi` \
       --host=$host_alias \
       --build=$build_alias
   fi
diff -Naur xbmc-26230/configure.in.orig xbmc-26230.patch/configure.in.orig
