diff -Naur xbmc-25850/configure.in xbmc-25850.patch/configure.in
--- xbmc-25850/configure.in	2009-12-19 19:08:26.227140824 +0100
+++ xbmc-25850.patch/configure.in	2009-12-19 19:08:01.262251286 +0100
@@ -1556,6 +1556,7 @@
       --host=$host_alias \
       --build=$build_alias
   else
       ./configure \
+      `if test "$cross_compiling" = "yes"; then echo ac_cv_openssl_linker_option=no; fi` \
       --host=$host_alias \
       --build=$build_alias
diff -Naur xbmc-25850/xbmc/cores/dvdplayer/Codecs/libbdnav/config.log xbmc-25850.patch/xbmc/cores/dvdplayer/Codecs/libbdnav/config.log
