diff -Naur xbmc-24229/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h xbmc-24229.patch/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h
--- xbmc-24229/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h	2009-11-03 10:51:21.827519184 +0100
+++ xbmc-24229.patch/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h	2009-11-03 12:31:01.038437018 +0100
@@ -29,10 +29,14 @@
 
 class CMutex;
 
-#if defined(_LINUX) && !defined(__APPLE__)
+#if defined(_LINUX) && !defined(__APPLE__) && !defined(__UCLIBC__)
 #define _file _fileno
 #endif
 
+#if defined(__UCLIBC__)
+#define _file __filedes
+#endif
+
 #define MAX_EMULATED_FILES    50
 #define FILE_WRAPPER_OFFSET   0x00000100
 
