diff -Naur xbmc-24572/configure.in xbmc-24572.patch/configure.in
--- xbmc-24572/configure.in	2009-11-13 20:37:01.756638545 +0100
+++ xbmc-24572.patch/configure.in	2009-11-13 20:41:14.259621759 +0100
@@ -352,6 +352,12 @@
 AC_CHECK_LIB([z],           [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
+AC_CHECK_LIB([pthread],     [pthread_yield], 
+		  AC_DEFINE([HAVE_PTHREAD_YIELD], [1],
+                  [Define to 1 if you have the 'pthread_yield' function]))
+AC_CHECK_LIB([pthread],     [sched_yield],
+                  AC_DEFINE([HAVE_SCHED_YIELD], [1],
+                  [Define to 1 if you have the 'sched_yield' function]))
 PKG_CHECK_MODULES([MAD],        [mad],
                   [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
                   AC_MSG_ERROR($missing_library))
diff -Naur xbmc-24572/xbmc/FileSystem/FileCurl.cpp xbmc-24572.patch/xbmc/FileSystem/FileCurl.cpp
--- xbmc-24572/xbmc/FileSystem/FileCurl.cpp	2009-11-13 18:19:50.000000000 +0100
+++ xbmc-24572.patch/xbmc/FileSystem/FileCurl.cpp	2009-11-13 20:38:20.330748551 +0100
@@ -19,6 +19,8 @@
  *
  */
 
+#include "config.h"
+
 #include "FileCurl.h"
 #include "Util.h"
 #include "URL.h"
@@ -1179,10 +1181,10 @@
         // happens especially on ftp during initial connection
 #ifndef _LINUX
         SwitchToThread();
-#elif __APPLE__
-        sched_yield();
-#else
+#elif HAVE_PTHREAD_YIELD
         pthread_yield();
+#elif HAVE_SCHED_YIELD
+        sched_yield();
 #endif
 
         FD_ZERO(&fdread);
