diff -Naur xbmc-24229/configure.in xbmc-24229.patch/configure.in
--- xbmc-24229/configure.in	2009-11-03 12:18:12.098521374 +0100
+++ xbmc-24229.patch/configure.in	2009-11-03 12:37:53.688395337 +0100
@@ -350,6 +350,12 @@
 AC_CHECK_LIB([pthread],     [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([lzo2],        [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([z],           [main],, AC_MSG_ERROR($missing_library))
+AC_CHECK_LIB([pthread],     [pthread_yield], 
+                  AC_DEFINE([HAVE_PTHREAD_YIELD], [1],
+                  [Define to 1 if you have the 'pthread_yield' function]))
+AC_CHECK_LIB([pthread],     [sched_yield],
+                  AC_DEFINE([HAVE_SCHED_YIELD], [1],
+                  [Define to 1 if you have the 'sched_yield' function]))
 PKG_CHECK_MODULES([MAD],        [mad],
                   [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
                   AC_MSG_ERROR($missing_library))
diff -Naur xbmc-24229/xbmc/FileSystem/FileCurl.cpp xbmc-24229.patch/xbmc/FileSystem/FileCurl.cpp
--- xbmc-24229/xbmc/FileSystem/FileCurl.cpp	2009-11-03 10:49:58.898396757 +0100
+++ xbmc-24229.patch/xbmc/FileSystem/FileCurl.cpp	2009-11-03 12:35:23.238521816 +0100
@@ -19,6 +19,8 @@
  *
  */
 
+#include "config.h"
+
 #include "FileCurl.h"
 #include "Util.h"
 #include "URL.h"
@@ -1178,10 +1180,10 @@
         // happens especially on ftp during initial connection
 #ifndef _LINUX
         SwitchToThread();
-#elif __APPLE__
-        sched_yield();
-#else
+#elif HAVE_PTHREAD_YIELD
         pthread_yield();
+#elif HAVE_SCHED_YIELD
+        sched_yield();
 #endif
 
         FD_ZERO(&fdread);
