diff -Naur xbmc-24229/configure.in xbmc-24229.patch/configure.in
--- xbmc-24229/configure.in	2009-11-03 12:40:25.588436224 +0100
+++ xbmc-24229.patch/configure.in	2009-11-03 12:43:02.056519683 +0100
@@ -359,6 +359,12 @@
 AC_CHECK_LIB([pthread],     [sched_yield],
                   AC_DEFINE([HAVE_SCHED_YIELD], [1],
                   [Define to 1 if you have the 'sched_yield' function]))
+AC_CHECK_LIB([c],           [__xstat64],
+                  AC_DEFINE([HAVE_XSTAT64], [1],
+                  [Define to 1 if you have the '__xstat64' function]))
+AC_CHECK_LIB([c],           [__lxstat64],
+                  AC_DEFINE([HAVE_LXSTAT64], [1],
+                  [Define to 1 if you have the '__lxstat64' function]))
 PKG_CHECK_MODULES([MAD],        [mad],
                   [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
                   AC_MSG_ERROR($missing_library))
diff -Naur xbmc-24229/xbmc/cores/DllLoader/exports/exports_python_linux.cpp xbmc-24229.patch/xbmc/cores/DllLoader/exports/exports_python_linux.cpp
--- xbmc-24229/xbmc/cores/DllLoader/exports/exports_python_linux.cpp	2009-11-03 10:51:21.801520187 +0100
+++ xbmc-24229.patch/xbmc/cores/DllLoader/exports/exports_python_linux.cpp	2009-11-03 12:40:50.204524744 +0100
@@ -19,6 +19,8 @@
  *
  */
 
+#include "config.h"
+
 #include "system.h"
 #include <sys/types.h>
 #include <utime.h>
@@ -224,32 +226,32 @@
   return opendir(strName.c_str());
 }
 
-#ifdef __APPLE__
+#ifdef HAVE_XSTAT64 && HAVE_LXSTAT64
 
-int xbp_stat(const char * path, struct stat * buf)
+int xbp__xstat64(int ver, const char *filename, struct stat64 *stat_buf)
 {
-  CStdString strName = _P(path);
-  return stat(strName.c_str(), buf);
+  CStdString strName = _P(filename);
+  return __xstat64(ver, strName.c_str(), stat_buf);
 }
 
-int xbp_lstat(const char * path, struct stat * buf)
+int xbp__lxstat64(int ver, const char *filename, struct stat64 *stat_buf)
 {
-  CStdString strName = _P(path);
-  return lstat(strName.c_str(), buf);
+  CStdString strName = _P(filename);
+  return __lxstat64(ver, strName.c_str(), stat_buf);
 }
 
 #else
 
-int xbp__xstat64(int ver, const char *filename, struct stat64 *stat_buf)
+int xbp_stat(const char * path, struct stat * buf)
 {
-  CStdString strName = _P(filename);
-  return __xstat64(ver, strName.c_str(), stat_buf);
+  CStdString strName = _P(path);
+  return stat(strName.c_str(), buf);
 }
 
-int xbp__lxstat64(int ver, const char *filename, struct stat64 *stat_buf)
+int xbp_lstat(const char * path, struct stat * buf)
 {
-  CStdString strName = _P(filename);
-  return __lxstat64(ver, strName.c_str(), stat_buf);
+  CStdString strName = _P(path);
+  return lstat(strName.c_str(), buf);
 }
 
 #endif
