diff -Naur xbmc-24233/xbmc/GUISettings.cpp xbmc-24233.patch/xbmc/GUISettings.cpp
--- xbmc-24233/xbmc/GUISettings.cpp	2009-11-03 13:43:17.810534312 +0100
+++ xbmc-24233.patch/xbmc/GUISettings.cpp	2009-11-03 14:57:33.414520535 +0100
@@ -493,20 +493,20 @@
   if (g_application.IsStandAlone())
   {
 #ifndef __APPLE__
-    AddString(0, "network.interface",775,"", SPIN_CONTROL_TEXT);
-    AddInt(0, "network.assignment", 715, NETWORK_DHCP, NETWORK_DHCP, 1, NETWORK_DISABLED, SPIN_CONTROL_TEXT);
-    AddString(0, "network.ipaddress", 719, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
-    AddString(0, "network.subnet", 720, "255.255.255.0", EDIT_CONTROL_IP_INPUT);
-    AddString(0, "network.gateway", 721, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
-    AddString(0, "network.dns", 722, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
-    AddString(0, "network.dnssuffix", 22002, "", EDIT_CONTROL_INPUT, true);
-    AddString(0, "network.essid", 776, "0.0.0.0", BUTTON_CONTROL_STANDARD);
-    AddInt(0, "network.enc", 778, ENC_NONE, ENC_NONE, 1, ENC_WPA2, SPIN_CONTROL_TEXT);
-    AddString(0, "network.key", 777, "0.0.0.0", EDIT_CONTROL_INPUT);
+    AddString(1, "network.interface",775,"", SPIN_CONTROL_TEXT);
+    AddInt(2, "network.assignment", 715, NETWORK_DHCP, NETWORK_DHCP, 1, NETWORK_DISABLED, SPIN_CONTROL_TEXT);
+    AddString(3, "network.ipaddress", 719, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
+    AddString(4, "network.subnet", 720, "255.255.255.0", EDIT_CONTROL_IP_INPUT);
+    AddString(5, "network.gateway", 721, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
+    AddString(6, "network.dns", 722, "0.0.0.0", EDIT_CONTROL_IP_INPUT);
+    AddString(7, "network.dnssuffix", 22002, "", EDIT_CONTROL_INPUT, true);
+    AddString(8, "network.essid", 776, "0.0.0.0", BUTTON_CONTROL_STANDARD);
+    AddInt(9, "network.enc", 778, ENC_NONE, ENC_NONE, 1, ENC_WPA2, SPIN_CONTROL_TEXT);
+    AddString(10, "network.key", 777, "0.0.0.0", EDIT_CONTROL_INPUT);
 #ifndef _WIN32
-    AddString(0, "network.save", 779, "", BUTTON_CONTROL_STANDARD);
+    AddString(11, "network.save", 779, "", BUTTON_CONTROL_STANDARD);
 #endif
-    AddSeparator(0, "network.sep1");
+    AddSeparator(12, "network.sep1");
 #endif
   }
   AddBool(13, "network.usehttpproxy", 708, false);
diff -Naur xbmc-24233/xbmc/GUIWindowSettingsCategory.cpp xbmc-24233.patch/xbmc/GUIWindowSettingsCategory.cpp
--- xbmc-24233/xbmc/GUIWindowSettingsCategory.cpp	2009-11-03 13:43:28.129521550 +0100
+++ xbmc-24233.patch/xbmc/GUIWindowSettingsCategory.cpp	2009-11-03 10:50:13.000000000 +0100
@@ -3702,7 +3702,12 @@
 
 void CGUIWindowSettingsCategory::NetworkInterfaceChanged(void)
 {
+  if (!g_application.IsStandAlone())
+    return;
+
+#ifdef __APPLE__
   return;
+#endif
 
    NetworkAssignment iAssignment;
    CStdString sIPAddress;
