diff -Naur xbmc-24572/xbmc/utils/Splash.cpp xbmc-24572.patch/xbmc/utils/Splash.cpp
--- xbmc-24572/xbmc/utils/Splash.cpp	2009-11-13 18:19:48.374746147 +0100
+++ xbmc-24572.patch/xbmc/utils/Splash.cpp	2009-11-14 16:58:52.677231191 +0100
@@ -53,9 +53,9 @@
   g_graphicsContext.Clear();
 
   g_graphicsContext.SetCameraPosition(CPoint(0, 0));
-  float w = g_graphicsContext.GetWidth() * 0.5f;
-  float h = g_graphicsContext.GetHeight() * 0.5f;
-  CGUIImage* image = new CGUIImage(0, 0, w*0.5f, h*0.5f, w, h, m_ImageName);
+  float w = g_graphicsContext.GetWidth();
+  float h = g_graphicsContext.GetHeight();
+  CGUIImage* image = new CGUIImage(0, 0, 0, 0, w, h, m_ImageName);
   image->SetAspectRatio(CAspectRatio::AR_KEEP);
   image->AllocResources();
 
