diff -Naur xbmc-28745/configure.in xbmc-28745.patch/configure.in
--- xbmc-28745/configure.in	2010-03-23 04:05:13.476967336 +0100
+++ xbmc-28745.patch/configure.in	2010-03-23 04:07:49.166967342 +0100
@@ -728,6 +728,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1269,6 +1270,7 @@
       --enable-protocol=http \
       --enable-pthreads \
       `if test "$host_cpu" = "powerpc" || test "$host_cpu" = "powerpc64"; then echo --enable-pic; fi` \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --cc="$CC"
   fi
 ], [$USE_EXTERNAL_FFMPEG])
