diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:24:56.419753331 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:25:57.248628460 +0200
@@ -1275,6 +1275,8 @@
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
