diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:35:03.335660103 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:36:09.419753169 +0200
@@ -1316,6 +1316,8 @@
     ./configure2 \
       --extra-cflags="$CFLAGS $DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --disable-strip \
