diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:43:26.857628415 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:44:02.530754199 +0200
@@ -1375,6 +1375,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
