diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:47:39.398628679 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:48:18.635753312 +0200
@@ -1432,6 +1432,8 @@
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --with-pic
   fi
