diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:50:17.774753500 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:50:47.902628564 +0200
@@ -1492,6 +1492,8 @@
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
+      --host=$host_alias \
+      --build=$build_alias \
       --without-xscreensaver \
       --disable-sound \
       --disable-cyclone \
