diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:51:28.016629554 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:52:26.372628570 +0200
@@ -1509,9 +1509,11 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
+    ./configure --with-ssl-dir=/opt/local/include
   else
-	./configure --without-openssl
+    ./configure --without-openssl \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
