diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 18:54:17.023663284 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 18:55:01.867627572 +0200
@@ -1552,7 +1552,9 @@
   if test "$host_vendor" = "apple" ; then
     ./configure --disable-nls --with-libiconv-prefix=/opt/local --with-libintl-prefix=/opt/local --without-readline
   else
-    ./configure --disable-nls
+    ./configure --disable-nls \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
