diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:25:37.000000000 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:46:16.387376489 +0200
@@ -23,6 +23,8 @@
 xrandr_disabled="== XRandR support disabled. SDL will be used for resolution support. =="
 goom_enabled="== GOOM enabled. =="
 goom_disabled="== GOOM disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -161,6 +163,12 @@
   [use_goom=$enableval],
   [use_goom=no])
 
+AC_ARG_ENABLE([xbmcprojectm],
+            [AS_HELP_STRING([--enable-xbmcprojectm],
+               [enable XBMCProjectM visualisation (default is yes)])],
+            [use_xbmcprojectm=$enableval],
+            [use_xbmcprojectm=yes])
+
 AC_ARG_ENABLE([ccache],
   [AS_HELP_STRING([--enable-ccache],
   [enable building with ccache feature (default is auto)])],
@@ -725,6 +733,15 @@
 AC_CHECK_HEADERS([librtmp/log.h librtmp/amf.h librtmp/rtmp.h],use_librtmp=yes,
 use_librtmp=no;AC_MSG_RESULT($librtmp_not_found))
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "yes"; then
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  BUILD_XBMCPROJECTM=0
+else
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  BUILD_XBMCPROJECTM=1
+fi
+
 ### External libraries checks
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
@@ -1132,6 +1149,12 @@
   final_message="$final_message\n  GOOM:\t\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_mid" = "yes"; then
   final_message="$final_message\n  MID Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DMID"
@@ -1401,6 +1424,7 @@
 AC_SUBST(SDL_DEFINES)
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
+AC_SUBST(BUILD_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
 AC_SUBST(USE_INTERNAL_LIBA52)
@@ -1788,7 +1812,7 @@
       -DSYSCONF_INSTALL_DIR:PATH="${sysconfdir}" -DSHARE_INSTALL_PREFIX:PATH="${datadir}" .
 set +x
   fi
-], [0])
+], [$BUILD_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
diff -Naur xbmc-31903/Makefile.in xbmc-31903.patch/Makefile.in
--- xbmc-31903/Makefile.in	2010-07-17 16:25:37.000000000 +0200
+++ xbmc-31903.patch/Makefile.in	2010-07-17 16:46:16.391376663 +0200
@@ -108,8 +108,10 @@
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
+VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 VIS_DIRS+=xbmc/visualizations/iTunes
 endif
@@ -307,7 +309,9 @@
 ifneq (arm, $(ARCH))
 	$(MAKE) -C xbmc/visualizations/OpenGLSpectrum
 	$(MAKE) -C xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
 	$(MAKE) -C xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 	$(MAKE) -C xbmc/visualizations/iTunes
 endif
