diff -Naur xbmc-30122/configure.in xbmc-30122.patch/configure.in
--- xbmc-30122/configure.in	2010-05-13 13:16:42.000000000 +0200
+++ xbmc-30122.patch/configure.in	2010-05-13 13:44:00.985198286 +0200
@@ -23,6 +23,8 @@
 xrandr_disabled="== XRandR support disabled. SDL will be used for resolution support. =="
 goom_enabled="== GOOM enabled. =="
 goom_disabled="== GOOM disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -170,6 +172,12 @@
             [use_goom=$enableval],
             [use_goom=no])
 
+AC_ARG_ENABLE([xbmcprojectm],
+            [AS_HELP_STRING([--enable-xbmcprojectm],
+               [enable XBMCProjectM visualisation (default is yes)])],
+            [use_xbmcprojectm=$enableval],
+            [use_xbmcprojectm=yes])
+
 AC_ARG_ENABLE([ccache],
             [AS_HELP_STRING([--enable-ccache],
                [enable building with ccache feature (default is auto)])],
@@ -608,6 +616,15 @@
   fi
 fi
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "yes"; then
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  BUILD_XBMCPROJECTM=0
+else
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  BUILD_XBMCPROJECTM=1
+fi
+
 ### External libraries checks
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
@@ -992,6 +1009,12 @@
   final_message="$final_message\n  GOOM:\t\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_mid" = "yes"; then
   final_message="$final_message\n  MID Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DMID"
@@ -1245,6 +1268,7 @@
 AC_SUBST(SDL_DEFINES)
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
+AC_SUBST(BUILD_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
 AC_SUBST(USE_EXTERNAL_LIBASS)
@@ -1626,7 +1650,7 @@
       -DSYSCONF_INSTALL_DIR:PATH="${sysconfdir}" -DSHARE_INSTALL_PREFIX:PATH="${datadir}" .
 set +x
   fi
-], [0])
+], [$BUILD_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
diff -Naur xbmc-30122/Makefile.in xbmc-30122.patch/Makefile.in
--- xbmc-30122/Makefile.in	2010-05-13 13:16:42.000000000 +0200
+++ xbmc-30122.patch/Makefile.in	2010-05-13 13:41:15.585324083 +0200
@@ -106,8 +106,10 @@
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
+VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 VIS_DIRS+=xbmc/visualizations/iTunes
 endif
@@ -307,7 +309,9 @@
 visualizations: exports
 	$(MAKE) -C xbmc/visualizations/OpenGLSpectrum
 	$(MAKE) -C xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
 	$(MAKE) -C xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 	$(MAKE) -C xbmc/visualizations/iTunes
 endif
