diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:47:46.246501133 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:47:17.801376519 +0200
@@ -25,6 +25,8 @@
 goom_disabled="== GOOM disabled. =="
 xbmcprojectm_enabled="== XBMCProjectM enabled. =="
 xbmxprojectm_disabled="== XBMCProjectM disabled. =="
+rsxs_enabled="== RSXS enabled. =="
+rsxs_disabled="== RSXS disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -169,6 +171,12 @@
             [use_xbmcprojectm=$enableval],
             [use_xbmcprojectm=yes])
 
+AC_ARG_ENABLE([rsxs],
+            [AS_HELP_STRING([--enable-rsxs],
+               [enable RSXS Screensaver (default is yes)])],
+            [use_rsxs=$enableval],
+            [use_rsxs=yes])
+
 AC_ARG_ENABLE([ccache],
   [AS_HELP_STRING([--enable-ccache],
   [enable building with ccache feature (default is auto)])],
@@ -742,6 +750,15 @@
   BUILD_XBMCPROJECTM=1
 fi
 
+# RSXS
+if test "$use_rsxs" = "yes"; then
+  AC_MSG_NOTICE($rsxs_enabled)
+  BUILD_RSXS=0
+else
+  AC_MSG_NOTICE($rsxs_disabled)
+  BUILD_RSXS=1
+fi
+
 ### External libraries checks
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
@@ -1155,6 +1172,12 @@
   final_message="$final_message\n  XBMCProjectM:\tNo"
 fi
 
+if test "$use_rsxs" = "yes"; then
+  final_message="$final_message\n  RSXS:\t\tYes"
+else
+  final_message="$final_message\n  RSXS:\t\tNo"
+fi
+
 if test "$use_mid" = "yes"; then
   final_message="$final_message\n  MID Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DMID"
@@ -1425,6 +1448,7 @@
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
 AC_SUBST(BUILD_XBMCPROJECTM)
+AC_SUBST(BUILD_RSXS)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
 AC_SUBST(USE_INTERNAL_LIBA52)
@@ -1866,7 +1890,7 @@
       --disable-skyrocket
   fi
 fi
-], [0])
+], [$BUILD_RSXS])
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
diff -Naur xbmc-31903/Makefile.in xbmc-31903.patch/Makefile.in
--- xbmc-31903/Makefile.in	2010-07-17 16:47:46.258501239 +0200
+++ xbmc-31903.patch/Makefile.in	2010-07-17 16:47:17.803376676 +0200
@@ -103,8 +103,10 @@
 	lib/xbmc-dll-symbols \
 	lib/jsoncpp/jsoncpp/src/lib_json
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
+ifneq (@BUILD_RSXS@,1)
+SS_DIRS=+xbmc/screensavers/rsxs-0.9/xbmc
+endif
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
@@ -323,8 +325,10 @@
 endif
 screensavers: exports
 ifneq (arm, $(ARCH))
+ifneq (@BUILD_RSXS@,1)
 	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
 endif
+endif
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
