diff -Naur xbmc-30122/configure.in xbmc-30122.patch/configure.in
--- xbmc-30122/configure.in	2010-05-13 13:51:13.813324070 +0200
+++ xbmc-30122.patch/configure.in	2010-05-13 13:52:55.358197998 +0200
@@ -1065,12 +1065,6 @@
   final_message="$final_message\n  MID Support:\tNo"
 fi
 
-if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
-else
-  INCLUDES="$INCLUDES"
-fi
-
 if test "x$use_ccache" != "xno"; then
   AC_PATH_PROG(CCACHE,ccache,none)
   if test "$ac_cv_path_CCACHE" = "none"; then
diff -Naur xbmc-30122/tools/TexturePacker/XBMCTex.cpp xbmc-30122.patch/tools/TexturePacker/XBMCTex.cpp
--- xbmc-30122/tools/TexturePacker/XBMCTex.cpp	2010-05-13 13:20:20.000000000 +0200
+++ xbmc-30122.patch/tools/TexturePacker/XBMCTex.cpp	2010-05-13 13:51:39.760198949 +0200
@@ -37,7 +37,7 @@
 #endif
 
 #ifdef _LINUX
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 #else
 #include "../../xbmc/lib/liblzo/LZO1X.H"
 #endif
