diff -Naur xbmc-28910/guilib/Makefile.in xbmc-28910.patch/guilib/Makefile.in
--- xbmc-28910/guilib/Makefile.in	2010-03-28 17:12:32.000000000 +0200
+++ xbmc-28910.patch/guilib/Makefile.in	2010-03-28 19:13:23.816629379 +0200
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-I../xbmc/cores/dvdplayer/Codecs/ffmpeg
 endif
diff -Naur xbmc-28910/xbmc/FileSystem/Makefile.in xbmc-28910.patch/xbmc/FileSystem/Makefile.in
--- xbmc-28910/xbmc/FileSystem/Makefile.in	2010-03-28 17:14:55.000000000 +0200
+++ xbmc-28910.patch/xbmc/FileSystem/Makefile.in	2010-03-28 19:13:23.834753245 +0200
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 ifeq ($(findstring osx,$(ARCH)), osx)
 INCLUDES+=-I../../lib/libcurl-OSX/include
diff -Naur xbmc-28910/xbmc/Makefile.in xbmc-28910.patch/xbmc/Makefile.in
--- xbmc-28910/xbmc/Makefile.in	2010-03-28 17:13:35.000000000 +0200
+++ xbmc-28910.patch/xbmc/Makefile.in	2010-03-28 19:13:23.837752973 +0200
@@ -1,4 +1,4 @@
-INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer
+INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer
 
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-Icores/dvdplayer/Codecs/ffmpeg
