diff -Naur xbmc-29428/configure.in xbmc-29428.patch/configure.in
--- xbmc-29428/configure.in	2010-04-22 00:56:44.000000000 +0200
+++ xbmc-29428.patch/configure.in	2010-04-22 02:22:21.771774020 +0200
@@ -474,7 +474,7 @@
 # check for library basenames
 if test "$host_vendor" != "apple"; then
   # libcurl
-  LIBCURL_BASENAME=$(gcc $CFLAGS -print-file-name=libcurl.so | \
+  LIBCURL_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libcurl.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -483,7 +483,7 @@
   fi
 
   # libFLAC
-  LIBFLAC_BASENAME=$(gcc $CFLAGS -print-file-name=libFLAC.so | \
+  LIBFLAC_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libFLAC.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -492,7 +492,7 @@
   fi
 
   # libvorbisfile
-  LIBVORBISFILE_BASENAME=$(gcc $CFLAGS -print-file-name=libvorbisfile.so | \
+  LIBVORBISFILE_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libvorbisfile.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -501,7 +501,7 @@
   fi
 
   # libmodplug
-  LIBMODPLUG_BASENAME=$(gcc $CFLAGS -print-file-name=libmodplug.so | \
+  LIBMODPLUG_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libmodplug.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -510,7 +510,7 @@
   fi
 
   # libfaad
-  LIBFAAD_BASENAME=$(gcc $CFLAGS -print-file-name=libfaad.so | \
+  LIBFAAD_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libfaad.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -519,7 +519,7 @@
   fi
 
   # libmad
-  LIBMAD_BASENAME=$(gcc $CFLAGS -print-file-name=libmad.so | \
+  LIBMAD_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libmad.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -528,7 +528,7 @@
   fi
 
   # libogg
-  LIBOGG_BASENAME=$(gcc $CFLAGS -print-file-name=libogg.so | \
+  LIBOGG_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libogg.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -537,7 +537,7 @@
   fi
 
   # libvorbisenc
-  LIBVORBISENC_BASENAME=$(gcc $CFLAGS -print-file-name=libvorbisenc.so | \
+  LIBVORBISENC_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libvorbisenc.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
@@ -546,7 +546,7 @@
   fi
 
   # libvorbis
-  LIBVORBIS_BASENAME=$(gcc $CFLAGS -print-file-name=libvorbis.so | \
+  LIBVORBIS_BASENAME=$($ac_cv_prog_CC $CFLAGS -print-file-name=libvorbis.so | \
                      while read output; do objdump -p $output | \
                      grep SONAME | \
                      awk 'BEGIN { FS = " " } ; { print $2 }'; done)
