diff -Naur xbmc-29740/configure.in xbmc-29740.patch/configure.in
--- xbmc-29740/configure.in	2010-05-03 04:08:41.115131030 +0200
+++ xbmc-29740.patch/configure.in	2010-05-03 04:12:20.116129219 +0200
@@ -499,7 +499,7 @@
 AC_DEFUN([XB_FIND_SONAME],
 [
   AC_MSG_CHECKING([for lib$2 soname])
-  $1_SONAME=$( gcc -print-file-name=lib$2.so | \
+  $1_SONAME=$( $CC -print-file-name=lib$2.so | \
                 while read output; do objdump -p $output | \
                   grep "SONAME" | \
                   sed -e 's/ \+SONAME \+//'; done 2> /dev/null )
