diff -Naur xbmc-30122/configure.in xbmc-30122.patch/configure.in
--- xbmc-30122/configure.in	2010-05-13 13:16:42.000000000 +0200
+++ xbmc-30122.patch/configure.in	2010-05-14 13:12:09.725479259 +0200
@@ -1603,9 +1603,7 @@
       --with-pic
   else
     ./configure \
-      CPPFLAGS="$CPPFLAGS" \
-      CFLAGS="$CFLAGS -fPIC" \ 
-      CXXFLAGS="$CXXFLAGS -fPIC" \
+      CFLAGS="-O3" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
       --host=$host_alias \
       --build=$build_alias \
