diff -Naur xbmc-30973/xbmc/Application.cpp xbmc-30973.patch/xbmc/Application.cpp
--- xbmc-30973/xbmc/Application.cpp	2010-06-08 18:41:51.000000000 +0200
+++ xbmc-30973.patch/xbmc/Application.cpp	2010-06-09 02:43:00.326087347 +0200
@@ -2121,7 +2121,8 @@
               g_infoManager.GetFPS(), strCores.c_str(), profiling.c_str());
 #else
     double dCPU = m_resourceCounter.GetCPUUsage();
-    info.Format("FreeMem %"PRIu64"/%"PRIu64" Kb, FPS %2.1f, %s. CPU-XBMC %4.2f%%%s", stat.dwAvailPhys/1024, stat.dwTotalPhys/1024,
+//    info.Format("FreeMem %"PRIu64"/%"PRIu64" Kb, FPS %2.1f, %s. CPU-XBMC %4.2f%%%s", stat.dwAvailPhys/1024, stat.dwTotalPhys/1024,
+    info.Format("FreeMem %d/%d Kb, FPS %2.1f, %s. CPU-XBMC %4.2f%%%s", (int)(stat.dwAvailPhys/1024), (int)(stat.dwTotalPhys/1024),
               g_infoManager.GetFPS(), strCores.c_str(), dCPU, profiling.c_str());
 #endif
 
