diff -Naur xbmc-30122/configure.in xbmc-30122.patch/configure.in
--- xbmc-30122/configure.in	2010-05-13 13:58:23.276198942 +0200
+++ xbmc-30122.patch/configure.in	2010-05-13 13:58:49.323198531 +0200
@@ -489,9 +489,6 @@
   PKG_CHECK_MODULES([XMU],          [xmu],
                     [INCLUDES="$INCLUDES $XMU_CFLAGS"; LIBS="$LIBS $XMU_LIBS"],
                     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],     [xinerama],
-                    [INCLUDES="$INCLUDES $XINERAMA_CFLAGS"; LIBS="$LIBS $XINERAMA_LIBS"],
-                    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],         [libcurl],
                     [INCLUDES="$INCLUDES $CURL_CFLAGS"; LIBS="$LIBS $CURL_LIBS"],
                     AC_MSG_ERROR($missing_library))
