diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 19:12:51.211753728 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 19:24:46.921753713 +0200
@@ -463,9 +463,6 @@
   PKG_CHECK_MODULES([XMU],          [xmu],
                     [INCLUDES+=" $XMU_CFLAGS"; LIBS+=" $XMU_LIBS"],
                     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],     [xinerama],
-                    [INCLUDES+=" $XINERAMA_CFLAGS"; LIBS+=" $XINERAMA_LIBS"],
-                    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],         [libcurl],
                     [INCLUDES+=" $CURL_CFLAGS"; LIBS+=" $CURL_LIBS"],
                     AC_MSG_ERROR($missing_library))
