diff -Naur xbmc-30587/configure.in xbmc-30587.patch/configure.in
--- xbmc-30587/configure.in	2010-05-26 12:27:28.620730606 +0200
+++ xbmc-30587.patch/configure.in	2010-05-26 12:29:17.460731872 +0200
@@ -531,9 +531,6 @@
   PKG_CHECK_MODULES([XMU],   [xmu],
     [INCLUDES="$INCLUDES $XMU_CFLAGS"; LIBS="$LIBS $XMU_LIBS"],
     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],[xinerama],
-    [INCLUDES="$INCLUDES $XINERAMA_CFLAGS"; LIBS="$LIBS $XINERAMA_LIBS"],
-    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],  [libcurl],
     [INCLUDES="$INCLUDES $CURL_CFLAGS"; LIBS="$LIBS $CURL_LIBS"],
     AC_MSG_ERROR($missing_library))
