diff -Naur xbmc-30587/configure.in xbmc-30587.patch/configure.in
--- xbmc-30587/configure.in	2010-05-26 12:30:34.583732752 +0200
+++ xbmc-30587.patch/configure.in	2010-05-26 12:33:26.657731562 +0200
@@ -941,6 +941,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1569,7 +1570,8 @@
       --enable-protocol=http \
       --enable-pthreads \
       `case $host_cpu in powerpc*|arm*) echo --enable-pic ;; esac` \
-      --cc="$CC"
+      --cc="$CC" \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi`
   fi
 ], [$USE_EXTERNAL_FFMPEG])
 
