diff -Naur xbmc-28910/configure.in xbmc-28910.patch/configure.in
--- xbmc-28910/configure.in	2010-03-28 19:26:02.516628518 +0200
+++ xbmc-28910.patch/configure.in	2010-03-28 19:26:44.961754003 +0200
@@ -764,6 +764,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1318,6 +1319,7 @@
       --enable-protocol=http \
       --enable-pthreads \
       `if test "$host_cpu" = "powerpc" || test "$host_cpu" = "powerpc64"; then echo --enable-pic; fi` \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --cc="$CC"
   fi
 ], [$USE_EXTERNAL_FFMPEG])
