diff -Naur xbmc-28910/xbmc/utils/Splash.cpp xbmc-28910.patch/xbmc/utils/Splash.cpp
--- xbmc-28910/xbmc/utils/Splash.cpp	2010-03-28 17:14:58.000000000 +0200
+++ xbmc-28910.patch/xbmc/utils/Splash.cpp	2010-03-28 19:29:51.138661142 +0200
@@ -53,9 +53,9 @@
   g_graphicsContext.Clear();
 
   g_graphicsContext.SetCameraPosition(CPoint(0, 0));
-  float w = g_graphicsContext.GetWidth() * 0.5f;
-  float h = g_graphicsContext.GetHeight() * 0.5f;
-  CGUIImage* image = new CGUIImage(0, 0, w*0.5f, h*0.5f, w, h, m_ImageName);
+  float w = g_graphicsContext.GetWidth();
+  float h = g_graphicsContext.GetHeight();
+  CGUIImage* image = new CGUIImage(0, 0, 0, 0, w, h, m_ImageName);
   image->SetAspectRatio(CAspectRatio::AR_KEEP);
   image->AllocResources();
 
