diff -Naur xbmc-13.alpha-65ad3ae/configure.in xbmc-13.alpha-65ad3ae.patch/configure.in
--- xbmc-13.alpha-65ad3ae/configure.in	2014-02-03 16:36:26.000000000 +0100
+++ xbmc-13.alpha-65ad3ae.patch/configure.in	2014-02-04 20:15:33.927101424 +0100
@@ -872,7 +872,7 @@
     fi
   fi
 fi
-if test "$target_platform" = "target_linux"; then
+if test "$target_platform" = "target_linux" -o "$target_platform" = "target_raspberry_pi"; then
   USE_STATIC_FFMPEG=1
   use_static_ffmpeg=yes
   AC_DEFINE([USE_STATIC_FFMPEG], [1], [link ffmpeg statically])
