From 249cccc2da097917995571b123ac22e30ed3f686 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Thu, 7 Mar 2013 12:50:57 +0000
Subject: [PATCH] [rbp] Enable Vsync as a default

---
 xbmc/settings/Settings.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/settings/Settings.h b/xbmc/settings/Settings.h
index fd44a0f..5ba1d36 100644
--- a/xbmc/settings/Settings.h
+++ b/xbmc/settings/Settings.h
@@ -36,7 +36,7 @@
 #ifdef MID
 #define DEFAULT_VSYNC       VSYNC_DISABLED
 #else  // MID
-#if defined(TARGET_DARWIN) || defined(_WIN32)
+#if defined(TARGET_DARWIN) || defined(_WIN32) || defined(TARGET_RASPBERRY_PI)
 #define DEFAULT_VSYNC       VSYNC_ALWAYS
 #else
 #define DEFAULT_VSYNC       VSYNC_DRIVER
-- 
1.7.10

