diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 14:17:38.576806450 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 14:22:07.235423909 +0100
@@ -914,6 +914,10 @@
     AC_LANG_POP([C++])
   fi
   CPPFLAGS="$SAVE_CPPFLAGS"
+
+  # check for yasm
+  AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
+
 else
   AC_MSG_NOTICE($external_ffmpeg_disabled)
   USE_EXTERNAL_FFMPEG=0
@@ -1688,6 +1692,7 @@
       `if test "$use_arch" != "no"; then echo --arch=$use_arch; fi`\
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       --target-os=$(tolower $(uname -s)) \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --enable-muxer=adts \
