diff -Naur xbmc-1677554/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h xbmc-1677554.patch/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h
--- xbmc-1677554/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h	2012-10-27 20:43:47.000000000 +0200
+++ xbmc-1677554.patch/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h	2012-10-28 16:39:35.842914672 +0100
@@ -21,6 +21,7 @@
 #include <sys/socket.h>
 #include <sys/time.h>
 //#include <math.h>
+#include <unistd.h>
 #ifdef DEB_PACK
 #include <xbmc/xbmcclient.h>
 #else
