diff -Naur xbmc-9389dc8/xbmc/settings/GUISettings.cpp xbmc-9389dc8.patch/xbmc/settings/GUISettings.cpp
--- xbmc-9389dc8/xbmc/settings/GUISettings.cpp	2011-09-01 23:34:05.267139356 +0200
+++ xbmc-9389dc8.patch/xbmc/settings/GUISettings.cpp	2011-09-01 23:34:52.098765331 +0200
@@ -423,7 +423,8 @@
   AddString(vs, "videoscreen.testpattern",226,"", BUTTON_CONTROL_STANDARD);
 #endif
 #if defined(_LINUX) && !defined(__APPLE__)
-  AddBool(NULL, "videoscreen.haslcd", 4501, false);
+  AddSeparator(vs, "videoscreen.sep2");
+  AddBool(vs, "videoscreen.haslcd", 4501, false);
 #endif
 
   CSettingsCategory* ao = AddCategory(4, "audiooutput", 772);
