diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 13:30:17.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-03-04 19:51:28.278102248 +0100
@@ -1646,6 +1646,7 @@
       --disable-static \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
       `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
+      `if test -n "$host_alias"; then echo --cross-prefix="${host_alias}-"; fi`\
       `if test "$use_arch" != "no"; then echo --arch=$use_arch; fi`\
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       --target-os=$(tolower $(uname -s)) \
diff -Naur xbmc-Dharma-10.1-c8405db/Makefile.in xbmc-Dharma-10.1-c8405db.patch/Makefile.in
--- xbmc-Dharma-10.1-c8405db/Makefile.in	2011-02-15 13:30:17.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/Makefile.in	2011-03-04 19:52:52.500328225 +0100
@@ -467,8 +467,8 @@
 	xbmc/cores/DllLoader/exports/exports.a
 
 ifeq (arm, $(ARCH))
-# Force external python2.5 for now!
-LIBS +=-lpython2.5
+# Force external python2.6 for now!
+LIBS +=-lpython2.6
 endif
 
 xbmc.bin: $(OBJSXBMC) $(DYNOBJSXBMC)
