diff -Naur xbmc-10.1-Dharma/xbmc/FileSystem/File.cpp xbmc-10.1-Dharma.patch/xbmc/FileSystem/File.cpp
--- xbmc-10.1-Dharma/xbmc/FileSystem/File.cpp	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/FileSystem/File.cpp	2011-05-04 17:59:12.093724017 +0200
@@ -217,7 +217,7 @@
         return false;
     }
 
-    if ( (flags & READ_NO_CACHE) == 0 && CUtil::IsInternetStream(strFileName) && !CUtil::IsPicture(strFileName) )
+    if ( (flags & READ_NO_CACHE) == 0 && ( CUtil::IsInternetStream(strFileName) || CUtil::IsOnLAN(strFileName) ) && !CUtil::IsPicture(strFileName) )
       m_flags |= READ_CACHED;
 
     CURL url(strFileName);
