diff --git a/xbmc/interfaces/legacy/ModuleXbmc.cpp b/xbmc/interfaces/legacy/ModuleXbmc.cpp
index 38564212d7d1ab7f15e486674e567119fcec8c8f..f9b02175db585e56a95895621536d504ebeb268a 100644
--- a/xbmc/interfaces/legacy/ModuleXbmc.cpp
+++ b/xbmc/interfaces/legacy/ModuleXbmc.cpp
@@ -337,6 +337,13 @@ namespace XBMCAddon
       }
     }
 
+    void stopSFX()
+    {
+      XBMC_TRACE;
+      g_audioManager.Stop();
+      g_audioManager.UnLoad();
+    }
+    
     void enableNavSounds(bool yesNo)
     {
       XBMC_TRACE;
diff --git a/xbmc/interfaces/legacy/ModuleXbmc.h b/xbmc/interfaces/legacy/ModuleXbmc.h
index 9e04cfa114dfa906d9fafb2d95aa324dd6e6ca22..e6e4b3cebb4971c4201c31216fb5343f2305c925 100644
--- a/xbmc/interfaces/legacy/ModuleXbmc.h
+++ b/xbmc/interfaces/legacy/ModuleXbmc.h
@@ -230,6 +230,14 @@ namespace XBMCAddon
     void playSFX(const char* filename);
 
     /**
+     * stopSFX() -- Stops wav file
+     *
+     * example:
+     *   - xbmc.stopSFX()
+     */
+    void stopSFX();
+    
+    /**
      * enableNavSounds(yesNo) -- Enables/Disables nav sounds
      *
      * yesNo          : integer - enable (True) or disable (False) nav sounds

