From cc39b66b38657787e99bf6369a77c993cd601c23 Mon Sep 17 00:00:00 2001
From: ulion <ulion2002@gmail.com>
Date: Sun, 24 Mar 2013 08:05:04 +0800
Subject: [PATCH] Only add to preDeps when it's not in there.

---
 xbmc/addons/AddonInstaller.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index 276467a..631a01c 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -343,13 +343,11 @@ bool CAddonInstaller::CheckDependencies(const AddonPtr &addon,
     }
     // at this point we have our dep, or the dep is optional (and we don't have it) so check that it's OK as well
     // TODO: should we assume that installed deps are OK?
-    if (dep)
+    if (dep && std::find(preDeps.begin(), preDeps.end(), dep->ID()) == preDeps.end())
     {
-      if (std::find(preDeps.begin(), preDeps.end(), dep->ID()) == preDeps.end() &&
-       !CheckDependencies(dep, preDeps))
+      if (!CheckDependencies(dep, preDeps))
         return false;
-      else
-        preDeps.push_back(dep->ID());
+      preDeps.push_back(dep->ID());
     }
   }
   return true;
-- 
1.8.1.5

