diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 14:23:09.427177685 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 14:23:28.731790834 +0100
@@ -562,8 +562,6 @@
 AC_CHECK_LIB([pthread],     [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([lzo2],        [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([z],           [main],, AC_MSG_ERROR($missing_library))
-AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
-AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([wavpack],     [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([mpeg2],       [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([mpeg2convert],[main],, AC_MSG_ERROR($missing_library))
@@ -975,6 +973,8 @@
   USE_EXTERNAL_PYTHON=1
   AC_DEFINE([USE_EXTERNAL_PYTHON], [1], [Whether to use external python library.])
 else
+  AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
+  AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
   AC_MSG_NOTICE($external_python_disabled)
   USE_EXTERNAL_PYTHON=0
 fi
