diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdcss/src/libdvdcss.pc.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdcss/src/libdvdcss.pc.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdcss/src/libdvdcss.pc.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdcss/src/libdvdcss.pc.in	2011-02-15 14:07:34.152020190 +0100
@@ -6,5 +6,5 @@
 Name: libdvdcss
 Description: DVD access and decryption library.
 Version: @VERSION@
-Libs: -L${libdir} -ldvdcss
-Cflags: -I{includedir} -I${includedir}/@PACKAGE@
+Libs: -ldvdcss
+Cflags: -I.
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config2.sh xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config2.sh
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config2.sh	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config2.sh	2011-02-15 14:07:34.153020170 +0100
@@ -56,17 +56,17 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $dvdread_cflags $extracflags $threadcflags
+      echo $dvdread_cflags $extracflags $threadcflags
 fi
 
 if test "$echo_minicflags" = "yes"; then
-      echo -I$prefix/include -I$prefix/include/dvdnav $extracflags $threadcflags
+      echo $extracflags $threadcflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir -ldvdnav $dvdread_libs $threadlib
+      echo -ldvdnav $dvdread_libs $threadlib
 fi      
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L$libdir -ldvdnavmini $threadlib
+      echo -ldvdnavmini $threadlib
 fi
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav-config.in	2011-02-15 14:07:34.154020150 +0100
@@ -76,13 +76,13 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@ -I@includedir@/libdvdread @THREAD_CFLAGS@
+      echo @THREAD_CFLAGS@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdnav -ldvdread @THREAD_LIBS@
+      echo -ldvdnav -ldvdread @THREAD_LIBS@
 fi
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L@libdir@ -ldvdnavmini @THREAD_LIBS@
+      echo -ldvdnavmini @THREAD_LIBS@
 fi      
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnavmini.pc.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnavmini.pc.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnavmini.pc.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnavmini.pc.in	2011-02-15 14:07:34.156020109 +0100
@@ -7,5 +7,5 @@
 Description: DVD Navigation mini library
 Version: @VERSION@
 
-Cflags: -I${includedir} @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav.pc.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav.pc.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav.pc.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdnav/misc/dvdnav.pc.in	2011-02-15 14:07:34.157020088 +0100
@@ -8,5 +8,5 @@
 Version: @VERSION@
 
 Requires.private: dvdread >= 4.1.2
-Cflags: -I${includedir} @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.in	2011-02-15 14:07:34.158020067 +0100
@@ -72,9 +72,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo ""
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -ldvdread
 fi
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.sh xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.sh
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.sh	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread-config.sh	2011-02-15 14:07:34.158020067 +0100
@@ -48,9 +48,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $extracflags
+      echo $extracflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir $dvdreadlib
+      echo $dvdreadlib
 fi      
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread.pc.in xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread.pc.in
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread.pc.in	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/Codecs/libdvd/libdvdread/misc/dvdread.pc.in	2011-02-15 14:07:34.159020047 +0100
@@ -7,5 +7,5 @@
 Description: Low level DVD access library
 Version: @VERSION@
 
-Cflags: -I${includedir}
-Libs: -L${libdir} -ldvdread
+Cflags: -I.
+Libs: -ldvdread
