diff -Naur xbmc-10.1-Dharma/xbmc/lib/libPython/xbmcmodule/pyutil.cpp xbmc-10.1-Dharma.patch/xbmc/lib/libPython/xbmcmodule/pyutil.cpp
--- xbmc-10.1-Dharma/xbmc/lib/libPython/xbmcmodule/pyutil.cpp	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/lib/libPython/xbmcmodule/pyutil.cpp	2011-06-10 21:14:07.881047162 +0200
@@ -23,6 +23,7 @@
 #include <wchar.h>
 #include <vector>
 #include "addons/Skin.h"
+#include "utils/log.h"
 #include "tinyXML/tinyxml.h"
 #include "utils/CharsetConverter.h"
 #include "CriticalSection.h"
@@ -202,7 +203,22 @@
     g_callQueue.erase(iter);
     lock.Leave();
     if (p.func)
+    {
       p.func(p.args);
+
+      // Since the callback is likely to make it into python, and since
+      // not all of the callback functions handle errors, the error state
+      // may remain set from the previous call. As a result subsequent calls
+      // to callback functions exhibit odd behavior difficult to debug.
+      if (PyErr_Occurred())
+      {
+        CLog::Log(LOGERROR,"Exception in python script callback execution");
+
+        // This clears the python error state and prints it to the log
+        PyErr_Print();
+      }
+
+    }
     //(*((*iter).first))((*iter).second);
     lock.Enter();
     iter = g_callQueue.begin();
