diff -Naur xbmc-1d9a8e0/configure.in xbmc-1d9a8e0.patch/configure.in
--- xbmc-1d9a8e0/configure.in	2011-08-09 06:37:07.000000000 +0200
+++ xbmc-1d9a8e0.patch/configure.in	2011-08-09 06:58:57.094093611 +0200
@@ -66,6 +66,8 @@
 xrandr_disabled="== XRandR support disabled. SDL will be used for resolution support. =="
 goom_enabled="== GOOM enabled. =="
 goom_disabled="== GOOM disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 dvdcss_enabled="== DVDCSS support enabled. =="
@@ -208,6 +210,12 @@
   [use_goom=$enableval],
   [use_goom=no])
 
+AC_ARG_ENABLE([xbmcprojectm],
+  [AS_HELP_STRING([--enable-xbmcprojectm],
+  [enable XBMCProjectM visualisation (default is yes)])],
+  [use_xbmcprojectm=$enableval],
+  [use_xbmcprojectm=yes])
+
 AC_ARG_ENABLE([ccache],
   [AS_HELP_STRING([--enable-ccache],
   [enable building with ccache feature (default is auto)])],
@@ -803,6 +811,15 @@
   fi
 fi
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "yes" && test "use_gl" == "yes"; then
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  DISABLE_XBMCPROJECTM=0
+else
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  DISABLE_XBMCPROJECTM=1
+fi
+
 # libRTMP
 if test "$use_librtmp" != "no"; then
   AC_CHECK_HEADERS([librtmp/log.h librtmp/amf.h librtmp/rtmp.h],,
@@ -1266,6 +1283,12 @@
   final_message="$final_message\n  GOOM:\t\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_libbluray" = "yes"; then
   final_message="$final_message\n  Bluray:\tYes"
 else
@@ -1495,6 +1518,7 @@
 AC_SUBST(SDL_DEFINES)
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(DISABLE_GOOM)
+AC_SUBST(DISABLE_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(PYTHON_VERSION)
 AC_SUBST(OUTPUT_FILES)
@@ -1739,7 +1763,7 @@
     fi
 set +x
 fi
-], [0])
+], [$DISABLE_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   ./configure  \
diff -Naur xbmc-1d9a8e0/Makefile.in xbmc-1d9a8e0.patch/Makefile.in
--- xbmc-1d9a8e0/Makefile.in	2011-08-09 06:37:11.000000000 +0200
+++ xbmc-1d9a8e0.patch/Makefile.in	2011-08-09 07:01:36.862090884 +0200
@@ -144,8 +144,10 @@
 ifeq (@USE_OPENGL@,1)
 VIS_DIRS+=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@DISABLE_XBMCPROJECTM@,1)
+VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 endif
 
 ifeq (($(findstring osx,@ARCH@), osx),(@USE_OPENGL@,1))
