diff -Naur xbmc-pvr-fc63e66/language/English/strings.xml xbmc-pvr-fc63e66.patch/language/English/strings.xml
--- xbmc-pvr-fc63e66/language/English/strings.xml	2012-03-19 21:20:31.531809910 +0100
+++ xbmc-pvr-fc63e66.patch/language/English/strings.xml	2012-03-19 21:21:56.576430020 +0100
@@ -1360,6 +1360,7 @@
   <string id="14096">Rip</string>
   <string id="14097">Audio CD Insert Action</string>
   <string id="14098">Play</string>
+  <string id="14099">Eject disc when Audio CD ripping is complete</string>
 
   <string id="15015">Remove</string>
   <string id="15016">Games</string>
diff -Naur xbmc-pvr-fc63e66/xbmc/cdrip/CDDARipper.cpp xbmc-pvr-fc63e66.patch/xbmc/cdrip/CDDARipper.cpp
--- xbmc-pvr-fc63e66/xbmc/cdrip/CDDARipper.cpp	2012-03-16 22:27:56.000000000 +0100
+++ xbmc-pvr-fc63e66.patch/xbmc/cdrip/CDDARipper.cpp	2012-03-19 21:21:07.725417655 +0100
@@ -332,6 +332,11 @@
   }
 
   CLog::Log(LOGINFO, "Ripped CD succesfull");
+  if (g_guiSettings.GetBool("audiocds.ejectonrip"))
+  {
+    CLog::Log(LOGINFO, "Ejecting CD");
+    CIoSupport::EjectTray();
+  }
   return true;
 }
 
diff -Naur xbmc-pvr-fc63e66/xbmc/settings/GUISettings.cpp xbmc-pvr-fc63e66.patch/xbmc/settings/GUISettings.cpp
--- xbmc-pvr-fc63e66/xbmc/settings/GUISettings.cpp	2012-03-19 21:20:31.533809950 +0100
+++ xbmc-pvr-fc63e66.patch/xbmc/settings/GUISettings.cpp	2012-03-19 21:21:07.734417841 +0100
@@ -349,6 +349,7 @@
   AddInt(acd, "audiocds.quality", 622, CDDARIP_QUALITY_CBR, qualities, SPIN_CONTROL_TEXT);
   AddInt(acd, "audiocds.bitrate", 623, 192, 128, 32, 320, SPIN_CONTROL_INT_PLUS, MASK_KBPS);
   AddInt(acd, "audiocds.compressionlevel", 665, 5, 0, 1, 8, SPIN_CONTROL_INT_PLUS);
+  AddBool(acd, "audiocds.ejectonrip", 14099, false);
 
 #ifdef HAS_KARAOKE
   CSettingsCategory* kar = AddCategory(3, "karaoke", 13327);
