diff --git a/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h b/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h
index dbdbb94..6f68715 100644
--- a/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h
+++ b/tools/EventClients/Clients/WiiRemote/WiiUse_WiiRemote.h
@@ -27,7 +27,7 @@
 #include "../../lib/c++/xbmcclient.h"
 #endif
 //#ifndef WIN32
-//	#include <unistd.h>
+	#include <unistd.h>
 //#endif
 #include "wiiuse.h"
 //#define ICON_PATH "../../"
