diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 13:30:17.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 13:57:02.105903186 +0100
@@ -65,6 +65,8 @@
 xrandr_disabled="== XRandR support disabled. SDL will be used for resolution support. =="
 goom_enabled="== GOOM enabled. =="
 goom_disabled="== GOOM disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -204,6 +206,12 @@
   [use_goom=$enableval],
   [use_goom=no])
 
+AC_ARG_ENABLE([xbmcprojectm],
+            [AS_HELP_STRING([--enable-xbmcprojectm],
+               [enable XBMCProjectM visualisation (default is yes)])],
+            [use_xbmcprojectm=$enableval],
+            [use_xbmcprojectm=yes])
+
 AC_ARG_ENABLE([ccache],
   [AS_HELP_STRING([--enable-ccache],
   [enable building with ccache feature (default is auto)])],
@@ -832,6 +840,15 @@
   AC_MSG_NOTICE($librtmp_disabled)
 fi
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "yes"; then
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  BUILD_XBMCPROJECTM=0
+else
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  BUILD_XBMCPROJECTM=1
+fi
+
 ### External libraries checks
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
@@ -1210,6 +1227,12 @@
   final_message="$final_message\n  Bluray:\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_mid" = "yes"; then
   final_message="$final_message\n  MID Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DMID"
@@ -1474,6 +1497,7 @@
 AC_SUBST(SDL_DEFINES)
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
+AC_SUBST(BUILD_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
 AC_SUBST(USE_INTERNAL_LIBA52)
@@ -1870,7 +1894,7 @@
       -DSYSCONF_INSTALL_DIR:PATH="${sysconfdir}" -DSHARE_INSTALL_PREFIX:PATH="${datadir}" .
 set +x
   fi
-], [0])
+], [$BUILD_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
diff -Naur xbmc-Dharma-10.1-c8405db/Makefile.in xbmc-Dharma-10.1-c8405db.patch/Makefile.in
--- xbmc-Dharma-10.1-c8405db/Makefile.in	2011-02-15 13:56:41.203329597 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/Makefile.in	2011-02-15 13:57:02.107903144 +0100
@@ -109,8 +109,10 @@
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
+VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 VIS_DIRS+=xbmc/visualizations/iTunes
 endif
@@ -308,7 +310,9 @@
 ifneq (arm, $(ARCH))
 	$(MAKE) -C xbmc/visualizations/OpenGLSpectrum
 	$(MAKE) -C xbmc/visualizations/WaveForm
+ifneq (@BUILD_XBMCPROJECTM@,1)
 	$(MAKE) -C xbmc/visualizations/XBMCProjectM
+endif
 ifeq ($(findstring osx,$(ARCH)), osx)
 	$(MAKE) -C xbmc/visualizations/iTunes
 endif
