From 4b7c1455582ccb6f8979681b4baed93d2852140c Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 11 Jul 2014 19:37:12 +0300
Subject: [PATCH] bye bye libflac

---
 configure.in | 3 ---
 1 file changed, 3 deletions(-)

diff -Naur xbmc-14-67f025d/configure.in xbmc-14-67f025d.patch/configure.in
--- xbmc-14-67f025d/configure.in	2014-07-25 06:45:49.000000000 +0200
+++ xbmc-14-67f025d.patch/configure.in	2014-07-25 06:54:20.396754173 +0200
@@ -1056,7 +1056,6 @@
 AC_CHECK_HEADER([ogg/ogg.h],,        AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([vorbis/vorbisfile.h],, AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([libmodplug/modplug.h],, AC_MSG_ERROR($missing_library))
-AC_CHECK_HEADER([FLAC/stream_decoder.h],, AC_MSG_ERROR($missing_library))
 
 AC_CHECK_HEADER([curl/curl.h],, AC_MSG_ERROR($missing_library))
 XB_FIND_SONAME([CURL], [curl])
@@ -1219,7 +1218,6 @@
 fi
 
 XB_FIND_SONAME([OGG],         [ogg])
-XB_FIND_SONAME([FLAC],        [FLAC])
 XB_FIND_SONAME([VORBIS],      [vorbis])
 XB_FIND_SONAME([VORBISFILE],  [vorbisfile])
 XB_FIND_SONAME([MODPLUG],     [modplug])
@@ -2596,7 +2594,6 @@
 AC_SUBST(HAVE_XBMC_NONFREE)
 AC_SUBST(USE_ASAP_CODEC)
 AC_SUBST(LIBCURL_BASENAME)
-AC_SUBST(LIBFLAC_BASENAME)
 AC_SUBST(LIBVORBISFILE_BASENAME)
 AC_SUBST(LIBMODPLUG_BASENAME)
 AC_SUBST(LIBOGG_BASENAME)
