From e441ca60e52ce8bdeb50538ef6b048ee196935e2 Mon Sep 17 00:00:00 2001
From: Voyager-xbmc <patrick.middag@telenet.be>
Date: Thu, 29 Dec 2011 08:34:08 +0100
Subject: [PATCH] Fix remaining dvd resume issue and soft eject problem

---
 xbmc/Autorun.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xbmc/Autorun.cpp b/xbmc/Autorun.cpp
index 4f49ff0..4bd823f 100644
--- a/xbmc/Autorun.cpp
+++ b/xbmc/Autorun.cpp
@@ -97,7 +97,7 @@ bool CAutorun::PlayDisc(const CStdString& path, bool bypassSettings, bool startF
     mediaPath = path;
 
 #ifdef _WIN32
-  if (mediaPath.IsEmpty())
+  if (mediaPath.IsEmpty() || mediaPath.CompareNoCase("iso9660://") == 0)
     mediaPath = g_mediaManager.TranslateDevicePath("");
 #endif
 
-- 
1.7.5.4

