diff -Naur xbmc-28622/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamMMS.cpp xbmc-28622.patch/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamMMS.cpp
--- xbmc-28622/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamMMS.cpp	2010-03-17 20:11:42.000000000 +0100
+++ xbmc-28622.patch/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamMMS.cpp	2010-03-19 04:04:41.074751644 +0100
@@ -35,6 +35,7 @@
 // some distros have already patched this but upstream @ https://launchpad.net/libmms
 // does still has the "this" usage.
 #define this instance
+#undef byte
 #include <libmms/mmsio.h> // FIXME: remove this header once the ubuntu headers is fixed (variable named this)
 #include <libmms/mms.h>
 #include <libmms/mmsh.h>
