diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 13:58:10.630505451 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 13:59:48.190515901 +0100
@@ -67,6 +67,8 @@
 goom_disabled="== GOOM disabled. =="
 xbmcprojectm_enabled="== XBMCProjectM enabled. =="
 xbmxprojectm_disabled="== XBMCProjectM disabled. =="
+rsxs_enabled="== RSXS enabled. =="
+rsxs_disabled="== RSXS disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -212,6 +214,12 @@
             [use_xbmcprojectm=$enableval],
             [use_xbmcprojectm=yes])
 
+AC_ARG_ENABLE([rsxs],
+            [AS_HELP_STRING([--enable-rsxs],
+               [enable RSXS Screensaver (default is yes)])],
+            [use_rsxs=$enableval],
+            [use_rsxs=yes])
+
 AC_ARG_ENABLE([ccache],
   [AS_HELP_STRING([--enable-ccache],
   [enable building with ccache feature (default is auto)])],
@@ -849,6 +857,15 @@
   BUILD_XBMCPROJECTM=1
 fi
 
+# RSXS
+if test "$use_rsxs" = "yes"; then
+  AC_MSG_NOTICE($rsxs_enabled)
+  BUILD_RSXS=0
+else
+  AC_MSG_NOTICE($rsxs_disabled)
+  BUILD_RSXS=1
+fi
+
 ### External libraries checks
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
@@ -1233,6 +1250,12 @@
   final_message="$final_message\n  XBMCProjectM:\tNo"
 fi
 
+if test "$use_rsxs" = "yes"; then
+  final_message="$final_message\n  RSXS:\t\tYes"
+else
+  final_message="$final_message\n  RSXS:\t\tNo"
+fi
+
 if test "$use_mid" = "yes"; then
   final_message="$final_message\n  MID Support:\tYes"
   SDL_DEFINES="$SDL_DEFINES -DMID"
@@ -1498,6 +1521,7 @@
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
 AC_SUBST(BUILD_XBMCPROJECTM)
+AC_SUBST(BUILD_RSXS)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
 AC_SUBST(USE_INTERNAL_LIBA52)
@@ -1952,7 +1976,7 @@
       --disable-skyrocket
   fi
 fi
-], [0])
+], [$BUILD_RSXS])
 
 XB_CONFIG_MODULE([xbmc/lib/libapetag], [
   if test "$host_vendor" = "apple" ; then
diff -Naur xbmc-Dharma-10.1-c8405db/Makefile.in xbmc-Dharma-10.1-c8405db.patch/Makefile.in
--- xbmc-Dharma-10.1-c8405db/Makefile.in	2011-02-15 13:58:10.631505430 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/Makefile.in	2011-02-15 13:59:48.191515880 +0100
@@ -104,8 +104,10 @@
 LIB_DIRS+=lib/libmicrohttpd
 endif
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
+ifneq (@BUILD_RSXS@,1)
+SS_DIRS=+xbmc/screensavers/rsxs-0.9/xbmc
+endif
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
@@ -324,8 +326,10 @@
 endif
 screensavers: exports
 ifneq (arm, $(ARCH))
+ifneq (@BUILD_RSXS@,1)
 	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
 endif
+endif
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
