diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp
--- xbmc-Dharma-10.1-c8405db/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2011-02-15 13:30:12.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2011-02-15 14:08:59.000292217 +0100
@@ -773,7 +773,7 @@
           if (!pBuffer)
           {
             // No free pre-allocated buffers so make one
-#ifdef _WIN32
+#if 1
             // force Windows to use YV12 until DX renderer gets NV12 or YUY2 capability.
             pBuffer = new CPictureBuffer(DVDVideoPicture::FMT_YUV420P, m_width, m_height);
 #else
