diff --git a/xbmc/storage/linux/UDisksProvider.cpp b/xbmc/storage/linux/UDisksProvider.cpp
index 6afc1a8..c940e69 100644
--- a/xbmc/storage/linux/UDisksProvider.cpp
+++ b/xbmc/storage/linux/UDisksProvider.cpp
@@ -356,7 +356,7 @@ void CUDisksProvider::DeviceChanged(const char *object, IStorageEventsCallback *
   {
     bool mounted = device->m_isMounted;
     device->Update();
-    if (g_advancedSettings.m_handleMounting)
+    if (g_advancedSettings.m_handleMounting && device->m_isOptical)
       device->Mount();
     if (!mounted && device->m_isMounted && callback)
       callback->OnStorageAdded(device->m_Label, device->m_MountPath);
