diff -Naur xbmc-04e78f4/configure.in xbmc-04e78f4.patch/configure.in
--- xbmc-04e78f4/configure.in	2011-04-26 19:38:51.000000000 +0200
+++ xbmc-04e78f4.patch/configure.in	2011-04-26 19:58:01.520765916 +0200
@@ -1344,7 +1344,6 @@
 
 OUTPUT_FILES="Makefile \
     Makefile.include \
-    addons/skin.confluence/media/Makefile \
     xbmc/Makefile \
     xbmc/cdrip/Makefile \
     xbmc/cores/Makefile \
diff -Naur xbmc-04e78f4/Makefile.in xbmc-04e78f4.patch/Makefile.in
--- xbmc-04e78f4/Makefile.in	2011-04-26 19:38:51.000000000 +0200
+++ xbmc-04e78f4.patch/Makefile.in	2011-04-26 19:58:43.768901469 +0200
@@ -144,13 +144,8 @@
 
 endif
 
-CONFLUENCE_MEDIA=addons/skin.confluence/media
-
-SKIN_DIRS=\
-	$(CONFLUENCE_MEDIA)
-
 DIRS= $(BIN_DIRS) $(EC_DIRS) $(XBMCTEX_DIRS) $(DVDPCODECS_DIRS) $(PAPCODECS_DIRS) \
-	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS) $(SKIN_DIRS)
+	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS)
 
 LIBS=@LIBS@
 CFLAGS=@CFLAGS@
@@ -187,12 +182,7 @@
 	@./bootstrap
 
 # skin textures
-ifneq ($(findstring arm,@ARCH@), arm)
-skins: tools/TexturePacker/TexturePacker force
-	 $(MAKE) -C $(CONFLUENCE_MEDIA)
-else
 skins:
-endif
 
 
 xbmc/guilib/guilib.a: force
