diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 14:06:18.343564347 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 14:16:56.717644949 +0100
@@ -685,9 +685,6 @@
   PKG_CHECK_MODULES([XMU],   [xmu],
     [INCLUDES="$INCLUDES $XMU_CFLAGS"; LIBS="$LIBS $XMU_LIBS"],
     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],[xinerama],
-    [INCLUDES="$INCLUDES $XINERAMA_CFLAGS"; LIBS="$LIBS $XINERAMA_LIBS"],
-    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],  [libcurl],
     [INCLUDES="$INCLUDES $CURL_CFLAGS"; LIBS="$LIBS $CURL_LIBS"],
     AC_MSG_ERROR($missing_library))
