diff -Naur xbmc-pvr-11.0.1/language/Dutch/strings.xml xbmc-pvr-11.0.1.patch/language/Dutch/strings.xml
--- xbmc-pvr-11.0.1/language/Dutch/strings.xml	2012-03-27 17:55:51.000000000 +0200
+++ xbmc-pvr-11.0.1.patch/language/Dutch/strings.xml	2012-05-14 14:13:28.656860932 +0200
@@ -1542,7 +1542,6 @@
   <string id="16322">Spline36</string>
   <string id="16323">Spline36 geoptimaliseerd</string>
   <string id="16324">Software menging</string>
-  <string id="16325">Auto - ION geoptimaliseerd</string>
 
   <string id="16400">Kwaliteitsverbetering video</string>
 
diff -Naur xbmc-pvr-11.0.1/language/English/strings.xml xbmc-pvr-11.0.1.patch/language/English/strings.xml
--- xbmc-pvr-11.0.1/language/English/strings.xml	2012-03-27 17:55:51.000000000 +0200
+++ xbmc-pvr-11.0.1.patch/language/English/strings.xml	2012-05-14 14:13:28.661861033 +0200
@@ -1539,9 +1539,7 @@
   <string id="16321">DXVA Best</string>
   <string id="16322">Spline36</string>
   <string id="16323">Spline36 optimized</string>
-
   <string id="16324">Software Blend</string>
-  <string id="16325">Auto - ION Optimized</string>
   
   <string id="16400">Post-processing</string>
 
diff -Naur xbmc-pvr-11.0.1/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp xbmc-pvr-11.0.1.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp
--- xbmc-pvr-11.0.1/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp	2012-05-14 14:11:52.791931577 +0200
+++ xbmc-pvr-11.0.1.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp	2012-05-14 14:13:28.663861073 +0200
@@ -531,8 +531,7 @@
 bool CVDPAU::Supports(EINTERLACEMETHOD method)
 {
   if(method == VS_INTERLACEMETHOD_VDPAU_BOB
-  || method == VS_INTERLACEMETHOD_AUTO
-  || method == VS_INTERLACEMETHOD_AUTO_ION)
+  || method == VS_INTERLACEMETHOD_AUTO)
     return true;
 
   for(SInterlaceMapping* p = g_interlace_mapping; p->method != VS_INTERLACEMETHOD_NONE; p++)
@@ -658,21 +657,8 @@
   }
   else
   {
-    if (method == VS_INTERLACEMETHOD_AUTO_ION)
-    {
-      if (vid_height <= 576)
-      {
-        VdpBool enabled[]={1,1,0};
-        vdp_st = vdp_video_mixer_set_feature_enables(videoMixer, ARSIZE(feature), feature, enabled);
-      }
-      else if (vid_height > 576)
-      {
-        VdpBool enabled[]={1,0,0};
-        vdp_st = vdp_video_mixer_set_feature_enables(videoMixer, ARSIZE(feature), feature, enabled);
-      }
-    }
-    else if (method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL
-         ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF)
+    if (method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL
+    ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF)
     {
       VdpBool enabled[]={1,0,0};
       vdp_st = vdp_video_mixer_set_feature_enables(videoMixer, ARSIZE(feature), feature, enabled);
@@ -1399,16 +1385,14 @@
     if (mode == VS_DEINTERLACEMODE_FORCE
     || (mode == VS_DEINTERLACEMODE_AUTO && m_DVDVideoPics.front().iFlags & DVP_FLAG_INTERLACED))
     {
-      if((method == VS_INTERLACEMETHOD_AUTO_ION
-      ||  method == VS_INTERLACEMETHOD_VDPAU_BOB
+      if((method == VS_INTERLACEMETHOD_VDPAU_BOB
       ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL
       ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF
       ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL
       ||  method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL_HALF
       ||  method == VS_INTERLACEMETHOD_VDPAU_INVERSE_TELECINE ))
       {
-        if((method == VS_INTERLACEMETHOD_AUTO_ION && vid_height > 576)
-        || method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF
+        if(method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_HALF
         || method == VS_INTERLACEMETHOD_VDPAU_TEMPORAL_SPATIAL_HALF
         || avctx->skip_frame == AVDISCARD_NONREF)
           m_mixerstep = 0;
diff -Naur xbmc-pvr-11.0.1/xbmc/video/dialogs/GUIDialogVideoSettings.cpp xbmc-pvr-11.0.1.patch/xbmc/video/dialogs/GUIDialogVideoSettings.cpp
--- xbmc-pvr-11.0.1/xbmc/video/dialogs/GUIDialogVideoSettings.cpp	2012-03-27 17:55:34.000000000 +0200
+++ xbmc-pvr-11.0.1.patch/xbmc/video/dialogs/GUIDialogVideoSettings.cpp	2012-05-14 14:13:28.664861093 +0200
@@ -113,7 +113,6 @@
     entries.push_back(make_pair(VS_INTERLACEMETHOD_VDPAU_INVERSE_TELECINE     , 16314));
     entries.push_back(make_pair(VS_INTERLACEMETHOD_DXVA_BOB                   , 16320));
     entries.push_back(make_pair(VS_INTERLACEMETHOD_DXVA_BEST                  , 16321));
-    entries.push_back(make_pair(VS_INTERLACEMETHOD_AUTO_ION                   , 16325));
 
     /* remove unsupported methods */
     for(vector<pair<int, int> >::iterator it = entries.begin(); it != entries.end();)
