From 5385b26621a8d966f187f7b63dfaa8f336beb9a5 Mon Sep 17 00:00:00 2001
From: Lars Op den Kamp <lars@opdenkamp.eu>
Date: Tue, 12 Mar 2013 10:20:27 +0100
Subject: [PATCH] [cec] extra guards around m_queryThread

---
 xbmc/peripherals/devices/PeripheralCecAdapter.cpp |   13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 4cdd3c4..6509c5d 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -99,8 +99,8 @@ class DllLibCEC : public DllDynamic, DllLibCECInterface
     m_bStop = true;
   }
 
-  SAFE_DELETE(m_queryThread);
   StopThread(true);
+  delete m_queryThread;
 
   if (m_dll && m_cecAdapter)
   {
@@ -389,7 +389,7 @@ void CPeripheralCecAdapter::Process(void)
       Sleep(5);
   }
 
-  SAFE_DELETE(m_queryThread);
+  m_queryThread->StopThread(true);
 
   bool bSendStandbyCommands(false);
   {
@@ -1105,9 +1105,12 @@ void CPeripheralCecAdapter::OnSettingChanged(const CStdString &strChangedSetting
   }
   else if (IsRunning())
   {
-    CLog::Log(LOGDEBUG, "%s - sending the updated configuration to libCEC", __FUNCTION__);
-    SetConfigurationFromSettings();
-    m_queryThread->UpdateConfiguration(&m_configuration);
+    if (m_queryThread->IsRunning())
+    {
+      CLog::Log(LOGDEBUG, "%s - sending the updated configuration to libCEC", __FUNCTION__);
+      SetConfigurationFromSettings();
+      m_queryThread->UpdateConfiguration(&m_configuration);
+    }
   }
   else
   {
-- 
1.7.10

