From e616214879880d4d7fd52d58f5f75660ab8c2930 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 16 Mar 2014 19:52:27 +0200
Subject: [PATCH] assume we are runing in a WM

---
 xbmc/windowing/X11/WinSystemX11.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xbmc/windowing/X11/WinSystemX11.cpp b/xbmc/windowing/X11/WinSystemX11.cpp
index c95f4ec..d1cc658 100644
--- a/xbmc/windowing/X11/WinSystemX11.cpp
+++ b/xbmc/windowing/X11/WinSystemX11.cpp
@@ -875,7 +875,7 @@ bool CWinSystemX11::SetWindow(int width, int height, bool fullscreen, const std:
     vi = glXChooseVisual(m_dpy, m_nScreen, att);
     cmap = XCreateColormap(m_dpy, RootWindow(m_dpy, vi->screen), vi->visual, AllocNone);
 
-    bool hasWM = HasWindowManager();
+    bool hasWM = true;
 
     int def_vis = (vi->visual == DefaultVisual(m_dpy, vi->screen));
     swa.override_redirect = hasWM ? False : True;
-- 
1.7.2.5

