diff -Naur xbmc-10.1-Dharma/configure.in xbmc-10.1-Dharma.patch/configure.in
--- xbmc-10.1-Dharma/configure.in	2011-09-14 15:17:03.499574627 +0200
+++ xbmc-10.1-Dharma.patch/configure.in	2011-09-14 15:20:45.627527971 +0200
@@ -92,6 +92,8 @@
 openmax_not_found="== Could not find libnvomx. OpenMax support disabled. =="
 libass_using_internal="== External Libass not found, using internal. =="
 ssh_not_found="== Could not find libssh. SSH FTP VFS support disabled. =="
+samba_not_found="== Could not find libsmbclient. SAMBA support disabled. =="
+samba_disabled="== SAMBA support disabled. =="
 librtmp_not_found="== Could not find libRTMP. RTMP support disabled. =="
 librtmp_disabled="== RTMP support disabled. =="
 
@@ -274,6 +276,12 @@
   [use_avahi=$enableval],
   [use_avahi=yes])
 
+AC_ARG_ENABLE([samba],
+  [AS_HELP_STRING([--disable-samba],
+  [disable SAMBA support (default is enabled if libsmbclient is found)])],
+  [use_samba=$enableval],
+  [use_samba=yes])
+
 AC_ARG_ENABLE([non-free],
   [AS_HELP_STRING([--disable-non-free],
   [disable componentents with non-compliant licenses])],
@@ -625,7 +633,6 @@
 if test "$host_vendor" = "apple" ; then
   AC_CHECK_LIB([iconv],     [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([mysqlclient], [main],, AC_MSG_ERROR($missing_library))
-  AC_CHECK_LIB([smbclient], [main],, AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([VORBISFILE],     [vorbisfile],
     [INCLUDES="$INCLUDES $VORBISFILE_CFLAGS"; LIBS="$LIBS $VORBISFILE_LIBS"],
     AC_MSG_ERROR($missing_library))
@@ -643,7 +650,6 @@
   AC_CHECK_LIB([mysqlclient],[main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([faad],       [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_HEADER([FLAC/stream_decoder.h],, AC_MSG_ERROR($missing_library))
-  AC_CHECK_LIB([smbclient],  [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([SDL_mixer],  [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([dl],         [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([resolv],     [main],, AC_MSG_ERROR($missing_library))
@@ -786,6 +792,18 @@
    fi
 fi
 
+# samba
+if test "$use_samba" = "yes"; then
+  AC_CHECK_LIB([smbclient], [main],,
+    use_samba=no;AC_MSG_RESULT($samba_not_found))
+else
+  AC_MSG_RESULT($samba_disabled)
+fi
+
+if test "$use_samba" = "yes"; then
+  AC_DEFINE([HAVE_LIBSMBCLIENT], [1], [Define to 1 if you have Samba installed])
+fi
+
 # XRandR
 if test "$host_vendor" = "apple" ; then
   use_xrandr="no"
@@ -1312,6 +1330,12 @@
   final_message="$final_message\n  Avahi:\tNo"
 fi
 
+if test "$use_samba" = "yes"; then
+  final_message="$final_message\n  Samba:\tYes"
+else
+  final_message="$final_message\n  Samba:\tNo"
+fi
+
 if test "$host_vendor" = "apple"; then
   echo "#define GIT_REV \"$GIT_REV\"" > git_revision.h
 else
diff -Naur xbmc-10.1-Dharma/guilib/system.h xbmc-10.1-Dharma.patch/guilib/system.h
--- xbmc-10.1-Dharma/guilib/system.h	2011-03-08 02:49:13.000000000 +0100
+++ xbmc-10.1-Dharma.patch/guilib/system.h	2011-09-14 15:19:35.257592410 +0200
@@ -58,7 +58,6 @@
 #endif
 
 #define HAS_FILESYSTEM
-#define HAS_FILESYSTEM_SMB
 #define HAS_FILESYSTEM_CDDA
 #define HAS_FILESYSTEM_RTV
 #define HAS_FILESYSTEM_DAAP
@@ -67,6 +66,10 @@
 #define HAS_FILESYSTEM_HTSP
 #define HAS_FILESYSTEM_MMS
 
+#ifdef HAVE_LIBSMBCLIENT
+#define HAS_FILESYSTEM_SMB
+#endif
+
 /**********************
  * Non-free Components
  **********************/
diff -Naur xbmc-10.1-Dharma/xbmc/GUIDialogMediaSource.cpp xbmc-10.1-Dharma.patch/xbmc/GUIDialogMediaSource.cpp
--- xbmc-10.1-Dharma/xbmc/GUIDialogMediaSource.cpp	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/GUIDialogMediaSource.cpp	2011-09-14 17:13:07.080677066 +0200
@@ -240,9 +240,11 @@
     share1.m_ignore = true;
     extraShares.push_back(share1);
 
+#ifdef HAS_FILESYSTEM_SMB
     share1.strPath = "smb://";
     share1.strName = g_localizeStrings.Get(20171);
     extraShares.push_back(share1);
+#endif
 
     share1.strPath = "upnp://";
     share1.strName = "UPnP Devices";
@@ -282,9 +284,11 @@
     share1.strName = "ReplayTV Devices";
     extraShares.push_back(share1);
 
+#ifdef HAS_FILESYSTEM_SMB
     share1.strPath = "smb://";
     share1.strName = g_localizeStrings.Get(20171);
     extraShares.push_back(share1);
+#endif
 
     share1.strPath = "hdhomerun://";
     share1.strName = "HDHomerun Devices";
@@ -313,9 +317,11 @@
       extraShares.push_back(share1);
     }
 
+#ifdef HAS_FILESYSTEM_SMB
     share1.strPath = "smb://";
     share1.strName = g_localizeStrings.Get(20171);
     extraShares.push_back(share1);
+#endif
 
     share1.strPath = "upnp://";
     share1.strName = "UPnP Devices";
diff -Naur xbmc-10.1-Dharma/xbmc/GUIDialogNetworkSetup.cpp xbmc-10.1-Dharma.patch/xbmc/GUIDialogNetworkSetup.cpp
--- xbmc-10.1-Dharma/xbmc/GUIDialogNetworkSetup.cpp	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/GUIDialogNetworkSetup.cpp	2011-09-14 16:49:00.902517185 +0200
@@ -128,7 +128,9 @@
     return;
 
   pSpin->Clear();
+#ifdef HAS_FILESYSTEM_SMB
   pSpin->AddLabel(g_localizeStrings.Get(20171), NET_PROTOCOL_SMB);
+#endif
   pSpin->AddLabel(g_localizeStrings.Get(20256), NET_PROTOCOL_HTSP);
   pSpin->AddLabel(g_localizeStrings.Get(20257), NET_PROTOCOL_VTP);
   pSpin->AddLabel(g_localizeStrings.Get(21331), NET_PROTOCOL_TUXBOX);
