From 35e4a2eb5515fe72d87b6d78b0567609df79df2b Mon Sep 17 00:00:00 2001
From: bobo1on1 <bob-nospam-@xbmc.org>
Date: Wed, 28 Mar 2012 21:30:25 +0200
Subject: [PATCH] fixed: possible deadlock

---
 xbmc/peripherals/devices/PeripheralCecAdapter.cpp |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 0cb6602..1c91341 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -140,8 +140,10 @@ void CPeripheralCecAdapter::Announce(AnnouncementFlag flag, const char *sender,
   else if (flag == System && !strcmp(sender, "xbmc") && !strcmp(message, "OnSleep"))
   {
     // this will also power off devices when we're the active source
-    CSingleLock lock(m_critSection);
-    m_bStop = true;
+    {
+      CSingleLock lock(m_critSection);
+      m_bStop = true;
+    }
     WaitForThreadExit(0);
   }
   else if (flag == System && !strcmp(sender, "xbmc") && !strcmp(message, "OnWake"))
-- 
1.7.5.4

