diff -Naur xbmc-11.0.1/xbmc/settings/GUIWindowSettingsCategory.cpp xbmc-11.0.1.patch/xbmc/settings/GUIWindowSettingsCategory.cpp
--- xbmc-11.0.1/xbmc/settings/GUIWindowSettingsCategory.cpp	2012-06-30 19:13:51.558262433 +0200
+++ xbmc-11.0.1.patch/xbmc/settings/GUIWindowSettingsCategory.cpp	2012-06-30 19:59:14.545818328 +0200
@@ -553,7 +553,7 @@
     {
       bool hasInterop = false;
 #ifdef GL_NV_vdpau_interop
-      hasInterop = true;
+      hasInterop = false;
 #endif
       CGUIControl *pControl = (CGUIControl *)GetControl(pSettingControl->GetID());
       if (pControl && hasInterop && glewIsSupported("GL_NV_vdpau_interop"))
