diff -Naur xbmc-10.1-Dharma/xbmc/lib/libPython/xbmcmodule/xbmcplugin.cpp xbmc-10.1-Dharma.patch/xbmc/lib/libPython/xbmcmodule/xbmcplugin.cpp
--- xbmc-10.1-Dharma/xbmc/lib/libPython/xbmcmodule/xbmcplugin.cpp	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/lib/libPython/xbmcmodule/xbmcplugin.cpp	2011-08-21 17:13:13.382281536 +0200
@@ -74,7 +74,7 @@
     if (!PyArg_ParseTupleAndKeywords(
       args,
       kwds,
-      (char*)"iOO|bl",
+      (char*)"iOO|bi",
       (char**)keywords,
       &handle,
       &pURL,
