From 0e26e69497c0977f6a0efbf48d199f6313c539cf Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 3 Aug 2014 13:03:44 +0300
Subject: [PATCH] bye bye libflac

---
 configure.in | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.in b/configure.in
index a9d5f46..702f086 100644
--- a/configure.in
+++ b/configure.in
@@ -1056,7 +1056,6 @@ AC_CHECK_HEADER([jpeglib.h],,        AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([ogg/ogg.h],,        AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([vorbis/vorbisfile.h],, AC_MSG_ERROR($missing_library))
 AC_CHECK_HEADER([libmodplug/modplug.h],, AC_MSG_ERROR($missing_library))
-AC_CHECK_HEADER([FLAC/stream_decoder.h],, AC_MSG_ERROR($missing_library))
 
 AC_CHECK_HEADER([curl/curl.h],, AC_MSG_ERROR($missing_library))
 XB_FIND_SONAME([CURL], [curl])
@@ -1219,7 +1218,6 @@ fi
 fi
 
 XB_FIND_SONAME([OGG],         [ogg])
-XB_FIND_SONAME([FLAC],        [FLAC])
 XB_FIND_SONAME([VORBIS],      [vorbis])
 XB_FIND_SONAME([VORBISFILE],  [vorbisfile])
 XB_FIND_SONAME([MODPLUG],     [modplug])
@@ -2596,7 +2594,6 @@ AC_SUBST(OUTPUT_FILES)
 AC_SUBST(HAVE_XBMC_NONFREE)
 AC_SUBST(USE_ASAP_CODEC)
 AC_SUBST(LIBCURL_BASENAME)
-AC_SUBST(LIBFLAC_BASENAME)
 AC_SUBST(LIBVORBISFILE_BASENAME)
 AC_SUBST(LIBMODPLUG_BASENAME)
 AC_SUBST(LIBOGG_BASENAME)
-- 
1.9.1

