--- xbmc-pvr-11.0.2/configure.in.orig	2012-08-29 01:02:48.633007939 +0200
+++ xbmc-pvr-11.0.2/configure.in	2012-08-29 01:03:40.593006545 +0200
@@ -1287,11 +1287,7 @@
   AC_DEFINE([USE_EXTERNAL_FFMPEG], [1], [Whether to use external FFmpeg libraries.])
 
   # Disable vdpau support if external libavcodec doesn't have it
-  AC_RUN_IFELSE(
-    AC_LANG_PROGRAM([[#include <libavcodec/avcodec.h>]],
-      [[avcodec_register_all();
-        AVCodec *codec = avcodec_find_decoder_by_name("vc1_vdpau");
-        return (codec) ? 0 : 1;]]),,
+  AC_CHECK_LIB([avcodec], [ff_vdpau_vc1_decode_picture],,
     [if test "x$use_vdpau" = "xyes"; then
       AC_MSG_ERROR($ffmpeg_vdpau_not_supported)
     else
