diff -Naur xbmc-9389dc8/configure.in xbmc-9389dc8.patch/configure.in
--- xbmc-9389dc8/configure.in	2011-09-01 17:21:47.000000000 +0200
+++ xbmc-9389dc8.patch/configure.in	2011-09-01 23:30:40.419399392 +0200
@@ -428,6 +428,14 @@
 esac
 AC_SUBST([ARCH])
 
+AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
+if test "$HAVE_GIT" = "yes" -a "$GIT_REV" = ""; then
+  GIT_REV=$(git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h %ci" HEAD | awk '{gsub("-", "");print $2"-"$1}')
+elif test "$GIT_REV" = ""; then
+  GIT_REV="Unknown"
+fi
+final_message="$final_message\n  git Rev.:\t${GIT_REV}"
+
 # platform debug flags
 if test "$use_debug" = "yes"; then
   final_message="$final_message\n  Debugging:\tYes"
@@ -1177,7 +1185,6 @@
   fi
 fi
 
-AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
 # Checks for header files.
 AC_HEADER_DIRENT
@@ -1408,12 +1415,6 @@
   final_message="$final_message\n  Avahi:\tNo"
 fi
 
-if test "$HAVE_GIT" = "yes"; then
-  GIT_REV=$(git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h %ci" HEAD | awk '{gsub("-", "");print $2"-"$1}')
-fi
-if test "$GIT_REV" = ""; then
-  GIT_REV="Unknown"
-fi
 if test "$host_vendor" = "apple"; then
   echo "#define GIT_REV \"$GIT_REV\"" > git_revision.h
 else
