diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/GUISettings.cpp xbmc-Dharma-10.1-c8405db.patch/xbmc/GUISettings.cpp
--- xbmc-Dharma-10.1-c8405db/xbmc/GUISettings.cpp	2011-02-15 14:32:35.774825404 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/GUISettings.cpp	2011-02-15 14:33:07.788183524 +0100
@@ -527,6 +527,7 @@
   AddBool(NULL, "videolibrary.seasonthumbs", 20382, true);
   AddBool(vdl, "videolibrary.actorthumbs", 20402, true);
   AddInt(vdl, "videolibrary.flattentvshows", 20412, 1, 0, 1, 2, SPIN_CONTROL_TEXT);
+  AddBool(NULL, "videolibrary.flattenmoviesets", 22002, false);
   AddBool(vdl, "videolibrary.updateonstartup", 22000, false);
   AddBool(vdl, "videolibrary.backgroundupdate", 22001, false);
   AddSeparator(vdl, "videolibrary.sep3");
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/VideoDatabase.cpp xbmc-Dharma-10.1-c8405db.patch/xbmc/VideoDatabase.cpp
--- xbmc-Dharma-10.1-c8405db/xbmc/VideoDatabase.cpp	2011-02-15 13:30:05.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/VideoDatabase.cpp	2011-02-15 14:33:07.792183444 +0100
@@ -4394,7 +4394,7 @@
     if (NULL == m_pDS.get()) return false;
 
     CStdString strSQL = "select * from movieview ";
-    if (fetchSets)
+    if (fetchSets && !g_guiSettings.GetBool("videolibrary.flattenmoviesets"))
     {
       // not getting a set, so grab all sets that match this where clause first
       CStdString setsWhere;
