From ff7b191ae381bd16a3b86fd2820d43a45a08fe4b Mon Sep 17 00:00:00 2001
From: xbmc <fernetmenta@online.de>
Date: Mon, 27 Feb 2012 18:27:47 +0100
Subject: [PATCH] linux: disable gui sounds before starting player

---
 xbmc/Application.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 9b362af..8a0fe87 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -3868,7 +3868,7 @@ bool CApplication::PlayFile(const CFileItem& item, bool bRestart)
 
   // Workaround for bug/quirk in SDL_Mixer on OSX.
   // TODO: Remove after GUI Sounds redux
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(_LINUX)
   g_audioManager.Enable(false);
 #endif
 
@@ -3932,7 +3932,7 @@ bool CApplication::PlayFile(const CFileItem& item, bool bRestart)
     }
 #endif
 
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(_LINUX)
     g_audioManager.Enable(false);
 #endif
   }
-- 
1.7.5.4

