--- a/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2013-03-28 00:07:58.432987577 +0100
+++ b/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2013-03-28 00:08:29.226987558 +0100
@@ -440,7 +440,7 @@
   if (streaminfo)
   {
     /* too speed up dvd switches, only analyse very short */
-    if(m_pInput->IsStreamType(DVDSTREAM_TYPE_DVD))
+    if(m_pInput->IsStreamType(DVDSTREAM_TYPE_DVD) || m_pInput->IsStreamType(DVDSTREAM_TYPE_PVRMANAGER))
       m_pFormatContext->max_analyze_duration = 500000;
 
 
