From 4f6188bc2bcee52ab3a150fff336b58c11f8928a Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 22 Mar 2014 22:18:28 +0200
Subject: [PATCH] dont set _NET_WM_STATE_FULLSCREEN

---
 xbmc/windowing/X11/WinSystemX11.cpp |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/xbmc/windowing/X11/WinSystemX11.cpp b/xbmc/windowing/X11/WinSystemX11.cpp
index c95f4ec..d12e050 100644
--- a/xbmc/windowing/X11/WinSystemX11.cpp
+++ b/xbmc/windowing/X11/WinSystemX11.cpp
@@ -903,12 +903,6 @@ bool CWinSystemX11::SetWindow(int width, int height, bool fullscreen, const std:
                     InputOutput, vi->visual,
                     mask, &swa);
 
-    if (fullscreen && hasWM)
-    {
-      Atom fs = XInternAtom(m_dpy, "_NET_WM_STATE_FULLSCREEN", True);
-      XChangeProperty(m_dpy, m_mainWindow, XInternAtom(m_dpy, "_NET_WM_STATE", True), XA_ATOM, 32, PropModeReplace, (unsigned char *) &fs, 1);
-    }
-
     // define invisible cursor
     Pixmap bitmapNoData;
     XColor black;
-- 
1.7.2.5

