From f3ffe4e25bfca41379e4e2ffba75ad55d13ba30e Mon Sep 17 00:00:00 2001
From: xbmc <fernetmenta@online.de>
Date: Thu, 21 Mar 2013 07:53:20 +0100
Subject: [PATCH] only free unused textures after a flip

---
 xbmc/Application.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index fa0e71d..5bd2d72 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2363,11 +2363,10 @@ void CApplication::Render()
   {
     g_graphicsContext.Flip(dirtyRegions);
     g_renderManager.NotifyDisplayFlip();
+    g_TextureManager.FreeUnusedTextures();
   }
   CTimeUtils::UpdateFrameTime(flip);
 
-  g_TextureManager.FreeUnusedTextures();
-
   g_renderManager.UpdateResolution();
   g_renderManager.ManageCaptures();
 }
-- 
1.8.1.5

