diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 13:30:17.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 19:08:46.445385047 +0100
@@ -387,6 +387,15 @@
      AC_MSG_ERROR(unsupported host ($host))
 esac
 AC_SUBST([ARCH])
+
+AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
+if test "$HAVE_GIT" = "yes" -a "$GIT_REV" = ""; then
+  GIT_REV=$(git rev-parse --short HEAD)
+elif test "$GIT_REV" = ""; then
+  GIT_REV="Unknown"
+fi
+final_message="$final_message\n  git Rev.:\t${GIT_REV}"
+
 if test "$use_debug" = "yes"; then
   final_message="$final_message\n  Debugging:\tYes"
   if test "$use_profiling" = "yes"; then
@@ -1097,7 +1106,6 @@
   fi
 fi
 
-AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
 # Checks for header files.
 AC_HEADER_DIRENT
@@ -1282,12 +1290,6 @@
   final_message="$final_message\n  Avahi:\tNo"
 fi
 
-if test "$HAVE_GIT" = "yes"; then
-  GIT_REV=$(git rev-parse --short HEAD)
-fi
-if test "$GIT_REV" = ""; then
-  GIT_REV="Unknown"
-fi
 if test "$host_vendor" = "apple"; then
   echo "#define GIT_REV \"$GIT_REV\"" > git_revision.h
 else
