diff -Naur xbmc-14-a29b21a/xbmc/windowing/X11/WinSystemX11.h xbmc-14-a29b21a.patch/xbmc/windowing/X11/WinSystemX11.h
--- xbmc-14-a29b21a/xbmc/windowing/X11/WinSystemX11.h	2014-08-24 17:41:42.000000000 +0200
+++ xbmc-14-a29b21a.patch/xbmc/windowing/X11/WinSystemX11.h	2014-08-31 13:44:50.250774848 +0200
@@ -47,6 +47,7 @@
   virtual bool ResizeWindow(int newWidth, int newHeight, int newLeft, int newTop);
   virtual bool SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, bool blankOtherDisplays);
   virtual void UpdateResolutions();
+  virtual bool CanDoWindowed() { return false; }
   virtual int  GetNumScreens() { return 1; }
   virtual int  GetCurrentScreen() { return m_nScreen; }
   virtual void ShowOSMouse(bool show);
