diff -Naur xbmc-10.1-Dharma/xbmc/FileSystem/FileSmb.cpp xbmc-10.1-Dharma.patch/xbmc/FileSystem/FileSmb.cpp
--- xbmc-10.1-Dharma/xbmc/FileSystem/FileSmb.cpp	2011-09-05 02:24:35.154994777 +0200
+++ xbmc-10.1-Dharma.patch/xbmc/FileSystem/FileSmb.cpp	2011-09-05 02:22:55.604582159 +0200
@@ -122,6 +122,8 @@
       // use the weaker LANMAN password hash in order to be compatible with older servers
       fprintf(f, "\tclient lanman auth = yes\n");
       fprintf(f, "\tlanman auth = yes\n");
+      fprintf(f, "\tsocket options = TCP_NODELAY IPTOS_LOWDELAY SO_RCVBUF=65536 SO_SNDBUF=65536\n");
+      fprintf(f, "\tlock directory = %s/.smb/\n", getenv("HOME"));
 
       // set wins server if there's one. name resolve order defaults to 'lmhosts host wins bcast'.
       // if no WINS server has been specified the wins method will be ignored.
