diff -Naur xbmc-eedceeb/xbmc/settings/GUISettings.cpp xbmc-eedceeb.patch/xbmc/settings/GUISettings.cpp
--- xbmc-eedceeb/xbmc/settings/GUISettings.cpp	2011-06-18 21:30:30.000000000 +0200
+++ xbmc-eedceeb.patch/xbmc/settings/GUISettings.cpp	2011-06-18 22:02:53.535611620 +0200
@@ -422,7 +422,8 @@
   AddString(vs, "videoscreen.testpattern",226,"", BUTTON_CONTROL_STANDARD);
 #endif
 #if defined(_LINUX) && !defined(__APPLE__)
-  AddBool(NULL, "videoscreen.haslcd", 4501, false);
+  AddSeparator(vs, "videoscreen.sep2");
+  AddBool(vs, "videoscreen.haslcd", 4501, false);
 #endif
 
   CSettingsCategory* ao = AddCategory(4, "audiooutput", 772);
