diff -Naur xbmc-13.alpha-73371a9/configure.in xbmc-13.alpha-73371a9.patch/configure.in
--- xbmc-13.alpha-73371a9/configure.in	2014-02-12 00:22:27.303734884 +0100
+++ xbmc-13.alpha-73371a9.patch/configure.in	2014-02-12 00:27:15.390961157 +0100
@@ -1746,21 +1746,25 @@
     AC_MSG_NOTICE("using ffmpeg: ${with_ffmpeg}")
     export PKG_CONFIG_PATH="${with_ffmpeg}/lib/pkgconfig:$PKG_CONFIG_PATH"
   fi
-fi
 
-if test "$FFMPEG_FOUND" != "true"; then
+  # get the libdir for static linking
+  FFMPEG_LIBDIR=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --variable=libdir libavcodec)
+  GNUTLS_ALL_LIBS=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --libs-only-l --silence-errors gnutls)
+  VORBISENC_ALL_LIBS=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --libs-only-l --silence-errors vorbisenc)
+
+  if test "$FFMPEG_FOUND" != "true"; then
+    PKG_CHECK_MODULES([FFMPEG], [$FFMPEG_LIBNAMES],
+      [INCLUDES="$INCLUDES $FFMPEG_CFLAGS"; LIBS="$LIBS $FFMPEG_LIBS"; FFMPEG_FOUND="true"],
+      [AC_MSG_ERROR("ffmpeg not found")])
+  fi
+else
   PKG_CHECK_MODULES([FFMPEG], [$FFMPEG_LIBNAMES],
-    [INCLUDES="$INCLUDES $FFMPEG_CFLAGS"; LIBS="$LIBS $FFMPEG_LIBS"; FFMPEG_FOUND="true"], 
+    [INCLUDES="$INCLUDES $FFMPEG_CFLAGS"; LIBS="$LIBS $FFMPEG_LIBS"; FFMPEG_FOUND="true"],
     [AC_MSG_ERROR("ffmpeg not found")])
 fi
 
-# get the libdir for static linking
-FFMPEG_LIBDIR=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --variable=libdir libavcodec)
-GNUTLS_ALL_LIBS=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --libs-only-l --silence-errors gnutls)
-VORBISENC_ALL_LIBS=$(PKG_CONFIG_SYSROOT_DIR="" ${PKG_CONFIG} --static --libs-only-l --silence-errors vorbisenc)
-
 # TODO: make this conditional
-USE_STATIC_FFMPEG=1 
+USE_STATIC_FFMPEG=1
 
 echo "Checking for SWIG installation"
 AC_PATH_PROG(SWIG_EXE, swig, "none")
diff -Naur xbmc-13.alpha-73371a9/Makefile.in xbmc-13.alpha-73371a9.patch/Makefile.in
--- xbmc-13.alpha-73371a9/Makefile.in	2014-02-11 22:25:40.000000000 +0100
+++ xbmc-13.alpha-73371a9.patch/Makefile.in	2014-02-12 00:21:17.374681719 +0100
@@ -488,17 +488,6 @@
 DYNOBJSXBMC+= xbmc/freebsd/freebsd.a
 endif
 
-ifeq (@USE_STATIC_FFMPEG@,1)
-DYNOBJSXBMC += @FFMPEG_LIBDIR@/libavcodec.a \
-               @FFMPEG_LIBDIR@/libavfilter.a \
-               @FFMPEG_LIBDIR@/libswresample.a \
-               @FFMPEG_LIBDIR@/libavformat.a \
-               @FFMPEG_LIBDIR@/libavutil.a \
-               @FFMPEG_LIBDIR@/libpostproc.a \
-               @FFMPEG_LIBDIR@/libswscale.a
-
-LIBS+= @GNUTLS_ALL_LIBS@ @VORBISENC_ALL_LIBS@
-endif
 OBJSXBMC:=$(filter-out $(DYNOBJSXBMC), $(OBJSXBMC))
 
 MAINOBJS=xbmc/xbmc.o
