diff -Naur xbmc-9389dc8/configure.in xbmc-9389dc8.patch/configure.in
--- xbmc-9389dc8/configure.in	2011-09-01 23:34:05.488142309 +0200
+++ xbmc-9389dc8.patch/configure.in	2011-09-01 23:51:26.371021628 +0200
@@ -68,6 +68,8 @@
 goom_disabled="== GOOM disabled. =="
 rsxs_enabled="== RSXS enabled. =="
 rsxs_disabled="== RSXS disabled. =="
+xbmcprojectm_enabled="== XBMCProjectM enabled. =="
+xbmxprojectm_disabled="== XBMCProjectM disabled. =="
 x11_enabled="== X11 enabled. =="
 x11_disabled="== X11 disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
@@ -219,6 +221,12 @@
   [use_rsxs=$enableval],
   [use_rsxs=yes])
 
+AC_ARG_ENABLE([xbmcprojectm],
+  [AS_HELP_STRING([--enable-xbmcprojectm],
+  [enable XBMCProjectM visualisation (default is yes)])],
+  [use_xbmcprojectm=$enableval],
+  [use_xbmcprojectm=yes])
+
 AC_ARG_ENABLE([x11],
   [AS_HELP_STRING([--enable-x11],
   [enable x11 (default is yes) 'Linux Only'])],
@@ -850,6 +858,15 @@
   fi
 fi
 
+# XBMCPROJECTM
+if test "$use_xbmcprojectm" = "no" || test "$use_gl" = "no"; then
+  AC_MSG_NOTICE($xbmcprojectm_disabled)
+  DISABLE_XBMCPROJECTM=1
+else
+  AC_MSG_NOTICE($xbmcprojectm_enabled)
+  DISABLE_XBMCPROJECTM=0
+fi
+
 # libRTMP
 if test "$use_librtmp" != "no"; then
   AC_CHECK_HEADERS([librtmp/log.h librtmp/amf.h librtmp/rtmp.h],,
@@ -1324,6 +1341,12 @@
   final_message="$final_message\n  RSXS:\t\tNo"
 fi
 
+if test "$use_xbmcprojectm" = "yes"; then
+  final_message="$final_message\n  XBMCProjectM:\tYes"
+else
+  final_message="$final_message\n  XBMCProjectM:\tNo"
+fi
+
 if test "$use_x11" = "yes"; then
   final_message="$final_message\n  X11:\t\tYes"
 else
@@ -1553,6 +1576,7 @@
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(DISABLE_GOOM)
 AC_SUBST(DISABLE_RSXS)
+AC_SUBST(DISABLE_XBMCPROJECTM)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(PYTHON_VERSION)
 AC_SUBST(OUTPUT_FILES)
@@ -1797,7 +1821,7 @@
     fi
 set +x
 fi
-], [0])
+], [$DISABLE_XBMCPROJECTM])
 
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   ./configure  \
diff -Naur xbmc-9389dc8/Makefile.in xbmc-9389dc8.patch/Makefile.in
--- xbmc-9389dc8/Makefile.in	2011-09-01 23:34:05.581143554 +0200
+++ xbmc-9389dc8.patch/Makefile.in	2011-09-01 23:52:59.832272157 +0200
@@ -144,8 +144,10 @@
 ifeq (@USE_OPENGL@,1)
 VIS_DIRS+=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+ifneq (@DISABLE_XBMCPROJECTM@,1)
+  VIS_DIRS+=xbmc/visualizations/XBMCProjectM
+endif
 endif
 
 ifeq (($(findstring osx,@ARCH@), osx),(@USE_OPENGL@,1))
