diff -Naur xbmc-12.0.6/xbmc/powermanagement/PowerManager.h xbmc-12.0.6.patch/xbmc/powermanagement/PowerManager.h
--- xbmc-12.0.6/xbmc/powermanagement/PowerManager.h	2013-03-11 18:01:45.000000000 +0100
+++ xbmc-12.0.6.patch/xbmc/powermanagement/PowerManager.h	2013-03-12 13:42:28.698769168 +0100
@@ -32,8 +32,8 @@
   virtual bool Reboot()       { return false; }
 
   virtual bool CanPowerdown() { return true; }
-  virtual bool CanSuspend()   { return true; }
-  virtual bool CanHibernate() { return true; }
+  virtual bool CanSuspend()   { return false; }
+  virtual bool CanHibernate() { return false; }
   virtual bool CanReboot()    { return true; }
 
   virtual int  BatteryLevel() { return 0; }
