diff -Naur xbmc-f43dcfa/configure.in xbmc-f43dcfa.patch/configure.in
--- xbmc-f43dcfa/configure.in	2011-09-10 02:31:22.000000000 +0200
+++ xbmc-f43dcfa.patch/configure.in	2011-09-10 08:16:46.068320869 +0200
@@ -963,7 +963,7 @@
 USE_LIBAFPCLIENT=0
 if test "$host_vendor" = "apple" ; then
   if test "x$use_libafpclient" != "xno"; then
-    AC_CHECK_HEADERS([afp/libafpclient.h],,
+    AC_CHECK_HEADERS([afpfs-ng/libafpclient.h],,
      [if test "x$use_libafpclient" = "xyes"; then
         AC_MSG_ERROR($libafpclient_not_found)
       elif test "x$use_libafpclient" != "xno"; then
diff -Naur xbmc-f43dcfa/xbmc/filesystem/DllLibAfp.h xbmc-f43dcfa.patch/xbmc/filesystem/DllLibAfp.h
--- xbmc-f43dcfa/xbmc/filesystem/DllLibAfp.h	2011-09-10 02:31:05.000000000 +0200
+++ xbmc-f43dcfa.patch/xbmc/filesystem/DllLibAfp.h	2011-09-10 08:17:55.057220975 +0200
@@ -27,9 +27,9 @@
 #ifdef __cplusplus
 extern "C" {
 #endif   
-#include <afp/libafpclient.h>
-#include <afp/map_def.h>
-#include <afp/midlevel.h>
+#include <afpfs-ng/libafpclient.h>
+#include <afpfs-ng/map_def.h>
+#include <afpfs-ng/midlevel.h>
 #ifdef __cplusplus
 }
 #endif
