diff -Naur xbmc-Dharma-10.1-c8405db/configure.in xbmc-Dharma-10.1-c8405db.patch/configure.in
--- xbmc-Dharma-10.1-c8405db/configure.in	2011-02-15 14:00:23.143803217 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/configure.in	2011-02-15 14:01:15.327739326 +0100
@@ -1263,12 +1263,6 @@
   final_message="$final_message\n  MID Support:\tNo"
 fi
 
-if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
-else
-  INCLUDES="$INCLUDES"
-fi
-
 if test "x$use_ccache" != "xno"; then
   AC_PATH_PROG(CCACHE,ccache,none)
   if test "$ac_cv_path_CCACHE" = "none"; then
diff -Naur xbmc-Dharma-10.1-c8405db/tools/TexturePacker/XBMCTex.cpp xbmc-Dharma-10.1-c8405db.patch/tools/TexturePacker/XBMCTex.cpp
--- xbmc-Dharma-10.1-c8405db/tools/TexturePacker/XBMCTex.cpp	2011-02-15 13:30:04.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/tools/TexturePacker/XBMCTex.cpp	2011-02-15 14:01:15.328739305 +0100
@@ -40,7 +40,7 @@
 #endif
 
 #ifdef _LINUX
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 #else
 #include "../../xbmc/lib/liblzo/LZO1X.H"
 #endif
