diff -Naur xbmc-10.0-Dharma/xbmc/cores/dvdplayer/Codecs/ffmpeg/libavformat/spdif.c xbmc-10.0-Dharma.patch/xbmc/cores/dvdplayer/Codecs/ffmpeg/libavformat/spdif.c
--- xbmc-10.0-Dharma/xbmc/cores/dvdplayer/Codecs/ffmpeg/libavformat/spdif.c	2010-12-17 20:13:36.000000000 +0100
+++ xbmc-10.0-Dharma.patch/xbmc/cores/dvdplayer/Codecs/ffmpeg/libavformat/spdif.c	2011-01-07 01:19:48.721555949 +0100
@@ -108,7 +108,7 @@
 static int spdif_header_ac3(AVFormatContext *s, AVPacket *pkt)
 {
     IEC958Context *ctx = s->priv_data;
-    int bitstream_mode = pkt->data[6] & 0x7;
+    int bitstream_mode = pkt->data[5] & 0x7;
 
     ctx->data_type  = IEC958_AC3 | (bitstream_mode << 8);
     ctx->pkt_offset = AC3_FRAME_SIZE << 2;
