diff -Naur xbmc-f70eb43/lib/enca/configure.ac xbmc-f70eb43.patch/lib/enca/configure.ac
--- xbmc-f70eb43/lib/enca/configure.ac	2013-01-11 17:06:57.000000000 +0100
+++ xbmc-f70eb43.patch/lib/enca/configure.ac	2013-01-12 08:44:23.957554815 +0100
@@ -24,7 +24,7 @@
   test/Makefile \
   tools/Makefile)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([1.8 gnits check-news dist-bzip2])
 AM_ACLOCAL_INCLUDE(m4)
 AM_MAINTAINER_MODE
@@ -40,7 +40,6 @@
 AC_AIX
 AC_ISC_POSIX
 AC_PROG_LIBTOOL
-AM_PROG_CC_STDC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_LN_S
diff -Naur xbmc-f70eb43/lib/libdvd/libdvdcss/configure.ac xbmc-f70eb43.patch/lib/libdvd/libdvdcss/configure.ac
--- xbmc-f70eb43/lib/libdvd/libdvdcss/configure.ac	2013-01-11 17:06:58.000000000 +0100
+++ xbmc-f70eb43.patch/lib/libdvd/libdvdcss/configure.ac	2013-01-12 08:45:45.361975774 +0100
@@ -5,7 +5,7 @@
 AC_CANONICAL_SYSTEM
 
 AM_INIT_AUTOMAKE(libdvdcss, 1.2.10)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_PROG_CC
 AC_STDC_HEADERS
