From 5c62df65cb2ef1c7dcebbf07bd6d180960ab6715 Mon Sep 17 00:00:00 2001
From: theuni <theuni-nospam-@xbmc.org>
Date: Wed, 4 Apr 2012 14:53:51 -0400
Subject: [PATCH] ffmpeg: disable ffmpeg's crystalhd implementation for now

---
 configure.in |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.in b/configure.in
index 0d80719..40e9fb5 100755
--- a/configure.in
+++ b/configure.in
@@ -2041,6 +2041,7 @@ XB_CONFIG_MODULE([lib/ffmpeg], [
       --disable-ffmpeg \
       --disable-doc \
       --disable-decoder=mpeg_xvmc \
+      --disable-crystalhd \
       --enable-postproc \
       --enable-gpl \
       --enable-protocol=http \
@@ -2080,6 +2081,7 @@ XB_CONFIG_MODULE([lib/ffmpeg], [
       --disable-ffplay \
       --disable-ffserver \
       --disable-ffmpeg \
+      --disable-crystalhd \
       --enable-shared \
       --disable-doc \
       --enable-postproc \
-- 
1.7.10

