diff -Naur xbmc-pvr-3513480/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp xbmc-pvr-3513480.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp
--- xbmc-pvr-3513480/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2011-04-25 02:36:33.000000000 +0200
+++ xbmc-pvr-3513480.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2011-04-25 05:07:26.043098079 +0200
@@ -771,7 +771,7 @@
           if (!pBuffer)
           {
             // No free pre-allocated buffers so make one
-#ifdef _WIN32
+#if 1
             // force Windows to use YV12 until DX renderer gets NV12 or YUY2 capability.
             pBuffer = new CPictureBuffer(DVDVideoPicture::FMT_YUV420P, m_width, m_height);
 #else
