From 5c61cc2c3ef7b047060742d736fd2a8cc037ae57 Mon Sep 17 00:00:00 2001
From: Jim Carroll <thecarrolls@jiminger.com>
Date: Sun, 3 Feb 2013 16:31:51 -0500
Subject: [PATCH] Remove some annoying log-spam.

---
 xbmc/interfaces/python/CallbackHandler.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/interfaces/python/CallbackHandler.cpp b/xbmc/interfaces/python/CallbackHandler.cpp
index 01707b9..61c0b5f 100644
--- a/xbmc/interfaces/python/CallbackHandler.cpp
+++ b/xbmc/interfaces/python/CallbackHandler.cpp
@@ -35,8 +35,8 @@
      */
     PythonCallbackHandler::PythonCallbackHandler() : RetardedAsynchCallbackHandler("PythonCallbackHandler")
     {
+      TRACE;
       objectThreadState = PyThreadState_Get();
-      CLog::Log(LOGDEBUG,"NEWADDON PythonCallbackHandler construction with PyThreadState 0x%lx",(long)objectThreadState);
     }
 
     /**
-- 
1.7.10

