diff -Naur xbmc-14-0b1a6eb/xbmc/windowing/X11/XRandR.cpp xbmc-14-0b1a6eb.patch/xbmc/windowing/X11/XRandR.cpp
--- xbmc-14-0b1a6eb/xbmc/windowing/X11/XRandR.cpp	2014-09-01 09:53:24.000000000 +0200
+++ xbmc-14-0b1a6eb.patch/xbmc/windowing/X11/XRandR.cpp	2014-09-01 17:39:11.816218099 +0200
@@ -74,7 +74,7 @@
   CStdString cmd;
   cmd  = getenv("XBMC_BIN_HOME");
   cmd += "/xbmc-xrandr";
-  cmd = StringUtils::Format("%s -q --screen %d --current", cmd.c_str(), screennum);
+  cmd = StringUtils::Format("%s -q --screen %d", cmd.c_str(), screennum);
 
   FILE* file = popen(cmd.c_str(),"r");
   if (!file)
