diff --git a/xbmc/windowing/WinEventsX11.cpp b/xbmc/windowing/WinEventsX11.cpp
index a5b4ba2..3586a6a 100644
--- a/xbmc/windowing/WinEventsX11.cpp
+++ b/xbmc/windowing/WinEventsX11.cpp
@@ -731,7 +731,7 @@ XBMCKey CWinEventsX11Imp::LookupXbmcKeySym(KeySym keysym)
 
   // try ascii mappings
   if (keysym>>8 == 0x00)
-    return (XBMCKey)(keysym & 0xFF);
+    return (XBMCKey)tolower(keysym & 0xFF);
 
   return (XBMCKey)keysym;
 }