diff -Naur xbmc-10.1-Dharma/xbmc/cdrip/DllVorbis.h xbmc-10.1-Dharma.patch/xbmc/cdrip/DllVorbis.h
--- xbmc-10.1-Dharma/xbmc/cdrip/DllVorbis.h	2011-03-08 02:49:14.000000000 +0100
+++ xbmc-10.1-Dharma.patch/xbmc/cdrip/DllVorbis.h	2011-03-29 16:37:38.432826576 +0200
@@ -61,7 +61,7 @@
 public:
     virtual ~DllVorbis() {};
     virtual void vorbis_info_init(vorbis_info *vi)
-        { vorbis_info_init(vi); }
+        { ::vorbis_info_init(vi); }
     virtual int vorbis_bitrate_flushpacket(vorbis_dsp_state *vd, ogg_packet *op)
         { return ::vorbis_bitrate_flushpacket(vd, op); }
     virtual int vorbis_bitrate_addblock(vorbis_block *vb)
