From 0be29d9a6e30b1cf1c50e12a9dea28f7dce6f1f7 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Mon, 1 Sep 2014 03:16:37 +0200
Subject: [PATCH] add support to read frequency output if using intel's pstate
 driver

---
 xbmc/utils/CPUInfo.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/utils/CPUInfo.cpp b/xbmc/utils/CPUInfo.cpp
index 93ca784..31b27ba 100644
--- a/xbmc/utils/CPUInfo.cpp
+++ b/xbmc/utils/CPUInfo.cpp
@@ -267,6 +267,8 @@ CCPUInfo::CCPUInfo(void)
     m_fProcTemperature = fopen("/sys/class/thermal/thermal_zone0/temp", "r");  // On Raspberry PIs
 
   m_fCPUFreq = fopen ("/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq", "r");
+  if (m_fCPUFreq == NULL)
+    m_fCPUFreq = fopen ("/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq", "r");
   if (!m_fCPUFreq)
   {
     m_cpuInfoForFreq = true;
