diff -Naur xbmc-10.1-Dharma/configure.in xbmc-10.1-Dharma.patch/configure.in
--- xbmc-10.1-Dharma/configure.in	2011-03-08 02:49:24.000000000 +0100
+++ xbmc-10.1-Dharma.patch/configure.in	2011-09-21 22:13:32.305639980 +0200
@@ -1613,6 +1613,7 @@
       --extra-cflags="$CFLAGS -w -D_DARWIN_C_SOURCE -Dattribute_deprecated=" \
       --disable-amd3dnow \
       --disable-static \
+      --disable-doc \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
@@ -1644,6 +1645,7 @@
     ./configure \
       --extra-cflags="$PASSED_CFLAGS $FFMPEG_EXTRACFLAGS" \
       --disable-static \
+      --disable-doc \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
       `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       `if test "$use_arch" != "no"; then echo --arch=$use_arch; fi`\
