From f025c39679b438cb91a0ef0f205736ff92fddeac Mon Sep 17 00:00:00 2001
From: Vic Lee <llyzs@163.com>
Date: Thu, 10 Feb 2011 09:36:15 +0800
Subject: [PATCH 5/5] UDisk: device must be remount in DeviceChanged event


Signed-off-by: Vic Lee <llyzs@163.com>
---
 xbmc/linux/UDisksProvider.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xbmc/storage/linux/UDisksProvider.cpp b/xbmc/storage/linux/UDisksProvider.cpp
index 8e9b378..e24502a 100644
--- a/xbmc/storage/linux/UDisksProvider.cpp
+++ b/xbmc/storage/linux/UDisksProvider.cpp
@@ -368,6 +368,8 @@ void CUDisksProvider::DeviceChanged(const char *object, IStorageEventsCallback *
   {
     bool mounted = device->m_isMounted;
     device->Update();
+    if (g_advancedSettings.m_handleMounting)
+      device->Mount();
     if (!mounted && device->m_isMounted && callback)
       callback->OnStorageAdded(device->m_Label, device->m_MountPath);
     else if (mounted && !device->m_isMounted && callback)
-- 
1.7.2.3

