diff -Naur xbmc-dab646e/configure.in xbmc-dab646e.patch/configure.in
--- xbmc-dab646e/configure.in	2011-09-27 15:56:42.000000000 +0200
+++ xbmc-dab646e.patch/configure.in	2011-09-27 17:31:10.639824226 +0200
@@ -987,27 +987,22 @@
 fi
 
 # libafpclient
-USE_LIBAFPCLIENT=0
-if test "$host_vendor" = "apple" ; then
-  if test "x$use_libafpclient" != "xno"; then
-    AC_CHECK_HEADERS([afpfs-ng/libafpclient.h],,
-     [if test "x$use_libafpclient" = "xyes"; then
+if test "x$use_libafpclient" != "xno"; then
+  AC_CHECK_HEADERS([afpfs-ng/libafpclient.h],,
+    [if test "x$use_libafpclient" = "xyes"; then
         AC_MSG_ERROR($libafpclient_not_found)
-      elif test "x$use_libafpclient" != "xno"; then
+     elif test "x$use_libafpclient" != "xno"; then
         AC_MSG_NOTICE($libafpclient_not_found)
         use_libafpclient="no"
-      fi
-     ])
+     fi
+    ])
     if test "x$use_libafpclient" != "xno"; then
       XB_FIND_SONAME([AFPCLIENT], [afpclient], [use_libafpclient])
       AC_DEFINE([HAVE_LIBAFPCLIENT], [1], [Whether to use libafpclient library.])
       USE_LIBAFPCLIENT=1
     fi
-  else
-    AC_MSG_NOTICE($libafpclient_disabled)
-  fi
 else
-  use_libafpclient="no"
+    AC_MSG_NOTICE($libafpclient_disabled)
 fi
 
 # libplist for airplay feature
