From 88e250d43dcb13ec2a05f64994c57a3d96b024b4 Mon Sep 17 00:00:00 2001
From: Jim Carroll <thecarrolls@jiminger.com>
Date: Mon, 11 Mar 2013 03:19:46 -0400
Subject: [PATCH] [fix] build issue with std::min on some platforms.

---
 xbmc/interfaces/legacy/File.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/xbmc/interfaces/legacy/File.h b/xbmc/interfaces/legacy/File.h
index d8b5d26..d3c17b3 100644
--- a/xbmc/interfaces/legacy/File.h
+++ b/xbmc/interfaces/legacy/File.h
@@ -27,6 +27,8 @@
 #include "LanguageHook.h"
 #include "commons/Buffer.h"
 
+#include <algorithm>
+
 namespace XBMCAddon
 {
 
@@ -70,7 +72,7 @@
       inline String read(unsigned long numBytes = 0) 
       { 
         XbmcCommons::Buffer b = readBytes(numBytes);
-        return b.getString(numBytes == 0 ? b.remaining() : std::min(b.remaining(),numBytes));
+        return b.getString(numBytes == 0 ? b.remaining() : std::min((unsigned long)b.remaining(),numBytes));
       }
 
       /**
-- 
1.7.10

