diff -Naur xbmc-13.alpha-fc8358e/configure.in xbmc-13.alpha-fc8358e.patch/configure.in
--- xbmc-13.alpha-fc8358e/configure.in	2014-02-02 04:41:09.000000000 +0100
+++ xbmc-13.alpha-fc8358e.patch/configure.in	2014-02-03 15:47:44.591058586 +0100
@@ -2794,6 +2794,7 @@
     CXXFLAGS="" \
     LDFLAGS="$(echo "$LDFLAGS" | sed "s/-Wl,-Bsymbolic-functions//g")" \
     ./configure \
+      --prefix="${prefix}" \
       --extra-cflags="$PASSED_CFLAGS $FFMPEG_EXTRACFLAGS" \
       --disable-static \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
