From ec9129fe58614016d98af521471ff097171ae444 Mon Sep 17 00:00:00 2001
From: Alasdair Campbell <alcoheca@gmail.com>
Date: Tue, 2 Apr 2013 04:41:48 +0100
Subject: [PATCH] upnp: hide unplayable items in fiels view fixes #13526

---
 xbmc/network/upnp/UPnPServer.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/xbmc/network/upnp/UPnPServer.cpp b/xbmc/network/upnp/UPnPServer.cpp
index 86f13eb..0518d14 100644
--- a/xbmc/network/upnp/UPnPServer.cpp
+++ b/xbmc/network/upnp/UPnPServer.cpp
@@ -13,6 +13,7 @@
 #include "filesystem/VideoDatabaseDirectory.h"
 #include "guilib/WindowIDs.h"
 #include "music/tags/MusicInfoTag.h"
+#include "settings/Settings.h"
 #include "settings/GUISettings.h"
 #include "utils/log.h"
 #include "utils/md5.h"
@@ -600,7 +601,14 @@ static NPT_String TranslateWMPObjectId(NPT_String id)
 
             items.Sort(SORT_METHOD_LABEL, SortOrderAscending);
         } else {
-            CDirectory::GetDirectory((const char*)parent_id, items);
+            // this is the only way to hide unplayable items in the 'files'
+            // view as we cannot tell what context (eg music vs video) the
+            // request came from
+            string supported = g_settings.m_pictureExtensions + "|"
+                             + g_settings.m_videoExtensions + "|"
+                             + g_settings.m_musicExtensions + "|"
+                             + g_settings.m_discStubExtensions;
+            CDirectory::GetDirectory((const char*)parent_id, items, supported);
             DefaultSortItems(items);
         }
 
-- 
1.8.1.5

