diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/GUISettings.cpp xbmc-Dharma-10.1-c8405db.patch/xbmc/GUISettings.cpp
--- xbmc-Dharma-10.1-c8405db/xbmc/GUISettings.cpp	2011-02-15 13:30:06.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/GUISettings.cpp	2011-02-15 14:31:51.689709299 +0100
@@ -617,6 +617,7 @@
   AddInt(vid, "myvideos.selectaction", 22079, SELECT_ACTION_PLAY_OR_RESUME, SELECT_ACTION_CHOOSE, 1, SELECT_ACTION_INFO, SPIN_CONTROL_TEXT);
   AddBool(NULL, "myvideos.treatstackasfile", 20051, true);
   AddBool(vid, "myvideos.extractflags",20433, true);
+  AddBool(vid, "myvideos.alwaysextractflags",20433, false);
   AddBool(vid, "myvideos.cleanstrings", 20418, false);
   AddBool(NULL, "myvideos.extractthumb",20433, true);
 
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/ThumbLoader.cpp xbmc-Dharma-10.1-c8405db.patch/xbmc/ThumbLoader.cpp
--- xbmc-Dharma-10.1-c8405db/xbmc/ThumbLoader.cpp	2011-02-15 13:30:05.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/ThumbLoader.cpp	2011-02-15 14:31:51.690709279 +0100
@@ -148,7 +148,7 @@
       m_item.SetThumbnailImage(m_target);
     }
   }
-  else if (m_item.HasVideoInfoTag() && !m_item.GetVideoInfoTag()->HasStreamDetails())
+  else if (m_item.HasVideoInfoTag() && (!m_item.GetVideoInfoTag()->HasStreamDetails() || g_guiSettings.GetBool("myvideos.alwaysextractflags")))
   {
     CLog::Log(LOGDEBUG,"%s - trying to extract filestream details from video file %s", __FUNCTION__, m_path.c_str());
     result = CDVDFileInfo::GetFileStreamDetails(&m_item);
