diff -Naur xbmc-Dharma-10.1-c8405db/guilib/Makefile.in xbmc-Dharma-10.1-c8405db.patch/guilib/Makefile.in
--- xbmc-Dharma-10.1-c8405db/guilib/Makefile.in	2011-02-15 13:30:05.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/guilib/Makefile.in	2011-02-15 14:06:40.345116165 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-I../xbmc/cores/dvdplayer/Codecs/ffmpeg
 endif
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/FileSystem/Makefile.in xbmc-Dharma-10.1-c8405db.patch/xbmc/FileSystem/Makefile.in
--- xbmc-Dharma-10.1-c8405db/xbmc/FileSystem/Makefile.in	2011-02-15 13:30:05.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/FileSystem/Makefile.in	2011-02-15 14:06:40.346116145 +0100
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 ifeq ($(findstring osx,$(ARCH)), osx)
 INCLUDES+=-I../../lib/libcurl-OSX/include
diff -Naur xbmc-Dharma-10.1-c8405db/xbmc/Makefile.in xbmc-Dharma-10.1-c8405db.patch/xbmc/Makefile.in
--- xbmc-Dharma-10.1-c8405db/xbmc/Makefile.in	2011-02-15 13:30:06.000000000 +0100
+++ xbmc-Dharma-10.1-c8405db.patch/xbmc/Makefile.in	2011-02-15 14:06:40.347116125 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer
+INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer
 
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-Icores/dvdplayer/Codecs/ffmpeg
