diff -Naur xbmc.orig/Makefile.in xbmc.patch/Makefile.in
--- xbmc.orig/Makefile.in	2013-12-24 17:50:51.128994014 +0100
+++ xbmc.patch/Makefile.in	2013-12-24 18:04:19.065302931 +0100
@@ -324,7 +324,7 @@
 endif
 
 ifeq (@USE_SSE4@,1)
-FINAL_TARGETS+=sse4
+LIBSSE4+=sse4
 sse4 : force
 	$(MAKE) -C xbmc/linux/sse4
 endif
@@ -464,7 +464,7 @@
 
 codecs: papcodecs dvdpcodecs dvdpextcodecs
 
-libs: libhdhomerun imagelib libexif system/libcpluff-@ARCH@.so $(CMYTH)
+libs: $(LIBSSE4) libhdhomerun imagelib libexif system/libcpluff-@ARCH@.so $(CMYTH)
 
 externals: codecs libs visualizations screensavers libaddon pvraddons
 
