diff -Naur xbmc-dc3bee7/configure.in xbmc-dc3bee7.patch/configure.in
--- xbmc-dc3bee7/configure.in	2011-09-09 01:27:11.000000000 +0200
+++ xbmc-dc3bee7.patch/configure.in	2011-09-09 03:37:55.671676295 +0200
@@ -20,7 +20,7 @@
 [
   if [[ "$host_vendor" != "apple" ]]; then
     AC_MSG_CHECKING([for lib$2 soname])
-    $1_FILENAME=$($LD $LDFLAGS -l$2 -M -o /dev/null 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
+    $1_FILENAME=$($LD -l$2 -M -o /dev/null 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
     if [[ ! -z $$1_FILENAME ]]; then
       $1_SONAME=$(objdump -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
     fi
