diff -Naur a/import.cpp b/import.cpp
--- a/import.cpp	2016-08-23 15:16:10.362347074 +0100
+++ b/import.cpp	2016-08-23 15:16:22.814431356 +0100
@@ -250,7 +250,7 @@
 char *cImport::Add2Description(char *description, const char *Name, int Value)
 {
     char *value=NULL;
-    if (asprintf(&value,"%i",Value)==-1) return false;
+    if (asprintf(&value,"%i",Value)==-1) return NULL;
     description = strcatrealloc(description,Name);
     description = strcatrealloc(description,": ");
     description = strcatrealloc(description,value);
