diff -ru vdr-plugin-xmltv2vdr.orig/dist/epgdata2xmltv/Makefile vdr-plugin-xmltv2vdr/dist/epgdata2xmltv/Makefile
--- vdr-plugin-xmltv2vdr.orig/dist/epgdata2xmltv/Makefile	2017-03-20 18:08:49.000000000 +0100
+++ vdr-plugin-xmltv2vdr/dist/epgdata2xmltv/Makefile	2020-06-17 12:34:35.472188515 +0200
@@ -19,7 +19,7 @@
 DEFINES += -D__STDC_CONSTANT_MACROS -D__USE_XOPEN_EXTENDED
 
 INCLUDES += $(shell $(PKG-CONFIG) --cflags $(PKG-INCLUDES))
-LIBS     += $(shell $(PKG-CONFIG) --libs $(PKG-LIBS)) 
+LIBS     += $(shell $(PKG-CONFIG) --libs --static $(PKG-LIBS)) 
 
 INCLUDES += -I..
 
