From ebcc884a7b96ee6aaa02529ea973c652e1295cbd Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Wed, 17 Jun 2020 12:37:50 +0200
Subject: [PATCH] fix epgdata2xmltv static linking issue

---
 dist/epgdata2xmltv/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dist/epgdata2xmltv/Makefile b/dist/epgdata2xmltv/Makefile
index 3a47fde..011a008 100644
--- a/dist/epgdata2xmltv/Makefile
+++ b/dist/epgdata2xmltv/Makefile
@@ -19,7 +19,7 @@ DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 DEFINES += -D__STDC_CONSTANT_MACROS -D__USE_XOPEN_EXTENDED
 
 INCLUDES += $(shell $(PKG-CONFIG) --cflags $(PKG-INCLUDES))
-LIBS     += $(shell $(PKG-CONFIG) --libs $(PKG-LIBS)) 
+LIBS     += $(shell $(PKG-CONFIG) --libs --static $(PKG-LIBS)) 
 
 INCLUDES += -I..
 
-- 
2.51.0

