From fbb3e491b38f8fb9bcf5bb3780f9e228defafe21 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Wed, 8 Nov 2023 13:47:25 +0000
Subject: [PATCH] add required include for vdr-plugin-xmltv2vdr 0.2.2

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 7d0536f..c02a9f5 100644
--- a/Makefile
+++ b/Makefile
@@ -53,6 +53,8 @@ SOFILE = libvdr-$(PLUGIN).so
 PKG-LIBS += libxml-2.0 sqlite3
 PKG-INCLUDES += libxml-2.0 sqlite3
 
+INCLUDES += -I$(VDRDIR)/include
+
 DEFINES += -D_GNU_SOURCE -D_XOPEN_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
 CXXFLAGS += $(shell $(PKG_CONFIG) --cflags $(PKG-INCLUDES)) -Wextra
-- 
2.51.0

