diff --git a/gfx/widgets/gfx_widget_achievement_popup.c b/gfx/widgets/gfx_widget_achievement_popup.c
index ba583a8..c9ef547 100644
--- a/gfx/widgets/gfx_widget_achievement_popup.c
+++ b/gfx/widgets/gfx_widget_achievement_popup.c
@@ -320,13 +320,13 @@ static void gfx_widget_achievement_popup_frame(void* data, void* userdata)
 
       /* Title */
       gfx_widgets_draw_text(
-         &p_dispwidget->gfx_widget_fonts.regular,
+         &p_dispwidget->gfx_widget_fonts.msg_queue,
          state->queue[state->queue_read_index].title,
          screen_pos_x + state->height - text_unfold_offset
          + p_dispwidget->simple_widget_padding,
          screen_pos_y
-         + p_dispwidget->gfx_widget_fonts.regular.line_height
-         + p_dispwidget->gfx_widget_fonts.regular.line_ascender,
+         + p_dispwidget->gfx_widget_fonts.msg_queue.line_height
+         + p_dispwidget->gfx_widget_fonts.msg_queue.line_ascender,
          video_width,
          video_height,
          TEXT_COLOR_FAINT,
@@ -337,13 +337,13 @@ static void gfx_widget_achievement_popup_frame(void* data, void* userdata)
 
       /* TODO: is a ticker necessary ? */
       gfx_widgets_draw_text(
-         &p_dispwidget->gfx_widget_fonts.regular,
+         &p_dispwidget->gfx_widget_fonts.msg_queue,
          state->queue[state->queue_read_index].subtitle,
          screen_pos_x + state->height - text_unfold_offset
          + p_dispwidget->simple_widget_padding,
          screen_pos_y + state->height
-         - p_dispwidget->gfx_widget_fonts.regular.line_height
-         - p_dispwidget->gfx_widget_fonts.regular.line_descender,
+         - p_dispwidget->gfx_widget_fonts.msg_queue.line_height
+         - p_dispwidget->gfx_widget_fonts.msg_queue.line_descender,
          video_width,
          video_height,
          TEXT_COLOR_INFO,
@@ -353,7 +353,7 @@ static void gfx_widget_achievement_popup_frame(void* data, void* userdata)
       if (is_folding)
       {
          gfx_widgets_flush_text(video_width, video_height,
-            &p_dispwidget->gfx_widget_fonts.regular);
+            &p_dispwidget->gfx_widget_fonts.msg_queue);
 
          if (dispctx && dispctx->scissor_end)
             dispctx->scissor_end(video_info->userdata,
@@ -551,13 +551,13 @@ static void gfx_widget_achievement_popup_start(
    const dispgfx_widget_t* p_dispwidget = (const dispgfx_widget_t*)
       state->dispwidget_ptr;
 
-   state->height = p_dispwidget->gfx_widget_fonts.regular.line_height * 4.0f;
+   state->height = p_dispwidget->gfx_widget_fonts.msg_queue.line_height * 4.0f;
    state->width = MAX(
       font_driver_get_message_width(
-         p_dispwidget->gfx_widget_fonts.regular.font,
+         p_dispwidget->gfx_widget_fonts.msg_queue.font,
          state->queue[state->queue_read_index].title, 0, 1.0f),
       font_driver_get_message_width(
-         p_dispwidget->gfx_widget_fonts.regular.font,
+         p_dispwidget->gfx_widget_fonts.msg_queue.font,
          state->queue[state->queue_read_index].subtitle, 0, 1.0f)
    );
    state->width += p_dispwidget->simple_widget_padding * 2;
