diff --git a/gnu/stdio.in.h b/gnu/stdio.in.h
index 88b368b..44f1087 100644
--- a/gnu/stdio.in.h
+++ b/gnu/stdio.in.h
@@ -139,7 +139,9 @@ _GL_WARN_ON_USE (fflush, "fflush is not always POSIX compliant - "
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 #undef gets
+#if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
+#endif
 
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
