diff -Naur glibc-2.27/aclocal.m4 glibc-2.27-2/aclocal.m4
--- glibc-2.27/aclocal.m4	2022-04-20 02:23:50.549365949 +0200
+++ glibc-2.27-2/aclocal.m4	2022-04-20 02:26:23.179843270 +0200
@@ -2,7 +2,7 @@
 dnl the internal functions defined and used by the main configure script
 dnl match those expected by the fragments.  When changing this version,
 dnl install.texi also needs to be updated.
-m4_define([GLIBC_AUTOCONF_VERSION], [2.69])
+m4_define([GLIBC_AUTOCONF_VERSION], [2.71])
 m4_if(m4_defn([AC_AUTOCONF_VERSION]), GLIBC_AUTOCONF_VERSION, [],
       [m4_fatal(m4_flatten(
 Exactly version GLIBC_AUTOCONF_VERSION of Autoconf is required but you have
