Index: Linux-PAM-1.0.2/Makefile.in
===================================================================
--- Linux-PAM-1.0.2.orig/Makefile.in	2008-08-29 01:14:12.000000000 -0700
+++ Linux-PAM-1.0.2/Makefile.in	2009-09-04 01:06:22.571466501 -0700
@@ -71,8 +71,7 @@
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = libpam tests libpamc libpam_misc modules po conf doc \
-	examples xtests
+DIST_SUBDIRS = libpam libpamc libpam_misc modules po conf
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -244,8 +243,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.9 gnu dist-bzip2 check-news
-@STATIC_MODULES_FALSE@SUBDIRS = libpam tests libpamc libpam_misc modules po conf doc examples xtests
-@STATIC_MODULES_TRUE@SUBDIRS = modules libpam libpamc libpam_misc tests po conf doc examples xtests
+@STATIC_MODULES_FALSE@SUBDIRS = libpam libpamc libpam_misc modules po conf
+@STATIC_MODULES_TRUE@SUBDIRS = modules libpam libpamc libpam_misc po conf
 CLEANFILES = *~
 M4_FILES = m4/codeset.m4 m4/gettext.m4 m4/glibc21.m4 m4/glibc2.m4 \
 	m4/iconv.m4 m4/intdiv0.m4 m4/intmax.m4 m4/inttypes_h.m4 \
Index: Linux-PAM-1.0.2/modules/pam_access/pam_access.c
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/Makefile.am
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/Makefile.in
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/pam_unix_passwd.c
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/support.c
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/yppasswd.h
===================================================================
Index: Linux-PAM-1.0.2/modules/pam_unix/yppasswd_xdr.c
===================================================================
Index: Linux-PAM-1.0.2/po/POTFILES.in
===================================================================
