commit 59c7f0369a67e1aecf365c92186bc5b45e0efcba
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Mon Oct 14 22:03:43 2013 +0300

    fix rpath

diff --git a/Makefile.rules b/Makefile.rules
index f0c542b..fc5fe64 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -639,11 +639,10 @@ endif
 ifneq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
   ifneq ($(HOST_OS), Darwin)
     ifdef TOOLNAME
-      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
       ifdef EXAMPLE_TOOL
-        LD.Flags += $(RPATH) -Wl,$(ExmplDir) $(DynamicFlag)
+        LD.Flags += $(DynamicFlag)
       else
-        LD.Flags += $(RPATH) -Wl,$(ToolDir) $(DynamicFlag)
+        LD.Flags += $(DynamicFlag)
     endif
   endif
 else
