diff -Naur llvm-3.1.src/autoconf/configure.ac llvm-3.1.src.patch/autoconf/configure.ac
--- llvm-3.1.src/autoconf/configure.ac	2012-05-11 22:48:57.000000000 +0200
+++ llvm-3.1.src.patch/autoconf/configure.ac	2012-11-09 21:39:16.897271776 +0100
@@ -402,6 +402,12 @@
        AC_CHECK_PROG(BUILD_CXX, c++, c++, , , /usr/ucb/c++)
      fi
   fi
+  AC_SUBST(BUILD_CPPFLAGS)
+  AC_SUBST(BUILD_CFLAGS)
+  AC_SUBST(BUILD_CXXFLAGS)
+  AC_SUBST(BUILD_LDFLAGS)
+  AC_SUBST(BUILD_AR)
+  AC_SUBST(BUILD_RANLIB)
 else
   AC_SUBST(LLVM_CROSS_COMPILING, [0])
 fi
diff -Naur llvm-3.1.src/autoconf/m4/build_exeext.m4 llvm-3.1.src.patch/autoconf/m4/build_exeext.m4
--- llvm-3.1.src/autoconf/m4/build_exeext.m4	2008-09-26 19:27:58.000000000 +0200
+++ llvm-3.1.src.patch/autoconf/m4/build_exeext.m4	2012-11-09 15:41:36.583964079 +0100
@@ -18,7 +18,7 @@
      fi
   fi
   test -z "$BUILD_CC" && AC_MSG_ERROR([no acceptable cc found in \$PATH])
-  ac_build_link='${BUILD_CC-cc} -o conftest $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&AS_MESSAGE_LOG_FD'
+  ac_build_link='${BUILD_CC-cc} -o conftest $BUILD_CFLAGS $BUILD_CPPFLAGS $BUILD_LDFLAGS conftest.$ac_ext $LIBS 1>&AS_MESSAGE_LOG_FD'
   rm -f conftest*
   echo 'int main () { return 0; }' > conftest.$ac_ext
   ac_cv_build_exeext=
diff -Naur llvm-3.1.src/Makefile llvm-3.1.src.patch/Makefile
--- llvm-3.1.src/Makefile	2012-01-17 03:56:49.000000000 +0100
+++ llvm-3.1.src.patch/Makefile	2012-11-09 21:56:28.931837126 +0100
@@ -109,11 +109,14 @@
 	$(Verb) if [ ! -f BuildTools/Makefile ]; then \
           $(MKDIR) BuildTools; \
 	  cd BuildTools ; \
-	  unset CFLAGS ; \
-	  unset CXXFLAGS ; \
-	  $(PROJ_SRC_DIR)/configure --build=$(BUILD_TRIPLE) \
-		--host=$(BUILD_TRIPLE) --target=$(BUILD_TRIPLE) \
-	        --disable-polly ; \
+	  CC="$(BUILD_CC)" \
+	  CXX="$(BUILD_CXX)" \
+	  CFLAGS="$(BUILD_CFLAGS)" \
+	  CXXFLAGS="$(BUILD_CXXFLAGS)" \
+	  LDFLAGS="$(BUILD_LDFLAGS)" \
+	  AR="$(BUILD_AR)" \
+	  RANLIB="$(BUILD_RANLIB)" \
+	  $(PROJ_SRC_DIR)/configure --disable-polly ; \
 	  cd .. ; \
 	fi; \
 	(unset SDKROOT; \
@@ -128,8 +131,13 @@
 	  DISABLE_ASSERTIONS=$(DISABLE_ASSERTIONS) \
 	  ENABLE_EXPENSIVE_CHECKS=$(ENABLE_EXPENSIVE_CHECKS) \
 	  ENABLE_LIBCPP=$(ENABLE_LIBCPP) \
-	  CFLAGS= \
-	  CXXFLAGS= \
+	  CC="$(BUILD_CC)" \
+	  CXX="$(BUILD_CXX)" \
+	  CFLAGS="$(BUILD_CFLAGS)" \
+	  CXXFLAGS="$(BUILD_CXXFLAGS)" \
+	  LDFLAGS="$(BUILD_LDFLAGS)" \
+	  AR="$(BUILD_AR)" \
+	  RANLIB="$(BUILD_RANLIB)" \
 	) || exit 1;
 endif
 
diff -Naur llvm-3.1.src/Makefile.config.in llvm-3.1.src.patch/Makefile.config.in
--- llvm-3.1.src/Makefile.config.in	2012-05-16 00:06:08.000000000 +0200
+++ llvm-3.1.src.patch/Makefile.config.in	2012-11-09 21:43:39.805643703 +0100
@@ -127,6 +127,12 @@
 # call if we're cross-compiling).
 BUILD_CC=@BUILD_CC@
 BUILD_CXX=@BUILD_CXX@
+BUILD_CPPFLAGS=@BUILD_CPPFLAGS@
+BUILD_CFLAGS=@BUILD_CFLAGS@
+BUILD_CXXFLAGS=@BUILD_CXXFLAGS@
+BUILD_LDFLAGS=@BUILD_LDFLAGS@
+BUILD_AR=@BUILD_AR@
+BUILD_RANLIB=@BUILD_RANLIB@
 
 # Triple for configuring build tools when cross-compiling
 BUILD_TRIPLE=@build@
diff -Naur llvm-3.1.src/projects/sample/autoconf/configure.ac llvm-3.1.src.patch/projects/sample/autoconf/configure.ac
--- llvm-3.1.src/projects/sample/autoconf/configure.ac	2012-04-05 21:34:15.000000000 +0200
+++ llvm-3.1.src.patch/projects/sample/autoconf/configure.ac	2012-11-09 21:40:36.105588918 +0100
@@ -342,6 +342,12 @@
        AC_CHECK_PROG(BUILD_CXX, c++, c++, , , /usr/ucb/c++)
      fi
   fi
+  AC_SUBST(BUILD_CPPFLAGS)
+  AC_SUBST(BUILD_CFLAGS)
+  AC_SUBST(BUILD_CXXFLAGS)
+  AC_SUBST(BUILD_LDFLAGS)
+  AC_SUBST(BUILD_AR)
+  AC_SUBST(BUILD_RANLIB)
 else
   AC_SUBST(LLVM_CROSS_COMPILING, [0])
 fi
diff -Naur llvm-3.1.src/projects/sample/autoconf/m4/build_exeext.m4 llvm-3.1.src.patch/projects/sample/autoconf/m4/build_exeext.m4
--- llvm-3.1.src/projects/sample/autoconf/m4/build_exeext.m4	2011-10-19 01:10:47.000000000 +0200
+++ llvm-3.1.src.patch/projects/sample/autoconf/m4/build_exeext.m4	2012-11-09 15:52:56.307603861 +0100
@@ -18,7 +18,7 @@
      fi
   fi
   test -z "$BUILD_CC" && AC_MSG_ERROR([no acceptable cc found in \$PATH])
-  ac_build_link='${BUILD_CC-cc} -o conftest $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&AS_MESSAGE_LOG_FD'
+  ac_build_link='${BUILD_CC-cc} -o conftest $BUILD_CFLAGS $BUILD_CPPFLAGS $BUILD_LDFLAGS conftest.$ac_ext $LIBS 1>&AS_MESSAGE_LOG_FD'
   rm -f conftest*
   echo 'int main () { return 0; }' > conftest.$ac_ext
   ac_cv_build_exeext=
diff -Naur llvm-3.1.src/projects/sample/Makefile.llvm.config.in llvm-3.1.src.patch/projects/sample/Makefile.llvm.config.in
--- llvm-3.1.src/projects/sample/Makefile.llvm.config.in	2012-03-26 04:09:01.000000000 +0200
+++ llvm-3.1.src.patch/projects/sample/Makefile.llvm.config.in	2012-11-09 21:43:00.000981773 +0100
@@ -95,6 +95,12 @@
 # call if we're cross-compiling).
 BUILD_CC=@BUILD_CC@
 BUILD_CXX=@BUILD_CXX@
+BUILD_CPPFLAGS=@BUILD_CPPFLAGS@
+BUILD_CFLAGS=@BUILD_CFLAGS@
+BUILD_CXXFLAGS=@BUILD_CXXFLAGS@
+BUILD_LDFLAGS=@BUILD_LDFLAGS@
+BUILD_AR=@BUILD_AR@
+BUILD_RANLIB=@BUILD_RANLIB@
 
 # Triple for configuring build tools when cross-compiling
 BUILD_TRIPLE=@build@
