diff -Naur a/scripts/sc-desktop.py b/scripts/sc-desktop.py
--- a/scripts/sc-desktop.py	2016-11-18 13:31:29.522462062 -0800
+++ b/scripts/sc-desktop.py	2016-11-18 13:33:09.268432661 -0800
@@ -80,7 +80,7 @@
         parser = argparse.ArgumentParser(description=__doc__)
         parser.add_argument('command', type=str, choices=['start', 'stop', 'restart', 'debug'])
         args = parser.parse_args()
-        daemon = SCDaemon('/tmp/steamcontroller.pid')
+        daemon = SCDaemon('/run/steamcontroller.pid')
 
         if 'start' == args.command:
             daemon.start()
diff -Naur a/scripts/sc-xbox.py b/scripts/sc-xbox.py
--- a/scripts/sc-xbox.py	2016-11-18 13:31:29.522462062 -0800
+++ b/scripts/sc-xbox.py	2016-11-18 13:33:09.266432642 -0800
@@ -79,7 +79,7 @@
         parser = argparse.ArgumentParser(description=__doc__)
         parser.add_argument('command', type=str, choices=['start', 'stop', 'restart', 'debug'])
         args = parser.parse_args()
-        daemon = SCDaemon('/tmp/steamcontroller.pid')
+        daemon = SCDaemon('/run/steamcontroller.pid')
 
         if 'start' == args.command:
             daemon.start()
