commit 1467b7cd9d153be6680cc818b10c66967a2f0951
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sat Jan 7 14:11:25 2017 -0800

    steam client will never be running

diff --git a/src/daemon.py b/src/daemon.py
index 964c765..fba7792 100644
--- a/src/daemon.py
+++ b/src/daemon.py
@@ -9,7 +9,6 @@ import time
 import atexit
 import signal
 import syslog
-import psutil
 import traceback
 import gc
 
@@ -92,16 +91,12 @@ class Daemon(object):
         self.daemonize()
         syslog.syslog(syslog.LOG_INFO, '{}: started'.format(os.path.basename(sys.argv[0])))
         while True:
-            # look if steam is running
-            if len([p for p in psutil.process_iter() if p.name() == 'steam']) == 0:
-                try:
-                    self.run()
-                except Exception as e: # pylint: disable=W0703
-                    syslog.syslog(syslog.LOG_ERR, '{}: {!s}'.format(os.path.basename(sys.argv[0]), e))
-                    syslog.syslog(syslog.LOG_ERR, traceback.format_exc())
-                    gc.collect()
-            else:
-                syslog.syslog(syslog.LOG_INFO, '{}: steam client is runing'.format(os.path.basename(sys.argv[0])))
+            try:
+                self.run()
+            except Exception as e: # pylint: disable=W0703
+                syslog.syslog(syslog.LOG_ERR, '{}: {!s}'.format(os.path.basename(sys.argv[0]), e))
+                syslog.syslog(syslog.LOG_ERR, traceback.format_exc())
+                gc.collect()
             time.sleep(2)
 
     def stop(self):
