diff -Naur a/src/__init__.py b/src/__init__.py
--- a/src/__init__.py	2016-09-14 13:00:35.000000000 -0700
+++ b/src/__init__.py	2016-11-17 23:40:57.137095353 -0800
@@ -116,18 +116,16 @@
         self._cmsg = []
         self._ctx = usb1.USBContext()
 
+        while self._handle is None:
+            for i in range(len(PRODUCT_ID)):
+                pid = PRODUCT_ID[i]
+                endpoint = ENDPOINT[i]
+                ccidx = CONTROLIDX[i]
 
-        for i in range(len(PRODUCT_ID)):
-            pid = PRODUCT_ID[i]
-            endpoint = ENDPOINT[i]
-            ccidx = CONTROLIDX[i]
-
-            self._handle = self._ctx.openByVendorIDAndProductID(
-                VENDOR_ID, pid,
-                skip_on_error=True,
-            )
-            if self._handle is not None:
-                break
+                self._handle = self._ctx.openByVendorIDAndProductID(
+                    VENDOR_ID, pid,
+                    skip_on_error=True,
+                )
 
         if self._handle is None:
             raise ValueError('SteamControler Device not found')
