diff -Naur a/scripts/sc-desktop.py b/scripts/sc-desktop.py
--- a/scripts/sc-desktop.py	2016-01-02 23:13:18.000000000 +0100
+++ b/scripts/sc-desktop.py	2016-01-04 22:00:18.536497613 +0100
@@ -22,6 +22,10 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 """Steam Controller Mouse, Keyboard mode"""
 
 from steamcontroller import SteamController, SCButtons
diff -Naur a/scripts/sc-xbox.py b/scripts/sc-xbox.py
--- a/scripts/sc-xbox.py	2016-01-02 23:13:18.000000000 +0100
+++ b/scripts/sc-xbox.py	2016-01-04 22:00:30.751519105 +0100
@@ -22,6 +22,10 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 """Steam Controller XBOX360 Gamepad Emulator"""
 
 from steamcontroller import \
