commit 8da1123bcbf3ad12b6c9d7975ebbb3f8663fd166
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sat Jan 7 13:50:30 2017 -0800

    fix include dir

diff --git a/src/uinput.py b/src/uinput.py
index 684a018..ffc2354 100644
--- a/src/uinput.py
+++ b/src/uinput.py
@@ -36,10 +36,10 @@ from steamcontroller.tools import get_so_extensions
 from collections import deque
 
 # Get All defines from linux headers
-if os.path.exists('/usr/include/linux/input-event-codes.h'):
-    CHEAD = defines('/usr/include', 'linux/input-event-codes.h')
+if os.path.exists('/storage/.kodi/addons/driver.steamcontroller/include/linux/input-event-codes.h'):
+    CHEAD = defines('/storage/.kodi/addons/driver.steamcontroller/include', 'linux/input-event-codes.h')
 else:
-    CHEAD = defines('/usr/include', 'linux/input.h')
+    CHEAD = defines('/storage/.kodi/addons/driver.steamcontroller/include', 'linux/input.h')
 
 # Keys enum contains all keys and button from linux/uinput.h (KEY_* BTN_*)
 Keys = IntEnum('Keys', {i: CHEAD[i] for i in CHEAD.keys() if (i.startswith('KEY_') or
