commit 5947f69820db3ce2d976ac533fb64319637d3f93
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sat Jan 7 13:53:36 2017 -0800

    use run for pid directory

diff --git a/scripts/sc-desktop.py b/scripts/sc-desktop.py
index e0b8ec7..ad494d8 100755
--- a/scripts/sc-desktop.py
+++ b/scripts/sc-desktop.py
@@ -87,9 +87,9 @@ if __name__ == '__main__':
         parser.add_argument('-i', '--index', type=int, choices=[0,1,2,3], default=None)
         args = parser.parse_args()
         if args.index != None:
-            daemon = SCDaemon('/tmp/steamcontroller{:d}.pid'.format(args.index))
+            daemon = SCDaemon('/run/steamcontroller{:d}.pid'.format(args.index))
         else:
-            daemon = SCDaemon('/tmp/steamcontroller.pid')
+            daemon = SCDaemon('/run/steamcontroller.pid')
 
         if 'start' == args.command:
             daemon.start()
diff --git a/scripts/sc-mixed.py b/scripts/sc-mixed.py
index 5e25af9..c9139e7 100755
--- a/scripts/sc-mixed.py
+++ b/scripts/sc-mixed.py
@@ -126,9 +126,9 @@ if __name__ == '__main__':
         parser.add_argument('-i', '--index', type=int, choices=[0,1,2,3], default=None)
         args = parser.parse_args()
         if args.index != None:
-            daemon = SCDaemon('/tmp/steamcontroller{:d}.pid'.format(args.index))
+            daemon = SCDaemon('/run/steamcontroller{:d}.pid'.format(args.index))
         else:
-            daemon = SCDaemon('/tmp/steamcontroller.pid')
+            daemon = SCDaemon('/run/steamcontroller.pid')
 
         if 'start' == args.command:
             daemon.start()
diff --git a/scripts/sc-xbox.py b/scripts/sc-xbox.py
index 89d4f55..0c61b4e 100755
--- a/scripts/sc-xbox.py
+++ b/scripts/sc-xbox.py
@@ -86,9 +86,9 @@ if __name__ == '__main__':
         parser.add_argument('-i', '--index', type=int, choices=[0,1,2,3], default=None)
         args = parser.parse_args()
         if args.index != None:
-            daemon = SCDaemon('/tmp/steamcontroller{:d}.pid'.format(args.index))
+            daemon = SCDaemon('/run/steamcontroller{:d}.pid'.format(args.index))
         else:
-            daemon = SCDaemon('/tmp/steamcontroller.pid')
+            daemon = SCDaemon('/run/steamcontroller.pid')
 
         if 'start' == args.command:
             daemon.start()
