commit b1bcba9b73d1e5e755ef6cda014ea3cc5a5ba270
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sat Jan 7 13:51:01 2017 -0800

    append python path

diff --git a/scripts/sc-callbacks.py b/scripts/sc-callbacks.py
index eaeb658..35a272a 100755
--- a/scripts/sc-callbacks.py
+++ b/scripts/sc-callbacks.py
@@ -25,6 +25,8 @@
 """Steam Controller Callback Mode example"""
 import sys
 
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 from steamcontroller import SteamController, SCButtons
 from steamcontroller.events import EventMapper, Pos
 from steamcontroller.uinput import Keys
diff --git a/scripts/sc-desktop.py b/scripts/sc-desktop.py
index 9eff430..e0b8ec7 100755
--- a/scripts/sc-desktop.py
+++ b/scripts/sc-desktop.py
@@ -22,6 +22,10 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 """Steam Controller Mouse, Keyboard mode"""
 
 from steamcontroller import SteamController, SCButtons
diff --git a/scripts/sc-dump.py b/scripts/sc-dump.py
index b6f2c24..b582771 100755
--- a/scripts/sc-dump.py
+++ b/scripts/sc-dump.py
@@ -25,6 +25,9 @@
 """Steam Controller USB Dumper"""
 
 import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 from steamcontroller import SteamController
 
 def dump(_, sci):
diff --git a/scripts/sc-mixed.py b/scripts/sc-mixed.py
index 4a9c3c0..5e25af9 100755
--- a/scripts/sc-mixed.py
+++ b/scripts/sc-mixed.py
@@ -22,6 +22,10 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 """Steam Controller XBOX360 Gamepad Emulator"""
 
 from steamcontroller import \
diff --git a/scripts/sc-test-cmsg.py b/scripts/sc-test-cmsg.py
index 08c8481..a59fbe9 100755
--- a/scripts/sc-test-cmsg.py
+++ b/scripts/sc-test-cmsg.py
@@ -26,6 +26,9 @@
 
 import sys
 import struct
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 from steamcontroller import SteamController
 
 def dump(_, sci):
diff --git a/scripts/sc-xbox.py b/scripts/sc-xbox.py
index 814c8bc..89d4f55 100755
--- a/scripts/sc-xbox.py
+++ b/scripts/sc-xbox.py
@@ -22,6 +22,10 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+
+sys.path.append('/storage/.kodi/addons/driver.steamcontroller/lib')
+
 """Steam Controller XBOX360 Gamepad Emulator"""
 
 from steamcontroller import \
