diff --git a/configure.ac b/configure.ac
index 7c053d2..dcb0c4d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -314,7 +314,7 @@ AC_MSG_RESULT($enable_libftdi)
 
 if test "$enable_libftdi" = "yes"; then
 	ifdef([PKG_CHECK_MODULES],
-	 	[PKG_CHECK_MODULES(LIBFTDI, libftdi >= 0.8,
+	 	[PKG_CHECK_MODULES(LIBFTDI, libftdi1 >= 0.8,
 			[AC_DEFINE(HAVE_LIBFTDI, [1], [Define to 1 if you have libftdi])],
 			[ enable_libftdi=no ])],
 		[AC_MSG_WARN([pkg-config not (fully) installed; drivers requiring libftdi may not be built])])
diff --git a/server/drivers/hd44780-low.h b/server/drivers/hd44780-low.h
index 47acf45..6faa830 100644
--- a/server/drivers/hd44780-low.h
+++ b/server/drivers/hd44780-low.h
@@ -26,7 +26,7 @@
 #endif
 
 #ifdef HAVE_LIBFTDI
-# include <ftdi.h>
+# include <libftdi1/ftdi.h>
 #endif
 
 /** \name Symbolic names for connection types
diff --git a/server/drivers/i2500vfd.c b/server/drivers/i2500vfd.c
index d896a69..7fe74f2 100644
--- a/server/drivers/i2500vfd.c
+++ b/server/drivers/i2500vfd.c
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "i2500vfd.h"
diff --git a/server/drivers/lis.c b/server/drivers/lis.c
index 5d3e97a..6c26a19 100644
--- a/server/drivers/lis.c
+++ b/server/drivers/lis.c
@@ -42,7 +42,7 @@
 #include <pthread.h>
 
 #include <usb.h>
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "lis.h"
diff --git a/server/drivers/ula200.c b/server/drivers/ula200.c
index cbdde40..a84eb49 100644
--- a/server/drivers/ula200.c
+++ b/server/drivers/ula200.c
@@ -31,7 +31,7 @@
 #include <errno.h>
 
 #include <usb.h>
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "ula200.h"
