diff --git a/server/drivers/hd44780-low.h b/server/drivers/hd44780-low.h
index 47acf45..6faa830 100644
--- a/server/drivers/hd44780-low.h
+++ b/server/drivers/hd44780-low.h
@@ -26,7 +26,7 @@
 #endif
 
 #ifdef HAVE_LIBFTDI
-# include <ftdi.h>
+# include <libftdi1/ftdi.h>
 #endif
 
 /** \name Symbolic names for connection types
diff --git a/server/drivers/i2500vfd.c b/server/drivers/i2500vfd.c
index d896a69..7fe74f2 100644
--- a/server/drivers/i2500vfd.c
+++ b/server/drivers/i2500vfd.c
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "i2500vfd.h"
diff --git a/server/drivers/lis.c b/server/drivers/lis.c
index 5d3e97a..6c26a19 100644
--- a/server/drivers/lis.c
+++ b/server/drivers/lis.c
@@ -42,7 +42,7 @@
 #include <errno.h>
 #include <pthread.h>
 
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "lis.h"
diff --git a/server/drivers/ula200.c b/server/drivers/ula200.c
index cbdde40..a84eb49 100644
--- a/server/drivers/ula200.c
+++ b/server/drivers/ula200.c
@@ -31,7 +31,7 @@
 #include <string.h>
 #include <errno.h>
 
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 #include "lcd.h"
 #include "ula200.h"
