From 689d136c2806188d4a8bdaf9b5a7a57c2d889050 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Fri, 21 Feb 2020 03:02:31 +0000
Subject: [PATCH] meson build: fails to build stlinkv3 when usb is disabled

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 375089c30..3c5296c5f 100644
--- a/meson.build
+++ b/meson.build
@@ -91,6 +91,7 @@ else
   config_digilent_spi = false
   config_developerbox_spi = false
   config_pickit2_spi = false
+  config_stlinkv3_spi = false
 endif
 
 # some programmers require libpci
