diff --git a/Makefile b/Makefile
index 6b70ab8..f602b45 100644
--- a/Makefile
+++ b/Makefile
@@ -436,7 +436,7 @@ NEED_PCI := yes
 endif
 
 ifeq ($(CONFIG_FT2232_SPI), yes)
-FTDILIBS := $(shell pkg-config --libs libftdi 2>/dev/null || printf "%s" "-lftdi -lusb")
+FTDILIBS := $(shell pkg-config --libs libftdi1 2>/dev/null || printf "%s" "-lftdi1 -lusb")
 # This is a totally ugly hack.
 FEATURE_CFLAGS += $(shell LC_ALL=C grep -q "FTDISUPPORT := yes" .features && printf "%s" "-D'CONFIG_FT2232_SPI=1'")
 FEATURE_LIBS += $(shell LC_ALL=C grep -q "FTDISUPPORT := yes" .features && printf "%s" "$(FTDILIBS)")
@@ -666,7 +666,7 @@ featuresavailable:
 endif
 
 define FTDI_TEST
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 struct ftdi_context *ftdic = NULL;
 int main(int argc, char **argv)
 {
diff --git a/ft2232_spi.c b/ft2232_spi.c
index a5d3828..d54cf6a 100644
--- a/ft2232_spi.c
+++ b/ft2232_spi.c
@@ -27,7 +27,7 @@
 #include "flash.h"
 #include "programmer.h"
 #include "spi.h"
-#include <ftdi.h>
+#include <libftdi1/ftdi.h>
 
 /* Please keep sorted by vendor ID, then device ID. */
 
