diff -Naur flashrom/Makefile flashrom.patch/Makefile
--- flashrom/Makefile	2009-04-14 03:43:10.000000000 +0200
+++ flashrom.patch/Makefile	2009-04-17 00:00:43.238159402 +0200
@@ -40,8 +40,7 @@
 
 # Set the flashrom version string from the highest revision number
 # of the checked out flashrom files.
-SVNDEF := -D'FLASHROM_VERSION="$(shell svnversion -cn . \
-          | sed -e "s/.*://" -e "s/\([0-9]*\).*/\1/")"'
+ SVNDEF := -D'FLASHROM_VERSION="$(shell cat version)"'
 
 $(PROGRAM): $(OBJS)
 	$(CC) -o $(PROGRAM) $(OBJS) $(LDFLAGS)
diff -Naur flashrom/version flashrom.patch/version
--- flashrom/version	1970-01-01 01:00:00.000000000 +0100
+++ flashrom.patch/version	2009-04-16 23:59:52.958410419 +0200
@@ -0,0 +1 @@
+4107
