--- a/gfx/common/vulkan_common.c	2019-05-06 19:20:30.904330830 +0200
+++ b/gfx/common/vulkan_common.c	2019-05-06 19:20:22.648359000 +0200
@@ -2068,9 +2068,9 @@
       res = vkCreateInstance(&info, NULL, &vk->context.instance);
    }
 
-   if (res == VK_ERROR_INCOMPATIBLE_DRIVER)
+   if (res != VK_SUCCESS)
    {
-      RARCH_ERR("Failed to create Vulkan instance.\n");
+      RARCH_ERR("Failed to create Vulkan instance (%d).\n", res);
       return false;
    }
 