diff -Naur retroarch.git/libretro-common/include/glsym/glsym_es2.h retroarch.patch/libretro-common/include/glsym/glsym_es2.h
--- retroarch.git/libretro-common/include/glsym/glsym_es2.h	2016-12-04 00:48:23.294181322 +0100
+++ retroarch.patch/libretro-common/include/glsym/glsym_es2.h	2016-12-04 00:50:17.086161391 +0100
@@ -23,7 +23,7 @@
 typedef GLint GLfixed;
 #endif
 
-#if (__STDC_VERSION__ <= 199409L) || (OSX && !MAC_OS_X_VERSION_10_7)
+#if (__STDC_VERSION__ <= 199901L) || (OSX && !MAC_OS_X_VERSION_10_7)
 #ifndef GLint64
 typedef long long int GLint64;
 #endif
