diff --git a/menu/drivers/xmb.c b/menu/drivers/xmb.c
index ce2c906..93afff6 100644
--- a/menu/drivers/xmb.c
+++ b/menu/drivers/xmb.c
@@ -713,6 +713,7 @@ static void xmb_render_keyboard(xmb_handle_t *xmb, const char *grid[], unsigned
 
       uintptr_t texture = xmb->textures.list[XMB_TEXTURE_KEY];
       if (i == id)
+      {
          texture = xmb->textures.list[XMB_TEXTURE_KEY_HOVER];
 
       menu_display_blend_begin();
@@ -726,6 +727,7 @@ static void xmb_render_keyboard(xmb_handle_t *xmb, const char *grid[], unsigned
             texture);
 
       menu_display_blend_end();
+      }
 
       menu_display_draw_text(xmb->font, grid[i],
             width/2.0 - (11*ptr_width)/2.0 + (i % 11) * ptr_width + ptr_width/2.0,
