diff --git a/retroarch.c b/retroarch.c
index b82cc1c540..22f3fe720c 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -35465,6 +35465,9 @@ bool retroarch_main_init(int argc, char *argv[])
    menu_driver_find_driver(p_rarch, settings,
          "menu driver", verbosity_enabled);
 #endif
+   /* Enforce stored brightness if needed */
+   if (frontend_driver_can_set_screen_brightness())
+      frontend_driver_set_screen_brightness(settings->uints.screen_brightness);
 
    /* Attempt to initialize core */
    if (p_rarch->has_set_core)
