diff --git a/menu/menu_displaylist.c b/menu/menu_displaylist.c
index 00b7995..f7a8b6e 100644
--- a/menu/menu_displaylist.c
+++ b/menu/menu_displaylist.c
@@ -2679,12 +2679,14 @@ static int menu_displaylist_parse_load_content_settings(
             MENU_ENUM_LABEL_LOAD_STATE,
             MENU_SETTING_ACTION_LOADSTATE, 0, 0);
 
+      if (settings->menu.show_advanced_settings)
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UNDO_LOAD_STATE),
             msg_hash_to_str(MENU_ENUM_LABEL_UNDO_LOAD_STATE),
             MENU_ENUM_LABEL_UNDO_LOAD_STATE,
             MENU_SETTING_ACTION_LOADSTATE, 0, 0);
 
+      if (settings->menu.show_advanced_settings)
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UNDO_SAVE_STATE),
             msg_hash_to_str(MENU_ENUM_LABEL_UNDO_SAVE_STATE),
@@ -2725,12 +2727,14 @@ static int menu_displaylist_parse_load_content_settings(
             MENU_SETTING_ACTION, 0, 0);
 #endif
 
+      if (settings->menu.show_advanced_settings)
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_SAVE_CURRENT_CONFIG_OVERRIDE_CORE),
             msg_hash_to_str(MENU_ENUM_LABEL_SAVE_CURRENT_CONFIG_OVERRIDE_CORE),
             MENU_ENUM_LABEL_SAVE_CURRENT_CONFIG_OVERRIDE_CORE,
             MENU_SETTING_ACTION, 0, 0);
 
+      if (settings->menu.show_advanced_settings)
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_SAVE_CURRENT_CONFIG_OVERRIDE_GAME),
             msg_hash_to_str(MENU_ENUM_LABEL_SAVE_CURRENT_CONFIG_OVERRIDE_GAME),
