diff --git a/configuration.h b/configuration.h
index 002318b724..f7c8b25667 100644
--- a/configuration.h
+++ b/configuration.h
@@ -334,6 +334,7 @@ typedef struct settings
       unsigned cpu_min_freq;
       unsigned cpu_max_freq;
 #endif
+
    } uints;
 
    struct
diff --git a/menu/cbs/menu_cbs_left.c b/menu/cbs/menu_cbs_left.c
index 35e7faedb4..eb7c62afe1 100644
--- a/menu/cbs/menu_cbs_left.c
+++ b/menu/cbs/menu_cbs_left.c
@@ -676,6 +676,7 @@ static int manual_content_scan_core_name_left(unsigned type, const char *label,
    return 0;
 }
 
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
 static int cpu_policy_mode_change(unsigned type, const char *label,
       bool wraparound)
@@ -790,6 +791,7 @@ static int cpu_policy_freq_tweak(unsigned type, const char *label,
    return 0;
 }
 #endif
+#endif
 
 static int core_setting_left(unsigned type, const char *label,
       bool wraparound)
diff --git a/menu/menu_driver.c b/menu/menu_driver.c
index 11bd084c7a..f08953d324 100644
--- a/menu/menu_driver.c
+++ b/menu/menu_driver.c
@@ -6699,9 +6699,11 @@ void menu_driver_toggle(
 
    if (on) 
    {
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_MENU);
 #endif
+#endif
 #ifdef HAVE_OVERLAY
       /* If an overlay was displayed before the toggle
        * and overlays are disabled in menu, need to
@@ -6720,9 +6722,11 @@ void menu_driver_toggle(
    }
    else
    {
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_CORE);
 #endif
+#endif
 #ifdef HAVE_OVERLAY
       /* Inhibits pointer 'select' and 'cancel' actions
        * (until the next time 'select'/'cancel' are released) */
diff --git a/retroarch.c b/retroarch.c
index 3f3b0eafa5..5c58f2c5a0 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -6573,16 +6573,20 @@ static void runloop_pause_checks(void)
       command_event(CMD_EVENT_DISCORD_UPDATE, &userdata);
 #endif
 
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_MENU);
+#endif
 #endif
    }
    else
    {
       RARCH_LOG("[Core]: %s\n", msg_hash_to_str(MSG_UNPAUSED));
 
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
       set_cpu_scaling_signal(CPUSCALING_EVENT_FOCUS_CORE);
+#endif
 #endif
    }
 
@@ -15915,9 +15919,11 @@ static void drivers_init(struct rarch_state *p_rarch,
    if (flags & DRIVER_MIDI_MASK)
       midi_driver_init(settings);
 
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
    cpu_scaling_driver_init();
 #endif
+#endif
 }
 
 /**
@@ -16014,9 +16020,11 @@ static void driver_uninit(struct rarch_state *p_rarch, int flags)
    if (flags & DRIVER_MIDI_MASK)
       midi_driver_free();
 
+#ifndef HAVE_LAKKA_SWITCH
 #ifdef HAVE_LAKKA
    cpu_scaling_driver_free();
 #endif
+#endif
 }
 
 static void retroarch_deinit_drivers(
