diff --git a/gfx/drivers_context/vc_egl_ctx.c b/gfx/drivers_context/vc_egl_ctx.c
index 646c79223b..7969138ae0 100644
--- a/gfx/drivers_context/vc_egl_ctx.c
+++ b/gfx/drivers_context/vc_egl_ctx.c
@@ -55,6 +55,13 @@
 
 #include "../../verbosity.h"
 
+typedef struct
+{
+   DISPMANX_ELEMENT_HANDLE_T element;
+   int width;
+   int height;
+} EGL_DISPMANX_WINDOW_T;
+
 typedef struct
 {
    bool smooth;
