--- a/retroarch.c	2020-02-08 22:27:40.576347000 +0100
+++ b/retroarch.c	2020-02-08 22:27:50.788166811 +0100
@@ -19563,6 +19563,7 @@
       if (ctx)
       {
          video_context_data = *ctx_data;
+         RARCH_LOG("Picking video context driver %s\n", ctx->ident);
          return ctx;
       }
    }
@@ -19576,6 +19577,7 @@
       if (ctx)
       {
          video_context_data = *ctx_data;
+         RARCH_LOG("Picking video context driver %s\n", ctx->ident);
          return ctx;
       }
    }
