From 923ab3b6f4cf34e055f18b0d159270831db8935b Mon Sep 17 00:00:00 2001
From: twinaphex <libretro@gmail.com>
Date: Tue, 27 Mar 2018 14:19:36 +0200
Subject: [PATCH] (DRM Ctx) Buildfix

---
 gfx/drivers_context/drm_ctx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gfx/drivers_context/drm_ctx.c b/gfx/drivers_context/drm_ctx.c
index 90fadbba4f..2300815c18 100644
--- a/gfx/drivers_context/drm_ctx.c
+++ b/gfx/drivers_context/drm_ctx.c
@@ -903,6 +903,7 @@ static void gfx_ctx_drm_set_flags(void *data, uint32_t flags)
 const gfx_ctx_driver_t gfx_ctx_drm = {
    gfx_ctx_drm_init,
    gfx_ctx_drm_destroy,
+   gfx_ctx_drm_get_api,
    gfx_ctx_drm_bind_api,
    gfx_ctx_drm_swap_interval,
    gfx_ctx_drm_set_video_mode,
