diff --git a/retroarch.c b/retroarch.c
index 8cbccb1b14..36cea4c27b 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -5477,6 +5477,10 @@ enum retro_language rarch_get_language_from_iso(const char *iso639)
       {"ar", RETRO_LANGUAGE_ARABIC},
       {"el", RETRO_LANGUAGE_GREEK},
    };
+   
+   if (iso639 == NULL) {
+      return lang;
+   }
 
    for (i = 0; i < sizeof(pairs) / sizeof(pairs[0]); i++)
    {
