diff -Naur retroarch/libretro-common/queues/task_queue.c retroarch.patch/libretro-common/queues/task_queue.c
--- retroarch/libretro-common/queues/task_queue.c	2016-07-31 00:07:35.497118479 +0200
+++ retroarch.patch/libretro-common/queues/task_queue.c	2016-08-02 22:18:01.040639520 +0200
@@ -82,7 +82,7 @@
       {
          if (task->error)
             task_queue_msg_push(1, 60, true, "%s: %s",
-               "Task failed\n", task->title);
+               "Task failed", task->title);
          else
             task_queue_msg_push(1, 60, true, "100%%: %s", task->title);
       }
diff -Naur retroarch/menu/menu_displaylist.c retroarch.patch/menu/menu_displaylist.c
--- retroarch/menu/menu_displaylist.c	2016-07-31 00:07:32.108998293 +0200
+++ retroarch.patch/menu/menu_displaylist.c	2016-07-31 00:59:53.660611939 +0200
@@ -2688,6 +2688,7 @@
             MENU_ENUM_LABEL_LOAD_STATE,
             MENU_SETTING_ACTION_LOADSTATE, 0, 0);
 
+#ifndef HAVE_LAKKA
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UNDO_LOAD_STATE),
             msg_hash_to_str(MENU_ENUM_LABEL_UNDO_LOAD_STATE),
@@ -2699,6 +2700,7 @@
             msg_hash_to_str(MENU_ENUM_LABEL_UNDO_SAVE_STATE),
             MENU_ENUM_LABEL_UNDO_SAVE_STATE,
             MENU_SETTING_ACTION_LOADSTATE, 0, 0);
+#endif
 
       menu_entries_append_enum(info->list,
             msg_hash_to_str(MENU_ENUM_LABEL_VALUE_CORE_OPTIONS),
diff -Naur retroarch/menu/menu_setting.c retroarch.patch/menu/menu_setting.c
--- retroarch/menu/menu_setting.c	2016-07-31 00:07:32.072997016 +0200
+++ retroarch.patch/menu/menu_setting.c	2016-08-09 14:16:29.955922130 +0200
@@ -4288,6 +4288,7 @@
                parent_group);
          menu_settings_list_current_add_enum_idx(list, list_info, MENU_ENUM_LABEL_MENU_SETTINGS);
 
+#ifndef HAVE_LAKKA
          CONFIG_ACTION(
                list, list_info,
                msg_hash_to_str(MENU_ENUM_LABEL_USER_INTERFACE_SETTINGS),
@@ -4296,6 +4297,7 @@
                &subgroup_info,
                parent_group);
          menu_settings_list_current_add_enum_idx(list, list_info, MENU_ENUM_LABEL_USER_INTERFACE_SETTINGS);
+#endif
 
          CONFIG_ACTION(
                list, list_info,
