diff --git a/menu/cbs/menu_cbs_ok.c b/menu/cbs/menu_cbs_ok.c
index 67652f4747..8bffa21635 100644
--- a/menu/cbs/menu_cbs_ok.c
+++ b/menu/cbs/menu_cbs_ok.c
@@ -2731,7 +2731,7 @@ static int action_ok_set_switch_cpu_profile(const char *path,
    char command[PATH_MAX_LENGTH] = {0};
 
 #ifdef HAVE_LAKKA_SWITCH
-   snprintf(command, sizeof(command), "cpu-profile set %s", profile_name);
+   snprintf(command, sizeof(command), "cpu-profile set '%s'", profile_name);
 
    system(command);
    snprintf(command, sizeof(command), "Current profile set to %s", profile_name);
diff --git a/switch_performance_profiles.h b/switch_performance_profiles.h
index 15dde6eec7..2b8a2ce3a7 100644
--- a/switch_performance_profiles.h
+++ b/switch_performance_profiles.h
@@ -20,7 +20,6 @@
 #if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LIBNX)
 #ifdef HAVE_LAKKA_SWITCH
 static char *SWITCH_GPU_PROFILES[] = {
-    "docked-overclock-3",
     "docked-overclock-2",
     "docked-overclock-1",
     "docked",
@@ -36,7 +35,6 @@ static char *SWITCH_GPU_PROFILES[] = {
 };
 
 static char *SWITCH_GPU_SPEEDS[] = {
-    "998 Mhz",
     "921 Mhz",
     "844 Mhz",
     "768 Mhz",
@@ -65,11 +63,13 @@ static int SWITCH_BRIGHTNESS[] = {
 
 static char *SWITCH_CPU_PROFILES[] = {
 #ifndef HAVE_LIBNX
-    "overclock-4",
-    "overclock-3",
-    "overclock-2",
-    "overclock-1",
-    "default",
+    "Maximum Performance",
+    "High Performance",
+    "Boost Performance",
+    "Stock Performance",
+    "Powersaving Mode 1",
+    "Powersaving Mode 2",
+    "Powersaving Mode 3",
 #else
     "Maximum Performance",
     "High Performance",
@@ -86,11 +86,13 @@ static char *SWITCH_CPU_PROFILES[] = {
 
 static char *SWITCH_CPU_SPEEDS[] = {
 #ifndef HAVE_LIBNX
-    "1912 MHz",
-    "1734 MHz",
-    "1530 MHz",
+    "1785 MHz",
+    "1581 MHz",
     "1224 MHz",
-    "1020 MHz"
+    "1020 MHz",
+    "918 MHz",
+    "816 MHz",
+    "714 MHz"
 #else
     "1785 MHz",
     "1581 MHz",
@@ -104,11 +106,13 @@ static char *SWITCH_CPU_SPEEDS[] = {
 
 static unsigned SWITCH_CPU_SPEEDS_VALUES[] = {
 #ifndef HAVE_LIBNX
-    1912000000,
-    1734000000,
-    1530000000,
+    1785000000,
+    1581000000,
     1224000000,
-    1020000000
+    1020000000,
+    918000000,
+    816000000,
+    714000000
 #else
     1785000000,
     1581000000,
