diff -Naur retroarch.git/gfx/drivers_context/drm_ctx.c retroarch.patch/gfx/drivers_context/drm_ctx.c
--- retroarch.git/gfx/drivers_context/drm_ctx.c	2016-05-26 23:11:52.632009156 +0200
+++ retroarch.patch/gfx/drivers_context/drm_ctx.c	2016-05-26 23:21:07.039178715 +0200
@@ -327,7 +327,10 @@
 
    if (drm->drm)
       if (g_drm_fd >= 0)
+      {
+	 drmDropMaster(g_drm_fd);
          filestream_close(drm->drm);
+      }
 
    g_gbm_surface      = NULL;
    g_gbm_dev          = NULL;
@@ -423,6 +426,8 @@
    drm->fb_width    = g_drm_connector->modes[0].hdisplay;
    drm->fb_height   = g_drm_connector->modes[0].vdisplay;
 
+   drmSetMaster(g_drm_fd);
+
    g_gbm_dev        = gbm_create_device(fd);
 
    if (!g_gbm_dev)
