diff -Naur retroarch-b3aef50/menu/menu_displaylist.c retroarch.patch/menu/menu_displaylist.c
--- retroarch-b3aef50/menu/menu_displaylist.c	2017-02-04 10:47:25.918945904 +0100
+++ retroarch.patch/menu/menu_displaylist.c	2017-02-07 02:27:15.723884551 +0100
@@ -3873,7 +3873,7 @@
             MENU_SETTING_ACTION, 0, 0);
    }
 
-#ifdef HAVE_NETWORKING
+#if 0
    if (info->download_core)
    {
       menu_entries_prepend(info->list,
