diff -Naur json-c-0.9/configure.in json-c-0.9.patch/configure.in
--- json-c-0.9/configure.in	2009-01-06 23:55:33.000000000 +0100
+++ json-c-0.9.patch/configure.in	2013-02-01 16:45:09.890830041 +0100
@@ -10,7 +10,7 @@
 # Checks for libraries.
 
 # Checks for header files.
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h limits.h strings.h syslog.h unistd.h [sys/param.h] stdarg.h)
 
