diff -Naur xwiimote-7341491-old/makefile xwiimote-7341491-new/makefile
--- xwiimote-7341491-old/makefile	2011-08-02 10:16:15.000000000 -0700
+++ xwiimote-7341491-new/makefile	2011-08-02 10:20:01.000000000 -0700
@@ -4,12 +4,14 @@
 # Dedicated to the Public Domain
 #
 
+CC ?= gcc
+
 .PHONY: build clean
 
 build: libxwiimote.so
 
 libxwiimote.so: lib/*.c
-	gcc -shared -o libxwiimote.so lib/*.c -fPIC -Wall -O2 -ludev
+	$(CC) -shared -o libxwiimote.so lib/*.c -fPIC -Wall -O2 -ludev $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 clean:
 	@rm -fv libxwiimote.so
diff -Naur xwiimote-7341491-old/tools/makefile xwiimote-7341491-new/tools/makefile
--- xwiimote-7341491-old/tools/makefile	2011-08-02 10:16:15.000000000 -0700
+++ xwiimote-7341491-new/tools/makefile	2011-08-02 10:21:44.000000000 -0700
@@ -4,6 +4,8 @@
 # Dedicated to the Public Domain
 #
 
+CC ?= gcc
+
 .PHONY: all clean
 
 all: xwiiconn xwii xwiishow xwiidump
@@ -12,16 +14,16 @@
 	@rm -fv xwiiconn xwii xwiishow xwiidump
 
 xwiiconn: xwiiconn.c
-	gcc -o xwiiconn xwiiconn.c -Wall -O0 -g -lbluetooth `pkg-config --libs --cflags dbus-glib-1 glib-2.0`
+	$(CC) -o xwiiconn xwiiconn.c -Wall -O0 -g -lbluetooth `pkg-config --libs --cflags dbus-glib-1 glib-2.0` $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 xwii: xwii.c ../libxwiimote.so
-	gcc -o xwii xwii.c -Wall -O0 -g ../libxwiimote.so -I../lib
+	$(CC) -o xwii xwii.c -Wall -O0 -g -L../ -lxwiimote -I../lib $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 xwiishow: xwiishow.c ../libxwiimote.so
-	gcc -o xwiishow xwiishow.c -Wall -O0 -g ../libxwiimote.so -I../lib
+	$(CC) -o xwiishow xwiishow.c -Wall -O0 -g -L../ -lxwiimote -I../lib $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 xwiidump: xwiidump.c
-	gcc -o xwiidump xwiidump.c -Wall -O0 -g
+	$(CC) -o xwiidump xwiidump.c -Wall -O0 -g $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 ../libxwiimote.so:
 	@echo Making library
