diff -Naur a/dtbTool b/dtbTool
--- a/dtbTool	2017-02-01 14:07:03.000000000 -0800
+++ b/dtbTool	2017-10-23 22:32:17.090949808 -0700
@@ -84,6 +84,7 @@
 		'apq8096'        : 291,
 		'msm8998'        : 292,
 		'msm8996sg'      : 305,
+		'ipq8074'        : 306,
 	}
 
 platform_type = {
@@ -108,6 +109,7 @@
 		'stp'         : 23,
 		'sbc'         : 24,
 		'cls'         : 29,
+		'hk01'        : 30,
 		}
 
 pmic_ids = {
