diff -Nur a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
--- a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2018-12-01 21:10:15.000000000 +0100
+++ b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2018-12-01 23:28:48.956901556 +0100
@@ -3815,12 +3815,18 @@
 			"Continuing without Mali regulator control\n");
 		/* Allow probe to continue without regulator */
 	}
+	err = regulator_enable(kbdev->regulator);
+	if (err) {
+		dev_err(kbdev->dev, "Failed to enable regulator\n");
+		return err;
+	}
 #endif /* LINUX_VERSION_CODE >= 3, 12, 0 */
 
 	kbdev->mali_rst = of_reset_control_get_by_index(kbdev->dev->of_node, 0);
 	if (IS_ERR(kbdev->mali_rst)) {
 		dev_err(kbdev->dev, "Couldn't get mali reset line\n");
 		err = PTR_ERR(kbdev->mali_rst);
+		kbdev->mali_rst = NULL;
 		goto fail;
 	}
 
@@ -3828,6 +3834,7 @@
 	if (IS_ERR(kbdev->bus_clk)) {
 		dev_err(kbdev->dev, "Couldn't get the mali bus clock\n");
 		err = PTR_ERR(kbdev->bus_clk);
+		kbdev->bus_clk = NULL;
 		goto fail;
 	}
 
