diff -Nur a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c
--- a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c	2019-04-18 22:11:29.000000000 +0200
+++ b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c	2019-04-18 22:15:09.982940904 +0200
@@ -1661,7 +1661,11 @@
 static int kbase_cpu_vm_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 {
 #else
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0))
 static int kbase_cpu_vm_fault(struct vm_fault *vmf)
+#else
+static vm_fault_t kbase_cpu_vm_fault(struct vm_fault *vmf)
+#endif
 {
 	struct vm_area_struct *vma = vmf->vma;
 #endif
