diff -Naur nss-3.29.5.orig/nspr/config/autoconf.mk.in nss-3.29.5/nspr/config/autoconf.mk.in
--- nss-3.29.5.orig/nspr/config/autoconf.mk.in	2017-09-08 11:03:27.572619156 +0200
+++ nss-3.29.5/nspr/config/autoconf.mk.in	2017-09-08 11:03:41.100520343 +0200
@@ -104,7 +104,7 @@
 RESOLVE_LINK_SYMBOLS = @RESOLVE_LINK_SYMBOLS@
 
 HOST_CC		= @HOST_CC@
-HOST_CFLAGS	= @HOST_CFLAGS@
+#HOST_CFLAGS	= @HOST_CFLAGS@
 HOST_LDFLAGS	= @HOST_LDFLAGS@
 
 DEFINES		= @DEFINES@ @DEFS@
