diff --git a/nss/lib/freebl/Makefile b/nss/lib/freebl/Makefile
index bff11c7..0c714c8 100644
--- a/nss/lib/freebl/Makefile
+++ b/nss/lib/freebl/Makefile
@@ -541,9 +541,7 @@ else
     ECL_SRCS += curve25519_32.c
 endif
 
-ifndef HAVE_INT128_SUPPORT
-    VERIFIED_SRCS += FStar.c
-endif
+VERIFIED_SRCS += FStar.c
 
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #
