diff --git a/CMakeLists.txt b/CMakeLists.txt
index 928667710..6f1015980 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -153,6 +153,6 @@ elseif(LIBRETRO)
 	set(CMAKE_SHARED_LIBRARY_PREFIX "")
 	set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 	target_compile_definitions(${PROJECT_NAME} PRIVATE LIBRETRO)
-	target_compile_options(${PROJECT_NAME} PRIVATE -Wno-misleading-indentation)
+	target_compile_options(${PROJECT_NAME} PRIVATE -Wno-misleading-indentation -g)
 	if(APPLE)
 		find_library(FOUNDATION Foundation)
 		target_link_libraries(${PROJECT_NAME} PRIVATE ${FOUNDATION})
@@ -209,7 +210,9 @@ if(WINDOWS_STORE)
 	set(USE_VULKAN OFF)
 endif()
 
-set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS_RELEASE -s)
+if(NOT LIBRETRO)
+	set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS_RELEASE -s)
+endif()
 if(MSVC)
 	target_compile_options(${PROJECT_NAME} PRIVATE /GR /GS-)
 	if(WINDOWS_STORE)
