diff --git a/configure.ac b/configure.ac
index b79026d..ac73873 100644
--- a/configure.ac
+++ b/configure.ac
@@ -947,7 +947,7 @@ if test "$os" = "linux"; then
 
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
 
-   MODULES="$MODULES vmxnet vmhgfs"
+   MODULES="$MODULES vmxnet"
    # See if we need vmci and vsock modules. Starting with 3.9 they made
    # their way into mainline kernel.
    if test "$osVersion" -lt 309000; then
