diff --git a/configure b/configure
index 3861c20..bec644c 100755
--- a/configure
+++ b/configure
@@ -2881,7 +2881,7 @@ if test "$with_kernel_modules" = "yes"; then
          if test ! -d "$LINUXDIR/kernel/"; then
              as_fn_error $? "$LINUXDIR/kernel does not exist" "$LINENO" 5
          fi
-         LINUXINCLUDE="$LINUXDIR/build/include"
+         LINUXINCLUDE="$LINUXDIR/include"
          if test ! -d "$LINUXINCLUDE"; then
             as_fn_error $? "Can't find include dir under $LINUXDIR" "$LINENO" 5
          fi
diff --git a/configure.ac b/configure.ac
index be4a7c0..2bf6779 100644
--- a/configure.ac
+++ b/configure.ac
@@ -148,7 +148,7 @@ if test "$with_kernel_modules" = "yes"; then
          if test ! -d "$LINUXDIR/kernel/"; then
              AC_MSG_ERROR([$LINUXDIR/kernel does not exist])
          fi
-         LINUXINCLUDE="$LINUXDIR/build/include"
+         LINUXINCLUDE="$LINUXDIR/include"
          if test ! -d "$LINUXINCLUDE"; then
             AC_MSG_ERROR([Can't find include dir under $LINUXDIR])
          fi
