diff -Naur joycond-031f04311a912514cea9deb020512ee6d7063398/CMakeLists.txt joycond-031f04311a912514cea9deb020512ee6d7063398-2/CMakeLists.txt
--- joycond-031f04311a912514cea9deb020512ee6d7063398/CMakeLists.txt	2022-07-23 07:18:46.000000000 +0200
+++ joycond-031f04311a912514cea9deb020512ee6d7063398-2/CMakeLists.txt	2022-10-05 03:59:01.052897338 +0200
@@ -25,7 +25,7 @@
 
 add_subdirectory(src)
 
-install(FILES joycond DESTINATION /usr/bin/
+install(FILES .aarch64-libreelec-linux-gnueabi/joycond DESTINATION /usr/bin/
         PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
         )
 install(FILES udev/89-joycond.rules udev/72-joycond.rules DESTINATION /lib/udev/rules.d/
