From 7dcf55c9560172db7e9daee2c20413cd8c9549dd Mon Sep 17 00:00:00 2001
From: xbmc <fernetmenta@online.de>
Date: Sun, 16 Jun 2013 13:23:19 +0200
Subject: [PATCH] renderer: delete fence on uninit

---
 xbmc/cores/VideoRenderers/LinuxRendererGL.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGL.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGL.cpp
index c663423..858d39d 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGL.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGL.cpp
@@ -1164,7 +1164,12 @@ void CLinuxRendererGL::UnInit()
 
   // YV12 textures
   for (int i = 0; i < NUM_BUFFERS; ++i)
+  {
     (this->*m_textureDelete)(i);
+    if (m_buffers[i].fence)
+      glDeleteSync(m_buffers[i].fence);
+    m_buffers[i].fence = None;
+  }
 
   // cleanup framebuffer object if it was in use
   m_fbo.fbo.Cleanup();
-- 
1.8.1.6

