diff -Naur xbmc-audioencoder-wav-91a2b72/CMakeLists.txt xbmc-audioencoder-wav-91a2b72.patch/CMakeLists.txt
--- xbmc-audioencoder-wav-91a2b72/CMakeLists.txt	2014-07-18 23:48:40.000000000 +0200
+++ xbmc-audioencoder-wav-91a2b72.patch/CMakeLists.txt	2014-10-20 13:08:19.017026869 +0200
@@ -6,7 +6,7 @@
 
 enable_language(CXX)
 
-find_package(xbmc REQUIRED)
+find_package(kodi REQUIRED)
 
 include_directories(${XBMC_INCLUDE_DIR})
 
diff -Naur xbmc-audioencoder-wav-91a2b72/src/EncoderWav.cpp xbmc-audioencoder-wav-91a2b72.patch/src/EncoderWav.cpp
--- xbmc-audioencoder-wav-91a2b72/src/EncoderWav.cpp	2014-07-18 23:48:40.000000000 +0200
+++ xbmc-audioencoder-wav-91a2b72.patch/src/EncoderWav.cpp	2014-10-20 02:55:43.985839271 +0200
@@ -18,7 +18,7 @@
  *
  */
 
-#include "xbmc/xbmc_audioenc_dll.h"
+#include "kodi/xbmc_audioenc_dll.h"
 #include <stdio.h>
 #include <string.h>
 
