diff --git a/Makefile.am b/Makefile.am
index 906140e..75888ae 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,13 +6,7 @@ endif
 if HAS_ANDROID_5_0_0
 SUBDIRS += libsync
 endif
-SUBDIRS += egl glesv1 glesv2 ui sf input camera vibrator media wifi
-
-if HAS_LIBNFC_NXP_HEADERS
-SUBDIRS += libnfc_nxp libnfc_ndef_nxp
-endif
-SUBDIRS += utils tests
-
+SUBDIRS += egl glesv1 glesv2
 
 MAINTAINERCLEANFILES = \
 	aclocal.m4 compile config.guess config.sub \
