diff -Naur mysql-5.1.67-old/configure.in mysql-5.1.67-new/configure.in
--- mysql-5.1.67-old/configure.in	2012-12-13 02:05:51.000000000 -0800
+++ mysql-5.1.67-new/configure.in	2012-12-30 08:39:00.000000000 -0800
@@ -22,7 +22,7 @@
 AM_INIT_AUTOMAKE([1.9 tar-ustar])
 AC_PROG_LIBTOOL
 
-AM_CONFIG_HEADER([include/config.h])
+AC_CONFIG_HEADERS([include/config.h])
 
 # Request support for automake silent-rules if available.
 # Default to verbose output. One can use the configure-time
