diff -Naur a/extra/CMakeLists.txt b/extra/CMakeLists.txt
--- a/extra/CMakeLists.txt	2016-09-21 23:16:59.458308368 -0700
+++ b/extra/CMakeLists.txt	2016-09-21 23:17:34.187663318 -0700
@@ -28,11 +28,9 @@
 # Default install component for the files is Server here
 SET(MYSQL_INSTALL_COMPONENT Server)
 
-IF(NOT CMAKE_CROSSCOMPILING)
- ADD_EXECUTABLE(comp_err comp_err.c)
- TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl)
- SET_TARGET_PROPERTIES(comp_err PROPERTIES LINKER_LANGUAGE CXX)
-ENDIF()
+ADD_EXECUTABLE(comp_err comp_err.c)
+TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl)
+SET_TARGET_PROPERTIES(comp_err PROPERTIES LINKER_LANGUAGE CXX)
 
 IF (BUILD_BUNDLED_LZ4)
   ADD_CONVENIENCE_LIBRARY(lz4_lib
