--- a/sql-common/client_authentication.cc	2017-05-06 16:38:34.349486565 +0200
+++ b/sql-common/client_authentication.cc	2017-05-06 16:41:32.286431336 +0200
@@ -84,7 +84,7 @@
 
   if (mysql->options.extension != NULL &&
       mysql->options.extension->server_public_key_path != NULL &&
-      mysql->options.extension->server_public_key_path != '\0')
+      *(mysql->options.extension->server_public_key_path) != '\0')
   {
     pub_key_file= fopen(mysql->options.extension->server_public_key_path,
                         "r");
