diff -up supertux-0.3.1/src/addon_manager.cpp~ supertux-0.3.1/src/addon_manager.cpp
--- supertux-0.3.1/src/addon_manager.cpp~	2008-07-10 10:58:11.000000000 +0200
+++ supertux-0.3.1/src/addon_manager.cpp	2008-07-10 10:58:11.000000000 +0200
@@ -25,6 +25,7 @@
 #include <physfs.h>
 #include <sys/stat.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "addon_manager.hpp"
 #include "config.h"
 #include "log.hpp"
diff -up supertux-0.3.1/src/console.hpp~ supertux-0.3.1/src/console.hpp
--- supertux-0.3.1/src/console.hpp~	2008-07-10 10:48:45.000000000 +0200
+++ supertux-0.3.1/src/console.hpp	2008-07-10 10:57:01.000000000 +0200
@@ -26,6 +26,7 @@
 #include <string>
 #include <sstream>
 #include <iostream>
+#include <memory>
 #include <squirrel.h>
 
 class Console;
diff -up supertux-0.3.1/src/lisp/lexer.cpp~ supertux-0.3.1/src/lisp/lexer.cpp
--- supertux-0.3.1/src/lisp/lexer.cpp~	2008-07-10 11:00:49.000000000 +0200
+++ supertux-0.3.1/src/lisp/lexer.cpp	2008-07-10 11:00:49.000000000 +0200
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <string.h>
 
 #include "lexer.hpp"
 
diff -up supertux-0.3.1/src/resources.cpp~ supertux-0.3.1/src/resources.cpp
diff -up supertux-0.3.1/src/sprite/sprite_manager.hpp~ supertux-0.3.1/src/sprite/sprite_manager.hpp
--- supertux-0.3.1/src/sprite/sprite_manager.hpp~	2008-07-10 10:58:59.000000000 +0200
+++ supertux-0.3.1/src/sprite/sprite_manager.hpp	2008-07-10 10:58:59.000000000 +0200
@@ -21,6 +21,7 @@
 #define SUPERTUX_SPRITE_MANAGER_H
 
 #include <map>
+#include <string>
 
 class SpriteData;
 class Sprite;
diff -up supertux-0.3.1/src/textscroller.hpp~ supertux-0.3.1/src/textscroller.hpp
--- supertux-0.3.1/src/textscroller.hpp~	2008-07-10 10:50:27.000000000 +0200
+++ supertux-0.3.1/src/textscroller.hpp	2008-07-10 10:57:18.000000000 +0200
@@ -24,6 +24,7 @@
 #include <vector>
 #include <string>
 #include <map>
+#include <memory>
 
 #include "screen.hpp"
 #include "math/vector.hpp"
diff -up supertux-0.3.1/src/title.cpp~ supertux-0.3.1/src/title.cpp
--- supertux-0.3.1/src/title.cpp~	2008-07-10 10:57:50.000000000 +0200
+++ supertux-0.3.1/src/title.cpp	2008-07-10 10:57:50.000000000 +0200
@@ -23,6 +23,7 @@
 #include <iostream>
 #include <sstream>
 #include <stdexcept>
+#include <algorithm>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -up supertux-0.3.1/src/video/sdl_texture.hpp~ supertux-0.3.1/src/video/sdl_texture.hpp
--- supertux-0.3.1/src/video/sdl_texture.hpp~	2008-07-10 11:01:13.000000000 +0200
+++ supertux-0.3.1/src/video/sdl_texture.hpp	2008-07-10 11:01:13.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <algorithm>
 #include <SDL.h>
 
 #include "texture.hpp"
