Fix build with GCC 4.4

Lubomir Rintel <lkundrak@v3.sk>

diff -up supertux-0.3.1/src/lisp/lisp.cpp.gcc44 supertux-0.3.1/src/lisp/lisp.cpp
--- supertux-0.3.1/src/lisp/lisp.cpp.gcc44	2009-03-01 15:13:40.000000000 +0100
+++ supertux-0.3.1/src/lisp/lisp.cpp	2009-03-01 15:13:53.000000000 +0100
@@ -19,6 +19,7 @@
 
 #include <config.h>
 
+#include <cstdio>
 #include "lisp.hpp"
 
 namespace lisp
diff -up supertux-0.3.1/src/random_generator.cpp.gcc44 supertux-0.3.1/src/random_generator.cpp
--- supertux-0.3.1/src/random_generator.cpp.gcc44	2009-03-01 15:14:05.000000000 +0100
+++ supertux-0.3.1/src/random_generator.cpp	2009-03-01 15:14:18.000000000 +0100
@@ -39,6 +39,7 @@
 #include <stdexcept>
 #include <time.h>
 #include <cassert>
+#include <cstdio>
 #include "random_generator.hpp"
 
 RandomGenerator systemRandom;               // global random number generator
