diff -Naur lutro.git/Makefile lutro.patch/Makefile
--- lutro.git/Makefile	2015-03-11 03:30:46.660525401 +0100
+++ lutro.patch/Makefile	2015-03-11 04:15:24.998942686 +0100
@@ -134,7 +134,7 @@
 	$(MAKE) -C deps/lua/src CC="$(CC)" CXX="$(CXX)" MYCFLAGS="$(LUA_MYCFLAGS) -w -g $(fpic)" MYLDFLAGS="$(LFLAGS) $(fpic)" SYSCFLAGS="$(LUA_SYSCFLAGS) $(fpic)" a
 
 deps/luajit/src/libluajit.a:
-	$(MAKE) -C deps/luajit/src BUILDMODE=static CFLAGS="$(LUA_MYCFLAGS) $(fpic)" Q= LDFLAGS="$(fpic)"
+	$(MAKE) -C deps/luajit/src HOST_CC="$(HOST_CC) $(PTR_SIZE)" CROSS="$(CROSS)" BUILDMODE=static CFLAGS="$(LUA_MYCFLAGS) $(fpic)" Q= LDFLAGS="$(fpic)"
 
 %.o: %.c
 	$(CC) $(CFLAGS) -c -o $@ $<
