diff -Naur libhid-0.2.16/m4/md_check_libusb018b.m4 libhid-0.2.16.patch/m4/md_check_libusb018b.m4
--- libhid-0.2.16/m4/md_check_libusb018b.m4	2004-05-26 02:37:41.000000000 +0200
+++ libhid-0.2.16.patch/m4/md_check_libusb018b.m4	2016-01-10 00:07:24.000000000 +0100
@@ -2,8 +2,8 @@
   [
     AC_CHECK_HEADERS([usb.h])
 
-    LIBUSB_CFLAGS="`libusb-config --cflags`"
-    LIBUSB_LIBS="`libusb-config --libs`"
+    LIBUSB_CFLAGS="`pkg-config --cflags libusb`"
+    LIBUSB_LIBS="`pkg-config --libs libusb`"
 
     AC_SUBST(LIBUSB_CFLAGS)
     AC_SUBST(LIBUSB_LIBS)
