diff -Naur libhid-0.2.16/configure.ac libhid-0.2.16.patch/configure.ac
--- libhid-0.2.16/configure.ac	2007-04-01 22:32:10.000000000 +0200
+++ libhid-0.2.16.patch/configure.ac	2013-01-12 17:23:25.129691249 +0100
@@ -21,7 +21,7 @@
 
 AC_INIT(MD_INIT_NAME, MD_INIT_VERSION, MD_INIT_ADDRESS)
 
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([include/hid.h])
 AC_CONFIG_AUX_DIR([.])
 
