From 4708fdeab67cae81ee7211a306c6adf0bed24d3c Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Tue, 3 Feb 2015 18:19:17 +0200
Subject: [PATCH] fix wirbelscancontrol

---
 common.c            |    2 +-
 extended_frontend.h |    7 +++----
 menusetup.c         |    2 +-
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/common.c b/common.c
index 04a1e6e..40194d2 100644
--- a/common.c
+++ b/common.c
@@ -54,7 +54,7 @@ void cMySetup::InitSystems(void) {
      }
 
   if (DVB_Type >= SCAN_NO_DEVICE || ! systems[DVB_Type]) {
-     for (DVB_Type = scantype_t(SCAN_UNDEFINED + 1); DVB_Type < SCAN_NO_DEVICE; DVB_Type = scantype_t(DVB_Type + 1)) {
+     for (DVB_Type = scantype_t(SCAN_TERRESTRIAL); DVB_Type < SCAN_NO_DEVICE; DVB_Type = scantype_t(DVB_Type + 1)) {
          if (systems[DVB_Type])
             break;
          }
diff --git a/extended_frontend.h b/extended_frontend.h
index 680594e..bcdb11f 100644
--- a/extended_frontend.h
+++ b/extended_frontend.h
@@ -152,13 +152,12 @@ typedef enum {
  * 20120107 wirbel
  */
 typedef enum {
-        SCAN_UNDEFINED,
-        SCAN_SATELLITE,
-        SCAN_CABLE,
         SCAN_TERRESTRIAL,
-        SCAN_TERRCABLE_ATSC,   /* I dislike this mixture of terr and cable. fix later, as it leads to problems now. */
+        SCAN_CABLE,
+        SCAN_SATELLITE,
         SCAN_PVRINPUT,
         SCAN_PVRINPUT_FM,
+        SCAN_TERRCABLE_ATSC,   /* I dislike this mixture of terr and cable. fix later, as it leads to problems now. */
         SCAN_NO_DEVICE,
         SCAN_TRANSPONDER=999
 } scantype_t;
diff --git a/menusetup.c b/menusetup.c
index daeccb1..46087be 100644
--- a/menusetup.c
+++ b/menusetup.c
@@ -22,7 +22,7 @@
 using namespace COUNTRY;
 
 static const char * ScannerDesc   = "wirbelscan scan thread";
-static const char * DVB_Types[]  = {"NOTHING","DVB-S/S2","DVB-C","DVB-T/T2","ATSC","pvrinput","pvrinput FM", "no device found"};
+static const char * DVB_Types[]  = {"DVB-T/T2", "DVB-C", "DVB-S/S2", "pvrinput", "pvrinput FM", "ATSC", "no device found"};
 
 cMenuScanning * MenuScanning    = NULL;   // pointer to actual menu
 cScanner      * Scanner         = NULL;
-- 
1.7.10.4

