From: awiouy <awiouy@gmail.com>
Date: Thu, 6 Jul 2017 18:51:24 +0200
Subject: [PATCH] htop: netbsd-curses support

diff --git a/RichString.c b/RichString.c
--- a/RichString.c
+++ b/RichString.c
@@ -18,21 +18,7 @@ in the source distribution for its full text.
 #include <ctype.h>
 
 #include <assert.h>
-#ifdef HAVE_NCURSESW_CURSES_H
-#include <ncursesw/curses.h>
-#elif HAVE_NCURSES_NCURSES_H
-#include <ncurses/ncurses.h>
-#elif HAVE_NCURSES_CURSES_H
-#include <ncurses/curses.h>
-#elif HAVE_NCURSES_H
-#include <ncurses.h>
-#elif HAVE_CURSES_H
 #include <curses.h>
-#endif
-
-#ifdef HAVE_LIBNCURSESW
-#include <wctype.h>
-#endif
 
 #define RichString_size(this) ((this)->chlen)
 #define RichString_sizeVal(this) ((this).chlen)
@@ -41,19 +27,11 @@ in the source distribution for its full text.
 #define RichString_beginAllocated(this) memset(&this, 0, sizeof(RichString)); (this).chptr = (this).chstr;
 #define RichString_end(this) RichString_prune(&(this));
 
-#ifdef HAVE_LIBNCURSESW
-#define RichString_printVal(this, y, x) mvadd_wchstr(y, x, (this).chptr)
-#define RichString_printoffnVal(this, y, x, off, n) mvadd_wchnstr(y, x, (this).chptr + off, n)
-#define RichString_getCharVal(this, i) ((this).chptr[i].chars[0] & 255)
-#define RichString_setChar(this, at, ch) do{ (this)->chptr[(at)] = (CharType) { .chars = { ch, 0 } }; } while(0)
-#define CharType cchar_t
-#else
 #define RichString_printVal(this, y, x) mvaddchstr(y, x, (this).chptr)
 #define RichString_printoffnVal(this, y, x, off, n) mvaddchnstr(y, x, (this).chptr + off, n)
 #define RichString_getCharVal(this, i) ((this).chptr[i])
 #define RichString_setChar(this, at, ch) do{ (this)->chptr[(at)] = ch; } while(0)
 #define CharType chtype
-#endif
 
 typedef struct RichString_ {
    int chlen;
@@ -87,41 +65,6 @@ static void RichString_extendLen(RichString* this, int len) {
 
 #define RichString_setLen(this, len) do{ if(len < RICHSTRING_MAXLEN && this->chlen < RICHSTRING_MAXLEN) { RichString_setChar(this,len,0); this->chlen=len; } else RichString_extendLen(this,len); }while(0)
 
-#ifdef HAVE_LIBNCURSESW
-
-static inline void RichString_writeFrom(RichString* this, int attrs, const char* data_c, int from, int len) {
-   wchar_t data[len+1];
-   len = mbstowcs(data, data_c, len);
-   if (len < 0)
-      return;
-   int newLen = from + len;
-   RichString_setLen(this, newLen);
-   for (int i = from, j = 0; i < newLen; i++, j++) {
-      this->chptr[i] = (CharType) { .attr = attrs & 0xffffff, .chars = { (iswprint(data[j]) ? data[j] : '?') } };
-   }
-}
-
-inline void RichString_setAttrn(RichString* this, int attrs, int start, int finish) {
-   cchar_t* ch = this->chptr + start;
-   for (int i = start; i <= finish; i++) {
-      ch->attr = attrs;
-      ch++;
-   }
-}
-
-int RichString_findChar(RichString* this, char c, int start) {
-   wchar_t wc = btowc(c);
-   cchar_t* ch = this->chptr + start;
-   for (int i = start; i < this->chlen; i++) {
-      if (ch->chars[0] == wc)
-         return i;
-      ch++;
-   }
-   return -1;
-}
-
-#else
-
 static inline void RichString_writeFrom(RichString* this, int attrs, const char* data_c, int from, int len) {
    int newLen = from + len;
    RichString_setLen(this, newLen);
@@ -148,8 +91,6 @@ int RichString_findChar(RichString* this, char c, int start) {
    return -1;
 }
 
-#endif
-
 void RichString_prune(RichString* this) {
    if (this->chlen > RICHSTRING_MAXLEN)
       free(this->chptr);

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -183,36 +183,6 @@
    ], [$4])
 ])
 
-AC_ARG_ENABLE(unicode, [AS_HELP_STRING([--enable-unicode], [enable Unicode support])], ,enable_unicode="yes")
-if test "x$enable_unicode" = xyes; then
-   HTOP_CHECK_SCRIPT([ncursesw6], [addnwstr], [HAVE_LIBNCURSESW], "ncursesw6-config",
-    HTOP_CHECK_SCRIPT([ncursesw], [addnwstr], [HAVE_LIBNCURSESW], "ncursesw5-config",
-     HTOP_CHECK_SCRIPT([ncurses], [addnwstr], [HAVE_LIBNCURSESW], "ncurses5-config",
-      HTOP_CHECK_LIB([ncursesw6], [addnwstr], [HAVE_LIBNCURSESW],
-       HTOP_CHECK_LIB([ncursesw], [addnwstr], [HAVE_LIBNCURSESW],
-        HTOP_CHECK_LIB([ncurses], [addnwstr], [HAVE_LIBNCURSESW],
-      missing_libraries="$missing_libraries libncursesw"
-      AC_MSG_ERROR([You may want to use --disable-unicode or install libncursesw.])
-   ))))))
-
-   AC_CHECK_HEADERS([ncursesw/curses.h],[:],
-      [AC_CHECK_HEADERS([ncurses/ncurses.h],[:],
-         [AC_CHECK_HEADERS([ncurses/curses.h],[:],
-            [AC_CHECK_HEADERS([ncurses.h],[:],[missing_headers="$missing_headers $ac_header"])])])])
-else
-   HTOP_CHECK_SCRIPT([ncurses6], [refresh], [HAVE_LIBNCURSES], "ncurses6-config",
-    HTOP_CHECK_SCRIPT([ncurses], [refresh], [HAVE_LIBNCURSES], "ncurses5-config",
-     HTOP_CHECK_LIB([ncurses6],  [refresh], [HAVE_LIBNCURSES],
-      HTOP_CHECK_LIB([ncurses],  [refresh], [HAVE_LIBNCURSES],
-      missing_libraries="$missing_libraries libncurses"
-   ))))
-   
-   AC_CHECK_HEADERS([curses.h],[:],
-      [AC_CHECK_HEADERS([ncurses/curses.h],[:],
-         [AC_CHECK_HEADERS([ncurses/ncurses.h],[:],
-            [AC_CHECK_HEADERS([ncurses.h],[:],[missing_headers="$missing_headers $ac_header"])])])])
-fi
-
 if test "$my_htop_platform" = "freebsd"; then
    AC_CHECK_LIB([kvm], [kvm_open], [], [missing_libraries="$missing_libraries libkvm"])
 fi

