diff -Naur glupen64-dd195c8/libretro-common/include/glsym/glsym_es2.h glupen64-dd195c8.patch/libretro-common/include/glsym/glsym_es2.h
--- glupen64-dd195c8/libretro-common/include/glsym/glsym_es2.h	2016-12-07 21:26:44.210012103 +0100
+++ glupen64-dd195c8.patch/libretro-common/include/glsym/glsym_es2.h	2016-12-07 21:27:40.214010330 +0100
@@ -23,7 +23,7 @@
 typedef GLint GLfixed;
 #endif
 
-#if (__STDC_VERSION__ <= 199901L) || (OSX && !MAC_OS_X_VERSION_10_7)
+#if 1
 #ifndef GLint64
 typedef long long int GLint64;
 #endif
