diff --git a/couchpotato/runner.py b/couchpotato/runner.py
index c0b7eb8..3a550f9 100644
--- a/couchpotato/runner.py
+++ b/couchpotato/runner.py
@@ -112,7 +112,7 @@ def runCouchPotato(options, base_path, args, data_dir = None, log_dir = None, En
     Env.set('data_dir', data_dir)
     Env.set('log_path', os.path.join(log_dir, 'CouchPotato.log'))
     Env.set('db_path', 'sqlite:///' + db_path)
-    Env.set('cache_dir', os.path.join(data_dir, 'cache'))
+    Env.set('cache_dir', os.path.join(data_dir, 'cache.cpv2'))
     Env.set('cache', FileSystemCache(os.path.join(Env.get('cache_dir'), 'python')))
     Env.set('console_log', options.console_log)
     Env.set('quiet', options.quiet)
