diff -Naur transmission-2.76/configure.ac transmission-2.76.patch/configure.ac
--- transmission-2.76/configure.ac	2013-01-09 03:32:09.534138180 +0200
+++ transmission-2.76.patch/configure.ac	2013-01-12 22:44:37.881897688 +0200
@@ -14,7 +14,7 @@
 
 AC_CONFIG_MACRO_DIR([m4])
 
-dnl AM_CONFIG_HEADER(config.h)
+dnl AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR(libtransmission/transmission.h)
 AM_INIT_AUTOMAKE([1.9 tar-pax])
 LT_INIT
