commit 17807066bc04cd28bba89fd176d4d0f69ead9728
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 12 16:18:50 2013 +0300

    disable minimize

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index b5d40c0..18bea9d 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -5448,7 +5448,6 @@ bool CApplication::SwitchToFullScreen()
 
 void CApplication::Minimize()
 {
-  g_Windowing.Minimize();
 }
 
 PLAYERCOREID CApplication::GetCurrentPlayer()
