diff -Naur vdr-1.7.5a/Makefile vdr-1.7.5b/Makefile
--- vdr-1.7.5a/Makefile	2009-04-14 13:46:01.000000000 +0200
+++ vdr-1.7.5b/Makefile	2009-04-14 13:51:40.000000000 +0200
@@ -20,7 +20,7 @@
 MANDIR   = $(PREFIX)/share/man
 BINDIR   = $(PREFIX)/bin
 LOCDIR   = ./locale
-LIBS     = -ljpeg -lpthread -ldl -lcap -lrt -lfreetype -lfontconfig
+LIBS     = -ljpeg -lpthread -ldl -lrt -lfreetype -lfontconfig
 INCLUDES = -I/usr/include/freetype2
 
 PLUGINDIR= ./PLUGINS
diff -Naur vdr-1.7.5a/vdr.c vdr-1.7.5b/vdr.c
--- vdr-1.7.5a/vdr.c	2009-04-14 13:45:27.000000000 +0200
+++ vdr-1.7.5b/vdr.c	2009-04-14 13:53:53.000000000 +0200
@@ -32,7 +32,6 @@
 #include <pwd.h>
 #include <signal.h>
 #include <stdlib.h>
-#include <sys/capability.h>
 #include <sys/prctl.h>
 #include <termios.h>
 #include <unistd.h>
@@ -114,28 +113,13 @@
 
 static bool DropCaps(void)
 {
-  // drop all capabilities except selected ones
-  cap_t caps = cap_from_text("= cap_sys_nice,cap_sys_time=ep");
-  if (!caps) {
-     fprintf(stderr, "vdr: cap_from_text failed: %s\n", strerror(errno));
-     return false;
-     }
-  if (cap_set_proc(caps) == -1) {
-     fprintf(stderr, "vdr: cap_set_proc failed: %s\n", strerror(errno));
-     cap_free(caps);
-     return false;
-     }
-  cap_free(caps);
+  // capabilites support removed
   return true;
 }
 
 static bool SetKeepCaps(bool On)
 {
-  // set keeping capabilities during setuid() on/off
-  if (prctl(PR_SET_KEEPCAPS, On ? 1 : 0, 0, 0, 0) != 0) {
-     fprintf(stderr, "vdr: prctl failed\n");
-     return false;
-     }
+  // capabilites support removed
   return true;
 }
 
