diff -Naur vdr-1.7.7/config.c vdr-1.7.7a/config.c
--- vdr-1.7.7/config.c	2009-05-17 14:38:41.000000000 +0200
+++ vdr-1.7.7a/config.c	2009-05-17 14:39:52.000000000 +0200
@@ -367,7 +367,7 @@
   DelTimeshiftRec = 0;
 #endif /* DELTIMESHIFTREC */
   MinEventTimeout = 30;
-  MinUserInactivity = 300;
+  MinUserInactivity = 0;
   NextWakeupTime = 0;
   MultiSpeedMode = 0;
   ShowReplayMode = 0;
@@ -736,7 +736,6 @@
   else if (!strcasecmp(Name, "DelTimeshiftRec"))     DelTimeshiftRec    = atoi(Value);
 #endif /* DELTIMESHIFTREC */
   else if (!strcasecmp(Name, "MinEventTimeout"))     MinEventTimeout    = atoi(Value);
-  else if (!strcasecmp(Name, "MinUserInactivity"))   MinUserInactivity  = atoi(Value);
   else if (!strcasecmp(Name, "NextWakeupTime"))      NextWakeupTime     = atoi(Value);
   else if (!strcasecmp(Name, "MultiSpeedMode"))      MultiSpeedMode     = atoi(Value);
   else if (!strcasecmp(Name, "ShowReplayMode"))      ShowReplayMode     = atoi(Value);
@@ -974,7 +973,6 @@
   Store("DelTimeshiftRec",    DelTimeshiftRec);
 #endif /* DELTIMESHIFTREC */
   Store("MinEventTimeout",    MinEventTimeout);
-  Store("MinUserInactivity",  MinUserInactivity);
   Store("NextWakeupTime",     NextWakeupTime);
 #ifdef USE_DDEPGENTRY
   Store("DoubleEpgAction",    DoubleEpgAction);
diff -Naur vdr-1.7.7/menu.c vdr-1.7.7a/menu.c
--- vdr-1.7.7/menu.c	2009-05-17 14:38:41.000000000 +0200
+++ vdr-1.7.7a/menu.c	2009-05-17 14:39:06.000000000 +0200
@@ -3831,7 +3831,6 @@
   Clear();
 #endif /* VOLCTRL */
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Min. event timeout (min)"),   &data.MinEventTimeout));
-  Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Min. user inactivity (min)"), &data.MinUserInactivity));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$SVDRP timeout (s)"),          &data.SVDRPTimeout));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Zap timeout (s)"),            &data.ZapTimeout));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Channel entry timeout (ms)"), &data.ChannelEntryTimeout, 0));
