diff -Naur vdr-1.7.9/Makefile vdr-1.7.9a/Makefile
--- vdr-1.7.9/Makefile	2009-10-03 08:53:31.000000000 +0200
+++ vdr-1.7.9a/Makefile	2009-10-03 08:53:59.000000000 +0200
@@ -21,7 +21,7 @@
 BINDIR   = $(PREFIX)/bin
 LOCDIR   = ./locale
 LIBS     = -ljpeg -lpthread -ldl -lrt -lfreetype -lfontconfig
-INCLUDES = -I/usr/include/freetype2
+INCLUDES =
 
 PLUGINDIR= ./PLUGINS
 PLUGINLIBDIR= $(PLUGINDIR)/lib
