diff -Naur vdr-1.7.9/tools.c vdr-1.7.9.patch/tools.c
--- vdr-1.7.9/tools.c	2009-05-31 13:43:24.000000000 +0200
+++ vdr-1.7.9.patch/tools.c	2009-10-06 22:12:39.421294729 +0200
@@ -484,7 +484,8 @@
 {
   if (!FileName)
      return NULL;
-  char *TargetName = canonicalize_file_name(FileName);
+  char *ResolvedName = (char*) malloc(1024);
+  char *TargetName = realpath(FileName, ResolvedName);
   if (!TargetName) {
      if (errno == ENOENT) // file doesn't exist
         TargetName = strdup(FileName);
