diff -Naur vdr-1.7.9/thread.c vdr-1.7.9.patch/thread.c
--- vdr-1.7.9/thread.c	2009-04-13 15:50:39.000000000 +0200
+++ vdr-1.7.9.patch/thread.c	2009-10-06 22:10:06.587419872 +0200
@@ -160,10 +160,17 @@
      if (!GetAbsTime(&abstime, TimeoutMs))
         TimeoutMs = 0;
      }
-  if (Write)
-     Result = TimeoutMs ? pthread_rwlock_timedwrlock(&rwlock, &abstime) : pthread_rwlock_wrlock(&rwlock);
-  else
-     Result = TimeoutMs ? pthread_rwlock_timedrdlock(&rwlock, &abstime) : pthread_rwlock_rdlock(&rwlock);
+
+  while (true) {
+    if (Write)
+       Result = pthread_rwlock_wrlock(&rwlock);
+    else
+       Result = pthread_rwlock_rdlock(&rwlock);
+    if ( Result == 0 | TimeoutMs == 0 )
+      break;
+    TimeoutMs--;
+    usleep(1000);
+  }
   return Result == 0;
 }
 
