From 973c66c3d4e7418e53932c719b4cb3345b38da9e Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 12 Oct 2013 21:40:12 +0300
Subject: [PATCH] decrease channels.conf autosave delay

---
 vdr.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vdr.c b/vdr.c
index c63eeca..1fb0c57 100644
--- a/vdr.c
+++ b/vdr.c
@@ -74,7 +74,7 @@
 #define SHUTDOWNCANCELPROMPT   5 // seconds to wait in user prompt to allow canceling shutdown
 #define RESTARTCANCELPROMPT    5 // seconds to wait in user prompt before restarting on SIGHUP
 #define MANUALSTART          600 // seconds the next timer must be in the future to assume manual start
-#define CHANNELSAVEDELTA     600 // seconds before saving channels.conf after automatic modifications
+#define CHANNELSAVEDELTA     120 // seconds before saving channels.conf after automatic modifications
 #define DEVICEREADYTIMEOUT    30 // seconds to wait until all devices are ready
 #define MENUTIMEOUT          120 // seconds of user inactivity after which an OSD display is closed
 #define TIMERCHECKDELTA       10 // seconds between checks for timers that need to see their channel
-- 
1.7.2.5

