diff -Naur vdr-1.7.9/dvbdevice.h vdr-1.7.9.patch/dvbdevice.h
--- vdr-1.7.9/dvbdevice.h	2009-06-06 13:16:47.000000000 +0200
+++ vdr-1.7.9.patch/dvbdevice.h	2009-10-06 20:36:09.827294633 +0200
@@ -16,8 +16,8 @@
 #include "device.h"
 #include "dvbspu.h"
 
-#if DVB_API_VERSION != 5 || DVB_API_VERSION_MINOR != 0
-#error VDR requires Linux DVB driver API version 5.0!
+#if DVB_API_VERSION != 5
+#error VDR requires Linux DVB driver API version 5.0 or higher!
 #endif
 
 #define MAXDVBDEVICES  8
