From 70ba5ad47b5bb49216c30d5321948d33dd7a3cfe Mon Sep 17 00:00:00 2001
From: Daniel Scheller <d.scheller@gmx.net>
Date: Wed, 14 Nov 2012 23:23:26 +0100
Subject: [PATCH] testing: disable all connection notifications

---
 script.xbmc.lcd/lcdmain.py |   36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/script.xbmc.lcd/lcdmain.py b/script.xbmc.lcd/lcdmain.py
index 585f87d..15a7e3f 100644
--- a/script.xbmc.lcd/lcdmain.py
+++ b/script.xbmc.lcd/lcdmain.py
@@ -125,26 +125,26 @@ def handleConnectLCD():
 
     ret = g_lcdproc.Initialize()
 
-    if not ret:
-      count = 10
-      if not g_failedConnectionNotified:
-        g_failedConnectionNotified = True
-	g_initialConnectAttempt = False
-        text = __settings__.getLocalizedString(500)
-        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
-      while (not xbmc.abortRequested) and (count > 0):
-        time.sleep(1)
-        count -= 1
-	ret = False
-    else:
-      text = __settings__.getLocalizedString(501)
-      if not g_failedConnectionNotified and not g_initialConnectAttempt:
-        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
-        g_failedConnectionNotified = True
-      break
+    #if not ret:
+    #  count = 10
+    #  if not g_failedConnectionNotified:
+    #    g_failedConnectionNotified = True
+    #    g_initialConnectAttempt = False
+    #    text = __settings__.getLocalizedString(500)
+    #    xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
+    #  while (not xbmc.abortRequested) and (count > 0):
+    #    time.sleep(1)
+    #    count -= 1
+    #    ret = False
+    #else:
+    #  text = __settings__.getLocalizedString(501)
+    #  if not g_failedConnectionNotified and not g_initialConnectAttempt:
+    #    xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
+    #    g_failedConnectionNotified = True
+    #  break
 
   # initial connection attempt done, update flag
-  g_initialConnectAttempt = False
+  # g_initialConnectAttempt = False
   return ret
 
 #MAIN - entry point
-- 
1.7.10

