diff -Naur script.xbmc.lcd-dc08f6e/script.xbmc.lcd/lcdmain.py script.xbmc.lcd-dc08f6e.patch/script.xbmc.lcd/lcdmain.py
--- script.xbmc.lcd-dc08f6e/script.xbmc.lcd/lcdmain.py	2012-01-22 19:49:39.000000000 +0100
+++ script.xbmc.lcd-dc08f6e.patch/script.xbmc.lcd/lcdmain.py	2012-01-23 10:58:54.317705282 +0100
@@ -133,10 +133,10 @@
         count -= 1
       if not g_failedConnectionNotified:
         g_failedConnectionNotified = True
-        text = __settings__.getLocalizedString(500)
+#        text = __settings__.getLocalizedString(500)
         xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
     else:
-      text = __settings__.getLocalizedString(501)
+#      text = __settings__.getLocalizedString(501)
       if not g_failedConnectionNotified:
         xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
         g_failedConnectionNotified = True
