diff -Naur script.xbmc.lcd-dc08f6e/script.xbmc.lcd/lcdmain.py script.xbmc.lcd-dc08f6e.patch/script.xbmc.lcd/lcdmain.py
--- script.xbmc.lcd-dc08f6e/script.xbmc.lcd/lcdmain.py	2012-01-22 19:49:39.000000000 +0100
+++ script.xbmc.lcd-dc08f6e.patch/script.xbmc.lcd/lcdmain.py	2012-01-28 19:34:23.222961732 +0100
@@ -134,11 +134,11 @@
       if not g_failedConnectionNotified:
         g_failedConnectionNotified = True
         text = __settings__.getLocalizedString(500)
-        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
+#        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
     else:
       text = __settings__.getLocalizedString(501)
       if not g_failedConnectionNotified:
-        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
+#        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,text,10,__icon__))
         g_failedConnectionNotified = True
       print "lcd: connected to LCD"
       break
