From 48056f2eb5ed03be0589bbe335dcdaa676604b28 Mon Sep 17 00:00:00 2001
From: theirix <theirix@gmail.com>
Date: Sat, 31 Oct 2020 11:57:33 +0300
Subject: [PATCH 1/2] Add CMake option DISABLE_SDL

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9f64fe6..c836a9a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,12 @@ include (${CMAKE_ROOT}/Modules/CheckIncludeFile.cmake)
 include(GNUInstallDirs)
 include(CheckFunctionExists)
 
-find_package(SDL)
+option(DISABLE_SDL "Disable SDL" OFF)
+
+if (NOT DISABLE_SDL)
+	find_package(SDL)
+endif()
+
 find_package(Threads REQUIRED)
 
 CHECK_INCLUDE_FILE(malloc.h HAVE_MALLOC_H)

From c754a0342fe9bbd74bbb90d7244777cb6370804a Mon Sep 17 00:00:00 2001
From: theirix <theirix@gmail.com>
Date: Mon, 2 Nov 2020 19:03:05 +0300
Subject: [PATCH 2/2] Use ENABLE_SDL instead of DISABLE_SDL

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c836a9a9..ac9d3ce8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,9 @@ include (${CMAKE_ROOT}/Modules/CheckIncludeFile.cmake)
 include(GNUInstallDirs)
 include(CheckFunctionExists)
 
-option(DISABLE_SDL "Disable SDL" OFF)
+option(ENABLE_SDL "Enable SDL" ON)
 
-if (NOT DISABLE_SDL)
+if (ENABLE_SDL)
 	find_package(SDL)
 endif()
 
