diff -uNr a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2026-01-23 19:21:37.000000000 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts	2026-02-08 22:35:35.635195476 +0900
@@ -342,10 +342,18 @@
 	};
 };
 
+&hdmi0_sound {
+	status = "okay";
+};
+
 &hdptxphy0 {
 	status = "okay";
 };
 
+&i2s5_8ch {
+	status = "okay";
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		/* RTL8211F */
