diff -Naur linux-cuboxi-3.14-aab1674/drivers/media/usb/dvb-usb-v2/rtl28xxu.c linux-cuboxi-3.14-aab1674.patch/drivers/media/usb/dvb-usb-v2/rtl28xxu.c
--- linux-cuboxi-3.14-aab1674/drivers/media/usb/dvb-usb-v2/rtl28xxu.c	2014-08-26 14:31:42.000000000 +0200
+++ linux-cuboxi-3.14-aab1674.patch/drivers/media/usb/dvb-usb-v2/rtl28xxu.c	2014-08-26 18:23:05.035432084 +0200
@@ -1441,6 +1441,8 @@
 		&rtl2832u_props, "Sveon STV20", NULL) },
 	{ DVB_USB_DEVICE(USB_VID_KWORLD_2, USB_PID_SVEON_STV27,
 		&rtl2832u_props, "Sveon STV27", NULL) },
+	{ DVB_USB_DEVICE(USB_VID_GTEK, 0xa803,
+		&rtl2832u_props, "Realtek RTL2832U reference design", NULL) },
 
 	/* RTL2832P devices: */
 	{ DVB_USB_DEVICE(USB_VID_HANFTEK, 0x0131,
