From 063c492b89cdbb8739d04b5190105cf2c9e12e6f Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime.ripard@free-electrons.com>
Date: Tue, 29 Nov 2016 10:11:12 +0100
Subject: [PATCH 1/4] makefile: Add install target and build the module by
 default

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
---
 src/devicedrv/mali/Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/devicedrv/mali/Makefile b/src/devicedrv/mali/Makefile
index 44c7bb83981a..2fee5e096b51 100755
--- a/src/devicedrv/mali/Makefile
+++ b/src/devicedrv/mali/Makefile
@@ -177,13 +177,18 @@ ifeq ($(MALI_MEM_SWAP_TRACKING),1)
 EXTRA_DEFINES += -DMALI_MEM_SWAP_TRACKING=1
 endif
 
-all: $(UMP_SYMVERS_FILE)
+modules: $(UMP_SYMVERS_FILE)
 	$(MAKE) ARCH=$(ARCH) -C $(KDIR) M=$(CURDIR) modules
 	@rm $(FILES_PREFIX)__malidrv_build_info.c $(FILES_PREFIX)__malidrv_build_info.o
 
+all: modules
+
 clean:
 	$(MAKE) ARCH=$(ARCH) -C $(KDIR) M=$(CURDIR) clean
 
+install: modules
+	$(MAKE) ARCH=$(ARCH) -C $(KDIR) M=$(CURDIR) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install
+
 kernelrelease:
 	$(MAKE) ARCH=$(ARCH) -C $(KDIR) kernelrelease
 
-- 
2.10.2

