diff --git a/bsnes/GNUmakefile b/bsnes/GNUmakefile
index 23efe065..7857157c 100644
--- a/bsnes/GNUmakefile
+++ b/bsnes/GNUmakefile
@@ -1,9 +1,9 @@
 target := bsnes
 binary := application
 build := performance
-openmp := true
-local := true
-flags += -I. -I..
+openmp := false
+local := false
+flags += -I. -I.. -Wno-narrowing -Wno-multichar -g -fPIC ${CXXFLAGS}
 
 # in order for this to work, obj/lzma.o must be omitted or bsnes will hang on startup.
 # further, only the X-Video driver works reliably. OpenGL 3.2, OpenGL 2.0, and XShm crash bsnes.
diff --git a/bsnes/target-libretro/GNUmakefile b/bsnes/target-libretro/GNUmakefile
index b0610c81..b76b4411 100644
--- a/bsnes/target-libretro/GNUmakefile
+++ b/bsnes/target-libretro/GNUmakefile
@@ -1,6 +1,6 @@
 name := bsnes_libretro
-local := false
-openmp := true
+openmp := false
+local := false
 flags += -Wno-narrowing -Wno-multichar -g -fPIC
 
 objects := libretro $(objects)
@@ -12,9 +8,9 @@ obj/libretro.o: target-libretro/libretro.cpp
 
 all: $(objects)
 ifeq ($(platform),linux)
-	$(strip $(compiler) -o out/$(name).so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/$(name).so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),windows)
-	$(strip $(compiler) -o out/$(name).dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/$(name).dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),macos)
 	$(strip $(compiler) -o out/bsnes_hd_libretro.dylib -shared $(objects) -lpthread -ldl)
 else ifeq ($(platform), ios-arm64)
