--- /drivers/net/wireless/bcmdhd/dhd_linux.c
+++ /drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -4320,13 +4320,17 @@
 	return DHD_FLAG_STA_MODE;
 }
 
-#define BCM4330_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm4330_bg.bin"
-#define BCM4330_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm4330_apsta_bg.bin"
-#define BCM4330_NVRAM_NAME	"/lib/firmware/brcm/brcmfmac4330-sdio.txt"
+#define BCM4329_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4329-sdio.bin"
+#define BCM4329_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4329-sdio.bin"
+#define BCM4329_NVRAM_NAME	"/lib/firmware/brcm/brcmfmac4329-sdio.txt"
+
+#define BCM4330_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4330-sdio.bin"
+#define BCM4330_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4330-sdio.bin"
+#define BCM4330_NVRAM_NAME	"/lib/firmware/brcm/brcmfmac4330-sdio.wand,imx6q-wandboard.txt"
 /* Ampak AP6335 */
-#define BCM4339_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm4339a0_ag.bin"
-#define BCM4339_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm4339a0_ag_apsta.bin"
-#define BCM4339_NVRAM_NAME	"/lib/firmware/brcm/nvram_ap6335.txt"
+#define BCM4339_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4339-sdio.bin"
+#define BCM4339_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/brcmfmac4339-sdio.bin"
+#define BCM4339_NVRAM_NAME	"/lib/firmware/brcm/brcmfmac4339-sdio.txt"
 /* Ampak AP6212 */
 #define BCM43430_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm43438a0.bin"
 #define BCM43430_AP_MODE_FIRMWARE_NAME	"/lib/firmware/brcm/fw_bcm43438a0_apsta.bin"
@@ -4344,6 +4348,7 @@
 
 
 static const struct dhd_firmware_names dhd_fwname_data[] = {
+{ BCM4329_CHIP_ID, DHD_FIRMWARE_NVRAM(BCM4329) },
 { BCM4330_CHIP_ID, DHD_FIRMWARE_NVRAM(BCM4330) },
 { BCM4339_CHIP_ID, DHD_FIRMWARE_NVRAM(BCM4339) },
 { BCM43430_CHIP_ID, DHD_FIRMWARE_NVRAM(BCM43430) }
