--- /arch/arm/boot/dts/imx6dl-wandboard-revc1.dts
+++ /arch/arm/boot/dts/imx6dl-wandboard-revc1.dts
@@ -38,7 +38,7 @@
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
 			regulator-name = "wlreg_on";
-			gpio = <&gpio6 0 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio1 26 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
 	};
@@ -52,9 +52,9 @@
 			dir   = "out";
 		};
 
-		wl_reg {
-			label = "wl_reg";
-			gpios = <&gpio1 26 1>;
+		bluetooth-reg-on {
+			label = "bluetooth-reg-on";
+			gpios = <&gpio5 30 1>;
 			dir   = "out";
 		};
 
@@ -66,7 +66,7 @@
 
 		bluetooth-wake {
 			label = "bluetooth-wake";
-			gpios = <&gpio5 30 1>;
+			gpios = <&gpio1 30 1>;
 			dir   = "in";
 		};
 
@@ -114,7 +114,6 @@
 
 		pinctrl_wifi_ctrl: wifi_ctrlgrp {
 			fsl,pins = <
-				MX6QDL_PAD_CSI0_DAT14__GPIO6_IO00 0x000b0			/* wifi-reset */
 				MX6QDL_PAD_CSI0_DAT13__GPIO5_IO31 0x000b0			/* wifi-on */
 				MX6QDL_PAD_ENET_RXD1__GPIO1_IO26  0x000b0			/* wifi-reg-on */
 				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29  0x80000000		/* wifi-host-wake */
@@ -124,8 +123,9 @@
 
 		pinctrl_bt_ctrl: bt_ctrlgrp {
 			fsl,pins = <
-				MX6QDL_PAD_CSI0_VSYNC__GPIO5_IO21	0x000b0		/* bluetooth-reg-on */
-				MX6QDL_PAD_CSI0_DAT12__GPIO5_IO30	0x80000000	/* bluetooth-wake */
+				MX6QDL_PAD_CSI0_VSYNC__GPIO5_IO21	0x000b0		/* bluetooth-reset */
+				MX6QDL_PAD_CSI0_DAT12__GPIO5_IO30	0x000b0		/* bluetooth-reg-on */
+				MX6QDL_PAD_ENET_TXD0__GPIO1_IO30	0x80000000	/* bluetooth-wake */
 				MX6QDL_PAD_CSI0_DATA_EN__GPIO5_IO20	0x80000000	/* bluetooth-host-wake */
 			>;
 		};
