diff -Naur gst-plugins-bad-0.10.11.2/ext/alsaspdif/alsaspdifsink.c gst-plugins-bad-0.10.11.2.patch/ext/alsaspdif/alsaspdifsink.c
--- gst-plugins-bad-0.10.11.2/ext/alsaspdif/alsaspdifsink.c	2009-01-21 11:15:41.000000000 +0100
+++ gst-plugins-bad-0.10.11.2.patch/ext/alsaspdif/alsaspdifsink.c	2009-05-12 02:58:42.000000000 +0200
@@ -313,8 +313,8 @@
 
   result = gst_util_uint64_scale_int (samples, GST_SECOND, sink->rate);
   GST_LOG_OBJECT (sink,
-      "Samples raw: %d, delay: %d, real: %d, Time: %" GST_TIME_FORMAT, raw,
-      delay, samples, GST_TIME_ARGS (result));
+      "Samples raw: %d, delay: %d, real: %d, Time: %" GST_TIME_FORMAT, (int)raw,
+      (int)delay, (int)samples, GST_TIME_ARGS (result));
   return result;
 }
 
diff -Naur gst-plugins-bad-0.10.11.2/ext/resindvd/gstmpegdemux.c gst-plugins-bad-0.10.11.2.patch/ext/resindvd/gstmpegdemux.c
--- gst-plugins-bad-0.10.11.2/ext/resindvd/gstmpegdemux.c	2009-05-11 23:18:13.000000000 +0200
+++ gst-plugins-bad-0.10.11.2.patch/ext/resindvd/gstmpegdemux.c	2009-05-12 03:04:47.000000000 +0200
@@ -1207,7 +1207,7 @@
     /* We keep the offset of this scr */
     demux->last_scr_offset = demux->adapter_offset + 12;
 
-    GST_DEBUG_OBJECT (demux, "SCR: 0x%08x SCRE: 0x%08x", scr, scr_ext);
+    GST_DEBUG_OBJECT (demux, "SCR: 0x%08x SCRE: 0x%08x", (unsigned int)scr, scr_ext);
 
     if (scr_ext) {
       scr = (scr * 300 + scr_ext % 300) / 300;
