diff -Naur linux-Switch-5.1.0/nvidia/platform/t210/nx/kernel-dts/nx-platforms/tegra210-nx-keys.dtsi linux-Switch-5.1.0-2/nvidia/platform/t210/nx/kernel-dts/nx-platforms/tegra210-nx-keys.dtsi
--- linux-Switch-5.1.0/nvidia/platform/t210/nx/kernel-dts/nx-platforms/tegra210-nx-keys.dtsi	2023-04-12 07:00:12.773160197 +0200
+++ linux-Switch-5.1.0-2/nvidia/platform/t210/nx/kernel-dts/nx-platforms/tegra210-nx-keys.dtsi	2023-04-12 07:02:08.027831616 +0200
@@ -37,14 +37,14 @@
 		volume_down {
 			label = "Volume Down";
 			gpios = <&gpio TEGRA_GPIO(X, 7) GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_VOLUMEDOWN>;
+			linux,code = <KEY_KPMINUS>;
 			debounce-interval = <16>;
 		};
 
 		volume_up {
 			label = "Volume Up";
 			gpios = <&gpio TEGRA_GPIO(X, 6) GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_VOLUMEUP>;
+			linux,code = <KEY_KPPLUS>;
 			debounce-interval = <16>;
 		};
 	};
