diff -Naur linux-Switch/nvidia/platform/t210/icosa/kernel-dts/icosa-platforms/tegra210-icosa-sensor.dtsi linux-Switch-2/nvidia/platform/t210/icosa/kernel-dts/icosa-platforms/tegra210-icosa-sensor.dtsi
--- linux-Switch/nvidia/platform/t210/icosa/kernel-dts/icosa-platforms/tegra210-icosa-sensor.dtsi	2021-10-16 22:41:32.935142657 -0700
+++ linux-Switch-2/nvidia/platform/t210/icosa/kernel-dts/icosa-platforms/tegra210-icosa-sensor.dtsi	2021-10-16 22:56:56.393018162 -0700
@@ -20,7 +20,7 @@
 	i2c@7000c400 {
 		bh1730fvc@29 {
 			/* Only enable on distros with proper auto-brightness support */
-			status = "disabled";
+			status = "okay";
 
 			compatible = "rohm,bh1730fvc";
 			reg = <0x29>;
