diff -Naur linux-Switch-5.1.2/nvidia/drivers/video/tegra/Kconfig linux-Switch-5.1.2-2/nvidia/drivers/video/tegra/Kconfig
--- linux-Switch-5.1.2/nvidia/drivers/video/tegra/Kconfig	2024-01-29 05:57:16.331738434 +0100
+++ linux-Switch-5.1.2-2/nvidia/drivers/video/tegra/Kconfig	2024-01-29 07:21:54.862883090 +0100
@@ -231,7 +231,7 @@
 
 config TEGRA_DP_BRANCH_STDP2550
 	bool "Enable DP branch STDP2550/RTD2172N support."
-	depends on TEGRA_DP && MEDIA_CEC
+	depends on TEGRA_DP && MEDIA_CEC_SUPPORT
 	default n
 	help
 	  Say Y here to enable the DP branch MegaChips STDP2550 and Realtek
