diff -Naur linux-Switch/nvidia/drivers/video/tegra/dc/dp_branch.c linux-Switch-2/nvidia/drivers/video/tegra/dc/dp_branch.c
--- linux-Switch/nvidia/drivers/video/tegra/dc/dp_branch.c	2022-01-05 18:00:36.325665889 +0100
+++ linux-Switch-2/nvidia/drivers/video/tegra/dc/dp_branch.c	2022-01-05 18:17:06.605838285 +0100
@@ -66,6 +66,15 @@
 	tegra_dc_dp_dpcd_write(dp, NV_DPCD_BRANCH_STDP_CEC_RX_INFO,
 			       dpcd_data);
 
+
+
+	printk(KERN_ALERT "dp: cec: rx data: ");
+	for (i = 0; i < msg_len; i++)
+		printk(KERN_CONT "%02X ", msg[i]);
+	printk(KERN_CONT "\n");
+
+
+
 	return msg_len;
 }
 
@@ -332,6 +341,11 @@
 		cec_transmit_done(data->adap, CEC_TX_STATUS_OK,
 				  0, 0, 0, 0);
 	}
+
+	printk(KERN_ALERT "dp: cec: tx data: ");
+	for (i = 0; i < data->cec_tx_len; i++)
+		printk(KERN_CONT "%02X ", data->cec_tx_buf[i]);
+	printk(KERN_CONT " (tries: %d/%d)\n", i, data->cec_tx_attempts);
 }
 
 static void dpb_stm_cec_rx_worker(struct work_struct *work)
