diff -Naur linux-Switch/nvidia/platform/t210/icosa/kernel-dts/tegra210-icosa-common.dtsi linux-Switch-2/nvidia/platform/t210/icosa/kernel-dts/tegra210-icosa-common.dtsi
--- linux-Switch/nvidia/platform/t210/icosa/kernel-dts/tegra210-icosa-common.dtsi	2022-10-14 16:26:22.015884735 +0200
+++ linux-Switch-2/nvidia/platform/t210/icosa/kernel-dts/tegra210-icosa-common.dtsi	2022-10-14 16:28:10.917657052 +0200
@@ -86,7 +86,6 @@
 	};
 
 	pmc@7000e400 {
-		/delete-property/ nvidia,secure-pmc;
 		nvidia,invert-interrupt;
 		nvidia,suspend-mode = <0>;
 		nvidia,cpu-pwr-good-time = <0>;
