diff -Naur gmp-4.2.3.orig/configure gmp-4.2.3/configure
--- gmp-4.2.3.orig/configure	2008-08-05 13:56:41.000000000 +0200
+++ gmp-4.2.3/configure	2008-08-05 13:57:06.000000000 +0200
@@ -1584,22 +1584,7 @@
 
 
 
-# If --target is not used then $target_alias is empty, but if say
-# "./configure athlon-pc-freebsd3.5" is used, then all three of
-# $build_alias, $host_alias and $target_alias are set to
-# "athlon-pc-freebsd3.5".
-#
-if test -n "$target_alias" && test "$target_alias" != "$host_alias"; then
-  { { echo "$as_me:$LINENO: error: --target is not appropriate for GMP
-Use --build=CPU-VENDOR-OS if you need to specify your CPU and/or system
-explicitly.  Use --host if cross-compiling (see \"Installing GMP\" in the
-manual for more on this)." >&5
-echo "$as_me: error: --target is not appropriate for GMP
-Use --build=CPU-VENDOR-OS if you need to specify your CPU and/or system
-explicitly.  Use --host if cross-compiling (see \"Installing GMP\" in the
-manual for more on this)." >&2;}
-   { (exit 1); exit 1; }; }
-fi
+
 
 gmp_configm4="config.m4"
 gmp_tmpconfigm4=cnfm4.tmp
