diff -Naur nfs-utils-1.1.0-old/tools/locktest/Makefile.am nfs-utils-1.1.0-new/tools/locktest/Makefile.am
--- nfs-utils-1.1.0-old/tools/locktest/Makefile.am	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/tools/locktest/Makefile.am	2007-08-11 19:18:59.000000000 -0700
@@ -5,8 +5,8 @@
 
 noinst_PROGRAMS = testlk
 testlk_SOURCES = testlk.c
-testlk_CFLAGS=$(CFLAGS_FOR_BUILD)
-testlk_CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
-testlk_LDFLAGS=$(LDFLAGS_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 
 MAINTAINERCLEANFILES = Makefile.in
diff -Naur nfs-utils-1.1.0-old/tools/rpcdebug/Makefile.am nfs-utils-1.1.0-new/tools/rpcdebug/Makefile.am
--- nfs-utils-1.1.0-old/tools/rpcdebug/Makefile.am	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/tools/rpcdebug/Makefile.am	2007-08-11 19:19:37.000000000 -0700
@@ -8,8 +8,8 @@
 
 sbin_PROGRAMS = rpcdebug
 rpcdebug_SOURCES = rpcdebug.c
-rpcdebug_CFLAGS=$(CFLAGS_FOR_BUILD)
-rpcdebug_CPPFLAGS=$(CPPFLAGS_FOR_BUILD) -I$(top_srcdir)/support/include
-rpcdebug_LDFLAGS=$(LDFLAGS_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPFLAGS=$(CPPFLAGS_FOR_BUILD) -I$(top_srcdir)/support/include
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 
 MAINTAINERCLEANFILES = Makefile.in
diff -Naur nfs-utils-1.1.0-old/tools/rpcgen/Makefile.am nfs-utils-1.1.0-new/tools/rpcgen/Makefile.am
--- nfs-utils-1.1.0-old/tools/rpcgen/Makefile.am	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0-new/tools/rpcgen/Makefile.am	2007-08-11 19:19:57.000000000 -0700
@@ -9,9 +9,9 @@
 		 rpc_util.c rpc_sample.c rpc_output.h rpc_parse.h \
 		 rpc_scan.h rpc_util.h
 
-rpcgen_CFLAGS=$(CFLAGS_FOR_BUILD)
-rpcgen_CPPLAGS=$(CPPFLAGS_FOR_BUILD)
-rpcgen_LDFLAGS=$(LDFLAGS_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPLAGS=$(CPPFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 
 MAINTAINERCLEANFILES = Makefile.in
 
