Fix USE_SCROLL_HINTS/USE_HASHMAP logic

--- a/ncurses/curses.priv.h	2018-12-16 02:16:58.000000000 +0100
+++ a/ncurses/curses.priv.h	2021-05-30 00:12:14.086528124 +0200
@@ -258,10 +258,9 @@ extern NCURSES_EXPORT(void *) _nc_memmov
 /*
  * Scroll hints are useless when hashmap is used
  */
-#if !USE_SCROLL_HINTS
-#if !USE_HASHMAP
-#define USE_SCROLL_HINTS 1
-#else
+#if USE_SCROLL_HINTS
+#if USE_HASHMAP
+#undef USE_SCROLL_HINTS
 #define USE_SCROLL_HINTS 0
 #endif
 #endif
