Fix freeing not allocated fallback entries by allocating a copy.

--- a/ncurses/tinfo/tinfo_driver.c	2018-11-24 23:17:03.000000000 +0100
+++ b/ncurses/tinfo/tinfo_driver.c	2018-12-29 10:27:19.000000000 +0100
@@ -180,6 +180,8 @@ drv_CanHandle(TERMINAL_CONTROL_BLOCK * T
     if (status != TGETENT_YES) {
 	const TERMTYPE2 *fallback = _nc_fallback2(tname);
 
+	if (fallback)
+            fallback = _nc_copy_entry(fallback);
 	if (fallback) {
 	    T(("found fallback entry"));
 	    TerminalType(termp) = *fallback;
