diff --git a/Makefile b/Makefile
index c7f3daa..a781624 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
-PREFIX = /usr/local
+# PREFIX = /usr/local
 BINDIR = $(PREFIX)/bin
 MANDIR = $(PREFIX)/share/man
-CFLAGS = -Wall -O2 -DHAVE_FDATASYNC=1 -DHAVE_SETXATTR=1
-LDFLAGS = 
+CFLAGS += -Wall -O2 -DHAVE_FDATASYNC=1
+LDFLAGS += 
 
 FUSE_CFLAGS = $(shell pkg-config --cflags fuse)
 FUSE_LIBS = $(shell pkg-config --libs fuse)
@@ -10,13 +10,15 @@ FUSE_LIBS = $(shell pkg-config --libs fuse)
 PCRE_CFLAGS = $(shell pkg-config --cflags libpcre)
 PCRE_LIBS = $(shell pkg-config --libs libpcre)
 
+CC=$(CROSS_COMPILE)gcc
+
 all: rewritefs
 
 rewritefs: rewritefs.o rewrite.o
-	gcc rewritefs.o rewrite.o $(FUSE_LIBS) $(PCRE_LIBS) $(LDFLAGS) -o $@
+	$(CC) rewritefs.o rewrite.o $(FUSE_LIBS) $(PCRE_LIBS) $(LDFLAGS) -o $@
 
 %.o: %.c
-	gcc $(CFLAGS) $(FUSE_CFLAGS) $(PCRE_CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(FUSE_CFLAGS) $(PCRE_CFLAGS) -c $< -o $@
 
 clean:
 	rm -f rewritefs *.o
