diff -Nur shared-mime-info-2.1.orig/data/meson.build shared-mime-info-2.1/data/meson.build
--- shared-mime-info-2.1.orig/data/meson.build	2021-01-01 06:12:16.000000000 +1100
+++ shared-mime-info-2.1/data/meson.build	2021-01-14 00:25:58.299790452 +1100
@@ -1,6 +1,4 @@
 
-install_man('update-mime-database.1')
-
 freedesktop_org_xml = i18n.merge_file(
     input: 'freedesktop.org.xml.in',
     output: 'freedesktop.org.xml',
@@ -15,15 +13,3 @@
   [ 'its/shared-mime-info.loc', 'its/shared-mime-info.its', ],
   install_dir : get_option('datadir') / 'gettext/its'
 )
-
-custom_target('shared-mime-info-spec-html',
-    input : 'shared-mime-info-spec.xml',
-    output: 'shared-mime-info-spec-html',
-    command: [
-        xmlto,
-        '-o', '@OUTPUT@',
-        'html-nochunks',
-        '@INPUT@',
-    ],
-    build_by_default: true,
-)
diff -Nur shared-mime-info-2.1.orig/meson.build shared-mime-info-2.1/meson.build
--- shared-mime-info-2.1.orig/meson.build	2021-01-01 06:12:16.000000000 +1100
+++ shared-mime-info-2.1/meson.build	2021-01-14 00:24:18.552142696 +1100
@@ -22,7 +22,6 @@
 
 itstool = find_program('itstool')
 xmllint = find_program('xmllint')
-xmlto   = find_program('xmlto')
 
 ###############################################################################
 # Find xdgmime
