From 553a75b739184397de8401758d99c3fe6fa93e79 Mon Sep 17 00:00:00 2001
From: perillamint <perillamint@quendi.moe>
Date: Tue, 12 Jun 2018 11:38:12 +0900
Subject: [PATCH] Add BCM4356A3 subver

---
 drivers/bluetooth/btbcm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bluetooth/btbcm.c b/drivers/bluetooth/btbcm.c
index 6659f113042c..e58fe59bd310 100644
--- a/drivers/bluetooth/btbcm.c
+++ b/drivers/bluetooth/btbcm.c
@@ -327,6 +327,7 @@ static const struct {
 	{ 0x2209, "BCM43430A1"  },	/* 001.002.009 */
 	{ 0x6119, "BCM4345C0"	},	/* 003.001.025 */
 	{ 0x230f, "BCM4356A2"	},	/* 001.003.015 */
+	{ 0x2409, "BCM4356A3"   },      /* 001.004.009 */
 	{ }
 };
 
-- 
2.17.1

