--- a/arch/arm64/boot/dts/nvidia/tegra210-nintendo-switch.dts	2018-05-28 18:14:12.387165000 +0200
+++ b/arch/arm64/boot/dts/nvidia/tegra210-nintendo-switch.dts	2018-05-28 18:14:12.387165000 +0200
@@ -1009,6 +1009,7 @@
 		cd-gpios = <&gpio TEGRA_GPIO(Z, 1) GPIO_ACTIVE_LOW>;
 		power-gpios = <&gpio TEGRA_GPIO(E, 4) GPIO_ACTIVE_HIGH>;
 		vqmmc-supply = <&vddio_sdmmc>;
+		max-frequency = <200000000>;
 	};
 
 	/* eMMC */
@@ -1016,6 +1017,7 @@
 		status = "okay";
 		bus-width = <8>;
 		non-removable;
+		max-frequency = <200000000>;
 	};
 
 	aconnect@702c0000 {
