diff --git a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_intr.c b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_intr.c
index c888d0aa474..27c06935918 100644
--- a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_intr.c
+++ b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_intr.c
@@ -1302,7 +1302,7 @@ static int32_t handle_hc_nak_intr(dwc_otg_hcd_t *hcd,
 			hcd->ssplit_lock = 0;
 
 		qtd->complete_split = 0;
-		if (qtd->error_count > 200) {
+		if (qtd->error_count > 20000) {
 			DWC_ERROR("Can not read device info from hub.We take it error\n");
 			halt_channel(hcd, hc, qtd, DWC_OTG_HC_XFER_XACT_ERR);
 		} else {
