diff --git a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_queue.c b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_queue.c
index 3dd3ba7cacb..d809d903450 100644
--- a/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_queue.c
+++ b/drivers/amlogic/usb/dwc_otg/310/dwc_otg_hcd_queue.c
@@ -367,13 +367,13 @@ static int check_periodic_bandwidth(dwc_otg_hcd_t *hcd, dwc_otg_qh_t *qh)
 		 * Max periodic usecs is 80% x 125 usec = 100 usec.
 		 */
 
-		max_claimed_usecs = 100 - qh->usecs;
+		max_claimed_usecs = 125 - qh->usecs;
 	else
 		/*
 		 * Full speed mode.
 		 * Max periodic usecs is 90% x 1000 usec = 900 usec.
 		 */
-		max_claimed_usecs = 900 - qh->usecs;
+		max_claimed_usecs = 1000 - qh->usecs;
 
 	if (hcd->periodic_usecs > max_claimed_usecs) {
 		DWC_INFO("%s: already claimed usecs %d, required usecs %d\n", __func__, hcd->periodic_usecs, qh->usecs);
@@ -724,4 +724,4 @@ int dwc_otg_hcd_qtd_add(dwc_otg_qtd_t *qtd,
 	return retval;
 }
 
-#endif /* DWC_DEVICE_ONLY */
\ No newline at end of file
+#endif /* DWC_DEVICE_ONLY */
