diff -Naur switch-u-boot-a78b224c6bccced805767022398875225fae031d/scripts/Makefile.lib switch-u-boot-a78b224c6bccced805767022398875225fae031d-2/scripts/Makefile.lib
--- switch-u-boot-a78b224c6bccced805767022398875225fae031d/scripts/Makefile.lib	2022-12-21 19:29:32.133017061 +0100
+++ switch-u-boot-a78b224c6bccced805767022398875225fae031d-2/scripts/Makefile.lib	2022-12-21 19:56:57.174600727 +0100
@@ -318,7 +318,7 @@
 # Modified for U-Boot
 # Bring in any U-Boot-specific include at the end of the file
 cmd_dtc = mkdir -p $(dir ${dtc-tmp}) ; \
-	(cat $<; $(if $(u_boot_dtsi),echo '\#include "$(u_boot_dtsi)"')) > $(pre-tmp); \
+	(cat $<; $(if $(u_boot_dtsi),echo '#include "$(u_boot_dtsi)"')) > $(pre-tmp); \
 	$(CPP) $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $(pre-tmp) ; \
 	$(DTC) -O dtb -o $@ -b 0 \
 		-i $(dir $<) $(DTC_FLAGS) \
