From 2da86f2595157c68c2d49e6c2b08a508750878ed Mon Sep 17 00:00:00 2001
From: Andrew Borodin <aborodin@vmail.ru>
Date: Tue, 9 Aug 2016 16:55:12 +0300
Subject: [PATCH] Define NCURSES_CONST if curses library doesn't define it.

Signed-off-by: Andrew Borodin <aborodin@vmail.ru>
---
 lib/tty/tty-ncurses.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/tty/tty-ncurses.h b/lib/tty/tty-ncurses.h
index 4f0e603..9962c7c 100644
--- a/lib/tty/tty-ncurses.h
+++ b/lib/tty/tty-ncurses.h
@@ -20,6 +20,11 @@
 #include <ncursesw/curses.h>
 #endif /* USE_NCURSESW */
 
+/* netbsd-libcurses doesn't define NCURSES_CONST */
+#ifndef NCURSES_CONST
+#define NCURSES_CONST const
+#endif
+
 /*** typedefs(not structures) and defined constants **********************************************/
 
 /*** enums ***************************************************************************************/

