--- a/configure.ac	2015-02-10 15:21:25.466408144 +0100
+++ b/configure.ac.new	2015-02-10 15:21:08.070408144 +0100
@@ -106,9 +106,9 @@
 
 AC_FUNC_ALLOCA
 AC_FUNC_FORK
-AC_FUNC_MALLOC
-AC_FUNC_REALLOC
-AC_CHECK_FUNCS([dup2 isascii memset pow regcomp setlocale strchr strtol])
+# AC_FUNC_MALLOC
+# AC_FUNC_REALLOC
+AC_CHECK_FUNCS([malloc realloc dup2 isascii memset pow regcomp setlocale strchr strtol])
 
 AC_CONFIG_FILES(
 Makefile
