diff -Naur gnutls-3.0.25/configure.ac gnutls-3.0.25.patch/configure.ac
--- gnutls-3.0.25/configure.ac	2012-10-12 17:08:55.000000000 +0200
+++ gnutls-3.0.25.patch/configure.ac	2013-01-12 02:31:30.475591948 +0100
@@ -27,7 +27,7 @@
 
 AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz dist-lzip -Wall -Werror -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_MSG_RESULT([***
 *** Checking for compilation programs...
