diff -Naur gnutls-3.1.6/configure.ac gnutls-3.1.6.patch/configure.ac
--- gnutls-3.1.6/configure.ac	2013-01-02 19:01:38.000000000 +0100
+++ gnutls-3.1.6.patch/configure.ac	2013-01-16 00:59:04.022275973 +0100
@@ -25,9 +25,9 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE([1.11.3 no-dist-gzip dist-xz dist-lzip -Wall -Werror -Wno-override])
+AM_INIT_AUTOMAKE([1.11.3 no-dist-gzip dist-xz dist-lzip -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_MSG_RESULT([***
 *** Checking for compilation programs...
@@ -35,6 +35,10 @@
 
 dnl Checks for programs.
 AC_PROG_CC
+
+# automake 1.12 seems to require this, but automake 1.11 doesn't recognize it
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
 AM_PROG_AS
 AC_PROG_CXX
 gl_EARLY
