diff -Naur gnutls-3.0.20/configure.ac gnutls-3.0.20.patch/configure.ac
--- gnutls-3.0.20/configure.ac	2012-05-26 09:31:17.000000000 +0200
+++ gnutls-3.0.20.patch/configure.ac	2012-06-13 22:02:17.364789323 +0200
@@ -25,7 +25,7 @@
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz dist-lzip -Wall -Werror -Wno-override])
+AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz dist-lzip -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AM_CONFIG_HEADER(config.h)
 
@@ -35,6 +35,10 @@
 
 dnl Checks for programs.
 AC_PROG_CC
+
+# automake 1.12 seems to require this, but automake 1.11 doesn't recognize it
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
 AM_PROG_AS
 AC_PROG_CXX
 gl_EARLY
