diff --git a/arch/arm/mach-imx/mkimage_fit_atf.sh b/arch/arm/mach-imx/mkimage_fit_atf.sh
index 2a17968794c1..2d6c46633c72 100755
--- a/arch/arm/mach-imx/mkimage_fit_atf.sh
+++ b/arch/arm/mach-imx/mkimage_fit_atf.sh
@@ -53,7 +53,7 @@ cat << __HEADER_EOF
 	description = "Configuration to load ATF before U-Boot";
 
 	images {
-		uboot@1 {
+		uboot_1 {
 			description = "U-Boot (64-bit)";
 			os = "u-boot";
 			data = /incbin/("$BL33");
@@ -68,7 +68,7 @@ cnt=1
 for dtname in $*
 do
 	cat << __FDT_IMAGE_EOF
-		fdt@$cnt {
+		fdt_$cnt {
 			description = "$(basename $dtname .dtb)";
 			data = /incbin/("$dtname");
 			type = "flat_dt";
@@ -79,7 +79,7 @@ cnt=$((cnt+1))
 done
 
 cat << __HEADER_EOF
-		atf@1 {
+		atf_1 {
 			description = "ARM Trusted Firmware";
 			os = "arm-trusted-firmware";
 			data = /incbin/("$BL31");
@@ -93,7 +93,7 @@ __HEADER_EOF
 
 if [ -f $BL32 ]; then
 cat << __HEADER_EOF
-		tee@1 {
+		tee_1 {
 			description = "TEE firmware";
 			data = /incbin/("$BL32");
 			type = "firmware";
@@ -108,7 +108,7 @@ fi
 cat << __CONF_HEADER_EOF
 	};
 	configurations {
-		default = "config@1";
+		default = "config_1";
 
 __CONF_HEADER_EOF
 
@@ -117,20 +117,20 @@ for dtname in $*
 do
 if [ -f $BL32 ]; then
 cat << __CONF_SECTION_EOF
-		config@$cnt {
+		config_$cnt {
 			description = "$(basename $dtname .dtb)";
-			firmware = "uboot@1";
-			loadables = "atf@1", "tee@1";
-			fdt = "fdt@$cnt";
+			firmware = "uboot_1";
+			loadables = "atf_1", "tee_1";
+			fdt = "fdt_$cnt";
 		};
 __CONF_SECTION_EOF
 else
 cat << __CONF_SECTION1_EOF
-		config@$cnt {
+		config_$cnt {
 			description = "$(basename $dtname .dtb)";
-			firmware = "uboot@1";
-			loadables = "atf@1";
-			fdt = "fdt@$cnt";
+			firmware = "uboot_1";
+			loadables = "atf_1";
+			fdt = "fdt_$cnt";
 		};
 __CONF_SECTION1_EOF
 fi
