diff -Naur linux-odroidc.git/fs/overlayfs/super.c linux-odroidc.patch/fs/overlayfs/super.c
--- linux-odroidc.git/fs/overlayfs/super.c	2016-10-19 12:20:02.501637960 +0200
+++ linux-odroidc.patch/fs/overlayfs/super.c	2016-10-19 14:46:10.474793888 +0200
@@ -514,7 +514,7 @@
 			break;
 
 		default:
-			return -EINVAL;
+			break;
 		}
 	}
 	return 0;
@@ -665,7 +665,7 @@
 
 static struct file_system_type ovl_fs_type = {
 	.owner		= THIS_MODULE,
-	.name		= "overlayfs",
+	.name		= "overlay",
 	.mount		= ovl_mount,
 	.kill_sb	= kill_anon_super,
 };
