diff -Naur linux-9d4e014.orig/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd linux-9d4e014/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
--- linux-9d4e014.orig/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd	2016-06-02 21:43:00.848754376 +0200
+++ linux-9d4e014/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd	2016-06-02 21:47:34.000000000 +0200
@@ -133,11 +133,11 @@
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x01800000  0x00100000>;
+		reserve-memory = <0x04800000 0x00100000>;
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
 		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
@@ -175,7 +175,7 @@
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-								reserve-memory = <0x04000000>; // 64M
+                reserve-memory = <0x10000000>;
 //		reserve-iomap = "true";
         };
 ///	-	picture decoder
@@ -1054,6 +1054,25 @@
 	};
 
 /// ***************************************************************************************
+///    -       Led
+//$$ MODULE="Led"
+//$$ DEVICE="led"
+//$$ L2 PROP_STR = "status"
+//$$ L2 PROP_U32 = "board_type"
+//$$ L2 PROP_STR = "red_led_gpio"
+//$$ L2 PROP_STR = "blue_led_gpio"
+//$$ L2 PROP_U32 = "led_mode"
+	led{
+		compatible = "amlogic-led";
+		dev_name = "aml-setio";
+		status = "okay";
+		red_led_gpio = "GPIO_TEST_N";/** if your board just need one gpio,please config red_led_gpio and blue_led_gpio the same */
+		blue_led_gpio = "GPIO_TEST_N";/**  */
+		red_led_mode = <1>; /** 1:H on L off , 0:H off L on */
+		blue_led_mode= <1>; /** 1:H on L off , 0:H off L on */
+	};
+
+/// ***************************************************************************************
 /// -   Audio
 //$$ MODULE="Audio"
 //$$ DEVICE="audio_i2s"
