diff -Naur linux-3.2-rc7/net/bluetooth/hci_event.c linux-3.2-rc7.patch/net/bluetooth/hci_event.c
--- linux-3.2-rc7/net/bluetooth/hci_event.c	2011-12-24 06:51:06.000000000 +0100
+++ linux-3.2-rc7.patch/net/bluetooth/hci_event.c	2012-01-01 15:50:48.745287438 +0100
@@ -695,7 +695,8 @@
 	if (rp->status)
 		return;
 
-	memcpy(hdev->extfeatures, rp->features, 8);
+	if (rp->page == 1)
+		memcpy(hdev->extfeatures, rp->features, 8);
 
 	hci_req_complete(hdev, HCI_OP_READ_LOCAL_EXT_FEATURES, rp->status);
 }
