From 2e056332a4ef0783f67b430bbb282f8c632984d5 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 12 Oct 2013 14:24:16 +0300
Subject: [PATCH 1/5] cleanup and crosscompile

---
 Makefile |   20 ++++++--------------
 dwm.c    |    4 ++--
 2 files changed, 8 insertions(+), 16 deletions(-)

diff --git a/Makefile b/Makefile
index b139b54..ca28790 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,6 @@
 # dwm - dynamic window manager
 # See LICENSE file for copyright and license details.
 
-include config.mk
-
 SRC = dwm.c
 OBJ = ${SRC:.c=.o}
 
@@ -42,19 +40,13 @@ dist: clean
 	@rm -rf dwm-${VERSION}
 
 install: all
-	@echo installing executable file to ${DESTDIR}${PREFIX}/bin
-	@mkdir -p ${DESTDIR}${PREFIX}/bin
-	@cp -f dwm ${DESTDIR}${PREFIX}/bin
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/dwm
-	@echo installing manual page to ${DESTDIR}${MANPREFIX}/man1
-	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
-	@sed "s/VERSION/${VERSION}/g" < dwm.1 > ${DESTDIR}${MANPREFIX}/man1/dwm.1
-	@chmod 644 ${DESTDIR}${MANPREFIX}/man1/dwm.1
+	@echo installing executable file to ${DESTDIR}${PREFIX}/usr/bin
+	@mkdir -p ${DESTDIR}${PREFIX}/usr/bin
+	@cp -f dwm ${DESTDIR}${PREFIX}/usr/bin
+	@chmod 755 ${DESTDIR}${PREFIX}/usr/bin/dwm
 
 uninstall:
-	@echo removing executable file from ${DESTDIR}${PREFIX}/bin
-	@rm -f ${DESTDIR}${PREFIX}/bin/dwm
-	@echo removing manual page from ${DESTDIR}${MANPREFIX}/man1
-	@rm -f ${DESTDIR}${MANPREFIX}/man1/dwm.1
+	@echo removing executable file from ${DESTDIR}${PREFIX}/usr/bin
+	@rm -f ${DESTDIR}${PREFIX}/usr/bin/dwm
 
 .PHONY: all options clean dist install uninstall
diff --git a/dwm.c b/dwm.c
index 1d78655..cb8be1e 100644
--- a/dwm.c
+++ b/dwm.c
@@ -2006,7 +2006,7 @@ updatetitle(Client *c) {
 void
 updatestatus(void) {
 	if(!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
-		strcpy(stext, "dwm-"VERSION);
+		strcpy(stext, "dwm");
 	drawbar(selmon);
 }
 
@@ -2129,7 +2129,7 @@ zoom(const Arg *arg) {
 int
 main(int argc, char *argv[]) {
 	if(argc == 2 && !strcmp("-v", argv[1]))
-		die("dwm-"VERSION", © 2006-2011 dwm engineers, see LICENSE for details\n");
+		die("dwm, © 2006-2011 dwm engineers, see LICENSE for details\n");
 	else if(argc != 1)
 		die("usage: dwm [-v]\n");
 	if(!setlocale(LC_CTYPE, "") || !XSupportsLocale())
-- 
1.7.2.5

