commit 1040e9440a68bfa50c070f3cf956e46546d54729
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 12 16:37:56 2013 +0300

    do not die if XAllocNamedColor fails

diff --git a/dwm.c b/dwm.c
index cb8be1e..5f9f419 100644
--- a/dwm.c
+++ b/dwm.c
@@ -932,7 +932,7 @@ getcolor(const char *colstr) {
 	XColor color;
 
 	if(!XAllocNamedColor(dpy, cmap, colstr, &color, &color))
-		die("error, cannot allocate color '%s'\n", colstr);
+		printf("error, cannot allocate color '%s'\n", colstr);
 	return color.pixel;
 }
 
