From 71091fbbdf97ad13c13b9f893dd12062e9fcf61d Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 12 Oct 2013 14:26:11 +0300
Subject: [PATCH 3/5] hide bar

---
 config.def.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/config.def.h b/config.def.h
index 77ff358..0c9b846 100644
--- a/config.def.h
+++ b/config.def.h
@@ -10,7 +10,7 @@ static const char selbgcolor[]      = "#005577";
 static const char selfgcolor[]      = "#eeeeee";
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
-static const Bool showbar           = True;     /* False means no bar */
+static const Bool showbar           = False;     /* False means no bar */
 static const Bool topbar            = True;     /* False means bottom bar */
 
 /* tagging */
-- 
1.7.2.5

