diff -Naur xorg-launch-helper-4/configure.ac xorg-launch-helper-4.patch/configure.ac
--- xorg-launch-helper-4/configure.ac	2012-10-23 21:17:54.000000000 +0200
+++ xorg-launch-helper-4.patch/configure.ac	2013-12-06 15:57:17.948725894 +0100
@@ -3,7 +3,7 @@
 
 AC_PREREQ([2.68])
 AC_INIT([xorg-launch-helper], [4], [auke-jan.h.kok@intel.com])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_SRCDIR([src/main.c])
 AC_CONFIG_HEADERS([config.h])
