diff --git a/configure.in b/configure.in
index 18165f3..97007b5 100644
--- a/configure.in
+++ b/configure.in
@@ -7,12 +7,13 @@ dnl
 dnl Original version Dug Song <dugsong@monkey.org>
 
 AC_PREREQ(2.59c)
-AC_INIT(event.c)
+AC_INIT([libevent], [2.0.21-stable])
+AC_CONFIG_SRCDIR([event.c])
 
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE(libevent,2.0.21-stable)
-AM_CONFIG_HEADER(config.h)
+AM_INIT_AUTOMAKE
+AC_CONFIG_HEADERS(config.h)
 AC_DEFINE(NUMERIC_VERSION, 0x02001500, [Numeric representation of the version])
 
 dnl Initialize prefix.
diff --git a/test/Makefile.am b/test/Makefile.am
index b10c41a..53e00e4 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -19,8 +19,6 @@ endif
 EXTRA_PROGRAMS = regress
 noinst_HEADERS = tinytest.h tinytest_macros.h regress.h tinytest_local.h
 
-TESTS = $(top_srcdir)/test/test.sh
-
 BUILT_SOURCES =
 if BUILD_REGRESS
 BUILT_SOURCES += regress.gen.c regress.gen.h
diff --git a/test/Makefile.in b/test/Makefile.in
index 451bd3c..939ad57 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -315,7 +315,6 @@ AUTOMAKE_OPTIONS = foreign
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/compat -I$(top_srcdir)/include -I../include -DTINYTEST_LOCAL
 EXTRA_DIST = regress.rpc regress.gen.h regress.gen.c rpcgen_wrapper.sh test.sh
 noinst_HEADERS = tinytest.h tinytest_macros.h regress.h tinytest_local.h
-TESTS = $(top_srcdir)/test/test.sh
 BUILT_SOURCES = $(am__append_2)
 test_init_SOURCES = test-init.c
 test_init_LDADD = ../libevent_core.la
