diff -Naur ppp-2.4.5/chat/Makefile.linux ppp-2.4.5.patch/chat/Makefile.linux
--- ppp-2.4.5/chat/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/chat/Makefile.linux	2014-01-10 08:57:16.699478234 +0100
@@ -10,7 +10,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe
+COPTS:=	-O2 -g -pipe
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
diff -Naur ppp-2.4.5/pppd/Makefile.linux ppp-2.4.5.patch/pppd/Makefile.linux
--- ppp-2.4.5/pppd/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppd/Makefile.linux	2014-01-10 08:55:22.876569576 +0100
@@ -32,7 +32,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS := -O2 -pipe -Wall -g
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
diff -Naur ppp-2.4.5/pppd/plugins/Makefile.linux ppp-2.4.5.patch/pppd/plugins/Makefile.linux
--- ppp-2.4.5/pppd/plugins/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppd/plugins/Makefile.linux	2014-01-10 08:56:05.739471826 +0100
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	:= -O2 -g
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
diff -Naur ppp-2.4.5/pppd/plugins/pppoatm/Makefile.linux ppp-2.4.5.patch/pppd/plugins/pppoatm/Makefile.linux
--- ppp-2.4.5/pppd/plugins/pppoatm/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppd/plugins/pppoatm/Makefile.linux	2014-01-10 08:56:36.406462256 +0100
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	:= -O2 -g
 CFLAGS	= $(COPTS) -I../.. -I../../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
diff -Naur ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux ppp-2.4.5.patch/pppd/plugins/pppol2tp/Makefile.linux
--- ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppd/plugins/pppol2tp/Makefile.linux	2014-01-10 08:56:54.647468343 +0100
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	:= -O2 -g
 CFLAGS	= $(COPTS) -I. -I../.. -I../../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
diff -Naur ppp-2.4.5/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.5.patch/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.5/pppd/plugins/rp-pppoe/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppd/plugins/rp-pppoe/Makefile.linux	2014-01-10 08:56:20.595464447 +0100
@@ -25,7 +25,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 RP_VERSION=3.8p
 
-COPTS=-O2 -g
+COPTS:=-O2 -g
 CFLAGS=$(COPTS) -I../../../include '-DRP_VERSION="$(RP_VERSION)"'
 all: rp-pppoe.so pppoe-discovery
 
diff -Naur ppp-2.4.5/pppstats/Makefile.linux ppp-2.4.5.patch/pppstats/Makefile.linux
--- ppp-2.4.5/pppstats/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5.patch/pppstats/Makefile.linux	2014-01-10 08:57:41.568497184 +0100
@@ -10,7 +10,7 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS := -O
 COMPILE_FLAGS = -I../include
 LIBS =
 
