diff -Naur ppp-2.4.7/pppd/Makefile.linux ppp-2.4.7.patch/pppd/Makefile.linux
--- ppp-2.4.7/pppd/Makefile.linux	2014-08-09 14:31:39.000000000 +0200
+++ ppp-2.4.7.patch/pppd/Makefile.linux	2014-09-10 13:31:20.731202543 +0200
@@ -48,7 +48,7 @@
 # Uncomment the next line to include support for PPP packet filtering.
 # This requires that the libpcap library and headers be installed
 # and that the kernel driver support PPP packet filtering.
-FILTER=y
+# FILTER=y
 
 # Uncomment the next line to enable multilink PPP (enabled by default)
 # Linux distributions: Please leave multilink ENABLED in your builds
