diff -Naur glibc-2.20/misc/sys/xattr.h glibc-2.20.patch/misc/sys/xattr.h
--- glibc-2.20/misc/sys/xattr.h	2014-09-07 10:09:09.000000000 +0200
+++ glibc-2.20.patch/misc/sys/xattr.h	2015-01-18 02:47:33.169110306 +0100
@@ -26,15 +26,8 @@
 
 /* The following constants should be used for the fifth parameter of
    `*setxattr'.  */
-#ifndef __USE_KERNEL_XATTR_DEFS
-enum
-{
-  XATTR_CREATE = 1,	/* set value, fail if attr already exists.  */
-#define XATTR_CREATE	XATTR_CREATE
-  XATTR_REPLACE = 2	/* set value, fail if attr does not exist.  */
-#define XATTR_REPLACE	XATTR_REPLACE
-};
-#endif
+#define XATTR_CREATE 1
+#define XATTR_REPLACE 2
 
 /* Set the attribute NAME of the file pointed to by PATH to VALUE (which
    is SIZE bytes long).  Return 0 on success, -1 for errors.  */
