diff --git a/Makefile b/Makefile
index fc2830ef4..95af92680 100644
--- a/Makefile
+++ b/Makefile
@@ -46,8 +46,11 @@ endif
 # Unix
 ifeq ($(platform), unix)
    TARGET := $(TARGET_NAME)_libretro.so
-   LDFLAGS += -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T -Wl,--gc-sections
-   fpic = -fPIC
+   ifneq ($(findstring --version-script=,$(LDFLAGS)),--version-script=)
+      LDFLAGS += -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T -Wl,--gc-sections
+      fpic = -fPIC
+   endif
+
 
 # Raspberry Pi 4
 else ifneq (,$(findstring rpi4,$(platform)))
