diff -Naur libgpg-error-1.10/configure.ac libgpg-error-1.10.patch/configure.ac
--- libgpg-error-1.10/configure.ac	2010-10-26 11:13:48.000000000 +0200
+++ libgpg-error-1.10.patch/configure.ac	2012-04-11 10:19:39.115582454 +0200
@@ -112,7 +112,7 @@
 
 
 # Note, that autogen.sh greps for the next line.
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.18])
 AM_GNU_GETTEXT([external])
 
 # Checks for header files.
