diff -Naur upFS-9a906da00d97d9b31fae91a431031d55ed99571c/Makefile upFS-9a906da00d97d9b31fae91a431031d55ed99571c-3/Makefile
--- upFS-9a906da00d97d9b31fae91a431031d55ed99571c/Makefile	2025-02-21 16:46:36.636480266 +0100
+++ upFS-9a906da00d97d9b31fae91a431031d55ed99571c-3/Makefile	2025-02-22 17:15:50.289720784 +0100
@@ -1,7 +1,6 @@
-CC=gcc
 ECFLAGS=-O3
 CFLAGS=-DUPFS_LNCP -DUPFS_PERMLOWERCASE -DUPFS_FATNAMES -D_FILE_OFFSET_BITS=64 $(ECFLAGS)
-FUSE_FLAGS=`pkg-config --cflags --libs fuse`
+FUSE_FLAGS=$(shell pkg-config --cflags --libs fuse)
 
 all: upfs upfs-ps mount.upfs mount.upfsps
 
@@ -18,10 +17,10 @@
 	$(CC) $(CFLAGS) mountupfsps.c -o mount.upfsps
 
 install: all
-	install upfs /usr/bin/upfs
-	install upfs-ps /usr/bin/upfs-ps
-	install mount.upfs /sbin/mount.upfs
-	install mount.upfsps /sbin/mount.upfsps
+	install upfs $(DESTDIR)/usr/bin/upfs
+	install upfs-ps $(DESTDIR)/usr/bin/upfs-ps
+	install mount.upfs $(DESTDIR)/sbin/mount.upfs
+	install mount.upfsps $(DESTDIR)/sbin/mount.upfsps
 
 clean:
 	rm -f upfs upfs-ps mount.upfs mount.upfsps
