diff -Naur upFS-9a906da00d97d9b31fae91a431031d55ed99571c/upfs.c upFS-9a906da00d97d9b31fae91a431031d55ed99571c-3/upfs.c
--- upFS-9a906da00d97d9b31fae91a431031d55ed99571c/upfs.c	2025-02-21 16:46:36.636480266 +0100
+++ upFS-9a906da00d97d9b31fae91a431031d55ed99571c-3/upfs.c	2025-02-22 17:21:16.107561511 +0100
@@ -558,7 +558,7 @@
     /* Copy it in the store */
     from_file_fd = openat(store_root, sfrom, O_RDONLY);
     if (from_file_fd < 0) goto error;
-    to_file_fd = openat(store_root, sto, O_WRONLY|O_CREAT|O_EXCL);
+    to_file_fd = openat(store_root, sto, O_WRONLY|O_CREAT|O_EXCL, S_IRUSR|S_IWUSR);
     if (to_file_fd < 0) goto error;
     buf = malloc(BUFSZ);
     if (!buf) goto error;
