--- xorg-server.orig/exa/exa.c	2008-10-14 23:24:44.000000000 +0200
+++ xorg-server/exa/exa.c	2008-10-14 23:25:52.000000000 +0200
@@ -882,6 +882,11 @@
     dixSetPrivate(&pScreen->devPrivates, exaScreenPrivateKey, pExaScr);
 
     pExaScr->migration = ExaMigrationAlways;
+    if (pScreenInfo->flags & EXA_MIGRATION_GREEDY) {
+        pExaScr->migration = ExaMigrationGreedy;
+        LogMessage(X_INFO, "EXA(%d): Forcing greedy migration option\n",
+		   pScreen->myNum);
+    }
 
     exaDDXDriverInit(pScreen);
 
--- xorg-server.orig/exa/exa.h	2008-10-14 23:24:44.000000000 +0200
+++ xorg-server/exa/exa.h	2008-10-14 23:25:52.000000000 +0200
@@ -737,6 +737,13 @@
 #define EXA_TWO_BITBLT_DIRECTIONS	(1 << 2)
 
 /**
+ * EXA_MIGRATION_GREEDY indicates to EXA that the driver prefers to
+ * use the "greedy" migration heuristic.  This is to work around
+ * issues with EXA on the Intel 965 chipset. (LP: #177492)
+ */
+#define EXA_MIGRATION_GREEDY            (1 << 3)
+
+/**
  * EXA_HANDLES_PIXMAPS indicates to EXA that the driver can handle
  * all pixmap addressing and migration.
  */
