diff -Naur xorg-server-1.5.2.orig/dix/registry.c xorg-server-1.5.2/dix/registry.c
--- xorg-server-1.5.2.orig/dix/registry.c	2008-10-11 14:11:31.000000000 +0200
+++ xorg-server-1.5.2/dix/registry.c	2008-10-11 14:12:27.000000000 +0200
@@ -32,7 +32,7 @@
 
 #define BASE_SIZE 16
 #define CORE "X11"
-#define FILENAME SERVER_MISC_CONFIG_PATH "/protocol.txt"
+#define FILENAME "/etc/X11/protocol.txt"
 
 #define PROT_COMMENT '#'
 #define PROT_REQUEST 'R'
