diff -Naur xorg-server-1.6.99.902.orig/Xi/exevents.c xorg-server-1.6.99.902/Xi/exevents.c
--- xorg-server-1.6.99.902.orig/Xi/exevents.c	2009-09-26 14:12:03.000000000 +0200
+++ xorg-server-1.6.99.902/Xi/exevents.c	2009-09-26 18:55:30.000000000 +0200
@@ -195,11 +195,15 @@
 static void
 CopyKeyClass(DeviceIntPtr device, DeviceIntPtr master)
 {
-    KeyClassPtr mk = master->key;
-    KeyClassPtr dk = device->key;
+    KeyClassPtr mk, dk;
     int i;
 
-    if (device == master)
+    if (device == master || device == NULL || master == NULL)
+        return;
+
+    mk = master->key;
+    dk = device->key;
+    if (dk == NULL || mk == NULL)
         return;
 
     mk->sourceid = device->id;
