diff -Nurp patched/include/misc.h working/include/misc.h
--- patched/include/misc.h	2009-02-05 01:09:03.000000000 -0800
+++ working/include/misc.h	2009-02-05 02:12:57.000000000 -0800
@@ -86,7 +86,7 @@ extern unsigned long serverGeneration;
 #ifndef MAXSCREENS
 #define MAXSCREENS	16
 #endif
-#define MAXCLIENTS	256
+#define MAXCLIENTS	512
 #define MAXEXTENSIONS   128
 #define MAXFORMATS	8
 #define MAXDEVICES	20 /* input devices */
