diff -Nurp patched/hw/xfree86/modes/xf86RandR12.c working/hw/xfree86/modes/xf86RandR12.c
--- patched/hw/xfree86/modes/xf86RandR12.c	2009-03-02 22:57:18.000000000 -0800
+++ working/hw/xfree86/modes/xf86RandR12.c	2009-03-02 22:57:24.000000000 -0800
@@ -944,7 +944,8 @@ xf86RandR12SetRotations (ScreenPtr pScre
     for (c = 0; c < config->num_crtc; c++) {
 	xf86CrtcPtr    crtc = config->crtc[c];
 
-	RRCrtcSetRotations (crtc->randr_crtc, rotations);
+        if (crtc != NULL)
+            RRCrtcSetRotations (crtc->randr_crtc, rotations);
     }
 #endif
     randrp->supported_rotations = rotations;
