Index: xorg-server-1.6.0/hw/xfree86/common/xf86Init.c
===================================================================
--- xorg-server-1.6.0.orig/hw/xfree86/common/xf86Init.c	2009-03-19 23:19:45.000000000 -0700
+++ xorg-server-1.6.0/hw/xfree86/common/xf86Init.c	2009-03-19 23:23:22.000000000 -0700
@@ -1345,8 +1345,12 @@
   signal(SIGCHLD, SIG_DFL);	/* Need to wait for child processes */
 #endif
 
-  if (!beenHere)
+  if (!beenHere) {
+    /* have glibc report internal abort traces to stderr instead of
+       the controlling terminal */
+    setenv("LIBC_FATAL_STDERR_","1",0);
     xf86LogInit();
+  }
 
         /* Set stderr to non-blocking. */
 #ifndef O_NONBLOCK
