From 8663524ce398b56af596ab1e7e96b62ca45b79e9 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 13 Apr 2009 17:49:00 +1000
Subject: [PATCH] xfree86: don't synthesise a mouse section if synaptics devices are found.

diff -Naur xorg-server-1.6.99.902.orig/hw/xfree86/common/xf86Config.c xorg-server-1.6.99.902/hw/xfree86/common/xf86Config.c
--- xorg-server-1.6.99.902.orig/hw/xfree86/common/xf86Config.c	2009-09-26 14:12:03.000000000 +0200
+++ xorg-server-1.6.99.902/hw/xfree86/common/xf86Config.c	2009-09-26 19:05:52.000000000 +0200
@@ -2406,6 +2406,7 @@
         {
             if (strcmp((*dev)->driver, "kbd") == 0 ||
                 strcmp((*dev)->driver, "mouse") == 0 ||
+                strcmp((*dev)->driver, "synaptics") == 0 ||
                 strcmp((*dev)->driver, "vmmouse") == 0)
             {
                 IDevPtr *current;
