diff -Naur xorg-server-1.5.3-old/os/osdep.h xorg-server-1.5.3-new/os/osdep.h
--- xorg-server-1.5.3-old/os/osdep.h	2008-11-05 08:52:17.000000000 -0800
+++ xorg-server-1.5.3-new/os/osdep.h	2008-11-05 14:26:40.000000000 -0800
@@ -56,7 +56,9 @@
 #define BUFSIZE 4096
 #define BUFWATERMARK 8192
 
+#if defined(XDMCP) || defined(HASXDMAUTH)
 #include <X11/Xdmcp.h>
+#endif
 
 #ifndef sgi	    /* SGI defines OPEN_MAX in a useless way */
 #ifndef X_NOT_POSIX
@@ -122,9 +124,11 @@
 
 #include <stddef.h>
 
+#if defined(XDMCP) || defined(HASXDMAUTH)
 typedef Bool (*ValidatorFunc)(ARRAY8Ptr Auth, ARRAY8Ptr Data, int packet_type);
 typedef Bool (*GeneratorFunc)(ARRAY8Ptr Auth, ARRAY8Ptr Data, int packet_type);
 typedef Bool (*AddAuthorFunc)(unsigned name_length, char *name, unsigned data_length, char *data);
+#endif
 
 typedef struct _connectionInput {
     struct _connectionInput *next;
@@ -257,6 +261,7 @@
 extern int  SecureRPCReset    (AuthRstCArgs);
 #endif
 
+#ifdef XDMCP
 /* in xdmcp.c */
 extern void XdmcpUseMsg (void);
 extern int XdmcpOptions(int argc, char **argv, int i);
@@ -281,6 +286,7 @@
 
 struct sockaddr_in;
 extern void XdmcpRegisterBroadcastAddress (struct sockaddr_in *addr);
+#endif
 
 #ifdef HASXDMAUTH
 extern void XdmAuthenticationInit (char *cookie, int cookie_length);
