diff -Naur xorg-server-1.4.0.90/hw/xfree86/os-support/linux/lnx_apm.c xorg-server-1.4.0.90.edit/hw/xfree86/os-support/linux/lnx_apm.c
--- xorg-server-1.4.0.90/hw/xfree86/os-support/linux/lnx_apm.c	2007-12-14 00:38:23.000000000 +0100
+++ xorg-server-1.4.0.90.edit/hw/xfree86/os-support/linux/lnx_apm.c	2007-12-14 00:38:18.000000000 +0100
@@ -35,6 +35,8 @@
 # define APM_SUSPEND_FAILED 0xf001
 #endif
 
+typedef unsigned short  apm_event_t;
+
 static PMClose lnxAPMOpen(void);
 static void lnxCloseAPM(void);
 static pointer APMihPtr = NULL;
