diff -Naur xorg-server-1.6.1.901.orig/configure xorg-server-1.6.1.901/configure
--- xorg-server-1.6.1.901.orig/configure	2009-05-09 13:58:58.000000000 +0200
+++ xorg-server-1.6.1.901/configure	2009-05-09 14:12:45.000000000 +0200
@@ -21949,12 +21949,12 @@
         pkg_cv_GL_CFLAGS="$GL_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9 gl >= 7.1.0\"") >&5
-  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9 gl >= 7.1.0") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9\"") >&5
+  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GL_CFLAGS=`$PKG_CONFIG --cflags "glproto >= 1.4.9 gl >= 7.1.0" 2>/dev/null`
+  pkg_cv_GL_CFLAGS=`$PKG_CONFIG --cflags "glproto >= 1.4.9" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21967,12 +21967,12 @@
         pkg_cv_GL_LIBS="$GL_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9 gl >= 7.1.0\"") >&5
-  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9 gl >= 7.1.0") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9\"") >&5
+  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GL_LIBS=`$PKG_CONFIG --libs "glproto >= 1.4.9 gl >= 7.1.0" 2>/dev/null`
+  pkg_cv_GL_LIBS=`$PKG_CONFIG --libs "glproto >= 1.4.9" 2>/dev/null`
 else
   pkg_failed=yes
 fi
