Index: xorg-server/hw/xfree86/os-support/linux/lnx_init.c
===================================================================
--- xorg-server.orig/hw/xfree86/os-support/linux/lnx_init.c	2008-06-18 10:28:16.000000000 +0300
+++ xorg-server/hw/xfree86/os-support/linux/lnx_init.c	2008-06-18 12:06:06.000000000 +0300
@@ -283,9 +283,7 @@
 	        FatalError("Unable to set screen info\n");
 	    close(fbfd);
 #endif
-        } else { /* ShareVTs */
-            close(xf86Info.consoleFd);
-        }
+	}
 	signal(SIGUSR2, xf86ReloadInputDevs);
     } else { 	/* serverGeneration != 1 */
         if (!ShareVTs && VTSwitch)
@@ -314,7 +312,11 @@
     int vtno = -1;
 #endif
 
-    if (ShareVTs) return;
+    if (ShareVTs)
+    {
+	    close(xf86Info.consoleFd);
+	    return;
+    }
 
 #if defined(DO_OS_FONTRESTORE)
     if (ioctl(xf86Info.consoleFd, VT_GETSTATE, &vts) < 0)
