diff -Naur xorg-server-1.8.99.903/hw/xfree86/common/xf86pciBus.c xorg-server-1.8.99.903.patch/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.8.99.903/hw/xfree86/common/xf86pciBus.c	2010-06-10 16:07:55.000000000 +0200
+++ xorg-server-1.8.99.903.patch/hw/xfree86/common/xf86pciBus.c	2010-06-27 01:06:57.977808296 +0200
@@ -1129,7 +1129,7 @@
 	    break;
 	case 0x102b:		    driverList[0] = "mga";	break;
 	case 0x10c8:		    driverList[0] = "neomagic"; break;
-	case 0x10de: case 0x12d2:   driverList[0] = "nv";	break;
+	case 0x10de: case 0x12d2:   driverList[0] = "nvidia";	break;
 	case 0x1106:		    driverList[0] = "openchrome"; break;
         case 0x1b36:		    driverList[0] = "qxl"; break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
