diff -Naur xorg-server-1.8.99.903/hw/xfree86/modes/xf86Crtc.c xorg-server-1.8.99.903.patch/hw/xfree86/modes/xf86Crtc.c
--- xorg-server-1.8.99.903/hw/xfree86/modes/xf86Crtc.c	2010-06-15 23:39:25.000000000 +0200
+++ xorg-server-1.8.99.903.patch/hw/xfree86/modes/xf86Crtc.c	2010-06-27 00:53:28.722808399 +0200
@@ -2379,10 +2379,10 @@
 	    xf86DrvMsg(i, X_INFO, "Using user preference for initial modes\n");
 	else if (xf86TargetPreferred(scrn, config, modes, enabled, width, height))
 	    xf86DrvMsg(i, X_INFO, "Using exact sizes for initial modes\n");
-	else if (xf86TargetAspect(scrn, config, modes, enabled, width, height))
-	    xf86DrvMsg(i, X_INFO, "Using fuzzy aspect match for initial modes\n");
 	else if (xf86TargetFallback(scrn, config, modes, enabled, width, height))
 	    xf86DrvMsg(i, X_INFO, "Using sloppy heuristic for initial modes\n");
+	else if (xf86TargetAspect(scrn, config, modes, enabled, width, height))
+	    xf86DrvMsg(i, X_INFO, "Using fuzzy aspect match for initial modes\n");
 	else
 	    xf86DrvMsg(i, X_WARNING, "Unable to find initial modes\n");
     }
