diff -Naur xorg-server-20091215/autom4te.cache/output.0 xorg-server-20091215.patch/autom4te.cache/output.0
diff -Naur xorg-server-20091215/autom4te.cache/requests xorg-server-20091215.patch/autom4te.cache/requests
diff -Naur xorg-server-20091215/autom4te.cache/traces.0 xorg-server-20091215.patch/autom4te.cache/traces.0
diff -Naur xorg-server-20091215/config.log xorg-server-20091215.patch/config.log
diff -Naur xorg-server-20091215/configure xorg-server-20091215.patch/configure
--- xorg-server-20091215/configure	2009-12-15 20:40:42.188321308 +0100
+++ xorg-server-20091215.patch/configure	2009-12-15 20:43:03.790321294 +0100
@@ -21110,39 +21110,6 @@
 $as_echo "${FONT100DPIDIR}" >&6; }
 
 
-as_ac_File=`$as_echo "ac_cv_file_${sysconfdir}/X11/fontpath.d" | $as_tr_sh`
-{ $as_echo "$as_me:$LINENO: checking for ${sysconfdir}/X11/fontpath.d" >&5
-$as_echo_n "checking for ${sysconfdir}/X11/fontpath.d... " >&6; }
-if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
-  $as_echo_n "(cached) " >&6
-else
-  test "$cross_compiling" = yes &&
-  { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-$as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r "${sysconfdir}/X11/fontpath.d"; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-ac_res=`eval 'as_val=${'$as_ac_File'}
-		 $as_echo "$as_val"'`
-	       { $as_echo "$as_me:$LINENO: result: $ac_res" >&5
-$as_echo "$ac_res" >&6; }
-as_val=`eval 'as_val=${'$as_ac_File'}
-		 $as_echo "$as_val"'`
-   if test "x$as_val" = x""yes; then
-  DEFAULT_FONT_PATH='catalogue:$(sysconfdir)/X11/fontpath.d'
-else
-
-		DEFAULT_FONT_PATH="${FONTMISCDIR}/,${FONTTTFDIR}/,${FONTOTFDIR}/,${FONTTYPE1DIR}/,${FONT100DPIDIR}/,${FONT75DPIDIR}/"
-		case $host_os in
-			darwin*)    DEFAULT_FONT_PATH="${DEFAULT_FONT_PATH},/Library/Fonts,/System/Library/Fonts" ;;
-		esac
-
-fi
-
 
 # Check whether --with-default-font-path was given.
 if test "${with_default_font_path+set}" = set; then
diff -Naur xorg-server-20091215/configure.ac xorg-server-20091215.patch/configure.ac
--- xorg-server-20091215/configure.ac	2009-12-15 20:40:42.190321618 +0100
+++ xorg-server-20091215.patch/configure.ac	2009-12-15 20:42:53.892447009 +0100
@@ -504,14 +504,6 @@
 
 dnl Uses --default-font-path if set, otherwise checks for /etc/X11/fontpath.d,
 dnl otherwise uses standard subdirectories of FONTROOTDIR
-AC_CHECK_FILE([${sysconfdir}/X11/fontpath.d],
-	[DEFAULT_FONT_PATH='catalogue:$(sysconfdir)/X11/fontpath.d'],
-	[
-		DEFAULT_FONT_PATH="${FONTMISCDIR}/,${FONTTTFDIR}/,${FONTOTFDIR}/,${FONTTYPE1DIR}/,${FONT100DPIDIR}/,${FONT75DPIDIR}/"
-		case $host_os in
-			darwin*)    DEFAULT_FONT_PATH="${DEFAULT_FONT_PATH},/Library/Fonts,/System/Library/Fonts" ;;
-		esac
-	])
 AC_ARG_WITH(default-font-path, AS_HELP_STRING([--with-default-font-path=PATH], [Comma separated list of font dirs]),
 				[ FONTPATH="$withval" ],
 				[ FONTPATH="${DEFAULT_FONT_PATH}" ])
diff -Naur xorg-server-20091215/include/do-not-use-config.h.in~ xorg-server-20091215.patch/include/do-not-use-config.h.in~
