diff -Naur xorg-server-1.7.99.1/hw/xfree86/common/xf86Config.c xorg-server-1.7.99.1.patch/hw/xfree86/common/xf86Config.c
--- xorg-server-1.7.99.1/hw/xfree86/common/xf86Config.c	2009-10-13 19:04:27.000000000 +0200
+++ xorg-server-1.7.99.1.patch/hw/xfree86/common/xf86Config.c	2009-12-10 15:29:23.098423510 +0100
@@ -118,7 +118,7 @@
     {.name = "record",   .toLoad = TRUE,    .load_opt=NULL},
 #endif
 #ifdef XF86DRI
-    {.name = "dri",      .toLoad = TRUE,    .load_opt=NULL},
+    {.name = "dri",      .toLoad = FALSE,    .load_opt=NULL},
 #endif
 #ifdef DRI2
     {.name = "dri2",     .toLoad = TRUE,    .load_opt=NULL},
