diff --git a/hw/xfree86/common/xf86Globals.c b/hw/xfree86/common/xf86Globals.c
index 04a96cb..1036839 100644
--- a/hw/xfree86/common/xf86Globals.c
+++ b/hw/xfree86/common/xf86Globals.c
@@ -105,7 +105,7 @@ xf86InfoRec xf86Info = {
     .lastEventTime              = -1,
     .vtRequestsPending          = FALSE,
     .dontVTSwitch               = FALSE,
-    .dontZap                    = FALSE,
+    .dontZap                    = TRUE,
     .dontZoom                   = FALSE,
     .notrapSignals              = FALSE,
     .caughtSignal               = FALSE,
