diff -Naur xorg-server-1.12.2/dix/window.c xorg-server-1.12.2.patch/dix/window.c
--- xorg-server-1.12.2/dix/window.c	2012-05-17 19:09:02.000000000 +0200
+++ xorg-server-1.12.2.patch/dix/window.c	2012-06-04 07:31:56.958440047 +0200
@@ -107,6 +107,7 @@
 #include "validate.h"
 #include "windowstr.h"
 #include "propertyst.h"
+#include "selection.h" /* DeleteWindowFromAnySelections */
 #include "input.h"
 #include "inputstr.h"
 #include "resource.h"
