--- xorg-server-1.6.1.901/hw/xfree86/common/xf86AutoConfig.c~	2009-07-03 11:59:30.000000000 -0700
+++ xorg-server-1.6.1.901/hw/xfree86/common/xf86AutoConfig.c	2009-07-03 11:59:30.000000000 -0700
@@ -516,7 +516,8 @@
     /* Fallback to platform default hardware */
     if (i < (nmatches - 1)) {
 #if defined(__i386__) || defined(__amd64__) || defined(__hurd__)
-	matches[i++] = xnfstrdup("vesa");
+	if (geteuid() == 0)
+		matches[i++] = xnfstrdup("vesa");
 #elif defined(__sparc__) && !defined(sun)
 	matches[i++] = xnfstrdup("sunffb");
 #endif
