diff -Naur xorg-server-1.8.2-old/xorg-server.m4 xorg-server-1.8.2-new/xorg-server.m4
--- xorg-server-1.8.2-old/xorg-server.m4	2010-07-01 10:35:53.000000000 -0700
+++ xorg-server-1.8.2-new/xorg-server.m4	2010-07-01 10:36:36.000000000 -0700
@@ -31,7 +31,8 @@
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
 	AC_REQUIRE([PKG_PROG_PKG_CONFIG])
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS -I`$PKG_CONFIG --variable=sdkdir xorg-server`"
+	m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+	CFLAGS="$CFLAGS -I$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=sdkdir xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
