diff -Naur xorg-server-1.11.0-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.11.0-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.11.0-old/hw/xfree86/common/xf86pciBus.c	2011-08-27 09:47:09.000000000 -0700
+++ xorg-server-1.11.0-new/hw/xfree86/common/xf86pciBus.c	2011-08-27 09:47:33.000000000 -0700
@@ -1107,7 +1107,7 @@
 	case 0x1142:		    driverList[0] = "apm"; break;
 	case 0xedd8:		    driverList[0] = "ark"; break;
 	case 0x1a03:		    driverList[0] = "ast"; break;
-	case 0x1002:		    driverList[0] = "ati"; break;
+	case 0x1002:		    driverList[0] = "radeon"; break;
 	case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
 	case 0x3d3d:		    driverList[0] = "glint"; break;
