diff -Naur xorg-server-1.11.0-old/hw/xfree86/common/xf86pciBus.c xorg-server-1.11.0-new/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.11.0-old/hw/xfree86/common/xf86pciBus.c	2011-08-27 09:46:27.000000000 -0700
+++ xorg-server-1.11.0-new/hw/xfree86/common/xf86pciBus.c	2011-08-27 09:47:09.000000000 -0700
@@ -1123,7 +1123,7 @@
 	    break;
 	case 0x102b:		    driverList[0] = "mga";	break;
 	case 0x10c8:		    driverList[0] = "neomagic"; break;
-	case 0x10de: case 0x12d2:   driverList[0] = "nv";	break;
+	case 0x10de: case 0x12d2:   driverList[0] = "nouveau";	break;
 	case 0x1106:		    driverList[0] = "openchrome"; break;
         case 0x1b36:		    driverList[0] = "qxl"; break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
