From 57a188877b0472edb12bdf707b1550abbeb85af8 Mon Sep 17 00:00:00 2001
From: Fedora X Ninjas <x@fedoraproject.org>
Date: Wed, 17 Feb 2010 11:12:36 +1000
Subject: [PATCH] Stick giant warning into logfiles about udev configuration changes.

Signed-off-by: Fedora X Ninjas <x@fedoraproject.org>
---
 hw/xfree86/common/xf86Config.c |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/hw/xfree86/common/xf86Config.c b/hw/xfree86/common/xf86Config.c
index 132e8bc..b1722cd 100644
--- a/hw/xfree86/common/xf86Config.c
+++ b/hw/xfree86/common/xf86Config.c
@@ -1472,6 +1472,19 @@ checkCoreInputDevices(serverLayoutPtr servlayoutp, Bool implicitLayout)
 	                "input devices.\n\tIf no devices become available, "
 	                "reconfigure %s or disable AutoAddDevices.\n",
 			config_backend, config_backend);
+
+	xf86Msg(X_INFO, "\n"
+        "***************************************************************\n"
+        "***************************************************************\n"
+        "** OpenELEC.tv switched to udev-based device detection with  **\n"
+        "** xorg-x11-server-1.7.99.901-1. Custom HAL configuration    **\n"
+        "** stored in /etc/hal/fdi/policy will not be seen by this    **\n"
+        "** server version. If you have such configuration you will   **\n"
+        "** need to update it to the new format.                      **\n"
+        "** For more information, see the Fedora wiki page            **\n"
+        "** https://fedoraproject.org/wiki/Input_device_configuration **\n"
+        "***************************************************************\n"
+        "***************************************************************\n");
 #else
 	xf86Msg(X_INFO, "Hotplugging is disabled and no input devices were configured.\n"
 			"\tTry disabling AllowEmptyInput.\n");
-- 
1.6.6.1

