--- ecore-r42497.orig/src/lib/ecore_con/ecore_con_info.c	2009-09-18 18:13:46.000000000 +0200
+++ ecore-r42497/src/lib/ecore_con/ecore_con_info.c	2009-09-18 18:14:08.000000000 +0200
@@ -243,7 +243,7 @@
 	  freeaddrinfo(result);
 	err = write(fd[1], "", 1);
 	close(fd[1]);
-# ifdef __USE_ISOC99
+# if 0
 	_Exit(0);
 # else
 	_exit(0);
--- ecore-r42497.orig/src/lib/ecore_con/ecore_con_dns.c	2009-09-18 18:14:35.000000000 +0200
+++ ecore-r42497/src/lib/ecore_con/ecore_con_dns.c	2009-09-18 18:14:43.000000000 +0200
@@ -135,7 +135,7 @@
 	     write(fd[1], "", 1);
 	  }
 	close(fd[1]);
-# ifdef __USE_ISOC99
+# if 0
 	_Exit(0);
 # else
 	_exit(0);
