diff -Naur a/gdk-pixbuf/meson.build b/gdk-pixbuf/meson.build
--- a/gdk-pixbuf/meson.build	2018-06-15 20:13:23.593361199 +0200
+++ b/gdk-pixbuf/meson.build	2018-06-15 20:13:42.601469091 +0200
@@ -232,15 +232,15 @@
 
 # The 'loaders.cache' used for testing, so we don't accidentally
 # load the installed cache; we always build it by default
-loaders_cache = custom_target('loaders.cache',
-                              output: 'loaders.cache',
-                              capture: true,
-                              command: [
-                                gdk_pixbuf_query_loaders,
-                                dynamic_loaders,
-                              ],
-                              build_by_default: true)
-loaders_dep = declare_dependency(sources: [ loaders_cache ])
+#loaders_cache = custom_target('loaders.cache',
+#                              output: 'loaders.cache',
+#                              capture: true,
+#                              command: [
+#                                gdk_pixbuf_query_loaders,
+#                                dynamic_loaders,
+#                              ],
+#                              build_by_default: true)
+#loaders_dep = declare_dependency(sources: [ loaders_cache ])
 
 build_gir = get_option('gir')
 if build_gir
diff -Naur a/meson.build b/meson.build
--- a/meson.build	2018-06-15 20:13:23.573361085 +0200
+++ b/meson.build	2018-06-15 20:13:32.761413236 +0200
@@ -398,7 +398,7 @@
 # i18n
 subdir('po')
 
-subdir('tests')
+#subdir('tests')
 subdir('thumbnailer')
 
 # Documentation
diff -Naur a/thumbnailer/meson.build b/thumbnailer/meson.build
--- a/thumbnailer/meson.build	2018-06-15 20:13:23.605361267 +0200
+++ b/thumbnailer/meson.build	2018-06-15 20:13:50.305512825 +0200
@@ -12,22 +12,22 @@
                                          c_args: common_cflags,
                                          dependencies: gdk_pixbuf_deps + [ gdkpixbuf_dep ])
 
-custom_target('thumbnailer',
-              input: 'gdk-pixbuf-thumbnailer.thumbnailer.in',
-              output: 'gdk-pixbuf-thumbnailer.thumbnailer',
-              command: [
-                gen_thumbnailer,
-                '--printer=@0@'.format(gdk_pixbuf_print_mime_types.full_path()),
-                '--pixdata=@0@'.format(gdk_pixbuf_pixdata.full_path()),
-                '--loaders=@0@'.format(loaders_cache.full_path()),
-                '--bindir=@0@'.format(gdk_pixbuf_bindir),
-                '@INPUT@',
-                '@OUTPUT@',
-              ],
-              depends: [
-                gdk_pixbuf_print_mime_types,
-                gdk_pixbuf_pixdata,
-                loaders_cache,
-              ],
-              install: true,
-              install_dir: join_paths(gdk_pixbuf_datadir, 'thumbnailers'))
+#custom_target('thumbnailer',
+#              input: 'gdk-pixbuf-thumbnailer.thumbnailer.in',
+#              output: 'gdk-pixbuf-thumbnailer.thumbnailer',
+#              command: [
+#                gen_thumbnailer,
+#                '--printer=@0@'.format(gdk_pixbuf_print_mime_types.full_path()),
+#                '--pixdata=@0@'.format(gdk_pixbuf_pixdata.full_path()),
+#                '--loaders=@0@'.format(loaders_cache.full_path()),
+#                '--bindir=@0@'.format(gdk_pixbuf_bindir),
+#                '@INPUT@',
+#                '@OUTPUT@',
+#              ],
+#              depends: [
+#                gdk_pixbuf_print_mime_types,
+#                gdk_pixbuf_pixdata,
+#                loaders_cache,
+#              ],
+#              install: true,
+#              install_dir: join_paths(gdk_pixbuf_datadir, 'thumbnailers'))
