diff -Naur libxcb-1.9.2/configure.ac libxcb-1.9.2.patch/configure.ac
--- libxcb-1.9.2/configure.ac	2013-11-07 23:03:07.000000000 +0100
+++ libxcb-1.9.2.patch/configure.ac	2013-12-06 15:33:09.888478982 +0100
@@ -2,20 +2,22 @@
 # Process this file with autoconf to produce a configure script.
 
 # Initialize Autoconf
-AC_PREREQ(2.57)
 AC_INIT([libxcb],
         1.9.2,
         [xcb@lists.freedesktop.org])
+
+AC_PREREQ(2.57)
+
+# Initialize Automake
+AM_INIT_AUTOMAKE([foreign dist-bzip2])
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+
 AC_CONFIG_SRCDIR([xcb.pc.in])
 # Set common system defines for POSIX extensions, such as _GNU_SOURCE
 # Must be called before any macros that run the compiler (like AC_PROG_LIBTOOL)
 # to avoid autoconf errors.
 AC_USE_SYSTEM_EXTENSIONS
 
-# Initialize Automake
-AM_INIT_AUTOMAKE([foreign dist-bzip2])
-m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-
 AM_PATH_PYTHON([2.6])
 
 PKG_CHECK_MODULES(CHECK, [check >= 0.9.4], [HAVE_CHECK=yes], [HAVE_CHECK=no])
