From cb5fd9a497a8ad624bf4f4950723d6da63a1f1d1 Mon Sep 17 00:00:00 2001
From: 5schatten <supervisedthinking@gmail.com>
Date: Thu, 29 Aug 2019 12:01:31 +0200
Subject: [PATCH] libretro/Makefile: add platform unix if platform armv

---
 yabause/src/libretro/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
index 5589969fd..0ca2752c1 100644
--- a/yabause/src/libretro/Makefile
+++ b/yabause/src/libretro/Makefile
@@ -51,6 +51,10 @@ else ifneq ($(findstring MINGW,$(shell uname -a)),)
 	system_platform = win
 endif
 
+ifneq (,$(findstring armv,$(platform)))
+	override platform += unix
+endif
+
 CORE_DIR := .
 
 TARGET_NAME = yabause
