diff -Naur vdr-plugin-restfulapi-0.2.1.4/Makefile vdr-plugin-restfulapi-0.2.1.4.patch/Makefile
--- vdr-plugin-restfulapi-0.2.1.4/Makefile	2015-03-18 23:12:01.000000000 +0100
+++ vdr-plugin-restfulapi-0.2.1.4.patch/Makefile	2016-01-07 05:24:24.036888492 +0100
@@ -49,7 +49,7 @@
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
-LIBS    += $(shell cxxtools-config --libs) -lcxxtools-http
+LIBS += $(shell pkg-config --libs cxxtools-http cxxtools)
 CONFDIR  = $(call PKGCFG,configdir)
 PLGCONFDIR = $(CONFDIR)/plugins/$(PLUGIN)
 
