From 0132877e6017740184a0da88059940a30ef1a4b6 Mon Sep 17 00:00:00 2001
From: cvh <namerp@googlemail.com>
Date: Tue, 13 Feb 2018 15:11:11 +0100
Subject: [PATCH] webapp path

---
 tools.cpp | 2 +-
 tools.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools.cpp b/tools.cpp
index 8a73a0f..293a576 100644
--- a/tools.cpp
+++ b/tools.cpp
@@ -214,7 +214,7 @@ void Settings::initDefault()
   SetIp((string)"0.0.0.0");
   SetEpgImageDirectory((string)"/var/cache/vdr/epgimages");
   SetChannelLogoDirectory((string)"/usr/share/vdr/channel-logos");
-  SetWebappDirectory((string)"/var/lib/vdr/plugins/restfulapi/webapp");
+  SetWebappDirectory((string)"/storage/.kodi/addons/service.multimedia.vdr-addon/res/plugins/restfulapi");
   SetHeaders((string)"true");
   webapp_filetypes_filename = "webapp_file_types.conf";
 }
diff --git a/tools.h b/tools.h
index ca38f5e..b7f7a78 100644
--- a/tools.h
+++ b/tools.h
@@ -41,7 +41,7 @@
 #define RESTFULAPI_EXTENSIONS
 
 #ifndef DOCUMENT_ROOT
-#define DOCUMENT_ROOT "/var/lib/vdr/plugins/restfulapi/"
+#define DOCUMENT_ROOT "/storage/.kodi/addons/service.multimedia.vdr-addon/res/plugins/restfulapi/"
 #endif
 
 class Settings
-- 
2.51.0

