--- a/tools.cpp
+++ b/tools.cpp
@@ -214,7 +214,7 @@ void Settings::initDefault()
   SetIp((string)"0.0.0.0");
   SetEpgImageDirectory((string)"/var/cache/vdr/epgimages");
   SetChannelLogoDirectory((string)"/usr/share/vdr/channel-logos");
-  SetWebappDirectory((string)"/var/lib/vdr/plugins/restfulapi/webapp");
+  SetWebappDirectory((string)"/storage/.kodi/addons/service.multimedia.vdr-addon/res/plugins/restfulapi");
   SetHeaders((string)"true");
   webapp_filetypes_filename = "webapp_file_types.conf";
 }
--- a/tools.h
+++ b/tools.h
@@ -40,7 +40,7 @@
 #define RESTFULAPI_EXTENSIONS
 
 #ifndef DOCUMENT_ROOT
-#define DOCUMENT_ROOT "/var/lib/vdr/plugins/restfulapi/"
+#define DOCUMENT_ROOT "/storage/.kodi/addons/service.multimedia.vdr-addon/res/plugins/restfulapi/"
 #endif
 
 class Settings
