From c03f04cab1c7ac901533b6503bf67fe8b8aa6d3c Mon Sep 17 00:00:00 2001
From: mglae <mglmail@arcor.de>
Date: Thu, 1 Jul 2021 23:43:23 +0200
Subject: [PATCH] fix static lib order

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f0b1467..dc91072 100644
--- a/Makefile
+++ b/Makefile
@@ -51,7 +51,7 @@ SOFILE = libvdr-$(PLUGIN).so
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
-LIBS    += $(shell cxxtools-config --libs) -lcxxtools-http
+LIBS    +=  -lcxxtools-http $(shell cxxtools-config --libs) -lssl
 CONFDIR  = $(call PKGCFG,configdir)
 PLGCONFDIR = $(CONFDIR)/plugins/$(PLUGIN)
 
-- 
2.51.0

