--- a/Makefile	2016-10-16 11:52:25.000000000 +0200
+++ b/Makefile	2021-06-23 22:49:33.447650255 +0200
@@ -51,7 +51,7 @@ SOFILE = libvdr-$(PLUGIN).so
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
-LIBS    += $(shell cxxtools-config --libs) -lcxxtools-http
+LIBS    +=  -lcxxtools-http $(shell cxxtools-config --libs)
 CONFDIR  = $(call PKGCFG,configdir)
 PLGCONFDIR = $(CONFDIR)/plugins/$(PLUGIN)
 