diff -Naur linux-3.7.2/drivers/media/rc/fintek-cir.c linux-3.7.2.patch/drivers/media/rc/fintek-cir.c
--- linux-3.7.2/drivers/media/rc/fintek-cir.c	2013-01-16 11:43:34.103841326 +0100
+++ linux-3.7.2.patch/drivers/media/rc/fintek-cir.c	2013-01-16 11:43:48.022744456 +0100
@@ -557,6 +557,8 @@
 	/* rx resolution is hardwired to 50us atm, 1, 25, 100 also possible */
 	rdev->rx_resolution = US_TO_NS(CIR_SAMPLE_PERIOD);
 
+	fintek->rdev = rdev;
+
 	ret = -EBUSY;
 	/* now claim resources */
 	if (!request_region(fintek->cir_addr,
@@ -572,7 +574,7 @@
 		goto exit_free_irq;
 
 	device_init_wakeup(&pdev->dev, true);
-	fintek->rdev = rdev;
+
 	fit_pr(KERN_NOTICE, "driver has been successfully loaded\n");
 	if (debug)
 		cir_dump_regs(fintek);
diff -Naur linux-3.7.2/drivers/media/rc/ite-cir.c linux-3.7.2.patch/drivers/media/rc/ite-cir.c
--- linux-3.7.2/drivers/media/rc/ite-cir.c	2013-01-16 11:43:34.103841326 +0100
+++ linux-3.7.2.patch/drivers/media/rc/ite-cir.c	2013-01-16 11:43:48.023744450 +0100
@@ -1591,6 +1591,8 @@
 	rdev->driver_name = ITE_DRIVER_NAME;
 	rdev->map_name = RC_MAP_RC6_MCE;
 
+	itdev->rdev = rdev;
+
 	ret = -EBUSY;
 	/* now claim resources */
 	if (!request_region(itdev->cir_addr,
diff -Naur linux-3.7.2/drivers/media/rc/nuvoton-cir.c linux-3.7.2.patch/drivers/media/rc/nuvoton-cir.c
--- linux-3.7.2/drivers/media/rc/nuvoton-cir.c	2013-01-16 11:43:34.103841326 +0100
+++ linux-3.7.2.patch/drivers/media/rc/nuvoton-cir.c	2013-01-16 11:43:48.024744444 +0100
@@ -1068,6 +1068,7 @@
 	/* tx bits */
 	rdev->tx_resolution = XYZ;
 #endif
+	nvt->rdev = rdev;
 
 	ret = -EBUSY;
 	/* now claim resources */
@@ -1092,7 +1093,7 @@
 		goto exit_free_wake_irq;
 
 	device_init_wakeup(&pdev->dev, true);
-	nvt->rdev = rdev;
+
 	nvt_pr(KERN_NOTICE, "driver has been successfully loaded\n");
 	if (debug) {
 		cir_dump_regs(nvt);
