diff --git a/src/compat/compat.h b/src/compat/compat.h
index 1aa0e7b..751ace7 100644
--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -43,7 +43,7 @@
 #error "WireGuard requires Linux >= 3.10"
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 7, 0)
 #error "WireGuard has been merged into Linux >= 5.6 and therefore this compatibility module is no longer required."
 #endif
 
