From 3cb23c7b05be36e130bd1930dad3382ce8c08c51 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 26 Oct 2014 00:26:09 +0300
Subject: [PATCH] meh glib

---
 gdk/Makefile.am |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gdk/Makefile.am b/gdk/Makefile.am
index c880f6e..7661530 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -307,7 +307,7 @@ lib_LTLIBRARIES = $(gdktargetlib)
 
 EXTRA_LTLIBRARIES = libgdk-x11-2.0.la libgdk-win32-2.0.la libgdk-quartz-2.0.la libgdk-directfb-2.0.la
 
-MAINTAINERCLEANFILES = $(gdk_built_sources) stamp-gdkenumtypes.h
+MAINTAINERCLEANFILES = $(gdk_built_sources)
 EXTRA_DIST += $(gdk_built_sources)
 EXTRA_HEADERS =
 
@@ -333,7 +333,7 @@ BUILT_SOURCES = \
 	$(gdk_built_sources)			\
 	gdkconfig.h
 
-gdkenumtypes.h: stamp-gdkenumtypes.h
+gdkenumtypes.h:
 	@true
 stamp-gdkenumtypes.h: @REBUILD@ $(gdk_public_h_sources) gdkenumtypes.h.template
 	( cd $(srcdir) && $(GLIB_MKENUMS) --template gdkenumtypes.h.template \
-- 
1.7.2.5

