From fca2ac06da064b133101f215efd5fbd738fb3ef9 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 27 Mar 2014 14:26:27 +0200
Subject: [PATCH] no gtk-doc

---
 Makefile.am  |    3 +--
 configure.ac |   12 ------------
 2 files changed, 1 insertions(+), 14 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 8e3a2f1..35e94d2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,7 @@
 include $(top_srcdir)/Makefile.decl
 
 SRC_SUBDIRS = gdk gtk modules demos tests perf
-SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
+SUBDIRS = po po-properties $(SRC_SUBDIRS) m4macros build
 
 # require automake 1.4
 AUTOMAKE_OPTIONS = 1.7
@@ -136,7 +136,6 @@ MAINTAINERCLEANFILES = \
 	$(srcdir)/mkinstalldirs \
 	$(srcdir)/omf.make \
 	$(srcdir)/xmldocs.make \
-	$(srcdir)/gtk-doc.make \
 	$(srcdir)/ChangeLog \
 	`find "$(srcdir)" -type f -name Makefile.in -print`
 
diff --git a/configure.ac b/configure.ac
index 245a87f..f6ede8a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1594,8 +1594,6 @@ GOBJECT_INTROSPECTION_CHECK([0.9.3])
 # Checks for gtk-doc and docbook-tools
 ##################################################
 
-GTK_DOC_CHECK([1.11])
-
 AC_CHECK_PROG(DB2HTML, db2html, true, false)
 AM_CONDITIONAL(HAVE_DOCBOOK, $DB2HTML)
 
@@ -1727,16 +1725,6 @@ demos/Makefile
 demos/gtk-demo/Makefile
 demos/gtk-demo/geninclude.pl
 tests/Makefile
-docs/Makefile
-docs/reference/Makefile
-docs/reference/gdk/Makefile
-docs/reference/gdk/version.xml
-docs/reference/gtk/Makefile
-docs/reference/gtk/version.xml
-docs/reference/libgail-util/Makefile
-docs/faq/Makefile
-docs/tools/Makefile
-docs/tutorial/Makefile
 build/Makefile
 build/win32/Makefile
 build/win32/vs9/Makefile
-- 
1.7.2.5

