diff -Naur libtirpc-0.2.2/doc/Makefile.am libtirpc-0.2.2.patch/doc/Makefile.am
--- libtirpc-0.2.2/doc/Makefile.am	2011-05-02 14:10:40.000000000 +0200
+++ libtirpc-0.2.2.patch/doc/Makefile.am	2014-01-10 06:17:51.829113901 +0100
@@ -1,5 +1,5 @@
 
 install: install-am
 	$(mkinstalldirs) $(DESTDIR)/$(sysconfdir)
-	cp -p ./etc_netconfig $(DESTDIR)/$(sysconfdir)/netconfig
+	cp -p $(top_srcdir)/doc/etc_netconfig $(DESTDIR)/$(sysconfdir)/netconfig
 	chmod 0644 $(DESTDIR)/$(sysconfdir)/netconfig
