diff --git a/yabause/src/core/video/opengl/common/src/commongl.c b/yabause/src/core/video/opengl/common/src/commongl.c
index 3194e9e..4069efd 100755
--- a/yabause/src/core/video/opengl/common/src/commongl.c
+++ b/yabause/src/core/video/opengl/common/src/commongl.c
@@ -809,6 +809,9 @@ void syncVdp1FBBuffer(u32 addr) {
 }
 
 void updateVdp1DrawingFBMem(int frame) {
+#ifndef GL_PIXEL_BUFFER_BARRIER_BIT
+#define GL_PIXEL_BUFFER_BARRIER_BIT       0x00000080
+#endif
     if (_Ygl->vdp1fb_write_buf[frame] != NULL) {
       glBindTexture(GL_TEXTURE_2D, _Ygl->vdp1AccessTex[frame]);
       glBindBuffer(GL_PIXEL_UNPACK_BUFFER, _Ygl->vdp1_pbo[frame]);
