diff --git a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
index 2e50ea7..9a05b49 100644
--- a/yabause/src/libretro/Makefile
+++ b/yabause/src/libretro/Makefile
@@ -14,6 +14,11 @@ HAVE_SYS_PARAM_H = 1
 HAVE_FSEEKO = 0
 USE_RTHREADS = 0
 HAVE_CDROM = 0
+HAVE_OPENGL ?= 0
+
+ifeq ($(HAVE_OPENGL),1)
+	LDFLAGS += -lGL
+endif
 
 ifeq ($(platform),)
 	platform = unix
@@ -56,7 +61,7 @@ ifneq (,$(findstring unix,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	# ARM
 	ifneq (,$(findstring armv,$(platform)))
@@ -90,7 +95,7 @@ else ifeq ($(platform), jetson-nano)
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 	FLAGS += -march=armv8-a+crc -mtune=cortex-a57
@@ -101,7 +106,7 @@ else ifeq ($(platform), arm64)
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 
@@ -111,7 +116,7 @@ else ifneq (,$(findstring rockpro64,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 	FORCE_GLES = 1
@@ -123,7 +128,7 @@ else ifeq ($(platform), odroid-n2)
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 	FORCE_GLES = 1
@@ -139,7 +144,7 @@ else ifneq (,$(findstring odroid,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-	LDFLAGS += -lpthread -lGL
+	LDFLAGS += -lpthread
 	ARCH_IS_LINUX = 1
 	BOARD := $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
 
