diff --git a/drivers/amlogic/display/vout/vout_serve.c b/drivers/amlogic/display/vout/vout_serve.c
index d91d40d..029b36d 100644
--- a/drivers/amlogic/display/vout/vout_serve.c
+++ b/drivers/amlogic/display/vout/vout_serve.c
@@ -596,10 +596,16 @@ static int __init get_vout_init_mode(char *str)
 	char *option;
 	int count = 3;
 	char find = 0;
+	char str2[1024];
+	char *ptr2 = str2;
 
 	if (NULL == str)
 		return -EINVAL;
 
+	strcpy(str2, str);
+	strcat(str2, ",en"); // logo was already displayed by uboot
+	ptr = ptr2 = str2;
+
 	do {
 		if (!isalpha(*ptr) && !isdigit(*ptr)) {
 			find = 1;
@@ -611,7 +617,7 @@ static int __init get_vout_init_mode(char *str)
 
 	sep[0] = *ptr;
 	sep[1] = '\0';
-	while ((count--) && (option = strsep(&str, sep))) {
+	while ((count--) && (option = strsep(&ptr2, sep))) {
 		/* vout_log_info("%s\n", option); */
 		str2lower(option);
 		vout_init_mode_parse(option);
@@ -619,7 +625,7 @@ static int __init get_vout_init_mode(char *str)
 
 	return 0;
 }
-__setup("vout=", get_vout_init_mode);
+__setup("hdmimode=", get_vout_init_mode);
 
 MODULE_AUTHOR("Platform-BJ <platform.bj@amlogic.com>");
 MODULE_DESCRIPTION("VOUT Server Module");
