diff --git a/drivers/media/rc/keymaps/rc-wetek-hub.c b/drivers/media/rc/keymaps/rc-wetek-hub.c
index 0955ecfcb77..0da1b237b09 100644
--- a/drivers/media/rc/keymaps/rc-wetek-hub.c
+++ b/drivers/media/rc/keymaps/rc-wetek-hub.c
@@ -12,6 +12,7 @@
 #include <linux/module.h>
 
 static struct rc_map_table wetek_hub[] = {
+	// WeTek Hub remote
 	{ 0x77f1, KEY_POWER },
 	{ 0x77f2, KEY_HOME },
 	{ 0x77f3, KEY_MUTE },
@@ -24,6 +25,46 @@ static struct rc_map_table wetek_hub[] = {
 	{ 0x77fa, KEY_MENU },
 	{ 0x77fb, KEY_VOLUMEUP },
 	{ 0x77fc, KEY_VOLUMEDOWN },
+
+	// WeTek Pro Remote
+	{ 0x010c, KEY_POWER },
+	{ 0x012a, KEY_RED },
+	{ 0x012b, KEY_GREEN },
+	{ 0x012c, KEY_YELLOW },
+	{ 0x012d, KEY_BLUE },
+	{ 0x0111, KEY_REWIND },
+	{ 0x0112, KEY_PLAYPAUSE },
+	{ 0x0113, KEY_FASTFORWARD },
+	{ 0x0114, KEY_RECORD },
+	{ 0x0117, KEY_MENU },
+	{ 0x011a, KEY_TAB },
+	{ 0x0129, KEY_HOME },
+	{ 0x011d, KEY_BACK },
+	{ 0x010F, KEY_SUBTITLE },
+	{ 0x0127, KEY_MUTE },
+	{ 0x0110, KEY_TEXT },
+	{ 0x010b, KEY_PVR },
+	{ 0x0128, KEY_INFO },
+	{ 0x010a, KEY_CONNECT },
+	{ 0x0122, KEY_UP },
+	{ 0x0123, KEY_DOWN },
+	{ 0x0124, KEY_LEFT },
+	{ 0x0125, KEY_RIGHT },
+	{ 0x0126, KEY_OK },
+	{ 0x011e, KEY_VOLUMEUP },
+	{ 0x011f, KEY_VOLUMEDOWN },
+	{ 0x0120, KEY_PAGEUP },
+	{ 0x0121, KEY_PAGEDOWN },
+	{ 0x0101, KEY_1 },
+	{ 0x0102, KEY_2 },
+	{ 0x0103, KEY_3 },
+	{ 0x0104, KEY_4 },
+	{ 0x0105, KEY_5 },
+	{ 0x0106, KEY_6 },
+	{ 0x0107, KEY_7 },
+	{ 0x0108, KEY_8 },
+	{ 0x0109, KEY_9 },
+	{ 0x0100, KEY_0 },
 };
 
 static struct rc_map_list wetek_hub_map = {
