From 7e9c3e465ed680cfea89a15f326f023d0514c586 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 5 Mar 2017 11:20:55 +0100
Subject: [PATCH 1/2] tools: do not build libfdt Python module

---
 tools/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/Makefile b/tools/Makefile
index 5db2a54225..1b60a7244a 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -233,7 +233,7 @@ always := $(hostprogs-y)
 
 # Build a libfdt Python module if swig is available
 # Use 'sudo apt-get install swig libpython-dev' to enable this
-always += $(if $(shell which swig 2> /dev/null),_libfdt.so)
+#always += $(if $(shell which swig 2> /dev/null),_libfdt.so)
 
 # Generated LCD/video logo
 LOGO_H = $(objtree)/include/bmp_logo.h
