diff -Naur marvell-libvmeta-3.8.3/Makefile marvell-libvmeta-3.8.3.patch/Makefile
--- marvell-libvmeta-3.8.3/Makefile	2011-03-07 18:51:17.000000000 +0100
+++ marvell-libvmeta-3.8.3.patch/Makefile	2012-12-27 01:14:34.043086675 +0100
@@ -22,7 +22,7 @@
 	uninstall-host uninstall-target
 
 .c.o:
-	${CC} -fPIC -o $@ -c $<
+	${CC} ${CFLAGS} -fPIC -o $@ -c $<
 
 all: compile install-host install-target 
 
@@ -32,7 +32,7 @@
 	${AR} -rcs $@ ${vmeta_lib_obj}
 
 libvmeta.so: ${vmeta_lib_obj}
-	${CC} -g -s -shared -Wl,-soname,$@ -o $@ ${vmeta_lib_obj} -lbmm
+	${CC} -g -s ${LDFLAGS} -shared -Wl,-soname,$@ -o $@ ${vmeta_lib_obj} -lbmm
 
 clean: clean-local
 
