diff --git a/arch/arm64/boot/dts/amlogic/gxl_p212_1g_lepotato.dts b/arch/arm64/boot/dts/amlogic/gxl_p212_1g_lepotato.dts
new file mode 100644
index 0000000..8272581
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/gxl_p212_1g_lepotato.dts
@@ -0,0 +1,25 @@
+#include "gxl_p212_1g.dts"
+
+/ {
+        leds: gpio_leds {
+                compatible = "gpio-leds";
+
+		system {
+			label = "librecomputer:system-status";
+			gpios = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+			panic-indicator;
+		};
+
+		blue {
+			label = "librecomputer:blue";
+			gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
+	gpio_keypad {
+		status = "disabled";
+	};
+
+};
diff --git a/arch/arm64/boot/dts/amlogic/gxl_p212_2g_lepotato.dts b/arch/arm64/boot/dts/amlogic/gxl_p212_2g_lepotato.dts
new file mode 100644
index 0000000..75f8bc7
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/gxl_p212_2g_lepotato.dts
@@ -0,0 +1,25 @@
+#include "gxl_p212_2g.dts"
+
+/ {
+        leds: gpio_leds {
+                compatible = "gpio-leds";
+
+		system {
+			label = "librecomputer:system-status";
+			gpios = <&gpio GPIODV_24 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+			panic-indicator;
+		};
+
+		blue {
+			label = "librecomputer:blue";
+			gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
+	gpio_keypad {
+		status = "disabled";
+	};
+
+};
