diff -Naur eglibc-2.15-17022/elf/rtld.c eglibc-2.15-17022.patch/elf/rtld.c
--- eglibc-2.15-17022/elf/rtld.c	2012-02-07 05:47:54.000000000 +0100
+++ eglibc-2.15-17022.patch/elf/rtld.c	2012-02-10 04:52:55.462289159 +0100
@@ -1398,7 +1398,7 @@
 	      char *copy = malloc (len);
 	      if (copy == NULL)
 		_dl_fatal_printf ("out of memory\n");
-	      l->l_libname->name = l->l_name = memcpy (copy, dsoname, len);
+	      l->l_libname->name = memcpy (copy, dsoname, len);
 	    }
 
 	  /* Add the vDSO to the object list.  */
