diff -Naur redream.git/CMakeLists.txt redream.patch/CMakeLists.txt
--- redream.git/CMakeLists.txt	2017-05-09 14:11:24.670774187 -0500
+++ redream.patch/CMakeLists.txt	2017-05-09 14:13:00.733822968 -0500
@@ -98,11 +98,13 @@
 list(APPEND REDREAM_DEFS GLEW_STATIC)
 
 # glfw
+if(NOT BUILD_LIBRETRO)
 set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "")
 set(GLFW_BUILD_TESTS OFF CACHE BOOL "")
 set(GLFW_BUILD_DOCS OFF CACHE BOOL "")
 add_subdirectory(deps/glfw EXCLUDE_FROM_ALL)
 list(APPEND REDREAM_INCLUDE_DIRS deps/glfw/include)
+endif()
 
 # inih
 add_library(inih STATIC deps/inih/ini.c)
