diff -Naur marvell-ipp-0.2.1/example/Rules.make marvell-ipp-0.2.1.patch/example/Rules.make
--- marvell-ipp-0.2.1/example/Rules.make	2011-03-07 18:51:40.000000000 +0100
+++ marvell-ipp-0.2.1.patch/example/Rules.make	2012-12-26 23:48:51.811467005 +0100
@@ -28,26 +28,26 @@
 #PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi/bin
 PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi-4.4.4/bin
 TOOLCHAIN_PREFIX=arm-marvell-linux-gnueabi
-CFLAGS= -mcpu=marvell-f 
+#CFLAGS= -mcpu=marvell-f 
 endif
 
 #==============================================================================
 # GNU           binaries             (server admin update)
 #==============================================================================
-CC=$(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
-CXX=$(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
-AR=$(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-ar
-AS=$(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-as
-LN=$(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
+CC ?= $(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
+CXX?= $(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
+AR ?= $(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-ar
+AS ?= $(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-as
+LN ?= $(PATH_GNU_BIN)/$(TOOLCHAIN_PREFIX)-gcc
 
 #==============================================================================
 # GNU build options:                (build engineer update)
 #==============================================================================
-CFLAGS+= -O3 -Wall -mabi=aapcs-linux -fPIC -D_IPP_LINUX
-CXXFLAGS= $(CFLAGS)
-ASFLAGS= -k -mcpu=iwmmxt2
-ARFLAGS=
-LNFLAGS= -lpthread -ldl -lm -lrt $(PGO_FLAGS) -Wl,-R ./
+CFLAGS += -O3 -Wall -mabi=aapcs-linux -fPIC -D_IPP_LINUX
+CXXFLAGS ?= $(CFLAGS)
+ASFLAGS ?= -k -mcpu=iwmmxt2
+ARFLAGS =
+LNFLAGS ?= -lpthread -ldl -lm -lrt $(PGO_FLAGS) -Wl,-R ./
 
 #==============================================================================
 # User root path												(user update)
