diff -Naur xf86-video-fglrx-legacy-12.6/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h xf86-video-fglrx-legacy-12.6.patch/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h
--- xf86-video-fglrx-legacy-12.6/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h	2012-07-18 17:58:01.000000000 +0200
+++ xf86-video-fglrx-legacy-12.6.patch/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h	2013-06-06 23:56:41.440839811 +0200
@@ -89,7 +89,6 @@
     FGL_ASIC_ID(0x677B),
     FGL_ASIC_ID(0x6772),
     FGL_ASIC_ID(0x6779),
-    FGL_ASIC_ID(0x6770),
     FGL_ASIC_ID(0x671F),
     FGL_ASIC_ID(0x6718),
     FGL_ASIC_ID(0x6719),
@@ -193,7 +192,6 @@
     FGL_ASIC_ID(0x68D1),
     FGL_ASIC_ID(0x68C7),
     FGL_ASIC_ID(0x68E0),
-    FGL_ASIC_ID(0x68E1),
     FGL_ASIC_ID(0x68F0),
     FGL_ASIC_ID(0x68F1),
     FGL_ASIC_ID(0x68E4),
