diff -Naur xf86-video-fglrx-legacy-12.6/common/lib/modules/fglrx/build_mod/firegl_public.c xf86-video-fglrx-legacy-12.6.patch/common/lib/modules/fglrx/build_mod/firegl_public.c
--- xf86-video-fglrx-legacy-12.6/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-07-04 21:43:47.000000000 +0200
+++ xf86-video-fglrx-legacy-12.6.patch/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-07-09 19:33:02.694483911 +0200
@@ -940,7 +940,7 @@
 
 static struct pci_driver fglrx_pci_driver = 
 {
-    .name           = "fglrx_pci",
+    .name           = "fglrx_legacy_pci",
     .id_table       = fglrx_pci_table,
     .probe          = fglrx_pci_probe,
 #ifdef CONFIG_PM
