diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-08-26 12:22:58.175457157 -0700
+++ b/CMakeLists.txt	2016-08-26 12:22:58.234457734 -0700
@@ -6,11 +6,12 @@
 
 enable_language(CXX)
 
-find_package(kodi REQUIRED)
+find_package(Kodi REQUIRED)
 find_package(libretro-craft REQUIRED)
 
 set(DEPLIBS craft)
 set(CRAFT_CUSTOM_BINARY ${CRAFT_LIB}
-                              game.libretro.craft${CMAKE_SHARED_LIBRARY_SUFFIX} craft)
+                           ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} craft)
+
+build_addon(${PROJECT_NAME} CRAFT DEPLIBS)
 
-build_addon(game.libretro.craft CRAFT DEPLIBS)
