From 1191268f518d298c8617a4a74644ee1bb627f614 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 25 Aug 2014 15:35:14 +0300
Subject: [PATCH] move magic.mgc to addon/data

---
 src/magic.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/magic.c b/src/magic.c
index e4bd12b..59c1456 100644
--- a/src/magic.c
+++ b/src/magic.c
@@ -100,7 +100,7 @@ get_default_magic(void)
 	if ((home = getenv("HOME")) == NULL)
 		return MAGIC;
 
-	if (asprintf(&hmagicpath, "%s/.magic.mgc", home) < 0)
+	if (asprintf(&hmagicpath, "%s/.kodi/addons/virtual.system-tools/data/magic.mgc", home) < 0)
 		return MAGIC;
 	if (stat(hmagicpath, &st) == -1) {
 		free(hmagicpath);
-- 
1.7.2.5

