diff --git a/src/apprentice.c b/src/apprentice.c
index 961e83d..b7d500c 100644
--- a/src/apprentice.c
+++ b/src/apprentice.c
@@ -348,11 +348,9 @@ apprentice_1(struct magic_set *ms, const char *fn, int action)
 protected void
 file_ms_free(struct magic_set *ms)
 {
-	size_t i;
 	if (ms == NULL)
 		return;
-	for (i = 0; i < MAGIC_SETS; i++)
-		mlist_free(ms->mlist[i]);
+	mlist_free(ms->mlist[0]);
 	free(ms->o.pbuf);
 	free(ms->o.buf);
 	free(ms->c.li);
