diff -uNr a/menu/menu_setting.c b/menu/menu_setting.c
--- a/menu/menu_setting.c	2025-03-08 15:10:42.948576500 +0900
+++ b/menu/menu_setting.c	2025-03-10 21:37:34.480367623 +0900
@@ -9565,6 +9565,14 @@
 {
    systemd_service_toggle(LAKKA_SAFESHUTDOWN_PATH, (char*)"retroflag_picase_safeshutdown.service",
          *setting->value.target.boolean);
+   if(*setting->value.target.boolean)
+   {
+      system("/usr/bin/retroflag_picase_install_gpio-poweroff_overlay.sh enable");
+   }
+   else
+   {
+      system("/usr/bin/retroflag_picase_install_gpio-poweroff_overlay.sh disable");
+   }
 }
 #endif
 
