diff -uNr a/intl/msg_hash_us.h b/intl/msg_hash_us.h
--- a/intl/msg_hash_us.h	2025-07-17 00:32:37.902028400 +0900
+++ b/intl/msg_hash_us.h	2025-07-17 10:20:08.413280125 +0900
@@ -16412,11 +16412,11 @@
 #ifdef HAVE_RETROFLAG
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_SAFESHUTDOWN_ENABLE,
-   "Retroflag Safe Shutdown"
+   "Retroflag Safe Shutdown (Reboot required)"
    )
 MSG_HASH(
    MENU_ENUM_SUBLABEL_SAFESHUTDOWN_ENABLE,
-   "For use with compatible Retroflag case."
+   "For use with compatible Retroflag case. Reboot is required when changing."
    )
 #endif
 MSG_HASH(
