diff -Naur elisa-plugins-bad-0.5.30/elisa/plugins/pigment/pigment_frontend.py elisa-plugins-bad-0.5.30.patch/elisa/plugins/pigment/pigment_frontend.py
--- elisa-plugins-bad-0.5.30/elisa/plugins/pigment/pigment_frontend.py	2009-03-02 10:14:06.000000000 +0100
+++ elisa-plugins-bad-0.5.30.patch/elisa/plugins/pigment/pigment_frontend.py	2009-03-10 16:57:52.000000000 +0100
@@ -453,7 +453,10 @@
             else:
                 self.gtk_window.unfullscreen()
         else:
-            viewport.fullscreen = fullscreen
+            viewport.set_position(0, 0)
+            resolution = viewport.get_screen_resolution()
+            viewport.set_size(resolution[0], resolution[1])
+
             if not headless:
                 viewport.show()
 
