Index: udev-136/rules/rules.d/50-udev-default.rules
===================================================================
--- udev-136.orig/rules/rules.d/50-udev-default.rules
+++ udev-136/rules/rules.d/50-udev-default.rules
@@ -34,7 +34,7 @@ KERNEL=="winradio*",		GROUP="video"
 
 # graphics
 KERNEL=="agpgart",		MODE="0600", GROUP="video"
-KERNEL=="card[0-9]*",		NAME="dri/%k"
+KERNEL=="card[0-9]*",		NAME="dri/%k" GROUP="video"
 KERNEL=="pmu",			GROUP="video"
 KERNEL=="nvidia*|nvidiactl*",	GROUP="video"
 SUBSYSTEM=="graphics",		GROUP="video"
