diff -Naur lirc-20100408/daemons/hardware.h lirc-20100408.patch/daemons/hardware.h
--- lirc-20100408/daemons/hardware.h	2009-01-04 23:17:39.000000000 +0100
+++ lirc-20100408.patch/daemons/hardware.h	2010-04-09 16:05:14.406356679 +0200
@@ -13,7 +13,7 @@
 #ifndef _HARDWARE_H
 #define _HARDWARE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "ir_remote_types.h"
 
 struct hardware
diff -Naur lirc-20100408/daemons/hw_accent.h lirc-20100408.patch/daemons/hw_accent.h
--- lirc-20100408/daemons/hw_accent.h	2007-07-29 20:20:06.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_accent.h	2010-04-09 16:05:14.356327542 +0200
@@ -27,7 +27,7 @@
 #ifndef _HW_ACCENT_H
 #define _HW_ACCENT_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int accent_decode (struct ir_remote *remote,
 		   ir_code *prep,
diff -Naur lirc-20100408/daemons/hw_bte.h lirc-20100408.patch/daemons/hw_bte.h
--- lirc-20100408/daemons/hw_bte.h	2007-07-29 20:20:06.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_bte.h	2010-04-09 16:05:14.362326501 +0200
@@ -26,7 +26,7 @@
 #ifndef HW_BTE_H
 #define HW_BTE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int bte_decode(struct ir_remote *remote,
 	       ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_caraca.h lirc-20100408.patch/daemons/hw_caraca.h
--- lirc-20100408/daemons/hw_caraca.h	2007-07-29 20:20:06.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_caraca.h	2010-04-09 16:05:14.400327269 +0200
@@ -13,7 +13,7 @@
 #ifndef HW_CARACA_H
 #define HW_CARACA_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int caraca_decode(struct ir_remote *remote,
 		  ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_creative.h lirc-20100408.patch/daemons/hw_creative.h
--- lirc-20100408/daemons/hw_creative.h	2007-07-29 20:20:07.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_creative.h	2010-04-09 16:05:14.393357176 +0200
@@ -13,7 +13,7 @@
 #ifndef _HW_CREATIVE_H
 #define _HW_CREATIVE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int creative_decode(struct ir_remote *remote,
 		    ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_ea65.h lirc-20100408.patch/daemons/hw_ea65.h
--- lirc-20100408/daemons/hw_ea65.h	2007-07-29 20:20:07.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_ea65.h	2010-04-09 16:05:14.346327762 +0200
@@ -24,7 +24,7 @@
 #ifndef HW_EA65_H
 #define HW_EA65_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int   ea65_decode(struct ir_remote *remote,
 		  ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_irlink.h lirc-20100408.patch/daemons/hw_irlink.h
--- lirc-20100408/daemons/hw_irlink.h	2008-01-13 21:50:43.000000000 +0100
+++ lirc-20100408.patch/daemons/hw_irlink.h	2010-04-09 16:05:14.350327534 +0200
@@ -23,7 +23,7 @@
 #ifndef HW_IRLINK_H
 #define HW_IRLINK_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int     irlink_decode                (struct ir_remote   *remote,
 				                      ir_code            *prep,
diff -Naur lirc-20100408/daemons/hw_irman.h lirc-20100408.patch/daemons/hw_irman.h
--- lirc-20100408/daemons/hw_irman.h	2007-07-29 20:20:08.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_irman.h	2010-04-09 16:05:14.368352769 +0200
@@ -13,7 +13,7 @@
 #ifndef _HW_IRMAN_H
 #define _HW_IRMAN_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int irman_decode(struct ir_remote *remote,
 		 ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_logitech.h lirc-20100408.patch/daemons/hw_logitech.h
--- lirc-20100408/daemons/hw_logitech.h	2007-07-29 20:20:08.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_logitech.h	2010-04-09 16:05:14.348327369 +0200
@@ -13,7 +13,7 @@
 #ifndef _HW_LOGITECH_H
 #define _HW_LOGITECH_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int logitech_decode(struct ir_remote *remote,
 		    ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_mouseremote.h lirc-20100408.patch/daemons/hw_mouseremote.h
--- lirc-20100408/daemons/hw_mouseremote.h	2007-07-29 20:20:08.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_mouseremote.h	2010-04-09 16:05:14.395328008 +0200
@@ -16,7 +16,7 @@
 #ifndef HW_MOUSEREMOTE_H
 #define HW_MOUSEREMOTE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int mouseremote_decode(struct ir_remote *remote,
 		       ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_mp3anywhere.h lirc-20100408.patch/daemons/hw_mp3anywhere.h
--- lirc-20100408/daemons/hw_mp3anywhere.h	2007-07-29 20:20:08.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_mp3anywhere.h	2010-04-09 16:05:14.364357606 +0200
@@ -15,7 +15,7 @@
 #ifndef HW_MP3ANYWHERE_H
 #define HW_MP3ANYWHERE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int mp3anywhere_decode(struct ir_remote *remote,
 		       ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_mplay.h lirc-20100408.patch/daemons/hw_mplay.h
--- lirc-20100408/daemons/hw_mplay.h	2007-12-22 13:33:16.000000000 +0100
+++ lirc-20100408.patch/daemons/hw_mplay.h	2010-04-09 16:05:14.389357823 +0200
@@ -25,7 +25,7 @@
 #ifndef HW_MPLAY_H
 #define HW_MPLAY_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 extern int mplay_decode(struct ir_remote *remote,
                         ir_code *prep,
diff -Naur lirc-20100408/daemons/hw_pcmak.h lirc-20100408.patch/daemons/hw_pcmak.h
--- lirc-20100408/daemons/hw_pcmak.h	2007-07-29 20:20:08.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_pcmak.h	2010-04-09 16:05:14.402362075 +0200
@@ -14,7 +14,7 @@
 #ifndef HW_PCMAK_H
 #define HW_PCMAK_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int pcmak_decode(struct ir_remote *remote,
 		 ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_pinsys.h lirc-20100408.patch/daemons/hw_pinsys.h
--- lirc-20100408/daemons/hw_pinsys.h	2007-07-29 20:20:09.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_pinsys.h	2010-04-09 16:05:14.352327630 +0200
@@ -14,7 +14,7 @@
 #ifndef _HW_PINSYS_H
 #define _HW_PINSYS_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int is_it_is_it_huh(int port);
 int autodetect(void);
diff -Naur lirc-20100408/daemons/hw_pixelview.h lirc-20100408.patch/daemons/hw_pixelview.h
--- lirc-20100408/daemons/hw_pixelview.h	2007-07-29 20:20:09.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_pixelview.h	2010-04-09 16:05:14.366327740 +0200
@@ -13,7 +13,7 @@
 #ifndef _HW_PIXELVIEW_H
 #define _HW_PIXELVIEW_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int pixelview_decode(struct ir_remote *remote,
 		     ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_silitek.h lirc-20100408.patch/daemons/hw_silitek.h
--- lirc-20100408/daemons/hw_silitek.h	2007-07-29 20:20:09.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_silitek.h	2010-04-09 16:05:14.358327986 +0200
@@ -12,7 +12,7 @@
 #ifndef _HW_SILITEK_H
 #define _HW_SILITEK_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int silitek_decode(struct ir_remote *remote,
 		   ir_code *prep,ir_code *codep,ir_code *postp,
diff -Naur lirc-20100408/daemons/hw_slinke.h lirc-20100408.patch/daemons/hw_slinke.h
--- lirc-20100408/daemons/hw_slinke.h	2007-07-29 20:20:09.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_slinke.h	2010-04-09 16:05:14.360356926 +0200
@@ -14,7 +14,7 @@
 #ifndef _HW_SLINKE_H
 #define _HW_SLINKE_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int slinke_decode(struct ir_remote *remote,
 		  ir_code          *prep,
diff -Naur lirc-20100408/daemons/hw_tira.h lirc-20100408.patch/daemons/hw_tira.h
--- lirc-20100408/daemons/hw_tira.h	2010-03-20 17:18:30.000000000 +0100
+++ lirc-20100408.patch/daemons/hw_tira.h	2010-04-09 16:05:14.391327119 +0200
@@ -28,7 +28,7 @@
 #ifndef HW_TIRA_H
 #define HW_TIRA_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int     tira_decode                  (struct ir_remote   *remote,
 				      ir_code            *prep,
diff -Naur lirc-20100408/daemons/hw_usbx.h lirc-20100408.patch/daemons/hw_usbx.h
--- lirc-20100408/daemons/hw_usbx.h	2007-07-29 20:20:12.000000000 +0200
+++ lirc-20100408.patch/daemons/hw_usbx.h	2010-04-09 16:05:14.354327656 +0200
@@ -23,7 +23,7 @@
 #ifndef HW_USBX_H
 #define HW_USBX_H
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 int     usbx_decode                  (struct ir_remote   *remote,
 				      ir_code            *prep,
diff -Naur lirc-20100408/daemons/irrecord.c lirc-20100408.patch/daemons/irrecord.c
--- lirc-20100408/daemons/irrecord.c	2010-03-20 17:18:30.000000000 +0100
+++ lirc-20100408.patch/daemons/irrecord.c	2010-04-09 16:05:14.387327137 +0200
@@ -44,7 +44,7 @@
 #include <signal.h>
 #include <syslog.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 #include "hardware.h"
 #include "hw-types.h"
diff -Naur lirc-20100408/daemons/ir_remote.c lirc-20100408.patch/daemons/ir_remote.c
--- lirc-20100408/daemons/ir_remote.c	2010-04-02 12:26:57.000000000 +0200
+++ lirc-20100408.patch/daemons/ir_remote.c	2010-04-09 16:05:14.413327192 +0200
@@ -22,7 +22,7 @@
 
 #include <sys/ioctl.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 #include "lircd.h"
 #include "ir_remote.h"
diff -Naur lirc-20100408/daemons/ir_remote.h lirc-20100408.patch/daemons/ir_remote.h
--- lirc-20100408/daemons/ir_remote.h	2009-05-24 12:46:52.000000000 +0200
+++ lirc-20100408.patch/daemons/ir_remote.h	2010-04-09 16:05:14.398356646 +0200
@@ -21,7 +21,7 @@
 #include <math.h>
 #include <stdlib.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "hardware.h"
 
 #include "ir_remote_types.h"
diff -Naur lirc-20100408/daemons/ir_remote_types.h lirc-20100408.patch/daemons/ir_remote_types.h
--- lirc-20100408/daemons/ir_remote_types.h	2010-04-02 12:26:57.000000000 +0200
+++ lirc-20100408.patch/daemons/ir_remote_types.h	2010-04-09 16:05:14.404327809 +0200
@@ -21,7 +21,7 @@
 #include <math.h>
 #include <stdlib.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 #ifdef LONG_IR_CODE
 typedef unsigned long long ir_code;
diff -Naur lirc-20100408/drivers/lirc_atiusb/lirc_atiusb.c lirc-20100408.patch/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-20100408/drivers/lirc_atiusb/lirc_atiusb.c	2010-03-17 15:16:15.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_atiusb/lirc_atiusb.c	2010-04-09 16:05:14.459351320 +0200
@@ -65,7 +65,7 @@
 #include <linux/wait.h>
 #include <linux/list.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_dev/lirc_dev.c lirc-20100408.patch/drivers/lirc_dev/lirc_dev.c
--- lirc-20100408/drivers/lirc_dev/lirc_dev.c	2010-03-17 15:16:15.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_dev/lirc_dev.c	2010-04-09 16:05:14.516356068 +0200
@@ -70,7 +70,7 @@
 #include <linux/device.h>
 #endif
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "lirc_dev.h"
 
 static int debug;
diff -Naur lirc-20100408/drivers/lirc_dev/lirc_dev.h lirc-20100408.patch/drivers/lirc_dev/lirc_dev.h
--- lirc-20100408/drivers/lirc_dev/lirc_dev.h	2010-01-30 16:01:29.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_dev/lirc_dev.h	2010-04-09 16:05:14.520327275 +0200
@@ -28,7 +28,7 @@
 #include <linux/kfifo.h>
 #endif
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 struct lirc_buffer {
 	wait_queue_head_t wait_poll;
diff -Naur lirc-20100408/drivers/lirc_ene0100/lirc_ene0100.h lirc-20100408.patch/drivers/lirc_ene0100/lirc_ene0100.h
--- lirc-20100408/drivers/lirc_ene0100/lirc_ene0100.h	2009-08-25 14:27:53.000000000 +0200
+++ lirc-20100408.patch/drivers/lirc_ene0100/lirc_ene0100.h	2010-04-09 16:05:14.447327629 +0200
@@ -20,7 +20,7 @@
  */
 
 #include "drivers/kcompat.h"
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 
 /* hardware address */
diff -Naur lirc-20100408/drivers/lirc_igorplugusb/lirc_igorplugusb.c lirc-20100408.patch/drivers/lirc_igorplugusb/lirc_igorplugusb.c
--- lirc-20100408/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2010-03-17 15:16:15.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2010-04-09 16:05:14.525356846 +0200
@@ -63,7 +63,7 @@
 #include <linux/time.h>
 
 #include "drivers/kcompat.h"
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 
 #if !defined(KERNEL_2_5)
diff -Naur lirc-20100408/drivers/lirc_imon/lirc_imon.c lirc-20100408.patch/drivers/lirc_imon/lirc_imon.c
--- lirc-20100408/drivers/lirc_imon/lirc_imon.c	2010-03-17 15:27:19.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_imon/lirc_imon.c	2010-04-09 16:05:14.500327646 +0200
@@ -54,7 +54,7 @@
 #include <linux/timer.h>
 
 #include "drivers/kcompat.h"
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 
 
diff -Naur lirc-20100408/drivers/lirc_it87/lirc_it87.c lirc-20100408.patch/drivers/lirc_it87/lirc_it87.c
--- lirc-20100408/drivers/lirc_it87/lirc_it87.c	2010-03-17 15:16:15.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_it87/lirc_it87.c	2010-04-09 16:05:14.467327188 +0200
@@ -71,7 +71,7 @@
 #include <linux/timer.h>
 #include <linux/pnp.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 #include "drivers/kcompat.h"
 
diff -Naur lirc-20100408/drivers/lirc_ite8709/lirc_ite8709.c lirc-20100408.patch/drivers/lirc_ite8709/lirc_ite8709.c
--- lirc-20100408/drivers/lirc_ite8709/lirc_ite8709.c	2010-01-13 20:56:13.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_ite8709/lirc_ite8709.c	2010-04-09 16:05:14.479327272 +0200
@@ -31,7 +31,7 @@
 #include <linux/io.h>
 #endif
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 #include "drivers/kcompat.h"
 
diff -Naur lirc-20100408/drivers/lirc_mceusb/lirc_mceusb.c lirc-20100408.patch/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-20100408/drivers/lirc_mceusb/lirc_mceusb.c	2010-03-31 06:51:02.000000000 +0200
+++ lirc-20100408.patch/drivers/lirc_mceusb/lirc_mceusb.c	2010-04-09 16:05:14.439352599 +0200
@@ -67,7 +67,7 @@
 #include <linux/wait.h>
 #include <linux/time.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_parallel/lirc_parallel.c lirc-20100408.patch/drivers/lirc_parallel/lirc_parallel.c
--- lirc-20100408/drivers/lirc_parallel/lirc_parallel.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_parallel/lirc_parallel.c	2010-04-09 16:05:14.420351761 +0200
@@ -70,7 +70,7 @@
 #include <linux/poll.h>
 #include <linux/parport.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_sasem/lirc_sasem.c lirc-20100408.patch/drivers/lirc_sasem/lirc_sasem.c
--- lirc-20100408/drivers/lirc_sasem/lirc_sasem.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_sasem/lirc_sasem.c	2010-04-09 16:05:14.534327175 +0200
@@ -58,7 +58,7 @@
 #include <linux/usb.h>
 
 #include "drivers/kcompat.h"
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 
 
diff -Naur lirc-20100408/drivers/lirc_serial/lirc_serial.c lirc-20100408.patch/drivers/lirc_serial/lirc_serial.c
--- lirc-20100408/drivers/lirc_serial/lirc_serial.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_serial/lirc_serial.c	2010-04-09 16:05:14.509327239 +0200
@@ -120,7 +120,7 @@
 #endif
 #endif
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_sir/lirc_sir.c lirc-20100408.patch/drivers/lirc_sir/lirc_sir.c
--- lirc-20100408/drivers/lirc_sir/lirc_sir.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_sir/lirc_sir.c	2010-04-09 16:05:14.429327188 +0200
@@ -95,7 +95,7 @@
 
 #include <linux/timer.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/lirc_dev/lirc_dev.h"
 #include "drivers/kcompat.h"
 
diff -Naur lirc-20100408/drivers/lirc_streamzap/lirc_streamzap.c lirc-20100408.patch/drivers/lirc_streamzap/lirc_streamzap.c
--- lirc-20100408/drivers/lirc_streamzap/lirc_streamzap.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_streamzap/lirc_streamzap.c	2010-04-09 16:05:14.486358057 +0200
@@ -52,7 +52,7 @@
 #endif
 #include <linux/usb.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_ttusbir/lirc_ttusbir.c lirc-20100408.patch/drivers/lirc_ttusbir/lirc_ttusbir.c
--- lirc-20100408/drivers/lirc_ttusbir/lirc_ttusbir.c	2009-03-15 10:34:01.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_ttusbir/lirc_ttusbir.c	2010-04-09 16:05:14.444327695 +0200
@@ -38,7 +38,7 @@
 #include <linux/slab.h>
 #include <linux/usb.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/drivers/lirc_wpc8769l/lirc_wpc8769l.c lirc-20100408.patch/drivers/lirc_wpc8769l/lirc_wpc8769l.c
--- lirc-20100408/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2010-03-17 15:16:16.000000000 +0100
+++ lirc-20100408.patch/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2010-04-09 16:05:14.475356554 +0200
@@ -66,7 +66,7 @@
 #include <linux/platform_device.h>
 #endif
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
diff -Naur lirc-20100408/tools/mode2.c lirc-20100408.patch/tools/mode2.c
--- lirc-20100408/tools/mode2.c	2009-12-28 14:05:30.000000000 +0100
+++ lirc-20100408.patch/tools/mode2.c	2010-04-09 16:05:14.543327535 +0200
@@ -33,7 +33,7 @@
 #include <syslog.h>
 #include <time.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "daemons/ir_remote.h"
 #include "daemons/hardware.h"
 #include "daemons/hw-types.h"
diff -Naur lirc-20100408/tools/smode2.c lirc-20100408.patch/tools/smode2.c
--- lirc-20100408/tools/smode2.c	2003-03-30 14:26:44.000000000 +0200
+++ lirc-20100408.patch/tools/smode2.c	2010-04-09 16:05:14.546353310 +0200
@@ -37,7 +37,7 @@
 #include <vga.h>
 #include <vgagl.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 
 GraphicsContext *screen;
 GraphicsContext *physicalscreen;
diff -Naur lirc-20100408/tools/xmode2.c lirc-20100408.patch/tools/xmode2.c
--- lirc-20100408/tools/xmode2.c	2009-12-27 20:04:10.000000000 +0100
+++ lirc-20100408.patch/tools/xmode2.c	2010-04-09 16:05:14.539352347 +0200
@@ -60,7 +60,7 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
-#include "drivers/lirc.h"
+#include <linux/lirc.h>
 #include "daemons/ir_remote.h"
 #include "daemons/hardware.h"
 #include "daemons/hw-types.h"
