diff -Naur xbmc-13.alpha-6edea14/xbmc/cores/dvdplayer/DVDPlayer.cpp xbmc-13.alpha-6edea14.patch/xbmc/cores/dvdplayer/DVDPlayer.cpp
--- xbmc-13.alpha-6edea14/xbmc/cores/dvdplayer/DVDPlayer.cpp	2013-08-02 16:28:51.000000000 +0200
+++ xbmc-13.alpha-6edea14.patch/xbmc/cores/dvdplayer/DVDPlayer.cpp	2013-08-02 17:19:34.162512140 +0200
@@ -479,7 +479,7 @@
 #endif
 
     Create();
-    if(!m_ready.WaitMSec(100))
+    if(!m_ready.WaitMSec(g_advancedSettings.m_videoBusyDialogDelay_ms))
     {
       CGUIDialogBusy* dialog = (CGUIDialogBusy*)g_windowManager.GetWindow(WINDOW_DIALOG_BUSY);
       if(dialog)
diff -Naur xbmc-13.alpha-6edea14/xbmc/cores/omxplayer/OMXPlayer.cpp xbmc-13.alpha-6edea14.patch/xbmc/cores/omxplayer/OMXPlayer.cpp
--- xbmc-13.alpha-6edea14/xbmc/cores/omxplayer/OMXPlayer.cpp	2013-08-02 16:28:51.000000000 +0200
+++ xbmc-13.alpha-6edea14.patch/xbmc/cores/omxplayer/OMXPlayer.cpp	2013-08-02 17:19:34.164512143 +0200
@@ -522,7 +522,7 @@
 #endif
 
     Create();
-    if(!m_ready.WaitMSec(100))
+    if(!m_ready.WaitMSec(g_advancedSettings.m_videoBusyDialogDelay_ms))
     {
       CGUIDialogBusy* dialog = (CGUIDialogBusy*)g_windowManager.GetWindow(WINDOW_DIALOG_BUSY);
       if(dialog)
diff -Naur xbmc-13.alpha-6edea14/xbmc/settings/AdvancedSettings.cpp xbmc-13.alpha-6edea14.patch/xbmc/settings/AdvancedSettings.cpp
--- xbmc-13.alpha-6edea14/xbmc/settings/AdvancedSettings.cpp	2013-08-02 16:28:51.000000000 +0200
+++ xbmc-13.alpha-6edea14.patch/xbmc/settings/AdvancedSettings.cpp	2013-08-02 17:21:43.772739525 +0200
@@ -184,6 +184,7 @@
 
   m_videoDefaultLatency = 0.0;
   m_videoDisableHi10pMultithreading = false;
+  m_videoBusyDialogDelay_ms = 100;
 
   m_musicUseTimeSeeking = true;
   m_musicTimeSeekForward = 10;
@@ -700,6 +701,10 @@
     //0 = disable fps detect, 1 = only detect on timestamps with uniform spacing, 2 detect on all timestamps
     XMLUtils::GetInt(pElement, "fpsdetect", m_videoFpsDetect, 0, 2);
 
+    // controls the delay, in milliseconds, until
+    // the busy dialog is shown when starting video playback.
+    XMLUtils::GetInt(pElement, "busydialogdelayms", m_videoBusyDialogDelay_ms, 0, 1000);
+
     // Store global display latency settings
     TiXmlElement* pVideoLatency = pElement->FirstChildElement("latency");
     if (pVideoLatency)
diff -Naur xbmc-13.alpha-6edea14/xbmc/settings/AdvancedSettings.h xbmc-13.alpha-6edea14.patch/xbmc/settings/AdvancedSettings.h
--- xbmc-13.alpha-6edea14/xbmc/settings/AdvancedSettings.h	2013-08-02 16:28:51.000000000 +0200
+++ xbmc-13.alpha-6edea14.patch/xbmc/settings/AdvancedSettings.h	2013-08-02 17:21:29.790714897 +0200
@@ -190,6 +190,7 @@
     bool m_DXVANoDeintProcForProgressive;
     int  m_videoFpsDetect;
     bool m_videoDisableHi10pMultithreading;
+    int  m_videoBusyDialogDelay_ms;
     StagefrightConfig m_stagefrightConfig;
 
     CStdString m_videoDefaultPlayer;
