From abcd8697b46924f8a18d733fc9d2bf884e901a46 Mon Sep 17 00:00:00 2001
From: leszekb <leszek@control-by.net>
Date: Mon, 14 May 2018 22:42:34 +0200
Subject: [PATCH] Update player_event_handler.rs

---
 src/player_event_handler.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/player_event_handler.rs b/src/player_event_handler.rs
index b6a653d..6064bc0 100644
--- a/src/player_event_handler.rs
+++ b/src/player_event_handler.rs
@@ -8,7 +8,7 @@ fn run_program(program: &str, env_vars: HashMap<&str, String>) {
     Command::new(&v.remove(0))
         .args(&v)
         .envs(env_vars.iter())
-        .spawn()
+        .status()
         .expect("program failed to start");
 }
 

