From e9bb269936ea26b1c0c698b8d05aaf68e2e79bcc Mon Sep 17 00:00:00 2001
From: awiouy <awiouy@gmail.com>
Date: Tue, 12 Sep 2017 09:41:14 +0200
Subject: [PATCH] use librespot_sink pulseadio sink

---
 src/audio_backend/pulseaudio.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/audio_backend/pulseaudio.rs b/src/audio_backend/pulseaudio.rs
index 3b9a09b..ce78062 100644
--- a/src/audio_backend/pulseaudio.rs
+++ b/src/audio_backend/pulseaudio.rs
@@ -23,12 +23,13 @@ impl Open for PulseAudioSink {
         
         let name = CString::new("librespot").unwrap();
         let description = CString::new("A spoty client library").unwrap();
+        let sink = CString::new("librespot_sink").unwrap();
 
         let s = unsafe {
             pa_simple_new(null(),               // Use the default server.
                           name.as_ptr(),        // Our application's name.
                           PA_STREAM_PLAYBACK,
-                          null(),               // Use the default device.
+                          sink.as_ptr(),        // Our sink.
                           description.as_ptr(), // Description of our stream.
                           &ss,                  // Our sample format.
                           null(),               // Use default channel map
