commit 6a9b509f4f85427f6392c08fc642afb9fc7bb8f2
Author: awiouy <awiouy@gmail.com>
Date:   Tue Dec 3 23:21:35 2019 +0100

    Notify Kodi

diff --git a/core/src/spotify_id.rs b/core/src/spotify_id.rs
index e6f0cdd..03f8912 100644
--- a/core/src/spotify_id.rs
+++ b/core/src/spotify_id.rs
@@ -8,6 +8,12 @@ pub enum SpotifyAudioType {
     NonPlayable,
 }
 
+impl fmt::Display for SpotifyAudioType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub struct SpotifyId {
     pub id: u128,
diff --git a/playback/src/config.rs b/playback/src/config.rs
index 0f71110..931167d 100644
--- a/playback/src/config.rs
+++ b/playback/src/config.rs
@@ -30,6 +30,7 @@ pub struct PlayerConfig {
     pub bitrate: Bitrate,
     pub normalisation: bool,
     pub normalisation_pregain: f32,
+    pub notify_kodi: bool,
 }
 
 impl Default for PlayerConfig {
@@ -38,6 +39,7 @@ impl Default for PlayerConfig {
             bitrate: Bitrate::default(),
             normalisation: false,
             normalisation_pregain: 0.0,
+            notify_kodi: false,
         }
     }
 }
diff --git a/playback/src/player.rs b/playback/src/player.rs
index a54a577..4e006ab 100644
--- a/playback/src/player.rs
+++ b/playback/src/player.rs
@@ -5,7 +5,8 @@ use futures::{future, Future};
 use std;
 use std::borrow::Cow;
 use std::cmp::max;
-use std::io::{Read, Result, Seek, SeekFrom};
+use std::fs::OpenOptions;
+use std::io::{Read, Result, Seek, SeekFrom, Write};
 use std::mem;
 use std::sync::mpsc::{RecvError, RecvTimeoutError, TryRecvError};
 use std::thread;
@@ -427,6 +428,14 @@ impl PlayerInternal {
         }
     }
 
+    fn notify_kodi(&mut self, id: &str, track_id: &SpotifyId) {
+        // println!("fifo = {} {}", id, track_id.to_base62());
+        if self.config.notify_kodi {
+            let mut file = OpenOptions::new().write(true).open("/tmp/librespot").unwrap();
+            writeln!(&mut file, "{}\n{} {}", id, track_id.audio_type.to_string(), track_id.to_base62()).unwrap();
+        }
+    }
+
     fn handle_command(&mut self, cmd: PlayerCommand) {
         debug!("command={:?}", cmd);
         match cmd {
@@ -451,11 +460,17 @@ impl PlayerInternal {
                                 | PlayerState::EndOfTrack {
                                     track_id: old_track_id,
                                     ..
-                                } => self.send_event(PlayerEvent::Changed {
-                                    old_track_id: old_track_id,
-                                    new_track_id: track_id,
-                                }),
-                                _ => self.send_event(PlayerEvent::Started { track_id }),
+                                } => {
+                                    self.send_event(PlayerEvent::Changed {
+                                        old_track_id: old_track_id,
+                                        new_track_id: track_id,
+                                    });
+                                    self.notify_kodi("1", &track_id)
+                                }
+                                _ => {
+                                    self.send_event(PlayerEvent::Started { track_id });
+                                    self.notify_kodi("2", &track_id)
+                                }
                             }
 
                             self.start_sink();
@@ -485,13 +500,17 @@ impl PlayerInternal {
                                 | PlayerState::EndOfTrack {
                                     track_id: old_track_id,
                                     ..
-                                } => self.send_event(PlayerEvent::Changed {
-                                    old_track_id: old_track_id,
-                                    new_track_id: track_id,
-                                }),
+                                } => {
+                                    self.send_event(PlayerEvent::Changed {
+                                        old_track_id: old_track_id,
+                                        new_track_id: track_id,
+                                    });
+                                    self.notify_kodi("3", &track_id)
+                                }
                                 _ => (),
                             }
                             self.send_event(PlayerEvent::Stopped { track_id });
+                            self.notify_kodi("4", &track_id)
                         }
                     }
 
@@ -547,6 +566,7 @@ impl PlayerInternal {
 
                     self.send_event(PlayerEvent::Started { track_id });
                     self.start_sink();
+                    self.notify_kodi("5", &track_id)
                 } else {
                     warn!("Player::play called from invalid state");
                 }
@@ -558,6 +578,7 @@ impl PlayerInternal {
 
                     self.stop_sink_if_running();
                     self.send_event(PlayerEvent::Stopped { track_id });
+                    self.notify_kodi("6", &track_id)
                 } else {
                     warn!("Player::pause called from invalid state");
                 }
@@ -570,6 +591,7 @@ impl PlayerInternal {
                     self.stop_sink_if_running();
                     self.send_event(PlayerEvent::Stopped { track_id });
                     self.state = PlayerState::Stopped;
+                    self.notify_kodi("7", &track_id)
                 }
                 PlayerState::Stopped => {
                     warn!("Player::stop called from invalid state");
diff --git a/src/main.rs b/src/main.rs
index e3718fb..a480480 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -184,6 +184,11 @@ fn setup(args: &[String]) -> Setup {
             "Pregain (dB) applied by volume normalisation",
             "PREGAIN",
         )
+        .optflag(
+            "",
+            "notify-kodi",
+            "Notify Kodi",
+        )
         .optflag(
             "",
             "linear-volume",
@@ -276,6 +281,8 @@ fn setup(args: &[String]) -> Setup {
         )
     };
 
+    let notify_kodi = matches.opt_present("notify-kodi");
+
     let session_config = {
         let device_id = device_id(&name);
 
@@ -319,6 +326,7 @@ fn setup(args: &[String]) -> Setup {
                 .opt_str("normalisation-pregain")
                 .map(|pregain| pregain.parse::<f32>().expect("Invalid pregain float value"))
                 .unwrap_or(PlayerConfig::default().normalisation_pregain),
+            notify_kodi: notify_kodi,
         }
     };
 
