commit 0f3141a0874566ce867139aeadbefdb076706baf
Author: awiouy <awiouy@gmail.com>
Date:   Tue Dec 3 23:22:55 2019 +0100

    Use own pulseaudio sink

diff --git a/playback/src/audio_backend/pulseaudio.rs b/playback/src/audio_backend/pulseaudio.rs
index 88f6280..702be66 100644
--- a/playback/src/audio_backend/pulseaudio.rs
+++ b/playback/src/audio_backend/pulseaudio.rs
@@ -76,6 +76,7 @@ impl Open for PulseAudioSink {
 
 impl Sink for PulseAudioSink {
     fn start(&mut self) -> io::Result<()> {
+        let sink = CString::new("librespot_sink").unwrap();
         if self.s == null_mut() {
             self.s = call_pulseaudio(
                 |err| unsafe {
@@ -83,7 +84,7 @@ impl Sink for PulseAudioSink {
                         null(),             // Use the default server.
                         self.name.as_ptr(), // Our application's name.
                         PA_STREAM_PLAYBACK,
-                        null(),             // Use the default device.
+                        sink.as_ptr(),      // Our sink.
                         self.desc.as_ptr(), // desc of our stream.
                         &self.ss,           // Our sample format.
                         null(),             // Use default channel map
