From 697926d2058e8650a9dd559d90a8a130950c417a Mon Sep 17 00:00:00 2001
From: Sasha Hilton <sashahilton00@users.noreply.github.com>
Date: Tue, 27 Jun 2017 12:45:34 +0100
Subject: [PATCH] Update system_information_string

---
 src/connection/mod.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/connection/mod.rs b/src/connection/mod.rs
index af134fc..cfb5000 100644
--- a/src/connection/mod.rs
+++ b/src/connection/mod.rs
@@ -41,7 +41,7 @@ pub fn authenticate(transport: Transport, credentials: Credentials, device_id: S
         system_info => {
             cpu_family: CpuFamily::CPU_UNKNOWN,
             os: Os::OS_UNKNOWN,
-            system_information_string: "This is not the client you are looking for".to_owned(),
+            system_information_string: "librespot".to_string() + "_" + &(version::short_sha()) + "_" + &(version::short_now()),
             device_id: device_id,
         },
         version_string: version::version_string(),
