commit 88ba7c3ad879e784ce55714c2c834911b2573ce1
Author: awiouy <awiouy@gmail.com>
Date:   Tue Dec 3 23:22:55 2019 +0100

    Use own pulseaudio sink

diff --git a/playback/src/audio_backend/pulseaudio.rs b/playback/src/audio_backend/pulseaudio.rs
index e844b0d..0320f65 100644
--- a/playback/src/audio_backend/pulseaudio.rs
+++ b/playback/src/audio_backend/pulseaudio.rs
@@ -80,6 +80,7 @@ impl Open for PulseAudioSink {
 
 impl Sink for PulseAudioSink {
     fn start(&mut self) -> io::Result<()> {
+        let sink = CString::new("librespot_sink").unwrap();
         if self.s == null_mut() {
             self.s = call_pulseaudio(
                 |err| unsafe {
@@ -87,7 +88,7 @@ impl Sink for PulseAudioSink {
                         null(),             // Use the default server.
                         self.name.as_ptr(), // Our application's name.
                         PA_STREAM_PLAYBACK,
-                        null(),             // Use the default device.
+                        sink.as_ptr(),      // Our sink.
                         self.desc.as_ptr(), // desc of our stream.
                         &self.ss,           // Our sample format.
                         null(),             // Use default channel map
