--- a/Cargo.toml	2021-02-22 00:37:28.000000000 +0000
+++ b/Cargo.toml	2021-03-25 11:30:44.000000000 +0000
@@ -90,3 +90,6 @@
     ["target/release/librespot", "usr/bin/", "755"],
     ["contrib/librespot.service", "lib/systemd/system/", "644"]
 ]
+
+[patch.crates-io]
+vorbis = { git = 'https://github.com/roderickvd/vorbis-rs', branch = 'fix-ub-panic' }
--- a/audio/Cargo.toml	2021-02-22 00:37:28.000000000 +0000
+++ b/audio/Cargo.toml	2021-03-25 11:46:31.000000000 +0000
@@ -23,7 +23,7 @@
 aes-ctr = "0.3"
 
 librespot-tremor = { version = "0.2.0", optional = true }
-vorbis = { version ="0.0.14", optional = true }
+vorbis = { version ="0.1.0", optional = true }
 
 [features]
 with-tremor = ["librespot-tremor"]
