diff -Naur squashfs4.2-old/squashfs-tools/compressor.c squashfs4.2-new/squashfs-tools/compressor.c
--- squashfs4.2-old/squashfs-tools/compressor.c	2011-02-11 07:49:24.000000000 -0800
+++ squashfs4.2-new/squashfs-tools/compressor.c	2011-03-04 10:27:47.000000000 -0800
@@ -27,7 +27,7 @@
 
 #ifndef GZIP_SUPPORT
 static struct compressor gzip_comp_ops =  {
-	NULL, NULL, NULL, NULL, NULL, NULL, ZLIB_COMPRESSION, "gzip", 0
+	NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, ZLIB_COMPRESSION, "gzip", 0
 };
 #else
 extern struct compressor gzip_comp_ops;
