diff -Naur a/os_dep/linux/os_intfs.c b/os_dep/linux/os_intfs.c
--- a/os_dep/linux/os_intfs.c	2016-04-19 11:17:44.465411644 -0700
+++ b/os_dep/linux/os_intfs.c	2016-04-19 11:23:24.742014169 -0700
@@ -958,7 +958,13 @@
 	return dscp >> 5;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
+static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb,
+			    void *accel_priv,
+			    select_queue_fallback_t fallback)
+#else
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb)
+#endif
 {
 	_adapter	*padapter = rtw_netdev_priv(dev);
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
