diff -Naur RTL8192CU-v4.0.2_9000.20130911/os_dep/linux/rtw_android.c RTL8192CU-v4.0.2_9000.20130911.patch/os_dep/linux/rtw_android.c
--- RTL8192CU-v4.0.2_9000.20130911/os_dep/linux/rtw_android.c	2013-09-11 05:56:55.000000000 +0200
+++ RTL8192CU-v4.0.2_9000.20130911.patch/os_dep/linux/rtw_android.c	2015-04-04 20:20:23.779119031 +0200
@@ -230,7 +230,11 @@
 {
 	int cmd_num;
 	for(cmd_num=0 ; cmd_num<ANDROID_WIFI_CMD_MAX; cmd_num++)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0))
+		if(!strncasecmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
+#else
 		if(0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
+#endif
 			break;
 		
 	return cmd_num;
