diff -Naur xine-lib-1.2-r10926.orig/src/Makefile.am xine-lib-1.2-r10926/src/Makefile.am
--- xine-lib-1.2-r10926.orig/src/Makefile.am	2009-04-25 11:58:16.000000000 +0200
+++ xine-lib-1.2-r10926/src/Makefile.am	2009-04-25 11:58:51.000000000 +0200
@@ -15,5 +15,4 @@
 	libw32dll \
 	libreal \
 	post \
-	combined \
-	vdr
+	combined
diff -Naur xine-lib-1.2-r10926.orig/src/Makefile.in xine-lib-1.2-r10926/src/Makefile.in
--- xine-lib-1.2-r10926.orig/src/Makefile.in	2009-04-25 11:58:16.000000000 +0200
+++ xine-lib-1.2-r10926/src/Makefile.in	2009-04-25 11:58:44.000000000 +0200
@@ -397,8 +397,7 @@
 	libw32dll \
 	libreal \
 	post \
-	combined \
-	vdr
+	combined
 
 all: all-recursive
 
