diff -Naur xine-lib-1.2-r10947/src/xine-engine/metronom.c xine-lib-1.2-r10947a/src/xine-engine/metronom.c
--- xine-lib-1.2-r10947/src/xine-engine/metronom.c	2009-04-26 09:07:51.000000000 +0200
+++ xine-lib-1.2-r10947a/src/xine-engine/metronom.c	2009-05-31 14:00:22.000000000 +0200
@@ -304,8 +304,6 @@
 
   /* video_vpts and audio_vpts adjustements */
   cur_time = this->xine->clock->get_current_time(this->xine->clock);
-  xprintf(this->xine, XINE_VERBOSITY_DEBUG,
-          "current time : %" PRId64 "\n", cur_time);
 
   switch (type) {
     case DISC_STREAMSTART:
@@ -322,12 +320,33 @@
 
     case DISC_ABSOLUTE:
     case DISC_RELATIVE:
-      if (this->video_vpts < this->audio_vpts) {
-        this->video_vpts = this->audio_vpts;
-        xprintf(this->xine, XINE_VERBOSITY_DEBUG, "video vpts adjusted to audio vpts %" PRId64 "\n", this->video_vpts);
+      if (this->video_vpts < cur_time) {
+        /* still frame */
+        if (this->audio_vpts > cur_time) {
+          /* still frame with audio */
+          this->video_vpts = this->audio_vpts;
+          xprintf(this->xine, XINE_VERBOSITY_DEBUG, "video vpts adjusted to audio vpts %" PRId64 "\n", this->video_vpts);
+        } else {
+          /* still frame, no audio */
+          this->video_vpts = this->prebuffer + cur_time;
+          this->audio_vpts = this->video_vpts;
+          this->audio_vpts_rmndr = 0;
+          this->force_video_jump = 1;
+          this->force_audio_jump = 1;
+          this->video_drift = 0;
+          xprintf(this->xine, XINE_VERBOSITY_DEBUG, "vpts adjusted with prebuffer to %" PRId64 "\n",
+	    this->video_vpts);
+        }
       } else {
-        this->audio_vpts = this->video_vpts;
-        xprintf(this->xine, XINE_VERBOSITY_DEBUG, "audio vpts adjusted to video vpts %" PRId64 "\n", this->video_vpts);
+        /* video */
+        if (this->audio_vpts < cur_time) {
+          /* video, no sound */
+          this->audio_vpts = this->video_vpts;
+          this->audio_vpts_rmndr = 0;
+          xprintf(this->xine, XINE_VERBOSITY_DEBUG, "audio vpts adjusted to video vpts %" PRId64 "\n", this->video_vpts);
+        } else {
+          /* video + audio */
+        }
       }
       break;
   }
