diff --git a/src/video_out/video_out_vdpau.c b/src/video_out/video_out_vdpau.c
--- a/src/video_out/video_out_vdpau.c
+++ b/src/video_out/video_out_vdpau.c
@@ -1386,7 +1394,7 @@ static void vdpau_display_frame (vo_driv
 
   XLockDisplay( this->display );
 
-  if ( frame->format==XINE_IMGFMT_VDPAU && this->deinterlace && non_progressive && stream_speed && frame_duration>2500 ) {
+  if ( frame->format==XINE_IMGFMT_VDPAU && this->deinterlace && non_progressive /* && stream_speed */ && frame_duration>2500 ) {
     VdpTime current_time = 0;
     VdpVideoSurface past[2];
     VdpVideoSurface future[1];
