diff -Naur expat-2.0.1-old/configure.in expat-2.0.1-new/configure.in
--- expat-2.0.1-old/configure.in	2007-05-09 06:26:48.000000000 -0700
+++ expat-2.0.1-new/configure.in	2010-08-17 13:12:17.000000000 -0700
@@ -31,6 +31,7 @@
 
 AC_CONFIG_SRCDIR(Makefile.in)
 AC_CONFIG_AUX_DIR(conftools)
+AC_CONFIG_MACRO_DIR([conftools])
 
 
 dnl
@@ -50,9 +51,6 @@
 
 AC_CONFIG_HEADER(expat_config.h)
 
-sinclude(conftools/libtool.m4)
-sinclude(conftools/ac_c_bigendian_cross.m4)
-
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
