diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-06-21 05:58:38.000000000 -0700
+++ b/CMakeLists.txt	2017-10-10 09:39:36.713079492 -0700
@@ -10,6 +10,7 @@
 set(PACKAGE_STRING "${PACKAGE_NAME} ${PACKAGE_VERSION}")
 set(PACKAGE_TARNAME "${PACKAGE_NAME}")
 
+option(BUILD_docs "build the docs for expat library" ON)
 option(BUILD_tools "build the xmlwf tool for expat library" ON)
 option(BUILD_examples "build the examples for expat library" ON)
 option(BUILD_tests "build the tests for expat library" ON)
@@ -92,8 +93,9 @@
 install(FILES lib/expat.h lib/expat_external.h DESTINATION include)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/expat.pc DESTINATION lib/pkgconfig)
 
-
-add_custom_command(TARGET expat PRE_BUILD COMMAND $(MAKE) -C doc xmlwf.1)
+if(BUILD_docs)
+    add_custom_command(TARGET expat PRE_BUILD COMMAND $(MAKE) -C doc xmlwf.1)
+endif()
 
 if(BUILD_tools AND NOT WINCE)
     set(xmlwf_SRCS
