diff -Naur xbmc-pvr-addons-259d150/addons/pvr.vdr.vnsi/addon/addon.xml.in xbmc-pvr-addons-259d150.patch/addons/pvr.vdr.vnsi/addon/addon.xml.in
--- xbmc-pvr-addons-259d150/addons/pvr.vdr.vnsi/addon/addon.xml.in	2014-10-06 03:49:06.000000000 +0200
+++ xbmc-pvr-addons-259d150.patch/addons/pvr.vdr.vnsi/addon/addon.xml.in	2014-10-14 23:33:09.834238654 +0200
@@ -8,7 +8,7 @@
     <c-pluff version="0.1"/>
     <import addon="xbmc.pvr" version="1.9.2"/>
     <import addon="xbmc.codec" version="1.0.0"/>
-    <import addon="xbmc.gui" version="5.0.1"/>
+    <import addon="xbmc.gui" version="5.3.0"/>
   </requires>
   <extension
     point="xbmc.pvrclient"
diff -Naur xbmc-pvr-addons-259d150/xbmc/libXBMC_gui.h xbmc-pvr-addons-259d150.patch/xbmc/libXBMC_gui.h
--- xbmc-pvr-addons-259d150/xbmc/libXBMC_gui.h	2014-10-06 03:49:06.000000000 +0200
+++ xbmc-pvr-addons-259d150.patch/xbmc/libXBMC_gui.h	2014-10-14 23:32:15.914304646 +0200
@@ -36,10 +36,10 @@
 #endif
 
 /* current ADDONGUI API version */
-#define XBMC_GUI_API_VERSION "5.0.1"
+#define XBMC_GUI_API_VERSION "5.3.0"
 
 /* min. ADDONGUI API version */
-#define XBMC_GUI_MIN_API_VERSION "5.0.1"
+#define XBMC_GUI_MIN_API_VERSION "5.3.0"
 
 #define ADDON_ACTION_PREVIOUS_MENU          10
 #define ADDON_ACTION_CLOSE_DIALOG           51
