From 615c1fe2ff904cdb9ebcd8edc0b2c42c3cd6218c Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Mon, 9 Sep 2013 12:13:15 +0200
Subject: [PATCH] ffmpeg backport: h264: do not discard NAL_SEI when skipping
 frames

---
 libavcodec/h264.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index 4f09c67..62ac8fb 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -4529,8 +4529,7 @@ static int decode_nal_units(H264Context *h, const uint8_t *buf, int buf_size,
                     first_slice = hx->nal_unit_type;
                 }
 
-            // FIXME do not discard SEI id
-            if (avctx->skip_frame >= AVDISCARD_NONREF && h->nal_ref_idc == 0)
+            if (avctx->skip_frame >= AVDISCARD_NONREF && h->nal_ref_idc == 0 && h->nal_unit_type != NAL_SEI)
                 continue;
 
 again:
-- 
1.8.4

