diff -Naur plymouth-lite-0.6.0/ply-image.c plymouth-lite-0.6.0.patch/ply-image.c
--- plymouth-lite-0.6.0/ply-image.c	2009-02-19 12:16:36.000000000 +0100
+++ plymouth-lite-0.6.0.patch/ply-image.c	2010-02-08 01:30:05.336856616 +0100
@@ -220,7 +220,7 @@
     png_set_palette_to_rgb (png);
 
   if ((color_type == PNG_COLOR_TYPE_GRAY) && (bits_per_pixel < 8))
-    png_set_gray_1_2_4_to_8 (png);
+    png_set_expand_gray_1_2_4_to_8 (png);
 
   if (png_get_valid (png, info, PNG_INFO_tRNS))
     png_set_tRNS_to_alpha (png);
