diff --git a/svn-current/trunk/src/burner/unzip.c b/svn-current/trunk/src/burner/unzip.c
index 879235f..0740960 100644
--- a/svn-current/trunk/src/burner/unzip.c
+++ b/svn-current/trunk/src/burner/unzip.c
@@ -1179,7 +1179,7 @@ extern int ZEXPORT unzOpenCurrentFile3 (file, method, level, raw, password)
     if (password != NULL)
     {
         int i;
-        s->pcrc_32_tab = get_crc_table();
+        s->pcrc_32_tab = (const long unsigned int *)get_crc_table();
         init_keys(password,s->keys,s->pcrc_32_tab);
         if (ZSEEK(s->z_filefunc, s->filestream,
                   s->pfile_in_zip_read->pos_in_zipfile +
