diff --git a/board/amlogic/configs/libretech_cc.h b/board/amlogic/configs/libretech_cc.h
index b3ccfae..6052a87 100644
--- a/board/amlogic/configs/libretech_cc.h
+++ b/board/amlogic/configs/libretech_cc.h
@@ -45,10 +45,10 @@
 #define CONFIG_CEC_OSD_NAME		"LibreTech-CC"
 #define CONFIG_CEC_WAKEUP
 
-#define CONFIG_INSTABOOT
+// #define CONFIG_INSTABOOT
 #define CONFIG_CMD_CFGLOAD
 /* configs for dtb in boot.img */
-//#define DTB_BIND_KERNEL
+#define DTB_BIND_KERNEL
 
 /* SMP Definitinos */
 #define CPU_RELEASE_ADDR		secondary_boot_func
@@ -209,7 +209,7 @@
             "if imgread kernel ${recovery_part} ${loadaddr} ${recovery_offset}; then wipeisb; bootm ${loadaddr}; fi;"\
             "\0"\
         "init_display="\
-            "osd open;osd clear;imgread pic logo bootup $loadaddr;bmp display $bootup_offset;bmp scale"\
+            "osd open;osd clear;"\
             "\0"\
         "cmdline_keys="\
             "if keyman init 0x1234; then "\
@@ -236,15 +236,8 @@
             "\0"\
 
 #define CONFIG_PREBOOT  \
-            "run bcb_cmd; "\
-            "run factory_reset_poweroff_protect;"\
-            "run upgrade_check;"\
-            "run init_display;"\
-            "run storeargs;"\
-            "run upgrade_key;" \
-            "run switch_bootmode;" \
-            "cfgload;"
-#define CONFIG_BOOTCOMMAND "run storeboot"
+            "run init_display;"
+#define CONFIG_BOOTCOMMAND "cfgload"
 
 //#define CONFIG_ENV_IS_NOWHERE  1
 #define CONFIG_ENV_SIZE   (64*1024)
@@ -298,7 +291,7 @@
 /* support for mtd */
 //#define CONFIG_AML_MTD 1
 /* support for nftl */
-#define CONFIG_AML_NAND	1
+//#define CONFIG_AML_NAND	1
 
 #if defined(CONFIG_AML_NAND) && defined(CONFIG_AML_MTD)
 #error CONFIG_AML_NAND/CONFIG_AML_MTD can not support at the sametime;
@@ -390,7 +383,7 @@
 #endif //#if defined(CONFIG_CMD_USB)
 
 //UBOOT fastboot config
-#define CONFIG_CMD_FASTBOOT 1
+// #define CONFIG_CMD_FASTBOOT 1
 #define CONFIG_FASTBOOT_FLASH_MMC_DEV 1
 #define CONFIG_FASTBOOT_FLASH 1
 #define CONFIG_USB_GADGET 1
