From 7c305a6fc9cd41a5eddf28cfbfc33bb98459c29e Mon Sep 17 00:00:00 2001
From: xbmc <fernetmenta@online.de>
Date: Thu, 2 May 2013 19:29:05 +0200
Subject: [PATCH] ffmpeg: fix vdpau vc1 field interlace

---
 libavcodec/vdpau_vc1.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libavcodec/vdpau_vc1.c b/libavcodec/vdpau_vc1.c
index f5da9bb..6831599 100644
--- a/libavcodec/vdpau_vc1.c
+++ b/libavcodec/vdpau_vc1.c
@@ -59,7 +59,12 @@ static int vdpau_vc1_start_frame(AVCodecContext *avctx,
     else
         info->picture_type  = s->pict_type - 1 + s->pict_type / 3;
 
-    info->frame_coding_mode = v->fcm;
+    if (v->fcm == PROGRESSIVE)
+        info->frame_coding_mode = 0;
+    else if (v->fcm == ILACE_FRAME)
+        info->frame_coding_mode = 2;
+    else if (v->fcm == ILACE_FIELD)
+        info->frame_coding_mode = 3;
     info->postprocflag      = v->postprocflag;
     info->pulldown          = v->broadcast;
     info->interlace         = v->interlace;
-- 
1.8.1.6

