From 53b61e3b71c358df2ca7605f8d8478878862ca07 Mon Sep 17 00:00:00 2001
From: fritsch <peter.fruehberger@gmail.com>
Date: Sat, 13 Apr 2013 11:30:39 +0200
Subject: [PATCH] XVBA: revisit draw functions

---
 libavcodec/xvba_h264.c | 2 +-
 libavcodec/xvba_vc1.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavcodec/xvba_h264.c b/libavcodec/xvba_h264.c
index ae45f3a..309d928 100644
--- a/libavcodec/xvba_h264.c
+++ b/libavcodec/xvba_h264.c
@@ -160,7 +160,7 @@ static int end_frame(AVCodecContext *avctx)
       h->got_first_iframe = 1;
   }
 
-  ff_draw_horiz_band(h->avctx, &h->dsp, h->cur_pic_ptr, NULL, 0, h->avctx->height, h->picture_structure, h->first_field, 0, 0, 0, 0);
+  ff_h264_draw_horiz_band(h, 0, h->avctx->height);
 
   return 0;
 }
diff --git a/libavcodec/xvba_vc1.c b/libavcodec/xvba_vc1.c
index bf3d9c2..ff35a28 100644
--- a/libavcodec/xvba_vc1.c
+++ b/libavcodec/xvba_vc1.c
@@ -142,7 +142,7 @@ static int end_frame(AVCodecContext *avctx)
         break;
     }
 
-    ff_draw_horiz_band(s->avctx, &s->dsp, s->current_picture_ptr, s->last_picture_ptr, 0, s->avctx->height, s->picture_structure, s->first_field, (s->unrestricted_mv && !s->intra_only), s->low_delay, s->v_edge_pos, s->h_edge_pos);
+    ff_mpeg_draw_horiz_band(s, 0, s->avctx->height);
 
     return 0;
 }
-- 
1.8.1.5

