--- zsnes-1.5-RC1-svn/src/configure.orig	2006-12-07 10:16:09.000000000 +0100
+++ zsnes-1.5-RC1-svn/src/configure	2006-12-07 10:39:59.000000000 +0100
@@ -1742,9 +1742,9 @@
 
 VERSION=pre1.43
 
-CFLAGS="$CFLAGS -pipe -I. -I/usr/local/include -I/usr/include"
+CFLAGS="$CFLAGS -pipe -I."
 
-LDFLAGS="$LDFLAGS -L/usr/local/lib -L/usr/lib"
+LDFLAGS="$LDFLAGS"
 
 ac_aux_dir=
 for ac_dir in "$srcdir" "$srcdir/.." "$srcdir/../.."; do
@@ -4026,17 +4026,7 @@
   { echo "$as_me:$LINENO: result: $with_zlib" >&5
 echo "${ECHO_T}$with_zlib" >&6; }
 fi
-if test x$with_zlib = xyes; then
-  ZLIB_VERSION=$(<conf.zlibtest)
-  :
-else
-  ZLIB_CFLAGS=""
-  ZLIB_LIBS=""
-  ZLIB_VERSION=""
-  { { echo "$as_me:$LINENO: error: zlib >= 1.2.3 is required" >&5
-echo "$as_me: error: zlib >= 1.2.3 is required" >&2;}
-   { (exit 1); exit 1; }; }
-fi
+ZLIB_VERSION=1.2.3
 LIBS="$tempLIBS"
 CFLAGS="$tempCFLAGS"
 rm conf.zlibtest
