diff -Naur zsnes_1_51b/src/Makefile.in zsnes_1_51ba/src/Makefile.in
--- zsnes_1_51b/src/Makefile.in	2007-01-24 21:54:12.000000000 +0100
+++ zsnes_1_51ba/src/Makefile.in	2009-01-06 18:04:23.000000000 +0100
@@ -104,32 +104,36 @@
 	rm -f version.o
 
 $(PSR): parsegen.cpp
-	@CXX@ @CXXFLAGS@ -o $@ $< -lz
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $< -lz
 
 TOOLSEXE=$(TOOL_D)/archopt $(TOOL_D)/cutrtype $(TOOL_D)/extraext\
 	$(TOOL_D)/macroll $(TOOL_D)/minwhite $(TOOL_D)/nreplace\
 	$(TOOL_D)/sec-test $(TOOL_D)/srccount $(TOOL_D)/varrep
 tools: $(TOOLSEXE) $(TOOL_D)/depbuild
+$(TOOL_D)/fileutil.o: $(TOOL_D)/fileutil.cpp
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ -c $<
+$(TOOL_D)/strutil.o: $(TOOL_D)/strutil.cpp
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ -c $<
 $(TOOL_D)/archopt: $(TOOL_D)/archopt.c
-	@CC@ @CFLAGS@ -m32 -o $@ $@.c
+	$(HOST_CXX) $(HOST_CXXFLAGS) -m32 -o $@ $@.c
 $(TOOL_D)/cutrtype: $(TOOL_D)/cutrtype.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/depbuild: $(TOOL_D)/depbuild.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/extraext: $(TOOL_D)/extraext.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/macroll: $(TOOL_D)/macroll.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/minwhite: $(TOOL_D)/minwhite.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_D)/fileutil.o
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_D)/fileutil.o
 $(TOOL_D)/nreplace: $(TOOL_D)/nreplace.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_D)/fileutil.o
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_D)/fileutil.o
 $(TOOL_D)/sec-test: $(TOOL_D)/sec-test.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/srccount: $(TOOL_D)/srccount.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_D)/fileutil.o
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_D)/fileutil.o
 $(TOOL_D)/varrep: $(TOOL_D)/varrep.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) $(HOST_CXXFLAGS) -o $@ $@.cpp $(TOOL_O)
 
 include makefile.dep
 makefile.dep: $(TOOL_D)/depbuild Makefile
