diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-10-18 13:25:07.000000000 -0700
+++ b/CMakeLists.txt	2016-11-17 11:16:11.723432153 -0800
@@ -6,7 +6,8 @@
 find_package(Kodi REQUIRED)
 find_package(libretro-prboom REQUIRED)
 
+set(DEPLIBS prboom)
 set(PRBOOM_CUSTOM_BINARY ${PRBOOM_LIB}
-                         ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
+                          ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
-build_addon(${PROJECT_NAME} PRBOOM "")
+build_addon(${PROJECT_NAME} PRBOOM DEPLIBS)
