diff -Naur nestopia-eeda3a8/source/core/NstCore.hpp nestopia.patch/source/core/NstCore.hpp
--- nestopia-eeda3a8/source/core/NstCore.hpp	2016-12-26 18:10:34.498411156 +0100
+++ nestopia.patch/source/core/NstCore.hpp	2016-12-26 18:11:15.010409873 +0100
@@ -279,7 +279,7 @@
 		template<typename T>
 		inline long signed_shl(T v,uint c)
 		{
-			enum {NATIVE = T(-7) << 1 == -14};
+			enum {NATIVE = false};
 			return Helper::ShiftSigned<T,NATIVE>::Left( v, c );
 		}
 
