diff -Naur Python-2.6.1/setup.py Python-2.6.1.patch/setup.py
--- Python-2.6.1/setup.py	2009-02-15 02:05:08.000000000 +0100
+++ Python-2.6.1.patch/setup.py	2009-02-15 02:14:15.000000000 +0100
@@ -17,7 +17,14 @@
 from distutils.command.install_lib import install_lib
 
 # This global variable is used to hold the list of modules to be disabled.
-disabled_module_list = []
+try:
+	disabled_module_list = os.environ["PYTHON_DISABLE_MODULES"].split()
+except KeyError:
+	disabled_module_list = []
+try:
+	disable_ssl = os.environ["PYTHON_DISABLE_SSL"]
+except KeyError:
+	disable_ssl = 0
 
 def add_dir_to_list(dirlist, dir):
     """Add the directory 'dir' to the list 'dirlist' (at the front) if
@@ -310,6 +317,7 @@
         return sys.platform
 
     def detect_modules(self):
+        global disable_ssl
         # Ensure that /usr/local is always used
         try:
             modules_include_dirs = os.environ["PYTHON_MODULES_INCLUDE"].split()
@@ -632,7 +640,8 @@
                                      ] )
 
         if (ssl_incs is not None and
-            ssl_libs is not None):
+            ssl_libs is not None and
+			not disable_ssl):
             exts.append( Extension('_ssl', ['_ssl.c'],
                                    include_dirs = ssl_incs,
                                    library_dirs = ssl_libs,
