diff -Naur Python-2.6.1/Python/import.c Python-2.6.1.patch/Python/import.c
--- Python-2.6.1/Python/import.c	2008-09-01 16:18:30.000000000 +0200
+++ Python-2.6.1.patch/Python/import.c	2009-02-17 18:57:55.000000000 +0100
@@ -880,8 +880,12 @@
 	FILE *fp;
 	time_t mtime = srcstat->st_mtime;
 	mode_t mode = srcstat->st_mode;
-
+	char *py_dontcompile = getenv("PYTHON_DONTCOMPILE");
+ 
+    if (!py_dontcompile) 
 	fp = open_exclusive(cpathname, mode);
+    else
+	fp = NULL;
 	if (fp == NULL) {
 		if (Py_VerboseFlag)
 			PySys_WriteStderr(
