diff -Naur Python-2.6.1/setup.py Python-2.6.1.patch/setup.py
--- Python-2.6.1/setup.py	2008-11-04 21:43:31.000000000 +0100
+++ Python-2.6.1.patch/setup.py	2009-02-20 21:29:21.000000000 +0100
@@ -1668,7 +1668,8 @@
                                          ffi_configfile):
                 from distutils.dir_util import mkpath
                 mkpath(ffi_builddir)
-                config_args = []
+                config_args = ['--host=%s' % os.environ.get('GNU_HOST'), 
+                '--build=%s' % os.environ.get('GNU_BUILD')]
 
                 # Pass empty CFLAGS because we'll just append the resulting
                 # CFLAGS to Python's; -g or -O2 is to be avoided.
