diff -Naur Python-2.6.6/setup.py Python-2.6.6.patch/setup.py
--- Python-2.6.6/setup.py	2011-05-27 13:46:39.977454633 +0200
+++ Python-2.6.6.patch/setup.py	2011-05-27 13:47:28.200047208 +0200
@@ -758,7 +758,7 @@
 
         if (ssl_incs is not None and
             ssl_libs is not None and
-            openssl_ver >= 0x00907000 and False):
+            openssl_ver >= 0x00907000):
             # The _hashlib module wraps optimized implementations
             # of hash functions from the OpenSSL library.
             exts.append( Extension('_hashlib', ['_hashopenssl.c'],
@@ -778,7 +778,7 @@
                             depends = ['md5.h']) )
             missing.append('_hashlib')
 
-        if (True or openssl_ver < 0x00908000):
+        if (openssl_ver < 0x00908000):
             # OpenSSL doesn't do these until 0.9.8 so we'll bring our own hash
             exts.append( Extension('_sha256', ['sha256module.c']) )
             exts.append( Extension('_sha512', ['sha512module.c']) )
