diff -Naur Python-2.6.2/Modules/_ctypes/libffi/include/ffi.h.in Python-2.6.2.patch/Modules/_ctypes/libffi/include/ffi.h.in
--- Python-2.6.2/Modules/_ctypes/libffi/include/ffi.h.in	2008-04-04 18:01:54.000000000 +0200
+++ Python-2.6.2.patch/Modules/_ctypes/libffi/include/ffi.h.in	2009-08-01 23:31:01.102336818 +0200
@@ -70,19 +70,8 @@
 #include <stddef.h>
 #include <limits.h>
 
-/* LONG_LONG_MAX is not always defined (not if STRICT_ANSI, for example).
-   But we can find it either under the correct ANSI name, or under GNU
-   C's internal name.  */
-#ifdef LONG_LONG_MAX
-# define FFI_LONG_LONG_MAX LONG_LONG_MAX
-#else
-# ifdef LLONG_MAX
-#  define FFI_LONG_LONG_MAX LLONG_MAX
-# else
-#  ifdef __GNUC__
-#   define FFI_LONG_LONG_MAX __LONG_LONG_MAX__
-#  endif
-# endif
+#if defined __USE_ISOC99 && defined __GNUC__
+#  define FFI_LONG_LONG_MAX __LONG_LONG_MAX__
 #endif
 
 /* The closure code assumes that this works on pointers, i.e. a size_t	*/
