diff -Naur Python-2.7.3/configure Python-2.7.3.patch/configure
--- Python-2.7.3/configure	2012-04-10 01:07:36.000000000 +0200
+++ Python-2.7.3.patch/configure	2012-07-28 04:48:41.720322418 +0200
@@ -5425,11 +5425,11 @@
 		# debug builds.
 		OPT="-g -O0 -Wall $STRICT_PROTO"
 	    else
-		OPT="-g $WRAP -O3 -Wall $STRICT_PROTO"
+		OPT="-g $WRAP $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="$STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
diff -Naur Python-2.7.3/configure.in Python-2.7.3.patch/configure.in
--- Python-2.7.3/configure.in	2012-04-10 01:07:36.000000000 +0200
+++ Python-2.7.3.patch/configure.in	2012-07-28 04:48:22.987067842 +0200
@@ -946,11 +946,11 @@
 		# debug builds.
 		OPT="-g -O0 -Wall $STRICT_PROTO"
 	    else
-		OPT="-g $WRAP -O3 -Wall $STRICT_PROTO"
+		OPT="-g $WRAP $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="$STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
