--- a/drivers/media/dvb-avl/aml_dmx.c	2017-12-10 19:34:57.000000000 +0100
+++ b/drivers/media/dvb-avl/aml_dmx.c	2017-12-12 12:55:21.255063140 +0100
@@ -547,7 +547,7 @@
 	struct dvb_demux_feed *feed = dmx->channel[f->chan_id].feed;
 
 	if (feed && feed->cb.sec)
-		feed->cb.sec(p, sec_len, NULL, 0, f->filter, DMX_OK);
+		feed->cb.sec(p, sec_len, NULL, 0, f->filter);
 }
 
 static void hardware_match_section(struct aml_dmx *dmx,
@@ -968,7 +968,7 @@
 		if (dmx->channel[2].feed && dmx->channel[2].feed->cb.ts) {
 			dmx->channel[2].feed->cb.ts(buffer1_virt, len1,
 						buffer2_virt, len2,
-						&dmx->channel[2].feed->feed.ts, DMX_OK);
+						&dmx->channel[2].feed->feed.ts);
 		}
 	}
 	WRITE_MPEG_REG(PARSER_SUB_RP, rd_ptr);
@@ -1167,7 +1167,7 @@
 			channel->dvr_feed->cb.ts(
 					(u8 *)afifo->pages+afifo->buf_read*size,
 					cnt*size, NULL, 0,
-					&channel->dvr_feed->feed.ts, DMX_OK);
+					&channel->dvr_feed->feed.ts);
 		afifo->buf_read = 0;
 	}
 
@@ -1186,7 +1186,7 @@
 			channel->dvr_feed->cb.ts(
 					(u8 *)afifo->pages+afifo->buf_read*size,
 					cnt*size, NULL, 0,
-					&channel->dvr_feed->feed.ts, DMX_OK);
+					&channel->dvr_feed->feed.ts);
 		afifo->buf_read = afifo->buf_toggle;
 	}
 
--
