diff --git a/drivers/media/amlogic/Kconfig b/drivers/media/amlogic/Kconfig
index 5203702..7af18b7 100644
--- a/drivers/media/amlogic/Kconfig
+++ b/drivers/media/amlogic/Kconfig
@@ -5,4 +5,6 @@ menu "Amlogic Device Drivers"
 
 source "drivers/media/amlogic/video_dev/Kconfig"
 
+source "drivers/media/amlogic/wetek/Kconfig"
+
 endmenu
diff --git a/drivers/media/amlogic/Makefile b/drivers/media/amlogic/Makefile
index 049cb81..99bbb31 100644
--- a/drivers/media/amlogic/Makefile
+++ b/drivers/media/amlogic/Makefile
@@ -3,3 +3,5 @@
 ##########################################
 
 obj-$(CONFIG_V4L_AMLOGIC_VIDEO) += video_dev/
+
+obj-$(CONFIG_WETEK) += wetek/
