--- a/include/uapi/linux/dvb/dmx.h	2017-01-23 18:15:29.000000000 +0100
+++ b/include/uapi/linux/dvb/dmx.h	2017-02-08 14:12:07.802477417 +0100
@@ -197,6 +197,20 @@
 	__u32           flags;
 };
 
+typedef enum dmx_source {
+	DMX_SOURCE_FRONT0 = 0,
+	DMX_SOURCE_FRONT1,
+	DMX_SOURCE_FRONT2,
+	DMX_SOURCE_FRONT3,
+	DMX_SOURCE_DVR0   = 16,
+	DMX_SOURCE_DVR1,
+	DMX_SOURCE_DVR2,
+	DMX_SOURCE_DVR3,
+	DMX_SOURCE_FRONT0_OFFSET = 100,
+	DMX_SOURCE_FRONT1_OFFSET,
+	DMX_SOURCE_FRONT2_OFFSET
+} dmx_source_t;
+
 /**
  * struct dmx_stc - Stores System Time Counter (STC) information.
  *
--
