diff --git a/xbmc/windowing/WinEventsSDL.cpp b/xbmc/windowing/WinEventsSDL.cpp
index c266c1e..9bfd6ac 100644
--- a/xbmc/windowing/WinEventsSDL.cpp
+++ b/xbmc/windowing/WinEventsSDL.cpp
@@ -439,7 +439,6 @@ bool CWinEventsSDL::ProcessLinuxShortcuts(SDL_Event& event)
     switch(event.key.keysym.sym)
     {
       case SDLK_TAB:  // ALT+TAB to minimize/hide
-        g_application.Minimize();
         return true;
       default:
         break;
