diff -Naur bluez-4.71/scripts/bluetooth.rules bluez-4.71.patch/scripts/bluetooth.rules
--- bluez-4.71/scripts/bluetooth.rules	2010-09-08 17:29:06.000000000 +0200
+++ bluez-4.71.patch/scripts/bluetooth.rules	2010-09-22 20:23:11.973264738 +0200
@@ -1,3 +1,4 @@
 # Run helper every time a Bluetooth device appears
 # On remove actions, bluetoothd should go away by itself
 ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="/usr/local/sbin/bluetoothd --udev"
+ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="/usr/local/sbin/bluetoothd --udev"
diff -Naur bluez-4.71/scripts/bluetooth.rules.in bluez-4.71.patch/scripts/bluetooth.rules.in
--- bluez-4.71/scripts/bluetooth.rules.in	2009-07-06 22:21:40.000000000 +0200
+++ bluez-4.71.patch/scripts/bluetooth.rules.in	2010-09-22 20:23:01.488265635 +0200
@@ -1,3 +1,4 @@
 # Run helper every time a Bluetooth device appears
 # On remove actions, bluetoothd should go away by itself
 ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
+ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
