diff -Naur a/tools/hciattach_bcm43xx.c b/tools/hciattach_bcm43xx.c
--- a/tools/hciattach_bcm43xx.c	2016-02-25 18:23:15.468717720 +0000
+++ b/tools/hciattach_bcm43xx.c	2016-02-25 18:24:08.028997243 +0000
@@ -368,9 +368,6 @@
 	if (bcm43xx_locate_patch(FIRMWARE_DIR, chip_name, fw_path)) {
 		fprintf(stderr, "Patch not found, continue anyway\n");
 	} else {
-		if (bcm43xx_set_speed(fd, ti, speed))
-			return -1;
-
 		if (bcm43xx_load_firmware(fd, fw_path))
 			return -1;
 
@@ -379,6 +376,7 @@
 			perror("Can't set host baud rate");
 			return -1;
 		}
+		sleep(1);
 
 		if (bcm43xx_reset(fd))
 			return -1;
