From 40c6e7c5a0d81a781c94e5a64661848018c1f8be Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Thu, 19 Sep 2019 22:55:09 +0100
Subject: [PATCH] sixaxis autotrust

---
 plugins/sixaxis.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/sixaxis.c b/plugins/sixaxis.c
index f6baea7..c166304 100644
--- a/plugins/sixaxis.c
+++ b/plugins/sixaxis.c
@@ -371,7 +371,7 @@ static bool setup_device(int fd, const char *sysfs_path,
 	else
 		btd_device_set_pnpid(device, cp->source, cp->vid, cp->pid, cp->version);
 
-	btd_device_set_trusted(device, false);
+	btd_device_set_trusted(device, true);
 	btd_device_set_temporary(device, true);
 
 	closure = g_new0(struct authentication_closure, 1);
-- 
2.7.4

