diff -Naur bluez-5.37/plugins/sixaxis.c bluez-5.37.patch/plugins/sixaxis.c
--- bluez-5.37/plugins/sixaxis.c	2015-03-31 15:41:57.000000000 +0200
+++ bluez-5.37.patch/plugins/sixaxis.c	2016-04-21 12:13:25.044666696 +0200
@@ -303,6 +303,7 @@
 	btd_device_set_pnpid(device, devices[index].source, devices[index].vid,
 				devices[index].pid, devices[index].version);
 	btd_device_set_temporary(device, false);
+	btd_device_set_trusted(device, true);
 
 	return true;
 }
