diff -Naur bluez-5.35/Makefile.tools bluez-5.35.patch/Makefile.tools
--- bluez-5.35/Makefile.tools	2015-09-04 03:19:36.000000000 +0200
+++ bluez-5.35.patch/Makefile.tools	2015-10-02 15:28:03.406835739 +0200
@@ -8,7 +8,7 @@
 					client/gatt.h client/gatt.c \
 					monitor/uuid.h monitor/uuid.c
 client_bluetoothctl_LDADD = gdbus/libgdbus-internal.la @GLIB_LIBS@ @DBUS_LIBS@ \
-				-lreadline
+				-lreadline -lncurses
 endif
 
 if MONITOR
@@ -340,12 +340,12 @@
 				attrib/utils.c src/log.c client/display.c \
 				client/display.h
 attrib_gatttool_LDADD = lib/libbluetooth-internal.la \
-			src/libshared-glib.la @GLIB_LIBS@ -lreadline
+			src/libshared-glib.la @GLIB_LIBS@ -lreadline -lncurses
 
 tools_obex_client_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 						tools/obex-client-tool.c
 tools_obex_client_tool_LDADD = lib/libbluetooth-internal.la \
-						@GLIB_LIBS@ -lreadline
+						@GLIB_LIBS@ -lreadline -lncurses
 
 tools_obex_server_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 						tools/obex-server-tool.c
@@ -354,16 +354,16 @@
 tools_bluetooth_player_SOURCES = tools/bluetooth-player.c \
 				client/display.h client/display.c
 tools_bluetooth_player_LDADD = gdbus/libgdbus-internal.la \
-				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline
+				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline -lncurses
 
 tools_obexctl_SOURCES = tools/obexctl.c \
 				client/display.h client/display.c
 tools_obexctl_LDADD = gdbus/libgdbus-internal.la \
-				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline
+				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline -lncurses
 
 tools_btmgmt_SOURCES = tools/btmgmt.c src/uuid-helper.c client/display.c
 tools_btmgmt_LDADD = lib/libbluetooth-internal.la src/libshared-mainloop.la \
-				-lreadline
+				-lreadline -lncurses
 endif
 
 if EXPERIMENTAL
