diff --git a/Makefile.obexd b/Makefile.obexd
index cd3ace4..89f1609 100644
--- a/Makefile.obexd
+++ b/Makefile.obexd
@@ -8,8 +8,6 @@ endif
 
 EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service
 
-if OBEX
-
 obex_plugindir = $(libdir)/obex/plugins
 
 obexd_builtin_modules =
@@ -33,6 +31,8 @@ obexd_builtin_sources += obexd/plugins/opp.c
 obexd_builtin_modules += ftp
 obexd_builtin_sources += obexd/plugins/ftp.c obexd/plugins/ftp.h
 
+if OBEX
+
 obexd_builtin_modules += irmc
 obexd_builtin_sources += obexd/plugins/irmc.c
 
@@ -47,6 +47,8 @@ obexd_builtin_sources += obexd/plugins/mas.c obexd/src/map_ap.h \
                                obexd/plugins/messages.h \
                                obexd/plugins/messages-dummy.c
 
+endif
+
 obexd_builtin_modules += mns
 obexd_builtin_sources += obexd/client/mns.c obexd/src/map_ap.h \
                                obexd/client/map-event.h
@@ -90,8 +92,6 @@ obexd_src_obexd_CFLAGS = $(AM_CFLAGS) @GLIB_CFLAGS@ @DBUS_CFLAGS@ \
 
 obexd_src_obexd_CPPFLAGS = -I$(builddir)/lib -I$(builddir)/obexd/src
 
-endif
-
 obexd_src_obexd_SHORTNAME = obexd
 
 obexd_builtin_files = obexd/src/builtin.h $(obexd_builtin_nodist)
