diff -Naur bluez-5.1/test/simple-agent bluez-5.1.patch/test/simple-agent
--- bluez-5.1/test/simple-agent	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/simple-agent	2013-02-02 05:24:47.752050167 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import sys
 import dbus
@@ -151,7 +151,7 @@
 	path = "/test/agent"
 	agent = Agent(bus, path)
 
-	mainloop = GObject.MainLoop()
+	mainloop = gobject.MainLoop()
 
 	obj = bus.get_object(BUS_NAME, "/org/bluez");
 	manager = dbus.Interface(obj, "org.bluez.AgentManager1")
diff -Naur bluez-5.1/test/test-device bluez-5.1.patch/test/test-device
--- bluez-5.1/test/test-device	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/test-device	2013-02-02 05:24:47.755050146 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import sys
 import dbus
@@ -13,7 +13,7 @@
 
 dbus.mainloop.glib.DBusGMainLoop(set_as_default=True)
 bus = dbus.SystemBus()
-mainloop = GObject.MainLoop()
+mainloop = gobject.MainLoop()
 
 option_list = [
 		make_option("-i", "--device", action="store",
diff -Naur bluez-5.1/test/test-discovery bluez-5.1.patch/test/test-discovery
--- bluez-5.1/test/test-discovery	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/test-discovery	2013-02-02 05:24:47.755050146 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import dbus
 import dbus.mainloop.glib
@@ -152,5 +152,5 @@
 
 	adapter.StartDiscovery()
 
-	mainloop = GObject.MainLoop()
+	mainloop = gobject.MainLoop()
 	mainloop.run()
diff -Naur bluez-5.1/test/test-hfp bluez-5.1.patch/test/test-hfp
--- bluez-5.1/test/test-hfp	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/test-hfp	2013-02-02 05:24:47.756050139 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import os
 import sys
@@ -217,7 +217,7 @@
 
 	(options, args) = parser.parse_args()
 
-	mainloop = GObject.MainLoop()
+	mainloop = gobject.MainLoop()
 
 	opts = {
 			"Version" : dbus.UInt16(0x0106),
diff -Naur bluez-5.1/test/test-manager bluez-5.1.patch/test/test-manager
--- bluez-5.1/test/test-manager	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/test-manager	2013-02-02 05:24:47.757050132 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import dbus
 import dbus.mainloop.glib
@@ -35,5 +35,5 @@
 	except:
 		print("No adapter found")
 
-	mainloop = GObject.MainLoop()
+	mainloop = gobject.MainLoop()
 	mainloop.run()
diff -Naur bluez-5.1/test/test-profile bluez-5.1.patch/test/test-profile
--- bluez-5.1/test/test-profile	2012-12-24 18:46:55.000000000 +0100
+++ bluez-5.1.patch/test/test-profile	2013-02-02 05:24:47.757050132 +0100
@@ -2,7 +2,7 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-from gi.repository import GObject
+import gobject
 
 import os
 import sys
@@ -93,7 +93,7 @@
 
 	profile = Profile(bus, options.path)
 
-	mainloop = GObject.MainLoop()
+	mainloop = gobject.MainLoop()
 
 	opts = {
 			"AutoConnect" :	options.auto_connect,
