From 20dfa1079c088236bcd63f1a986956b488177ebf Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Tue, 6 Aug 2013 11:18:09 +0300
Subject: [PATCH] obexd: use system bus

---
 obexd/src/manager.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/obexd/src/manager.c b/obexd/src/manager.c
index dbfbef8..9a29f8c 100644
--- a/obexd/src/manager.c
+++ b/obexd/src/manager.c
@@ -569,7 +569,7 @@ gboolean manager_init(void)
 
 	dbus_error_init(&err);
 
-	connection = g_dbus_setup_bus(DBUS_BUS_SESSION, OBEXD_SERVICE, &err);
+	connection = g_dbus_setup_bus(DBUS_BUS_SYSTEM, OBEXD_SERVICE, &err);
 	if (connection == NULL) {
 		if (dbus_error_is_set(&err) == TRUE) {
 			fprintf(stderr, "%s\n", err.message);
-- 
1.7.2.5

