--- a/v4l/scripts/make_kconfig.pl
+++ b/v4l/scripts/make_kconfig.pl
@@ -626,6 +626,11 @@ ($$)
 close OUT;
 
 # These options should default to off
+disable_config('MEDIA_ANALOG_TV_SUPPORT');
+disable_config('MEDIA_CAMERA_SUPPORT');
+disable_config('MEDIA_CEC_SUPPORT');
+disable_config('SOC_CAMERA');
+disable_config('CEC_CORE');
 disable_config('DVB_AV7110_FIRMWARE');
 disable_config('DVB_CINERGYT2_TUNING');
 disable_config('VIDEO_HELPER_CHIPS_AUTO');
--- a/v4l/versions.txt
+++ b/v4l/versions.txt
@@ -2,6 +2,9 @@
 [9.255.255]
 VIDEO_IPU3_IMGU
 VIDEO_DW9714
+CEC_NOTIFIER
+MEDIA_CEC_RC
+MEDIA_CEC_SUPPORT
 
 [4.19.0]
 # Needs union ec_response_get_next_data_v1.cec_message
