diff -Nur a/arch/arm/Kconfig usb/arch/arm/Kconfig
--- a/arch/arm/Kconfig	2016-09-12 16:05:51.000000000 +0200
+++ usb/arch/arm/Kconfig	2016-09-21 19:19:44.533396743 +0200
@@ -590,9 +590,8 @@
 	select SPL_SYS_THUMB_BUILD if !ARM64
 	select SYS_NS16550
 	select SYS_THUMB_BUILD if !ARM64
 	select USB if DISTRO_DEFAULTS
-	select USB_KEYBOARD if DISTRO_DEFAULTS
 	select USB_STORAGE if DISTRO_DEFAULTS
 	select USE_TINY_PRINTF
 	imply CMD_DM
 	imply CMD_GPT
