--- u-boot/include/configs/sunxi-common.h	2016-08-24 12:16:55.774029888 +0200
+++ u-boot.patched/include/configs/sunxi-common.h	2016-08-24 11:50:59.000000000 +0200
@@ -497,7 +497,14 @@ extern int soft_i2c_gpio_scl;
 		"ext2load mmc 0 0x48000000 uImage && " \
 		"bootm 0x48000000\0"
 #else
-#define BOOTCMD_SUNXI_COMPAT
+#define BOOTCMD_SUNXI_COMPAT \
+		"bootcmd_sunxi=" \
+			"if load mmc 0 0x44000000 uEnv.txt; then " \
+				"echo Loaded environment from uEnv.txt; " \
+				"env import -t 0x44000000 ${filesize}; " \
+			"fi;" \
+		"run bootcmd;\0"
+#define CONFIG_BOOTCOMMAND "run bootcmd_sunxi"
 #endif
 
 #include <config_distro_bootcmd.h>
