diff --git a/dvbdevice.c b/dvbdevice.c
index 42ec5b1..9c397d7 100644
--- a/dvbdevice.c
+++ b/dvbdevice.c
@@ -839,7 +839,11 @@ bool cDvbTuner::SetFrontend(void)
      SETCMD(DTV_HIERARCHY, dtp.Hierarchy());
      if (frontendType == SYS_DVBT2) {
         // DVB-T2
+#ifdef DTV_DVBT2_PLP_ID
         SETCMD(DTV_DVBT2_PLP_ID, dtp.PlpId());
+#else
+        SETCMD(DTV_DVBT2_PLP_ID_LEGACY, dtp.PlpId());
+#endif
         }
 
      tuneTimeout = DVBT_TUNE_TIMEOUT;
