diff --git a/dvbdevice.h b/dvbdevice.h
index 2a34806..dc95443 100644
--- a/dvbdevice.h
+++ b/dvbdevice.h
@@ -67,10 +67,6 @@ enum {
 
 // --- End of definitions for older DVB API versions -------------------------
 
-#if (DVB_API_VERSION << 8 | DVB_API_VERSION_MINOR) < 0x0503
-#error VDR requires Linux DVB driver API version 5.3 or higher!
-#endif
-
 #define MAXDVBDEVICES  8
 #define MAXDELIVERYSYSTEMS 8
 
