diff -Nru vdr-1.7.15-vanilla/eitscan.c vdr-1.7.15-disable_eitscan/eitscan.c
--- vdr-1.7.15-vanilla/eitscan.c	2010-06-06 19:11:14.000000000 +0300
+++ vdr-1.7.15-disable_eitscan/eitscan.c	2010-06-06 20:44:07.000000000 +0300
@@ -146,7 +146,7 @@
                if (Device) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (!Device->Receiving()) {
