From 092b87b7789206f041de68016fb9a900577a6681 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 13 Oct 2013 20:40:36 +0300
Subject: [PATCH] no locale support

---
 vdr.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/vdr.c b/vdr.c
index ba59f07..c1b0541 100644
--- a/vdr.c
+++ b/vdr.c
@@ -637,10 +637,6 @@ int main(int argc, char *argv[])
      cCharSetConv::SetSystemCharacterTable(CodeSet);
      }
 
-  // Initialize internationalization:
-
-  I18nInitialize(LocaleDirectory);
-
   // Main program loop variables - need to be here to have them initialized before any EXIT():
 
   cEpgDataReader EpgDataReader;
-- 
1.7.2.5

