diff --git a/dvbdevice.c b/dvbdevice.c
index 42ec5b1..69a6f0c 100644
--- a/dvbdevice.c
+++ b/dvbdevice.c
@@ -839,7 +839,7 @@ bool cDvbTuner::SetFrontend(void)
      SETCMD(DTV_HIERARCHY, dtp.Hierarchy());
      if (frontendType == SYS_DVBT2) {
         // DVB-T2
-        SETCMD(DTV_DVBT2_PLP_ID, dtp.PlpId());
+        SETCMD(DTV_DVBT2_PLP_ID_LEGACY, dtp.PlpId());
         }
 
      tuneTimeout = DVBT_TUNE_TIMEOUT;
