diff -Naur gdb-6.8/bfd/configure gdb-6.8a/bfd/configure
--- gdb-6.8/bfd/configure	2008-02-14 16:20:25.000000000 +0100
+++ gdb-6.8a/bfd/configure	2009-07-19 09:14:08.000000000 +0200
@@ -5034,6 +5034,11 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+linux-uclibc*)
+  lt_cv_deplibs_check_method=pass_all
+  lt_cv_file_magic_test_file=`echo /lib/libuClibc-*.so`
+  ;;
+
 netbsd*)
   if echo __ELF__ | $CC -E - | $GREP __ELF__ > /dev/null; then
     lt_cv_deplibs_check_method='match_pattern /lib[^/]+(\.so\.[0-9]+\.[0-9]+|_pic\.a)$'
diff -Naur gdb-6.8/libtool.m4 gdb-6.8a/libtool.m4
--- gdb-6.8/libtool.m4	2008-02-16 16:26:46.000000000 +0100
+++ gdb-6.8a/libtool.m4	2009-07-19 09:15:44.000000000 +0200
@@ -2937,6 +2937,11 @@
   fi
   ;;
 
+linux-uclibc*)
+  lt_cv_deplibs_check_method=pass_all
+  lt_cv_file_magic_test_file=`echo /lib/libuClibc-*.so`
+  ;;
+
 newos6*)
   lt_cv_deplibs_check_method='file_magic ELF [[0-9]][[0-9]]*-bit [[ML]]SB (executable|dynamic lib)'
   lt_cv_file_magic_cmd=/usr/bin/file
diff -Naur gdb-6.8/opcodes/configure gdb-6.8a/opcodes/configure
--- gdb-6.8/opcodes/configure	2008-02-14 13:33:17.000000000 +0100
+++ gdb-6.8a/opcodes/configure	2009-07-19 09:17:14.000000000 +0200
@@ -5038,6 +5038,11 @@
   fi
   ;;
 
+linux-uclibc*)
+  lt_cv_deplibs_check_method=pass_all
+  lt_cv_file_magic_test_file=`echo /lib/libuClibc-*.so`
+  ;;
+
 newos6*)
   lt_cv_deplibs_check_method='file_magic ELF [0-9][0-9]*-bit [ML]SB (executable|dynamic lib)'
   lt_cv_file_magic_cmd=/usr/bin/file
