diff --git a/magic/Magdir/archive b/magic/Magdir/archive
index c998e5b..4da532a 100644
--- a/magic/Magdir/archive
+++ b/magic/Magdir/archive
@@ -688,6 +688,7 @@
 >>4	byte		0x0b		\b, at least v1.1 to extract
 >>0x161	string		WINZIP		\b, WinZIP self-extracting
 >>4	byte		0x14		\b, at least v2.0 to extract
+>>4     byte            0x2d            Zip64, at least v3.0 to extract
 
 # Zoo archiver
 20	lelong		0xfdc4a7dc	Zoo archive data
