From 19f5096d0b4c1a49af6cc4e0d086d6128c493bd8 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Wed, 18 May 2011 18:11:23 +0000
Subject: [PATCH] Fixed 0 file problem.

---
 magic/Magdir/msdos |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/magic/Magdir/msdos b/magic/Magdir/msdos
index 731d50b..f8852cb 100644
--- a/magic/Magdir/msdos
+++ b/magic/Magdir/msdos
@@ -389,10 +389,11 @@
 >>4	uleshort&0x4842			>0			\bsupport
 >0	ubyte				x			\b)
 # DOS driver cmd640x.sys has 0x12 instead of 0xffffffff for pointer field to next device header 
-0	ulequad&0x000007a0ffffffed	0x0000000000000000	DOS-executable (
->4	uleshort&0x8000			0x8000			\bcharacter device driver
->>10	string				x			%-.8s
->4	uleshort&0x4000			0x4000			\b,control strings-support)
+# Too weak, matches files that only contain 0's
+#0	ulequad&0x000007a0ffffffed	0x0000000000000000	DOS-executable (
+#>4	uleshort&0x8000			0x8000			\bcharacter device driver
+#>>10	string				x			%-.8s
+#>4	uleshort&0x4000			0x4000			\b,control strings-support)
 
 # test too generic ?
 0	byte		0x8c		DOS executable (COM)
-- 
1.7.4.1

