From e990dc53f658cb67ec7b33c32e0bc24e65801792 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Sat, 14 May 2011 15:04:15 +0000
Subject: [PATCH] small archive

---
 magic/Magdir/archive |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/magic/Magdir/archive b/magic/Magdir/archive
index e84c8e3..5ad1ce3 100644
--- a/magic/Magdir/archive
+++ b/magic/Magdir/archive
@@ -654,6 +654,9 @@
 >>>>78	string	-template		Template
 !:mime	application/vnd.oasis.opendocument.image-template
 
+>26	byte	x			Zip archive data
+
+
 # StarView Metafile
 # From Pierre Ducroquet <pinaraf@pinaraf.info>
 0	string	VCLMTF	StarView MetaFile
-- 
1.7.4.1

