From a81c0756c48621fadd18597bb337e08c4c3529d0 Mon Sep 17 00:00:00 2001
From: Yan Li <yan.i.li@intel.com>
Date: Mon, 5 Jan 2009 17:16:03 +0800
Subject: [PATCH] Linux should use evdev xkb rule


Signed-off-by: Yan Li <yan.i.li@intel.com>
---
 fdi/policy/10osvendor/10-keymap.fdi |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/fdi/policy/10osvendor/10-keymap.fdi b/fdi/policy/10osvendor/10-keymap.fdi
index c9ad87c..1971e8b 100644
--- a/fdi/policy/10osvendor/10-keymap.fdi
+++ b/fdi/policy/10osvendor/10-keymap.fdi
@@ -13,6 +13,7 @@
       <merge key="input.xkb.model" type="string">keyboard</merge>
       <match key="/org/freedesktop/Hal/devices/computer:system.kernel.name"
              string="Linux">
+        <merge key="input.xkb.rules" type="string">evdev</merge>
         <merge key="input.xkb.model" type="string">evdev</merge>
       </match>
 
-- 
1.5.6.5

