diff -ur a/binutils/Makefile.in b/binutils/Makefile.in
--- a/binutils/Makefile.in	2020-02-01 12:50:05.000000000 +0100
+++ b/binutils/Makefile.in	2020-02-20 00:45:18.792253182 +0100
@@ -560,7 +560,7 @@
 zlibinc = @zlibinc@
 AUTOMAKE_OPTIONS = dejagnu no-dist foreign subdir-objects
 ACLOCAL_AMFLAGS = -I .. -I ../config -I ../bfd
-SUBDIRS = doc po
+SUBDIRS =
 tooldir = $(exec_prefix)/$(target_alias)
 
 # Automake 1.10+ disables lex and yacc output file regeneration if
