diff -Naur sshfs-fuse-2.2/sshfs.c sshfs-fuse-2.2.patch/sshfs.c
--- sshfs-fuse-2.2/sshfs.c	2008-10-20 14:10:09.000000000 +0200
+++ sshfs-fuse-2.2.patch/sshfs.c	2011-03-03 21:23:02.721900708 +0100
@@ -2920,7 +2920,7 @@
 		return 1;
 
 	case KEY_PORT:
-		tmp = g_strdup_printf("-oPort=%s", arg + 2);
+		tmp = g_strdup_printf("-p %s", arg + 2);
 		ssh_add_arg(tmp);
 		g_free(tmp);
 		return 0;
@@ -3183,9 +3183,9 @@
 	sshfs.ptyfd = -1;
 	sshfs.ptyslavefd = -1;
 	ssh_add_arg("ssh");
-	ssh_add_arg("-x");
-	ssh_add_arg("-a");
-	ssh_add_arg("-oClearAllForwardings=yes");
+//	ssh_add_arg("-x");
+//	ssh_add_arg("-a");
+//	ssh_add_arg("-oClearAllForwardings=yes");
 
 	if (fuse_opt_parse(&args, &sshfs, sshfs_opts, sshfs_opt_proc) == -1 ||
 	    parse_workarounds() == -1)
