diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce19c4cc..07131a02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -670,52 +670,6 @@ if (${PLAYER_BUILD_EXECUTABLE})
 	endif()
 endif()
 
-# manpage
-# Handled as extra target "man"
-set(MAN_NAME easyrpg-player.6)
-find_program(A2X_EXECUTABLE NAMES a2x a2x.py)
-if(NOT A2X_EXECUTABLE STREQUAL "A2X_EXECUTABLE-NOTFOUND")
-	add_custom_command(OUTPUT resources/${MAN_NAME}
-		COMMAND ${A2X_EXECUTABLE} -a player_version="${PROJECT_VERSION}" -f manpage -D ${CMAKE_CURRENT_BINARY_DIR}/resources ${CMAKE_CURRENT_SOURCE_DIR}/resources/${MAN_NAME}.adoc
-		DEPENDS resources/${MAN_NAME}.adoc
-		COMMENT "(Re-)building manpage ${MAN_NAME}"
-		VERBATIM)
-	if(UNIX)
-		add_custom_target(man ALL DEPENDS resources/${MAN_NAME})
-		install(FILES resources/${MAN_NAME} DESTINATION ${CMAKE_INSTALL_MANDIR}/man6)
-	else()
-		add_custom_target(man DEPENDS resources/${MAN_NAME})
-	endif()
-else()
-	# no a2x, distribution archive?
-	if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/resources/${MAN_NAME})
-		install(FILES resources/${MAN_NAME} DESTINATION share/man/man6)
-	endif()
-endif()
-
-# Doxygen
-# Handled as extra target "doc"
-find_package(Doxygen)
-if(DOXYGEN_FOUND)
-	# fake autotools variables
-	set(DX_DOCDIR ${CMAKE_CURRENT_BINARY_DIR}/doc)
-	set(srcdir ${CMAKE_CURRENT_SOURCE_DIR})
-	configure_file(resources/Doxyfile.in resources/Doxyfile @ONLY)
-	unset(DX_DOCDIR)
-	unset(srcdir)
-
-	add_custom_target(player_doc
-		${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/resources/Doxyfile
-		DEPENDS ${PROJECT_NAME}
-		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/resources
-		COMMENT "Generating API documentation with Doxygen" VERBATIM)
-	if(NOT TARGET doc)
-		# Handle liblcf doc target
-		add_custom_target(doc)
-	endif()
-	add_dependencies(doc player_doc)
-endif()
-
 # Unit tests
 option(PLAYER_ENABLE_TESTS "Execute unit tests after compilation finishes" ON)
 