From 2b06ab6d8603b511150d965e876ece1c24ee80e3 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 7 Mar 2015 00:16:48 +0200
Subject: [PATCH] disable network mode on new installs

---
 DVBAPISetup.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DVBAPISetup.cpp b/DVBAPISetup.cpp
index b776f21..b540ebf 100644
--- a/DVBAPISetup.cpp
+++ b/DVBAPISetup.cpp
@@ -19,7 +19,7 @@
 #include "DVBAPISetup.h"
 
 int LogLevel = 2;
-int OSCamNetworkMode = 1;
+int OSCamNetworkMode = 0;
 char OSCamHost[HOST_NAME_MAX] = "localhost";
 int OSCamPort = 2000;
 
-- 
1.7.10.4

