From 7d51cc457823156d446d06c6ae40f850958fa735 Mon Sep 17 00:00:00 2001
From: Mariusz Bialonczyk <manio@skyboo.net>
Date: Sat, 25 Nov 2017 18:57:08 +0100
Subject: [PATCH] Fix compilation for kernels >= 4.14

Closes #122
---
 CA.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CA.h b/CA.h
index 7130be6..019cd1c 100644
--- a/CA.h
+++ b/CA.h
@@ -9,4 +9,12 @@ typedef struct ca_descr_aes {
   unsigned char cw[16];
 } ca_descr_aes_t;
 
+#ifndef CA_SET_PID /* removed in kernel 4.14 */
+typedef struct ca_pid {
+        unsigned int pid;
+        int index;          /* -1 == disable */
+} ca_pid_t;
+#define CA_SET_PID _IOW('o', 135, struct ca_pid)
 #endif
+
+#endif // __CA_H
