From 1a84f218bdf914b1622442ef4abd380ee1878388 Mon Sep 17 00:00:00 2001
From: CvH <namerp@googlemail.com>
Date: Fri, 18 Mar 2016 11:04:06 +0100
Subject: [PATCH] crosscompiling

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c6531ec..e367247 100644
--- a/Makefile
+++ b/Makefile
@@ -141,7 +141,7 @@ $(SOFILE): $(OBJS) $(FFDECSA)
 ifndef LIBDVBCSA
 $(FFDECSA): $(FFDECSADIR)/*.c $(FFDECSADIR)/*.h
 	@echo CC $@
-	$(Q)@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CXXFLAGS) $(LDFLAGS) $(CSAFLAGS)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
+	$(Q)@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CXXFLAGS) $(LDFLAGS) $(CSAFLAGS)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) FFdecsa.o
 endif
 
 install-lib: $(SOFILE)
-- 
2.51.0

