diff -Naur elfutils-0.155/Makefile.am elfutils-0.155.patch/Makefile.am
--- elfutils-0.155/Makefile.am	2012-08-27 20:27:31.000000000 +0200
+++ elfutils-0.155.patch/Makefile.am	2013-03-05 18:38:47.565095550 +0100
@@ -23,8 +23,7 @@
 pkginclude_HEADERS = version.h
 
 # Add doc back when we have some real content.
-SUBDIRS = config m4 lib libelf libebl libdwfl libdw libcpu libasm backends \
-	  src po tests
+SUBDIRS = config m4 lib libelf libebl libdwfl libdw
 
 EXTRA_DIST = elfutils.spec GPG-KEY NOTES CONTRIBUTING \
 	     COPYING COPYING-GPLV2 COPYING-LGPLV3
diff -Naur elfutils-0.155/Makefile.in elfutils-0.155.patch/Makefile.in
--- elfutils-0.155/Makefile.in	2012-08-27 20:29:45.000000000 +0200
+++ elfutils-0.155.patch/Makefile.in	2013-03-05 18:39:11.176974119 +0100
@@ -256,8 +256,7 @@
 pkginclude_HEADERS = version.h
 
 # Add doc back when we have some real content.
-SUBDIRS = config m4 lib libelf libebl libdwfl libdw libcpu libasm backends \
-	  src po tests
+SUBDIRS = config m4 lib libelf libebl libdwfl libdw
 
 EXTRA_DIST = elfutils.spec GPG-KEY NOTES CONTRIBUTING \
 	     COPYING COPYING-GPLV2 COPYING-LGPLV3
