diff -Naur elfutils-0.155/libasm/Makefile.am elfutils-0.155.patch/libasm/Makefile.am
--- elfutils-0.155/libasm/Makefile.am	2012-08-27 20:27:31.000000000 +0200
+++ elfutils-0.155.patch/libasm/Makefile.am	2013-03-05 18:11:43.841545009 +0100
@@ -69,7 +69,6 @@
 		-Wl,--version-script,$(srcdir)/libasm.map,--no-undefined \
 		-Wl,--soname,$@.$(VERSION) \
 		../libebl/libebl.a ../libelf/libelf.so  $(libasm_so_LDLIBS)
-	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 	ln -fs $@ $@.$(VERSION)
 
 install: install-am libasm.so
diff -Naur elfutils-0.155/libasm/Makefile.in elfutils-0.155.patch/libasm/Makefile.in
--- elfutils-0.155/libasm/Makefile.in	2012-08-27 20:29:45.000000000 +0200
+++ elfutils-0.155.patch/libasm/Makefile.in	2013-03-05 18:12:05.604432305 +0100
@@ -654,7 +654,6 @@
 @MUDFLAP_FALSE@		-Wl,--version-script,$(srcdir)/libasm.map,--no-undefined \
 @MUDFLAP_FALSE@		-Wl,--soname,$@.$(VERSION) \
 @MUDFLAP_FALSE@		../libebl/libebl.a ../libelf/libelf.so  $(libasm_so_LDLIBS)
-@MUDFLAP_FALSE@	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 @MUDFLAP_FALSE@	ln -fs $@ $@.$(VERSION)
 
 @MUDFLAP_FALSE@install: install-am libasm.so
diff -Naur elfutils-0.155/libdw/Makefile.am elfutils-0.155.patch/libdw/Makefile.am
--- elfutils-0.155/libdw/Makefile.am	2012-08-27 20:27:31.000000000 +0200
+++ elfutils-0.155.patch/libdw/Makefile.am	2013-03-05 18:12:26.753322780 +0100
@@ -112,7 +112,6 @@
 		-Wl,--version-script,$<,--no-undefined \
 		-Wl,--whole-archive $(filter-out $<,$^) -Wl,--no-whole-archive\
 		-ldl $(zip_LIBS)
-	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 	ln -fs $@ $@.$(VERSION)
 
 install: install-am libdw.so
diff -Naur elfutils-0.155/libdw/Makefile.in elfutils-0.155.patch/libdw/Makefile.in
--- elfutils-0.155/libdw/Makefile.in	2012-08-27 20:29:45.000000000 +0200
+++ elfutils-0.155.patch/libdw/Makefile.in	2013-03-05 18:12:41.272246520 +0100
@@ -844,7 +844,6 @@
 @MUDFLAP_FALSE@		-Wl,--version-script,$<,--no-undefined \
 @MUDFLAP_FALSE@		-Wl,--whole-archive $(filter-out $<,$^) -Wl,--no-whole-archive\
 @MUDFLAP_FALSE@		-ldl $(zip_LIBS)
-@MUDFLAP_FALSE@	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 @MUDFLAP_FALSE@	ln -fs $@ $@.$(VERSION)
 
 @MUDFLAP_FALSE@install: install-am libdw.so
diff -Naur elfutils-0.155/libelf/Makefile.am elfutils-0.155.patch/libelf/Makefile.am
--- elfutils-0.155/libelf/Makefile.am	2012-08-27 20:27:31.000000000 +0200
+++ elfutils-0.155.patch/libelf/Makefile.am	2013-03-05 18:13:23.328024851 +0100
@@ -106,7 +106,6 @@
 	$(LINK) -shared -o $@ -Wl,--whole-archive,$<,--no-whole-archive \
 		-Wl,--version-script,$(srcdir)/libelf.map,--no-undefined \
 		-Wl,--soname,$@.$(VERSION),-z,defs,-z,relro $(libelf_so_LDLIBS)
-	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 	ln -fs $@ $@.$(VERSION)
 
 install: install-am libelf.so
diff -Naur elfutils-0.155/libelf/Makefile.in elfutils-0.155.patch/libelf/Makefile.in
--- elfutils-0.155/libelf/Makefile.in	2012-08-27 20:29:45.000000000 +0200
+++ elfutils-0.155.patch/libelf/Makefile.in	2013-03-05 18:13:37.270951393 +0100
@@ -830,7 +830,6 @@
 @MUDFLAP_FALSE@	$(LINK) -shared -o $@ -Wl,--whole-archive,$<,--no-whole-archive \
 @MUDFLAP_FALSE@		-Wl,--version-script,$(srcdir)/libelf.map,--no-undefined \
 @MUDFLAP_FALSE@		-Wl,--soname,$@.$(VERSION),-z,defs,-z,relro $(libelf_so_LDLIBS)
-@MUDFLAP_FALSE@	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 @MUDFLAP_FALSE@	ln -fs $@ $@.$(VERSION)
 
 @MUDFLAP_FALSE@install: install-am libelf.so
diff -Naur elfutils-0.155/src/Makefile.in elfutils-0.155.patch/src/Makefile.in
--- elfutils-0.155/src/Makefile.in	2012-08-27 20:29:46.000000000 +0200
+++ elfutils-0.155.patch/src/Makefile.in	2013-03-05 18:14:22.361713943 +0100
@@ -867,7 +867,6 @@
 @NATIVE_LD_FALSE@	$(LINK) -shared -o $@ -Wl,--whole-archive,$<,--no-whole-archive \
 @NATIVE_LD_FALSE@		$(libelf) $(libeu) \
 @NATIVE_LD_FALSE@		-Wl,--version-script,$(srcdir)/libld_elf_i386.map
-@NATIVE_LD_FALSE@	$(textrel_check)
 
 # Special rule to make it possible to define libld_elf_a_SOURCES as we do.
 # Otherwise make would complain.
