--- elfutils-0.176/Makefile.am.orig	2019-04-01 12:59:25.141077178 +0200
+++ elfutils-0.176/Makefile.am	2019-04-01 12:59:34.157100775 +0200
@@ -28,7 +28,7 @@
 
 # Add doc back when we have some real content.
 SUBDIRS = config m4 lib libelf libebl libdwelf libdwfl libdw libcpu libasm \
-	  backends src po tests
+	  backends po tests
 
 EXTRA_DIST = elfutils.spec GPG-KEY NOTES CONTRIBUTING \
 	     COPYING COPYING-GPLV2 COPYING-LGPLV3
