--- a/arch/arm/boot/dts/bcm2708-rpi.dtsi	2017-11-18 04:56:28.149405247 +0000
+++ b/arch/arm/boot/dts/bcm2708-rpi.dtsi	2017-11-18 05:35:40.265159491 +0000
@@ -21,7 +21,7 @@
 		gpio = &gpio;
 		uart0 = &uart0;
 		sdhost = &sdhost;
-		mmc0 = &sdhost;
+		mmc1 = &sdhost;
 		i2s = &i2s;
 		spi0 = &spi0;
 		i2c0 = &i2c0;
@@ -29,7 +29,7 @@
 		spi1 = &spi1;
 		spi2 = &spi2;
 		mmc = &mmc;
-		mmc1 = &mmc;
+		mmc0 = &mmc;
 		i2c1 = &i2c1;
 		i2c2 = &i2c2;
 		usb = &usb;
