Index: lirc-0.8.7pre2/remotes/atiusb/lircd.conf.atiusb
===================================================================
--- lirc-0.8.7pre2.orig/remotes/atiusb/lircd.conf.atiusb
+++ lirc-0.8.7pre2/remotes/atiusb/lircd.conf.atiusb
@@ -1865,7 +1865,7 @@ end remote
 
 begin remote
 
-  name  Medion_X10
+  name  Medion_X10_V2
   bits           16
   eps            30
   aeps          100
@@ -1943,7 +1943,7 @@ end remote
 
 begin remote
 
-  name  Medion_X10
+  name  Medion_X10_V3
   bits           16
   eps            30
   aeps          100
Index: lirc-0.8.7pre2/remotes/bestbuy/lircd.conf.bestbuy2
===================================================================
--- lirc-0.8.7pre2.orig/remotes/bestbuy/lircd.conf.bestbuy2
+++ lirc-0.8.7pre2/remotes/bestbuy/lircd.conf.bestbuy2
@@ -11,7 +11,7 @@
 
 begin remote
 
-  name  BESTBUY
+  name  BESTBUY2
   bits            7
   eps            30
   aeps          100
Index: lirc-0.8.7pre2/remotes/dvico/lircd.conf.fusionHDTV
===================================================================
--- lirc-0.8.7pre2.orig/remotes/dvico/lircd.conf.fusionHDTV
+++ lirc-0.8.7pre2/remotes/dvico/lircd.conf.fusionHDTV
@@ -153,7 +153,7 @@ end remote
 
 begin remote
 
-  name  DVICO_MCE
+  name  DVICO_MCE2
   bits            8
   eps            30
   aeps          100
Index: lirc-0.8.7pre2/remotes/generic/NEC-pulse.conf
===================================================================
--- lirc-0.8.7pre2.orig/remotes/generic/NEC-pulse.conf
+++ lirc-0.8.7pre2/remotes/generic/NEC-pulse.conf
@@ -7,7 +7,7 @@
 
 begin remote
 
-  name  NEC
+  name  NEC_PULSE
   bits           16
   flags SPACE_ENC|CONST_LENGTH
   eps            30
Index: lirc-0.8.7pre2/remotes/generic/NEC-short-pulse.conf
===================================================================
--- lirc-0.8.7pre2.orig/remotes/generic/NEC-short-pulse.conf
+++ lirc-0.8.7pre2/remotes/generic/NEC-short-pulse.conf
@@ -9,7 +9,7 @@
 
 begin remote
 
-  name  NEC
+  name  NEC_SHORT_PULSE
   bits           16
   flags SPACE_ENC|CONST_LENGTH
   eps            30
Index: lirc-0.8.7pre2/remotes/generic/SONY12.conf
===================================================================
--- lirc-0.8.7pre2.orig/remotes/generic/SONY12.conf
+++ lirc-0.8.7pre2/remotes/generic/SONY12.conf
@@ -6,7 +6,7 @@
 
 begin remote
 
-  name  SONY
+  name  SONY_12
   bits            7
   flags SPACE_ENC|CONST_LENGTH
   eps            30
Index: lirc-0.8.7pre2/remotes/generic/SONY20.conf
===================================================================
--- lirc-0.8.7pre2.orig/remotes/generic/SONY20.conf
+++ lirc-0.8.7pre2/remotes/generic/SONY20.conf
@@ -6,7 +6,7 @@
 
 begin remote
 
-  name  SONY
+  name  SONY_20
   bits            7
   flags SPACE_ENC|CONST_LENGTH
   eps            30
Index: lirc-0.8.7pre2/remotes/hauppauge/lircd.conf.hauppauge
===================================================================
--- lirc-0.8.7pre2.orig/remotes/hauppauge/lircd.conf.hauppauge
+++ lirc-0.8.7pre2/remotes/hauppauge/lircd.conf.hauppauge
@@ -278,14 +278,14 @@ end remote
 #
 # contributed by Matthew Wright
 #
-# brand:  Hauppauge (HRV-1600 RT Remote)
+# brand:  Hauppauge (HVR-1600 RT Remote)
 # model no. of remote control: A415-HPG-A
 # devices being controlled by this remote:
 #
 
 begin remote
 
-  name  Hauppauge
+  name  Hauppauge_HVR_1600
   bits           13
   flags RC5|CONST_LENGTH
   eps            30
Index: lirc-0.8.7pre2/remotes/leadtek/lircd.conf.PVR2000
===================================================================
--- lirc-0.8.7pre2.orig/remotes/leadtek/lircd.conf.PVR2000
+++ lirc-0.8.7pre2/remotes/leadtek/lircd.conf.PVR2000
@@ -74,7 +74,7 @@ end remote
 
 begin remote
 
-  name  PVR2000
+  name  PVR2000_V2
   bits           8
   flags  SHIFT_ENC
   gap       120000
