From 4022c521c26e339152c6cc8febd6d737725de8ee Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Wed, 22 Jan 2014 15:59:14 +0200
Subject: [PATCH] meh alsa...

---
 configure.ac |   25 -------------------------
 1 files changed, 0 insertions(+), 25 deletions(-)

diff --git a/configure.ac b/configure.ac
index 73a353f..74ccba8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -342,31 +342,6 @@ AC_CHECK_HEADERS(portaudio.h,[
     ],,${portaudio_lib_other}
   )]
 )
-dnl audio_alsa driver requires ALSA library installed and some linker flags
-have_alsa=no
-AC_CHECK_HEADERS(alsa/asoundlib.h,[
-  AC_CHECK_LIB(asound, snd_pcm_open,[
-    AC_DEFINE(HAVE_LIBALSA)
-    have_alsa=yes
-    possible_drivers="${possible_drivers} (audio_alsa)"
-    ],,
-  )]
-)
-dnl alsa_usb driver requires an ALSA that knows about our hwdep device
-if test "$have_alsa" = "yes"; then
-  AC_MSG_CHECKING(for ALSA SB RC hwdep support)
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#include <alsa/asoundlib.h>
-]], [[
-  if (snd_hwdep_info_get_iface(NULL) == SND_HWDEP_IFACE_SB_RC) {}
-]])],[
-    AC_MSG_RESULT(yes)
-    AC_DEFINE(HAVE_ALSA_SB_RC)
-    possible_drivers="${possible_drivers} (alsa_usb)"
-    ],[
-    AC_MSG_RESULT(no)
-  ])
-fi
 dnl creative_infracd driver requires linux scsi api
 AC_CHECK_HEADERS(scsi/sg.h,[
   AC_DEFINE(HAVE_SCSI)
-- 
1.7.2.5

