diff -Naur lirc-0.9.0-old/configure.ac lirc-0.9.0-new/configure.ac
--- lirc-0.9.0-old/configure.ac	2012-02-29 21:54:01.000000000 -0800
+++ lirc-0.9.0-new/configure.ac	2012-02-29 21:55:06.000000000 -0800
@@ -164,7 +164,6 @@
 	(lirc_dev lirc_atiusb) \
 	(lirc_dev lirc_bt829) \
 	(lirc_dev lirc_gpio) \
-	(lirc_dev lirc_i2c) \
 	(lirc_dev lirc_igorplugusb) \
 	(lirc_dev lirc_ttusbir) \
 	(lirc_dev lirc_imon) \
@@ -420,32 +419,31 @@
 AC_ARG_WITH(driver,
 [  --with-driver=X         specify the driver to use, where X is one of:
                           all, none, userspace, accent, act200l,
-                          act220l, adaptec, alsa_usb, animax, asusdh,
+                          act220l, alsa_usb, animax, asusdh,
                           atilibusb, atiusb, atwf83, audio, audio_alsa
                           avermedia, avermedia_vdomate, avermedia98,
-                          awlibusb, bestbuy, bestbuy2, breakoutbox,
+                          awlibusb, bestbuy, bestbuy2,
                           bte, bw6130, caraca, chronos, commandir,
                           comX, cph06x, creative, creative_infracd,
                           devinput, dfclibusb, digimatrix, dsp, dvico, ea65,
                           exaudio, flyvideo, ftdi, gvbctv5pci,
-                          hauppauge, hauppauge_dvb,
-                          hercules_smarttv_stereo, i2cuser,
+                          i2cuser,
                           igorplugusb, iguanaIR, imon, imon_24g,
                           imon_knob, imon_lcd, imon_pad, imon_rsc,
                           irdeo, irdeo_remote, irlink, irman, irreal,
-                          knc_one, kworld,
-                          leadtek_0007, leadtek_0010, leadtek_pvr2000,
+                          kworld,
+                          leadtek_0007, leadtek_0010,
                           livedrive_midi, livedrive_seq, logitech,
                           lptX, macmini, mediafocusI,
                           mouseremote, mouseremote_ps2, mp3anywhere,
                           mplay, nslu2,packard_bell, parallel, pcmak,
-                          pcmak_usb, pctv, pixelview_bt878,
-                          pixelview_pak, pixelview_pro, provideo,
+                          pcmak_usb, pctv,
+                          pixelview_pak, pixelview_pro,
                           realmagic, remotemaster, sa1100, samsung,
                           sasem, sb0540, serial, silitek, sir, slinke,
                           srm7500libusb, tekram,
                           tekram_bt829, tira, tira_raw, ttusbir,
-                          tuxbox, tvbox, udp, uirt2, uirt2_raw,
+                          tuxbox, udp, uirt2, uirt2_raw,
                           usb_uirt_raw, usbx, wpc8769l],
 driver=${withval},
 driver="unset"
@@ -592,9 +590,6 @@
 	lirc_dev-lirc_gpio)
 		#lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
 		;;
-	lirc_dev-lirc_i2c)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
-		;;
 	lirc_dev-lirc_igorplugusb)
 		;;
 	lirc_dev-lirc_ttusbir)
@@ -721,12 +716,6 @@
   AC_DEFINE(LIRC_SIR_ACTISYS_ACT220L)
 fi
     
-if test "$driver" = "adaptec"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="adaptec/lircd.conf.AVC-2410"
-#  lircmd_conf="adaptec/lircmd.conf.AVC-2410"
-fi
-
 if test "$driver" = "alsa_usb"; then
   lirc_driver="alsa_usb"
   hw_module="hw_alsa_usb.o"
@@ -828,11 +817,6 @@
   lircmd_conf="bestbuy/lircmd.conf.bestbuy2"
 fi
 
-if test "$driver" = "breakoutbox"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="creative/lircd.conf.breakoutbox"
-fi
-
 if test "$driver" = "bte"; then
   lirc_driver="none"
   hw_module="hw_bte.o  serial.o"
@@ -983,23 +967,6 @@
 #  lircmd_conf="iodata/lircmd.conf.gvbctv5pci"
 fi
 
-if test "$driver" = "hauppauge"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="hauppauge/lircd.conf.hauppauge"
-  lircmd_conf="hauppauge/lircmd.conf.hauppauge"
-fi
-
-if test "$driver" = "hauppauge_dvb"; then
-  lirc_driver="lirc_dev"
-  lircd_conf="hauppauge/lircd.conf.hauppauge"
-  lircmd_conf="hauppauge/lircmd.conf.hauppauge"
-fi
-
-if test "$driver" = "hercules_smarttv_stereo"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="hercules/lircd.conf.smarttv_stereo"
-fi
-
 if test "$driver" = "i2cuser"; then
   lirc_driver="none"
   hw_module="hw_i2cuser.o receive.o"
@@ -1084,12 +1051,6 @@
   HW_DEFAULT="hw_irreal"
 fi
 
-if test "$driver" = "knc_one"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="knc_one/lircd.conf.knc_one"
-#  lircmd_conf="knc_one/lircmd.conf.knc_one"
-fi
-
 if test "$driver" = "kworld"; then
   lirc_driver="lirc_dev lirc_gpio"
   lircd_conf="kworld/lircd.conf.kworld"
@@ -1108,12 +1069,6 @@
   lircmd_conf="leadtek/lircmd.conf.RM-0010"
 fi
 
-if test "$driver" = "leadtek_pvr2000"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="leadtek/lircd.conf.PVR2000"
-#  lircmd_conf="leadtek/lircmd.conf.PVR2000"
-fi
-
 if test "$driver" = "livedrive_midi"; then
   lirc_driver="none"
   hw_module="hw_livedrive_common.o hw_livedrive_midi.o"
@@ -1230,12 +1185,6 @@
 #  lircmd_conf="pinnacle_systems/lircmd.conf.pctv"
 fi
 
-if test "$driver" = "pixelview_bt878"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="pixelview/lircd.conf.playtv_bt878"
-#  lircmd_conf="pixelview/lircmd.conf.pixelview_bt878"
-fi
-
 if test "$driver" = "pixelview_pak"; then
   lirc_driver="lirc_dev lirc_gpio"
 # the config files are the same as for the PlayTV pro
@@ -1249,12 +1198,6 @@
   lircmd_conf="pixelview/lircmd.conf.playtv_pro"
 fi
 
-if test "$driver" = "provideo"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="provideo/lircd.conf.pv951"
-#  lircmd_conf="provideo/lircmd.conf.pv951"
-fi
-
 if test "$driver" = "realmagic"; then
   lirc_driver="none"
   hw_module="hw_pixelview.o serial.o"
@@ -1372,11 +1315,6 @@
   HW_DEFAULT="hw_default"
 fi
 
-if test "$driver" = "tvbox"; then
-  lirc_driver="lirc_dev lirc_i2c"
-  lircd_conf="asus/lircd.conf.asus"
-fi
-
 if test "$driver" = "udp"; then
   lirc_driver="none"
   hw_module="hw_udp.o receive.o"
@@ -1512,7 +1450,6 @@
   lirc_driver="lirc_dev \
 	lirc_atiusb \
 	lirc_bt829 \
-	lirc_i2c \
 	lirc_igorplugusb \
 	lirc_imon \
 	lirc_parallel \
@@ -1779,7 +1716,6 @@
   drivers/lirc_bt829/Makefile
   drivers/lirc_dev/Makefile
   drivers/lirc_gpio/Makefile
-  drivers/lirc_i2c/Makefile
   drivers/lirc_igorplugusb/Makefile
   drivers/lirc_ttusbir/Makefile
   drivers/lirc_imon/Makefile
@@ -1808,9 +1744,6 @@
 elif test "$driver" = "alsa_usb"; then
   echo "You will have to use the snd-usb-audio driver from:"
   echo "http://www.alsa-project.org/"
-elif test "$driver" = "hauppauge_dvb"; then
-  echo "You will have to use the DVB driver from:"
-  echo "http://www.linuxtv.org"
 elif test "$kernel_module" = ""; then
   echo "Your hardware does not require a special LIRC kernel module."
   if test "${irtty}" != ""; then
