Index: lirc-0.8.7pre2/remotes/atiusb/lircd.conf.atiusb
===================================================================
--- lirc-0.8.7pre2.orig/remotes/atiusb/lircd.conf.atiusb
+++ lirc-0.8.7pre2/remotes/atiusb/lircd.conf.atiusb
@@ -2195,7 +2195,7 @@ end remote
 # this config file was automatically generated
 # using lirc-0.8.2(default) on Wed Feb  6 01:32:32 2008
 #
-# contributed by 
+# contributed by http://www.mythtv.org/wiki/Snapstream_Firefly
 #
 # brand:                       Snapstream Firefly
 # model no. of remote control: R1000
@@ -2215,63 +2215,111 @@ begin remote
   pre_data       0x14
   post_data_bits  16
   post_data      0x0
-  gap          235978
+  gap          147992
+  min_repeat      5
   toggle_bit_mask 0x80800000
 
       begin codes
-          maximize                 0x81AC
-          close                    0xD702
-          1                        0x628D
-          2                        0xE30E
-          3                        0x648F
-          4                        0xE510
-          5                        0x6691
-          6                        0xE712
-          7                        0x6893
-          8                        0xE914
-          9                        0x6A95
-          0                        0xEC17
-          back                     0x6B96
-          ent                      0xED18
-          volup                    0x5E89
-          voldown                  0xDD08
-          mute                     0x5F8A
-          chup                     0xE00B
-          chdown                   0x618C
-          firefly                  0xD500
-          info                     0x83AE
-          option                   0x042F
-          up                       0x6F9A
-          down                     0xF722
-          left                     0x729D
-          right                    0xF41F
-          ok                       0x739E
-          menu                     0xF11C
-          exit                     0x75A0
-          rec                      0xFC27
-          play                     0x7AA5
-          stop                     0xFD28
-          rew                      0x79A4
-          prev                     0x002B
-          pause                    0x7EA9
-          ffwd                     0xFB26
-          next                     0x7FAA
-          music                    0xDB06
-          photo                    0x5A85
-          dvd                      0xD904
-          tv                       0x5883
-          video                    0xDC07
-          help                     0x5681
-          mouse                    0x022D
-          a                        0x6E99
-          b                        0xF01B
-          c                        0x76A1
-          d                        0xF823
+	  KEY_MAX                  0x812C
+	  KEY_MAX                  0x01AC
+	  KEY_CLOSE                0x5702
+	  KEY_CLOSE                0xD782
+	  KEY_1                    0x620D
+	  KEY_1                    0xE28D
+	  KEY_2                    0x630E
+	  KEY_2                    0xE38E
+	  KEY_3                    0x640F
+	  KEY_3                    0xE48F
+	  KEY_4                    0x6510
+	  KEY_4                    0xE590
+	  KEY_5                    0x6611
+	  KEY_5                    0xE691
+	  KEY_6                    0x6712
+	  KEY_6                    0xE792
+	  KEY_7                    0x6813
+	  KEY_7                    0xE893
+	  KEY_8                    0x6914
+	  KEY_8                    0xE994
+	  KEY_9                    0x6A15
+	  KEY_9                    0xEA95
+	  KEY_0                    0x6C17
+	  KEY_0                    0xEC97
+	  KEY_BACK                 0x6B16
+	  KEY_BACK                 0xEB96
+	  KEY_ENTER                0x6D18
+	  KEY_ENTER                0xED98
+	  KEY_VOLUMEUP             0x5E09
+	  KEY_VOLUMEUP             0xDE89
+	  KEY_VOLUMEDOWN           0x5D08
+	  KEY_VOLUMEDOWN           0xDD88
+	  KEY_MUTE                 0x5F0A
+	  KEY_MUTE                 0xDF8A
+	  KEY_VENDOR               0x5500
+	  KEY_VENDOR               0xD580
+	  KEY_CHANNELUP            0x600B
+	  KEY_CHANNELUP            0xE08B
+	  KEY_CHANNELDOWN          0x610C
+	  KEY_CHANNELDOWN          0xE18C
+	  KEY_INFO                 0x832E
+	  KEY_INFO                 0x03AE
+	  KEY_OPTION               0x842F
+	  KEY_OPTION               0x04AF
+	  KEY_UP                   0x6F1A
+	  KEY_UP                   0xEF9A
+	  KEY_LEFT                 0x721D
+	  KEY_LEFT                 0xF29D
+	  KEY_DOWN                 0x7722
+	  KEY_DOWN                 0xF7A2
+	  KEY_RIGHT                0x741F
+	  KEY_RIGHT                0xF49F
+	  KEY_OK                   0x731E
+	  KEY_OK                   0xF39E
+	  KEY_MENU                 0x711C
+	  KEY_MENU                 0xF19C
+	  KEY_EXIT                 0x7520
+	  KEY_EXIT                 0xF5A0
+	  KEY_RECORD               0x7C27
+	  KEY_RECORD               0xFCA7
+	  KEY_PLAY                 0x7A25
+	  KEY_PLAY                 0xFAA5
+	  KEY_STOP                 0x7D28
+	  KEY_STOP                 0xFDA8
+	  KEY_REWIND               0x7924
+	  KEY_REWIND               0xF9A4
+	  KEY_FORWARD              0x7B26
+	  KEY_FORWARD              0xFBA6
+	  KEY_PREVIOUS             0x802B
+	  KEY_PREVIOUS             0x00AB
+	  KEY_PAUSE                0x7E29
+	  KEY_PAUSE                0xFEA9
+	  KEY_NEXT                 0x7F2A
+	  KEY_NEXT                 0xFFAA
+	  KEY_AUDIO                0x5B06
+	  KEY_AUDIO                0xDB86
+	  PHOTOS                   0x5A05
+	  PHOTOS                   0xDA85
+	  KEY_DVD                  0x5904
+	  KEY_DVD                  0xD984
+	  KEY_TV                   0x5803
+	  KEY_TV                   0xD883
+	  KEY_VIDEO                0x5C07
+	  KEY_VIDEO                0xDC87
+	  KEY_HELP                 0x5601
+	  KEY_HELP                 0xD681
+	  MOUSE                    0x822D
+	  MOUSE                    0x02AD
+	  KEY_A                    0x6E19
+	  KEY_A                    0xEE99
+	  KEY_B                    0x701B
+	  KEY_B                    0xF09B
+	  KEY_C                    0x7621
+	  KEY_C                    0xF6A1
+	  KEY_D                    0x7823
+	  KEY_D                    0xF8A3
       end codes
 
 end remote
 
-
 #
 # this config file was automatically generated
 # using lirc-0.8.2(default) on Thu Oct  4 17:42:33 2007
