diff -Naur lirc-0.9.0-old-old/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.9.0-old-new/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.9.0-old-old/drivers/lirc_atiusb/lirc_atiusb.c	2012-05-01 12:09:19.000000000 -0700
+++ lirc-0.9.0-old-new/drivers/lirc_atiusb/lirc_atiusb.c	2012-05-01 12:14:52.000000000 -0700
@@ -138,23 +138,23 @@
 
 static struct usb_device_id usb_remote_table[] = {
 	/* X10 USB Firecracker Interface */
-	{ USB_DEVICE(VENDOR_ATI1, 0x0002) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0002) }, */
 
 	/* X10 VGA Video Sender */
-	{ USB_DEVICE(VENDOR_ATI1, 0x0003) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0003) }, */
 
 	/* ATI Wireless Remote Receiver */
-	{ USB_DEVICE(VENDOR_ATI1, 0x0004) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0004) }, */
 
 	/* NVIDIA Wireless Remote Receiver */
-	{ USB_DEVICE(VENDOR_ATI1, 0x0005) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0005) }, */
 
 	/* ATI Wireless Remote Receiver */
-	{ USB_DEVICE(VENDOR_ATI1, 0x0006) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0006) }, */
 
 	/* X10 USB Wireless Transceivers */
 	{ USB_DEVICE(VENDOR_ATI1, 0x0007) },
-	{ USB_DEVICE(VENDOR_ATI1, 0x0008) },
+	/* { USB_DEVICE(VENDOR_ATI1, 0x0008) }, */
 	{ USB_DEVICE(VENDOR_ATI1, 0x0009) },
 	{ USB_DEVICE(VENDOR_ATI1, 0x000A) },
 	{ USB_DEVICE(VENDOR_ATI1, 0x000B) },
