diff -Naur fontconfig-2.10.0-old/fontconfig.pc.in fontconfig-2.10.0-new/fontconfig.pc.in
--- fontconfig-2.10.0-old/fontconfig.pc.in	2012-05-31 19:26:10.000000000 -0700
+++ fontconfig-2.10.0-new/fontconfig.pc.in	2012-07-19 08:27:33.000000000 -0700
@@ -5,7 +5,7 @@
 sysconfdir=@sysconfdir@
 localstatedir=@localstatedir@
 PACKAGE=@PACKAGE@
-confdir=@confdir@
+confdir=@baseconfigdir@
 cachedir=@fc_cachedir@
 
 Name: Fontconfig
