diff -Naur libpcap-1.7.4/configure.in libpcap-1.7.4.patch/configure.in
--- libpcap-1.7.4/configure.in	2015-04-22 22:19:15.000000000 +0200
+++ libpcap-1.7.4.patch/configure.in	2016-03-26 10:36:25.193757606 +0100
@@ -456,7 +456,7 @@
 	if test x$with_libnl != xno ; then
 		have_any_nl="no"
 
-                incdir=-I/usr/include/libnl3
+#                incdir=-I/usr/include/libnl3
                 libnldir=
                 if test x$withval != x ; then
                   libnldir=-L${withval}/lib/.libs
