diff -Naur linux-b425b70a6379f3415ed349f5c5719d4a6f315523/arch/arm64/boot/dts/qcom/pm8998.dtsi linux-b425b70a6379f3415ed349f5c5719d4a6f315523-2/arch/arm64/boot/dts/qcom/pm8998.dtsi
--- linux-b425b70a6379f3415ed349f5c5719d4a6f315523/arch/arm64/boot/dts/qcom/pm8998.dtsi	2022-07-02 07:55:12.633536880 +0200
+++ linux-b425b70a6379f3415ed349f5c5719d4a6f315523-2/arch/arm64/boot/dts/qcom/pm8998.dtsi	2022-07-02 08:01:11.968113266 +0200
@@ -50,7 +50,7 @@
 				interrupts = <0x0 0x8 0 IRQ_TYPE_EDGE_BOTH>;
 				debounce = <15625>;
 				bias-pull-up;
-				linux,code = <KEY_POWER>;
+				linux,code = <KEY_SLEEP>;
 			};
 		};
 
diff -Naur linux-b425b70a6379f3415ed349f5c5719d4a6f315523/arch/arm64/boot/dts/qcom/sdm845-ayn-odin.dts linux-b425b70a6379f3415ed349f5c5719d4a6f315523-2/arch/arm64/boot/dts/qcom/sdm845-ayn-odin.dts
--- linux-b425b70a6379f3415ed349f5c5719d4a6f315523/arch/arm64/boot/dts/qcom/sdm845-ayn-odin.dts	2022-07-02 07:55:12.637536952 +0200
+++ linux-b425b70a6379f3415ed349f5c5719d4a6f315523-2/arch/arm64/boot/dts/qcom/sdm845-ayn-odin.dts	2022-07-02 08:00:35.659442976 +0200
@@ -37,7 +37,7 @@
 
 		vol-up {
 			label = "Volume Up";
-			linux,code = <KEY_VOLUMEUP>;
+			linux,code = <KEY_KPPLUS>;
 			gpios = <&pm8998_gpio 6 GPIO_ACTIVE_LOW>;
 		};
 	};
@@ -574,7 +574,7 @@
 		interrupts = <0x0 0x8 1 IRQ_TYPE_EDGE_BOTH>;
 		debounce = <15625>;
 		bias-pull-up;
-		linux,code = <KEY_VOLUMEDOWN>;
+		linux,code = <KEY_KPMINUS>;
 	};
 };
 
