diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2015-10-16 19:01:09.000000000 +0200
+++ b/Makefile.am	2016-01-06 21:22:34.118008794 +0100
@@ -1 +1,8 @@
+README: README.md
+        fgrep -v "[Build Status]" $< \
+                | markdown \
+                | html2text -style pretty -nobs \
+                | sed -e 's:&gt;:>:g' \
+                > $@.tmp
+
 SUBDIRS = src examples
diff -Naur a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2015-10-16 19:01:09.000000000 +0200
+++ b/src/Makefile.am	2016-01-06 21:26:12.352336266 +0100
@@ -49,9 +49,9 @@
 iperf3_profile_SOURCES  = main.c \
 						  $(libiperf_la_SOURCES)
 
-iperf3_profile_CFLAGS   = -pg -g
+iperf3_profile_CFLAGS   = -g
 iperf3_profile_LDADD    = libiperf.la
-iperf3_profile_LDFLAGS  = -pg -g
+iperf3_profile_LDFLAGS  = -g
 
 # Specify the sources and various flags for the test cases
 t_timer_SOURCES         = t_timer.c
