diff -Naur busybox-1.13.2/init/halt.c busybox-1.13.2a/init/halt.c
--- busybox-1.13.2/init/halt.c	2008-11-09 18:28:19.000000000 +0100
+++ busybox-1.13.2a/init/halt.c	2009-02-01 16:38:37.000000000 +0100
@@ -79,7 +79,7 @@
 
 	/* Perform action. */
 	rc = 1;
-	if (!(flags & 4)) { /* no -f */
+	if (ENABLE_INIT && !(flags & 4)) { /* no -f */
 //TODO: I tend to think that signalling linuxrc is wrong
 // pity original author didn't comment on it...
 		if (ENABLE_FEATURE_INITRD) {
