diff -Naur a/loginutils/passwd.c b/loginutils/passwd.c
--- a/loginutils/passwd.c	2017-07-06 08:14:57.000000000 -0700
+++ b/loginutils/passwd.c	2017-09-11 17:06:07.572805135 -0700
@@ -220,7 +220,7 @@
 	if (rc > 0)
 		/* password in /etc/shadow was updated */
 		newp = (char*) "x";
-	if (rc >= 0)
+	if (rc == 0)
 		/* 0 = /etc/shadow missing (not an error), >0 = passwd changed in /etc/shadow */
 #endif
 	{
