diff -Naur scummvm-0.9.1.orig/configure scummvm-0.9.1/configure
--- scummvm-0.9.1.orig/configure	2007-01-13 13:45:19.000000000 +0100
+++ scummvm-0.9.1/configure	2007-01-13 13:45:42.000000000 +0100
@@ -753,6 +753,23 @@
 if test -n "$_host"; then
 	# Cross-compiling mode - add your target here if needed
 	case "$_host" in
+		geexbox-i386)
+			echo "Cross-compiling to $_host, forcing endianness and type sizes"
+			DEFINES="$DEFINES -DUNIX"
+			_def_endianness='#define SCUMM_LITTLE_ENDIAN'
+			type_1_byte='char'
+			type_2_byte='short'
+			type_4_byte='int'
+			;;
+		geexbox-powerpc)
+			echo "Cross-compiling to $_host, forcing endianness, alignment and type sizes"
+			DEFINES="$DEFINES -DUNIX"
+			_def_endianness='#define SCUMM_BIG_ENDIAN'
+			_def_align='#define SCUMM_NEED_ALIGNMENT'
+			type_1_byte='char'
+			type_2_byte='short'
+			type_4_byte='int'
+			;;
 		linupy|arm-riscos)
 			echo "Cross-compiling to $_host, forcing endianness, alignment and type sizes"
 			DEFINES="$DEFINES -DUNIX"
