diff -Naur xbmc-audioencoder-lame-736b5f4/CMakeLists.txt xbmc-audioencoder-lame-736b5f4.patch/CMakeLists.txt
--- xbmc-audioencoder-lame-736b5f4/CMakeLists.txt	2014-07-18 23:48:35.000000000 +0200
+++ xbmc-audioencoder-lame-736b5f4.patch/CMakeLists.txt	2014-10-20 13:06:32.787416257 +0200
@@ -6,7 +6,7 @@
 
 enable_language(CXX)
 
-find_package(xbmc REQUIRED)
+find_package(kodi REQUIRED)
 find_package(Lame REQUIRED)
 if(WIN32)
   find_package(MPGHIP REQUIRED)
diff -Naur xbmc-audioencoder-lame-736b5f4/src/EncoderLame.cpp xbmc-audioencoder-lame-736b5f4.patch/src/EncoderLame.cpp
--- xbmc-audioencoder-lame-736b5f4/src/EncoderLame.cpp	2014-07-18 23:48:35.000000000 +0200
+++ xbmc-audioencoder-lame-736b5f4.patch/src/EncoderLame.cpp	2014-10-20 02:51:53.710444163 +0200
@@ -19,7 +19,7 @@
  */
 
 #include <lame/lame.h>
-#include "xbmc/xbmc_audioenc_dll.h"
+#include "kodi/xbmc_audioenc_dll.h"
 #include <string.h>
 #include <stdlib.h>
 #include <algorithm>
