diff -Naur asix-ax887xx-4.1.0/Makefile asix-ax887xx-4.1.0.patch/Makefile
--- asix-ax887xx-4.1.0/Makefile	2011-02-22 01:54:38.000000000 +0100
+++ asix-ax887xx-4.1.0.patch/Makefile	2011-06-14 03:58:46.714369862 +0200
@@ -2,7 +2,7 @@
 TARGET	= asix
 OBJS	= asix.o
 MDIR	= drivers/net/usb
-KDIR	= /lib/modules/$(CURRENT)/build
+KDIR	:= /lib/modules/$(CURRENT)/build
 SUBLEVEL= $(shell uname -r | cut -d '.' -f 3 | cut -d '.' -f 1 | cut -d '-' -f 1 | cut -d '_' -f 1)
 
 ifneq (,$(filter $(SUBLEVEL),14 15 16 17 18 19 20 21))
