diff -Naur asix-ax887xx-4.2.0/asix.c asix-ax887xx-4.2.0.patch/asix.c
--- asix-ax887xx-4.2.0/asix.c	2011-09-19 14:48:58.000000000 +0200
+++ asix-ax887xx-4.2.0.patch/asix.c	2011-12-20 03:39:27.929024533 +0100
@@ -837,7 +837,7 @@
 	.ndo_do_ioctl		= ax8817x_ioctl,
 	.ndo_set_mac_address		= ax8817x_set_mac_addr,
 	.ndo_validate_addr		= eth_validate_addr,
-	.ndo_set_multicast_list	= ax8817x_set_multicast,
+	.ndo_set_rx_mode	= ax8817x_set_multicast,
 };
 #endif
 
@@ -1530,7 +1530,7 @@
 	.ndo_get_stats		= axusbnet_get_stats,
 	.ndo_set_mac_address 	= ax8817x_set_mac_addr,
 	.ndo_validate_addr		= eth_validate_addr,
-	.ndo_set_multicast_list = ax88772b_set_multicast,
+	.ndo_set_rx_mode = ax88772b_set_multicast,
 };
 #endif
 
