diff -ur alsa.old/src/Makefile.in alsa.dev/src/Makefile.in
--- alsa.old/src/Makefile.in	2006-04-19 09:50:27.000000000 +0200
+++ alsa.dev/src/Makefile.in	2006-10-14 17:16:10.000000000 +0200
@@ -360,7 +360,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libasound.la: $(libasound_la_OBJECTS) $(libasound_la_DEPENDENCIES) 
-	$(LINK) -rpath $(libdir) $(libasound_la_LDFLAGS) $(libasound_la_OBJECTS) $(libasound_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(DESTDIR)$(libdir) $(libasound_la_LDFLAGS) $(libasound_la_OBJECTS) $(libasound_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
diff -ur alsa.old/src/pcm/scopes/Makefile.in alsa.dev/src/pcm/scopes/Makefile.in
--- alsa.old/src/pcm/scopes/Makefile.in	2006-04-19 09:50:38.000000000 +0200
+++ alsa.dev/src/pcm/scopes/Makefile.in	2006-10-14 17:16:38.000000000 +0200
@@ -317,7 +317,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 scope-level.la: $(scope_level_la_OBJECTS) $(scope_level_la_DEPENDENCIES) 
-	$(LINK) -rpath $(pkglibdir) $(scope_level_la_LDFLAGS) $(scope_level_la_OBJECTS) $(scope_level_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(DESTDIR)$(pkglibdir) $(scope_level_la_LDFLAGS) $(scope_level_la_OBJECTS) $(scope_level_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
