diff -Naur alsa-lib-1.0.4.orig/include/error.h alsa-lib-1.0.4/include/error.h
--- alsa-lib-1.0.4.orig/include/error.h	2004-04-27 00:16:40.000000000 +0200
+++ alsa-lib-1.0.4/include/error.h	2004-04-27 00:19:31.000000000 +0200
@@ -61,11 +61,11 @@
 extern int snd_lib_error_set_handler(snd_lib_error_handler_t handler);
 
 #if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ > 95)
-#define SNDERR(...) snd_lib_error(__FILE__, __LINE__, __FUNCTION__, 0, __VA_ARGS__) /**< Shows a sound error message. */
-#define SYSERR(...) snd_lib_error(__FILE__, __LINE__, __FUNCTION__, errno, __VA_ARGS__) /**< Shows a system error message (related to \c errno). */
+#define SNDERR(...)
+#define SYSERR(...)
 #else
-#define SNDERR(args...) snd_lib_error(__FILE__, __LINE__, __FUNCTION__, 0, ##args) /**< Shows a sound error message. */
-#define SYSERR(args...) snd_lib_error(__FILE__, __LINE__, __FUNCTION__, errno, ##args) /**< Shows a system error message (related to \c errno). */
+#define SNDERR(args...)
+#define SYSERR(args...)
 #endif
 
 /** \} */
