diff --git a/src/conf/cards/aliases.conf b/src/conf/cards/aliases.conf
--- a/src/conf/cards/aliases.conf 2016-07-20 20:57:09.510009951 +0200
+++ b/src/conf/cards/aliases.conf 2016-07-20 20:57:41.757691480 +0200
@@ -55,6 +55,8 @@
 AV200 cards.CMI8788
 CMI8786 cards.CMI8788
 CMI8787 cards.CMI8788
+'On-board_SPDIF' cards.imx-spdif
+'Integrated_SPDI' cards.imx-spdif

 <confdir:pcm/default.conf>
 <confdir:pcm/dmix.conf>
