diff -rupN a/src/conf/cards/HDA-Intel.conf b/src/conf/cards/HDA-Intel.conf
--- a/src/conf/cards/HDA-Intel.conf	2012-09-06 10:55:14.000000000 +0200
+++ b/src/conf/cards/HDA-Intel.conf	2012-11-24 14:25:41.457981812 +0100
@@ -211,7 +211,7 @@ HDA-Intel.pcm.hdmi.0 {
 			"cards.HDA-Intel.pcm.hdmi.common:"
 			"CARD=" $CARD ","
 			"DEVICE=3,"
-			"CTLINDEX=0,"
+			"CTLINDEX=1,"
 			"AES0=" $AES0 ","
 			"AES1=" $AES1 ","
 			"AES2=" $AES2 ","
@@ -234,7 +234,7 @@ HDA-Intel.pcm.hdmi.1 {
 			"cards.HDA-Intel.pcm.hdmi.common:"
 			"CARD=" $CARD ","
 			"DEVICE=7,"
-			"CTLINDEX=1,"
+			"CTLINDEX=2,"
 			"AES0=" $AES0 ","
 			"AES1=" $AES1 ","
 			"AES2=" $AES2 ","
@@ -257,7 +257,7 @@ HDA-Intel.pcm.hdmi.2 {
 			"cards.HDA-Intel.pcm.hdmi.common:"
 			"CARD=" $CARD ","
 			"DEVICE=8,"
-			"CTLINDEX=2,"
+			"CTLINDEX=3,"
 			"AES0=" $AES0 ","
 			"AES1=" $AES1 ","
 			"AES2=" $AES2 ","
@@ -280,7 +280,7 @@ HDA-Intel.pcm.hdmi.3 {
 			"cards.HDA-Intel.pcm.hdmi.common:"
 			"CARD=" $CARD ","
 			"DEVICE=9,"
-			"CTLINDEX=3,"
+			"CTLINDEX=4,"
 			"AES0=" $AES0 ","
 			"AES1=" $AES1 ","
 			"AES2=" $AES2 ","
