diff -Naur alsa-lib-1.0.26-old/configure.in alsa-lib-1.0.26-new/configure.in
--- alsa-lib-1.0.26-old/configure.in	2012-09-06 01:55:14.000000000 -0700
+++ alsa-lib-1.0.26-new/configure.in	2012-12-29 06:07:44.000000000 -0800
@@ -53,7 +53,7 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AM_CONFIG_HEADER(include/config.h)
+AC_CONFIG_HEADERS(include/config.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
