From 982c628f484266c8db4f70dc1e856bc7844c1873 Mon Sep 17 00:00:00 2001
From: Patrick Rasmussen <patrickrasmussen1988@gmail.com>
Date: Fri, 13 Mar 2015 14:52:56 +0100
Subject: [PATCH] Add USB Sound Blaster HD

---
 src/conf/cards/USB-Audio.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/conf/cards/USB-Audio.conf b/src/conf/cards/USB-Audio.conf
index 4cbf85e..2591d00 100644
--- a/src/conf/cards/USB-Audio.conf
+++ b/src/conf/cards/USB-Audio.conf
@@ -62,6 +62,7 @@ USB-Audio.pcm.iec958_device {
 # Second iec958 device number, if any.
 USB-Audio.pcm.iec958_2_device {
 	"PHIREE U2" 1  # 0 = PCM S/PDIF, 1 = non-PCM S/PDIF
+	"USB Sound Blaster HD" 1
 }
 
 
-- 
2.1.4
