diff -Naur yenc-0.4.0/setup.py yenc-0.4.0.patch/setup.py
--- yenc-0.4.0/setup.py	2011-09-06 09:04:33.000000000 +0200
+++ yenc-0.4.0.patch/setup.py	2012-11-06 20:15:14.369125798 +0100
@@ -21,7 +21,7 @@
  # 
 ##=============================================================================
 
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 
 setup(	
 	name		 = "yenc",
