diff -Naur irserver-05.02.10/lowlevel.c irserver-05.02.10.patch/lowlevel.c
--- irserver-05.02.10/lowlevel.c	2010-02-05 01:14:44.000000000 +0100
+++ irserver-05.02.10.patch/lowlevel.c	2010-04-26 13:22:40.171936190 +0200
@@ -2638,10 +2638,17 @@
 #endif
 #ifdef LINUX
 		//LINUX Autofind USB devices
-		if ((!strncmp (st + q,"usb",3) || !strncmp (st + q,"USB",3)) && !autoflag) {
+		if( !strncmp( st, "/dev/irtrans", 12 ) && !autoflag ) {
+			autoflag = 1;
+			res = 0;
+			get_detail_deviceinfo ("",st,IF_USB);
+		}
+		else if ((!strncmp (st + q,"usb",3) || !strncmp (st + q,"USB",3)) && !autoflag) {
 			autoflag = 1;
 			res = 0;
 			for (i=0;i < 16;i++) {
+				sprintf (dst,"/dev/irtrans%d",i);
+				if (get_detail_deviceinfo ("",dst,IF_USB)) {
 				sprintf (dst,"/dev/ttyUSB%d",i);
 				if (get_detail_deviceinfo ("",dst,IF_USB)) {
 					sprintf (dst,"/dev/usb/ttyUSB%d",i);
@@ -2655,6 +2662,7 @@
 				}
 			}
 		}
+		}
 		else if ((!strncmp (st + q,"/dev/usb/tty",12) || !strncmp (st + q,"/dev/ttyUSB",11) || !strncmp (st + q,"/dev/usb/tts",12) ||
 				  !strncmp (st + q,"/dev/ttyusb",11) || !strncmp (st + q,"/dev/tts/USB",12)) && !autoflag) {
 			if (st[q + strlen (st+q) - 1] == ']') {
