From 902ee661f1864aaf8325621085f6a1b5a6a3673a Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 05 Dec 2011 21:24:48 +0000
Subject: test/radeon: add missing files for dist

---
diff --git a/tests/radeon/Makefile.am b/tests/radeon/Makefile.am
index 021ca72..1775669 100644
--- a/tests/radeon/Makefile.am
+++ b/tests/radeon/Makefile.am
@@ -9,4 +9,6 @@ noinst_PROGRAMS = \
 
 radeon_ttm_SOURCES = \
 	rbo.c \
+	rbo.h \
+	list.h \
 	radeon_ttm.c
--
cgit v0.9.0.2-2-gbebe
