From c8a437f4c76527b3c8385699ccee07f35fe3f166 Mon Sep 17 00:00:00 2001
From: Michel Dänzer <michel.daenzer@amd.com>
Date: Tue, 26 Nov 2013 09:16:03 +0000
Subject: radeon: Update unaligned offset for 2D->1D tiling transition on SI

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=71983

Tested-by: Arek Ruśniak <arek.rusi@gmail.com>
---
diff --git a/radeon/radeon_surface.c b/radeon/radeon_surface.c
index b528a48..dcbbfdc 100644
--- a/radeon/radeon_surface.c
+++ b/radeon/radeon_surface.c
@@ -1667,7 +1667,7 @@ static int si_surface_init_2d(struct radeon_surface_manager *surf_man,
             return si_surface_init_1d(surf_man, surf, level, bpe, tile_mode, offset, i);
         }
         /* level0 and first mipmap need to have alignment */
-        aligned_offset = surf->bo_size;
+        aligned_offset = offset = surf->bo_size;
         if ((i == 0)) {
             aligned_offset = ALIGN(aligned_offset, surf->bo_alignment);
         }
--
cgit v0.9.0.2-2-gbebe
