diff -Naur libdrm-2.4.40-old/include/drm/drm.h libdrm-2.4.40-new/include/drm/drm.h
--- libdrm-2.4.40-old/include/drm/drm.h	2012-07-29 06:39:21.000000000 -0700
+++ libdrm-2.4.40-new/include/drm/drm.h	2012-11-14 12:54:07.000000000 -0800
@@ -618,7 +618,11 @@
 	__u64 value;
 };
 
+#ifdef O_CLOEXEC
 #define DRM_CLOEXEC O_CLOEXEC
+#else
+#define DRM_CLOEXEC 0
+#endif
 struct drm_prime_handle {
 	__u32 handle;
 
