diff --git a/libdrm/nouveau/nouveau_bo.c b/libdrm/nouveau/nouveau_bo.c
index 023c6be..66466e3 100644
--- a/libdrm/nouveau/nouveau_bo.c
+++ b/libdrm/nouveau/nouveau_bo.c
@@ -404,7 +404,8 @@ nouveau_bo_handle_ref(struct nouveau_device *dev, uint32_t handle,
 		nvbo->size = req.size;
 		nvbo->handle = req.handle;
 	}
- 
+
+	nvbo->base.handle = nvbo->handle;
 	return 0;
 } 
 
diff --git a/libdrm/nouveau/nouveau_dma.c b/libdrm/nouveau/nouveau_dma.c
index 23da64b..b084f70 100644
--- a/libdrm/nouveau/nouveau_dma.c
+++ b/libdrm/nouveau/nouveau_dma.c
@@ -41,6 +41,7 @@ WRITE_PUT(struct nouveau_channel_priv *nvchan, uint32_t val)
 	volatile int dum;
 
 	NOUVEAU_DMA_BARRIER;
+	dum = nvchan->pushbuf[0];
 	dum = READ_GET(nvchan);
 
 	*nvchan->put = put;
