diff -Nur a/common/cmd_fat.c b/common/cmd_fat.c
--- a/common/cmd_fat.c	2015-03-03 20:29:10.000000000 +0000
+++ b/common/cmd_fat.c	2015-03-04 15:04:36.856261977 +0000
@@ -149,3 +149,74 @@
 	"      to 'dev' on 'interface'"
 );
 #endif
+
+int do_fat_bootini(cmd_tbl_t *cmdtp, int flag, int argc, char *argv[])
+{
+	unsigned long   filesize = 0;
+	unsigned char   *fp = (unsigned char *)0x5FFF0000, cmd[512], skip = 0, first = 1;
+	unsigned int    wpos = 0;
+
+	// file check & update
+	setenv("msgload", "Loading boot.ini from ");
+	setenv("filesize", "0");
+	run_command("if fatload mmc 0:1 5FFF0000 boot.ini;" \
+		"   then echo $msgload FAT;" \
+		"   fi", 0);
+
+	if((filesize = getenv_ulong("filesize", 16, 0))) {
+		if(filesize > 64 * 1024) {
+			printf("File Size Error! Max file size 64Kbytes. filesize = %d\n", filesize);
+			return  0;
+		}
+
+		while(1) {
+			if(*fp == '#')  skip = 1;
+			else {
+				skip = 0;   wpos = 0;
+				memset(cmd, 0x00, sizeof(cmd));
+			}
+
+			while(*fp != 0x0A) {
+				if((*fp != 0x0D) && (!skip))	cmd[wpos++] = *fp;
+
+				fp++;
+				if(filesize)	filesize--;
+				else		break;
+			}
+
+			if(wpos) {
+				if(wpos < sizeof(cmd)) {
+					if(first) {
+						if(!strncmp(cmd, "ODROID-XU3-UBOOT-CONFIG", sizeof("ODROID-XU3-UBOOT-CONFIG"))) {
+							printf("Find boot.ini file from FAT Area!!\n");	first = 0;
+						}
+						else {
+							printf("Find boot.ini file. But This file is not ODROID-XU3 config file!\n");
+							return  0;
+						}
+					}
+				else {
+					printf("boot.ini command = %s\n", cmd);
+					run_command(cmd, 0);
+				}
+			}
+			wpos = 0;
+		}
+		fp++;
+		if(filesize)	filesize--;
+		else		break;
+		}
+		return  1;
+	}
+	return  0;
+}
+
+U_BOOT_CMD(
+	bootini, 1, 0, do_fat_bootini,
+	"bootini - boot.ini textfile load from FAT32\n",
+	"<interface(only support mmc 0:1/0:2)>\n"
+	"	- boot.ini file load from FAT32 on 'interface'\n"
+	"Will attempt boot from: \n"
+	"	fat mmc 0:1 /boot.ini\n"
+        "on the above sequence\n"
+);

