diff -Naur glew-1.5.1/config/Makefile.linux glew-1.5.1.patch/config/Makefile.linux
--- glew-1.5.1/config/Makefile.linux	2008-11-03 07:25:20.000000000 +0100
+++ glew-1.5.1.patch/config/Makefile.linux	2009-10-13 19:24:17.305658392 +0200
@@ -1,19 +1,13 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
+CC = $(CC)
+LD = $(CC)
 ifneq (undefined, $(origin GLEW_MX))
 CFLAGS.EXTRA = -DGLEW_MX
 endif
 PICFLAG = -fPIC
 LDFLAGS.SO = -shared -Wl,-soname=$(LIB.SONAME)
 M_ARCH ?= $(shell arch)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
 LIBDIR = $(GLEW_DEST)/lib
-endif
 LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
