diff -Naur glew-1.5.2/config/Makefile.linux glew-1.5.2.patch/config/Makefile.linux
--- glew-1.5.2/config/Makefile.linux	2009-12-31 18:06:19.000000000 +0100
+++ glew-1.5.2.patch/config/Makefile.linux	2010-02-25 17:56:39.061037970 +0100
@@ -1,19 +1,13 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
+CC = $(CC)
+LD = $(CC)
 ifneq (undefined, $(origin GLEW_MX))
 CFLAGS.EXTRA = -DGLEW_MX
 endif
 PICFLAG = -fPIC
 LDFLAGS.SO = -shared -Wl,-soname=$(LIB.SONAME)
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
 LIBDIR = $(GLEW_DEST)/lib
-endif
 LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
