From ba254faae130080d5045d8fd97dc1f232b391d2f Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 16 Aug 2014 17:03:32 +0300
Subject: [PATCH] fix pkgconf

---
 glew.pc.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/glew.pc.in b/glew.pc.in
index 4c934af..35ef2ac 100644
--- a/glew.pc.in
+++ b/glew.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+includedir=${prefix}/include
+libdir=${exec_prefix}/lib
 
 Name: glew
 Description: The OpenGL Extension Wrangler library
-- 
1.7.2.5

