diff -Naur glew-1.7.0/config/Makefile.linux glew-1.7.0.patch/config/Makefile.linux
--- glew-1.7.0/config/Makefile.linux	2011-08-25 16:17:55.000000000 +0200
+++ glew-1.7.0.patch/config/Makefile.linux	2012-01-27 18:52:20.764675791 +0100
@@ -1,15 +1,7 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
 PICFLAG = -fPIC
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
 LIBDIR = $(GLEW_DEST)/lib
-endif
 LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
diff -Naur glew-1.7.0/Makefile glew-1.7.0.patch/Makefile
--- glew-1.7.0/Makefile	2012-01-27 18:50:02.004347735 +0100
+++ glew-1.7.0.patch/Makefile	2012-01-27 18:50:27.970848891 +0100
@@ -56,7 +56,7 @@
 
 AR = ar
 INSTALL = install
-STRIP = strip
+STRIP ?= strip
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
