--- glew-1.6.0/config/Makefile.linux.orig	2011-05-07 12:08:25.427908940 -0600
+++ glew-1.6.0/config/Makefile.linux	2011-05-07 12:08:50.345126195 -0600
@@ -1,15 +1,9 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
+CC = $(CC)
+LD = $(CC)
 PICFLAG = -fPIC
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
 LIBDIR = $(GLEW_DEST)/lib
-endif
 LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
