diff -Naur pixman-0.28.2-old/configure.ac pixman-0.28.2-new/configure.ac
--- pixman-0.28.2-old/configure.ac	2012-12-10 03:46:36.000000000 -0800
+++ pixman-0.28.2-new/configure.ac	2012-12-29 05:30:54.000000000 -0800
@@ -64,7 +64,7 @@
 # Suppress verbose compile lines
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_CANONICAL_HOST
 
