diff -Naur xbmc-dharma-32669/configure.in xbmc-dharma-32669.patch/configure.in
--- xbmc-dharma-32669/configure.in	2010-08-10 14:23:07.000000000 +0200
+++ xbmc-dharma-32669.patch/configure.in	2010-08-11 17:55:44.131212442 +0200
@@ -1619,6 +1619,7 @@
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
+      --enable-runtime-cpudetect \
       `case $host_cpu in powerpc*|arm*|amd64*|x86_64*|ia64*) echo --enable-pic ;; esac` \
       --cc="$CC" &&
       sed -i -e "s#define HAVE_SYMVER 1#define HAVE_SYMVER 0#" config.h &&
diff -Naur xbmc-dharma-32669/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp xbmc-dharma-32669.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
--- xbmc-dharma-32669/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp	2010-08-10 14:25:06.000000000 +0200
+++ xbmc-dharma-32669.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp	2010-08-11 17:57:03.591337432 +0200
@@ -241,7 +241,7 @@
   }
 
   // set acceleration
-  m_pCodecContext->dsp_mask = FF_MM_FORCE | FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE;
+  m_pCodecContext->dsp_mask = 0;//FF_MM_FORCE | FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE;
 
   // advanced setting override for skip loop filter (see avcodec.h for valid options)
   // TODO: allow per video setting?
