diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:48:30.812376923 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:49:06.305376455 +0200
@@ -61,6 +61,8 @@
 external_liba52_disabled="== Use of external liba52 disabled. =="
 external_libdts_enabled="== Use of external libdts enabled. =="
 external_libdts_disabled="== Use of external libdts disabled. =="
+external_libz_enabled="== Use of external libz enabled. =="
+external_libz_disabled="== Use of external libz disabled. =="
 external_python_enabled="== Use of external python enabled. =="
 external_python_disabled="== Use of external python disabled. =="
 ffmpeg_vdpau_not_supported="== External ffmpeg doesn't support VDPAU. VDPAU support disabled. =="
@@ -287,6 +289,12 @@
   [use_external_libdts=$enableval],
   [use_external_libdts=$use_external_libraries])
 
+AC_ARG_ENABLE([external-libz],
+            [AS_HELP_STRING([--enable-external-libz],
+            [enable use of external libz library (default is no)])],
+            [use_external_libz=$enableval],
+            [use_external_libz=no])
+
 AC_ARG_ENABLE([external-python],
   [AS_HELP_STRING([--enable-external-python],
   [enable use of external python library (default is no) 'Linux only'])],
@@ -848,6 +856,18 @@
   USE_EXTERNAL_LIBDTS=0
 fi
 
+# External libz
+if test "$use_external_libraries" = "yes" || test "$use_external_libz" = "yes"; then
+  AC_CHECK_LIB([z],          [main],, AC_MSG_ERROR($missing_library))
+  AC_CHECK_HEADERS([zlib.h],, AC_MSG_ERROR([$missing_headers]))
+  AC_MSG_NOTICE($external_libz_enabled)
+  USE_EXTERNAL_LIBZ=1
+  AC_DEFINE([USE_EXTERNAL_LIBZ], [1], [Whether to use external libz library/zlib.])
+else
+  AC_MSG_NOTICE($external_libz_disabled)
+  USE_EXTERNAL_LIBZ=0
+fi
+
 # External Python
 if test "$use_external_python" = "yes"; then
   AC_CHECK_LIB([python2.6], [main],
@@ -1353,6 +1373,11 @@
     final_message="$final_message\n  External libdts:\tNo"
   fi
 fi
+if test "$use_external_libz" = "yes"; then
+  final_message="$final_message\n  External zlib:\tYes"
+else
+  final_message="$final_message\n  External zlib:\tNo"
+fi
 if test "$use_external_python" = "yes"; then
   final_message="$final_message\n  External Python:\tYes"
 else
@@ -1791,7 +1816,7 @@
     CFLAGS="$CFLAGS -fno-common" \
     ./configure
   fi
-], [0])
+], [$USE_EXTERNAL_LIBZ])
 
 XB_CONFIG_MODULE([lib/libass], [
   if test "$host_vendor" = "apple" ; then
