diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:53:28.281376543 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:55:45.481375135 +0200
@@ -1045,6 +1045,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1639,6 +1640,7 @@
       `if test "$use_arch" != "no"; then echo --arch=$use_arch; fi`\
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       --target-os=$(tolower $(uname -s)) \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --enable-muxer=adts \
