diff -Naur xbmc-dharma-35100/Makefile.in xbmc-dharma-35100.patch/Makefile.in
--- xbmc-dharma-35100/Makefile.in	2010-10-30 05:36:41.000000000 +0200
+++ xbmc-dharma-35100.patch/Makefile.in	2010-11-01 04:53:48.542705631 +0100
@@ -471,11 +471,6 @@
 	xbmc/cores/DllLoader/exports/util/exports_utils.a \
 	xbmc/cores/DllLoader/exports/exports.a
 
-ifeq (arm, $(ARCH))
-# Force external python2.5 for now!
-LIBS +=-lpython2.5
-endif
-
 xbmc.bin: $(OBJSXBMC) $(DYNOBJSXBMC)
 ifeq ($(findstring osx,$(ARCH)), osx)
 	$(CXX) $(LDFLAGS) -o xbmc.bin -Wl,-all_load,-ObjC $(DYNOBJSXBMC) $(OBJSXBMC) $(LIBS) -rdynamic
