diff -Naur xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/Makefile xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/Makefile
--- xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/Makefile	2010-09-19 14:54:08.000000000 +0200
+++ xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/Makefile	2010-09-21 20:31:58.649299449 +0200
@@ -1,3 +1,4 @@
+CXX ?= g++
 CFLAGS = -Wall -pipe -fPIC -funroll-loops
 OBJ_DIR = release-$(shell $(CC) -v 2>&1|grep ^Target:|cut -d' ' -f2)
 OBJS = wiiuse_v0.12/src/$(OBJ_DIR)/libwiiuse.so
@@ -6,7 +7,7 @@
 
 all:
 	@$(MAKE) -C wiiuse_$(VERSION)/src $@
-	g++ $(CFLAGS) WiiUse_WiiRemote.cpp $(OBJS) -o $(BIN)
+	$(CXX) $(CFLAGS) -I./wiiuse_$(VERSION)/src WiiUse_WiiRemote.cpp $(OBJS) -o $(BIN)
 wiiuse:
 	@$(MAKE) -C wiiuse_$(VERSION)/src
 clean:
diff -Naur xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example/Makefile xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example/Makefile
--- xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example/Makefile	2010-09-19 14:54:08.000000000 +0200
+++ xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example/Makefile	2010-09-21 20:28:15.388265664 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
diff -Naur xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example-sdl/Makefile xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example-sdl/Makefile
--- xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example-sdl/Makefile	2010-09-19 14:54:08.000000000 +0200
+++ xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/example-sdl/Makefile	2010-09-21 20:28:22.476390252 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
diff -Naur xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/src/Makefile xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/src/Makefile
--- xbmc-dharma-33971/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/src/Makefile	2010-09-19 14:54:08.000000000 +0200
+++ xbmc-dharma-33971.patch/tools/EventClients/Clients/WiiRemote/wiiuse_v0.12/src/Makefile	2010-09-21 20:28:32.447265813 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
