diff -Naur xbmc-dharma-35003/xbmc/GUISettings.cpp xbmc-dharma-35003.patch/xbmc/GUISettings.cpp
--- xbmc-dharma-35003/xbmc/GUISettings.cpp	2010-10-25 15:36:18.000000000 +0200
+++ xbmc-dharma-35003.patch/xbmc/GUISettings.cpp	2010-10-25 23:34:14.131974258 +0200
@@ -618,6 +618,7 @@
   AddInt(vid, "myvideos.selectaction", 22079, SELECT_ACTION_PLAY_OR_RESUME, SELECT_ACTION_CHOOSE, 1, SELECT_ACTION_INFO, SPIN_CONTROL_TEXT);
   AddBool(NULL, "myvideos.treatstackasfile", 20051, true);
   AddBool(vid, "myvideos.extractflags",20433, true);
+  AddBool(vid, "myvideos.alwaysextractflags",20433, false);
   AddBool(vid, "myvideos.cleanstrings", 20418, false);
   AddBool(NULL, "myvideos.extractthumb",20433, true);
 
diff -Naur xbmc-dharma-35003/xbmc/ThumbLoader.cpp xbmc-dharma-35003.patch/xbmc/ThumbLoader.cpp
--- xbmc-dharma-35003/xbmc/ThumbLoader.cpp	2010-10-25 15:37:06.000000000 +0200
+++ xbmc-dharma-35003.patch/xbmc/ThumbLoader.cpp	2010-10-25 23:34:04.223857821 +0200
@@ -148,7 +148,7 @@
       m_item.SetThumbnailImage(m_target);
     }
   }
-  else if (m_item.HasVideoInfoTag() && !m_item.GetVideoInfoTag()->HasStreamDetails())
+  else if (m_item.HasVideoInfoTag() && (!m_item.GetVideoInfoTag()->HasStreamDetails() || g_guiSettings.GetBool("myvideos.alwaysextractflags")))
   {
     CLog::Log(LOGDEBUG,"%s - trying to extract filestream details from video file %s", __FUNCTION__, m_path.c_str());
     result = CDVDFileInfo::GetFileStreamDetails(&m_item);
