diff -Naur xbmc-dharma-35611/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp xbmc-dharma-35611.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp
--- xbmc-dharma-35611/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp	2010-12-10 14:45:40.000000000 +0100
+++ xbmc-dharma-35611.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/VDPAU.cpp	2010-12-10 16:15:52.642486771 +0100
@@ -662,7 +662,9 @@
   if (dl_vdp_device_create_x11)
   {
     CSingleLock lock(g_graphicsContext);
-    m_Display = g_Windowing.GetDisplay();
+    //m_Display = g_Windowing.GetDisplay();
+    m_Display = XOpenDisplay(NULL);
+
   }
 
   int mScreen = DefaultScreen(m_Display);
