From 10c0b2f9b4388318d185f97d530344750f528e64 Mon Sep 17 00:00:00 2001
From: fritsch <Peter.Fruehberger@gmail.com>
Date: Sun, 12 Apr 2015 19:40:37 +0200
Subject: [PATCH] CIR: Dump workaround for intel cir5

---
 drivers/media/rc/nuvoton-cir.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/rc/nuvoton-cir.c b/drivers/media/rc/nuvoton-cir.c
index 9c2c863..8a6832a 100644
--- a/drivers/media/rc/nuvoton-cir.c
+++ b/drivers/media/rc/nuvoton-cir.c
@@ -992,7 +992,7 @@ static int nvt_probe(struct pnp_dev *pdev, const struct pnp_device_id *dev_id)
 
 	/* validate pnp resources */
 	if (!pnp_port_valid(pdev, 0) ||
-	    pnp_port_len(pdev, 0) < CIR_IOREG_LENGTH) {
+	    pnp_port_len(pdev, 0) < 8) {
 		dev_err(&pdev->dev, "IR PNP Port not valid!\n");
 		goto exit_free_dev_rdev;
 	}
@@ -1003,7 +1003,7 @@ static int nvt_probe(struct pnp_dev *pdev, const struct pnp_device_id *dev_id)
 	}
 
 	if (!pnp_port_valid(pdev, 1) ||
-	    pnp_port_len(pdev, 1) < CIR_IOREG_LENGTH) {
+	    pnp_port_len(pdev, 1) < 8) {
 		dev_err(&pdev->dev, "Wake PNP Port not valid!\n");
 		goto exit_free_dev_rdev;
 	}
-- 
1.9.1

