diff --git a/src/os/libretro/Makefile b/src/os/libretro/Makefile
index d7092ed..22565e1 100644
--- a/src/os/libretro/Makefile
+++ b/src/os/libretro/Makefile
@@ -57,7 +57,6 @@ ifneq ($(GIT_VERSION)," unknown")
    CXXFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
 endif
 
-GET_STATIC_ARG=$(if $(filter $(shell $(CXX) -print-file-name=$1.a),$1.a),,-static-$1)
 # Unix
 ifneq (,$(findstring unix,$(platform)))
    GPP_MAJOR := $(shell $(CXX) -dumpversion | cut -d '.' -f 1)
@@ -68,7 +67,7 @@ ifneq (,$(findstring unix,$(platform)))
       CXXFLAGS := $(subst -std=c++20,-std=c++2a,$(CXXFLAGS))
    endif
    CXXFLAGS += $(LTO)
-   LDFLAGS += $(LTO) $(PTHREAD_FLAGS) $(call GET_STATIC_ARG,libgcc) $(call GET_STATIC_ARG,libstdc++)
+   LDFLAGS += $(LTO) $(PTHREAD_FLAGS)
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    ifneq ($(findstring SunOS,$(shell uname -a)),)
