diff -Naur xorg-server-1.4.orig/hw/xfree86/dri/dri.c xorg-server-1.4/hw/xfree86/dri/dri.c
--- xorg-server-1.4.orig/hw/xfree86/dri/dri.c	2008-01-14 19:25:12.000000000 +0100
+++ xorg-server-1.4/hw/xfree86/dri/dri.c	2008-01-14 19:25:41.000000000 +0100
@@ -74,7 +74,7 @@
 
 #define PCI_BUS_NO_DOMAIN(bus) ((bus) & 0xffu)
 
-#if !defined(PANORAMIX)
+#ifdef PANORAMIX
 extern Bool noPanoramiXExtension;
 #endif
 
@@ -308,7 +308,6 @@
     drm_context_t *       reserved;
     int                 reserved_count;
     int                 i;
-    Bool                xineramaInCore = FALSE;
     DRIEntPrivPtr       pDRIEntPriv;
     ScrnInfoPtr         pScrn = xf86Screens[pScreen->myNum];
     DRIContextFlags	flags    = 0;
@@ -321,21 +320,18 @@
 	return FALSE;
     }
 
+#ifdef PANORAMIX
     /*
      * If Xinerama is on, don't allow DRI to initialise.  It won't be usable
      * anyway.
      */
-    if (xf86LoaderCheckSymbol("noPanoramiXExtension"))
-	xineramaInCore = TRUE;
-
-    if (xineramaInCore) {
-	if (!noPanoramiXExtension) {
-	    DRIDrvMsg(pScreen->myNum, X_WARNING,
+    if (!noPanoramiXExtension) {
+      DRIDrvMsg(pScreen->myNum, X_WARNING, 
 		"Direct rendering is not supported when Xinerama is enabled\n");
 	    return FALSE;
 	}
-    }
-
+#endif /* PANORAMIX */
+    
     if (!DRIOpenDRMMaster(pScrn, pDRIInfo->SAREASize,
 			  pDRIInfo->busIdString,
 			  pDRIInfo->drmDriverName))
