--- xorg-server-1.4.0.90.orig/hw/xfree86/os-support/linux/Makefile.in	2008-02-22 16:56:43.000000000 +0100
+++ xorg-server-1.4.0.90/hw/xfree86/os-support/linux/Makefile.in	2008-02-22 16:57:07.000000000 +0100
@@ -412,7 +412,7 @@
                      $(PLATFORM_PCI_SUPPORT)
 
 AM_CFLAGS = -DUSESTDRES -DHAVE_SYSV_IPC $(DIX_CFLAGS) $(XORG_CFLAGS) $(PLATFORM_DEFINES)
-INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES) -I/usr/include/drm # FIXME this last part is crack
+INCLUDES = $(XORG_INCS) $(PLATFORM_INCLUDES)
 
 # FIXME: These need to be added to the build
 LNX_EXTRA_SRCS = \
