Subject: [PATCH] tvheadend: HACK ffmpeg 3.1.5 instead of 3.3

diff --git a/Makefile.ffmpeg b/Makefile.ffmpeg
--- a/Makefile.ffmpeg
+++ b/Makefile.ffmpeg
@@ -94,10 +94,10 @@ LIBMFX_URL     = https://github.com/lu-zero/mfx_dispatch/archive/$(LIBMFX_TB)
 LIBMFX_SHA1    = 84dbdf4a6b409067e863eb9564bb2efdec6d39ce
 LIBMFX_DIFFS   = libmfx.linux.path.diff
 
-FFMPEG         = ffmpeg-3.3
-FFMPEG_TB      = $(FFMPEG).tar.bz2
+FFMPEG         = ffmpeg-3.1.5
+FFMPEG_TB      = $(FFMPEG).tar.bz2
 FFMPEG_URL     = http://ffmpeg.org/releases/$(FFMPEG_TB)
-FFMPEG_SHA1    = 2590af8c754ce8d55745b1e5647e498bf271dc77
+FFMPEG_SHA1    = 053dfea8c9e057fdf237885decd03147cb2412b1
 
 
 # ##############################################################################
