diff -Naur a/Makefile.libretro b/Makefile.libretro
--- a/Makefile.libretro	2016-01-07 21:04:18.000000000 +0100
+++ b/Makefile.libretro	2016-01-07 21:46:25.556864218 +0100
@@ -96,7 +96,7 @@
    TARGET := $(TARGET_NAME)_libretro.so
    SHARED := -shared -Wl,--version-script=libretro/link.T
    fpic := -fPIC
-   CC = gcc
+   CC ?= gcc
    DESMUME_JIT_ARM = 1
 ifneq (,$(findstring cortexa8,$(platform)))
    CPPFLAGS += -marm -mcpu=cortex-a8
