diff -Naur bash-4.2/execute_cmd.c bash-4.2.patch/execute_cmd.c
--- bash-4.2/execute_cmd.c	2011-02-09 23:32:25.000000000 +0100
+++ bash-4.2.patch/execute_cmd.c	2011-02-16 14:30:36.517222120 +0100
@@ -2202,7 +2202,11 @@
   /* If the `lastpipe' option is set with shopt, and job control is not
      enabled, execute the last element of non-async pipelines in the
      current shell environment. */
+#if defined (JOB_CONTROL)
   if (lastpipe_opt && job_control == 0 && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+#else
+  if (lastpipe_opt && asynchronous == 0 && pipe_out == NO_PIPE && prev > 0)
+#endif /* JOB_CONTROL */
     {
       lstdin = move_to_high_fd (0, 0, 255);
       if (lstdin > 0)
