diff -Nur lirc-0.7.2pre3.orig/configure lirc-0.7.2pre3/configure
--- lirc-0.7.2pre3.orig/configure	Mon Aug  8 14:43:04 2005
+++ lirc-0.7.2pre3/configure	Mon Aug  8 14:42:32 2005
@@ -7870,16 +7870,12 @@
   kernelext=ko
   no_kernel=yes
 
-  if test `uname` != "Linux"; then
-    kerneldir="not running Linux"
-  else
-    for dir in /usr/src/kernel-source-`uname -r` /usr/src/linux-`uname -r` /usr/src/linux /lib/modules/`uname -r`/source /lib/modules/`uname -r`/build ${ac_kerneldir}; do
+    for dir in ${ac_kerneldir}; do
       if test -d $dir; then
         kerneldir=`dirname $dir/Makefile`/
         no_kernel=no
       fi;
     done
-  fi
 
   if test x${no_kernel} != xyes; then
     if test -f ${kerneldir}/Makefile; then
