diff -Naur lirc-0.7.0.orig/configure lirc-0.7.0/configure
--- lirc-0.7.0.orig/configure	2004-12-30 01:36:49.918207016 +0100
+++ lirc-0.7.0/configure	2004-12-30 01:36:56.205251240 +0100
@@ -22813,8 +22813,8 @@
       echo "	echo \$(VERSION)" >>${ac_pkss_makefile}
       echo "lirc_tell_me_what_patchlevel_is:" >>${ac_pkss_makefile}
       echo "	echo \$(PATCHLEVEL)" >>${ac_pkss_makefile}
-      version=$(make -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_version_is M=$(pwd))
-      patchlevel=$(make -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_patchlevel_is M=$(pwd))
+      version=$(${MAKE-make} -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_version_is M=$(pwd))
+      patchlevel=$(${MAKE-make} -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_patchlevel_is M=$(pwd))
       if test ${version} -eq 2; then
         if test ${patchlevel} -lt 5; then
           kernelext=o
