diff -Nur lirc-0.7.2pre3.orig/daemons/lircd.c lirc-0.7.2pre3/daemons/lircd.c
--- lirc-0.7.2pre3.orig/daemons/lircd.c	Sun Jul 10 11:34:12 2005
+++ lirc-0.7.2pre3/daemons/lircd.c	Fri Aug 12 08:41:21 2005
@@ -493,6 +493,14 @@
 	{
 		if(hw.init_func)
 		{
+			struct stat s;
+			int i;
+			for (i = 0; i < 10 * 5; i++)
+			{
+				if (!stat(hw.device, &s))
+					break;
+				usleep(100000);
+			}
 			if(!hw.init_func())
 			{
 				shutdown(clis[0],2);
