diff -Nur lirc-0.7.2pre3.orig/tools/irpty.c lirc-0.7.2pre3/tools/irpty.c
--- lirc-0.7.2pre3.orig/tools/irpty.c	Fri Aug 12 13:41:13 2005
+++ lirc-0.7.2pre3/tools/irpty.c	Fri Aug 12 13:44:32 2005
@@ -84,10 +84,12 @@
 	if ((child = fork()) < 0) {
 		die("fork error");
 	} else if (!child) {
+		int lircd_alive = 1;
 		fd_set fds;
 		while (1) {
 			FD_ZERO(&fds);
-			FD_SET(lsock, &fds);
+			if (lircd_alive)
+				FD_SET(lsock, &fds);
 			FD_SET(STDIN_FILENO, &fds);
 			select(lsock + 1, &fds, NULL, NULL, NULL);
 			
@@ -99,7 +101,7 @@
 				if (write(ptym, buf, nread) != nread)
 					die("writen error to master pty");
 			}
-			if (FD_ISSET(lsock, &fds))
+			if (lircd_alive && FD_ISSET(lsock, &fds))
 			{
 				char *ir;
 				char *irchars;
@@ -118,7 +120,8 @@
 					free(ir);
 					if(ret==-1) break;
 				}
-				if(ret==-1) break;
+				if(ret==-1)
+					lircd_alive = 0;
 			}
 		}
 		if (!ignoreeof)
