diff -Naur lirc-0.8.5.orig/drivers/lirc_wpc8769l/lirc_wpc8769l.c lirc-0.8.5/drivers/lirc_wpc8769l/lirc_wpc8769l.c
--- lirc-0.8.5.orig/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2009-05-31 16:53:36.000000000 +0200
+++ lirc-0.8.5/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2009-05-31 16:54:14.000000000 +0200
@@ -891,6 +891,7 @@
 	.owner		= THIS_MODULE,
 };
 
+#ifdef CONFIG_ACPI
 static acpi_status wec_parse_resources(struct acpi_resource *resource,
 	void *context)
 {
@@ -934,6 +935,7 @@
 	else
 		return 0;
 }
+#endif
 
 #ifdef MODULE
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 18)
@@ -1024,6 +1026,7 @@
 
 	/* If needed, read the resource information for the ACPI device
 	 * description. */
+#ifdef CONFIG_ACPI
 	if (!skip_probe) {
 		rc = wpc8769l_acpi_detect();
 		if (rc) {
@@ -1032,6 +1035,7 @@
 			return rc;
 		}
 	}
+#endif
 
 	/* Check that we got some resource info to work with. */
 	if (!baseport1 || !irq) {
