diff -Naur lirc-0.8.5pre2.orig/drivers/lirc_parallel/lirc_parallel.c lirc-0.8.5pre2/drivers/lirc_parallel/lirc_parallel.c
--- lirc-0.8.5pre2.orig/drivers/lirc_parallel/lirc_parallel.c	2009-04-25 19:04:58.000000000 +0200
+++ lirc-0.8.5pre2/drivers/lirc_parallel/lirc_parallel.c	2009-04-25 19:06:56.000000000 +0200
@@ -35,9 +35,6 @@
 #endif
 
 #include <linux/autoconf.h>
-#ifdef CONFIG_SMP
-#error "--- Sorry, this driver is not SMP safe. ---"
-#endif
 
 #include <linux/module.h>
 #include <linux/sched.h>
