diff -Naur gmp-5.0.5/configure.in gmp-5.0.5.patch/configure.in
--- gmp-5.0.5/configure.in	2012-05-06 13:19:49.000000000 +0200
+++ gmp-5.0.5.patch/configure.in	2013-01-12 02:25:31.897257189 +0100
@@ -65,8 +65,8 @@
 dnl  Note that there's a copy of these options in the top-level Makefile.am,
 dnl  so update there too if changing anything.
 dnl
-AM_INIT_AUTOMAKE([1.8 gnu no-dependencies $(top_builddir)/ansi2knr])
-AM_CONFIG_HEADER(config.h:config.in)
+AM_INIT_AUTOMAKE([1.8 gnu no-dependencies])
+AC_CONFIG_HEADERS(config.h:config.in)
 AM_MAINTAINER_MODE
 
 
@@ -2228,9 +2228,6 @@
 echo "      MPN_PATH=\"$path\""
 
 
-# Automake ansi2knr support.
-AM_C_PROTOTYPES
-
 CL_AS_NOEXECSTACK
 
 GMP_PROG_AR
diff -Naur gmp-5.0.5/Makefile.am gmp-5.0.5.patch/Makefile.am
--- gmp-5.0.5/Makefile.am	2012-05-06 13:19:49.000000000 +0200
+++ gmp-5.0.5.patch/Makefile.am	2013-01-12 02:25:09.603364249 +0100
@@ -25,7 +25,7 @@
 # Makefiles in subdirectories, but here we must omit it so automake gives
 # the actual ansi2knr build rule, not "cd $(top_builddir) && make ansi2knr".
 #
-AUTOMAKE_OPTIONS = 1.8 gnu no-dependencies ansi2knr
+AUTOMAKE_OPTIONS = 1.8 gnu no-dependencies
 
 
 # Libtool -version-info for libgmp.la and libmp.la.  See "Versioning" in the
