--- a/egl/wayland-egl.pc.in	2020-02-05 00:33:12.000000000 +0100
+++ b/egl/wayland-egl.pc.in	2020-02-10 22:45:04.592575524 +0100
@@ -4,8 +4,9 @@
 includedir=@includedir@
 
 Name: wayland-egl
-Description: Frontend wayland-egl library
-Version: 18.1.0
-Requires: wayland-client
-Libs: -L${libdir} -lwayland-egl
+Description: Mali EGL library
+Requires: wayland-client
+Version: 18.1.0
+Libs: -L${libdir} -lMali
+Libs.private: -lm -lpthread
 Cflags: -I${includedir}
