diff -Naur lame-3.99.5/configure.in lame-3.99.5.patch/configure.in
--- lame-3.99.5/configure.in	2012-02-28 19:50:27.000000000 +0100
+++ lame-3.99.5.patch/configure.in	2012-06-13 18:26:00.741781534 +0200
@@ -77,9 +77,6 @@
 	AC_MSG_RESULT(${GCC_version})
 fi
 
-dnl more automake stuff
-AM_C_PROTOTYPES
-
 AC_CHECK_HEADER(dmalloc.h)
 if test "${ac_cv_header_dmalloc_h}" = "yes"; then
 	AM_WITH_DMALLOC
diff -Naur lame-3.99.5/doc/html/Makefile.am lame-3.99.5.patch/doc/html/Makefile.am
--- lame-3.99.5/doc/html/Makefile.am	2010-09-30 22:58:40.000000000 +0200
+++ lame-3.99.5.patch/doc/html/Makefile.am	2012-06-13 18:28:01.586276747 +0200
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.7 2010/09/30 20:58:40 jaz001 Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 docdir = $(datadir)/doc
 pkgdocdir = $(docdir)/$(PACKAGE)
diff -Naur lame-3.99.5/doc/man/Makefile.am lame-3.99.5.patch/doc/man/Makefile.am
--- lame-3.99.5/doc/man/Makefile.am	2000-10-22 13:39:44.000000000 +0200
+++ lame-3.99.5.patch/doc/man/Makefile.am	2012-06-13 18:27:52.160085394 +0200
@@ -1,6 +1,6 @@
 ## $Id: Makefile.am,v 1.1 2000/10/22 11:39:44 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
+AUTOMAKE_OPTIONS = foreign
 
 man_MANS = lame.1
 EXTRA_DIST = ${man_MANS}
diff -Naur lame-3.99.5/libmp3lame/i386/Makefile.am lame-3.99.5.patch/libmp3lame/i386/Makefile.am
--- lame-3.99.5/libmp3lame/i386/Makefile.am	2011-04-04 11:42:34.000000000 +0200
+++ lame-3.99.5.patch/libmp3lame/i386/Makefile.am	2012-06-13 18:27:23.761463827 +0200
@@ -1,7 +1,5 @@
 ## $Id: Makefile.am,v 1.26 2011/04/04 09:42:34 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign $(top_srcdir)/ansi2knr
-
 DEFS = @DEFS@ @CONFIG_DEFS@
 
 ECHO ?= echo
