diff -Naur pcre-8.30/Makefile.am pcre-8.30.patch/Makefile.am
--- pcre-8.30/Makefile.am	2012-01-27 12:10:10.000000000 +0100
+++ pcre-8.30.patch/Makefile.am	2012-02-29 20:50:54.271154676 +0100
@@ -521,31 +521,31 @@
 
 # Arrange for the per-function man pages to have 16-bit names as well.
 install-data-hook:
-	ln -s pcre_assign_jit_stack.3		$(DESTDIR)$(man3dir)/pcre16_assign_jit_stack.3
-	ln -s pcre_compile.3			$(DESTDIR)$(man3dir)/pcre16_compile.3
-	ln -s pcre_compile2.3			$(DESTDIR)$(man3dir)/pcre16_compile2.3
-	ln -s pcre_config.3			$(DESTDIR)$(man3dir)/pcre16_config.3
-	ln -s pcre_copy_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_copy_named_substring.3
-	ln -s pcre_copy_substring.3		$(DESTDIR)$(man3dir)/pcre16_copy_substring.3
-	ln -s pcre_dfa_exec.3			$(DESTDIR)$(man3dir)/pcre16_dfa_exec.3
-	ln -s pcre_exec.3			$(DESTDIR)$(man3dir)/pcre16_exec.3
-	ln -s pcre_free_study.3			$(DESTDIR)$(man3dir)/pcre16_free_study.3
-	ln -s pcre_free_substring.3		$(DESTDIR)$(man3dir)/pcre16_free_substring.3
-	ln -s pcre_free_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_free_substring_list.3
-	ln -s pcre_fullinfo.3			$(DESTDIR)$(man3dir)/pcre16_fullinfo.3
-	ln -s pcre_get_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_get_named_substring.3
-	ln -s pcre_get_stringnumber.3		$(DESTDIR)$(man3dir)/pcre16_get_stringnumber.3
-	ln -s pcre_get_stringtable_entries.3	$(DESTDIR)$(man3dir)/pcre16_get_stringtable_entries.3
-	ln -s pcre_get_substring.3		$(DESTDIR)$(man3dir)/pcre16_get_substring.3
-	ln -s pcre_get_substring_list.3		$(DESTDIR)$(man3dir)/pcre16_get_substring_list.3
-	ln -s pcre_jit_stack_alloc.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_alloc.3
-	ln -s pcre_jit_stack_free.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_free.3
-	ln -s pcre_maketables.3			$(DESTDIR)$(man3dir)/pcre16_maketables.3
-	ln -s pcre_pattern_to_host_byte_order.3 $(DESTDIR)$(man3dir)/pcre16_pattern_to_host_byte_order.3
-	ln -s pcre_refcount.3			$(DESTDIR)$(man3dir)/pcre16_refcount.3
-	ln -s pcre_study.3			$(DESTDIR)$(man3dir)/pcre16_study.3
-	ln -s pcre_utf16_to_host_byte_order.3	$(DESTDIR)$(man3dir)/pcre16_utf16_to_host_byte_order.3
-	ln -s pcre_version.3			$(DESTDIR)$(man3dir)/pcre16_version.3
+	ln -sf pcre_assign_jit_stack.3		$(DESTDIR)$(man3dir)/pcre16_assign_jit_stack.3
+	ln -sf pcre_compile.3			$(DESTDIR)$(man3dir)/pcre16_compile.3
+	ln -sf pcre_compile2.3			$(DESTDIR)$(man3dir)/pcre16_compile2.3
+	ln -sf pcre_config.3			$(DESTDIR)$(man3dir)/pcre16_config.3
+	ln -sf pcre_copy_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_copy_named_substring.3
+	ln -sf pcre_copy_substring.3		$(DESTDIR)$(man3dir)/pcre16_copy_substring.3
+	ln -sf pcre_dfa_exec.3			$(DESTDIR)$(man3dir)/pcre16_dfa_exec.3
+	ln -sf pcre_exec.3			$(DESTDIR)$(man3dir)/pcre16_exec.3
+	ln -sf pcre_free_study.3		$(DESTDIR)$(man3dir)/pcre16_free_study.3
+	ln -sf pcre_free_substring.3		$(DESTDIR)$(man3dir)/pcre16_free_substring.3
+	ln -sf pcre_free_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_free_substring_list.3
+	ln -sf pcre_fullinfo.3			$(DESTDIR)$(man3dir)/pcre16_fullinfo.3
+	ln -sf pcre_get_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_get_named_substring.3
+	ln -sf pcre_get_stringnumber.3		$(DESTDIR)$(man3dir)/pcre16_get_stringnumber.3
+	ln -sf pcre_get_stringtable_entries.3	$(DESTDIR)$(man3dir)/pcre16_get_stringtable_entries.3
+	ln -sf pcre_get_substring.3		$(DESTDIR)$(man3dir)/pcre16_get_substring.3
+	ln -sf pcre_get_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_get_substring_list.3
+	ln -sf pcre_jit_stack_alloc.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_alloc.3
+	ln -sf pcre_jit_stack_free.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_free.3
+	ln -sf pcre_maketables.3		$(DESTDIR)$(man3dir)/pcre16_maketables.3
+	ln -sf pcre_pattern_to_host_byte_order.3 $(DESTDIR)$(man3dir)/pcre16_pattern_to_host_byte_order.3
+	ln -sf pcre_refcount.3			$(DESTDIR)$(man3dir)/pcre16_refcount.3
+	ln -sf pcre_study.3			$(DESTDIR)$(man3dir)/pcre16_study.3
+	ln -sf pcre_utf16_to_host_byte_order.3	$(DESTDIR)$(man3dir)/pcre16_utf16_to_host_byte_order.3
+	ln -sf pcre_version.3			$(DESTDIR)$(man3dir)/pcre16_version.3
 
 pcrecpp_man = doc/pcrecpp.3
 EXTRA_DIST += $(pcrecpp_man)
diff -Naur pcre-8.30/Makefile.in pcre-8.30.patch/Makefile.in
--- pcre-8.30/Makefile.in	2012-02-04 15:35:53.000000000 +0100
+++ pcre-8.30.patch/Makefile.in	2012-02-29 20:51:36.067990716 +0100
@@ -1799,31 +1799,31 @@
 
 # Arrange for the per-function man pages to have 16-bit names as well.
 install-data-hook:
-	ln -s pcre_assign_jit_stack.3		$(DESTDIR)$(man3dir)/pcre16_assign_jit_stack.3
-	ln -s pcre_compile.3			$(DESTDIR)$(man3dir)/pcre16_compile.3
-	ln -s pcre_compile2.3			$(DESTDIR)$(man3dir)/pcre16_compile2.3
-	ln -s pcre_config.3			$(DESTDIR)$(man3dir)/pcre16_config.3
-	ln -s pcre_copy_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_copy_named_substring.3
-	ln -s pcre_copy_substring.3		$(DESTDIR)$(man3dir)/pcre16_copy_substring.3
-	ln -s pcre_dfa_exec.3			$(DESTDIR)$(man3dir)/pcre16_dfa_exec.3
-	ln -s pcre_exec.3			$(DESTDIR)$(man3dir)/pcre16_exec.3
-	ln -s pcre_free_study.3			$(DESTDIR)$(man3dir)/pcre16_free_study.3
-	ln -s pcre_free_substring.3		$(DESTDIR)$(man3dir)/pcre16_free_substring.3
-	ln -s pcre_free_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_free_substring_list.3
-	ln -s pcre_fullinfo.3			$(DESTDIR)$(man3dir)/pcre16_fullinfo.3
-	ln -s pcre_get_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_get_named_substring.3
-	ln -s pcre_get_stringnumber.3		$(DESTDIR)$(man3dir)/pcre16_get_stringnumber.3
-	ln -s pcre_get_stringtable_entries.3	$(DESTDIR)$(man3dir)/pcre16_get_stringtable_entries.3
-	ln -s pcre_get_substring.3		$(DESTDIR)$(man3dir)/pcre16_get_substring.3
-	ln -s pcre_get_substring_list.3		$(DESTDIR)$(man3dir)/pcre16_get_substring_list.3
-	ln -s pcre_jit_stack_alloc.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_alloc.3
-	ln -s pcre_jit_stack_free.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_free.3
-	ln -s pcre_maketables.3			$(DESTDIR)$(man3dir)/pcre16_maketables.3
-	ln -s pcre_pattern_to_host_byte_order.3 $(DESTDIR)$(man3dir)/pcre16_pattern_to_host_byte_order.3
-	ln -s pcre_refcount.3			$(DESTDIR)$(man3dir)/pcre16_refcount.3
-	ln -s pcre_study.3			$(DESTDIR)$(man3dir)/pcre16_study.3
-	ln -s pcre_utf16_to_host_byte_order.3	$(DESTDIR)$(man3dir)/pcre16_utf16_to_host_byte_order.3
-	ln -s pcre_version.3			$(DESTDIR)$(man3dir)/pcre16_version.3
+	ln -sf pcre_assign_jit_stack.3		$(DESTDIR)$(man3dir)/pcre16_assign_jit_stack.3
+	ln -sf pcre_compile.3			$(DESTDIR)$(man3dir)/pcre16_compile.3
+	ln -sf pcre_compile2.3			$(DESTDIR)$(man3dir)/pcre16_compile2.3
+	ln -sf pcre_config.3			$(DESTDIR)$(man3dir)/pcre16_config.3
+	ln -sf pcre_copy_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_copy_named_substring.3
+	ln -sf pcre_copy_substring.3		$(DESTDIR)$(man3dir)/pcre16_copy_substring.3
+	ln -sf pcre_dfa_exec.3			$(DESTDIR)$(man3dir)/pcre16_dfa_exec.3
+	ln -sf pcre_exec.3			$(DESTDIR)$(man3dir)/pcre16_exec.3
+	ln -sf pcre_free_study.3		$(DESTDIR)$(man3dir)/pcre16_free_study.3
+	ln -sf pcre_free_substring.3		$(DESTDIR)$(man3dir)/pcre16_free_substring.3
+	ln -sf pcre_free_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_free_substring_list.3
+	ln -sf pcre_fullinfo.3			$(DESTDIR)$(man3dir)/pcre16_fullinfo.3
+	ln -sf pcre_get_named_substring.3	$(DESTDIR)$(man3dir)/pcre16_get_named_substring.3
+	ln -sf pcre_get_stringnumber.3		$(DESTDIR)$(man3dir)/pcre16_get_stringnumber.3
+	ln -sf pcre_get_stringtable_entries.3	$(DESTDIR)$(man3dir)/pcre16_get_stringtable_entries.3
+	ln -sf pcre_get_substring.3		$(DESTDIR)$(man3dir)/pcre16_get_substring.3
+	ln -sf pcre_get_substring_list.3	$(DESTDIR)$(man3dir)/pcre16_get_substring_list.3
+	ln -sf pcre_jit_stack_alloc.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_alloc.3
+	ln -sf pcre_jit_stack_free.3		$(DESTDIR)$(man3dir)/pcre16_jit_stack_free.3
+	ln -sf pcre_maketables.3		$(DESTDIR)$(man3dir)/pcre16_maketables.3
+	ln -sf pcre_pattern_to_host_byte_order.3 $(DESTDIR)$(man3dir)/pcre16_pattern_to_host_byte_order.3
+	ln -sf pcre_refcount.3			$(DESTDIR)$(man3dir)/pcre16_refcount.3
+	ln -sf pcre_study.3			$(DESTDIR)$(man3dir)/pcre16_study.3
+	ln -sf pcre_utf16_to_host_byte_order.3	$(DESTDIR)$(man3dir)/pcre16_utf16_to_host_byte_order.3
+	ln -sf pcre_version.3			$(DESTDIR)$(man3dir)/pcre16_version.3
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
