diff -Naur libsndfile-1.0.25-old/configure.ac libsndfile-1.0.25-new/configure.ac
--- libsndfile-1.0.25-old/configure.ac	2011-07-13 01:59:57.000000000 -0700
+++ libsndfile-1.0.25-new/configure.ac	2011-07-13 17:04:14.000000000 -0700
@@ -18,9 +18,6 @@
 AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
 AM_SILENT_RULES([yes])
 
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
-
 AC_LANG([C])
 
 AC_PROG_CC
diff -Naur libsndfile-1.0.25-old/Makefile.am libsndfile-1.0.25-new/Makefile.am
--- libsndfile-1.0.25-old/Makefile.am	2011-07-07 02:47:24.000000000 -0700
+++ libsndfile-1.0.25-new/Makefile.am	2011-07-13 17:04:14.000000000 -0700
@@ -2,6 +2,8 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
+ACLOCAL_AMFLAGS = -I M4
+
 if BUILD_OCTAVE_MOD
 octave_dir = Octave
 endif
