diff --git a/libretro-common/gfx/gl_capabilities.c b/libretro-common/gfx/gl_capabilities.c
index 0cc8eda59..270fc0968 100644
--- a/libretro-common/gfx/gl_capabilities.c
+++ b/libretro-common/gfx/gl_capabilities.c
@@ -300,7 +300,6 @@ bool gl_check_capability(enum gl_capability_enum enum_idx)
       case GL_CAPS_BGRA8888:
 #ifdef HAVE_OPENGLES
          /* There are both APPLE and EXT variants. */
-         if (gl_query_extension("BGRA8888"))
             return true;
 #else
          return true;
