diff -Nur a/include/ieee80211.h b/include/ieee80211.h
--- a/include/ieee80211.h	2016-04-19 18:42:01.000000000 +0200
+++ b/include/ieee80211.h	2016-04-19 18:46:08.596497819 +0200
@@ -1399,7 +1399,7 @@
         (addr[3] == 0xff) && (addr[4] == 0xff) && (addr[5] == 0xff));
 }
 
-extern __inline int is_zero_mac_addr(const u8 *addr)
+static __inline int is_zero_mac_addr(const u8 *addr)
 {
     return ((addr[0] == 0x00) && (addr[1] == 0x00) && (addr[2] == 0x00) &&   \
         (addr[3] == 0x00) && (addr[4] == 0x00) && (addr[5] == 0x00));
