diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc0676d3d..0d13f62c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -813,7 +813,7 @@ elseif(WIN32)
 		link_directories(dx9sdk/Lib/x86)
 	endif()
 	set(nativeExtra ${nativeExtra} ext/native/base/logging.cpp)
-elseif(TARGET SDL2::SDL2)
+elseif(TARGET SDL2::SDL2 AND NOT LIBRETRO)
 	set(TargetBin PPSSPPSDL)
 	# Require SDL
 	add_definitions(-DSDL)
