diff --git a/libretro/Makefile b/libretro/Makefile
index 4fdb2ea..0c7ef54 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -83,13 +83,13 @@ else ifneq (,$(findstring rpi,$(platform)))
 	INCFLAGS += -I/opt/vc/include
 	CPUFLAGS += -DARMv5_ONLY 
 	PLATFORM_EXT := unix
-	TARGET_ARCH = arm
+	override TARGET_ARCH := arm
 	LDFLAGS += -lrt -ldl
 	
 # ARM64
 else ifneq (,$(findstring arm64,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
-	TARGET_ARCH = arm64
+	override TARGET_ARCH := arm64
 	fpic := -fPIC
 	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
 	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/aarch64/include
@@ -95,7 +119,7 @@ else ifneq (,$(findstring imx6,$(platform)))
 	GL_LIB := -lGLESv2
 	CPUFLAGS += 
 	PLATFORM_EXT := unix
-	TARGET_ARCH = arm
+	override TARGET_ARCH = arm
 	HAVE_NEON=1
 	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/$(TARGET_ARCH)/include
 	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/$(TARGET_ARCH)/lib
@@ -151,7 +175,7 @@ else ifneq (,$(findstring ios,$(platform)))
 	PLATCFLAGS += -DIOS -DHAVE_POSIX_MEMALIGN
 	CPUFLAGS += -DARMv5_ONLY -DARM
 	PLATFORM_EXT := unix
-	TARGET_ARCH = arm
+	override TARGET_ARCH = arm
 
 # Android
 else ifneq (,$(findstring android,$(platform)))
@@ -162,7 +186,7 @@ else ifneq (,$(findstring android,$(platform)))
 
 	CC = arm-linux-androideabi-gcc
 	CXX = arm-linux-androideabi-g++
-	TARGET_ARCH = arm
+	override TARGET_ARCH = arm
 	GLES = 1
 	PLATCFLAGS += -DANDROID 
 	CPUCFLAGS  += 
@@ -188,7 +212,7 @@ else ifeq ($(platform), qnx)
 	AS = qcc -Vgcc_ntoarmv7le
 	CXX = QCC -Vgcc_ntoarmv7le
 	AR = QCC -Vgcc_ntoarmv7le
-	TARGET_ARCH = arm
+	override TARGET_ARCH = arm
 	GLES = 1
 	PLATCFLAGS += -D__BLACKBERRY_QNX__
 	HAVE_NEON = 1
@@ -200,11 +224,11 @@ else ifeq ($(platform), qnx)
 # ARM
 else ifneq (,$(findstring armv,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
-	TARGET_ARCH = armv7
+	override TARGET_ARCH = arm
 	fpic := -fPIC
 	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
-	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/$(TARGET_ARCH)/include
-	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/$(TARGET_ARCH)/lib
+	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/arm/include
+	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/arm/lib
 	FFMPEGLDFLAGS += -L$(FFMPEGLIBDIR) -lavformat -lavcodec -lavutil -lswresample -lswscale
 	INCFLAGS += -I.
 	ifneq (,$(findstring gles,$(platform)))
diff --git a/libretro/Makefile.common b/libretro/Makefile.common
index c732d9a..f1620da 100644
--- a/libretro/Makefile.common
+++ b/libretro/Makefile.common
@@ -87,6 +87,20 @@ ifeq ($(PLATFORM_EXT), android)
 		$(NATIVEDIR)/gfx_es2/gl3stub.c \
 		$(NATIVEDIR)/math/fast/fast_math.c \
 		$(NATIVEDIR)/math/fast/fast_matrix.c
+else ifeq ($(TARGET_ARCH),arm)
+	SOURCES_C +=\
+		$(NATIVEDIR)/gfx_es2/gl3stub.c \
+		$(NATIVEDIR)/math/fast/fast_math.c \
+		$(NATIVEDIR)/math/fast/fast_matrix.c
+	SOURCES_CXX +=\
+		$(COREDIR)/Util/AudioFormatNEON.cpp
+else ifeq ($(TARGET_ARCH),arm64)
+	SOURCES_C +=\
+		$(NATIVEDIR)/gfx_es2/gl3stub.c \
+		$(NATIVEDIR)/math/fast/fast_math.c \
+		$(NATIVEDIR)/math/fast/fast_matrix.c
+	SOURCES_CXX +=\
+		$(COREDIR)/Util/AudioFormatNEON.cpp
 endif
 
 SOURCES_C += $(EXTDIR)/glew/glew.c
@@ -563,10 +574,6 @@ endif
 		SOURCES_C   += $(NATIVEDIR)/math/fast/fast_matrix_sse.c
    endif
 endif
-SOURCES_CXX += \
-   $(COMMONDIR)/ArmEmitter.cpp \
-   $(COMMONDIR)/Arm64Emitter.cpp \
-   $(COREDIR)/Util/DisArm64.cpp
 
 #UDIS86
 # Compiled and linked even on ARM for now
