diff --git a/libretro/Makefile b/libretro/Makefile
index 8970ddc..732d899 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -85,6 +85,30 @@ else ifneq (,$(findstring rpi,$(platform)))
 	PLATFORM_EXT := unix
 	TARGET_ARCH = arm
 	LDFLAGS += -lrt -ldl
+	
+# ARM64
+else ifneq (,$(findstring arm64,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	TARGET_ARCH = arm64
+	fpic := -fPIC
+	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
+	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/aarch64/include
+	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/aarch64/lib
+	FFMPEGLDFLAGS += -L$(FFMPEGLIBDIR) -lavformat -lavcodec -lavutil -lswresample -lswscale
+	INCFLAGS += -I.
+	HAVE_NEON=1
+	ifneq (,$(findstring gles,$(platform)))
+		GLES := 1
+		GL_LIB := -lGLESv2 -lEGL
+		LDFLAGS += -lGLESv2 -lEGL
+	else
+		GL_LIB := -lGL
+
+	endif
+	CPUFLAGS += -D__arm64__ -DARM64_ASM -D__NEON_OPT
+	PLATCFLAGS += -DARM64
+	LDFLAGS += -lrt -ldl
+	PLATFORM_EXT := unix
 
 # i.MX6
 else ifneq (,$(findstring imx6,$(platform)))
@@ -200,11 +224,11 @@ else ifeq ($(platform), qnx)
 # ARM
 else ifneq (,$(findstring armv,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
-	TARGET_ARCH = armv7
+	TARGET_ARCH = arm
 	fpic := -fPIC
 	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
-	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/$(TARGET_ARCH)/include
-	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/$(TARGET_ARCH)/lib
+	FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/arm/include
+	FFMPEGLIBDIR := $(FFMPEGDIR)/linux/arm/lib
 	FFMPEGLDFLAGS += -L$(FFMPEGLIBDIR) -lavformat -lavcodec -lavutil -lswresample -lswscale
 	INCFLAGS += -I.
 	ifneq (,$(findstring gles,$(platform)))
diff --git a/libretro/Makefile.common b/libretro/Makefile.common
index b0751e2..fdd5a87 100644
--- a/libretro/Makefile.common
+++ b/libretro/Makefile.common
@@ -87,6 +87,16 @@ ifeq ($(PLATFORM_EXT), android)
 		$(NATIVEDIR)/gfx_es2/gl3stub.c \
 		$(NATIVEDIR)/math/fast/fast_math.c \
 		$(NATIVEDIR)/math/fast/fast_matrix.c
+else ifeq ($(TARGET_ARCH),arm)
+	SOURCES_C +=\
+		$(NATIVEDIR)/gfx_es2/gl3stub.c \
+		$(NATIVEDIR)/math/fast/fast_math.c \
+		$(NATIVEDIR)/math/fast/fast_matrix.c
+else ifeq ($(TARGET_ARCH),arm64)
+	SOURCES_C +=\
+		$(NATIVEDIR)/gfx_es2/gl3stub.c \
+		$(NATIVEDIR)/math/fast/fast_math.c \
+		$(NATIVEDIR)/math/fast/fast_matrix.c
 endif
 
 SOURCES_C += $(EXTDIR)/glew/glew.c
@@ -564,10 +574,6 @@ endif
 		SOURCES_C   += $(NATIVEDIR)/math/fast/fast_matrix_sse.c
    endif
 endif
-SOURCES_CXX += \
-   $(COMMONDIR)/ArmEmitter.cpp \
-   $(COMMONDIR)/Arm64Emitter.cpp \
-   $(COREDIR)/Util/DisArm64.cpp
 
 #UDIS86
 # Compiled and linked even on ARM for now
