diff -Naur ppsspp-00051fd/ext/glew/glew.c ppsspp-00051fd.patch/ext/glew/glew.c
--- ppsspp-00051fd/ext/glew/glew.c	2018-12-24 10:48:13.869315087 +0100
+++ ppsspp-00051fd.patch/ext/glew/glew.c	2018-12-24 10:50:23.969537401 +0100
@@ -20872,8 +20872,6 @@
   return r;
 #elif defined(_WIN32)
   return wglewInit();
-#elif !defined(__APPLE__) || defined(GLEW_APPLE_GLX) /* _UNIX */
-  return glxewInit();
 #else
   return r;
 #endif /* _WIN32 */
