diff -Naur libmms-0.5/src/mmsh.h libmms-0.5.patch/src/mmsh.h
--- libmms-0.5/src/mmsh.h	2009-10-31 13:35:28.000000000 +0100
+++ libmms-0.5.patch/src/mmsh.h	2010-01-25 12:11:31.862959172 +0100
@@ -41,20 +41,20 @@
 
 int      mmsh_read (mms_io_t *io, mmsh_t *instance, char *data, int len);
 int      mmsh_time_seek (mms_io_t *io, mmsh_t *instance, double time_sec);
-mms_off_t mmsh_seek (mms_io_t *io, mmsh_t *this, mms_off_t offset, int origin);
+mms_off_t mmsh_seek (mms_io_t *io, mmsh_t *instance, mms_off_t offset, int origin);
 uint32_t mmsh_get_length (mmsh_t *instance);
-double   mmsh_get_time_length (mmsh_t *this);
-uint64_t mmsh_get_raw_time_length (mmsh_t *this);
+double   mmsh_get_time_length (mmsh_t *instance);
+uint64_t mmsh_get_raw_time_length (mmsh_t *instance);
 mms_off_t mmsh_get_current_pos (mmsh_t *instance);
 void     mmsh_close (mmsh_t *instance);
 
 int      mmsh_peek_header (mmsh_t *instance, char *data, int maxsize);
 
-uint32_t mmsh_get_asf_header_len (mmsh_t *this);
+uint32_t mmsh_get_asf_header_len (mmsh_t *instance);
 
-uint32_t mmsh_get_asf_packet_len (mmsh_t *this);
+uint32_t mmsh_get_asf_packet_len (mmsh_t *instance);
 
-int      mmsh_get_seekable (mmsh_t *this);
+int      mmsh_get_seekable (mmsh_t *instance);
 
 #ifdef __cplusplus
 }
diff -Naur libmms-0.5/src/mmsx.h libmms-0.5.patch/src/mmsx.h
--- libmms-0.5/src/mmsx.h	2009-10-31 13:35:28.000000000 +0100
+++ libmms-0.5.patch/src/mmsx.h	2010-01-25 12:11:51.451960516 +0100
@@ -55,11 +55,11 @@
 
 mms_off_t mmsx_get_current_pos (mmsx_t *instance);
 
-uint32_t  mmsx_get_asf_header_len (mmsx_t *this);
+uint32_t  mmsx_get_asf_header_len (mmsx_t *instance);
 
-uint64_t  mmsx_get_asf_packet_len (mmsx_t *this);
+uint64_t  mmsx_get_asf_packet_len (mmsx_t *instance);
 
-int       mmsx_get_seekable (mmsx_t *this);
+int       mmsx_get_seekable (mmsx_t *instance);
 
 #ifdef __cplusplus
 }
