diff -Naur gcc-4.9.0/gcc/config/i386/linux64.h gcc-4.9.0.patch/gcc/config/i386/linux64.h
--- gcc-4.9.0/gcc/config/i386/linux64.h	2014-01-02 23:23:26.000000000 +0100
+++ gcc-4.9.0.patch/gcc/config/i386/linux64.h	2014-04-22 20:27:05.853832561 +0200
@@ -28,5 +28,5 @@
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 "/lib/ld-linux-x32.so.2"
diff -Naur gcc-4.9.0/gcc/config/mips/linux64.h gcc-4.9.0.patch/gcc/config/mips/linux64.h
--- gcc-4.9.0/gcc/config/mips/linux64.h	2014-01-02 23:23:26.000000000 +0100
+++ gcc-4.9.0.patch/gcc/config/mips/linux64.h	2014-04-22 20:28:54.699515966 +0200
@@ -25,9 +25,9 @@
 #define GLIBC_DYNAMIC_LINKER32 \
   "%{mnan=2008:/lib/ld-linux-mipsn8.so.1;:/lib/ld.so.1}"
 #define GLIBC_DYNAMIC_LINKER64 \
-  "%{mnan=2008:/lib64/ld-linux-mipsn8.so.1;:/lib64/ld.so.1}"
+  "%{mnan=2008:/lib64/ld-linux-mipsn8.so.1;:/lib/ld.so.1}"
 #define GLIBC_DYNAMIC_LINKERN32 \
-  "%{mnan=2008:/lib32/ld-linux-mipsn8.so.1;:/lib32/ld.so.1}"
+  "%{mnan=2008:/lib32/ld-linux-mipsn8.so.1;:/lib/ld.so.1}"
 
 #undef UCLIBC_DYNAMIC_LINKER32
 #define UCLIBC_DYNAMIC_LINKER32 \
@@ -36,7 +36,7 @@
 #define UCLIBC_DYNAMIC_LINKER64 \
   "%{mnan=2008:/lib/ld64-uClibc-mipsn8.so.0;:/lib/ld64-uClibc.so.0}"
 #define UCLIBC_DYNAMIC_LINKERN32 \
-  "%{mnan=2008:/lib32/ld-uClibc-mipsn8.so.0;:/lib32/ld-uClibc.so.0}"
+  "%{mnan=2008:/lib32/ld-uClibc-mipsn8.so.0;:/lib/ld-uClibc.so.0}"
 
 #define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
diff -Naur gcc-4.9.0/gcc/config/rs6000/linux64.h gcc-4.9.0.patch/gcc/config/rs6000/linux64.h
--- gcc-4.9.0/gcc/config/rs6000/linux64.h	2014-01-02 23:23:26.000000000 +0100
+++ gcc-4.9.0.patch/gcc/config/rs6000/linux64.h	2014-04-22 20:30:39.567210839 +0200
@@ -369,9 +369,9 @@
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
 #ifdef LINUX64_DEFAULT_ABI_ELFv2
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv1:/lib64/ld64.so.1;:/lib64/ld64.so.2}"
+#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv1:/lib/ld64.so.1;:/lib/ld64.so.2}"
 #else
-#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv2:/lib64/ld64.so.2;:/lib64/ld64.so.1}"
+#define GLIBC_DYNAMIC_LINKER64 "%{mabi=elfv2:/lib/ld64.so.2;:/lib/ld64.so.1}"
 #endif
 #define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
diff -Naur gcc-4.9.0/gcc/config/sparc/linux64.h gcc-4.9.0.patch/gcc/config/sparc/linux64.h
--- gcc-4.9.0/gcc/config/sparc/linux64.h	2014-01-02 23:23:26.000000000 +0100
+++ gcc-4.9.0.patch/gcc/config/sparc/linux64.h	2014-04-22 20:27:05.854832558 +0200
@@ -93,7 +93,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -201,7 +201,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
