Index: gcc-4.4+svnr145550/gcc/incpath.c
===================================================================
--- gcc-4.4+svnr145550.orig/gcc/incpath.c	2009-04-04 13:48:31.000000000 -0700
+++ gcc-4.4+svnr145550/gcc/incpath.c	2009-04-04 14:49:29.000000000 -0700
@@ -417,6 +417,26 @@
   p->construct = 0;
   p->user_supplied_p = user_supplied_p;
 
+    /* A common error when cross compiling is including
+    host headers. This code below will try to fail fast
+    for cross compiling. Currently we consider /usr/include,
+    /opt/include and /sw/include as harmful.  */
+
+	{
+		/* printf("Adding Path: %s\n", p->name ); */
+		if( strstr(p->name, "/usr/include" ) == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /usr/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		} else if( strstr(p->name, "/sw/include") == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /sw/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		} else if( strstr(p->name, "/opt/include") == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /opt/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		 }
+	}
+
+
   add_cpp_dir_path (p, chain);
 }
 
