diff -ur gcc-6.2.0/gcc/ubsan.c gcc-6.2.0-new/gcc/ubsan.c
--- gcc-6.2.0/gcc/ubsan.c	2016-08-12 12:57:35.000000000 -0700
+++ gcc-6.2.0-new/gcc/ubsan.c	2018-01-07 17:02:27.368403661 -0800
@@ -1469,7 +1469,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
