From aae3b1193025ccec332da4165d46502f9685aea4 Mon Sep 17 00:00:00 2001
From: "Sascha Kuehndel (InuSasha)" <dev@inusasha.de>
Date: Sat, 6 May 2017 16:08:14 +0200
Subject: [PATCH] make fallthru explicit, fixes compile error in gcc7

---
 src/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.c b/src/main.c
index f6c50fa..d0fb25a 100644
--- a/src/main.c
+++ b/src/main.c
@@ -105,6 +105,7 @@ int main(int argc,char **argv)
                 {
                     syslog(LOG_WARNING, "the highest verbosity level is -vvv\n");
                 }
+                /* FALLTHRU */
             case 'f':
                 foreground = true;
                 break;
