--- a/src/input.c	2018-01-23 17:39:35.000000000 +0000
+++ b/src/input.c	2022-07-19 13:44:12.924701772 +0000
@@ -1100,8 +1100,8 @@
                 if (device->repeat_filter == true)
                 {
 		     if (evkey_type[device->current.event_out.code] == EVENTLIRCD_EVKEY_TYPE_KEY) {
-			     time_delta = 1000000 * (device->current.event_out.time.tv_sec  - previous->event_out.time.tv_sec ) +
-			                            (device->current.event_out.time.tv_usec - previous->event_out.time.tv_usec);
+			     time_delta = 1000000 * (device->current.event_out.input_event_sec  - previous->event_out.input_event_sec ) +
+			                            (device->current.event_out.input_event_usec - previous->event_out.input_event_usec);
 			     if (((previous->repeat_count == 0) && (time_delta <  900000)) ||
 			         ((previous->repeat_count == 1) && (time_delta <  500000)) ||
 			         ((previous->repeat_count == 2) && (time_delta <  300000)) ||
@@ -1116,7 +1116,8 @@
 		     }
                 }
 		previous->repeat_count++;
-		previous->event_out.time = device->current.event_out.time;
+		previous->event_out.input_event_sec = device->current.event_out.input_event_sec;
+		previous->event_out.input_event_usec = device->current.event_out.input_event_usec;
 		device->current.repeat_count = previous->repeat_count;
 		break;
 	/*
