diff -Naur atk-2.20.0/atk/atkobject.c atk-2.20.0.patch/atk/atkobject.c
--- atk-2.20.0/atk/atkobject.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atkobject.c	2016-03-26 01:10:01.058238623 +0100
@@ -26,7 +26,6 @@
 #include <glib/gi18n-lib.h>
 
 #include "atk.h"
-#include "atkmarshal.h"
 #include "atkprivate.h"
 
 /**
@@ -610,27 +609,6 @@
                   G_TYPE_POINTER);
 
   /**
-   * AtkObject::state-change:
-   * @atkobject: the object which received the signal.
-   * @arg1: The name of the state which has changed
-   * @arg2: A boolean which indicates whether the state has been set or unset.
-   *
-   * The "state-change" signal is emitted when an object's state
-   * changes.  The detail value identifies the state type which has
-   * changed.
-   */
-  atk_object_signals[STATE_CHANGE] =
-    g_signal_new ("state_change",
-                  G_TYPE_FROM_CLASS (klass),
-                  G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED,
-                  G_STRUCT_OFFSET (AtkObjectClass, state_change),
-                  (GSignalAccumulator) NULL, NULL,
-                  atk_marshal_VOID__STRING_BOOLEAN,
-                  G_TYPE_NONE, 2,
-                  G_TYPE_STRING,
-                  G_TYPE_BOOLEAN);
-
-  /**
    * AtkObject::visible-data-changed:
    * @atkobject: the object which received the signal.
    *
diff -Naur atk-2.20.0/atk/atktable.c atk-2.20.0.patch/atk/atktable.c
--- atk-2.20.0/atk/atktable.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atktable.c	2016-03-26 01:11:35.242363570 +0100
@@ -20,7 +20,6 @@
 #include "config.h"
 
 #include "atktable.h"
-#include "atkmarshal.h"
 
 /**
  * SECTION:atktable
@@ -102,78 +101,6 @@
   if (!initialized)
     {
       /**
-       * AtkTable::row-inserted:
-       * @atktable: the object which received the signal.
-       * @arg1: The index of the first row inserted.
-       * @arg2: The number of rows inserted.
-       *
-       * The "row-inserted" signal is emitted by an object which
-       * implements the AtkTable interface when a row is inserted.
-       */
-      atk_table_signals[ROW_INSERTED] =
-	g_signal_new ("row_inserted",
-		      ATK_TYPE_TABLE,
-		      G_SIGNAL_RUN_LAST,
-		      G_STRUCT_OFFSET (AtkTableIface, row_inserted),
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT,
-		      G_TYPE_NONE,
-		      2, G_TYPE_INT, G_TYPE_INT);
-      /**
-       * AtkTable::column-inserted:
-       * @atktable: the object which received the signal.
-       * @arg1: The index of the column inserted.
-       * @arg2: The number of colums inserted.
-       *
-       * The "column-inserted" signal is emitted by an object which
-       * implements the AtkTable interface when a column is inserted.
-       */
-      atk_table_signals[COLUMN_INSERTED] =
-	g_signal_new ("column_inserted",
-		      ATK_TYPE_TABLE,
-		      G_SIGNAL_RUN_LAST,
-		      G_STRUCT_OFFSET (AtkTableIface, column_inserted),
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT,
-		      G_TYPE_NONE,
-		      2, G_TYPE_INT, G_TYPE_INT);
-      /**
-       * AtkTable::row-deleted:
-       * @atktable: the object which received the signal.
-       * @arg1: The index of the first row deleted.
-       * @arg2: The number of rows deleted.
-       *
-       * The "row-deleted" signal is emitted by an object which
-       * implements the AtkTable interface when a row is deleted.
-       */
-      atk_table_signals[ROW_DELETED] =
-	g_signal_new ("row_deleted",
-		      ATK_TYPE_TABLE,
-		      G_SIGNAL_RUN_LAST,
-		      G_STRUCT_OFFSET (AtkTableIface, row_deleted),
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT,
-		      G_TYPE_NONE,
-		      2, G_TYPE_INT, G_TYPE_INT);
-      /**
-       * AtkTable::column-deleted:
-       * @atktable: the object which received the signal.
-       * @arg1: The index of the first column deleted.
-       * @arg2: The number of columns deleted.
-       *
-       * The "column-deleted" signal is emitted by an object which
-       * implements the AtkTable interface when a column is deleted.
-       */
-      atk_table_signals[COLUMN_DELETED] =
-	g_signal_new ("column_deleted",
-		      ATK_TYPE_TABLE,
-		      G_SIGNAL_RUN_LAST,
-		      G_STRUCT_OFFSET (AtkTableIface, column_deleted),
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT,
-		      G_TYPE_NONE,
-		      2, G_TYPE_INT, G_TYPE_INT);
-      /**
        * AtkTable::row-reordered:
        * @atktable: the object which received the signal.
        *
diff -Naur atk-2.20.0/atk/atktext.c atk-2.20.0.patch/atk/atktext.c
--- atk-2.20.0/atk/atktext.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atktext.c	2016-03-26 01:12:57.099480029 +0100
@@ -20,7 +20,6 @@
 #include "config.h"
 
 #include "atk.h"
-#include "atkmarshal.h"
 
 #include <string.h>
 
@@ -190,75 +189,6 @@
       class->get_bounded_ranges = atk_text_real_get_bounded_ranges; 
 
       /**
-       * AtkText::text-changed:
-       * @atktext: the object which received the signal.
-       * @arg1: The position (character offset) of the insertion or deletion.
-       * @arg2: The length (in characters) of text inserted or deleted.
-       *
-       * The "text-changed" signal is emitted when the text of the
-       * object which implements the AtkText interface changes, This
-       * signal will have a detail which is either "insert" or
-       * "delete" which identifies whether the text change was an
-       * insertion or a deletion.
-       *
-       * Deprecated: 2.9.4: Use #AtkObject::text-insert or
-       * #AtkObject::text-remove instead.
-       */
-      atk_text_signals[TEXT_CHANGED] =
-	g_signal_new ("text_changed",
-		      ATK_TYPE_TEXT,
-		      G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED,
-		      G_STRUCT_OFFSET (AtkTextIface, text_changed), 
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT,
-		      G_TYPE_NONE,
-		      2, G_TYPE_INT, G_TYPE_INT);
-
-      /**
-       * AtkText::text-insert:
-       * @atktext: the object which received the signal.
-       * @arg1: The position (character offset) of the insertion.
-       * @arg2: The length (in characters) of text inserted.
-       * @arg3: The new text inserted
-       *
-       * The "text-insert" signal is emitted when a new text is
-       * inserted. If the signal was not triggered by the user
-       * (e.g. typing or pasting text), the "system" detail should be
-       * included.
-       */
-      atk_text_signals[TEXT_INSERT] =
-	g_signal_new ("text_insert",
-		      ATK_TYPE_TEXT,
-		      G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED,
-		      0,
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT_STRING,
-		      G_TYPE_NONE,
-		      3, G_TYPE_INT, G_TYPE_INT, G_TYPE_STRING);
-
-      /**
-       * AtkText::text-remove:
-       * @atktext: the object which received the signal.
-       * @arg1: The position (character offset) of the removal.
-       * @arg2: The length (in characters) of text removed.
-       * @arg3: The old text removed
-       *
-       * The "text-remove" signal is emitted when a new text is
-       * removed. If the signal was not triggered by the user
-       * (e.g. typing or pasting text), the "system" detail should be
-       * included.
-       */
-      atk_text_signals[TEXT_REMOVE] =
-	g_signal_new ("text_remove",
-		      ATK_TYPE_TEXT,
-		      G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED,
-		      0,
-		      (GSignalAccumulator) NULL, NULL,
-		      atk_marshal_VOID__INT_INT_STRING,
-		      G_TYPE_NONE,
-		      3, G_TYPE_INT, G_TYPE_INT, G_TYPE_STRING);
-
-      /**
        * AtkText::text-caret-moved:
        * @atktext: the object which received the signal.
        * @arg1: The new position of the text caret.
diff -Naur atk-2.20.0/atk/atkutil.c atk-2.20.0.patch/atk/atkutil.c
--- atk-2.20.0/atk/atkutil.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atkutil.c	2016-03-26 01:13:04.481490538 +0100
@@ -20,7 +20,6 @@
 #include "config.h"
 
 #include "atkutil.h"
-#include "atkmarshal.c"
 
 /**
  * SECTION:atkutil
diff -Naur atk-2.20.0/atk/atkvalue.c atk-2.20.0.patch/atk/atkvalue.c
--- atk-2.20.0/atk/atkvalue.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atkvalue.c	2016-03-26 01:18:03.268897883 +0100
@@ -23,7 +23,6 @@
 #include <glib/gi18n-lib.h>
 
 #include "atkvalue.h"
-#include "atkmarshal.h"
 #include "atk-enum-types.h"
 #include "atkprivate.h"
 
@@ -322,42 +321,6 @@
 atk_value_base_init (AtkValueIface *class)
 {
   static gboolean initialized = FALSE;
-  if (!initialized)
-    {
-      /**
-       * AtkValue::value-changed:
-       * @atkvalue: the object on which the signal was emitted.
-       * @value: the new value in a numerical form.
-       * @text: human readable text alternative (also called
-       * description) of this object. NULL if not available.
-       *
-       * The 'value-changed' signal is emitted when the current value
-       * that represent the object changes. @value is the numerical
-       * representation of this new value.  @text is the human
-       * readable text alternative of @value, and can be NULL if it is
-       * not available. Note that if there is a textual description
-       * associated with the new numeric value, that description
-       * should be included regardless of whether or not it has also
-       * changed.
-       *
-       * Example: a password meter whose value changes as the user
-       * types their new password. Appropiate value text would be
-       * "weak", "acceptable" and "strong".
-       *
-       * Since: 2.12
-       */
-      atk_value_signals[VALUE_CHANGED] =
-        g_signal_new ("value_changed",
-                      ATK_TYPE_VALUE,
-                      G_SIGNAL_RUN_LAST,
-                      0,
-                      (GSignalAccumulator) NULL, NULL,
-                      atk_marshal_VOID__DOUBLE_STRING,
-                      G_TYPE_NONE,
-                      2, G_TYPE_DOUBLE, G_TYPE_STRING);
-
-      initialized = TRUE;
-    }
 }
 
 /**
diff -Naur atk-2.20.0/atk/atkwindow.c atk-2.20.0.patch/atk/atkwindow.c
--- atk-2.20.0/atk/atkwindow.c	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/atkwindow.c	2016-03-26 01:13:13.448503294 +0100
@@ -20,7 +20,6 @@
 #include "config.h"
 
 #include "atkwindow.h"
-#include "atkmarshal.h"
 
 /**
  * SECTION:atkwindow
diff -Naur atk-2.20.0/atk/Makefile.am atk-2.20.0.patch/atk/Makefile.am
--- atk-2.20.0/atk/Makefile.am	2016-02-15 12:29:12.000000000 +0100
+++ atk-2.20.0.patch/atk/Makefile.am	2016-03-26 01:08:29.075118340 +0100
@@ -115,50 +115,7 @@
 
 # ---------- Handle built sources ----------
 
-BUILT_SOURCES = atkmarshal.h atkmarshal.c atk-enum-types.h atk-enum-types.c
-
-atkmarshal.h: stamp-atkmarshal.h
-	@true
-stamp-atkmarshal.h: @REBUILD@ atkmarshal.list
-	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=atk_marshal $(srcdir)/atkmarshal.list --header >> xgen-gmh \
-	&& (cmp -s xgen-gmh atkmarshal.h || cp xgen-gmh atkmarshal.h) \
-	&& rm -f xgen-gmh xgen-gmh~					\
-	&& echo timestamp > $(@F)
-
-atkmarshal.c: stamp-atkmarshal.c
-	@true
-stamp-atkmarshal.c: @REBUILD@ atkmarshal.list
-	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=atk_marshal $(srcdir)/atkmarshal.list --body >> xgen-gmc \
-	&& (cmp -s xgen-gmc atkmarshal.c || cp xgen-gmc atkmarshal.c) \
-	&& rm -f xgen-gmc xgen-gmc~					\
-	&& echo timestamp > $(@F)
-
-atk-enum-types.h: s-enum-types-h
-	@true
-s-enum-types-h: @REBUILD@ $(atk_headers) Makefile
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			--fhead "#if defined(ATK_DISABLE_SINGLE_INCLUDES) && !defined (__ATK_H_INSIDE__) && !defined (ATK_COMPILATION)\n#error \"Only <atk/atk.h> can be included directly.\"\n#endif\n\n#ifndef __ATK_ENUM_TYPES_H__\n#define __ATK_ENUM_TYPES_H__\n\n#include <atk/atkversion.h>\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
-			--fprod "/* enumerations from \"@filename@\" */\n" \
-			--vhead "ATK_AVAILABLE_IN_ALL\nGType @enum_name@_get_type (void);\n#define ATK_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n"         \
-			--ftail "G_END_DECLS\n\n#endif /* __ATK_ENUM_TYPES_H__ */" \
-		$(atk_headers) ) > tmp-atk-enum-types.h \
-	&& (cmp -s tmp-atk-enum-types.h atk-enum-types.h || cp tmp-atk-enum-types.h atk-enum-types.h ) \
-	&& rm -f tmp-atk-enum-types.h \
-	&& echo timestamp > $(@F)
-
-atk-enum-types.c: s-enum-types-c
-	@true
-s-enum-types-c: @REBUILD@ $(atk_headers) Makefile
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			--fhead "#include \"config.h\"\n\n#include <atk.h>" \
-			--fprod "\n/* enumerations from \"@filename@\" */" \
-			--vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {"     \
-			--vprod "      { @VALUENAME@, \"@VALUENAME@\", \"@valuenick@\" }," \
-			--vtail "      { 0, NULL, NULL }\n    };\n    etype = g_@type@_register_static (\"@EnumName@\", values);\n  }\n  return etype;\n}\n" \
-		$(atk_headers) ) > tmp-atk-enum-types.c \
-	&& (cmp -s tmp-atk-enum-types.c atk-enum-types.c || cp tmp-atk-enum-types.c atk-enum-types.c ) \
-	&& rm -f tmp-atk-enum-types.c \
-	&& echo timestamp > $(@F)
+BUILT_SOURCES = atk-enum-types.h atk-enum-types.c
 
 
 if HAVE_INTROSPECTION
