diff -Naur mpfr-3.1.0/configure.ac mpfr-3.1.0.patch/configure.ac
--- mpfr-3.1.0/configure.ac	2011-10-03 10:17:15.000000000 +0200
+++ mpfr-3.1.0.patch/configure.ac	2012-06-13 11:44:03.396985843 +0200
@@ -301,7 +301,6 @@
 fi
 
 AM_PROG_CC_C_O
-AM_C_PROTOTYPES
 
 case $host in
   *-apple-darwin*)
diff -Naur mpfr-3.1.0/tests/Makefile.am mpfr-3.1.0.patch/tests/Makefile.am
--- mpfr-3.1.0/tests/Makefile.am	2011-10-03 10:17:14.000000000 +0200
+++ mpfr-3.1.0.patch/tests/Makefile.am	2012-06-13 12:47:50.874799018 +0200
@@ -9,7 +9,7 @@
 # PARTICULAR PURPOSE.
 
 
-AUTOMAKE_OPTIONS = 1.6 gnu $(top_builddir)/ansi2knr
+AUTOMAKE_OPTIONS = 1.6 gnu
 
 # tversion is run
 #   * at the beginning so that the user gets a possible version error
