diff -Naurp linux-amlogic-3.10-c954850/arch/arm/Makefile linux-amlogic-3.10-c954850-patch/arch/arm/Makefile
--- linux-amlogic-3.10-c954850/arch/arm/Makefile	2016-02-27 02:12:06.000000000 +0200
+++ linux-amlogic-3.10-c954850-patch/arch/arm/Makefile	2016-09-18 10:07:59.000000000 +0300
@@ -99,6 +99,7 @@ tune-$(CONFIG_CPU_FEROCEON)	:=$(call cc-
 tune-$(CONFIG_CPU_V6)		:=$(call cc-option,-mtune=arm1136j-s,-mtune=strongarm)
 tune-$(CONFIG_CPU_V6K)		:=$(call cc-option,-mtune=arm1136j-s,-mtune=strongarm)
 tune-$(CONFIG_ARCH_MESON8B)	:=-mtune=cortex-a5 -mfloat-abi=softfp -mfpu=neon-vfpv4
+tune-$(CONFIG_ARCH_MESON8)	:=-mtune=cortex-a9 -mfloat-abi=softfp -mfpu=neon-fp16
 
 ifeq ($(CONFIG_AEABI),y)
 CFLAGS_ABI	:=-mabi=aapcs-linux -mno-thumb-interwork
