diff -Naurp linux-fb5b898/drivers/amlogic/amports/vh264_4k2k.c linux-fb5b898-patch/drivers/amlogic/amports/vh264_4k2k.c
--- linux-fb5b898/drivers/amlogic/amports/vh264_4k2k.c	2016-12-15 18:49:08.000000000 +0200
+++ linux-fb5b898-patch/drivers/amlogic/amports/vh264_4k2k.c	2016-12-21 10:11:33.000000000 +0200
@@ -97,7 +97,7 @@ static u32 frame_width, frame_height, fr
 static struct timer_list recycle_timer;
 static u32 stat;
 static u32 error_watchdog_count;
-static uint error_recovery_mode = 0;
+static uint error_recovery_mode = 1;
 static u32 sync_outside;
 static u32 vh264_4k2k_rotation;
 static u32 first_i_recieved;
diff -Naurp linux-fb5b898/drivers/amlogic/amports/vh264.c linux-fb5b898-patch/drivers/amlogic/amports/vh264.c
--- linux-fb5b898/drivers/amlogic/amports/vh264.c	2016-12-15 18:49:08.000000000 +0200
+++ linux-fb5b898-patch/drivers/amlogic/amports/vh264.c	2016-12-21 10:11:07.000000000 +0200
@@ -237,7 +237,7 @@ static struct vframe_s *p_last_vf;
 static u32 last_pts, last_pts_remainder;
 static bool check_pts_discontinue;
 static u32 wait_buffer_counter;
-static uint error_recovery_mode = 0;
+static uint error_recovery_mode = 1;
 static uint error_recovery_mode_in = 3;
 static uint error_recovery_mode_use = 3;
 
