diff -Naur linux-5df008c.orig/arch/arm/boot/dts/amlogic/meson8_m8_2G.dtd linux-5df008c/arch/arm/boot/dts/amlogic/meson8_m8_2G.dtd
--- linux-5df008c.orig/arch/arm/boot/dts/amlogic/meson8_m8_2G.dtd	2016-07-13 14:09:54.058314295 +0200
+++ linux-5df008c/arch/arm/boot/dts/amlogic/meson8_m8_2G.dtd	2016-07-13 14:15:10.000000000 +0200
@@ -133,11 +133,11 @@
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x01800000  0x00100000>;
+		reserve-memory = <0x04800000 0x00100000>;
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
 		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
