diff -Naur linux-83ba060.orig/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd linux-83ba060/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd
--- linux-83ba060.orig/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd	2016-06-11 15:59:55.210940179 +0200
+++ linux-83ba060/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd	2016-06-11 16:04:36.000000000 +0200
@@ -133,11 +133,11 @@
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x01800000  0x00100000>;
+    reserve-memory = <0x04800000 0x00100000>;
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
 		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
