--- a/arch/arm/boot/dts/amlogic/minix_x8h.dtd	2017-08-17 11:21:25.000000000 +0200
+++ b/arch/arm/boot/dts/amlogic/minix_x8h.dtd	2017-08-17 00:14:49.000000000 +0200
@@ -0,0 +1,1101 @@
+/dts-v1/;
+
+/ {
+	amlogic-dt-id = "m8_k200_2gsdhc";
+	compatible = "AMLOGIC,8726_M8";
+	model = "AMLOGIC";
+	interrupt-parent = <0x1>;
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+
+	cpus {
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <0x200>;
+		};
+
+		cpu@1 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <0x1>;
+		};
+
+		cpu@2 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <0x2>;
+		};
+
+		cpu@3 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a9";
+			reg = <0x3>;
+		};
+	};
+
+	cache-controller {
+		compatible = "arm,meson-pl310-cache";
+		reg = <0xc4200000 0x1000>;
+		arm,data-latency = <0x3 0x3 0x3>;
+		arm,tag-latency = <0x2 0x2 0x2>;
+		arm,filter-ranges = <0x100000 0xc0000000>;
+		cache-unified;
+		cache-level = <0x2>;
+		aux-instruction_prefetch;
+		aux-data_prefetch;
+		aux-ns_lockdown;
+		aux-force_no_write_alloc;
+		aux-cache_replace_policy_round_robin;
+		aux-early_write_response;
+		aux-full_line_of_zero;
+		aux-ns_int_ctrl;
+		aux-share_override;
+		prefetch-double_line_fill;
+		prefetch-prefetch_drop;
+		prefetch-prefetch_offset = <0x7>;
+	};
+
+	memory {
+		device_type = "memory";
+		aml_reserved_start = <0x6000000>;
+		aml_reserved_end = <0x5000000>;
+		phys_offset = <0x200000>;
+		linux,total-memory = <0x7fe00000>;
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+
+		region@0 {
+			region_name = "cma_0";
+			reg = <0x0 0x1200000>;
+			linux,contiguous-region;
+			linux,phandle = <0x2>;
+			phandle = <0x2>;
+		};
+	};
+
+	interrupt-controller {
+		compatible = "arm,cortex-a9-gic";
+		reg = <0xc4301000 0x1000 0xc4300100 0x100>;
+		interrupt-controller;
+		#interrupt-cells = <0x3>;
+		#address-cells = <0x0>;
+		linux,phandle = <0x1>;
+		phandle = <0x1>;
+	};
+
+	vpu {
+		compatible = "amlogic,vpu";
+		dev_name = "vpu";
+		status = "ok";
+		clk_level = <0x7>;
+	};
+
+	mesonfb {
+		compatible = "amlogic,mesonfb";
+		dev_name = "mesonfb";
+		status = "okay";
+		reserve-memory = <0x1800000 0x100000>;
+		reserve-iomap = "true";
+		vmode = <0x3>;
+		scale_mode = <0x1>;
+		4k2k_fb = <0x0>;
+		display_size_default = <0x780 0x438 0x780 0xca8 0x20>;
+	};
+
+	deinterlace {
+		compatible = "amlogic,deinterlace";
+		dev_name = "deinterlace";
+		status = "ok";
+		reserve-memory = <0x2100000>;
+	};
+
+	mesonstream {
+		compatible = "amlogic,mesonstream";
+		dev_name = "mesonstream.0";
+		status = "okay";
+		reserve-memory = <0x2000000>;
+		reserve-iomap = "true";
+	};
+
+	vdec {
+		compatible = "amlogic,vdec";
+		dev_name = "vdec.0";
+		status = "okay";
+		reserve-memory = <0x10000000>;
+		reserve-iomap = "true";
+	};
+
+	picdec {
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";
+		status = "okay";
+		share-memory-name = "vdec0";
+		share-memory-offset = <0x0>;
+		share-memory-size = <0x4000000>;
+	};
+
+	ppmgr {
+		compatible = "amlogic,ppmgr";
+		dev_name = "ppmgr";
+		status = "okay";
+		reserve-memory = <0x1000000>;
+	};
+
+	amvenc_avc {
+		compatible = "amlogic,amvenc_avc";
+		dev_name = "amvenc_avc.0";
+		status = "okay";
+		linux,contiguous-region = <0x2>;
+	};
+
+	vdin1 {
+		compatible = "amlogic,vdin";
+		dev_name = "vdin1";
+		status = "ok";
+		reserve-memory = <0x1000000>;
+		miracast_size = <0x780 0x438>;
+		reserve-iomap = "true";
+		irq = <0x75>;
+		vdin_id = <0x1>;
+	};
+
+	amlvideo2 {
+		compatible = "amlogic,amlvideo2";
+		dev_name = "amlvideo2.0";
+		status = "okay";
+		reserve-memory = <0x1800000>;
+		reserve-iomap = "true";
+	};
+
+	ion_dev {
+		compatible = "amlogic,ion_dev";
+		dev_name = "ion_dev";
+		status = "ok";
+		share-memory-name = "ppmgr0";
+		share-memory-offset = <0x0>;
+		share-memory-size = <0x2000000>;
+	};
+
+	mesonvout {
+		compatible = "amlogic,mesonvout";
+		dev_name = "mesonvout";
+		status = "okay";
+	};
+
+	rtc {
+		compatible = "amlogic,aml_rtc";
+		status = "disabled";
+	};
+
+	uart_ao {
+		compatible = "amlogic,aml_uart";
+		port_name = "uart_ao";
+		status = "okay";
+		dev_name = "uart_ao";
+		pinctrl-names = "default";
+		pinctrl-0 = <0x3>;
+	};
+
+	uart_0 {
+		compatible = "amlogic,aml_uart";
+		port_name = "uart_a";
+		status = "okay";
+		dev_name = "uart_0";
+		pinctrl-names = "default";
+		pinctrl-0 = <0x4>;
+	};
+
+	uart_1 {
+		compatible = "amlogic,aml_uart";
+		port_name = "uart_b";
+		status = "disabled";
+		dev_name = "uart_1";
+	};
+
+	uart_2 {
+		compatible = "amlogic,aml_uart";
+		port_name = "uart_c";
+		status = "disabled";
+		dev_name = "uart_2";
+	};
+
+	uart_3 {
+		compatible = "amlogic,aml_uart";
+		port_name = "uart_d";
+		status = "ok";
+		dev_name = "uart_3";
+	};
+
+	bt-dev {
+		compatible = "amlogic,bt-dev";
+		dev_name = "bt-dev";
+		status = "ok";
+		gpio_reset = "GPIOX_20";
+	};
+
+	wifi {
+		compatible = "amlogic,aml_broadcm_wifi";
+		dev_name = "aml_broadcm_wifi";
+		status = "okay";
+		interrupt_pin = "GPIOX_21";
+		irq_num = <0x4>;
+		irq_trigger_type = "GPIO_IRQ_HIGH";
+		power_on_pin = "GPIOAO_6";
+		power_on_pin2 = "GPIOX_11";
+		clock_32k_pin = "GPIOX_10";
+	};
+
+	wifi_power {
+		compatible = "amlogic,wifi_power";
+		dev_name = "wifi_power";
+		status = "okay";
+		power_gpio = "GPIOAO_6";
+		power_gpio2 = "GPIOX_11";
+	};
+
+	sdio {
+		compatible = "amlogic,aml_sdio";
+		dev_name = "aml_sdio.0";
+		status = "okay";
+		reg = <0xc1108c20 0x20>;
+		pinctrl-names = "sd_clk_cmd_pins", "sd_all_pins", "emmc_clk_cmd_pins", "emmc_all_pins", "sdio_clk_cmd_pins", "sdio_all_pins", "sd_1bit_pins";
+		pinctrl-0 = <0x5>;
+		pinctrl-1 = <0x6>;
+		pinctrl-2 = <0x7>;
+		pinctrl-3 = <0x8>;
+		pinctrl-4 = <0x9>;
+		pinctrl-5 = <0xa>;
+		pinctrl-6 = <0xb>;
+
+		sd {
+			status = "okay";
+			port = <0x1>;
+			pinname = "sd";
+			ocr_avail = <0x200000>;
+			caps = "MMC_CAP_4_BIT_DATA", "MMC_CAP_MMC_HIGHSPEED", "MMC_CAP_SD_HIGHSPEED";
+			f_min = <0x493e0>;
+			f_max = <0x2faf080>;
+			max_req_size = <0x20000>;
+			gpio_dat3 = "CARD_4";
+			jtag_pin = "CARD_0";
+			gpio_cd = "CARD_6";
+			gpio_ro = "GPIOZ_0";
+			irq_in = <0x3>;
+			irq_out = <0x5>;
+			card_type = <0x5>;
+		};
+
+		sdio {
+			status = "ok";
+			port = <0x0>;
+			pinname = "sdio";
+			ocr_avail = <0x200000>;
+			caps = "MMC_CAP_4_BIT_DATA", "MMC_CAP_MMC_HIGHSPEED", "MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE";
+			f_min = <0x493e0>;
+			f_max = <0x2faf080>;
+			max_req_size = <0x20000>;
+			card_type = <0x3>;
+		};
+	};
+
+	sdhc {
+		compatible = "amlogic,aml_sdhc";
+		dev_name = "aml_sdhc.0";
+		reg = <0xc1108e00 0x3c>;
+		pinctrl-names = "sdhc_sd_clk_cmd_pins", "sdhc_sd_all_pins", "sdhc_emmc_clk_cmd_pins", "sdhc_emmc_all_pins", "sdhc_sdio_clk_cmd_pins", "sdhc_sdio_all_pins";
+		pinctrl-0 = <0xc>;
+		pinctrl-1 = <0xd>;
+		pinctrl-2 = <0xe>;
+		pinctrl-3 = <0xf>;
+		pinctrl-4 = <0x10>;
+		pinctrl-5 = <0x11>;
+
+		emmc {
+			status = "okay";
+			port = <0x5>;
+			pinname = "emmc";
+			ocr_avail = <0x200000>;
+			caps = "MMC_CAP_8_BIT_DATA", "MMC_CAP_MMC_HIGHSPEED", "MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE", "MMC_CAP_ERASE", "MMC_CAP_HW_RESET";
+			f_min = <0x493e0>;
+			f_max = <0x5f5e100>;
+			max_req_size = <0x20000>;
+			gpio_dat3 = "BOOT_3";
+			card_type = <0x1>;
+		};
+	};
+
+	i2c@c8100500 {
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-AO";
+		status = "ok";
+		reg = <0xc8100500 0x1d>;
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		device_id = <0x0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x12>;
+		use_pio = <0x0>;
+		master_i2c_speed = <0x61a80>;
+	};
+
+	i2c@c1108500 {
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-A";
+		status = "ok";
+		reg = <0xc1108500 0x20>;
+		device_id = <0x1>;
+		pinctrl-names = "default";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		use_pio = <0x0>;
+		master_i2c_speed = <0x493e0>;
+	};
+
+	i2c@c11087c0 {
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-B";
+		status = "ok";
+		reg = <0xc11087c0 0x20>;
+		device_id = <0x2>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x13>;
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		use_pio = <0x0>;
+		master_i2c_speed = <0x493e0>;
+	};
+
+	i2c@c11087e0 {
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-C";
+		status = "ok";
+		reg = <0xc11087e0 0x20>;
+		device_id = <0x3>;
+		pinctrl-names = "default";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		use_pio = <0x0>;
+		master_i2c_speed = <0x493e0>;
+	};
+
+	i2c@c1108d20 {
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-D";
+		status = "ok";
+		reg = <0xc1108d20 0x20>;
+		device_id = <0x4>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x14>;
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+	};
+
+	dvfs {
+		compatible = "amlogic, amlogic-dvfs";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		status = "ok";
+
+		vcck_dvfs {
+			dvfs_id = <0x1>;
+			table_count = <0xe>;
+			dvfs_table = <0x17700 0xc96a8 0xc96a8 0x2ee00 0xc96a8 0xc96a8 0x4c2c0 0xc96a8 0xc96a8 0x639c0 0xc96a8 0xc96a8 0x7b0c0 0xc96a8 0xc96a8 0x927c0 0xcf850 0xcf850 0xafc80 0xcf850 0xcf850 0xc7380 0xd59f8 0xd59f8 0xf6180 0xe1d48 0xe1d48 0x124f80 0xee098 0xee098 0x159b40 0xfa3e8 0xfa3e8 0x188940 0x10c8e0 0x10c8e0 0x1b7740 0x112a88 0x112a88 0x1e6540 0x118c30 0x118c30>;
+		};
+	};
+
+	arm_pmu {
+		compatible = "arm,cortex-a9-pmu";
+		status = "ok";
+		interrupts = <0x0 0x89 0x4 0x0 0x8a 0x4 0x0 0x99 0x4 0x0 0x9a 0x4>;
+	};
+
+	pmu {
+		compatible = "amlogic, aml_pmu_prober";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+
+		rn5t618 {
+			compatible = "ricoh_pmu";
+			sub_type = "rn5t618";
+			i2c_bus = "i2c_bus_ao";
+			reg = <0x32>;
+			soft_limit_to99 = <0x0>;
+			ddr_voltage = <0x16e360>;
+			vddao_voltage = <0x124f80>;
+			status = "okay";
+		};
+
+		rtc8563 {
+			compatible = "rtc8563";
+			i2c_bus = "i2c_bus_ao";
+			reg = <0x51>;
+			gpio_interrupt = "GPIOX_12";
+			status = "ok";
+		};
+	};
+
+	usb_con {
+		lm-compatible = "logicmodule-bus";
+
+		usb_b {
+			lm-compatible = "amlogic,usb";
+			lm-periph-id = <0x1>;
+			clock-src = "usb1";
+			port-id = <0x1>;
+			port-type = <0x1>;
+			port-speed = <0x0>;
+			port-config = <0x0>;
+			port-dma = <0x0>;
+			port-id-mode = <0x1>;
+			status = "okay";
+		};
+
+		usb_a {
+			lm-compatible = "amlogic,usb";
+			lm-periph-id = <0x0>;
+			clock-src = "usb0";
+			port-id = <0x0>;
+			port-type = <0x0>;
+			port-speed = <0x0>;
+			port-config = <0x0>;
+			port-dma = <0x0>;
+			port-id-mode = <0x0>;
+			gpio-vbus-power = "GPIOZ_1";
+			gpio-work-mask = <0x1>;
+			status = "okay";
+		};
+	};
+
+	spi@cc000000 {
+		compatible = "amlogic,apollo_spi_nor";
+		status = "ok";
+		reg = <0xcc000000 0x4000000>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x15>;
+		nr-parts = <0x2>;
+		nr-part-0 = <0x16>;
+		nr-part-1 = <0x17>;
+
+		bootloader {
+			offset = <0x0>;
+			size = <0x100000>;
+			linux,phandle = <0x16>;
+			phandle = <0x16>;
+		};
+
+		ubootenv {
+			offset = <0x100000>;
+			size = <0x10000>;
+			linux,phandle = <0x17>;
+			phandle = <0x17>;
+		};
+	};
+
+	nand {
+		compatible = "amlogic,aml_nand";
+		dev_name = "nand";
+		status = "ok";
+		reg = <0xd0048600 0x24>;
+		pinctrl-names = "nand_rb_mod", "nand_norb_mod";
+		pinctrl-0 = <0x18 0x19 0x1a 0x1b>;
+		pinctrl-1 = <0x18 0x19 0x1a 0x1b>;
+		device_id = <0x0>;
+		plat-names = "nandnormal";
+		plat-num = <0x1>;
+		plat-part-0 = <0x1c>;
+
+		normal {
+			enable_pad = "ce0", "ce1", "ce2", "ce3";
+			busy_pad = "rb0";
+			linux,phandle = <0x1c>;
+			phandle = <0x1c>;
+		};
+	};
+
+	efuse {
+		compatible = "amlogic,efuse";
+		dev_name = "efuse";
+		status = "okay";
+		plat-pos = <0x0 0x1c6>;
+		plat-count = <0x3a>;
+		usid-min = <0x8>;
+		usid-max = <0x1f>;
+	};
+
+	thermal {
+		compatible = "amlogic-thermal";
+		#thermal-cells = <0x7>;
+		dev_name = "aml_thermal";
+		trip_point = <0x46 0x188941 0x188941 0x1ff 0x1ff 0x3 0x5 0x50 0x124f81 0x124f81 0x1b3 0x1b3 0x2 0x4 0x5a 0xc3501 0xc3501 0x148 0x148 0x2 0x3 0x104 0xffffffff 0xffffffff 0xffffffff 0xffffffff 0xffffffff 0xffffffff>;
+		cpu_cali_a = <0x0>;
+		idle_interval = <0x3e8>;
+		use_virtual_thermal;
+		freq_sample_period = <0x1e>;
+		report_time = <0x1 0xa 0x1e 0x3c>;
+		cpu_virtual = <0x7a120 0x28 0x28 0x28 0x28 0x927c0 0x2d 0x37 0x3c 0x41 0xc3500 0x32 0x3c 0x41 0x46 0xf4240 0x37 0x41 0x46 0x4b 0x124f80 0x3c 0x46 0x4b 0x50 0x155cc0 0x41 0x4b 0x50 0x55>;
+		gpu_virtual = <0xb7 0x28 0x28 0x28 0x28 0xfb 0x28 0x2d 0x32 0x37 0x13f 0x32 0x3c 0x41 0x46 0x1aa 0x37 0x41 0x46 0x4b 0x1ff 0x3c 0x46 0x4b 0x50>;
+	};
+
+	audio_platform {
+		compatible = "amlogic,aml-i2s";
+		dev_name = "aml-i2s.0";
+		status = "okay";
+	};
+
+	audio_dai {
+		compatible = "amlogic,aml-i2s-dai";
+		dev_name = "aml-i2s-dai.0";
+		status = "okay";
+	};
+
+	audio_spdif_dai {
+		compatible = "amlogic,aml-spdif-dai";
+		dev_name = "aml-spdif-dai.0";
+		status = "okay";
+	};
+
+	audio_spdif_codec {
+		compatible = "amlogic,aml-spdif-codec";
+		dev_name = "spdif-dit.0";
+		pinctrl-names = "aml_audio_spdif";
+		pinctrl-0 = <0x1d>;
+		status = "okay";
+	};
+
+	audio_pcm2BT_codec {
+		compatible = "amlogic,pcm2BT-codec";
+		dev_name = "pcm2bt.0";
+		status = "okay";
+	};
+
+	audio_pcm_dai {
+		compatible = "amlogic,aml-pcm-dai";
+		dev_name = "aml-pcm-dai.0";
+		status = "okay";
+	};
+
+	audio_pcm {
+		compatible = "amlogic,aml-pcm";
+		dev_name = "aml-pcm.0";
+		status = "okay";
+	};
+
+	audio_m8_audio {
+		compatible = "amlogic,m8_audio_codec";
+		dev_name = "aml_m8_codec.0";
+		status = "okay";
+	};
+
+	dummy_codec {
+		compatible = "amlogic,aml_dummy_codec";
+		dev_name = "dummy_codec.0";
+		status = "okay";
+	};
+
+	aml_audio_codec {
+		compatible = "amlogic,audio_codec";
+
+		es8323 {
+			codec_name = "es8323";
+			i2c_addr = <0x10>;
+			i2c_bus = "i2c_bus_d";
+			id_reg = <0x0>;
+			id_val = <0x6>;
+			capless = <0x0>;
+			status = "okay";
+		};
+	};
+
+	aml_m8_sound_card {
+		compatible = "sound_card, aml_snd_m8";
+		aml,sound_card = "AML-M8AUDIO";
+		aml,codec_dai = "AML-M8", "es8323-hifi";
+		aml,audio-routing-rt5616 = "Ext Spk", "LOUTL", "Ext Spk", "LOUTR", "HP", "HPOL", "HP", "HPOR", "micbias1", "MAIN MIC", "IN2P", "micbias1";
+		aml,audio-routing-amlm8 = "Ext Spk", "LINEOUTL", "Ext Spk", "LINEOUTR", "HP", "HP_L", "HP", "HP_R", "MICBIAS", "MAIN MIC", "LINPUT1", "MICBIAS";
+		aml,audio-routing-dummy = "Ext Spk", "LOUTL", "Ext Spk", "LOUTR";
+		aml,audio-routing-es8323 = "Ext Spk", "LOUTL", "Ext Spk", "LOUTR", "HP", "HPOL", "HP", "HPOR";
+		mute_gpio = "GPIO_BSD_EN";
+		hp_disable;
+		hp_paraments = <0x320 0x12c 0x0 0x5 0x1>;
+		pinctrl-names = "aml_snd_m8";
+		pinctrl-0 = <0x1e>;
+		status = "okay";
+	};
+
+	gpio_keypad {
+		compatible = "amlogic,gpio_keypad";
+		status = "okay";
+		scan_period = <0x14>;
+		key_num = <0x1>;
+		key_name = "power";
+		key_code = <0x74>;
+		key_pin = "GPIOAO_3";
+		irq_keyup = <0x6>;
+		irq_keydown = <0x7>;
+	};
+
+	saradc {
+		compatible = "amlogic,saradc";
+		status = "okay";
+	};
+
+	meson-remote {
+		compatible = "amlogic,aml_remote";
+		dev_name = "meson-remote";
+		status = "ok";
+		ao_baseaddr = <0xf3100480>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x1f>;
+	};
+
+	securitykey {
+		compatible = "amlogic,aml_keys";
+		status = "ok";
+	};
+
+	unifykey {
+		compatible = "amlogic,unifykey";
+		status = "ok";
+		unifykey-num = <0x6>;
+		unifykey-index-0 = <0x20>;
+		unifykey-index-1 = <0x21>;
+		unifykey-index-2 = <0x22>;
+		unifykey-index-3 = <0x23>;
+		unifykey-index-4 = <0x24>;
+		unifykey-index-5 = <0x25>;
+
+		key_0 {
+			key-name = "usid";
+			key-device = "nandkey";
+			key-dataformat = "allascii";
+			key-permit = "read", "write", "del";
+			linux,phandle = <0x20>;
+			phandle = <0x20>;
+		};
+
+		key_1 {
+			key-name = "mac";
+			key-device = "nandkey";
+			key-dataformat = "hexdata";
+			key-permit = "read", "write", "del";
+			linux,phandle = <0x21>;
+			phandle = <0x21>;
+		};
+
+		key_2 {
+			key-name = "hdcp";
+			key-device = "nandkey";
+			key-dataformat = "hexdata";
+			key-permit = "read", "write", "del";
+			linux,phandle = <0x22>;
+			phandle = <0x22>;
+		};
+
+		key_3 {
+			key-name = "secure_boot_set";
+			key-device = "efusekey";
+			key-dataformat = "hexdata";
+			key-permit = "read", "write";
+			linux,phandle = <0x23>;
+			phandle = <0x23>;
+		};
+
+		key_4 {
+			key-name = "mac_bt";
+			key-device = "nandkey";
+			key-dataformat = "hexdata";
+			key-permit = "read", "write", "del";
+			linux,phandle = <0x24>;
+			phandle = <0x24>;
+		};
+
+		key_5 {
+			key-name = "mac_wifi";
+			key-device = "nandkey";
+			key-dataformat = "hexdata";
+			key-permit = "read", "write", "del";
+			linux,phandle = <0x25>;
+			phandle = <0x25>;
+		};
+	};
+
+	amhdmitx {
+		compatible = "amlogic,amhdmitx";
+		dev_name = "amhdmitx";
+		status = "ok";
+		vend-data = <0x26>;
+		pwr-ctrl = <0x27>;
+
+		vend_data {
+			vendor_name = "Amlogic";
+			vendor_id = <0x0>;
+			product_desc = "X8";
+			cec_osd_string = "X8";
+			linux,phandle = <0x26>;
+			phandle = <0x26>;
+		};
+
+		pwr_ctrl {
+			pwr_5v_on = [00];
+			pwr_5v_off = [00];
+			pwr_3v3_on = [00];
+			pwr_3v3_off = [00];
+			pwr_hpll_vdd_on = [00];
+			pwr_hpll_vdd_off = [00];
+			linux,phandle = <0x27>;
+			phandle = <0x27>;
+		};
+	};
+
+	aml_pm {
+		compatible = "amlogic,pm-m8";
+		dev_name = "aml_pm_m8";
+		status = "okay";
+	};
+
+	cpufreq-meson {
+		compatible = "amlogic,cpufreq-meson";
+		status = "okay";
+	};
+
+	crypto_device {
+		compatible = "amlogic,crypto-device";
+		dev_name = "crypto_device";
+	};
+
+	gpio {
+		compatible = "amlogic,m8-gpio";
+		dev_name = "gpio";
+		#gpio-cells = <0x2>;
+	};
+
+	pinmux {
+		compatible = "amlogic,pinmux-m8";
+		dev_name = "pinmux";
+		#pinmux-cells = <0x2>;
+
+		ao_uart {
+			amlogic,setmask = <0xa 0x1800>;
+			amlogic,pins = "GPIOAO_0", "GPIOAO_1";
+			linux,phandle = <0x3>;
+			phandle = <0x3>;
+		};
+
+		a_uart {
+			amlogic,setmask = <0x4 0x3c00>;
+			amlogic,pins = "GPIOX_12", "GPIOX_13", "GPIOX_14", "GPIOX_15";
+			linux,phandle = <0x4>;
+			phandle = <0x4>;
+		};
+
+		nand_input {
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_4", "BOOT_5", "BOOT_6", "BOOT_7", "BOOT_12", "BOOT_13", "BOOT_8", "BOOT_9", "BOOT_10", "BOOT_11", "BOOT_14", "BOOT_15", "BOOT_16", "BOOT_17";
+			amlogic,enable-output = <0x1>;
+			linux,phandle = <0x18>;
+			phandle = <0x18>;
+		};
+
+		conf_nand {
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_4", "BOOT_5", "BOOT_6", "BOOT_7", "BOOT_15";
+			amlogic,pullup = <0x0>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x19>;
+			phandle = <0x19>;
+		};
+
+		conf_nand1 {
+			amlogic,pins = "BOOT_8", "BOOT_10";
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x1a>;
+			phandle = <0x1a>;
+		};
+
+		nand {
+			amlogic,setmask = <0x2 0x7fe0000>;
+			amlogic,clrmask = <0x5 0xe 0x6 0x3f000000 0x4 0x7c000000>;
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_4", "BOOT_5", "BOOT_6", "BOOT_7", "BOOT_8", "BOOT_9", "BOOT_10", "BOOT_12", "BOOT_13", "BOOT_14", "BOOT_15", "BOOT_16", "BOOT_17";
+			linux,phandle = <0x1b>;
+			phandle = <0x1b>;
+		};
+
+		sdhc_b_pin {
+			amlogic,setmask = <0x2 0xfc00>;
+			amlogic,clrmask = <0x2 0xf0 0x8 0x600>;
+			amlogic,pins = "CARD_0", "CARD_1", "CARD_2", "CARD_3", "CARD_4", "CARD_5";
+		};
+
+		sdhc_c_pin {
+			amlogic,setmask = <0x6 0x3f000000>;
+			amlogic,clrmask = <0x4 0x6c000000 0x2 0x4c00000>;
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_10", "BOOT_11";
+		};
+
+		sdhc_a_pin {
+			amlogic,setmask = <0x8 0x3f>;
+			amlogic,clrmask = <0x5 0x6c00>;
+			amlogic,pins = "GPIOX_0", "GPIOX_1", "GPIOX_2", "GPIOX_3", "GPIOX_8", "GPIOX_9";
+		};
+
+		sdio_all_pins {
+			amlogic,setmask = <0x8 0x3f>;
+			amlogic,clrmask = <0x6 0x3f000000 0x2 0xfc00 0x5 0x6c00>;
+			amlogic,pins = "GPIOX_0", "GPIOX_1", "GPIOX_2", "GPIOX_3", "GPIOX_8", "GPIOX_9";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xa>;
+			phandle = <0xa>;
+		};
+
+		sdio_clk_cmd_pins {
+			amlogic,setmask = <0x8 0x3>;
+			amlogic,clrmask = <0x6 0x3f000000 0x2 0xfc00 0x5 0x6c00>;
+			amlogic,pins = "GPIOX_8", "GPIOX_9";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x9>;
+			phandle = <0x9>;
+		};
+
+		sd_all_pins {
+			amlogic,setmask = <0x2 0xfc00>;
+			amlogic,clrmask = <0x6 0x3f000000 0x8 0x63f 0x2 0xf0>;
+			amlogic,pins = "CARD_0", "CARD_1", "CARD_2", "CARD_3", "CARD_4", "CARD_5";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x6>;
+			phandle = <0x6>;
+		};
+
+		sd_1bit_pins {
+			amlogic,setmask = <0x2 0x8c00>;
+			amlogic,clrmask = <0x6 0x3f000000 0x8 0x3f 0x2 0xf0>;
+			amlogic,pins = "CARD_0", "CARD_1", "CARD_2", "CARD_3", "CARD_4", "CARD_5";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xb>;
+			phandle = <0xb>;
+		};
+
+		sd_clk_cmd_pins {
+			amlogic,setmask = <0x2 0xc00>;
+			amlogic,clrmask = <0x6 0x3f000000 0x8 0x3f 0x2 0xf0>;
+			amlogic,pins = "CARD_2", "CARD_3";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x5>;
+			phandle = <0x5>;
+		};
+
+		emmc_all_pins {
+			amlogic,setmask = <0x6 0x3f000000>;
+			amlogic,clrmask = <0x2 0x4c0fc00 0x8 0x3f 0x4 0x6c000000>;
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_16", "BOOT_17";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x8>;
+			phandle = <0x8>;
+		};
+
+		emmc_clk_cmd_pins {
+			amlogic,setmask = <0x6 0x3000000>;
+			amlogic,clrmask = <0x2 0x4c0fc00 0x8 0x3f 0x4 0x6c000000>;
+			amlogic,pins = "BOOT_16", "BOOT_17";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x7>;
+			phandle = <0x7>;
+		};
+
+		sdhc_sd_clk_cmd_pins {
+			amlogic,setmask = <0x2 0x30>;
+			amlogic,clrmask = <0x5 0x7c00 0x4 0x7c000000 0x2 0xfc00 0x8 0x600>;
+			amlogic,pins = "CARD_2", "CARD_3";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xc>;
+			phandle = <0xc>;
+		};
+
+		sdhc_sd_all_pins {
+			amlogic,setmask = <0x2 0xf0>;
+			amlogic,clrmask = <0x5 0x7c00 0x4 0x7c000000 0x2 0xfc00 0x8 0x600>;
+			amlogic,pins = "CARD_0", "CARD_1", "CARD_2", "CARD_3", "CARD_4", "CARD_5";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xd>;
+			phandle = <0xd>;
+		};
+
+		sdhc_emmc_clk_cmd_pins {
+			amlogic,setmask = <0x4 0xc000000>;
+			amlogic,clrmask = <0x2 0x4c000f0 0x5 0x7c00 0x6 0x3f000000>;
+			amlogic,pins = "BOOT_16", "BOOT_17";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xe>;
+			phandle = <0xe>;
+		};
+
+		sdhc_emmc_all_pins {
+			amlogic,setmask = <0x4 0x7c000000>;
+			amlogic,clrmask = <0x2 0x4c000f0 0x5 0x7c00 0x6 0x3f000000>;
+			amlogic,pins = "BOOT_0", "BOOT_1", "BOOT_2", "BOOT_3", "BOOT_4", "BOOT_5", "BOOT_6", "BOOT_7", "BOOT_16", "BOOT_17";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0xf>;
+			phandle = <0xf>;
+		};
+
+		sdhc_sdio_clk_cmd_pins {
+			amlogic,setmask = <0x5 0xc00>;
+			amlogic,clrmask = <0x2 0x58000f0 0x4 0x7c000000 0x8 0x3f>;
+			amlogic,pins = "GPIOX_8", "GPIOX_9";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x10>;
+			phandle = <0x10>;
+		};
+
+		sdhc_sdio_all_pins {
+			amlogic,setmask = <0x5 0x6c00>;
+			amlogic,clrmask = <0x2 0x58000f0 0x4 0x7c000000 0x8 0x3f>;
+			amlogic,pins = "GPIOX_0", "GPIOX_1", "GPIOX_2", "GPIOX_3", "GPIOX_8", "GPIOX_9";
+			amlogic,enable-output = <0x1>;
+			amlogic,pullup = <0x1>;
+			amlogic,pullupen = <0x1>;
+			linux,phandle = <0x11>;
+			phandle = <0x11>;
+		};
+
+		ao_i2c {
+			amlogic,setmask = <0xa 0x60>;
+			amlogic,clrmask = <0xa 0x1800006>;
+			amlogic,pins = "GPIOAO_4", "GPIOAO_5";
+			linux,phandle = <0x12>;
+			phandle = <0x12>;
+		};
+
+		b_i2c {
+			amlogic,setmask = <0x5 0xc000000>;
+			amlogic,pins = "GPIOZ_2", "GPIOZ_3";
+			linux,phandle = <0x13>;
+			phandle = <0x13>;
+		};
+
+		d_i2c {
+			amlogic,setmask = <0x4 0xc>;
+			amlogic,pins = "GPIOH_7", "GPIOH_8";
+			linux,phandle = <0x14>;
+			phandle = <0x14>;
+		};
+
+		remote_pin {
+			amlogic,setmask = <0xa 0x1>;
+			amlogic,pins = "GPIOAO_7";
+			linux,phandle = <0x1f>;
+			phandle = <0x1f>;
+		};
+
+		audio_pin {
+			amlogic,setmask = <0x1 0x20 0x3 0x78000000 0xa 0x78000000>;
+			amlogic,clrmask = <0x4 0x3c000 0x5 0x1000>;
+			amlogic,pins = "GPIOY_5", "GPIOX_4", "GPIOX_5", "GPIOX_6", "GPIOX_7", "GPIOAO_8", "GPIOAO_9", "GPIOAO_10", "GPIOAO_11";
+			linux,phandle = <0x1e>;
+			phandle = <0x1e>;
+		};
+
+		audio_pin1 {
+			amlogic,setmask = <0x1 0x80>;
+			amlogic,clrmask = <0x1 0x10000>;
+			amlogic,pins = "GPIOY_3";
+			linux,phandle = <0x1d>;
+			phandle = <0x1d>;
+		};
+
+		aml_spi_nor_pins {
+			amlogic,setmask = <0x5 0xf>;
+			amlogic,clrmask = <0x2 0x380000>;
+			amlogic,pins = "BOOT_11", "BOOT_12", "BOOT_13", "BOOT_18";
+			linux,phandle = <0x15>;
+			phandle = <0x15>;
+		};
+
+		dvb_p_ts0_pins {
+			amlogic,setmask = <0x3 0x3f>;
+			amlogic,clrmask = <0x1 0xcce41 0x4 0x2000000 0x7 0x20000000 0x9 0x431f>;
+			amlogic,pins = "GPIOY_9", "GPIOY_10", "GPIOY_11", "GPIOY_12", "GPIOY_13", "GPIOY_14", "GPIOY_15", "GPIOY_16", "GPIOY_0", "GPIOY_1", "GPIOY_4", "GPIOY_8";
+			linux,phandle = <0x28>;
+			phandle = <0x28>;
+		};
+
+		dvb_s_ts0_pins {
+			amlogic,setmask = <0x3 0x1f>;
+			amlogic,clrmask = <0x1 0xcce41 0x4 0x2000000 0x9 0x310>;
+			amlogic,pins = "GPIOY_9", "GPIOY_0", "GPIOY_1", "GPIOY_4", "GPIOY_8";
+			linux,phandle = <0x29>;
+			phandle = <0x29>;
+		};
+
+		dvb_fe_pins {
+			amlogic,clrmask = <0x1 0x38 0x3 0x18000 0x4 0x1800000 0x9 0xc0>;
+			amlogic,pins = "GPIOY_5", "GPIOY_6";
+			linux,phandle = <0x2a>;
+			phandle = <0x2a>;
+		};
+	};
+
+	meson-eth {
+		compatible = "amlogic,meson-eth";
+		dev_name = "meson-eth";
+		status = "okay";
+		ethbaseaddr = <0xfe0c0000>;
+		interruptnum = <0x28>;
+		new_maclogic = <0x1>;
+	};
+
+	dvb {
+		compatible = "amlogic,dvb";
+		ts0 = "parallel";
+		ts0_control = <0x0>;
+		ts0_invert = <0x0>;
+		pinctrl-names = "p_ts0", "s_ts0";
+		pinctrl-0 = <0x28>;
+		pinctrl-1 = <0x29>;
+	};
+
+	dvbfe {
+		compatible = "amlogic,dvbfe";
+		dtv_demod0 = "Avl6211";
+		dtv_demod0_i2c_adap_id = <0x2>;
+		dtv_demod0_i2c_addr = <0xc0>;
+		dtv_demod0_reset_value = <0x0>;
+		dtv_demod0_reset_gpio = "GPIOY_5";
+		fe0_dtv_demod = <0x0>;
+		fe0_ts = <0x0>;
+		fe0_dev = <0x0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <0x2a>;
+	};
+
+	meson-rng {
+		compatible = "amlogic,meson-rng";
+		status = "ok";
+	};
+};
