diff --git a/dts/Kconfig b/dts/Kconfig
index 0cef225ba9..e4e3455d4c 100644
--- a/dts/Kconfig
+++ b/dts/Kconfig
@@ -13,7 +13,6 @@ config PYLIBFDT
 
 config DTOC
 	bool
-	select PYLIBFDT
 
 config BINMAN
 	bool
diff --git a/tools/dtoc/fdt.py b/tools/dtoc/fdt.py
index dbc338653b..04f3c5935c 100644
--- a/tools/dtoc/fdt.py
+++ b/tools/dtoc/fdt.py
@@ -10,7 +10,7 @@ import struct
 import sys
 
 import fdt_util
-import libfdt
+from pylibfdt import libfdt
 from libfdt import QUIET_NOTFOUND
 
 # This deals with a device tree, presenting it as an assortment of Node and
