diff -Naur pango-1.38.1/Makefile.am pango-1.38.1.patch/Makefile.am
--- pango-1.38.1/Makefile.am	2015-11-14 23:05:52.000000000 +0100
+++ pango-1.38.1.patch/Makefile.am	2015-11-14 23:19:07.548943525 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango pango-view examples tools tests build
+SUBDIRS= pango build
 
 EXTRA_DIST = 			\
 	autogen.sh		\
diff -Naur pango-1.38.1/pango/Makefile.am pango-1.38.1.patch/pango/Makefile.am
--- pango-1.38.1/pango/Makefile.am	2015-09-01 00:47:11.000000000 +0200
+++ pango-1.38.1.patch/pango/Makefile.am	2015-11-14 23:21:14.081145117 +0100
@@ -31,7 +31,7 @@
 	$(LIBTHAI_CFLAGS)					\
 	$(X_CFLAGS)
 
-BUILT_SOURCES = pango-enum-types.h pango-enum-types.c pango-features.h
+BUILT_SOURCES = pango-features.h
 
 pangoincludedir=$(includedir)/pango-1.0/pango
 
@@ -127,22 +127,6 @@
 	pango-features.h   \
 	pango-enum-types.h
 
-pango-enum-types.h: s-enum-types-h
-s-enum-types-h: $(pango_headers) Makefile.am pango-enum-types.h.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			 --template pango-enum-types.h.template \
-			$(pango_headers) ) > pango-enum-types.h.tmp \
-			&& mv pango-enum-types.h.tmp pango-enum-types.h \
-			&& echo timestamp > $@
-
-pango-enum-types.c: s-enum-types-c
-s-enum-types-c: $(pango_headers) Makefile.am pango-enum-types.c.template
-	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) \
-			 --template pango-enum-types.c.template \
-			$(pango_headers) ) > pango-enum-types.c.tmp \
-			&& mv pango-enum-types.c.tmp pango-enum-types.c \
-			&& echo timestamp > $@
-
 pango-features.h:
 	$(AM_V_GEN) cd $(top_builddir) && ./config.status pango/pango-features.h
 
diff -Naur pango-1.38.1/pango/pango-context.c pango-1.38.1.patch/pango/pango-context.c
--- pango-1.38.1/pango/pango-context.c	2015-09-13 17:05:39.000000000 +0200
+++ pango-1.38.1.patch/pango/pango-context.c	2015-11-14 23:22:57.512225981 +0100
@@ -38,6 +38,7 @@
 
 #include "pango-engine-private.h"
 #include "pango-script-private.h"
+#include "pango-enum-types.h"
 
 /**
  * PangoContext:
diff -Naur pango-1.38.1/pango/pangofc-font.c pango-1.38.1.patch/pango/pangofc-font.c
--- pango-1.38.1/pango/pangofc-font.c	2015-04-15 19:37:43.000000000 +0200
+++ pango-1.38.1.patch/pango/pangofc-font.c	2015-11-14 23:23:49.840316980 +0100
@@ -45,6 +45,7 @@
 #include "pango-engine.h"
 #include "pango-layout.h"
 #include "pango-impl-utils.h"
+#include "pango-enum-types.h"
 
 #include <fontconfig/fcfreetype.h>
 
diff -Naur pango-1.38.1/pango/pango-layout.c pango-1.38.1.patch/pango/pango-layout.c
--- pango-1.38.1/pango/pango-layout.c	2015-09-01 18:20:54.000000000 +0200
+++ pango-1.38.1.patch/pango/pango-layout.c	2015-11-14 23:23:19.280262638 +0100
@@ -77,6 +77,7 @@
 #include <string.h>
 
 #include "pango-layout-private.h"
+#include "pango-enum-types.h"
 
 
 typedef struct _Extents Extents;
