diff -Naur linux-3.10.10/include/uapi/linux/ptrace.h linux-3.10.10.patch/include/uapi/linux/ptrace.h
--- linux-3.10.10/include/uapi/linux/ptrace.h	2013-08-29 18:47:51.000000000 +0200
+++ linux-3.10.10.patch/include/uapi/linux/ptrace.h	2013-09-04 16:38:10.182685149 +0200
@@ -55,11 +55,13 @@
 
 #define PTRACE_PEEKSIGINFO	0x4209
 
+#ifdef __KERNEL__
 struct ptrace_peeksiginfo_args {
 	__u64 off;	/* from which siginfo to start */
 	__u32 flags;
 	__s32 nr;	/* how may siginfos to take */
 };
+#endif /* __KERNEL__ */
 
 /* Read signals from a shared (process wide) queue */
 #define PTRACE_PEEKSIGINFO_SHARED	(1 << 0)
