diff --git a/GLideN64/src/Graphics/OpenGLContext/opengl_Utils.cpp b/GLideN64/src/Graphics/OpenGLContext/opengl_Utils.cpp
index 0279af9..1b1d7cc 100644
--- a/GLideN64/src/Graphics/OpenGLContext/opengl_Utils.cpp
+++ b/GLideN64/src/Graphics/OpenGLContext/opengl_Utils.cpp
@@ -53,6 +53,9 @@ bool Utils::isEGLExtensionSupported(const char * extension)
 
 	const char* extensions = eglQueryString(eglGetDisplay(EGL_DEFAULT_DISPLAY), EGL_EXTENSIONS);
 
+	if (!extensions)
+		return false;
+
 	const char* start = extensions;
 	for (;;) {
 		where = strstr(start, extension);
