diff -uNr a/Makefile b/Makefile
--- a/Makefile	2026-04-18 19:12:56.000000000 +0900
+++ b/Makefile	2026-04-21 20:12:58.011943683 +0900
@@ -171,10 +171,12 @@
 RGBLINK := $(RGBDS)rgblink
 RGBGFX  := $(RGBDS)rgbgfx
 
+ifneq ("$(wildcard $(RGBASM))", "")
 # RGBASM 0.7+ deprecate and remove `-h`
 RGBASM_FLAGS := $(if $(filter $(shell echo 'println __RGBDS_MAJOR__ || (!__RGBDS_MAJOR__ && __RGBDS_MINOR__ > 6)' | $(RGBASM) -), $$0), -h,) --include $(OBJ)/BootROMs/ --include BootROMs/
 # RGBGFX 0.6+ replace `-h` with `-Z`, and need `-c embedded`
 RGBGFX_FLAGS := $(if $(filter $(shell echo 'println __RGBDS_MAJOR__ || (!__RGBDS_MAJOR__ && __RGBDS_MINOR__ > 5)' | $(RGBASM) -), $$0), -h -u, -Z -u -c embedded)
+endif
 
 
 # Set compilation and linkage flags based on target, platform and configuration
