diff -Naur libproxy-svn-334/libproxy-1.0.pc.in libproxy-svn-334.patch/libproxy-1.0.pc.in
--- libproxy-svn-334/libproxy-1.0.pc.in	2009-04-21 15:45:40.000000000 +0200
+++ libproxy-svn-334.patch/libproxy-1.0.pc.in	2009-04-21 18:34:08.000000000 +0200
@@ -2,6 +2,7 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
+modulesdir=${libdir}/libproxy/@VERSION@/modules
 
 Name: libproxy
 Description: Proxy Configuration Library
