From 812680199b0e474c83cdc8bd331178afcaaad862 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 4 May 2024 12:07:33 +0000
Subject: [PATCH] fix build with gcc-14

add missing header
---
 src/arg_int.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arg_int.c b/src/arg_int.c
index 29c20e5..bc5ab06 100644
--- a/src/arg_int.c
+++ b/src/arg_int.c
@@ -27,6 +27,7 @@ USA.
 /* #ifdef HAVE_STDLIB_H */
 #include <stdlib.h>
 /* #endif */
+#include <ctype.h>
 
 #include "argtable2.h"
 #include <limits.h>
