diff -Naur gcc-4.7.0-old/gcc/config/i386/linux64.h gcc-4.7.0-new/gcc/config/i386/linux64.h
--- gcc-4.7.0-old/gcc/config/i386/linux64.h	2011-07-07 08:38:34.000000000 -0700
+++ gcc-4.7.0-new/gcc/config/i386/linux64.h	2012-04-23 21:03:08.000000000 -0700
@@ -29,5 +29,5 @@
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 "/lib/ld-linux-x32.so.2"
diff -Naur gcc-4.7.0-old/gcc/config/mips/iris6.h gcc-4.7.0-new/gcc/config/mips/iris6.h
--- gcc-4.7.0-old/gcc/config/mips/iris6.h	2012-03-05 02:43:22.000000000 -0800
+++ gcc-4.7.0-new/gcc/config/mips/iris6.h	2012-04-23 21:02:07.000000000 -0700
@@ -256,27 +256,27 @@
 #define STARTFILE_SPEC \
   "%{!shared: \
      %{mabi=n32: \
-       %{mips4:%{pg:/usr/lib32/mips4/gcrt1.o%s} \
-         %{!pg:%{p:/usr/lib32/mips4/mcrt1.o%s /usr/lib32/mips4/libprof1.a%s} \
-           %{!p:/usr/lib32/mips4/crt1.o%s}}} \
-       %{!mips4:%{pg:/usr/lib32/mips3/gcrt1.o%s} \
-         %{!pg:%{p:/usr/lib32/mips3/mcrt1.o%s /usr/lib32/mips3/libprof1.a%s} \
-           %{!p:/usr/lib32/mips3/crt1.o%s}}}} \
+       %{mips4:%{pg:/usr/lib/mips4/gcrt1.o%s} \
+         %{!pg:%{p:/usr/lib/mips4/mcrt1.o%s /usr/lib/mips4/libprof1.a%s} \
+           %{!p:/usr/lib/mips4/crt1.o%s}}} \
+       %{!mips4:%{pg:/usr/lib/mips3/gcrt1.o%s} \
+         %{!pg:%{p:/usr/lib/mips3/mcrt1.o%s /usr/lib/mips3/libprof1.a%s} \
+           %{!p:/usr/lib/mips3/crt1.o%s}}}} \
      %{mabi=64: \
-       %{mips4:%{pg:/usr/lib64/mips4/gcrt1.o} \
-         %{!pg:%{p:/usr/lib64/mips4/mcrt1.o /usr/lib64/mips4/libprof1.a} \
-           %{!p:/usr/lib64/mips4/crt1.o}}} \
-       %{!mips4:%{pg:/usr/lib64/mips3/gcrt1.o} \
-         %{!pg:%{p:/usr/lib64/mips3/mcrt1.o /usr/lib64/mips3/libprof1.a} \
-           %{!p:/usr/lib64/mips3/crt1.o}}}}} \
+       %{mips4:%{pg:/usr/lib/mips4/gcrt1.o} \
+         %{!pg:%{p:/usr/lib/mips4/mcrt1.o /usr/lib/mips4/libprof1.a} \
+           %{!p:/usr/lib/mips4/crt1.o}}} \
+       %{!mips4:%{pg:/usr/lib/mips3/gcrt1.o} \
+         %{!pg:%{p:/usr/lib/mips3/mcrt1.o /usr/lib/mips3/libprof1.a} \
+           %{!p:/usr/lib/mips3/crt1.o}}}}} \
   irix-crti.o%s crtbegin.o%s"
 
 #undef LIB_SPEC
 #define LIB_SPEC \
-  "%{mabi=n32: %{mips4:-L/usr/lib32/mips4} %{!mips4:-L/usr/lib32/mips3} \
-     -L/usr/lib32} \
-   %{mabi=64: %{mips4:-L/usr/lib64/mips4} %{!mips4:-L/usr/lib64/mips3} \
-     -L/usr/lib64} \
+  "%{mabi=n32: %{mips4:-L/usr/lib/mips4} %{!mips4:-L/usr/lib/mips3} \
+     -L/usr/lib} \
+   %{mabi=64: %{mips4:-L/usr/lib/mips4} %{!mips4:-L/usr/lib/mips3} \
+     -L/usr/lib} \
    %{!shared:" \
      SUBTARGET_DONT_WARN_UNUSED_SPEC \
      " %{pthread:-lpthread} %{p:libprof1.a%s}%{pg:libprof1.a%s} -lc " \
@@ -293,10 +293,10 @@
   "%{Ofast|ffast-math|funsafe-math-optimizations:crtfastmath.o%s} \
    crtend.o%s irix-crtn.o%s \
    %{!shared: \
-     %{mabi=n32:%{mips4:/usr/lib32/mips4/crtn.o%s}\
-       %{!mips4:/usr/lib32/mips3/crtn.o%s}}\
-     %{mabi=64:%{mips4:/usr/lib64/mips4/crtn.o%s}\
-       %{!mips4:/usr/lib64/mips3/crtn.o%s}}}"
+     %{mabi=n32:%{mips4:/usr/lib/mips4/crtn.o%s}\
+       %{!mips4:/usr/lib/mips3/crtn.o%s}}\
+     %{mabi=64:%{mips4:/usr/lib/mips4/crtn.o%s}\
+       %{!mips4:/usr/lib/mips3/crtn.o%s}}}"
 
 /* Generic part of the LINK_SPEC.  */
 #undef LINK_SPEC
diff -Naur gcc-4.7.0-old/gcc/config/mips/linux64.h gcc-4.7.0-new/gcc/config/mips/linux64.h
--- gcc-4.7.0-old/gcc/config/mips/linux64.h	2011-07-19 11:00:27.000000000 -0700
+++ gcc-4.7.0-new/gcc/config/mips/linux64.h	2012-04-23 21:02:07.000000000 -0700
@@ -24,9 +24,9 @@
 #define GNU_USER_LINK_EMULATIONN32 "elf32%{EB:b}%{EL:l}tsmipn32"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKERN32 "/lib/ld-uClibc.so.0"
 #define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32, \
diff -Naur gcc-4.7.0-old/gcc/config/rs6000/linux64.h gcc-4.7.0-new/gcc/config/rs6000/linux64.h
--- gcc-4.7.0-old/gcc/config/rs6000/linux64.h	2012-03-12 09:16:51.000000000 -0700
+++ gcc-4.7.0-new/gcc/config/rs6000/linux64.h	2012-04-23 21:02:07.000000000 -0700
@@ -359,7 +359,7 @@
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 #define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
diff -Naur gcc-4.7.0-old/gcc/config/sparc/linux64.h gcc-4.7.0-new/gcc/config/sparc/linux64.h
--- gcc-4.7.0-old/gcc/config/sparc/linux64.h	2011-11-11 17:05:07.000000000 -0800
+++ gcc-4.7.0-new/gcc/config/sparc/linux64.h	2012-04-23 21:02:07.000000000 -0700
@@ -94,7 +94,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -113,7 +113,7 @@
       %{static:-static}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -202,7 +202,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
