diff -Naur gcc-4.5.1/configure gcc-4.5.1.patch/configure
--- gcc-4.5.1/configure	2010-06-24 23:06:37.000000000 +0200
+++ gcc-4.5.1.patch/configure	2010-09-02 17:12:08.807742278 +0200
@@ -5781,7 +5781,7 @@
 
 # Check for PPL
 ppl_major_version=0
-ppl_minor_version=10
+ppl_minor_version=11
 ppllibs=" -lppl_c -lppl -lgmpxx"
 pplinc=
 
diff -Naur gcc-4.5.1/configure.ac gcc-4.5.1.patch/configure.ac
--- gcc-4.5.1/configure.ac	2010-06-24 23:06:37.000000000 +0200
+++ gcc-4.5.1.patch/configure.ac	2010-09-02 17:12:22.748618664 +0200
@@ -1511,7 +1511,7 @@
 
 # Check for PPL
 ppl_major_version=0
-ppl_minor_version=10
+ppl_minor_version=11
 ppllibs=" -lppl_c -lppl -lgmpxx"
 pplinc=
 
