diff -Naur gcc-4.7.0-old/gcc/config/i386/t-linux64 gcc-4.7.0-new/gcc/config/i386/t-linux64
--- gcc-4.7.0-old/gcc/config/i386/t-linux64	2012-03-22 08:21:36.000000000 -0700
+++ gcc-4.7.0-new/gcc/config/i386/t-linux64	2012-03-22 08:24:29.000000000 -0700
@@ -31,9 +31,9 @@
 # 	/lib64 has x86-64 libraries.
 # 	/libx32 has x32 libraries.
 #
-comma=,
-MULTILIB_OPTIONS    = $(subst $(comma),/,$(TM_MULTILIB_CONFIG))
-MULTILIB_DIRNAMES   = $(patsubst m%, %, $(subst /, ,$(MULTILIB_OPTIONS)))
-MULTILIB_OSDIRNAMES = m64=../lib64
-MULTILIB_OSDIRNAMES+= m32=$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
-MULTILIB_OSDIRNAMES+= mx32=../libx32
+comma=,
+MULTILIB_OPTIONS    = $(subst $(comma),/,$(TM_MULTILIB_CONFIG))
+MULTILIB_DIRNAMES   = $(patsubst m%, %, $(subst /, ,$(MULTILIB_OPTIONS)))
+#MULTILIB_OSDIRNAMES = m64=../lib64
+#MULTILIB_OSDIRNAMES+= m32=$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+#MULTILIB_OSDIRNAMES+= mx32=../libx32
