diff -Naur gcc-4.8.0/configure gcc-4.8.0.patch/configure
--- gcc-4.8.0/configure	2013-02-15 18:45:54.000000000 +0100
+++ gcc-4.8.0.patch/configure	2013-03-23 00:16:47.286139411 +0100
@@ -6036,8 +6036,8 @@
     CFLAGS="${_cloog_saved_CFLAGS} ${clooginc} ${islinc} ${gmpinc}"
     LDFLAGS="${_cloog_saved_LDFLAGS} ${clooglibs} ${isllibs} ${gmplib}"
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.17.0 of CLooG" >&5
-$as_echo_n "checking for version 0.17.0 of CLooG... " >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.18.0 of CLooG" >&5
+$as_echo_n "checking for version 0.18.0 of CLooG... " >&6; }
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include "cloog/version.h"
@@ -6045,7 +6045,7 @@
 main ()
 {
 #if CLOOG_VERSION_MAJOR != 0 \
-    || CLOOG_VERSION_MINOR != 17 \
+    || CLOOG_VERSION_MINOR != 18 \
     || CLOOG_VERSION_REVISION < 0
     choke me
    #endif
