diff -Naur gcc-4.7.0-old/libstdc++-v3/config.h.in gcc-4.7.0-new/libstdc++-v3/config.h.in
--- gcc-4.7.0-old/libstdc++-v3/config.h.in	2012-03-22 08:25:54.000000000 -0700
+++ gcc-4.7.0-new/libstdc++-v3/config.h.in	2012-03-22 08:28:32.000000000 -0700
@@ -130,7 +130,7 @@
 #undef HAVE_FABSL
 
 /* Define to 1 if you have the <fenv.h> header file. */
-#undef HAVE_FENV_H
+#define HAVE_FENV_H 1
 
 /* Define to 1 if you have the `finite' function. */
 #undef HAVE_FINITE
@@ -184,7 +184,7 @@
 #undef HAVE_HYPOTL
 
 /* Define if you have the iconv() function. */
-#undef HAVE_ICONV
+#define HAVE_ICONV 1
 
 /* Define to 1 if you have the <ieeefp.h> header file. */
 #undef HAVE_IEEEFP_H
@@ -223,7 +223,7 @@
 #undef HAVE_ISWBLANK
 
 /* Define if LC_MESSAGES is available in <locale.h>. */
-#undef HAVE_LC_MESSAGES
+#define HAVE_LC_MESSAGES 1
 
 /* Define to 1 if you have the `ldexpf' function. */
 #undef HAVE_LDEXPF
@@ -304,7 +304,7 @@
 #undef HAVE_QFPCLASS
 
 /* Define to 1 if you have the `setenv' function. */
-#undef HAVE_SETENV
+#define HAVE_SETENV 1
 
 /* Define to 1 if you have the `sincos' function. */
 #undef HAVE_SINCOS
@@ -346,7 +346,7 @@
 #undef HAVE_STDLIB_H
 
 /* Define if strerror_l is available in <string.h>. */
-#undef HAVE_STRERROR_L
+#define HAVE_STRERROR_L 1
 
 /* Define if strerror_r is available in <string.h>. */
 #undef HAVE_STRERROR_R
@@ -377,7 +377,7 @@
 #undef HAVE_SYS_IOCTL_H
 
 /* Define to 1 if you have the <sys/ipc.h> header file. */
-#undef HAVE_SYS_IPC_H
+#define HAVE_SYS_IPC_H 1
 
 /* Define to 1 if you have the <sys/isa_defs.h> header file. */
 #undef HAVE_SYS_ISA_DEFS_H
@@ -389,10 +389,10 @@
 #undef HAVE_SYS_PARAM_H
 
 /* Define to 1 if you have the <sys/resource.h> header file. */
-#undef HAVE_SYS_RESOURCE_H
+#define HAVE_SYS_RESOURCE_H 1
 
 /* Define to 1 if you have the <sys/sem.h> header file. */
-#undef HAVE_SYS_SEM_H
+#define HAVE_SYS_SEM_H 1
 
 /* Define to 1 if you have the <sys/stat.h> header file. */
 #undef HAVE_SYS_STAT_H
@@ -401,7 +401,7 @@
 #undef HAVE_SYS_SYSINFO_H
 
 /* Define to 1 if you have the <sys/time.h> header file. */
-#undef HAVE_SYS_TIME_H
+#define HAVE_SYS_TIME_H 1
 
 /* Define to 1 if you have the <sys/types.h> header file. */
 #undef HAVE_SYS_TYPES_H
@@ -431,7 +431,7 @@
 #undef HAVE_TGMATH_H
 
 /* Define to 1 if the target supports thread-local storage. */
-#undef HAVE_TLS
+#define HAVE_TLS 1
 
 /* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
@@ -644,7 +644,7 @@
 #undef HAVE__TANL
 
 /* Define as const if the declaration of iconv() needs const. */
-#undef ICONV_CONST
+#define ICONV_CONST 1
 
 /* Define to the sub-directory in which libtool stores uninstalled libraries.
    */
