diff -up SDL-1.2.13/include/SDL_stdinc.h~ SDL-1.2.13/include/SDL_stdinc.h
--- SDL-1.2.13/include/SDL_stdinc.h~	2007-12-31 05:48:36.000000000 +0100
+++ SDL-1.2.13/include/SDL_stdinc.h	2009-02-13 11:01:34.000000000 +0100
@@ -320,7 +320,7 @@ do {								\
 #define SDL_memcpy4(dst, src, len)	SDL_memcpy(dst, src, (len) << 2)
 #endif
 
-#if defined(__GNUC__) && defined(i386)
+#if 0
 #define SDL_revcpy(dst, src, len)			\
 do {							\
 	int u0, u1, u2;					\

