diff -Nur SDL-1.2.14/configure.in SDL-1.2.14-new/configure.in
--- SDL-1.2.14/configure.in	2009-10-18 01:35:03.000000000 +0800
+++ SDL-1.2.14-new/configure.in	2010-01-08 15:14:09.000000000 +0800
@@ -745,7 +745,6 @@
             esac
         fi
 
-        AC_PATH_PROG(NASM, yasm)
         echo "%ifidn __OUTPUT_FORMAT__,elf" > unquoted-sections
         echo "section .note.GNU-stack noalloc noexec nowrite progbits" >> unquoted-sections
         echo "%endif" >> unquoted-sections
