diff -Naur SDL-1.2.14-old/configure.in SDL-1.2.14-new/configure.in
--- SDL-1.2.14-old/configure.in	2009-10-17 10:35:03.000000000 -0700
+++ SDL-1.2.14-new/configure.in	2010-08-17 16:42:33.000000000 -0700
@@ -3,6 +3,7 @@
 AC_CONFIG_HEADER(include/SDL_config.h)
 AC_GNU_SOURCE
 AC_CONFIG_AUX_DIRS($srcdir/build-scripts)
+AC_CONFIG_MACRO_DIR([acinclude])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 #
