--- SDL-1.2.14/src/audio/SDL_audio.c	2009-10-13 18:25:35.000000000 +0800
+++ SDL-1.2.14-new/src/audio/SDL_audio.c	2010-01-08 15:29:35.000000000 +0800
@@ -36,15 +36,15 @@
 
 /* Available audio drivers */
 static AudioBootStrap *bootstrap[] = {
-#if SDL_AUDIO_DRIVER_BSD
-	&BSD_AUDIO_bootstrap,
-#endif
 #if SDL_AUDIO_DRIVER_PULSE
 	&PULSE_bootstrap,
 #endif
 #if SDL_AUDIO_DRIVER_ALSA
 	&ALSA_bootstrap,
 #endif
+#if SDL_AUDIO_DRIVER_BSD
+	&BSD_AUDIO_bootstrap,
+#endif
 #if SDL_AUDIO_DRIVER_OSS
 	&DSP_bootstrap,
 	&DMA_bootstrap,
