From edd203bae0939789cf9f34281ad04f103b495e16 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 17 Jan 2023 12:13:29 -0800
Subject: [PATCH 1/2] libretro: Fix missing library version

---
 src/libretro.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libretro.cpp b/src/libretro.cpp
index 5b2830a..1aeb141 100644
--- a/src/libretro.cpp
+++ b/src/libretro.cpp
@@ -114,7 +114,7 @@ void retro_get_system_info(struct retro_system_info *info) {
 #else
 #define __GIT_VERSION ""
 #endif
-	info->library_version = __GIT_VERSION;
+	info->library_version = SCUMMVM_VERSION;
 	info->valid_extensions = "scummvm";
 	info->need_fullpath = true;
 	info->block_extract = false;
-- 
2.34.1

