From 9d4e078e96c950f6b2d8977d08390844cd9aa62c Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Thu, 28 Oct 2021 16:00:20 -0700
Subject: [PATCH 2/2] libretro: Move scummvm.ini to writable save directory

---
 src/libretro-os.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libretro-os.cpp b/src/libretro-os.cpp
index 27a442d..3b52467 100644
--- a/src/libretro-os.cpp
+++ b/src/libretro-os.cpp
@@ -742,7 +742,7 @@ public:
 	}
 
 	virtual Common::String getDefaultConfigFileName() {
-		return s_systemDir + "/scummvm.ini";
+		return s_saveDir + "/scummvm.ini";
 	}
 
 	virtual void logMessage(LogMessageType::Type type, const char *message) {
-- 
2.34.1

