diff -Naur a/backends/platform/libretro/build/Makefile b/backends/platform/libretro/build/Makefile
--- a/backends/platform/libretro/build/Makefile	2016-05-09 19:28:52.000000000 -0700
+++ b/backends/platform/libretro/build/Makefile	2016-05-11 11:55:30.434658218 -0700
@@ -24,9 +24,9 @@
 
 TARGET_NAME := scummvm
 
-LD        = $(CXX)
-AR        = ar cru
-RANLIB    = ranlib
+LD        ?= $(CXX)
+AR        ?= ar cru
+RANLIB    ?= ranlib
 
 ifeq ($(platform), unix)
    TARGET  := $(TARGET_NAME)_libretro.so
