--- shairplay-0.9.0.orig/src/shairplay.c
+++ shairplay-0.9.0/src/shairplay.c
@@ -313,9 +313,12 @@ main(int argc, char *argv[])
 
 	raop = raop_init_from_keyfile(10, &raop_cbs, "airport.key", NULL);
 	if (raop == NULL) {
-		fprintf(stderr, "Could not initialize the RAOP service\n");
-		fprintf(stderr, "Please make sure the airport.key file is in the current directory.\n");
-		return -1;
+		raop = raop_init_from_keyfile(10, &raop_cbs, "/etc/shairplay/airport.key", NULL);
+		if (raop == NULL) {
+			fprintf(stderr, "Could not initialize the RAOP service\n");
+			fprintf(stderr, "Please make sure the airport.key file is in the current directory.\n");
+			return -1;
+		}
 	}
 
 	if (strlen(options.password)) {

