From d20a6d6bcd625d0162cc7158b597e04ef3ee1379 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Sat, 11 Jun 2011 23:22:19 +0200
Subject: [PATCH] Disable run of ldconfig if crosscompiling too

Signed-off-by: Stephan Raue <stephan@openelec.tv>
---
 configure.ac |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9973c92..363e671 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,6 +64,10 @@ AC_ARG_WITH([ldconfig],
 	    [  --without-ldconfig  Disable call of ldconfig on make install],
 	    [ldconfig='echo ldconfig disabled'])
 
+if test x"$cross_compiling" = xyes; then
+  ldconfig='echo crosscompiling: ldconfig disabled'
+fi
+
 AC_CONFIG_HEADER(config.h)
 
 EXTRA_OBJ=""
-- 
1.7.5.4

