diff -Naur a/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp b/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp
--- a/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp	2019-04-12 00:30:46.000000000 -0700
+++ b/media_driver/linux/common/codec/ddi/media_libva_decoder.cpp	2019-04-12 22:55:05.146368984 -0700
@@ -41,10 +41,6 @@
 #include "media_interfaces.h"
 #include "media_ddi_decode_const.h"
 
-#ifndef ANDROID
-#include <X11/Xutil.h>
-#endif
-
 #include <linux/fb.h>
 
 typedef MediaDdiFactory<DdiMediaDecode, DDI_DECODE_CONFIG_ATTR> DdiDecodeFactory;
