From fa3f88530ec9083ff15d9637a9019a8a9408435b Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Sat, 9 Nov 2019 10:02:43 +0000
Subject: [PATCH 12/12] WIP: v4l2-request: rolling timestamps

---
 libavcodec/v4l2_request.c | 10 +++++++---
 libavcodec/v4l2_request.h |  1 +
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/libavcodec/v4l2_request.c b/libavcodec/v4l2_request.c
index 1dabf77689..611c22f8b6 100644
--- a/libavcodec/v4l2_request.c
+++ b/libavcodec/v4l2_request.c
@@ -105,12 +105,14 @@ static int v4l2_request_queue_buffer(V4L2RequestContext *ctx, int request_fd, V4
         .type = buf->buffer.type,
         .memory = buf->buffer.memory,
         .index = buf->index,
-        .timestamp.tv_usec = buf->index + 1,
+        .timestamp.tv_usec = ctx->timestamp,
         .bytesused = buf->used,
         .request_fd = request_fd,
         .flags = ((request_fd >= 0) ? V4L2_BUF_FLAG_REQUEST_FD : 0) | flags,
     };
 
+    buf->buffer.timestamp = buffer.timestamp;
+
     if (V4L2_TYPE_IS_MULTIPLANAR(buf->buffer.type)) {
         planes[0].bytesused = buf->used;
         buffer.bytesused = 0;
@@ -200,6 +202,9 @@ static int v4l2_request_queue_decode(AVCodecContext *avctx, AVFrame *frame, stru
 
     av_log(avctx, AV_LOG_DEBUG, "%s: avctx=%p used=%u controls=%d index=%d fd=%d request_fd=%d first_slice=%d last_slice=%d\n", __func__, avctx, req->output.used, count, req->capture.index, req->capture.fd, req->request_fd, first_slice, last_slice);
 
+    if (first_slice)
+        ctx->timestamp++;
+
     ret = v4l2_request_set_controls(ctx, req->request_fd, control, count);
     if (ret < 0) {
         av_log(avctx, AV_LOG_ERROR, "%s: set controls failed for request %d, %s (%d)\n", __func__, req->request_fd, strerror(errno), errno);
@@ -651,6 +656,7 @@ int ff_v4l2_request_init(AVCodecContext *avctx, uint32_t pixelformat, uint32_t b
 
     ctx->media_fd = -1;
     ctx->video_fd = -1;
+    ctx->timestamp = 0;
 
     udev = udev_new();
     if (!udev) {
@@ -784,8 +790,6 @@ static int v4l2_request_buffer_alloc(AVCodecContext *avctx, V4L2RequestBuffer *b
         return ret;
     }
 
-    buf->buffer.timestamp.tv_usec = buf->index + 1;
-
     if (V4L2_TYPE_IS_OUTPUT(type)) {
         void *addr = mmap(NULL, buf->size, PROT_READ | PROT_WRITE, MAP_SHARED, ctx->video_fd, V4L2_TYPE_IS_MULTIPLANAR(type) ? buf->buffer.m.planes[0].m.mem_offset : buf->buffer.m.offset);
         if (addr == MAP_FAILED) {
diff --git a/libavcodec/v4l2_request.h b/libavcodec/v4l2_request.h
index d4146bd4ee..72698f6f3c 100644
--- a/libavcodec/v4l2_request.h
+++ b/libavcodec/v4l2_request.h
@@ -28,6 +28,7 @@ typedef struct V4L2RequestContext {
     int media_fd;
     enum v4l2_buf_type output_type;
     struct v4l2_format format;
+    int timestamp;
 } V4L2RequestContext;
 
 typedef struct V4L2RequestBuffer {
