From b72cd16bafbb8b54ae9449248d1e0d559fa6f02d Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Mon, 22 Jun 2020 23:16:41 +0200
Subject: [PATCH] fix compilation with gcc10

---
 vice/src/arch/libretro/archdep.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vice/src/arch/libretro/archdep.c b/vice/src/arch/libretro/archdep.c
index dc3491f2ace4..0465c098f031 100644
--- a/vice/src/arch/libretro/archdep.c
+++ b/vice/src/arch/libretro/archdep.c
@@ -64,7 +64,7 @@
 #include "util.h"
 #include "keyboard.h"
 
-#define LIBRETROHACK 1
+//#define LIBRETROHACK 1
 #ifdef LIBRETROHACK
 //FIXME remove me!
 BYTE plus4tcbm_outputa[2], plus4tcbm_outputb[2], plus4tcbm_outputc[2];
@@ -715,4 +715,4 @@ void archdep_vice_exit(int excode)
 {
     exit(excode);
 }
-*/
\ No newline at end of file
+*/
-- 
2.27.0

