From 0079354fada76965346bf626cbd40c9ac25f0ce3 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 21 Sep 2013 21:14:31 +0300
Subject: [PATCH] support /storage/.config/modprobe.d

---
 libkmod/libkmod.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libkmod/libkmod.c b/libkmod/libkmod.c
index ef83e31..a0d4bda 100644
--- a/libkmod/libkmod.c
+++ b/libkmod/libkmod.c
@@ -62,6 +62,7 @@ static const char *default_config_paths[] = {
 	SYSCONFDIR "/modprobe.d",
 	"/run/modprobe.d",
 	"/lib/modprobe.d",
+	"/storage/.config/modprobe.d",
 	NULL
 };
 
-- 
1.7.2.5

