diff -Naur read-edid-1.4.1.orig/lrmi.c read-edid-1.4.1/lrmi.c
--- read-edid-1.4.1.orig/lrmi.c	2008-06-28 14:22:33.000000000 +0200
+++ read-edid-1.4.1/lrmi.c	2008-06-28 14:22:47.000000000 +0200
@@ -169,7 +169,7 @@
 	}
 
 
-#define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
+#define DEFAULT_VM86_FLAGS 	(X86_EFLAGS_IF | X86_EFLAGS_IOPL)
 #define DEFAULT_STACK_SIZE 	0x1000
 #define RETURN_TO_32_INT 	255
 
@@ -781,7 +781,7 @@
 
 			context.vm.regs.cs = get_int_seg(v);
 			context.vm.regs.eip = get_int_off(v);
-			context.vm.regs.eflags &= ~(VIF_MASK | TF_MASK);
+			context.vm.regs.eflags &= ~(X86_EFLAGS_VIF | X86_EFLAGS_TF);
 
 			continue;
 			}
