diff -Naur hdparm-9.39/Makefile hdparm-9.39.patch/Makefile
--- hdparm-9.39/Makefile	2012-01-06 17:05:37.000000000 +0100
+++ hdparm-9.39.patch/Makefile	2012-07-28 05:22:53.271127964 +0200
@@ -13,7 +13,7 @@
 CC ?= gcc
 STRIP ?= strip
 
-CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
+CFLAGS := -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
 
 LDFLAGS = -s
 #LDFLAGS = -s -static
