--- a/makefile	2020-01-31 06:50:07.043158899 +0100
+++ b/makefile	2020-01-31 06:43:35.696492947 +0100
@@ -1713,12 +1713,11 @@
 
 generate: \
 		genie \
-		$(GEN_FOLDERS) \
 		$(GENDIR)/version.cpp \
 		$(patsubst %.po,%.mo,$(call rwildcard, language/, *.po)) \
 		$(patsubst $(SRC)/%.lay,$(GENDIR)/%.lh,$(LAYOUTS)) \
 		$(GENDIR)/mame/machine/mulcd.hxx \
-		$(GENDIR)/includes/SDL2
+		$(GENDIR)/includes/SDL2 | $(GEN_FOLDERS)
 
 $(GENDIR)/includes/SDL2:
 	-$(call MKDIR,$@)
@@ -1754,7 +1753,7 @@
 	@echo Compressing $<...
 	$(SILENT)$(PYTHON) scripts/build/complay.py $< $@ layout_$(basename $(notdir $<))
 
-$(GENDIR)/mame/machine/mulcd.hxx: $(SRC)/mame/machine/mulcd.ppm scripts/build/file2str.py
+$(GENDIR)/mame/machine/mulcd.hxx: $(SRC)/mame/machine/mulcd.ppm scripts/build/file2str.py | $(GEN_FOLDERS)
 	@echo Converting $<...
 	$(SILENT)$(PYTHON) scripts/build/file2str.py $< $@ mulcd_bkg uint8_t
 
--- a/3rdparty/genie/src/actions/make/make_cpp.lua	2020-01-09 10:56:49.931595408 +0100
+++ b/3rdparty/genie/src/actions/make/make_cpp.lua	2020-01-09 10:59:04.237732440 +0100
@@ -560,7 +560,7 @@
 
 				_p('')
 			elseif (path.getextension(file) == ".rc") then
-				_p('$(OBJDIR)/%s.res: %s', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
+				_p('$(OBJDIR)/%s.res: %s | $(OBJDIR)', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
 				if prj.msgresource then
 					_p('\t@echo ' .. prj.msgresource)
 				else
