--- a/3rdparty/genie/src/actions/make/make_cpp.lua	2020-01-09 10:56:49.931595408 +0100
+++ b/3rdparty/genie/src/actions/make/make_cpp.lua	2020-01-09 10:59:04.237732440 +0100
@@ -560,7 +560,7 @@
 
 				_p('')
 			elseif (path.getextension(file) == ".rc") then
-				_p('$(OBJDIR)/%s.res: %s', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
+				_p('$(OBJDIR)/%s.res: %s | $(OBJDIR)', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
 				if prj.msgresource then
 					_p('\t@echo ' .. prj.msgresource)
 				else
