--- a/3rdparty/genie/build/gmake.linux/genie.make	2020-01-14 06:27:22.077161784 +0100
+++ b/3rdparty/genie/build/gmake.linux/genie.make	2020-01-14 06:27:49.856857764 +0100
@@ -28,8 +28,12 @@
   RM    = $(SILENT) del /F "$(subst /,\\,$(1))" 2> nul || exit 0
 endif
 
-CC  = gcc
-CXX = g++
+CC  = $(HOST_CC)
+CXX = $(HOST_CXX)
+CPPFLAGS = $(HOST_CPPFLAGS)
+CFLAGS = $(HOST_CFLAGS)
+CXXFLAGS = $(HOST_CXXFLAGS)
+LDFLAGS = $(HOST_LDFLAGS)
 AR  = ar
 
 ifndef RESCOMP
