diff -Naur a/3rdparty/genie/build/gmake.linux/genie.make b/3rdparty/genie/build/gmake.linux/genie.make
--- a/3rdparty/genie/build/gmake.linux/genie.make	2017-06-28 12:47:18.000000000 -0700
+++ b/3rdparty/genie/build/gmake.linux/genie.make	2017-07-14 12:21:57.302218319 -0700
@@ -28,10 +28,6 @@
   RM    = $(SILENT) del /F "$(subst /,\\,$(1))" 2> nul || exit 0
 endif
 
-CC  = gcc
-CXX = g++
-AR  = ar
-
 ifndef RESCOMP
   ifdef WINDRES
     RESCOMP = $(WINDRES)
