diff --git a/3rdparty/genie/src/actions/make/make_cpp.lua b/3rdparty/genie/src/actions/make/make_cpp.lua
index 880d9d35bf..69fe6f06bf 100644
--- a/3rdparty/genie/src/actions/make/make_cpp.lua
+++ b/3rdparty/genie/src/actions/make/make_cpp.lua
@@ -632,7 +632,7 @@
 
 				_p('')
 			elseif (path.getextension(file) == ".rc") then
-				_p('$(OBJDIR)/%s.res: %s', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
+				_p('$(OBJDIR)/%s.res: %s | $(OBJDIR)', _MAKE.esc(path.getbasename(file)), _MAKE.esc(file))
 				if prj.msgresource then
 					_p('\t@echo ' .. prj.msgresource)
 				else
diff --git a/makefile b/makefile
index 49a30ef20b..c31ee8e318 100644
--- a/makefile
+++ b/makefile
@@ -1710,12 +1710,11 @@ genie: $(GENIE)
 
 generate: \
 		genie \
-		$(GEN_FOLDERS) \
 		$(GENDIR)/version.cpp \
 		$(patsubst %.po,%.mo,$(call rwildcard, language/, *.po)) \
 		$(patsubst $(SRC)/%.lay,$(GENDIR)/%.lh,$(LAYOUTS)) \
 		$(GENDIR)/mame/machine/mulcd.hxx \
-		$(GENDIR)/includes/SDL2
+		$(GENDIR)/includes/SDL2 | $(GEN_FOLDERS)
 
 $(GENDIR)/includes/SDL2:
 	-$(call MKDIR,$@)
@@ -1751,7 +1750,7 @@ $(GENDIR)/%.lh: $(SRC)/%.lay scripts/build/complay.py | $(GEN_FOLDERS)
 	@echo Compressing $<...
 	$(SILENT)$(PYTHON) scripts/build/complay.py $< $@ layout_$(basename $(notdir $<))
 
-$(GENDIR)/mame/machine/mulcd.hxx: $(SRC)/mame/machine/mulcd.ppm scripts/build/file2str.py
+$(GENDIR)/mame/machine/mulcd.hxx: $(SRC)/mame/machine/mulcd.ppm scripts/build/file2str.py | $(GEN_FOLDERS)
 	@echo Converting $<...
 	$(SILENT)$(PYTHON) scripts/build/file2str.py $< $@ mulcd_bkg uint8_t
 
-- 

