From e549da0f4ab03fc9a98bcc5104314e7d430d1dd5 Mon Sep 17 00:00:00 2001
From: Ntemis <ierokipides.dem@gmail.com>
Date: Tue, 14 Jul 2020 23:07:06 +0300
Subject: [PATCH 1/1] bump submodule lib-common

---
 libretro/libretro-common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libretro/libretro-common b/libretro/libretro-common
index 6f62594..6a8f6fc 160000
--- a/libretro/libretro-common
+++ b/libretro/libretro-common
@@ -1 +1 @@
-Subproject commit 6f625948d1b985702cb92f4907c7db0e21b6eebc
+Subproject commit 6a8f6fc1e7e20cd1c9715fc0a7926f39d24aad91
-- 
2.25.1

