diff -Naur lzo-2.07/configure lzo-2.07.patch/configure
--- lzo-2.07/configure	2014-06-25 15:47:13.000000000 +0200
+++ lzo-2.07.patch/configure	2014-06-29 04:46:18.795013873 +0200
@@ -13953,7 +13953,7 @@
 /* end confdefs.h.  */
 #define LZO_CFG_NO_CONFIG_HEADER 1
 #define LZO_WANT_ACC_INCD_H 1
-#include "include/lzo/lzodefs.h"
+#include "include/lzo/lzoconf.h"
 #include "src/lzo_supp.h"
 
 #undef  LZOCHK_ASSERT
diff -Naur lzo-2.07/configure.ac lzo-2.07.patch/configure.ac
--- lzo-2.07/configure.ac	2014-06-25 15:47:13.000000000 +0200
+++ lzo-2.07.patch/configure.ac	2014-06-29 04:47:08.138074223 +0200
@@ -164,7 +164,7 @@
 # // Write output files
 # ************************************************************************/
 
-mfx_LZO_LZOCHK(["-I$srcdir"],["include/lzo/lzodefs.h"],["src/lzo_supp.h"])
+mfx_LZO_LZOCHK(["-I$srcdir"],["include/lzo/lzoconf.h"],["src/lzo_supp.h"])
 
 if test -r .Conf.settings2; then
     . ./.Conf.settings2
