--- uClibc.orig/libc/sysdeps/linux/common/bits/sigcontext.h	2008-02-28 03:06:39.000000000 +0100
+++ uClibc/libc/sysdeps/linux/common/bits/sigcontext.h	2008-02-28 03:06:52.000000000 +0100
@@ -24,6 +24,7 @@
 /* Kernel headers before 2.1.1 define a struct sigcontext_struct, but
    we need sigcontext.  */
 # define sigcontext_struct sigcontext
+# define __user
 
 # include <asm/sigcontext.h>
 #endif
