diff -Naur uClibc-20090902/Makefile.in uClibc-20090902.patch/Makefile.in
--- uClibc-20090902/Makefile.in	2009-08-31 10:48:04.000000000 +0200
+++ uClibc-20090902.patch/Makefile.in	2009-09-02 11:25:11.698689044 +0200
@@ -167,7 +167,7 @@
 	$(Q)set -e; \
 	tmp=`mktemp $(top_builddir)include/bits/sysnum.h.XXXXXX 2>/dev/null || true`; \
 	[ -z "$$tmp" ] && tmp='$(top_builddir)include/bits/sysnum.h.new'; \
-	KERNEL_HEADERS="${KERNEL_HEADERS}" top_builddir="$(top_builddir)" CC="$(CC) $(CPU_CFLAGS)" $(SHELL) $< > $$tmp; \
+	KERNEL_HEADERS="${KERNEL_HEADERS}" top_builddir="$(top_builddir)" CC="$(HOSTCC) $(CPU_CFLAGS)" $(SHELL) $< > $$tmp; \
 	if cmp $(top_builddir)include/bits/sysnum.h $$tmp >/dev/null 2>&1; then \
 		$(RM) $$tmp; \
 	else \
