--- uClibc-0.9.30/include/net/if_arp.h~	2004-10-30 14:56:00.000000000 -0300
+++ uClibc-0.9.30/include/net/if_arp.h	2009-09-02 16:20:24.000000000 -0300
@@ -127,6 +127,9 @@ struct arphdr
 #define ARPHRD_IEEE802_TR 800		/* Magic type ident for TR.  */
 #define ARPHRD_IEEE80211 801		/* IEEE 802.11.  */
 
+#define ARPHRD_VOID	0xFFFF		/* Void type, nothing is known.  */
+#define ARPHRD_NONE	0xFFFE		/* Zero header length.  */
+
 /* ARP ioctl request.  */
 struct arpreq
   {
