diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-02-18 21:17:52.000000000 +0100
+++ b/configure.ac	2016-02-18 23:22:37.384875431 +0100
@@ -417,7 +417,6 @@
 GETTEXT_PACKAGE=avahi
 AC_SUBST([GETTEXT_PACKAGE])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[Gettext package])
-AM_GLIB_GNU_GETTEXT
 
 avahilocaledir='${prefix}/${DATADIRNAME}/locale'
 AC_SUBST(avahilocaledir)
