diff -Naur avahi-0.6.30/service-type-database/Makefile.am avahi-0.6.30.patch/service-type-database/Makefile.am
--- avahi-0.6.30/service-type-database/Makefile.am	2010-06-29 05:30:35.000000000 +0200
+++ avahi-0.6.30.patch/service-type-database/Makefile.am	2012-01-16 09:44:27.232178796 +0100
@@ -18,13 +18,12 @@
 EXTRA_DIST=build-db.in service-types
 
 pkgdata_DATA=service-types
-pkglib_DATA=
 
 if HAVE_PYTHON
 if HAVE_GDBM
 
 noinst_SCRIPTS=build-db
-pkglib_DATA+=service-types.db
+pkgdata_DATA+=service-types.db
 
 build-db: build-db.in
 	$(AM_V_GEN)sed -e 's,@PYTHON\@,$(PYTHON),g' \
@@ -41,7 +40,7 @@
 if HAVE_DBM
 
 noinst_SCRIPTS=build-db
-pkglib_DATA+=service-types.db.pag service-types.db.dir
+pkgdata_DATA+=service-types.db.pag service-types.db.dir
 
 build-db: build-db.in
 	$(AM_V_GEN)sed -e 's,@PYTHON\@,$(PYTHON),g' \
