diff -Naur avahi-0.6.31/avahi-daemon/Makefile.am avahi-0.6.31.patch/avahi-daemon/Makefile.am
--- avahi-0.6.31/avahi-daemon/Makefile.am	2010-08-26 02:51:38.000000000 +0200
+++ avahi-0.6.31.patch/avahi-daemon/Makefile.am	2012-06-13 18:44:23.910169467 +0200
@@ -169,7 +169,7 @@
 	done
 
 install-data-local:
-	test -z "$(localstatedir)/run" || $(mkdir_p) "$(DESTDIR)$(localstatedir)/run"
+	test -z "$(localstatedir)/run" || $(MKDIR_P) "$(DESTDIR)$(localstatedir)/run"
 
 update-systemd:
 	curl http://cgit.freedesktop.org/systemd/plain/src/sd-daemon.c > sd-daemon.c
