diff -Naur melonds-1ad65728476d7b9594c4ff91a1ba60460a0a30e7/Makefile melonds-1ad65728476d7b9594c4ff91a1ba60460a0a30e7-2/Makefile
--- melonds-1ad65728476d7b9594c4ff91a1ba60460a0a30e7/Makefile	2022-02-23 05:16:02.377658636 +0100
+++ melonds-1ad65728476d7b9594c4ff91a1ba60460a0a30e7-2/Makefile	2022-02-23 05:27:31.772948432 +0100
@@ -240,7 +240,20 @@
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
-   
+
+# Tegra210
+else ifeq ($(platform), tegra210)
+   EXT ?= so
+   CPUFLAGS += -march=armv8-a -mtune=cortex-a57 -mtp=soft -mcpu=cortex-a57+crc+fp+simd
+   HAVE_NEON = 1
+   TARGET := $(TARGET_NAME)_libretro.$(EXT)
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+   LIBS += -lpthread -lGL -lrt
+   HAVE_OPENGL = 1
+   HAVE_THREADS = 1
+   JIT_ARCH = aarch64
+
 # Odroid C4
 else ifeq ($(platform), odroidc4)
    EXT ?= so
