diff --git a/Makefile b/Makefile
index 492eae6..bbf0216 100644
--- a/Makefile
+++ b/Makefile
@@ -42,6 +42,10 @@ TARGET_NAME := melonds
 LIBS		    = -lm
 DEFINES :=
 
+ifeq ($(HAVE_OPENGL), 1)
+LIBS += -lGL
+endif
+
 ifneq ($(findstring Haiku,$(shell uname -s)),)
 LIBS += -lnetwork
 endif
@@ -76,8 +80,8 @@ ifeq ($(platform), unix)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS +=-lpthread -lGL
-   HAVE_OPENGL=1
+   LIBS +=-lpthread
+   # HAVE_OPENGL=1
    HAVE_THREADS=1
 else ifeq ($(platform), linux-portable)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
