--- a/modules/remote.py
+++ b/modules/remote.py
@@ -13,7 +13,7 @@ connected = False
 def update_xbmc_object():
     host = get_setting_value('server_hostname')
     try:
-        icon = os.path.abspath('static/images/maraschino_logo.png')
+        icon = os.path.abspath('/storage/.xbmc/addons/service.web.maraschino/static/images/maraschino_logo.png')
 
         if not os.path.exists(icon):
             icon = os.path.abspath('maraschino/static/images/maraschino_logo.png')


