diff -Naur live-0.3.0/css/Makefile live-0.3.0.patch/css/Makefile
--- live-0.3.0/css/Makefile	2013-04-04 22:16:40.000000000 +0200
+++ live-0.3.0.patch/css/Makefile	2016-01-07 06:49:35.954078949 +0100
@@ -4,7 +4,7 @@
 
 ### Additional options to silence TNTNET warnings
 TNTFLAGS   ?= -Wno-overloaded-virtual -Wno-unused-function
-TNTVERSION ?= $(shell tntnet-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+TNTVERSION ?= $(shell pkg-config --modversion tntnet | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
 
 ### The C++ compiler and options:
 
diff -Naur live-0.3.0/httpd/Makefile live-0.3.0.patch/httpd/Makefile
--- live-0.3.0/httpd/Makefile	2013-04-04 22:16:40.000000000 +0200
+++ live-0.3.0.patch/httpd/Makefile	2016-01-07 06:51:05.349223371 +0100
@@ -3,7 +3,7 @@
 
 CXXFLAGS ?= -O2 -Woverloaded-virtual -Wall -fPIC
 
-CXXFLAGS += `tntnet-config --cxxflags`
+CXXFLAGS += `pkg-config --cflags tntnet`
 
 ### Includes and Defines (add further entries here):
 
diff -Naur live-0.3.0/javascript/Makefile live-0.3.0.patch/javascript/Makefile
--- live-0.3.0/javascript/Makefile	2013-04-04 22:16:40.000000000 +0200
+++ live-0.3.0.patch/javascript/Makefile	2016-01-07 06:49:51.561104158 +0100
@@ -4,7 +4,7 @@
 
 ### Additional options to silence TNTNET warnings
 TNTFLAGS   ?= -Wno-overloaded-virtual -Wno-unused-function
-TNTVERSION ?= $(shell tntnet-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+TNTVERSION ?= $(shell pkg-config --modversion tntnet | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
 
 ### The C++ compiler and options:
 
diff -Naur live-0.3.0/Makefile live-0.3.0.patch/Makefile
--- live-0.3.0/Makefile	2013-04-04 22:16:40.000000000 +0200
+++ live-0.3.0.patch/Makefile	2016-01-07 06:48:29.683971910 +0100
@@ -26,7 +26,7 @@
 LDFLAGS	 ?= -fPIC -g
 
 ### Check for libpcre c++ wrapper
-HAVE_LIBPCRECPP = $(shell pcre-config --libs-cpp)
+HAVE_LIBPCRECPP = $(shell pkg-config --libs libpcrecpp libpcre)
 
 ### The directory environment:
 
@@ -46,18 +46,18 @@
 
 APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
 I18NTARG   = $(shell if [ `echo $(APIVERSION) | tr [.] [0]` -ge "10507" ]; then echo "i18n"; fi)
-TNTVERSION = $(shell tntnet-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
-CXXTOOLVER = $(shell cxxtools-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+TNTVERSION = $(shell pkg-config --modversion tntnet | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+CXXTOOLVER = $(shell pkg-config --modversion cxxtools | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
 TNTVERS7   = $(shell ver=$(TNTVERSION); if [ $$ver -ge "1606" ]; then echo "yes"; fi)
 
-CXXFLAGS  += $(shell tntnet-config --cxxflags)
-LIBS      += $(shell tntnet-config --libs)
+CXXFLAGS  += $(shell pkg-config --cflags tntnet)
+LIBS      += $(shell pkg-config --libs tntnet)
 
 ### Optional configuration features
 PLUGINFEATURES =
 ifneq ($(HAVE_LIBPCRECPP),)
 	PLUGINFEATURES += -DHAVE_LIBPCRECPP
-	CXXFLAGS       += $(shell pcre-config --cflags)
+	CXXFLAGS       += $(shell pkg-config --cflags libpcreposix libpcre)
 	LIBS           += $(HAVE_LIBPCRECPP)
 endif
 
diff -Naur live-0.3.0/pages/Makefile live-0.3.0.patch/pages/Makefile
--- live-0.3.0/pages/Makefile	2013-04-04 22:16:40.000000000 +0200
+++ live-0.3.0.patch/pages/Makefile	2016-01-07 06:50:04.234124627 +0100
@@ -4,8 +4,8 @@
 
 ### Additional options to silence TNTNET warnings
 TNTFLAGS   ?= -Wno-overloaded-virtual -Wno-unused-variable
-TNTVERSION ?= $(shell tntnet-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
-CXXTOOLVER ?= $(shell cxxtools-config --version | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+TNTVERSION ?= $(shell pkg-config --modversion tntnet | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
+CXXTOOLVER ?= $(shell pkg-config --modversion cxxtools | sed -e's/\.//g' | sed -e's/pre.*//g' | awk '/^..$$/ { print $$1."000"} /^...$$/ { print $$1."00"} /^....$$/ { print $$1."0" } /^.....$$/ { print $$1 }')
 
 ### The C++ compiler and options:
 
