Index: vdr-plugin-live-0.2.0/thread.cpp
===================================================================
--- vdr-plugin-live-0.2.0.orig/thread.cpp	2011-08-25 21:58:33.000000000 +0200
+++ vdr-plugin-live-0.2.0/thread.cpp	2011-08-25 21:58:37.000000000 +0200
@@ -3,6 +3,7 @@
 #include <stdexcept>
 #include <vdr/tools.h>
 #include <tnt/tntnet.h>
+#include <tnt/tntconfig.h>
 #include "thread.h"
 #include "tntconfig.h"
 
