From c3ecf9fabb674bc154f5a08cdd62e74e2b5d0e36 Mon Sep 17 00:00:00 2001
From: Christian Ruppert
Date: Sun, 03 Jun 2012 23:42:31 +0000
Subject: Compile fix for >=vdr-1.7.28

---
diff --git a/tasks.cpp b/tasks.cpp
index 9f20190..2a2b8d5 100644
--- a/tasks.cpp
+++ b/tasks.cpp
@@ -60,9 +60,15 @@ void PlayRecordingTask::Action()
 
 	const char *current = NowReplaying();
 	if (!current || (0 != strcmp(current, recording->FileName()))) {
+#if VDRVERSNUM >= 10728
+		cReplayControl::SetRecording( 0 );
+		cControl::Shutdown();
+		cReplayControl::SetRecording( recording->FileName() );
+#else
 		cReplayControl::SetRecording( 0, 0 );
 		cControl::Shutdown();
 		cReplayControl::SetRecording( recording->FileName(), recording->Title() );
+#endif
 		cControl::Launch( new cReplayControl );
 		cControl::Attach();
 	}
@@ -122,7 +128,11 @@ void StopRecordingTask::Action()
 		return;
 	}
 
+#if VDRVERSNUM >= 10728
+	cReplayControl::SetRecording( 0 );
+#else
 	cReplayControl::SetRecording( 0, 0 );
+#endif
 	cControl::Shutdown();
 }
 
--
cgit v0.9.0.2-40-g7c79
