From e9c2cc8e656e7a346cb225e91ee86394f0807415 Mon Sep 17 00:00:00 2001
From: BartolomeyKant <notevill@gmail.com>
Date: Mon, 26 May 2025 15:44:39 +0500
Subject: [PATCH] fix cmake minimum required

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d71526c..5e22441 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@
 ##
 ################################################################################
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.10..3.31)
 project(gcem)
 
 set(GCEM_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include)
