diff -Naur libvdpau-0.5-old/configure.ac libvdpau-0.5-new/configure.ac
--- libvdpau-0.5-old/configure.ac	2012-09-04 10:26:33.000000000 -0700
+++ libvdpau-0.5-new/configure.ac	2012-12-29 05:04:20.000000000 -0800
@@ -4,7 +4,7 @@
 AM_INIT_AUTOMAKE([dist-bzip2 foreign])
 AM_MAINTAINER_MODE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Disable static libraries by default.  Use --enable-static if you really want
 # them.
