--- ftgl.orig/unix/Makefile	2014-08-04 17:21:22.210742862 +0200
+++ ftgl/unix/Makefile	2003-01-12 22:27:04.000000000 +0100
@@ -1,7 +1,7 @@
 top_builddir=.
 -include $(top_builddir)/Make.conf
 
-SUBDIRS = src demo docs
+SUBDIRS = src demo
 
 AUTOGEN = Make.conf config.log config.status libtool autom4te.cache ftgl.pc
 AUTOGEN_MAINT = aclocal.m4 acinclude.m4 config.guess config.sub configure ltmain.sh install-sh

