diff -Naur Mesa-openchrome-svn-20090302/bin/mklib Mesa-openchrome-svn-20090302.patch/bin/mklib
--- Mesa-openchrome-svn-20090302/bin/mklib	2009-02-26 12:02:26.000000000 +0100
+++ Mesa-openchrome-svn-20090302.patch/bin/mklib	2009-03-03 01:48:26.000000000 +0100
@@ -217,9 +217,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
diff -Naur Mesa-openchrome-svn-20090302/configs/linux-dri Mesa-openchrome-svn-20090302.patch/configs/linux-dri
--- Mesa-openchrome-svn-20090302/configs/linux-dri	2009-02-26 12:02:26.000000000 +0100
+++ Mesa-openchrome-svn-20090302.patch/configs/linux-dri	2009-03-03 01:49:56.000000000 +0100
@@ -14,6 +14,7 @@
 #MKDEP_OPTIONS = -MF depend
 
 OPT_FLAGS  = -O2 -g
+HOST_OPT_FLAGS = $(OPT_FLAGS)
 PIC_FLAGS  = -fPIC
 
 # Add '-DGLX_USE_TLS' to ARCH_FLAGS to enable TLS support.
@@ -30,6 +31,9 @@
 CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
 	$(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
 
+HOST_CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
+	$(HOST_OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
+
 CXXFLAGS = -Wall $(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES)
 
 GLUT_CFLAGS = -fexceptions
