From 561186ae0cd8814a8d518eed424ae453a1dae1d3 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Mon, 27 Jun 2016 01:37:55 +0200
Subject: [PATCH] drivers/of: put extended cmdline at the end

---
 drivers/of/fdt.c | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/drivers/of/fdt.c b/drivers/of/fdt.c
index 2e9f7d96617..a29163d21f4 100644
--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -894,20 +894,12 @@ int __init early_init_dt_scan_chosen(unsigned long node, const char *uname,
 		p = (char *)of_get_flat_dt_prop(node, "bootargs", &l);
 
 	if (p != NULL && l > 0) {
-		if (concat_cmdline) {
-			int cmdline_len;
-			int copy_len;
-			strlcat(cmdline, " ", COMMAND_LINE_SIZE);
-			cmdline_len = strlen(cmdline);
-			copy_len = COMMAND_LINE_SIZE - cmdline_len - 1;
-			copy_len = min((int)l, copy_len);
-			strncpy(cmdline + cmdline_len, p, copy_len);
-			cmdline[cmdline_len + copy_len] = '\0';
-		} else {
-			strlcpy(cmdline, p, min((int)l, COMMAND_LINE_SIZE));
-		}
+		strlcpy(cmdline, p, min((int)l, COMMAND_LINE_SIZE));
 	}
 
+	if (concat_cmdline)
+		strlcat(cmdline, config_cmdline, COMMAND_LINE_SIZE);
+
 	pr_debug("Command line is: %s\n", (char*)data);
 
 	/* break now */
