--- a/Makefile	2020-02-27 19:24:52.878981499 +0100
+++ b/Makefile	2020-02-27 19:24:31.671275000 +0100
@@ -86,7 +86,7 @@
   $(shell $(MKDIR) $(dirname)))
 
 # Complie and link variables. LD_LIBS means the dynamic or static library needed for the object file.
-CFLAGS     	:= $(if $(DEBUG),-g -Wall, -O2 -Wall)
+CFLAGS     	:= -fPIC
 CFLAGS     	+= $(if $(GEN_DYN_LIB), $(addprefix -fPIC -I ,$(sort $(dir $(SRC_H)))), $(addprefix -I ,$(sort $(dir $(SRC_H)))))
 CXXFLAGS   	= $(CFLAGS)
 LDFLAGS    	:=
