diff -Naur wiringPi.git/wiringPi/Makefile wiringPi.patch/wiringPi/Makefile
--- wiringPi.git/wiringPi/Makefile	2017-01-08 18:29:40.981110547 +0100
+++ wiringPi.patch/wiringPi/Makefile	2017-01-08 18:30:13.802268055 +0100
@@ -36,7 +36,7 @@
 
 #DEBUG	= -g -O0
 DEBUG	= -O2
-CC	= gcc
+#CC	= gcc
 INCLUDE	= -I.
 DEFS	= -D_GNU_SOURCE
 CFLAGS	= $(DEBUG) $(DEFS) -Wformat=2 -Wall -Wextra -Winline $(INCLUDE) -pipe -fPIC
