fix libhdhomerun includes

--- a/configure
+++ b/configure
@@ -419,7 +419,7 @@ if enabled hdhomerun_static; then
 else
 
   if enabled_or_auto hdhomerun_client; then
-    if check_cc_header 'libhdhomerun/hdhomerun' libhdhomerun; then
+    if check_cc_header 'hdhomerun' libhdhomerun; then
       enable  hdhomerun_client
       LDFLAGS="$LDFLAGS -lhdhomerun"
     fi
--- a/src/input/mpegts/tvhdhomerun/tvhdhomerun.c
+++ b/src/input/mpegts/tvhdhomerun/tvhdhomerun.c
@@ -17,7 +17,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "libhdhomerun/hdhomerun.h"
+#include "hdhomerun.h"
 
 #include "tvheadend.h"
 #include "input.h"
--- a/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c
+++ b/src/input/mpegts/tvhdhomerun/tvhdhomerun_frontend.c
@@ -17,7 +17,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "libhdhomerun/hdhomerun.h"
+#include "hdhomerun.h"
 
 #include <fcntl.h>
 #include "tvheadend.h"
