From 73f3ba388b0a1109d039e71c0b0c859a1f5d0262 Mon Sep 17 00:00:00 2001
From: CvH <namerp@googlemail.com>
Date: Sat, 27 Feb 2021 07:03:24 +0100
Subject: [PATCH] dvb scan path

---
 src/input/mpegts/scanfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/input/mpegts/scanfile.c b/src/input/mpegts/scanfile.c
index 8a16c44ff..05725a90b 100644
--- a/src/input/mpegts/scanfile.c
+++ b/src/input/mpegts/scanfile.c
@@ -922,7 +922,7 @@ scanfile_init ( const char *muxconf_path, int lock )
 #elif defined(PLATFORM_FREEBSD)
     path = "/usr/local/share/dtv-scan-tables";
 #else
-    path = "/usr/share/dvb";
+    path = "/storage/.kodi/addons/service.tvheadend43/dvb-scan";
 #endif
 
   if (!initialized) {
-- 
2.51.0

