diff -Naur pymetar-0.15/setup.py pymetar-0.15.patch/setup.py
--- pymetar-0.15/setup.py	2009-06-28 18:39:44.000000000 +0200
+++ pymetar-0.15.patch/setup.py	2009-07-19 12:35:02.850182624 +0200
@@ -27,5 +27,5 @@
       py_modules=["pymetar"],
       scripts = [ "bin/weather.py" ], 
       data_files = [("share/doc/pymetar-%s/"%pymetar.__version__,
-       ['README', 'COPYING', 'THANKS', 'TODO', 'bin/example.py', 'librarydoc.txt'])]
+       ['README', 'COPYING', 'THANKS', 'librarydoc.txt'])]
       )
