diff -Naur vdr-dvbapi-plugin-6e8cad7/Makefile vdr-dvbapi-plugin-6e8cad7.patch/Makefile
--- vdr-dvbapi-plugin-6e8cad7/Makefile	2011-12-25 14:24:33.000000000 +0100
+++ vdr-dvbapi-plugin-6e8cad7.patch/Makefile	2011-12-29 18:59:21.847652339 +0100
@@ -80,7 +80,7 @@
 	gcc -O -fbuiltin -fomit-frame-pointer -fPIC -shared -o $@ $< -ldl
 	
 $(FFDECSA): $(FFDECSADIR)/*.c $(FFDECSADIR)/*.h
-	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CSAFLAGS) -march=$(CPUOPT)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
+	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CSAFLAGS)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) FFdecsa.o
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
