diff -uNr a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
--- a/yabause/src/libretro/Makefile	2026-02-06 22:48:31.000000000 +0900
+++ b/yabause/src/libretro/Makefile	2026-02-08 18:07:03.847967150 +0900
@@ -127,6 +127,28 @@
 	DYNAREC_DEVMIYAX = 1
 	FLAGS += -march=armv8-a+crc -mtune=cortex-a73.cortex-a53 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad
 
+# Rockchip RK356X / RK3576 / RK3588
+else ifneq (,$(findstring RK35,$(platform)))
+	override platform += unix
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+	LDFLAGS += -lpthread
+	FLAGS += -DAARCH64 -march=armv8-a+crc+fp+simd
+	ARCH_IS_LINUX = 1
+	HAVE_SSE = 0
+	FORCE_GLES = 1
+	USE_AARCH64_DRC = 1
+	DYNAREC_DEVMIYAX = 1
+	
+	ifneq (,$(findstring RK356X,$(platform)))
+		FLAGS += -mcpu=cortex-a55 -mtune=cortex-a55
+	else ifneq (,$(findstring RK3576,$(platform)))
+		FLAGS += -mcpu=cortex-a72.cortex-a53 -mtune=cortex-a72.cortex-a53
+	else ifneq (,$(findstring RK3588,$(platform)))
+		FLAGS += -mcpu=cortex-a76.cortex-a55 -mtune=cortex-a76.cortex-a55
+	endif
+
 # Rockchip RK3288 / RK3399 - 32-bit userspace
 else ifneq (,$(findstring RK,$(platform)))
 	override platform += unix
