From b0f33807f0dadf11d1b007204607d98d9f17d8f7 Mon Sep 17 00:00:00 2001
From: lufi <lutz@fiebach.de>
Date: Sun, 18 Aug 2013 22:16:45 +0200
Subject: [PATCH 1/2] Update defaults.py

---
 defaults.py | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/defaults.py b/defaults.py
index 7a582de..16e78e8 100755
--- a/defaults.py
+++ b/defaults.py
@@ -144,11 +144,11 @@
 
 _services = \
     {
-        "sshd"      : ["sshd.service"],
-        "avahi"     : ["avahi-daemon.service"],
-        "samba"     : ["nmbd.service", "smbd.service"],
-        "bluetooth" : ["bluetooth.service"],
-        "obexd"     : ["obex.service"],
-        "syslog"    : ["syslog.service"],
-        "cron"      : ["cron.service"],
-    }
\ No newline at end of file
+        "sshd"   : ["sshd.service"],
+        "avahi"  : ["avahi-daemon.service"],
+        "samba"  : ["nmbd.service", "smbd.service"],
+        "bluez"  : ["bluetooth.service"],
+        "obexd"  : ["obex.service"],
+        "syslog" : ["syslog.service"],
+        "cron"   : ["cron.service"],
+    }
-- 
1.8.1.6


From 26d02eb4b2ca01e975a4c0b4c8a91a8010147cbf Mon Sep 17 00:00:00 2001
From: lufi <lutz@fiebach.de>
Date: Sun, 18 Aug 2013 22:21:28 +0200
Subject: [PATCH 2/2] change old service config files to .cache/services/ dir

---
 oe.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/oe.py b/oe.py
index 7df2ba2..d3db4b4 100755
--- a/oe.py
+++ b/oe.py
@@ -213,7 +213,7 @@ def set_service_option(service, option, value):
         lines = []        
         changed = False
 
-        conf_file_name = '%s/service_%s.conf' % (CONFIG_CACHE, service)
+        conf_file_name = '%s/services/%s.conf' % (CONFIG_CACHE, service)
                         
         if os.path.isfile(conf_file_name):
             with open(conf_file_name, "r") as conf_file:
@@ -239,7 +239,7 @@ def get_service_option(service, option, default=None):
         conf_file_name = ''
         
         if not SYSTEMD:
-            conf_file_name = '%s/service_%s.conf' % (CONFIG_CACHE, service)
+            conf_file_name = '%s/services/%s.conf' % (CONFIG_CACHE, service)
         else:
             if os.path.exists('%s/services/%s.conf' % (CONFIG_CACHE, service)):
                 conf_file_name = '%s/services/%s.conf' % (CONFIG_CACHE, service)
-- 
1.8.1.6

