From a9293db50e645a44b8cc2c4b82f9dc521c27d076 Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Tue, 12 May 2009 14:29:24 -0600
Subject: [PATCH 09/14] es: update window title

---
 progs/es1/xegl/tri.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/progs/es1/xegl/tri.c b/progs/es1/xegl/tri.c
index ca37c2d..233c021 100644
--- a/progs/es1/xegl/tri.c
+++ b/progs/es1/xegl/tri.c
@@ -427,7 +427,7 @@ main(int argc, char *argv[])
    printf("EGL_CLIENT_APIS = %s\n", s);
 
    make_x_window(x_dpy, egl_dpy,
-                 "tri", 0, 0, winWidth, winHeight,
+                 "OpenGL ES 1.x tri", 0, 0, winWidth, winHeight,
                  &win, &egl_ctx, &egl_surf);
 
    XMapWindow(x_dpy, win);
-- 
1.6.2.4

