From b8e64cc4390fb2f499f659374d8b4705846c7a19 Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Tue, 12 May 2009 14:29:33 -0600
Subject: [PATCH 10/14] es: updated Makefile comment

---
 progs/es1/xegl/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/progs/es1/xegl/Makefile b/progs/es1/xegl/Makefile
index fe0b05c..4b21ee3 100644
--- a/progs/es1/xegl/Makefile
+++ b/progs/es1/xegl/Makefile
@@ -1,4 +1,4 @@
-# progs/gles/xegl/Makefile
+# progs/es1/xegl/Makefile
 
 TOP = ../../..
 include $(TOP)/configs/current
-- 
1.6.2.4

