From ea625e90c1f9058a67e1a7906c8dce3224e2ee13 Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Tue, 12 May 2009 14:30:40 -0600
Subject: [PATCH 12/14] es: fix comment

---
 progs/es2/xegl/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/progs/es2/xegl/Makefile b/progs/es2/xegl/Makefile
index 8904203..bd5611a 100644
--- a/progs/es2/xegl/Makefile
+++ b/progs/es2/xegl/Makefile
@@ -1,4 +1,4 @@
-# progs/es/es2/xegl/Makefile
+# progs/es2/xegl/Makefile
 
 TOP = ../../..
 include $(TOP)/configs/current
-- 
1.6.2.4

