diff --git a/configure b/configure
index a58ee46..69d8e56 100755
--- a/configure
+++ b/configure
@@ -3357,7 +3357,7 @@ rm -f conftest*
 
 
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
+LIBS="$LIBS"
 { echo "$as_me:$LINENO: checking SVR4..." >&5
 echo "$as_me: checking SVR4..." >&6;}
 echo "$as_me:$LINENO: checking for ANSI C header files" >&5
diff --git a/configure.in b/configure.in
index 34c9372..b189549 100644
--- a/configure.in
+++ b/configure.in
@@ -196,7 +196,7 @@ AC_EGREP_CPP(yes,
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
+LIBS="$LIBS"
 AC_CHECKING(SVR4)
 AC_TRY_LINK([#include <utmpx.h>
 ],,
