diff -Naur boom3-72fc7f24852689287a9fcf7628e8f2e665bd1b33/neo/Makefile boom3-72fc7f24852689287a9fcf7628e8f2e665bd1b33-2/neo/Makefile
--- boom3-72fc7f24852689287a9fcf7628e8f2e665bd1b33/neo/Makefile	2022-03-02 06:57:12.016089897 +0100
+++ boom3-72fc7f24852689287a9fcf7628e8f2e665bd1b33-2/neo/Makefile	2022-03-02 07:05:00.863947991 +0100
@@ -117,6 +117,16 @@
    AR = arm-vita-eabi-ar
    CXXFLAGS += -Wl,-q -Wall -O3
    STATIC_LINKING = 1
+else ifeq ($(platform), tegra210)
+   EXT ?= so
+   TARGET := $(TARGET_NAME)_libretro.$(EXT)
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+   LIBS := -ldl -lpthread
+   CFLAGS      += -march=armv8-a -mtune=cortex-a57 -mcpu=cortex-a57+crc+fp+simd
+   CXXFLAGS := $(ASFLAGS) $(CFLAGS) -std=gnu++11
+   GL_LIB := -lGL
+   D3XP := 1
 # Nintendo Switch (libnx)
 else ifeq ($(platform), libnx)
     include $(DEVKITPRO)/libnx/switch_rules

