diff -Naur xorg-server-1.5.99.902/hw/xfree86/loader/loaderProcs.h xorg-server-1.5.99.902.patch/hw/xfree86/loader/loaderProcs.h
--- xorg-server-1.5.99.902/hw/xfree86/loader/loaderProcs.h	2009-01-31 05:21:50.000000000 +0100
+++ xorg-server-1.5.99.902.patch/hw/xfree86/loader/loaderProcs.h	2009-02-19 05:16:33.000000000 +0100
@@ -56,7 +56,6 @@
 #undef IN_LOADER
 #define IN_LOADER
 #include "xf86Module.h"
-#include <X11/fonts/fontmod.h>
 
 typedef struct module_desc {
     struct module_desc *child;
