diff -Naur xf86-video-fglrx-11.5/common/lib/modules/fglrx/build_mod/drmP.h xf86-video-fglrx-11.5.patch/common/lib/modules/fglrx/build_mod/drmP.h
--- xf86-video-fglrx-11.5/common/lib/modules/fglrx/build_mod/drmP.h	2011-05-03 17:19:14.000000000 +0200
+++ xf86-video-fglrx-11.5.patch/common/lib/modules/fglrx/build_mod/drmP.h	2011-05-11 17:39:25.549917899 +0200
@@ -57,7 +57,9 @@
 #include <linux/pci.h>
 #include <linux/version.h>
 #include <linux/sched.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,39)
 #include <linux/smp_lock.h>	/* For (un)lock_kernel */
+#endif
 #include <linux/mm.h>
 #include <linux/pagemap.h>
 #if defined(__alpha__) || defined(__powerpc__)
diff -Naur xf86-video-fglrx-11.5/common/lib/modules/fglrx/build_mod/firegl_public.c xf86-video-fglrx-11.5.patch/common/lib/modules/fglrx/build_mod/firegl_public.c
--- xf86-video-fglrx-11.5/common/lib/modules/fglrx/build_mod/firegl_public.c	2011-05-03 17:19:14.000000000 +0200
+++ xf86-video-fglrx-11.5.patch/common/lib/modules/fglrx/build_mod/firegl_public.c	2011-05-11 17:52:40.421297182 +0200
@@ -114,7 +114,9 @@
 #include <linux/pci.h>
 #include <linux/wait.h>
 #include <linux/miscdevice.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,39)
 #include <linux/smp_lock.h>
+#endif
 // newer SuSE kernels need this
 #include <linux/highmem.h>
 
@@ -1049,7 +1051,11 @@
     dev->pubdev.signature = FGL_DEVICE_SIGNATURE;
 
     for (i = 0; i < __KE_MAX_SPINLOCKS; i++)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,39)
         dev->spinlock[i] = SPIN_LOCK_UNLOCKED;
+#else
+        dev->spinlock[i] = __SPIN_LOCK_UNLOCKED(old_style_spin_init);
+#endif
 
     for (i=0; i < __KE_MAX_SEMAPHORES; i++)
         sema_init(&dev->struct_sem[i], 1);
@@ -1900,13 +1906,17 @@
 /** \brief Grab global kernel lock */
 void ATI_API_CALL KCL_GlobalKernelLock(void)
 {
+#ifdef CONFIG_KERNEL_LOCK
     lock_kernel();
+#endif
 }
 
 /** \brief Release global kernel lock */
 void ATI_API_CALL KCL_GlobalKernelUnlock(void)
 {
+#ifdef CONFIG_KERNEL_LOCK
     unlock_kernel();
+#endif
 }
 
 /*****************************************************************************/
