diff --git a/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h b/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h
index b358d0b..d45a78a 100755
--- a/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h
+++ b/common/lib/modules/fglrx/build_mod/fglrxko_pci_ids.h
@@ -97,7 +97,6 @@
     FGL_ASIC_ID(0x677B),
     FGL_ASIC_ID(0x6772),
     FGL_ASIC_ID(0x6779),
-    FGL_ASIC_ID(0x6770),
     FGL_ASIC_ID(0x671F),
     FGL_ASIC_ID(0x6718),
     FGL_ASIC_ID(0x6719),
@@ -147,7 +146,6 @@
     FGL_ASIC_ID(0x68D1),
     FGL_ASIC_ID(0x68C7),
     FGL_ASIC_ID(0x68E0),
-    FGL_ASIC_ID(0x68E1),
     FGL_ASIC_ID(0x68F0),
     FGL_ASIC_ID(0x68F1),
     FGL_ASIC_ID(0x68E4),
