diff -Naur uxlaunch-0.46/xserver.c uxlaunch-0.46.patch/xserver.c
--- uxlaunch-0.46/xserver.c	2010-03-04 19:51:44.000000000 +0100
+++ uxlaunch-0.46.patch/xserver.c	2010-03-24 17:51:31.408908456 +0100
@@ -230,12 +230,12 @@
 	if (!ret && (statbuf.st_mode & S_ISUID)) {
 		execl(xserver, xserver,  displayname,
 		      "-nolisten", "tcp", "-dpi", "120", "-noreset",
-		      "-auth", xauth_cookie_file,
+		      "-auth", xauth_cookie_file, "-nocursor", "-nohwaccess",
 		      vt, NULL);
 	} else {
 		execl(xserver, xserver,  displayname,
 		      "-nolisten", "tcp", "-dpi", "120", "-noreset",
-		      "-auth", user_xauth_path,
+		      "-auth", user_xauth_path, "-nocursor", "-nohwaccess",
 		      "-logfile", xorg_log,
 		      vt, NULL);
 	}
