commit 43aba5ffffc6d35ed97db035c5818c76652b06de
Author: Matthias Reichl <hias@horus.com>
Date:   Mon Sep 28 23:55:13 2020 +0200

    move /etc/systemd/system to /storage/.config/system.d

diff --git a/meson.build b/meson.build
index dbbddb68e2..4592cd1094 100644
--- a/meson.build
+++ b/meson.build
@@ -205,7 +205,7 @@ memory_accounting_default = get_option('memory-accounting-default')
 status_unit_format_default = get_option('status-unit-format-default')
 
 conf.set_quoted('PKGSYSCONFDIR',                              pkgsysconfdir)
-conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     join_paths(pkgsysconfdir, 'system'))
+conf.set_quoted('SYSTEM_CONFIG_UNIT_DIR',                     '/storage/.config/system.d')
 conf.set_quoted('SYSTEM_DATA_UNIT_PATH',                      systemunitdir)
 conf.set_quoted('SYSTEM_SYSVINIT_PATH',                       sysvinit_path)
 conf.set_quoted('SYSTEM_SYSVRCND_PATH',                       sysvrcnd_path)
