diff -Naur wayland-20090913/Makefile.in wayland-20090913.patch/Makefile.in
--- wayland-20090913/Makefile.in	2009-05-29 02:23:31.000000000 +0200
+++ wayland-20090913.patch/Makefile.in	2009-09-18 16:14:57.000000000 +0200
@@ -27,7 +27,7 @@
 $(libs) : CFLAGS += -fPIC @FFI_CFLAGS@
 $(libs) : LDLIBS += @FFI_LIBS@
 $(libs) :
-	gcc -shared $^ $(LDLIBS)  -o $@
+	$(CC) -shared $^ $(LDLIBS)  -o $@
 
 wayland-system-compositor :			\
 	wayland-system-compositor.o		\
@@ -54,11 +54,11 @@
 $(cairo_clients) : LDLIBS += -L. -lwayland @CAIRO_CLIENT_LIBS@ -lrt
 
 install : $(libs) $(compositors)
-	install -d @libdir@ @libdir@/pkgconfig ${udev_rules_dir}
-	install $(libs) @libdir@
-	install wayland-server.pc wayland.pc @libdir@/pkgconfig
-	install wayland-client.h @includedir@
-	install 70-wayland.rules ${udev_rules_dir}
+	install -d ${DESTDIR}@libdir@ ${DESTDIR}@libdir@/pkgconfig ${DESTDIR}${udev_rules_dir}
+	install $(libs) ${DESTDIR}/@libdir@
+	install wayland-server.pc wayland.pc ${DESTDIR}@libdir@/pkgconfig
+	install wayland-client.h ${DESTDIR}@includedir@
+	install 70-wayland.rules ${DESTDIR}${udev_rules_dir}
 
 clean :
 	rm -f $(clients) $(compositors) *.o *.so
