diff -Naur eventlircd-17/src/input.c eventlircd-17.patch/src/input.c
--- eventlircd-17/src/input.c	2010-10-17 22:18:18.000000000 +0200
+++ eventlircd-17.patch/src/input.c	2010-10-29 01:16:56.118197221 +0200
@@ -1067,12 +1067,12 @@
             {
                 time_delta = 1000000 * (device->current.event_out.time.tv_sec  - previous->event_out.time.tv_sec ) +
                                        (device->current.event_out.time.tv_usec - previous->event_out.time.tv_usec);
-                if (((previous->repeat_count == 0) && (time_delta <  900000)) ||
-                    ((previous->repeat_count == 1) && (time_delta <  500000)) ||
-                    ((previous->repeat_count == 2) && (time_delta <  300000)) ||
-                    ((previous->repeat_count == 3) && (time_delta <  200000)) ||
-                    ((previous->repeat_count == 4) && (time_delta <  150000)) ||
-                    ((previous->repeat_count >= 5) && (time_delta <  100000)))
+                if (((previous->repeat_count == 0) && (time_delta <  180000)) ||
+                    ((previous->repeat_count == 1) && (time_delta <  125000)) ||
+                    ((previous->repeat_count == 2) && (time_delta <  75000)) ||
+                    ((previous->repeat_count == 3) && (time_delta <  50000)) ||
+                    ((previous->repeat_count == 4) && (time_delta <  35000)) ||
+                    ((previous->repeat_count >= 5) && (time_delta <  25000)))
                 {
                     memset(&(device->current.event_out), 0, sizeof(struct input_event));
                     device->current.event_out.type = LIRCUDEVD_EV_NULL;
