diff --git a/udev/lircd_helper.in b/udev/lircd_helper.in
index f580cc6..6befeed 100644
--- a/udev/lircd_helper.in
+++ b/udev/lircd_helper.in
@@ -41,7 +41,11 @@ case "${ACTION}" in
             daemon="${daemon} --uinput"
-            daemon="${daemon} --output=@localstatedir@/run/lirc/lircd-${devname_instance}"
-            daemon="${daemon} --pidfile=@localstatedir@/run/lirc/lircd-${devname_instance}.pid"
-            daemon="${daemon} ${lircd_conf}"
+            daemon="${daemon} --output=@localstatedir@/run/lirc/lircd"
+            daemon="${daemon} --pidfile=@localstatedir@/run/lirc/lircd-${devname_instance}.pid"
+            if test -e "/storage/.config/lircd.conf" ; then
+                daemon="${daemon} /storage/.config/lircd.conf"
+            else
+                daemon="${daemon} ${lircd_conf}"
+            fi
             ${daemon}
             for devlink in ${DEVLINKS} ; do
                 devlink_instance=`echo ${devlink} | /bin/sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
