diff -Naur eventlircd-17/udev/lircd_helper.in eventlircd-17.patch/udev/lircd_helper.in
--- eventlircd-17/udev/lircd_helper.in	2010-10-17 22:18:18.000000000 +0200
+++ eventlircd-17.patch/udev/lircd_helper.in	2010-10-17 22:23:51.144592363 +0200
@@ -53,7 +53,7 @@
     remove)
         instance=`echo $DEVNAME | sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
         if test -e "@localstatedir@/run/lirc/lircd-${instance}.pid" ; then
-            pid=`cat @localstate@/run/lirc/lircd-${instance}.pid`
+            pid=`cat @localstatedir@/run/lirc/lircd-${instance}.pid`
             if test ! "x${pid}" = "x" ; then
                 kill ${pid}
             fi
diff -Naur eventlircd-17/udev/wakeup_enable.in eventlircd-17.patch/udev/wakeup_enable.in
--- eventlircd-17/udev/wakeup_enable.in	2010-10-17 22:18:18.000000000 +0200
+++ eventlircd-17.patch/udev/wakeup_enable.in	2010-10-17 22:24:39.559156689 +0200
@@ -5,13 +5,13 @@
 
 devpath=`readlink -f "/sys/${DEVPATH}"`
 while echo "${devpath}" | /bin/grep -q '^/sys/devices/' > /dev/null 2>&1 ; do
-    if test -e "${devpath}/power/wakeup" ; then                                 
-        state=`cat "${devpath}/power/wakeup"`                                       
+    if test -e "${devpath}/power/wakeup" ; then
+        state=`cat "${devpath}/power/wakeup"`
         if test "x${state}" = "xdisabled" ; then
             echo "enabled" > "${devpath}/power/wakeup"
-        fi                                                 
-    fi                                                     
+        fi
+    fi
     devpath=`dirname "${devpath}"`
-done                                       
-                                           
+done
+
 exit 0
