diff -Naur eventlircd-17/etc/ircore.evmap eventlircd-17.patch/etc/ircore.evmap
--- eventlircd-17/etc/ircore.evmap	1970-01-01 01:00:00.000000000 +0100
+++ eventlircd-17.patch/etc/ircore.evmap	2010-09-18 19:43:56.000000000 +0200
@@ -0,0 +1,22 @@
+KEY_1          = KEY_NUMERIC_1
+KEY_2          = KEY_NUMERIC_2
+KEY_3          = KEY_NUMERIC_3
+KEY_4          = KEY_NUMERIC_4
+KEY_5          = KEY_NUMERIC_5
+KEY_6          = KEY_NUMERIC_6
+KEY_7          = KEY_NUMERIC_7
+KEY_8          = KEY_NUMERIC_8
+KEY_9          = KEY_NUMERIC_9
+KEY_0          = KEY_NUMERIC_0
+KEY_E          = KEY_BACK
+KEY_KP1        = KEY_NUMERIC_1
+KEY_KP2        = KEY_NUMERIC_2
+KEY_KP3        = KEY_NUMERIC_3
+KEY_KP4        = KEY_NUMERIC_4
+KEY_KP5        = KEY_NUMERIC_5
+KEY_KP6        = KEY_NUMERIC_6
+KEY_KP7        = KEY_NUMERIC_7
+KEY_KP8        = KEY_NUMERIC_8
+KEY_KP9        = KEY_NUMERIC_9
+KEY_KP0        = KEY_NUMERIC_0
+KEY_KPASTERISK = KEY_NUMERIC_STAR
diff -Naur eventlircd-17/udev/lircd_helper.in eventlircd-17.patch/udev/lircd_helper.in
--- eventlircd-17/udev/lircd_helper.in	2010-09-18 19:43:56.000000000 +0200
+++ eventlircd-17.patch/udev/lircd_helper.in	2010-09-27 19:46:24.948885231 +0200
@@ -53,7 +53,7 @@
     remove)
         instance=`echo $DEVNAME | sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
         if test -e "@localstatedir@/run/lirc/lircd-${instance}.pid" ; then
-            pid=`cat @localstate@/run/lirc/lircd-${instance}.pid`
+            pid=`cat @localstatedir@/run/lirc/lircd-${instance}.pid`
             if test ! "x${pid}" = "x" ; then
                 kill ${pid}
             fi
diff -Naur eventlircd-17/udev/rules.d/98-eventlircd.rules.disabled.in eventlircd-17.patch/udev/rules.d/98-eventlircd.rules.disabled.in
--- eventlircd-17/udev/rules.d/98-eventlircd.rules.disabled.in	2010-09-18 19:43:56.000000000 +0200
+++ eventlircd-17.patch/udev/rules.d/98-eventlircd.rules.disabled.in	2010-09-27 21:21:25.123009701 +0200
@@ -51,6 +51,13 @@
   ENV{eventlircd_enable}="true"
 
 #-------------------------------------------------------------------------------
+# Ask eventlircd to handle input event devices created by ir-core (kernel).
+#-------------------------------------------------------------------------------
+SUBSYSTEMS=="rc", \
+  ENV{eventlircd_enable}="true"
+  ENV{eventlircd_evmap}="ircore.evmap"
+
+#-------------------------------------------------------------------------------
 # Ask eventlircd to handle USB HID devices that show up as event devices and are
 # known to be remote controls. For simplicity, the event map file names have the
 # format <BUSTYPE>_<VENDOR>_<PRODUCT>.evmap.
diff -Naur eventlircd-17/udev/wakeup_enable.in eventlircd-17.patch/udev/wakeup_enable.in
--- eventlircd-17/udev/wakeup_enable.in	2010-09-18 19:43:56.000000000 +0200
+++ eventlircd-17.patch/udev/wakeup_enable.in	2010-09-27 19:53:32.215884397 +0200
@@ -5,13 +5,13 @@
 
 devpath=`readlink -f "/sys/${DEVPATH}"`
 while echo "${devpath}" | /bin/grep -q '^/sys/devices/' > /dev/null 2>&1 ; do
-    if test -e "${devpath}/power/wakeup" ; then                                 
-        state=`cat "${devpath}/power/wakeup"`                                       
+    if test -e "${devpath}/power/wakeup" ; then
+        state=`cat "${devpath}/power/wakeup"`
         if test "x${state}" = "xdisabled" ; then
             echo "enabled" > "${devpath}/power/wakeup"
-        fi                                                 
-    fi                                                     
+        fi
+    fi
     devpath=`dirname "${devpath}"`
-done                                       
-                                           
+done
+
 exit 0
