diff -Naur eventlircd-42/udev/wakeup_enable.in eventlircd-42.patch/udev/wakeup_enable.in
--- eventlircd-42/udev/wakeup_enable.in	2011-04-01 16:42:27.000000000 +0200
+++ eventlircd-42.patch/udev/wakeup_enable.in	2011-11-28 02:05:59.743690557 +0100
@@ -4,7 +4,7 @@
 ################################################################################
 
 devpath=`readlink -f "/sys/${DEVPATH}"`
-while echo "${devpath}" | /bin/grep -q '^/sys/devices/' > /dev/null 2>&1 ; do
+while echo "${devpath}" | grep -q '^/sys/devices/' > /dev/null 2>&1 ; do
     if test -e "${devpath}/power/wakeup" ; then
         state=`cat "${devpath}/power/wakeup"`
         if test "x${state}" = "xdisabled" ; then
