diff --git a/udev/lircd_helper.in b/udev/lircd_helper.in
index f580cc6..d4d7f3f 100644
--- a/udev/lircd_helper.in
+++ b/udev/lircd_helper.in
@@ -41,7 +41,13 @@
             daemon="${daemon} --uinput"
             daemon="${daemon} --output=@localstatedir@/run/lirc/lircd-${devname_instance}"
             daemon="${daemon} --pidfile=@localstatedir@/run/lirc/lircd-${devname_instance}.pid"
-            daemon="${daemon} ${lircd_conf}"
+            if test -e "/storage/.config/lircd.conf" ; then
+                daemon="${daemon} /storage/.config/lircd.conf"
+            elif test -e "${lircd_conf}" ; then
+                daemon="${daemon} ${lircd_conf}"
+            else
+                exit 1
+            fi
             ${daemon}
             for devlink in ${DEVLINKS} ; do
                 devlink_instance=`echo ${devlink} | /bin/sed -e 's/\/\+/~/g' -e 's/^~dev~//'`
