diff -Naur eventlircd-17/etc/ircore.evmap eventlircd-17.patch/etc/ircore.evmap
--- eventlircd-17/etc/ircore.evmap	1970-01-01 01:00:00.000000000 +0100
+++ eventlircd-17.patch/etc/ircore.evmap	2010-09-28 17:44:57.652885417 +0200
@@ -0,0 +1,33 @@
+KEY_1            = KEY_NUMERIC_1
+KEY_2            = KEY_NUMERIC_2
+KEY_3            = KEY_NUMERIC_3
+KEY_4            = KEY_NUMERIC_4
+KEY_5            = KEY_NUMERIC_5
+KEY_6            = KEY_NUMERIC_6
+KEY_7            = KEY_NUMERIC_7
+KEY_8            = KEY_NUMERIC_8
+KEY_9            = KEY_NUMERIC_9
+KEY_0            = KEY_NUMERIC_0
+KEY_STAR         = KEY_NUMERIC_STAR
+KEY_POUND        = KEY_NUMERIC_POUND
+KEY_KP1          = KEY_NUMERIC_1
+KEY_KP2          = KEY_NUMERIC_2
+KEY_KP3          = KEY_NUMERIC_3
+KEY_KP4          = KEY_NUMERIC_4
+KEY_KP5          = KEY_NUMERIC_5
+KEY_KP6          = KEY_NUMERIC_6
+KEY_KP7          = KEY_NUMERIC_7
+KEY_KP8          = KEY_NUMERIC_8
+KEY_KP9          = KEY_NUMERIC_9
+KEY_KP0          = KEY_NUMERIC_0
+KEY_KPASTERISK   = KEY_NUMERIC_STAR
+
+KEY_E            = KEY_EXIT
+KEY_BACK         = KEY_EXIT
+KEY_STOPCD       = KEY_STOP
+KEY_NEXTSONG     = KEY_NEXT
+KEY_PREVIOUSSONG = KEY_PREVIOUS
+KEY_PROPS        = KEY_INFO
+KEY_ANGLE        = KEY_ZOOM
+KEY_MP3          = KEY_AUDIO
+KEY_TEXT         = KEY_EPG
diff -Naur eventlircd-17/udev/lircd_helper.in eventlircd-17.patch/udev/lircd_helper.in
diff -Naur eventlircd-17/udev/rules.d/98-eventlircd.rules.disabled.in eventlircd-17.patch/udev/rules.d/98-eventlircd.rules.disabled.in
--- eventlircd-17/udev/rules.d/98-eventlircd.rules.disabled.in	2010-09-18 19:43:56.000000000 +0200
+++ eventlircd-17.patch/udev/rules.d/98-eventlircd.rules.disabled.in	2010-09-30 17:28:57.741250130 +0200
@@ -51,6 +51,13 @@
   ENV{eventlircd_enable}="true"
 
 #-------------------------------------------------------------------------------
+# Ask eventlircd to handle input event devices created by ir-core (kernel).
+#-------------------------------------------------------------------------------
+SUBSYSTEMS=="rc", \
+  ENV{eventlircd_enable}="true"
+  ENV{eventlircd_evmap}="ircore.evmap"
+
+#-------------------------------------------------------------------------------
 # Ask eventlircd to handle USB HID devices that show up as event devices and are
 # known to be remote controls. For simplicity, the event map file names have the
 # format <BUSTYPE>_<VENDOR>_<PRODUCT>.evmap.
@@ -77,9 +84,9 @@
   ENV{eventlircd_enable}="true", \
   ENV{eventlircd_evmap}="03_$env{ID_VENDOR_ID}_$env{ID_MODEL_ID}.evmap"
 
+LABEL="end-usb"
+
 # Enable wake-on-usb for the USB remotes.
 ENV{eventlircd_enable}=="true", RUN+="wakeup_enable"
 
-LABEL="end-usb"
-
 LABEL="end"
diff -Naur eventlircd-17/udev/wakeup_enable.in eventlircd-17.patch/udev/wakeup_enable.in
