diff --git a/configure.ac b/configure.ac
index 671241e..b6fece6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -294,42 +294,6 @@ if test x$libpcap_prefix = x ; then
     libpcap_prefix="/usr /usr/local /opt /software"
 fi
 
-AC_MSG_CHECKING([where to find pcap.h])
-foundpcaph=0
-oldCPPFLAGS=$CPPFLAGS
-for test_prefix in "" $libpcap_prefix ; do
-    for x in "" /pcap ; do
-        if test x$test_prefix != x ; then
-            CPPFLAGS="$oldCPPFLAGS -I$test_prefix/include$x"
-        fi
-        AC_TRY_CPP([
-#include <pcap.h>
-        ], [
-        AC_MSG_RESULT([$test_prefix/include$x])
-        foundpcaph=1
-        break
-        ])
-    done
-    if test $foundpcaph = 1 ; then
-        break
-    fi
-done
-
-if test $foundpcaph = 0 ; then
-    AC_MSG_RESULT([no idea])
-    AC_MSG_ERROR([can't find pcap.h
-  You're not going to get very far without libpcap.])
-else
-    dnl assume that -lpcap is under $test_prefix/lib
-    if test x$test_prefix != x ; then
-        LDFLAGS="$LDFLAGS -L$test_prefix/lib"
-    fi
-    AC_CHECK_LIB(pcap, pcap_open_live, , [
-            AC_MSG_ERROR([can't find libpcap
-  You're not going to get very far without libpcap.])
-        ])
-fi
-
 foundpcap=0
 AC_CHECK_HEADERS([pcap.h pcap/pcap.h], [
     foundpcap=1
diff --git a/edline.c b/edline.c
index 75f1a5a..aed80cf 100644
--- a/edline.c
+++ b/edline.c
@@ -7,7 +7,7 @@
 static const char rcsid[] = "$Id: edline.c,v 1.2 2002/11/04 12:27:35 chris Exp $";
 
 #include <ctype.h>
-#include <curses.h>
+#include <ncurses.h>
 #include <string.h>
 
 #include "iftop.h"
diff --git a/iftop.c b/iftop.c
index a090dcf..5851c86 100644
--- a/iftop.c
+++ b/iftop.c
@@ -23,7 +23,7 @@
 #endif
 
 #include <pthread.h>
-#include <curses.h>
+#include <ncurses.h>
 #include <signal.h>
 #include <string.h>
 #include <unistd.h>
