From bba53cbeaf3cd17b261c49e6c9a440710990a8c6 Mon Sep 17 00:00:00 2001
From: natinusala <natinusala@gmail.com>
Date: Mon, 18 Nov 2019 13:18:13 +0100
Subject: [PATCH] Makefile: properly link against GL on unix

---
 Makefile | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7146d4a..a5123d2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 STATIC_LINKING := 0
 AR             := ar
 HAVE_OPENGL    := 0
+LIBGL          := 
 
 ifneq ($(V),1)
    Q := @
@@ -75,7 +76,8 @@ ifeq ($(platform), unix)
 	EXT ?= so
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
-	HAVE_OPENGL = 1
+   HAVE_OPENGL = 1
+   LIBGL = -lGL
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
 else ifeq ($(platform), linux-portable)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
@@ -137,11 +139,15 @@ else
    TARGET := $(TARGET_NAME)_libretro.dll
    HAVE_OPENGL = 1
    SHARED := -shared -static-libgcc -static-libstdc++ -s -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LDFLAGS += -lopengl32
+   LIBGL = -lopengl32
 endif
 
 LDFLAGS += $(LIBM)
 
+ifeq ($(HAVE_OPENGL),1)
+LDFLAGS += $(LIBGL)
+endif
+
 ifeq ($(DEBUG), 1)
    CFLAGS += -O0 -g -DDEBUG
    CXXFLAGS += -O0 -g -DDEBUG
