--- a/config.sh
+++ b/config.sh
@@ -698,7 +698,7 @@ do
 		break
 	;;
 	'-r'|'--oscam-revision')
-		revision=`(svnversion -n . 2>/dev/null || printf 0) | sed 's/.*://; s/[^0-9]*$//; s/^$/0/'`
+		echo "$OSCAM_ADDON_VERSION"
 		if [ "$revision" = "0" ]
 		then
 			which git > /dev/null 2>&1 && revision=`git log -10 --pretty=%B | grep git-svn-id | head -n 1 | sed -n -e 's/^.*trunk@\([0-9]*\) .*$/\1/p'`
