From ead91a9770db872c74a1f3b142e61aa29561ff51 Mon Sep 17 00:00:00 2001
From: CvH <namerp@googlemail.com>
Date: Mon, 4 Apr 2016 19:07:14 +0200
Subject: [PATCH] link with ludev

---
 CMakeLists.txt       | 2 +-
 utils/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b08cfc0..2eadd8e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -626,7 +626,7 @@ else (NOT USE_COMPRESS EQUAL 1)
 	set (exe_name "oscam-upx")
 endif (NOT USE_COMPRESS EQUAL 1)
 add_executable (${exe_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo)
+target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo udev)
 if (WITH_SIGNING EQUAL 1)
 	SIGN_COMMAND_OSCAM()
 endif (WITH_SIGNING EQUAL 1)
diff --git a/utils/CMakeLists.txt b/utils/CMakeLists.txt
index 49649863..1c368dac 100644
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -12,7 +12,7 @@ file (GLOB all_srcs ${exe_srcs})
 
 set (util_name "list_smargo")
 add_executable (${util_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link})
+target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link} udev)
 #----------------------- printout resume -----------------------------
 
 message (STATUS "Utils: operating system: ${OSCamOperatingSystem}")
-- 
2.51.0

