From c3d3cc2706717f91eabfdc4dccc6377e32cc3c6d Mon Sep 17 00:00:00 2001
From: CvH <namerp@googlemail.com>
Date: Mon, 4 Apr 2016 19:07:14 +0200
Subject: [PATCH 2/4] pcsc pthread

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2eadd8e2..9e8b0fc9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -701,6 +701,7 @@ if (HAVE_PCSC)
 if (NOT OSCamOperatingSystem MATCHES "Mac OS X")
 if (NOT OSCamOperatingSystem MATCHES "Windows/Cygwin")
 	target_link_libraries (${exe_name} pcsclite)
+	target_link_libraries (${exe_name} pthread)
 endif (NOT OSCamOperatingSystem MATCHES "Windows/Cygwin")
 endif (NOT OSCamOperatingSystem MATCHES "Mac OS X")
 endif (HAVE_PCSC)
-- 
2.51.0

