diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 8eff4d7..2bf3318 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -1716,11 +1716,8 @@ case "$6" in
                 PATCHLEVEL=$(grep "^PATCHLEVEL =" $MAKEFILE | cut -d " " -f 3)
                 SUBLEVEL=$(grep "^SUBLEVEL =" $MAKEFILE | cut -d " " -f 3)
 
-                if [ -n "$PATCHLEVEL" -a $PATCHLEVEL -ge 6 \
-                        -a -n "$SUBLEVEL" -a $SUBLEVEL -le 5 ]; then
                     SELECTED_MAKEFILE=Makefile.kbuild
                     RET=0
-                fi
             fi
         fi
 
