by ultraman

--- a/kernel/nv-procfs.c	2013-03-27 22:26:51.000000000 +0100
+++ b/kernel/nv-procfs.c	2013-07-05 12:08:43.000000000 +0200
@@ -789,18 +791,18 @@
             if (!proc_nvidia_agp)
                 goto failed;
 
-            entry = NV_CREATE_PROC_FILE("status", proc_nvidia_agp,
-                nv_procfs_read_agp_status, NULL, NULL, nv);
+            entry = proc_create_data("status", NV_PROC_RO, proc_nvidia_agp,
+                nv_procfs_read_agp_status, nv);
             if (!entry)
                 goto failed;
 
-            entry = NV_CREATE_PROC_FILE("host-bridge", proc_nvidia_agp,
-                nv_procfs_read_agp_info, NULL, NULL, NULL);
+            entry = proc_create_data("host-bridge", NV_PROC_RO, proc_nvidia_agp,
+                nv_procfs_read_agp_info, NULL);
             if (!entry)
                 goto failed;
 
-            entry = NV_CREATE_PROC_FILE("gpu", proc_nvidia_agp,
-                nv_procfs_read_agp_info, NULL, NULL, nv);
+            entry = proc_create_data("gpu", NV_PROC_RO, proc_nvidia_agp,
+                nv_procfs_read_agp_info, nv);
             if (!entry)
                 goto failed;
         }
