diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index 36c588bad..f70cb55b8 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -38,6 +38,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	fe-pi-audio.dtbo \
 	goodix.dtbo \
 	googlevoicehat-soundcard.dtbo \
+	gamegirl.dtbo \
 	gpio-fan.dtbo \
 	gpio-ir.dtbo \
 	gpio-ir-tx.dtbo \
diff --git a/arch/arm/boot/dts/overlays/gamegirl.dts b/arch/arm/boot/dts/overlays/gamegirl.dts
index e69de29bb..949e13db5 100644
--- a/arch/arm/boot/dts/overlays/gamegirl-overlay.dts
+++ b/arch/arm/boot/dts/overlays/gamegirl-overlay.dts
@@ -0,0 +1,21 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "brcm,bcm2708";
+
+    fragment@0 {
+        target = <&gpio>;
+        __overlay__ {
+
+            pinctrl-names = "default";
+            pinctrl-0 = <&gamegirl>;
+
+            gamegirl: gamegirl {
+                brcm,pins = <18 19>;   /* gpio no. */
+                brcm,function = <2 2>; /* 0: GPIO in, 1: GPIO out, 2: alt5, 3: alt4, 4: alt0, 5: alt1, 6: alt2, 7: alt3 */
+                brcm,pull = <2 2>;     /* 2:up 1:down 0:none */
+            };
+        };
+    };
+};
