--- a/drivers/video/mxc/ldb.c	2015-05-06 19:05:42.000000000 +0200
+++ b/drivers/video/mxc/ldb.c	2015-06-18 17:14:52.241638700 +0200
@@ -159,6 +159,22 @@
 	 0,
 	 FB_VMODE_NONINTERLACED,
 	 FB_MODE_IS_DETAILED,},
+	{
+	 "LDB-WVGA-UDOO", 57, 800, 480, 30060, // Rif. 800x480 Panel DATAVISION dtfs070d0shlx
+	 56, 50, 
+	 23, 20,
+	 150, 2,
+	 0,
+	 FB_VMODE_NONINTERLACED,
+	 FB_MODE_IS_DETAILED,},
+	{
+	 "LDB-WXGA-UDOO", 60, 1368, 768, 12960, // Rif. 1366x768 Panel G156XW01V0
+	 9, 3,
+	 2, 7,
+	 200, 38,
+	 0,
+	 FB_VMODE_NONINTERLACED,
+	 FB_MODE_IS_DETAILED,},
 };
 static int ldb_modedb_sz = ARRAY_SIZE(ldb_modedb);
 
