diff -urN a/drivers/of/fdt.c b/drivers/of/fdt.c
--- a/drivers/of/fdt.c	2014-09-25 08:33:34.727318450 +0200
+++ b/drivers/of/fdt.c	2014-09-25 08:46:52.192711249 +0200
@@ -822,12 +822,21 @@
 	 * CONFIG_CMDLINE is meant to be a default in case nothing else
 	 * managed to set the command line, unless CONFIG_CMDLINE_FORCE
 	 * is set in which case we override whatever was found earlier.
+	 *
+	 * But we do append CONFIG_CMDLINE to bootloader arguments.
 	 */
 #ifdef CONFIG_CMDLINE
 #ifndef CONFIG_CMDLINE_FORCE
 	if (!((char *)data)[0])
-#endif
 		strlcpy(data, CONFIG_CMDLINE, COMMAND_LINE_SIZE);
+	else {
+		/* append CONFIG_CMDLINE to bootloader arguments */
+		strlcat(data, " ", COMMAND_LINE_SIZE);	
+		strlcat(data, CONFIG_CMDLINE, COMMAND_LINE_SIZE);		
+	}
+#else
+	strlcpy(data, CONFIG_CMDLINE, COMMAND_LINE_SIZE);
+#endif /* CONFIG_CMDLINE_FORCE */
 #endif /* CONFIG_CMDLINE */
 
 	pr_debug("Command line is: %s\n", (char*)data);
