diff -Naur uzebox.git/tools/uzem/avr8.h uzebox.patch/tools/uzem/avr8.h
--- uzebox.git/tools/uzem/avr8.h	2016-06-26 21:59:20.243240043 +0200
+++ uzebox.patch/tools/uzem/avr8.h	2016-06-27 22:56:09.836039916 +0200
@@ -37,15 +37,6 @@
 #include "input_driver.h"
 #include "video_driver.h"
 
-#if defined(__WIN32__)
-    #include <windows.h> // Win32 memory mapped I/O
-    #include <winioctl.h>
-#elif defined(LINUX) || defined(__APPLE__)
-    #include <sys/mman.h> // Linux memory mapped I/O
-#else
-    #include <sys/mmap.h> // Unix memory mapped I/O
-#endif
-
 // If you're building from the command line or on a non-MS compiler you'll need
 // -lSDL or somesuch.
 #include <SDL2/SDL.h>
