--- fluidsynth-1.1.6/CMakeLists.txt.orig	2015-04-06 13:12:55.151874342 +0200
+++ fluidsynth-1.1.6/CMakeLists.txt	2015-04-06 13:14:32.931876573 +0200
@@ -85,11 +85,11 @@
 endif ( ${CMAKE_SYSTEM} MATCHES "OS2" )
 
 # Initialize the library directory name suffix.
-if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-  set ( _init_lib_suffix "64" )
-else ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#  set ( _init_lib_suffix "64" )
+#else ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
   set ( _init_lib_suffix "" )
-endif ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#endif ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
 set ( LIB_SUFFIX ${_init_lib_suffix} CACHE STRING
       "library directory name suffix (32/64/nothing)" )
 mark_as_advanced ( LIB_SUFFIX )
