diff --git a/sshconnect2.c b/sshconnect2.c
index d6af0b9..22c0aa6 100644
--- a/sshconnect2.c
+++ b/sshconnect2.c
@@ -1320,8 +1320,7 @@ load_identity_file(char *filename, int userprovided)
 	struct stat st;
 
 	if (stat(id->filename, &st) < 0) {
-		(id->userprovided ? logit : debug3)("no such identity: %s: %s",
-		    id->filename, strerror(errno));
+		debug3("no such identity: %s", id->filename);
 		return NULL;
 	}
 	private = key_load_private_type(KEY_UNSPEC, filename, "", NULL, &perm_ok);
