diff --git a/sshconnect2.c b/sshconnect2.c
index d6af0b9..22c0aa6 100644
--- a/sshconnect2.c
+++ b/sshconnect2.c
@@ -1524,9 +1524,7 @@
 	struct stat st;
 
 	if (stat(id->filename, &st) == -1) {
-		do_log2(id->userprovided ?
-		    SYSLOG_LEVEL_INFO : SYSLOG_LEVEL_DEBUG3,
-		    "no such identity: %s: %s", id->filename, strerror(errno));
+		debug3("no such identity: %s", id->filename);
 		return NULL;
 	}
 	snprintf(prompt, sizeof prompt,
