diff -ru a/openbsd-compat/openssl-compat.c b/openbsd-compat/openssl-compat.c
--- a/openbsd-compat/openssl-compat.c	2018-10-16 20:01:20.000000000 -0400
+++ b/openbsd-compat/openssl-compat.c	2018-12-06 04:40:34.296087875 -0500
@@ -76,7 +76,7 @@
 	ENGINE_load_builtin_engines();
 	ENGINE_register_all_complete();
 
-#if OPENSSL_VERSION_NUMBER < 0x10001000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	OPENSSL_config(NULL);
 #else
 	OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS |
