diff -Naur openssh-5.8p1-old/configure.ac openssh-5.8p1-new/configure.ac
--- openssh-5.8p1-old/configure.ac	2011-02-21 17:41:23.000000000 -0800
+++ openssh-5.8p1-new/configure.ac	2011-02-21 17:42:20.000000000 -0800
@@ -41,7 +41,7 @@
 AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_PROG_EGREP
-AC_PATH_PROG(AR, ar)
+AC_PATH_TOOL(AR, ar)
 AC_PATH_PROG(CAT, cat)
 AC_PATH_PROG(KILL, kill)
 AC_PATH_PROGS(PERL, perl5 perl)
@@ -1188,25 +1188,25 @@
 
 AC_CHECK_DECLS(GLOB_NOMATCH, , , [#include <glob.h>])
 
-AC_MSG_CHECKING([whether struct dirent allocates space for d_name])
-AC_RUN_IFELSE(
-	[AC_LANG_SOURCE([[
+AC_CACHE_CHECK(
+	[whether struct dirent allocates space for d_name],
+	[ac_cv_have_space_for_dirent_d_name],
+	[AC_RUN_IFELSE(
+		[AC_LANG_SOURCE([[
 #include <sys/types.h>
 #include <dirent.h>
 int main(void){struct dirent d;exit(sizeof(d.d_name)<=sizeof(char));}
-	]])],
-	[AC_MSG_RESULT(yes)],
-	[
-		AC_MSG_RESULT(no)
-		AC_DEFINE(BROKEN_ONE_BYTE_DIRENT_D_NAME, 1,
-			[Define if your struct dirent expects you to
-			allocate extra space for d_name])
-	],
-	[
-		AC_MSG_WARN([cross compiling: assuming BROKEN_ONE_BYTE_DIRENT_D_NAME])
-		AC_DEFINE(BROKEN_ONE_BYTE_DIRENT_D_NAME)
-	]
+		]])],
+		[ ac_cv_have_space_for_dirent_d_name="yes" ],
+		[ ac_cv_have_space_for_dirent_d_name="no"  ],
+		[ ac_cv_have_space_for_dirent_d_name="no"  ]
+	)]
 )
+if test "x$ac_cv_have_space_for_dirent_d_name" = "xno" ; then
+	AC_DEFINE(BROKEN_ONE_BYTE_DIRENT_D_NAME, 1,
+		[Define if your struct dirent expects you to
+		allocate extra space for d_name])
+fi
 
 AC_MSG_CHECKING([for /proc/pid/fd directory])
 if test -d "/proc/$$/fd" ; then
