diff -Naur moonlight-20d2d77/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp moonlight-20d2d77-2/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp
--- moonlight-20d2d77/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp	2021-06-11 00:23:20.472732135 +0000
+++ moonlight-20d2d77-2/src/streaming/ffmpeg/FFmpegVideoDecoder.cpp	2021-06-11 00:26:21.582274704 +0000
@@ -36,10 +36,10 @@
     
     switch (video_format) {
         case VIDEO_FORMAT_H264:
-            m_decoder = avcodec_find_decoder_by_name("h264");
+            m_decoder = avcodec_find_decoder_by_name("h264_nvmpi");
             break;
         case VIDEO_FORMAT_H265:
-            m_decoder = avcodec_find_decoder_by_name("hevc");
+            m_decoder = avcodec_find_decoder_by_name("hevc_nvmpi");
             break;
     }
     
