diff -Naur a/utils/Makefile b/utils/Makefile
--- a/utils/Makefile	2016-04-19 13:20:50.000000000 -0700
+++ b/utils/Makefile	2016-10-20 00:31:34.238941511 -0700
@@ -27,7 +27,7 @@
 
 CFLAGS		?= -g -O2 -Wall
 
-PROGRAMS	= inputattach jstest jscal fftest ffmvforce ffset \
+PROGRAMS	= inputattach jstest jscal fftest ffset \
 		  ffcfstress jscal-restore jscal-store evdev-joystick
 
 PREFIX          ?= /usr/local
