From 0cf5d433583ead5a634f7bea9e08236bc1805318 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Mon, 3 Feb 2020 23:41:55 +0000
Subject: [PATCH] replace obsolete stime after glibc-2.31

---
 eit.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/eit.c b/eit.c
index 50d8229..60c295d 100644
--- a/eit.c
+++ b/eit.c
@@ -391,7 +391,10 @@ cTDT::cTDT(const u_char *Data)
   if (abs(diff) > MAX_TIME_DIFF) {
      mutex.Lock();
      if (abs(diff) > MAX_ADJ_DIFF) {
-        if (stime(&dvbtim) == 0)
+        struct timespec ts;
+        ts.tv_sec = dvbtim;
+        ts.tv_nsec = 0;
+        if (clock_settime(CLOCK_REALTIME, &ts) == 0)
            isyslog("system time changed from %s (%ld) to %s (%ld)", *TimeToString(loctim), loctim, *TimeToString(dvbtim), dvbtim);
         else
            esyslog("ERROR while setting system time: %m");
-- 
2.20.1

