From: Stefan Saraev <stefan@saraev.ca>
Subject: [PATCH] decrease channels.conf autosave delay

--- a/vdr.c
+++ b/vdr.c
@@ -74,7 +74,7 @@
 #define SHUTDOWNCANCELPROMPT   5 // seconds to wait in user prompt to allow canceling shutdown
 #define RESTARTCANCELPROMPT    5 // seconds to wait in user prompt before restarting on SIGHUP
 #define MANUALSTART          600 // seconds the next timer must be in the future to assume manual start
-#define CHANNELSAVEDELTA     600 // seconds before saving channels.conf after automatic modifications
+#define CHANNELSAVEDELTA     120 // seconds before saving channels.conf after automatic modifications
 #define DEVICEREADYTIMEOUT    30 // seconds to wait until all devices are ready
 #define MENUTIMEOUT          120 // seconds of user inactivity after which an OSD display is closed
 #define TIMERCHECKDELTA       10 // seconds between checks for timers that need to see their channel

