From 6700ee3b4b14a21f946b2c27cd25c8a9930d7a59 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Tue, 13 Feb 2018 15:11:11 +0100
Subject: [PATCH 2/2] decrease channels.conf autosave delay

---
 vdr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vdr.c b/vdr.c
index df119cdf..1e134075 100644
--- a/vdr.c
+++ b/vdr.c
@@ -82,7 +82,7 @@
 #define SHUTDOWNCANCELPROMPT   5 // seconds to wait in user prompt to allow canceling shutdown
 #define RESTARTCANCELPROMPT    5 // seconds to wait in user prompt before restarting on SIGHUP
 #define MANUALSTART          600 // seconds the next timer must be in the future to assume manual start
-#define CHANNELSAVEDELTA     600 // seconds before saving channels.conf after automatic modifications
+#define CHANNELSAVEDELTA     120 // seconds before saving channels.conf after automatic modifications
 #define DEVICEREADYTIMEOUT    30 // seconds to wait until all devices are ready
 #define MENUTIMEOUT          120 // seconds of user inactivity after which an OSD display is closed
 #define TIMERCHECKDELTA        5 // seconds between checks for timers that need to see their channel
-- 
2.51.0

