diff -Naur a/libretro/Makefile b/libretro/Makefile
--- a/libretro/Makefile	2016-01-19 10:11:20.000000000 +0100
+++ b/libretro/Makefile	2016-01-19 10:27:00.009866912 +0100
@@ -201,8 +201,8 @@
 
 # ARM
 else ifneq (,$(findstring armv,$(platform)))
-	CC = gcc
-	CXX = g++
+	CC ?= gcc
+	CXX ?= g++
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
