diff -Naur a/libretro/Makefile b/libretro/Makefile
--- a/libretro/Makefile	2016-05-10 10:13:03.235599098 -0700
+++ b/libretro/Makefile	2016-05-10 10:41:33.975635164 -0700
@@ -237,6 +237,23 @@
 	PLATCFLAGS += -DARM
 	LDFLAGS += -lrt -ldl
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+        CC ?= gcc
+        CXX ?= g++
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        LDFLAGS += -shared -Wl,--version-script=link.T -Wl,--no-undefined
+        FFMPEGINCFLAGS += -I$(FFMPEGDIR)/linux/$(FFMPEG_ARCH)/include
+        FFMPEGLIBDIR := $(FFMPEGDIR)/linux/$(FFMPEG_ARCH)/lib
+        FFMPEGLDFLAGS += -L$(FFMPEGLIBDIR) -lavformat -lavcodec -lavutil -lswresample -lswscale
+        INCFLAGS += -I.
+        WITH_DYNAREC=arm64
+        GLES := 1
+        GL_LIB := -lGLESv2 -lEGL
+        LDFLAGS += -lGLESv2 -lEGL
+        LDFLAGS += -lrt -ldl
+
 # emscripten
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_emscripten.bc
