diff -Naur xf86-video-openchrome-0.2.903/configure xf86-video-openchrome-0.2.903.patch/configure
--- xf86-video-openchrome-0.2.903/configure	2009-05-18 21:39:54.297241592 +0200
+++ xf86-video-openchrome-0.2.903.patch/configure	2009-05-18 21:39:22.851250949 +0200
@@ -12859,7 +12859,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 $as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
@@ -12887,7 +12887,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 $as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
@@ -12915,7 +12915,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   $as_echo_n "(cached) " >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { $as_echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 $as_echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
