diff -Naur gst-plugins-good-0.10.14.2/sys/v4l2/v4l2_calls.c gst-plugins-good-0.10.14.2.patch/sys/v4l2/v4l2_calls.c
--- gst-plugins-good-0.10.14.2/sys/v4l2/v4l2_calls.c	2009-05-11 23:06:23.000000000 +0200
+++ gst-plugins-good-0.10.14.2.patch/sys/v4l2/v4l2_calls.c	2009-05-12 03:15:18.000000000 +0200
@@ -141,7 +141,7 @@
     GST_LOG_OBJECT (e, "   name:      '%s'", input.name);
     GST_LOG_OBJECT (e, "   type:      %08x", input.type);
     GST_LOG_OBJECT (e, "   audioset:  %08x", input.audioset);
-    GST_LOG_OBJECT (e, "   std:       %016x", input.std);
+    GST_LOG_OBJECT (e, "   std:       %016x", (unsigned int)input.std);
     GST_LOG_OBJECT (e, "   status:    %08x", input.status);
 
     v4l2channel = g_object_new (GST_TYPE_V4L2_TUNER_CHANNEL, NULL);
