From fc19cd1da6161ec5602e692488dfc8e670797afe Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 15 Aug 2014 22:10:03 +0300
Subject: [PATCH] meh help2man

---
 Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 5a336df..2d2029b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -701,7 +701,7 @@ update_mans = \
   $(AM_V_GEN): \
     && $(MKDIR_P) doc \
     && $(extend_PATH) \
-    && $(PERL) $(srcdir)/doc/help2man --output=$@
+    && $(PERL) $(srcdir)/doc/help2man --output=$@ --no-discard-stderr
 
 amhello_sources = \
   doc/amhello/configure.ac \
-- 
1.7.2.5

