diff -ur automake-1.15/bin/automake.in automake-1.15-new/bin/automake.in
--- automake-1.15/bin/automake.in	2015-01-05 11:25:55.000000000 -0800
+++ automake-1.15-new/bin/automake.in	2018-01-07 16:34:38.899448825 -0800
@@ -3878,7 +3878,7 @@
 sub substitute_ac_subst_variables
 {
   my ($text) = @_;
-  $text =~ s/\${([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
+  $text =~ s/\$\{([^ \t=:+{}]+)\}/substitute_ac_subst_variables_worker ($1)/ge;
   return $text;
 }
 
