From 2657cd8325b0f092fc986a307b9de7e186fe8b9f Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 25 Aug 2018 07:49:26 -0700
Subject: [PATCH] memory: tegra: drop optimization to O1 for tegra210

to work around a compiler issue.

Signed-off-by: Matt Madison <matt@madison.systems>
---
 drivers/memory/tegra/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/memory/tegra/Makefile b/drivers/memory/tegra/Makefile
index 79722d32b4cb..935363c342e7 100644
--- a/drivers/memory/tegra/Makefile
+++ b/drivers/memory/tegra/Makefile
@@ -10,6 +10,7 @@ obj-$(CONFIG_TEGRA_MC) += tegra-mc.o
 
 obj-$(CONFIG_TEGRA124_EMC) += tegra124-emc.o
 
+ccflags-$(CONFIG_TEGRA210_EMC) += -O1
 obj-$(CONFIG_TEGRA210_EMC) += t210-emc.o
 t210-emc-objs += tegra210-emc.o tegra210-emc-cc-r21015.o tegra210-emc-cc-r21012.o tegra210-emc-cc-r21021.o tegra210-dt-parse.o
 