diff --git a/file_path_str.c b/file_path_str.c
index d1b0c781c1..3442b173cc 100644
--- a/file_path_str.c
+++ b/file_path_str.c
@@ -171,7 +171,7 @@ const char *file_path_str(enum file_path_enum enum_idx)
          str = "http://thumbnailpacks.libretro.com";
          break;
       case FILE_PATH_LAKKA_URL:
-         str = "http://le.builds.lakka.tv";
+         str = "http://nightly.builds.lakka.tv/.updater";
          break;
       case FILE_PATH_SHADERS_GLSL_ZIP:
          str = "shaders_glsl.zip";
