--- pulseaudio-0.9.21/src/pulse/context.c.orig	2009-11-22 22:57:06.000000000 -0500
+++ pulseaudio-0.9.21/src/pulse/context.c	2010-05-10 10:57:57.000000000 -0400
@@ -603,11 +603,6 @@
         return NULL;
     }
 
-    if (st.st_uid != getuid()) {
-        pa_xfree(p);
-        return NULL;
-    }
-
     return p;
 }
 
@@ -625,11 +620,6 @@
         return NULL;
     }
 
-    if (st.st_uid != getuid()) {
-        pa_xfree(p);
-        return NULL;
-    }
-
     return p;
 }
 #endif
--- pulseaudio-0.9.21/src/pulsecore/core-util.c.orig	2009-11-22 22:57:07.000000000 -0500
+++ pulseaudio-0.9.21/src/pulsecore/core-util.c	2010-05-10 10:59:12.000000000 -0400
@@ -235,8 +235,6 @@
 
 #ifndef OS_IS_WIN32
     if (!S_ISDIR(st.st_mode) ||
-        (st.st_uid != uid) ||
-        (st.st_gid != gid) ||
         ((st.st_mode & 0777) != m)) {
         errno = EACCES;
         goto fail;
@@ -1293,12 +1291,6 @@
         goto finish;
     }
 
-    if (st.st_uid != getuid()) {
-        pa_log_error("Home directory %s not ours.", h);
-        errno = EACCES;
-        goto finish;
-    }
-
     ret = pa_sprintf_malloc("%s" PA_PATH_SEP ".pulse", h);
 
 finish:
@@ -1524,7 +1516,6 @@
         } else {
 
             if (S_ISDIR(st.st_mode) &&
-                (st.st_uid == getuid()) &&
                 ((st.st_mode & 0777) == 0700)) {
 
                 pa_xfree(p);
