From c9d7dcaa87f9ada49d75483a7c604dba27a0d71a Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Tue, 7 Jan 2014 13:22:48 +0200
Subject: [PATCH] never append -dirty to server/module version

---
 git-version-gen |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/git-version-gen b/git-version-gen
index 9d65b80..fcf11fd 100755
--- a/git-version-gen
+++ b/git-version-gen
@@ -152,6 +152,7 @@ v=`echo "$v" |sed 's/^v//'`
 git status > /dev/null 2>&1
 
 dirty=`sh -c 'git diff-index --name-only HEAD' 2>/dev/null` || dirty=
+dirty=
 case "$dirty" in
     '') ;;
     *) # Append the suffix only if there isn't one already.
-- 
1.7.2.5

