diff --git a/build-aux/ltmain.sh b/build-aux/ltmain.sh
index 63ae69d..cfde522 100644
--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -6918,7 +6918,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      #add_dir="-L$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
