From d8ceac3c27e55c39af6d6956479564682b466980 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Mon, 10 Feb 2020 11:01:39 +0000
Subject: [PATCH] fix build with kernel 5.6-rc1

---
 nuc_led.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/nuc_led.c b/nuc_led.c
index c975f22..1d9a272 100644
--- a/nuc_led.c
+++ b/nuc_led.c
@@ -37,6 +37,7 @@
 #include <linux/acpi.h>
 #include <linux/vmalloc.h>
 #include <linux/uaccess.h>
+#include <linux/version.h>
 
 MODULE_AUTHOR("Miles Peterson");
 MODULE_DESCRIPTION("Intel NUC LED Control WMI Driver");
@@ -443,11 +444,18 @@ static ssize_t acpi_proc_read(struct file *filp, char __user *buff,
         return ret;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0))
+static struct proc_ops proc_acpi_operations = {
+        .proc_read     = acpi_proc_read,
+        .proc_write    = acpi_proc_write,
+};
+#else
 static struct file_operations proc_acpi_operations = {
         .owner    = THIS_MODULE,
         .read     = acpi_proc_read,
         .write    = acpi_proc_write,
 };
+#endif
 
 /* Init & unload */
 static int __init init_nuc_led(void)
-- 
2.20.1

