diff -Naur wpa_supplicant-2.0/src/drivers/drivers.mak wpa_supplicant-2.0.patch/src/drivers/drivers.mak
--- wpa_supplicant-2.0/src/drivers/drivers.mak	2013-01-12 16:42:53.000000000 +0100
+++ wpa_supplicant-2.0.patch/src/drivers/drivers.mak	2013-01-21 02:15:18.034917696 +0100
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 else
   ifdef CONFIG_LIBNL_TINY
     DRV_LIBS += -lnl-tiny
diff -Naur wpa_supplicant-2.0/src/drivers/drivers.mk wpa_supplicant-2.0.patch/src/drivers/drivers.mk
--- wpa_supplicant-2.0/src/drivers/drivers.mk	2013-01-12 16:42:53.000000000 +0100
+++ wpa_supplicant-2.0.patch/src/drivers/drivers.mk	2013-01-21 02:15:39.677784809 +0100
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 else
   ifdef CONFIG_LIBNL_TINY
     DRV_LIBS += -lnl-tiny
