diff -up wpa_supplicant-1.0-rc1/wpa_supplicant/wpa_gui-qt4/wpagui.cpp.getopt wpa_supplicant-1.0-rc1/wpa_supplicant/wpa_gui-qt4/wpagui.cpp
--- wpa_supplicant-1.0-rc1/wpa_supplicant/wpa_gui-qt4/wpagui.cpp.getopt	2012-01-10 16:28:43.451307760 -0600
+++ wpa_supplicant-1.0-rc1/wpa_supplicant/wpa_gui-qt4/wpagui.cpp	2012-01-10 16:28:53.999175893 -0600
@@ -12,10 +12,7 @@
  * See README and COPYING for more details.
  */
 
-#ifdef __MINGW32__
-/* Need to get getopt() */
 #include <unistd.h>
-#endif
 
 #ifdef CONFIG_NATIVE_WINDOWS
 #include <windows.h>
