diff -Naur wpa_supplicant-0.7.3/src/drivers/drivers.mak wpa_supplicant-0.7.3.patch/src/drivers/drivers.mak
--- wpa_supplicant-0.7.3/src/drivers/drivers.mak	2010-09-07 17:43:39.000000000 +0200
+++ wpa_supplicant-0.7.3.patch/src/drivers/drivers.mak	2011-11-30 13:37:22.187964539 +0100
@@ -31,10 +31,10 @@
 NEED_AP_MLME=y
 NEED_NETLINK=y
 NEED_LINUX_IOCTL=y
-DRV_LIBS += -lnl
+DRV_LIBS += -lnl-3
 
 ifdef CONFIG_LIBNL20
-DRV_LIBS += -lnl-genl
+DRV_LIBS += -lnl-genl-3
 DRV_CFLAGS += -DCONFIG_LIBNL20
 endif
 endif
