diff -Naur wpa_supplicant-2.3/src/drivers/drivers.mak wpa_supplicant-2.3.patch/src/drivers/drivers.mak
--- wpa_supplicant-2.3/src/drivers/drivers.mak	2014-10-09 16:41:31.000000000 +0200
+++ wpa_supplicant-2.3.patch/src/drivers/drivers.mak	2014-10-10 17:54:54.030025202 +0200
@@ -35,7 +35,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 ifdef CONFIG_LIBNL3_ROUTE
   DRV_LIBS += -lnl-route-3
   DRV_CFLAGS += -DCONFIG_LIBNL3_ROUTE
diff -Naur wpa_supplicant-2.3/src/drivers/drivers.mk wpa_supplicant-2.3.patch/src/drivers/drivers.mk
--- wpa_supplicant-2.3/src/drivers/drivers.mk	2014-10-09 16:41:31.000000000 +0200
+++ wpa_supplicant-2.3.patch/src/drivers/drivers.mk	2014-10-10 17:54:32.261065224 +0200
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 ifdef CONFIG_LIBNL3_ROUTE
   DRV_LIBS += -lnl-route-3
   DRV_CFLAGS += -DCONFIG_LIBNL3_ROUTE
