diff -Naur wpa_supplicant-0.7.3/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service wpa_supplicant-0.7.3.patch/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service
--- wpa_supplicant-0.7.3/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service	2010-09-07 17:43:39.000000000 +0200
+++ wpa_supplicant-0.7.3.patch/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service	2011-11-06 22:34:15.331582673 +0100
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=fi.epitest.hostap.WPASupplicant
-Exec=/sbin/wpa_supplicant -u
+Exec=/usr/bin/wpa_supplicant -u
 User=root
diff -Naur wpa_supplicant-0.7.3/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service wpa_supplicant-0.7.3.patch/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service
--- wpa_supplicant-0.7.3/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service	2010-09-07 17:43:39.000000000 +0200
+++ wpa_supplicant-0.7.3.patch/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service	2011-11-06 22:34:24.216699484 +0100
@@ -1,4 +1,4 @@
 [D-BUS Service]
 Name=fi.w1.wpa_supplicant1
-Exec=/sbin/wpa_supplicant -u
+Exec=/usr/bin/wpa_supplicant -u
 User=root
