From 8e396c0586e511b746e377f36817e32413b7c5aa Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 4 Jul 2015 22:21:19 +0300
Subject: [PATCH] move datapath to /storage

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d73d069..1043f1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(GLEW REQUIRED)
 find_package(ProjectM REQUIRED)
 
 add_definitions(-DHAS_SDL_OPENGL)
-set(DATA_PATH "${CMAKE_INSTALL_PREFIX}/share/xbmc/addons/visualization.projectm/resources")
+set(DATA_PATH "/storage/.kodi/addons/visualization.projectm/resources")
 add_definitions(-DDATAPATH="${DATA_PATH}")
 
 set(INCLUDES ${OpenGL_INCLUDE_DIR}
-- 
2.1.4

