From 6e83396c8360a8457af4d5348a852b6f333ac29f Mon Sep 17 00:00:00 2001
From: Arne Morten Kvarving <arne.morten.kvarving@sintef.no>
Date: Wed, 18 Jan 2017 12:33:09 +0100
Subject: [PATCH] remove glew

---
 CMakeLists.txt | 3 ++-
 src/Main.cpp   | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b323a82..56348b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ find_package(p8-platform REQUIRED)
 find_package(OpenGL REQUIRED)
 
 if(NOT APPLE)
-  find_package(GLEW REQUIRED)
+  #find_package(GLEW REQUIRED)
   set(EXTRA_INCLUDES ${GLEW_INCLUDE_DIR})
   set(EXTRA_LIBRARIES ${GLEW_LIBRARIES})
 endif()
@@ -28,6 +28,7 @@ set(INCLUDES ${OpenGL_INCLUDE_DIR}
              ${PROJECTM_INCLUDE_DIRS})
 
 set(DEPLIBS ${OPENGL_LIBRARIES}
+  #${GLEW_LIBRARIES}
             ${EXTRA_LIBRARIES}
             ${PROJECTM_LIBS}
             ${extra_libs})
diff --git a/src/Main.cpp b/src/Main.cpp
index e9e1345..ae1042e 100644
--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -59,7 +59,7 @@ d4rk@xbmc.org
 #include <threads/mutex.h>
 
 #if !defined(__APPLE__)
-#include <GL/glew.h>
+#include <GL/gl.h>
 #endif
 
 #include "libprojectM/projectM.hpp"
-- 
2.7.4

