diff --git a/libavutil/utils.c b/libavutil/utils.c
index 971b48b..b57078e 100644
--- a/libavutil/utils.c
+++ b/libavutil/utils.c
@@ -28,7 +28,7 @@
 
 unsigned avutil_version(void)
 {
-    av_assert0(PIX_FMT_VDA_VLD == 81); //check if the pix fmt enum has not had anything inserted or removed by mistake
+    av_assert0(PIX_FMT_VDA_VLD == 82); //check if the pix fmt enum has not had anything inserted or removed by mistake
     av_assert0(AV_SAMPLE_FMT_DBLP == 9);
     av_assert0(AVMEDIA_TYPE_ATTACHMENT == 4);
     av_assert0(AV_PICTURE_TYPE_BI == 7);
