--- module-init-tools-3.8/configure.org	2009-05-24 11:34:42.000000000 +0200
+++ module-init-tools-3.8/configure	2009-05-24 11:34:58.000000000 +0200
@@ -3854,14 +3854,6 @@
 # between a broken cc and a working cc but missing libz.a.
 LDADD="$LDADD $zlib_flags"
 
-
-case $target in
-*-*-linux*) ;;
-*) { { $as_echo "$as_me:$LINENO: error: Linux only, dude!" >&5
-$as_echo "$as_me: error: Linux only, dude!" >&2;}
-   { (exit 1); exit 1; }; };;
-esac
-
 # Thanks to Martin Pool
 if test x"$GCC" = xyes
 then
