diff -Naur libX11-1.2.1.orig/src/util/Makefile.in libX11-1.2.1/src/util/Makefile.in
--- libX11-1.2.1.orig/src/util/Makefile.in	2009-04-08 19:00:52.000000000 +0200
+++ libX11-1.2.1/src/util/Makefile.in	2009-04-08 19:03:07.000000000 +0200
@@ -51,17 +51,17 @@
 makekeys_OBJECTS = makekeys-makekeys.$(OBJEXT)
 makekeys_LDADD = $(LDADD)
 makekeys_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(makekeys_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	--mode=link host-gcc $(makekeys_CFLAGS) $(HOST_CFLAGS) $(AM_LDFLAGS) \
+	$(HOST_LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I. -I$(top_builddir)/src -I$(top_builddir)/include/X11@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
-COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+COMPILE = $(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(CPPFLAGS) $(AM_CFLAGS) $(HOST_CFLAGS)
+CCLD = $(HOST_CC)
 LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	--mode=link host-gcc $(AM_CFLAGS) $(HOST_CFLAGS) $(AM_LDFLAGS) \
+	$(HOST_LDFLAGS) -o $@
 SOURCES = makekeys.c
 DIST_SOURCES = makekeys.c
 ETAGS = etags
@@ -80,10 +80,10 @@
 AWK = @AWK@
 BIGFONT_CFLAGS = @BIGFONT_CFLAGS@
 BIGFONT_LIBS = @BIGFONT_LIBS@
-CC = @CC_FOR_BUILD@
+CC = host-gcc
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS = $(HOST_CFLAGS)
 CHANGELOG_CMD = @CHANGELOG_CMD@
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
