diff -Naur libX11-1.1.99.2.orig/configure libX11-1.1.99.2/configure
--- libX11-1.1.99.2.orig/configure	2008-11-09 15:36:00.000000000 +0100
+++ libX11-1.1.99.2/configure	2008-11-09 15:34:54.000000000 +0100
@@ -12371,6 +12371,7 @@
 case "$ac_cv_use_xcb" in
 no)
 	X11_REQUIRES="xau xcmiscproto bigreqsproto"
+	X11_EXTRA_DEPS="xau"
 
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for XDMCP" >&5
@@ -12430,58 +12431,18 @@
 	# Put the nasty error message in config.log where it belongs
 	echo "$XDMCP_PKG_ERRORS" >&5
 
-	{ { echo "$as_me:$LINENO: error: Package requirements (xdmcp) were not met:
-
-$XDMCP_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables XDMCP_CFLAGS
-and XDMCP_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-" >&5
-echo "$as_me: error: Package requirements (xdmcp) were not met:
-
-$XDMCP_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables XDMCP_CFLAGS
-and XDMCP_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-" >&2;}
-   { (exit 1); exit 1; }; }
+	{ echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6; }
+                { echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6; }
 elif test $pkg_failed = untried; then
-	{ { echo "$as_me:$LINENO: error: The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables XDMCP_CFLAGS
-and XDMCP_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details." >&5
-echo "$as_me: error: The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables XDMCP_CFLAGS
-and XDMCP_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+	{ echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6; }
 else
 	XDMCP_CFLAGS=$pkg_cv_XDMCP_CFLAGS
 	XDMCP_LIBS=$pkg_cv_XDMCP_LIBS
         { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
-	:
-fi
 	{ echo "$as_me:$LINENO: checking for XdmcpWrap in -lXdmcp" >&5
 echo $ECHO_N "checking for XdmcpWrap in -lXdmcp... $ECHO_C" >&6; }
 if test "${ac_cv_lib_Xdmcp_XdmcpWrap+set}" = set; then
@@ -12544,11 +12505,84 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_Xdmcp_XdmcpWrap" >&5
 echo "${ECHO_T}$ac_cv_lib_Xdmcp_XdmcpWrap" >&6; }
 if test $ac_cv_lib_Xdmcp_XdmcpWrap = yes; then
+
+	        	{ echo "$as_me:$LINENO: checking for XdmcpWrap in -lXdmcp" >&5
+echo $ECHO_N "checking for XdmcpWrap in -lXdmcp... $ECHO_C" >&6; }
+if test "${ac_cv_lib_Xdmcp_XdmcpWrap+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lXdmcp $XDMCP_LIBS $LIBS"
+cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char XdmcpWrap ();
+int
+main ()
+{
+return XdmcpWrap ();
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext conftest$ac_exeext
+if { (ac_try="$ac_link"
+case "(($ac_try" in
+  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
+  *) ac_try_echo=$ac_try;;
+esac
+eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
+  (eval "$ac_link") 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } && {
+	 test -z "$ac_c_werror_flag" ||
+	 test ! -s conftest.err
+       } && test -s conftest$ac_exeext &&
+       $as_test_x conftest$ac_exeext; then
+  ac_cv_lib_Xdmcp_XdmcpWrap=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+	ac_cv_lib_Xdmcp_XdmcpWrap=no
+fi
+
+rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
+      conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ echo "$as_me:$LINENO: result: $ac_cv_lib_Xdmcp_XdmcpWrap" >&5
+echo "${ECHO_T}$ac_cv_lib_Xdmcp_XdmcpWrap" >&6; }
+if test $ac_cv_lib_Xdmcp_XdmcpWrap = yes; then
   xdmauth="yes"
 else
   xdmauth="no"
 fi
 
+	        	X11_EXTRA_DEPS="$X11_EXTRA_DEPS xdmcp"
+
+else
+
+			XDMCP_CFLAGS=
+			XDMCP_LIBS=
+
+fi
+
+fi
 
 cat >>confdefs.h <<\_ACEOF
 #define USE_XCB 0
diff -Naur libX11-1.1.99.2.orig/configure.ac libX11-1.1.99.2/configure.ac
--- libX11-1.1.99.2.orig/configure.ac	2008-11-09 15:33:41.000000000 +0100
+++ libX11-1.1.99.2/configure.ac	2008-11-09 15:33:56.000000000 +0100
@@ -48,8 +48,18 @@
 case "$ac_cv_use_xcb" in
 no)
 	X11_REQUIRES="xau xcmiscproto bigreqsproto"
-	PKG_CHECK_MODULES(XDMCP, xdmcp)
-	AC_CHECK_LIB(Xdmcp, XdmcpWrap, [xdmauth="yes"], [xdmauth="no"], [$XDMCP_LIBS])
+	X11_EXTRA_DEPS="xau"
+	PKG_CHECK_MODULES(XDMCP, xdmcp,
+		AC_CHECK_LIB(Xdmcp, XdmcpWrap,
+			[
+	        	AC_CHECK_LIB(Xdmcp, XdmcpWrap, [xdmauth="yes"], [xdmauth="no"], [$XDMCP_LIBS])
+	        	X11_EXTRA_DEPS="$X11_EXTRA_DEPS xdmcp"
+			],
+			[
+			XDMCP_CFLAGS=
+			XDMCP_LIBS=
+			], [$XDMCP_LIBS]),
+		[AC_MSG_RESULT(no)])
 	AC_DEFINE(USE_XCB, 0, [Use XCB for low-level protocol implementation])
 	;;
 *)
diff -Naur libX11-1.1.99.2.orig/src/ConnDis.c libX11-1.1.99.2/src/ConnDis.c
--- libX11-1.1.99.2.orig/src/ConnDis.c	2008-11-09 15:33:41.000000000 +0100
+++ libX11-1.1.99.2/src/ConnDis.c	2008-11-09 15:33:56.000000000 +0100
@@ -39,7 +39,9 @@
 #include <X11/Xlibint.h>
 #include <X11/Xtrans/Xtrans.h>
 #include <X11/Xauth.h>
+#ifdef HASXDMAUTH
 #include <X11/Xdmcp.h>
+#endif
 #include <stdio.h>
 #include <ctype.h>
 #include <unistd.h>
diff -Naur libX11-1.1.99.2.orig/x11.pc.in libX11-1.1.99.2/x11.pc.in
--- libX11-1.1.99.2.orig/x11.pc.in	2008-11-09 15:33:41.000000000 +0100
+++ libX11-1.1.99.2/x11.pc.in	2008-11-09 15:33:56.000000000 +0100
@@ -9,7 +9,7 @@
 Description: X Library
 Version: @PACKAGE_VERSION@
 Requires: xproto @XKBPROTO_REQUIRES@
-Requires.private: xau xdmcp @X11_EXTRA_DEPS@
+Requires.private: @X11_EXTRA_DEPS@
 Cflags: -I${includedir} @XTHREAD_CFLAGS@
 Libs: -L${libdir} -lX11
 Libs.private: @XTHREADLIB@
