diff -Naur libX11-1.3.4-old/configure.ac libX11-1.3.4-new/configure.ac
--- libX11-1.3.4-old/configure.ac	2010-06-03 15:21:17.000000000 -0700
+++ libX11-1.3.4-new/configure.ac	2010-06-09 16:06:19.000000000 -0700
@@ -35,6 +35,7 @@
 DOLT
 AC_PROG_CC
 PKG_PROG_PKG_CONFIG
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
 
 if test x"$CC_FOR_BUILD" = x; then
 	if test x"$cross_compiling" = xyes; then
@@ -355,7 +356,7 @@
 # Find keysymdef.h
 #
 AC_MSG_CHECKING([keysymdef.h])
-dir=`$PKG_CONFIG --variable=includedir xproto`
+dir=$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=includedir xproto`
 KEYSYMDEF="$dir/X11/keysymdef.h"
 if test -f "$KEYSYMDEF"; then
         AC_MSG_RESULT([$KEYSYMDEF])
