From cd61ed7becee71277789de0a9cf81627682f64d1 Mon Sep 17 00:00:00 2001
From: natinusala <natinusala@gmail.com>
Date: Mon, 18 Nov 2019 14:10:46 +0100
Subject: [PATCH] Makefile: link against GL on unix

---
 .gitignore | 1 +
 Makefile   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 1e75026..4484e0d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -7,6 +7,7 @@
 *.vpk
 *.pk3
 *.zip
+*.so
 datafiles/*
 
 # Windows image file caches
diff --git a/Makefile b/Makefile
index 779c403..de52787 100644
--- a/Makefile
+++ b/Makefile
@@ -67,6 +67,7 @@ ifeq ($(platform), unix)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+   LDFLAGS += -lGL
 else ifeq ($(platform), linux-portable)
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC -nostdlib
