diff -Naur compcache-20101022/zram_drv.c compcache-20101022.patch/zram_drv.c
--- compcache-20101022/zram_drv.c	2010-10-22 04:13:51.000000000 +0200
+++ compcache-20101022.patch/zram_drv.c	2010-10-22 04:31:48.407299221 +0200
@@ -458,7 +458,7 @@
 		return 0;
 	}
 
-	if (unlikely(bio_rw_flagged(bio, BIO_RW_DISCARD))) {
+	if (unlikely(bio->bi_rw & REQ_DISCARD)) {
 		zram_discard(zram, bio);
 		return 0;
 	}
