diff -Naur systemd-215/src/shared/missing.h systemd-215.patch/src/shared/missing.h
--- systemd-215/src/shared/missing.h	2014-07-21 14:28:37.277575885 +0200
+++ systemd-215.patch/src/shared/missing.h	2014-07-21 14:27:39.940232879 +0200
@@ -381,6 +381,16 @@
 #define IFLA_MACVLAN_MAX (__IFLA_MACVLAN_MAX - 1)
 #endif
 
+#define IFLA_VTI_UNSPEC 0
+#define IFLA_VTI_LINK 1
+#define IFLA_VTI_IKEY 2
+#define IFLA_VTI_OKEY 3
+#define IFLA_VTI_LOCAL 4
+#define IFLA_VTI_REMOTE 5
+#define __IFLA_VTI_MAX 6
+
+#define IFLA_VTI_MAX    (__IFLA_VTI_MAX - 1)
+
 #if !HAVE_DECL_IFLA_PHYS_PORT_ID
 #undef IFLA_PROMISCUITY
 #define IFLA_PROMISCUITY 30
