diff -Naur xvba-video-0.8.0/src/Makefile.am xvba-video-0.8.0.patch/src/Makefile.am
--- xvba-video-0.8.0/src/Makefile.am	2011-06-14 13:07:13.000000000 +0200
+++ xvba-video-0.8.0.patch/src/Makefile.am	2011-07-13 20:00:21.496624808 +0200
@@ -75,7 +75,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
-	    ln -s xvba_drv_video.so $${drv}_drv_video.so;	\
+	    ln -sf xvba_drv_video.so $${drv}_drv_video.so;	\
 	done
 
 # Extra clean files so that maintainer-clean removes *everything*
diff -Naur xvba-video-0.8.0/src/Makefile.in xvba-video-0.8.0.patch/src/Makefile.in
--- xvba-video-0.8.0/src/Makefile.in	2011-06-14 13:07:29.000000000 +0200
+++ xvba-video-0.8.0.patch/src/Makefile.in	2011-07-13 20:00:34.705800165 +0200
@@ -727,7 +727,7 @@
 install-data-hook:
 	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
 	for drv in $(DRIVERS); do				\
-	    ln -s xvba_drv_video.so $${drv}_drv_video.so;	\
+	    ln -sf xvba_drv_video.so $${drv}_drv_video.so;	\
 	done
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
