diff -Naur xbmc-audioencoder-flac-7aee9d0/CMakeLists.txt xbmc-audioencoder-flac-7aee9d0.patch/CMakeLists.txt
--- xbmc-audioencoder-flac-7aee9d0/CMakeLists.txt	2014-07-18 23:48:22.000000000 +0200
+++ xbmc-audioencoder-flac-7aee9d0.patch/CMakeLists.txt	2014-10-20 13:03:49.565070141 +0200
@@ -6,7 +6,7 @@
 
 enable_language(CXX)
 
-find_package(xbmc REQUIRED)
+find_package(kodi REQUIRED)
 find_package(FLAC REQUIRED)
 find_package(Ogg REQUIRED)
 
diff -Naur xbmc-audioencoder-flac-7aee9d0/src/EncoderFlac.cpp xbmc-audioencoder-flac-7aee9d0.patch/src/EncoderFlac.cpp
--- xbmc-audioencoder-flac-7aee9d0/src/EncoderFlac.cpp	2014-07-18 23:48:22.000000000 +0200
+++ xbmc-audioencoder-flac-7aee9d0.patch/src/EncoderFlac.cpp	2014-10-20 02:13:58.806998666 +0200
@@ -20,7 +20,7 @@
 
 #include <FLAC/stream_encoder.h>
 #include <FLAC/metadata.h>
-#include "xbmc/xbmc_audioenc_dll.h"
+#include "kodi/xbmc_audioenc_dll.h"
 #include <string.h>
 
 extern "C" {
