diff -Naur libcdio-0.80.orig/lib/driver/gnu_linux.c libcdio-0.80/lib/driver/gnu_linux.c
--- libcdio-0.80.orig/lib/driver/gnu_linux.c	2010-02-17 23:21:04.260757228 +0100
+++ libcdio-0.80/lib/driver/gnu_linux.c	2010-02-17 23:21:28.273257431 +0100
@@ -30,6 +30,9 @@
 
 static const char _rcsid[] = "$Id: gnu_linux.c,v 1.28 2007/03/07 04:35:47 rocky Exp $";
 
+#define u8 uint8_t
+#include <inttypes.h>
+
 #include <string.h>
 #include <limits.h>
 #include <sys/types.h>
