diff -Naur libcdio-0.90-old/configure.ac libcdio-0.90-new/configure.ac
--- libcdio-0.90-old/configure.ac	2012-10-27 08:07:45.000000000 -0700
+++ libcdio-0.90-new/configure.ac	2012-12-30 14:12:26.000000000 -0800
@@ -26,7 +26,7 @@
 AM_INIT_AUTOMAKE([foreign])
 
 AC_CANONICAL_HOST
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
 # Enable silent build rules by default (Automake v1.11 or later).
