diff -Naur libcdio-0.83/src/cd-paranoia/doc/ja/Makefile.am libcdio-0.83.patch/src/cd-paranoia/doc/ja/Makefile.am
--- libcdio-0.83/src/cd-paranoia/doc/ja/Makefile.am	2008-11-29 05:52:43.000000000 +0100
+++ libcdio-0.83.patch/src/cd-paranoia/doc/ja/Makefile.am	2012-06-13 17:50:45.487977958 +0200
@@ -24,7 +24,7 @@
 
 install-man1: $(man_MANS)
 	@$(NORMAL_INSTALL)
-	test -z "$(man1dir)" || $(mkdir_p) "$(DESTDIR)$(mandir)$(mansubdir)"
+	test -z "$(man1dir)" || $(MKDIR_P) "$(DESTDIR)$(mandir)$(mansubdir)"
 	@list='$(man1_MANS)'; \
 	l2='$(man_MANS)'; for i in $$l2; do \
 	  case "$$i" in \
