commit 179bbd6adf94ac21c990bfc09616c3f61406e360
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Sep 24 11:59:36 2009 +0000

    [G45] Implement vaQueryDisplayAttributes() as a no-op.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index d26942d..375d532 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -1031,8 +1031,10 @@ i965_QueryDisplayAttributes(VADriverContextP ctx,
                             VADisplayAttribute *attr_list,	/* out */
                             int *num_attributes)		/* out */
 {
-    /* TODO */
-    return VA_STATUS_ERROR_UNKNOWN;
+    if (num_attributes)
+        *num_attributes = 0;
+
+    return VA_STATUS_SUCCESS;
 }
 
 /* 
