commit 41e88c17745a6e8553ed39e627a0ec22840b8100
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Sep 24 12:54:40 2009 +0000

    [G45] Fix VA driver version.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index 8d28547..800a642 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -1322,8 +1322,8 @@ __vaDriverInit_0_31(  VADriverContextP ctx )
     struct i965_driver_data *i965;
     int result;
 
-    ctx->version_major = 0;
-    ctx->version_minor = 29;
+    ctx->version_major = VA_MAJOR_VERSION;
+    ctx->version_minor = VA_MINOR_VERSION;
     ctx->max_profiles = I965_MAX_PROFILES;
     ctx->max_entrypoints = I965_MAX_ENTRYPOINTS;
     ctx->max_attributes = I965_MAX_CONFIG_ATTRIBUTES;
