commit 390fab410fe30e0de0f47a539d1d15d022fb6902
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Sep 24 12:55:11 2009 +0000

    [G45] Fix vaAssociateSubpicture() arguments.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index 800a642..6582c71 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -489,8 +489,6 @@ i965_AssociateSubpicture(VADriverContextP ctx,
                          short dest_y,
                          unsigned short dest_width,
                          unsigned short dest_height,
-                         unsigned short width,
-                         unsigned short height,
                          /*
                           * whether to enable chroma-keying or global-alpha
                           * see VA_SUBPICTURE_XXX values
