commit d6e726b80766c669dffe846fcd31fcfe5dc89847
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Nov 4 12:13:41 2009 +0000

    [G45] Fix subpictures formats array terminator.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index 6a60ecc..e476ca2 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -76,7 +76,7 @@ static const i965_subpic_format_map_t *
 get_subpic_format(const VAImageFormat *va_format)
 {
     unsigned int i;
-    for (i = 0; i < sizeof(i965_subpic_formats_map)/sizeof(i965_subpic_formats_map[0]); i++) {
+    for (i = 0; i965_subpic_formats_map[i].type != 0; i++) {
         const i965_subpic_format_map_t * const m = &i965_subpic_formats_map[i];
         if (m->va_format.fourcc == va_format->fourcc &&
             (m->type == I965_SURFACETYPE_RGBA ?
