commit ff263433c32aac08574b58b26f3f4be7abe05bdf
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Nov 4 13:23:40 2009 +0000

    [G45] Don't return VA_STATUS_SUCCESS for unimplemented functions.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index 637e304..9e8502a 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -496,7 +496,8 @@ i965_SetSubpictureImage(VADriverContextP ctx,
                         VASubpictureID subpicture,
                         VAImageID image)
 {
-    return VA_STATUS_SUCCESS;
+    /* TODO */
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 VAStatus 
@@ -506,7 +507,8 @@ i965_SetSubpictureChromakey(VADriverContextP ctx,
                             unsigned int chromakey_max,
                             unsigned int chromakey_mask)
 {
-    return VA_STATUS_SUCCESS;
+    /* TODO */
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 VAStatus 
@@ -514,7 +516,8 @@ i965_SetSubpictureGlobalAlpha(VADriverContextP ctx,
                               VASubpictureID subpicture,
                               float global_alpha)
 {
-    return VA_STATUS_SUCCESS;
+    /* TODO */
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 VAStatus 
@@ -1103,7 +1106,7 @@ i965_GetDisplayAttributes(VADriverContextP ctx,
                           int num_attributes)
 {
     /* TODO */
-    return VA_STATUS_ERROR_UNKNOWN;
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 /* 
@@ -1118,7 +1121,7 @@ i965_SetDisplayAttributes(VADriverContextP ctx,
                           int num_attributes)
 {
     /* TODO */
-    return VA_STATUS_ERROR_UNKNOWN;
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 VAStatus 
@@ -1128,7 +1131,7 @@ i965_DbgCopySurfaceToBuffer(VADriverContextP ctx,
                             unsigned int *stride)       /* out */
 {
     /* TODO */
-    return VA_STATUS_ERROR_UNKNOWN;
+    return VA_STATUS_ERROR_UNIMPLEMENTED;
 }
 
 static VAStatus 
@@ -1270,7 +1273,8 @@ VAStatus i965_DeriveImage(VADriverContextP ctx,
                           VASurfaceID surface,
                           VAImage *image)        /* out */
 {
-    return VA_STATUS_SUCCESS;
+    /* TODO */
+    return VA_STATUS_ERROR_OPERATION_FAILED;
 }
 
 static void 
