diff -Naur libva-0.31.0/config.h.in libva-0.31.0.diff/config.h.in
--- libva-0.31.0/config.h.in	2009-10-10 23:13:39.445657680 +0200
+++ libva-0.31.0.diff/config.h.in	2009-10-10 23:12:44.000000000 +0200
@@ -1,5 +1,8 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
+/* Defined to __attribute__((visibility("hidden"))) when available */
+#undef ATTRIBUTE_HIDDEN
+
 /* Define to 1 if you have the <dlfcn.h> header file. */
 #undef HAVE_DLFCN_H
 
diff -Naur libva-0.31.0/configure libva-0.31.0.diff/configure
--- libva-0.31.0/configure	2009-10-10 23:13:39.576782883 +0200
+++ libva-0.31.0.diff/configure	2009-10-10 23:12:44.390782788 +0200
@@ -13164,6 +13164,38 @@
 fi
 
 
+{ $as_echo "$as_me:$LINENO: checking whether __attribute__((visibility())) is supported" >&5
+$as_echo_n "checking whether __attribute__((visibility())) is supported... " >&6; }
+if test "${libva_cv_visibility_attribute+set}" = set; then
+  $as_echo_n "(cached) " >&6
+else
+  cat > conftest.c <<EOF
+int foo __attribute__ ((visibility ("hidden"))) = 1;
+int bar __attribute__ ((visibility ("protected"))) = 1;
+EOF
+    libva_cv_visibility_attribute=no
+    if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
+        if grep '\.hidden.*foo' conftest.s >/dev/null; then
+            if grep '\.protected.*bar' conftest.s >/dev/null; then
+	       libva_cv_visibility_attribute=yes
+            fi
+        fi
+    fi
+    rm -f conftest.cs
+
+fi
+{ $as_echo "$as_me:$LINENO: result: $libva_cv_visibility_attribute" >&5
+$as_echo "$libva_cv_visibility_attribute" >&6; }
+ATTRIBUTE_HIDDEN=""
+if test $libva_cv_visibility_attribute = yes; then
+    ATTRIBUTE_HIDDEN="__attribute__((visibility(\"hidden\")))"
+fi
+
+cat >>confdefs.h <<_ACEOF
+#define ATTRIBUTE_HIDDEN $ATTRIBUTE_HIDDEN
+_ACEOF
+
+
 # We only need the headers, we don't link against the DRM libraries
 LIBVA_CFLAGS="$DRM_CFLAGS"
 
diff -Naur libva-0.31.0/configure.ac libva-0.31.0.diff/configure.ac
--- libva-0.31.0/configure.ac	2009-09-10 17:18:49.000000000 +0200
+++ libva-0.31.0.diff/configure.ac	2009-10-10 23:12:32.556667509 +0200
@@ -100,6 +100,30 @@
 fi
 AM_CONDITIONAL(BUILD_I965_DRIVER, test x$enable_i965_driver = xyes)
 
+dnl Check for __attribute__((visibility()))
+AC_CACHE_CHECK([whether __attribute__((visibility())) is supported],
+    libva_cv_visibility_attribute,
+    [cat > conftest.c <<EOF
+int foo __attribute__ ((visibility ("hidden"))) = 1;
+int bar __attribute__ ((visibility ("protected"))) = 1;
+EOF
+    libva_cv_visibility_attribute=no
+    if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
+        if grep '\.hidden.*foo' conftest.s >/dev/null; then
+            if grep '\.protected.*bar' conftest.s >/dev/null; then
+	       libva_cv_visibility_attribute=yes
+            fi
+        fi
+    fi
+    rm -f conftest.[cs]
+])
+ATTRIBUTE_HIDDEN=""
+if test $libva_cv_visibility_attribute = yes; then
+    ATTRIBUTE_HIDDEN="__attribute__((visibility(\"hidden\")))"
+fi
+AC_DEFINE_UNQUOTED([ATTRIBUTE_HIDDEN], [$ATTRIBUTE_HIDDEN],
+    [Defined to __attribute__((visibility("hidden"))) when available])
+
 # We only need the headers, we don't link against the DRM libraries
 LIBVA_CFLAGS="$DRM_CFLAGS"
 AC_SUBST(LIBVA_CFLAGS)
