commit a07468c4f0e30b5f207a2bdf1b362f5584d748a7
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Tue Sep 22 12:29:56 2009 +0000

    Fix vainfo dependencies (Konstantin Pavlov).

diff --git a/test/Makefile.am b/test/Makefile.am
index d7e1b1f..8785259 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -25,9 +25,8 @@ bin_PROGRAMS = vainfo
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/test/basic -I$(top_srcdir)/src/x11
 
-
-vainfo_LDADD = $(top_srcdir)/va/$(libvabackendlib)
-vainfo_DEPENDENCIES = $(top_srcdir)/va/$(libvabackendlib)
+vainfo_LDADD = $(top_srcdir)/va/libva.la $(top_srcdir)/va/$(libvabackendlib)
+vainfo_DEPENDENCIES = $(top_srcdir)/va/libva.la $(top_srcdir)/va/$(libvabackendlib)
 vainfo_SOURCES = vainfo.c
 
 SUBDIRS = basic decode encode putsurface
