diff -Naur libva-0.31.0/test/Makefile.am libva-0.31.0.diff/test/Makefile.am
--- libva-0.31.0/test/Makefile.am	2009-09-10 17:18:48.000000000 +0200
+++ libva-0.31.0.diff/test/Makefile.am	2009-10-10 23:14:32.350657987 +0200
@@ -31,10 +31,10 @@
 
 TESTS = $(check_PROGRAMS)
 
-TEST_LIBS = $(top_srcdir)/src/$(libvabackendlib)
+TEST_LIBS = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
 
-vainfo_LDADD = $(top_srcdir)/src/$(libvabackendlib)
-vainfo_DEPENDENCIES = $(top_srcdir)/src/$(libvabackendlib)
+vainfo_LDADD = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
+vainfo_DEPENDENCIES = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
 vainfo_SOURCES = vainfo.c
 
 test_01_LDADD = $(TEST_LIBS)
diff -Naur libva-0.31.0/test/Makefile.in libva-0.31.0.diff/test/Makefile.in
--- libva-0.31.0/test/Makefile.in	2009-10-10 23:13:39.621783024 +0200
+++ libva-0.31.0.diff/test/Makefile.in	2009-10-10 23:14:48.470783134 +0200
@@ -275,9 +275,9 @@
 testdir = $(bindir)
 AM_CFLAGS = -I$(top_srcdir)/../../include/external/ -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
 TESTS = $(check_PROGRAMS)
-TEST_LIBS = $(top_srcdir)/src/$(libvabackendlib)
-vainfo_LDADD = $(top_srcdir)/src/$(libvabackendlib)
-vainfo_DEPENDENCIES = $(top_srcdir)/src/$(libvabackendlib)
+TEST_LIBS = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
+vainfo_LDADD = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
+vainfo_DEPENDENCIES = $(top_srcdir)/src/libva.la $(top_srcdir)/src/$(libvabackendlib)
 vainfo_SOURCES = vainfo.c
 test_01_LDADD = $(TEST_LIBS)
 test_01_SOURCES = test_01.c
