diff -Naur libva-0.31.0/configure libva-0.31.0.diff/configure
--- libva-0.31.0/configure	2009-10-10 23:14:15.667658343 +0200
+++ libva-0.31.0.diff/configure	2009-10-10 23:16:27.484782476 +0200
@@ -2652,7 +2652,7 @@
 if test "${with_drivers_path+set}" = set; then
   withval=$with_drivers_path;
 else
-  with_drivers_path="$libdir/dri"
+  with_drivers_path="$libdir/va/drivers"
 fi
 
 
diff -Naur libva-0.31.0/configure.ac libva-0.31.0.diff/configure.ac
--- libva-0.31.0/configure.ac	2009-10-10 23:14:15.668657233 +0200
+++ libva-0.31.0.diff/configure.ac	2009-10-10 23:16:15.673657877 +0200
@@ -73,7 +73,7 @@
 
 AC_ARG_WITH(drivers-path,
 	    [AC_HELP_STRING([--with-drivers-path=[[path]]], [drivers path])],,
-	    [with_drivers_path="$libdir/dri"])
+	    [with_drivers_path="$libdir/va/drivers"])
 
 LIBVA_DRIVERS_PATH="$with_drivers_path"
 AC_SUBST(LIBVA_DRIVERS_PATH)
