commit 43becedcfa2b9b040006868a63c686a131c9ba74
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Jun 24 08:12:13 2009 +0000

    Install drivers into pkglibdir/va/.

diff --git a/configure.ac b/configure.ac
index f2d0ea4..ebe5df6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ AC_ARG_ENABLE(i965-driver,
 
 AC_ARG_WITH(drivers-path,
 	    [AC_HELP_STRING([--with-drivers-path=[[path]]], [drivers path])],,
-	    [with_drivers_path="$libdir/dri"])
+	    [with_drivers_path="$libdir/va/drivers"])
 
 LIBVA_DRIVERS_PATH="$with_drivers_path"
 AC_SUBST(LIBVA_DRIVERS_PATH)
