diff -Naur libva-0.31.0/configure libva-0.31.0.diff/configure
--- libva-0.31.0/configure	2009-10-10 23:17:18.356783363 +0200
+++ libva-0.31.0.diff/configure	2009-10-10 23:19:28.956657592 +0200
@@ -859,6 +859,7 @@
 BUILD_DUMMY_DRIVER_TRUE
 LIBVA_LT_LDFLAGS
 LIBVA_LT_VERSION
+LIBVA_SDS_VERSION
 LIBVA_VERSION
 LIBVA_MICRO_VERSION
 LIBVA_MINOR_VERSION
@@ -2615,11 +2616,14 @@
 
 
 
+LIBVA_SDS_VERSION=3
+
+
 LIBVA_LT_CURRENT=1
 LIBVA_LT_REV=0
 LIBVA_LT_AGE=0
 LIBVA_LT_VERSION="$LIBVA_LT_CURRENT:$LIBVA_LT_REV:$LIBVA_LT_AGE"
-LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION"
+LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION -release $LIBVA_VERSION.$LIBVA_SDS_VERSION"
 
 
 
@@ -4514,13 +4518,13 @@
 else
   lt_cv_nm_interface="BSD nm"
   echo "int some_variable = 0;" > conftest.$ac_ext
-  (eval echo "\"\$as_me:4517: $ac_compile\"" >&5)
+  (eval echo "\"\$as_me:4521: $ac_compile\"" >&5)
   (eval "$ac_compile" 2>conftest.err)
   cat conftest.err >&5
-  (eval echo "\"\$as_me:4520: $NM \\\"conftest.$ac_objext\\\"\"" >&5)
+  (eval echo "\"\$as_me:4524: $NM \\\"conftest.$ac_objext\\\"\"" >&5)
   (eval "$NM \"conftest.$ac_objext\"" 2>conftest.err > conftest.out)
   cat conftest.err >&5
-  (eval echo "\"\$as_me:4523: output\"" >&5)
+  (eval echo "\"\$as_me:4527: output\"" >&5)
   cat conftest.out >&5
   if $GREP 'External.*some_variable' conftest.out > /dev/null; then
     lt_cv_nm_interface="MS dumpbin"
@@ -5726,7 +5730,7 @@
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 5729 "configure"' > conftest.$ac_ext
+  echo '#line 5733 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -7549,11 +7553,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7552: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:7556: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:7556: \$? = $ac_status" >&5
+   echo "$as_me:7560: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -7888,11 +7892,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7891: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:7895: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:7895: \$? = $ac_status" >&5
+   echo "$as_me:7899: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -7993,11 +7997,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7996: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:8000: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:8000: \$? = $ac_status" >&5
+   echo "$as_me:8004: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -8048,11 +8052,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:8051: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:8055: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:8055: \$? = $ac_status" >&5
+   echo "$as_me:8059: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -10851,7 +10855,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 10854 "configure"
+#line 10858 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -10947,7 +10951,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 10950 "configure"
+#line 10954 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
diff -Naur libva-0.31.0/configure.ac libva-0.31.0.diff/configure.ac
--- libva-0.31.0/configure.ac	2009-10-10 23:17:18.356783363 +0200
+++ libva-0.31.0.diff/configure.ac	2009-10-10 23:19:15.805657564 +0200
@@ -28,6 +28,9 @@
 m4_define([libva_version],
           [libva_major_version.libva_minor_version.libva_micro_version])
 
+# increase this number for each API change
+m4_define([libva_sds_version], [3])
+
 # if the library source code has changed, increment revision
 m4_define([libva_lt_revision], [0])
 # if any interface was added/removed/changed, then inc current, reset revision
@@ -52,11 +55,14 @@
 AC_SUBST(LIBVA_MICRO_VERSION)
 AC_SUBST(LIBVA_VERSION)
 
+LIBVA_SDS_VERSION=libva_sds_version
+AC_SUBST(LIBVA_SDS_VERSION)
+
 LIBVA_LT_CURRENT=libva_lt_current
 LIBVA_LT_REV=libva_lt_revision
 LIBVA_LT_AGE=libva_lt_age
 LIBVA_LT_VERSION="$LIBVA_LT_CURRENT:$LIBVA_LT_REV:$LIBVA_LT_AGE"
-LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION"
+LIBVA_LT_LDFLAGS="-version-info $LIBVA_LT_VERSION -release $LIBVA_VERSION.$LIBVA_SDS_VERSION"
 AC_SUBST(LIBVA_LT_VERSION)
 AC_SUBST(LIBVA_LT_LDFLAGS)
 
diff -Naur libva-0.31.0/dummy_drv_video/Makefile.in libva-0.31.0.diff/dummy_drv_video/Makefile.in
--- libva-0.31.0/dummy_drv_video/Makefile.in	2009-10-10 23:13:39.578782409 +0200
+++ libva-0.31.0.diff/dummy_drv_video/Makefile.in	2009-10-10 23:19:30.911783933 +0200
@@ -161,6 +161,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/i965_drv_video/Makefile.in libva-0.31.0.diff/i965_drv_video/Makefile.in
--- libva-0.31.0/i965_drv_video/Makefile.in	2009-10-10 23:13:39.580782565 +0200
+++ libva-0.31.0.diff/i965_drv_video/Makefile.in	2009-10-10 23:19:31.021782765 +0200
@@ -202,6 +202,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/i965_drv_video/shaders/Makefile.in libva-0.31.0.diff/i965_drv_video/shaders/Makefile.in
--- libva-0.31.0/i965_drv_video/shaders/Makefile.in	2009-10-10 23:13:39.582782720 +0200
+++ libva-0.31.0.diff/i965_drv_video/shaders/Makefile.in	2009-10-10 23:19:31.088782868 +0200
@@ -132,6 +132,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/i965_drv_video/shaders/mpeg2/Makefile.in libva-0.31.0.diff/i965_drv_video/shaders/mpeg2/Makefile.in
--- libva-0.31.0/i965_drv_video/shaders/mpeg2/Makefile.in	2009-10-10 23:13:39.583782588 +0200
+++ libva-0.31.0.diff/i965_drv_video/shaders/mpeg2/Makefile.in	2009-10-10 23:19:31.153783373 +0200
@@ -132,6 +132,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/i965_drv_video/shaders/mpeg2/vld/Makefile.in libva-0.31.0.diff/i965_drv_video/shaders/mpeg2/vld/Makefile.in
--- libva-0.31.0/i965_drv_video/shaders/mpeg2/vld/Makefile.in	2009-10-10 23:13:39.585782953 +0200
+++ libva-0.31.0.diff/i965_drv_video/shaders/mpeg2/vld/Makefile.in	2009-10-10 23:19:31.220782567 +0200
@@ -92,6 +92,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/i965_drv_video/shaders/render/Makefile.in libva-0.31.0.diff/i965_drv_video/shaders/render/Makefile.in
--- libva-0.31.0/i965_drv_video/shaders/render/Makefile.in	2009-10-10 23:13:39.586782472 +0200
+++ libva-0.31.0.diff/i965_drv_video/shaders/render/Makefile.in	2009-10-10 23:19:31.290782972 +0200
@@ -92,6 +92,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/Makefile.in libva-0.31.0.diff/Makefile.in
--- libva-0.31.0/Makefile.in	2009-10-10 23:13:39.598782635 +0200
+++ libva-0.31.0.diff/Makefile.in	2009-10-10 23:19:31.708782943 +0200
@@ -196,6 +196,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/src/Makefile.in libva-0.31.0.diff/src/Makefile.in
--- libva-0.31.0/src/Makefile.in	2009-10-10 23:19:00.350783347 +0200
+++ libva-0.31.0.diff/src/Makefile.in	2009-10-10 23:19:31.388782687 +0200
@@ -206,6 +206,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/src/va.c libva-0.31.0.diff/src/va.c
--- libva-0.31.0/src/va.c	2009-09-10 17:18:49.000000000 +0200
+++ libva-0.31.0.diff/src/va.c	2009-10-10 23:19:15.806658410 +0200
@@ -40,6 +40,7 @@
 
 
 #define DRIVER_INIT_FUNC	"__vaDriverInit_0_31"
+#define DRIVER_INIT_FUNC_SDS	"__vaDriverInit_0_31_sds"
 
 #define DRIVER_EXTENSION	"_drv_video.so"
 
@@ -170,6 +171,11 @@
             init_func = (VADriverInit) dlsym(handle, DRIVER_INIT_FUNC);
             if (!init_func)
             {
+                /* Then try SDS extensions (VDPAU and XvBA backends) */
+                init_func = (VADriverInit) dlsym(handle, DRIVER_INIT_FUNC_SDS);
+            }
+            if (!init_func)
+            {
                 va_errorMessage("%s has no function %s\n", driver_path, DRIVER_INIT_FUNC);
                 dlclose(handle);
             }
diff -Naur libva-0.31.0/src/va_version.h.in libva-0.31.0.diff/src/va_version.h.in
--- libva-0.31.0/src/va_version.h.in	2009-09-10 17:18:49.000000000 +0200
+++ libva-0.31.0.diff/src/va_version.h.in	2009-10-10 23:19:15.807658068 +0200
@@ -47,6 +47,13 @@
 #define VA_MICRO_VERSION   (@LIBVA_MICRO_VERSION@)
 
 /**
+ * VA_SDS_VERSION:
+ *
+ * The version of the SDS API extensions to the VA library
+ */
+#define VA_SDS_VERSION     (@LIBVA_SDS_VERSION@)
+
+/**
  * VA_VERSION:
  *
  * The full version of the VA library, like 1.2.3
@@ -59,7 +66,7 @@
  * The full version of the VA library, in string form (suited for
  * string concatenation)
  */
-#define VA_VERSION_S       "@LIBVA_VERSION@"
+#define VA_VERSION_S       "@LIBVA_VERSION@-sds@LIBVA_SDS_VERSION@"
 
 /**
  * VA_VERSION_HEX:
diff -Naur libva-0.31.0/src/x11/Makefile.in libva-0.31.0.diff/src/x11/Makefile.in
--- libva-0.31.0/src/x11/Makefile.in	2009-10-10 23:19:00.355783107 +0200
+++ libva-0.31.0.diff/src/x11/Makefile.in	2009-10-10 23:19:31.476782185 +0200
@@ -159,6 +159,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
diff -Naur libva-0.31.0/test/Makefile.in libva-0.31.0.diff/test/Makefile.in
--- libva-0.31.0/test/Makefile.in	2009-10-10 23:16:01.445782945 +0200
+++ libva-0.31.0.diff/test/Makefile.in	2009-10-10 23:19:31.623782491 +0200
@@ -185,6 +185,7 @@
 LIBVA_MAJOR_VERSION = @LIBVA_MAJOR_VERSION@
 LIBVA_MICRO_VERSION = @LIBVA_MICRO_VERSION@
 LIBVA_MINOR_VERSION = @LIBVA_MINOR_VERSION@
+LIBVA_SDS_VERSION = @LIBVA_SDS_VERSION@
 LIBVA_VERSION = @LIBVA_VERSION@
 LIPO = @LIPO@
 LN_S = @LN_S@
