commit c682fb5ebd1964638f90590f66edf49629e7481d
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Fri Aug 21 11:34:50 2009 +0000

    Move vaPutSurface() flags to <va/va.h> header.

diff --git a/src/va.h b/src/va.h
index 0519f6f..6b4bbd4 100755
--- a/src/va.h
+++ b/src/va.h
@@ -127,6 +127,22 @@ typedef int VAStatus;	/* Return status type from functions */
 #define VA_STATUS_ERROR_RESOLUTION_NOT_SUPPORTED 0x00000013
 #define VA_STATUS_ERROR_UNKNOWN			0xFFFFFFFF
 
+/* De-interlacing flags for vaPutSurface() */
+#define VA_FRAME_PICTURE        0x00000000 
+#define VA_TOP_FIELD            0x00000001
+#define VA_BOTTOM_FIELD         0x00000002
+
+/*
+ * Clears the drawable with background color.
+ * for hardware overlay based implementation this flag
+ * can be used to turn off the overlay
+ */
+#define VA_CLEAR_DRAWABLE       0x00000008 
+
+/* Color space conversion flags for vaPutSurface() */
+#define VA_SRC_BT601            0x00000010
+#define VA_SRC_BT709            0x00000020
+
 /*
  * Returns a short english description of error_status
  */
diff --git a/src/x11/va_x11.h b/src/x11/va_x11.h
index 2171ac6..5ab5200 100644
--- a/src/x11/va_x11.h
+++ b/src/x11/va_x11.h
@@ -27,22 +27,6 @@ VADisplay vaGetDisplay (
  * color space conversion and scaling to the destination
  * rectangle
  */
-/* de-interlacing flags for vaPutSurface */
-#define VA_FRAME_PICTURE	0x00000000 
-#define VA_TOP_FIELD		0x00000001
-#define VA_BOTTOM_FIELD		0x00000002
-
-/* 
- * clears the drawable with background color.
- * for hardware overlay based implementation this flag
- * can be used to turn off the overlay
- */
-#define VA_CLEAR_DRAWABLE	0x00000008 
-
-/* color space conversion flags for vaPutSurface */
-#define VA_SRC_BT601		0x00000010
-#define VA_SRC_BT709		0x00000020
-
 VAStatus vaPutSurface (
     VADisplay dpy,
     VASurfaceID surface,	
