diff -Naur libva-1.1.0-old/configure.ac libva-1.1.0-new/configure.ac
--- libva-1.1.0-old/configure.ac	2012-10-04 06:30:51.000000000 -0700
+++ libva-1.1.0-new/configure.ac	2012-12-30 15:06:37.000000000 -0800
@@ -86,7 +86,7 @@
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([dist-bzip2])
 
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 TODAY="`LC_ALL=C date +'%a, %d %b %Y %X %z'`"
