diff --git a/va/va.h b/va/va.h
index d9e4c7e..dc0f092 100644
--- a/va/va.h
+++ b/va/va.h
@@ -1905,6 +1905,8 @@ VAStatus vaSyncSurface (
     VASurfaceID render_target
 );
 
+#define HAVE_VA_SURFACE_STATUS_EMPTY 1
+
 typedef enum
 {
     VASurfaceRendering	= 1, /* Rendering in progress */ 
@@ -1912,7 +1914,8 @@ typedef enum
                              /* this status is useful if surface is used as the source */
                              /* of an overlay */
     VASurfaceReady	= 4, /* not being rendered or displayed */
-    VASurfaceSkipped	= 8  /* Indicate a skipped frame during encode */
+    VASurfaceSkipped	= 8, /* Indicate a skipped frame during encode */
+    VASurfaceEmpty	= 16 /* contains no actual data */
 } VASurfaceStatus;
 
 /*
