From f342855b2b489de4141f7d23cfd66b7f243b74b7 Mon Sep 17 00:00:00 2001
From: Stephen <elfarto@elfarto.com>
Date: Tue, 23 Nov 2021 09:10:01 +0000
Subject: [PATCH] Add nvidia-drm to the drm driver map

---
 va/drm/va_drm_utils.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/va/drm/va_drm_utils.c b/va/drm/va_drm_utils.c
index 467d9ac9..b3422d19 100644
--- a/va/drm/va_drm_utils.c
+++ b/va/drm/va_drm_utils.c
@@ -45,6 +45,7 @@ static const struct driver_name_map g_driver_name_map[] = {
     { "nouveau",    7, "nouveau"  }, // Mesa Gallium driver
     { "radeon",     6, "r600"     }, // Mesa Gallium driver
     { "amdgpu",     6, "radeonsi" }, // Mesa Gallium driver
+    { "nvidia-drm",10, "nvidia"   }, // NVIDIA driver
     { NULL,         0, NULL }
 };
 
