From ad99ba3a8d3116e4eb8d78e7b7e4bb30e9464064 Mon Sep 17 00:00:00 2001
From: Xiang, Haihao <haihao.xiang@intel.com>
Date: Thu, 19 May 2011 03:10:46 +0000
Subject: i965_drv_video: initialize context for H.264 decoding on G4x

---
diff --git a/i965_drv_video/i965_media.c b/i965_drv_video/i965_media.c
index d6a56d2..12a036a 100644
--- a/i965_drv_video/i965_media.c
+++ b/i965_drv_video/i965_media.c
@@ -341,6 +341,9 @@ g4x_dec_hw_context_init(VADriverContextP ctx, VAProfile profile)
     case VAProfileH264Baseline:
     case VAProfileH264Main:
     case VAProfileH264High:
+        i965_media_h264_dec_context_init(ctx, media_context);
+        break;
+
     case VAProfileVC1Simple:
     case VAProfileVC1Main:
     case VAProfileVC1Advanced:
--
cgit v0.8.3-6-g21f6
