Don't install some useeless tests programs


--- libva-0.31.0/test/basic/Makefile.am~	2010-02-12 09:42:10.000000000 -0500
+++ libva-0.31.0/test/basic/Makefile.am	2010-03-10 13:36:50.449834525 -0500
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
+check_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
 		test_07 test_08 test_09 test_10 test_11
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
--- libva-0.31.0/test/decode/Makefile.am~	2010-02-12 09:42:10.000000000 -0500
+++ libva-0.31.0/test/decode/Makefile.am	2010-03-10 13:37:09.636843464 -0500
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = mpeg2vldemo
+check_PROGRAMS = mpeg2vldemo
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
 
--- libva-0.31.0/test/encode/Makefile.am~	2010-02-12 09:42:10.000000000 -0500
+++ libva-0.31.0/test/encode/Makefile.am	2010-03-10 13:37:26.701846294 -0500
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = h264encode
+check_PROGRAMS = h264encode
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
 
--- libva-0.31.0/test/putsurface/Makefile.am~	2010-03-10 13:18:55.270697040 -0500
+++ libva-0.31.0/test/putsurface/Makefile.am	2010-03-10 13:38:08.106713556 -0500
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = putsurface
+check_PROGRAMS = putsurface
 
 AM_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/src/x11 -DIN_LIBVA
 
