--- fakeroot-1.25.3/configure.ac	2020-10-08 17:13:18.000000000 +0000
+++ fakeroot-1.25.3/configure.ac	2021-01-10 03:57:50.295218481 +0000
@@ -597,9 +597,7 @@
 AC_CONFIG_FILES([
    Makefile
    scripts/Makefile
-   doc/Makefile
-   doc/de/Makefile doc/es/Makefile doc/fr/Makefile doc/nl/Makefile doc/pt/Makefile doc/sv/Makefile
-   test/Makefile test/defs])
+   ])
 AC_OUTPUT
 
 dnl Local variables:
--- fakeroot-1.25.3/Makefile.am	2020-10-08 17:13:18.000000000 +0000
+++ fakeroot-1.25.3/Makefile.am	2021-01-10 04:02:06.703081728 +0000
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS=foreign
 ACLOCAL_AMFLAGS = -I build-aux
-SUBDIRS=doc scripts test
+SUBDIRS=scripts
 
 noinst_LTLIBRARIES = libcommunicate.la libmacosx.la
 libcommunicate_la_SOURCES = communicate.c
