diff -Naur binutils-2.20.51.0.3.orig/ld/emultempl/elf32.em binutils-2.20.51.0.3/ld/emultempl/elf32.em
--- binutils-2.20.51.0.3.orig/ld/emultempl/elf32.em	2009-11-17 21:44:38.142032544 +0100
+++ binutils-2.20.51.0.3/ld/emultempl/elf32.em	2009-11-17 21:46:37.073216252 +0100
@@ -1250,7 +1250,9 @@
 	      && command_line.rpath == NULL)
 	    {
 	      lib_path = (const char *) getenv ("LD_RUN_PATH");
-	      if (gld${EMULATION_NAME}_search_needed (lib_path, &n,
+              if ((lib_path) && (strlen (lib_path) == 0))
+                lib_path = NULL;
+              if (gld${EMULATION_NAME}_search_needed (lib_path, &n,
 						      force))
 		break;
 	    }
@@ -1476,6 +1478,8 @@
   rpath = command_line.rpath;
   if (rpath == NULL)
     rpath = (const char *) getenv ("LD_RUN_PATH");
+  if ((rpath) && (strlen (rpath) == 0))
+    rpath = NULL;
 
   for (abfd = link_info.input_bfds; abfd; abfd = abfd->link_next)
     {
