--- binutils-2.20.51.0.11/bfd/compress.c.orig	2010-08-11 23:52:42.000000000 +0200
+++ binutils-2.20.51.0.11/bfd/compress.c	2010-09-26 11:08:50.196564435 +0200
@@ -94,7 +94,7 @@
   bfd_set_error (bfd_error_invalid_operation);
   return FALSE;
 #else
-  bfd_size_type compressed_size;
+  uLong compressed_size;
   bfd_byte *compressed_buffer;
 
   compressed_size = compressBound (uncompressed_size) + 12;
