diff -Naur binutils-2.22/configure binutils-2.22.patch/configure
--- binutils-2.22/configure	2011-08-14 14:28:15.000000000 +0200
+++ binutils-2.22.patch/configure	2012-05-08 05:57:26.093455382 +0200
@@ -5946,8 +5946,7 @@
 main ()
 {
 #if CLOOG_VERSION_MAJOR != 0 \
-    || CLOOG_VERSION_MINOR != 16 \
-    || CLOOG_VERSION_REVISION < 1
+    || CLOOG_VERSION_MINOR != 18
     choke me
    #endif
   ;
diff -Naur binutils-2.22/configure.ac binutils-2.22.patch/configure.ac
--- binutils-2.22/configure.ac	2011-11-21 12:58:27.000000000 +0100
+++ binutils-2.22.patch/configure.ac	2012-05-08 05:54:27.974921129 +0200
@@ -1589,7 +1589,7 @@
   dnl
   dnl If we use CLooG-Legacy, the provided version information is
   dnl ignored.
-  CLOOG_CHECK_VERSION(0,16,1)
+  CLOOG_CHECK_VERSION(0,18,0)
 
   dnl Only execute fail-action, if CLooG has been requested.
   CLOOG_IF_FAILED([
