diff -Naur binutils-2.20.51.0.9-old/bfd/configure binutils-2.20.51.0.9-new/bfd/configure
--- binutils-2.20.51.0.9-old/bfd/configure	2010-05-27 11:22:20.000000000 -0700
+++ binutils-2.20.51.0.9-new/bfd/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10320,7 +10320,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/binutils/configure binutils-2.20.51.0.9-new/binutils/configure
--- binutils-2.20.51.0.9-old/binutils/configure	2010-05-27 11:22:22.000000000 -0700
+++ binutils-2.20.51.0.9-new/binutils/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10114,7 +10114,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/gas/configure binutils-2.20.51.0.9-new/gas/configure
--- binutils-2.20.51.0.9-old/gas/configure	2010-04-13 06:26:42.000000000 -0700
+++ binutils-2.20.51.0.9-new/gas/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10101,7 +10101,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/gprof/configure binutils-2.20.51.0.9-new/gprof/configure
--- binutils-2.20.51.0.9-old/gprof/configure	2010-04-13 06:26:45.000000000 -0700
+++ binutils-2.20.51.0.9-new/gprof/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10039,7 +10039,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/ld/configure binutils-2.20.51.0.9-new/ld/configure
--- binutils-2.20.51.0.9-old/ld/configure	2010-05-27 11:22:31.000000000 -0700
+++ binutils-2.20.51.0.9-new/ld/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10528,7 +10528,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/libiberty/Makefile.in binutils-2.20.51.0.9-new/libiberty/Makefile.in
--- binutils-2.20.51.0.9-old/libiberty/Makefile.in	2010-05-27 11:22:33.000000000 -0700
+++ binutils-2.20.51.0.9-new/libiberty/Makefile.in	2010-05-27 16:15:59.000000000 -0700
@@ -327,7 +327,7 @@
 # multilib-specific flags, it's overridden by FLAGS_TO_PASS from the
 # default multilib, so we have to take CFLAGS into account as well,
 # since it will be passed the multilib flags.
-MULTIOSDIR = `$(CC) $(CFLAGS) -print-multi-os-directory`
+MULTIOSDIR = '../lib'
 install_to_libdir: all
 	${mkinstalldirs} $(DESTDIR)$(libdir)/$(MULTIOSDIR)
 	$(INSTALL_DATA) $(TARGETLIB) $(DESTDIR)$(libdir)/$(MULTIOSDIR)/$(TARGETLIB)n
diff -Naur binutils-2.20.51.0.9-old/libtool.m4 binutils-2.20.51.0.9-new/libtool.m4
--- binutils-2.20.51.0.9-old/libtool.m4	2010-01-15 09:05:56.000000000 -0800
+++ binutils-2.20.51.0.9-new/libtool.m4	2010-05-27 16:15:59.000000000 -0700
@@ -2043,7 +2043,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
diff -Naur binutils-2.20.51.0.9-old/opcodes/configure binutils-2.20.51.0.9-new/opcodes/configure
--- binutils-2.20.51.0.9-old/opcodes/configure	2010-04-13 06:26:47.000000000 -0700
+++ binutils-2.20.51.0.9-new/opcodes/configure	2010-05-27 16:15:59.000000000 -0700
@@ -10050,7 +10050,7 @@
   # Ok, now we have the path, separated by spaces, we can step through it
   # and add multilib dir if necessary.
   lt_tmp_lt_search_path_spec=
-  lt_multi_os_dir=`$CC $CPPFLAGS $CFLAGS $LDFLAGS -print-multi-os-directory 2>/dev/null`
+  lt_multi_os_dir='../lib'
   for lt_sys_path in $lt_search_path_spec; do
     if test -d "$lt_sys_path/$lt_multi_os_dir"; then
       lt_tmp_lt_search_path_spec="$lt_tmp_lt_search_path_spec $lt_sys_path/$lt_multi_os_dir"
