diff --git a/libs/sectioncrc/Makefile b/libs/sectioncrc/Makefile
index 87e4627..ad8a0e5 100755
--- a/libs/sectioncrc/Makefile
+++ b/libs/sectioncrc/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  = -g -MD -Wall -I. -I../../include 
 
 OBJ = sectioncrc.o 
diff --git a/tools/dsmcc-receive/Makefile b/tools/dsmcc-receive/Makefile
index 891a50c..ee2d82e 100644
--- a/tools/dsmcc-receive/Makefile
+++ b/tools/dsmcc-receive/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc -lz
 
diff --git a/tools/eitsecactualtoanother/Makefile b/tools/eitsecactualtoanother/Makefile
index be3a790..723c857 100644
--- a/tools/eitsecactualtoanother/Makefile
+++ b/tools/eitsecactualtoanother/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/eitsecfilter/Makefile b/tools/eitsecfilter/Makefile
index f4e5fc4..cc94b71 100644
--- a/tools/eitsecfilter/Makefile
+++ b/tools/eitsecfilter/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/eitsecmapper/Makefile b/tools/eitsecmapper/Makefile
index 9c6ce8f..29dbc39 100644
--- a/tools/eitsecmapper/Makefile
+++ b/tools/eitsecmapper/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/esaudio2pes/Makefile b/tools/esaudio2pes/Makefile
index 349b187..607f0a3 100644
--- a/tools/esaudio2pes/Makefile
+++ b/tools/esaudio2pes/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/esaudioinfo/Makefile b/tools/esaudioinfo/Makefile
index 2c014b2..69a38a0 100644
--- a/tools/esaudioinfo/Makefile
+++ b/tools/esaudioinfo/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/esvideompeg2info/Makefile b/tools/esvideompeg2info/Makefile
index aaa8065..ac91058 100644
--- a/tools/esvideompeg2info/Makefile
+++ b/tools/esvideompeg2info/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/esvideompeg2pes/Makefile b/tools/esvideompeg2pes/Makefile
index a905e7b..7020503 100644
--- a/tools/esvideompeg2pes/Makefile
+++ b/tools/esvideompeg2pes/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/i13942ts/Makefile b/tools/i13942ts/Makefile
index 72e4b2d..0c020f8 100644
--- a/tools/i13942ts/Makefile
+++ b/tools/i13942ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include  -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 
 OBJS = i13942ts.o
diff --git a/tools/ip2sec/Makefile b/tools/ip2sec/Makefile
index 3a90c82..7580ca3 100644
--- a/tools/ip2sec/Makefile
+++ b/tools/ip2sec/Makefile
@@ -1,4 +1,3 @@
-CC      = gcc
 CFLAGS  += -g -O2 -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lpcap
 
diff --git a/tools/m2ts2cbrts/Makefile b/tools/m2ts2cbrts/Makefile
index a2ed89f..778efb8 100644
--- a/tools/m2ts2cbrts/Makefile
+++ b/tools/m2ts2cbrts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/mpe2sec/Makefile b/tools/mpe2sec/Makefile
index 5c29efd..55cbe4d 100644
--- a/tools/mpe2sec/Makefile
+++ b/tools/mpe2sec/Makefile
@@ -1,4 +1,3 @@
-CC      = gcc
 CFLAGS  += -g -O2 -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 
 OBJS    = mpe.o sectioncrc.o
diff --git a/tools/mpeg2videovbv/Makefile b/tools/mpeg2videovbv/Makefile
index 8635e67..4074c93 100644
--- a/tools/mpeg2videovbv/Makefile
+++ b/tools/mpeg2videovbv/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/oddparity/Makefile b/tools/oddparity/Makefile
index 7970590..6839a36 100644
--- a/tools/oddparity/Makefile
+++ b/tools/oddparity/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pes2es/Makefile b/tools/pes2es/Makefile
index a598df9..c3e803f 100644
--- a/tools/pes2es/Makefile
+++ b/tools/pes2es/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pes2txt/Makefile b/tools/pes2txt/Makefile
index d65922e..50057a1 100644
--- a/tools/pes2txt/Makefile
+++ b/tools/pes2txt/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pesaudio2ts/Makefile b/tools/pesaudio2ts/Makefile
index 6fec027..ab75796 100644
--- a/tools/pesaudio2ts/Makefile
+++ b/tools/pesaudio2ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pesdata2ts/Makefile b/tools/pesdata2ts/Makefile
index 3e180f7..d541485 100644
--- a/tools/pesdata2ts/Makefile
+++ b/tools/pesdata2ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pesinfo/Makefile b/tools/pesinfo/Makefile
index 5c9cbb3..3ffc1d7 100644
--- a/tools/pesinfo/Makefile
+++ b/tools/pesinfo/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/pesvideo2ts/Makefile b/tools/pesvideo2ts/Makefile
index a8de20e..a2ebab0 100644
--- a/tools/pesvideo2ts/Makefile
+++ b/tools/pesvideo2ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/sec2ts/Makefile b/tools/sec2ts/Makefile
index 0607bcf..df22126 100644
--- a/tools/sec2ts/Makefile
+++ b/tools/sec2ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/ts2m2ts/Makefile b/tools/ts2m2ts/Makefile
index b70c018..28104fd 100644
--- a/tools/ts2m2ts/Makefile
+++ b/tools/ts2m2ts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/ts2pes/Makefile b/tools/ts2pes/Makefile
index 61301f4..349c29e 100644
--- a/tools/ts2pes/Makefile
+++ b/tools/ts2pes/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/ts2sec/Makefile b/tools/ts2sec/Makefile
index ebbe4be..baad160 100644
--- a/tools/ts2sec/Makefile
+++ b/tools/ts2sec/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tscbrmuxer/Makefile b/tools/tscbrmuxer/Makefile
index aa93c07..38cd632 100644
--- a/tools/tscbrmuxer/Makefile
+++ b/tools/tscbrmuxer/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsccc/Makefile b/tools/tsccc/Makefile
index 44340fb..85ec0f8 100755
--- a/tools/tsccc/Makefile
+++ b/tools/tsccc/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tscrypt/Makefile b/tools/tscrypt/Makefile
index 3ce49f5..9cec259 100644
--- a/tools/tscrypt/Makefile
+++ b/tools/tscrypt/Makefile
@@ -1,4 +1,3 @@
-CC      = gcc
 CFLAGS  += -g -O2 -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -ldvbcsa
 
diff --git a/tools/tsdiscont/Makefile b/tools/tsdiscont/Makefile
index bed498e..78329c1 100644
--- a/tools/tsdiscont/Makefile
+++ b/tools/tsdiscont/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsdoubleoutput/Makefile b/tools/tsdoubleoutput/Makefile
index e35c8a2..e8927fe 100644
--- a/tools/tsdoubleoutput/Makefile
+++ b/tools/tsdoubleoutput/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsfilter/Makefile b/tools/tsfilter/Makefile
index 5e8a3fc..9397197 100755
--- a/tools/tsfilter/Makefile
+++ b/tools/tsfilter/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsfixcc/Makefile b/tools/tsfixcc/Makefile
index cdf93ae..ebd894f 100755
--- a/tools/tsfixcc/Makefile
+++ b/tools/tsfixcc/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsinputswitch/Makefile b/tools/tsinputswitch/Makefile
index 5b2f205..3a08c7a 100644
--- a/tools/tsinputswitch/Makefile
+++ b/tools/tsinputswitch/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsloop/Makefile b/tools/tsloop/Makefile
index 0781c2c..71dc3c6 100755
--- a/tools/tsloop/Makefile
+++ b/tools/tsloop/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsmask/Makefile b/tools/tsmask/Makefile
index afe3fe8..914a1ef 100755
--- a/tools/tsmask/Makefile
+++ b/tools/tsmask/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsmodder/Makefile b/tools/tsmodder/Makefile
index 72a4fde..2199b8a 100644
--- a/tools/tsmodder/Makefile
+++ b/tools/tsmodder/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsnullfiller/Makefile b/tools/tsnullfiller/Makefile
index fd7bf5a..3b6898f 100644
--- a/tools/tsnullfiller/Makefile
+++ b/tools/tsnullfiller/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsnullshaper/Makefile b/tools/tsnullshaper/Makefile
index 2071129..3b89b7e 100644
--- a/tools/tsnullshaper/Makefile
+++ b/tools/tsnullshaper/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc -lrt
 
diff --git a/tools/tsororts/Makefile b/tools/tsororts/Makefile
index 408c924..0687c55 100644
--- a/tools/tsororts/Makefile
+++ b/tools/tsororts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsorts/Makefile b/tools/tsorts/Makefile
index 1b4d287..96db276 100644
--- a/tools/tsorts/Makefile
+++ b/tools/tsorts/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsoutputswitch/Makefile b/tools/tsoutputswitch/Makefile
index c2d0b09..6abfddb 100644
--- a/tools/tsoutputswitch/Makefile
+++ b/tools/tsoutputswitch/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tspcrmeasure/Makefile b/tools/tspcrmeasure/Makefile
index 819a25e..f39efd3 100644
--- a/tools/tspcrmeasure/Makefile
+++ b/tools/tspcrmeasure/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tspcrrestamp/Makefile b/tools/tspcrrestamp/Makefile
index 3f05709..43f37b4 100644
--- a/tools/tspcrrestamp/Makefile
+++ b/tools/tspcrrestamp/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tspcrstamp/Makefile b/tools/tspcrstamp/Makefile
index 5520ef3..6c050bc 100644
--- a/tools/tspcrstamp/Makefile
+++ b/tools/tspcrstamp/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tspidmapper/Makefile b/tools/tspidmapper/Makefile
index c02853a..f237a77 100644
--- a/tools/tspidmapper/Makefile
+++ b/tools/tspidmapper/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsstamp/Makefile b/tools/tsstamp/Makefile
index 80393fe..f288f09 100644
--- a/tools/tsstamp/Makefile
+++ b/tools/tsstamp/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tstcpreceive/Makefile b/tools/tstcpreceive/Makefile
index 4f1cac7..ebbba5f 100644
--- a/tools/tstcpreceive/Makefile
+++ b/tools/tstcpreceive/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tstcpsend/Makefile b/tools/tstcpsend/Makefile
index 8894859..fb8a573 100644
--- a/tools/tstcpsend/Makefile
+++ b/tools/tstcpsend/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc -lrt
 
diff --git a/tools/tstdt/Makefile b/tools/tstdt/Makefile
index 1873adb..c5eb23d 100644
--- a/tools/tstdt/Makefile
+++ b/tools/tstdt/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tstimedwrite/Makefile b/tools/tstimedwrite/Makefile
index 3ad1432..7a34191 100644
--- a/tools/tstimedwrite/Makefile
+++ b/tools/tstimedwrite/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc -lrt
 
diff --git a/tools/tstimeout/Makefile b/tools/tstimeout/Makefile
index 0dbc314..c37b97e 100644
--- a/tools/tstimeout/Makefile
+++ b/tools/tstimeout/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsudpreceive/Makefile b/tools/tsudpreceive/Makefile
index 2e08cee..3a6c3cb 100644
--- a/tools/tsudpreceive/Makefile
+++ b/tools/tsudpreceive/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/tsudpsend/Makefile b/tools/tsudpsend/Makefile
index 0d55bd4..bd6f39e 100644
--- a/tools/tsudpsend/Makefile
+++ b/tools/tsudpsend/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc -lrt
 
diff --git a/tools/tsvbr2cbr/Makefile b/tools/tsvbr2cbr/Makefile
index 01cc992..65d670c 100644
--- a/tools/tsvbr2cbr/Makefile
+++ b/tools/tsvbr2cbr/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/txt2pes/Makefile b/tools/txt2pes/Makefile
index f506da7..55f0d5e 100644
--- a/tools/txt2pes/Makefile
+++ b/tools/txt2pes/Makefile
@@ -1,4 +1,3 @@
-C      = gcc
 CFLAGS  += -g -MD -Wall -I. -I../../include $(CPPFLAGS)
 LDFLAGS  += -lc
 
diff --git a/tools/zpipe/Makefile b/tools/zpipe/Makefile
index 5997129..707647d 100644
--- a/tools/zpipe/Makefile
+++ b/tools/zpipe/Makefile
@@ -1,4 +1,3 @@
-CC      = gcc
 CFLAGS  += -g -O2 -MD -Wall -I. -I../../include -D_FILE_OFFSET_BITS=64 $(CPPFLAGS)
 LDFLAGS  += -lz
 
