--- snapcast-0.14.0/client/Makefile	2018-04-27 19:43:25.000000000 +0200
+++ snapcast-0.14.0.makefiles/client/Makefile	2018-05-21 13:14:12.388757526 +0200
@@ -83,7 +83,6 @@
 
 else
 
-CXX       = g++
 CXXFLAGS += -pthread -DHAS_VORBIS -DHAS_ALSA -DHAS_AVAHI -DHAS_DAEMON
-LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -latomic
+LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -logg
 OBJ      += ../common/daemon.o player/alsa_player.o browseZeroConf/browse_avahi.o
diff -Naur snapcast-0.14.0/server/Makefile snapcast-0.14.0.makefiles/server/Makefile
--- snapcast-0.14.0/server/Makefile	2018-04-27 19:43:25.000000000 +0200
+++ snapcast-0.14.0.makefiles/server/Makefile	2018-05-21 13:14:56.881206277 +0200
@@ -88,7 +88,6 @@
 
 else
 
-CXX       = g++
 CXXFLAGS += -DHAS_AVAHI -DHAS_DAEMON -DHAS_ALSA -pthread
-LDFLAGS  += -lrt -lasound -lavahi-client -lavahi-common -latomic
+LDFLAGS  += -lrt -lasound -lavahi-client -lavahi-common
 OBJ      += ../common/daemon.o publishZeroConf/publish_avahi.o streamreader/alsa_stream.o
