diff -Naur snapcast-0.13.0/client/Makefile snapcast-0.13.0.Makefiles/client/Makefile
--- snapcast-0.13.0/client/Makefile	2018-03-04 21:43:32.000000000 +0100
+++ snapcast-0.13.0.Makefiles/client/Makefile	2018-03-05 13:33:36.686709820 +0100
@@ -78,10 +78,8 @@
 
 else
 
-CXX       = g++
-STRIP     = strip
 CXXFLAGS += -pthread -DHAS_VORBIS -DHAS_ALSA -DHAS_AVAHI -DHAS_DAEMON
-LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -static-libgcc -static-libstdc++ -latomic
+LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -logg
 OBJ      += ../common/daemon.o player/alsaPlayer.o browseZeroConf/browseAvahi.o
 
 endif
diff -Naur snapcast-0.13.0/server/Makefile snapcast-0.13.0.Makefiles/server/Makefile
--- snapcast-0.13.0/server/Makefile	2018-03-04 21:43:32.000000000 +0100
+++ snapcast-0.13.0.Makefiles/server/Makefile	2018-03-05 13:34:03.530858350 +0100
@@ -84,10 +84,8 @@
 
 else
 
-CXX       = g++
-STRIP     = strip
 CXXFLAGS += -DHAS_AVAHI -DHAS_DAEMON -pthread
-LDFLAGS   = -lrt -lvorbis -lvorbisenc -logg -lFLAC -lavahi-client -lavahi-common -static-libgcc -static-libstdc++
+LDFLAGS   = -lrt -lvorbis -lvorbisenc -logg -lFLAC -lavahi-client -lavahi-common
 OBJ      += ../common/daemon.o publishZeroConf/publishAvahi.o 
 
 endif
