diff -Naur cxxtools-2.2/configure cxxtools-2.2.patch/configure
--- cxxtools-2.2/configure	2013-04-21 21:13:11.000000000 +0200
+++ cxxtools-2.2.patch/configure	2014-01-03 20:34:53.404660480 +0100
@@ -17316,8 +17316,8 @@
 
 
 
-CXXTOOLS_CXXFLAGS='-I${includedir}'
-CXXTOOLS_LDFLAGS='-L${libdir} -lcxxtools'
+CXXTOOLS_CXXFLAGS=''
+CXXTOOLS_LDFLAGS='-lcxxtools'
 
 
 
diff -Naur cxxtools-2.2/configure.in cxxtools-2.2.patch/configure.in
--- cxxtools-2.2/configure.in	2013-04-20 23:31:50.000000000 +0200
+++ cxxtools-2.2.patch/configure.in	2014-01-03 20:26:32.064005192 +0100
@@ -106,8 +106,8 @@
 
 AC_PROG_LIBTOOL
 
-CXXTOOLS_CXXFLAGS='-I${includedir}'
-CXXTOOLS_LDFLAGS='-L${libdir} -lcxxtools'
+CXXTOOLS_CXXFLAGS=''
+CXXTOOLS_LDFLAGS='-lcxxtools'
 
 AC_SUBST(CXXTOOLS_CXXFLAGS)
 AC_SUBST(CXXTOOLS_LDFLAGS)
