--- a/xbmc/cores/AudioEngine/Sinks/AESinkPi.cpp	2016-09-17 16:35:20.000000000 +0100
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkPi.cpp	2016-10-01 19:26:40.470553260 +0100
@@ -527,6 +527,7 @@
   m_info.m_wantsIECPassthrough = true;
   list.push_back(m_info);
 
+#if 0
   m_info.m_channels.Reset();
   m_info.m_dataFormats.clear();
   m_info.m_streamTypes.clear();
@@ -548,6 +549,7 @@
 
   m_info.m_wantsIECPassthrough = true;
   list.push_back(m_info);
+#endif
 
   m_info.m_channels.Reset();
   m_info.m_dataFormats.clear();
