diff --git a/xbmc/windows/CMakeLists.txt b/xbmc/windows/CMakeLists.txt
index 3700602..c75f78f 100644
--- a/xbmc/windows/CMakeLists.txt
+++ b/xbmc/windows/CMakeLists.txt
@@ -1,3 +1,5 @@
+find_package(PNG REQUIRED)
+
 set(SOURCES GUIMediaWindow.cpp
             GUIWindowDebugInfo.cpp
             GUIWindowFileManager.cpp
@@ -9,7 +11,8 @@ set(SOURCES GUIMediaWindow.cpp
             GUIWindowSplash.cpp
             GUIWindowStartup.cpp
             GUIWindowSystemInfo.cpp
-            GUIWindowWeather.cpp)
+            GUIWindowWeather.cpp
+            readpng.c)
 
 set(HEADERS GUIMediaWindow.h
             GUIWindowDebugInfo.h
@@ -22,6 +25,9 @@ set(HEADERS GUIMediaWindow.h
             GUIWindowSplash.h
             GUIWindowStartup.h
             GUIWindowSystemInfo.h
-            GUIWindowWeather.h)
+            GUIWindowWeather.h
+            readpng.h)
 
 core_add_library(windows)
+
+target_link_libraries(windows PRIVATE ${PNG_LIBRARIES})
