diff -Naur libamlplayer-m1-e7ba300/amadec/Makefile libamlplayer-m1-e7ba300.patch/amadec/Makefile
--- libamlplayer-m1-e7ba300/amadec/Makefile	2013-05-26 01:08:34.000000000 +0200
+++ libamlplayer-m1-e7ba300.patch/amadec/Makefile	2013-05-26 06:42:39.800719267 +0200
@@ -14,18 +14,16 @@
 
 LDFLAGS += -shared -lpthread -lm -lasound -lrt
 
-TARGET_DIR=$(shell pwd)/../../../target
-STAGING_DIR=$(shell pwd)/../../../staging
-CFLAGS+=-O2
-CFLAGS+=-I$(shell pwd) -I$(shell pwd)/include -I${STAGING_DIR}/include -I${STAGING_DIR}/usr/include -L${TARGET_DIR}/lib -L${TARGET_DIR}/usr/lib
+CFLAGS+=-O2 -fPIC
+CFLAGS+=-I$(shell pwd) -I$(shell pwd)/include
 CFLAGS+=-DALSA_OUT -DENABLE_WAIT_FORMAT
 
 export CC CFLAGS
 
-all: $(TARGET) 
+all: $(TARGET)
 
 $(TARGET): $(TARGET_OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(TARGET_OBJS:%.o=%.c) -o $@  
+	$(CC) $(CFLAGS) $(LDFLAGS) $(TARGET_OBJS:%.o=%.c) -o $@
 
 install:$(TARGET)
 	-install -m 555 ${TARGET} $(INSTALL_DIR)
diff -Naur libamlplayer-m1-e7ba300/amcodec/include/amports/aformat.h libamlplayer-m1-e7ba300.patch/amcodec/include/amports/aformat.h
--- libamlplayer-m1-e7ba300/amcodec/include/amports/aformat.h	2013-05-26 01:08:34.000000000 +0200
+++ libamlplayer-m1-e7ba300.patch/amcodec/include/amports/aformat.h	2013-05-26 06:42:39.821719205 +0200
@@ -76,7 +76,7 @@
 								 ||(afmt == AFORMAT_AMR)\
 								 ||(afmt == AFORMAT_ALAC)\
 								 ||(afmt == AFORMAT_AC3) \
-								 ||(afmt == AFORMAT_EAC3) \								 
+								 ||(afmt == AFORMAT_EAC3) \
 								 ||(afmt == AFORMAT_APE) \
 								 ||(afmt == AFORMAT_FLAC) )
 
diff -Naur libamlplayer-m1-e7ba300/amcodec/Makefile libamlplayer-m1-e7ba300.patch/amcodec/Makefile
--- libamlplayer-m1-e7ba300/amcodec/Makefile	2013-05-26 01:08:34.000000000 +0200
+++ libamlplayer-m1-e7ba300.patch/amcodec/Makefile	2013-05-26 06:42:39.827719188 +0200
@@ -35,9 +35,6 @@
 INCLUDE=${SRCTREE}/include/ 
 #CFLAGS=-O0  -gdwarf-2  -g
 CFLAGS=-O2   -I$(SRC)/include  -I${SRCTREE}/audio_ctl -I${SRCTREE}/codec -I${SRCTREE}/../amadec/include -fPIC
-CFLAGS+=-mfloat-abi=softfp -mfpu=neon -mtune=cortex-a9 -march=armv7-a
-
-
 
 
 target_all=  $(TARGET)$(LIB_VERSION)$(LIB_SUB_VERSION)
@@ -51,9 +48,6 @@
 $(target_all):$(DIROBJ)
 	$(call ld_mk,$(OUT_DIR)$@,$^)
 
-
-
-
 include $(SRCTREE)/rules.mk
 export CC CPP AS AR LD TOPDIR SRCTREE  CFLAGS LDFLAGS Q
 
diff -Naur libamlplayer-m1-e7ba300/amcodec/rules.mk libamlplayer-m1-e7ba300.patch/amcodec/rules.mk
--- libamlplayer-m1-e7ba300/amcodec/rules.mk	2013-05-26 01:08:34.000000000 +0200
+++ libamlplayer-m1-e7ba300.patch/amcodec/rules.mk	2013-05-26 06:42:39.828719185 +0200
@@ -19,10 +19,10 @@
 
 ld_mk=$(Q)echo "LD $(1) $(2)  $(3)";\
 		$(CC) $(LDFLAGS)  \
-		--start-group	\
+		-Wl,--start-group	\
 		$(2:%-dir=%/build-in.o) \
 		$(3:%-dir=%/build-in.o) \
-		--end-group    \
+		-Wl,--end-group    \
 		-o $(1) 	
 				
 clr_mk=$(Q)echo "CLEAN  $(1)";\
diff -Naur libamlplayer-m1-e7ba300/amplayer/Makefile.mk libamlplayer-m1-e7ba300.patch/amplayer/Makefile.mk
--- libamlplayer-m1-e7ba300/amplayer/Makefile.mk	2013-05-26 01:08:33.000000000 +0200
+++ libamlplayer-m1-e7ba300.patch/amplayer/Makefile.mk	2013-05-26 06:43:36.590551945 +0200
@@ -33,7 +33,7 @@
 
 
 
-LDFLAGS+= -L$(PREFIX)/lib -lavutil -lavformat -lswscale -lavcodec -lavutil -lm  -lpthread -lamcodec 
+LDFLAGS+= -L$(PREFIX)/lib -lavutil_aml -lavformat_aml -lswscale_aml -lavcodec_aml -lavutil_aml -lm  -lpthread -lamcodec
 
 INSTALL_DIR?=$(PREFIX)/lib
 LDFLAGS+=-shared 
@@ -48,9 +48,8 @@
 	CFLAGS+=-I$(SRC)/player/include
 endif
 
-CFLAGS+=-I${SRCTREE}/../amffmpeg -I${SRCTREE}/../amcodec/include -I${SRCTREE}/../amadec/include
+CFLAGS+=-I${SRCTREE}/../amffmpeg -I${SRCTREE}/../amcodec/include -I${SRCTREE}/../amadec/include -I${SRCTREE}/../amavutils/include
 CFLAGS+=-fPIC -DENABLE_FREE_SCALE
-CFLAGS+=-mfloat-abi=softfp -mfpu=neon -mtune=cortex-a9 -march=armv7-a
 target_all=  $(TARGET)
 
 
