From 0b2b2e26bbae3d06580080a66a85e5e0d90ea9f6 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 10 Aug 2014 14:21:19 +0300
Subject: [PATCH] link static

---
 prog/sensors/Module.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/prog/sensors/Module.mk b/prog/sensors/Module.mk
index b656441..ad59874 100644
--- a/prog/sensors/Module.mk
+++ b/prog/sensors/Module.mk
@@ -40,7 +40,7 @@ REMOVESENSORSMAN := $(patsubst $(MODULE_DIR)/%,$(DESTDIR)$(PROGSENSORSMAN1DIR)/%
 LIBICONV := $(shell if /sbin/ldconfig -p | grep -q '/libiconv\.so$$' ; then echo \-liconv; else echo; fi)
 
 $(PROGSENSORSTARGETS): $(PROGSENSORSSOURCES:.c=.ro) lib/$(LIBSHBASENAME)
-	$(CC) $(EXLDFLAGS) -o $@ $(PROGSENSORSSOURCES:.c=.ro) $(LIBICONV) -Llib -lsensors
+	$(CC) $(EXLDFLAGS) -o $@ $(PROGSENSORSSOURCES:.c=.ro) $(LIBICONV) -Llib -Wl,-Bstatic -lsensors -Wl,-Bdynamic -lm
 
 all-prog-sensors: $(PROGSENSORSTARGETS)
 user :: all-prog-sensors
-- 
1.7.2.5

