diff -Naur libvorbis-1.3.2-old/configure.ac libvorbis-1.3.2-new/configure.ac
--- libvorbis-1.3.2-old/configure.ac	2010-11-01 12:27:12.000000000 -0700
+++ libvorbis-1.3.2-new/configure.ac	2010-11-01 14:50:03.000000000 -0700
@@ -8,6 +8,7 @@
 AC_INIT([libvorbis],[1.3.2],[vorbis-dev@xiph.org])
 
 AC_CONFIG_SRCDIR([lib/mdct.c])
+AC_CONFIG_MACRO_DIR([m4])
 
 AC_CANONICAL_TARGET([])
 
@@ -16,7 +17,6 @@
 AM_CONFIG_HEADER([config.h])
 
 dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I m4")
 
 dnl Library versioning
 dnl - library source changed -> increment REVISION
diff -Naur libvorbis-1.3.2-old/Makefile.am libvorbis-1.3.2-new/Makefile.am
--- libvorbis-1.3.2-old/Makefile.am	2010-03-25 18:56:25.000000000 -0700
+++ libvorbis-1.3.2-new/Makefile.am	2010-11-01 14:50:03.000000000 -0700
@@ -2,6 +2,8 @@
 
 AUTOMAKE_OPTIONS = 1.7 foreign dist-zip dist-bzip2
 
+ACLOCAL_AMFLAGS = -I m4
+
 SUBDIRS = m4 include vq lib test doc
 
 if BUILD_EXAMPLES 
