diff -Naur libvorbis-1.3.3-old/configure.ac libvorbis-1.3.3-new/configure.ac
--- libvorbis-1.3.3-old/configure.ac	2012-02-03 14:00:34.000000000 -0800
+++ libvorbis-1.3.3-new/configure.ac	2012-02-09 12:56:02.000000000 -0800
@@ -8,6 +8,7 @@
 AC_INIT([libvorbis],[1.3.3],[vorbis-dev@xiph.org])
 
 AC_CONFIG_SRCDIR([lib/mdct.c])
+AC_CONFIG_MACRO_DIR([m4])
 
 AC_CANONICAL_TARGET([])
 
@@ -16,7 +17,6 @@
 AC_CONFIG_HEADERS([config.h])
 
 dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I m4")
 
 dnl Library versioning
 dnl - library source changed -> increment REVISION
diff -Naur libvorbis-1.3.3-old/Makefile.am libvorbis-1.3.3-new/Makefile.am
--- libvorbis-1.3.3-old/Makefile.am	2010-12-21 00:46:03.000000000 -0800
+++ libvorbis-1.3.3-new/Makefile.am	2012-02-09 12:56:02.000000000 -0800
@@ -3,6 +3,8 @@
 #AUTOMAKE_OPTIONS = 1.7 foreign dist-zip dist-bzip2
 AUTOMAKE_OPTIONS = foreign 1.11 dist-zip dist-xz
 
+ACLOCAL_AMFLAGS = -I m4
+
 SUBDIRS = m4 include vq lib test doc
 
 if BUILD_EXAMPLES 
