diff -Naur xbmc-master-14-687ac52/xbmc/dialogs/GUIDialogSeekBar.cpp xbmc-master-14-687ac52.patch/xbmc/dialogs/GUIDialogSeekBar.cpp
--- xbmc-master-14-687ac52/xbmc/dialogs/GUIDialogSeekBar.cpp	2014-07-06 16:16:21.000000000 +0200
+++ xbmc-master-14-687ac52.patch/xbmc/dialogs/GUIDialogSeekBar.cpp	2014-07-06 19:45:14.102908695 +0200
@@ -68,7 +68,7 @@
   }
 
   // update controls
-  if (!g_application.GetSeekHandler()->InProgress() && !g_infoManager.m_performingSeek)
+  if (!g_application.GetSeekHandler()->InProgress() && !g_infoManager.m_performingSeek && g_infoManager.GetTotalPlayTime())
   { // position the bar at our current time
     CONTROL_SELECT_ITEM(POPUP_SEEK_LABEL, (unsigned int)(g_infoManager.GetPlayTime()/g_infoManager.GetTotalPlayTime() * 0.1f));
     SET_CONTROL_LABEL(POPUP_SEEK_LABEL, g_infoManager.GetCurrentPlayTime());
