diff -Naur xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh
--- xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh	2014-05-01 16:53:36.000000000 +0200
+++ xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh	2014-05-01 17:54:59.643325427 +0200
@@ -56,17 +56,17 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $dvdread_cflags $extracflags $threadcflags
+      echo $dvdread_cflags $extracflags $threadcflags
 fi
 
 if test "$echo_minicflags" = "yes"; then
-      echo -I$prefix/include -I$prefix/include/dvdnav $extracflags $threadcflags
+      echo $extracflags $threadcflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir -ldvdnav $dvdread_libs $threadlib
+      echo -ldvdnav $dvdread_libs $threadlib
 fi      
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L$libdir -ldvdnavmini $threadlib
+      echo -ldvdnavmini $threadlib
 fi
diff -Naur xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in
--- xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in	2014-05-01 16:53:36.000000000 +0200
+++ xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in	2014-05-01 17:56:12.030447794 +0200
@@ -7,5 +7,5 @@
 Description: DVD Navigation mini library
 Version: @VERSION@
 
-Cflags: -I${includedir} @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff -Naur xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnav.pc.in xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnav.pc.in
--- xbmc-master-14-088e9fa/lib/libdvd/libdvdnav/misc/dvdnav.pc.in	2014-05-01 16:53:36.000000000 +0200
+++ xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdnav/misc/dvdnav.pc.in	2014-05-01 17:55:50.427406667 +0200
@@ -8,5 +8,5 @@
 Version: @VERSION@
 
 Requires.private: dvdread >= 4.1.2
-Cflags: -I${includedir} @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff -Naur xbmc-master-14-088e9fa/lib/libdvd/libdvdread/misc/dvdread-config.sh xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdread/misc/dvdread-config.sh
--- xbmc-master-14-088e9fa/lib/libdvd/libdvdread/misc/dvdread-config.sh	2014-05-01 16:53:36.000000000 +0200
+++ xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdread/misc/dvdread-config.sh	2014-05-01 17:56:55.745553577 +0200
@@ -48,9 +48,9 @@
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $extracflags
+      echo $extracflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir $dvdreadlib
+      echo $dvdreadlib
 fi      
diff -Naur xbmc-master-14-088e9fa/lib/libdvd/libdvdread/misc/dvdread.pc.in xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdread/misc/dvdread.pc.in
--- xbmc-master-14-088e9fa/lib/libdvd/libdvdread/misc/dvdread.pc.in	2014-05-01 16:53:36.000000000 +0200
+++ xbmc-master-14-088e9fa.patch/lib/libdvd/libdvdread/misc/dvdread.pc.in	2014-05-01 17:57:28.731937685 +0200
@@ -7,5 +7,5 @@
 Description: Low level DVD access library
 Version: @VERSION@
 
-Cflags: -I${includedir}
-Libs: -L${libdir} -ldvdread
+Cflags: -I.
+Libs: -ldvdread
