diff --git a/xbmc/windowing/WinEventsX11.cpp b/xbmc/windowing/WinEventsX11.cpp
index ed31c04..34fcfae 100644
--- a/xbmc/windowing/WinEventsX11.cpp
+++ b/xbmc/windowing/WinEventsX11.cpp
@@ -709,7 +709,6 @@ bool CWinEventsX11::ProcessShortcuts(XBMC_Event& event)
     switch(event.key.keysym.sym)
     {
       case XBMCK_TAB:  // ALT+TAB to minimize/hide
-        g_application.Minimize();
         return true;
 
       default:
