diff --git a/xbmc/settings/GUIWindowSettingsCategory.cpp b/xbmc/settings/GUIWindowSettingsCategory.cpp
index cbf0acb..f789e18 100644
--- a/xbmc/settings/GUIWindowSettingsCategory.cpp
+++ b/xbmc/settings/GUIWindowSettingsCategory.cpp
@@ -2434,7 +2434,7 @@ DisplayMode CGUIWindowSettingsCategory::FillInScreens(CStdString strSetting, RES
     pControl->Clear();
 
     CStdString strScreen;
-    if (g_advancedSettings.m_canWindowed)
+    if (g_advancedSettings.m_canWindowed && !g_application.IsStandAlone())
       pControl->AddLabel(g_localizeStrings.Get(242), -1);
 
 #if !defined(HAS_GLX)
