diff -Naur libmicrohttpd-0.9.24/configure.ac libmicrohttpd-0.9.24.patch/configure.ac
--- libmicrohttpd-0.9.24/configure.ac	2012-12-25 19:36:51.000000000 +0100
+++ libmicrohttpd-0.9.24.patch/configure.ac	2013-01-12 03:26:44.632253887 +0100
@@ -23,7 +23,7 @@
 AC_PREREQ(2.57)
 AC_INIT([libmicrohttpd], [0.9.24],[libmicrohttpd@gnu.org])
 AM_INIT_AUTOMAKE([silent-rules])
-AM_CONFIG_HEADER([MHD_config.h])
+AC_CONFIG_HEADERS([MHD_config.h])
 AC_CONFIG_MACRO_DIR([m4])
 AH_TOP([#define _GNU_SOURCE  1])
 
