--- libupnp-1.4.1.old/upnp/src/soap/soap_device.c
+++ libupnp-1.4.1/upnp/src/soap/soap_device.c
@@ -69,9 +69,6 @@
 static const char *Soap_Action_Failed = "Action Failed";
 static const char *Soap_Invalid_Var = "Invalid Var";
 
-const char *ContentTypeHeader =
-    "CONTENT-TYPE: text/xml; charset=\"utf-8\"\r\n";
-
 /****************************************************************************
 *	Function :	get_request_type
 *
@@ -1074,3 +1071,8 @@
 #endif // EXCLUDE_SOAP
 
 #endif // INCLUDE_DEVICE_APIS
+
+#if (!defined INCLUDE_DEVICE_APIS && defined INCLUDE_CLIENT_APIS) || defined INCLUDE_DEVICE_APIS
+    const char *ContentTypeHeader =
+        "CONTENT-TYPE: text/xml; charset=\"utf-8\"\r\n";
+#endif
