diff --git a/mk_arcade_joystick_rpi.c b/mk_arcade_joystick_rpi.c
index 344a475..66550f7 100755
--- a/mk_arcade_joystick_rpi.c
+++ b/mk_arcade_joystick_rpi.c
@@ -47,6 +47,14 @@ MODULE_LICENSE("GPL");
 #define HAVE_TIMER_SETUP
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0)
+#define del_timer_sync timer_delete_sync
+#endif
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 16, 0)
+#define from_timer timer_container_of
+#endif
+
 #define MK_MAX_DEVICES		2
 #define MK_MAX_BUTTONS      13
 
