From d8f15e55b4d357ac0a2b384888080b120600d09d Mon Sep 17 00:00:00 2001
From: vpeter4 <peter.vicman@gmail.com>
Date: Wed, 8 Apr 2015 18:27:06 +0200
Subject: [PATCH] cflags

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c904eaf..df0fe3e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CPPFLAGS	:= -I.
-CFLAGS		:= -g -Wall -Werror
+CFLAGS		+= -g -Wall -Werror
 INSTALL		:= install
 DESTDIR		:=
 SPECFILE	:= keyutils.spec
-- 
1.8.1.2

