diff -Naur libnl-1.1/include/netlink-local.h libnl-1.1.patch/include/netlink-local.h
--- libnl-1.1/include/netlink-local.h	2008-01-14 16:48:45.000000000 +0100
+++ libnl-1.1.patch/include/netlink-local.h	2010-04-13 20:34:38.502955161 +0200
@@ -15,6 +15,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
