diff -Naur afpfs-ng-0.8.1/include/afpfs-ng/dsi.h afpfs-ng-0.8.1.patch/include/afpfs-ng/dsi.h
--- afpfs-ng-0.8.1/include/afpfs-ng/dsi.h	2011-09-10 12:14:34.170716482 +0200
+++ afpfs-ng-0.8.1.patch/include/afpfs-ng/dsi.h	2011-09-10 12:22:28.790093211 +0200
@@ -6,13 +6,15 @@
 
 struct dsi_request
 {
-	unsigned short requestid;
-	unsigned char subcommand;
-	void * other;
-	unsigned char wait;
-	pthread_cond_t  condition_cond;
-	struct dsi_request * next;
-	int return_code;
+        unsigned short requestid;
+        unsigned char subcommand;
+        void * other;
+        int wait;
+        int done_waiting;
+        pthread_cond_t  waiting_cond;
+        pthread_mutex_t waiting_mutex;
+        struct dsi_request * next;
+        int return_code;
 };
 
 int dsi_receive(struct afp_server * server, void * data, int size);
