diff -Naur afpfs-ng-0.8.1/configure.ac afpfs-ng-0.8.1.patch/configure.ac
--- afpfs-ng-0.8.1/configure.ac	2011-09-10 12:14:34.164716403 +0200
+++ afpfs-ng-0.8.1.patch/configure.ac	2011-09-10 12:15:13.266241771 +0200
@@ -51,21 +51,6 @@
 	case $host in
 		*-*-darwin*) 
 		AC_MSG_CHECKING([for correct gcrypt version])
-		AC_RUN_IFELSE(
-			[AC_LANG_PROGRAM([
-				#include <gcrypt.h>
-				#include <stdio.h>],[
-				char*p= GCRYPT_VERSION;
-				unsigned int vers;
-				vers=atoi(p)*10000;
-				p=strchr(p,'.')+1;
-				vers+=atoi(p)*100;
-				p=strchr(p,'.')+1;
-				vers+=atoi(p);
-				if (vers<10400) return 1;
-			])],
-			[AC_MSG_RESULT([yes])],
-			[AC_MSG_ERROR([version is < 1.4.0])])
 		AM_CONDITIONAL(HAVE_LIBGCRYPT, true)
 		AC_DEFINE([HAVE_LIBGCRYPT], [1] )
 		;;
