diff -Naur afpfs-ng-0.8.1/fuse/client.c afpfs-ng-0.8.1.patch/fuse/client.c
--- afpfs-ng-0.8.1/fuse/client.c	2008-03-08 03:44:16.000000000 +0100
+++ afpfs-ng-0.8.1.patch/fuse/client.c	2011-05-22 19:08:53.938152499 +0200
@@ -61,8 +61,8 @@
 			snprintf(filename, PATH_MAX,
 				"/usr/local/bin/%s",AFPFSD_FILENAME);
 			if (access(filename,X_OK)) {
-				snprintf(filename, "/usr/bin/%s",
-					AFPFSD_FILENAME);
+				snprintf(filename, PATH_MAX,
+				"/usr/bin/%s", AFPFSD_FILENAME);
 				if (access(filename,X_OK)) {
 					printf("Could not find server (%s)\n",
 						filename);
