--- a/drivers/gpu/drm/i915/intel_pm.c	2013-05-30 22:57:09.098080381 +0200
+++ b/drivers/gpu/drm/i915/intel_pm.c	2013-05-30 23:31:31.644308032 +0200
@@ -2503,8 +2503,8 @@
 	if (i915_enable_rc6 >= 0)
 		return i915_enable_rc6;
 
-	/* Disable RC6 on Ironlake */
-	if (INTEL_INFO(dev)->gen == 5)
+	/* Disable RC6 on Ironlake and Sandybridge GT1 */
+	if (INTEL_INFO(dev)->gen == 5 || IS_SNB_GT1(dev))
 		return 0;
 
 	if (IS_HASWELL(dev)) {
