diff -Naur libtool-2.4-old/libltdl/m4/libtool.m4 libtool-2.4-new/libltdl/m4/libtool.m4
--- libtool-2.4-old/libltdl/m4/libtool.m4	2010-09-22 01:41:19.000000000 -0700
+++ libtool-2.4-new/libltdl/m4/libtool.m4	2010-09-23 20:25:25.000000000 -0700
@@ -6905,6 +6905,9 @@
 *\ -flto*\ *) CFLAGS="$CFLAGS -fno-lto" ;;
 *\ -fwhopr*\ *) CFLAGS="$CFLAGS -fno-whopr" ;;
 esac
+case "$CC $CFLAGS " in #(
+*\ -fuse-linker-plugin*\ *) CFLAGS="$CFLAGS -fno-use-linker-plugin" ;;
+esac
 
 dnl Parse the compiler output and extract the necessary
 dnl objects, libraries and library flags.
