diff --git a/build-aux/ltmain.sh b/build-aux/ltmain.sh
index 5eca4ae..23901ff 100644
--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -8333,7 +8333,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      #add_dir=-L$libdir
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
