diff -Naur libtool-2.4-old/libltdl/config/ltmain.m4sh libtool-2.4-new/libltdl/config/ltmain.m4sh
--- libtool-2.4-old/libltdl/config/ltmain.m4sh	2010-09-22 13:16:41.000000000 -0700
+++ libtool-2.4-new/libltdl/config/ltmain.m4sh	2010-09-22 13:21:52.000000000 -0700
@@ -5051,7 +5051,7 @@
       # -O*, -flto*, -fwhopr*, -fuse-linker-plugin GCC link-time optimization
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*|--sysroot=*| \
-      -O*|-flto*|-fwhopr*|-fuse-linker-plugin)
+      -O*|-flto*|-fwhopr*|-fuse-linker-plugin|-fuse-ld=*)
         func_quote_for_eval "$arg"
 	arg="$func_quote_for_eval_result"
         func_append compile_command " $arg"
