From 88aefc6c5b8f271872d4c99355f0a7d888596d73 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Mon, 22 Jun 2020 19:13:57 +0200
Subject: [PATCH] Fix build with gcc 10

---
 lib/in_wsr/ws_audio.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/in_wsr/ws_audio.h b/lib/in_wsr/ws_audio.h
index 11d80da0b6cc..2d3d661ca498 100644
--- a/lib/in_wsr/ws_audio.h
+++ b/lib/in_wsr/ws_audio.h
@@ -10,6 +10,6 @@ void ws_audio_port_write(BYTE port,BYTE value);
 BYTE ws_audio_port_read(BYTE port);
 void ws_audio_process(void);
 void ws_audio_sounddma(void);
-int WaveAdrs;
+extern int WaveAdrs;
 
 #endif
-- 
2.27.0

