diff -Naur libtool-2.2.10-old/libltdl/config/ltmain.m4sh libtool-2.2.10-new/libltdl/config/ltmain.m4sh
--- libtool-2.2.10-old/libltdl/config/ltmain.m4sh	2010-07-04 16:25:41.000000000 -0700
+++ libtool-2.2.10-new/libltdl/config/ltmain.m4sh	2010-07-04 16:37:36.000000000 -0700
@@ -4492,7 +4492,7 @@
       # -O*, -flto*, -fwhopr, -fuse-linker-plugin GCC link-time optimization
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*| \
-      -O*|-flto*|-fwhopr|-fuse-linker-plugin)
+      -O*|-flto*|-fwhopr|-fuse-linker-plugin|-fuse-ld=*)
         func_quote_for_eval "$arg"
 	arg="$func_quote_for_eval_result"
         func_append compile_command " $arg"
