diff -Naur libaacs-0.5.0-old/configure.ac libaacs-0.5.0-new/configure.ac
--- libaacs-0.5.0-old/configure.ac	2012-08-17 04:28:42.000000000 -0700
+++ libaacs-0.5.0-new/configure.ac	2013-01-03 21:08:26.000000000 -0800
@@ -24,7 +24,7 @@
 AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_SYSTEM
 AM_INIT_AUTOMAKE([foreign])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
 SET_FEATURES="-D_ISOC99_SOURCE -D_POSIX_C_SOURCE=200112L"
