diff --git a/configure.ac b/configure.ac
index 466f7f9..0dec17f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -27,7 +27,7 @@ AM_INIT_AUTOMAKE([foreign dist-bzip2 no-dist-gzip])
 AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
-SET_FEATURES="-D_ISOC99_SOURCE -D_POSIX_C_SOURCE=200112L"
+SET_FEATURES="-D_ISOC99_SOURCE -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=500"
 
 case "${host_os}" in
   "")

