diff -Naur screen-4.3.1/configure screen-4.3.1.patch/configure
--- screen-4.3.1/configure	2015-06-28 23:42:40.000000000 +0200
+++ screen-4.3.1.patch/configure	2016-03-26 17:52:12.807054501 +0100
@@ -4156,7 +4156,6 @@
 
 
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking SVR4..." >&5
 $as_echo "$as_me: checking SVR4..." >&6;}
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
diff -Naur screen-4.3.1/configure.ac screen-4.3.1.patch/configure.ac
--- screen-4.3.1/configure.ac	2015-06-28 23:22:55.000000000 +0200
+++ screen-4.3.1.patch/configure.ac	2016-03-26 17:52:03.538036990 +0100
@@ -203,7 +203,6 @@
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
 AC_CHECKING(SVR4)
 AC_TRY_LINK([#include <utmpx.h>
 ],,
