diff -uNr a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts	2026-02-09 06:03:27.000000000 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts	2026-03-09 17:17:05.722368387 +0900
@@ -11,6 +11,25 @@
 / {
 	model = "Anbernic RG351M";
 	compatible = "anbernic,rg351m", "rockchip,rk3326";
+
+	battery: battery {
+			compatible = "simple-battery";
+			charge-full-design-microamp-hours = <3500000>;
+			charge-term-current-microamp = <350000>;
+			constant-charge-current-max-microamp = <2000000>;
+			constant-charge-voltage-max-microvolt = <4200000>;
+			factory-internal-resistance-micro-ohms = <180000>;
+			voltage-max-design-microvolt = <4100000>;
+			voltage-min-design-microvolt = <3500000>;
+
+			ocv-capacity-celsius = <20>;
+			ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+						<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+						<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+						<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+						<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+						<3574170 0>;
+	};
 };
 
 &internal_display {
@@ -19,3 +38,9 @@
 	rotation = <270>;
 	vdd-supply = <&vcc_lcd>;
 };
+
+&rk817_charger {
+	/* This device does not have a current sense resistor */
+	//rockchip,resistor-sense-micro-ohms = <0>;
+	monitored-battery = <&battery>;
+};
diff -uNr a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi	2026-02-09 06:03:27.000000000 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi	2026-03-09 17:37:00.940816921 +0900
@@ -15,6 +15,7 @@
 / {
 	aliases {
 		mmc0 = &sdmmc;
+		mmc1 = &sdio;
 	};
 
 	chosen {
@@ -341,6 +342,12 @@
 			};
 		};
 
+		rk817_charger: charger {
+			rockchip,resistor-sense-micro-ohms = <10000>;
+			rockchip,sleep-enter-current-microamp = <300000>;
+			rockchip,sleep-filter-current-microamp = <100000>;
+		};
+
 		rk817_codec: codec {
 			rockchip,mic-in-differential;
 		};
@@ -393,6 +400,19 @@
 	status = "okay";
 };
 
+&sdio {
+	cap-sd-highspeed;
+	card-detect-delay = <200>;
+	cd-gpios = <&gpio3 RK_PB4 GPIO_ACTIVE_LOW>;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vmmc-supply = <&vcc_sd>;
+	vqmmc-supply = <&vccio_sd>;
+	status = "disabled";
+};
+
 &sfc {
 	#address-cells = <1>;
 	pinctrl-0 = <&sfc_clk &sfc_cs0 &sfc_bus2>;
diff -uNr a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts	2026-02-09 06:03:27.000000000 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts	2026-03-09 17:26:18.882440826 +0900
@@ -7,9 +7,11 @@
 	model = "Anbernic RG351V";
 	compatible = "anbernic,rg351v", "rockchip,rk3326";
 
-	gpio_keys_vol: gpio-keys-vol {
+	gpio_keys: odroidgo3-keys {
 		compatible = "gpio-keys";
 		autorepeat;
+		pinctrl-0 = <&btn_pins_vol>;
+		pinctrl-names = "default";
 
 		button-vol-down {
 			gpios = <&gpio2 RK_PA1 GPIO_ACTIVE_LOW>;
@@ -23,10 +25,30 @@
 			linux,code = <KEY_VOLUMEUP>;
 		};
 	};
+
+	battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <3500000>;
+		charge-term-current-microamp = <350000>;
+		constant-charge-current-max-microamp = <2000000>;
+		constant-charge-voltage-max-microvolt = <4200000>;
+		factory-internal-resistance-micro-ohms = <180000>;
+		voltage-max-design-microvolt = <4100000>;
+		voltage-min-design-microvolt = <3500000>;
+
+		ocv-capacity-celsius = <20>;
+		ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+					<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+					<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+					<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+					<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+					<3574170 0>;
+	};
 };
 
 &internal_display {
 	compatible = "anbernic,rg351v-panel", "newvision,nv3051d";
+	iovcc-supply = <&vcc_lcd>;
 	vdd-supply = <&vcc_lcd>;
 };
 
@@ -42,3 +64,22 @@
 &vccio_sd {
 	regulator-max-microvolt = <1800000>;
 };
+
+&sdio {
+	status = "okay";
+};
+
+&rk817_charger {
+       /* This device does not have a current sense resistor */
+       rockchip,resistor-sense-micro-ohms = <0>;
+       monitored-battery = <&battery>;
+};
+
+&pinctrl {
+	btns {
+		btn_pins_vol: btn-pins-vol {
+			rockchip,pins = <2 RK_PA0 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+};
