diff -uNr a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi	2026-03-01 20:34:20.594973900 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dtsi	2026-03-01 23:09:56.589847184 +0900
@@ -182,6 +182,7 @@
 		pinctrl-0 = <&pmic_int>, <&i2s1_2ch_mclk>;
 		pinctrl-names = "default";
 		#sound-dai-cells = <0>;
+		rockchip,system-power-controller;
 		wakeup-source;
 
 		vcc1-supply = <&vccsys>;
