diff --git a/src/gapi_gl.h b/src/gapi_gl.h
index 4e9b47d..3a672f6 100644
--- a/src/gapi_gl.h
+++ b/src/gapi_gl.h
@@ -752,7 +752,7 @@ namespace GAPI {
             bool isVolume = (opt & OPT_VOLUME)  != 0;
             bool isShadow = fmt == FMT_SHADOW;
 
-            target = isVolume ? GL_TEXTURE_3D : (isCube ? GL_TEXTURE_CUBE_MAP : GL_TEXTURE_2D);
+            target = isVolume ? GL_TEXTURE_3D_OES : (isCube ? GL_TEXTURE_CUBE_MAP : GL_TEXTURE_2D);
 
             glGenTextures(1, &ID);
 
