diff -Naur libplayer-20090929/configure libplayer-20090929.patch/configure
--- libplayer-20090929/configure	2009-09-30 03:22:00.000000000 +0200
+++ libplayer-20090929.patch/configure	2009-10-01 04:44:24.830578857 +0200
@@ -805,19 +805,18 @@
 #   check for libX11
 #################################################
 if test "$x11" = "auto" -o "$x11" = "yes"; then
-  if test "$wrapper_xine" = "yes" -o "$wrapper_mplayer" = "yes"; then
-    x11="no"
-    echolog "Checking for libX11 ..."
-    if [ -n "$libX11dir" ]; then
-      check_cflags -I$libX11dir/include
-      check_ldflags -L$libX11dir/lib
-    fi
-    check_lib X11/X.h XInitThreads -lX11
-    if [ "$?" = 0 ]; then
-      add_extracflags -DUSE_X11
-      add_pkgconfig_requires x11
-      x11="yes"
-    fi
+  x11="no"
+  echolog "Checking for libX11 ..."
+  if [ -n "$libX11dir" ]; then
+    check_cflags -I$libX11dir/include
+    check_ldflags -L$libX11dir/lib
+  fi
+  check_lib X11/X.h XInitThreads -lX11
+  if [ "$?" = 0 ]; then
+    add_extracflags -DUSE_X11
+    add_pkgconfig_requires x11
+    add_extralibs -lX11 -lm
+    x11="yes"
   fi
 fi
 
