--- /dev/null	2015-06-05 20:37:04.396503995 +0200
+++ b/Makefile	2016-01-27 22:47:09.795982961 +0100
@@ -0,0 +1,44 @@
+CC ?= gcc
+INSTALL_FILE = install -p -m 644
+INSTALL_DIR = install -p -d
+LN = ln -s
+RM = rm -fv
+CFLAGS += -c -O2 -Wall
+LDFLAGS +=
+
+SRC = src
+CFILES = $(SRC)/image_DXT.c $(SRC)/image_helper.c $(SRC)/SOIL.c $(SRC)/stb_image_aug.c
+OFILES = $(CFILES:.c=.o)
+LIBNAME = libSOIL
+VERSION = 1.07-20071110
+MAJOR = 1
+
+HFILES = $(SRC)/SOIL.h $(SRC)/image_DXT.h $(SRC)/image_helper.h \
+  $(SRC)/stbi_DDS_aug.h $(SRC)/stbi_DDS_aug_c.h $(SRC)/stb_image_aug.h
+AFILE = libSOIL.a
+INCLUDEDIR = /usr/include/SOIL
+LIBDIR = /usr/lib
+
+all: $(OFILES) lib
+
+%.o: %.c
+	$(CC) $(CFLAGS) $< -o $@
+
+lib: $(OFILES)
+	# create static library
+	$(AR) -cvq $(LIBNAME).a $(OFILES)
+
+install:
+	$(INSTALL_DIR) $(DESTDIR)/$(INCLUDEDIR)
+	$(INSTALL_FILE) $(HFILES) $(DESTDIR)/$(INCLUDEDIR)
+	$(INSTALL_DIR) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL_FILE) $(AFILE) $(DESTDIR)/$(LIBDIR)
+
+clean:
+	$(RM) *.o
+	$(RM) *~
+
+distclean:
+	$(RM) $(AFILE) $(SOFILE)
+
+.PHONY: all lib clean distclean
