commit 45b2e4ce57e345a9b9344d5b8d8f46155b872a27
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 19 21:48:49 2013 +0300

    vdr 2.1.2 buildfix

diff --git a/src/config/config.c b/src/config/config.c
index e058e96..1aaf9b8 100644
--- a/src/config/config.c
+++ b/src/config/config.c
@@ -45,7 +45,7 @@ cXVDRServerConfig::cXVDRServerConfig()
 }
 
 void cXVDRServerConfig::Load() {
-  cLiveQueue::SetTimeShiftDir(VideoDirectory);
+  cLiveQueue::SetTimeShiftDir(cVideoDirectory::Name());
   cRecordingsCache::GetInstance().LoadResumeData();
 
   if(!cConfig<cSetupLine>::Load(AddDirectory(ConfigDirectory, GENERAL_CONFIG_FILE), true, false))
diff --git a/src/xvdr/xvdrclient.c b/src/xvdr/xvdrclient.c
index a1a9460..e15dd8f 100644
--- a/src/xvdr/xvdrclient.c
+++ b/src/xvdr/xvdrclient.c
@@ -1385,7 +1385,7 @@ bool cXVDRClient::processTIMER_Update() /* OPCODE 85 */
 bool cXVDRClient::processRECORDINGS_GetDiskSpace() /* OPCODE 100 */
 {
   int FreeMB;
-  int Percent = VideoDiskSpace(&FreeMB);
+  int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB);
   int Total   = (FreeMB / (100 - Percent)) * 100;
 
   m_resp->put_U32(Total);
