From: Kornel <kornel@geekhood.net>
Date: Mon, 14 May 2018 10:37:34 +0100
Subject: [PATCH] Non-writable /tmp is a thing

---
 configure | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 7a609659..0a2b73e8 100755
--- a/configure
+++ b/configure
@@ -271,13 +271,15 @@ error() {
 echo
 
 # /tmp, because mingw has problems opening /dev/null and gives false negative
-if ! echo "int main(){}" | "$CC" -xc -std=c99 -o /tmp/gcccheck - > /dev/null; then
+if ! echo "int main(){}" | "$CC" -xc -std=c99 -o pngquant-gcccheck - > /dev/null; then
+    rm -f pngquant-gcccheck
     if [ "$SKIP_CC_CHECK" -eq 1 ]; then
         status "Compiler" "$CC failed to compile anything (make sure it's installed and supports C99)"
     else
         error "Compiler" "$CC failed to compile anything (make sure it's installed and supports C99)"
     fi
 fi
+rm -f pngquant-gcccheck
 
 status "Compiler" "$CC"
 
