diff -Naur linux-6ce48b3/include/linux/compiler-gcc.h linux-6ce48b3.patch/include/linux/compiler-gcc.h
--- linux-6ce48b3/include/linux/compiler-gcc.h	2016-09-08 12:53:14.157013900 +0200
+++ linux-6ce48b3.patch/include/linux/compiler-gcc.h	2016-09-08 12:56:03.187022963 +0200
@@ -50,9 +50,10 @@
  */
 #if !defined(CONFIG_ARCH_SUPPORTS_OPTIMIZED_INLINING) || \
     !defined(CONFIG_OPTIMIZE_INLINING) || (__GNUC__ < 4)
-# define inline		inline		__attribute__((always_inline))
-# define __inline__	__inline__	__attribute__((always_inline))
-# define __inline	__inline	__attribute__((always_inline))
+/* XXX: check __GNUC_STDC_INLINE__, fix line length */
+# define inline                inline          __attribute__((always_inline)) __attribute__((__gnu_inline__))
+# define __inline__    __inline__      __attribute__((always_inline)) __attribute__((__gnu_inline__))
+# define __inline      __inline        __attribute__((always_inline)) __attribute__((__gnu_inline__))
 #else
 /* A lot of inline functions can cause havoc with function tracing */
 # define inline		inline		notrace
