--- a/include/rtmp_def.h 2016-01-15 09:13:33.299185050 +0100
+++ b/include/rtmp_def.h 2016-01-15 09:15:02.146281249 +0100
@@ -1621,14 +1621,14 @@
 #define INF_MBSSID_DEV_NAME		"wlan"
 #else
 #if CONFIG_RT_FIRST_CARD == 7610
-#define INF_MAIN_DEV_NAME		"ra"
-#define INF_MBSSID_DEV_NAME		"ra"
+#define INF_MAIN_DEV_NAME		"wlan"
+#define INF_MBSSID_DEV_NAME		"wlan"
 #elif CONFIG_RT_SECOND_CARD == 7610
-#define INF_MAIN_DEV_NAME		"rai"
-#define INF_MBSSID_DEV_NAME		"rai"
+#define INF_MAIN_DEV_NAME		"wlani"
+#define INF_MBSSID_DEV_NAME		"wlani"
 #else
-#define INF_MAIN_DEV_NAME		"ra"
-#define INF_MBSSID_DEV_NAME		"ra"
+#define INF_MAIN_DEV_NAME		"wlan"
+#define INF_MBSSID_DEV_NAME		"wlan"
 #endif
 #endif /* ANDROID_SUPPORT */
 #if CONFIG_RT_FIRST_CARD == 7610
