diff -Naur a/include/configs/odroidc2.h b/include/configs/odroidc2.h
--- a/include/configs/odroidc2.h	2016-06-17 12:29:58.790889191 -0700
+++ b/include/configs/odroidc2.h	2016-06-17 12:31:40.319838401 -0700
@@ -83,20 +83,12 @@
 	"loadaddr=0x20000000\0"		\
 	"dtbaddr=0x1000000\0"		\
 	"fdt_high=0x20000000\0"		\
-	"hdmimode=1080p60hz\0"		\
-	"cecconfig=cec0xf\0"		\
-	"bootargs=root=/dev/mmcblk0p2 rw init=/init rootwait "	\
-		"console=ttyS0,115200 "				\
-		"hdmimode=1080p60hz hdmitx=cecf "		\
-		"logo=osd1,loaded,0x3f800000,1080p60hz "		\
-		"androidboot.hardware=odroidc2 androidboot.serialno=${fbt_id#} "	\
-		"androidboot.selinux=disabled  \0"		\
-	"bootcmd=cfgload; showlogo ${hdmimode}; movi read dtb 0 ${dtbaddr}; movi read boot 0 ${loadaddr}; booti ${loadaddr} - ${dtbaddr}\0"
+	"bootcmd=cfgload\0"
 
 #define CONFIG_PREBOOT
 #define CONFIG_BOOTCOMMAND
 
-//#define CONFIG_ENV_IS_NOWHERE  1
+#define CONFIG_ENV_IS_NOWHERE  1
 #define CONFIG_ENV_SIZE			(32 * SZ_1K)	/* 32kB */
 #define CONFIG_ENV_OFFSET		(720 * SZ_1K)	/* FIXME: should be
 							   close to U-boot image
@@ -176,7 +168,7 @@
 #define CONFIG_LZO			1
 
 /* storage: emmc/nand/sd */
 #define CONFIG_ENV_OVERWRITE
-#define CONFIG_ENV_IS_IN_MMC		1
+#define CONFIG_ENV_IS_IN_MMC		0
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #define CONFIG_AML_SD_EMMC		1
 
