diff --git a/kernel/dvb_hdhomerun_init.c b/kernel/dvb_hdhomerun_init.c
index d02a322..bd97d5e 100644
--- a/kernel/dvb_hdhomerun_init.c
+++ b/kernel/dvb_hdhomerun_init.c
@@ -143,7 +143,7 @@ static int dvb_hdhomerun_stop_feed(struct dvb_demux_feed *feed)
 	return ret;
 }
 
-static int __devinit dvb_hdhomerun_register(struct dvb_hdhomerun *hdhomerun)
+static int  dvb_hdhomerun_register(struct dvb_hdhomerun *hdhomerun)
 {
 	struct dvb_adapter *dvb_adapter;
 	struct dvb_demux *dvbdemux;
@@ -284,7 +284,7 @@ static void dvb_hdhomerun_unregister(struct dvb_hdhomerun *hdhomerun)
 }
 
 
-static int __devinit dvb_hdhomerun_probe(struct platform_device *plat_dev)
+static int dvb_hdhomerun_probe(struct platform_device *plat_dev)
 {
 	int ret;
 	struct dvb_hdhomerun *hdhomerun;
