diff -uNr dvbhdhomerun-0.0.15-orig/userhdhomerun/hdhomerun_tuner.cpp dvbhdhomerun-0.0.15/userhdhomerun/hdhomerun_tuner.cpp
--- dvbhdhomerun-0.0.15-orig/userhdhomerun/hdhomerun_tuner.cpp	2013-02-17 22:37:34.000000000 +0100
+++ dvbhdhomerun-0.0.15/userhdhomerun/hdhomerun_tuner.cpp	2013-03-02 10:25:15.000000000 +0100
@@ -97,12 +97,28 @@
          string type(tmp);
          LOG() << "Type of device: " << type << endl;
          if(type == "hdhomerun_dvbt") {
-            LOG() << "Notice, setting to DVB-C!! Use /etc/dvbhdhomerun to change that." << endl;
-            m_type = HdhomerunTuner::DVBC;
+            m_type = HdhomerunTuner::DVBT;
          }
          else if(type == "hdhomerun_atsc") {
             m_type = HdhomerunTuner::ATSC;
          }
+         else if(type == "hdhomerun3_dvbt") {
+            m_type = HdhomerunTuner::DVBT;
+         }        
+         else if(type.find("dvbt") != string::npos) {
+            m_type = HdhomerunTuner::DVBT;
+         }
+         else if(type.find("dvbc") != string::npos) {
+            m_type = HdhomerunTuner::DVBC;
+         }
+         else if(type.find("atsc") != string::npos) {
+            m_type = HdhomerunTuner::ATSC;
+         }
+
+         if (m_type != HdhomerunTuner::NOT_SET) {
+            LOG() << "Auto detecting tuner type set to \"" << type 
+                  << "\" based on auto detecting" << endl;
+         }
       }
       else {
          ERR() << "get_model_str from HDHomeRun failed!" << endl;
