diff -Naur dvbhdhomerun-20120831/kernel/Makefile dvbhdhomerun-20120831.patch/kernel/Makefile
--- dvbhdhomerun-20120831/kernel/Makefile	2012-08-31 14:03:28.000000000 +0200
+++ dvbhdhomerun-20120831.patch/kernel/Makefile	2013-01-17 02:37:39.115879705 +0100
@@ -3,7 +3,7 @@
 
 PWD		:= $(shell pwd)
 
-EXTRA_CFLAGS += -Idrivers/media/dvb/dvb-core/ -Idrivers/media/dvb/frontends
+EXTRA_CFLAGS += -Idrivers/media/dvb/dvb-core/ -Idrivers/media/dvb-core/ -Idrivers/media/dvb/frontends
 
 dvb_hdhomerun-objs := dvb_hdhomerun_init.o dvb_hdhomerun_data.o dvb_hdhomerun_control.o
 obj-m += dvb_hdhomerun.o
diff -Naur dvbhdhomerun-20120831/v4l-dvb/hdhomerun/Makefile dvbhdhomerun-20120831.patch/v4l-dvb/hdhomerun/Makefile
--- dvbhdhomerun-20120831/v4l-dvb/hdhomerun/Makefile	2012-08-31 14:03:28.000000000 +0200
+++ dvbhdhomerun-20120831.patch/v4l-dvb/hdhomerun/Makefile	2013-01-17 02:38:01.408751902 +0100
@@ -3,4 +3,4 @@
 
 obj-$(CONFIG_DVB_HDHOMERUN) += dvb_hdhomerun_core.o
 
-EXTRA_CFLAGS += -Idrivers/media/dvb/dvb-core/ -Idrivers/media/dvb/frontends
+EXTRA_CFLAGS += -Idrivers/media/dvb/dvb-core/ -Idrivers/media/dvb-core/ -Idrivers/media/dvb/frontends
