diff -Naur megaglest-source-3.3.5/Jamfile megaglest-source-3.3.5.patch/Jamfile
--- megaglest-source-3.3.5/Jamfile	2010-06-26 01:56:31.000000000 +0200
+++ megaglest-source-3.3.5.patch/Jamfile	2010-07-15 07:10:13.563768426 +0200
@@ -83,50 +83,50 @@
 ExternalLibs glest.bin : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL LUA JPEG PNG CURL X11  ;
 IncludeDir glest.bin : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_DIRS) ;
 
-#### Editor ####
-if $(WX_AVAILABLE) = "yes" {
-  SubDir TOP glest_map_editor ;
-
-  GLEST_MAP_DIRS = . ;
-  for i in $(GLEST_DIRS) {
-    GLEST_MAP_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
-  }
-
-  Application glest_editor : $(GLEST_MAP_SOURCES) ;
-  LinkWith glest_editor : glestlib strefloplib ;
-  ExternalLibs glest_editor : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL WX CURL ;
-  IncludeDir glest_editor : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_MAP_DIRS) ;
-}
-
-### Viewer ###
-if $(WX_AVAILABLE) = "yes" {
-  SubDir TOP g3d_viewer ;
-
-  GLEST_VIEWER_DIRS = 
-    . 
-
-;
-
-  for i in $(GLEST_VIEWER_DIRS) {
-    GLEST_VIEWER_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
-  }
-
-  Application glest_g3dviewer : $(GLEST_VIEWER_SOURCES) ../glest_game/graphics/unit_particle_type.cpp ../glest_game/graphics/particle_type.cpp ;
-  LinkWith glest_g3dviewer : glestlib strefloplib ;
-  ExternalLibs glest_g3dviewer : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL LUA JPEG PNG CURL WX X11  ;
-  IncludeDir glest_g3dviewer : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_VIEWER_DIRS)  ../glest_game/graphics ../glest_game/global ../glest_game/sound ../glest_game/game ;
-}
-
-### Configurator ###
-  SubDir TOP configurator ;
-
-  GLEST_CONFIG_DIRS = . ;
-  for i in $(GLEST_DIRS) {
-    GLEST_CONFIG_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
-  }
-
-  Application glest_configurator : $(GLEST_CONFIG_SOURCES) ;
-  LinkWith glest_configurator : glestlib strefloplib ;
-  ExternalLibs glest_configurator : SDL GL GLU XERCES WX CURL ;
-  IncludeDir glest_configurator : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_CONFIG_DIRS) ;
+# #### Editor ####
+# if $(WX_AVAILABLE) = "yes" {
+#   SubDir TOP glest_map_editor ;
+# 
+#   GLEST_MAP_DIRS = . ;
+#   for i in $(GLEST_DIRS) {
+#     GLEST_MAP_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
+#   }
+# 
+#   Application glest_editor : $(GLEST_MAP_SOURCES) ;
+#   LinkWith glest_editor : glestlib strefloplib ;
+#   ExternalLibs glest_editor : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL WX CURL ;
+#   IncludeDir glest_editor : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_MAP_DIRS) ;
+# }
+
+# ### Viewer ###
+# if $(WX_AVAILABLE) = "yes" {
+#   SubDir TOP g3d_viewer ;
+# 
+#   GLEST_VIEWER_DIRS = 
+#     . 
+# 
+# ;
+# 
+#   for i in $(GLEST_VIEWER_DIRS) {
+#     GLEST_VIEWER_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
+#   }
+# 
+#   Application glest_g3dviewer : $(GLEST_VIEWER_SOURCES) ../glest_game/graphics/unit_particle_type.cpp ../glest_game/graphics/particle_type.cpp ;
+#   LinkWith glest_g3dviewer : glestlib strefloplib ;
+#   ExternalLibs glest_g3dviewer : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL LUA JPEG PNG CURL WX X11  ;
+#   IncludeDir glest_g3dviewer : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_VIEWER_DIRS)  ../glest_game/graphics ../glest_game/global ../glest_game/sound ../glest_game/game ;
+# }
+
+# ### Configurator ###
+#   SubDir TOP configurator ;
+# 
+#   GLEST_CONFIG_DIRS = . ;
+#   for i in $(GLEST_DIRS) {
+#     GLEST_CONFIG_SOURCES += [ Wildcard $(i) : *.cpp *.h ] ;
+#   }
+# 
+#   Application glest_configurator : $(GLEST_CONFIG_SOURCES) ;
+#   LinkWith glest_configurator : glestlib strefloplib ;
+#   ExternalLibs glest_configurator : SDL GL GLU XERCES WX CURL ;
+#   IncludeDir glest_configurator : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_CONFIG_DIRS) ;
 
