diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_0.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_0.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_0.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_0.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_0[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"                ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n1.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_n1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXX   X XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...       ......",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n2.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_n2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"  XXXXXXXXX     ",
+".. XXXXXXX .....",
+"...       ......",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_n3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"  XXXXXXXXX     ",
+".. XXXXXXX .....",
+"... XXXXX ......",
+"....     .......",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_n4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+" XXXXXXXXXXX    ",
+". XXXXXXXXX ....",
+".. XXXXXXX .....",
+"... XXXXX ......",
+"....     .......",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_n5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_n5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_n5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+" XXXXXXXXXXX    ",
+". XXXXXXXXX ....",
+".. XXXXXXX .....",
+"... XXXXX ......",
+".... XXX .......",
+".....   ........",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p1.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_p1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXX       XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p2.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_p2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXX       XXXXXX",
+"XX ....... XXXXX",
+"  .........     ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_p3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXX     XXXXXXX",
+"XXX ..... XXXXXX",
+"XX ....... XXXXX",
+"  .........     ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_p4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXX     XXXXX X",
+"XXX ..... XXXXXX",
+"XX ....... XXXXX",
+"X ......... XXXX",
+" ...........    ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_gradient_p5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_gradient_p5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_gradient_p5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #FF0000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXXXX   XXXXXX X",
+"XXXX ... XX    X",
+"XXX ..... XXXXXX",
+"XX ....... XXXXX",
+"X ......... XXXX",
+" ...........    ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_0.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_0.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_0.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_0.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_0[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"                ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n1.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_n1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXX   X XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...       ......",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n2.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_n2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...XXXXXXX......",
+"...       ......",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_n3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...       ......",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_n4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...       ......",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_n5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_n5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_n5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXX   X    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"   XXXXXXX      ",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...XXXXXXX......",
+"...       ......",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p1.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_p1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXX XX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXX       XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p2.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_p2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXXXX",
+"XXX       XXXXXX",
+"XXX.......XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_p3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXXXXXXXXXX    X",
+"XXX       XXXXXX",
+"XXX.......XXXXXX",
+"XXX.......XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_p4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX XX X",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXXXXX X",
+"XXX       XXXX X",
+"XXX.......XXXXXX",
+"XXX.......XXXXXX",
+"XXX.......XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_height_p5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_height_p5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_height_p5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #0000FF",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXX    X",
+"XXXXXXXXXXX XXXX",
+"XXXXXXXXXXX    X",
+"XXX       XXXX X",
+"XXX.......X    X",
+"XXX.......XXXXXX",
+"XXX.......XXXXXX",
+"XXX.......XXXXXX",
+"   .......      ",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_none.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_none.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_none.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_none.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_none[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #FF0000",
+". c #008000",
+"X c #000080",
+"o c None",
+/* pixels */
+"oo  oooXXooo  oo",
+"oo   oXXXXo   oo",
+"ooo   XXXX   ooo",
+"oooo   XX   oooo",
+"ooooo      ooooo",
+"oooooo    oooooo",
+"oooooo    oooooo",
+"ooooo      ooooo",
+"oooo   XX   oooo",
+"ooo   XXXX   ooo",
+"oo   oXXXXo   oo",
+"oo  ooXXXXoo  oo",
+"ooooo......ooooo",
+"................",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_bush.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_bush.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_bush.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_bush.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,27 @@
+/* XPM */
+static const char *brush_object_bush[] = {
+"20 20 4 1",
+" 	c None",
+".	c #0000FF",
+"+	c #008000",
+"@	c #00FF00",
+"....................",
+"....................",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..      +@+ +     ..",
+"..     +@+@+@@    ..",
+"..     @+@+@+@    ..",
+"..    +@ + + ++   ..",
+"..    @+@+@++ +   ..",
+"..    +@+ +@@+    ..",
+"..     + +@++@    ..",
+".. ++++++++++++++ ..",
+"..                ..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c1-bigtree.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c1-bigtree.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c1-bigtree.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c1-bigtree.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,28 @@
+/* XPM */
+static const char *brush_object_c1_bigtree[] = {
+"20 20 5 1",
+" 	c None",
+".	c #FFCC7F",
+"+	c #808000",
+"@	c #008000",
+"#	c #800000",
+"....................",
+"....................",
+"..      ++@+      ..",
+"..    @+++++@+    ..",
+"..   +++@+@+++@   ..",
+"..   +@++@++@++   ..",
+"..  ++++++++++@+  ..",
+"..  +@@+@++@@@+@  ..",
+"..   ++++@+++++   ..",
+"..   ++@+++@@+@   ..",
+"..    ++++@+++    ..",
+"..      @+++      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..     ######     ..",
+"..    #########   ..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c2-hanged.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c2-hanged.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c2-hanged.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c2-hanged.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,31 @@
+/* XPM */
+static const char *brush_object_c2_hanged[] = {
+"20 20 8 1",
+" 	c None",
+".	c #00FFFF",
+"+	c #800000",
+"@	c #C0C0C0",
+"#	c #808080",
+"$	c #FFFF00",
+"%	c #0000FF",
+"&	c #008000",
+"....................",
+"....................",
+"..       +        ..",
+"..   +++++++++    ..",
+"..   @ + + + @    ..",
+"..  ##  +++  @    ..",
+"..  @@@  +  ###   ..",
+".. @#@#@ + # @ #  ..",
+".. @@#@@ + #+@ #  ..",
+".. $@@@$ + #+@+#  ..",
+"..  %$%  + #+@+#  ..",
+"..  %$%  + # @+#  ..",
+"..  @@@  + ###+#+ ..",
+"..  @ @  +    + + ..",
+"..       +        ..",
+"..       +        ..",
+"..      &+#       ..",
+"..&&&&&&&&&&&&&&&&..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c3-statue.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c3-statue.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c3-statue.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c3-statue.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,29 @@
+/* XPM */
+static const char *brush_object_c3_statue[] = {
+"20 20 6 1",
+" 	c None",
+".	c #B2194C",
+"+	c #C0C0C0",
+"@	c #808080",
+"#	c #808000",
+"$	c #008000",
+"....................",
+"....................",
+"..                ..",
+"..     +          ..",
+"..      +@@       ..",
+"..       +@       ..",
+"..     ###+@      ..",
+"..     #+#@+@     ..",
+"..     ###@       ..",
+"..      @@@       ..",
+"..      @@@       ..",
+"..      @ @       ..",
+"..      @  @      ..",
+"..      @  @      ..",
+"..     ++++++     ..",
+"..      @@@@      ..",
+"..      @@@@      ..",
+"..$$$$$$$$$$$$$$$$..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c4-bigrock.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c4-bigrock.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c4-bigrock.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c4-bigrock.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,28 @@
+/* XPM */
+static const char *brush_object_c4_bigrock[] = {
+"20 20 5 1",
+" 	c None",
+".	c #7fff19",
+"+	c #808080",
+"@	c #C0C0C0",
+"#	c #008000",
+"....................",
+"....................",
+"..                ..",
+"..                ..",
+"..                ..",
+"..     +@         ..",
+"..    @+++   +@   ..",
+"..    +++@###++   ..",
+"..    +@+@###++   ..",
+"..   @++++@###+   ..",
+"..   ++@+@+###+@  ..",
+"..   ++##+++##++  ..",
+"..   +@##++++@++  ..",
+"..  @++##+@++@+@  ..",
+"..  ++++#++###++  ..",
+"..  +@++++++####  ..",
+"..  +++++###++++  ..",
+"..################..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c5-blocking.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c5-blocking.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_c5-blocking.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_c5-blocking.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,27 @@
+/* XPM */
+static const char *brush_object_c5_blocking[] = {
+"20 20 4 1",
+" 	c None",
+".	c #FF33CC",
+"+	c #C0C0C0",
+"@	c #008000",
+"....................",
+"....................",
+"..                ..",
+"..  ++++++++++++  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..  +          +  ..",
+"..@@@@@@@@@@@@@@@@..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_dead-tree.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_dead-tree.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_dead-tree.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_dead-tree.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_object_dead_tree[] = {
+"20 20 3 1",
+" 	c None",
+".	c #FFFFFF",
+"+	c #800000",
+"....................",
+"....................",
+"..                ..",
+"..  +  +          ..",
+"..   + +   + +    ..",
+"..    +     ++    ..",
+"..  ++++  +  + +  ..",
+"..     +++   ++   ..",
+"..   +  ++  +++   ..",
+"..   +  ++++   +  ..",
+"..  +++  +++      ..",
+"..     +++++  ++  ..",
+"..      ++++++    ..",
+"..      ++++      ..",
+"..      ++++      ..",
+"..      ++++      ..",
+"..     ++++++     ..",
+"..    +++++++++   ..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_stone.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_stone.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_stone.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_stone.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,28 @@
+/* XPM */
+static const char *brush_object_stone[] = {
+"20 20 5 1",
+" 	c None",
+".	c #7F7FFF",
+"+	c #808080",
+"@	c #C0C0C0",
+"#	c #008000",
+"....................",
+"....................",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..    +@          ..",
+"..   +++@++       ..",
+"..   @++++@+@     ..",
+"..  +++++++++@    ..",
+"..  +++@+++++@@   ..",
+"..   ++++++@++@@  ..",
+"..   +@++++++++@  ..",
+".. ############## ..",
+"..                ..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_tree.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_tree.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_tree.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_tree.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,28 @@
+/* XPM */
+static const char * brush_object_tree[] = {
+"20 20 5 1",
+" 	c None",
+".	c #FF0000",
+"+	c #008000",
+"@	c #00FF00",
+"#	c #800000",
+"....................",
+"....................",
+"..       ++       ..",
+"..     ++++++     ..",
+"..    ++++@+++    ..",
+"..    ++++++++    ..",
+"..   +@++@@++@+   ..",
+"..   ++++++@+++   ..",
+"..    +++@++++    ..",
+"..    ++++++++    ..",
+"..     ++++++     ..",
+"..      #++#      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..      ####      ..",
+"..     ######     ..",
+"..    #########   ..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_object_water-object.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_water-object.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_object_water-object.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_object_water-object.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,28 @@
+/* XPM */
+static const char *brush_object_water_object[] = {
+"20 20 5 1",
+" 	c None",
+".	c #7F7F7F",
+"+	c #008000",
+"@	c #00FFFF",
+"#	c #0000FF",
+"....................",
+"....................",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..                ..",
+"..       +        ..",
+"..       +   +    ..",
+"..     + + + +    ..",
+"..     + ++ +     ..",
+"..     +  + +     ..",
+"..      + ++      ..",
+"..@@@@@@@++@@@@@@@..",
+"..@#@@@#@@@#@@@#@@..",
+"..#@#@#@#@#@#@#@#@..",
+"..################..",
+"....................",
+"...................."};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_blue.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_blue.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_blue.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_blue.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_blue[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #000080",
+"X c #0000FF",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"oo XX.XXXX.XX oo",
+"oo X...XX...X oo",
+"oo XX.XXXX.XX oo",
+"oo XXXXXXXXXX oo",
+"oo XXXXXXXXXX oo",
+"oo X........X oo",
+"oo XXXXXXXXXX oo",
+"oo XXXXXXXXXX oo",
+"oo XXXX..XXXX oo",
+"ooo XX....XX ooo",
+"ooo X......X ooo",
+"oooo  ....  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_cyan.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_cyan.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_cyan.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_cyan.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_cyan[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #008080",
+"X c #00FFFF",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo .......... oo",
+"oo ....XX.... oo",
+"oo ...XXXX... oo",
+"oo ...XXXX... oo",
+"oo ....XX.... oo",
+"oo .......... oo",
+"oo .XX....XX. oo",
+"oo ..XXXXXX.. oo",
+"oo ...XXXX... oo",
+"oo .......... oo",
+"oo .......... oo",
+"ooo ........ ooo",
+"ooo ........ ooo",
+"oooo  ....  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_green.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_green.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_green.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_green.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_green[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #008000",
+"X c #00FF00",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .X.X..X.X. oo",
+"oo .......... oo",
+"oo .XXX..XXX. oo",
+"oo ...X..X... oo",
+"ooo ..XXXX.. ooo",
+"ooo ........ ooo",
+"oooo  ....  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_orange.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_orange.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_orange.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_orange.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_orange[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #804000",
+"X c #FF8000",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo XXXX.X.XXX oo",
+"oo XXX.X.XXXX oo",
+"oo XX.X.XXXXX oo",
+"oo X.X.XXXXXX oo",
+"oo .X.XXXXXXX oo",
+"oo X.XXXXX..X oo",
+"oo .XXXXX...X oo",
+"oo XXXXX...XX oo",
+"oo XXX....XXX oo",
+"oo XXXX..XXXX oo",
+"oo XXX.X.XXXX oo",
+"ooo XXXXXXXX ooo",
+"ooo XXXXXXXX ooo",
+"oooo  XXXX  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_pink.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_pink.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_pink.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_pink.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_pink[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #800080",
+"X c #FF00FF",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo ..X....X.. oo",
+"oo X...XX...X oo",
+"oo XX.XXXX.XX oo",
+"oo X...XX...X oo",
+"oo ..X....X.. oo",
+"oo X...XX...X oo",
+"oo XX.XXXX.XX oo",
+"oo X...XX...X oo",
+"oo ..X....X.. oo",
+"oo X...XX...X oo",
+"oo XX.XXXX.XX oo",
+"ooo ...XX... ooo",
+"ooo .X....X. ooo",
+"oooo  .XX.  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_player.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_player.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_player.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_player.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_players_player[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #FFFF00",
+". c #008080",
+"X c None",
+/* pixels */
+"XXXXX......XXXXX",
+"XXX..........XXX",
+"XX............XX",
+"X..............X",
+"X....      ....X",
+"......  ..  ....",
+"......  ..  ....",
+"......  ..  ....",
+"......     .....",
+"......  ........",
+"......  ........",
+"X.....  .......X",
+"X....    ......X",
+"XX............XX",
+"XXX..........XXX",
+"XXXXX......XXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_red.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_red.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_red.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_red.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_red[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #800000",
+"X c #FF0000",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo .......... oo",
+"oo .......... oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"oo .......... oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"oo XXXXXXXXXX oo",
+"oo .......... oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"ooo ........ ooo",
+"ooo ........ ooo",
+"oooo  ....  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_white.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_white.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_white.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_white.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_white[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo XXXXXXXXXX oo",
+"oo ....XX.... oo",
+"oo X.X.XX.X.X oo",
+"oo ....XX.... oo",
+"oo XXXXXXXXXX oo",
+"oo XX..XX..XX oo",
+"oo XXXXXXXXXX oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"oo .......... oo",
+"oo XXXXXXXXXX oo",
+"ooo XXXXXXXX ooo",
+"ooo XXXXXXXX ooo",
+"oooo  XXXX  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_players_yellow.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_yellow.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_players_yellow.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_players_yellow.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *brush_players_yellow[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808000",
+"X c #FFFF00",
+"o c None",
+/* pixels */
+"oo            oo",
+"oo .........X oo",
+"oo .XXX....X. oo",
+"oo .XXX...X.. oo",
+"oo .XXX..X..X oo",
+"oo .....X..X. oo",
+"oo ....X..X.. oo",
+"oo ...X..X..X oo",
+"oo ..X..X..X. oo",
+"oo .X..X..X.. oo",
+"oo X..X..X... oo",
+"oo ..X..X.... oo",
+"ooo X..X.... ooo",
+"ooo ..X..... ooo",
+"oooo  ....  oooo",
+"oooooo    oooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_1-gold.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_1-gold.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_1-gold.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_1-gold.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *brush_resource_1_gold[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #dddd00",
+". c None",
+/* pixels */
+"................",
+".  ..........  .",
+".   ........   .",
+"..   ......   ..",
+"...   ....   ...",
+"....   ..   ....",
+".....      .....",
+"......    ......",
+"......    ......",
+".....      .....",
+"....   ..   ....",
+"...   ....   ...",
+"..   ......   ..",
+".   ........   .",
+".  ..........  .",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_2-stone.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_2-stone.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_2-stone.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_2-stone.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *brush_resource_2_stone[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #808080",
+". c None",
+/* pixels */
+"................",
+".  ..........  .",
+".   ........   .",
+"..   ......   ..",
+"...   ....   ...",
+"....   ..   ....",
+".....      .....",
+"......    ......",
+"......    ......",
+".....      .....",
+"....   ..   ....",
+"...   ....   ...",
+"..   ......   ..",
+".   ........   .",
+".  ..........  .",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *brush_resource_3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #FF0000",
+". c None",
+/* pixels */
+"................",
+".  ..........  .",
+".   ........   .",
+"..   ......   ..",
+"...   ....   ...",
+"....   ..   ....",
+".....      .....",
+"......    ......",
+"......    ......",
+".....      .....",
+"....   ..   ....",
+"...   ....   ...",
+"..   ......   ..",
+".   ........   .",
+".  ..........  .",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *brush_resource_4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #0000FF",
+". c None",
+/* pixels */
+"................",
+".  ..........  .",
+".   ........   .",
+"..   ......   ..",
+"...   ....   ...",
+"....   ..   ....",
+".....      .....",
+"......    ......",
+"......    ......",
+".....      .....",
+"....   ..   ....",
+"...   ....   ...",
+"..   ......   ..",
+".   ........   .",
+".  ..........  .",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_resource_5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_resource_5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *brush_resource_5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #008080",
+". c None",
+/* pixels */
+"................",
+".  ..........  .",
+".   ........   .",
+"..   ......   ..",
+"...   ....   ...",
+"....   ..   ....",
+".....      .....",
+"......    ......",
+"......    ......",
+".....      .....",
+"....   ..   ....",
+"...   ....   ...",
+"..   ......   ..",
+".   ........   .",
+".  ..........  .",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_custom.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_custom.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_custom.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_custom.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_surface_custom[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #7c4645",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"                ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+"                ",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_grass1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_grass1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_grass1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_grass1.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_surface_grass1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #00a000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"                ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+"                ",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_grass2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_grass2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_grass2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_grass2.xpm	2010-05-05 00:55:22.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_surface_grass2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #536e0b",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"                ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+"                ",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_road.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_road.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_road.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_road.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_surface_road[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #800000",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"                ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+"                ",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_stone.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_stone.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/brush_surface_stone.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/brush_surface_stone.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *brush_surface_stone[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #808080",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"                ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+" .............. ",
+"                ",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/edit_randomize-heights.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_randomize-heights.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/edit_randomize-heights.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_randomize-heights.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *edit_randomize_heights[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #008000",
+". c None",
+/* pixels */
+"................",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................",
+"................",
+"...... .....   .",
+"...      .     .",
+"                ",
+"                ",
+"                ",
+"                ",
+"                ",
+"                "
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/edit_randomize.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_randomize.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/edit_randomize.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_randomize.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *edit_randomize[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #008000",
+". c #00FF00",
+"X c #00FFFF",
+/* pixels */
+"                ",
+" XX    XX       ",
+"  XX  XX     .  ",
+"   XXXX         ",
+"    XX          ",
+"   XXXX        .",
+"  XX  XX       .",
+" XX    XX     ..",
+"              . ",
+"               .",
+"           .    ",
+"          ... . ",
+"           .    ",
+"               .",
+"    .   ....    ",
+"       ......   "
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/edit_redo.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_redo.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/edit_redo.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_redo.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *edit_redo[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #000000",
+". c None",
+/* pixels */
+"................",
+"........   .....",
+".........   ....",
+"..........   ...",
+"...........   ..",
+"....           .",
+"...            .",
+"..   ......   ..",
+".   ......   ...",
+".  ......   ....",
+".  .....   .....",
+".  .............",
+".  .............",
+".. .............",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/edit_switch-surfaces.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_switch-surfaces.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/edit_switch-surfaces.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_switch-surfaces.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *edit_switch_surfaces[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #008000",
+". c #00FF00",
+/* pixels */
+"........        ",
+"........        ",
+"........        ",
+"........        ",
+"........        ",
+"..... ..  .     ",
+"....  ..  ..    ",
+"...     .....   ",
+"...     .....   ",
+"....  ..  ..    ",
+"..... ..  .     ",
+"........        ",
+"........        ",
+"........        ",
+"........        ",
+"........        "
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/edit_undo.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_undo.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/edit_undo.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/edit_undo.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,24 @@
+/* XPM */
+static const char *edit_undo[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 2 1",
+"  c #000000",
+". c None",
+/* pixels */
+"................",
+".....   ........",
+"....   .........",
+"...   ..........",
+"..   ...........",
+".           ....",
+".            ...",
+"..   ......   ..",
+"...   ......   .",
+"....   ......  .",
+".....   .....  .",
+".............  .",
+".............  .",
+"............. ..",
+"................",
+"................"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_1.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_1.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_1.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_1.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *radius_1[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #C0C0C0",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXX..XXXXXXX",
+"XXXXXX.  .XXXXXX",
+"XXXXXX.  .XXXXXX",
+"XXXXXXX..XXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_2.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_2.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_2.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_2.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_2[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooXXooooooo",
+"oooooo.  .oooooo",
+"oooooX    Xooooo",
+"oooooX    Xooooo",
+"oooooo.  .oooooo",
+"oooooooXXooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_3.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_3.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_3.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_3.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *radius_3[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #808080",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXX.  .XXXXXX",
+"XXXXX.    .XXXXX",
+"XXXXX      XXXXX",
+"XXXXX      XXXXX",
+"XXXXX.    .XXXXX",
+"XXXXXX.  .XXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX",
+"XXXXXXXXXXXXXXXX"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_4.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_4.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_4.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_4.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_4[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"ooooooX..Xoooooo",
+"ooooo.    .ooooo",
+"ooooX      Xoooo",
+"oooo.      .oooo",
+"oooo.      .oooo",
+"ooooX      Xoooo",
+"ooooo.    .ooooo",
+"ooooooX..Xoooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_5.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_5.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_5.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_5.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_5[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooXXooooooo",
+"oooooX.  .Xooooo",
+"ooooX      Xoooo",
+"oooo.      .oooo",
+"oooX        Xooo",
+"oooX        Xooo",
+"oooo.      .oooo",
+"ooooX      Xoooo",
+"oooooX.  .Xooooo",
+"oooooooXXooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_6.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_6.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_6.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_6.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_6[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"ooooooX..Xoooooo",
+"ooooo.    .ooooo",
+"oooo.      .oooo",
+"oooX        Xooo",
+"ooo.        .ooo",
+"ooo.        .ooo",
+"oooX        Xooo",
+"oooo.      .oooo",
+"ooooo.    .ooooo",
+"ooooooX..Xoooooo",
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_7.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_7.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_7.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_7.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_7[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooooXXooooooo",
+"ooooo.    .ooooo",
+"oooo        oooo",
+"ooo.        .ooo",
+"ooo          ooo",
+"ooX          Xoo",
+"ooX          Xoo",
+"ooo          ooo",
+"ooo.        .ooo",
+"oooo        oooo",
+"ooooo.    .ooooo",
+"oooooooXXooooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_8.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_8.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_8.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_8.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,26 @@
+/* XPM */
+static const char *radius_8[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 4 1",
+"  c #000000",
+". c #808080",
+"X c #C0C0C0",
+"o c None",
+/* pixels */
+"oooooooooooooooo",
+"oooooooooooooooo",
+"oooooo.  .oooooo",
+"ooooX      Xoooo",
+"oooX        Xooo",
+"ooo          ooo",
+"oo.          .oo",
+"oo            oo",
+"oo            oo",
+"oo.          .oo",
+"ooo          ooo",
+"oooX        Xooo",
+"ooooX      Xoooo",
+"oooooo.  .oooooo",
+"oooooooooooooooo",
+"oooooooooooooooo"
+};
diff -Naur megaglest-source-3.3.5/glest_map_editor/icons/radius_9.xpm megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_9.xpm
--- megaglest-source-3.3.5/glest_map_editor/icons/radius_9.xpm	1970-01-01 01:00:00.000000000 +0100
+++ megaglest-source-3.3.5.patch/glest_map_editor/icons/radius_9.xpm	2010-05-04 02:22:11.000000000 +0200
@@ -0,0 +1,25 @@
+/* XPM */
+static const char *radius_9[] = {
+/* columns rows colors chars-per-pixel */
+"16 16 3 1",
+"  c #000000",
+". c #808080",
+"X c None",
+/* pixels */
+"XXXXXXXXXXXXXXXX",
+"XXXXXXX..XXXXXXX",
+"XXXXX.    .XXXXX",
+"XXXX        XXXX",
+"XXX          XXX",
+"XX.          .XX",
+"XX            XX",
+"X.            .X",
+"X.            .X",
+"XX            XX",
+"XX.          .XX",
+"XXX          XXX",
+"XXXX        XXXX",
+"XXXXX.    .XXXXX",
+"XXXXXXX..XXXXXXX",
+"XXXXXXXXXXXXXXXX"
+};
