diff -Naur htop-0.8.3/configure.ac htop-0.8.3.patch/configure.ac
--- htop-0.8.3/configure.ac	2009-06-23 15:53:18.000000000 +0200
+++ htop-0.8.3.patch/configure.ac	2010-08-31 15:26:52.545129346 +0200
@@ -94,9 +94,6 @@
    AC_CHECK_HEADERS([curses.h],[:],[missing_headers="$missing_headers $ac_header"])
 fi
 
-AC_CHECK_FILE($PROCDIR/stat,,AC_MSG_ERROR(Cannot find /proc/stat. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
-AC_CHECK_FILE($PROCDIR/meminfo,,AC_MSG_ERROR(Cannot find /proc/meminfo. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
-
 PLPA_INCLUDED(plpa-1.1)
 PLPA_INIT(plpa_happy=yes, plpa_happy=no)
 AM_CONDITIONAL([HAVE_PLPA], [test "$plpa_happy" = "yes"])
