diff -Naur Python-2.7.10.orig/configure Python-2.7.10/configure
--- Python-2.7.10.orig/configure	2015-08-18 12:24:03.016649265 -0700
+++ Python-2.7.10/configure	2015-08-18 12:23:41.716695873 -0700
@@ -14215,7 +14215,7 @@
 
 # first curses configure check
 ac_save_cppflags="$CPPFLAGS"
-CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+CPPFLAGS="$CPPFLAGS -I${prefix}/usr/include/ncursesw"
 
 for ac_header in curses.h ncurses.h
 do :
