diff -ur populatefs-1.0.orig/src/mod_path.c populatefs-1.0/src/mod_path.c
--- populatefs-1.0.orig/src/mod_path.c	2019-08-14 00:14:18.411252751 +0200
+++ populatefs-1.0/src/mod_path.c	2019-08-14 00:14:03.978642084 +0200
@@ -3,6 +3,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <unistd.h>
 #include <dirent.h>
 #include <ext2fs/ext2fs.h>
