From: https://github.com/crazycat69/linux_media/commit/b9a64b0bf203030b3203916789ca4dfe55825edc
From: CrazyCat <crazycat69@narod.ru>
Date: Mon, 18 Jul 2016 16:33:31 +0300
Subject: [PATCH] Fixes in si2168 configuration.

--- a/drivers/media/usb/dvb-usb/cxusb.c
+++ b/drivers/media/usb/dvb-usb/cxusb.c
@@ -1386,6 +1386,7 @@ static int cxusb_mygica_t230_frontend_attach(struct dvb_usb_adapter *adap)
 		usb_rcvbulkpipe(d->udev, d->props.adapter[0].fe[0].stream.endpoint));
 
 	/* attach frontend */
+	memset(&si2168_config, 0, sizeof(si2168_config));
 	si2168_config.i2c_adapter = &adapter;
 	si2168_config.fe = &adap->fe_adap[0].fe;
 	si2168_config.ts_mode = SI2168_TS_PARALLEL;
@@ -1641,6 +1642,7 @@ static void cxusb_disconnect(struct usb_interface *intf)
 {
 	struct dvb_usb_device *d = usb_get_intfdata(intf);
 	struct cxusb_state *st = d->priv;
+#if 0
 	struct i2c_client *client;
 
 	/* remove I2C client for tuner */
@@ -1656,7 +1658,7 @@ static void cxusb_disconnect(struct usb_interface *intf)
 		module_put(client->dev.driver->owner);
 		i2c_unregister_device(client);
 	}
-
+#endif
 	dvb_usb_device_exit(intf);
 }
 
-- 
2.7.4

