diff -Naur mrxvt-0.5.4/configure.ac mrxvt-0.5.4.patch/configure.ac
--- mrxvt-0.5.4/configure.ac	2008-05-12 19:21:56.000000000 +0200
+++ mrxvt-0.5.4.patch/configure.ac	2013-01-12 01:54:55.819541351 +0100
@@ -28,7 +28,7 @@
 AC_INIT(mrxvt, 0.5.4, [jimmyzhou@users.sourceforge.net])
 AM_INIT_AUTOMAKE(mrxvt, 0.5.4)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR([src/main.c])
 
 
