--- projectm/libprojectM.pc.in.orig	2015-03-16 14:02:55.733369674 +0100
+++ projectm/libprojectM.pc.in	2015-03-16 14:03:42.685370237 +0100
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@LIB_INSTALL_DIR@
+libdir=@CMAKE_INSTALL_PREFIX@/lib
 includedir=@CMAKE_INSTALL_PREFIX@/include
 pkgdatadir=@CMAKE_INSTALL_PREFIX@/@RESOURCE_PREFIX@
 sysconfdir=@CMAKE_INSTALL_PREFIX@/@RESOURCE_PREFIX@
@@ -9,5 +9,5 @@
 Version: 2.0.0
 Description: projectM - OpenGL Milkdrop
 Requires:
-Libs: -L${libdir} -lprojectM 
+Libs: -L${libdir} -lprojectM -lMilkdropPresetFactory -lprojectM -lNativePresetFactory -lRenderer -lm
 Cflags: -I${includedir} 
