diff -x '*orig' -rup a/Common.hpp b/Common.hpp
--- a/Common.hpp	2012-05-20 04:43:26.000000000 +0100
+++ b/Common.hpp	2016-03-17 05:00:14.176905540 +0000
@@ -63,7 +63,8 @@ extern FILE *fmemopen(void *buf, size_t
 
 #ifdef LINUX
 #include <cstdlib>
-#define projectM_isnan isnan
+#include <cmath>
+#define projectM_isnan std::isnan
 
 #endif
 
diff -x '*orig' -rup a/MilkdropPresetFactory/Parser.cpp b/MilkdropPresetFactory/Parser.cpp
--- a/MilkdropPresetFactory/Parser.cpp	2012-05-20 04:43:26.000000000 +0100
+++ b/MilkdropPresetFactory/Parser.cpp	2016-03-17 05:00:14.177905556 +0000
@@ -1406,7 +1406,7 @@ PerFrameEqn * Parser::parse_implicit_per
   PerFrameEqn * per_frame_eqn;
   GenExpr * gen_expr;
 
-  if (fs == NULL)
+  if (!fs)
     return NULL;
   if (param_string == NULL)
     return NULL;
@@ -1561,7 +1561,7 @@ InitCond * Parser::parse_per_frame_init_
 
   if (preset == NULL)
     return NULL;
-  if (fs == NULL)
+  if (!fs)
     return NULL;
 
   if ((token = parseToken(fs, name)) != tEq)
@@ -1875,7 +1875,7 @@ int Parser::parse_shapecode(char * token
   /* Null argument checks */
   if (preset == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (token == NULL)
     return PROJECTM_FAILURE;
@@ -2166,7 +2166,7 @@ int Parser::parse_wave(char * token, std
 
   if (token == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
@@ -2348,7 +2348,7 @@ int Parser::parse_shape(char * token, st
   if (token == NULL)
 
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
