--- projectm/cmake/FindFreetype2.orig.cmake	2013-07-17 11:47:46.844351322 +0200
+++ projectm/cmake/FindFreetype2.cmake	2013-07-17 11:48:14.912350387 +0200
@@ -64,7 +64,7 @@
       /sw/include/freetype2
       ${VISUS_INCLUDE}
       ${VISUS_INCLUDE}/freetype2
-      NO_DEFAULT_PATH
+      #      NO_DEFAULT_PATH
 )
 
 FIND_PATH(FREETYPE2_FT2BUILD ft2build.h
@@ -74,7 +74,7 @@
       /usr/X11R6/include
       /sw/include
       ${VISUS_INCLUDE}
-      NO_DEFAULT_PATH
+      #      NO_DEFAULT_PATH
 )
 
 FIND_PATH(FREETYPE2_FT_CONFIG bin/freetype-config
@@ -84,7 +84,7 @@
       /usr/X11R6
       /sw/bin
       ${VISUS_BINARY_DIR}
-      NO_DEFAULT_PATH
+      #      NO_DEFAULT_PATH
 )
 
 IF (NOT WIN32)
@@ -95,7 +95,7 @@
       /usr/X11R6/lib
       /sw/lib
       ${VISUS_LIBRARIES}   
-      NO_DEFAULT_PATH
+      #      NO_DEFAULT_PATH
 )
 
 ELSE (NOT WIN32)
