--- projectm/MilkdropPresetFactory/CMakeLists.txt.orig	2015-03-16 13:27:41.557344326 +0100
+++ projectm/MilkdropPresetFactory/CMakeLists.txt	2015-03-16 13:31:00.769346714 +0100
@@ -15,3 +15,6 @@
 ADD_LIBRARY(MilkdropPresetFactory STATIC ${MilkdropPresetFactory_SOURCES})
 SET_TARGET_PROPERTIES(MilkdropPresetFactory PROPERTIES VERSION 2.00 SOVERSION 2)
 TARGET_LINK_LIBRARIES(MilkdropPresetFactory Renderer)
+IF(BUILD_PROJECTM_STATIC)
+  INSTALL(TARGETS MilkdropPresetFactory DESTINATION lib${LIB_SUFFIX})
+endif()
