--- projectm/NativePresetFactory/CMakeLists.txt.orig	2015-03-16 13:31:17.965346920 +0100
+++ projectm/NativePresetFactory/CMakeLists.txt	2015-03-16 13:31:38.921347172 +0100
@@ -13,3 +13,6 @@
 ADD_LIBRARY(NativePresetFactory STATIC ${NativePresetFactory_SOURCES})
 SET_TARGET_PROPERTIES(NativePresetFactory PROPERTIES VERSION 2.00 SOVERSION 2)
 TARGET_LINK_LIBRARIES(NativePresetFactory Renderer m)
+IF(BUILD_PROJECTM_STATIC)
+  INSTALL(TARGETS NativePresetFactory DESTINATION lib${LIB_SUFFIX})
+endif()
