diff -Naur SABnzbd-0.6.4/interfaces/Plush/templates/config_directories.tmpl SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_directories.tmpl
--- SABnzbd-0.6.4/interfaces/Plush/templates/config_directories.tmpl	2011-06-08 00:02:34.000000000 +0200
+++ SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_directories.tmpl	2011-06-13 16:37:49.027611932 +0200
@@ -76,16 +76,6 @@
           <span class="component-desc">$T('explain-dirscan_speed')</span>
         </label>
       </div>
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="script_dir">
-          <span class="component-title">$T('opt-script_dir')</span>
-          <input type="text" size="50" name="script_dir" id="script_dir" value="$script_dir">
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-script_dir')</span>
-        </label>
-      </div>
       <div class="field-pair alt">
         <label class="nocheck clearfix" for="email_dir">
           <span class="component-title">$T('opt-email_dir')</span>
@@ -109,49 +99,6 @@
     </fieldset>
   </div><!-- /component-group1 -->
 
-  <div id="core-component-group2" class="component-group clearfix">
-    <div class="component-group-desc">
-      <h3>$T('systemFolders')</h3>
-      <p><small>$T('base-folder'):<br/>$my_lcldata</small></p>
-    </div>
-    <fieldset class="component-group-list">
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="admin_dir">
-          <span class="component-title">$T('opt-admin_dir')</span>
-          <input type="text" size="50" name="admin_dir" id="admin_dir" value="$admin_dir">
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-admin_dir1')</span>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-admin_dir2')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="log_dir">
-          <span class="component-title">$T('opt-log_dir')</span>
-          <input type="text" size="50" name="log_dir" id="log_dir" value="$log_dir">
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-log_dir')</span>
-        </label>
-      </div>
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="nzb_backup_dir">
-          <span class="component-title">$T('opt-nzb_backup_dir')</span>
-          <input type="text" size="50" name="nzb_backup_dir" id="nzb_backup_dir" value="$nzb_backup_dir">
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-nzb_backup_dir')</span>
-        </label>
-      </div>
-    </fieldset>
-  </div><!-- /component-group2 -->
-
   <div class="component-group-last clearfix">
     <div class="component-group-desc">
       <h3>&nbsp;</h3>
diff -Naur SABnzbd-0.6.4/interfaces/Plush/templates/config_general.tmpl SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_general.tmpl
--- SABnzbd-0.6.4/interfaces/Plush/templates/config_general.tmpl	2011-06-08 00:02:34.000000000 +0200
+++ SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_general.tmpl	2011-06-13 16:35:26.411903234 +0200
@@ -14,82 +14,6 @@
     </div>
     <fieldset class="component-group-list">
       <div class="field-pair">
-        <label class="nocheck clearfix" for="host">
-          <span class="component-title">$T('opt-host')</span>
-          <input type="text" name="host" id="host" value="$host"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-host')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="port">
-          <span class="component-title">$T('opt-port')</span>
-          <input type="text" name="port" id="port" size=8 value="$port"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-port')</span>
-        </label>
-      </div>
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="username">
-          <span class="component-title">$T('opt-web_username')</span>
-          <input type="text" name="username" id="username" value="$username"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-web_username')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="password">
-          <span class="component-title">$T('opt-web_password')</span>
-          <input type="password" name="password" id="password" value="$password"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-web_password')</span>
-        </label>
-      </div>
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="web_dir">
-          <span class="component-title">$T('opt-web_dir')</span>
-          <select name="web_dir" id="web_dir">
-          <!--#for $webline in $web_list#-->
-          <!--#if $webline.lower() == $web_dir.lower()#-->
-            <option value="$webline" selected>$webline</option>
-          <!--#else#-->
-            <option value="$webline">$webline</option>
-          <!--#end if#-->
-          <!--#end for#-->
-          </select>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-web_dir')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="web_dir2">
-          <span class="component-title">$T('opt-web_dir2')</span>
-          <select name="web_dir2" id="web_dir2">
-          <!--#for $webline in $web_list2#-->
-          <!--#if $webline.lower() == $web_dir2.lower()#-->
-            <option value="$webline" selected>$webline</option>
-          <!--#else#-->
-            <option value="$webline">$webline</option>
-          <!--#end if#-->
-          <!--#end for#-->
-          </select>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-web_dir2')</span>
-        </label>
-      </div>
-      <div class="field-pair">
         <label class="nocheck clearfix" for="language">
           <span class="component-title">$T('opt-language')</span>
           <select name="language" id="language" id="language" class="select">
@@ -129,13 +53,6 @@
           <span class="component-desc">$T('explain-nzbkey')</span>
         </label>
       </div>
-      <div class="field-pair alt">
-        <label class="clearfix" for="disable_api_key">
-          <input type="checkbox" name="disable_api_key" id="disable_api_key" value="1" <!--#if $disable_api_key > 0 then 'checked="1"' else ""#--> />
-          <span class="component-title">$T('opt-disableApikey')</span>
-          <span class="component-desc">$T('explain-disableApikey')</span>
-        </label>
-      </div>
     </fieldset>
   </div><!-- /component-group1 -->
 
@@ -152,36 +69,6 @@
           <span class="component-desc">$T('explain-enable_https')</span>
         </label>
       </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="https_port">
-          <span class="component-title">$T('opt-https_port')</span>
-          <input type="text" name="https_port" id="https_port" size="8" value="$https_port"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-https_port')</span>
-        </label>
-      </div>
-      <div class="field-pair">
-        <label class="nocheck clearfix" for="https_cert">
-          <span class="component-title">$T('opt-https_cert')</span>
-          <input type="text" name="https_cert" id="https_cert" value="$https_cert"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-https_cert')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <label class="nocheck clearfix" for="https_key">
-          <span class="component-title">$T('opt-https_key')</span>
-          <input type="text" name="https_key" id="https_key" value="$https_key"/>
-        </label>
-        <label class="nocheck clearfix">
-          <span class="component-title">&nbsp;</span>
-          <span class="component-desc">$T('explain-https_key')</span>
-        </label>
-      </div>
     </fieldset>
   </div><!-- /component-group2 -->
 
diff -Naur SABnzbd-0.6.4/interfaces/Plush/templates/config_switches.tmpl SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_switches.tmpl
--- SABnzbd-0.6.4/interfaces/Plush/templates/config_switches.tmpl	2011-06-08 00:02:34.000000000 +0200
+++ SABnzbd-0.6.4.patch/interfaces/Plush/templates/config_switches.tmpl	2011-06-13 16:35:26.446903654 +0200
@@ -12,20 +12,6 @@
       <h3>$T('swtag-general')</h3>
     </div>
     <fieldset class="component-group-list">
-      <div class="field-pair">
-        <input type="checkbox" name="check_new_rel" id="check_new_rel" value="1" <!--#if $check_new_rel > 0 then "checked=1" else ""#--> />
-        <label class="clearfix" for="check_new_rel">
-          <span class="component-title">$T('opt-check_new_rel')</span>
-          <span class="component-desc">$T('explain-check_new_rel')</span>
-        </label>
-      </div>
-      <div class="field-pair alt">
-        <input type="checkbox" name="auto_browser" id="auto_browser" value="1" <!--#if $auto_browser > 0 then "checked=1" else ""#--> />
-        <label class="clearfix" for="auto_browser">
-          <span class="component-title">$T('opt-auto_browser')</span>
-          <span class="component-desc">$T('explain-auto_browser')</span>
-        </label>
-      </div>
       <!--#if $have_ampm#-->
       <div class="field-pair">
         <input type="checkbox" name="ampm" id="ampm" value="1" <!--#if $ampm > 0 then "checked=1" else ""#--> />
