diff -Naur dbus-python-1.1.1/configure.ac dbus-python-1.1.1.patch/configure.ac
--- dbus-python-1.1.1/configure.ac	2012-06-25 18:02:12.000000000 +0200
+++ dbus-python-1.1.1.patch/configure.ac	2013-01-12 01:09:51.235915152 +0100
@@ -29,7 +29,7 @@
 AC_SUBST(DBUS_PYTHON_MICRO_VERSION, dbus_python_micro_version)
 
 AC_CONFIG_SRCDIR([_dbus_bindings/module.c])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AM_INIT_AUTOMAKE([1.9 -Wno-portability])
 
