diff -Naur udev-151.orig/extras/ata_id/ata_id.c udev-151/extras/ata_id/ata_id.c
--- udev-151.orig/extras/ata_id/ata_id.c	2010-02-16 21:25:09.530757189 +0100
+++ udev-151/extras/ata_id/ata_id.c	2010-02-16 21:39:06.082007681 +0100
@@ -20,6 +20,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#define u8 uint8_t
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
diff -Naur udev-151.orig/extras/scsi_id/scsi_serial.c udev-151/extras/scsi_id/scsi_serial.c
--- udev-151.orig/extras/scsi_id/scsi_serial.c	2010-02-16 21:25:09.530757189 +0100
+++ udev-151/extras/scsi_id/scsi_serial.c	2010-02-16 21:38:56.010757438 +0100
@@ -14,6 +14,8 @@
  *	Free Software Foundation version 2 of the License.
  */
 
+#define u8 uint8_t
+
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
