diff -Nur udev-145/udev/udevd.c udev-145-new/udev/udevd.c
--- udev-145/udev/udevd.c	2009-07-13 09:19:51.000000000 +0800
+++ udev-145-new/udev/udevd.c	2009-09-03 22:49:08.000000000 +0800
@@ -654,14 +654,16 @@
 	if ((ioctl(pfd[FD_INOTIFY].fd, FIONREAD, &nbytes) < 0) || (nbytes <= 0))
 		return 0;
 
-	buf = malloc(nbytes);
+        info(udev, "buf allocated size: %d", nbytes); 
+	buf = malloc(nbytes*2);
 	if (buf == NULL) {
 		err(udev, "error getting buffer for inotify\n");
 		return -1;
 	}
-
+        
 	nbytes = read(pfd[FD_INOTIFY].fd, buf, nbytes);
-
+        info(udev, "buf read size: %d", nbytes);
+        
 	for (pos = 0; pos < nbytes; pos += sizeof(struct inotify_event) + ev->len) {
 		struct udev_device *dev;
 
