diff -up udev-116/extras/rule_generator/write_net_rules.nettype udev-116/extras/rule_generator/write_net_rules
--- udev-116/extras/rule_generator/write_net_rules.nettype	2007-10-18 21:16:28.000000000 +0200
+++ udev-116/extras/rule_generator/write_net_rules	2007-10-18 21:20:55.000000000 +0200
@@ -23,6 +23,10 @@
 
 RULES_FILE='/etc/udev/rules.d/70-persistent-net.rules'
 
+if [ "$MATCHIFTYPE" ]; then
+	[ $[MATCHIFTYPE] -gt 256 ] && return 0
+fi
+
 . /lib/udev/rule_generator.functions
 
 interface_name_taken() {
