diff -up udev-120/etc/udev/rules.d/60-persistent-storage.rules.sysconf udev-120/etc/udev/rules.d/60-persistent-storage.rules
--- udev-120/rules/rules.d/60-persistent-storage.rules.sysconf	2008-04-03 21:12:53.000000000 +0200
+++ udev-120/rules/rules.d/60-persistent-storage.rules	2008-04-04 14:11:35.000000000 +0200
@@ -3,6 +3,8 @@
 # persistent storage links: /dev/disk/{by-id,by-uuid,by-label,by-path}
 # scheme based on "Linux persistent device names", 2004, Hannes Reinecke <hare@suse.de>
 
+ENV{UDEV_NO_PERSISTENT_STORAGE}=="1", GOTO="persistent_storage_end"
+
 ACTION!="add|change", GOTO="persistent_storage_end"
 SUBSYSTEM!="block", GOTO="persistent_storage_end"
 
diff -up udev-120/etc/udev/rules.d/60-persistent-storage-tape.rules.sysconf udev-120/etc/udev/rules.d/60-persistent-storage-tape.rules
--- udev-120/rules/rules.d/60-persistent-storage-tape.rules.sysconf	2008-04-03 21:12:53.000000000 +0200
+++ udev-120/rules/rules.d/60-persistent-storage-tape.rules	2008-04-04 14:10:46.000000000 +0200
@@ -2,6 +2,8 @@
 
 # persistent storage links: /dev/tape/{by-id,by-path}
 
+ENV{UDEV_NO_PERSISTENT_STORAGE}=="1", GOTO="persistent_storage_tape_end"
+
 ACTION!="add|change", GOTO="persistent_storage_tape_end"
 
 KERNEL=="nst[0-9]", SUBSYSTEMS=="scsi", IMPORT{program}="scsi_id --export --whitelisted --fallback-to-sysfs -s %p -d $tempnode", SYMLINK+="tape/by-id/$env{ID_BUS}-$env{ID_SERIAL}-nst"
