diff -urN udev-150/rules/rules.d/60-persistent-storage.rules udev-150-new/rules/rules.d/60-persistent-storage.rules
--- udev-150/rules/rules.d/60-persistent-storage.rules	2009-12-09 01:43:09.000000000 +0800
+++ udev-150-new/rules/rules.d/60-persistent-storage.rules	2010-01-09 21:53:59.000000000 +0800
@@ -3,6 +3,8 @@
 # persistent storage links: /dev/disk/{by-id,by-uuid,by-label,by-path}
 # scheme based on "Linux persistent device names", 2004, Hannes Reinecke <hare@suse.de>
 
+ENV{UDEV_NO_PERSISTENT_STORAGE}=="1", GOTO="persistent_storage_end"
+
 # forward scsi device event to corresponding block device
 ACTION=="change", SUBSYSTEM=="scsi", ENV{DEVTYPE}=="scsi_device", TEST=="block", ATTR{block/*/uevent}="change"
 
diff -urN udev-150/rules/rules.d/60-persistent-storage-tape.rules udev-150-new/rules/rules.d/60-persistent-storage-tape.rules
--- udev-150/rules/rules.d/60-persistent-storage-tape.rules	2009-12-03 20:45:03.000000000 +0800
+++ udev-150-new/rules/rules.d/60-persistent-storage-tape.rules	2010-01-09 21:54:26.000000000 +0800
@@ -2,6 +2,8 @@
 
 # persistent storage links: /dev/tape/{by-id,by-path}
 
+ENV{UDEV_NO_PERSISTENT_STORAGE}=="1", GOTO="persistent_storage_tape_end"
+
 ACTION!="add|change", GOTO="persistent_storage_tape_end"
 
 # type 8 devices are "Medium Changers"
