diff -Naur udev-162/configure.ac udev-162.patch/configure.ac
--- udev-162/configure.ac	2010-09-03 10:11:37.000000000 +0200
+++ udev-162.patch/configure.ac	2010-09-03 23:14:46.552571286 +0200
@@ -84,9 +84,6 @@
 	AC_SUBST([GLIB_CFLAGS])
 	AC_SUBST([GLIB_LIBS])
 
-	AC_CHECK_LIB([acl], [acl_init], [:], AC_MSG_ERROR([libacl not found]))
-	AC_CHECK_HEADER([acl/libacl.h], [:], AC_MSG_ERROR([libacl header not found]))
-
 	PKG_CHECK_MODULES(LIBUSB, libusb >= 0.1.12)
 	AC_SUBST(LIBUSB_CFLAGS)
 	AC_SUBST(LIBUSB_LIBS)
diff -Naur udev-162/Makefile.am udev-162.patch/Makefile.am
--- udev-162/Makefile.am	2010-09-03 13:19:05.000000000 +0200
+++ udev-162.patch/Makefile.am	2010-09-03 23:16:04.628571420 +0200
@@ -441,21 +441,6 @@
 libexec_PROGRAMS += extras/hid2hci/hid2hci
 
 # ------------------------------------------------------------------------------
-# udev_acl - apply ACLs for users with local forground sessions
-# ------------------------------------------------------------------------------
-extras_udev_acl_udev_acl_SOURCES = extras/udev-acl/udev-acl.c
-extras_udev_acl_udev_acl_CPPFLAGS = $(AM_CPPFLAGS) $(GLIB_CFLAGS)
-extras_udev_acl_udev_acl_LDADD = libudev/libudev-private.la -lacl $(GLIB_LIBS)
-dist_udevrules_DATA += extras/udev-acl/70-acl.rules
-libexec_PROGRAMS += extras/udev-acl/udev-acl
-
-udevacl-install-hook:
-	mkdir -p $(DESTDIR)$(prefix)/lib/ConsoleKit/run-seat.d
-	ln -sf $(libexecdir)/udev-acl $(DESTDIR)$(prefix)/lib/ConsoleKit/run-seat.d/udev-acl.ck
-
-INSTALL_EXEC_HOOKS += udevacl-install-hook
-
-# ------------------------------------------------------------------------------
 # usb-db - read USB vendor/device string database
 # ------------------------------------------------------------------------------
 extras_usb_db_usb_db_SOURCES = extras/usb-db/usb-db.c
@@ -502,106 +487,6 @@
 
 libexec_PROGRAMS += extras/mtd_probe/mtd_probe
  
-
-# ------------------------------------------------------------------------------
-# keymap - map custom hardware's multimedia keys
-# ------------------------------------------------------------------------------
-extras_keymap_keymap_SOURCES = extras/keymap/keymap.c
-extras_keymap_keymap_CPPFLAGS = $(AM_CPPFLAGS) -I extras/keymap
-nodist_extras_keymap_keymap_SOURCES = \
-	extras/keymap/keys-from-name.h \
-	extras/keymap/keys-to-name.h
-BUILT_SOURCES += $(nodist_extras_keymap_keymap_SOURCES)
-
-EXTRA_DIST += extras/keymap/check-keymaps.sh
-dist_udevrules_DATA += extras/keymap/95-keymap.rules
-dist_udevrules_DATA += extras/keymap/95-keyboard-force-release.rules
-libexec_PROGRAMS += extras/keymap/keymap
-dist_libexec_SCRIPTS += extras/keymap/findkeyboards
-dist_libexec_SCRIPTS += extras/keymap/keyboard-force-release.sh
-dist_doc_DATA = extras/keymap/README.keymap.txt
-
-CLEANFILES += \
-	extras/keymap/keys.txt \
-	extras/keymap/keys-from-name.gperf
-
-udevkeymapdir = $(libexecdir)/keymaps
-dist_udevkeymap_DATA = \
-	extras/keymap/keymaps/acer \
-	extras/keymap/keymaps/acer-aspire_5720 \
-	extras/keymap/keymaps/acer-aspire_5920g \
-	extras/keymap/keymaps/acer-aspire_6920 \
-	extras/keymap/keymaps/acer-travelmate_c300 \
-	extras/keymap/keymaps/asus \
-	extras/keymap/keymaps/compaq-e_evo \
-	extras/keymap/keymaps/dell \
-	extras/keymap/keymaps/everex-xt5000 \
-	extras/keymap/keymaps/fujitsu-amilo_pa_2548 \
-	extras/keymap/keymaps/fujitsu-amilo_pro_edition_v3505 \
-	extras/keymap/keymaps/fujitsu-amilo_pro_v3205 \
-	extras/keymap/keymaps/fujitsu-amilo_si_1520 \
-	extras/keymap/keymaps/fujitsu-esprimo_mobile_v5 \
-	extras/keymap/keymaps/fujitsu-esprimo_mobile_v6 \
-	extras/keymap/keymaps/hewlett-packard \
-	extras/keymap/keymaps/hewlett-packard-2510p_2530p \
-	extras/keymap/keymaps/hewlett-packard-compaq_elitebook \
-	extras/keymap/keymaps/hewlett-packard-pavilion \
-	extras/keymap/keymaps/hewlett-packard-presario-2100 \
-	extras/keymap/keymaps/hewlett-packard-tablet \
-	extras/keymap/keymaps/hewlett-packard-tx2 \
-	extras/keymap/keymaps/ibm-thinkpad-usb-keyboard-trackpoint \
-	extras/keymap/keymaps/inventec-symphony_6.0_7.0 \
-	extras/keymap/keymaps/lenovo-3000 \
-	extras/keymap/keymaps/lenovo-ideapad \
-	extras/keymap/keymaps/lenovo-thinkpad-usb-keyboard-trackpoint \
-	extras/keymap/keymaps/lenovo-thinkpad_x6_tablet \
-	extras/keymap/keymaps/lenovo-thinkpad_x200_tablet \
-	extras/keymap/keymaps/lg-x110 \
-	extras/keymap/keymaps/logitech-wave \
-	extras/keymap/keymaps/logitech-wave-cordless \
-	extras/keymap/keymaps/logitech-wave-pro-cordless \
-	extras/keymap/keymaps/maxdata-pro_7000 \
-	extras/keymap/keymaps/medion-fid2060 \
-	extras/keymap/keymaps/medionnb-a555 \
-	extras/keymap/keymaps/micro-star \
-	extras/keymap/keymaps/module-asus-w3j \
-	extras/keymap/keymaps/module-ibm \
-	extras/keymap/keymaps/module-lenovo \
-	extras/keymap/keymaps/module-sony \
-	extras/keymap/keymaps/module-sony-old \
-	extras/keymap/keymaps/olpc-xo \
-	extras/keymap/keymaps/onkyo \
-	extras/keymap/keymaps/oqo-model2 \
-	extras/keymap/keymaps/samsung-other \
-	extras/keymap/keymaps/samsung-sq1us \
-	extras/keymap/keymaps/samsung-sx20s \
-	extras/keymap/keymaps/toshiba-satellite_a100 \
-	extras/keymap/keymaps/toshiba-satellite_a110 \
-	extras/keymap/keymaps/toshiba-satellite_m30x \
-	extras/keymap/keymaps/zepto-znote
-
-udevkeymapforcereldir = $(libexecdir)/keymaps/force-release
-dist_udevkeymapforcerel_DATA = \
-	extras/keymap/force-release-maps/samsung-other \
-	extras/keymap/force-release-maps/common-volume-keys
-
-extras/keymap/keys.txt: $(INCLUDE_PREFIX)/linux/input.h
-	$(AM_V_at)mkdir -p extras/keymap
-	$(AM_V_GEN)$(AWK) '/^#define.*KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' < $< | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
-
-extras/keymap/keys-from-name.gperf: extras/keymap/keys.txt
-	$(AM_V_GEN)$(AWK) 'BEGIN{ print "struct key { const char* name; unsigned short id; };"; print "%null-strings"; print "%%";} { print $$1 ", " $$1 }' < $< > $@
-
-extras/keymap/keys-from-name.h: extras/keymap/keys-from-name.gperf Makefile
-	$(AM_V_GEN)$(GPERF) -L ANSI-C -t --ignore-case -N lookup_key -H hash_key_name -p -C < $< > $@
-
-extras/keymap/keys-to-name.h: extras/keymap/keys.txt Makefile
-	$(AM_V_GEN)$(AWK) 'BEGIN{ print "const char* const key_names[KEY_CNT] = { "} { print "[" $$1 "] = \"" $$1 "\"," } END{print "};"}' < $< > $@
-
-keymaps-distcheck-hook: extras/keymap/keys.txt
-	$(top_srcdir)/extras/keymap/check-keymaps.sh $(top_srcdir) $^
-DISTCHECK_HOOKS += keymaps-distcheck-hook
-
 endif # ENABLE_EXTRAS
 
 # ------------------------------------------------------------------------------
