diff -urN udev-150/rules/rules.d/50-udev-default.rules udev-150-new/rules/rules.d/50-udev-default.rules
--- udev-150/rules/rules.d/50-udev-default.rules	2009-12-15 23:17:02.000000000 +0800
+++ udev-150-new/rules/rules.d/50-udev-default.rules	2010-01-09 21:03:09.000000000 +0800
@@ -22,9 +22,9 @@
 KERNEL=="mem|kmem|port|nvram",	GROUP="kmem", MODE="0640"
 
 # input
-KERNEL=="mouse*|mice|event*",	MODE="0640"
-KERNEL=="ts[0-9]*|uinput",	MODE="0640"
-KERNEL=="js[0-9]*",		MODE="0644"
+KERNEL=="mouse*|mice|event*",	MODE="0660", GROUP="video"
+KERNEL=="ts[0-9]*|uinput",	MODE="0660", GROUP="video"
+KERNEL=="js[0-9]*",		MODE="0664", GROUP="video"
 
 # video4linux
 SUBSYSTEM=="video4linux",	GROUP="video"
