diff -urN udev-150/rules/rules.d/50-udev-default.rules udev-150-new/rules/rules.d/50-udev-default.rules
--- udev-150/rules/rules.d/50-udev-default.rules	2009-12-15 23:17:02.000000000 +0800
+++ udev-150-new/rules/rules.d/50-udev-default.rules	2010-01-09 21:08:31.000000000 +0800
@@ -7,7 +7,7 @@
 KERNEL=="tty[pqrstuvwxyzabcdef][0123456789abcdef]", GROUP="tty", MODE="0660"
 KERNEL=="ptmx",			GROUP="tty", MODE="0666"
 KERNEL=="tty",			GROUP="tty", MODE="0666"
-KERNEL=="tty[0-9]*",		GROUP="tty", MODE="0620"
+KERNEL=="tty[0-9]*",            GROUP="video", MODE="0660"
 KERNEL=="console",		MODE="0600"
 KERNEL=="vcs|vcs[0-9]*|vcsa|vcsa[0-9]*", GROUP="tty"
 
