diff -Naur udev-175/extras/rule_generator/write_cd_rules udev-175.patch/extras/rule_generator/write_cd_rules
--- udev-175/src/extras/rule_generator/write_cd_rules	2010-06-25 11:37:36.000000000 +0200
+++ udev-175.patch/src/extras/rule_generator/write_cd_rules	2011-12-04 17:11:04.704237462 +0100
@@ -27,9 +27,9 @@
 	fi
 fi
 
-RULES_FILE="/etc/udev/rules.d/70-persistent-cd.rules"
+RULES_FILE="/run/udev/70-persistent-cd.rules"
 
-. /lib/udev/rule_generator.functions
+. /lib/udev/rule_generator.functions
 
 find_next_available() {
 	raw_find_next_available "$(find_all_rules 'SYMLINK\+=' "$1")"
diff -Naur udev-175/extras/rule_generator/write_net_rules udev-175.patch/extras/rule_generator/write_net_rules
--- udev-175/src/extras/rule_generator/write_net_rules	2010-08-11 10:30:57.000000000 +0200
+++ udev-175.patch/src/extras/rule_generator/write_net_rules	2011-12-04 17:11:22.696592805 +0100
@@ -38,9 +38,9 @@
 	fi
 fi
 
-RULES_FILE='/etc/udev/rules.d/70-persistent-net.rules'
+RULES_FILE='/run/udev/70-persistent-net.rules'
 
-. /lib/udev/rule_generator.functions
+. /lib/udev/rule_generator.functions
 
 interface_name_taken() {
 	local value="$(find_all_rules 'NAME=' $INTERFACE)"
