diff -urN udev-150/rules/rules.d/50-udev-default.rules udev-150-new/rules/rules.d/50-udev-default.rules
--- udev-150/rules/rules.d/50-udev-default.rules	2009-12-15 23:17:02.000000000 +0800
+++ udev-150-new/rules/rules.d/50-udev-default.rules	2010-01-09 21:05:59.000000000 +0800
@@ -34,6 +34,7 @@
 
 # graphics
 KERNEL=="agpgart",		MODE="0600", GROUP="video"
+KERNEL=="card[0-9]*",           GROUP="video"
 KERNEL=="pmu",			GROUP="video"
 KERNEL=="nvidia*|nvidiactl*",	GROUP="video"
 SUBSYSTEM=="graphics",		GROUP="video"
