diff -Naur samba-3.0.25.orig/source/include/debug.h samba-3.0.25/source/include/debug.h
--- samba-3.0.25.orig/source/include/debug.h	2007-05-17 12:28:16.000000000 +0200
+++ samba-3.0.25/source/include/debug.h	2007-05-17 12:31:32.000000000 +0200
@@ -162,54 +162,18 @@
  * will remove the extra conditional test.
  */
 
-#define DEBUGLVL( level ) \
-  ( ((level) <= MAX_DEBUG_LEVEL) && \
-     ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-     (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-      DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) ) )
-
-
-#define DEBUGLVLC( dbgc_class, level ) \
-  ( ((level) <= MAX_DEBUG_LEVEL) && \
-     ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-     (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-      DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) ) )
-
-
-#define DEBUG( level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
-
-#define DEBUGC( dbgc_class, level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-	    DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
-
-#define DEBUGADD( level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
-
-#define DEBUGADDC( dbgc_class, level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-          ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
-
-/* Print a separator to the debug log. */
-#define DEBUGSEP(level)\
-	DEBUG((level),("===============================================================\n"))
+#define DEBUGLVL( level )  (0)
+
+#define DEBUGLVLC( dbgc_class, level )  (0)
+
+#define DEBUG( level, body )  ((void)0)
+
+#define DEBUGC( dbgc_class, level, body )  ((void)0)
+
+#define DEBUGADD( level, body )  ((void)0)
+
+#define DEBUGADDC( dbgc_class, level, body )  ((void)0)
+
+#define DEBUGSEP( level )  ((void)0)
 
 #endif
