diff -Naur samba-3.0.25.orig/source/libsmb/clierror.c samba-3.0.25/source/libsmb/clierror.c
--- samba-3.0.25.orig/source/libsmb/clierror.c	2007-05-17 12:28:16.000000000 +0200
+++ samba-3.0.25/source/libsmb/clierror.c	2007-05-17 12:32:35.000000000 +0200
@@ -159,7 +159,7 @@
 	if (flgs2 & FLAGS2_32_BIT_ERROR_CODES) {
 		NTSTATUS status = NT_STATUS(IVAL(cli->inbuf,smb_rcls));
 
-		return nt_errstr(status);
+		return "NT_STATUS_UNSUCCESSFUL";
         }
 
 	cli_dos_error(cli, &errclass, &errnum);
