diff -Naur samba-3.0.20.orig/source/configure samba-3.0.20/source/configure
--- samba-3.0.20.orig/source/configure	2005-08-19 22:09:01.000000000 +0300
+++ samba-3.0.20/source/configure	2005-08-23 13:51:06.000000000 +0300
@@ -18731,11 +18731,7 @@
     *linux*)
        # glibc <= 2.3.2 has a broken getgrouplist
        if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+           linux_getgrouplist_ok=no
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
