--- SAMBA_3_2/source/modules/vfs_default.c	(revision 23961)
+++ SAMBA_3_2/source/modules/vfs_default.c	(revision 23962)
@@ -826,10 +826,11 @@
 
 	START_PROFILE(syscall_linux_setlease);
 
-#ifdef LINUX
+#ifdef HAVE_KERNEL_OPLOCKS_LINUX
 	/* first set the signal handler */
-	if(linux_set_lease_sighandler(fd) == -1)
+	if(linux_set_lease_sighandler(fd) == -1) {
 		return -1;
+	}
 
 	result = linux_setlease(fd, leasetype);
 #else

