diff -Naur openvpn-2.3.0/configure.ac openvpn-2.3.0.patch/configure.ac
--- openvpn-2.3.0/configure.ac	2012-11-29 21:47:57.000000000 +0100
+++ openvpn-2.3.0.patch/configure.ac	2013-01-16 03:36:51.567034700 +0100
@@ -32,7 +32,7 @@
 AC_DEFINE([OPENVPN_VERSION_RESOURCE], [PRODUCT_VERSION_RESOURCE], [Version in windows resource format])
 
 AC_CONFIG_AUX_DIR([.])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([src/openvpn/syshead.h])
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE
