diff -Naur openvpn-2.2.2/configure.ac openvpn-2.2.2.patch/configure.ac
--- openvpn-2.2.2/configure.ac	2011-12-13 17:58:56.000000000 +0100
+++ openvpn-2.2.2.patch/configure.ac	2013-01-12 00:50:12.273055490 +0100
@@ -27,7 +27,7 @@
 
 m4_include(version.m4)
 AC_INIT([OpenVPN], [PRODUCT_VERSION], [openvpn-users@lists.sourceforge.net], [openvpn])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR(syshead.h)
 
 dnl Guess host type.
