commit e8bb2e8e0ef3602fc8cb252cee541bed5d45fc5c
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 19 22:19:10 2013 +0300

    vdr 2.1.2 buildfix

diff --git a/uservars.h b/uservars.h
index f098750..a0e047c 100644
--- a/uservars.h
+++ b/uservars.h
@@ -538,7 +538,7 @@ class cVideodirVar : public cInternalVar {
     cVideodirVar() : cInternalVar("videodir") {}
     string Evaluate(const cEvent*, bool escapeStrings = false)
 	{
-	    return VideoDirectory;
+	    return cVideoDirectory::Name();
 	}
 };
 
