From 15d08ffb09de612ae1eb53b2aeca2f906f7076a6 Mon Sep 17 00:00:00 2001
From: Christian Wieninger
Date: Sun, 04 Mar 2012 08:44:19 +0000
Subject: support for vdr 1.7.25

---
diff --git a/Makefile b/Makefile
index 0397217..4a72b41 100644
--- a/Makefile
+++ b/Makefile
@@ -64,7 +64,7 @@ CXXFLAGS ?= -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses -Wno-format-y2k
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
+#DVBDIR = ../../../../DVB
 VDRDIR = ../../..
 LIBDIR = ../../lib
 TMPDIR = /tmp
diff --git a/conflictcheck.h b/conflictcheck.h
index fcc6dcb..7e6fb7a 100644
--- a/conflictcheck.h
+++ b/conflictcheck.h
@@ -163,12 +163,20 @@ class cConflictCheckDevice
 				result = true;
 #ifdef DO_REC_AND_PLAY_ON_PRIMARY_DEVICE
 			    else
+#if APIVERSNUM < 10725
 				result = Priority >= Setup.PrimaryLimit;
+#else
+				result = Priority >= 0;
+#endif
 #endif
 #endif
 			}
 			else
+#if APIVERSNUM < 10725
 			    result = !IsPrimaryDevice() || Priority >= Setup.PrimaryLimit;
+#else
+			    result = !IsPrimaryDevice() || Priority >= 0;
+#endif
 		    }
 		    else
 			needsDetachReceivers = true;
--
cgit v0.9.0.2-39-g756e
