From 61c795e71df00f4d3c22d2be405ceb0eb0e0e49c Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 22 Nov 2013 21:45:56 +0200
Subject: [PATCH] don't build unused plugins

---
 Makefile |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 118e379..5f15ebe 100644
--- a/Makefile
+++ b/Makefile
@@ -16,9 +16,9 @@
 ### edit one of these lines to '1', if you don't want the addon epgsearchonly, 
 ### conflictcheckonly or quickepgsearch
 
-WITHOUT_EPGSEARCHONLY=0
-WITHOUT_CONFLICTCHECKONLY=0
-WITHOUT_QUICKSEARCH=0
+WITHOUT_EPGSEARCHONLY=1
+WITHOUT_CONFLICTCHECKONLY=1
+WITHOUT_QUICKSEARCH=1
 
 ### edit this to '0' if you don't want epgsearch to auto config itself
 AUTOCONFIG=1
-- 
1.7.2.5

