diff -Naur newt-0.52.14/Makefile.in newt-0.52.14.patch/Makefile.in
--- newt-0.52.14/Makefile.in	2011-11-11 12:00:15.000000000 +0100
+++ newt-0.52.14.patch/Makefile.in	2013-05-30 23:40:30.104284277 +0200
@@ -5,7 +5,7 @@
 CPP = @CPP@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
-CPPFLAGS = -D_GNU_SOURCE -I/usr/include/slang @CPPFLAGS@ 
+CPPFLAGS = -D_GNU_SOURCE @CPPFLAGS@ 
 GNU_LD = @GNU_LD@
 
 VERSION = @VERSION@
