diff -Naur a/arch/arm/boot/dts/imx6dl-hummingboard.dts b/arch/arm/boot/dts/imx6dl-hummingboard.dts
--- a/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-22 18:54:01.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-22 19:27:28.561585329 +0100
@@ -58,13 +58,13 @@
 			regulator-max-microvolt = <5000000>;
 		};
 	};
-
+/*
 	codec: spdif-transmitter {
 		compatible = "linux,spdif-dit";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_hummingboard_spdif>;
 	};
-
+*/
 	imx-drm {
 		compatible = "fsl,imx-drm";
 		crtcs = <&ipu1 0>, <&ipu1 1>;
@@ -74,9 +74,10 @@
 	sound-spdif {
 		compatible = "fsl,imx-audio-spdif";
 		model = "imx-spdif";
-		/* IMX6 doesn't implement this yet */
 		spdif-controller = <&spdif>;
 		spdif-out;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_hummingboard_spdif>;
 	};
 
         sound-hdmi {
diff -Naur a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
--- a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-22 18:54:01.000000000 +0100
+++ b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-03-22 18:28:38.000000000 +0100
@@ -63,13 +63,13 @@
 			regulator-max-microvolt = <5000000>;
 		};
 	};
-
+/*
 	codec: spdif-transmitter {
 		compatible = "linux,spdif-dit";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_cubox_i_spdif>;
 	};
-
+*/
 	imx-drm {
 		compatible = "fsl,imx-drm";
 		crtcs = <&ipu1 0>, <&ipu1 1>;
@@ -81,6 +81,8 @@
 		model = "imx-spdif";
 		spdif-controller = <&spdif>;
 		spdif-out;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_cubox_i_spdif>;
 	};
 
         sound-hdmi {
