--- a/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-02-28 09:35:11.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-03-17 07:01:02.258818227 +0100
@@ -127,8 +127,6 @@
 };
 
 &hdmi {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hummingboard_hdmi>;
 	ddc = <&i2c2>;
 	status = "okay";
 	crtcs = <&ipu1 0>;
@@ -138,14 +136,8 @@
         status = "okay";
 };
 
-&hdmi_cec {
-        pinctrl-names = "default";
-        pinctrl-0 = <&pinctrl_hummingboard_hdmi>;
-        status = "okay";
-};
-
 &hdmi_core {
-        ipu_id = <1>;
+        ipu_id = <0>;
         disp_id = <0>;
         status = "okay";
 };
@@ -178,7 +170,7 @@
 	clock-frequency = <100000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c2_2>;
-	status = "okay";
+	status = "disable";
 };
 
 &iomuxc {
@@ -196,12 +188,6 @@
 			>;
 		};
 
-		pinctrl_hummingboard_hdmi: hummingboard-hdmi {
-			fsl,pins = <
-				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x1f8b0
-			>;
-		};
-
 		pinctrl_hummingboard_spdif: hummingboard-spdif {
 			fsl,pins = <MX6QDL_PAD_GPIO_17__SPDIF_OUT 0x1b0b0>;
 		};
