diff -Naur linux-imx_3.0.35_4.1.0-0f2006c/include/linux/fsl_devices.h linux-imx_3.0.35_4.1.0-0f2006c.patch/include/linux/fsl_devices.h
--- linux-imx_3.0.35_4.1.0-0f2006c/include/linux/fsl_devices.h	2014-01-12 18:38:34.000000000 +0100
+++ linux-imx_3.0.35_4.1.0-0f2006c.patch/include/linux/fsl_devices.h	2014-01-13 03:49:09.472692153 +0100
@@ -18,7 +18,10 @@
 #define _FSL_DEVICE_H_
 
 #include <linux/types.h>
+
+#ifdef __KERNEL__
 #include <linux/cdev.h>
+#endif /* __KERNEL__ */
 
 /*
  * Some conventions on how we handle peripherals on Freescale chips
