diff -Naur linux-cuboxi-a5afa23/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c linux-cuboxi-a5afa23.patch/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c
--- linux-cuboxi-a5afa23/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c	2014-08-10 15:00:10.000000000 +0200
+++ linux-cuboxi-a5afa23.patch/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_driver.c	2014-08-10 16:03:01.233093741 +0200
@@ -1327,7 +1327,7 @@
 };
 MODULE_DEVICE_TABLE(of, mxs_gpu_dt_ids);
 
-#ifdef CONFIG_PM
+#ifdef CONFIG_SUSPEND
 static int gpu_runtime_suspend(struct device *dev)
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 5, 7)
@@ -1377,7 +1377,7 @@
         .name   = DEVICE_NAME,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0)
 		.of_match_table = mxs_gpu_dt_ids,
-#if CONFIG_PM
+#ifdef CONFIG_SUSPEND
 		.pm		= &gpu_pm_ops,
 #endif
 #endif
