--- linux-linux-imx+bdde708ebf/drivers/mxc/ipu3/ipu_param_mem.h	2013-08-16 04:01:35.000000000 +0200
+++ linux-linux-imx+bdde708ebf/drivers/mxc/ipu3/ipu_param_mem.h-new	2013-10-16 07:49:35.916193147 +0200
@@ -382,7 +382,7 @@
 			ipu_ch_param_set_field(&params, 1, 78, 7, 15);  /* burst size */
 			uv_stride = uv_stride*2;
 		} else {
-			ipu_ch_param_set_field(&params, 1, 78, 7, 31);  /* burst size */
+			ipu_ch_param_set_field(&params, 1, 78, 7, 63);  /* burst size */
 		}
 		break;
 	case IPU_PIX_FMT_YVU420P:
