diff -Naur a/sound/soc/codecs/mxc_spdif.h b/sound/soc/codecs/mxc_spdif.h
--- a/sound/soc/codecs/mxc_spdif.h	2013-12-06 16:18:39.000000000 +0100
+++ b/sound/soc/codecs/mxc_spdif.h	2014-01-21 18:44:54.301081422 +0100
@@ -150,9 +150,13 @@
 					 SNDRV_PCM_RATE_64000 | \
 					 SNDRV_PCM_RATE_96000)
 
+// !!! temporary workaround for broken 24bit mode !!!
+#define MXC_SPDIF_FORMATS_PLAYBACK	(SNDRV_PCM_FMTBIT_S16_LE)
+/*
 #define MXC_SPDIF_FORMATS_PLAYBACK	(SNDRV_PCM_FMTBIT_S16_LE | \
 					 SNDRV_PCM_FMTBIT_S20_3LE | \
 					 SNDRV_PCM_FMTBIT_S24_LE)
+*/
 
 #define MXC_SPDIF_FORMATS_CAPTURE	(SNDRV_PCM_FMTBIT_S24_LE)
 
