diff -Naur linux-cuboxi-592b2d9/arch/arm/boot/dts/imx6dl-hummingboard.dts linux-cuboxi-592b2d9.patch/arch/arm/boot/dts/imx6dl-hummingboard.dts
--- linux-cuboxi-592b2d9/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-04-13 01:48:52.000000000 +0200
+++ linux-cuboxi-592b2d9.patch/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-04-21 19:18:25.799869731 +0200
@@ -23,6 +23,7 @@
 		gpios = <&gpio1 2 1>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_hummingboard_gpio1_2>;
+		linux,rc-map-name = "rc-rc6-mce";
 	};
 
 	regulators {
diff -Naur linux-cuboxi-592b2d9/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi linux-cuboxi-592b2d9.patch/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi
--- linux-cuboxi-592b2d9/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-04-13 01:48:52.000000000 +0200
+++ linux-cuboxi-592b2d9.patch/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-04-21 19:18:00.475889579 +0200
@@ -14,6 +14,7 @@
 		gpios = <&gpio3 9 1>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_cubox_i_ir>;
+		linux,rc-map-name = "rc-rc6-mce";
 	};
 
 	pwmleds {
