--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi	2014-08-26 14:31:43.000000000 +0200
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi	2014-08-27 22:14:04.000000000 +0200
@@ -21,6 +21,14 @@
 		mxcfb3 = &mxcfb4;
 	};
 
+	ir_recv: ir-receiver {
+		compatible = "gpio-ir-receiver";
+		gpios = <&gpio3 18 1>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_sabresd_ir>;
+		linux,rc-map-name = "rc-rc6-mce";
+	};
+	
 	chosen {
 		stdout-path = &uart1;
 	};
@@ -132,6 +140,12 @@
 		red {
 		        gpios = <&gpio1 2 0>;
 		        default-state = "on";
+		        linux,default-trigger = "heartbeat";
+		};
+
+		fan {
+			gpios = <&gpio3 28 0>;
+			default-state = "off";
 		};
 	};
 
@@ -475,6 +489,12 @@
 			>;
 		};
 
+		pinctrl_sabresd_ir: sabresd-ir {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D18__GPIO3_IO18 0x80000000
+			>;
+		};
+		
 		pinctrl_hdmi_cec: hdmi_cecgrp {
 			fsl,pins = <
 				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x1f8b0
@@ -618,6 +638,7 @@
 		pinctrl_gpio_leds: gpioledsgrp {
 			fsl,pins = <
 				MX6QDL_PAD_GPIO_2__GPIO1_IO02 0x80000000
+				MX6QDL_PAD_EIM_D28__GPIO3_IO28 0x80000000
 			>;
 		};
 	};
