diff -Naur linux-imx_3.0.35_4.1.0-0f2006c/arch/arm/mach-mx6/board-mx6q_cubox-i.c linux-imx_3.0.35_4.1.0-0f2006c.patch/arch/arm/mach-mx6/board-mx6q_cubox-i.c
--- linux-imx_3.0.35_4.1.0-0f2006c/arch/arm/mach-mx6/board-mx6q_cubox-i.c	2014-01-12 18:38:33.000000000 +0100
+++ linux-imx_3.0.35_4.1.0-0f2006c.patch/arch/arm/mach-mx6/board-mx6q_cubox-i.c	2014-01-22 22:28:14.749986987 +0100
@@ -53,6 +53,7 @@
 #include <linux/regulator/fixed.h>
 #ifdef CONFIG_IR_GPIO_CIR
 #include <media/gpio-ir-recv.h>
+#include <media/rc-map.h>
 #endif
 
 #include <mach/common.h>
@@ -257,6 +258,7 @@
 static struct gpio_ir_recv_platform_data cubox_i_ir_data = {
 	.gpio_nr = GPIO_IR_IN,
 	.active_low = 1,
+	.map_name = RC_MAP_RC6_MCE,
 };
 
 static struct platform_device cubox_i_ir = {
