--- a/tools/env/fw_env.c	2016-03-22 16:09:17.949471217 +0100
+++ b/tools/env/fw_env.c	2016-03-22 16:32:45.301980868 +0100
@@ -888,12 +888,13 @@
 				}
 		}
 
-		if (lseek (fd, blockstart, SEEK_SET) == -1) {
-			fprintf (stderr,
-				 "Seek error on %s: %s\n",
-				 DEVNAME (dev), strerror (errno));
-			return -1;
-		}
+		if (blockstart != 0)
+			if (lseek (fd, blockstart, SEEK_SET) == -1) {
+				fprintf (stderr,
+					 "Seek error on %s: %s\n",
+					 DEVNAME (dev), strerror (errno));
+				return -1;
+			}
 
 #ifdef DEBUG
 		fprintf(stderr, "Write 0x%x bytes at 0x%llx\n", erasesize,
@@ -1036,9 +1037,8 @@
 		    mtdinfo.type != MTD_NANDFLASH &&
 		    mtdinfo.type != MTD_DATAFLASH &&
 		    mtdinfo.type != MTD_UBIVOLUME) {
-			fprintf (stderr, "Unsupported flash type %u on %s\n",
-				 mtdinfo.type, DEVNAME(dev_current));
-			return -1;
+			memset(&mtdinfo, 0, sizeof(mtdinfo));
+			mtdinfo.type = MTD_ABSENT;
 		}
 	} else {
 		memset(&mtdinfo, 0, sizeof(mtdinfo));
