--- a/gtk/gen-gtk-gresources-xml.py	2021-02-24 19:13:19.000000000 +0000
+++ b/gtk/gen-gtk-gresources-xml.py	2021-04-03 23:52:35.000000000 +0000
@@ -23,11 +23,6 @@
     <file>theme/Adwaita/gtk-contained-dark.css</file>
 '''
 
-for f in get_files('theme/Adwaita/assets', '.png'):
-  xml += '    <file preprocess=\'to-pixdata\'>theme/Adwaita/assets/{0}</file>\n'.format(f)
-
-xml += '\n'
-
 for f in get_files('theme/Adwaita/assets', '.svg'):
   xml += '    <file>theme/Adwaita/assets/{0}</file>\n'.format(f)
 
@@ -38,11 +33,6 @@
     <file>theme/HighContrast/gtk-contained-inverse.css</file>
 '''
 
-for f in get_files('theme/HighContrast/assets', '.png'):
-  xml += '    <file preprocess=\'to-pixdata\'>theme/HighContrast/assets/{0}</file>\n'.format(f)
-
-xml += '\n'
-
 for f in get_files('theme/HighContrast/assets', '.svg'):
   xml += '    <file>theme/HighContrast/assets/{0}</file>\n'.format(f)
 
