diff --git a/intl/msg_hash_ar.h b/intl/msg_hash_ar.h
index 159212b..baca1af 100644
--- a/intl/msg_hash_ar.h
+++ b/intl/msg_hash_ar.h
@@ -1304,7 +1304,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB Rating")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Reboot")
+      "Reboot (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Recording Config")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_chs.h b/intl/msg_hash_chs.h
index b9a669a..c9ada96 100644
--- a/intl/msg_hash_chs.h
+++ b/intl/msg_hash_chs.h
@@ -1207,7 +1207,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB 评分")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "重启")
+      "重启 (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "录像配置目录")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_cht.h b/intl/msg_hash_cht.h
index 1de5b97..d774e78 100644
--- a/intl/msg_hash_cht.h
+++ b/intl/msg_hash_cht.h
@@ -1207,7 +1207,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB 評分")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "重啟")
+      "重啟 (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "錄影設定目錄")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_de.h b/intl/msg_hash_de.h
index 4b0c014..7696016 100644
--- a/intl/msg_hash_de.h
+++ b/intl/msg_hash_de.h
@@ -1230,7 +1230,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB-Bewertung")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Neustart")
+      "Neustart (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Aufnahme-Konfigurationen")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_eo.h b/intl/msg_hash_eo.h
index c9e686d..bba65ba 100644
--- a/intl/msg_hash_eo.h
+++ b/intl/msg_hash_eo.h
@@ -1109,7 +1109,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB Rating")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Reboot")
+      "Reboot (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Recording Config")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_es.h b/intl/msg_hash_es.h
index 73a94de..dc0f007 100644
--- a/intl/msg_hash_es.h
+++ b/intl/msg_hash_es.h
@@ -2067,7 +2067,7 @@ MSG_HASH(
 	)
 MSG_HASH(
 	MENU_ENUM_LABEL_VALUE_REBOOT,
-	"Reiniciar"
+	"Reiniciar (RCM)"
 	)
 MSG_HASH(
 	MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
diff --git a/intl/msg_hash_fr.h b/intl/msg_hash_fr.h
index 4a56fc7..39075d5 100644
--- a/intl/msg_hash_fr.h
+++ b/intl/msg_hash_fr.h
@@ -1218,7 +1218,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Classification TGDB")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Redémarrer")
+      "Redémarrer (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Configurations d'enregistrement vidéo")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_it.h b/intl/msg_hash_it.h
index d9ff765..15813b5 100644
--- a/intl/msg_hash_it.h
+++ b/intl/msg_hash_it.h
@@ -1230,7 +1230,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Valutazione TGDB ")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Riavvia")
+      "Riavvia (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Configurazione per la registrazione")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_ja.h b/intl/msg_hash_ja.h
index b907273..385daed 100644
--- a/intl/msg_hash_ja.h
+++ b/intl/msg_hash_ja.h
@@ -1312,7 +1312,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB評価")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "再起動")
+      "再起動 (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "録画設定")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_ko.h b/intl/msg_hash_ko.h
index 308f282..151687f 100644
--- a/intl/msg_hash_ko.h
+++ b/intl/msg_hash_ko.h
@@ -1204,7 +1204,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB 등급")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "재시작")
+      "재시작 (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "녹화 설정")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_nl.h b/intl/msg_hash_nl.h
index c996452..cf5ade3 100644
--- a/intl/msg_hash_nl.h
+++ b/intl/msg_hash_nl.h
@@ -1107,7 +1107,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB Rating")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Herstart")
+      "Herstart (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Opname Config Map")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_pl.h b/intl/msg_hash_pl.h
index e900485..98bd829 100644
--- a/intl/msg_hash_pl.h
+++ b/intl/msg_hash_pl.h
@@ -1308,7 +1308,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Ocena TGDB")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Restart")
+      "Restart (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Konfig. Nagrywania")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_pt_br.h b/intl/msg_hash_pt_br.h
index 7cc4f38..09c9aba 100644
--- a/intl/msg_hash_pt_br.h
+++ b/intl/msg_hash_pt_br.h
@@ -1312,7 +1312,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Classificação TGDB")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Reiniciar")
+      "Reiniciar (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Configuração de Gravação")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_pt_pt.h b/intl/msg_hash_pt_pt.h
index 3d0a4e7..7206f1c 100644
--- a/intl/msg_hash_pt_pt.h
+++ b/intl/msg_hash_pt_pt.h
@@ -1198,7 +1198,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Classificação TGDB")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Reinicializar")
+      "Reinicializar (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Pasta de armazenamento das configuração de gravação")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_ru.h b/intl/msg_hash_ru.h
index dfdc7c4..3c33cfb 100644
--- a/intl/msg_hash_ru.h
+++ b/intl/msg_hash_ru.h
@@ -1227,7 +1227,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "Рейтинг TGDB")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Перезагрузить")
+      "Перезагрузить (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Конфигурация записи")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_us.h b/intl/msg_hash_us.h
index 8df0d0d..c55f1be 100644
--- a/intl/msg_hash_us.h
+++ b/intl/msg_hash_us.h
@@ -1312,7 +1312,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB Rating")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Reboot")
+      "Reboot (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Recording Config")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/intl/msg_hash_vn.h b/intl/msg_hash_vn.h
index 4df2bb2..1012886 100644
--- a/intl/msg_hash_vn.h
+++ b/intl/msg_hash_vn.h
@@ -1218,7 +1218,7 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_START_CONTENT,
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RDB_ENTRY_TGDB_RATING,
       "TGDB Rating")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_REBOOT,
-      "Khởi động lại")
+      "Khởi động lại (RCM)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_CONFIG_DIRECTORY,
       "Recording Config")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_RECORDING_OUTPUT_DIRECTORY,
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index eabdc98..7685a13 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -2208,14 +2208,14 @@ static bool setting_append_list(
                parent_group);
          menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_REBOOT);
 
-         CONFIG_ACTION(
+         /*CONFIG_ACTION(
                list, list_info,
                MENU_ENUM_LABEL_SHUTDOWN,
                MENU_ENUM_LABEL_VALUE_SHUTDOWN,
                &group_info,
                &subgroup_info,
                parent_group);
-         menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_SHUTDOWN);
+         menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_SHUTDOWN);*/
 #endif
 
          CONFIG_ACTION(
