diff --git a/wifi/drivers/connmanctl.c b/wifi/drivers/connmanctl.c
index 2909ec3..992fe9c 100644
--- a/wifi/drivers/connmanctl.c
+++ b/wifi/drivers/connmanctl.c
@@ -93,6 +93,13 @@ static void connmanctl_get_ssids(struct string_list* ssids)
       strlcpy(ssid, line+4, sizeof(ssid));
       string_list_append(ssids, ssid, attr);
    }
+   
+   // if there is only one network and its SSID is not printable
+   // then a reboot is needed to fix Wi-Fi
+   if (ssids->size == 1 && strlen(ssids->elems[0].data) >= 1 && !isprint(ssids->elems[0].data[0]))
+   {
+   	runloop_msg_queue_push("Please reboot to enable Wi-Fi", 1, 180, true);
+   }
 }
 
 static bool connmanctl_ssid_is_online(unsigned i)
