diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index 95c5470459..4bc69e3a0b 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -1085,7 +1085,11 @@ static bool frontend_unix_powerstate_check_acpi_sysfs(
    {
       const char *node = retro_dirent_get_name(entry);
 
+#ifdef HAVE_LAKKA_SWITCH
+      if (node && strstr(node, "max170xx_battery"))
+#else
       if (node && strstr(node, "BAT"))
+#endif
          check_proc_acpi_sysfs_battery(node,
                &have_battery, &charging, seconds, percent);
    }
