diff -Naur xbmc-addon-xvdr-8d57661/src/xvdr/XVDRSession.cpp xbmc-addon-xvdr-8d57661.patch/src/xvdr/XVDRSession.cpp
--- xbmc-addon-xvdr-8d57661/src/xvdr/XVDRSession.cpp	2012-03-22 13:38:00.000000000 +0100
+++ xbmc-addon-xvdr-8d57661.patch/src/xvdr/XVDRSession.cpp	2012-03-30 04:04:13.345131222 +0200
@@ -27,6 +27,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #include "responsepacket.h"
 #include "requestpacket.h"
