diff -Naur squashfs-a56f344/squashfs-tools/Makefile squashfs-a56f344.patch/squashfs-tools/Makefile
--- squashfs-a56f344/squashfs-tools/Makefile	2011-02-21 23:58:19.000000000 +0100
+++ squashfs-a56f344.patch/squashfs-tools/Makefile	2011-02-22 00:05:33.197451875 +0100
@@ -77,7 +77,7 @@
 # If your C library or build/target environment doesn't support XATTRs then
 # comment out the next line to build Mksquashfs and Unsquashfs without XATTR
 # support
-XATTR_SUPPORT = 1
+#XATTR_SUPPORT = 1
 
 # Select whether you wish xattrs to be stored by Mksquashfs and extracted
 # by Unsquashfs by default.  If selected users can disable xattr support by
@@ -85,7 +85,7 @@
 #
 # If unselected, Mksquashfs/Unsquashfs won't store and extract xattrs by
 # default.  Users can enable xattrs by using the -xattrs option.
-XATTR_DEFAULT = 1
+#XATTR_DEFAULT = 1
 
 
 ###############################################
@@ -100,7 +100,7 @@
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
 	unsquash-4.o swap.o compressor.o
 
-CFLAGS = $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
+CFLAGS += $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"  -O2 -Wall
 
 LIBS = -lpthread -lm
@@ -196,7 +196,7 @@
 all: mksquashfs unsquashfs
 
 mksquashfs: $(MKSQUASHFS_OBJS)
-	$(CC) $(MKSQUASHFS_OBJS) $(LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MKSQUASHFS_OBJS) $(LIBS) -o $@
 
 mksquashfs.o: mksquashfs.c squashfs_fs.h mksquashfs.h sort.h squashfs_swap.h \
 	xattr.h pseudo.h compressor.h
@@ -227,7 +227,7 @@
 xz_wrapper.o: xz_wrapper.c compressor.h squashfs_fs.h
 
 unsquashfs: $(UNSQUASHFS_OBJS)
-	$(CC) $(UNSQUASHFS_OBJS) $(LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(UNSQUASHFS_OBJS) $(LIBS) -o $@
 
 unsquashfs.o: unsquashfs.h unsquashfs.c squashfs_fs.h squashfs_swap.h \
 	squashfs_compat.h xattr.h read_fs.h compressor.h
