diff -Naur squashfs4.2/squashfs-tools/Makefile squashfs4.2.patch/squashfs-tools/Makefile
--- squashfs4.2/squashfs-tools/Makefile	2011-02-28 21:04:15.000000000 +0100
+++ squashfs4.2.patch/squashfs-tools/Makefile	2011-03-03 04:07:03.566708237 +0100
@@ -78,7 +78,7 @@
 # If your C library or build/target environment doesn't support XATTRs then
 # comment out the next line to build Mksquashfs and Unsquashfs without XATTR
 # support
-XATTR_SUPPORT = 1
+#XATTR_SUPPORT = 1
 
 # Select whether you wish xattrs to be stored by Mksquashfs and extracted
 # by Unsquashfs by default.  If selected users can disable xattr support by
@@ -86,7 +86,7 @@
 #
 # If unselected, Mksquashfs/Unsquashfs won't store and extract xattrs by
 # default.  Users can enable xattrs by using the -xattrs option.
-XATTR_DEFAULT = 1
+#XATTR_DEFAULT = 1
 
 
 ###############################################
