diff -Naur squashfs4.0-20091205/squashfs-tools/Makefile squashfs4.0-20091205.patch/squashfs-tools/Makefile
--- squashfs4.0-20091205/squashfs-tools/Makefile	2009-10-14 05:32:57.000000000 +0200
+++ squashfs4.0-20091205.patch/squashfs-tools/Makefile	2009-12-05 18:24:48.241450813 +0100
@@ -2,8 +2,8 @@
 # Building LZMA support
 # Download LZMA sdk (4.65 used in development, other versions may work),
 # set LZMA_DIR to unpacked source, and uncomment next line
-LZMA_SUPPORT = 1
-LZMA_DIR = ../../../LZMA/lzma465
+#LZMA_SUPPORT = 1
+#LZMA_DIR = ../../../LZMA/lzma465
 
 #Compression default.
 COMP_DEFAULT = gzip
