diff -Naur elisa-0.5.31/elisa/core/plugin_registry.py elisa-0.5.31.patch/elisa/core/plugin_registry.py
--- elisa-0.5.31/elisa/core/plugin_registry.py	2009-03-09 09:31:34.000000000 +0100
+++ elisa-0.5.31.patch/elisa/core/plugin_registry.py	2009-03-09 20:32:10.000000000 +0100
@@ -156,8 +156,8 @@
     default_config = \
         {'repository': 'http://elisa-plugins.fluendo.com/plugin_list',
          'update_plugin_cache': True,
-         'auto_update_plugins': True,
-         'auto_install_new_recommended_plugins': True}
+         'auto_update_plugins': False,
+         'auto_install_new_recommended_plugins': False}
     config_doc = \
         {'repository': 'The plugin repository to query for new plugins and ' \
             'plugin updates.',
