diff -Naur linux-3.17.3/drivers/media/rc/ir-rc6-decoder.c linux-3.17.3.patch/drivers/media/rc/ir-rc6-decoder.c
--- linux-3.17.3/drivers/media/rc/ir-rc6-decoder.c	2014-11-14 19:10:57.000000000 +0100
+++ linux-3.17.3.patch/drivers/media/rc/ir-rc6-decoder.c	2014-11-20 00:59:40.082608912 +0100
@@ -259,8 +259,8 @@
 			case 32:
 				if ((scancode & RC6_6A_LCC_MASK) == RC6_6A_MCE_CC) {
 					protocol = RC_TYPE_RC6_MCE;
-					scancode &= ~RC6_6A_MCE_TOGGLE_MASK;
 					toggle = !!(scancode & RC6_6A_MCE_TOGGLE_MASK);
+					scancode &= ~RC6_6A_MCE_TOGGLE_MASK;
 				} else {
 					protocol = RC_BIT_RC6_6A_32;
 					toggle = 0;
