From 7460d8ebca84bd5e7c26e88143f14f4909598b68 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <skeggsb@gmail.com>
Date: Mon, 13 Apr 2009 20:20:39 +1000
Subject: [PATCH 5/6] nv50/f11: disable acceleration on NVAx chipsets

---
 src/nv_driver.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/src/nv_driver.c b/src/nv_driver.c
index a280257..f097fb9 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -1144,6 +1144,14 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
 		pNv->ShadowFB = TRUE;
 	}
 
+	if ((pNv->NVArch & 0xf0) == 0xa0) {
+		xf86DrvMsg(pScrn->scrnIndex, X_NOTICE,
+			   "Acceleration disabled, currently non-functional "
+			   "on this chipset\n");
+		pNv->NoAccel = TRUE;
+		pNv->ShadowFB = TRUE;
+	}
+
 	/* Save current console video mode */
 	if (pNv->Architecture >= NV_ARCH_50 && pNv->pInt10 && !pNv->kms_enable) {
 		const xf86Int10InfoPtr pInt10 = pNv->pInt10;
-- 
1.6.2.2

