diff --git a/src/nv_bios.c b/src/nv_bios.c
index fe6368e..7b56148 100644
--- a/src/nv_bios.c
+++ b/src/nv_bios.c
@@ -4104,6 +4104,8 @@ parse_dcb20_entry(ScrnInfoPtr pScrn, struct bios_parsed_dcb *bdcb,
 			mask = ~0x5;
 			if (conf & 0x4)
 				entry->lvdsconf.use_power_scripts = true;
+			if ((conf & 0xf0) == 0x60)
+				mask &= ~0xf0;
 		}
 		if (conf & mask) {
 			/* I'm bored of getting this reported; left as a reminder for someone to fix it */
