diff -Naur mame-0138s/makefile mame-0138s.patch/makefile
--- mame-0138s/makefile	2010-05-13 11:12:54.000000000 +0200
+++ mame-0138s.patch/makefile	2010-07-13 01:04:52.722999679 +0200
@@ -206,10 +206,10 @@
 # BIGENDIAN = 1
 
 # uncomment next line to build expat as part of MAME build
-BUILD_EXPAT = 1
+# BUILD_EXPAT = 1
 
 # uncomment next line to build zlib as part of MAME build
-BUILD_ZLIB = 1
+# BUILD_ZLIB = 1
 
 # uncomment next line to include the symbols
 # SYMBOLS = 1
@@ -439,7 +439,7 @@
 ifneq ($(OPTIMIZE),0)
 ifneq ($(TARGETOS),os2)
 ifndef NOWERROR
-CCOMFLAGS += -Werror -fno-strict-aliasing $(ARCHOPTS)
+CCOMFLAGS += -fno-strict-aliasing $(ARCHOPTS)
 else
 CCOMFLAGS += -fno-strict-aliasing $(ARCHOPTS)
 endif
@@ -706,12 +706,12 @@
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(FILE2STR)
 	@echo Converting $<...
-	@$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
+	@mame-file2str $< $@ layout_$(basename $(notdir $<))
 
 $(OBJ)/%.fh: $(SRC)/%.png $(PNG2BDC) $(FILE2STR)
 	@echo Converting $<...
-	@$(PNG2BDC) $< $(OBJ)/temp.bdc
-	@$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
+	@mame-png2bdc  $< $(OBJ)/temp.bdc
+	@mame-file2str $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
 
 $(OBJ)/%.a:
 	@echo Archiving $@...
diff -Naur mame-0138s/src/emu/cpu/cpu.mak mame-0138s.patch/src/emu/cpu/cpu.mak
--- mame-0138s/src/emu/cpu/cpu.mak	2010-02-03 22:20:20.000000000 +0100
+++ mame-0138s.patch/src/emu/cpu/cpu.mak	2010-07-12 23:04:03.416999399 +0200
@@ -1019,7 +1019,7 @@
 # rule to generate the C files
 $(CPUOBJ)/m68000/m68kops.c: $(M68KMAKE) $(CPUSRC)/m68000/m68k_in.c
 	@echo Generating M68K source files...
-	$(M68KMAKE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
+	mame-m68kmake$(EXE) $(CPUOBJ)/m68000 $(CPUSRC)/m68000/m68k_in.c
 
 # rule to build the generator
 ifneq ($(CROSS_BUILD),1)
@@ -1548,7 +1548,7 @@
 # rule to generate the C file
 $(CPUOBJ)/tms57002/tms57002.inc: $(TMSMAKE) $(CPUSRC)/tms57002/tmsinstr.lst
 	@echo Generating TMS57002 source file...
-	$(TMSMAKE) $(CPUSRC)/tms57002/tmsinstr.lst $@
+	mame-tmsmake$(EXE) $(CPUSRC)/tms57002/tmsinstr.lst $@
 
 # rule to build the generator
 ifneq ($(CROSS_BUILD),1)
diff -Naur mame-0138s/src/osd/sdl/sdl.mak mame-0138s.patch/src/osd/sdl/sdl.mak
--- mame-0138s/src/osd/sdl/sdl.mak	2010-02-14 04:47:30.000000000 +0100
+++ mame-0138s.patch/src/osd/sdl/sdl.mak	2010-07-13 01:07:47.158999892 +0200
@@ -454,14 +454,14 @@
 
 # the new debugger relies on GTK+ in addition to the base SDLMAME needs
 # Non-X11 builds can not use the debugger
-CCOMFLAGS += `pkg-config --cflags gtk+-2.0` `pkg-config --cflags gconf-2.0`
-LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
+#CCOMFLAGS += `pkg-config --cflags gtk+-2.0` `pkg-config --cflags gconf-2.0`
+#LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
 #CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
 
 # some systems still put important things in a different prefix
-LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
+#LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
 # make sure we can find X headers
-CCOMFLAGS += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
+# CCOMFLAGS += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
 endif # NO_X11
 
 #-------------------------------------------------
