diff --git a/src/input_common/udp/client.cpp b/src/input_common/udp/client.cpp
index ce2fb78c..c173c690 100644
--- a/src/input_common/udp/client.cpp
+++ b/src/input_common/udp/client.cpp
@@ -97,7 +97,7 @@ private:
         auto pad_message = Request::Create(pad_data, client_id);
         std::memcpy(send_buffer2.data(), &pad_message, PAD_DATA_SIZE);
         std::size_t len2 = socket.send_to(boost::asio::buffer(send_buffer2), send_endpoint);
-        StartSend(timer.expiry());
+        StartSend(timer.expires_at());
     }
 
     SocketCallback callback;
