diff --git a/Makefile b/Makefile
index 3be779d..0c822bf 100644
--- a/Makefile
+++ b/Makefile
@@ -281,7 +281,7 @@ endif
 
 # Set ffmpeg configure options
 ifeq ($(HAVE_FFMPEG_STATIC), 1)
-FFMPEG_CONF_OPTS =--disable-encoders --disable-decoders --enable-decoder=aac --enable-decoder=aac_fixed --enable-decoder=aac_latm --disable-programs
+FFMPEG_CONF_OPTS =--disable-encoders --disable-decoders --enable-decoder=aac --enable-decoder=aac_fixed --enable-decoder=aac_latm --disable-programs --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages
 ifeq ($(FFMPEG_DISABLE_VDPAU), 1)
 FFMPEG_CONF_OPTS += --disable-vdpau
 endif
