diff -Naur efibootmgr-ceb177a/src/lib/scsi_ioctls.c efibootmgr-ceb177a.patch/src/lib/scsi_ioctls.c
--- efibootmgr-ceb177a/src/lib/scsi_ioctls.c	2014-10-21 22:17:27.000000000 +0200
+++ efibootmgr-ceb177a.patch/src/lib/scsi_ioctls.c	2016-03-26 13:49:59.119669675 +0100
@@ -24,7 +24,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
-#include <linux/nvme.h>
+#include <linux/nvme_ioctl.h>
 #include "scsi_ioctls.h"
 
 int
