diff -Naur retroarch.git/menu/drivers/xmb.c retroarch.patch/menu/drivers/xmb.c
--- retroarch.git/menu/drivers/xmb.c	2017-01-07 21:21:53.262938761 +0100
+++ retroarch.patch/menu/drivers/xmb.c	2017-01-07 21:23:41.962790836 +0100
@@ -2170,7 +2170,8 @@
       menu_animation_ctl(MENU_ANIMATION_CTL_TICKER, &ticker);
 
       label_offset = xmb->margins.label.top;
-      if (i == current && menu_entry_get_sublabel(i, entry_sublabel, sizeof(entry_sublabel)))
+      /*if (i == current && menu_entry_get_sublabel(i, entry_sublabel, sizeof(entry_sublabel)))*/
+      if (0)
       {
          label_offset = - xmb->margins.label.top;
 
