From edbabcc7df6cd9e282991bd550d125d7f75bd738 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Mon, 22 Jun 2020 19:42:17 +0200
Subject: [PATCH] Fix building with gcc10

---
 si_types.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/si_types.h b/si_types.h
index 5fc7835d80c7..41ce2e1e6603 100644
--- a/si_types.h
+++ b/si_types.h
@@ -114,7 +114,7 @@ struct service {
   uint32_t logical_channel_number;
   uint8_t  running;
   void   * priv;
-} service_t, * p_service_t;
+};
 
 /*******************************************************************************
 /* transponder type.
@@ -201,7 +201,7 @@ struct transponder {
   /*----------------------------*/
   char * network_name;
   network_change_t network_change;
-} __attribute__((packed))  transponder_t, * p_transponder_t;
+} __attribute__((packed));
 
 /*******************************************************************************
 /* satellite channel routing type.
-- 
2.27.0

