From a2196e61c14dd0437754406d007ae6e8273eb9a6 Mon Sep 17 00:00:00 2001
From: Anthony Nash <nash.ant@gmail.com>
Date: Wed, 20 Apr 2011 16:00:07 +0100
Subject: [PATCH 2/2] Set defaults

---
 app/config/configApp.py |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/app/config/configApp.py b/app/config/configApp.py
index fb0e2d0..8247df6 100644
--- a/app/config/configApp.py
+++ b/app/config/configApp.py
@@ -39,11 +39,11 @@ class configApp():
         self.addSection('global')
         self.setDefault('global', 'server.environment', 'production')
         self.setDefault('global', 'host', '0.0.0.0')
-        self.setDefault('global', 'port', 5000)
+        self.setDefault('global', 'port', 8083)
         self.setDefault('global', 'username', '')
         self.setDefault('global', 'password', '')
-        self.setDefault('global', 'launchbrowser', True)
-        self.setDefault('global', 'updater', True)
+        self.setDefault('global', 'launchbrowser', False)
+        self.setDefault('global', 'updater', False)
         self.setDefault('global', 'git', 'git')
         self.setDefault('global', 'urlBase', '')
         self.setDefault('global', 'ignoreWords', '')
-- 
1.7.4.4

