--- a/desmume/src/frontend/libretro/Makefile.libretro	2018-09-09 17:25:30.299207550 +0200
+++ b/desmume/src/frontend/libretro/Makefile.libretro	2018-09-09 17:21:15.835543000 +0200
@@ -45,10 +45,10 @@
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=$(CORE_DIR)/frontend/libretro/link.T
-   LIBS := -lpthread -lGL -lpcap
-   CXXFLAGS += -DHAVE_OPENGL -std=gnu++11
-   DESMUME_OPENGL = 1
-	DESMUME_OPENGL_CORE = 1
+   LIBS := -lpthread -lpcap
+   CXXFLAGS += -std=gnu++11
+   # DESMUME_OPENGL = 1
+	# DESMUME_OPENGL_CORE = 1
 
    ifneq (,$(findstring clang,$(platform)))
       CC = clang
