diff -Naur feedparser-4.1/setup.py feedparser-4.1.patch/setup.py
--- feedparser-4.1/setup.py	2006-01-10 19:13:27.000000000 +0100
+++ feedparser-4.1.patch/setup.py	2010-10-19 01:30:43.326660941 +0200
@@ -1,4 +1,4 @@
-from distutils.core import setup
+from setuptools import setup
 
 # patch distutils if it can't cope with the "classifiers" or "download_url"
 # keywords (prior to python 2.3.0).
