diff -Naur xbmc-rpi-c767513/xbmc/windows/GUIWindowSystemInfo.cpp xbmc-rpi-c767513.patch/xbmc/windows/GUIWindowSystemInfo.cpp
--- xbmc-rpi-c767513/xbmc/windows/GUIWindowSystemInfo.cpp	2012-05-17 13:59:53.000000000 +0200
+++ xbmc-rpi-c767513.patch/xbmc/windows/GUIWindowSystemInfo.cpp	2012-05-22 03:28:07.651129799 +0200
@@ -136,11 +136,14 @@
     SetControlLabel(i++, "%s %s", 22023, SYSTEM_RENDER_VENDOR);
     SetControlLabel(i++, "%s %s", 22024, SYSTEM_RENDER_VERSION);
 #endif
+#ifndef TARGET_RASPBERRY_PI
     SetControlLabel(i++, "%s %s", 22010, SYSTEM_GPU_TEMPERATURE);
+#endif
   }
   else if (m_section == CONTROL_BT_HARDWARE)
   {
     SET_CONTROL_LABEL(40,g_localizeStrings.Get(20160));
+#ifndef TARGET_RASPBERRY_PI
 #ifdef HAS_SYSINFO
     SET_CONTROL_LABEL(i++, g_sysinfo.GetXBVerInfo());
     SetControlLabel(i++, "%s %s", 22011, SYSTEM_CPU_TEMPERATURE);
@@ -148,6 +151,7 @@
 #endif
     SetControlLabel(i++, "%s %s", 13271, SYSTEM_CPU_USAGE);
     i++; // empty line
+#endif
     SetControlLabel(i++, "%s: %s", 22012, SYSTEM_TOTAL_MEMORY);
     SetControlLabel(i++, "%s: %s", 158, SYSTEM_FREE_MEMORY);
   }
