From 23ce92d80eebf17f22f352045163ceda896682f8 Mon Sep 17 00:00:00 2001
From: gimli <gimli@xbmc.org>
Date: Sat, 2 Jun 2012 00:18:25 +0200
Subject: [PATCH] [rbp] disabled supported codec check. doesn't work on the PI

---
 xbmc/cores/omxplayer/OMXVideo.cpp |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/xbmc/cores/omxplayer/OMXVideo.cpp b/xbmc/cores/omxplayer/OMXVideo.cpp
index b29859b..7e4c097 100644
--- a/xbmc/cores/omxplayer/OMXVideo.cpp
+++ b/xbmc/cores/omxplayer/OMXVideo.cpp
@@ -290,6 +290,7 @@ bool COMXVideo::Open(CDVDStreamInfo &hints, OMXClock *clock, bool deinterlace, b
   }
 
   OMX_VIDEO_PARAM_PORTFORMATTYPE formatType;
+  /*
   OMX_INIT_STRUCTURE(formatType);
   formatType.nPortIndex = m_omx_decoder.GetInputPort();
   OMX_U32 nIndex = 1;
@@ -311,10 +312,10 @@ bool COMXVideo::Open(CDVDStreamInfo &hints, OMXClock *clock, bool deinterlace, b
 
   if(!bFound)
   {
-    CLog::Log(LOGINFO, "COMXVideo::Open coding : %s supported\n", m_video_codec_name.c_str());
-    printf("COMXVideo::Open coding : %s supported\n", m_video_codec_name.c_str());
+    CLog::Log(LOGINFO, "COMXVideo::Open coding : %s not supported\n", m_video_codec_name.c_str());
     return false;
   }
+  */
 
   if(clock == NULL)
     return false;
-- 
1.7.10

