diff -Naur xbmc-rpi-c2c161c/xbmc/guilib/Texture.h xbmc-rpi-c2c161c.patch/xbmc/guilib/Texture.h
--- xbmc-rpi-c2c161c/xbmc/guilib/Texture.h	2012-04-26 16:26:57.000000000 +0200
+++ xbmc-rpi-c2c161c.patch/xbmc/guilib/Texture.h	2012-04-29 01:54:25.479859785 +0200
@@ -45,6 +45,7 @@
 #endif
 
 #ifdef TARGET_RASPBERRY_PI
+#define EGL_EGLEXT_PROTOTYPES
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
 #include "xbmc/filesystem/File.h"
