diff -Naur a/xbmc/filesystem/CurlFile.cpp b/xbmc/filesystem/CurlFile.cpp
--- a/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp	2012-07-03 14:34:17.000000000 +0100
+++ b/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp	2012-07-03 14:49:14.000000000 +0100
@@ -227,6 +227,12 @@
   VECPLAYERCORES vecCores;
   GetPlayers(item, vecCores);
 
+#if defined(HAVE_OMXPLAYER)
+    for (int i = 0; i < vecCores.size();i++)
+        if (vecCores.at(i) == PCID_OMXPLAYER)
+            return EPC_OMXPLAYER;
+#endif
+
   //If we have any players return the first one
   if( vecCores.size() > 0 ) return vecCores.at(0);

