diff -Naur xbmc-rpi-0bcbaa3/lib/DllAvFormat.h xbmc-rpi-0bcbaa3.patch/lib/DllAvFormat.h
--- xbmc-rpi-0bcbaa3/lib/DllAvFormat.h	2012-06-01 21:10:03.000000000 +0200
+++ xbmc-rpi-0bcbaa3.patch/lib/DllAvFormat.h	2012-06-05 07:06:49.123867584 +0200
@@ -119,7 +119,7 @@
   virtual int url_feof(AVIOContext *s) { return ::url_feof(s); }
   virtual void avformat_close_input(AVFormatContext **s) { ::avformat_close_input(s); }
   virtual int av_read_frame(AVFormatContext *s, AVPacket *pkt) { return ::av_read_frame(s, pkt); }
-  virtual void av_read_frame_flush(AVFormatContext *s) { ::av_read_frame_flush_internal(s); }
+  virtual void av_read_frame_flush(AVFormatContext *s) { ::av_read_frame_flush(s); }
   virtual int av_read_play(AVFormatContext *s) { return ::av_read_play(s); }
   virtual int av_read_pause(AVFormatContext *s) { return ::av_read_pause(s); }
   virtual int av_seek_frame(AVFormatContext *s, int stream_index, int64_t timestamp, int flags) { return ::av_seek_frame(s, stream_index, timestamp, flags); }
