diff -Naur xbmc-rpi-cc57efc/configure.in xbmc-rpi-cc57efc.patch/configure.in
--- xbmc-rpi-cc57efc/configure.in	2012-02-02 01:42:56.000000000 +0100
+++ xbmc-rpi-cc57efc.patch/configure.in	2012-02-02 22:33:23.641531080 +0100
@@ -872,11 +872,12 @@
     PKG_CHECK_MODULES([ALSA],  [alsa],
       [INCLUDES="$INCLUDES $ALSA_CFLAGS"; LIBS="$LIBS $ALSA_LIBS"; use_alsa=yes],
       AC_MSG_NOTICE($alsa_not_found); use_alsa=no)
+  fi
+
     PKG_CHECK_MODULES([DBUS],    [dbus-1],
       [INCLUDES="$INCLUDES $DBUS_CFLAGS"; LIBS="$LIBS $DBUS_LIBS"; use_dbus=yes]; \
       AC_DEFINE([HAVE_DBUS],[1],["Define to 1 if dbus is installed"]),
       AC_MSG_NOTICE($missing_library); use_dbus=no)
-  fi
 
   if test "x$use_sdl" != "xno"; then
     PKG_CHECK_MODULES([SDL],   [sdl],
