diff -Naur xbmc-rpi-c767513/xbmc/TextureCache.cpp xbmc-rpi-c767513.patch/xbmc/TextureCache.cpp
--- xbmc-rpi-c767513/xbmc/TextureCache.cpp	2012-05-17 13:59:46.000000000 +0200
+++ xbmc-rpi-c767513.patch/xbmc/TextureCache.cpp	2012-05-22 18:41:37.116511846 +0200
@@ -92,7 +92,7 @@
   CTextureDetails details;
   if (GetCachedTexture(url, details))
   {
-    IncrementUseCount(details);
+//    IncrementUseCount(details);
     return GetCachedPath(details.file);
   }
   return "";
