diff -Naur xbmc-rpi-c2c161c/xbmc/cores/omxplayer/OMXAudioCodecOMX.cpp xbmc-rpi-c2c161c.patch/xbmc/cores/omxplayer/OMXAudioCodecOMX.cpp
--- xbmc-rpi-c2c161c/xbmc/cores/omxplayer/OMXAudioCodecOMX.cpp	2012-04-26 16:26:58.000000000 +0200
+++ xbmc-rpi-c2c161c.patch/xbmc/cores/omxplayer/OMXAudioCodecOMX.cpp	2012-04-26 18:15:42.988848797 +0200
@@ -310,7 +310,7 @@
   else
   {
     CLog::Log(LOGINFO, "COMXAudioCodecOMX::GetChannelMap - FFmpeg reported %d channels, but the layout contains %d ignoring", m_pCodecContext->channels, bits);
-    layout = m_dllAvCodec.avcodec_guess_channel_layout(m_pCodecContext->channels, m_pCodecContext->codec_id, NULL);
+    layout = m_dllAvUtil.av_get_default_channel_layout(m_pCodecContext->channels);
   }
 
   int index = 0;
