--- a/tools/llvm-config/llvm-config.cpp	2019-12-15 20:54:21.898634608 +0100
+++ b/tools/llvm-config/llvm-config.cpp	2019-12-15 20:56:12.280223943 +0100
@@ -285,6 +285,10 @@
   CurrentExecPrefix =
       sys::path::parent_path(sys::path::parent_path(CurrentPath)).str();
 
+#ifdef LLVM_CONFIG_EXEC_PREFIX
+  CurrentExecPrefix = LLVM_CONFIG_EXEC_PREFIX;
+#endif
+
   // Check to see if we are inside a development tree by comparing to possible
   // locations (prefix style or CMake style).
   if (sys::fs::equivalent(CurrentExecPrefix, LLVM_OBJ_ROOT)) {
