diff -Naur llvm-3.5.0.src/Makefile.rules llvm-3.5.0.src.patch/Makefile.rules
--- llvm-3.5.0.src/Makefile.rules	2014-06-20 21:00:41.000000000 +0200
+++ llvm-3.5.0.src.patch/Makefile.rules	2014-09-05 11:04:20.267554714 +0200
@@ -594,16 +594,6 @@
   endif
 endif
 
-ifdef SHARED_LIBRARY
-ifneq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
-ifneq ($(HOST_OS),Darwin)
-  LD.Flags += $(RPATH) -Wl,'$$ORIGIN'
-else
-  LD.Flags += -Wl,-install_name  -Wl,"@rpath/lib$(LIBRARYNAME)$(SHLIBEXT)"
-endif
-endif
-endif
-
 ifdef TOOL_VERBOSE
   C.Flags += -v
   CXX.Flags += -v
@@ -650,14 +640,7 @@
   ifndef TOOL_NO_EXPORTS
     LD.Flags += $(RDYNAMIC)
   endif
-  ifneq ($(HOST_OS), Darwin)
-    ifdef TOOLNAME
-      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
-    endif
-  else
-    ifneq ($(DARWIN_MAJVERS),4)
-      LD.Flags += $(RPATH) -Wl,@executable_path/../lib
-    endif
+  ifeq ($(HOST_OS), Darwin)
     ifeq ($(RC_XBS),YES)
       TempFile := $(shell mkdir -p ${OBJROOT}/dSYMs ; mktemp ${OBJROOT}/dSYMs/llvm-lto.XXXXXX)
       LD.Flags += -Wl,-object_path_lto -Wl,$(TempFile)
