diff -Naur llvm.src/Makefile.rules llvm.src.patch/Makefile.rules
--- llvm.src/Makefile.rules	2013-11-15 00:51:29.000000000 +0100
+++ llvm.src.patch/Makefile.rules	2014-01-04 16:33:49.254066226 +0100
@@ -577,14 +577,6 @@
   endif
 endif
 
-ifdef SHARED_LIBRARY
-ifneq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
-ifneq ($(HOST_OS),Darwin)
-  LD.Flags += $(RPATH) -Wl,'$$ORIGIN'
-endif
-endif
-endif
-
 ifdef TOOL_VERBOSE
   C.Flags += -v
   CXX.Flags += -v
@@ -613,14 +605,7 @@
   ifndef TOOL_NO_EXPORTS
     LD.Flags += $(RDYNAMIC)
   endif
-  ifneq ($(HOST_OS), Darwin)
-    ifdef TOOLNAME
-      LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
-    endif
-  else
-    ifneq ($(DARWIN_MAJVERS),4)
-      LD.Flags += $(RPATH) -Wl,@executable_path/../lib
-    endif
+  ifeq ($(HOST_OS), Darwin)
     ifeq ($(RC_XBS),YES)
       TempFile := $(shell mkdir -p ${OBJROOT}/dSYMs ; mktemp ${OBJROOT}/dSYMs/llvm-lto.XXXXXX)
       LD.Flags += -Wl,-object_path_lto -Wl,$(TempFile)
