diff --git a/desmume/Makefile.libretro b/desmume/Makefile.libretro
index 7e8f3d20..b6bbbe18 100644
--- a/desmume/Makefile.libretro
+++ b/desmume/Makefile.libretro
@@ -39,7 +39,16 @@ ifneq (,$(findstring unix,$(platform)))
       CXX = clang++
       AR = clang
    endif
-
+   
+   # ARM64
+   ifneq (,$(findstring arm64,$(platform)))
+      DESMUME_JIT = 0
+      DESMUME_JIT_ARM = 0
+      override TARGET_ARCH := arm64
+      CXXFLAGS += -DUSE_POSIX_MEMALIGN
+      CXXFLAGS += -DARM64
+   endif
+   
    # Generic ARM
    ifneq (,$(findstring armv,$(platform)))
       DESMUME_JIT_ARM = 1
