diff -Naur xresprobe-0.4.24ubuntu3.orig/lcdsize.sh xresprobe-0.4.24ubuntu3/lcdsize.sh
--- xresprobe-0.4.24ubuntu3.orig/lcdsize.sh	2007-05-08 18:17:18.000000000 +0200
+++ xresprobe-0.4.24ubuntu3/lcdsize.sh	2007-05-08 18:17:29.000000000 +0200
@@ -21,7 +21,7 @@
 # License, version 2, can be found in /usr/share/common-licenses/GPL-2.
 
 getres () {
-  RESLINE="$(egrep "$EGREPLINE" "$LOGFILE" | head -1)"
+  RESLINE="$(egrep "$EGREPLINE" "$LOGFILE" | head -n 1)"
   RES="$(echo "$RESLINE" | sed -e "$SEDLINE")"
 }
 
