diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-08-26 10:41:32.949343201 -0700
+++ b/CMakeLists.txt	2016-08-26 10:41:33.088344559 -0700
@@ -6,11 +6,12 @@
 
 enable_language(CXX)
 
-find_package(kodi REQUIRED)
+find_package(Kodi REQUIRED)
 find_package(libretro-mame2003 REQUIRED)
 
 set(DEPLIBS mame2003)
 set(MAME2003_CUSTOM_BINARY ${MAME2003_LIB}
-                       ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} mame2003)
+                           ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} mame2003)
 
 build_addon(${PROJECT_NAME} MAME2003 DEPLIBS)
+
