diff -Naur jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/CMakeLists.txt jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/CMakeLists.txt
--- jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/CMakeLists.txt	2021-05-30 17:43:49.026903628 +0000
+++ jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/CMakeLists.txt	2021-05-30 18:01:34.312953931 +0000
@@ -5,11 +5,19 @@
 set(CMAKE_CXX_FLAGS“$ {CMAKE_CXX_FLAGS} -fPIC”)
 #set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/lib")
 #set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/aarch64-linux-gnu")
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/aarch64-linux-gnu/tegra")
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/local/cuda/lib64")
+#set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/")
+if ( LAKKA_BUILD_LIBS )
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=${LAKKA_BUILD_LIBS}")
+endif()
 
-find_library(LIB_NVBUF nvbuf_utils PATHS /usr/lib/aarch64-linux-gnu/tegra)
-find_library(LIB_V4L2 nvv4l2 PATHS /usr/lib/aarch64-linux-gnu/tegra)
+if ( LAKKA_API_PATH )
+  set(LAKKA_API_PATH "${LAKKA_API_PATH}")
+endif()
+ 
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/")
+
+find_library(LIB_NVBUF nvbuf_utils PATHS /usr/lib/)
+find_library(LIB_V4L2 nvv4l2 PATHS /usr/lib/)
 find_package (Threads)
 #find_library(LIB_DRM drm PATHS /usr/lib/aarch64-linux-gnu/tegra)
 #find_library(LIB_EGL EGL PATHS /usr/lib/aarch64-linux-gnu/tegra)
@@ -17,14 +25,15 @@
 add_library(nvmpi SHARED
     nvmpi_dec.cpp
     nvmpi_enc.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
+    
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
     #common/NvVideoConverter.cpp
     #common/NvApplicationProfiler.cpp
     #common/NvEglRenderer.cpp
@@ -34,14 +43,14 @@
 add_library(nvmpi_static STATIC
     nvmpi_dec.cpp
     nvmpi_enc.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
     #common/NvVideoConverter.cpp
     #common/NvApplicationProfiler.cpp
     #common/NvEglRenderer.cpp
@@ -53,6 +62,10 @@
 set_target_properties(nvmpi PROPERTIES SOVERSION 1)
 set_target_properties(nvmpi nvmpi_static PROPERTIES PUBLIC_HEADER nvmpi.h)
 target_link_libraries(nvmpi PRIVATE ${CMAKE_THREAD_LIBS_INIT} ${LIB_NVBUF}  ${LIB_V4L2})
+if (LAKKA_API_PATH)
+  target_include_directories(nvmpi PRIVATE ${LAKKA_API_PATH}/usr/include)
+endif ()
+
 target_include_directories(nvmpi PRIVATE /usr/src/jetson_multimedia_api/include)
 target_include_directories(nvmpi PRIVATE /usr/local/cuda/include)
 target_include_directories(nvmpi_static PRIVATE /usr/src/jetson_multimedia_api/include)
