diff -Naur jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/nvmpi_dec.cpp jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/nvmpi_dec.cpp
--- jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef/nvmpi_dec.cpp	2021-10-17 12:50:06.940056217 -0700
+++ jetson-ffmpeg-20067187641389ba309bd3ca51933718b6b475ef-2/nvmpi_dec.cpp	2021-10-17 13:05:54.983392168 -0700
@@ -498,7 +498,7 @@
 	ctx->dec->capture_plane.setStreamStatus(false);
 	
 	if (ctx->dec_capture_loop) {
-		ctx->dec_capture_loop->join();
+		//ctx->dec_capture_loop->join();
 		delete ctx->dec_capture_loop;
 		ctx->dec_capture_loop = nullptr;
 	}
