diff -Naur jetson-ffmpeg/CMakeLists.txt jetson-ffmpeg-2/CMakeLists.txt
--- jetson-ffmpeg/CMakeLists.txt	2021-10-24 15:45:57.133395456 -0700
+++ jetson-ffmpeg-2/CMakeLists.txt	2021-10-25 18:36:52.699330067 -0700
@@ -5,15 +5,23 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 #set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/lib")
 #set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/aarch64-linux-gnu")
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/aarch64-linux-gnu/tegra")
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/local/cuda/lib64")
+#set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/aarch64-linux-gnu/tegra")
+#set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/local/cuda/lib64")
 
-find_library(LIB_NVBUF nvbuf_utils PATHS /usr/lib/aarch64-linux-gnu/tegra)
-find_package(PkgConfig)
-pkg_check_modules(LIB_V4L2 REQUIRED libnvv4l2)
-find_library(LIB_V4L2 nvv4l2 PATHS /usr/lib/aarch64-linux-gnu/tegra)
-find_library(LIB_NVJPEG nvjpeg PATHS /usr/lib/aarch64-linux-gnu/tegra)
+if ( LAKKA_BUILD_LIBS )
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=${LAKKA_BUILD_LIBS}")
+endif()
 
+if ( LAKKA_API_PATH )
+  set(LAKKA_API_PATH "${LAKKA_API_PATH}")
+endif()
+ 
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath-link=/usr/lib/")
+find_library(LIB_NVBUF nvbuf_utils PATHS /usr/lib/)
+find_package(PkgConfig)
+#pkg_check_modules(LIB_V4L2 REQUIRED libnvv4l2)
+find_library(LIB_V4L2 nvv4l2 PATHS /usr/lib/)
+find_library(LIB_NVJPEG nvjpeg PATHS /usr/lib/)
 find_package (Threads)
 #find_library(LIB_DRM drm PATHS /usr/lib/aarch64-linux-gnu/tegra)
 #find_library(LIB_EGL EGL PATHS /usr/lib/aarch64-linux-gnu/tegra)
@@ -21,14 +29,15 @@
 add_library(nvmpi SHARED
     nvmpi_dec.cpp
     nvmpi_enc.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
+    
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
     #common/NvVideoConverter.cpp
     #common/NvApplicationProfiler.cpp
     #common/NvEglRenderer.cpp
@@ -38,14 +47,14 @@
 add_library(nvmpi_static STATIC
     nvmpi_dec.cpp
     nvmpi_enc.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
-    /usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvBuffer.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2Element.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvV4l2ElementPlane.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoDecoder.cpp
+    ${LAKKA_API_PATH}/usr/src/jetson_multimedia_api/samples/common/classes/NvVideoEncoder.cpp
     #common/NvVideoConverter.cpp
     #common/NvApplicationProfiler.cpp
     #common/NvEglRenderer.cpp
