diff -Naur polkit-0.104/configure.ac polkit-0.104.patch/configure.ac
--- polkit-0.104/configure.ac	2012-01-03 17:25:49.000000000 +0100
+++ polkit-0.104.patch/configure.ac	2013-01-12 09:45:00.437875775 +0100
@@ -3,7 +3,7 @@
 AC_PREREQ(2.59c)
 AC_INIT(polkit, 0.104, http://lists.freedesktop.org/mailman/listinfo/polkit-devel)
 AM_INIT_AUTOMAKE(polkit, 0.104)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
@@ -24,7 +24,6 @@
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AM_PROG_LIBTOOL
 AC_PROG_MAKE_SET
