diff -Naur libffi-3.1/include/Makefile.am libffi-3.1.patch/include/Makefile.am
--- libffi-3.1/include/Makefile.am	2014-04-25 19:45:13.000000000 +0200
+++ libffi-3.1.patch/include/Makefile.am	2014-07-03 14:30:29.019161705 +0200
@@ -5,5 +5,6 @@
 DISTCLEANFILES=ffitarget.h
 EXTRA_DIST=ffi.h.in ffi_common.h
 
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includesdir = $(includedir)/@PACKAGE_NAME@
+
 nodist_includes_HEADERS = ffi.h ffitarget.h
diff -Naur libffi-3.1/libffi.pc.in libffi-3.1.patch/libffi.pc.in
--- libffi-3.1/libffi.pc.in	2014-04-25 19:45:13.000000000 +0200
+++ libffi-3.1.patch/libffi.pc.in	2014-07-03 14:30:56.047220839 +0200
@@ -2,7 +2,7 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 toolexeclibdir=@toolexeclibdir@
-includedir=${libdir}/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includedir=@includedir@/@PACKAGE_NAME@
 
 Name: @PACKAGE_NAME@
 Description: Library supporting Foreign Function Interfaces
