From a2d10693e352db0cdfc22550f23f8294564ce939 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 16 Aug 2014 12:26:53 +0300
Subject: [PATCH] fix pkgconf

---
 libffi.pc.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libffi.pc.in b/libffi.pc.in
index 1477a66..1dd4ebd 100644
--- a/libffi.pc.in
+++ b/libffi.pc.in
@@ -7,5 +7,5 @@ includedir=@includedir@/@PACKAGE_NAME@
 Name: @PACKAGE_NAME@
 Description: Library supporting Foreign Function Interfaces
 Version: @PACKAGE_VERSION@
-Libs: -L${toolexeclibdir} -lffi
+Libs: -L${libdir} -lffi
 Cflags: -I${includedir}
-- 
1.7.2.5

