diff -Naur RTL8812AU-4.3.20.orig/Makefile RTL8812AU-4.3.20/Makefile
--- RTL8812AU-4.3.20.orig/Makefile	2016-11-24 09:28:58.380699074 +0100
+++ RTL8812AU-4.3.20/Makefile	2016-11-24 09:29:31.144699689 +0100
@@ -6,7 +6,7 @@
 #EXTRA_CFLAGS += -Werror
 #EXTRA_CFLAGS += -pedantic
 #EXTRA_CFLAGS += -Wshadow -Wpointer-arith -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes
-EXTRA_CFLAGS += -Werror=incompatible-pointer-types
+EXTRA_CFLAGS += -Wno-incompatible-pointer-types

 EXTRA_CFLAGS += -Wno-unused-variable
 EXTRA_CFLAGS += -Wno-unused-value
