From b7634effd997ce14c534a3ea437b4c20ee778d14 Mon Sep 17 00:00:00 2001
From: mglae <mglmail@arcor.de>
Date: Sun, 24 Jan 2021 15:57:32 +0100
Subject: [PATCH] support locale i18n

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 2e27a1e..57a5017 100644
--- a/Makefile
+++ b/Makefile
@@ -70,12 +70,14 @@ all: client server
 client:
 	$(Q)$(MAKE) --no-print-directory -C ./tools
 	$(Q)$(MAKE) --no-print-directory -C ./client
+	$(Q)$(MAKE) --no-print-directory -C ./client install-i18n
 
 server:
 	$(Q)$(MAKE) --no-print-directory -C ./tools
 	$(Q)$(MAKE) --no-print-directory -C ./libdvbmpeg
 	$(Q)$(MAKE) --no-print-directory -C ./remux
 	$(Q)$(MAKE) --no-print-directory -C ./server
+	$(Q)$(MAKE) --no-print-directory -C ./server install-i18n
 
 install-client: client
 	$(Q)$(MAKE) --no-print-directory -C ./client install
-- 
2.51.0

