--- a/Makefile	2017-09-30 21:31:48.000000000 +0200
+++ b/Makefile	2021-01-12 23:26:57.857536391 +0100
@@ -71,12 +71,14 @@ all: client server
 client:
 	$(MAKE) -C ./tools
 	$(MAKE) -C ./client
+	$(MAKE) -C ./client install-i18n
 
 server:
 	$(MAKE) -C ./tools
 	$(MAKE) -C ./libdvbmpeg
 	$(MAKE) -C ./remux
 	$(MAKE) -C ./server
+	$(MAKE) -C ./server install-i18n
 
 install-client: client
 	$(MAKE) -C ./client install
