diff -r -Bc a/server/connectionVTP.c b/server/connectionVTP.c
*** a/server/connectionVTP.c	2021-05-22 10:33:01.633332698 +0200
--- b/server/connectionVTP.c	2021-05-22 10:27:50.000000000 +0200
***************
*** 1882,1888 ****
  #endif
  									}
  									else {
! 										cDevice::SetCurrentChannel(CurrentChannel);
  									}
  								}
  								isyslog("channel %d moved to %d", FromNumber, ToNumber);
--- 1882,1888 ----
  #endif
  									}
  									else {
! 										cDevice::SetCurrentChannel(CurrentChannel->Number());
  									}
  								}
  								isyslog("channel %d moved to %d", FromNumber, ToNumber);
***************
*** 1983,1989 ****
  							Channels.SwitchTo(CurrentChannel->Number());
  #endif
  						else
! 							cDevice::SetCurrentChannel(CurrentChannel);
  					}
  					Reply(250, "Channel \"%s\" deleted", Option);
  				}
--- 1983,1989 ----
  							Channels.SwitchTo(CurrentChannel->Number());
  #endif
  						else
! 							cDevice::SetCurrentChannel(CurrentChannel->Number());
  					}
  					Reply(250, "Channel \"%s\" deleted", Option);
  				}
