diff -Naur xbmc-74fa128/xbmc/screensavers/rsxs-0.9/lib/stdio.h xbmc-74fa128.patch/xbmc/screensavers/rsxs-0.9/lib/stdio.h
diff -Naur xbmc-74fa128/xbmc/screensavers/rsxs-0.9/lib/stdio.in.h xbmc-74fa128.patch/xbmc/screensavers/rsxs-0.9/lib/stdio.in.h
--- xbmc-74fa128/xbmc/screensavers/rsxs-0.9/lib/stdio.in.h	2013-04-13 23:19:12.183315057 +0200
+++ xbmc-74fa128.patch/xbmc/screensavers/rsxs-0.9/lib/stdio.in.h	2013-04-13 23:42:24.552302369 +0200
@@ -699,26 +699,14 @@
 # endif
 #endif
 
-#if @GNULIB_GETS@
-# if @REPLACE_STDIO_READ_FUNCS@ && @GNULIB_STDIO_H_NONBLOCKING@
-#  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
-#   undef gets
-#   define gets rpl_gets
-#  endif
-_GL_FUNCDECL_RPL (gets, char *, (char *s) _GL_ARG_NONNULL ((1)));
-_GL_CXXALIAS_RPL (gets, char *, (char *s));
-# else
-_GL_CXXALIAS_SYS (gets, char *, (char *s));
-#  undef gets
-# endif
-_GL_CXXALIASWARN (gets);
 /* It is very rare that the developer ever has full control of stdin,
-   so any use of gets warrants an unconditional warning.  Assume it is
-   always declared, since it is required by C89.  */
+   so any use of gets warrants an unconditional warning; besides, C11
+   removed it.  */
+#undef gets
+#if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
 
-
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
 struct obstack;
 /* Grow an obstack with formatted output.  Return the number of
diff -Naur xbmc-74fa128/xbmc/screensavers/rsxs-0.9/m4/stdio_h.m4 xbmc-74fa128.patch/xbmc/screensavers/rsxs-0.9/m4/stdio_h.m4
--- xbmc-74fa128/xbmc/screensavers/rsxs-0.9/m4/stdio_h.m4	2013-04-13 23:19:12.216314905 +0200
+++ xbmc-74fa128.patch/xbmc/screensavers/rsxs-0.9/m4/stdio_h.m4	2013-04-13 23:37:54.379416637 +0200
@@ -18,7 +18,6 @@
   GNULIB_GETC=1
   GNULIB_GETCHAR=1
   GNULIB_FGETS=1
-  GNULIB_GETS=1
   GNULIB_FREAD=1
   dnl This ifdef is necessary to avoid an error "missing file lib/stdio-read.c"
   dnl "expected source file, required through AC_LIBSOURCES, not found". It is
@@ -113,7 +112,6 @@
   GNULIB_GETCHAR=0;              AC_SUBST([GNULIB_GETCHAR])
   GNULIB_GETDELIM=0;             AC_SUBST([GNULIB_GETDELIM])
   GNULIB_GETLINE=0;              AC_SUBST([GNULIB_GETLINE])
-  GNULIB_GETS=0;                 AC_SUBST([GNULIB_GETS])
   GNULIB_OBSTACK_PRINTF=0;       AC_SUBST([GNULIB_OBSTACK_PRINTF])
   GNULIB_OBSTACK_PRINTF_POSIX=0; AC_SUBST([GNULIB_OBSTACK_PRINTF_POSIX])
   GNULIB_PCLOSE=0;               AC_SUBST([GNULIB_PCLOSE])
