diff -Naur linux-xu3/arch/arm/mach-exynos/exynos-ss.c linux-xu3.patch/arch/arm/mach-exynos/exynos-ss.c
--- linux-xu3/arch/arm/mach-exynos/exynos-ss.c	2017-01-20 00:03:08.164350238 +0100
+++ linux-xu3.patch/arch/arm/mach-exynos/exynos-ss.c	2017-01-20 00:07:24.504134302 +0100
@@ -257,6 +257,13 @@
 static unsigned int ess_virt_addr = 0;
 static unsigned int ess_size = 0;
 
+enum ess_cause_emerg_events {
+	CAUSE_INVALID_DUMP = 0x00000000,
+	CAUSE_KERNEL_PANIC = 0x00000001,
+	CAUSE_FORCE_DUMP   = 0x0000000D,
+	CAUSE_FORCE_REBOOT = 0x000000FF,
+};
+
 DEFINE_PER_CPU(struct exynos_ss_core_reg *, ess_core_reg);
 DEFINE_PER_CPU(struct exynos_ss_mmu_reg *, ess_mmu_reg);
 DEFINE_PER_CPU(enum ess_cause_emerg_events, ess_cause_emerg);
@@ -493,13 +500,6 @@
 }
 #endif
 
-enum ess_cause_emerg_events {
-	CAUSE_INVALID_DUMP = 0x00000000,
-	CAUSE_KERNEL_PANIC = 0x00000001,
-	CAUSE_FORCE_DUMP   = 0x0000000D,
-	CAUSE_FORCE_REBOOT = 0x000000FF,
-};
-
 static void exynos_ss_scratch_reg(unsigned int val)
 {
 	__raw_writel(val, S5P_VA_SS_SCRATCH);
