--- a/Makefile.in	2018-12-04 21:56:06.000000000 +0100
+++ b/Makefile.in	2020-01-09 07:06:03.472529409 +0100
@@ -380,7 +380,7 @@
 ecc-521.$(OBJEXT): ecc-521.h
 ecc-25519.$(OBJEXT): ecc-25519.h
 
-.asm.$(OBJEXT): $(srcdir)/asm.m4 machine.m4 config.m4
+%.$(OBJEXT): %.asm $(srcdir)/asm.m4 machine.m4 config.m4
 	$(M4) $(srcdir)/asm.m4 machine.m4 config.m4 $< >$*.s
 	$(COMPILE) -c $*.s
 	@echo "$@ : $< $(srcdir)/asm.m4 machine.m4 config.m4" >$@.d 
