diff -Naur nettle-2.7.1/Makefile.in nettle-2.7.1.patch/Makefile.in
--- nettle-2.7.1/Makefile.in	2013-05-28 16:21:53.000000000 +0200
+++ nettle-2.7.1.patch/Makefile.in	2014-05-30 17:52:21.402473749 +0200
@@ -20,7 +20,7 @@
 
 include config.make
 
-PRE_CPPFLAGS = -I.
+PRE_CPPFLAGS = -I. -I$(srcdir)
 # FIXME: Add configuration of LIBEXT?
 LIBTARGETS = @IF_STATIC@ libnettle.a @IF_HOGWEED@ libhogweed.a
 SHLIBTARGETS = @IF_SHARED@ $(LIBNETTLE_FORLINK) @IF_HOGWEED@ $(LIBHOGWEED_FORLINK)
@@ -229,27 +229,27 @@
 # executable. Avoid object file targets to make it easy to run the
 # right compiler.
 aesdata$(EXEEXT_FOR_BUILD): aesdata.c
-	$(CC_FOR_BUILD) `test -f aesdata.c || echo '$(srcdir)/'`aesdata.c \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f aesdata.c || echo '$(srcdir)/'`aesdata.c \
 	-o aesdata$(EXEEXT_FOR_BUILD)
 
 desdata$(EXEEXT_FOR_BUILD): desdata.c
-	$(CC_FOR_BUILD) `test -f desdata.c || echo '$(srcdir)/'`desdata.c \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f desdata.c || echo '$(srcdir)/'`desdata.c \
 	-o desdata$(EXEEXT_FOR_BUILD)
 
 twofishdata$(EXEEXT_FOR_BUILD): twofishdata.c
-	$(CC_FOR_BUILD) `test -f twofishdata.c || echo '$(srcdir)/'`twofishdata.c \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f twofishdata.c || echo '$(srcdir)/'`twofishdata.c \
 	-o twofishdata$(EXEEXT_FOR_BUILD)
 
 shadata$(EXEEXT_FOR_BUILD): shadata.c
-	$(CC_FOR_BUILD) `test -f shadata.c || echo '$(srcdir)/'`shadata.c -lm \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f shadata.c || echo '$(srcdir)/'`shadata.c -lm \
 	-o shadata$(EXEEXT_FOR_BUILD)
 
 gcmdata$(EXEEXT_FOR_BUILD): gcmdata.c
-	$(CC_FOR_BUILD) `test -f gcmdata.c || echo '$(srcdir)/'`gcmdata.c \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f gcmdata.c || echo '$(srcdir)/'`gcmdata.c \
 	-o gcmdata$(EXEEXT_FOR_BUILD)
 
 eccdata$(EXEEXT_FOR_BUILD): eccdata.c mini-gmp.c mini-gmp.h
-	$(CC_FOR_BUILD) `test -f eccdata.c || echo '$(srcdir)/'`eccdata.c \
+	$(CC_FOR_BUILD) $(PRE_CPPFLAGS) `test -f eccdata.c || echo '$(srcdir)/'`eccdata.c \
 	-o eccdata$(EXEEXT_FOR_BUILD)
 
 # desCore rules
