--- a/arch/arm/boot/dts/imx6q-tbs2910.dts	2016-09-17 15:27:28.950265621 +0200
+++ b/arch/arm/boot/dts/imx6q-tbs2910.dts	2016-09-17 15:28:25.984089674 +0200
@@ -84,17 +84,14 @@
 			gpios = <&gpio1 2 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "heartbeat";
 		};
-	};
 
-	fan {
-		compatible = "gpio-fan";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_gpio_fan>;
-		gpios = <&gpio3 28 GPIO_ACTIVE_HIGH>;
-		gpio-fan,speed-map = <0    0
-		                      3000 1>;
+		fan {
+			gpios = <&gpio3 28 0>;
+			default-state = "off";
+		};
 	};
 
+
 	sound {
 		compatible = "fsl,imx-audio-sgtl5000";
 		model = "imx-sgtl5000";
