diff -Naur linux-cuboxi-3.14-ea83bda.orig/drivers/media/usb/dvb-usb-v2/dvbsky.c linux-cuboxi-3.14-ea83bda/drivers/media/usb/dvb-usb-v2/dvbsky.c
--- linux-cuboxi-3.14-ea83bda.orig/drivers/media/usb/dvb-usb-v2/dvbsky.c	2015-05-12 07:55:11.516604741 +0200
+++ linux-cuboxi-3.14-ea83bda/drivers/media/usb/dvb-usb-v2/dvbsky.c	2015-05-12 07:58:19.750871492 +0200
@@ -280,7 +280,11 @@
 	.i2c_wr_max = 33,
 	.clock_out = 0,
 	.ts_mode = M88DS3103_TS_CI,
+	.ts_clk = 16000,
+	.ts_clk_pol = 0,
 	.agc = 0x99,
+	.lnb_hv_pol = 1,
+	.lnb_en_pol = 1,
 };
 
 static int dvbsky_s960_attach(struct dvb_usb_adapter *adap)
