--- a/scripts/dtc/pylibfdt/Makefile	2018-07-09 16:24:14.000000000 +0200
+++ b/scripts/dtc/pylibfdt/Makefile	2018-09-02 22:06:07.945632774 +0200
@@ -18,7 +18,7 @@
 		CC="$(HOSTCC)" LDSHARED="$(HOSTCC) -shared " \
 		LDFLAGS="$(HOSTLDFLAGS)" \
 		VERSION="u-boot-$(UBOOTVERSION)" \
-		CPPFLAGS="$(HOSTCFLAGS) -I$(LIBFDT_srcdir)" OBJDIR=$(obj) \
+		CPPFLAGS="$(HOSTCFLAGS) -I$(LIBFDT_srcdir) `python-config --includes`" OBJDIR=$(obj) \
 		SOURCES="$(PYLIBFDT_srcs)" \
 		SWIG_OPTS="-I$(LIBFDT_srcdir) -I$(LIBFDT_srcdir)/.." \
 		$(PYTHON) $< --quiet build_ext --inplace
