diff --git a/src/od-pandora/pandora.cpp b/src/od-pandora/pandora.cpp
index d8c7a0e..b454ec6 100644
--- a/src/od-pandora/pandora.cpp
+++ b/src/od-pandora/pandora.cpp
@@ -498,7 +498,7 @@ int target_cfgfile_load (struct uae_prefs *p, const char *filename, int type, in
 	char *ptr;
 #ifdef RP9_SUPPORT
 	ptr = strstr((char *)filename, ".rp9");
-  if(ptr > 0)
+  if(ptr != 0)
   {
     // Load rp9 config
     result = rp9_parse_file(p, filename);
@@ -509,7 +509,7 @@ int target_cfgfile_load (struct uae_prefs *p, const char *filename, int type, in
 #endif
 	{
   	ptr = strstr((char *)filename, ".uae");
-    if(ptr > 0)
+    if(ptr != 0)
     {
       int type = CONFIG_TYPE_HARDWARE | CONFIG_TYPE_HOST;
       result = cfgfile_load(p, filename, &type, 0, 1);
@@ -555,7 +555,7 @@ int check_configfile(char *file)
   
   strncpy(tmp, file, MAX_PATH);
 	char *ptr = strstr(tmp, ".uae");
-	if(ptr > 0)
+	if(ptr != 0)
   {
     *(ptr + 1) = '\0';
     strncat(tmp, "conf", MAX_PATH);
