From 901f0feb8b6b97b445b0623cd5d105544911f060 Mon Sep 17 00:00:00 2001
From: Ronald Brown <rbrown4014@yahoo.com>
Date: Mon, 30 Jan 2023 08:46:42 -0800
Subject: [PATCH 3/7] Lakka: Always show Power Management Settings menu

---
 menu/menu_setting.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 5e8e514331..89fdda3fe9 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -10315,7 +10315,6 @@ static bool setting_append_list(
                &group_info,
                &subgroup_info,
                parent_group);
-         SETTINGS_DATA_LIST_CURRENT_ADD_FLAGS(list, list_info, SD_FLAG_LAKKA_ADVANCED);
 
          CONFIG_ACTION(
                list, list_info,
-- 
2.25.1

