diff -urN '--exclude=.git' /tmp/pristine-base/menu/cbs/menu_cbs_get_value.c /tmp/network-fix-tree/menu/cbs/menu_cbs_get_value.c
--- a/menu/cbs/menu_cbs_get_value.c	2026-05-19 18:23:50.078223754 +0000
+++ b/menu/cbs/menu_cbs_get_value.c	2026-05-19 18:24:06.482091822 +0000
@@ -2141,8 +2141,7 @@
             BIND_ACTION_GET_VALUE(cbs,
                   menu_action_setting_disp_set_label_manual_content_scan_core_name);
             break;
-#ifdef HAVE_NETWORKING
-         #ifdef HAVE_LAKKA_SWITCH
+#ifdef HAVE_LAKKA_SWITCH
          case MENU_ENUM_LABEL_GPU_PERF_MODE:
             BIND_ACTION_GET_VALUE(cbs,
                   menu_action_setting_disp_gpu_gov_mode);
@@ -2151,7 +2150,8 @@
          case MENU_ENUM_LABEL_GPU_MANAGED_MAX_FREQ:
             BIND_ACTION_GET_VALUE(cbs, menu_action_gpu_managed_freq_label);
             break;
-         #endif
+#endif
+#ifdef HAVE_NETWORKING
          case MENU_ENUM_LABEL_CORE_UPDATER_ENTRY:
             BIND_ACTION_GET_VALUE(cbs,
                   menu_action_setting_disp_set_label_core_updater_entry);
