From 48cf264260f7e8e123cc8b53d27ee67370329cbc Mon Sep 17 00:00:00 2001
From: Ronald Brown <rbrown4014@yahoo.com>
Date: Sun, 24 May 2026 17:17:57 +0200
Subject: [PATCH 1/2] sys-clk more_logging

---
 misc/sys-clk/sys-clk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/misc/sys-clk/sys-clk.c b/misc/sys-clk/sys-clk.c
index 5a17c47..d1c261d 100644
--- a/misc/sys-clk/sys-clk.c
+++ b/misc/sys-clk/sys-clk.c
@@ -933,6 +933,8 @@ void sys_clk_driver_init(enum sys_clk_domain dom)
             (unsigned)dom);
       return;
    }
+   RARCH_LOG("[sys-clk]: domain %u has usable device; enabling.\n",
+         (unsigned)dom);
    sys_clk_set_mode(dom, st->cur_mode, NULL);
 }
 
-- 
2.43.0

