From 4989e17fedc741f70e69a2e6bf4bf2ab5723157c Mon Sep 17 00:00:00 2001
From: Ronald Brown <rbrown4014@yahoo.com>
Date: Sun, 24 May 2026 19:55:17 +0200
Subject: [PATCH] move init

---
 retroarch.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/retroarch.c b/retroarch.c
index 225385e..b4077cb 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -1729,6 +1729,11 @@ void drivers_init(
       gfx_display_init_first_driver(p_disp, video_is_threaded);
    }
 
+#ifdef HAVE_LAKKA
+   sys_clk_driver_init(SYS_CLK_DOMAIN_CPU);
+   sys_clk_driver_init(SYS_CLK_DOMAIN_GPU);
+#endif
+
 #ifdef HAVE_MENU
    if (flags & DRIVER_VIDEO_MASK)
    {
@@ -1774,11 +1779,6 @@ void drivers_init(
    if (flags & DRIVER_MIDI_MASK)
       midi_driver_init(settings);
 
-#ifdef HAVE_LAKKA
-   sys_clk_driver_init(SYS_CLK_DOMAIN_CPU);
-   sys_clk_driver_init(SYS_CLK_DOMAIN_GPU);
-#endif
-
 #ifdef HAVE_LAKKA_SWITCH
    {
       /* Parse /flash/bootloader contents, then restore the saved
-- 
2.43.0

