From 6012645d1104daf06e17847233bcca66fa5e97e1 Mon Sep 17 00:00:00 2001
From: Ronald Brown <rbrown4014@yahoo.com>
Date: Mon, 25 May 2026 06:23:09 +0200
Subject: [PATCH] filter out non gpu devices that have devfreq

---
 misc/sys-clk/sys-clk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/misc/sys-clk/sys-clk.c b/misc/sys-clk/sys-clk.c
index 1905872..04fbf92 100644
--- a/misc/sys-clk/sys-clk.c
+++ b/misc/sys-clk/sys-clk.c
@@ -356,6 +356,8 @@ static bool resolve_root(sys_clk_state_t *st)
       dir_list_sort(devs, false);
       for (i = 0; i < (int)devs->size; i++)
       {
+         if (strstr(devs->elems[i].data, ".gpu") == NULL)
+            continue;
          char probe[PATH_MAX_LENGTH];
          fill_pathname_join(probe, devs->elems[i].data,
                st->be->file_cur_freq, sizeof(probe));
-- 
2.43.0

