diff -Naur RetroArch/cores/libretro-ffmpeg/ffmpeg_core.c RetroArch-2/cores/libretro-ffmpeg/ffmpeg_core.c
--- RetroArch/cores/libretro-ffmpeg/ffmpeg_core.c	2021-10-17 16:50:23.332408856 -0700
+++ RetroArch-2/cores/libretro-ffmpeg/ffmpeg_core.c	2021-10-18 17:01:27.814162473 -0700
@@ -1131,8 +1131,32 @@
 static bool open_codec(AVCodecContext **ctx, enum AVMediaType type, unsigned index)
 {
    int ret = 0;
+   AVCodec *codec;
+   switch (fctx->streams[index]->codec->codec_id)
+   {
+      case AV_CODEC_ID_H264: //h264
+         codec = avcodec_find_decoder_by_name("h264_nvmpi");
+         break;
+      case AV_CODEC_ID_HEVC: //hevc
+         codec = avcodec_find_decoder_by_name("hevc_nvmpi");
+         break;
+      case AV_CODEC_ID_VP8: //vp8
+         codec = avcodec_find_decoder_by_name("vp8_nvmpi");
+         break;
+      case AV_CODEC_ID_VP9: //vp9
+         codec = avcodec_find_decoder_by_name("vp9_nvmpi");
+         break;
+      case AV_CODEC_ID_MPEG4: //MPEG4
+         codec = avcodec_find_decoder_by_name("mpeg4_nvmpi");
+         break;
+      case AV_CODEC_ID_MPEG2VIDEO: //MPEG2
+         codec = avcodec_find_decoder_by_name("mpeg2_nvmpi");
+         break;
+      default: // everything else
+         codec = avcodec_find_decoder(fctx->streams[index]->codec->codec_id);
+         break;
+   }
 
-   AVCodec *codec = avcodec_find_decoder(fctx->streams[index]->codec->codec_id);
    if (!codec)
    {
       log_cb(RETRO_LOG_ERROR, "[FFMPEG] Couldn't find suitable decoder\n");
