From 0650bbb7d79c6db34bc54dfb73320303d83a8f56 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Tue, 14 Jan 2020 21:35:57 +0000
Subject: [PATCH] use sysroot path for pkgconfig and libtool

---
 mkspecs/features/qt_module.prf | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mkspecs/features/qt_module.prf b/mkspecs/features/qt_module.prf
index 828a9621b9..0f3472fffc 100644
--- a/mkspecs/features/qt_module.prf
+++ b/mkspecs/features/qt_module.prf
@@ -275,12 +275,12 @@ load(qt_targets)
     host_build: \
         QMAKE_PKGCONFIG_LIBDIR = $$[QT_HOST_LIBS]
     else: \
-        QMAKE_PKGCONFIG_LIBDIR = $$[QT_INSTALL_LIBS/raw]
+        QMAKE_PKGCONFIG_LIBDIR = $$[QT_INSTALL_LIBS]
     lib_bundle {
-        QMAKE_PKGCONFIG_INCDIR = $$[QT_INSTALL_LIBS/raw]/$${MODULE_INCNAME}.framework/Headers
+        QMAKE_PKGCONFIG_INCDIR = $$[QT_INSTALL_LIBS]/$${MODULE_INCNAME}.framework/Headers
         QMAKE_PKGCONFIG_CFLAGS = -D$$MODULE_DEFINE
     } else {
-        QMAKE_PKGCONFIG_INCDIR = $$[QT_INSTALL_HEADERS/raw]
+        QMAKE_PKGCONFIG_INCDIR = $$[QT_INSTALL_HEADERS]
         QMAKE_PKGCONFIG_CFLAGS = -D$$MODULE_DEFINE -I${includedir}/$$MODULE_INCNAME
         for(inc, MODULE_AUX_INCLUDES): \
             QMAKE_PKGCONFIG_CFLAGS += -I${includedir}/$$section(inc, /, 1, 1)
@@ -308,7 +308,7 @@ load(qt_targets)
     host_build: \
         QMAKE_LIBTOOL_LIBDIR = $$[QT_HOST_LIBS]
     else: \
-        QMAKE_LIBTOOL_LIBDIR = "=$$[QT_INSTALL_LIBS/raw]"
+        QMAKE_LIBTOOL_LIBDIR = "=$$[QT_INSTALL_LIBS]"
     !isEmpty(lib_replace0.match) {
         ltlib_replace0.match = $$lib_replace0.match
         ltlib_replace0.replace = $$QMAKE_LIBTOOL_LIBDIR/
-- 
2.17.1

