diff -Naur a/arch/arm64/boot/dts/qcom/msm8916.dtsi b/arch/arm64/boot/dts/qcom/msm8916.dtsi
--- a/arch/arm64/boot/dts/qcom/msm8916.dtsi	2017-05-10 06:07:46.000000000 -0700
+++ b/arch/arm64/boot/dts/qcom/msm8916.dtsi	2017-06-01 11:45:44.805705640 -0700
@@ -1177,6 +1177,7 @@
 						interrupts = <0 145 IRQ_TYPE_LEVEL_HIGH>,
 							     <0 146 IRQ_TYPE_LEVEL_HIGH>;
 						interrupt-names = "tx", "rx";
+						local-mac-address = [02 00 27 60 25 28];
 
 						qcom,smem-states = <&apps_smsm 10>, <&apps_smsm 9>;
 						qcom,smem-state-names = "tx-enable", "tx-rings-empty";
