diff -Naur lcdproc-0.5.5-old/server/drivers/Makefile.am lcdproc-0.5.5-new/server/drivers/Makefile.am
--- lcdproc-0.5.5-old/server/drivers/Makefile.am	2011-08-14 05:29:16.000000000 -0700
+++ lcdproc-0.5.5-new/server/drivers/Makefile.am	2012-01-13 17:58:57.000000000 -0800
@@ -20,7 +20,7 @@
 
 ## Keep the lists sorted!
 
-pkglib_PROGRAMS = @DRIVERS@
+pkglibexec_PROGRAMS = @DRIVERS@
 EXTRA_PROGRAMS = bayrad CFontz CFontz633 CFontzPacket curses debug CwLnx ea65 EyeboxOne g15 glcdlib glk hd44780 icp_a106 imon imonlcd IOWarrior irman joy lb216 lcdm001 lcterm lirc lis MD8800 mdm166a ms6931 mtc_s16209x MtxOrb mx5000 NoritakeVFD picolcd pyramid sed1330 sed1520 serialPOS serialVFD shuttleVFD stv5730 SureElec svga t6963 text tyan sli ula200 xosd i2500vfd irtrans
 noinst_LIBRARIES = libLCD.a libbignum.a
 
