diff -Naur go-1.9.2.orig/src/crypto/x509/root_unix.go go-1.9.2/src/crypto/x509/root_unix.go
--- go-1.9.2.orig/src/crypto/x509/root_unix.go	2017-11-03 14:58:53.655965257 +0100
+++ go-1.9.2/src/crypto/x509/root_unix.go	2017-11-03 14:59:16.923786983 +0100
@@ -19,6 +19,7 @@
 	"/usr/local/share/certs",       // FreeBSD
 	"/etc/pki/tls/certs",           // Fedora/RHEL
 	"/etc/openssl/certs",           // NetBSD
+	"/etc/ssl",                     // LibreELEC
 }
 
 const (
