diff -Naur a/src/crypto/x509/root_unix.go b/src/crypto/x509/root_unix.go
--- a/src/crypto/x509/root_unix.go	2016-04-19 16:27:35.000000000 -0700
+++ b/src/crypto/x509/root_unix.go	2016-04-22 01:21:02.884977461 -0700
@@ -13,6 +13,7 @@
 var certDirectories = []string{
 	"/etc/ssl/certs",               // SLES10/SLES11, https://golang.org/issue/12139
 	"/system/etc/security/cacerts", // Android
+	"/etc/ssl",                     // LibreELEC
 }
 
 func (c *Certificate) systemVerify(opts *VerifyOptions) (chains [][]*Certificate, err error) {
