diff --git a/bsnes/GNUmakefile b/bsnes/GNUmakefile
index 9edfe86a..c226952e 100644
--- a/bsnes/GNUmakefile
+++ b/bsnes/GNUmakefile
@@ -1,8 +1,8 @@
 target ?= bsnes
 binary ?= application
 build := performance
-openmp := true
-local ?= true
+openmp := false
+local ?= false
 flags += -I. -I..
 
 # in order for this to work, obj/lzma.o must be omitted or bsnes will hang on startup.
@@ -37,7 +37,6 @@ else ifeq ($(platform),macos)
 else ifneq ($(filter $(platform),linux bsd),)
   ifeq ($(binary),application)
     options += -Wl,-export-dynamic
-    options += -lX11 -lXext
   else ifeq ($(binary),library)
     flags += -fPIC
     options += -shared
diff --git a/bsnes/target-libretro/GNUmakefile b/bsnes/target-libretro/GNUmakefile
index edbc831d..9ec4a1c4 100644
--- a/bsnes/target-libretro/GNUmakefile
+++ b/bsnes/target-libretro/GNUmakefile
@@ -1,6 +1,6 @@
 name := bsnes_libretro
 local := false
-openmp := true
+openmp := false
 flags += -Wno-narrowing -Wno-multichar -g -fPIC
 
 ifeq ($(platform), ios-arm64)
@@ -23,7 +23,7 @@ obj/libretro.o: target-libretro/libretro.cpp
 
 all: $(objects)
 ifeq ($(platform),linux)
-	$(strip $(compiler) -o out/$(name).so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/$(name).so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),windows)
 	$(strip $(compiler) -o out/$(name).dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
 else ifeq ($(platform),libnx)
