--- a/kernel/nvidia-drm/nvidia-dma-resv-helper.h	2022-05-12 22:42:02.000000000 +0000
+++ b/kernel/nvidia-drm/nvidia-dma-resv-helper.h	2022-06-29 08:19:44.859696269 +0000
@@ -23,6 +23,7 @@
 #ifndef __NVIDIA_DMA_RESV_HELPER_H__
 #define __NVIDIA_DMA_RESV_HELPER_H__
 
+#include <linux/version.h>
 #include "nvidia-drm-conftest.h"
 
 #if defined(NV_DRM_FENCE_AVAILABLE)
@@ -69,7 +70,11 @@
                                               nv_dma_fence_t *fence)
 {
 #if defined(NV_LINUX_DMA_RESV_H_PRESENT)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 19, 0))
+    dma_resv_add_fence(obj, fence, DMA_RESV_USAGE_WRITE);
+#else
     dma_resv_add_excl_fence(obj, fence);
+#endif
 #else
     reservation_object_add_excl_fence(obj, fence);
 #endif
