diff --git a/kernel/nv-linux.h b/kernel/nv-linux.h
index 801c88c..78f8889 100644
--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -268,6 +268,11 @@ extern int nv_pat_mode;
 #if !defined(NV_VMWARE) && defined(CONFIG_ACPI)
 #include <acpi/acpi.h>
 #include <acpi/acpi_drivers.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+#include <linux/acpi.h>
+#endif
+
 #if defined(NV_ACPI_DEVICE_OPS_HAS_MATCH) || defined(ACPI_VIDEO_HID)
 #define NV_LINUX_ACPI_EVENTS_SUPPORTED 1
 #endif
