diff --git a/kernel/nv-drm.c b/kernel/nv-drm.c
index f4011c2..929a187 100644
--- a/kernel/nv-drm.c
+++ b/kernel/nv-drm.c
@@ -17,6 +17,7 @@
 #if defined(NV_DRM_AVAILABLE)
 
 #include <drm/drmP.h>
+#include <drm/drm_gem.h>
 
 extern nv_linux_state_t *nv_linux_devices;
 
@@ -114,6 +115,7 @@ static struct drm_driver nv_drm_driver = {
     .driver_features = DRIVER_GEM | DRIVER_PRIME,
     .load = nv_drm_load,
     .unload = nv_drm_unload,
+    .set_busid = drm_pci_set_busid,
     .fops = &nv_drm_fops,
 
     .gem_free_object = nv_gem_free,
