From 18f51127373375ce189be498a64cc286114f142f Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 19 Oct 2013 21:40:14 +0300
Subject: [PATCH] vdr 2.1.2 buildfix

---
 server/connectionVTP.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/server/connectionVTP.c b/server/connectionVTP.c
index 55c92de..57bb64b 100644
--- a/server/connectionVTP.c
+++ b/server/connectionVTP.c
@@ -1377,7 +1377,7 @@ bool cConnectionVTP::CmdSTAT(const char *Option)
 	if (*Option) {
 		if (strcasecmp(Option, "DISK") == 0) {
 			int FreeMB, UsedMB;
-			int Percent = VideoDiskSpace(&FreeMB, &UsedMB);
+			int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB, &UsedMB);
 			Reply(250, "%dMB %dMB %d%%", FreeMB + UsedMB, FreeMB, Percent);
 		}
 		else if (strcasecmp(Option, "NAME") == 0) {
-- 
1.7.2.5

