commit 949a46ec2ea06d97a7fcd47adc33153215399ac5
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sun Mar 4 16:39:58 2012 +0200

    vdr-streamdev: adjust makefile for openelec

diff --git a/Makefile b/Makefile
index b375844..231041f 100644
--- a/Makefile
+++ b/Makefile
@@ -21,8 +21,6 @@ VERSION = $(shell grep 'const char \*VERSION *=' common.c | awk '{ print $$5 }'
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
 TMPDIR = /tmp
 
 ### The version number of VDR (taken from VDR's "config.h"):
