diff -Naur pyOpenSSL-0.10/setup.py pyOpenSSL-0.10.patch/setup.py
--- pyOpenSSL-0.10/setup.py	2009-11-13 16:36:16.000000000 +0100
+++ pyOpenSSL-0.10.patch/setup.py	2010-10-19 00:24:51.858811333 +0200
@@ -15,7 +15,7 @@
 distutils.log.set_verbosity(3)
 
 import sys, os
-from distutils.core import Extension, setup
+from setuptools import Extension, setup
 from distutils.errors import DistutilsFileError
 from distutils.command.build_ext import build_ext
 
