diff -Naur imx-lib-3.5.7-1.0.0/vpu/Makefile imx-lib-3.5.7-1.0.0.patch/vpu/Makefile
--- imx-lib-3.5.7-1.0.0/vpu/Makefile	2013-06-21 15:22:18.000000000 +0200
+++ imx-lib-3.5.7-1.0.0.patch/vpu/Makefile	2014-01-13 08:07:56.431868081 +0100
@@ -1,6 +1,8 @@
 CC=$(CROSS_COMPILE)gcc
 AR=$(CROSS_COMPILE)ar
 
+CFLAGS := -Wall -O2
+
 # list of platforms which want this test case
 INCLUDE_LIST:= IMX27ADS IMX51 IMX53 IMX6Q
 
@@ -33,7 +35,7 @@
 endif
 
 %.o: %.c
-	$(CC) -D$(PLATFORM) -Wall -O2 -fPIC -c $^ -o $@
+	$(CC) -D$(PLATFORM) $(CFLAGS) -fPIC -c $^ -o $@
 
 $(LIBNAME).so.$(SONAMEVERSION): $(OBJ)
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ $^ -o $@ -lpthread
