diff -Naur pptp-1.7.2/Makefile pptp-1.7.2.patch/Makefile
--- pptp-1.7.2/Makefile	2008-05-14 08:33:55.000000000 +0200
+++ pptp-1.7.2.patch/Makefile	2012-08-17 14:06:16.929624069 +0200
@@ -4,25 +4,25 @@
 
 #################################################################
 # CHANGE THIS LINE to point to the location of your pppd binary.
-PPPD = /usr/sbin/pppd
+PPPD := /usr/sbin/pppd
 #################################################################
 
 BINDIR=$(DESTDIR)/usr/sbin
 MANDIR=$(DESTDIR)/usr/share/man/man8
 PPPDIR=$(DESTDIR)/etc/ppp
 
-CC	= gcc
+CC	:= gcc
 RM	= rm -f
 OPTIMIZE= -O0
 DEBUG	= -g
 INCLUDE =
-CFLAGS  = -Wall $(OPTIMIZE) $(DEBUG) $(INCLUDE)
+CFLAGS  := -Wall $(OPTIMIZE) $(DEBUG) $(INCLUDE)
 LIBS	= -lutil
 # Solaris 10
 # LIBS	= -lnsl -lsocket -lresolv
 # Solaris Nevada build 14 or above
 # LIBS    = -lnsl -lsocket
-LDFLAGS	=
+LDFLAGS	:=
 
 PPTP_BIN = pptp
 
