diff -Naur linux-amlogic-3.10-fa4c242/drivers/amlogic/gpu/mali/Kbuild linux-amlogic-3.10-fa4c242.patch/drivers/amlogic/gpu/mali/Kbuild
--- linux-amlogic-3.10-fa4c242/drivers/amlogic/gpu/mali/Kbuild	2015-01-22 21:04:55.000000000 +0100
+++ linux-amlogic-3.10-fa4c242.patch/drivers/amlogic/gpu/mali/Kbuild	2015-01-23 20:17:06.225678942 +0100
@@ -220,32 +220,32 @@
 # Get subversion revision number, fall back to only ${MALI_RELEASE_NAME} if no svn info is available
 MALI_RELEASE_NAME=$(shell cat $(TOP_KBUILD_SRC)$(DRIVER_DIR)/.version 2> /dev/null)
 
-SVN_INFO = (cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); svn info 2>/dev/null)
+#SVN_INFO = (cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); svn info 2>/dev/null)
 
-ifneq ($(shell $(SVN_INFO) 2>/dev/null),)
-# SVN detected
-SVN_REV := $(shell $(SVN_INFO) | grep '^Revision: '| sed -e 's/^Revision: //' 2>/dev/null)
-DRIVER_REV := $(MALI_RELEASE_NAME)-r$(SVN_REV)
-CHANGE_DATE := $(shell $(SVN_INFO) | grep '^Last Changed Date: ' | cut -d: -f2- | cut -b2-)
-CHANGED_REVISION := $(shell $(SVN_INFO) | grep '^Last Changed Rev: ' | cut -d: -f2- | cut -b2-)
-REPO_URL := $(shell $(SVN_INFO) | grep '^URL: ' | cut -d: -f2- | cut -b2-)
-
-else # SVN
-GIT_REV := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git describe --always 2>/dev/null)
-ifneq ($(GIT_REV),)
-# Git detected
-DRIVER_REV := $(MALI_RELEASE_NAME)-$(GIT_REV)
-CHANGE_DATE := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git log -1 --format="%ci")
-CHANGED_REVISION := $(GIT_REV)
-REPO_URL := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git describe --all --always 2>/dev/null)
+#ifneq ($(shell $(SVN_INFO) 2>/dev/null),)
+## SVN detected
+#SVN_REV := $(shell $(SVN_INFO) | grep '^Revision: '| sed -e 's/^Revision: //' 2>/dev/null)
+#DRIVER_REV := $(MALI_RELEASE_NAME)-r$(SVN_REV)
+#CHANGE_DATE := $(shell $(SVN_INFO) | grep '^Last Changed Date: ' | cut -d: -f2- | cut -b2-)
+#CHANGED_REVISION := $(shell $(SVN_INFO) | grep '^Last Changed Rev: ' | cut -d: -f2- | cut -b2-)
+#REPO_URL := $(shell $(SVN_INFO) | grep '^URL: ' | cut -d: -f2- | cut -b2-)
+
+#else # SVN
+#GIT_REV := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git describe --always 2>/dev/null)
+#ifneq ($(GIT_REV),)
+## Git detected
+#DRIVER_REV := $(MALI_RELEASE_NAME)-$(GIT_REV)
+#CHANGE_DATE := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git log -1 --format="%ci")
+#CHANGED_REVISION := $(GIT_REV)
+#REPO_URL := $(shell cd $(TOP_KBUILD_SRC)$(DRIVER_DIR); git describe --all --always 2>/dev/null)
 
-else # Git
+#else # Git
 # No Git or SVN detected
 DRIVER_REV := $(MALI_RELEASE_NAME)
 CHANGE_DATE := $(MALI_RELEASE_NAME)
 CHANGED_REVISION := $(MALI_RELEASE_NAME)
-endif
-endif
+#endif
+#endif
 
 ccflags-y += -DSVN_REV_STRING=\"$(DRIVER_REV)\"
 
diff -Naur linux-amlogic-3.10-fa4c242/drivers/amlogic/gpu/ump/Kbuild linux-amlogic-3.10-fa4c242.patch/drivers/amlogic/gpu/ump/Kbuild
--- linux-amlogic-3.10-fa4c242/drivers/amlogic/gpu/ump/Kbuild	2015-01-22 21:04:55.000000000 +0100
+++ linux-amlogic-3.10-fa4c242.patch/drivers/amlogic/gpu/ump/Kbuild	2015-01-23 20:18:02.544746215 +0100
@@ -32,7 +32,8 @@
 UDD_FILE_PREFIX = ../mali/
 
 # Get subversion revision number, fall back to 0000 if no svn info is available
-SVN_REV := $(shell ((svnversion | grep -qv exported && echo -n 'Revision: ' && svnversion) || git svn info | sed -e 's/$$$$/M/' | grep '^Revision: ' || echo ${MALI_RELEASE_NAME}) 2>/dev/null | sed -e 's/^Revision: //')
+#SVN_REV := $(shell ((svnversion | grep -qv exported && echo -n 'Revision: ' && svnversion) || git svn info | sed -e 's/$$$$/M/' | grep '^Revision: ' || echo ${MALI_RELEASE_NAME}) 2>/dev/null | sed -e 's/^Revision: //')
+SVN_REV := $(shell (echo ${MALI_RELEASE_NAME}) 2>/dev/null | sed -e 's/^Revision: //')
 
 ccflags-y += -DSVN_REV=$(SVN_REV)
 ccflags-y += -DSVN_REV_STRING=\"$(SVN_REV)\"
