diff --git a/arch/arm/mach-meson6/pm.c b/arch/arm/mach-meson6/pm.c
index 45c0500968f..c5de8f8ca3e 100755
--- a/arch/arm/mach-meson6/pm.c
+++ b/arch/arm/mach-meson6/pm.c
@@ -1143,6 +1143,7 @@ static struct platform_driver meson_pm_driver = {
 static int __init meson_pm_init(void)
 {
 	printk("enter %s\n",__func__);
+	pm_power_off = meson_pm_suspend;
     return platform_driver_probe(&meson_pm_driver, meson_pm_probe);
 }
 late_initcall(meson_pm_init);
