diff -Naur a/drivers/amlogic/amports/esparser.c b/drivers/amlogic/amports/esparser.c
--- a/drivers/amlogic/amports/esparser.c	2015-01-04 18:07:57.000000000 +0100
+++ b/drivers/amlogic/amports/esparser.c	2015-01-01 16:42:13.000000000 +0100
@@ -296,11 +296,14 @@
         WRITE_MPEG_REG(RESET1_REGISTER, RESET_PARSER);
 
         /* TS data path */
+		/* Ignore FEC control for WetekPlay */
+#ifndef CONFIG_WETEK
 #ifndef CONFIG_AM_DVB
         WRITE_MPEG_REG(FEC_INPUT_CONTROL, 0);
 #else
         tsdemux_set_reset_flag();
 #endif
+#endif
         CLEAR_MPEG_REG_MASK(TS_HIU_CTL, 1 << USE_HI_BSF_INTERFACE);
         CLEAR_MPEG_REG_MASK(TS_HIU_CTL_2, 1 << USE_HI_BSF_INTERFACE);
         CLEAR_MPEG_REG_MASK(TS_HIU_CTL_3, 1 << USE_HI_BSF_INTERFACE);
diff -Naur a/drivers/amlogic/amports/psparser.c b/drivers/amlogic/amports/psparser.c
--- a/drivers/amlogic/amports/psparser.c	2015-01-04 18:07:58.000000000 +0100
+++ b/drivers/amlogic/amports/psparser.c	2015-01-01 16:43:58.000000000 +0100
@@ -833,11 +833,15 @@
     WRITE_MPEG_REG(RESET1_REGISTER, RESET_PARSER);
 
     /* TS data path */
+		/* Ignore FEC control for WetekPlay */
+#ifndef CONFIG_WETEK
 #ifndef CONFIG_AM_DVB
     WRITE_MPEG_REG(FEC_INPUT_CONTROL, 0);
 #else
     tsdemux_set_reset_flag();
 #endif
+#endif
+
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL, 1 << USE_HI_BSF_INTERFACE);
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL_2, 1 << USE_HI_BSF_INTERFACE);
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL_3, 1 << USE_HI_BSF_INTERFACE);
diff -Naur a/drivers/amlogic/amports/rmparser.c b/drivers/amlogic/amports/rmparser.c
--- a/drivers/amlogic/amports/rmparser.c	2015-01-04 18:07:58.000000000 +0100
+++ b/drivers/amlogic/amports/rmparser.c	2015-01-01 16:44:39.000000000 +0100
@@ -80,11 +80,14 @@
     WRITE_MPEG_REG(RESET1_REGISTER, RESET_PARSER);
 
     /* TS data path */
+	/* Ignore FEC control for WetekPlay */
+#ifndef CONFIG_WETEK
 #ifndef CONFIG_AM_DVB
     WRITE_MPEG_REG(FEC_INPUT_CONTROL, 0);
 #else
     tsdemux_set_reset_flag();
 #endif
+#endif
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL, 1 << USE_HI_BSF_INTERFACE);
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL_2, 1 << USE_HI_BSF_INTERFACE);
     CLEAR_MPEG_REG_MASK(TS_HIU_CTL_3, 1 << USE_HI_BSF_INTERFACE);
