diff -Naur a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
--- a/drivers/media/dvb-core/dvb_frontend.c	2014-12-08 17:33:43.262657721 +0100
+++ b/drivers/media/dvb-core/dvb_frontend.c	2014-12-08 17:34:37.126207052 +0100
@@ -2416,6 +2416,9 @@
 	if(cmd == FE_SET_PROPERTY) {
 		tvps = (struct dtv_properties __user *)parg;
 
+		if(fe->ops.set_mode)
+			fe->ops.set_mode(fe, FE_QPSK);
+
 		dev_dbg(fe->dvb->device, "%s: properties.num = %d\n", __func__, tvps->num);
 		dev_dbg(fe->dvb->device, "%s: properties.props = %p\n", __func__, tvps->props);
 
