diff --git a/arch/arm/boot/dts/amlogic/wetek_play.dtd b/arch/arm/boot/dts/amlogic/wetek_play.dtd
old mode 100755
new mode 100644
index ce4f2e8..e53f762
--- a/arch/arm/boot/dts/amlogic/wetek_play.dtd
+++ b/arch/arm/boot/dts/amlogic/wetek_play.dtd
@@ -231,6 +231,8 @@ void root_func(){
 		port_name = "uart_a";
 		status = "okay";
 		dev_name = "uart_0";
+		pinctrl-names = "default";
+		pinctrl-0 = <&a_uart_pins>;
 	};
 	
 //$$ DEVICE="uart_1"
@@ -256,7 +258,7 @@ void root_func(){
 	uart_3{
 		compatible = "amlogic,aml_uart";
 		port_name = "uart_d";
-		status = "ok";
+		status = "disabled";
 		dev_name = "uart_3";
 	};
 
@@ -267,9 +269,11 @@ void root_func(){
     bt-dev{
         compatible = "amlogic,bt-dev";
         dev_name = "bt-dev";
-        gpio_reset = "GPIOE_11";
+        gpio_wake = "GPIOX_10";
         gpio_en = "GPIOE_11";
         status = "ok";
+		pinctrl-names = "default";
+		pinctrl-0 = <&btpcm_pins>;
     };
 
 /// ***************************************************************************************
@@ -985,6 +989,11 @@ void root_func(){
 		amlogic,pins = "GPIOX_17","GPIOX_18","GPIOX_19","GPIOX_20";
 		};
 		
+		btpcm_pins:btpcm_pins{
+		amlogic,setmask = <3 0x78000000>;
+		amlogic,pins = "GPIOX_4","GPIOX_5","GPIOX_6", "GPIOX_7";
+		};
+		
 //$$ MATCH "uart_ao_pin_match" = "&ao_uart_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_STR 2 = "amlogic,pins"
