--- a/arch/arm/mach-meson6/cpu.c.orig	2015-01-18 19:50:04.897840428 +0100
+++ b/arch/arm/mach-meson6/cpu.c	2015-01-18 19:50:30.241839679 +0100
@@ -59,6 +59,14 @@
 int mali_revb_flag = -1;
 //int mali_version(void)
 
+int (*get_cpu_temperature_celius)(void) = NULL;
+EXPORT_SYMBOL_GPL(get_cpu_temperature_celius);
+
+int get_cpu_temperature(void)
+{
+    return get_cpu_temperature_celius ? get_cpu_temperature_celius() : -1;
+}
+
 EXPORT_SYMBOL_GPL(mali_revb_flag);
 static int __init maliversion(char *str)
 {
