diff --git a/arch/arm/boot/dts/amlogic/wetek_play.dtd b/arch/arm/boot/dts/amlogic/wetek_play.dtd
index efc7f22..9994d09 100755
--- a/arch/arm/boot/dts/amlogic/wetek_play.dtd
+++ b/arch/arm/boot/dts/amlogic/wetek_play.dtd
@@ -794,45 +794,9 @@ void root_func(){
 			t_rea = <20>;
 			t_rhoh = <15>;
 			chip_num = <2>;
-			part_num = <9>;
-			partition = <&nand_partitions>;
+			part_num = <0>;
 			rb_detect = <1>;
 		};
-
-		nand_partitions:nand_partition{
-			logo{
-				offset=<0x0 0x4800000>;
-				size=<0x0 0x800000>;
-			};
-			aml_logo{
-				offset=<0x0 0x5800000>;
-				size=<0x0 0x800000>;
-			};
-			recovery{
-				offset=<0x0 0x6800000>;
-				size=<0x0 0x800000>;
-			};
-			boot{
-				offset=<0x0 0x8800000>;
-				size=<0x0 0x800000>;
-			};
-			system{
-				offset=<0x0 0xa800000>;
-				size=<0x0 0x40000000>;
-			};
-			cache{
-				offset=<0x0 0x4a800000>;
-				size=<0x0 0x20000000>;
-			};
-			backup{
-				offset=<0x0 0x6a800000>;
-				size=<0x0 0x10000000>;
-			};
-			userdata{
-				offset=<0xffffffff 0xffffffff>;
-				size=<0x0 0x0>;
-			};
-		};
 	};
 
 /// ***************************************************************************************
