diff -Naur iptables-1.4.20/iptables/Makefile.am iptables-1.4.20.patch/iptables/Makefile.am
--- iptables-1.4.20/iptables/Makefile.am	2013-08-06 17:48:43.000000000 +0200
+++ iptables-1.4.20.patch/iptables/Makefile.am	2013-09-01 05:30:03.118428151 +0200
@@ -38,7 +38,7 @@
 v6_sbin_links  = ip6tables ip6tables-restore ip6tables-save
 endif
 
-iptables-extensions.8: ${srcdir}/iptables-extensions.8.tmpl ../extensions/matches.man ../extensions/targets.man
+iptables-extensions.8: iptables-extensions.8.tmpl ../extensions/matches.man ../extensions/targets.man
 	${AM_VERBOSE_GEN} sed \
 		-e '/@MATCH@/ r ../extensions/matches.man' \
 		-e '/@TARGET@/ r ../extensions/targets.man' $< >$@;
