diff -Naur iptables-1.4.12.2/extensions/GNUmakefile.in iptables-1.4.12.2.patch/extensions/GNUmakefile.in
--- iptables-1.4.12.2/extensions/GNUmakefile.in	2012-01-02 18:19:09.000000000 +0100
+++ iptables-1.4.12.2.patch/extensions/GNUmakefile.in	2012-01-07 03:27:07.237205980 +0100
@@ -68,7 +68,7 @@
 
 .PHONY: all install clean distclean FORCE
 
-all: ${targets} check
+all: ${targets}
 
 check: ${targets}
 	@echo "  CHECK    unknown symbols in .so files"; \
