diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-10-18 13:22:58.000000000 -0700
+++ b/CMakeLists.txt	2016-11-16 22:21:34.314477541 -0800
@@ -4,40 +4,10 @@
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
 
 find_package(Kodi REQUIRED)
+find_package(libretro-beetle-supergrafx REQUIRED)
 
-include(ExternalProject)
+set(DEPLIBS beetle-supergrafx)
+set(BEETLE-SUPERGRAFX_CUSTOM_BINARY ${BEETLE_SUPERGRAFX_LIB}
+                          ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-  set(LIBRETRO_DEBUG DEBUG=1)
-endif()
-
-if("${CORE_SYSTEM_NAME}" STREQUAL "windows")
-  set(BUILD_COMMAND mingw32-make platform=win ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "linux")
-  set(BUILD_COMMAND make ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "darwin")
-  set(BUILD_COMMAND make platform=osx ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "ios")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "android")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "rbpi")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "freebsd")
-# TODO
-endif()
-
-externalproject_add(beetle-supergrafx
-                    GIT_REPOSITORY https://github.com/libretro/beetle-supergrafx-libretro
-                    GIT_TAG 516ede8
-                    PREFIX beetle-supergrafx
-                    CONFIGURE_COMMAND ""
-                    INSTALL_COMMAND ""
-                    BUILD_COMMAND ${BUILD_COMMAND}
-                    BUILD_IN_SOURCE 1)
-
-set(BEETLESGX_CUSTOM_BINARY ${PROJECT_BINARY_DIR}/beetle-supergrafx/src/beetle-supergrafx/mednafen_supergrafx_libretro${CMAKE_SHARED_LIBRARY_SUFFIX}
-                            ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
-
-build_addon(${PROJECT_NAME} BEETLESGX DEPLIBS)
-add_dependencies(${PROJECT_NAME} beetle-supergrafx)
+build_addon(${PROJECT_NAME} BEETLE-SUPERGRAFX DEPLIBS)
