diff -Naur build.LibreELEC-Generic.x86_64-9.0-devel/org_hyperion-355a324/include/utils/Image.h build.LibreELEC-Generic.x86_64-9.0-devel/hyperion-355a324/include/utils/Image.h
--- org_hyperion-355a324/include/utils/Image.h	2016-09-06 22:58:55.000000000 +0200
+++ hyperion-355a324/include/utils/Image.h	2017-05-30 18:18:06.043345976 +0200
@@ -2,6 +2,7 @@
 
 // STL includes
 #include <vector>
+#include <cassert>
 #include <cstdint>
 #include <cstring>
 #include <algorithm>
