From 1c134f52389f93b94fa6db291b79ba3933a031f7 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Fri, 8 Jun 2018 17:19:35 +0200
Subject: [PATCH] XBMCVideoChecker: use OnAVStart event for Kodi 18

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 libsrc/xbmcvideochecker/XBMCVideoChecker.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libsrc/xbmcvideochecker/XBMCVideoChecker.cpp b/libsrc/xbmcvideochecker/XBMCVideoChecker.cpp
index f638888..054e462 100644
--- a/libsrc/xbmcvideochecker/XBMCVideoChecker.cpp
+++ b/libsrc/xbmcvideochecker/XBMCVideoChecker.cpp
@@ -62,7 +62,8 @@ void XBMCVideoChecker::receiveReply()
 // silence - no "debug" code should be at the log
 //	std::cout << "KODICHECK INFO: Kodi Message: " << reply.toStdString() << std::endl;
 
-	if (reply.contains("\"method\":\"Player.OnPlay\""))
+	if ( (_xbmcVersion < 18 && reply.contains("\"method\":\"Player.OnPlay\"")) ||
+	     (_xbmcVersion >= 18 && reply.contains("\"method\":\"Player.OnAVStart\"")) )
 	{
 		// send a request for the current player state
 		_socket.write(_activePlayerRequest.toUtf8());
-- 
2.11.0

