diff -Naur a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2014-03-04 03:04:17.000000000 +0100
+++ b/tests/Makefile.am	2016-01-22 21:22:38.742833399 +0100
@@ -3,7 +3,7 @@
 
 libconfig_tests_SOURCES = tests.c
 
-libconfig_tests_CPPFLAGS = -I$(top_srcdir)/tinytest -I../lib
+libconfig_tests_CPPFLAGS = -I$(top_srcdir)/tinytest -I$(top_srcdir)/lib
 
 libconfig_tests_LDADD = -L$(top_builddir)/tinytest -ltinytest \
 	-L$(top_builddir)/lib/.libs -lconfig
