--- libconfig-1.7.2/Makefile.am.orig	2020-12-30 11:29:05.944066542 +0000
+++ libconfig-1.7.2/Makefile.am	2020-12-30 11:29:27.477746250 +0000
@@ -1,7 +1,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = lib doc tinytest tests
+SUBDIRS = lib
 
 if BUILDEXAMPLES
 SUBDIRS += examples
--- libconfig-1.7.2/configure.ac.orig	2020-12-30 11:27:52.216245652 +0000
+++ libconfig-1.7.2/configure.ac	2020-12-30 11:28:13.199901992 +0000
@@ -129,11 +129,8 @@
 	lib/libconfig++.pc
 	lib/libconfigConfig.cmake
 	lib/libconfig++Config.cmake
-	doc/Makefile
 	examples/Makefile
 	examples/c/Makefile
 	examples/c++/Makefile
-	tinytest/Makefile
-	tests/Makefile
 	libconfig.spec
 	)
