diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 8becb0e31814..b9c387e93955 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -3015,6 +3015,45 @@ static const struct panel_desc okaya_rs800480t_7x0gp = {
 	.bus_format = MEDIA_BUS_FMT_RGB666_1X18,
 };
 
+static const struct drm_display_mode retroflag_gpi_mode = {
+	.clock = 6400,
+	.hdisplay = 240,
+	.hsync_start = 240 + 38,
+	.hsync_end = 240 + 38 + 10,
+	.htotal = 240 + 38 + 10 + 20,
+	.vdisplay = 320,
+	.vsync_start = 320 + 20,
+	.vsync_end = 320 + 20 + 4,
+	.vtotal = 320 + 20 + 4 + 4,
+	.flags = DRM_MODE_FLAG_PVSYNC | DRM_MODE_FLAG_PHSYNC,
+};
+
+static const struct display_timing retroflag_gpi_timing = {
+	.pixelclock = { 6400000 },
+	.hactive = { 240 },
+	.hfront_porch = { 38 },
+	.hback_porch = { 20 },
+	.hsync_len = { 10 },
+	.vactive = { 320 },
+	.vfront_porch = { 20 },
+	.vback_porch = { 4 },
+	.vsync_len = { 4 },
+	.flags = DRM_MODE_FLAG_PVSYNC | DRM_MODE_FLAG_PHSYNC,
+};
+
+static const struct panel_desc retroflag_gpi = {
+	.modes = &retroflag_gpi_mode,
+	.num_modes = 1,
+	.bpc = 8,
+	.size = {
+		.width = 30,
+		.height = 40,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB666_1X24_CPADHI,
+	.bus_flags = DRM_BUS_FLAG_DE_LOW | DRM_BUS_FLAG_SYNC_DRIVE_NEGEDGE,
+	.connector_type = DRM_MODE_CONNECTOR_DPI,
+};
+
 static const struct drm_display_mode olimex_lcd_olinuxino_43ts_mode = {
 	.clock = 9000,
 	.hdisplay = 480,
@@ -4243,6 +4282,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "ontat,yx700wv03",
 		.data = &ontat_yx700wv03,
+	}, {
+		.compatible = "retroflag_gpi",
+		.data = &retroflag_gpi,
 	}, {
 		.compatible = "ortustech,com37h3m05dtc",
 		.data = &ortustech_com37h3m,
diff --git a/drivers/gpu/drm/vc4/vc4_dpi.c b/drivers/gpu/drm/vc4/vc4_dpi.c
index dde2c6014748..3aa326f1bce3 100644
--- a/drivers/gpu/drm/vc4/vc4_dpi.c
+++ b/drivers/gpu/drm/vc4/vc4_dpi.c
@@ -192,6 +192,8 @@ static void vc4_dpi_encoder_enable(struct drm_encoder *encoder)
 		dpi_c |= VC4_SET_FIELD(DPI_FORMAT_18BIT_666_RGB_1, DPI_FORMAT);
 	}
 
+	dpi_c |= DPI_OUTPUT_ENABLE_INVERT | DPI_HSYNC_NEGATE | DPI_VSYNC_NEGATE;
+
 	if (mode->flags & DRM_MODE_FLAG_NHSYNC)
 		dpi_c |= DPI_HSYNC_INVERT;
 	else if (!(mode->flags & DRM_MODE_FLAG_PHSYNC))
