From 30517cb122d0991f0ba51155c9afff6bdc2d8899 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Wed, 13 Mar 2013 21:30:55 +0200
Subject: [PATCH] cwc: syslog: move some log messages to debug

---
 src/capmt.c |    4 ++--
 src/cwc.c   |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/capmt.c b/src/capmt.c
index ac507bd..9e12e86 100644
--- a/src/capmt.c
+++ b/src/capmt.c
@@ -547,7 +547,7 @@ handle_ca0(capmt_t* capmt) {
 #endif
 
       if(ct->ct_keystate != CT_RESOLVED)
-        tvhlog(LOG_INFO, "capmt", "Obtained key for service \"%s\"",t->s_svcname);
+        tvhlog(LOG_DEBUG, "capmt", "Obtained key for service \"%s\"",t->s_svcname);
 
       ct->ct_keystate = CT_RESOLVED;
     }
@@ -851,7 +851,7 @@ capmt_table_input(struct th_descrambler *td, struct service *t,
           cce->cce_ecmsize = len;
 
           if(ct->ct_keystate != CT_RESOLVED)
-            tvhlog(LOG_INFO, "capmt",
+            tvhlog(LOG_DEBUG, "capmt",
               "Trying to obtain key for service \"%s\"",t->s_svcname);
 
           buf[9] = pmtversion;
diff --git a/src/cwc.c b/src/cwc.c
index ebafd4f..904213f 100644
--- a/src/cwc.c
+++ b/src/cwc.c
@@ -873,7 +873,7 @@ forbid:
     }
     
     if(ct->cs_keystate != CS_RESOLVED)
-      tvhlog(LOG_INFO, "cwc",
+      tvhlog(LOG_DEBUG, "cwc",
 	     "Obtained key for service \"%s\" in %"PRId64" ms, from %s:%i",
 	     t->s_svcname, delay, ct->cs_cwc->cwc_hostname,
 	     ct->cs_cwc->cwc_port);
-- 
1.7.10.4

