From 67e38551bc3acadf27b3ed4922e897348a612fdb Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Wed, 15 Oct 2014 13:35:09 +0300
Subject: [PATCH] remove werror

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index ddcc47e..0087a6c 100644
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@ PROG    := $(BUILDDIR)/tvheadend
 #
 
 CFLAGS  += -g -O2 -Wunused-result
-CFLAGS  += -Wall -Werror -Wwrite-strings -Wno-deprecated-declarations
+CFLAGS  += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
 CFLAGS  += -D_FILE_OFFSET_BITS=64
-- 
1.7.2.5

