diff -Naur tvheadend-3.9.1050/Makefile tvheadend-3.9.1050.patch/Makefile
--- tvheadend-3.9.1050/Makefile	2014-07-07 21:39:24.000000000 +0200
+++ tvheadend-3.9.1050.patch/Makefile	2014-07-14 17:57:09.483454814 +0200
@@ -28,7 +28,7 @@
 #
 
 CFLAGS  += -g -O2
-CFLAGS  += -Wall -Werror -Wwrite-strings -Wno-deprecated-declarations
+CFLAGS  += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
 CFLAGS  += -D_FILE_OFFSET_BITS=64
