diff -Naur melonds-0e1f06da626cbe67215c3f06f6bdf510dd4e4649/Makefile melonds-0e1f06da626cbe67215c3f06f6bdf510dd4e4649-2/Makefile
--- melonds-0e1f06da626cbe67215c3f06f6bdf510dd4e4649/Makefile	2022-12-15 08:33:23.910984766 +0100
+++ melonds-0e1f06da626cbe67215c3f06f6bdf510dd4e4649-2/Makefile	2022-12-15 08:34:18.552243209 +0100
@@ -236,7 +236,7 @@
    TARGET := $(TARGET_NAME)_libretro.$(EXT)
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
-   LIBS += -lpthread -lGLESv2
+   LIBS += -lpthread -lGLESv2 -lrt
    HAVE_OPENGLES3 = 1
    HAVE_THREADS = 1
    JIT_ARCH = aarch64
