diff --git a/Makefile b/Makefile
index ede55b3..86ccc10 100644
--- a/Makefile
+++ b/Makefile
@@ -50,6 +50,14 @@ TARGET_NAME := melonds
 LIBS		    = -lm
 DEFINES :=
 
+ifeq ($(HAVE_OPENGL), 1)
+LIBS += -lGL
+endif
+
+ifeq ($(HAVE_OPENGLES3), 1)
+LIBS += -lGLESv2
+endif
+
 # GIT HASH
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
@@ -72,10 +80,6 @@ ifeq ($(platform), unix)
    CFLAGS += -D_GNU_SOURCE
    LIBS += -lpthread -lrt
    HAVE_THREADS=1
-   ifneq ($(filter $(ARCH),x86 x86_64),)
-     LIBS += -lGL
-     HAVE_OPENGL=1
-   endif
    ifeq ($(ARCH),x86_64)
       JIT_ARCH=x64
    else ifeq ($(ARCH),arm64)
