diff --git a/Makefile b/Makefile
index ede55b3..86ccc10 100644
--- a/Makefile
+++ b/Makefile
@@ -50,6 +50,10 @@ TARGET_NAME := melonds
 LIBS		    = -lm
 DEFINES :=
 
+ifeq ($(HAVE_OPENGL), 1)
+LIBS += -lGL
+endif
+
 ifneq ($(findstring Haiku,$(shell uname -s)),)
 LIBS += -lnetwork
 endif
@@ -72,10 +76,6 @@ ifeq ($(platform), unix)
    CFLAGS += -D_GNU_SOURCE
    LIBS += -lpthread -lrt
    HAVE_THREADS=1
-   ifneq ($(filter $(ARCH),x86 x86_64),)
-     LIBS += -lGL
-     HAVE_OPENGL=1
-   endif
    ifeq ($(ARCH),x86_64)
       JIT_ARCH=x64
    else ifeq ($(ARCH),arm64)
