diff -Naur a/Makefile.libretro b/Makefile.libretro
--- a/Makefile.libretro	2020-03-17 06:53:23.000000000 +0100
+++ b/Makefile.libretro	2020-06-22 16:00:10.616134362 +0200
@@ -380,6 +380,14 @@
 	ARCH = arm
 	DYNAREC = ari64
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	SHARED := -shared -Wl,--no-undefined
+	fpic := -fPIC
+	DRC_CACHE_BASE = 0
+	ARCH = aarch64
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
