From d79df6249723586473108c92ab5ecf2ac3d9341d Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Wed, 10 Feb 2021 13:53:55 +0100
Subject: [PATCH] Makefile.libretro: disable THREAD_RENDERING for
 ARCH=arm/aarch64

---
 Makefile.libretro | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index 60feab41..ffb2c7f3 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -48,7 +48,6 @@ EXTRA_LDFLAGS =
 ifeq ($(platform), unix)
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
-	THREAD_RENDERING = 1
 ifneq ($(findstring SunOS,$(shell uname -s)),)
 	CC = gcc
 endif
@@ -59,6 +58,12 @@ ifneq ($(findstring __thumb__,$(shell $(CC) -E -dD $(CFLAGS) include/arm_feature
 endif
 endif
 
+ifeq ($(ARCH), $(filter $(ARCH), arm aarch64))
+	THREAD_RENDERING = 0
+else
+	THREAD_RENDERING = 1
+endif
+
 # ODROIDN2
 else ifneq (,$(findstring CortexA73_G12B,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
