diff -Naur a/Makefile.libretro b/Makefile.libretro
--- a/Makefile.libretro	2016-05-05 20:28:57.000000000 -0700
+++ b/Makefile.libretro	2016-05-10 10:10:33.447118136 -0700
@@ -246,6 +246,14 @@
 	ARCH = arm
 	USE_DYNAREC = 1
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+        TARGET := $(TARGET_NAME)_libretro.so
+        SHARED := -shared -Wl,--no-undefined
+        fpic := -fPIC
+        DRC_CACHE_BASE = 0
+        ARCH = aarch64
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
