diff -Naur pyvorbis-1.4/setup.py pyvorbis-1.4.patch/setup.py
--- pyvorbis-1.4/setup.py	2003-12-19 08:55:56.000000000 +0100
+++ pyvorbis-1.4.patch/setup.py	2009-04-03 00:41:53.588072664 +0200
@@ -10,14 +10,6 @@
 VERSION_MINOR = 4
 pyvorbis_version = str(VERSION_MAJOR) + '.' + str(VERSION_MINOR)
 
-try:
-    import ogg._ogg
-except ImportError:
-    print '''You must have the Ogg Python bindings
-installed in order to build and install
-these bindings. Import of ogg._ogg failed.'''
-    sys.exit(1)
-
 def get_setup():
     data = {}
     r = re.compile(r'(\S+)\s*?=\s*(.+)')
