From e1695e99974953cdbefdb145c5271c1c573893ed Mon Sep 17 00:00:00 2001
From: Patrick Wood <patrickhwood@gmail.com>
Date: Sat, 11 Jan 2014 01:16:23 -0500
Subject: [PATCH] 1 second boot delay

---
 include/configs/sunxi-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index a0c41ac..4601a22 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -272,7 +272,7 @@
 	  "\0" \
 	""
 
-#define CONFIG_BOOTDELAY	3
+#define CONFIG_BOOTDELAY	0
 #define CONFIG_SYS_BOOT_GET_CMDLINE
 #define CONFIG_AUTO_COMPLETE
 
