From 82f37399e10b472109dc614ade08e5d9ebd9ae49 Mon Sep 17 00:00:00 2001
From: Patrick Wood <patrickhwood@gmail.com>
Date: Sat, 11 Jan 2014 01:15:52 -0500
Subject: [PATCH] dual cubieboard2 cubietruck dram setup

---
 board/sunxi/dram_cubietruck.c | 29 +++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/board/sunxi/dram_cubietruck.c b/board/sunxi/dram_cubietruck.c
index 1e7c94a..65dfa07 100644
--- a/board/sunxi/dram_cubietruck.c
+++ b/board/sunxi/dram_cubietruck.c
@@ -3,7 +3,29 @@
 #include <common.h>
 #include <asm/arch/dram.h>
 
-static struct dram_para dram_para = {
+static struct dram_para dram_para_cb2 = {
+	.clock = 480,
+	.type = 3,
+	.rank_num = 1,
+	.density = 4096,
+	.io_width = 16,
+	.bus_width = 32,
+	.cas = 9,
+	.zq = 0x7f,
+	.odt_en = 0,
+	.size = 1024,
+	.tpr0 = 0x42d899b7,
+	.tpr1 = 0xa090,
+	.tpr2 = 0x22a00,
+	.tpr3 = 0x0,
+	.tpr4 = 0x1,
+	.tpr5 = 0x0,
+	.emr1 = 0x4,
+	.emr2 = 0x10,
+	.emr3 = 0x0,
+};
+
+static struct dram_para dram_para_ct = {
 	.clock = 432,
 	.type = 3,
 	.rank_num = 1,
@@ -27,5 +49,8 @@ static struct dram_para dram_para = {
 
 unsigned long sunxi_dram_init(void)
 {
-	return dramc_init(&dram_para);
+	unsigned long ram_size = dramc_init(&dram_para_ct);
+	if (ram_size < 2000000000)
+		return dramc_init(&dram_para_cb2);
+	return ram_size;
 }
