diff --git a/gdk-pixbuf/Makefile.am b/gdk-pixbuf/Makefile.am
index 8319b49..8f27c5c 100644
--- a/gdk-pixbuf/Makefile.am
+++ b/gdk-pixbuf/Makefile.am
@@ -617,51 +617,21 @@ MAINTAINERCLEANFILES = 		\
 #
 # gdk-pixbuf-enum-types.h
 #
-gdk-pixbuf-enum-types.h: s-enum-types-h
+gdk-pixbuf-enum-types.h:
 	@true
 
-s-enum-types-h: @REBUILD@ $(gdk_pixbuf_headers) gdk-pixbuf-enum-types.h.template
-	$(AM_V_GEN)( cd $(srcdir) && $(GLIB_MKENUMS) --template gdk-pixbuf-enum-types.h.template \
-		$(gdk_pixbuf_headers) ) > tmp-gdk-pixbuf-enum-types.h \
-	&& (cmp -s tmp-gdk-pixbuf-enum-types.h gdk-pixbuf-enum-types.h || cp tmp-gdk-pixbuf-enum-types.h gdk-pixbuf-enum-types.h ) \
-	&& rm -f tmp-gdk-pixbuf-enum-types.h \
-	&& echo timestamp > $(@F)
-
 CLEANFILES += tmp-gdk-pixbuf-enum-types.h
 MAINTAINERCLEANFILES += s-enum-types-h
 
 #
-# gdk-pixbuf-enum-types.c
-#
-gdk-pixbuf-enum-types.c: @REBUILD@ $(gdk_pixbuf_headers) gdk-pixbuf-enum-types.c.template
-	$(AM_V_GEN)(cd $(srcdir) && $(GLIB_MKENUMS) --template gdk-pixbuf-enum-types.c.template \
-		  $(gdk_pixbuf_headers)) > gdk-pixbuf-enum-types.c
-
-#
 # gdk-pixbuf-marshal.h
 #
-gdk-pixbuf-marshal.h: @REBUILD@ stamp-gdk-pixbuf-marshal.h
+gdk-pixbuf-marshal.h:
 	@true
 
-stamp-gdk-pixbuf-marshal.h: @REBUILD@ $(srcdir)/gdk-pixbuf-marshal.list
-	$(AM_V_GEN)echo "#if !defined(GDK_PIXBUF_DISABLE_DEPRECATED) || defined(GDK_PIXBUF_COMPILATION)" > xgen-gmh \
-	&& $(GLIB_GENMARSHAL) --prefix=_gdk_pixbuf_marshal $(srcdir)/gdk-pixbuf-marshal.list --header >> xgen-gmh \
-	&& echo "#endif /* !GDK_PIXBUF_DISABLE_DEPRECATED || GDK_PIXBUF_COMPILATION */" >> xgen-gmh \
-	&& (cmp -s xgen-gmh gdk-pixbuf-marshal.h || cp xgen-gmh gdk-pixbuf-marshal.h) \
-	&& rm -f xgen-gmh xgen-gmh~ \
-	&& echo timestamp > $(@F)
-
 CLEANFILES += xgen-gmh
 MAINTAINERCLEANFILES += stamp-gdk-pixbuf-marshal.h
 
-#
-# gdk-pixbuf-marshal.c
-#
-$(srcdir)/gdk-pixbuf-marshal.c: @REBUILD@ $(srcdir)/gdk-pixbuf-marshal.list
-	$(AM_V_GEN)(echo -e "#include <gdk-pixbuf/gdk-pixbuf.h>\n" | $(GLIB_GENMARSHAL) --prefix=_gdk_pixbuf_marshal $(srcdir)/gdk-pixbuf-marshal.list --body ) >> xgen-gmc \
-	&& cp xgen-gmc gdk-pixbuf-marshal.c \
-	&& rm -f xgen-gmc xgen-gmc~
-
 CLEANFILES += xgen-gmc
 
 # if srcdir!=builddir, clean out maintainer-clean files from builddir
