diff -Naur iftop-1.0pre2/configure.in iftop-1.0pre2.patch/configure.in
--- iftop-1.0pre2/configure.in	2011-10-03 23:55:33.000000000 +0200
+++ iftop-1.0pre2.patch/configure.in	2013-03-05 11:17:10.423257608 +0100
@@ -28,7 +28,7 @@
 
 AC_CANONICAL_SYSTEM
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE(iftop, "1.0pre2")
 
 AC_DEFINE_UNQUOTED(IFTOP_VERSION, "$VERSION", [The iftop version number])
