diff -Naur splashutils-1.5.4.4/src/fbsplash.h splashutils-1.5.4.4.patch/src/fbsplash.h
--- splashutils-1.5.4.4/src/fbsplash.h	2011-06-12 23:45:22.000000000 +0200
+++ splashutils-1.5.4.4.patch/src/fbsplash.h	2012-02-16 07:54:55.028245671 +0100
@@ -6,7 +6,7 @@
 #include <linux/kd.h>
 #include <linux/input.h>
 
-#define FBSPLASH_DIR		"//lib/splash"
+#define FBSPLASH_DIR		"/run/splash"
 #define FBSPLASH_CACHEDIR	FBSPLASH_DIR"/cache"
 #define FBSPLASH_PIDFILE	FBSPLASH_CACHEDIR"/daemon.pid"
 #define FBSPLASH_PROFILE	FBSPLASH_CACHEDIR"/profile"
diff -Naur splashutils-1.5.4.4/src/fbsplash.h.in splashutils-1.5.4.4.patch/src/fbsplash.h.in
--- splashutils-1.5.4.4/src/fbsplash.h.in	2011-06-12 23:43:39.000000000 +0200
+++ splashutils-1.5.4.4.patch/src/fbsplash.h.in	2012-02-16 07:54:39.069954498 +0100
@@ -6,7 +6,7 @@
 #include <linux/kd.h>
 #include <linux/input.h>
 
-#define FBSPLASH_DIR		"@libdir@/splash"
+#define FBSPLASH_DIR		"/run/splash"
 #define FBSPLASH_CACHEDIR	FBSPLASH_DIR"/cache"
 #define FBSPLASH_PIDFILE	FBSPLASH_CACHEDIR"/daemon.pid"
 #define FBSPLASH_PROFILE	FBSPLASH_CACHEDIR"/profile"
