From 6d64d70a46b8f238d2706017a084f30bd681f291 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 31 Aug 2013 13:44:53 +0300
Subject: [PATCH] show all removable disks  mounted under /media

---
 xbmc/storage/linux/UDevProvider.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xbmc/storage/linux/UDevProvider.cpp b/xbmc/storage/linux/UDevProvider.cpp
index c20facc..c1044c8 100644
--- a/xbmc/storage/linux/UDevProvider.cpp
+++ b/xbmc/storage/linux/UDevProvider.cpp
@@ -151,7 +151,7 @@ void CUDevProvider::GetDisks(VECSOURCES& disks, bool removable)
     if (removable  &&
       ((bus        && strstr(bus, "usb")) ||
        (cdrom      && strstr(cdrom,"1"))  ||
-       (mountpoint && strstr(mountpoint, "usb"))))
+       (mountpoint && strstr(mountpoint, "/media/"))))
     {
       const char *label = udev_device_get_property_value(device, "ID_FS_LABEL");
       if (!label)
-- 
1.7.2.5

