Workaround for 'Argument list too long' error.

--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -394,7 +394,7 @@
 #
 
 quiet_cmd_ar_builtin = AR      $@
-      cmd_ar_builtin = rm -f $@; $(AR) cDPrST $@ $(real-prereqs)
+      cmd_ar_builtin = rm -f $@; $(if $(findstring drivers/base/firmware_loader/builtin/built-in.a,$@),$(AR) cDPrST $@ $(sort $(foreach f,$(real-prereqs),$(dir $(f))*.o)),$(AR) cDPrST $@ $(real-prereqs))
 
 $(obj)/built-in.a: $(real-obj-y) FORCE
 	$(call if_changed,ar_builtin)
