--- a/arch/x86/kernel/cpu/mtrr/main.c	2015-10-18 08:10:52.951166026 +0100
+++ b/arch/x86/kernel/cpu/mtrr/main.c	2015-10-18 08:11:54.651493742 +0100
@@ -448,6 +448,7 @@
 	return mtrr_add_page(base >> PAGE_SHIFT, size >> PAGE_SHIFT, type,
 			     increment);
 }
+EXPORT_SYMBOL(mtrr_add);
 
 /**
  * mtrr_del_page - delete a memory type region
@@ -536,6 +537,7 @@
 		return -EINVAL;
 	return mtrr_del_page(reg, base >> PAGE_SHIFT, size >> PAGE_SHIFT);
 }
+EXPORT_SYMBOL(mtrr_del);
 
 /**
  * arch_phys_wc_add - add a WC MTRR and handle errors if PAT is unavailable
