--- a/Utilities/cmcurl/lib/vtls/openssl.c.orig	2016-10-19 19:17:49.615923691 +0200
+++ b/Utilities/cmcurl/lib/vtls/openssl.c	2016-10-19 19:22:50.849565684 +0200
@@ -134,6 +134,11 @@
 #define HAVE_SSL_COMP_FREE_COMPRESSION_METHODS 1
 #endif
 
+#if !defined(OPENSSL_NO_COMP) || \
+  OPENSSL_VERSION_NUMBER >= 0x10100000L /* 1.1.0+ has no free compression methods */ 
+#undef HAVE_SSL_COMP_FREE_COMPRESSION_METHODS /* undef */
+#endif
+
 #if (OPENSSL_VERSION_NUMBER < 0x0090808fL)
 /* not present in older OpenSSL */
 #define OPENSSL_load_builtin_modules(x)
