--- a/Makefile.am
+++ b/Makefile.am
@@ -22,8 +22,6 @@ SUBDIRS = \
 	src \
 	$(proxy_dirs) \
 	contrib \
-	examples \
-	test \
 	po \
 	doc
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -78,7 +78,7 @@ pkgconfigDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(pkgconfig_DATA)
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = m4 src daemon contrib examples test po doc
+DIST_SUBDIRS = m4 src daemon contrib po doc
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -268,8 +268,6 @@ SUBDIRS = \
 	src \
 	$(proxy_dirs) \
 	contrib \
-	examples \
-	test \
 	po \
 	doc
 
--- a/configure
+++ b/configure
@@ -18591,7 +18591,7 @@ else
 fi
 
 
-ac_config_files="$ac_config_files Makefile contrib/Makefile examples/Makefile daemon/Makefile daemon/zvbid.init doc/Doxyfile doc/Makefile m4/Makefile src/Makefile src/dvb/Makefile test/Makefile po/Makefile.in zvbi.spec zvbi-0.2.pc"
+ac_config_files="$ac_config_files Makefile contrib/Makefile daemon/Makefile daemon/zvbid.init doc/Doxyfile doc/Makefile m4/Makefile src/Makefile src/dvb/Makefile po/Makefile.in zvbi.spec zvbi-0.2.pc"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -19726,7 +19726,6 @@ do
     "po-directories") CONFIG_COMMANDS="$CONFIG_COMMANDS po-directories" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
     "contrib/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/Makefile" ;;
-    "examples/Makefile") CONFIG_FILES="$CONFIG_FILES examples/Makefile" ;;
     "daemon/Makefile") CONFIG_FILES="$CONFIG_FILES daemon/Makefile" ;;
     "daemon/zvbid.init") CONFIG_FILES="$CONFIG_FILES daemon/zvbid.init" ;;
     "doc/Doxyfile") CONFIG_FILES="$CONFIG_FILES doc/Doxyfile" ;;
@@ -19734,7 +19733,6 @@ do
     "m4/Makefile") CONFIG_FILES="$CONFIG_FILES m4/Makefile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "src/dvb/Makefile") CONFIG_FILES="$CONFIG_FILES src/dvb/Makefile" ;;
-    "test/Makefile") CONFIG_FILES="$CONFIG_FILES test/Makefile" ;;
     "po/Makefile.in") CONFIG_FILES="$CONFIG_FILES po/Makefile.in" ;;
     "zvbi.spec") CONFIG_FILES="$CONFIG_FILES zvbi.spec" ;;
     "zvbi-0.2.pc") CONFIG_FILES="$CONFIG_FILES zvbi-0.2.pc" ;;
--- a/configure.in
+++ b/configure.in
@@ -384,7 +384,6 @@ AM_CONDITIONAL(BUILD_STATIC_LIB, [test "x$enable_static" = xyes])
 AC_OUTPUT([
   Makefile
   contrib/Makefile
-  examples/Makefile
   daemon/Makefile
   daemon/zvbid.init
   doc/Doxyfile
@@ -392,7 +391,6 @@ AC_OUTPUT([
   m4/Makefile
   src/Makefile
   src/dvb/Makefile
-  test/Makefile
   po/Makefile.in
   zvbi.spec
   zvbi-0.2.pc
