diff -Nur a/include/os/rt_linux.h b/include/os/rt_linux.h
--- a/include/os/rt_linux.h	2016-04-12 21:56:36.000000000 +0200
+++ b/include/os/rt_linux.h	2016-04-12 21:59:47.571744543 +0200
@@ -277,8 +277,13 @@
 
 typedef struct _OS_FS_INFO_
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,5,0)
+  	uid_t				fsuid;
+	gid_t				fsgid;
+#else
 	kuid_t				fsuid;
 	kgid_t				fsgid;
+#endif
 	mm_segment_t	fs;
 } OS_FS_INFO;
 
