diff -Naur retroarch.git/gfx/drivers_context/mali_fbdev_ctx.c retroarch.patch/gfx/drivers_context/mali_fbdev_ctx.c
--- retroarch.git/gfx/drivers_context/mali_fbdev_ctx.c	2015-12-15 04:30:48.451470567 +0100
+++ retroarch.patch/gfx/drivers_context/mali_fbdev_ctx.c	2015-12-17 05:12:21.710615201 +0100
@@ -32,7 +32,7 @@
 typedef struct {
    egl_ctx_data_t egl;
 
-   struct mali_native_window native_window;
+   struct fbdev_window native_window;
    bool resize;
    unsigned width, height;
 } mali_ctx_data_t;
