diff -Naur busybox-1.15.0/networking/telnetd.c busybox-1.15.0a/networking/telnetd.c
--- busybox-1.15.0/networking/telnetd.c	2009-08-21 00:26:14.000000000 +0200
+++ busybox-1.15.0a/networking/telnetd.c	2009-08-30 11:38:57.000000000 +0200
@@ -63,7 +63,7 @@
 };
 #define G (*(struct globals*)&bb_common_bufsiz1)
 #define INIT_G() do { \
-	G.loginpath = "/bin/login"; \
+	G.loginpath = "/bin/sh"; \
 	G.issuefile = "/etc/issue.net"; \
 } while (0)
 
