From e7f1e5ca1de60db1336f0ec85b1c9df55f6402b7 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Tue, 04 May 2010 15:34:37 +0000
Subject: bootchartd: add missing parenthesis

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/init/bootchartd.c b/init/bootchartd.c
index fccaace..9c43e33 100644
--- a/init/bootchartd.c
+++ b/init/bootchartd.c
@@ -175,7 +175,7 @@ static void do_logging(void)
 	//FILE *proc_netdev = xfopen("proc_netdev.log", "w");
 	FILE *proc_ps = xfopen("proc_ps.log", "w");
 	int look_for_login_process = (getppid() == 1);
-	unsigned count = 60*1000*1000 / 200*1000; /* ~1 minute */
+	unsigned count = 60*1000*1000 / (200*1000); /* ~1 minute */
 
 	while (--count && !bb_got_signal) {
 		char *p;
@@ -200,8 +200,8 @@ static void do_logging(void)
 			/* dump_procs saw a getty or {g,k,x}dm
 			 * stop logging in 2 seconds:
 			 */
-			if (count > 2*1000*1000 / 200*1000)
-				count = 2*1000*1000 / 200*1000;
+			if (count > 2*1000*1000 / (200*1000))
+				count = 2*1000*1000 / (200*1000);
 		}
 		fflush_all();
  wait_more:
--
cgit v0.8.2.1
