From b8ba6b66f5d730efcf13dbbb3f9362dd6840d93b Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Tue, 04 May 2010 22:40:15 +0000
Subject: bootchartd: better wording in comment

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
diff --git a/init/bootchartd.c b/init/bootchartd.c
index 9c43e33..9faf14d 100644
--- a/init/bootchartd.c
+++ b/init/bootchartd.c
@@ -306,11 +306,12 @@ int bootchartd_main(int argc UNUSED_PARAM, char **argv)
 			/* Inform parent that we are ready */
 			raise(SIGSTOP);
 
-		/* If we started by kernel, PATH might be not set.
-		 * And in order to run tar we may need PATH to be set:
+		/* If we are started by kernel, PATH might be unset.
+		 * In order to find "tar", let's set some sane PATH:
 		 */
 		if (cmd == CMD_PID1 && !getenv("PATH"))
 			putenv((char*)bb_PATH_root_path);
+
 		tempdir = make_tempdir(cmd == CMD_START ? argv[2] : NULL);
 		do_logging();
 		finalize(tempdir);
--
cgit v0.8.2.1
