diff -Naur busybox-1.24.0/scripts/trylink busybox-1.24.0.patch/scripts/trylink
--- busybox-1.24.0/scripts/trylink	2015-10-12 14:37:58.000000000 +0200
+++ busybox-1.24.0.patch/scripts/trylink	2015-10-13 18:49:50.537508072 +0200
@@ -49,7 +49,7 @@
     local tempname="$(mktemp)"
     # Can use "-o /dev/null", but older gcc tend to *unlink it* on failure! :(
     # "-xc": C language. "/dev/null" is an empty source file.
-    if $CC $CPPFLAGS $CFLAGS $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
+    if $CC $CPPFLAGS $CFLAGS $LDFLAGS $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
 	echo "$1";
     else
 	echo "$2";
