diff -urpN busybox-1.17.0/archival/tar.c busybox-1.17.0-cosmetics/archival/tar.c
--- busybox-1.17.0/archival/tar.c	2010-07-04 15:32:24.000000000 +0200
+++ busybox-1.17.0-cosmetics/archival/tar.c	2010-07-19 00:26:04.000000000 +0200
@@ -237,7 +237,7 @@ static int writeTarHeader(struct TarBall
 	struct tar_header_t header;
 
 	memset(&header, 0, sizeof(header));
- 
+
 	strncpy(header.name, header_name, sizeof(header.name));
 
 	/* POSIX says to mask mode with 07777. */
diff -urpN busybox-1.17.0/coreutils/realpath.c busybox-1.17.0-cosmetics/coreutils/realpath.c
--- busybox-1.17.0/coreutils/realpath.c	2010-06-24 04:40:43.000000000 +0200
+++ busybox-1.17.0-cosmetics/coreutils/realpath.c	2010-07-19 00:26:04.000000000 +0200
@@ -23,7 +23,7 @@ int realpath_main(int argc UNUSED_PARAM,
 
 	do {
 		char *resolved_path = xmalloc_realpath(*argv);
-	       	if (resolved_path != NULL) {
+		if (resolved_path != NULL) {
 			puts(resolved_path);
 			free(resolved_path);
 		} else {
diff -urpN busybox-1.17.0/util-linux/fdisk.c busybox-1.17.0-cosmetics/util-linux/fdisk.c
--- busybox-1.17.0/util-linux/fdisk.c	2010-06-24 04:40:43.000000000 +0200
+++ busybox-1.17.0-cosmetics/util-linux/fdisk.c	2010-07-19 00:26:04.000000000 +0200
@@ -2805,7 +2805,7 @@ list_devs_in_proc_partititons(void)
 				&ma, &mi, &sz, ptname) != 4)
 			continue;
 		for (s = ptname; *s; s++)
-    			continue;
+			continue;
 		/* note: excluding '0': e.g. mmcblk0 is not a partition name! */
 		if (s[-1] >= '1' && s[-1] <= '9')
 			continue;
diff -urpN busybox-1.17.0/util-linux/more.c busybox-1.17.0-cosmetics/util-linux/more.c
--- busybox-1.17.0/util-linux/more.c	2010-06-24 04:40:43.000000000 +0200
+++ busybox-1.17.0-cosmetics/util-linux/more.c	2010-07-19 00:26:04.000000000 +0200
@@ -31,7 +31,7 @@ struct globals {
 
 #define setTermSettings(fd, argp) do { \
 		if (ENABLE_FEATURE_USE_TERMIOS) tcsetattr(fd, TCSANOW, argp); \
-	} while(0)
+	} while (0)
 #define getTermSettings(fd, argp) tcgetattr(fd, argp)
 
 static void gotsig(int sig UNUSED_PARAM)
@@ -46,7 +46,7 @@ static void gotsig(int sig UNUSED_PARAM)
 int more_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int more_main(int argc UNUSED_PARAM, char **argv)
 {
-	int c = c; /* for gcc */
+	int c = c; /* for compiler */
 	int lines;
 	int input = 0;
 	int spaces = 0;
