diff -urpN busybox-1.16.1/networking/httpd_indexcgi.c busybox-1.16.1-indexcgi/networking/httpd_indexcgi.c
--- busybox-1.16.1/networking/httpd_indexcgi.c	2010-03-20 03:58:07.000000000 +0100
+++ busybox-1.16.1-indexcgi/networking/httpd_indexcgi.c	2010-04-15 17:39:01.000000000 +0200
@@ -315,7 +315,7 @@ int main(int argc, char *argv[])
 		if (S_ISREG(cdir->dl_mode))
 			fmt_ull(cdir->dl_size);
 		fmt_str("<td class=dt>");
-		tm = gmtime(&cdir->dl_mtime);
+		ptm = gmtime(&cdir->dl_mtime);
 		fmt_04u(1900 + ptm->tm_year); *dst++ = '-';
 		fmt_02u(ptm->tm_mon + 1); *dst++ = '-';
 		fmt_02u(ptm->tm_mday); *dst++ = ' ';
