diff -Naur busybox-1.18.0/modutils/modprobe.c busybox-1.18.0.patch/modutils/modprobe.c
--- busybox-1.18.0/modutils/modprobe.c	2010-11-22 21:43:22.000000000 +0100
+++ busybox-1.18.0.patch/modutils/modprobe.c	2010-11-30 08:46:36.961605729 +0100
@@ -560,6 +560,7 @@
 
 	read_config("/etc/modprobe.conf");
 	read_config("/etc/modprobe.d");
+	read_config("/storage/.config/modprobe.d");
 	if (ENABLE_FEATURE_MODUTILS_SYMBOLS && G.need_symbols)
 		read_config("modules.symbols");
 	load_modules_dep();
