diff -Naur busybox-1.17.3-old/scripts/trylink busybox-1.17.3-new/scripts/trylink
--- busybox-1.17.3-old/scripts/trylink	2010-09-16 16:05:35.000000000 -0700
+++ busybox-1.17.3-new/scripts/trylink	2010-10-10 07:38:03.000000000 -0700
@@ -49,7 +49,7 @@
     local tempname="/tmp/temp.$$.$RANDOM"
     # Can use "-o /dev/null", but older gcc tend to *unlink it* on failure! :(
     # "-xc": C language. "/dev/null" is an empty source file.
-    if $CC $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
+    if $CC $CFLAGS $LDFLAGS $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
 	echo "$1";
     else
 	echo "$2";
