diff -Naur busybox-1.19.1/libbb/match_fstype.c busybox-1.19.1.patch/libbb/match_fstype.c
--- busybox-1.19.1/libbb/match_fstype.c	2011-08-28 12:59:56.000000000 +0200
+++ busybox-1.19.1.patch/libbb/match_fstype.c	2011-09-03 02:10:51.480283260 +0200
@@ -10,10 +10,10 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 
-#ifdef HAVE_MNTENT_H
-
 #include "libbb.h"
 
+#ifdef HAVE_MNTENT_H
+
 int FAST_FUNC match_fstype(const struct mntent *mt, const char *t_fstype)
 {
 	int match = 1;
