diff -urN a/include/configs/udoo.h b/include/configs/udoo.h
--- a/include/configs/udoo.h	2014-11-29 18:58:38.930773257 +0100
+++ b/include/configs/udoo.h	2014-11-29 19:04:06.005449364 +0100
@@ -126,8 +126,9 @@
 #define CONFIG_DEFAULT_FDT_FILE		"imx6q-udoo.dtb"
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"zImage=KERNEL\0" \
 	"script=boot.scr\0" \
-	"image=zImage\0" \
+	"image=${zImage}\0" \
 	"console=ttymxc1\0" \
 	"splashpos=m,m\0" \
 	"fdt_high=0xffffffff\0" \
@@ -168,9 +169,9 @@
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
-	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
-	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} /dtbs/${fdt_file}\0" \
+	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${zImage}\0" \
+	"loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${zImage}\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"bootz ${loadaddr} - ${fdt_addr};\0" \
