diff -aurN a/arch/arm/cpu/armv7/cpu.c b/arch/arm/cpu/armv7/cpu.c
--- a/arch/arm/cpu/armv7/cpu.c	2015-05-31 21:32:52.821314218 +0200
+++ b/arch/arm/cpu/armv7/cpu.c	2015-05-31 21:27:12.292623257 +0200
@@ -77,7 +77,7 @@
 	cpu_cache_initialization();
 
 #if (defined(MACH_TYPE_UDOO) || defined(CONFIG_VIDEO_ENABLED))
-	do_disable_lvds();
+	/*do_disable_lvds();*/
 #endif
 
 	return 0;
diff -aurN u-boot-imx6-144b1e9/board/udoo/udoo.c a-u/board/udoo/udoo.c
--- u-boot-imx6-144b1e9/board/udoo/udoo.c	2015-05-31 21:32:52.821314218 +0200
+++ a-u/board/udoo/udoo.c	2015-05-31 21:36:41.524566717 +0200
@@ -260,8 +260,8 @@
 static void do_enable_lvds(struct display_info_t const *dev)
 {
 	SETUP_IOMUX_PADS(lvds_pads);
-	gpio_direction_output(IMX_GPIO_NR(1, 2), 1); /* LVDS power On */
-	gpio_direction_output(IMX_GPIO_NR(1, 4), 1); /* LVDS backlight On */
+	/*gpio_direction_output(IMX_GPIO_NR(1, 2), 1);*/ /* LVDS power On */
+	/*gpio_direction_output(IMX_GPIO_NR(1, 4), 1);*/ /* LVDS backlight On */
         return;
 }
 