diff -Naur mythtv-21776/libs/libmyth/getloadavg.c mythtv-21776.patch/libs/libmyth/getloadavg.c
--- mythtv-21776/libs/libmyth/getloadavg.c	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-21776.patch/libs/libmyth/getloadavg.c	2009-09-11 19:45:43.072580514 +0200
@@ -0,0 +1,30 @@
+#include <stdio.h>
+
+// Quick and dirty getloadavg() replacement
+int getloadavg (double loadavg[], int nelem)
+{
+    FILE * pla;
+
+    if (nelem < 0)
+        return -1;
+    pla = fopen ("/proc/loadavg", "r");
+    if (pla == NULL)
+        return -1;
+
+    switch (nelem)
+    {
+        case 0:
+            break;
+        case 1:
+            nelem = fscanf (pla, "%lf ", &loadavg[0]);
+        case 2:
+            nelem = fscanf (pla, "%lf %lf ", &loadavg[0], &loadavg[1]);
+        case 3:
+        default:
+            nelem = fscanf (pla, "%lf %lf %lf ", &loadavg[0], &loadavg[1], &loadavg[2]);
+    }
+
+    fclose (pla);
+
+    return nelem;
+}
diff -Naur mythtv-21776/libs/libmyth/libmyth.pro mythtv-21776.patch/libs/libmyth/libmyth.pro
--- mythtv-21776/libs/libmyth/libmyth.pro	2009-09-11 18:20:40.000000000 +0200
+++ mythtv-21776.patch/libs/libmyth/libmyth.pro	2009-09-11 19:44:52.084581460 +0200
@@ -126,6 +126,9 @@
 }
 
 uclibc {
+    QMAKE_CFLAGS += -include mythuclibcfix.h
+    QMAKE_CXXFLAGS += -include mythuclibcfix.h
+    SOURCES += getloadavg.c
     SOURCES += mediamonitor-uclibc.cpp
     HEADERS += mediamonitor-uclibc.h
 }
diff -Naur mythtv-21776/libs/libmyth/mythuclibcfix.h mythtv-21776.patch/libs/libmyth/mythuclibcfix.h
--- mythtv-21776/libs/libmyth/mythuclibcfix.h	1970-01-01 01:00:00.000000000 +0100
+++ mythtv-21776.patch/libs/libmyth/mythuclibcfix.h	2009-09-11 19:46:50.926705617 +0200
@@ -0,0 +1,60 @@
+#ifndef _UCLIBC_HACK_H
+#define _UCLIBC_HACK_H 1
+
+#include <math.h>
+
+static inline double _uc_hack_sin(double d) {
+	return sin(d);
+}
+
+static inline double _uc_hack_cos(double d) {
+	return cos(d);
+}
+
+static inline double _uc_hack_acos(double d) {
+	return acos(d);
+}
+
+static inline double _uc_hack_asin(double d) {
+	return asin(d);
+}
+
+static inline double _uc_hack_tan(double d) {
+	return tan(d);
+}
+
+static inline double _uc_hack_atan(double d) {
+	return atan(d);
+}
+
+static inline double _uc_hack_pow(double x, double y) {
+	return pow(x, y);
+}
+
+static inline double _uc_hack_rint(double d) {
+	return rint(d);
+}
+
+static inline double _uc_hack_round(double d) {
+	return floor(d) + (d - floor(d) < 0.5 ? 0 : 1);
+}
+
+#define sin(x) (_uc_hack_sin((x)))
+#define sinf(x) (_uc_hack_sin((x)))
+#define cos(x) (_uc_hack_cos((x)))
+#define cosf(x) (_uc_hack_cos((x)))
+#define acos(x) (_uc_hack_acos((x)))
+#define acosf(x) (_uc_hack_acos((x)))
+#define asin(x) (_uc_hack_asin((x)))
+#define asinf(x) (_uc_hack_asin((x)))
+#define tan(x) (_uc_hack_tan((x)))
+#define tanf(x) (_uc_hack_tan((x)))
+#define atan(x) (_uc_hack_atan((X)))
+#define atanf(x) (_uc_hack_atan((x)))
+#define pow(x,y) (_uc_hack_pow((x),(y)))
+#define powf(x,y) (_uc_hack_pow((x),(y)))
+#define rint(x) (_uc_hack_rint((x)))
+#define rintf(x) (_uc_hack_rint((x)))
+#define round(x) (_uc_hack_round((x)))
+#define roundf(x) (_uc_hack_round((x)))
+#endif
diff -Naur mythtv-21776/libs/libmythdb/compat.h mythtv-21776.patch/libs/libmythdb/compat.h
--- mythtv-21776/libs/libmythdb/compat.h	2009-09-11 14:54:22.000000000 +0200
+++ mythtv-21776.patch/libs/libmythdb/compat.h	2009-09-11 20:16:08.986705264 +0200
@@ -254,4 +254,16 @@
     #define lseek64(f,o,w) lseek(f,o,w)
 #endif
 
+#if defined(__UCLIBC__)
+  #ifdef __cplusplus
+    extern "C" {
+  #endif
+
+  int getloadavg(double loadavg[], int nelem);
+
+  #ifdef __cplusplus
+    }
+  #endif
+#endif
+
 #endif // __COMPAT_H__
diff -Naur mythtv-21776/programs/mythfrontend/statusbox.cpp mythtv-21776.patch/programs/mythfrontend/statusbox.cpp
--- mythtv-21776/programs/mythfrontend/statusbox.cpp	2009-09-11 14:53:37.000000000 +0200
+++ mythtv-21776.patch/programs/mythfrontend/statusbox.cpp	2009-09-11 20:18:04.656580151 +0200
@@ -14,6 +14,7 @@
 #include "util.h"
 
 #include "config.h"
+#include "compat.h"
 #include "remoteutil.h"
 #include "tv.h"
 #include "jobqueue.h"
