diff -Naur mythtv-21776/libs/libmythtv/util-xv.cpp mythtv-21776.patch/libs/libmythtv/util-xv.cpp
--- mythtv-21776/libs/libmythtv/util-xv.cpp	2009-09-11 14:54:12.000000000 +0200
+++ mythtv-21776.patch/libs/libmythtv/util-xv.cpp	2009-09-12 14:12:50.974705750 +0200
@@ -21,7 +21,7 @@
 
 void close_all_xv_ports_signal_handler(int sig)
 {
-    cerr<<"Signal: "<<sys_siglist[sig]<<endl;
+    cerr<<"Signal: "<<strsignal[sig]<<endl;
     QMap<int,port_info>::iterator it;
     for (it = open_xv_ports.begin(); it != open_xv_ports.end(); ++it)
     {
