From 540dc69955cddd5db24a8a9e16db8162fc0ddd10 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 25 Apr 2014 13:02:35 +0300
Subject: [PATCH] hack: avoid potential SIGFPE in Menu::updateMenu()

---
 src/FbTk/Menu.cc |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/FbTk/Menu.cc b/src/FbTk/Menu.cc
index 3632fc9..75a5b4a 100644
--- a/src/FbTk/Menu.cc
+++ b/src/FbTk/Menu.cc
@@ -396,6 +396,7 @@ void Menu::enableTitle() {
 }
 
 void Menu::updateMenu() {
+    return; // because nobody cares
     if (m_title_vis) {
         m_item_w = theme()->titleFont().textWidth(m_label);
         m_item_w += (theme()->bevelWidth() * 2);
-- 
1.7.2.5

