Define FRIBIDI_CHUNK_SIZE only if using allocation pools

diff -Nur fribidi2-cvs-20061020.orig/lib/common.h fribidi2-cvs-20061020/lib/common.h
--- fribidi2-cvs-20061020.orig/lib/common.h	Tue Jan 31 05:23:12 2006
+++ fribidi2-cvs-20061020/lib/common.h	Sat Oct 21 16:42:41 2006
@@ -131,6 +131,8 @@
 
 /* FRIBIDI_CHUNK_SIZE is the number of bytes in each chunk of memory being
  * allocated for data structure pools. */
+#if USE_SIMPLE_MALLOC+0
+#else
 #ifndef FRIBIDI_CHUNK_SIZE
 # if HAVE_ASM_PAGE_H
 #  ifndef __FRIBIDI_DOC
@@ -145,6 +147,7 @@
 #  error FRIBIDI_CHUNK_SIZE now should define the size of a chunk in bytes.
 # endif	/* FRIBIDI_CHUNK_SIZE < 256 */
 #endif /* FRIBIDI_CHUNK_SIZE */
+#endif /* USE_SIMPLE_MALLOC */
 
 /* FRIBIDI_BEGIN_STMT should be used at the beginning of your macro
  * definitions that are to behave like simple statements.  Use
