diff -Naur a/game.libretro.desmume/addon.xml.in b/game.libretro.desmume/addon.xml.in
--- a/game.libretro.desmume/addon.xml.in	2016-07-16 04:22:14.000000000 -0700
+++ b/game.libretro.desmume/addon.xml.in	2016-08-26 12:49:51.513260529 -0700
@@ -5,7 +5,6 @@
 		provider-name="Libretro">
 	<requires>
 		<import addon="game.libretro" version="1.0.0"/>
-		<import addon="game.controller.nds" version="1.0.0"/>
 	</requires>
 	<extension point="kodi.gameclient"
                    library_@PLATFORM@="@LIBRARY_FILENAME@">
