diff -Naur Python-2.7.13.orig/Lib/platform.py Python-2.7.13/Lib/platform.py
--- Python-2.7.13.orig/Lib/platform.py	2017-01-08 11:57:36.615886244 +0100
+++ Python-2.7.13/Lib/platform.py	2017-01-08 12:12:44.567848271 +0100
@@ -1250,6 +1250,9 @@
         system = 'Windows'
         release = 'Vista'

+    if machine == 'aarch64' and architecture()[0] == '32bit':
+        machine = 'armv7l'
+
     _uname_cache = system,node,release,version,machine,processor
     return _uname_cache

