diff -Nurp patched/src/i830_quirks.c working/src/i830_quirks.c
--- patched/src/i830_quirks.c	2009-01-12 23:59:17.000000000 -0800
+++ working/src/i830_quirks.c	2009-01-12 23:59:58.000000000 -0800
@@ -308,6 +308,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_I855_GM, 0x1028, 0x00c8, quirk_pipea_force },
     /* Intel 855GM hardware (See Novell Bugzilla #406123) */
     { PCI_CHIP_I855_GM, 0x10cf, 0x1215, quirk_pipea_force },
+    /* Intel 855GM mv890 hardware (see LP: #305269) */
+    { PCI_CHIP_I855_GM, 0x8086, 0x3582, quirk_ignore_lvds },
     /* HP Pavilion ze4944ea needs pipe A force quirk (See LP: #242389) */
     { PCI_CHIP_I855_GM, 0x103c, 0x3084, quirk_pipea_force },
 
