diff -Nurp patched/src/i830_quirks.c working/src/i830_quirks.c
--- patched/src/i830_quirks.c	2009-03-02 15:49:37.000000000 -0800
+++ working/src/i830_quirks.c	2009-03-02 15:52:51.000000000 -0800
@@ -290,6 +290,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_GM45_GM, 0x103c, 0x30e8, quirk_ignore_tv },
     /* HP Compaq 2730p needs pipe A force quirk (LP: #291555) */
     { PCI_CHIP_GM45_GM, 0x103c, 0x30eb, quirk_pipea_force },
+    /* HP Mini needs pipe A force quirk (LP: #322104) */
+    { PCI_CHIP_I945_GME,0x103c, 0x361a, quirk_pipea_force },
 
     /* Thinkpad R31 needs pipe A force quirk */
     { PCI_CHIP_I830_M, 0x1014, 0x0505, quirk_pipea_force },
