diff -Naur xf86-video-intel-2.8.0.orig/src/i830_driver.c xf86-video-intel-2.8.0/src/i830_driver.c
--- xf86-video-intel-2.8.0.orig/src/i830_driver.c	2009-07-25 12:21:15.000000000 +0200
+++ xf86-video-intel-2.8.0/src/i830_driver.c	2009-07-25 12:23:42.000000000 +0200
@@ -1310,6 +1310,10 @@
 	xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "DRI is disabled because it "
 		"runs only at depths 16 and 24.\n");
 	pI830->directRenderingType = DRI_DISABLED;
+    } else if (IS_845G(pI830) || IS_I865G(pI830)) {
+	xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "DRI is disabled because it "
+		"fails to run on 845G and 865G chips. (See LP 304871)\n");
+	pI830->directRenderingType = DRI_DISABLED;
     }
 }
 
