diff -Nurp patched/src/i830_driver.c working/src/i830_driver.c
--- patched/src/i830_driver.c	2009-03-27 16:48:16.000000000 -0700
+++ working/src/i830_driver.c	2009-03-27 17:40:57.000000000 -0700
@@ -1659,7 +1659,12 @@ I830AccelMethodInit(ScrnInfoPtr pScrn)
 	xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "DRI is disabled because it "
 		"runs only at depths 16 and 24.\n");
 	pI830->directRenderingType = DRI_DISABLED;
+    } else if (IS_845G(pI830) || IS_I865G(pI830)) {
+	xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "DRI is disabled because it "
+		"fails to run on 845G and 865G chips. (See LP 304871)\n");
+	pI830->directRenderingType = DRI_DISABLED;
     }
+
 #endif /* XF86DRI */
 
     I830MapMMIO(pScrn);
