diff -Naur xf86-video-intel-2.8.0.901/src/i830_driver.c.org xf86-video-intel-2.8.0.901/src/i830_driver.c
--- xf86-video-intel-2.8.0.901/src/i830_driver.c.org	2009-08-13 22:07:07.000000000 +0200
+++ xf86-video-intel-2.8.0.901/src/i830_driver.c	2009-09-12 13:34:06.000000000 +0200
@@ -2685,7 +2695,9 @@
        return FALSE;
    }
 
-   i830_fixup_mtrrs(pScrn);
+   /* In a GEM config, the kernel manages MTRRs */
+   if (!pI830->memory_manager)
+       i830_fixup_mtrrs(pScrn);
 
    pI830->starting = TRUE;
 
