--- xf86-video-intel-2.5.1/src/i830_driver.c.org	2008-11-15 17:29:56.000000000 +0100
+++ xf86-video-intel-2.5.1/src/i830_driver.c	2008-11-15 17:30:52.000000000 +0100
@@ -1696,7 +1696,7 @@
     pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
-   pI830->XvPreferOverlay = xf86ReturnOptValBool(pI830->Options, OPTION_PREFER_OVERLAY, FALSE);
+   pI830->XvPreferOverlay = TRUE;
 
 #ifdef I830_XV
     if (xf86GetOptValInteger(pI830->Options, OPTION_VIDEO_KEY,
