diff -Naur xf86-video-intel-2.5.0/src/i830_driver.c xf86-video-intel-2.5.0.edit/src/i830_driver.c
--- xf86-video-intel-2.5.0/src/i830_driver.c	2008-10-20 23:02:38.000000000 +0200
+++ xf86-video-intel-2.5.0.edit/src/i830_driver.c	2008-10-25 17:53:33.000000000 +0200
@@ -3309,8 +3313,7 @@
        vgaHWSetMmioFuncs(hwp, pI830->MMIOBase, 0);
        vgaHWGetIOBase(hwp);
        DPRINTF(PFX, "assert( if(!vgaHWMapMem(pScrn)) )\n");
-       if (!vgaHWMapMem(pScrn))
-	   return FALSE;
+       vgaHWMapMem(pScrn);
    }
 
    i830_disable_render_standby(pScrn);
