diff -Naur xf86-video-intel-2.7.0.901/src/i830_hdmi.c.org xf86-video-intel-2.7.0.901/src/i830_hdmi.c
--- xf86-video-intel-2.7.0.901/src/i830_hdmi.c.org	2009-05-11 20:34:54.000000000 +0200
+++ xf86-video-intel-2.7.0.901/src/i830_hdmi.c	2009-05-11 20:35:50.000000000 +0200
@@ -248,8 +248,8 @@
                    "RRConfigureOutputProperty error, %d\n", err);
         return;
     }
-    /* Set the current value of the broadcast property as full range */
-    dev_priv->broadcast_rgb = 0;
+    /* Set the current value of the broadcast property as limited range */
+    dev_priv->broadcast_rgb = 1;
     err = RRChangeOutputProperty(output->randr_output,
                                  broadcast_atom,
                                  XA_INTEGER, 32, PropModeReplace,
diff -Naur xf86-video-intel-2.7.0.901/src/i830_sdvo.c.org xf86-video-intel-2.7.0.901/src/i830_sdvo.c
--- xf86-video-intel-2.7.0.901/src/i830_sdvo.c.org	2009-05-11 20:35:02.000000000 +0200
+++ xf86-video-intel-2.7.0.901/src/i830_sdvo.c	2009-05-11 20:43:47.000000000 +0200
@@ -1963,8 +1963,8 @@
                    "RRConfigureOutputProperty error, %d\n", err);
         return;
     }
-    /* Set the current value of the broadcast property as full range */
-    dev_priv->broadcast_rgb = 0;
+    /* Set the current value of the broadcast property as limited range */
+    dev_priv->broadcast_rgb = 1;
     err = RRChangeOutputProperty(output->randr_output,
                                  broadcast_atom,
                                  XA_INTEGER, 32, PropModeReplace,
