--- xf86-video-intel-2.7.0.901/src/i830_driver.c.org	2009-05-11 21:57:27.000000000 +0200
+++ xf86-video-intel-2.7.0.901/src/i830_driver.c	2009-05-11 21:58:30.000000000 +0200
@@ -2836,10 +2836,8 @@
    xf86DPMSInit(pScreen, xf86DPMSSet, 0);

 #ifdef INTEL_XVMC
-    pI830->XvMCEnabled = FALSE;
-    from = ((pI830->directRenderingType == DRI_DRI2) &&
-            xf86GetOptValBool(pI830->Options, OPTION_XVMC,
-                              &pI830->XvMCEnabled) ? X_CONFIG : X_DEFAULT);
+    pI830->XvMCEnabled = TRUE;
+    from = X_CONFIG;
     xf86DrvMsg(pScrn->scrnIndex, from, "Intel XvMC decoder %sabled\n",
               pI830->XvMCEnabled ? "en" : "dis");
 #endif
