--- xf86-video-intel-2.5.1/src/i830_quirks.c.org	2008-11-13 23:24:50.000000000 +0100
+++ xf86-video-intel-2.5.1/src/i830_quirks.c	2008-11-13 23:36:08.000000000 +0100
@@ -235,7 +235,10 @@
     /* Apple Mac mini has no lvds, but macbook pro does */
     { PCI_CHIP_I945_GM, 0x8086, 0x7270, quirk_mac_mini },
 
-    /* Transtec Senyo 610 mini pc */
+    /* Asus Eee PC B202 (See LP: #233787) */
+    { PCI_CHIP_I945_GME, 0x1043, 0x1252, quirk_ignore_lvds },
+
+    /* Transtec Senyo 610 mini pc (See LP: #233787) */
     { PCI_CHIP_I965_GM, 0x1509, 0x2f15, quirk_ignore_lvds },
 
     /* Clevo M720R has no tv output */
@@ -255,13 +258,15 @@
     { PCI_CHIP_I965_GM, 0x1028, 0x0286, quirk_ignore_tv },
     /* Dell Vostro A840 (LP: #235155) */
     { PCI_CHIP_I965_GM, 0x1028, 0x0298, quirk_ignore_tv },
-    /* Dell Studio Hybrid */
+    /* Dell Studio Hybrid (LP: #267945) */
     { PCI_CHIP_I965_GM, 0x1028, 0x0279, quirk_ignore_lvds },
 
     /* Lenovo Napa TV (use dmi)*/
     { PCI_CHIP_I945_GM, 0x17aa, SUBSYS_ANY, quirk_lenovo_tv_dmi },
     /* Lenovo 3000 v200 */
     { PCI_CHIP_I965_GM, 0x17aa, 0x3c18, quirk_ignore_tv },
+    /* Lenovo T60/R60 needs pipe A force quirk (See LP: #243405) */
+    { PCI_CHIP_I945_GM, 0x17aa, 0x201a, quirk_pipea_force },
 
     /* Panasonic Toughbook CF-Y4 has no TV output */
     { PCI_CHIP_I915_GM, 0x10f7, 0x8338, quirk_ignore_tv },
@@ -286,9 +291,13 @@
     /* HP Compaq 6730s has no TV output */
     { PCI_CHIP_GM45_GM, 0x103c, 0x30e8, quirk_ignore_tv },
 
+    /* Thinkpad G40 needs pipe A force quirk (See LP: #240457) */
+    { PCI_CHIP_I855_GM, 0x1014, 0x0543, quirk_pipea_force },
     /* Thinkpad R31 needs pipe A force quirk */
     { PCI_CHIP_I830_M, 0x1014, 0x0505, quirk_pipea_force },
-    /* Dell Latitude D400 needs pipe A force quirk (LP: #228519) */
+    /* Dell Latitude C400 needs pipe A force quirk (See LP: #216490) */
+    { PCI_CHIP_I855_GM, 0x1028, 0x00c8, quirk_pipea_force },
+    /* Dell Latitude D400 needs pipe A force quirk (See LP: #228519) *
     { PCI_CHIP_I855_GM, 0x1028, 0x0139, quirk_pipea_force },
     /* Dell Latitude D500 needs pipe A force quirk */
     { PCI_CHIP_I855_GM, 0x1028, 0x0152, quirk_pipea_force },
@@ -322,6 +331,9 @@
     /* Sony vaio VGN-SZ4MN (See LP: #212163) */
     { PCI_CHIP_I830_M, 0x104d, 0x81e6, quirk_pipea_force },
 
+    /* Quanta Gigabyte W251U (See LP: #244242) */
+    { PCI_CHIP_I945_GM, 0x152d, 0x0755, quirk_pipea_force },
+
     /* Ordi Enduro UW31 (See LP: #152416) */
     { PCI_CHIP_I945_GM, 0x1584, 0x9900, quirk_ignore_tv },
 
