From cb66b0945fcb86027cf74d22c1c061bf611564c4 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Fri, 31 Jul 2009 21:24:12 +0200
Subject: [PATCH] disable uxa render accel on i8xx

---
 src/i830_uxa.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/i830_uxa.c b/src/i830_uxa.c
index 2050c48..5ab266f 100644
--- a/src/i830_uxa.c
+++ b/src/i830_uxa.c
@@ -709,8 +709,8 @@ i830_uxa_init (ScreenPtr pScreen)
 
     /* Composite */
     if (!IS_I9XX(i830)) {
-    	i830->uxa_driver->check_composite = i830_check_composite;
-    	i830->uxa_driver->prepare_composite = i830_prepare_composite;
+    	i830->uxa_driver->check_composite = NULL;
+    	i830->uxa_driver->prepare_composite = NULL;
     	i830->uxa_driver->composite = i830_composite;
     	i830->uxa_driver->done_composite = i830_done_composite;
     } else if (IS_I915G(i830) || IS_I915GM(i830) ||
-- 
1.6.3.3

