diff --git a/src/bios_reader/Makefile.am b/src/bios_reader/Makefile.am
index f2f9dc0..4be476e 100644
--- a/src/bios_reader/Makefile.am
+++ b/src/bios_reader/Makefile.am
@@ -1,7 +1,7 @@
 AM_CFLAGS = @CWARNFLAGS@ @XORG_CFLAGS@ @PCIACCESS_CFLAGS@ \
 	-DREG_DUMPER
 
-noinst_PROGRAMS = bios_reader  $(BIOS_DUMPER) $(SWF_DUMPER)
+bin_PROGRAMS = bios_reader  $(BIOS_DUMPER) $(SWF_DUMPER)
 
 BIOS_DUMPER = bios_dumper
 
diff --git a/src/reg_dumper/Makefile.am b/src/reg_dumper/Makefile.am
index 8f00abe..1634a4f 100644
--- a/src/reg_dumper/Makefile.am
+++ b/src/reg_dumper/Makefile.am
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	intel_gtt \
 	intel_statuspage \
 	intel_hotplug \
