diff -Naur xf86-video-intel-20100912/src/intel_display.c xf86-video-intel-20100912.patch/src/intel_display.c
--- xf86-video-intel-20100912/src/intel_display.c	2010-09-10 14:19:12.000000000 +0200
+++ xf86-video-intel-20100912.patch/src/intel_display.c	2010-09-12 00:26:32.993390599 +0200
@@ -1528,10 +1528,14 @@
 	gp.value = &has_flipping;
 	(void)drmCommandWriteRead(intel->drmSubFD, DRM_I915_GETPARAM, &gp,
 				  sizeof(gp));
+	xf86DrvMsg(scrn->scrnIndex, X_INFO,
+		"Pageflipping %s in kernel, %s disabled in X\n",
+		has_flipping ? "enabled" : "disabled",
+		has_flipping ? "but" : "and");
 	if (has_flipping) {
 		xf86DrvMsg(scrn->scrnIndex, X_INFO,
-			   "Kernel page flipping support detected, enabling\n");
-		intel->use_pageflipping = TRUE;
+			   "Don't panic: https://bugzilla.redhat.com/588421\n");
+		intel->use_pageflipping = FALSE;
 
 		mode->event_context.version = DRM_EVENT_CONTEXT_VERSION;
 		mode->event_context.vblank_handler = intel_vblank_handler;
diff -Naur xf86-video-intel-20100912/src/intel_dri.c xf86-video-intel-20100912.patch/src/intel_dri.c
--- xf86-video-intel-20100912/src/intel_dri.c	2010-09-10 14:19:12.000000000 +0200
+++ xf86-video-intel-20100912.patch/src/intel_dri.c	2010-09-12 00:24:12.960390827 +0200
@@ -1057,7 +1057,7 @@
 
 	info.CopyRegion = I830DRI2CopyRegion;
 #if DRI2INFOREC_VERSION >= 4
-	if (intel->use_pageflipping) {
+	if (intel->use_pageflipping || 1) {
 	    info.version = 4;
 	    info.ScheduleSwap = I830DRI2ScheduleSwap;
 	    info.GetMSC = I830DRI2GetMSC;
