diff -up xf86-video-intel-20090908/src/drmmode_display.c.jx xf86-video-intel-20090908/src/drmmode_display.c
--- xf86-video-intel-20090908/src/drmmode_display.c.jx	2009-09-08 14:30:55.000000000 -0400
+++ xf86-video-intel-20090908/src/drmmode_display.c	2009-09-08 14:31:09.000000000 -0400
@@ -1195,8 +1195,19 @@ drmmode_output_get_property(xf86OutputPt
     return TRUE;
 }
 
+#ifdef RANDR_GET_CRTC_INTERFACE
+static xf86CrtcPtr
+drmmode_get_crtc(xf86OutputPtr output)
+{
+    return output->crtc;
+}
+#endif
+
 static const xf86OutputFuncsRec drmmode_output_funcs = {
 	.create_resources = drmmode_output_create_resources,
+#ifdef RANDR_GET_CRTC_INTERFACE
+	.get_crtc = drmmode_get_crtc,
+#endif
 #ifdef RANDR_12_INTERFACE
 	.set_property = drmmode_output_set_property,
 	.get_property = drmmode_output_get_property,
