diff -up xf86-video-intel-2.11.0/src/drmmode_display.c.no-flip xf86-video-intel-2.11.0/src/drmmode_display.c
--- xf86-video-intel-2.11.0/src/drmmode_display.c.no-flip	2010-05-03 15:30:19.000000000 -0400
+++ xf86-video-intel-2.11.0/src/drmmode_display.c	2010-05-03 15:30:59.000000000 -0400
@@ -1504,10 +1504,15 @@ Bool drmmode_pre_init(ScrnInfoPtr scrn, 
 	gp.value = &has_flipping;
 	(void)drmCommandWriteRead(intel->drmSubFD, DRM_I915_GETPARAM, &gp,
 				  sizeof(gp));
+
+        xf86DrvMsg(scrn->scrnIndex, X_INFO,
+                   "Pageflipping %s in kernel, %s disabled in X\n",
+                   has_flipping ? "enabled" : "disabled",
+                   has_flipping ? "but" : "and");
 	if (has_flipping) {
 		xf86DrvMsg(scrn->scrnIndex, X_INFO,
-			   "Kernel page flipping support detected, enabling\n");
-		intel->use_pageflipping = TRUE;
+			   "Don't panic: https://bugzilla.redhat.com/588421\n");
+		intel->use_pageflipping = FALSE;
 		drmmode->flip_count = 0;
 		drmmode->event_context.version = DRM_EVENT_CONTEXT_VERSION;
 		drmmode->event_context.vblank_handler = drmmode_vblank_handler;
diff -up xf86-video-intel-2.11.0/src/i830_dri.c.no-flip xf86-video-intel-2.11.0/src/i830_dri.c
--- xf86-video-intel-2.11.0/src/i830_dri.c.no-flip	2010-03-29 14:23:02.000000000 -0400
+++ xf86-video-intel-2.11.0/src/i830_dri.c	2010-05-03 15:30:19.000000000 -0400
@@ -1013,7 +1013,7 @@ Bool I830DRI2ScreenInit(ScreenPtr screen
 
 	info.CopyRegion = I830DRI2CopyRegion;
 #if DRI2INFOREC_VERSION >= 4
-	if (intel->use_pageflipping) {
+	if (intel->use_pageflipping || 1) {
 	    info.version = 4;
 	    info.ScheduleSwap = I830DRI2ScheduleSwap;
 	    info.GetMSC = I830DRI2GetMSC;
