diff -up xf86-video-intel-2.11.0/src/drmmode_display.c.jx xf86-video-intel-2.11.0/src/drmmode_display.c
--- xf86-video-intel-2.11.0/src/drmmode_display.c.jx	2010-04-16 10:28:08.000000000 -0400
+++ xf86-video-intel-2.11.0/src/drmmode_display.c	2010-04-26 15:28:58.000000000 -0400
@@ -689,6 +689,12 @@ drmmode_output_mode_valid(xf86OutputPtr 
 		else
 			return MODE_OK;
 	}
+
+	/* I don't actually believe any VGA is this good */
+	if (koutput->connector_type == DRM_MODE_CONNECTOR_VGA)
+		if (pModes->Clock > 250000)
+			return MODE_CLOCK_HIGH;
+
 	return MODE_OK;
 }
 
