Index: xf86-video-intel-2.11.0/src/drmmode_display.c
===================================================================
--- xf86-video-intel-2.11.0/src/drmmode_display.c
+++ xf86-video-intel-2.11.0/src/drmmode_display.c	2010-04-29 00:18:34.000000000 +0200
@@ -103,6 +103,7 @@
     "asus-laptop",
     "eeepc",
     "thinkpad_screen",
+    "mbp_backlight",
     "acpi_video1",
     "acpi_video0",
     "fujitsu-laptop",
