diff -Naur xf86-video-intel-2.99.902/configure.ac xf86-video-intel-2.99.902.patch/configure.ac
--- xf86-video-intel-2.99.902/configure.ac	2013-09-07 09:47:33.000000000 +0200
+++ xf86-video-intel-2.99.902.patch/configure.ac	2013-09-07 22:55:59.418980080 +0200
@@ -394,9 +394,6 @@
 AC_MSG_CHECKING([whether to include DRI2 support])
 AM_CONDITIONAL(DRI2, test "x$DRI2" = "xyes")
 AC_MSG_RESULT([$DRI2])
-if test "x$enable_dri" = "xyes" -a "x$KMS" = "xyes"; then
-	AC_MSG_ERROR([DRI2 requested but prerequisites not found])
-fi
 if test "x$DRI2" != "xno"; then
         AC_DEFINE(HAVE_DRI2,1,[Enable DRI2 driver support])
 	dri_msg="$dri_msg DRI2"
