diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-05-10 12:18:51.824846056 -0700
+++ b/Makefile	2016-05-10 13:12:56.463629991 -0700
@@ -344,6 +344,29 @@
 	PLATCFLAGS += -DARM
 	LIBS += -lstdc++ -lpthread -ldl
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+        armplatform := 1
+        TARGETLIB := $(TARGET_NAME)_libretro.so
+        TARGETOS=linux
+        fpic := -fPIC
+        SHARED := -shared -Wl,--version-script=src/osd/retro/link.T -Wl,--no-undefined
+        CCOMFLAGS += $(fpic)
+#        PLATCFLAGS += -march=armv7-a -DALIGN_INTS -DALIGN_SHORTS -DLSB_FIRST -fstrict-aliasing -fno-merge-constants -DSDLMAME_NO64BITIO -DSDLMAME_ARM -DRETRO_SETJMP_HACK
+        LDFLAGS += -Wl,--no-as-needed $(fpic) $(SHARED)
+        REALCC   ?= gcc
+        CC ?= g++
+        AR ?= @ar
+        LD ?= g++
+        CCOMFLAGS += $(PLATCFLAGS)
+        PTR64 = 1
+
+#        PLATCFLAGS += -DHAVE_GL
+        LIBS += -lGLESv2
+        GLES = 1
+
+        LIBS += -lstdc++ -lpthread -ldl
+
 # Windows cross compiler
 else ifeq ($(platform), wincross)
 	TARGETLIB := $(TARGET_NAME)_libretro.dll
@@ -557,7 +580,7 @@
 
 # CFLAGS is defined based on C or C++ targets
 # (remember, expansion only happens when used, so doing it here is ok)
-CFLAGS = $(CCOMFLAGS) $(CPPONLYFLAGS)
+CFLAGS += $(CCOMFLAGS) $(CPPONLYFLAGS)
 
 # we compile C-only to C89 standard with GNU extensions
 # we compile C++ code to C++98 standard with GNU extensions
