diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-01-14 22:44:37.000000000 +0100
+++ b/Makefile	2016-01-18 20:53:35.439383675 +0100
@@ -115,25 +115,23 @@
 		LIBS += -lGL
 	endif
 	LDFLAGS +=  $(fpic) $(SHARED)
-	REALCC   = gcc
-	NATIVECC = g++
-	NATIVECFLAGS = -std=gnu99
+	REALCC   ?= gcc
 	BASELIBS +=  -lpthread
-	CC = g++
-	AR = @ar
-	LD = g++
+	CC ?= g++
+	AR ?= @ar
+	LD ?= g++
 	CCOMFLAGS += $(PLATCFLAGS)
 	LIBS += -lstdc++ -lpthread -ldl
-	ifeq ($(firstword $(filter x86_64,$(UNAME))),x86_64)
+	ifeq ($(firstword $(filter x86_64,$(ARCH))),x86_64)
 		PTR64 = 1
 	endif
-	ifeq ($(firstword $(filter amd64,$(UNAME))),amd64)
+	ifeq ($(firstword $(filter amd64,$(ARCH))),amd64)
 		PTR64 = 1
 	endif
-	ifeq ($(firstword $(filter ppc64,$(UNAME))),ppc64)
+	ifeq ($(firstword $(filter ppc64,$(ARCH))),ppc64)
 		PTR64 = 1
 	endif
-	ifneq (,$(findstring ppc,$(UNAME)))
+	ifneq (,$(findstring ppc,$(ARCH)))
 		BIGENDIAN=1
 	else
 		PLATCFLAGS += -DLSB_FIRST
@@ -316,12 +314,10 @@
 	CCOMFLAGS += $(fpic) -mstructure-size-boundary=32 -falign-functions=16 -fsigned-char -finline -fno-common -fno-builtin -fweb -frename-registers -falign-functions=16
 	PLATCFLAGS += -march=armv7-a -DALIGN_INTS -DALIGN_SHORTS -DLSB_FIRST -fstrict-aliasing -fno-merge-constants -DSDLMAME_NO64BITIO -DSDLMAME_ARM -DRETRO_SETJMP_HACK
 	LDFLAGS += -Wl,--fix-cortex-a8 -Wl,--no-as-needed $(fpic) $(SHARED)
-	REALCC   = gcc
-	NATIVECC = g++
-	NATIVECFLAGS = -std=gnu99
-	CC = g++
-	AR = @ar
-	LD = g++
+	REALCC   ?= gcc
+	CC ?= g++
+	AR ?= @ar
+	LD ?= g++
 	CCOMFLAGS += $(PLATCFLAGS)
 
 	ifneq (,$(findstring cortexa8,$(platform)))
@@ -492,7 +488,7 @@
 # utilities
 MD = -mkdir$(EXE_EXT)
 RM = @rm -f
-OBJDUMP = @objdump
+OBJDUMP ?= @objdump
 PYTHON ?= @python2
 
 #-------------------------------------------------
