diff --git a/Makefile b/Makefile
index ea8b3f5..605cb7e 100644
--- a/Makefile
+++ b/Makefile
@@ -488,8 +488,8 @@ else
    FLAGS += -O0 -g
 endif
 
-LDFLAGS += $(fpic) $(SHARED)
-FLAGS += $(fpic) $(NEW_GCC_FLAGS)
+LDFLAGS += -fPIC $(SHARED)
+FLAGS += -fPIC $(NEW_GCC_FLAGS)
 FLAGS += $(INCFLAGS) $(INCFLAGS_PLATFORM)
 
 FLAGS += $(ENDIANNESS_DEFINES) $(WARNINGS) $(CORE_DEFINE) -DSTDC_HEADERS -D__STDC_LIMIT_MACROS -D__LIBRETRO__ $(EXTRA_INCLUDES) $(SOUND_DEFINE)
