diff --git a/Makefile.libretro b/Makefile.libretro
index 1409ac5..1822d87 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -53,9 +53,10 @@ endif
 ifneq (,$(findstring unix,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
+   CFLAGS := $(filter-out -O2,$(CFLAGS))
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T -Wl,--no-undefined
    ENDIANNESS_DEFINES := -DLSB_FIRST -DBYTE_ORDER=LITTLE_ENDIAN
    PLATFORM_DEFINES := -DHAVE_ZLIB
 
    # Raspberry Pi
    ifneq (,$(findstring rpi,$(platform)))
@@ -571,11 +572,11 @@ else
 	CXXFLAGS += -MD
 endif
 
-CFLAGS += -O2 -DNDEBUG
-CXXFLAGS += -O2 -DNDEBUG
+CFLAGS += -O0 -DNDEBUG
+CXXFLAGS += -O0 -DNDEBUG
 else
-	CFLAGS += -O2 -DNDEBUG
-	CXXFLAGS += -O2 -DNDEBUG
+	CFLAGS += -O0 -DNDEBUG
+	CXXFLAGS += -O0 -DNDEBUG
 endif
 endif
 
