diff -Naur libbmm-0.2.0/Makefile_general libbmm-0.2.0.patch/Makefile_general
--- libbmm-0.2.0/Makefile_general	2011-03-07 18:48:46.000000000 +0100
+++ libbmm-0.2.0.patch/Makefile_general	2012-12-24 20:19:47.791960702 +0100
@@ -30,20 +30,20 @@
 	uninstall-host uninstall-target
 
 .c.o:
-	${CC} -fPIC -o $@ -c $<
+	${CC} -fPIC ${CFLAGS} -o $@ -c $<
 
 all: compile install-host install-target 
 
 compile: bmm_test libbmm.so libbmm.a
 
 bmm_test: libbmm.so.$(ABI_VERSION) ${bmm_test_obj}
-	${CC} -o $@ ${bmm_test_obj} -L. -lbmm
+	${CC} ${LDLAGS} -o $@ ${bmm_test_obj} -L. -lbmm
 
 libbmm.a: ${bmm_lib_obj}
 	${AR} -rcs $@ ${bmm_lib_obj}
 
 libbmm.so.$(ABI_VERSION): ${bmm_lib_obj}
-	${CC} -g -s -shared -Wl,-soname,$@ -o $@.$(ABI_REVISION) ${bmm_lib_obj}
+	${CC} -g -s ${LDFLAGS} -shared -Wl,-soname,$@ -o $@.$(ABI_REVISION) ${bmm_lib_obj}
 	ln -s $@.$(ABI_REVISION) $@
 	ln -s $@.$(ABI_REVISION) libbmm.so
 
