diff -Naur libbluray-894441d1931e25f6250ae2d15d174d0266c98462-old/configure.ac libbluray-894441d1931e25f6250ae2d15d174d0266c98462-new/configure.ac
--- libbluray-894441d1931e25f6250ae2d15d174d0266c98462-old/configure.ac	2010-12-14 09:07:24.000000000 -0800
+++ libbluray-894441d1931e25f6250ae2d15d174d0266c98462-new/configure.ac	2010-12-14 09:08:31.000000000 -0800
@@ -33,7 +33,7 @@
 using_normal_linking="Using libaacs and libbdplus via normal linking."
 
 # configure options
-AC_ARG_ENABLE([dlopen-crypto-libs],
+AC_ARG_WITH([dlopen-crypto-libs],
   [AS_HELP_STRING([--with-dlopen-crypto-libs],
   [use libaacs and libbdplus via dlopen (default is auto)])],
   [use_dlopen_crypto_libs=$withval],
