From 1a02e84589efb3f8717d50bdc78d3f369b799198 Mon Sep 17 00:00:00 2001
From: Alessandro Ghedini <al3xbio@gmail.com>
Date: Mon, 15 Oct 2012 16:06:54 +0200
Subject: [PATCH] gnutls: put reset code into else block

Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=690551
---
 lib/gtls.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/lib/gtls.c b/lib/gtls.c
index 3b4dc40..f5f95ae 100644
--- a/lib/gtls.c
+++ b/lib/gtls.c
@@ -309,10 +309,11 @@ static CURLcode handshake(struct connectdata *conn,
       failf(data, "gnutls_handshake() failed: %s", gnutls_strerror(rc));
       return CURLE_SSL_CONNECT_ERROR;
     }
-
-    /* Reset our connect state machine */
-    connssl->connecting_state = ssl_connect_1;
-    return CURLE_OK;
+    else {
+      /* Reset our connect state machine */
+      connssl->connecting_state = ssl_connect_1;
+      return CURLE_OK;
+    }
   }
 }
 
-- 
1.7.10

