diff -Naur curl-7.28.1/m4/curl-override.m4 curl-7.28.1.patch/m4/curl-override.m4
--- curl-7.28.1/m4/curl-override.m4	2012-03-08 20:35:25.000000000 +0100
+++ curl-7.28.1.patch/m4/curl-override.m4	2013-01-12 02:57:05.467821530 +0100
@@ -88,13 +88,3 @@
 m4_defun([_LT_AC_LIBTOOL_SYS_PATH_SEPARATOR],
 [CURL_CHECK_PATH_SEPARATOR
 m4_define([$0],[])])
-
-dnl Override Autoconf's AC_CONFIG_MACRO_DIR (DIR)
-dnl -------------------------------------------------
-dnl This is an emulation of Autoconf's 2.61 macro.
-dnl This is done to use fixed macro across Autoconf
-dnl versions, and avoid warnings from modern libtool
-dnl which traces usage of this macro.
-
-AC_DEFUN([AC_CONFIG_MACRO_DIR],[:])
-
