diff -Naur curl-7.28.1-old/configure.ac curl-7.28.1-new/configure.ac
--- curl-7.28.1-old/configure.ac	2012-09-08 13:39:18.000000000 -0700
+++ curl-7.28.1-new/configure.ac	2012-12-29 10:47:09.000000000 -0800
@@ -35,7 +35,7 @@
 terms of the curl license; see COPYING for more details])
 
 AC_CONFIG_SRCDIR([lib/urldata.h])
-AM_CONFIG_HEADER(lib/curl_config.h include/curl/curlbuild.h)
+AC_CONFIG_HEADERS(lib/curl_config.h include/curl/curlbuild.h)
 AC_CONFIG_MACRO_DIR([m4])
 AM_MAINTAINER_MODE
 
