diff -Naur cloog-ppl-0.15.8/configure cloog-ppl-0.15.8.patch/configure
--- cloog-ppl-0.15.8/configure	2010-02-11 19:05:30.000000000 +0100
+++ cloog-ppl-0.15.8.patch/configure	2010-02-20 14:25:38.957509566 +0100
@@ -12684,7 +12684,7 @@
 POLYHEDRAL_BACKEND=polylib
 
 ppl_major_version=0
-ppl_minor_version=10
+ppl_minor_version=11
 
 
 
diff -Naur cloog-ppl-0.15.8/configure.in cloog-ppl-0.15.8.patch/configure.in
--- cloog-ppl-0.15.8/configure.in	2010-02-11 19:05:31.000000000 +0100
+++ cloog-ppl-0.15.8.patch/configure.in	2010-02-20 14:26:01.755539644 +0100
@@ -311,7 +311,7 @@
 
 dnl Checking for PPL
 ppl_major_version=0
-ppl_minor_version=10
+ppl_minor_version=11
 AC_SUBST(PPL_CPPFLAGS)
 AC_SUBST(PPL_LDFLAGS)
 AC_SUBST(PPL_LIBS)
