diff -Nur a/driver/src/devicedrv/mali/linux/mali_memory.c b/driver/src/devicedrv/mali/linux/mali_memory.c
--- a/driver/src/devicedrv/mali/linux/mali_memory.c	2018-09-13 08:52:20.000000000 +0200
+++ b/driver/src/devicedrv/mali/linux/mali_memory.c	2019-04-18 19:58:27.389785486 +0200
@@ -70,7 +70,9 @@
 	}
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
+static vm_fault_t mali_mem_vma_fault(struct vm_fault *vmf)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
 static int mali_mem_vma_fault(struct vm_fault *vmf)
 #else
 static int mali_mem_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
