diff -Nur a/driver/src/devicedrv/mali/linux/mali_ukk_mem.c b/driver/src/devicedrv/mali/linux/mali_ukk_mem.c
--- a/driver/src/devicedrv/mali/linux/mali_ukk_mem.c	2018-09-13 08:52:20.000000000 +0200
+++ b/driver/src/devicedrv/mali/linux/mali_ukk_mem.c	2019-02-09 20:10:06.867561465 +0100
@@ -207,10 +207,17 @@
 	kargs.ctx = (uintptr_t)session_data;
 
 	/* Check if we can access the buffers */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
+	if (!access_ok(kargs.dest, kargs.size)
+	    || !access_ok(kargs.src, kargs.size)) {
+		return -EINVAL;
+	}
+#else
 	if (!access_ok(VERIFY_WRITE, kargs.dest, kargs.size)
 	    || !access_ok(VERIFY_READ, kargs.src, kargs.size)) {
 		return -EINVAL;
 	}
+#endif
 
 	/* Check if size wraps */
 	if ((kargs.size + kargs.dest) <= kargs.dest
@@ -266,8 +273,13 @@
 		goto err_exit;
 
 	user_buffer = (void __user *)(uintptr_t)kargs.buffer;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,0,0)
+	if (!access_ok(user_buffer, kargs.size))
+		goto err_exit;
+#else
 	if (!access_ok(VERIFY_WRITE, user_buffer, kargs.size))
 		goto err_exit;
+#endif
 
 	/* allocate temporary buffer (kernel side) to store mmu page table info */
 	if (kargs.size <= 0)
