diff -Naur Cheetah-2.0.1/SetupTools.py Cheetah-2.0.1.patch/SetupTools.py
--- Cheetah-2.0.1/SetupTools.py	2007-11-03 20:44:38.000000000 +0100
+++ Cheetah-2.0.1.patch/SetupTools.py	2010-10-19 00:34:45.377616641 +0200
@@ -19,18 +19,7 @@
 import string
 import traceback
 
-from distutils.core import setup
-if 'CHEETAH_USE_SETUPTOOLS' in os.environ:
-    # @@TR: Please note that this is for testing purposes only!  PEAK setuptools
-    # is not required or recommended for installing Cheetah.  Downstream
-    # package managers (linux distros, etc.) should *not* enable this.
-    try:
-        # use http://peak.telecommunity.com/DevCenter/setuptools if it's installed
-        # requires Py >=2.3
-        from setuptools import setup
-    except ImportError:   
-        from distutils.core import setup
-
+from setuptools import setup
 from distutils.core import Command
 from distutils.command.install_data import install_data
 
