diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-11-13 15:05:10.000000000 -0800
+++ b/CMakeLists.txt	2016-11-17 11:20:49.421085972 -0800
@@ -6,7 +6,8 @@
 find_package(Kodi REQUIRED)
 find_package(libretro-scummvm REQUIRED)
 
+set(DEPLIBS scummvm)
 set(SCUMMVM_CUSTOM_BINARY ${SCUMMVM_LIB}
                           ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
-build_addon(${PROJECT_NAME} SCUMMVM "")
+build_addon(${PROJECT_NAME} SCUMMVM DEPLIBS)
