diff -Naur reicast.git/Makefile reicast.patch/Makefile
--- reicast.git/Makefile	2017-09-30 03:33:24.991589770 -0500
+++ reicast.patch/Makefile	2017-09-30 03:45:10.575416919 -0500
@@ -13,17 +13,10 @@
 
 TARGET_NAME   := reicast
 
-CXX      = ${CC_PREFIX}g++
-CC       = ${CC_PREFIX}gcc
-CC_AS    = ${CC_PREFIX}as
-
 MFLAGS   := 
 ASFLAGS  := 
-LDFLAGS  :=
 INCFLAGS :=
 LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
 
 UNAME=$(shell uname -a)
 
@@ -509,7 +502,7 @@
 PREFIX        ?= /usr/local
 
 ifneq (,$(findstring arm, $(ARCH)))
-	CC_AS    = ${CC_PREFIX}gcc #The ngen_arm.S must be compiled with gcc, not as
+	CC_AS    = ${CC_PREFIX}${CC} #The ngen_arm.S must be compiled with gcc, not as
 	ASFLAGS  += $(CFLAGS)
 endif
