diff --git a/Makefile b/Makefile
index 44aa40d..472773f 100644
--- a/Makefile
+++ b/Makefile
@@ -20,18 +20,21 @@ HAVE_CHD      := 1
 
 TARGET_NAME   := reicast
 
-CXX      = ${CC_PREFIX}g++
-CC       = ${CC_PREFIX}gcc
-CC_AS    = ${CC_PREFIX}as
+ifeq (,$(findstring arm, $(ARCH)))
+	CXX      = ${CC_PREFIX}g++
+	CC       = ${CC_PREFIX}gcc
+	CC_AS    = ${CC_PREFIX}as
+
+	LDFLAGS  :=
+	CFLAGS   :=
+	CXXFLAGS :=
+endif
 
 MFLAGS   := 
 ASFLAGS  := 
-LDFLAGS  :=
 LDFLAGS_END :=
 INCFLAGS :=
 LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
 
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
@@ -835,7 +838,7 @@ LIBS     += -lm
 PREFIX        ?= /usr/local
 
 ifneq (,$(findstring arm, $(ARCH)))
-	CC_AS    = ${CC_PREFIX}gcc #The ngen_arm.S must be compiled with gcc, not as
+	CC_AS    = ${CC_PREFIX}${CC} #The ngen_arm.S must be compiled with gcc, not as
 	ASFLAGS  += $(CFLAGS)
 endif
 
