--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ SOURCES := \
   src/*/*/*.cpp
 
 CPUFLAGS := $(MAKE_CPUFLAGS)
-STRIPCMD := strip --strip-all
+STRIPCMD := ${STRIP}
 ifneq ($(ISWIN),)
   OUTNAME := dosbox_pure_libretro.dll
   CXX     ?= g++
@@ -152,6 +152,9 @@ else
   CXX     ?= g++
   LDFLAGS := -Wl,--gc-sections -fno-ident
   COMMONFLAGS += -pthread
+ifeq (1,1) # always
+  CPUFLAGS :=
+else
   ifeq ($(CPUFLAGS),)
     # ARM optimizations
     PROCCPU := $(shell cat /proc/cpuinfo))
@@ -180,6 +183,7 @@ else
     CXX := g++-9
   endif
 endif
+endif
 
 ifeq ($(BUILD),DEBUG)
   BUILDDIR := debug
