diff -Naur portmap_4-old/Makefile portmap_4-new/Makefile
--- portmap_4-old/Makefile	2008-12-24 11:02:07.000000000 -0800
+++ portmap_4-new/Makefile	2008-12-24 11:02:52.000000000 -0800
@@ -75,10 +75,10 @@
 all:	portmap pmap_dump pmap_set
 
 $(OBJECTS): %.o: %.c
-	$(CC) $(CFLAGS) -fpie -c $<
+	$(CC) $(CFLAGS) -fpic -c $<
 
 portmap: $(OBJECTS) 
-	$(CC) $(CFLAGS) -o $@ -pie $(OBJECTS) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(OBJECTS) $(LIBS)
 
 pmap_dump: pmap_dump.c
 	$(CC) $(CFLAGS) -o $@ $? $(LIBS)
