--- portmap_4/Makefile.orig	Fri May 31 09:50:40 1996
+++ portmap_4/Makefile	Fri Aug  9 17:40:43 1996
@@ -67,9 +67,9 @@
 
 SHELL	= /bin/sh
 
-COPT	= -Dconst= -Dperror=xperror $(HOSTS_ACCESS) $(CHECK_PORT) \
+COPT	= -Dconst= $(HOSTS_ACCESS) $(CHECK_PORT) \
 	$(SYS) -DFACILITY=$(FACILITY) $(ULONG) $(ZOMBIES) $(SA_LEN)
-CFLAGS	= $(COPT) -O $(NSARCHS) $(SETPGRP)
+CFLAGS	= $(COPT) $(RPM_OPT_FLAGS) $(NSARCHS) $(SETPGRP)
 OBJECTS	= portmap.o pmap_check.o from_local.o $(AUX)
 
 all:	portmap pmap_dump pmap_set
--- portmap_4/Makefile.bad	1996-05-31 09:50:40.000000000 -0400
+++ portmap_4/Makefile	2002-12-12 09:54:00.000000000 -0500
@@ -74,8 +74,8 @@
 
 all:	portmap pmap_dump pmap_set
 
-portmap: $(OBJECTS) $(WRAP_DIR)/libwrap.a
-	$(CC) $(CFLAGS) -o $@ $(OBJECTS) $(WRAP_DIR)/libwrap.a $(LIBS)
+portmap: $(OBJECTS) 
+	$(CC) $(CFLAGS) -o $@ $(OBJECTS) $(LIBS)
 
 pmap_dump: pmap_dump.c
 	$(CC) $(CFLAGS) -o $@ $? $(LIBS)
