From 9cd13bb39f2a8ce5d458f5a33010b5c9c5edf131 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 20 Feb 2014 16:34:57 +0200
Subject: [PATCH] dont check for libsystemd-daemon

---
 configure.ac |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index ceb8dd8..bc199c5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,9 +20,6 @@ AC_CHECK_LIB([rt], [main], ,
 	     AC_MSG_ERROR([librt is required but was not found]))
 
 PKG_CHECK_MODULES([SYSTEMD], [systemd])
-PKG_CHECK_MODULES([LIBSYSTEMD_DAEMON], [libsystemd-daemon])
-AC_SUBST(LIBSYSTEMD_DAEMON_CFLAGS)
-AC_SUBST(LIBSYSTEMD_DAEMON_LIBS)
 
 AC_ARG_WITH([systemduserunitdir], AC_HELP_STRING([--with-systemduserunitdir=DIR],
 	[path to systemd user service directory]), [path_systemduserunit=${withval}],
-- 
1.7.2.5

