From 1e220cb2947f5c9ec5d5e35c7e2eb8b7dd728fca Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 22 Nov 2013 14:18:56 +0200
Subject: [PATCH] increase timeout

---
 src/main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index 49e2d43..3c556b5 100644
--- a/src/main.c
+++ b/src/main.c
@@ -85,14 +85,14 @@ int main(int argc, char **argv)
 
 		/* setup sighandler for main thread */
 		clock_gettime(CLOCK_REALTIME, &tv);
-		tv.tv_sec += 10;
+		tv.tv_sec += 30;
 
 		pthread_mutex_lock(&notify_mutex);
 		err = pthread_cond_timedwait(&notify_condition, &notify_mutex, &tv);
 		pthread_mutex_unlock(&notify_mutex);
 
 		if (err == ETIMEDOUT) {
-			fprintf(stderr, "X server startup timed out (10secs). This indicates an"
+			fprintf(stderr, "X server startup timed out (30secs). This indicates an"
 				"an issue in the server configuration or drivers.\n");
 			exit(EXIT_FAILURE);
 		}
-- 
1.7.2.5

