diff -Naur a/Makefile.libretro b/Makefile.libretro
--- a/Makefile.libretro	2016-03-18 14:07:33.000000000 -0700
+++ b/Makefile.libretro	2016-05-10 13:41:01.138087395 -0700
@@ -274,6 +274,16 @@
 		CPUFLAGS += -mfloat-abi=hard
 	endif
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+        TARGET := $(TARGET_NAME)_libretro.so
+        LDFLAGS += -shared -Wl,--version-script=$(ROOT_DIR)/link.T -Wl,--no-undefined
+        fpic := -fPIC
+        CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM
+        PLATCFLAGS += -DARM
+        GLES = 1
+        GL_LIB := -lGLESv2
+
 # emscripten
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_emscripten.bc
