diff -Naur lm_sensors-3.2.0-old/CHANGES lm_sensors-3.2.0-new/CHANGES
--- lm_sensors-3.2.0-old/CHANGES	2010-11-29 10:42:35.000000000 -0800
+++ lm_sensors-3.2.0-new/CHANGES	2010-11-29 10:43:06.000000000 -0800
@@ -4,6 +4,7 @@
 3.2.0 (2010-10-10)
   sensors-detect: Fix error seen if I2C bus numbers are not sequential
   libsensors: Increase MAX_SENSORS_PER_TYPE to 24
+  sensors.conf.defauit: Add a section for the W83627THF
               Map individual beep features
               Change license to LGPL
   pwmconfig: Sample all fan speeds before returning to full speed (#2380)
diff -Naur lm_sensors-3.2.0-old/etc/sensors.conf.default lm_sensors-3.2.0-new/etc/sensors.conf.default
--- lm_sensors-3.2.0-old/etc/sensors.conf.default	2010-07-04 00:22:11.000000000 -0700
+++ lm_sensors-3.2.0-new/etc/sensors.conf.default	2010-11-29 10:43:06.000000000 -0800
@@ -266,6 +266,25 @@
     label temp1 "M/B Temp"
 
 
+chip "w83627thf-*"
+
+    label in3 "+5V"
+    label in7 "5VSB"
+    label in8 "Vbat"
+
+    # Internal resistors
+    compute in3  @ * (1 + 34/51), @ / (1 + 34/51)
+    compute in7  @ * (1 + 34/51), @ / (1 + 34/51)
+
+    set in3_min  5.0 * 0.90
+    set in3_max  5.0 * 1.10
+    set in7_min  5.0 * 0.90
+    set in7_max  5.0 * 1.10
+# The battery voltage may or may not be monitored.
+#    set in8_min  3.0 * 0.90
+#    set in8_max  3.0 * 1.10
+
+
 chip "w83627ehf-*" "w83627dhg-*" "w83667hg-*"
 
     label in0 "Vcore"
