diff -Naur lm_sensors-3.2.0-old/CHANGES lm_sensors-3.2.0-new/CHANGES
--- lm_sensors-3.2.0-old/CHANGES	2010-12-13 18:45:21.000000000 -0800
+++ lm_sensors-3.2.0-new/CHANGES	2010-12-13 18:46:50.000000000 -0800
@@ -5,6 +5,7 @@
   sensors-detect: Fix error seen if I2C bus numbers are not sequential
   libsensors: Increase MAX_SENSORS_PER_TYPE to 24
   sensors.conf.defauit: Add a section for the W83627THF
+                        Add a section for the SCH5127
               Map individual beep features
               Change license to LGPL
   pwmconfig: Sample all fan speeds before returning to full speed (#2380)
diff -Naur lm_sensors-3.2.0-old/etc/sensors.conf.default lm_sensors-3.2.0-new/etc/sensors.conf.default
--- lm_sensors-3.2.0-old/etc/sensors.conf.default	2010-12-13 18:45:21.000000000 -0800
+++ lm_sensors-3.2.0-new/etc/sensors.conf.default	2010-12-13 18:46:50.000000000 -0800
@@ -402,6 +402,20 @@
     set in6_max  3.0 * 1.10
 
 
+chip "sch5127-*"
+
+    label in2 "+3.3V"
+    label in5 "3VSB"
+    label in6 "Vbat"
+
+    set in2_min  3.3 * 0.90
+    set in2_max  3.3 * 1.10
+    set in5_min  3.3 * 0.90
+    set in5_max  3.3 * 1.10
+    set in6_min  3.0 * 0.90
+    set in6_max  3.0 * 1.10
+
+
 chip "f71882fg-*"
 
     label in0 "+3.3V"
