diff -Naur lm_sensors-3.2.0-old/prog/detect/sensors-detect lm_sensors-3.2.0-new/prog/detect/sensors-detect
--- lm_sensors-3.2.0-old/prog/detect/sensors-detect	2010-11-21 07:20:09.000000000 -0800
+++ lm_sensors-3.2.0-new/prog/detect/sensors-detect	2010-11-21 07:21:05.000000000 -0800
@@ -1791,7 +1791,7 @@
 		features => FEAT_IN | FEAT_FAN | FEAT_TEMP,
 	}, {
 		name => "Nuvoton W83667HG-B Super IO Sensors",
-		driver => "to-be-written",	# Probably w83627ehf
+		driver => "w83627ehf",
 		devid => 0xB350,
 		devid_mask => 0xFFF0,
 		logdev => 0x0b,
