diff -Naur lm_sensors-3.2.0-old/etc/sensors.conf.default lm_sensors-3.2.0-new/etc/sensors.conf.default
--- lm_sensors-3.2.0-old/etc/sensors.conf.default	2011-02-07 19:37:38.000000000 -0800
+++ lm_sensors-3.2.0-new/etc/sensors.conf.default	2011-02-07 19:38:19.000000000 -0800
@@ -285,7 +285,7 @@
 #    set in8_max  3.0 * 1.10
 
 
-chip "w83627ehf-*" "w83627dhg-*" "w83667hg-*"
+chip "w83627ehf-*" "w83627dhg-*" "w83667hg-*" "nct6775-*" "nct6776-*"
 
     label in0 "Vcore"
     label in2 "AVCC"
