diff -Naur lm_sensors-3.2.0-old/CHANGES lm_sensors-3.2.0-new/CHANGES
--- lm_sensors-3.2.0-old/CHANGES	2010-10-10 12:15:54.000000000 -0700
+++ lm_sensors-3.2.0-new/CHANGES	2010-10-19 20:13:47.000000000 -0700
@@ -2,6 +2,7 @@
 -----------------------
 
 3.2.0 (2010-10-10)
+  sensors-detect: Fix error seen if I2C bus numbers are not sequential
   libsensors: Increase MAX_SENSORS_PER_TYPE to 24
               Map individual beep features
               Change license to LGPL
diff -Naur lm_sensors-3.2.0-old/prog/detect/sensors-detect lm_sensors-3.2.0-new/prog/detect/sensors-detect
--- lm_sensors-3.2.0-old/prog/detect/sensors-detect	2010-09-21 08:21:05.000000000 -0700
+++ lm_sensors-3.2.0-new/prog/detect/sensors-detect	2010-10-19 20:13:47.000000000 -0700
@@ -5930,6 +5930,7 @@
 	# the adapter drivers so that the numbers will be the same. If not, then
 	# we only load the adapter drivers which are useful.
 	foreach $adap (@i2c_adapters) {
+		next unless defined $adap;
 		next if $adap->{autoload};
 		next if $adap->{driver} eq 'UNKNOWN';
 		next if not defined $configfile and not $adap->{used};
