diff -Naur pango-1.32.5/configure.ac pango-1.32.5.patch/configure.ac
--- pango-1.32.5/configure.ac	2012-12-07 03:43:25.000000000 +0100
+++ pango-1.32.5.patch/configure.ac	2013-01-12 18:24:57.601881568 +0100
@@ -54,7 +54,7 @@
 dnl usage of GNU Make specific features.
 AM_INIT_AUTOMAKE(1.9 gnits dist-xz no-dist-gzip -Wno-portability)
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 
 PANGO_VERSION_MAJOR=pango_version_major()
