--- makemkv_v1.6.8_oss/makefile.linux	2011-04-25 08:44:43.000000000 +0200
+++ makemkv_v1.6.8_oss.patch/makefile.linux	2011-04-25 21:26:11.262636081 +0200
@@ -14,7 +14,7 @@
 BUILDINFO_BUILD_DATE=$(shell date)
 
 
-all: out/libdriveio.so.0 out/libmakemkv.so.1 out/makemkv
+all: out/libdriveio.so.0 out/libmakemkv.so.1
 	@echo "type \"sudo make -f makefile.linux install\" to install"
 
 clean:
@@ -23,13 +23,11 @@
 	rm -f out/*
 	rm -f tmp/*
 
-install: out/libdriveio.so.0 out/libmakemkv.so.1 out/makemkv
+install: out/libdriveio.so.0 out/libmakemkv.so.1
 	rm -f /usr/lib/libdriveio.so.0
 	rm -f /usr/lib/libmakemkv.so.1
-	rm -f /usr/bin/makemkv
 	install -t /usr/lib out/libdriveio.so.0 out/libmakemkv.so.1
 	ldconfig
-	install -t /usr/bin out/makemkv
 
 out/%: out/%.full
 	objcopy --strip-all --strip-debug --strip-unneeded --discard-all $< $@ 
@@ -42,7 +40,7 @@
 
 out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 	mkdir -p out
-	$(GCC) $(CXXOPTS) -D_REENTRANT -shared -Wl,-z,defs -o$@ $(LIBEBML_INC) $(LIBMATROSKA_INC) \
+	$(GCC) $(CXXOPTS) -L$(PKG_CONFIG_SYSROOT_DIR)/usr/lib -DOPENSSL_NO_EC -D_REENTRANT -shared -Wl,-z,defs -o$@ $(LIBEBML_INC) $(LIBMATROSKA_INC) \
 	$(LIBMAKEMKV_INC) $(SSTRING_INC) $(MAKEMKVGUI_INC) $(LIBABI_INC) $(LIBFFCODEC_INC) \
 	$(LIBEBML_SRC) $(LIBMATROSKA_SRC) $(LIBMAKEMKV_SRC) $(GLIBC_SRC) $(SSTRING_SRC) \
 	$(LIBABI_SRC) $(LIBABI_SRC_LINUX) $(LIBFFCODEC_SRC) \
