--- makemkv_v1.6.5_oss/makefile.linux	2011-02-16 05:16:27.000000000 +0100
+++ makemkv_v1.6.5_oss.patch/makefile.linux	2011-03-21 00:30:21.946735001 +0100
@@ -12,7 +12,7 @@
 BUILDINFO_BUILD_DATE=$(shell date)
 
 
-all: out/libdriveio.so.0 out/libmakemkv.so.1 out/makemkv
+all: out/libdriveio.so.0 out/libmakemkv.so.1
 	@echo "type \"sudo make -f makefile.linux install\" to install"
 
 clean:
@@ -40,7 +40,7 @@
 
 out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 	mkdir -p out
-	$(GCC) -D_REENTRANT -shared -Wl,-z,defs -o$@ $(LIBEBML_INC) $(LIBMATROSKA_INC) \
+	$(GCC) -DOPENSSL_NO_EC -D_REENTRANT -shared -Wl,-z,defs -o$@ -L$(SYSROOT_PREFIX)/usr/lib $(LIBEBML_INC) $(LIBMATROSKA_INC) \
 	$(LIBMAKEMKV_INC) $(SSTRING_INC) $(MAKEMKVGUI_INC) $(LIBABI_INC) $(LIBFFCODEC_INC) \
 	$(LIBEBML_SRC) $(LIBMATROSKA_SRC) $(LIBMAKEMKV_SRC) $(GLIBC_SRC) $(SSTRING_SRC) \
 	$(LIBABI_SRC) $(LIBABI_SRC_LINUX) $(LIBFFCODEC_SRC) \
