diff -Naur MPlayer-export-2009-06-13.orig/command.c MPlayer-export-2009-06-13/command.c
--- MPlayer-export-2009-06-13.orig/command.c	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/command.c	2009-06-13 22:55:21.000000000 +0200
@@ -2510,7 +2510,12 @@
 	    exit_player_with_rc(EXIT_QUIT,
 				(cmd->nargs > 0) ? cmd->args[0].v.i : 0);
 
-	case MP_CMD_PLAY_TREE_STEP:{
+	case MP_CMD_SET_OPTION: {
+	    extern struct m_config* mconfig;
+	    m_config_set_option(mconfig, cmd->args[0].v.s, cmd->args[1].v.s);
+	} break;
+
+        case MP_CMD_PLAY_TREE_STEP:{
 		int n = cmd->args[0].v.i == 0 ? 1 : cmd->args[0].v.i;
 		int force = cmd->args[1].v.i;
 
diff -Naur MPlayer-export-2009-06-13.orig/input/input.c MPlayer-export-2009-06-13/input/input.c
--- MPlayer-export-2009-06-13.orig/input/input.c	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/input/input.c	2009-06-13 22:55:21.000000000 +0200
@@ -140,6 +140,7 @@
   { MP_CMD_SWITCH_AUDIO, "switch_audio", 0, { { MP_CMD_ARG_INT,{-1} }, {-1,{0}} } },
   { MP_CMD_SWITCH_ANGLE, "switch_angle", 0, { { MP_CMD_ARG_INT,{-1} }, {-1,{0}} } },
   { MP_CMD_SWITCH_TITLE, "switch_title", 0, { { MP_CMD_ARG_INT,{-1} }, {-1,{0}} } },
+  { MP_CMD_SET_OPTION, "set_option", 2, { { MP_CMD_ARG_STRING, {0}}, { MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
 #ifdef CONFIG_TV
   { MP_CMD_TV_START_SCAN, "tv_start_scan", 0,  { {-1,{0}} }},
   { MP_CMD_TV_STEP_CHANNEL, "tv_step_channel", 1,  { { MP_CMD_ARG_INT ,{0}}, {-1,{0}} }},
diff -Naur MPlayer-export-2009-06-13.orig/input/input.h MPlayer-export-2009-06-13/input/input.h
--- MPlayer-export-2009-06-13.orig/input/input.h	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/input/input.h	2009-06-13 22:55:21.000000000 +0200
@@ -129,6 +129,7 @@
   MP_CMD_ASS_USE_MARGINS,
   MP_CMD_SWITCH_TITLE,
   MP_CMD_STOP,
+  MP_CMD_SET_OPTION,
 
   /// DVDNAV commands
   MP_CMD_DVDNAV_UP = 1000,
