diff -Naur MPlayer-export-2009-06-13.orig/cfg-mplayer.h MPlayer-export-2009-06-13/cfg-mplayer.h
--- MPlayer-export-2009-06-13.orig/cfg-mplayer.h	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/cfg-mplayer.h	2009-06-13 22:56:04.000000000 +0200
@@ -315,6 +315,8 @@
 	{"guiwid", &guiWinID, CONF_TYPE_INT, 0, 0, 0, NULL},
 #endif
 
+	{"bgvideo", &bg_video, CONF_TYPE_STRING, 0, 0, 0, NULL},
+
 	{"noloop", &mpctx_s.loop_times, CONF_TYPE_FLAG, 0, 0, -1, NULL},
 	{"loop", &mpctx_s.loop_times, CONF_TYPE_INT, CONF_RANGE, -1, 10000, NULL},
 	{"playlist", NULL, CONF_TYPE_STRING, 0, 0, 0, NULL},
diff -Naur MPlayer-export-2009-06-13.orig/mp_core.h MPlayer-export-2009-06-13/mp_core.h
--- MPlayer-export-2009-06-13.orig/mp_core.h	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/mp_core.h	2009-06-13 22:56:04.000000000 +0200
@@ -60,6 +60,7 @@
 
     stream_t *stream;
     demuxer_t *demuxer;
+    demuxer_t *bg_demuxer;
     sh_audio_t *sh_audio;
     sh_video_t *sh_video;
     demux_stream_t *d_audio;
diff -Naur MPlayer-export-2009-06-13.orig/mplayer.c MPlayer-export-2009-06-13/mplayer.c
--- MPlayer-export-2009-06-13.orig/mplayer.c	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/mplayer.c	2009-06-13 22:58:14.000000000 +0200
@@ -271,6 +271,8 @@
 int forced_subs_only=0;
 int file_filter=1;
 
+static char* bg_video = NULL;
+
 // cache2:
        int stream_cache_size=-1;
 #ifdef CONFIG_STREAM_CACHE
@@ -608,6 +610,13 @@
 	free_demuxer(mpctx->demuxer);
     }
     mpctx->demuxer=NULL;
+    current_module="free_bg_demuxer";
+    if(mpctx->bg_demuxer) {
+      stream_t* bg_s = mpctx->bg_demuxer->stream;
+      free_demuxer(mpctx->bg_demuxer);
+      mpctx->bg_demuxer = NULL;
+      free_stream(bg_s);
+    }
   }
 
   // kill the cache process:
@@ -1528,6 +1537,8 @@
             int percentage = -1;
             char percentage_text[10];
             int pts = demuxer_get_current_time(mpctx->demuxer);
+            if (mpctx->bg_demuxer)
+                pts = playing_audio_pts(mpctx->sh_audio, mpctx->d_audio, mpctx->audio_out);
 
             if (mpctx->osd_show_percentage)
                 percentage = demuxer_get_percent_pos(mpctx->demuxer);
@@ -1743,10 +1754,17 @@
 	current_module = "video_read_frame";
 	in_size = ds_get_packet_pts(d_video, &start, &pts);
 	if (in_size < 0) {
+	  if(mpctx->bg_demuxer) {
+	    if(!demux_seek(mpctx->bg_demuxer,0,0,1))
+	      hit_eof = 1;
+	    else
+	      continue;
+	  } else {
 	    // try to extract last frames in case of decoder lag
 	    in_size = 0;
 	    pts = 1e300;
 	    hit_eof = 1;
+	  }
 	}
 	if (in_size > max_framesize)
 	    max_framesize = in_size;
@@ -1939,6 +1957,13 @@
     current_module="av_sync";
 
     if(mpctx->sh_audio){
+     if(mpctx->bg_demuxer) {
+      if(!quiet) mp_msg(MSGT_AVSYNC,MSGL_STATUS,"A:%6.1f %4.1f%% %d%%   \r"
+ 		       ,mpctx->delay - mpctx->audio_out->get_delay()
+ 		       ,(mpctx->delay>0.5)?100.0*audio_time_usage/(double)mpctx->delay:0
+ 		       ,cache_fill_status
+ 		       );
+     } else {
 	double a_pts, v_pts;
 
 	if (autosync)
@@ -1989,6 +2014,7 @@
 	    if(!quiet)
 		print_status(a_pts - audio_delay, AV_delay, c_total);
 	}
+    }
 
     } else {
 	// No audio:
@@ -3348,6 +3374,34 @@
 mpctx->sh_audio=mpctx->d_audio->sh;
 mpctx->sh_video=mpctx->d_video->sh;
 
+while(mpctx->sh_audio && !mpctx->sh_video && bg_video) {
+  int bg_file_format = 0;
+  stream_t* bg_s;
+  mp_msg(MSGT_CPLAYER,MSGL_INFO, "Open bgvideo: %s\n",bg_video);
+  bg_s = open_stream(bg_video,NULL,&bg_file_format);
+  if(!bg_s) {
+    mp_msg(MSGT_CPLAYER,MSGL_ERR, "Couldn't open the stream for back ground video: %s\n",bg_video);
+    break;
+  }
+  mpctx->bg_demuxer = demux_open(bg_s,bg_file_format,-2,video_id,dvdsub_id,bg_video);
+  if(!mpctx->bg_demuxer) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"Failed to open background video\n");
+    free_stream(bg_s);
+    break;
+  }
+  if(!mpctx->bg_demuxer->video && !mpctx->bg_demuxer->video->sh) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"The background video don't have a video stream\n");
+    free_demuxer(mpctx->bg_demuxer);
+    mpctx->bg_demuxer = NULL;
+    free_stream(bg_s);
+    break;
+  }
+  mpctx->d_video = mpctx->bg_demuxer->video;
+  mpctx->sh_video = mpctx->d_video->sh;
+  mp_msg(MSGT_DEMUXER,MSGL_INFO,"Background video should work ;)\n");
+  break;
+}
+
 if(mpctx->sh_video){
 
   current_module="video_read_properties";
@@ -3665,6 +3719,7 @@
 	if (!mpctx->sh_video)
 	    mpctx->eof = PT_NEXT_ENTRY;
 
+if(mpctx->bg_demuxer && mpctx->d_audio->eof) mpctx->eof=1;
 
 if(!mpctx->sh_video) {
   // handle audio-only case:
@@ -3696,6 +3751,11 @@
 	  mpctx->eof = 1; goto goto_next_file;
       }
       if (frame_time < 0)
+        if(mpctx->bg_demuxer) {
+          if(!demux_seek(mpctx->bg_demuxer,0,0,1))
+            mpctx->eof = PT_NEXT_ENTRY;
+        }
+        else
 	  mpctx->eof = 1;
       else {
 	  // might return with !eof && !blit_frame if !correct_pts
