diff -Naur MPlayer-r28852.orig/libao2/ao_mpegpes.c MPlayer-r28852/libao2/ao_mpegpes.c
--- MPlayer-r28852.orig/libao2/ao_mpegpes.c	2009-03-06 20:57:55.000000000 +0100
+++ MPlayer-r28852/libao2/ao_mpegpes.c	2009-06-01 22:29:16.000000000 +0200
@@ -50,6 +50,7 @@
 #include <ost/audio.h>
 audioMixer_t dvb_mixer={255,255};
 #else
+#include <linux/types.h>
 #include <linux/dvb/audio.h>
 audio_mixer_t dvb_mixer={255,255};
 #endif
diff -Naur MPlayer-r28852.orig/libvo/vo_mpegpes.c MPlayer-r28852/libvo/vo_mpegpes.c
--- MPlayer-r28852.orig/libvo/vo_mpegpes.c	2009-03-06 20:57:55.000000000 +0100
+++ MPlayer-r28852/libvo/vo_mpegpes.c	2009-06-01 22:29:26.000000000 +0200
@@ -60,6 +60,7 @@
 #include <stdio.h>
 #include <time.h>
 
+#include <linux/types.h>
 #include <linux/dvb/dmx.h>
 #include <linux/dvb/frontend.h>
 #include <linux/dvb/video.h>
