diff -Naur MPlayer-export-2009-06-13.orig/libmpcodecs/vd_ffmpeg.c MPlayer-export-2009-06-13/libmpcodecs/vd_ffmpeg.c
--- MPlayer-export-2009-06-13.orig/libmpcodecs/vd_ffmpeg.c	2009-06-13 22:53:58.000000000 +0200
+++ MPlayer-export-2009-06-13/libmpcodecs/vd_ffmpeg.c	2009-06-14 00:32:06.000000000 +0200
@@ -145,7 +145,16 @@
         avcodec_flush_buffers(avctx);
         return CONTROL_TRUE;
     case VDCTRL_QUERY_UNSEEN_FRAMES:
-        return avctx->has_b_frames + 10;
+#ifdef FF_THREAD_FRAME
+    {
+      int thread_delay = 0;
+      if (avctx->active_thread_type & FF_THREAD_FRAME)
+        thread_delay = avctx->thread_count - 1;
+      return avctx->has_b_frames + thread_delay + 10;
+    }
+#else
+       return avctx->has_b_frames + 10;
+#endif
     }
     return CONTROL_UNKNOWN;
 }
