diff -Nurp a/input/input.c b/input/input.c
--- a/input/input.c	2009-05-28 22:11:18.000000000 +0800
+++ b/input/input.c	2009-05-28 21:48:56.000000000 +0800
@@ -317,6 +317,13 @@ static const mp_key_name_t key_names[] =
   { JOY_BTN7, "JOY_BTN7" },
   { JOY_BTN8, "JOY_BTN8" },
   { JOY_BTN9, "JOY_BTN9" },
+  { JOY_BTN10, "JOY_BTN10" },
+  { JOY_BTN11, "JOY_BTN11" },
+  { JOY_BTN12, "JOY_BTN12" },
+  { JOY_BTN13, "JOY_BTN13" },
+  { JOY_BTN14, "JOY_BTN14" },
+  { JOY_BTN15, "JOY_BTN15" },
+  { JOY_BTN16, "JOY_BTN16" },
 
   { AR_PLAY, "AR_PLAY" },
   { AR_PLAY_HOLD, "AR_PLAY_HOLD" },
diff -Nurp a/input/joystick.c b/input/joystick.c
--- a/input/joystick.c	2008-12-14 02:28:00.000000000 +0800
+++ b/input/joystick.c	2009-05-28 22:09:07.000000000 +0800
@@ -45,6 +45,7 @@
 
 int axis[256];
 int btns = 0;
+char name[128];
 
 int mp_input_joystick_init(char* dev) {
   int fd,l=0;
@@ -68,6 +69,9 @@ int mp_input_joystick_init(char* dev) {
 	  continue;
 	else if(errno == EAGAIN) {
 	  initialized = 1;
+	if (ioctl(fd, JSIOCGNAME(sizeof(name)), name) < 0)
+	  strncpy(name, "Unknown", sizeof(name));
+	  mp_msg(MSGT_INPUT,MSGL_INFO,"Using joystick: %s\n",name);
 	  break;
 	}
 	mp_msg(MSGT_INPUT,MSGL_ERR,MSGTR_INPUT_JOYSTICK_ErrReading,strerror(errno));
@@ -76,6 +79,7 @@ int mp_input_joystick_init(char* dev) {
       }	
       l += r;
     }
+
     if((unsigned int)l < sizeof(struct js_event)) {
       if(l > 0)
 	mp_msg(MSGT_INPUT,MSGL_WARN,MSGTR_INPUT_JOYSTICK_LoosingBytes,l);	  
@@ -141,6 +145,8 @@ int mp_input_joystick_read(int fd) {
     else
       return JOY_BTN0 + ev.number;
   } else if(ev.type & JS_EVENT_AXIS) {
+    if (!strncmp(name, "Sony PLAYSTATION(R)3 Controller", 31))
+      return MP_INPUT_NOTHING; // input doesn't handle axis events well, ignore them all
     if(ev.value < -JOY_AXIS_DELTA && axis[ev.number] != -1) {
       axis[ev.number] = -1;
       return (JOY_AXIS0_MINUS+(2*ev.number)) | MP_KEY_DOWN;
diff -Nurp a/input/joystick.h b/input/joystick.h
--- a/input/joystick.h	2008-12-14 02:28:00.000000000 +0800
+++ b/input/joystick.h	2009-05-28 21:47:52.000000000 +0800
@@ -52,6 +52,13 @@
 #define JOY_BTN7 (JOY_BTN_BASE+7)
 #define JOY_BTN8 (JOY_BTN_BASE+8)
 #define JOY_BTN9 (JOY_BTN_BASE+9)
+#define JOY_BTN10 (JOY_BTN_BASE+10)
+#define JOY_BTN11 (JOY_BTN_BASE+11)
+#define JOY_BTN12 (JOY_BTN_BASE+12)
+#define JOY_BTN13 (JOY_BTN_BASE+13)
+#define JOY_BTN14 (JOY_BTN_BASE+14)
+#define JOY_BTN15 (JOY_BTN_BASE+15)
+#define JOY_BTN16 (JOY_BTN_BASE+16)
 
 int mp_input_joystick_init(char* dev);
 
