diff -Naur atv-bootloader-r518/Makefile atv-bootloader-r518.patch/Makefile
--- atv-bootloader-r518/Makefile	2010-03-09 20:17:22.000000000 +0100
+++ atv-bootloader-r518.patch/Makefile	2011-05-21 14:04:05.592144598 +0200
@@ -11,12 +11,12 @@
 
 # if Linxu, use the darwin-cross tools to compile/link
 ifeq ($(OSTYPE),Linux)
-  CC  = /opt/darwin-cross/bin/i386-apple-darwin8-gcc-4.0
-  LD  = /opt/darwin-cross/bin/i386-apple-darwin8-ld
+  CC  := /opt/darwin-cross/bin/i386-apple-darwin8-gcc-4.0
+  LD  := /opt/darwin-cross/bin/i386-apple-darwin8-ld
 else
-  CC  = /usr/bin/gcc-4.0
-  LD  = /usr/bin/ld
-	LDFLAGS = -classic_linker
+  CC  := /usr/bin/gcc-4.0
+  LD  := /usr/bin/ld
+  LDFLAGS = -classic_linker
 endif
 
 # start.o must be 1st in the link order (ld below)
@@ -53,7 +53,4 @@
 clean:
 	rm -f *.o $(KERN_OBJ) mach_kernel
 
-
-
 #xxd mach_kernel | sed -e "s/ffff ffff 1000/0100 0000 1000/" | xxd -r - mach_kernel
-
