diff -Naur atv-bootloader-r518/Makefile atv-bootloader-r518.patch/Makefile
--- atv-bootloader-r518/Makefile	2010-03-09 20:17:22.000000000 +0100
+++ atv-bootloader-r518.patch/Makefile	2011-05-21 15:26:26.509551389 +0200
@@ -22,7 +22,7 @@
 # start.o must be 1st in the link order (ld below)
 OBJ	= start.o vsprintf.o console.o utils.o elilo_code.o darwin_code.o linux_code.o boot_loader.o
 
-KERN_OBJ= vmlinuz.obj initrd.obj
+KERN_OBJ:= vmlinuz.obj initrd.obj
 
 mach_kernel: $(KERN_OBJ) $(OBJ)
 	$(LD) $(LDFLAGS) -arch $(ARCH) -o mach_kernel $(OBJ) $(KERN_OBJ) \
