--- a/tools/Makefile	2019-05-05 18:38:23.284581772 +0200
+++ b/tools/Makefile	2019-05-05 18:38:39.844358910 +0200
@@ -120,7 +120,7 @@
 libfdt:
 
 tools/_libfdt.so: $(patsubst %.o,%.c,$(LIBFDT_OBJS)) tools/libfdt_wrap.c
-	LDFLAGS="$(HOSTLDFLAGS)" CFLAGS= ${PYTHON} $(srctree)/lib/libfdt/setup.py \
+	LDFLAGS="$(HOSTLDFLAGS)" CFLAGS=-I${TOOLCHAIN}/include/python2.7 ${PYTHON} $(srctree)/lib/libfdt/setup.py \
 		"$(_hostc_flags)" $^
 	mv _libfdt.so $@
 
