diff -Naur faad2-2.7-old/common/mp4ff/Makefile.am faad2-2.7-new/common/mp4ff/Makefile.am
--- faad2-2.7-old/common/mp4ff/Makefile.am	2009-02-06 06:24:21.000000000 -0800
+++ faad2-2.7-new/common/mp4ff/Makefile.am	2009-02-11 05:36:14.000000000 -0800
@@ -1,7 +1,7 @@
-lib_LIBRARIES = libmp4ff.a
-include_HEADERS = mp4ff.h mp4ffint.h
+lib_LTLIBRARIES = libmp4ff.la
+include_HEADERS = mp4ff.h mp4ffint.h mp4ff_int_types.h
 
-libmp4ff_a_CFLAGS = -DUSE_TAGGING=1
+libmp4ff_la_CFLAGS = -DUSE_TAGGING=1
 
-libmp4ff_a_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
-		     mp4tagupdate.c mp4ff.h mp4ffint.h mp4ff_int_types.h
+libmp4ff_la_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
+                      mp4tagupdate.c mp4ff.h mp4ffint.h mp4ff_int_types.h
diff -Naur faad2-2.7-old/frontend/Makefile.am faad2-2.7-new/frontend/Makefile.am
--- faad2-2.7-old/frontend/Makefile.am	2009-02-06 08:03:37.000000000 -0800
+++ faad2-2.7-new/frontend/Makefile.am	2009-02-11 05:35:52.000000000 -0800
@@ -5,7 +5,7 @@
 	   -I$(top_srcdir)/common/mp4ff
 
 faad_LDADD = $(top_builddir)/libfaad/libfaad.la \
-	     $(top_builddir)/common/mp4ff/libmp4ff.a
+	     $(top_builddir)/common/mp4ff/libmp4ff.la
 
 faad_SOURCES = main.c \
 	       audio.c audio.h \
