diff -Naur faad2-2.7-old/configure.in faad2-2.7-new/configure.in
--- faad2-2.7-old/configure.in	2009-02-04 16:51:03.000000000 -0800
+++ faad2-2.7-new/configure.in	2012-12-30 14:58:33.000000000 -0800
@@ -25,7 +25,7 @@
 AC_PROG_MAKE_SET
 AC_CHECK_PROGS(RPMBUILD, rpmbuild, rpm)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_ARG_WITH(xmms,[  --with-xmms             compile XMMS-1 plugin],
 	     WITHXMMS=$withval, WITHXMMS=no)
