diff -Naur kexec-tools-2.0.2/purgatory/Makefile kexec-tools-2.0.2.patch/purgatory/Makefile
--- kexec-tools-2.0.2/purgatory/Makefile	2010-07-29 06:19:59.000000000 +0200
+++ kexec-tools-2.0.2.patch/purgatory/Makefile	2011-10-30 18:38:31.462335639 +0100
@@ -56,7 +56,7 @@
 			-I$(srcdir)/include \
 			-I$(shell $(CC) -print-file-name=include)
 $(PURGATORY): LDFLAGS=$($(ARCH)_PURGATORY_EXTRA_CFLAGS)\
-			--no-undefined -nostartfiles -nostdlib -nodefaultlibs \
+			-nostartfiles -nostdlib -nodefaultlibs \
 			-e purgatory_start -r
 
 $(PURGATORY): $(PURGATORY_OBJS)
