diff -Naur sshguard-1.5/configure.ac sshguard-1.5.patch/configure.ac
--- sshguard-1.5/configure.ac	2011-02-09 13:01:47.000000000 +0100
+++ sshguard-1.5.patch/configure.ac	2013-01-12 01:13:22.735102536 +0100
@@ -5,7 +5,7 @@
 AC_INIT(sshguard, 1.5, mij@sshguard.net)
 AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/simclist.c])
-AM_CONFIG_HEADER([src/config.h])
+AC_CONFIG_HEADERS([src/config.h])
 # test if we are on Solaris, which requires outstanding practices
 AC_CANONICAL_BUILD
 AS_CASE([$build_os], 
