diff -Naur mergerfs-2.24.0/Makefile mergerfs-2.24.0-2/Makefile
--- mergerfs-2.24.0/Makefile	2023-11-02 23:58:29.373436471 +0100
+++ mergerfs-2.24.0-2/Makefile	2023-11-02 23:59:14.686439139 +0100
@@ -12,7 +12,7 @@
 #  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 #  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
-PKGCONFIG =  	$(shell which pkg-config)
+PKGCONFIG :=  	$(shell which pkg-config)
 GIT 	  = 	$(shell which git)
 TAR 	  = 	$(shell which tar)
 MKDIR     = 	$(shell which mkdir)
