diff -Naur lirc-20100104/drivers/lirc_dev/lirc_dev.c lirc-20100104.patch/drivers/lirc_dev/lirc_dev.c
--- lirc-20100104/drivers/lirc_dev/lirc_dev.c	2009-12-28 16:20:32.000000000 +0100
+++ lirc-20100104.patch/drivers/lirc_dev/lirc_dev.c	2010-01-04 15:42:28.410579354 +0100
@@ -32,7 +32,7 @@
 #error "**********************************************************"
 #endif
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
diff -Naur lirc-20100104/drivers/lirc_mceusb/lirc_mceusb.c lirc-20100104.patch/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-20100104/drivers/lirc_mceusb/lirc_mceusb.c	2009-12-18 08:15:12.000000000 +0100
+++ lirc-20100104.patch/drivers/lirc_mceusb/lirc_mceusb.c	2010-01-04 15:57:51.868549793 +0100
@@ -47,7 +47,7 @@
 #error "Sorry, this driver needs kernel version 2.6.5 or higher"
 #error "*******************************************************"
 #endif
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
