diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-06-08 13:06:09.757102470 -0700
+++ b/configure.ac	2016-06-08 13:07:04.620705345 -0700
@@ -40,9 +40,9 @@
 AM_CONDITIONAL(HAVE_DOXYGEN, test x$DOXYGEN = xyes)
 AC_PROG_LN_S
 LT_INIT([disable-static])
-AM_PATH_PYTHON([3.1],,)
-AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != ""])
-AC_CHECK_PROG([HAVE_PYTHON3],[python3],[yes],[no])
+AM_PATH_PYTHON([3.1],,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [false])
+AC_CHECK_PROG([HAVE_PYTHON3],[python3],[no],[no])
 LIBUSB_CONFIG="pkg-config libusb"
 
 dnl Checks for header files.
diff -Naur a/tools/Makefile.am b/tools/Makefile.am
--- a/tools/Makefile.am	2016-05-23 10:18:48.000000000 -0700
+++ b/tools/Makefile.am	2016-06-08 13:15:25.711341537 -0700
@@ -114,22 +114,7 @@
 
 install-exec-hook:
 	$(SED) -i -e 's/@version@/$(VERSION)/' \
-	    $(DESTDIR)$(bindir)/lirc-config-tool \
-	    $(DESTDIR)$(bindir)/irdb-get
-	rm -f $(DESTDIR)/$(bindir)/lirc-setup
-	$(srcdir)/make_rel_symlink.py \
-	    $(DESTDIR)/$(pkgpythondir)/lirc-setup \
-	    $(DESTDIR)/$(bindir)/lirc-setup
-
-install-data-hook:
-	rm -f $(DESTDIR)/$(pkgpythondir)/configs
-	$(srcdir)/make_rel_symlink.py \
-	    $(DESTDIR)$(pkgdatadir)/configs \
-	    $(DESTDIR)$(pkgpythondir)/configs
-
-uninstall-hook:
-	rm -f $(DESTDIR)/$(bindir)/lirc-setup \
-	      $(DESTDIR)/$(pkgpythondir)/configs
+	    $(DESTDIR)$(bindir)/lirc-config-tool
 
 clean-local:
 	-find . -name __pycache__ -exec rm -r {} \;
