From e2fcade8acd9280873bea676cbcf2f5e43a70e2a Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hiassoft@users.sf.net>
Date: Wed, 5 Apr 2017 13:40:27 +0200
Subject: [PATCH] lib: Fix bad driver initialization (#279).

---
 lib/drv_admin.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/drv_admin.c b/lib/drv_admin.c
index 66046ad..1c35560 100644
--- a/lib/drv_admin.c
+++ b/lib/drv_admin.c
@@ -259,6 +259,7 @@ int hw_choose_driver(const char* name)
 
 	if (name == NULL) {
 		memcpy(&drv, &drv_null, sizeof(struct driver));
+		drv.fd = -1;
 		return 0;
 	}
 	if (strcasecmp(name, "dev/input") == 0)
@@ -267,6 +268,7 @@ int hw_choose_driver(const char* name)
 	found = for_each_driver(match_hw_name, (void*)name, NULL);
 	if (found != (struct driver*)NULL) {
 		memcpy(&drv, found, sizeof(struct driver));
+		drv.fd = -1;
 		return 0;
 	}
 	return -1;
-- 
2.1.4

