--- a/plugins/devinput.c	2017-09-10 08:52:19.000000000 +0000
+++ b/plugins/devinput.c	2022-07-19 13:59:01.653286222 +0000
@@ -459,7 +459,7 @@
 		return 0;
 	}
 
-	log_trace("time %ld.%06ld  type %d  code %d  value %d", event.time.tv_sec, event.time.tv_usec, event.type,
+	log_trace("time %ld.%06ld  type %d  code %d  value %d", event.input_event_sec, event.input_event_usec, event.type,
 		  event.code, event.value);
 
 	value = (unsigned)event.value;
