From: Alec Leamas <leamas.alec@gmail.com>
Date: Fri, 23 Nov 2018 23:04:35 -0500
Subject: [PATCH] logging: Don't use broken LOG_CONS syslog flag.

---
 lib/lirc_log.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/lirc_log.c b/lib/lirc_log.c
index bea357b..900beb3 100644
--- a/lib/lirc_log.c
+++ b/lib/lirc_log.c
@@ -102,9 +102,9 @@ int lirc_log_open(const char* _progname, int _nodaemon, loglevel_t level)
 
 	if (use_syslog) {
 		if (nodaemon)
-			openlog(syslogident, LOG_CONS | LOG_PID | LOG_PERROR, LOG_LOCAL0);
+			openlog(syslogident, LOG_PID | LOG_PERROR, LOG_LOCAL0);
 		else
-			openlog(syslogident, LOG_CONS | LOG_PID, LOG_LOCAL0);
+			openlog(syslogident, LOG_PID, LOG_LOCAL0);
 	} else {
 		lf = fopen(logfile, "a");
 		if (lf == NULL) {
