diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-05-23 10:18:48.000000000 -0700
+++ b/configure.ac	2016-06-08 02:09:11.335251580 -0700
@@ -33,7 +33,7 @@
     AC_MSG_ERROR([
       xsltproc is required to build. It usually comes with libxml])
 fi
-AM_CONDITIONAL(HAVE_UINPUT, test -e /dev/uinput)
+AM_CONDITIONAL(HAVE_UINPUT, true)
 AC_CHECK_PROG([MAN2HTML],[man2html],[yes],[no])
 AM_CONDITIONAL(HAVE_MAN2HTML, test x$MAN2HTML = xyes)
 AC_CHECK_PROG([DOXYGEN],[doxygen],[yes],[no])
