--- a/lib/Makefile.am.org	2016-05-23 19:18:48.000000000 +0200
+++ a/lib/Makefile.am	2016-06-22 22:15:26.511048224 +0200
@@ -31,9 +31,11 @@
 
 libirrecord_la_LDFLAGS      = -llirc
 libirrecord_la_SOURCES      = irrecord.c
+libirrecord_la_DEPENDENCIES = liblirc.la
 
 liblirc_client_la_LDFLAGS   = -version-info 4:0:4
 liblirc_client_la_SOURCES   = lirc_client.c lirc_client.h lirc_log.c lirc_log.h
+liblirc_client_la_DEPENDENCIES = liblirc.la
 
 liblirc_driver_la_LDFLAGS   = -version-info 0:0:0
 liblirc_driver_la_SOURCES   = driver.h \
@@ -49,6 +51,7 @@
                               serial.c \
                               transmit.c \
                               transmit.h
+liblirc_driver_la_DEPENDENCIES = liblirc.la
 
 dist_include_HEADERS        = lirc_client.h \
                               lirc_driver.h \
