diff --git a/Makefile.am b/Makefile.am
index ba7150c..609b050 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -39,7 +39,7 @@ if WITH_SYSTEMDSYSTEMUNITDIR
     SYSTEMD_DIR         = systemd
 endif
 
-SUBDIRS                 = lib daemons tools plugins configs doc $(SYSTEMD_DIR)
+SUBDIRS                 = lib daemons tools plugins $(SYSTEMD_DIR)
 
 if INSTALL_ETC
 
@@ -60,6 +60,7 @@ endif
 
 dist_doc_DATA           = VERSION
 
+if HAVE_PYTHON
 nodist_pkgdata_DATA     = $(PYTHON_TARBALL)
 
 pkgconfigdir            = $(libdir)/pkgconfig
@@ -133,6 +134,8 @@ all-local:
 	    $(if $(VERBOSE),,-q) build
 endif
 
+endif
+
 install-data-hook:
 	$(SED) -i -e '/^plugindir/s|/usr/lib|$(libdir)|' \
 	    $(DESTDIR)$(lirc_confdir)/lirc_options.conf
diff --git a/configure.ac b/configure.ac
index 4268e31..b43904c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,7 +43,8 @@ AC_CHECK_PROG([DOXYGEN],[doxygen],[yes],[no])
 AM_CONDITIONAL(HAVE_DOXYGEN, test x$DOXYGEN = xyes)
 LT_INIT([disable-static])
 
-AM_PATH_PYTHON([3.1],,)
+AM_PATH_PYTHON([3.1],,[:])
+AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 PKG_CHECK_MODULES([PYTHON],[python-${PYTHON_VERSION}m],,[true])
 test -z "$PYTHON_LIBS" && \
     PKG_CHECK_MODULES([PYTHON], [python-$PYTHON_VERSION],,[true])
@@ -577,6 +578,7 @@ AX_REPORT_CONDITIONAL([DEVEL])
 AX_REPORT_CONDITIONAL([LINUX_KERNEL])
 AX_REPORT_CONDITIONAL([HAVE_DEVINPUT])
 AX_REPORT_CONDITIONAL([WITH_SYSTEMDSYSTEMUNITDIR])
+AX_REPORT_CONDITIONAL([HAVE_PYTHON])
 AX_REPORT_CONDITIONAL([HAVE_PYTHON35])
 
 echo
