diff -Naur kodi-platform-15edaf7.orig/CMakeLists.txt kodi-platform-15edaf7/CMakeLists.txt
--- kodi-platform-15edaf7.orig/CMakeLists.txt	2015-10-22 17:44:43.034540766 -0700
+++ kodi-platform-15edaf7/CMakeLists.txt	2015-10-22 17:46:38.851326343 -0700
@@ -9,7 +9,6 @@
 find_package(TinyXML REQUIRED)
 find_package(Threads REQUIRED)
 find_package(platform REQUIRED)
-include(UseMultiArch.cmake)
 include(CheckAtomic.cmake)
 
 set(kodiplatform_NAME kodiplatform)
@@ -43,7 +42,7 @@
 set_target_properties(kodiplatform PROPERTIES VERSION ${kodiplatform_VERSION_MAJOR}.${kodiplatform_VERSION_MINOR}.${kodiplatform_VERSION_PATCH}
                                               SOVERSION ${kodiplatform_VERSION_MAJOR}.0)
 
-install(TARGETS kodiplatform DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS kodiplatform DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 install(FILES src/util/XMLUtils.h
         DESTINATION include/kodi/util)
 
@@ -57,14 +56,14 @@
                                  ${CMAKE_INSTALL_PREFIX}/include)
 
   install(FILES ${CMAKE_BINARY_DIR}/kodiplatform.pc
-          DESTINATION ${CMAKE_INSTALL_LIBDIR_NOARCH}/pkgconfig)
+          DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 ENDIF(NOT WIN32)
 
 # config mode
 configure_file (kodiplatform-config.cmake.in
                 kodiplatform-config.cmake @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/kodiplatform-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR_NOARCH}/kodiplatform)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/kodiplatform)
 
 # Quell warnings with in-tree builds
 set(KODI_BUILD_DIR ${KODI_BUILD_DIR})
