diff -Nur rar2fs-1.29.3.orig/Makefile.am rar2fs-1.29.3/Makefile.am
--- rar2fs-1.29.3.orig/Makefile.am	2020-11-08 19:41:41.000000000 +0000
+++ rar2fs-1.29.3/Makefile.am	2021-01-01 11:33:42.609539322 +0000
@@ -1,4 +1,4 @@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src man
+SUBDIRS = src
 
 EXTRA_DIST = get-version.sh rarconfig.example
diff -Nur rar2fs-1.29.3.orig/Makefile.in rar2fs-1.29.3/Makefile.in
--- rar2fs-1.29.3.orig/Makefile.in	2020-12-15 13:06:30.000000000 +0000
+++ rar2fs-1.29.3/Makefile.in	2021-01-01 11:34:20.496533359 +0000
@@ -233,7 +233,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src man
+SUBDIRS = src
 EXTRA_DIST = get-version.sh rarconfig.example
 all: all-recursive
 
