--- a/Makefile
+++ b/Makefile
@@ -146,23 +146,23 @@
    AR ?= @ar
    LD := $(CXX)
    LIBS += -lstdc++ -lpthread -ldl
-   ifeq ($(firstword $(filter x86_64,$(UNAME))),x86_64)
+   ifeq ($(firstword $(filter x86_64,$(ARCH))),x86_64)
       PTR64 = 1
    endif
-   ifeq ($(firstword $(filter amd64,$(UNAME))),amd64)
+   ifeq ($(firstword $(filter amd64,$(ARCH))),amd64)
       PTR64 = 1
    endif
-   ifeq ($(firstword $(filter aarch64,$(UNAME))),aarch64)
+   ifeq ($(firstword $(filter aarch64,$(ARCH))),aarch64)
       PTR64 = 1
       PLATCFLAGS += -DSDLMAME_NO64BITIO -DSDLMAME_ARM -DFORCE_DRC_C_BACKEND
    endif
-   ifeq ($(firstword $(filter ppc64,$(UNAME))),ppc64)
+   ifeq ($(firstword $(filter ppc64,$(ARCH))),ppc64)
       PTR64 = 1
    endif
-   ifneq (,$(findstring ppc,$(UNAME)))
+   ifneq (,$(findstring ppc,$(ARCH)))
       BIGENDIAN=1
    endif
-   ifneq (,$(findstring armv,$(UNAME) $(platform)))
+   ifneq (,$(findstring armv,$(ARCH) $(platform)))
       CCOMFLAGS += -mstructure-size-boundary=32
       PLATCFLAGS += -DSDLMAME_NO64BITIO -DSDLMAME_ARM -DRETRO_SETJMP_HACK -DARM
       LDFLAGS += -Wl,--fix-cortex-a8 -Wl,--no-as-needed
