diff -Naur citra.git/src/citra_libretro/input/mouse_tracker.cpp citra.patch/src/citra_libretro/input/mouse_tracker.cpp
--- citra.git/src/citra_libretro/input/mouse_tracker.cpp	2017-09-18 07:43:04.264522312 -0500
+++ citra.patch/src/citra_libretro/input/mouse_tracker.cpp	2017-09-18 07:43:54.768068280 -0500
@@ -3,6 +3,7 @@
 // Refer to the license.txt file included.
 
 #include <glad/glad.h>
+#include <cmath>
 
 #include "citra_libretro/input/mouse_tracker.h"
 #include "citra_libretro/core_settings.h"
