--- a/countries.c	2016-07-31 10:25:01.000000000 +0200
+++ b/countries.c	2021-01-21 18:02:54.336375432 +0100
@@ -557,7 +557,7 @@ int atsc_is_qam(int atsc) {
 struct cCountry country_list[] = {
 /*- ISO 3166-1 - unique id - long country name                 alpha-3 numeric */
        {"AF", AF, "AFGHANISTAN"},                                  /*AFG, 4  },*/
-       {"AX", AX, "ÅLAND ISLANDS"},                                /*ALA, 248},*/
+       {"AX", AX, "Ã…LAND ISLANDS"},                                /*ALA, 248},*/
        {"AL", AL, "ALBANIA"},                                      /*ALB, 8  },*/
        {"DZ", DZ, "ALGERIA"},                                      /*DZA, 12 },*/
        {"AS", AS, "AMERICAN SAMOA"},                               /*ASM, 16 },*/
@@ -610,10 +610,10 @@ struct cCountry country_list[] = {
        {"CD", CD, "CONGO, THE DEMOCRATIC REPUBLIC OF THE"},        /*COD, 180},*/
        {"CK", CK, "COOK ISLANDS"},                                 /*COK, 184},*/
        {"CR", CR, "COSTA RICA"},                                   /*CRI, 188},*/
-       {"CI", CI, "CÔTE D'IVOIRE"},                                /*CIV, 384},*/
+       {"CI", CI, "CÃ”TE D'IVOIRE"},                                /*CIV, 384},*/
        {"HR", HR, "CROATIA"},                                      /*HRV, 191},*/
        {"CU", CU, "CUBA"},                                         /*CUB, 192},*/
-       {"CW", CW, "CURAÇAO"},                                      /*CUW, 531},*/
+       {"CW", CW, "CURAÃ‡AO"},                                      /*CUW, 531},*/
        {"CY", CY, "CYPRUS"},                                       /*CYP, 196},*/
        {"CZ", CZ, "CZECH REPUBLIC"},                               /*CZE, 203},*/
        {"DK", DK, "DENMARK"},                                      /*DNK, 208},*/
@@ -736,11 +736,11 @@ struct cCountry country_list[] = {
        {"PT", PT, "PORTUGAL"},                                     /*PRT, 620},*/
        {"PR", PR, "PUERTO RICO"},                                  /*PRI, 630},*/
        {"QA", QA, "QATA"},                                         /*QAT, 634},*/
-       {"RE", RE, "RÉUNION"},                                      /*REU, 638},*/
+       {"RE", RE, "RÃ‰UNION"},                                      /*REU, 638},*/
        {"RO", RO, "ROMANIA"},                                      /*ROU, 642},*/
        {"RU", RU, "RUSSIAN FEDERATION"},                           /*RUS, 643},*/
        {"RW", RW, "RWANDA"},                                       /*RWA, 646},*/
-       {"BL", BL, "SAINT BARTHÉLEMY"},                             /*BLM, 652},*/
+       {"BL", BL, "SAINT BARTHÃ‰LEMY"},                             /*BLM, 652},*/
        {"SH", SH, "SAINT HELENA"},                                 /*SHN, 654},*/
        {"KN", KN, "SAINT KITTS AND NEVIS"},                        /*KNA, 659},*/
        {"LC", LC, "SAINT LUCIA"},                                  /*LCA, 662},*/
