From 60a9d1a6e1f0ea6ab74f04fb515e18b65e3f0f9e Mon Sep 17 00:00:00 2001
From: mglae <mglmail@arcor.de>
Date: Fri, 9 Nov 2018 19:13:36 +0100
Subject: [PATCH] interface channel count

---
 wirbelscan.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/wirbelscan.cpp b/wirbelscan.cpp
index f2dda24..9ee66c2 100644
--- a/wirbelscan.cpp
+++ b/wirbelscan.cpp
@@ -20,6 +20,7 @@ class cScanner;
 extern int channelcount;     // menusetup.c
 extern int nextTransponders; // scanfilter.c
 extern cScanner* Scanner;
+extern TChannels NewChannels;
 
 const char* WIRBELSCAN_VERSION        = "2024.09.15"; /* YYYY.MM.DD */
 const char* WIRBELSCAN_DESCRIPTION    = "DVB channel scan for VDR";
@@ -221,8 +222,8 @@ bool cPluginWirbelscan::Service(const char* id, void* Data) {
         strcpy(s->transponder, lTransponder.length()? lTransponder.c_str():"none");
         s->progress = s->status == StatusScanning?lProgress:0;
         s->strength = s->status == StatusScanning?lStrength:0;
-        s->numChannels = 0;              // Channels.Count(); // not possible any longer.
-        s->newChannels = channelcount;   // ((Channels.Count() - channelcount) > 0) && channelcount?Channels.Count() - channelcount:0;
+        s->numChannels = cChannels::MaxNumber();
+        s->newChannels = (NewChannels.Count() > cChannels::MaxNumber()) ? NewChannels.Count() - cChannels::MaxNumber():0;
         s->nextTransponders = nextTransponders;
         return true;
         }
-- 
2.51.0

