--- a/wirbelscan.c
+++ b/wirbelscan.c
@@ -13,6 +13,8 @@
 #include "countries.h"
 #include "satellites.h"
 
+extern TChannels NewChannels;
+
 static const char *VERSION        = "2017.06.04";
 static const char *DESCRIPTION    = "DVB channel scan for VDR";
 static const char *MAINMENUENTRY  = NULL; /* main menu -> use wirbelscancontrol plugin */
@@ -201,8 +203,8 @@ bool cPluginWirbelscan::Service(const ch
         strcpy(s->transponder, lTransponder.length()? lTransponder.c_str():"none");
         s->progress = s->status == StatusScanning?lProgress:0;
         s->strength = s->status == StatusScanning?lStrength:0;
-        s->numChannels = 0;              // Channels.Count(); // not possible any longer.
-        s->newChannels = channelcount;   // ((Channels.Count() - channelcount) > 0) && channelcount?Channels.Count() - channelcount:0;
+        s->numChannels = cChannels::MaxNumber();
+        s->newChannels = (NewChannels.Count() > cChannels::MaxNumber()) ? NewChannels.Count() - cChannels::MaxNumber():0;
         s->nextTransponders = nextTransponders;
         return true;
         }
