--- a/common.c
+++ b/common.c
@@ -46,20 +46,10 @@ void cMySetup::InitSystems(void) {
   for(int i = 0; i < cDevice::NumDevices(); i++) {
      cDevice* device = cDevice::GetDevice(i);
      if (device == NULL) continue;
-     std::string t = *device->DeviceType();
-     if (t == "DVB-C" ) systems[SCAN_CABLE         ] = 1;
-     if (t == "DVB-T" ) systems[SCAN_TERRESTRIAL   ] = 1;
-     if (t == "DSS"   ) systems[SCAN_SATELLITE     ] = 1;
-     if (t == "DVB-S" ) systems[SCAN_SATELLITE     ] = 1;
-     if (t == "DVB-S2") systems[SCAN_SATELLITE     ] = 1;
-     if (t == "ATSC"  ) systems[SCAN_TERRCABLE_ATSC] = 1;
-     if (t == "DVB-T2") systems[SCAN_TERRESTRIAL   ] = 1;
-     if (t == "TURBO" ) systems[SCAN_SATELLITE     ] = 1;
-     if (t == "SAT>IP") {
-        if (device->ProvidesSource(cSource::FromString("S"))) systems[SCAN_SATELLITE     ] = 1;
-        if (device->ProvidesSource(cSource::FromString("T"))) systems[SCAN_TERRESTRIAL   ] = 1;
-        if (device->ProvidesSource(cSource::FromString("C"))) systems[SCAN_CABLE         ] = 1;
-        }
+     if (device->ProvidesSource(cSource::stSat))   systems[SCAN_SATELLITE] = 1;
+     if (device->ProvidesSource(cSource::stTerr))  systems[SCAN_TERRESTRIAL] = 1;
+     if (device->ProvidesSource(cSource::stCable)) systems[SCAN_CABLE] = 1;
+     if (device->ProvidesSource(cSource::stAtsc))  systems[SCAN_TERRCABLE_ATSC] = 1;
      }
 
   if (DVB_Type >= SCAN_NO_DEVICE || ! systems[DVB_Type]) {
