--- a/statemachine.c
+++ b/statemachine.c
@@ -309,7 +309,8 @@ void cStateMachine::Action(void) {
                  else
                     newState = eAddChannels;
               if (time(0) != tm) {
-                 MenuScanning->SetProgress(lProgress);
+                 if (MenuScanning)
+                    MenuScanning->SetProgress(lProgress);
                  tm = time(0);
                  }
               }
