diff -Naur xbmc-pvr-addons-master-1db308b/addons/Makefile.include.am xbmc-pvr-addons-master-1db308b.patch/addons/Makefile.include.am
--- xbmc-pvr-addons-master-1db308b/addons/Makefile.include.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/Makefile.include.am	2014-06-15 03:44:26.914810581 +0200
@@ -36,7 +36,9 @@
 	mkdir -m 755 -p $(DESTDIR)@LIBDIR@/$(ADDONNAME)
 	mkdir -m 755 -p $(DESTDIR)@DATADIR@/$(ADDONNAME)
 	cp -f @BINPREFIX@$(ADDONBINNAME)@BIN_EXT@ $(DESTDIR)@LIBDIR@/$(ADDONNAME) ; chmod 655 $(DESTDIR)@LIBDIR@/$(ADDONNAME)/@BINPREFIX@$(ADDONBINNAME)@BIN_EXT@
-	cp -r -f @abs_top_srcdir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME) ; chmod -R o+rx $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	cp -r -f @abs_top_srcdir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	cp -r -f @abs_top_builddir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	chmod -R o+rx $(DESTDIR)@DATADIR@/$(ADDONNAME)
 	rm -f $(DESTDIR)@DATADIR@/$(ADDONNAME)/addon.xml.in
 endif
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.argustv/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.argustv/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.argustv/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.argustv/Makefile.am	2014-06-15 03:41:47.992491868 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libargustv-addon
 lib_LTLIBRARIES = libargustv-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/jsoncpp/libjsoncpp.la
+LIBS            = @abs_top_builddir@/lib/jsoncpp/libjsoncpp.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc -Isrc/lib/filesystem -I@abs_top_srcdir@/lib/jsoncpp/include
+INCLUDES+=-I$(srcdir)/src -I$(srcdir)/src/lib/filesystem -I@abs_top_srcdir@/lib/jsoncpp/include
 
 libargustv_addon_la_SOURCES = src/activerecording.cpp \
                                    src/channel.cpp \
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.demo/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.demo/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.demo/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.demo/Makefile.am	2014-06-15 03:41:47.993491837 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libpvrdemo-addon
 lib_LTLIBRARIES = libpvrdemo-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.dvblink/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.dvblink/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.dvblink/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.dvblink/Makefile.am	2014-06-15 03:41:47.994491808 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libdvblink-addon
 lib_LTLIBRARIES = libdvblink-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/libdvblinkremote/libdvblinkremote.la
+LIBS            = @abs_top_builddir@/lib/libdvblinkremote/libdvblinkremote.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc
+INCLUDES+=-I$(srcdir)/src
 
 libdvblink_addon_la_SOURCES = src/client.cpp \
                               src/base64.cpp \
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.dvbviewer/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.dvbviewer/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.dvbviewer/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.dvbviewer/Makefile.am	2014-06-15 03:41:47.994491808 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libdvbviewer-addon
 lib_LTLIBRARIES = libdvbviewer-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.hts/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.hts/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.hts/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.hts/Makefile.am	2014-06-15 03:41:47.995491780 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libtvheadend-addon
 lib_LTLIBRARIES = libtvheadend-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/libhts/libhts.la -ldl
+LIBS            = @abs_top_builddir@/lib/libhts/libhts.la -ldl
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.mediaportal.tvserver/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.mediaportal.tvserver/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.mediaportal.tvserver/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.mediaportal.tvserver/Makefile.am	2014-06-15 03:41:47.995491780 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libmediaportal-addon
 lib_LTLIBRARIES = libmediaportal-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc
+INCLUDES+=-I$(srcdir)/src
 
 libmediaportal_addon_la_SOURCES = src/Cards.cpp \
                                   src/channels.cpp \
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.mythtv.cmyth/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.mythtv.cmyth/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.mythtv.cmyth/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.mythtv.cmyth/Makefile.am	2014-06-15 03:41:47.996491755 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libmythtvcmyth-addon
 lib_LTLIBRARIES = libmythtvcmyth-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/cmyth/libcmyth.la
+LIBS            = @abs_top_builddir@/lib/cmyth/libcmyth.la
 
 AM_CPPFLAGS = -I$(abs_top_srcdir)/lib/cmyth/include
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.nextpvr/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.nextpvr/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.nextpvr/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.nextpvr/Makefile.am	2014-06-15 03:41:47.997491730 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libnextpvr-addon
 lib_LTLIBRARIES = libnextpvr-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc
+INCLUDES+=-I$(srcdir)/src
 
 libnextpvr_addon_la_SOURCES = src/client.cpp \
                                   src/pvrclient-nextpvr.cpp \
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.njoy/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.njoy/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.njoy/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.njoy/Makefile.am	2014-06-15 03:41:47.997491730 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libpvrnjoy-addon
 lib_LTLIBRARIES = libpvrnjoy-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.vuplus/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.vuplus/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.vuplus/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.vuplus/Makefile.am	2014-06-15 03:41:47.998491708 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libvuplus-addon
 lib_LTLIBRARIES = libvuplus-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/addons/pvr.wmc/Makefile.am xbmc-pvr-addons-master-1db308b.patch/addons/pvr.wmc/Makefile.am
--- xbmc-pvr-addons-master-1db308b/addons/pvr.wmc/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/addons/pvr.wmc/Makefile.am	2014-06-15 03:41:47.998491708 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libpvrwmc-addon
 lib_LTLIBRARIES = libpvrwmc-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-master-1db308b/lib/cmyth/libcmyth/Makefile.am xbmc-pvr-addons-master-1db308b.patch/lib/cmyth/libcmyth/Makefile.am
--- xbmc-pvr-addons-master-1db308b/lib/cmyth/libcmyth/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/lib/cmyth/libcmyth/Makefile.am	2014-06-15 03:41:47.999491687 +0200
@@ -26,7 +26,7 @@
                      storagegroup.c \
                      epginfo.c
 
-INCLUDES=-I../include/ $(MYSQL_INCLUDES)
+INCLUDES=-I$(srcdir)/../include/ $(MYSQL_INCLUDES)
 
 $(LIB): libcmyth.la
 	cp -f .libs/libcmyth.a .
diff -Naur xbmc-pvr-addons-master-1db308b/lib/cmyth/librefmem/Makefile.am xbmc-pvr-addons-master-1db308b.patch/lib/cmyth/librefmem/Makefile.am
--- xbmc-pvr-addons-master-1db308b/lib/cmyth/librefmem/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/lib/cmyth/librefmem/Makefile.am	2014-06-15 03:41:47.999491687 +0200
@@ -3,7 +3,7 @@
 librefmem_la_SOURCES = alloc.c \
                       debug_refmem.c
 
-INCLUDES=-I../include/ -I../libcmyth/
+INCLUDES=-I$(srcdir)/../include/ -I$(srcdir)/../libcmyth/
 
 $(LIB): librefmem.la
 	cp -f .libs/librefmem.a .
diff -Naur xbmc-pvr-addons-master-1db308b/lib/jsoncpp/Makefile.am xbmc-pvr-addons-master-1db308b.patch/lib/jsoncpp/Makefile.am
--- xbmc-pvr-addons-master-1db308b/lib/jsoncpp/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/lib/jsoncpp/Makefile.am	2014-06-15 03:41:48.000491668 +0200
@@ -4,7 +4,7 @@
                         src/lib_json/json_value.cpp \
                         src/lib_json/json_writer.cpp
 
-INCLUDES=-Iinclude/
+INCLUDES=-I$(srcdir)/include/
 
 $(LIB): libjsoncpp.la
 	cp -f .libs/libjsoncpp.a .
diff -Naur xbmc-pvr-addons-master-1db308b/lib/libdvblinkremote/Makefile.am xbmc-pvr-addons-master-1db308b.patch/lib/libdvblinkremote/Makefile.am
--- xbmc-pvr-addons-master-1db308b/lib/libdvblinkremote/Makefile.am	2014-06-15 02:39:46.000000000 +0200
+++ xbmc-pvr-addons-master-1db308b.patch/lib/libdvblinkremote/Makefile.am	2014-06-15 03:41:48.000491668 +0200
@@ -26,10 +26,10 @@
 			util.cpp \
 			xml_object_serializer_factory.cpp
 
-LIBS= @abs_top_srcdir@/lib/tinyxml2/libtinyxml2.la
+LIBS= @abs_top_builddir@/lib/tinyxml2/libtinyxml2.la
  
-INCLUDES=-I..
+INCLUDES=-I$(srcdir)/..
 
 $(LIB): libdvblinkremote.la
 	cp -f .libs/libdvblinkremote.a .
-	cp -f .libs/libdvblinkremote.la $(LIB)
\ Kein Zeilenumbruch am Dateiende.
+	cp -f .libs/libdvblinkremote.la $(LIB)
