diff --git a/nss/cmd/shlibsign/Makefile b/nss/cmd/shlibsign/Makefile
index 83f90fa..177feee 100644
--- a/nss/cmd/shlibsign/Makefile
+++ b/nss/cmd/shlibsign/Makefile
@@ -95,5 +95,3 @@ else
     endif
 endif
 
-libs install :: $(CHECKLOC)
-
diff --git a/nss/coreconf/Linux.mk b/nss/coreconf/Linux.mk
index a586e5e..e815b72 100644
--- a/nss/coreconf/Linux.mk
+++ b/nss/coreconf/Linux.mk
@@ -16,12 +16,6 @@ ifeq ($(USE_PTHREADS),1)
 	IMPL_STRATEGY = _PTH
 endif
 
-CC			= gcc
-CCC			= g++
-RANLIB			= ranlib
-
-DEFAULT_COMPILER = gcc
-
 ifeq ($(OS_TARGET),Android)
 ifndef ANDROID_NDK
 	$(error Must set ANDROID_NDK to the path to the android NDK first)
