From 59bd0267cc8a581ff6f3388756a5524159a75b3b Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 13 Dec 2014 14:36:57 +0200
Subject: [PATCH 5/5] link against ncurses

---
 Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 41efc1f..d1d3ddc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -151,7 +151,7 @@
 			client/monitor.h client/monitor.c \
 			client/commands.c client/main.c
 
-client_connmanctl_LDADD = @DBUS_LIBS@ @GLIB_LIBS@ -lreadline -ldl
+client_connmanctl_LDADD = @DBUS_LIBS@ @GLIB_LIBS@ -lreadline -lncurses -ltermcap -ldl
 endif
 
 if WISPR

-- 
1.7.10.4

