From b7923c77913d8f8047b694a03494b2bb092337e1 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 13 Dec 2014 14:32:41 +0200
Subject: [PATCH 1/5] do not cleanup routes

---
 src/device.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/src/device.c b/src/device.c
index c0683ab..8ecfc86 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1445,8 +1445,6 @@ int __connman_device_init(const char *device, const char *nodevice)
 	if (nodevice)
 		nodevice_filter = g_strsplit(nodevice, ",", -1);
 
-	cleanup_devices();
-
 	return 0;
 }
 
-- 
1.7.10.4

