commit 707a0d73d7231b1821072712a7771c7aef140e21
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Tue Jul 23 11:28:10 2013 +0300

    ipv6 disabled by default

diff --git a/src/ipconfig.c b/src/ipconfig.c
index fbeff8f..3eb61c4 100644
--- a/src/ipconfig.c
+++ b/src/ipconfig.c
@@ -1123,10 +1123,7 @@ static struct connman_ipconfig *create_ipv6config(int index)
 	ipv6config->enabled = false;
 	ipv6config->type = CONNMAN_IPCONFIG_TYPE_IPV6;
 
-	if (!is_ipv6_supported)
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
-	else
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_AUTO;
+	ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
 
 	ipdevice = g_hash_table_lookup(ipdevice_hash, GINT_TO_POINTER(index));
 	if (ipdevice != NULL)
