From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Subject: [PATCH v2 1/3] conf: Add description of StartTetheringAutomatically
Date: Tue, 23 Apr 2013 15:17:43 +0300

---
 src/main.conf | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/main.conf b/src/main.conf
index a0608d4..7e509f7 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -85,3 +85,8 @@
 # This is a dangerous setting so the default value is false.
 # Do not activate this unless you really know what you are doing.
 # AllowEthernetTethering = false
+
+# Restore earlier tethering status when returning from offline mode,
+# re-enabling a technology, and after restarts and reboots.
+# Default value is false.
+# PersistentTetheringMode = false
-- 
1.7.11.7
