From 59bd0267cc8a581ff6f3388756a5524159a75b3b Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 13 Dec 2014 14:36:57 +0200
Subject: [PATCH 5/5] link against ncurses

---
 Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 41efc1f..d1d3ddc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -252,7 +252,7 @@ client_connmanctl_SOURCES = client/dbus_helpers.h client/dbus_helpers.c \
 			client/main.c
 
 client_connmanctl_LDADD = gdbus/libgdbus-internal.la @DBUS_LIBS@ @GLIB_LIBS@ \
-				-lreadline -ldl
+				-lreadline -lncurses -ldl
 endif
 
 noinst_PROGRAMS += unit/test-pbkdf2-sha1 unit/test-prf-sha1 unit/test-ippool
-- 
1.7.10.4

