From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Subject: [PATCH v2 2/3] main: Add support for StartTetheringAutomatically configuration variable
Date: Tue, 23 Apr 2013 15:17:44 +0300

---
 src/main.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/main.c b/src/main.c
index 016e52d..5e738cf 100644
--- a/src/main.c
+++ b/src/main.c
@@ -71,6 +71,7 @@ static struct {
 	connman_bool_t allow_hostname_updates;
 	connman_bool_t single_tech;
 	connman_bool_t allow_ethernet_tethering;
+	connman_bool_t persistent_tethering_mode;
 } connman_settings  = {
 	.bg_scan = TRUE,
 	.pref_timeservers = NULL,
@@ -83,6 +84,7 @@ static struct {
 	.allow_hostname_updates = TRUE,
 	.single_tech = FALSE,
 	.allow_ethernet_tethering = FALSE,
+	.persistent_tethering_mode = FALSE,
 };
 
 #define CONF_BG_SCAN                    "BackgroundScanning"
@@ -96,6 +98,7 @@ static struct {
 #define CONF_ALLOW_HOSTNAME_UPDATES     "AllowHostnameUpdates"
 #define CONF_SINGLE_TECH                "SingleConnectedTechnology"
 #define CONF_ALLOW_ETHERNET_TETHERING   "AllowEthernetTethering"
+#define CONF_PERSISTENT_TETHERING_MODE  "PersistentTetheringMode"
 
 static const char *supported_options[] = {
 	CONF_BG_SCAN,
@@ -109,6 +112,7 @@ static const char *supported_options[] = {
 	CONF_ALLOW_HOSTNAME_UPDATES,
 	CONF_SINGLE_TECH,
 	CONF_ALLOW_ETHERNET_TETHERING,
+	CONF_PERSISTENT_TETHERING_MODE,
 	NULL
 };
 
@@ -339,6 +343,14 @@ static void parse_config(GKeyFile *config)
 		connman_settings.allow_ethernet_tethering = boolean;
 
 	g_clear_error(&error);
+
+	boolean = g_key_file_get_boolean(config, "General",
+					CONF_PERSISTENT_TETHERING_MODE,
+					&error);
+	if (error == NULL)
+		connman_settings.persistent_tethering_mode = boolean;
+
+	g_clear_error(&error);
 }
 
 static int config_init(const char *file)
@@ -513,6 +525,9 @@ connman_bool_t connman_setting_get_bool(const char *key)
 	if (g_str_equal(key, CONF_ALLOW_ETHERNET_TETHERING) == TRUE)
 		return connman_settings.allow_ethernet_tethering;
 
+	if (g_str_equal(key, CONF_PERSISTENT_TETHERING_MODE) == TRUE)
+		return connman_settings.persistent_tethering_mode;
+
 	return FALSE;
 }
 
-- 1.7.11.7 

