diff -Naur connman-1.25/Makefile.am connman-1.25.patch/Makefile.am
--- connman-1.25/Makefile.am	2014-08-18 14:20:06.000000000 +0200
+++ connman-1.25.patch/Makefile.am	2014-09-24 19:25:39.972371273 +0200
@@ -250,7 +250,7 @@
 			client/main.c
 
 client_connmanctl_LDADD = gdbus/libgdbus-internal.la @DBUS_LIBS@ @GLIB_LIBS@ \
-				-lreadline -ldl
+				-lreadline -lncurses -ldl
 endif
 
 noinst_PROGRAMS += unit/test-pbkdf2-sha1 unit/test-prf-sha1 unit/test-ippool
