diff -Naur connman-24180d1243bf005c721be6f9d8d6da67bb669c55/src/connman-dbus.conf connman-24180d1243bf005c721be6f9d8d6da67bb669c55-2/src/connman-dbus.conf
--- connman-24180d1243bf005c721be6f9d8d6da67bb669c55/src/connman-dbus.conf	2024-02-02 04:49:53.238565581 +0100
+++ connman-24180d1243bf005c721be6f9d8d6da67bb669c55-2/src/connman-dbus.conf	2024-02-02 04:52:06.209049700 +0100
@@ -1,6 +1,14 @@
 <!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
 <busconfig>
+    <policy user="%DISTRO%">
+        <allow own="net.connman"/>
+        <allow send_destination="net.connman"/>
+        <allow send_interface="net.connman.Agent"/>
+        <allow send_interface="net.connman.Counter"/>
+        <allow send_interface="net.connman.Notification"/>
+    </policy>
+
     <policy user="root">
         <allow own="net.connman"/>
         <allow send_destination="net.connman"/>
