From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Subject: [PATCH 2/6] conf: Add description of AllowEthernetTethering
Date: Tue, 23 Apr 2013 15:17:37 +0300

---
 src/main.conf | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/main.conf b/src/main.conf
index 2985613..a0608d4 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -76,3 +76,12 @@
 # setting enabled applications will notice more network breaks than
 # normal. Default value is false.
 # SingleConnectedTechnology = false
+
+# Allow ethernet technology devices to be used when tethering.
+# If this is set to true and if ethernet tethering is activated,
+# a DHCP server is started to ethernet interface. The ethernet tethered
+# device should never be connected to corporate and home networks as
+# the device might confuse other network devices and services.
+# This is a dangerous setting so the default value is false.
+# Do not activate this unless you really know what you are doing.
+# AllowEthernetTethering = false
-- 1.7.11.7

