diff -Naur a/src/firewall.c b/src/firewall.c
--- a/src/firewall.c	2015-08-31 08:50:11.000000000 -0700
+++ b/src/firewall.c	2016-04-12 13:34:31.048946661 -0700
@@ -19,6 +19,10 @@
  *
  */
 
+#warning "Temporary work-around for broken glibc vs. linux kernel header definitions"
+#warning "This really should be removed sooner rather than later, when this is fixed upstream"
+#define _NET_IF_H 1
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -26,6 +30,15 @@
 #include <errno.h>
 
 #include <xtables.h>
+
+#include <net/if.h>
+#include <linux/if.h>
+#ifndef IFNAMSIZ
+#undef _NET_IF_H
+/* Let's make sure to include this one, too, if IFNAMSIZ isn't defined yet, as it is for kernels <= 4.2 */
+#include <net/if.h>
+#endif
+
 #include <linux/netfilter_ipv4/ip_tables.h>
 
 #include "connman.h"
diff -Naur a/src/iptables.c b/src/iptables.c
--- a/src/iptables.c	2016-04-12 13:33:35.630483664 -0700
+++ b/src/iptables.c	2016-04-12 13:34:39.216014906 -0700
@@ -19,6 +19,10 @@
  *
  */
 
+#warning "Temporary work-around for broken glibc vs. linux kernel header definitions"
+#warning "This really should be removed sooner rather than later, when this is fixed upstream"
+#define _NET_IF_H 1
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -33,6 +37,14 @@
 #include <xtables.h>
 #include <inttypes.h>
 
+#include <net/if.h>
+#include <linux/if.h>
+#ifndef IFNAMSIZ
+#undef _NET_IF_H
+/* Let's make sure to include this one, too, if IFNAMSIZ isn't defined yet, as it is for kernels <= 4.2 */
+#include <net/if.h>
+#endif
+
 #include <linux/netfilter_ipv4/ip_tables.h>
 
 #include "connman.h"
