diff --git a/vpn/vpn-provider.c b/vpn/vpn-provider.c
index 0336636..40450eb 100644
--- a/vpn/vpn-provider.c
+++ b/vpn/vpn-provider.c
@@ -1869,9 +1869,14 @@ static const char *get_string(GHashTable *settings, const char *key)
 static GSList *parse_user_networks(const char *network_str)
 {
 	GSList *networks = NULL;
-	char **elems = g_strsplit(network_str, ",", 0);
+	char **elems;
 	int i = 0;
 
+	if (network_str == NULL)
+		return NULL;
+
+	elems = g_strsplit(network_str, ",", 0);
+
 	if (elems == NULL)
 		return NULL;
 
