From 6bfcac5ebe1c29c78176d9560268f256512a0436 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 13 Dec 2014 14:36:08 +0200
Subject: [PATCH 4/5] ipv6 disabled by default

---
 src/ipconfig.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/ipconfig.c b/src/ipconfig.c
index 3d0c387..e45480e 100644
--- a/src/ipconfig.c
+++ b/src/ipconfig.c
@@ -1149,10 +1149,7 @@ static struct connman_ipconfig *create_ipv6config(int index)
 	ipv6config->enabled = false;
 	ipv6config->type = CONNMAN_IPCONFIG_TYPE_IPV6;
 
-	if (!is_ipv6_supported)
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
-	else
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_AUTO;
+	ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
 
 	ipdevice = g_hash_table_lookup(ipdevice_hash, GINT_TO_POINTER(index));
 	if (ipdevice)
-- 
1.7.10.4

