From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Subject: [PATCH 3/6] main: Add support for AllowEthernetTethering configuration variable
Date: Tue, 23 Apr 2013 15:17:38 +0300

---

rc/main.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/main.c b/src/main.c
index a76ec3b..016e52d 100644
--- a/src/main.c
+++ b/src/main.c
@@ -70,6 +70,7 @@ static struct {
 	char **blacklisted_interfaces;
 	connman_bool_t allow_hostname_updates;
 	connman_bool_t single_tech;
+	connman_bool_t allow_ethernet_tethering;
 } connman_settings  = {
 	.bg_scan = TRUE,
 	.pref_timeservers = NULL,
@@ -81,6 +82,7 @@ static struct {
 	.blacklisted_interfaces = NULL,
 	.allow_hostname_updates = TRUE,
 	.single_tech = FALSE,
+	.allow_ethernet_tethering = FALSE,
 };
 
 #define CONF_BG_SCAN                    "BackgroundScanning"
@@ -93,6 +95,7 @@ static struct {
 #define CONF_BLACKLISTED_INTERFACES     "NetworkInterfaceBlacklist"
 #define CONF_ALLOW_HOSTNAME_UPDATES     "AllowHostnameUpdates"
 #define CONF_SINGLE_TECH                "SingleConnectedTechnology"
+#define CONF_ALLOW_ETHERNET_TETHERING   "AllowEthernetTethering"
 
 static const char *supported_options[] = {
 	CONF_BG_SCAN,
@@ -105,6 +108,7 @@ static const char *supported_options[] = {
 	CONF_BLACKLISTED_INTERFACES,
 	CONF_ALLOW_HOSTNAME_UPDATES,
 	CONF_SINGLE_TECH,
+	CONF_ALLOW_ETHERNET_TETHERING,
 	NULL
 };
 
@@ -327,6 +331,14 @@ static void parse_config(GKeyFile *config)
 		connman_settings.single_tech = boolean;
 
 	g_clear_error(&error);
+
+	boolean = g_key_file_get_boolean(config, "General",
+					CONF_ALLOW_ETHERNET_TETHERING,
+					&error);
+	if (error == NULL)
+		connman_settings.allow_ethernet_tethering = boolean;
+
+	g_clear_error(&error);
 }
 
 static int config_init(const char *file)
@@ -498,6 +510,9 @@ connman_bool_t connman_setting_get_bool(const char *key)
 	if (g_str_equal(key, CONF_SINGLE_TECH) == TRUE)
 		return connman_settings.single_tech;
 
+	if (g_str_equal(key, CONF_ALLOW_ETHERNET_TETHERING) == TRUE)
+		return connman_settings.allow_ethernet_tethering;
+
 	return FALSE;
 }
 
-- 1.7.11.7 

