From 976a1c2f3d6c2c2d4b0ef8a43a3ef128936b0dd6 Mon Sep 17 00:00:00 2001
From: Gwenole Beauchesne <gwenole.beauchesne@intel.com>
Date: Wed, 18 Jun 2014 13:11:48 +0200
Subject: [PATCH] decoder: h264: fix RefPicList0/1 without frame in DPB.

---
 src/i965_decoder_utils.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/i965_decoder_utils.c b/src/i965_decoder_utils.c
index 0539e08..d79b2b3 100644
--- a/src/i965_decoder_utils.c
+++ b/src/i965_decoder_utils.c
@@ -526,6 +526,8 @@ intel_update_avc_frame_store_index(
             continue;
 
         GenAvcSurface * const avc_surface = obj_surface->private_data;
+        if (!avc_surface)
+            continue;
         if (avc_surface->frame_store_id >= 0) {
             GenFrameStore * const fs =
                 &frame_store[avc_surface->frame_store_id];
@@ -559,6 +561,8 @@ intel_update_avc_frame_store_index(
             continue;
 
         GenAvcSurface * const avc_surface = obj_surface->private_data;
+        if (!avc_surface)
+            continue;
         if (n < num_free_refs) {
             GenFrameStore * const fs = free_refs[n++];
             fs->surface_id = obj_surface->base.id;
-- 
1.7.9.5

