diff --git a/drivers/media/dvb-frontends/m88ds3103.c b/drivers/media/dvb-frontends/m88ds3103.c
index 315809d..54dff7c 100644
--- a/drivers/media/dvb-frontends/m88ds3103.c
+++ b/drivers/media/dvb-frontends/m88ds3103.c
@@ -1244,6 +1244,8 @@ static int m88ds3103_set_frontend(struct dvb_frontend *fe)
 	dprintk("symbol rate = %d\n", c->symbol_rate);
 	dprintk("delivery system = %d\n", c->delivery_system);
 	
+	state->delivery_system = c->delivery_system;
+	
 	realFreq = c->frequency;
 	lpf_offset_KHz = 0;
 	if(c->symbol_rate < 5000000){
@@ -1501,7 +1503,7 @@ static int m88ds3103_set_frontend(struct dvb_frontend *fe)
                 }
 		msleep(20);
 	}
-	
+/*	
 	if((status & FE_HAS_LOCK) == 0){
 		state->delivery_system = (state->delivery_system == SYS_DVBS) ? SYS_DVBS2 : SYS_DVBS;
 		m88ds3103_demod_connect(fe, offset_khz);
@@ -1514,7 +1516,7 @@ static int m88ds3103_set_frontend(struct dvb_frontend *fe)
 			msleep(20);
 		}
 	}
-	
+*/	
 	if (status & FE_HAS_LOCK){
 		if(state->config->ci_mode == 2)
 			m88ds3103_set_clock_ratio(state);
