diff -Naur pmtools/madt/Makefile pmtools.patch/madt/Makefile
--- pmtools/madt/Makefile	2005-11-11 18:07:04.000000000 +0100
+++ pmtools.patch/madt/Makefile	2007-11-16 19:48:56.000000000 +0100
@@ -1,6 +1,16 @@
+PROG= madt
+SRCS=	madt.c
+
+all: madt
+$(PROG) : $(SRCS)
+	$(CC) $(CFLAGS) $(SRCS) -o $(PROG)
+
 test: madt
-	./madt < APIC
-madt: madt.c acpi.h tables.c
-	cc -o madt madt.c
-clean:
-	rm madt
+	./madt < example.APIC.dat
+	./madt < example.APIC.bad.dat
+
+CLEANFILES= $(PROG)
+
+clean : 
+	rm -f $(CLEANFILES) $(patsubst %.c,%.o, $(SRCS)) *~
+
