diff -Naur vdr-plugin-epgfixer-1ad4aaf/Makefile vdr-plugin-epgfixer-1ad4aaf.patch/Makefile
--- vdr-plugin-epgfixer-1ad4aaf/Makefile	2015-01-01 11:15:50.000000000 +0100
+++ vdr-plugin-epgfixer-1ad4aaf.patch/Makefile	2016-01-07 04:40:37.742655989 +0100
@@ -70,8 +70,8 @@
 OBJS = $(PLUGIN).o blacklist.o charset.o config.o epgclone.o epghandler.o regexp.o setup_menu.o tools.o
 
 ifeq ($(REGEXLIB), pcre)
-LIBS += $(shell pcre-config --libs-posix)
-INCLUDES += $(shell pcre-config --cflags)
+LIBS += $(shell pkg-config --libs libpcreposix libpcre)
+INCLUDES += $(shell pkg-config --cflags libpcre)
 DEFINES += -DHAVE_PCREPOSIX
 endif
 
