diff -Naur imx-lib-3.5.7-1.0.0/ipu/Makefile imx-lib-3.5.7-1.0.0.patch/ipu/Makefile
--- imx-lib-3.5.7-1.0.0/ipu/Makefile	2013-06-21 15:22:18.000000000 +0200
+++ imx-lib-3.5.7-1.0.0.patch/ipu/Makefile	2014-01-13 08:07:33.322840816 +0100
@@ -1,6 +1,8 @@
 CC=$(CROSS_COMPILE)gcc
 AR=$(CROSS_COMPILE)ar
 
+CFLAGS := -Wall -O2
+
 ifeq ($(PLATFORM), IMX6Q)
 OBJS = mxc_ipu_hl_lib_dummy.o
 else
@@ -37,7 +39,7 @@
 ifeq "$(PLATFORM)" ""
 	$(error "Unspecified PLATFORM variable")
 endif
-	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -fPIC -c $^ -o $@
+	$(CC) -D$(PLATFORM) $(INCLUDE) $(CFLAGS) -fPIC -c $^ -o $@
 
 .PHONY: clean
 clean:
