CTL_PROC was removed in linux 2.6.18

diff -Nur strace-4.5.12.orig/system.c strace-4.5.12/system.c
--- strace-4.5.12.orig/system.c	Wed Jun  1 22:22:06 2005
+++ strace-4.5.12/system.c	Sat Sep 23 11:18:04 2006
@@ -1572,7 +1572,6 @@
 	{ CTL_KERN, "CTL_KERN" },
 	{ CTL_VM, "CTL_VM" },
 	{ CTL_NET, "CTL_NET" },
-	{ CTL_PROC, "CTL_PROC" },
 	{ CTL_FS, "CTL_FS" },
 	{ CTL_DEBUG, "CTL_DEBUG" },
 	{ CTL_DEV, "CTL_DEV" },
