diff -Naur strace-4.7-old/configure.ac strace-4.7-new/configure.ac
--- strace-4.7-old/configure.ac	2012-05-02 05:34:56.000000000 -0700
+++ strace-4.7-new/configure.ac	2013-01-03 21:01:04.000000000 -0800
@@ -3,7 +3,7 @@
 AC_INIT([strace],[4.7])
 AC_CONFIG_SRCDIR([strace.c])
 AC_CONFIG_AUX_DIR([.])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([foreign check-news dist-xz no-dist-gzip silent-rules])
 AM_MAINTAINER_MODE
 AC_CANONICAL_HOST
