diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-08-26 09:40:47.403061474 -0700
+++ b/CMakeLists.txt	2016-08-26 09:40:47.464062073 -0700
@@ -6,11 +6,12 @@
 
 enable_language(CXX)
 
-find_package(kodi REQUIRED)
+find_package(Kodi REQUIRED)
 find_package(libretro-mame2010 REQUIRED)
 
 set(DEPLIBS mame2010)
 set(MAME2010_CUSTOM_BINARY ${MAME2010_LIB}
-                       ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} mame2010)
+                           ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} mame2010)
 
 build_addon(${PROJECT_NAME} MAME2010 DEPLIBS)
+
