diff -Naur xbmc-13-482ccd3/xbmc/input/linux/LinuxInputDevices.cpp xbmc-13-482ccd3.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- xbmc-13-482ccd3/xbmc/input/linux/LinuxInputDevices.cpp	2014-04-19 01:36:22.000000000 +0200
+++ xbmc-13-482ccd3.patch/xbmc/input/linux/LinuxInputDevices.cpp	2014-04-20 19:47:29.725117958 +0200
@@ -1222,8 +1222,9 @@
 
     if ((now - m_lastHotplugCheck) >= 10)
     {
-      CheckHotplugged();
-      m_lastHotplugCheck = now;
+     // Comment for now as it has visible issue on video decoding when USB devices are available
+     //CheckHotplugged(); 
+     m_lastHotplugCheck = now;
     }
   }
 
