diff -Naur xf86-video-openchrome-20090302/Makefile.am xf86-video-openchrome-20090302.patch/Makefile.am
--- xf86-video-openchrome-20090302/Makefile.am	2008-10-29 10:54:10.000000000 +0100
+++ xf86-video-openchrome-20090302.patch/Makefile.am	2009-03-24 09:58:46.040036529 +0100
@@ -19,6 +19,6 @@
 #  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = src man
+SUBDIRS = src man libxvmc
 
 EXTRA_DIST = COPYING NEWS README
diff -Naur xf86-video-openchrome-20090302/Makefile.in xf86-video-openchrome-20090302.patch/Makefile.in
--- xf86-video-openchrome-20090302/Makefile.in	2009-03-03 01:13:33.000000000 +0100
+++ xf86-video-openchrome-20090302.patch/Makefile.in	2009-03-24 09:58:59.013295184 +0100
@@ -215,7 +215,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = src man
+SUBDIRS = src man libxvmc
 EXTRA_DIST = COPYING NEWS README
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
