diff -Naur speex-1.2rc1/libspeex/Makefile.am speex-1.2rc1.patch/libspeex/Makefile.am
--- speex-1.2rc1/libspeex/Makefile.am	2008-06-04 09:31:35.000000000 +0200
+++ speex-1.2rc1.patch/libspeex/Makefile.am	2009-03-10 20:40:51.000000000 +0100
@@ -4,7 +4,7 @@
 
 EXTRA_DIST=echo_diagnostic.m
 
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@ @FFT_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)
 
 lib_LTLIBRARIES = libspeex.la libspeexdsp.la
 
diff -Naur speex-1.2rc1/libspeex/Makefile.in speex-1.2rc1.patch/libspeex/Makefile.in
--- speex-1.2rc1/libspeex/Makefile.in	2008-07-21 04:17:19.000000000 +0200
+++ speex-1.2rc1.patch/libspeex/Makefile.in	2009-03-10 20:41:25.000000000 +0100
@@ -277,7 +277,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 EXTRA_DIST = echo_diagnostic.m
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@ @FFT_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)
 lib_LTLIBRARIES = libspeex.la libspeexdsp.la
 
 # Sources for compilation in the library
