diff -Naur speex-1.2rc1/Makefile.in speex-1.2rc1a/Makefile.in
--- speex-1.2rc1/Makefile.in	2008-07-21 04:17:19.000000000 +0200
+++ speex-1.2rc1a/Makefile.in	2010-05-18 14:31:25.229200674 +0200
@@ -213,7 +213,7 @@
 EXTRA_DIST = Speex.spec Speex.spec.in Speex.kdevelop speex.m4 speex.pc.in README.blackfin README.symbian README.TI-DSP
 
 #Fools KDevelop into including all files
-SUBDIRS = libspeex include @src@ doc win32 symbian ti
+SUBDIRS = libspeex include doc win32 symbian ti
 DIST_SUBDIRS = libspeex include src doc win32 symbian ti
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
