From 55d5d57a1d5a10ee2d120fad0b4c23553054b586 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Mon, 22 Jun 2020 20:12:13 +0200
Subject: [PATCH] Fix building with gcc10

---
 image_utils.c | 2 +-
 metadata.c    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/image_utils.c b/image_utils.c
index 24cfd0841954..e8d9635dc6e0 100644
--- a/image_utils.c
+++ b/image_utils.c
@@ -190,7 +190,7 @@ jpeg_memory_src(j_decompress_ptr cinfo, const unsigned char * buffer, size_t buf
 	src->pub.bytes_in_buffer = bufsize;
 }
 
-jmp_buf setjmp_buffer;
+static jmp_buf setjmp_buffer;
 /* Don't exit on error like libjpeg likes to do */
 static void
 libjpeg_error_handler(j_common_ptr cinfo)
diff --git a/metadata.c b/metadata.c
index 9cd86dc64779..4781db716d1a 100644
--- a/metadata.c
+++ b/metadata.c
@@ -502,7 +502,7 @@ GetAudioMetadata(const char *path, const char *name)
 }
 
 /* For libjpeg error handling */
-jmp_buf setjmp_buffer;
+static jmp_buf setjmp_buffer;
 static void
 libjpeg_error_handler(j_common_ptr cinfo)
 {
-- 
2.27.0

