From 4b5990098c71ff0a5e636e75485f3bf82c07eb42 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Tue, 2 Jul 2019 14:40:57 +0100
Subject: [PATCH] Remove unused EGL headers files

---
 src/main.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index 918046b..d921054 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -22,8 +22,6 @@
 #include "xbmc_vis_types.h"
 #if defined(HAS_GLES)
 #include <GLES2/gl2.h>
-#include <EGL/egl.h>
-#include <EGL/eglext.h>
 #include <assert.h>
 #define TO_STRING(...) #__VA_ARGS__
 #else
