diff --git a/desmume/src/frontend/libretro/Makefile.libretro b/desmume/src/frontend/libretro/Makefile.libretro
index d640fbb6..85840255 100644
--- a/desmume/src/frontend/libretro/Makefile.libretro
+++ b/desmume/src/frontend/libretro/Makefile.libretro
@@ -56,10 +56,10 @@ ifneq (,$(findstring unix,$(platform)))
          CXX = g++-x86
       endif
    else
-      LIBS := -lpthread -lGL -lpcap
-      CXXFLAGS += -DHAVE_OPENGL -std=gnu++11
-      DESMUME_OPENGL = 1
-	DESMUME_OPENGL_CORE = 1
+      LIBS := -lpthread -lpcap
+      CXXFLAGS += -std=gnu++11
+      # DESMUME_OPENGL = 1
+	# DESMUME_OPENGL_CORE = 1
    endif
 
    ifneq (,$(findstring clang,$(platform)))
