diff -Naur a/service.openelec.settings/oe.py b/service.openelec.settings/oe.py
--- a/service.openelec.settings/oe.py	2013-02-12 16:37:57.000000000 +0100
+++ b/service.openelec.settings/oe.py	2013-02-14 16:22:25.902628266 +0100
@@ -542,7 +542,7 @@
     dictModules = {}
       
     for strFilename in sorted(os.listdir(__cwd__+"/resources/lib/modules")):
-      if not strFilename.startswith("__") and strFilename.endswith(".py"):
+      if not strFilename.startswith("__") and strFilename.endswith(".pyo"):
 
         strModule, strExtension = strFilename.split(".")
         try:
