diff -Naur dbus-1.8.0/configure.ac dbus-1.8.0.patch/configure.ac
--- dbus-1.8.0/configure.ac	2014-01-20 14:40:59.000000000 +0100
+++ dbus-1.8.0.patch/configure.ac	2014-02-20 18:10:36.413817342 +0100
@@ -1141,7 +1141,7 @@
     have_systemd=no;
 else
     PKG_CHECK_MODULES(SYSTEMD,
-        [libsystemd-login >= 32, libsystemd-daemon >= 32, libsystemd-journal >= 32],
+        [libsystemd >= 209],
         have_systemd=yes,
         have_systemd=no)
 fi
