diff -Naur dbus-1.2.24-old/bus/Makefile.am dbus-1.2.24-new/bus/Makefile.am
--- dbus-1.2.24-old/bus/Makefile.am	2010-03-23 12:01:27.000000000 -0700
+++ dbus-1.2.24-new/bus/Makefile.am	2010-03-25 15:59:19.000000000 -0700
@@ -1,7 +1,7 @@
 
 configdir=$(sysconfdir)/dbus-1
 
-INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES=-I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDAEMON_NAME=\"dbus-daemon\"  -DDBUS_COMPILATION
 
@@ -81,7 +81,7 @@
 	$(top_builddir)/dbus/libdbus-convenience.la	\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS=@R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 
 LAUNCH_HELPER_SOURCES=				\
 	$(XML_SOURCES)				\
diff -Naur dbus-1.2.24-old/bus/Makefile.in dbus-1.2.24-new/bus/Makefile.in
--- dbus-1.2.24-old/bus/Makefile.in	2010-03-23 12:11:23.000000000 -0700
+++ dbus-1.2.24-new/bus/Makefile.in	2010-03-25 15:59:19.000000000 -0700
@@ -459,7 +459,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 configdir = $(sysconfdir)/dbus-1
-INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS) @PIE_CFLAGS@		\
+INCLUDES = -I$(top_srcdir) $(DBUS_BUS_CFLAGS)             		\
 	-DDBUS_SYSTEM_CONFIG_FILE=\""$(configdir)/system.conf"\"	\
 	-DDAEMON_NAME=\"dbus-daemon\"  -DDBUS_COMPILATION
 
@@ -523,7 +523,7 @@
 	$(top_builddir)/dbus/libdbus-convenience.la	\
 	$(DBUS_BUS_LIBS)
 
-dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@ @PIE_LDFLAGS@
+dbus_daemon_LDFLAGS = @R_DYNAMIC_LDFLAG@ @SECTION_LDFLAGS@
 LAUNCH_HELPER_SOURCES = \
 	$(XML_SOURCES)				\
 	config-parser-common.c			\
