diff -Naur dbus-1.4.16/dbus/dbus-message.c dbus-1.4.16.patch/dbus/dbus-message.c
--- dbus-1.4.16/dbus/dbus-message.c	2011-09-21 13:16:16.000000000 +0200
+++ dbus-1.4.16.patch/dbus/dbus-message.c	2011-10-29 02:09:16.927338774 +0200
@@ -3963,7 +3963,7 @@
   DBusValidationMode mode;
   dbus_uint32_t n_unix_fds = 0;
 
-  mode = DBUS_VALIDATION_MODE_DATA_IS_UNTRUSTED;
+  mode = DBUS_VALIDATION_MODE_WE_TRUST_THIS_DATA_ABSOLUTELY;
   
   oom = FALSE;
 
