diff -Naur libusb-1.0.9-old/configure.ac libusb-1.0.9-new/configure.ac
--- libusb-1.0.9-old/configure.ac	2012-04-19 23:44:27.000000000 -0700
+++ libusb-1.0.9-new/configure.ac	2012-12-29 06:46:36.000000000 -0800
@@ -31,7 +31,7 @@
 
 AC_CONFIG_SRCDIR([libusb/core.c])
 AC_CONFIG_MACRO_DIR([m4])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
 AC_PREREQ([2.50])
