diff -Naur a/linux/Makefile b/linux/Makefile
--- a/linux/Makefile	2017-12-30 22:01:38.000000000 +0200
+++ b/linux/Makefile	2018-02-23 13:25:54.025763853 +0200
@@ -112,6 +112,21 @@
 
 untar: linux-media.tar.bz2
 	tar xfj linux-media.tar.bz2
+	mkdir -p drivers/media/amlogic/
+	# Copy amlvideodri module
+	cp -a "$(SRCDIR)/drivers/amlogic/video_dev" "drivers/media/amlogic"
+	sed -i 's,common/,,g; s,"trace/,",g' `find drivers/media/amlogic/video_dev/ -type f`
+	# Copy videobuf-res module
+	cp -a "$(SRCDIR)/drivers/media/v4l2-core/videobuf-res.c" "drivers/media/v4l2-core/"
+	cp -a "$(SRCDIR)/include/media/videobuf-res.h" "include/media/"
+
+	# Copy WeTek Play DVB driver
+	cp -a "$(SRCDIR)/drivers/amlogic/wetek" "drivers/media/amlogic"
+
+	# Copy dvb-avl driver
+	cp -a $(SRCDIR)/drivers/amlogic/dvb-avl "drivers/media"
+	echo "obj-y += dvb-avl/" >> "drivers/media/Makefile"
+
 	-rm -f .patches_applied .linked_dir .git_log.md5
 
 clean:
