diff -Naur usbmuxd-1.0.6/Modules/describe.sh usbmuxd-1.0.6.patch/Modules/describe.sh
--- usbmuxd-1.0.6/Modules/describe.sh	2010-09-28 03:32:44.000000000 +0200
+++ usbmuxd-1.0.6.patch/Modules/describe.sh	2011-01-05 14:28:41.461148284 +0100
@@ -1,17 +1,8 @@
-#!/bin/bash
+#!/bin/sh
 
-# Check for git and a git repo.
-if head=`git rev-parse --verify HEAD 2>/dev/null`; then
-	/bin/echo -n `git describe`
-
-	# Are there uncommitted changes?
-	git update-index --refresh --unmerged > /dev/null
-	git diff-index --quiet HEAD || /bin/echo -n -dirty
-else
 # Check for version tag
 	if [ -e version.tag ]; then
 		/bin/echo -n `cat version.tag`
 	fi
-fi
 
 echo
