diff --git a/retroarch.c b/retroarch.c
index cc551e2827..774e1b92ad 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -5191,7 +5191,7 @@ TODO: Add a setting for these tweaks */
          runloop_msg_queue_push(msg_hash_to_str(MSG_VALUE_SHUTTING_DOWN), 1, 180, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
          command_event(CMD_EVENT_MENU_SAVE_CURRENT_CONFIG, NULL);
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -P now");
+         system("shutdown -P now &");
 #endif
          break;
       case CMD_EVENT_REBOOT:
@@ -5199,7 +5199,7 @@ TODO: Add a setting for these tweaks */
          runloop_msg_queue_push(msg_hash_to_str(MSG_VALUE_REBOOTING), 1, 180, true, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO);
          command_event(CMD_EVENT_MENU_SAVE_CURRENT_CONFIG, NULL);
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -r now");
+         system("shutdown -r now &");
 #endif
          break;
       case CMD_EVENT_RESUME:
