diff --git a/Makefile.common b/Makefile.common
index 580beba7eb..f8b9a77d10 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -1077,6 +1077,10 @@ ifeq ($(HAVE_LAKKA_NIGHTLY), 1)
    DEFINES += -DHAVE_LAKKA_NIGHTLY
 endif
 
+ifeq ($(HAVE_LAKKA_DEVBUILD), 1)
+   DEFINES += -DHAVE_LAKKA_DEVBUILD
+endif
+
 ifeq ($(HAVE_MENU_COMMON), 1)
    OBJ += menu/menu_setting.o \
           menu/menu_driver.o \
diff --git a/file_path_special.h b/file_path_special.h
index a303f7ca49..56b728559c 100644
--- a/file_path_special.h
+++ b/file_path_special.h
@@ -75,8 +75,10 @@ RETRO_BEGIN_DECLS
 #define FILE_PATH_LOBBY_LIBRETRO_URL "http://lobby.libretro.com/"
 #define FILE_PATH_CORE_THUMBNAILS_URL "http://thumbnails.libretro.com"
 #define FILE_PATH_CORE_THUMBNAILPACKS_URL "http://thumbnailpacks.libretro.com"
-#ifdef HAVE_LAKKA_NIGHTLY
+#if defined(HAVE_LAKKA_NIGHTLY)
 #define FILE_PATH_LAKKA_URL "http://nightly.builds.lakka.tv/.updater"
+#elif defined(HAVE_LAKKA_DEVBUILD)
+#define FILE_PATH_LAKKA_URL "http://nightly.builds.lakka.tv/.devbuilds"
 #else
 #define FILE_PATH_LAKKA_URL "http://le.builds.lakka.tv"
 #endif
