diff --git a/Makefile.common b/Makefile.common
index 2a715f4786..cd2f49518b 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -788,6 +788,7 @@ endif
 ifeq ($(HAVE_LAKKA), 1)
    OBJ += network/drivers_wifi/connmanctl.o
    OBJ += misc/cpufreq/cpufreq.o
+   LIBS += -lsystemd
 endif
 
 ifeq ($(HAVE_WIFI), 1)
diff --git a/retroarch.c b/retroarch.c
index 8137c2e358..3c766dd6cb 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -215,6 +215,7 @@
 
 #ifdef HAVE_LAKKA
 #include "lakka.h"
+#include <systemd/sd-daemon.h>
 #endif
 
 #define _PSUPP(var, name, desc) printf("  %s:\n\t\t%s: %s\n", name, desc, var ? "yes" : "no")
@@ -5933,6 +5934,9 @@ int rarch_main(int argc, char *argv[], void *data)
 #if HAVE_CLOUDSYNC
    task_push_cloud_sync();
 #endif
+#ifdef HAVE_LAKKA
+   sd_notify(0, "READY=1");
+#endif
 #if !defined(HAVE_MAIN) || defined(HAVE_QT)
    for (;;)
    {
