diff --git a/src/i965_drv_video.c b/src/i965_drv_video.c
index 31dafa2..a58eb7b 100755
--- a/src/i965_drv_video.c
+++ b/src/i965_drv_video.c
@@ -2358,7 +2358,11 @@ i965_QuerySurfaceStatus(VADriverContextP ctx,
             *status = VASurfaceReady;
         }
     } else {
+#ifdef HAVE_VA_SURFACE_STATUS_EMPTY
+        *status = VASurfaceReady | VASurfaceEmpty;
+#else
         *status = VASurfaceReady;
+#endif
     }
 
     return VA_STATUS_SUCCESS;
