diff --git a/src/gen75_vpp_vebox.c b/src/gen75_vpp_vebox.c
index a973ed4..ff1dd11 100644
--- a/src/gen75_vpp_vebox.c
+++ b/src/gen75_vpp_vebox.c
@@ -140,7 +140,7 @@ void hsw_veb_dndi_table(VADriverContextP ctx, struct intel_vebox_context *proc_c
         assert(di_param);
 
         progressive_dn = 0;
-        dndi_top_first = !(di_param->flags & VA_DEINTERLACING_BOTTOM_FIELD_FIRST);
+        dndi_top_first = !(di_param->flags & VA_DEINTERLACING_BOTTOM_FIELD);
     }
 
     /*
diff --git a/src/i965_post_processing.c b/src/i965_post_processing.c
index 536eb64..ccdd54c 100755
--- a/src/i965_post_processing.c
+++ b/src/i965_post_processing.c
@@ -3215,7 +3215,7 @@ pp_nv12_dndi_initialize(VADriverContextP ctx, struct i965_post_processing_contex
     int dndi_top_first = 1;
     VAProcFilterParameterBufferDeinterlacing *di_filter_param = (VAProcFilterParameterBufferDeinterlacing *)filter_param;
 
-    if (di_filter_param->flags & VA_DEINTERLACING_BOTTOM_FIELD_FIRST)
+    if (di_filter_param->flags & VA_DEINTERLACING_BOTTOM_FIELD)
         dndi_top_first = 0;
     else
         dndi_top_first = 1;
@@ -3617,7 +3617,7 @@ gen7_pp_nv12_dndi_initialize(VADriverContextP ctx, struct i965_post_processing_c
     VAProcFilterParameterBufferDeinterlacing *di_filter_param = (VAProcFilterParameterBufferDeinterlacing *)filter_param;
     int is_first_frame = (pp_dndi_context->frame_order == -1);
 
-    if (di_filter_param->flags & VA_DEINTERLACING_BOTTOM_FIELD_FIRST)
+    if (di_filter_param->flags & VA_DEINTERLACING_BOTTOM_FIELD)
         dndi_top_first = 0;
     else
         dndi_top_first = 1;
