diff -Naur intel-driver-1.0.19-old/configure.ac intel-driver-1.0.19-new/configure.ac
--- intel-driver-1.0.19-old/configure.ac	2012-11-08 17:43:34.000000000 -0800
+++ intel-driver-1.0.19-new/configure.ac	2013-01-03 17:19:13.000000000 -0800
@@ -22,7 +22,7 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([1.9 tar-ustar])
 
-AM_CONFIG_HEADER([src/config.h])
+AC_CONFIG_HEADERS([src/config.h])
 
 INTEL_DRIVER_MAJOR_VERSION=intel_driver_major_version
 INTEL_DRIVER_MINOR_VERSION=intel_driver_minor_version
