From 6a39049e34eeefeeb821970d83e1994870af8f3e Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 1 Nov 2007 14:56:25 -0400
Subject: [PATCH] Don't backfill bg=None windows in Composite.

---
 composite/compalloc.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/composite/compalloc.c b/composite/compalloc.c
index 006e808..67d830d 100644
--- a/composite/compalloc.c
+++ b/composite/compalloc.c
@@ -478,6 +478,7 @@ compNewPixmap (WindowPtr pWin, int x, int y, int w, int h)
 	 * Copy bits from the parent into the new pixmap so that it will
 	 * have "reasonable" contents in case for background None areas.
 	 */
+#if 0
 	if (pGC)
 	{
 	    XID val = IncludeInferiors;
@@ -492,6 +493,7 @@ compNewPixmap (WindowPtr pWin, int x, int y, int w, int h)
 				   w, h, 0, 0);
 	    FreeScratchGC (pGC);
 	}
+#endif
     }
     else
     {
-- 
1.5.3.4

