From d8f0b7e388f61a9ae528466dafac1bdfaf5b77ca Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Dec 2007 15:25:01 -0500
Subject: [PATCH] Poison {DE,}ALLOCATE_LOCAL so we don't build broken drivers.

---
 include/os.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/include/os.h b/include/os.h
index c53a9ea..0960d0c 100644
--- a/include/os.h
+++ b/include/os.h
@@ -50,6 +50,10 @@ SOFTWARE.
 #define OS_H
 
 #include "misc.h"
+
+#pragma GCC poison ALLOCATE_LOCAL
+#pragma GCC poison DEALLOCATE_LOCAL
+
 #include <stdarg.h>
 
 #define NullFID ((FID) 0)
-- 
1.5.2.4

