diff -Naur pm-utils-1.4.1/configure.ac pm-utils-1.4.1.patch/configure.ac
--- pm-utils-1.4.1/configure.ac	2010-07-04 16:50:59.000000000 +0200
+++ pm-utils-1.4.1.patch/configure.ac	2013-01-12 16:48:21.954203172 +0100
@@ -2,7 +2,7 @@
 AC_INIT(pm-utils, 1.4.1)
 AC_CONFIG_SRCDIR(src)
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_PROG_CC
 AC_PROG_LN_S
