diff --git a/installer/src/Makefile b/installer/src/Makefile
index 7679c65..a83fb0b 100644
--- a/installer/src/Makefile
+++ b/installer/src/Makefile
@@ -56,7 +56,7 @@ h826d-objs := $(COBJS) $(PREBUILD_OBJ)
 averusbh826d-objs := aver/averusb-mod.o
 default:: links
 	$(MAKE) -C $(KERNELSRC) O=$(KERNELOBJ) SUBDIRS=`pwd` $(TARGET)
-	$(CROSS_COMPILE)strip --strip-debug *.ko
+	$(STRIP) --strip-debug *.ko
 
 
 
