diff -Naur AVERMEDIA-Linux-x64-H826D-0.10-beta/installer/src/driver-core.c AVERMEDIA-Linux-x64-H826D-0.10-beta.patch/installer/src/driver-core.c
--- AVERMEDIA-Linux-x64-H826D-0.10-beta/installer/src/driver-core.c	2009-11-13 11:54:59.000000000 +0100
+++ AVERMEDIA-Linux-x64-H826D-0.10-beta.patch/installer/src/driver-core.c	2011-12-21 02:32:24.008225956 +0100
@@ -81,6 +81,7 @@
 #include	"aver/id.h"
 #include    "driver-core.h" 
 #include	<linux/version.h>
+#include        <linux/module.h>
 
 
 
