diff -Nr -u0 original/installer/installer.sh modified/installer/installer.sh
--- original/installer/installer.sh     2009-11-13 11:47:33.000000000 +0100
+++ modified/installer/installer.sh     2011-08-28 20:58:38.000000000 +0200
@@ -177,13 +176,0 @@
-	if [[ "$kversion" != "2" || "$kpatchlevel" != "6" ]]; then
-		dialog --backtitle "$BACKTITLE" \
-		--title "Kernel version error" \
-		--msgbox "Installer cannot determine kernel version or the running kernel is not 2.6.x kernel. \n\
-Installer will abort now.
-" \
-		10 $WIDTH
-		log "generate_kdep_string: wrong kernel version ${kversion}.${kpatchlevel}.${ksublevel}. Abort."
-        log_from_file $ksrc/Makefile #s016
-        log_from_file $kobj/Makefile #s016
-		exit
-	fi
-
diff -Nr -u0 original/installer/src/aver/osdep.c modified/installer/src/aver/osdep.c
--- original/installer/src/aver/osdep.c 2009-11-13 11:54:56.000000000 +0100
+++ modified/installer/src/aver/osdep.c 2011-08-28 20:19:19.000000000 +0200
@@ -89 +88,0 @@
-#include <linux/smp_lock.h>
@@ -435 +434 @@
-		init_MUTEX(tmp);
+		sema_init(tmp, 1);
@@ -439 +438 @@
-		init_MUTEX_LOCKED(tmp);
+		sema_init(tmp, 0);
@@ -1069,0 +1069 @@
+static DEFINE_MUTEX(dvbdev_mutex);
@@ -1072 +1072 @@
-	lock_kernel();
+	mutex_lock(&dvbdev_mutex);
@@ -1077 +1077 @@
-	unlock_kernel();
+	mutex_unlock(&dvbdev_mutex);
diff -Nr -u0 original/installer/src/aver/osdep_dvb.c modified/installer/src/aver/osdep_dvb.c
--- original/installer/src/aver/osdep_dvb.c     2009-11-13 11:54:57.000000000 +0100
+++ modified/installer/src/aver/osdep_dvb.c     2011-08-28 20:19:36.000000000 +0200
@@ -79 +78,0 @@
-#include <linux/smp_lock.h>
@@ -233 +232 @@
-		init_MUTEX(&p->feedlock);
+		sema_init(&p->feedlock, 1);
diff -Nr -u0 original/installer/src/aver/osdep_th2.c modified/installer/src/aver/osdep_th2.c
--- original/installer/src/aver/osdep_th2.c     2009-11-13 11:54:57.000000000 +0100
+++ modified/installer/src/aver/osdep_th2.c     2011-08-28 20:20:00.000000000 +0200
@@ -78 +77,0 @@
-#include <linux/smp_lock.h>
@@ -80,0 +80 @@
+#include "osdep.h"
@@ -90 +90 @@
-        lock_kernel();
+        SysLockKernel();
@@ -100 +100 @@
-        unlock_kernel();
+        SysUnlockKernel();
diff -Nr -u0 original/installer/src/aver/osdep_v4l2.c modified/installer/src/aver/osdep_v4l2.c
--- original/installer/src/aver/osdep_v4l2.c    2009-11-13 11:54:58.000000000 +0100
+++ modified/installer/src/aver/osdep_v4l2.c    2011-08-28 20:20:19.000000000 +0200
@@ -75,0 +76,2 @@
+#define VFL_TYPE_VTX 3
+
@@ -84 +85,0 @@
-#include <linux/smp_lock.h>
