--- a/drivers/rtc/rtc-pcf8523.c 2016-10-01 10:16:30.259771931 +0100
+++ b/drivers/rtc/rtc-pcf8523.c	2016-10-01 10:21:29.762638800 +0100
@@ -291,7 +291,13 @@ static int pcf8523_probe(struct i2c_clie
 	if (!pcf)
 		return -ENOMEM;
 
-	err = pcf8523_select_capacitance(client, true);
+	if (of_property_read_bool(client->dev.of_node, "nxp,xtalcap-7pf")) {
+		printk(KERN_ERR "PCF8523 - set 7pF crystal load");
+		err = pcf8523_select_capacitance(client, false);
+	} else {
+		printk(KERN_ERR "PCF8523 - set 12pF crystal load");
+		err = pcf8523_select_capacitance(client, true);
+	}
 	if (err < 0)
 		return err;
 
