diff -Naur strace-4.8/configure.ac strace-4.8.patch/configure.ac
--- strace-4.8/configure.ac	2013-05-04 21:51:57.000000000 +0200
+++ strace-4.8.patch/configure.ac	2013-09-04 03:30:12.478057723 +0200
@@ -1,8 +1,6 @@
 dnl Process this file with autoconf to create configure.  Use autoreconf.
 AC_PREREQ(2.57)
-AC_INIT([strace],
-	m4_esyscmd([./git-version-gen .tarball-version]),
-	[strace-devel@lists.sourceforge.net])
+AC_INIT([strace],[4.8])
 AC_CONFIG_SRCDIR([strace.c])
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_HEADERS([config.h])
