diff --git a/Makefile b/Makefile
index 8bdbfa7..cb9674f 100644
--- a/Makefile
+++ b/Makefile
@@ -192,9 +194,9 @@ else
   ifeq ($(platform),vita)
     CFLAGS   := -DNDEBUG -O3 -fno-ident -fno-partial-inlining
   else
-    CFLAGS   := -DNDEBUG -O2 -fno-ident
+    CFLAGS   := -DNDEBUG -O3 -fno-ident
   endif
-  LDFLAGS  += -O2
+  LDFLAGS  += -O3 -fno-ident
 endif
 
 CFLAGS  += $(CPUFLAGS) -std=gnu++11 -fomit-frame-pointer -fno-exceptions -fno-non-call-exceptions -Wno-address-of-packed-member -Wno-format -Wno-switch -Wno-psabi
@@ -209,7 +211,8 @@ CFLAGS  += -D__LIBRETRO__ -Iinclude -D_FILE_OFFSET_BITS=64
 CFLAGS  += $(COMMONFLAGS)
 #CFLAGS  += -fdata-sections #saves around 32 bytes on most platforms but wrongfully adds up to 60MB on msys2
 
-LDFLAGS += $(CPUFLAGS) -lpthread -shared
+LDFLAGS += $(CPUFLAGS) -shared
+LDLIBS  := -lpthread
 #LDFLAGS += -static-libstdc++ -static-libgcc #adds 1MB to output and still dynamically links against libc and libm
 
 .PHONY: all clean
@@ -234,7 +237,7 @@ ifeq ($(STATIC_LINKING), 1)
 	$(AR) rcs $@ $^
 else
 	$(info Linking $@ ...)
-	$(CXX) $(LDFLAGS) -o $@ $^
+	$(CXX) $(LDFLAGS) -o $@ $^ $(LDLIBS)
 ifneq ($(BUILD),DEBUG)
 	@-/opt/gcw0-toolchain/usr/mipsel-gcw0-linux-uclibc/bin/strip --strip-all $@ $(PIPETONULL);true # gcw0
 	@-strip --strip-all $@ $(PIPETONULL);true # others
