--- diffutils-3.5/Makefile.am.org	2017-05-24 18:54:09.404508347 +0200
+++ diffutils-3.5/Makefile.am	2017-05-24 18:46:25.238456499 +0200
@@ -19,7 +19,7 @@
 ALL_RECURSIVE_TARGETS =
 
 EXTRA_DIST = bootstrap exgettext ChangeLog-2008 cfg.mk dist-check.mk
-SUBDIRS = lib src tests doc man po gnulib-tests
+SUBDIRS = lib src tests doc po gnulib-tests
 
 ACLOCAL_AMFLAGS = -I m4
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
