diff -Naur emux.git/libretro/Makefile.lakka emux.patch/libretro/Makefile.lakka
--- emux.git/libretro/Makefile.lakka	1970-01-01 01:00:00.000000000 +0100
+++ emux.patch/libretro/Makefile.lakka	2016-05-12 08:46:58.338888783 +0200
@@ -0,0 +1,65 @@
+HOST_PLATFORM = linux
+
+TARGET_PLATFORM = lakka
+
+OBJEXT = .o
+SOEXT  = .so
+
+################
+# Platform setup
+
+STATIC_LINKING = 0
+platform       = unix
+PLATDEFS       = 
+PLATCFLAGS     = -fpic -fstrict-aliasing
+PLATCXXFLAGS   = -fpic -fstrict-aliasing
+PLATLDFLAGS    = -shared -lm -Wl,-version-script=$(BUILD_DIR)/link.T -Wl,-no-undefined
+PLATLDXFLAGS   = -shared -lm -Wl,-version-script=$(BUILD_DIR)/link.T -Wl,-no-undefined
+
+################
+# libretro setup
+
+RETRODEFS     = -D__LIBRETRO__
+RETROCFLAGS   =
+RETROCXXFLAGS =
+RETROLDFLAGS  =
+RETROLDXFLAGS =
+
+#################
+# Final variables
+
+DEFINES  += $(PLATDEFS) $(RETRODEFS)
+CFLAGS   += $(PLATCFLAGS) $(RETROCFLAGS) $(DEFINES) $(INCLUDES)
+CXXFLAGS += $(PLATCXXFLAGS) $(RETROCXXFLAGS) $(DEFINES) $(INCLUDES)
+LDFLAGS  += $(PLATLDFLAGS) $(RETROLDFLAGS)
+LDXFLAGS += $(PLATLDXFLAGS) $(RETROLDXFLAGS)
+
+########
+# Tuning
+
+ifneq ($(DEBUG),)
+  CFLAGS   += -O0 -g
+  CXXFLAGS += -O0 -g
+else
+  CFLAGS   += -O3 -DNDEBUG
+  CXXFLAGS += -O3 -DNDEBUG
+endif
+
+ifneq ($(LOG_PERFORMANCE),)
+  CFLAGS   += -DLOG_PERFORMANCE
+  CXXFLAGS += -DLOG_PERFORMANCE
+endif
+
+####################################
+# Variable setup for Makefile.common
+
+CORE_DIR  ?= ..
+BUILD_DIR ?= .
+INCLUDES   = 
+
+include $(BUILD_DIR)/Makefile.common
+
+###############
+# Include rules
+
+include $(BUILD_DIR)/Makefile.rules
