diff --git a/Makefile b/Makefile
index db13ab8e..7d1cec95 100644
--- a/Makefile
+++ b/Makefile
@@ -22,24 +22,16 @@ HAVE_CLANG    := 0
 TARGET_NAME   := reicast
 
 ifeq ($(HAVE_CLANG),1)
-	CXX      = ${CC_PREFIX}clang++
-	CC       = ${CC_PREFIX}clang
 	SHARED   := -fuse-ld=lld
 else
-	CXX      = ${CC_PREFIX}g++
-	CC       = ${CC_PREFIX}gcc
 	SHARED   :=
 endif
-CC_AS    = ${CC_PREFIX}as
 
 MFLAGS   := 
 ASFLAGS  := 
-LDFLAGS  :=
 LDFLAGS_END :=
 INCFLAGS :=
 LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
 
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
@@ -856,8 +870,8 @@ CXXFLAGS       += $(CORE_DEFINES)
 
 PREFIX        ?= /usr/local
 
 ifneq (,$(findstring arm, $(ARCH)))
-	CC_AS    = ${CC_PREFIX}gcc #The ngen_arm.S must be compiled with gcc, not as
+	CC_AS    = ${CC_PREFIX}${CC} #The ngen_arm.S must be compiled with gcc, not as
 	ASFLAGS  += $(CFLAGS)
 endif
 
