diff --git a/configure.ac b/configure.ac
index 50c0fe2..1d67de7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -159,10 +159,6 @@ PKG_CHECK_MODULES(LIBPARTED, [libparted >= 1.8.8])
 AC_SUBST(LIBPARTED_CFLAGS)
 AC_SUBST(LIBPARTED_LIBS)
 
-PKG_CHECK_MODULES(DEVMAPPER, [devmapper >= 1.02])
-AC_SUBST(DEVMAPPER_CFLAGS)
-AC_SUBST(DEVMAPPER_LIBS)
-
 have_lvm2=no
 AC_ARG_ENABLE(lvm2, AS_HELP_STRING([--enable-lvm2], [enable LVM2 support]))
 if test "x$enable_lvm2" = "xyes"; then
diff --git a/src/probers/Makefile.am b/src/probers/Makefile.am
index 06bb566..578b8ae 100644
--- a/src/probers/Makefile.am
+++ b/src/probers/Makefile.am
@@ -22,9 +22,7 @@ INCLUDES = \
 
 udevhelperdir = $(slashlibdir)/udev
 udevhelper_PROGRAMS =	udisks-part-id 			\
-			udisks-dm-export 		\
 		      	udisks-probe-ata-smart 		\
-			udisks-probe-sas-expander 	\
 			$(NULL)
 
 if HAVE_LVM2
