diff -Naur udisks-1.0.2/src/device.c udisks-1.0.2.patch/src/device.c
--- udisks-1.0.2/src/device.c	2010-11-06 17:41:23.000000000 +0100
+++ udisks-1.0.2.patch/src/device.c	2011-03-08 23:25:46.382977610 +0100
@@ -6307,7 +6307,7 @@
   options = prepend_default_mount_options (fsmo, caller_uid, given_options);
 
   /* validate mount options and check for authorizations */
-  s = g_string_new ("uhelper=udisks,nodev,nosuid");
+  s = g_string_new ("nodev,nosuid");
   for (n = 0; options[n] != NULL; n++)
     {
       const char *option = options[n];
