diff -Naur udisks-1.0.4/configure.ac udisks-1.0.4.patch/configure.ac
--- udisks-1.0.4/configure.ac	2011-08-26 17:08:35.000000000 +0200
+++ udisks-1.0.4.patch/configure.ac	2013-01-12 16:43:38.483117569 +0100
@@ -3,12 +3,11 @@
 AC_PREREQ(2.59c)
 AC_INIT(udisks, 1.0.4, http://bugs.freedesktop.org/enter_bug.cgi?product=udisks)
 AM_INIT_AUTOMAKE(udisks, 1.0.4)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AM_PROG_LIBTOOL
 AC_PROG_MAKE_SET
