diff -Naur linux-Switch-5.1.2/drivers/scsi/libiscsi_tcp.c linux-Switch-5.1.2-2/drivers/scsi/libiscsi_tcp.c
--- linux-Switch-5.1.2/drivers/scsi/libiscsi_tcp.c	2024-05-01 03:59:48.712685224 +0200
+++ linux-Switch-5.1.2-2/drivers/scsi/libiscsi_tcp.c	2024-05-01 04:11:55.495873360 +0200
@@ -100,7 +100,7 @@
 	segment->sg = sg;
 	segment->sg_offset = offset;
 	segment->size = min(sg->length - offset,
-			    segment->total_size - segment->total_copied);
+			    (size_t) segment->total_size - segment->total_copied);
 	segment->data = NULL;
 }
 
