diff -Naur pkg-config-0.25-old/pkg.m4 pkg-config-0.25-new/pkg.m4
--- pkg-config-0.25-old/pkg.m4	2009-12-06 12:39:02.000000000 -0800
+++ pkg-config-0.25-new/pkg.m4	2010-05-28 16:55:19.000000000 -0700
@@ -26,7 +26,15 @@
 # ----------------------------------
 AC_DEFUN([PKG_PROG_PKG_CONFIG],
 [m4_pattern_forbid([^_?PKG_[A-Z_]+$])
-m4_pattern_allow([^PKG_CONFIG(_PATH)?$])
+m4_pattern_allow([^PKG_CONFIG$])
+m4_pattern_allow([^PKG_CONFIG_ALLOW_SYSTEM_CFLAGS$])
+m4_pattern_allow([^PKG_CONFIG_ALLOW_SYSTEM_LIBS$])
+m4_pattern_allow([^PKG_CONFIG_DEBUG_SPEW$])
+m4_pattern_allow([^PKG_CONFIG_DISABLE_UNINSTALLED$])
+m4_pattern_allow([^PKG_CONFIG_LIBDIR$])
+m4_pattern_allow([^PKG_CONFIG_PATH$])
+m4_pattern_allow([^PKG_CONFIG_SYSROOT_DIR$])
+m4_pattern_allow([^PKG_CONFIG_TOP_BUILD_DIR$])
 AC_ARG_VAR([PKG_CONFIG], [path to pkg-config utility])
 AC_ARG_VAR([PKG_CONFIG_PATH], [directories to add to pkg-config's search path])
 AC_ARG_VAR([PKG_CONFIG_LIBDIR], [path overriding pkg-config's built-in search path])
