diff -NaurwB xmltv2vdr-0.1.1_orig/dist/epgdata2xmltv/epgdata2xmltv.h xmltv2vdr-0.1.1/dist/epgdata2xmltv/epgdata2xmltv.h
--- xmltv2vdr-0.1.1_orig/dist/epgdata2xmltv/epgdata2xmltv.h	2012-05-07 07:34:21.000000000 +0200
+++ xmltv2vdr-0.1.1/dist/epgdata2xmltv/epgdata2xmltv.h	2012-05-07 07:34:36.000000000 +0200
@@ -7,7 +7,6 @@
 #define __EPGDATA2XMLTV_H
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <libxslt/transform.h>
