--- a/libopenmpt/libopenmpt.pc.in	2016-07-09 18:17:00.000000000 +0000
+++ b/libopenmpt/libopenmpt.pc.in	2021-12-25 00:53:54.189008995 +0000
@@ -8,7 +8,7 @@
 Description: Tracker module player based on OpenMPT
 Version: @VERSION@
 Requires.private: @LIBOPENMPT_REQUIRES_PRIVATE@
-Libs: -L${libdir} -lopenmpt
+Libs: -L${libdir} -lopenmpt -lvorbisfile
 Libs.private: @LIBOPENMPT_LIBS_PRIVATE@
 Cflags: -I${includedir}
 
