--- a/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp	2013-10-02 18:39:38.933364683 +0200
+++ b/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp	2013-10-02 18:39:38.945364681 +0200
@@ -223,7 +223,7 @@
       return true;
 
 
-  if (m_dev->Ioctl(HDMICEC_IOC_SETLOGICALADDRESS, log_addr) != 0)
+  if (m_dev->Ioctl(HDMICEC_IOC_SETLOGICALADDRESS, (void *)log_addr) != 0)
   {
     LIB_CEC->AddLog(CEC_LOG_ERROR, "%s: HDMICEC_IOC_SETLOGICALADDRESS failed !", __func__);
     return false;
