diff -up libshout-2.2.2/src/Makefile.in.fix_speex libshout-2.2.2/src/Makefile.in
--- libshout-2.2.2/src/Makefile.in.fix_speex	2007-12-07 15:03:46.000000000 +0100
+++ libshout-2.2.2/src/Makefile.in	2007-12-07 15:04:43.000000000 +0100
@@ -65,7 +65,8 @@ LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_2 =
 libshout_la_DEPENDENCIES = net/libicenet.la timing/libicetiming.la \
 	avl/libiceavl.la httpp/libicehttpp.la $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_2) $(am__DEPENDENCIES_2)
+	$(am__DEPENDENCIES_2) $(am__DEPENDENCIES_2) \
+	$(am__DEPENDENCIES_2)
 am__libshout_la_SOURCES_DIST = shout.c util.c ogg.c vorbis.c mp3.c \
 	theora.c speex.c
 @HAVE_THEORA_TRUE@am__objects_1 = theora.lo
@@ -128,6 +129,7 @@ EXEEXT = @EXEEXT@
 F77 = @F77@
 FFLAGS = @FFLAGS@
 FGREP = @FGREP@
+GREP = @GREP@
 HAVE_PKGCONFIG_FALSE = @HAVE_PKGCONFIG_FALSE@
 HAVE_PKGCONFIG_TRUE = @HAVE_PKGCONFIG_TRUE@
 HAVE_SPEEX_FALSE = @HAVE_SPEEX_FALSE@
@@ -199,13 +201,9 @@ VORBIS_LIBS = @VORBIS_LIBS@
 VORBIS_PREFIX = @VORBIS_PREFIX@
 XIPH_CFLAGS = @XIPH_CFLAGS@
 XIPH_CPPFLAGS = @XIPH_CPPFLAGS@
-ac_ct_AR = @ac_ct_AR@
 ac_ct_CC = @ac_ct_CC@
 ac_ct_CXX = @ac_ct_CXX@
 ac_ct_F77 = @ac_ct_F77@
-ac_ct_RANLIB = @ac_ct_RANLIB@
-ac_ct_STRIP = @ac_ct_STRIP@
-ac_pt_PKG_CONFIG = @ac_pt_PKG_CONFIG@
 acx_pthread_config = @acx_pthread_config@
 am__fastdepCC_FALSE = @am__fastdepCC_FALSE@
 am__fastdepCC_TRUE = @am__fastdepCC_TRUE@
@@ -223,23 +221,30 @@ build_cpu = @build_cpu@
 build_os = @build_os@
 build_vendor = @build_vendor@
 datadir = @datadir@
+datarootdir = @datarootdir@
+docdir = @docdir@
+dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
 host_alias = @host_alias@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
+htmldir = @htmldir@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
+localedir = @localedir@
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 oldincludedir = @oldincludedir@
+pdfdir = @pdfdir@
 prefix = @prefix@
 program_transform_name = @program_transform_name@
+psdir = @psdir@
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 shout_cflags = @shout_cflags@
@@ -261,7 +266,7 @@ noinst_HEADERS = shout_ogg.h shout_priva
 libshout_la_SOURCES = shout.c util.c ogg.c vorbis.c mp3.c $(MAYBE_THEORA) $(MAYBE_SPEEX)
 AM_CFLAGS = @XIPH_CFLAGS@
 libshout_la_LIBADD = net/libicenet.la timing/libicetiming.la avl/libiceavl.la\
-		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS)
+		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS) $(SPEEX_LIBS)
 
 INCLUDES = -I$(top_builddir)/include
 all: all-recursive
diff -up libshout-2.2.2/src/Makefile.am.fix_speex libshout-2.2.2/src/Makefile.am
--- libshout-2.2.2/src/Makefile.am.fix_speex	2007-12-07 15:03:39.000000000 +0100
+++ libshout-2.2.2/src/Makefile.am	2007-12-07 15:04:35.000000000 +0100
@@ -26,7 +26,7 @@ libshout_la_SOURCES = shout.c util.c ogg
 AM_CFLAGS = @XIPH_CFLAGS@
 
 libshout_la_LIBADD = net/libicenet.la timing/libicetiming.la avl/libiceavl.la\
-		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS)
+		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS) $(SPEEX_LIBS)
 
 INCLUDES = -I$(top_builddir)/include
 
