diff -Naur RTL8188EU-fcc9034/os_dep/rtw_android.c RTL8188EU-fcc9034.patch/os_dep/rtw_android.c
--- RTL8188EU-fcc9034/os_dep/rtw_android.c	2014-03-25 15:05:52.000000000 +0100
+++ RTL8188EU-fcc9034.patch/os_dep/rtw_android.c	2015-04-04 20:44:25.386732963 +0200
@@ -20,6 +20,7 @@
 
 #include <linux/module.h>
 #include <linux/netdevice.h>
+#include <linux/version.h>
 
 #include <rtw_android.h>
 #include <osdep_service.h>
@@ -80,8 +81,13 @@
 {
 	int cmd_num;
 	for (cmd_num = 0; cmd_num < ANDROID_WIFI_CMD_MAX; cmd_num++)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0))
+		if (!strncasecmp(cmdstr , android_wifi_cmd_str[cmd_num],
+				  strlen(android_wifi_cmd_str[cmd_num])))
+#else
 		if (0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num],
 				  strlen(android_wifi_cmd_str[cmd_num])))
+#endif
 			break;
 	return cmd_num;
 }
