diff -Naur RTL8188EU-fcc9034/os_dep/usb_intf.c RTL8188EU-fcc9034.patch/os_dep/usb_intf.c
--- RTL8188EU-fcc9034/os_dep/usb_intf.c	2014-03-25 15:05:52.000000000 +0100
+++ RTL8188EU-fcc9034.patch/os_dep/usb_intf.c	2014-04-28 17:42:46.059538024 +0200
@@ -865,7 +865,6 @@
 	RT_TRACE(_module_hci_intfs_c_, _drv_err_, ("+rtw_drv_entry\n"));
 
 	DBG_88E(DRV_NAME " driver version=%s\n", DRIVERVERSION);
-	DBG_88E("build time: %s %s\n", __DATE__, __TIME__);
 
 	rtw_suspend_lock_init();
 
