diff -Naur linux-gamgirl.git/arch/arm/boot/dts/overlays/gamegirl-overlay.dts linux-gamgirl.patch/arch/arm/boot/dts/overlays/gamegirl-overlay.dts
--- linux-gamgirl.git/arch/arm/boot/dts/overlays/gamegirl-overlay.dts	1970-01-01 01:00:00.000000000 +0100
+++ linux-gamgirl.patch/arch/arm/boot/dts/overlays/gamegirl-overlay.dts	2017-01-09 10:02:38.145041732 +0100
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "brcm,bcm2708";
+
+    fragment@0 {
+        target = <&gpio>;
+        __overlay__ {                                                                                                                                                                         pinctrl-names = "default";
+            pinctrl-0 = <&gamegirl>;
+
+            gamegirl: gamegirl {
+                brcm,pins = <18 19>;     /* gpio no. */
+                brcm,function = <2 2>; /* 0: GPIO in, 1: GPIO out, 2: alt5, 3: alt4, 4: alt0, 5: alt1, 6: alt2, 7: alt3 */
+                brcm,pull = <2 2>;     /* 2:up 1:down 0:none */
+            };  
+        };  
+    };  
+};
diff -Naur linux-gamgirl.git/arch/arm/boot/dts/overlays/Makefile linux-gamgirl.patch/arch/arm/boot/dts/overlays/Makefile
--- linux-gamgirl.git/arch/arm/boot/dts/overlays/Makefile	2017-01-09 09:59:00.625048619 +0100
+++ linux-gamgirl.patch/arch/arm/boot/dts/overlays/Makefile	2017-01-09 10:04:06.805038925 +0100
@@ -18,6 +18,7 @@
 	enc28j60-spi2.dtbo \
 	fe-pi-audio.dtbo \
 	googlevoicehat-soundcard.dtbo \
+	gamegirl.dtbo \
 	gpio-ir.dtbo \
 	gpio-poweroff.dtbo \
 	hifiberry-amp.dtbo \
