diff -Naur a/config.h.in b/config.h.in
--- a/config.h.in	2016-05-18 18:19:56.000000000 +0200
+++ b/config.h.in	2016-12-26 22:49:43.514203152 +0100
@@ -21,6 +21,9 @@
 /* Define to 1 if you have the `mkfifo' function. */
 #undef HAVE_MKFIFO
 
+/* defined if poll.h is available */
+#undef HAVE_POLL_H
+
 /* Define to 1 if you have the `select' function. */
 #undef HAVE_SELECT
 
@@ -48,6 +51,9 @@
 /* Define to 1 if you have the <sys/file.h> header file. */
 #undef HAVE_SYS_FILE_H
 
+/* defined if sys/poll.h is available */
+#undef HAVE_SYS_POLL_H
+
 /* Define to 1 if you have the <sys/stat.h> header file. */
 #undef HAVE_SYS_STAT_H
 
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-05-18 18:19:41.000000000 +0200
+++ b/configure.ac	2016-12-26 22:48:40.190031280 +0100
@@ -20,7 +20,7 @@
 
 dnl Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS(fcntl.h sys/time.h unistd.h termios.h memory.hi sys/file.h)
+AC_CHECK_HEADERS(fcntl.h sys/time.h unistd.h termios.h memory.hi sys/file.h poll.h sys/poll.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_TYPE_PID_T
