diff -Naur ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e/CMakeLists.txt ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e-2/CMakeLists.txt
--- ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e/CMakeLists.txt	2023-05-30 22:01:29.680358154 +0200
+++ ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e-2/CMakeLists.txt	2023-05-30 22:06:10.316228814 +0200
@@ -393,6 +393,7 @@
 	elseif(APPLE AND NOT CMAKE_CROSSCOMPILING)
 		if(LIBRETRO AND ARM64)
 			set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14")
+           add_definitions(-mno-outline-atomics)
 		else()
 			set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
 		endif()
