diff -Naur ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e/ffmpeg/linux_arm64.sh ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e-2/ffmpeg/linux_arm64.sh
--- ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e/ffmpeg/linux_arm64.sh	2023-06-04 05:30:16.719535828 +0200
+++ ppsspp-845b6b7f3e8060c1ab462f3680e371082ff4253e-2/ffmpeg/linux_arm64.sh	2023-06-04 07:28:16.981119457 +0200
@@ -2,13 +2,29 @@
 
 set -e
 
-GENERAL="\
-   --enable-cross-compile \
-   --extra-libs="-lgcc" \
-   --arch=aarch64 \
-   --cc=aarch64-linux-gnu-gcc \
-   --cross-prefix=aarch64-linux-gnu- \
-   --nm=aarch64-linux-gnu-gcc-nm"
+#GENERAL="\
+#   --enable-cross-compile \
+#   --extra-libs="-lgcc" \
+#   --arch=aarch64 \
+#   --cc=aarch64-linux-gnu-gcc \
+#   --cross-prefix=aarch64-linux-gnu- \
+#   --nm=aarch64-linux-gnu-gcc-nm"
+
+#GENERAL="\
+#             --cpu=\"${TARGET_CPU}\" \
+#              --arch=\"${TARGET_ARCH}\" \
+#              --enable-cross-compile \
+#              --sysinclude=\"${SYSROOT_PREFIX}/usr/include\" \
+#              --nm=\"${NM}\" \
+#              --ar=\"${AR}\" \
+#              --as=\"${CC}\" \
+#              --cc=\"${CC}\" \
+#              --ld=\"${CC}\" \
+#              --host-cc=\"${HOST_CC}\" \
+#              --extra-libs=\"-lgcc\""
+
+
+GENERAL="--cpu="cortex-a57" --arch="aarch64" --enable-cross-compile --sysinclude="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/include" --nm="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-nm" --ar="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-ar" --as="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --cc="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --ld="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --host-cc="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/host-gcc" --extra-libs="-lgcc""
 
 #~ GENERAL="\
    #~ --enable-cross-compile \
@@ -83,10 +99,13 @@
 
 function build_ARM64
 {
+PWD="$(pwd)"
+ORIG_PWD="${PWD}"
+cd ${PKG_BUILD}/ffmpeg
 ./configure --target-os=linux \
     --prefix=./linux/aarch64 \
     ${GENERAL} \
-    --extra-cflags=" -O3 -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 " \
+    --extra-cflags=" -O3 -fasm -Wno-psabi -mno-outline-atomics -fno-short-enums -fno-strict-aliasing -finline-limit=300 " \
     --disable-shared \
     --enable-static \
     --enable-zlib \
@@ -102,8 +121,11 @@
     ${PARSERS}
 make clean
 make install
+cd "${ORIG_PWD}"
+PWD=""
+ORIG_PWD=""
 }
 
 build_ARM64
 
-echo Linux ARM builds finished
+echo Linux ARM64 build finished
