diff -Naur ppsspp-d18dea4f6a8064fef7b8a36c3eaf1012e6b25d9c/ffmpeg/linux_lakka.sh ppsspp-d18dea4f6a8064fef7b8a36c3eaf1012e6b25d9c-2/ffmpeg/linux_lakka.sh
--- ppsspp-d18dea4f6a8064fef7b8a36c3eaf1012e6b25d9c/ffmpeg/linux_lakka.sh	1970-01-01 01:00:00.000000000 +0100
+++ ppsspp-d18dea4f6a8064fef7b8a36c3eaf1012e6b25d9c-2/ffmpeg/linux_lakka.sh	2023-09-30 21:39:12.419941225 +0200
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+set -e
+
+#GENERAL="\
+#   --enable-cross-compile \
+#   --extra-libs="-lgcc" \
+#   --arch=aarch64 \
+#   --cc=aarch64-linux-gnu-gcc \
+#   --cross-prefix=aarch64-linux-gnu- \
+#   --nm=aarch64-linux-gnu-gcc-nm"
+
+GENERAL="\
+             --cpu=${TARGET_CPU} \
+              --arch=${TARGET_ARCH} \
+              --enable-cross-compile \
+              --sysinclude=${SYSROOT_PREFIX}/usr/include \
+              --nm=${NM} \
+              --ar=${AR} \
+              --as=${CC} \
+              --cc=${CC} \
+              --ld=${CC} \
+              --host-cc=${HOST_CC} \
+              --extra-libs=-lgcc"
+
+#GENERAL="--cpu="cortex-a57" --arch="aarch64" --enable-cross-compile --sysinclude="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/include" --nm="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-nm" --ar="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-ar" --as="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --cc="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --ld="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/aarch64-libreelec-linux-gnu-gcc" --host-cc="/home/gavin/Lakka-LibreELEC/build.Lakka-Switch.aarch64-5.0-devel/toolchain/bin/host-gcc" --extra-libs="-lgcc""
+
+#~ GENERAL="\
+   #~ --enable-cross-compile \
+   #~ --extra-libs="-lgcc" \
+   #~ --arch=arm \
+   #~ --cc=arm-linux-gnueabi-gcc \
+   #~ --cross-prefix=arm-linux-gnueabi- \
+   #~ --nm=arm-linux-gnueabi-nm"
+
+MODULES="\
+   --disable-avdevice \
+   --disable-filters \
+   --disable-programs \
+   --disable-network \
+   --disable-avfilter \
+   --disable-postproc \
+   --disable-encoders \
+   --disable-protocols \
+   --disable-hwaccels \
+   --disable-doc"
+
+VIDEO_DECODERS="\
+   --enable-decoder=h264 \
+   --enable-decoder=mpeg4 \
+   --enable-decoder=mpeg2video \
+   --enable-decoder=mjpeg \
+   --enable-decoder=mjpegb"
+
+AUDIO_DECODERS="\
+    --enable-decoder=aac \
+    --enable-decoder=aac_latm \
+    --enable-decoder=atrac3 \
+    --enable-decoder=atrac3p \
+    --enable-decoder=mp3 \
+    --enable-decoder=pcm_s16le \
+    --enable-decoder=pcm_s8"
+  
+DEMUXERS="\
+    --enable-demuxer=h264 \
+    --enable-demuxer=m4v \
+    --enable-demuxer=mpegvideo \
+    --enable-demuxer=mpegps \
+    --enable-demuxer=mp3 \
+    --enable-demuxer=avi \
+    --enable-demuxer=aac \
+    --enable-demuxer=pmp \
+    --enable-demuxer=oma \
+    --enable-demuxer=pcm_s16le \
+    --enable-demuxer=pcm_s8 \
+    --enable-demuxer=wav"
+
+VIDEO_ENCODERS="\
+	  --enable-encoder=huffyuv \
+	  --enable-encoder=ffv1 \
+	  --enable-encoder=mjpeg"
+
+AUDIO_ENCODERS="\
+	  --enable-encoder=pcm_s16le"
+
+MUXERS="\
+  	  --enable-muxer=avi"
+
+
+PARSERS="\
+    --enable-parser=h264 \
+    --enable-parser=mpeg4video \
+    --enable-parser=mpegaudio \
+    --enable-parser=mpegvideo \
+    --enable-parser=aac \
+    --enable-parser=aac_latm"
+
+
+function build_ARM64
+{
+PWD="$(pwd)"
+ORIG_PWD="${PWD}"
+cd ${PKG_BUILD}/ffmpeg
+./configure --target-os=linux \
+    --prefix=./linux/aarch64 \
+    ${GENERAL} \
+    --extra-cflags=" -O3 -fasm -Wno-psabi -mno-outline-atomics -fno-short-enums -fno-strict-aliasing -finline-limit=300 " \
+    --disable-shared \
+    --enable-static \
+    --enable-zlib \
+    --enable-pic \
+    --disable-everything \
+    ${MODULES} \
+    ${VIDEO_DECODERS} \
+    ${AUDIO_DECODERS} \
+    ${VIDEO_ENCODERS} \
+    ${AUDIO_ENCODERS} \
+    ${DEMUXERS} \
+    ${MUXERS} \
+    ${PARSERS}
+make clean
+make install
+cd "${ORIG_PWD}"
+PWD=""
+ORIG_PWD=""
+}
+
+build_ARM64
+
+echo Linux ARM64 build finished
