diff -Naur util-linux-ng-2.16/shlibs/blkid/blkid.pc util-linux-ng-2.16.patch/shlibs/blkid/blkid.pc
--- util-linux-ng-2.16/shlibs/blkid/blkid.pc	2009-07-16 01:18:03.000000000 +0200
+++ util-linux-ng-2.16.patch/shlibs/blkid/blkid.pc	2009-07-19 06:07:08.592182694 +0200
@@ -6,6 +6,6 @@
 Name: blkid
 Description: Block device id library
 Version: 2.16.0
-Requires.private: uuid
+Requires.private:
 Cflags: -I${includedir}/blkid
 Libs: -L${libdir} -lblkid
diff -Naur util-linux-ng-2.16/shlibs/blkid/blkid.pc.in util-linux-ng-2.16.patch/shlibs/blkid/blkid.pc.in
--- util-linux-ng-2.16/shlibs/blkid/blkid.pc.in	2009-07-04 01:20:03.000000000 +0200
+++ util-linux-ng-2.16.patch/shlibs/blkid/blkid.pc.in	2009-07-19 06:07:16.846092573 +0200
@@ -6,6 +6,6 @@
 Name: blkid
 Description: Block device id library
 Version: @LIBBLKID_VERSION@
-Requires.private: uuid
+Requires.private:
 Cflags: -I${includedir}/blkid
 Libs: -L${libdir} -lblkid
