diff -Naur util-linux-ng-2.15/libs/blkid/blkid.pc.in util-linux-ng-2.15.patch/libs/blkid/blkid.pc.in
--- util-linux-ng-2.15/libs/blkid/blkid.pc.in	2009-05-05 21:16:46.000000000 +0200
+++ util-linux-ng-2.15.patch/libs/blkid/blkid.pc.in	2009-05-13 01:31:30.223830997 +0200
@@ -6,6 +6,5 @@
 Name: blkid
 Description: Block device id library
 Version: @LIBBLKID_VERSION@
-Requires.private: uuid
 Cflags: -I${includedir}/blkid
 Libs: -L${libdir} -lblkid
diff -Naur util-linux-ng-2.15/libs/blkid/src/config.c util-linux-ng-2.15.patch/libs/blkid/src/config.c
--- util-linux-ng-2.15/libs/blkid/src/config.c	2009-04-20 11:47:46.000000000 +0200
+++ util-linux-ng-2.15.patch/libs/blkid/src/config.c	2009-05-13 01:23:18.578582126 +0200
@@ -24,9 +24,6 @@
 #include <errno.h>
 #endif
 #include <stdint.h>
-#ifdef HAVE_LIBUUID
-#include <uuid/uuid.h>
-#endif
 #include <stdarg.h>
 
 #include "blkdev.h"
diff -Naur util-linux-ng-2.15/libs/blkid/src/evaluate.c util-linux-ng-2.15.patch/libs/blkid/src/evaluate.c
--- util-linux-ng-2.15/libs/blkid/src/evaluate.c	2009-05-05 21:16:46.000000000 +0200
+++ util-linux-ng-2.15.patch/libs/blkid/src/evaluate.c	2009-05-13 01:23:30.362581321 +0200
@@ -27,9 +27,6 @@
 #include <errno.h>
 #endif
 #include <stdint.h>
-#ifdef HAVE_LIBUUID
-#include <uuid/uuid.h>
-#endif
 #include <stdarg.h>
 
 #include "pathnames.h"
diff -Naur util-linux-ng-2.15/libs/blkid/src/read.c util-linux-ng-2.15.patch/libs/blkid/src/read.c
--- util-linux-ng-2.15/libs/blkid/src/read.c	2009-04-20 11:48:57.000000000 +0200
+++ util-linux-ng-2.15.patch/libs/blkid/src/read.c	2009-05-13 01:23:51.666581008 +0200
@@ -25,7 +25,6 @@
 #endif
 
 #include "blkidP.h"
-#include "uuid/uuid.h"
 
 #ifdef HAVE_STRTOULL
 #define STRTOULL strtoull /* defined in stdlib.h if you try hard enough */
