diff -Naur rtl8192se-0017.0705.2010/rtllib/rtllib.h rtl8192se-0017.0705.2010.patch/rtllib/rtllib.h
--- rtl8192se-0017.0705.2010/rtllib/rtllib.h	2010-05-07 04:31:32.000000000 +0200
+++ rtl8192se-0017.0705.2010.patch/rtllib/rtllib.h	2010-10-10 21:24:50.861572432 +0200
@@ -35,6 +35,7 @@
 #endif
 #include <linux/timer.h>
 #include <linux/sched.h>
+#include <linux/semaphore.h>
 
 #include <linux/delay.h>
 #include <linux/wireless.h>
