diff -Naur orig/HAL/rtl8192/rtl_dm.c rtl8192se-0018.1025.2010/HAL/rtl8192/rtl_dm.c
--- orig/HAL/rtl8192/rtl_dm.c	2010-10-13 07:24:00.000000000 +0200
+++ rtl8192se-0018.1025.2010/HAL/rtl8192/rtl_dm.c	2010-12-04 23:19:57.529063406 +0100
@@ -3826,7 +3826,7 @@
 	if(priv->pwrdown)
 	{
 		u1Tmp = read_nic_byte(dev, 0x06);
-		printk("pwrdown, 0x6(BIT6)=%02x\n", u1Tmp);
+		//printk("pwrdown, 0x6(BIT6)=%02x\n", u1Tmp);
 		retval = (u1Tmp & BIT6) ? eRfOn: eRfOff;
 		
 		if(u1Tmp == 0xff)
