diff -Naur rtl8192se-0019.1207.2010.orig/HAL/rtl8192/rtl_dm.c rtl8192se-0019.1207.2010/HAL/rtl8192/rtl_dm.c
--- rtl8192se-0019.1207.2010.orig/HAL/rtl8192/rtl_dm.c	2010-12-07 04:53:18.000000000 +0100
+++ rtl8192se-0019.1207.2010/HAL/rtl8192/rtl_dm.c	2011-01-21 22:40:35.937060273 +0100
@@ -3834,7 +3834,7 @@
 	if(priv->pwrdown)
 	{
 		u1Tmp = read_nic_byte(dev, 0x06);
-		printk("pwrdown, 0x6(BIT6)=%02x\n", u1Tmp);
+//		printk("pwrdown, 0x6(BIT6)=%02x\n", u1Tmp);
 		retval = (u1Tmp & BIT6) ? eRfOn: eRfOff;
 		
 		if(u1Tmp == 0xff)
