diff -Naur xbmc-pvr-27135/xbmc/lib/libPython/xbmcmodule/player.cpp xbmc-pvr-27135.patch/xbmc/lib/libPython/xbmcmodule/player.cpp
--- xbmc-pvr-27135/xbmc/lib/libPython/xbmcmodule/player.cpp	2010-01-24 11:53:57.108631848 +0100
+++ xbmc-pvr-27135.patch/xbmc/lib/libPython/xbmcmodule/player.cpp	2010-01-24 17:57:58.453446334 +0100
@@ -470,12 +470,15 @@
     char *cLine = NULL;
     if (!PyArg_ParseTuple(args, (char*)"s", &cLine)) return NULL;
 
-    int nStream = g_application.m_pPlayer->GetSubtitleCount();
+    if (g_application.m_pPlayer)
+    {
+      int nStream = g_application.m_pPlayer->GetSubtitleCount();
 
-    g_settings.m_currentVideoSettings.m_SubtitleOn = true;
-    g_application.m_pPlayer->SetSubtitleVisible(true);
-    g_application.m_pPlayer->AddSubtitle(cLine);
-    g_application.m_pPlayer->SetSubtitle(nStream);
+      g_settings.m_currentVideoSettings.m_SubtitleOn = true;
+      g_application.m_pPlayer->SetSubtitleVisible(true);
+      g_application.m_pPlayer->AddSubtitle(cLine);
+      g_application.m_pPlayer->SetSubtitle(nStream);
+    }
 
     Py_INCREF(Py_None);
     return Py_None;
