diff -Naur xbmc-pvr-27135/xbmc/FileItem.cpp xbmc-pvr-27135.patch/xbmc/FileItem.cpp
--- xbmc-pvr-27135/xbmc/FileItem.cpp	2010-01-24 11:53:29.580755057 +0100
+++ xbmc-pvr-27135.patch/xbmc/FileItem.cpp	2010-01-24 18:02:09.346444342 +0100
@@ -33,11 +33,9 @@
 #include "FileSystem/MultiPathDirectory.h"
 #include "FileSystem/MusicDatabaseDirectory.h"
 #include "FileSystem/VideoDatabaseDirectory.h"
-#include "FileSystem/IDirectory.h"
 #include "FileSystem/FactoryDirectory.h"
 #include "MusicInfoTagLoaderFactory.h"
 #include "CueDocument.h"
-#include "utils/fstrcmp.h"
 #include "VideoDatabase.h"
 #include "MusicDatabase.h"
 #include "SortFileItem.h"
@@ -57,7 +55,6 @@
 #include "GUISettings.h"
 #include "AdvancedSettings.h"
 #include "Settings.h"
-#include "utils/TimeUtils.h"
 #include "utils/RegExp.h"
 #include "utils/log.h"
 #include "karaoke/karaokelyricsfactory.h"
diff -Naur xbmc-pvr-27135/xbmc/FileItem.cpp.orig xbmc-pvr-27135.patch/xbmc/FileItem.cpp.orig
diff -Naur xbmc-pvr-27135/xbmc/GUIWindowVideoBase.cpp xbmc-pvr-27135.patch/xbmc/GUIWindowVideoBase.cpp
--- xbmc-pvr-27135/xbmc/GUIWindowVideoBase.cpp	2010-01-24 11:54:22.566755297 +0100
+++ xbmc-pvr-27135.patch/xbmc/GUIWindowVideoBase.cpp	2010-01-24 18:02:09.343444194 +0100
@@ -36,7 +36,6 @@
 #include "Application.h"
 #include "NfoFile.h"
 #include "Picture.h"
-#include "utils/fstrcmp.h"
 #include "PlayListPlayer.h"
 #include "GUIPassword.h"
 #include "FileSystem/ZipManager.h"
diff -Naur xbmc-pvr-27135/xbmc/ProgramDatabase.cpp xbmc-pvr-27135.patch/xbmc/ProgramDatabase.cpp
--- xbmc-pvr-27135/xbmc/ProgramDatabase.cpp	2010-01-24 11:53:30.581257422 +0100
+++ xbmc-pvr-27135.patch/xbmc/ProgramDatabase.cpp	2010-01-24 18:02:09.344444453 +0100
@@ -20,7 +20,6 @@
  */
 
 #include "ProgramDatabase.h"
-#include "utils/fstrcmp.h"
 #include "Util.h"
 #include "GUIWindowFileManager.h"
 #include "FileItem.h"
diff -Naur xbmc-pvr-27135/xbmc/VideoDatabase.cpp xbmc-pvr-27135.patch/xbmc/VideoDatabase.cpp
--- xbmc-pvr-27135/xbmc/VideoDatabase.cpp	2010-01-24 11:53:29.525632072 +0100
+++ xbmc-pvr-27135.patch/xbmc/VideoDatabase.cpp	2010-01-24 18:02:09.342444075 +0100
@@ -21,7 +21,6 @@
 
 #include "VideoDatabase.h"
 #include "GUIWindowVideoBase.h"
-#include "utils/fstrcmp.h"
 #include "utils/RegExp.h"
 #include "utils/GUIInfoManager.h"
 #include "Util.h"
