diff -Naur xbmc-pvr-27135/xbmc/utils/SystemInfo.cpp xbmc-pvr-27135.patch/xbmc/utils/SystemInfo.cpp
--- xbmc-pvr-27135/xbmc/utils/SystemInfo.cpp	2010-01-24 11:54:23.303629749 +0100
+++ xbmc-pvr-27135.patch/xbmc/utils/SystemInfo.cpp	2010-01-24 18:00:16.991476525 +0100
@@ -577,9 +577,27 @@
 #if defined(_LINUX) && !defined(__APPLE__)
 CStdString CSysInfo::GetLinuxDistro()
 {
-  CStdString result = "";
+  static const char* release_file[] = { "/etc/debian_version",
+                                        "/etc/SuSE-release",
+                                        "/etc/mandrake-release",
+                                        "/etc/fedora-release",
+                                        "/etc/redhat-release",
+                                        "/etc/gentoo-release",
+                                        "/etc/slackware-version",
+                                        "/etc/arch-release",
+                                        NULL };
 
   FILE* pipe = popen("unset PYTHONHOME; unset PYTHONPATH; lsb_release -d | cut -f2", "r");
+  
+  for (int i = 0; !pipe && release_file[i]; i++)
+  {
+    CStdString cmd = "cat ";
+    cmd += release_file[i];
+
+    pipe = popen(cmd.c_str(), "r");
+  }
+
+  CStdString result = "";
   if (pipe)
   {
     char buffer[256] = {'\0'};
@@ -589,7 +607,6 @@
       CLog::Log(LOGWARNING, "Unable to determine Linux distribution");
     pclose(pipe);
   }
-
   return result.Trim();
 }
 #endif
