diff -Naur xbmc-pvr-27686/configure.in xbmc-pvr-27686.patch/configure.in
--- xbmc-pvr-27686/configure.in	2010-02-12 09:03:40.000000000 +0100
+++ xbmc-pvr-27686.patch/configure.in	2010-02-12 09:30:08.351016359 +0100
@@ -1121,6 +1121,8 @@
     cp pyconfig.h Include	
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1185,6 +1187,7 @@
       --disable-static \
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --disable-encoders \
@@ -1200,6 +1203,7 @@
       --enable-postproc \
       --enable-gpl \
       `if test "$use_vdpau" = "yes"; then echo --enable-vdpau; else echo --disable-vdpau; fi` \
+      --disable-vaapi \
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
@@ -1219,6 +1223,8 @@
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
@@ -1236,6 +1242,8 @@
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
@@ -1254,6 +1262,8 @@
     cp `pwd`/../libdvdread/src/*.h `pwd`/../includes/dvdread
   else
     ./configure2 \
+      --host=$host_alias \
+      --build=$build_alias \
       --extra-cflags="$CFLAGS $DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
       --enable-static \
@@ -1279,6 +1289,8 @@
       --cc="$CC"
   else
     ./configure2 \
+      --host=$host_alias \
+      --build=$build_alias \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
@@ -1311,6 +1323,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1346,7 +1360,9 @@
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4
   else
     CFLAGS="$CFLAGS -fPIC" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [$USE_EXTERNAL_LIBASS])
 
@@ -1360,6 +1376,8 @@
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       CPPFLAGS="$CPPFLAGS" \
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1386,6 +1404,8 @@
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
     ./configure  \
+      --host=$host_alias \
+      --build=$build_alias \
         CPPFLAGS="$CPPFLAGS" \
         CFLAGS="$CFLAGS -fPIC" \ 
         CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1416,6 +1436,8 @@
       --disable-skyrocket
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       CPPFLAGS="$CPPFLAGS" \
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1435,9 +1457,11 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
+    ./configure --with-ssl-dir=/opt/local/include
   else
-	./configure --without-openssl
+    ./configure --without-openssl \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
@@ -1446,7 +1470,9 @@
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --disable-shared
   else
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
@@ -1456,7 +1482,9 @@
     CXXFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic --disable-shared --enable-static
   else
-    ./configure --with-pic --enable-static
+    ./configure --with-pic --enable-static \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
