diff -Naur xbmc-pvr-26651/configure.in xbmc-pvr-26651.patch/configure.in
--- xbmc-pvr-26651/configure.in	2010-01-11 03:33:19.645347327 +0100
+++ xbmc-pvr-26651.patch/configure.in	2010-01-11 04:20:15.677347493 +0100
@@ -1116,6 +1116,8 @@
 XB_CONFIG_MODULE([xbmc/lib/libPython/Python], [
   if test "$host_vendor" = "apple" ; then
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1138,6 +1140,8 @@
     cp pyconfig.h Include	
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1155,6 +1159,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \    
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 -C \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-asm-optimizations \
       --disable-xmms-plugin \
@@ -1169,7 +1175,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/libvorbis], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1178,7 +1186,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/ogg], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1191,6 +1201,7 @@
       --disable-amd3dnow \
       --disable-static \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-devices \
@@ -1217,6 +1228,7 @@
       --disable-static \
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-decoder=mpeg_xvmc \
@@ -1229,6 +1241,7 @@
       --enable-postproc \
       --enable-gpl \
       `if test "$use_vdpau" = "yes"; then echo --enable-vdpau; else echo --disable-vdpau; fi` \
+      --disable-vaapi \
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
@@ -1242,12 +1255,16 @@
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     LDFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
@@ -1260,11 +1277,15 @@
   if test "$host_vendor" = "apple" ; then
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
@@ -1275,6 +1296,8 @@
   if test "$host_vendor" = "apple" ; then
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -D__DARWIN__ -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1284,6 +1307,8 @@
   else
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1299,6 +1324,8 @@
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1307,6 +1334,8 @@
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1328,6 +1357,10 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1337,6 +1370,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1352,6 +1387,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1360,6 +1397,8 @@
       --without-x
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1373,6 +1412,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic
   else
     true
@@ -1383,17 +1424,23 @@
   if test "$host_vendor" = "apple" ; then
     export MACOSX_DEPLOYMENT_TARGET=10.4
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
 XB_CONFIG_MODULE([xbmc/lib/libass], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias
   else
     CFLAGS="$CFLAGS -fPIC" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [$USE_EXTERNAL_LIBASS])
 
@@ -1402,12 +1449,16 @@
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     CPPFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
   else
     CFLAGS=-O3 \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
@@ -1425,6 +1476,8 @@
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
     ./configure  \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-shared \
       --enable-static \
       --with-pic
@@ -1439,6 +1492,8 @@
       CXXFLAGS="-fPIC -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
       LDFLAGS="-L/opt/local/lib" \
       LIBS="-lz" \
+      --host=$host_alias \
+      --build=$build_alias \
       --x-includes=/usr/X11/include --x-libraries=/usr/X11/lib \
       --without-xscreensaver \
       --disable-sound \
@@ -1454,6 +1509,8 @@
     ./configure \
       CFLAGS="-fPIC" \ 
       CXXFLAGS="-fPIC" \
+      --host=$host_alias \
+      --build=$build_alias \
       --without-xscreensaver \
       --disable-sound \
       --disable-cyclone \
@@ -1469,9 +1526,13 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
-  else
-	./configure
+      ./configure --with-ssl-dir=/opt/local/include \
+      --host=$host_alias \
+      --build=$build_alias
+  else
+      ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
diff -Naur xbmc-pvr-26651/configure.in.orig xbmc-pvr-26651.patch/configure.in.orig
