diff -Naur xbmc-pvr-28368/configure.in xbmc-pvr-28368.patch/configure.in
--- xbmc-pvr-28368/configure.in	2010-03-04 14:12:37.000000000 +0100
+++ xbmc-pvr-28368.patch/configure.in	2010-03-04 14:55:05.805134767 +0100
@@ -1161,7 +1161,9 @@
     cp pyconfig.h Include	
   else
     ./configure \
-      --enable-ipv6 \
+      --host=$host_alias \
+      --build=$build_alias \
+      --disable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
       --enable-shared \
@@ -1226,6 +1228,7 @@
       --disable-static \
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --enable-muxer=adts \
@@ -1242,6 +1245,7 @@
       --enable-postproc \
       --enable-gpl \
       `if test "$use_vdpau" = "yes"; then echo --enable-vdpau; else echo --disable-vdpau; fi` \
+      --disable-vaapi \
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
@@ -1261,6 +1265,8 @@
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
@@ -1278,6 +1284,8 @@
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
@@ -1296,6 +1304,8 @@
     cp `pwd`/../libdvdread/src/*.h `pwd`/../includes/dvdread
   else
     ./configure2 \
+      --host=$host_alias \
+      --build=$build_alias \
       --extra-cflags="$CFLAGS $DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
       --prefix="${prefix}" --includedir="${includedir}" --libdir="${libdir}" --datadir="${datadir}" \
       --enable-static \
@@ -1321,6 +1331,8 @@
       --cc="$CC"
   else
     ./configure2 \
+      --host=$host_alias \
+      --build=$build_alias \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
@@ -1353,6 +1365,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1388,7 +1402,9 @@
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4
   else
     CFLAGS="$CFLAGS -fPIC" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [$USE_EXTERNAL_LIBASS])
 
@@ -1402,6 +1418,8 @@
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       CPPFLAGS="$CPPFLAGS" \
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1428,6 +1446,8 @@
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
     ./configure  \
+      --host=$host_alias \
+      --build=$build_alias \
         CPPFLAGS="$CPPFLAGS" \
         CFLAGS="$CFLAGS -fPIC" \ 
         CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1458,6 +1478,8 @@
       --disable-skyrocket
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       CPPFLAGS="$CPPFLAGS" \
       CFLAGS="$CFLAGS -fPIC" \ 
       CXXFLAGS="$CXXFLAGS -fPIC" \
@@ -1477,9 +1499,11 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
+    ./configure --with-ssl-dir=/opt/local/include
   else
-	./configure --without-openssl
+    ./configure --without-openssl \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
@@ -1488,7 +1512,9 @@
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --disable-shared
   else
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
