diff -Naur xbmc-26784/configure.in xbmc-26784.patch/configure.in
--- xbmc-26784/configure.in	2010-01-14 06:01:36.002732210 +0100
+++ xbmc-26784.patch/configure.in	2010-01-14 07:36:53.996945233 +0100
@@ -1125,6 +1125,8 @@
 XB_CONFIG_MODULE([xbmc/lib/libPython/Python], [
   if test "$host_vendor" = "apple" ; then
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1147,6 +1149,8 @@
     cp pyconfig.h Include	
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-ipv6 \
       --enable-unicode=ucs4 \
       --without-cxx \
@@ -1164,6 +1168,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \    
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 -C \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-asm-optimizations \
       --disable-xmms-plugin \
@@ -1178,7 +1184,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/libvorbis], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1187,7 +1195,9 @@
 XB_CONFIG_MODULE([xbmc/cores/paplayer/vorbisfile/ogg], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic \
+      --host=$host_alias \
+      --build=$build_alias
   else
     true
   fi
@@ -1200,6 +1210,7 @@
       --disable-amd3dnow \
       --disable-static \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-devices \
@@ -1226,6 +1237,7 @@
       --disable-static \
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
+      `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-decoder=mpeg_xvmc \
@@ -1238,6 +1250,7 @@
       --enable-postproc \
       --enable-gpl \
       `if test "$use_vdpau" = "yes"; then echo --enable-vdpau; else echo --disable-vdpau; fi` \
+      --disable-vaapi \
       --enable-libfaad \
       --enable-protocol=http \
       --enable-pthreads \
@@ -1251,12 +1264,16 @@
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     LDFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
       --disable-win
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-oss \
       --disable-solaris-audio \
       --disable-al-audio \
@@ -1269,11 +1286,15 @@
   if test "$host_vendor" = "apple" ; then
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-doc \
       --enable-static \
       --with-pic
@@ -1284,6 +1305,8 @@
   if test "$host_vendor" = "apple" ; then
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -D__DARWIN__ -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1293,6 +1316,8 @@
   else
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../libdvdcss/src" \
+      --host=$host_alias \
+      --build=$build_alias \
       --enable-static \
       --disable-shared \
       --cc="$CC" &&
@@ -1308,6 +1333,8 @@
     ./configure2 MACOSX_DEPLOYMENT_TARGET=10.4 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1316,6 +1343,8 @@
     ./configure2 \
       --extra-cflags="$DVDREAD_CFLAGS -fPIC -DPIC -I`pwd`/../includes" \
       --extra-ldflags="-L`pwd`/../libdvdread/obj" \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-dvdread-config="`pwd`/../libdvdread/obj/dvdread-config" \
       --enable-static \
       --disable-shared \
@@ -1337,6 +1366,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1346,6 +1377,8 @@
       --disable-al-audio
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1361,6 +1394,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1369,6 +1404,8 @@
       --without-x
   else
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic \
       --disable-static \
       --enable-shared \
@@ -1382,6 +1419,8 @@
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --with-pic
   else
     true
@@ -1392,17 +1431,23 @@
   if test "$host_vendor" = "apple" ; then
     export MACOSX_DEPLOYMENT_TARGET=10.4
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
 XB_CONFIG_MODULE([xbmc/lib/libass], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="-fPIC -fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4
+    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias
   else
     CFLAGS="$CFLAGS -fPIC" \
-    ./configure
+    ./configure \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [$USE_EXTERNAL_LIBASS])
 
@@ -1411,12 +1456,16 @@
     CFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     CPPFLAGS="-isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4 -fno-common" \
     ./configure MACOSX_DEPLOYMENT_TARGET=10.4 \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
   else
     CFLAGS=-O3 \
     ./configure \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-static \
       --disable-debugging \
       --with-pic
@@ -1434,6 +1483,8 @@
 XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   if test "$host_vendor" != "apple" ; then
     ./configure  \
+      --host=$host_alias \
+      --build=$build_alias \
       --disable-shared \
       --enable-static \
       --with-pic
@@ -1448,6 +1499,8 @@
       CXXFLAGS="-fPIC -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4" \
       LDFLAGS="-L/opt/local/lib" \
       LIBS="-lz" \
+      --host=$host_alias \
+      --build=$build_alias \
       --x-includes=/usr/X11/include --x-libraries=/usr/X11/lib \
       --without-xscreensaver \
       --disable-sound \
@@ -1463,6 +1516,8 @@
     ./configure \
       CFLAGS="-fPIC" \ 
       CXXFLAGS="-fPIC" \
+      --host=$host_alias \
+      --build=$build_alias \
       --without-xscreensaver \
       --disable-sound \
       --disable-cyclone \
@@ -1478,9 +1533,13 @@
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
-	./configure --with-ssl-dir=/opt/local/include
-  else
-	./configure --without-openssl
+    ./configure --with-ssl-dir=/opt/local/include \
+      --host=$host_alias \
+      --build=$build_alias
+  else
+    ./configure --without-openssl \
+      --host=$host_alias \
+      --build=$build_alias
   fi
 ], [0])
 
diff -Naur xbmc-26784/configure.in.orig xbmc-26784.patch/configure.in.orig
diff -Naur xbmc-26784/configure.in.rej xbmc-26784.patch/configure.in.rej
