diff -Naur xbmc-pvr-27686/configure.in xbmc-pvr-27686.patch/configure.in
--- xbmc-pvr-27686/configure.in	2010-02-12 10:07:39.181174683 +0100
+++ xbmc-pvr-27686.patch/configure.in	2010-02-12 10:06:08.192017722 +0100
@@ -35,6 +35,8 @@
 goom_disabled="== GOOM disabled. =="
 xbmcprojectm_enabled="== XBMCProjectM enabled. =="
 xbmxprojectm_disabled="== XBMCProjectM disabled. =="
+rsxs_enabled="== RSXS enabled. =="
+rsxs_disabled="== RSXS disabled. =="
 pulse_not_found="== Could not find libpulse. PulseAudio support disabled. =="
 pulse_disabled="== PulseAudio support manually disabled. =="
 faac_not_found="== Could not find libfaac. FAAC support disabled. =="
@@ -151,6 +153,12 @@
             [use_xbmcprojectm=$enableval],
             [use_xbmcprojectm=yes])
 
+AC_ARG_ENABLE([rsxs],
+            [AS_HELP_STRING([--enable-rsxs],
+               [enable RSXS Screensaver (default is yes)])],
+            [use_rsxs=$enableval],
+            [use_rsxs=yes])
+
 AC_PATH_PROG(CCACHE,ccache,none)
 AC_ARG_ENABLE([ccache],
             [AS_HELP_STRING([--enable-ccache],
@@ -522,6 +530,15 @@
   BUILD_XBMCPROJECTM=1
 fi
 
+# RSXS
+if test "$use_rsxs" = "yes"; then
+  AC_MSG_NOTICE($rsxs_enabled)
+  BUILD_RSXS=0
+else
+  AC_MSG_NOTICE($rsxs_disabled)
+  BUILD_RSXS=1
+fi
+
 ### External libraries checks
 if test "$use_external_libraries" = "yes"; then
   AC_MSG_NOTICE($external_libraries_enabled)
@@ -850,6 +867,12 @@
   final_message="$final_message\n  XBMCProjectM:\tNo"
 fi
 
+if test "$use_rsxs" = "yes"; then
+  final_message="$final_message\n  RSXS:\t\tYes"
+else
+  final_message="$final_message\n  RSXS:\t\tNo"
+fi
+
 if test "$use_pcre" = "yes"; then
   final_message="$final_message\n  PCRE Support:\tYes"
 else
@@ -1081,6 +1104,7 @@
 AC_SUBST(BUILD_DVDCSS)
 AC_SUBST(BUILD_GOOM)
 AC_SUBST(BUILD_XBMCPROJECTM)
+AC_SUBST(BUILD_RSXS)
 AC_SUBST(USE_EXTERNAL_LIBRARIES)
 AC_SUBST(USE_EXTERNAL_FFMPEG)
 AC_SUBST(USE_EXTERNAL_LIBA52)
@@ -1477,7 +1501,7 @@
       --disable-lattice \
       --disable-skyrocket
   fi
-], [0])
+], [$BUILD_RSXS])
 
 XB_CONFIG_MODULE([xbmc/cores/dvdplayer/Codecs/libbdnav/], [
   if test "$host_vendor" = "apple" ; then
diff -Naur xbmc-pvr-27686/Makefile.in xbmc-pvr-27686.patch/Makefile.in
--- xbmc-pvr-27686/Makefile.in	2010-02-12 10:07:39.182172289 +0100
+++ xbmc-pvr-27686.patch/Makefile.in	2010-02-12 10:03:25.738140979 +0100
@@ -101,8 +101,10 @@
 	lib/libmodplug \
 	lib/xbmc-dll-symbols
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
+ifneq (@BUILD_RSXS@,1)
+SS_DIRS=+xbmc/screensavers/rsxs-0.9/xbmc
+endif
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
@@ -305,7 +307,9 @@
 	$(MAKE) -C xbmc/pvrclients/vdr-streamdev
 	$(MAKE) -C xbmc/pvrclients/tvheadend
 screensavers: exports
+ifneq (@BUILD_RSXS@,1)
 	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
+endif
 libaddon: exports
 	$(MAKE) -C xbmc/addons/lib/addonlib++
 	$(MAKE) -C xbmc/addons/lib/pvrlib++
