diff -Naur xbmc-27135/guilib/Makefile.in xbmc-27135.patch/guilib/Makefile.in
--- xbmc-27135/guilib/Makefile.in	2010-01-24 11:46:42.066643776 +0100
+++ xbmc-27135.patch/guilib/Makefile.in	2010-01-24 13:14:01.800579073 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-I../xbmc/cores/dvdplayer/Codecs/ffmpeg
 endif
diff -Naur xbmc-27135/xbmc/FileSystem/Makefile.in xbmc-27135.patch/xbmc/FileSystem/Makefile.in
--- xbmc-27135/xbmc/FileSystem/Makefile.in	2010-01-24 11:47:36.842631867 +0100
+++ xbmc-27135.patch/xbmc/FileSystem/Makefile.in	2010-01-24 13:14:59.573578187 +0100
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 ifeq ($(findstring osx,$(ARCH)), osx)
 INCLUDES+=-I../../lib/libcurl-OSX/include
diff -Naur xbmc-27135/xbmc/Makefile.in xbmc-27135.patch/xbmc/Makefile.in
--- xbmc-27135/xbmc/Makefile.in	2010-01-24 11:47:03.973632064 +0100
+++ xbmc-27135.patch/xbmc/Makefile.in	2010-01-24 13:14:32.079579280 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer
+INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer
 
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
   INCLUDES+=-Icores/dvdplayer/Codecs/ffmpeg
