diff -Naur xbmc-pvr-26651/guilib/Makefile xbmc-pvr-26651.patch/guilib/Makefile
--- xbmc-pvr-26651/guilib/Makefile	2010-01-11 03:33:19.802350966 +0100
+++ xbmc-pvr-26651.patch/guilib/Makefile	2010-01-11 04:38:11.069345520 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores/dvdplayer/Codecs/ffmpeg -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores/dvdplayer/Codecs/ffmpeg -I../xbmc/linux -I../xbmc/utils -I../xbmc/lib/libsquish
 
 SRCS=AnimatedGif.cpp \
      AudioContext.cpp \
diff -Naur xbmc-pvr-26651/xbmc/FileSystem/Makefile xbmc-pvr-26651.patch/xbmc/FileSystem/Makefile
--- xbmc-pvr-26651/xbmc/FileSystem/Makefile	2010-01-11 03:41:38.093347308 +0100
+++ xbmc-pvr-26651.patch/xbmc/FileSystem/Makefile	2010-01-11 04:38:11.069345520 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I../lib
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I../lib
 INCLUDES+=-I../lib/libcdio/libcdio/include
 
 CXXFLAGS+=-D__STDC_FORMAT_MACROS \
diff -Naur xbmc-pvr-26651/xbmc/Makefile xbmc-pvr-26651.patch/xbmc/Makefile
--- xbmc-pvr-26651/xbmc/Makefile	2010-01-11 03:34:23.448346973 +0100
+++ xbmc-pvr-26651.patch/xbmc/Makefile	2010-01-11 04:38:11.076345842 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/dvdplayer/Codecs/ffmpeg
+INCLUDES+=-I../ -I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer -Icores/dvdplayer/Codecs/ffmpeg
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
