diff -Naur xbmc-pvr-26651/configure.in xbmc-pvr-26651.patch/configure.in
--- xbmc-pvr-26651/configure.in	2010-01-11 04:37:50.978472580 +0100
+++ xbmc-pvr-26651.patch/configure.in	2010-01-11 04:38:58.055345557 +0100
@@ -1573,6 +1573,7 @@
       --build=$build_alias
   else
       ./configure \
+      `if test "$cross_compiling" = "yes"; then echo ac_cv_openssl_linker_option=no; fi` \
       --host=$host_alias \
       --build=$build_alias
   fi
diff -Naur xbmc-pvr-26651/configure.in.orig xbmc-pvr-26651.patch/configure.in.orig
