diff -Naur xbmc-23782/Makefile.include.in xbmc-23782.patch/Makefile.include.in
--- xbmc-23782/Makefile.include.in	2009-10-17 07:12:34.000000000 +0200
+++ xbmc-23782.patch/Makefile.include.in	2009-10-18 00:22:42.195445274 +0200
@@ -11,6 +11,7 @@
 CXXFLAGS+=@DEBUG_FLAGS@
 CFLAGS+=@CFLAGS@
 CFLAGS+=@DEBUG_FLAGS@
+LDFLAGS+=@LDFLAGS@
 INCLUDES+=@INCLUDES@
 DEFINES+= \
 	-D_LINUX \
@@ -33,7 +34,7 @@
 
 .cpp.o:
 	@rm -f ${<:.cpp=.o}
-	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.cpp=.o}
+	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.cpp=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
@@ -41,7 +42,7 @@
 
 .cc.o:
 	@rm -f ${<:.cc=.o}
-	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.cc=.o}
+	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.cc=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
@@ -49,7 +50,7 @@
 
 .c.o:
 	@rm -f ${<:.c=.o}
-	$(CC) -MD -c $(CFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.c=.o}
+	$(CC) -MD -c $(CFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.c=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
@@ -57,7 +58,7 @@
 
 .C.o:
 	@rm -f ${<:.C=.o}
-	$(CC) -MD -c $(CFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.C=.o}
+	$(CC) -MD -c $(CFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.C=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
@@ -65,14 +66,14 @@
 
 .S.o:
 	rm -f ${<:.S=.o}
-	$(CC) -c $(CFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.S=.o}
+	$(CC) -c $(CFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.S=.o}
 
 %.h.gch : %.h
-	$(CXX) -c $(CFLAGS) $(DEFINES) $(INCLUDES) $< -o $@
+	$(CXX) -c $(CFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o $@
 
 .m.o:
 	@rm -f ${<:.m=.o}
-	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.m=.o}
+	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.m=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
@@ -80,7 +81,7 @@
 
 .mm.o:
 	@rm -f ${<:.mm=.o}
-	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $< -o ${<:.mm=.o}
+	$(CXX) -MD -c $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(LDFLAGS) $< -o ${<:.mm=.o}
 	@cp $*.d $*.P; \
 	  sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 	      -e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $*.P; \
