diff -Naur xbmc-pvr-26651/configure.in xbmc-pvr-26651.patch/configure.in
--- xbmc-pvr-26651/configure.in	2010-01-11 04:39:36.433347449 +0100
+++ xbmc-pvr-26651.patch/configure.in	2010-01-11 04:39:55.195347999 +0100
@@ -39,6 +39,8 @@
 faac_disabled="== FAAC support manually disabled. =="
 dvdcss_enabled="== DVDCSS support enabled. =="
 dvdcss_disabled="== DVDCSS support disabled. =="
+hal_not_found="== Could not find hal and hal-storage. HAL support disabled. =="
+hal_disabled="== HAL support disabled. =="
 avahi_not_found="== Could not find libavahi-common or libavahi-client. Avahi support disabled. =="
 avahi_disabled="== Avahi support disabled. =="
 vdpau_not_found="== Could not find libvdpau. VDPAU support disabled. =="
@@ -195,6 +197,11 @@
                [enable MID support (default is no)])],
             [use_mid=$enableval],
             [use_mid=no])
+AC_ARG_ENABLE([hal],
+             [AS_HELP_STRING([--disable-hal],
+                [disable HAL support (default is enabled if hal and hal-storage is found)])],
+             [use_hal=$enableval],
+             [use_hal=yes])
 AC_ARG_ENABLE([avahi],
              [AS_HELP_STRING([--disable-avahi],
                 [disable Avahi support (default is enabled if libavahi-common and libavahi-client is found)])],
@@ -446,15 +453,9 @@
   PKG_CHECK_MODULES([DBUS],         [dbus-1],
                     [INCLUDES+=" $DBUS_CFLAGS"; LIBS+=" $DBUS_LIBS"],
                     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([HAL_STORAGE],  [hal-storage],
-                    [INCLUDES+=" $HAL_STORAGE_CFLAGS"; LIBS+=" $HAL_STORAGE_LIBS"], 
-                    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([SDL],          [sdl],
                     [INCLUDES+=" $SDL_CFLAGS"; LIBS+=" $SDL_LIBS"],
                     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([HAL],          [hal],
-                    [INCLUDES+=" $HAL_CFLAGS"; LIBS+=" $HAL_LIBS"],
-                    AC_MSG_ERROR($missing_library))
 fi
 
 # PulseAudio
@@ -483,6 +484,24 @@
   fi
 fi
 
+# HAL
+if test "$host_vendor" = "apple" ; then
+  use_hal="no"
+  AC_MSG_RESULT($hal_disabled)
+else
+  if test "$use_hal" = "yes"; then
+    PKG_CHECK_MODULES([HAL],  [hal],
+                    [INCLUDES+=" $HAL_CFLAGS"; LIBS+=" $HAL_LIBS"],
+                    use_hal=no;AC_MSG_ERROR($missing_library))
+    PKG_CHECK_MODULES([HAL_STORAGE],  [hal-storage],
+                    [INCLUDES+=" $HAL_STORAGE_CFLAGS"; LIBS+=" $HAL_STORAGE_LIBS"],
+                    use_hal=no;AC_MSG_ERROR($missing_library))
+    AC_DEFINE([HAS_HAL], [1], [Define to 1 if you have HAL installed])
+  else
+AC_MSG_RESULT($hal_disabled)
+  fi
+fi
+
 # avahi
 if test "$host_vendor" = "apple" ; then
   use_avahi="no"
diff -Naur xbmc-pvr-26651/configure.in.orig xbmc-pvr-26651.patch/configure.in.orig
diff -Naur xbmc-pvr-26651/guilib/system.h xbmc-pvr-26651.patch/guilib/system.h
--- xbmc-pvr-26651/guilib/system.h	2010-01-11 03:33:19.672356894 +0100
+++ xbmc-pvr-26651.patch/guilib/system.h	2010-01-11 04:39:55.196347910 +0100
@@ -116,7 +116,6 @@
 #define HAS_AVAHI
 #endif
 #define HAS_LCD
-#define HAS_HAL
 #define HAS_DBUS
 #define HAS_DBUS_SERVER
 #define HAS_GL
diff -Naur xbmc-pvr-26651/guilib/system.h.orig xbmc-pvr-26651.patch/guilib/system.h.orig
