diff -Naur xbmc-23757/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h xbmc-23757.patch/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h
--- xbmc-23757/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h	2009-10-16 18:44:08.000000000 +0200
+++ xbmc-23757.patch/xbmc/cores/DllLoader/exports/util/EmuFileWrapper.h	2009-10-16 20:08:13.747643850 +0200
@@ -29,10 +29,14 @@
 
 class CMutex;
 
-#if defined(_LINUX) && !defined(__APPLE__)
+#if defined(_LINUX) && !defined(__APPLE__) && !defined(__UCLIBC__)
 #define _file _fileno
 #endif
 
+#if defined(__UCLIBC__)
+#define _file __filedes
+#endif
+
 #define MAX_EMULATED_FILES    50
 #define FILE_WRAPPER_OFFSET   0x00000100
 
