diff -Naur xbmc-23782/configure.in xbmc-23782.patch/configure.in
--- xbmc-23782/configure.in	2009-10-17 07:13:04.000000000 +0200
+++ xbmc-23782.patch/configure.in	2009-10-18 19:30:30.994320947 +0200
@@ -327,6 +327,15 @@
 AC_CHECK_LIB([lzo2],        [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([z],           [main],, AC_MSG_ERROR($missing_library))
 
+AC_CHECK_LIB([pthread],[pthread_yield], 
+            AC_DEFINE([HAVE_PTHREAD_YIELD], [1],
+            [Define to 1 if you have the 'pthread_yield' function]))
+
+AC_CHECK_LIB([pthread],[sched_yield],
+            AC_DEFINE([HAVE_SCHED_YIELD], [1],
+            [Define to 1 if you have the 'sched_yield' function]))
+
+
 PKG_CHECK_MODULES([MAD],        [mad],
                   [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
                   AC_MSG_ERROR($missing_library))
diff -Naur xbmc-23782/xbmc/FileSystem/FileCurl.cpp xbmc-23782.patch/xbmc/FileSystem/FileCurl.cpp
--- xbmc-23782/xbmc/FileSystem/FileCurl.cpp	2009-10-17 07:09:13.000000000 +0200
+++ xbmc-23782.patch/xbmc/FileSystem/FileCurl.cpp	2009-10-18 19:24:45.938445349 +0200
@@ -19,6 +19,8 @@
  *
  */
 
+#include "config.h"
+
 #include "FileCurl.h"
 #include "Util.h"
 #include "URL.h"
@@ -1178,10 +1180,10 @@
         // happens especially on ftp during initial connection
 #ifndef _LINUX
         SwitchToThread();
-#elif __APPLE__
-        sched_yield();
-#else
+#elif HAVE_PTHREAD_YIELD
         pthread_yield();
+#elif HAVE_SCHED_YIELD
+        sched_yield();
 #endif
 
         FD_ZERO(&fdread);
