diff -Naur xbmc-23782/configure.in xbmc-23782.patch/configure.in
--- xbmc-23782/configure.in	2009-10-18 19:54:36.828319917 +0200
+++ xbmc-23782.patch/configure.in	2009-10-18 19:55:57.901319613 +0200
@@ -335,6 +335,9 @@
             AC_DEFINE([HAVE_SCHED_YIELD], [1],
             [Define to 1 if you have the 'sched_yield' function]))
 
+AC_CHECK_LIB([pthread],[pthread_getcpuclockid],
+            AC_DEFINE([HAVE_PTHREAD_GETCPUCLOCKID], [1],
+            [Define to 1 if you have the 'pthread_getcpuclockid' function]))
 
 PKG_CHECK_MODULES([MAD],        [mad],
                   [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
diff -Naur xbmc-23782/xbmc/linux/XThreadUtils.cpp xbmc-23782.patch/xbmc/linux/XThreadUtils.cpp
--- xbmc-23782/xbmc/linux/XThreadUtils.cpp	2009-10-17 07:09:25.000000000 +0200
+++ xbmc-23782.patch/xbmc/linux/XThreadUtils.cpp	2009-10-18 20:01:02.496322443 +0200
@@ -19,6 +19,8 @@
  *
  */
 
+#include "config.h"
+
 #include "PlatformDefs.h"
 #include "XHandle.h"
 #include "XThreadUtils.h"
@@ -189,7 +191,7 @@
     if (lpKernelTime)
       lpKernelTime->dwLowDateTime = lpKernelTime->dwHighDateTime = 0;
   }
-#elif _POSIX_THREAD_CPUTIME != -1
+#elif HAVE_PTHREAD_GETCPUCLOCKID && _POSIX_THREAD_CPUTIME != -1 
     if(lpUserTime)
     {
       lpUserTime->dwLowDateTime = 0;
