diff -Naur xbmc-23998/xbmc/Settings.h xbmc-23998.patch/xbmc/Settings.h
--- xbmc-23998/xbmc/Settings.h	2009-10-25 09:55:25.000000000 +0100
+++ xbmc-23998.patch/xbmc/Settings.h	2009-10-28 07:50:19.663615696 +0100
@@ -27,7 +27,7 @@
 #define DEFAULT_VSYNC       VSYNC_DISABLED
 #define DEFAULT_THUMB_SIZE  256
 #else  // MID
-#define DEFAULT_SKIN        "PM3.HD"
+#define DEFAULT_SKIN        "PM3.HD"
 #ifdef __APPLE__
 #define DEFAULT_VSYNC       VSYNC_ALWAYS
 #else
