diff -Naur xbmc-23998/configure.in xbmc-23998.patch/configure.in
--- xbmc-23998/configure.in	2009-10-25 13:30:40.869660379 +0100
+++ xbmc-23998.patch/configure.in	2009-10-25 13:33:30.376657368 +0100
@@ -692,6 +692,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1177,6 +1178,7 @@
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
       `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-decoder=mpeg_xvmc \
