diff -Naur xbmc-pvr-26651/configure.in xbmc-pvr-26651.patch/configure.in
--- xbmc-pvr-26651/configure.in	2010-01-11 04:40:33.487469992 +0100
+++ xbmc-pvr-26651.patch/configure.in	2010-01-11 04:42:38.159472611 +0100
@@ -770,6 +770,7 @@
   fi
 fi
 
+AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 AC_CHECK_PROG(HAVE_SVNVERSION,svnversion,"yes","no",)
 AC_CHECK_PROG(HAVE_GIT,git,"yes","no",)
 
@@ -1290,6 +1291,7 @@
       --disable-ipv6 \
       `if test "$use_debug" = "no"; then echo --disable-debug; fi` \
       `if test "$cross_compiling" = "yes"; then echo --enable-cross-compile; fi` \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --disable-encoders \
       --disable-decoder=mpeg_xvmc \
diff -Naur xbmc-pvr-26651/configure.in.orig xbmc-pvr-26651.patch/configure.in.orig
diff -Naur xbmc-pvr-26651/guilib/system.h.orig xbmc-pvr-26651.patch/guilib/system.h.orig
