diff -Naur xbmc-pvr-28391/configure.in xbmc-pvr-28391.patch/configure.in
--- xbmc-pvr-28391/configure.in	2010-03-06 13:44:06.000000000 +0100
+++ xbmc-pvr-28391.patch/configure.in	2010-03-07 23:26:55.998017036 +0100
@@ -438,9 +438,6 @@
   PKG_CHECK_MODULES([XMU],          [xmu],
                     [INCLUDES+=" $XMU_CFLAGS"; LIBS+=" $XMU_LIBS"],
                     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],     [xinerama],
-                    [INCLUDES+=" $XINERAMA_CFLAGS"; LIBS+=" $XINERAMA_LIBS"],
-                    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],         [libcurl],
                     [INCLUDES+=" $CURL_CFLAGS"; LIBS+=" $CURL_LIBS"],
                     AC_MSG_ERROR($missing_library))
