diff -Naur xbmc-f3b0020/xbmc/peripherals/Peripherals.cpp xbmc-f3b0020.patch/xbmc/peripherals/Peripherals.cpp
--- xbmc-f3b0020/xbmc/peripherals/Peripherals.cpp	2011-12-10 22:16:17.000000000 +0100
+++ xbmc-f3b0020.patch/xbmc/peripherals/Peripherals.cpp	2011-12-11 18:25:13.143643374 +0100
@@ -19,6 +19,7 @@
  *
  */
 
+#include "system.h"
 #include "Peripherals.h"
 #include "bus/PeripheralBus.h"
 #include "devices/PeripheralBluetooth.h"
