diff -Naur xbmc-11.0.1/xbmc/threads/platform/pthreads/ThreadImpl.h xbmc-11.0.1.patch/xbmc/threads/platform/pthreads/ThreadImpl.h
--- xbmc-11.0.1/xbmc/threads/platform/pthreads/ThreadImpl.h	2012-03-23 07:22:04.215885089 +0100
+++ xbmc-11.0.1.patch/xbmc/threads/platform/pthreads/ThreadImpl.h	2012-03-23 07:24:07.875383186 +0100
@@ -22,6 +22,7 @@
 #pragma once
 
 #include <pthread.h>
+#include <unistd.h>
 
 struct threadOpaque
 {
