diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:50:57.045376680 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:52:57.912501212 +0200
@@ -585,9 +585,6 @@
   PKG_CHECK_MODULES([XMU],   [xmu],
     [INCLUDES="$INCLUDES $XMU_CFLAGS"; LIBS="$LIBS $XMU_LIBS"],
     AC_MSG_ERROR($missing_library))
-  PKG_CHECK_MODULES([XINERAMA],[xinerama],
-    [INCLUDES="$INCLUDES $XINERAMA_CFLAGS"; LIBS="$LIBS $XINERAMA_LIBS"],
-    AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],  [libcurl],
     [INCLUDES="$INCLUDES $CURL_CFLAGS"; LIBS="$LIBS $CURL_LIBS"],
     AC_MSG_ERROR($missing_library))
