From 2f91d0ca3735ab9b1abbfe6e3c6613e56951b7d2 Mon Sep 17 00:00:00 2001
From: wsoltys <wiso@xbmc.org>
Date: Mon, 2 Jan 2012 11:13:02 +0100
Subject: [PATCH] fixed: Linux compiler error (hopefully)

---
 xbmc/linux/PlatformDefs.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/linux/PlatformDefs.h b/xbmc/linux/PlatformDefs.h
index aaed576..33d1c96 100644
--- a/xbmc/linux/PlatformDefs.h
+++ b/xbmc/linux/PlatformDefs.h
@@ -428,8 +428,8 @@ struct _stati64 {
   DWORD dwLength;
   DWORD dwMemoryLoad;
 
-  uint64_t dwTotalPhys;
-  uint64_t dwAvailPhys;
+  uint64_t ullTotalPhys;
+  uint64_t ullAvailPhys;
   uint64_t ullTotalPageFile;
   uint64_t ullAvailPageFile;
   uint64_t ullTotalVirtual;
-- 
1.7.5.4

