diff -Naur xbmc-10.0-Dharma/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp xbmc-10.0-Dharma.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp
--- xbmc-10.0-Dharma/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2010-12-17 20:14:04.000000000 +0100
+++ xbmc-10.0-Dharma.patch/xbmc/cores/dvdplayer/DVDCodecs/Video/CrystalHD.cpp	2011-01-21 13:52:11.238550146 +0100
@@ -773,7 +773,7 @@
           if (!pBuffer)
           {
             // No free pre-allocated buffers so make one
-#ifdef _WIN32
+#if 1
             // force Windows to use YV12 until DX renderer gets NV12 or YUY2 capability.
             pBuffer = new CPictureBuffer(DVDVideoPicture::FMT_YUV420P, m_width, m_height);
 #else
