diff -Naur xbmc-10.0-Dharma/configure.in xbmc-10.0-Dharma.patch/configure.in
--- xbmc-10.0-Dharma/configure.in	2010-12-17 20:30:53.000000000 +0100
+++ xbmc-10.0-Dharma.patch/configure.in	2011-01-06 03:41:05.387941174 +0100
@@ -546,8 +546,6 @@
 AC_CHECK_LIB([pthread],     [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([lzo2],        [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([z],           [main],, AC_MSG_ERROR($missing_library))
-AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
-AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([wavpack],     [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([mpeg2],       [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([mpeg2convert],[main],, AC_MSG_ERROR($missing_library))
@@ -940,6 +938,8 @@
   USE_EXTERNAL_PYTHON=1
   AC_DEFINE([USE_EXTERNAL_PYTHON], [1], [Whether to use external python library.])
 else
+  AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
+  AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
   AC_MSG_NOTICE($external_python_disabled)
   USE_EXTERNAL_PYTHON=0
 fi
