From 895d659a6accdae7eb668eebe9426e4820f5928a Mon Sep 17 00:00:00 2001
From: vdrfan <vdrfan-nospam-@xbmc.org>
Date: Fri, 30 Dec 2011 00:29:16 +0100
Subject: [PATCH] fixed: service addons are not started after installation
 (fixes #12378)

---
 xbmc/addons/AddonInstaller.cpp |   16 ++++++++++++++++
 1 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/xbmc/addons/AddonInstaller.cpp b/xbmc/addons/AddonInstaller.cpp
index 7de9eee..9272386 100644
--- a/xbmc/addons/AddonInstaller.cpp
+++ b/xbmc/addons/AddonInstaller.cpp
@@ -20,6 +20,7 @@
  */
 
 #include "AddonInstaller.h"
+#include "Service.h"
 #include "utils/log.h"
 #include "utils/URIUtils.h"
 #include "Util.h"
@@ -419,6 +420,14 @@ bool CAddonInstallJob::OnPreInstall()
     g_application.getApplicationMessenger().ExecBuiltIn("UnloadSkin", true);
     return true;
   }
+
+  if (m_addon->Type() == ADDON_SERVICE)
+  {
+    boost::shared_ptr<CService> service = boost::dynamic_pointer_cast<CService>(m_addon);
+    if (service)
+      service->Stop();
+    return true;
+  }
   return false;
 }
 
@@ -511,6 +520,13 @@ void CAddonInstallJob::OnPostInstall(bool reloadAddon)
       g_application.getApplicationMessenger().ExecBuiltIn("ReloadSkin");
     }
   }
+
+  if (m_addon->Type() == ADDON_SERVICE)
+  {
+    boost::shared_ptr<CService> service = boost::dynamic_pointer_cast<CService>(m_addon);
+    if (service)
+      service->Start();
+  }
 }
 
 void CAddonInstallJob::ReportInstallError(const CStdString& addonID,
-- 
1.7.5.4

