diff -Naur xbmc-f3b0020/xbmc/network/AirTunesServer.cpp xbmc-f3b0020.patch/xbmc/network/AirTunesServer.cpp
--- xbmc-f3b0020/xbmc/network/AirTunesServer.cpp	2011-12-10 22:16:17.000000000 +0100
+++ xbmc-f3b0020.patch/xbmc/network/AirTunesServer.cpp	2011-12-11 18:22:56.423945075 +0100
@@ -279,7 +279,7 @@
   }
 }
 
-CAirTunesServer::CAirTunesServer(int port, bool nonlocal)
+CAirTunesServer::CAirTunesServer(int port, bool nonlocal) : CThread("CAirTunesServer")
 {
   m_port = port;
   m_pLibShairport = new DllLibShairport();
