diff -Naur xbmc-31903/configure.in xbmc-31903.patch/configure.in
--- xbmc-31903/configure.in	2010-07-17 16:50:02.838501479 +0200
+++ xbmc-31903.patch/configure.in	2010-07-17 16:50:20.633376618 +0200
@@ -1205,12 +1205,6 @@
   final_message="$final_message\n  MID Support:\tNo"
 fi
 
-if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
-else
-  INCLUDES="$INCLUDES"
-fi
-
 if test "x$use_ccache" != "xno"; then
   AC_PATH_PROG(CCACHE,ccache,none)
   if test "$ac_cv_path_CCACHE" = "none"; then
diff -Naur xbmc-31903/tools/TexturePacker/XBMCTex.cpp xbmc-31903.patch/tools/TexturePacker/XBMCTex.cpp
--- xbmc-31903/tools/TexturePacker/XBMCTex.cpp	2010-07-17 16:28:40.000000000 +0200
+++ xbmc-31903.patch/tools/TexturePacker/XBMCTex.cpp	2010-07-17 16:50:20.634375824 +0200
@@ -37,7 +37,7 @@
 #endif
 
 #ifdef _LINUX
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 #else
 #include "../../xbmc/lib/liblzo/LZO1X.H"
 #endif
