diff -Naur xbmc-pvr-3513480/configure.in xbmc-pvr-3513480.patch/configure.in
--- xbmc-pvr-3513480/configure.in	2011-04-25 03:24:35.056907267 +0200
+++ xbmc-pvr-3513480.patch/configure.in	2011-04-25 04:31:28.068476385 +0200
@@ -1368,7 +1368,6 @@
 
 OUTPUT_FILES="Makefile \
     Makefile.include \
-    addons/skin.confluence/media/Makefile \
     xbmc/Makefile \
     xbmc/cdrip/Makefile \
     xbmc/cores/Makefile \
diff -Naur xbmc-pvr-3513480/Makefile.in xbmc-pvr-3513480.patch/Makefile.in
--- xbmc-pvr-3513480/Makefile.in	2011-04-25 03:15:20.988746910 +0200
+++ xbmc-pvr-3513480.patch/Makefile.in	2011-04-25 04:23:50.615799946 +0200
@@ -166,13 +166,8 @@
 	lib/addons/library.xbmc.pvr \
 	lib/addons/library.xbmc.gui \
 
-CONFLUENCE_MEDIA=addons/skin.confluence/media
-
-SKIN_DIRS=\
-	$(CONFLUENCE_MEDIA)
-
 DIRS= $(BIN_DIRS) $(EC_DIRS) $(XBMCTEX_DIRS) $(DVDPCODECS_DIRS) $(PAPCODECS_DIRS) \
-	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS) $(PVR_DIRS) $(LIBADDON_DIRS) $(SKIN_DIRS)
+	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS) $(PVR_DIRS) $(LIBADDON_DIRS)
 
 LIBS=@LIBS@
 CFLAGS=@CFLAGS@
@@ -209,12 +204,7 @@
 	@./bootstrap
 
 # skin textures
-ifneq ($(findstring arm,@ARCH@), arm)
-skins: tools/TexturePacker/TexturePacker force
-	 $(MAKE) -C $(CONFLUENCE_MEDIA)
-else
 skins:
-endif
 
 
 xbmc/guilib/guilib.a: force
