device tree for udoo LVDS display

diff -aurN a/arch/arm/boot/dts/imx6dl-udoo-15lvds.dts b/arch/arm/boot/dts/imx6dl-udoo-15lvds.dts
--- a/arch/arm/boot/dts/imx6dl-udoo-15lvds.dts	1970-01-01 01:00:00.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-udoo-15lvds.dts	2016-02-02 15:33:57.956493214 +0100
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ * Author: Ettore Chimenti <ettore.chimenti@udoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+/dts-v1/;
+#include "imx6dl.dtsi"
+#include "imx6qdl-udoo.dtsi"
+
+/ {
+	model = "Udoo i.MX6 Dual-lite Board";
+	compatible = "udoo,imx6dl-udoo", "fsl,imx6dl";
+
+	mxcfb2: fb@1 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB24";
+		mode_str = "1360x768M@60";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <1>;
+		status = "okay";
+	};
+	
+	regulators {
+		reg_lcd0_pwr: regulator@1 {
+			status = "okay";
+		};
+
+		reg_lcd0_backlight: regulator@2 {
+			status = "okay";
+		};
+	};
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+		fsl,data-width = <24>;
+
+		display-timings {
+			native-mode = <&timing2>;
+		};
+	};
+};
diff -aurN a/arch/arm/boot/dts/imx6dl-udoo-7lvds.dts b/arch/arm/boot/dts/imx6dl-udoo-7lvds.dts
--- a/arch/arm/boot/dts/imx6dl-udoo-7lvds.dts	1970-01-01 01:00:00.000000000 +0100
+++ b/arch/arm/boot/dts/imx6dl-udoo-7lvds.dts	2016-02-02 15:33:57.956493214 +0100
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ * Author: Ettore Chimenti <ettore.chimenti@udoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+/dts-v1/;
+#include "imx6dl.dtsi"
+#include "imx6qdl-udoo.dtsi"
+
+/ {
+	model = "Udoo i.MX6 Dual-lite Board";
+	compatible = "udoo,imx6dl-udoo", "fsl,imx6dl";
+
+	mxcfb2: fb@1 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB666";
+		mode_str = "800x480M@60";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "okay";
+	};
+	
+	regulators {
+		reg_lcd0_pwr: regulator@1 {
+			status = "okay";
+		};
+
+		reg_lcd0_backlight: regulator@2 {
+			status = "okay";
+		};
+	};
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+	};
+};
diff -aurN a/arch/arm/boot/dts/imx6qdl-udoo.dtsi b/arch/arm/boot/dts/imx6qdl-udoo.dtsi
--- a/arch/arm/boot/dts/imx6qdl-udoo.dtsi	2016-02-02 15:32:24.208082208 +0100
+++ b/arch/arm/boot/dts/imx6qdl-udoo.dtsi	2016-02-02 15:35:33.695723590 +0100
@@ -254,9 +254,9 @@
 				vsync-len = <2>;
 			};
 			
-			timing2: 1366x768 {	// 15.6" display
+			timing2: 1360x768 {	// 15.6" display
 				clock-frequency = <76000000>;
-				hactive = <1366>;
+				hactive = <1360>;	// IPU needs x and y modulo 8
 				vactive = <768>;
 				hback-porch = <220>;
 				hfront-porch = <40>;
diff -aurN a/arch/arm/boot/dts/imx6q-udoo-15lvds.dts b/arch/arm/boot/dts/imx6q-udoo-15lvds.dts
--- a/arch/arm/boot/dts/imx6q-udoo-15lvds.dts	1970-01-01 01:00:00.000000000 +0100
+++ b/arch/arm/boot/dts/imx6q-udoo-15lvds.dts	2016-02-02 15:33:57.956493214 +0100
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ * Author: Ettore Chimenti <ettore.chimenti@udoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6qdl-udoo.dtsi"
+
+/ {
+	model = "Udoo i.MX6 Quad Board";
+	compatible = "udoo,imx6q-udoo", "fsl,imx6q";
+
+	mxcfb2: fb@1 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB24";
+		mode_str = "1360x768M@60";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <1>;
+		status = "okay";
+	};
+	
+	regulators {
+		reg_lcd0_pwr: regulator@1 {
+			status = "okay";
+		};
+
+		reg_lcd0_backlight: regulator@2 {
+			status = "okay";
+		};
+	};
+};
+
+&sata {
+	status = "okay";
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+		fsl,data-width = <24>;
+
+		display-timings {
+			native-mode = <&timing2>;
+		};
+	};
+};
diff -aurN a/arch/arm/boot/dts/imx6q-udoo-7lvds.dts b/arch/arm/boot/dts/imx6q-udoo-7lvds.dts
--- a/arch/arm/boot/dts/imx6q-udoo-7lvds.dts	1970-01-01 01:00:00.000000000 +0100
+++ b/arch/arm/boot/dts/imx6q-udoo-7lvds.dts	2016-02-02 15:33:57.956493214 +0100
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ * Author: Ettore Chimenti <ettore.chimenti@udoo.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6qdl-udoo.dtsi"
+
+/ {
+	model = "Udoo i.MX6 Quad Board";
+	compatible = "udoo,imx6q-udoo", "fsl,imx6q";
+
+	mxcfb2: fb@1 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB666";
+		mode_str = "800x480M@60";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "okay";
+	};
+	
+	regulators {
+		reg_lcd0_pwr: regulator@1 {
+			status = "okay";
+		};
+
+		reg_lcd0_backlight: regulator@2 {
+			status = "okay";
+		};
+	};
+};
+
+&sata {
+	status = "okay";
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		status = "okay";
+	};
+};
diff -aurN a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
--- a/arch/arm/boot/dts/Makefile	2016-02-02 15:33:47.144907214 +0100
+++ b/arch/arm/boot/dts/Makefile	2016-02-02 15:33:57.956493214 +0100
@@ -308,6 +308,8 @@
 	imx6dl-tx6u-801x.dtb \
 	imx6dl-tx6u-811x.dtb \
 	imx6dl-udoo.dtb \
+	imx6dl-udoo-7lvds.dtb \
+	imx6dl-udoo-15lvds.dtb \
 	imx6dl-sbc-fx6.dtb \
 	imx6dl-sbc-fx6m.dtb \
 	imx6dl-wandboard.dtb \
@@ -351,6 +353,8 @@
 	imx6q-tx6q-1020-comtft.dtb \
 	imx6q-tx6q-1110.dtb \
 	imx6q-udoo.dtb \
+	imx6q-udoo-7lvds.dtb \
+	imx6q-udoo-15lvds.dtb \
 	imx6q-wandboard.dtb \
 	imx6q-wandboard-revb1.dtb \
 	imx6dl-vero.dtb
