diff -Naur xbmc-pvr-addons-259d150/configure.ac xbmc-pvr-addons-259d150.patch/configure.ac
--- xbmc-pvr-addons-259d150/configure.ac	2014-10-06 03:49:06.000000000 +0200
+++ xbmc-pvr-addons-259d150.patch/configure.ac	2014-10-20 03:09:15.933083541 +0200
@@ -293,18 +293,18 @@
 if test "x${libdir}" != 'x${exec_prefix}/lib'; then
   LIBDIR=${libdir}
 elif test "$prefix" = "NONE"; then
-  LIBDIR=$PREFIX_DEFAULT/lib/xbmc/addons
+  LIBDIR=$PREFIX_DEFAULT/lib/kodi/addons
 else
-  LIBDIR=$prefix/lib/xbmc/addons
+  LIBDIR=$prefix/lib/kodi/addons
 fi
 AC_SUBST(LIBDIR)
 
 if test "x${datadir}" != 'x${datarootdir}' && test "x${datarootdir}" = 'x${prefix}/share'; then
   DATADIR=${datadir}
 elif test "$prefix" = "NONE"; then
-  DATADIR=$PREFIX_DEFAULT/share/xbmc/addons
+  DATADIR=$PREFIX_DEFAULT/share/kodi/addons
 else
-  DATADIR=$prefix/share/xbmc/addons 
+  DATADIR=$prefix/share/kodi/addons 
 fi
 AC_SUBST(DATADIR)
 
