diff -Naur xbmc-pvr-addons-a6693d0/addons/Makefile.am xbmc-pvr-addons-a6693d0.patch/addons/Makefile.am
--- xbmc-pvr-addons-a6693d0/addons/Makefile.am	2012-10-02 02:47:01.000000000 +0200
+++ xbmc-pvr-addons-a6693d0.patch/addons/Makefile.am	2012-10-02 12:51:48.538463220 +0200
@@ -1,4 +1,5 @@
 SUBDIRS = pvr.demo \
+          pvr.fortherecord.argus \
           pvr.hts \
           pvr.mediaportal.tvserver \
           pvr.vdr.vnsi \
