diff -Naur xbmc-pvr-addons-94c9f6f/configure.ac xbmc-pvr-addons-94c9f6f.patch/configure.ac
--- xbmc-pvr-addons-94c9f6f/configure.ac	2012-09-05 20:43:31.000000000 +0200
+++ xbmc-pvr-addons-94c9f6f.patch/configure.ac	2012-09-05 21:24:01.619154672 +0200
@@ -75,10 +75,12 @@
 use_libcurl="no"
 PKG_CHECK_MODULES([CURL],[libcurl],use_libcurl="yes",AC_MSG_WARN("CURL was not found, N7 add-on will not be available"))
 
-### Check for Intree building
-checkpath=".."
-AC_CHECK_FILE([$checkpath/xbmc/xbmc.h], [AM_CONDITIONAL(IS_INTREE_BUILD, true) intree=true], [AM_CONDITIONAL(IS_INTREE_BUILD, false) intree=false])
-echo "Intree build: $intree"
+# ### Check for Intree building
+# checkpath=".."
+# AC_CHECK_FILE([$checkpath/xbmc/xbmc.h], [AM_CONDITIONAL(IS_INTREE_BUILD, true) intree=true], [AM_CONDITIONAL(IS_INTREE_BUILD, false) intree=false])
+# echo "Intree build: $intree"
+
+AM_CONDITIONAL(IS_INTREE_BUILD, false)
 
 HOST_CXXFLAGS="-Wall -Wextra -Wno-missing-field-initializers -Woverloaded-virtual -Wno-parentheses -fPIC $HOST_CXXFLAGS"
 
