diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/Makefile.include.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/Makefile.include.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/Makefile.include.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/Makefile.include.am	2013-09-09 18:54:06.000000000 +0200
@@ -36,7 +36,9 @@
 	mkdir -m 755 -p $(DESTDIR)@LIBDIR@/$(ADDONNAME)
 	mkdir -m 755 -p $(DESTDIR)@DATADIR@/$(ADDONNAME)
 	cp -f @BINPREFIX@$(ADDONBINNAME)@BIN_EXT@ $(DESTDIR)@LIBDIR@/$(ADDONNAME) ; chmod 655 $(DESTDIR)@LIBDIR@/$(ADDONNAME)/@BINPREFIX@$(ADDONBINNAME)@BIN_EXT@
-	cp -r -f @abs_top_srcdir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME) ; chmod -R o+rx $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	cp -r -f @abs_top_srcdir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	cp -r -f @abs_top_builddir@/addons/$(ADDONNAME)/addon/* $(DESTDIR)@DATADIR@/$(ADDONNAME)
+	chmod -R o+rx $(DESTDIR)@DATADIR@/$(ADDONNAME)
 endif
 
 all: @BUILD_TYPE@
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.argustv/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.argustv/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.argustv/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.argustv/Makefile.am	2013-09-09 03:48:49.756563520 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libargustv-addon
 lib_LTLIBRARIES = libargustv-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/jsoncpp/libjsoncpp.la
+LIBS            = @abs_top_builddir@/lib/jsoncpp/libjsoncpp.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc -Isrc/lib/filesystem -I@abs_top_srcdir@/lib/jsoncpp/include
+INCLUDES+=-I$(srcdir)/src -I$(srcdir)/src/lib/filesystem -I@abs_top_srcdir@/lib/jsoncpp/include
 
 libargustv_addon_la_SOURCES = src/activerecording.cpp \
                                    src/channel.cpp \
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.demo/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.demo/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.demo/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.demo/Makefile.am	2013-09-09 03:34:14.482815303 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libpvrdemo-addon
 lib_LTLIBRARIES = libpvrdemo-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.hts/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.hts/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.hts/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.hts/Makefile.am	2013-09-09 03:39:48.924418632 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libtvheadend-addon
 lib_LTLIBRARIES = libtvheadend-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/libhts/libhts.la -ldl
+LIBS            = @abs_top_builddir@/lib/libhts/libhts.la -ldl
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.mediaportal.tvserver/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.mediaportal.tvserver/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.mediaportal.tvserver/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.mediaportal.tvserver/Makefile.am	2013-09-09 03:45:22.546295558 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libmediaportal-addon
 lib_LTLIBRARIES = libmediaportal-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc
+INCLUDES+=-I$(srcdir)/src
 
 libmediaportal_addon_la_SOURCES = src/Cards.cpp \
                                   src/channels.cpp \
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.mythtv.cmyth/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.mythtv.cmyth/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.mythtv.cmyth/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.mythtv.cmyth/Makefile.am	2013-09-09 03:41:06.664969675 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libmythtvcmyth-addon
 lib_LTLIBRARIES = libmythtvcmyth-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/cmyth/libcmyth.la
+LIBS            = @abs_top_builddir@/lib/cmyth/libcmyth.la
 
 AM_CPPFLAGS = -I$(abs_top_srcdir)/lib/cmyth/include
 
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.nextpvr/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.nextpvr/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.nextpvr/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.nextpvr/Makefile.am	2013-09-09 03:49:09.311590467 +0200
@@ -10,11 +10,11 @@
 LIBNAME         = libnextpvr-addon
 lib_LTLIBRARIES = libnextpvr-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
-INCLUDES+=-Isrc
+INCLUDES+=-I$(srcdir)/src
 
 libnextpvr_addon_la_SOURCES = src/client.cpp \
                                   src/pvrclient-nextpvr.cpp \
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.njoy/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.njoy/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.njoy/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.njoy/Makefile.am	2013-09-09 03:36:15.792970520 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libpvrnjoy-addon
 lib_LTLIBRARIES = libpvrnjoy-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-frodo-910d7e7/addons/pvr.vuplus/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.vuplus/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/addons/pvr.vuplus/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/addons/pvr.vuplus/Makefile.am	2013-09-09 03:35:22.527902278 +0200
@@ -10,7 +10,7 @@
 LIBNAME         = libvuplus-addon
 lib_LTLIBRARIES = libvuplus-addon.la
 
-LIBS            = @abs_top_srcdir@/lib/tinyxml/libtinyxml.la
+LIBS            = @abs_top_builddir@/lib/tinyxml/libtinyxml.la
 
 include ../Makefile.include.am
 
diff -Naur xbmc-pvr-addons-frodo-910d7e7/lib/cmyth/libcmyth/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/lib/cmyth/libcmyth/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/lib/cmyth/libcmyth/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/lib/cmyth/libcmyth/Makefile.am	2013-09-09 03:02:37.362378511 +0200
@@ -25,7 +25,7 @@
                      channel.c \
                      storagegroup.c
 
-INCLUDES=-I../include/ $(MYSQL_INCLUDES)
+INCLUDES=-I$(srcdir)/../include/ $(MYSQL_INCLUDES)
 
 $(LIB): libcmyth.la
 	cp -f .libs/libcmyth.a .
diff -Naur xbmc-pvr-addons-frodo-910d7e7/lib/cmyth/librefmem/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/lib/cmyth/librefmem/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/lib/cmyth/librefmem/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/lib/cmyth/librefmem/Makefile.am	2013-09-09 02:58:20.664045817 +0200
@@ -3,7 +3,7 @@
 librefmem_la_SOURCES = alloc.c \
                       debug_refmem.c
 
-INCLUDES=-I../include/ -I../libcmyth/
+INCLUDES=-I$(srcdir)/../include/ -I$(srcdir)/../libcmyth/
 
 $(LIB): librefmem.la
 	cp -f .libs/librefmem.a .
diff -Naur xbmc-pvr-addons-frodo-910d7e7/lib/jsoncpp/Makefile.am xbmc-pvr-addons-frodo-910d7e7.patch/lib/jsoncpp/Makefile.am
--- xbmc-pvr-addons-frodo-910d7e7/lib/jsoncpp/Makefile.am	2013-09-09 01:56:00.000000000 +0200
+++ xbmc-pvr-addons-frodo-910d7e7.patch/lib/jsoncpp/Makefile.am	2013-09-09 02:53:57.575708599 +0200
@@ -4,7 +4,7 @@
                         src/lib_json/json_value.cpp \
                         src/lib_json/json_writer.cpp
 
-INCLUDES=-Iinclude/
+INCLUDES=-I$(srcdir)/include/
 
 $(LIB): libjsoncpp.la
 	cp -f .libs/libjsoncpp.a .
