From 822abd913ab38a742f710173f551bac22477eebd Mon Sep 17 00:00:00 2001
From: Lee Pollock <scudlee@gmail.com>
Date: Wed, 8 May 2013 18:04:41 +0100
Subject: [PATCH] [Fix] Re-get details from nfo file after advancing to first
 episodedetails

---
 xbmc/NfoFile.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/NfoFile.cpp b/xbmc/NfoFile.cpp
index 973e4d0..791c92e 100644
--- a/xbmc/NfoFile.cpp
+++ b/xbmc/NfoFile.cpp
@@ -71,6 +71,7 @@
     {
       int infos=0;
       m_headofdoc = strstr(m_headofdoc,"<episodedetails");
+      bNfo = GetDetails(details);
       while (m_headofdoc && details.m_iEpisode != episode)
       {
         m_headofdoc = strstr(m_headofdoc+1,"<episodedetails");
-- 
1.8.1.6

