From ef82d9b6177a0576548e8e9cdc0eb17cf10524af Mon Sep 17 00:00:00 2001
From: ulion <ulion2002@gmail.com>
Date: Thu, 9 May 2013 13:11:59 +0800
Subject: [PATCH] [OSX] Fix always on top after restore from fullscreen.

---
 xbmc/windowing/osx/WinSystemOSX.mm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/xbmc/windowing/osx/WinSystemOSX.mm b/xbmc/windowing/osx/WinSystemOSX.mm
index e45693c..cb86e0a 100644
--- a/xbmc/windowing/osx/WinSystemOSX.mm
+++ b/xbmc/windowing/osx/WinSystemOSX.mm
@@ -710,6 +710,7 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   static NSView* last_view = NULL;
   static NSSize last_view_size;
   static NSPoint last_view_origin;
+  static NSInteger last_window_level = NSNormalWindowLevel;
   bool was_fullscreen = m_bFullScreen;
   static int lastDisplayNr = res.iScreen;
   NSOpenGLContext* cur_context;
@@ -779,6 +780,7 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     last_view_origin = [last_view frame].origin;
     last_window_screen = [[last_view window] screen];
     last_window_origin = [[last_view window] frame].origin;
+    last_window_level = [[last_view window] level];
 
     if (CSettings::Get().GetBool("videoscreen.fakefullscreen"))
     {
@@ -891,7 +893,7 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     if (CSettings::Get().GetBool("videoscreen.fakefullscreen"))
     {
       // restore the windowed window level
-      [[last_view window] setLevel:NSNormalWindowLevel];
+      [[last_view window] setLevel:last_window_level];
 
       // Get rid of the new window we created.
       if (windowedFullScreenwindow != NULL)
-- 
1.8.1.6

