From 86a63c5040e307dcb633a5934ab1b3f9114009f5 Mon Sep 17 00:00:00 2001
From: arnova <arnova@void.org>
Date: Sun, 23 Jun 2013 23:16:55 +0200
Subject: [PATCH] changed: Make more obvious we're passing a CFileItem

---
 xbmc/music/infoscanner/MusicInfoScanner.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/xbmc/music/infoscanner/MusicInfoScanner.cpp b/xbmc/music/infoscanner/MusicInfoScanner.cpp
index 84a155c..f93d613 100644
--- a/xbmc/music/infoscanner/MusicInfoScanner.cpp
+++ b/xbmc/music/infoscanner/MusicInfoScanner.cpp
@@ -1009,8 +1009,10 @@ void CMusicInfoScanner::FindArtForAlbums(VECALBUMS &albums, const CStdString &pa
     }
   }
   if (albums.size() == 1 && !albumArt.empty())
-  { // assign to folder thumb as well
-    CMusicThumbLoader::SetCachedImage(path, "thumb", albumArt);
+  {
+    // assign to folder thumb as well
+    CFileItem albumItem(path, true);
+    CMusicThumbLoader::SetCachedImage(albumItem, "thumb", albumArt);
   }
 }
 
-- 
1.8.1.6

