diff -Naur xbmc-12.2.0/lib/libdvd/libdvdread/src/dvd_reader.c xbmc-12.2.0.patch/lib/libdvd/libdvdread/src/dvd_reader.c
--- xbmc-12.2.0/lib/libdvd/libdvdread/src/dvd_reader.c	2013-05-02 17:00:11.000000000 +0200
+++ xbmc-12.2.0.patch/lib/libdvd/libdvdread/src/dvd_reader.c	2013-06-18 07:08:23.973913805 +0200
@@ -423,6 +423,9 @@
               /* Also WIN32 does not have symlinks, so we don't need this bit of code. */
 
 	/* Resolve any symlinks and get the absolut dir name. */
+#if defined(_XBMC) /* for XBMC, only do symlink resolution for (real) non-xbmc-VFS paths */
+	if ( path[0] == '/' )
+#endif // _XBMC
 	{
 	    char *new_path;
 	    int cdir = open( ".", O_RDONLY );
