diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-05-08 08:20:29.000000000 -0700
+++ b/Makefile	2016-05-10 15:08:05.548597094 -0700
@@ -304,6 +304,17 @@
 		CPUFLAGS += -mfloat-abi=hard
 	endif
 
+# aarch64
+else ifneq (,$(findstring aarch64,$(platform)))
+        TARGET := $(TARGET_NAME)_libretro.so
+        LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined
+        fpic := -fPIC
+#        CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -DNOSSE
+        WITH_DYNAREC=arm64
+#        PLATCFLAGS += -DARM
+        GLES = 1
+        GL_LIB := -lGLESv2
+
 # emscripten
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_emscripten.bc
