--- a/hciattach_rtk.c	2016-06-03 06:59:52.251480348 +0200
+++ b/hciattach_rtk.c	2016-06-03 07:05:59.554167327 +0200
@@ -1433,7 +1433,7 @@
 	int ret = 0;
 	struct stat st;
 
-	ret = sprintf(firmware_file_name, FIRMWARE_DIRECTORY"rtlbt_fw");
+	ret = sprintf(firmware_file_name, FIRMWARE_DIRECTORY"rtl8723b_fw");
 
 	return firmware_file_name;
 }
@@ -1608,7 +1608,7 @@
 #endif
 
 GET_CONFIG:
-	ret = sprintf(bt_config_file_name, BT_CONFIG_DIRECTORY"rtlbt_config"); 
+	ret = sprintf(bt_config_file_name, BT_CONFIG_DIRECTORY"rtl8723b_config"); 
 	if (stat(bt_config_file_name, &st) < 0) {
 		RS_ERR("can't access bt config file:%s, errno:%d\n", bt_config_file_name, errno);
 		return -1;
