--- a/x86-32/src/wl/sys/wl_linux.c
+++ b/x86-32/src/wl/sys/wl_linux.c
@@ -385,7 +385,11 @@ static const struct net_device_ops wl_netdev_ops =
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 2, 0)
 	.ndo_set_multicast_list = wl_set_multicast_list,
+#else
+	.ndo_set_rx_mode = wl_set_multicast_list,
+#endif
 	.ndo_do_ioctl = wl_ioctl
 };

--- a/x86-64/src/wl/sys/wl_linux.c
+++ b/x86-64/src/wl/sys/wl_linux.c
@@ -385,7 +385,11 @@ static const struct net_device_ops wl_netdev_ops =
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 2, 0)
 	.ndo_set_multicast_list = wl_set_multicast_list,
+#else
+	.ndo_set_rx_mode = wl_set_multicast_list,
+#endif
 	.ndo_do_ioctl = wl_ioctl
 };
