--- a/x86-32/src/wl/sys/wl_cfg80211.c	2013-03-29 18:12:43.447176381 +0100
+++ b/x86-32/src/wl/sys/wl_cfg80211.c	2013-03-29 18:12:56.595176453 +0100
@@ -2013,8 +2013,12 @@
 		ie_len = (size_t)(ies->len);
 		rcu_read_unlock();
 #endif
-		cfg80211_put_bss(bss);
-	}
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 9, 0)
+ 		cfg80211_put_bss(bss);
+#else
+		cfg80211_put_bss(wl_to_wiphy(wl), bss);
+#endif
+ 	}
 
 	tim = bcm_parse_tlvs(ie, ie_len, WLAN_EID_TIM);
 	if (tim) {

--- a/x86-64/src/wl/sys/wl_cfg80211.c	2013-03-29 18:12:43.447176381 +0100
+++ a/x86-64/src/wl/sys/wl_cfg80211.c	2013-03-29 18:12:56.595176453 +0100
@@ -2013,8 +2013,12 @@
 		ie_len = (size_t)(ies->len);
 		rcu_read_unlock();
 #endif
-		cfg80211_put_bss(bss);
-	}
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 9, 0)
+ 		cfg80211_put_bss(bss);
+#else
+		cfg80211_put_bss(wl_to_wiphy(wl), bss);
+#endif
+ 	}
 
 	tim = bcm_parse_tlvs(ie, ie_len, WLAN_EID_TIM);
 	if (tim) {
