--- a/x86-32/src/wl/sys/wl_linux.c	2010-12-15 02:58:07.000000000 +0100
+++ b/x86-32/src/wl/sys/wl_linux.c	2010-12-27 22:14:38.679428984 +0100
@@ -251,7 +251,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
--- a/x86-64/src/wl/sys/wl_linux.c	2010-12-15 03:01:09.000000000 +0100
+++ b/x86-64/src/wl/sys/wl_linux.c	2010-12-27 22:15:18.273865927 +0100
@@ -251,7 +251,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
