--- a/x86-32/src/wl/sys/wl_linux.c
+++ b/x86-32/src/wl/sys/wl_linux.c
@@ -1773,8 +1773,8 @@
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
-	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
-		__DATE__, __TIME__, EPI_VERSION_STR);
+	bcm_bprintf(b, "wl%d: version %s\n", wl->pub->unit,
+        EPI_VERSION_STR);
 }
 
 #if defined(BCMDBG)

--- a/x86-64/src/wl/sys/wl_linux.c
+++b/x86-64/src/wl/sys/wl_linux.c
@@ -1773,8 +1773,8 @@
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
-	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
-		__DATE__, __TIME__, EPI_VERSION_STR);
+	bcm_bprintf(b, "wl%d: version %s\n", wl->pub->unit,
+		EPI_VERSION_STR);
 }
 
 #if defined(BCMDBG)
