diff --git a/arch/arm/mach-meson8/pm.c b/arch/arm/mach-meson8/pm.c
index 754f7274b2e..9bda172c450 100755
--- a/arch/arm/mach-meson8/pm.c
+++ b/arch/arm/mach-meson8/pm.c
@@ -421,6 +421,7 @@ static struct platform_driver meson_pm_driver = {
 static int __init meson_pm_init(void)
 {
 	printk("enter %s\n",__func__);
+	pm_power_off = meson_pm_suspend;
 	return platform_driver_probe(&meson_pm_driver, meson_pm_probe);
 }
 late_initcall(meson_pm_init);
