diff -Naur libdrm-newttm-svn-20090329.2/linux-core/drm_sysfs.c libdrm-newttm-svn-20090329.2.patch/linux-core/drm_sysfs.c
--- libdrm-newttm-svn-20090329.2/linux-core/drm_sysfs.c	2009-04-01 00:17:47.000000000 +0200
+++ libdrm-newttm-svn-20090329.2.patch/linux-core/drm_sysfs.c	2009-04-28 22:36:01.230011104 +0200
@@ -351,8 +351,8 @@
 	DRM_DEBUG("adding \"%s\" to sysfs\n",
 		  drm_get_connector_name(connector));
 
-	snprintf(connector->kdev.bus_id, BUS_ID_SIZE, "card%d-%s",
-		 dev->primary->index, drm_get_connector_name(connector));
+        dev_set_name(&connector->kdev, "card%d-%s",
+                    dev->primary->index, drm_get_connector_name(connector));
 	ret = device_register(&connector->kdev);
 
 	if (ret) {
@@ -487,12 +487,7 @@
 	else
 		minor_str = "card%d";
 	
-	snprintf(minor->kdev.bus_id, BUS_ID_SIZE, minor_str, minor->index);
-
-	/* Shouldn't register more than once */
-	BUG_ON(device_is_registered(&minor->kdev));
-
-	DRM_DEBUG("registering DRM device \"%s\"\n", minor->kdev.bus_id);
+        dev_set_name(&minor->kdev, minor_str, minor->index);
 
 	err = device_register(&minor->kdev);
 	if (err) {
diff -Naur libdrm-newttm-svn-20090329.2/linux-core/ttm/ttm_pat_compat.h libdrm-newttm-svn-20090329.2.patch/linux-core/ttm/ttm_pat_compat.h
--- libdrm-newttm-svn-20090329.2/linux-core/ttm/ttm_pat_compat.h	2009-04-01 00:17:47.000000000 +0200
+++ libdrm-newttm-svn-20090329.2.patch/linux-core/ttm/ttm_pat_compat.h	2009-04-28 22:38:20.546022042 +0200
@@ -33,6 +33,7 @@
 #ifndef _TTM_PAT_COMPAT_
 #define _TTM_PAT_COMPAT_
 #include <asm/page.h>
+#include <asm/pgtable_types.h>
 
 extern void ttm_pat_init(void);
 extern void ttm_pat_takedown(void);
