From 5f4085ddb16c206b1dc36d481c2aa6684ba392ee Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 18 Jul 2013 19:26:21 +0300
Subject: [PATCH] silence crond startup logging

---
 miscutils/crond.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff -Naur busybox-1.23.0/miscutils/crond.c busybox-1.23.0.patch/miscutils/crond.c
--- busybox-1.23.0/miscutils/crond.c	2014-10-04 22:35:59.000000000 +0200
+++ busybox-1.23.0.patch/miscutils/crond.c	2014-12-26 23:07:53.386118290 +0100
@@ -945,7 +945,6 @@
 
 	reopen_logfile_to_stderr();
 	xchdir(G.crontab_dir_name);
-	log8("crond (busybox "BB_VER") started, log level %d", G.log_level);
 	rescan_crontab_dir();
 	write_pidfile(CONFIG_PID_FILE_PATH "/crond.pid");
 
