From 575ee0c44577f8a1010fc565c5b4b7f4486b5dab Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 26 Jan 2014 16:45:18 +0200
Subject: [PATCH] dont run rmmod.pl check

---
 v4l/Makefile |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/v4l/Makefile b/v4l/Makefile
index 41f7283..21911e0 100644
--- a/v4l/Makefile
+++ b/v4l/Makefile
@@ -51,7 +51,6 @@ default:: prepare firmware
 	@echo Kernel build directory is $(OUTDIR)
 	$(MAKE) -C ../linux apply_patches
 	$(MAKE) -C $(OUTDIR) SUBDIRS=$(PWD) $(MYCFLAGS) modules
-	./scripts/rmmod.pl check
 #	$(MAKE) checkpatch
 
 mismatch:: prepare firmware
-- 
1.7.2.5

