diff -Naur gettext-0.18.1.1-old/gettext-tools/misc/gettextize.in gettext-0.18.1.1-new/gettext-tools/misc/gettextize.in
--- gettext-0.18.1.1-old/gettext-tools/misc/gettextize.in	2010-06-06 05:49:57.000000000 -0700
+++ gettext-0.18.1.1-new/gettext-tools/misc/gettextize.in	2010-08-18 06:52:51.000000000 -0700
@@ -1272,10 +1272,10 @@
     5) count="five paragraphs";;
     *) count="$count paragraphs";;
   esac
-  echo "Press Return to acknowledge the previous $count."
-  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
-  # non-interactive tools.
-  read dummy < /dev/tty
+#  echo "Press Return to acknowledge the previous $count."
+#  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
+#  # non-interactive tools.
+#  read dummy < /dev/tty
 fi
 
 exit 0
