--- a/core/rtw_br_ext.c	2021-10-10 13:28:28.410328649 +1100
+++ b/core/rtw_br_ext.c	2021-10-10 13:28:28.410328649 +1100
@@ -22,7 +22,10 @@
 #ifdef __KERNEL__
 #include <linux/if_arp.h>
 #include <net/ip.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
 #include <net/ipx.h>
+#endif
 #include <linux/atalk.h>
 #include <linux/udp.h>
 #include <linux/if_pppox.h>
@@ -175,6 +175,7 @@
 }
 
 
+#ifdef _NET_INET_IPX_H_
 static __inline__ void __nat25_generate_ipx_network_addr_with_node(unsigned char *networkAddr,
 				unsigned int *ipxNetAddr, unsigned char *ipxNodeAddr)
 {
@@ -195,6 +196,7 @@
 	memcpy(networkAddr+1, (unsigned char *)ipxNetAddr, 4);
 	memcpy(networkAddr+5, (unsigned char *)ipxSocketAddr, 2);
 }
+#endif
 
 
 static __inline__ void __nat25_generate_apple_network_addr(unsigned char *networkAddr,
@@ -341,6 +343,7 @@
 
 		return x & (NAT25_HASH_SIZE - 1);
 	}
+#ifdef _NET_INET_IPX_H_
 	else if(networkAddr[0] == NAT25_IPX)
 	{
 		unsigned long x;
@@ -350,6 +353,7 @@
 
 		return x & (NAT25_HASH_SIZE - 1);
 	}
+#endif
 	else if(networkAddr[0] == NAT25_APPLE)
 	{
 		unsigned long x;
@@ -932,6 +936,7 @@
 		}
 	}
 
+#ifdef _NET_INET_IPX_H_
 	/*---------------------------------------------------*/
 	/*         Handle IPX and Apple Talk frame           */
 	/*---------------------------------------------------*/
@@ -1195,6 +1200,7 @@
 
 		return -1;
 	}
+#endif
 
 	/*---------------------------------------------------*/
 	/*                Handle PPPoE frame                 */
