diff -Naur RTL8192DU-085e206/os_dep/usb_intf.c RTL8192DU-085e206.patch/os_dep/usb_intf.c
--- RTL8192DU-085e206/os_dep/usb_intf.c	2014-03-25 15:48:57.000000000 +0100
+++ RTL8192DU-085e206.patch/os_dep/usb_intf.c	2014-04-11 09:23:12.097223906 +0200
@@ -137,7 +137,7 @@
 	{USB_DEVICE(USB_VENDER_ID_REALTEK, 0x8194)},/* 8192DU-VS */ \
 	{USB_DEVICE(USB_VENDER_ID_REALTEK, 0x8111)},/* Realtek 5G dongle for WiFi Display */ \
 	{USB_DEVICE(USB_VENDER_ID_REALTEK, 0x0193)},/* 8192DE-VAU */ \
-	{USB_DEVICE(USB_VENDER_ID_REALTEK, 0x8171)},/* 8192DU-VC */ \
+/*	{USB_DEVICE(USB_VENDER_ID_REALTEK, 0x8171)}, 8192DU-VC overlaps with in-kernel rtl8192se */ \
 	/*=== Customer ID ===*/ \
 	/****** 8192DU-VC ********/ \
 	{USB_DEVICE(0x2019, 0xAB2C)},/* PCI - Abocm */ \
