diff -Naur a/os_dep/ioctl_cfg80211.c b/os_dep/ioctl_cfg80211.c
--- a/os_dep/ioctl_cfg80211.c	2016-09-11 01:51:12.796375013 +0100
+++ b/os_dep/ioctl_cfg80211.c	2016-09-11 01:52:39.942348529 +0100
@@ -1425,7 +1425,14 @@
 		}
 		else
 		{
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,8,0)
+			struct cfg80211_scan_info info = {
+				.aborted = aborted
+			};
+			cfg80211_scan_done(pwdev_priv->scan_request, &info);
+#else
 			cfg80211_scan_done(pwdev_priv->scan_request, aborted);
+#endif
 		}
 
 		pwdev_priv->scan_request = NULL;
