diff -Naur RTL8192DU-085e206/os_dep/rtw_android.c RTL8192DU-085e206.patch/os_dep/rtw_android.c
--- RTL8192DU-085e206/os_dep/rtw_android.c	2014-03-25 15:48:57.000000000 +0100
+++ RTL8192DU-085e206.patch/os_dep/rtw_android.c	2015-04-04 20:23:03.589408056 +0200
@@ -219,7 +219,11 @@
 {
 	int cmd_num;
 	for (cmd_num=0 ; cmd_num<ANDROID_WIFI_CMD_MAX; cmd_num++)
-		if (0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0))
+		if (!strncasecmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])))
+#else
+		if (0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])))
+#endif
 			break;
 
 	return cmd_num;
