diff -Naur eagle-20090913/eagle.c eagle-20090913.patch/eagle.c
--- eagle-20090913/eagle.c	2009-09-12 18:15:07.000000000 +0200
+++ eagle-20090913.patch/eagle.c	2009-09-19 03:44:02.000000000 +0200
@@ -41,41 +41,6 @@
 static EGLContext	currentContext;
 
 static void
-(*_mesa_TextureExternalMESA)(GLint target,
-			     GLint glx_texture_format,
-			     GLuint cpp,
-			     GLuint width, GLuint height,
-			     GLuint pitch, GLuint handle);
-static void
-(*_mesa_RenderbufferExternalMESA)(GLint target,
-				  GLint internalFormat,
-				  GLuint width, GLuint height,
-				  GLuint pitch, GLuint handle);
-
-EAGLE_EXPORT void
-glTextureExternalMESA(GLint target,
-		      GLint glx_texture_format,
-		      GLuint cpp,
-		      GLuint width, GLuint height,
-		      GLuint pitch, GLuint handle)
-{
-	_mesa_TextureExternalMESA(target, glx_texture_format, cpp,
-				  width, height, pitch, handle);
-}
-
-EAGLE_EXPORT void
-glRenderbufferExternalMESA(GLint target,
-			   GLint internalFormat,
-			   GLuint width, GLuint height,
-			   GLuint pitch, GLuint handle)
-{
-	_mesa_RenderbufferExternalMESA(target, internalFormat,
-				       width, height, pitch, handle);
-}
-
-
-
-static void
 eglSetError(EGLint error)
 {
 	lastError = error;
@@ -140,25 +105,6 @@
 
 static const char driDriverPath[] = EAGLE_DRIVER_PATH;
 
-static void
-lookup_hacks(EGLDisplay display)
-{
-	_mesa_TextureExternalMESA =
-		dlsym(display->driver, "_mesa_TextureExternalMESA");
-	if (_mesa_TextureExternalMESA == NULL) {
-		fprintf(stderr, "_mesa_TextureExternalMESA not found\n");
-		exit(1);
-	}
-
-	_mesa_RenderbufferExternalMESA =
-		dlsym(display->driver, "_mesa_RenderbufferExternalMESA");
-	if (_mesa_RenderbufferExternalMESA == NULL) {
-		fprintf(stderr, "_mesa_RenderbufferExternalMESA not found\n");
-		exit(1);
-	}
-
-}
-
 static int
 eglLoadDriver(EGLDisplay display, const char *driverName)
 {
@@ -185,8 +131,6 @@
 		return -1;
 	}
 
-	lookup_hacks(display);
-
 	extensions = dlsym(display->driver, __DRI_DRIVER_EXTENSIONS);
 	if (extensions == NULL) {
 		fprintf(stderr, "dlsym: %s\n", dlerror());
diff -Naur eagle-20090913/eagle.h eagle-20090913.patch/eagle.h
--- eagle-20090913/eagle.h	2009-09-12 18:15:07.000000000 +0200
+++ eagle-20090913.patch/eagle.h	2009-09-19 03:44:39.000000000 +0200
@@ -342,16 +342,4 @@
 
 extern void (*eglGetProcAddress(const char *procname))(void);
 
-
-extern void glTextureExternalMESA(GLint target,
-				  GLint glx_texture_format,
-				  GLuint cpp,
-				  GLuint width, GLuint height,
-				  GLuint pitch, GLuint handle);
-
-extern void glRenderbufferExternalMESA(GLint target,
-				       GLint internalFormat,
-				       GLuint width, GLuint height,
-				       GLuint pitch, GLuint handle);
-
 #endif
