diff --git a/kernel/nv-acpi.c b/kernel/nv-acpi.c
index b0138cd..756f330 100644
--- a/kernel/nv-acpi.c
+++ b/kernel/nv-acpi.c
@@ -303,8 +303,11 @@ static int nv_acpi_remove(struct acpi_device *device, int type)
 
     if (pNvAcpiObject->notify_handler_installed)
     {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
+/* beginning with 3.13, acpi_remove_notify_handler() waits for events to finish */
         NV_ACPI_OS_WAIT_EVENTS_COMPLETE();
 
+#endif
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
     }
