diff -Naur xf86-video-nvidia/usr/src/nv/conftest.sh xf86-video-nvidia.patch/usr/src/nv/conftest.sh
--- xf86-video-nvidia/usr/src/nv/conftest.sh	2009-12-19 00:10:53.000000000 +0100
+++ xf86-video-nvidia.patch/usr/src/nv/conftest.sh	2010-01-21 14:20:31.000000000 +0100
@@ -33,14 +33,14 @@
     # CONFIG_XEN and CONFIG_PARAVIRT are present, text_xen() treats
     # the kernel as a stand-alone kernel.
     #
-    FILE="linux/autoconf.h"
+    FILE="generated/autoconf.h"
 
     if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
         #
         # We are looking at a configured source tree; verify
         # that it's not a Xen kernel.
         #
-        echo "#include <linux/autoconf.h>
+        echo "#include <generated/autoconf.h>
         #if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT)
         #error CONFIG_XEN defined!
         #endif
@@ -110,7 +110,7 @@
     fi
 }
 
-CONFTEST_PREAMBLE="#include <linux/autoconf.h>
+CONFTEST_PREAMBLE="#include <generated/autoconf.h>
     #if defined(CONFIG_XEN) && \
         defined(CONFIG_XEN_INTERFACE_VERSION) &&  !defined(__XEN_INTERFACE_VERSION__)
     #define __XEN_INTERFACE_VERSION__ CONFIG_XEN_INTERFACE_VERSION
@@ -1365,7 +1365,7 @@
         #
         RET=1
         VERBOSE=$7
-        FILE="linux/autoconf.h"
+        FILE="generated/autoconf.h"
 
         if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
             #
@@ -1419,7 +1419,7 @@
         #
         RET=1
         VERBOSE=$7
-        FILE="linux/autoconf.h"
+        FILE="generated/autoconf.h"
 
         if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
             #
