diff -urN NVIDIA-Linux-x86_64-190.53-pkg2/usr/src/nv/conftest.sh NVIDIA-Linux-x86_64-190.53-pkg2.new/usr/src/nv/conftest.sh
--- NVIDIA-Linux-x86_64-190.53-pkg2/usr/src/nv/conftest.sh	2009-12-09 18:41:17.000000000 -0600
+++ NVIDIA-Linux-x86_64-190.53-pkg2.new/usr/src/nv/conftest.sh	2009-12-22 04:37:34.000000000 -0600
@@ -32,14 +32,14 @@
     # CONFIG_XEN and CONFIG_PARAVIRT are present, text_xen() treats
     # the kernel as a stand-alone kernel.
     #
-    FILE="linux/autoconf.h"
+    FILE="generated/autoconf.h"
 
     if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
         #
         # We are looking at a configured source tree; verify
         # that it's not a Xen kernel.
         #
-        echo "#include <linux/autoconf.h>
+        echo "#include <generated/autoconf.h>
         #if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT)
         #error CONFIG_XEN defined!
         #endif
@@ -111,7 +111,7 @@
     fi
 }
 
-CONFTEST_PREAMBLE="#include <linux/autoconf.h>
+CONFTEST_PREAMBLE="#include <generated/autoconf.h>
     #if defined(CONFIG_XEN) && \
         defined(CONFIG_XEN_INTERFACE_VERSION) &&  !defined(__XEN_INTERFACE_VERSION__)
     #define __XEN_INTERFACE_VERSION__ CONFIG_XEN_INTERFACE_VERSION
@@ -1375,7 +1375,7 @@
         #
         RET=1
         VERBOSE=$6
-        FILE="linux/autoconf.h"
+        FILE="generated/autoconf.h"
 
         if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
             #
@@ -1429,7 +1429,7 @@
         #
         RET=1
         VERBOSE=$6
-        FILE="linux/autoconf.h"
+        FILE="generated/autoconf.h"
 
         if [ -f $HEADERS/$FILE -o -f $OUTPUT/include/$FILE ]; then
             #
