--- a/kernel/conftest.sh	2013-02-04 15:10:20.669633666 -0500
+++ b/kernel/conftest.sh	2013-02-04 15:10:34.283334673 -0500
@@ -1701,7 +1701,7 @@
                 SUBLEVEL=$(grep "^SUBLEVEL =" $MAKEFILE | cut -d " " -f 3)
 
                 if [ -n "$PATCHLEVEL" -a $PATCHLEVEL -ge 6 \
-                        -a -n "$SUBLEVEL" -a $SUBLEVEL -le 5 ]; then
+                        -a -n "$SUBLEVEL" -a $SUBLEVEL -le 6 ]; then
                     SELECTED_MAKEFILE=Makefile.kbuild
                     RET=0
                 fi
