diff --git a/kernel/nv-drm.c b/kernel/nv-drm.c
index 1e41c6e..fb8d1bd 100644
--- a/kernel/nv-drm.c
+++ b/kernel/nv-drm.c
@@ -201,12 +201,7 @@ RM_STATUS NV_API_CALL nv_alloc_os_descriptor_handle(
     memset(&nv_obj->base, 0, sizeof(nv_obj->base));
     nv_obj->pages = pages;
 
-    ret = drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
-    if (ret)
-    {
-        status = RM_ERR_OPERATING_SYSTEM;
-        goto done;
-    }
+    drm_gem_private_object_init(nvl->drm, &nv_obj->base, size);
 
     ret = drm_gem_handle_create(file_priv, &nv_obj->base, handle);
     if (ret)
diff --git a/kernel/nv-linux.h b/kernel/nv-linux.h
index baa3e38..c99c5dc 100644
--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -1037,7 +1037,7 @@ static inline int nv_execute_on_all_cpus(void (*func)(void *info), void *info)
 #endif
 
 #if !defined(NV_VMWARE)
-#define NV_NUM_PHYSPAGES                num_physpages
+#define NV_NUM_PHYSPAGES                get_num_physpages()
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
