--- NVIDIA-Linux-x86_64-190.42-pkg2/usr/src/nv/nvacpi.c.orig	2009-12-14 02:52:35.000000000 +0100
+++ NVIDIA-Linux-x86_64-190.42-pkg2/usr/src/nv/nvacpi.c	2009-12-14 02:51:45.000000000 +0100
@@ -507,8 +507,14 @@
 
     *handlesPresent = 0;
 
+    #if ACPI_CA_VERSION < 0x20091112
     acpi_walk_namespace(ACPI_TYPE_DEVICE, ACPI_ROOT_OBJECT, 
                         ACPI_UINT32_MAX, nv_acpi_find_methods, NULL, NULL);
+    #else
+    acpi_walk_namespace(ACPI_TYPE_DEVICE, ACPI_ROOT_OBJECT, 
+                        ACPI_UINT32_MAX, nv_acpi_find_methods, NULL, NULL, NULL);
+    #endif
+
 
     if (nvif_handle)
     {
