diff -Naur NVIDIA-Linux-x86_64-256.53-no-compat32/kernel/nv.c NVIDIA-Linux-x86_64-256.53-no-compat32.patch/kernel/nv.c
--- NVIDIA-Linux-x86_64-256.53-no-compat32/kernel/nv.c	2010-08-28 05:28:03.000000000 +0200
+++ NVIDIA-Linux-x86_64-256.53-no-compat32.patch/kernel/nv.c	2010-10-27 13:30:54.202322884 +0200
@@ -423,7 +423,6 @@
 static struct file_operations nv_fops = {
     .owner     = THIS_MODULE,
     .poll      = nv_kern_poll,
-    .ioctl     = nv_kern_ioctl,
 #if defined(HAVE_UNLOCKED_IOCTL)
     .unlocked_ioctl = nv_kern_unlocked_ioctl,
 #endif
