diff -Naur NVIDIA-Linux-x86-270.30/kernel/nv-linux.h NVIDIA-Linux-x86-270.30.patch/kernel/nv-linux.h
--- NVIDIA-Linux-x86-270.30/kernel/nv-linux.h	2011-02-25 23:34:52.000000000 +0100
+++ NVIDIA-Linux-x86-270.30.patch/kernel/nv-linux.h	2011-04-10 19:30:26.606841460 +0200
@@ -87,7 +87,6 @@
 #if !defined(KERNEL_2_4)
 #include <linux/sched.h>            /* suser(), capable() replacement   */
 #include <linux/moduleparam.h>      /* module_param()                   */
-#include <linux/smp_lock.h>         /* kernel_locked                    */
 #include <asm/tlbflush.h>           /* flush_tlb(), flush_tlb_all()     */
 #include <asm/kmap_types.h>         /* page table entry lookup          */
 #endif
