diff -Naur NVIDIA-Linux-x86_64-270.41.19-no-compat32/kernel/conftest.sh NVIDIA-Linux-x86_64-270.41.19-no-compat32.patch/kernel/conftest.sh
--- NVIDIA-Linux-x86_64-270.41.19-no-compat32/kernel/conftest.sh	2011-05-17 08:32:19.000000000 +0200
+++ NVIDIA-Linux-x86_64-270.41.19-no-compat32.patch/kernel/conftest.sh	2011-07-09 18:21:30.602664961 +0200
@@ -76,7 +76,7 @@
 }
 
 build_cflags() {
-    BASE_CFLAGS="-D__KERNEL__ \
+    BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
 -nostdinc -isystem $ISYSTEM"
 
diff -Naur NVIDIA-Linux-x86_64-270.41.19-no-compat32/kernel/nv-linux.h NVIDIA-Linux-x86_64-270.41.19-no-compat32.patch/kernel/nv-linux.h
--- NVIDIA-Linux-x86_64-270.41.19-no-compat32/kernel/nv-linux.h	2011-05-17 08:32:19.000000000 +0200
+++ NVIDIA-Linux-x86_64-270.41.19-no-compat32.patch/kernel/nv-linux.h	2011-07-09 18:22:16.317252983 +0200
@@ -34,6 +34,8 @@
 #  error This driver does not support 2.5 kernels!
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(2, 7, 0)
 #  define KERNEL_2_6
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(3, 1, 0)
+#  define KERNEL_2_6
 #else
 #  error This driver does not support development kernels!
 #endif
