diff -Naur cloog-0.17.0/Makefile.am cloog-0.17.0.patch/Makefile.am
--- cloog-0.17.0/Makefile.am	2011-12-08 12:03:08.000000000 +0100
+++ cloog-0.17.0.patch/Makefile.am	2012-06-13 14:05:03.611679108 +0200
@@ -145,7 +145,7 @@
 	examples
 
 install-data-local:
-	@test -z "$(pkgconfig_libdir)" || $(mkdir_p) "$(DESTDIR)$(pkgconfig_libdir)"
+	@test -z "$(pkgconfig_libdir)" || $(MKDIR_P) "$(DESTDIR)$(pkgconfig_libdir)"
 	$(INSTALL_DATA) $(pkgconfig_libfile) "$(DESTDIR)$(pkgconfig_libdir)/$(pkgconfig_libfile)"
 
 uninstall-local:
