diff -Naur xbmc-8ad691a/lib/libdvd/libdvdcss/configure.ac xbmc-8ad691a.patch/lib/libdvd/libdvdcss/configure.ac
--- xbmc-8ad691a/lib/libdvd/libdvdcss/configure.ac	2013-04-08 12:03:51.000000000 +0200
+++ xbmc-8ad691a.patch/lib/libdvd/libdvdcss/configure.ac	2013-04-08 14:20:14.793113682 +0200
@@ -5,7 +5,7 @@
 AC_CANONICAL_SYSTEM
 
 AM_INIT_AUTOMAKE(libdvdcss, 1.2.12)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_PROG_CC
 AC_STDC_HEADERS
