diff -Nurp libdvdcss-1.2.9.orig/src/libdvdcss.c libdvdcss-1.2.9/src/libdvdcss.c
--- libdvdcss-1.2.9.orig/src/libdvdcss.c	2004-08-13 21:40:18.000000000 +0800
+++ libdvdcss-1.2.9/src/libdvdcss.c	2008-07-02 00:09:24.000000000 +0800
@@ -417,7 +417,7 @@ extern dvdcss_t dvdcss_open ( char *psz_
         }
 
         if( p_sector[0] == 0x00 && p_sector[1] == 0x00
-             && p_sector[2] == 0x01 && p_sector[3] == 0xba )
+             && p_sector[2] == 0x01 && p_sector[3] == 0xba || !strncmp(p_sector,"DVDVIDEO",8))
         {
             goto nocache;
         }
