diff -Naur libdvdcss-1.2.10.orig/Makefile.am libdvdcss-1.2.10/Makefile.am
--- libdvdcss-1.2.10.orig/Makefile.am	2008-09-11 20:50:40.000000000 +0200
+++ libdvdcss-1.2.10/Makefile.am	2008-09-11 20:51:37.000000000 +0200
@@ -4,9 +4,3 @@
 EXTRA_DIST = libdvdcss.spec bootstrap
 
 AUTOMAKE_OPTIONS = foreign dist-bzip2 subdir-objects
-
-doc-dummy:
-
-doc: doc-dummy
-	$(MAKE) -C doc doc
-
diff -Naur libdvdcss-1.2.10.orig/Makefile.in libdvdcss-1.2.10/Makefile.in
--- libdvdcss-1.2.10.orig/Makefile.in	2008-09-11 20:50:40.000000000 +0200
+++ libdvdcss-1.2.10/Makefile.in	2008-09-11 20:52:05.000000000 +0200
@@ -189,7 +189,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src test doc
+SUBDIRS = src test
 DIST_SUBDIRS = $(SUBDIRS)
 EXTRA_DIST = libdvdcss.spec bootstrap
 AUTOMAKE_OPTIONS = foreign dist-bzip2 subdir-objects
@@ -642,10 +642,6 @@
 	tags tags-recursive uninstall uninstall-am
 
 
-doc-dummy:
-
-doc: doc-dummy
-	$(MAKE) -C doc doc
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
