diff -Naur xbmc-12.2-83bda98/xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp xbmc-12.2-83bda98.patch/xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp
--- xbmc-12.2-83bda98/xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp	2013-08-25 21:45:30.000000000 +0200
+++ xbmc-12.2-83bda98.patch/xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp	2013-08-26 01:26:41.950633846 +0200
@@ -277,7 +277,9 @@
       case 0: g_guiSettings.SetInt("audiooutput.mode", AUDIO_ANALOG ); break;
       case 1: g_guiSettings.SetInt("audiooutput.mode", AUDIO_IEC958 ); bitstream = true; break;
       case 2: g_guiSettings.SetInt("audiooutput.mode", AUDIO_HDMI   ); bitstream = true; break;
+#ifdef TARGET_RASPBERRY_PI
       case 3: g_guiSettings.SetInt("audiooutput.mode", AUDIO_BOTH   ); break;
+#endif
     }
 
     EnableSettings(AUDIO_SETTINGS_OUTPUT_TO_ALL_SPEAKERS, bitstream);
