--- a/kernel-dts/tegra210-icosa-common.dtsi	2019-04-07 16:09:43.174359334 +0200
+++ b/kernel-dts/tegra210-icosa-common.dtsi	2019-04-07 16:09:25.910889000 +0200
@@ -394,8 +394,15 @@
 	};
 	/* right joycon */
 	serial@70006040 {
-		compatible = "nvidia,tegra20-uart";
+		//compatible = "nvidia,tegra20-uart";
 		status = "okay";
+		nvidia,invert-txd;
+		nvidia,invert-rts;
+
+       joycon {
+			compatible = "nintendo,joycon-uart";
+			charge-gpios = <&gpio TEGRA_GPIO(K, 3) GPIO_ACTIVE_HIGH>;
+		};
 	};
 	serial@70006300 {
 		dma-names = "tx";
@@ -411,7 +418,15 @@
 
 	/* left joycon */
 	serial@70006200 {
-		status = "disabled";
+		status = "okay";
+
+		nvidia,invert-txd;
+		nvidia,invert-rts;
+
+		joycon {
+			compatible = "nintendo,joycon-uart";
+			charge-gpios = <&gpio TEGRA_GPIO(CC, 3) GPIO_ACTIVE_HIGH>;
+		};
 	};
 
 	mailbox@70098000 {
