diff -Naur nxengine-381af8c/nxengine/player.cpp nxengine-patch/nxengine/player.cpp
--- nxengine-381af8c/nxengine/player.cpp	2016-12-21 02:16:56.665899948 +0100
+++ nxengine-patch/nxengine/player.cpp	2016-12-21 02:16:23.237901007 +0100
@@ -733,7 +733,7 @@
 	if (player->lookaway)
 	{
 		// keys which deactivate lookaway when you are facing away from player
-		static const char actionkeys[] = \
+		static const signed char actionkeys[] = \
 			{ LEFTKEY, RIGHTKEY, UPKEY, JUMPKEY, FIREKEY, -1 };
 		
 		// stop looking away if any keys are pushed
