diff --git a/Makefile b/Makefile
index a108525..4b1847d 100644
--- a/vendor/janet/Makefile
+++ b/vendor/janet/Makefile
@@ -161,10 +161,10 @@ JANET_BOOT_OBJECTS=$(patsubst src/%.c,build/%.boot.o,$(JANET_CORE_SOURCES) $(JAN
 $(JANET_BOOT_OBJECTS): $(JANET_BOOT_HEADERS)
 
 build/%.boot.o: src/%.c $(JANET_HEADERS) $(JANET_LOCAL_HEADERS) Makefile
-	$(CC) $(BOOT_CFLAGS) -o $@ -c $<
+	$(HOST_CC) $(BOOT_CFLAGS) -o $@ -c $<
 
 build/janet_boot: $(JANET_BOOT_OBJECTS)
-	$(CC) $(BOOT_CFLAGS) -o $@ $(JANET_BOOT_OBJECTS) $(CLIBS)
+	$(HOST_CC) $(BOOT_CFLAGS) -o $@ $(JANET_BOOT_OBJECTS) $(CLIBS)
 
 # Now the reason we bootstrap in the first place
 build/c/janet.c: build/janet_boot src/boot/boot.janet
