diff -urN libgcrypt-1.4.1/Makefile.am libgcrypt-1.4.1.new/Makefile.am
--- libgcrypt-1.4.1/Makefile.am	2007-05-04 12:11:28.000000000 +0200
+++ libgcrypt-1.4.1.new/Makefile.am	2008-07-13 18:18:04.000000000 +0200
@@ -22,8 +22,8 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-random-daemon \
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
-DIST_SUBDIRS = m4 mpi cipher src doc tests
-SUBDIRS = mpi cipher src doc tests
+DIST_SUBDIRS = m4 mpi cipher src
+SUBDIRS = mpi cipher src
 EXTRA_DIST = BUGS autogen.sh README.SVN
 DISTCLEANFILES = 
 
diff -urN libgcrypt-1.4.1/Makefile.in libgcrypt-1.4.1.new/Makefile.in
--- libgcrypt-1.4.1/Makefile.in	2008-04-25 17:53:24.000000000 +0200
+++ libgcrypt-1.4.1.new/Makefile.in	2008-07-13 18:17:54.000000000 +0200
@@ -236,8 +236,8 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-random-daemon \
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
-DIST_SUBDIRS = m4 mpi cipher src doc tests
-SUBDIRS = mpi cipher src doc tests
+DIST_SUBDIRS = m4 mpi cipher src
+SUBDIRS = mpi cipher src
 EXTRA_DIST = BUGS autogen.sh README.SVN
 DISTCLEANFILES = 
 all: config.h
