diff -Naur libgcrypt-1.4.6/cipher/Makefile.am libgcrypt-1.4.6.patch/cipher/Makefile.am
--- libgcrypt-1.4.6/cipher/Makefile.am	2009-12-11 16:31:38.000000000 +0100
+++ libgcrypt-1.4.6.patch/cipher/Makefile.am	2011-05-08 03:21:56.463021968 +0200
@@ -68,7 +68,7 @@
 camellia.c camellia.h camellia-glue.c
 
 if ENABLE_O_FLAG_MUNGING
-o_flag_munging = sed -e 's/-O\([2-9s][2-9s]*\)/-O1/' -e 's/-Ofast/-O1/g'
+o_flag_munging = sed -e 's/-O\([2-9s][2-9s]\)/-O1/' -e 's/-Ofast/-O1/g'
 else
 o_flag_munging = cat
 endif
diff -Naur libgcrypt-1.4.6/cipher/Makefile.in libgcrypt-1.4.6.patch/cipher/Makefile.in
--- libgcrypt-1.4.6/cipher/Makefile.in	2010-07-13 17:42:20.000000000 +0200
+++ libgcrypt-1.4.6.patch/cipher/Makefile.in	2011-05-08 03:22:12.059208971 +0200
@@ -274,7 +274,7 @@
 camellia.c camellia.h camellia-glue.c
 
 @ENABLE_O_FLAG_MUNGING_FALSE@o_flag_munging = cat
-@ENABLE_O_FLAG_MUNGING_TRUE@o_flag_munging = sed -e 's/-O\([2-9s][2-9s]*\)/-O1/' -e 's/-Ofast/-O1/g'
+@ENABLE_O_FLAG_MUNGING_TRUE@o_flag_munging = sed -e 's/-O\([2-9s][2-9s]\)/-O1/' -e 's/-Ofast/-O1/g'
 all: all-am
 
 .SUFFIXES:
