diff -Naur libgcrypt-1.6.4/Makefile.am libgcrypt-1.6.4.patch/Makefile.am
--- libgcrypt-1.6.4/Makefile.am	2015-09-07 14:05:57.000000000 +0200
+++ libgcrypt-1.6.4.patch/Makefile.am	2016-01-03 00:00:27.392411468 +0100
@@ -24,8 +24,8 @@
 # (A suitable gitlog-to-changelog script can be found in GnuPG master.)
 GITLOG_TO_CHANGELOG=gitlog-to-changelog
 
-DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS =         compat mpi cipher random src doc tests
+DIST_SUBDIRS = m4 compat mpi cipher random src
+SUBDIRS =         compat mpi cipher random src
 
 EXTRA_DIST = autogen.sh autogen.rc README.GIT LICENSES      	           \
              ChangeLog-2011 build-aux/ChangeLog-2011 doc/ChangeLog-2011    \
