diff -Naur libgcrypt-1.5.0-old/configure.ac libgcrypt-1.5.0-new/configure.ac
--- libgcrypt-1.5.0-old/configure.ac	2011-06-29 01:53:22.000000000 -0700
+++ libgcrypt-1.5.0-new/configure.ac	2013-01-03 21:04:51.000000000 -0800
@@ -59,7 +59,7 @@
 
 AC_CONFIG_SRCDIR([src/libgcrypt.vers])
 AM_INIT_AUTOMAKE([])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_LIBOBJ_DIR([compat])
 AC_CANONICAL_HOST
