diff -Naur libgcrypt-1.4.6-old/tests/Makefile.am libgcrypt-1.4.6-new/tests/Makefile.am
--- libgcrypt-1.4.6-old/tests/Makefile.am	2010-07-13 07:05:56.000000000 -0700
+++ libgcrypt-1.4.6-new/tests/Makefile.am	2010-08-10 10:00:44.000000000 -0700
@@ -35,6 +35,7 @@
 # a built header.
 AM_CPPFLAGS = -I../src -I$(top_srcdir)/src
 AM_CFLAGS = $(GPG_ERROR_CFLAGS)
+AM_LDFLAGS = $(GPG_ERROR_LIBS)
 
 LDADD = ../src/libgcrypt.la $(DL_LIBS)
 
diff -Naur libgcrypt-1.4.6-old/tests/Makefile.in libgcrypt-1.4.6-new/tests/Makefile.in
--- libgcrypt-1.4.6-old/tests/Makefile.in	2010-07-13 08:42:21.000000000 -0700
+++ libgcrypt-1.4.6-new/tests/Makefile.in	2010-08-10 10:02:01.000000000 -0700
@@ -338,6 +338,7 @@
 # a built header.
 AM_CPPFLAGS = -I../src -I$(top_srcdir)/src
 AM_CFLAGS = $(GPG_ERROR_CFLAGS)
+AM_LDFLAGS = $(GPG_ERROR_LIBS)
 LDADD = ../src/libgcrypt.la $(DL_LIBS)
 EXTRA_DIST = README rsa-16k.key cavs_tests.sh cavs_driver.pl
 all: all-am
