--- kodi-rockchip_18.2rc1-Leia/xbmc/peripherals/bus/virtual/PeripheralBusCEC.cpp        2019-04-15 22:19:08.000000000 +0200
+++ kodi-rockchip_18.2rc1-Leia.orig/xbmc/peripherals/bus/virtual/PeripheralBusCEC.cpp   2019-05-08 22:10:51.803079157 +0200
@@ -38,6 +38,10 @@
     result.m_strLocation = deviceList[iDevicePtr].strComName;
     result.m_type        = PERIPHERAL_CEC;
 
+    // can't use the switch below, as on Asus Tinker S both the adapterType as well as the vendorId seem to be 0 (UNKNOWN) according to libcec
+    // for now, hard code it to false, as the adapter can't be removed, just as on PI's case.
+    m_bNeedsPolling = false;
+
     // override the bus type, so users don't have to reconfigure their adapters
     switch(deviceList[iDevicePtr].adapterType)
     {
