diff -Naur a/arch/arm64/include/uapi/asm/sigcontext.h b/arch/arm64/include/uapi/asm/sigcontext.h
--- a/arch/arm64/include/uapi/asm/sigcontext.h	2016-04-06 23:18:37.000000000 -0700
+++ b/arch/arm64/include/uapi/asm/sigcontext.h	2016-04-07 16:00:47.870241223 -0700
@@ -58,7 +58,7 @@
 
 struct esr_context {
 	struct _aarch64_ctx head;
-	u64 esr;
+	__u64 esr;
 };
 
 #endif /* _UAPI__ASM_SIGCONTEXT_H */
