diff -Naur a/drivers/amlogic/display/vout/tv_vout.c b/drivers/amlogic/display/vout/tv_vout.c
--- a/drivers/amlogic/display/vout/tv_vout.c	2016-04-06 23:18:37.000000000 -0700
+++ b/drivers/amlogic/display/vout/tv_vout.c	2016-04-14 02:05:33.041542772 -0700
@@ -70,8 +70,8 @@
 #ifdef CONFIG_AML_VOUT_FRAMERATE_AUTOMATION
 #define DEFAULT_POLICY_FR_AUTO	1
 static enum vmode_e mode_by_user = VMODE_INIT_NULL;
-static int fr_auto_policy = DEFAULT_POLICY_FR_AUTO;
-static int fr_auto_policy_hold = DEFAULT_POLICY_FR_AUTO;
+static int fr_auto_policy = 0;
+static int fr_auto_policy_hold = 2;
 static int fps_playing_flag;
 static enum vmode_e fps_target_mode = VMODE_INIT_NULL;
 static void policy_framerate_automation_switch_store(char *para);
@@ -1532,8 +1532,8 @@
 	}
 
 #ifdef CONFIG_AML_VOUT_FRAMERATE_AUTOMATION
-	sprintf(policy_fr_auto, "%d", DEFAULT_POLICY_FR_AUTO);
-	sprintf(policy_fr_auto_switch, "%d", DEFAULT_POLICY_FR_AUTO);
+	sprintf(policy_fr_auto, "%d", 0);
+	sprintf(policy_fr_auto_switch, "%d", 2);
 #endif
 	return 0;
 }
