From ea455841e5d6081b33170fe3ea32aed9b7b2b72c Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Mon, 10 Jul 2017 19:44:54 +0100
Subject: [PATCH] add new cards

---
 os_dep/linux/usb_intf.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/os_dep/linux/usb_intf.c b/os_dep/linux/usb_intf.c
index ef79d2e..b098e56 100644
--- a/os_dep/linux/usb_intf.c
+++ b/os_dep/linux/usb_intf.c
@@ -195,6 +195,13 @@ static struct usb_device_id rtw_usb_id_tbl[] = {
 	{USB_DEVICE(0x0846, 0x9052), .driver_info = RTL8821}, /* Netgear - A6100 */
 #endif
 
+/*=== Patched ID ===*/
+#ifdef CONFIG_RTL8812A
+	{USB_DEVICE(0x2357, 0x010d),.driver_info = RTL8812}, /* TP-Link - T4U v2 */
+	{USB_DEVICE(0x148F, 0x9097),.driver_info = RTL8812}, /* Amped Wireless ACA1 */
+	{USB_DEVICE(0x056e, 0x4007),.driver_info = RTL8821}, /* Elecom - WDC-433DU2HBK */
+#endif
+
 #ifdef CONFIG_RTL8192E
 	/*=== Realtek demoboard ===*/
 	{USB_DEVICE_AND_INTERFACE_INFO(USB_VENDER_ID_REALTEK, 0x818B, 0xff, 0xff, 0xff), .driver_info = RTL8192E}, /* Default ID */
-- 
2.7.4

