From 4037784fc342dc81c71f7b950f4f5d94c0c21e5c Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 8 Sep 2015 18:36:02 +0100
Subject: [PATCH] Fix potential issue for timeout when connecting to an AP.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 include/rtw_mlme_ext.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/rtw_mlme_ext.h b/include/rtw_mlme_ext.h
index d1d5894..7dd78ad 100644
--- a/include/rtw_mlme_ext.h
+++ b/include/rtw_mlme_ext.h
@@ -27,7 +27,7 @@
 //  So, this driver tried to extend the dwell time for each scanning channel.
 //  This will increase the chance to receive the probe response from SoftAP.
 
-#define SURVEY_TO       (100)
+#define SURVEY_TO       (0)
 #define REAUTH_TO       (300) //(50)
 #define REASSOC_TO      (300) //(50)
 //#define DISCONNECT_TO (3000)
