From 3224e74ad9c230b74a658e80dad66ab95c9e2ef5 Mon Sep 17 00:00:00 2001
From: Vladimir Menshakov <vladimir.menshakov@gmail.com>
Date: Tue, 16 May 2017 13:40:44 +0100
Subject: [PATCH] added missing includes for linux-4.11.x

---
 include/osdep_service_linux.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/osdep_service_linux.h b/include/osdep_service_linux.h
index 12fdd7e..560762d 100644
--- a/include/osdep_service_linux.h
+++ b/include/osdep_service_linux.h
@@ -68,6 +68,11 @@
 	#include <linux/limits.h>
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0))
+	#include <linux/signal.h>
+	#include <linux/sched/signal.h>
+#endif
+
 #ifdef RTK_DMP_PLATFORM
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12))
 	#include <linux/pageremap.h>

