From b68705a507c47dc2ff2cc6121830af315fbcdc83 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 11 Jul 2015 02:31:07 +0100
Subject: [PATCH] Disable power saving by default

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e97fbbf..7981e59 100644
--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ CONFIG_SDIO_HCI = n
 CONFIG_GSPI_HCI = n
 
 CONFIG_MP_INCLUDED = y
-CONFIG_POWER_SAVING = y
+CONFIG_POWER_SAVING = n
 CONFIG_USB_AUTOSUSPEND = n
 CONFIG_HW_PWRP_DETECTION = n
 CONFIG_WIFI_TEST = n
