From 14eb40c9bcbb68044361334d5f95a4f63ebc8775 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Mon, 13 Mar 2017 19:47:38 +0000
Subject: [PATCH] RTL8812AU: Fix build for API changes in kernel 4.11

---
 include/osdep_service.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/osdep_service.h b/include/osdep_service.h
index 82e27c5..d866551 100644
--- a/include/osdep_service.h
+++ b/include/osdep_service.h
@@ -20,6 +20,10 @@
 #ifndef __OSDEP_SERVICE_H_
 #define __OSDEP_SERVICE_H_
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#include <linux/sched/signal.h>
+#endif
 
 #define _FAIL		0
 #define _SUCCESS	1
-- 
2.7.4

