diff -Naur RTL8812AU-b22cbdf/os_dep/linux/usb_intf.c RTL8812AU-b22cbdf.patch/os_dep/linux/usb_intf.c
--- RTL8812AU-b22cbdf/os_dep/linux/usb_intf.c	2014-04-12 01:49:09.000000000 +0200
+++ RTL8812AU-b22cbdf.patch/os_dep/linux/usb_intf.c	2014-09-14 15:05:39.598324411 +0200
@@ -285,6 +285,7 @@
 	/*=== Customer ID ===*/
 	{USB_DEVICE(0x7392, 0xA811),.driver_info = RTL8821}, /* Edimax - Edimax */
 	{USB_DEVICE(0x2001, 0x3314),.driver_info = RTL8821}, /* D-Link - Cameo */
+	{USB_DEVICE(0x0846, 0x9052),.driver_info = RTL8821}, /* Netgear - A6100 */
 #endif
 
 #ifdef CONFIG_RTL8192E
