diff -Naur linux-3.10.16/drivers/staging/rtl8712/usb_intf.c linux-3.10.16.patch/drivers/staging/rtl8712/usb_intf.c
--- linux-3.10.16/drivers/staging/rtl8712/usb_intf.c	2013-10-14 01:08:56.000000000 +0200
+++ linux-3.10.16.patch/drivers/staging/rtl8712/usb_intf.c	2013-10-16 13:27:44.032951265 +0200
@@ -92,6 +92,7 @@
 	{USB_DEVICE(0x0DF6, 0x005B)},
 	{USB_DEVICE(0x0DF6, 0x005D)},
 	{USB_DEVICE(0x0DF6, 0x0063)},
+	{USB_DEVICE(0x0DF6, 0x006C)},
 	/* Sweex */
 	{USB_DEVICE(0x177F, 0x0154)},
 	/* Thinkware */
