diff --git a/Makefile b/Makefile
index 130afd8..480032e 100644
--- a/Makefile
+++ b/Makefile
@@ -43,8 +43,8 @@ clean:
 	rm -f $(OBJS) $(TARGET)
 
 install: $(TARGET)
-	install -D -g root -o root $(TARGET) $(TARGET_DIR)/sbin/$(TARGET)
-	install -D -g root -o root $(TARGET).1 $(TARGET_DIR)/share/man/man1/$(TARGET).1
+	install -D $(TARGET) $(TARGET_DIR)/sbin/$(TARGET)
+	install -D $(TARGET).1 $(TARGET_DIR)/share/man/man1/$(TARGET).1
 
 hd-idle.o:     hd-idle.c
 
