diff -Naur ntp-4.2.6p5-old/configure.ac ntp-4.2.6p5-new/configure.ac
--- ntp-4.2.6p5-old/configure.ac	2011-09-22 20:17:25.000000000 -0700
+++ ntp-4.2.6p5-new/configure.ac	2013-01-03 19:05:39.000000000 -0800
@@ -40,7 +40,7 @@
 dnl the 'host' machine is where the resulting stuff runs.
 AC_DEFINE_UNQUOTED([STR_SYSTEM], ["$host"],
     [canonical system (cpu-vendor-os) of where we should run])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 dnl AC_ARG_PROGRAM
 
 ntp_atom_ok=${ntp_atom_ok=no}
