diff --git a/setup.py b/setup.py
index 8eb0b3c..277e44a 100755
--- a/setup.py
+++ b/setup.py
@@ -341,9 +341,7 @@
         return True if value in configuration.get(option, []) else None
 
     def initialize_options(self):
-        self.disable_platform_guessing = self.check_configuration(
-            "platform-guessing", "disable"
-        )
+        self.disable_platform_guessing = True
         self.add_imaging_libs = ""
         build_ext.initialize_options(self)
         for x in self.feature:
@@ -1007,6 +1005,7 @@
 
 try:
     setup(
+        version=PILLOW_VERSION,
         cmdclass={"build_ext": pil_build_ext},
         ext_modules=ext_modules,
         zip_safe=not (debug_build() or PLATFORM_MINGW),
