diff --git a/setup.py b/setup.py
index 8eb0b3c..277e44a 100755
--- a/setup.py
+++ b/setup.py
@@ -253,7 +253,7 @@ class pil_build_ext(build_ext):
     ] + [('add-imaging-libs=', None, 'Add libs to _imaging build')]
 
     def initialize_options(self):
-        self.disable_platform_guessing = None
+        self.disable_platform_guessing = True
         self.add_imaging_libs = ""
         build_ext.initialize_options(self)
         for x in self.feature:
