diff --git a/setup.py b/setup.py
index 4a0ad86c5..c07e1a938 100755
--- a/setup.py
+++ b/setup.py
@@ -194,7 +194,7 @@ class pil_build_ext(build_ext):
     ]

     def initialize_options(self):
-        self.disable_platform_guessing = None
+        self.disable_platform_guessing = True
         build_ext.initialize_options(self)
         for x in self.feature:
             setattr(self, 'disable_%s' % x, None)
