commit 0f8a281027f852ed5968361bcb7420ed242d0476
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 19 22:15:14 2013 +0300

    vdr 2.1.2 buildfix

diff --git a/pages/recordings.ecpp b/pages/recordings.ecpp
index 80f145b..6da5285 100644
--- a/pages/recordings.ecpp
+++ b/pages/recordings.ecpp
@@ -79,7 +79,7 @@ for (deletions_type::const_iterator it = deletions.begin(); it != deletions.end(
 deletions.clear();
 
 int FreeMB, UsedMB;
-int Percent = VideoDiskSpace(&FreeMB, &UsedMB);
+int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB, &UsedMB);
 int Minutes = int(double(FreeMB) / MB_PER_MINUTE);
 int Hours = Minutes / 60;
 Minutes %= 60;
diff --git a/recman.cpp b/recman.cpp
index 6a2fc00..9ed88f0 100644
--- a/recman.cpp
+++ b/recman.cpp
@@ -112,7 +112,7 @@ namespace vdrlive {
 		if (found == string::npos)
 			return false;
 
-		string newname = string(VideoDirectory) + "/" + name + oldname.substr(found);
+		string newname = string(cVideoDirectory::Name()) + "/" + name + oldname.substr(found);
 
 		if (!MoveDirectory(oldname.c_str(), newname.c_str(), copy)) {
 			esyslog("[LIVE]: renaming failed from '%s' to '%s'", oldname.c_str(), newname.c_str());
diff --git a/tntconfig.cpp b/tntconfig.cpp
index 3325776..161fdc4 100644
--- a/tntconfig.cpp
+++ b/tntconfig.cpp
@@ -279,7 +279,7 @@ namespace vdrlive {
 		// inserted by 'tadi' -- not verified, not counterchecked yet!
 		//app.mapUrl("^/vlc/(.+)", "static@tntnet")
 		//	.setPathInfo("/$1")
-		//	.pushArg(string("DocumentRoot=") + VideoDirectory);
+		//	.pushArg(string("DocumentRoot=") + cVideoDirectory::Name());
 
 		// the following selects the theme specific 'theme.css' file
 		// inserted by 'tadi' -- verified with above, but not counterchecked yet!
diff --git a/tools.cpp b/tools.cpp
index c744d06..cb3529e 100644
--- a/tools.cpp
+++ b/tools.cpp
@@ -365,7 +365,7 @@ namespace vdrlive {
 			stat(source.c_str(), &st1);
 			stat(target.c_str(),&st2);
 			if (!copy && (st1.st_dev == st2.st_dev)) {
-				if (!RenameVideoFile(source.c_str(), target.c_str())) {
+				if (!cVideoDirectory::RenameVideoFile(source.c_str(), target.c_str())) {
 					esyslog("[LIVE]: rename failed from %s to %s", source.c_str(), target.c_str());
 					return false;
 				}
@@ -461,7 +461,7 @@ namespace vdrlive {
 						size_t found = source.find_last_of(delim);
 						if (found != std::string::npos) {
 							source = source.substr(0, found);
-							while (source != VideoDirectory) {
+							while (source != cVideoDirectory::Name()) {
 								found = source.find_last_of(delim);
 								if (found == std::string::npos)
 									break;
@@ -478,7 +478,7 @@ namespace vdrlive {
 					size_t found = target.find_last_of(delim);
 					if (found != std::string::npos) {
 						target = target.substr(0, found);
-						while (target != VideoDirectory) {
+						while (target != cVideoDirectory::Name()) {
 							found = target.find_last_of(delim);
 							if (found == std::string::npos)
 								break;
