From d95f6d719298b4586fc4b889612423be67b52e88 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 16 Jun 2014 12:06:29 +0300
Subject: [PATCH] dont build i18n

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index ab2f2b5..923d184 100644
--- a/Makefile
+++ b/Makefile
@@ -94,7 +94,7 @@ WEBLIBS	   = pages/libpages.a css/libcss.a javascript/libjavascript.a
 
 ### Default rules:
 
-all: libvdr-$(PLUGIN).so $(I18NTARG)
+all: libvdr-$(PLUGIN).so
 
 .PHONY: all dist clean subdirs $(SUBDIRS) PAGES
 
-- 
1.7.2.5

