diff -Naur RTL2832-2.2.2_kernel-3.0.0/Makefile RTL2832-2.2.2_kernel-3.0.0.patch/Makefile
--- RTL2832-2.2.2_kernel-3.0.0/Makefile	2011-12-20 19:27:43.000000000 +0100
+++ RTL2832-2.2.2_kernel-3.0.0.patch/Makefile	2011-12-20 20:30:31.412007850 +0100
@@ -4,10 +4,10 @@
 # Choose here wich include file to use: from kernel 3.0.0 (good for 3.1.0) or from kernel 3.2.0
 
 # kernel 3.0.0 / 3.1.0
-INCLUDE_EXTRA_DVB := include-300
+#INCLUDE_EXTRA_DVB := include-300
 
 # kernel 3.2.0
-#INCLUDE_EXTRA_DVB := include-320
+INCLUDE_EXTRA_DVB := include-320
 
 # ----------------------------------------
 
