--- a/drivers/media/Kconfig
+++ b/drivers/media/Kconfig
@@ -235,5 +235,6 @@ source "drivers/media/i2c/Kconfig"
 source "drivers/media/spi/Kconfig"
 source "drivers/media/tuners/Kconfig"
 source "drivers/media/dvb-frontends/Kconfig"
+source "drivers/media/amlogic/Kconfig"
 
 endif # MEDIA_SUPPORT
--- a/drivers/media/Makefile
+++ b/drivers/media/Makefile
@@ -25,7 +25,7 @@
 obj-y += rc/
 
 obj-$(CONFIG_CEC_CORE) += cec/
-
+obj-y += amlogic/
 #
 # Finally, merge the drivers that require the core
 #
--- /dev/null
+++ b/drivers/media/amlogic/Kconfig
@@ -0,0 +1,8 @@
+#
+# Amlogic driver configuration
+#
+menu "Amlogic Device Drivers"
+
+source "drivers/media/amlogic/video_dev/Kconfig"
+
+endmenu
--- /dev/null
+++ b/drivers/media/amlogic/Makefile
@@ -0,0 +1,5 @@
+##########################################
+########## Amlogic Drivers ###############
+##########################################
+
+obj-$(CONFIG_V4L_AMLOGIC_VIDEO) += video_dev/
