--- a/drivers/video/sunxi/disp2/disp/dev_fb.c	2015-01-27 03:29:32.000000000 +0100
+++ b/drivers/video/sunxi/disp2/disp/dev_fb.c	2015-12-28 10:45:14.276418445 +0100
@@ -874,7 +874,7 @@
 
 	case FBIO_WAITFORVSYNC:
 	{
-		//ret = fb_wait_for_vsync(info);
+		ret = fb_wait_for_vsync(info);
 		break;
 	}
 
diff -Nur a/drivers/video/sunxi/disp2/disp/dev_disp.c b/drivers/video/sunxi/disp2/disp/dev_disp.c
--- a/drivers/video/sunxi/disp2/disp/dev_disp.c	2016-04-12 12:21:19.000000000 +0200
+++ b/drivers/video/sunxi/disp2/disp/dev_disp.c	2016-05-23 22:06:41.937693271 +0200
@@ -932,7 +932,7 @@
 	}
 
 	para->disp_int_process       = disp_sync_finish_process;
-	//para->vsync_event            = drv_disp_vsync_event;
+	para->vsync_event            = drv_disp_vsync_event;
 	para->start_process          = start_process;
 	//para.capture_event          = capture_event;
 
