diff --git a/linux-tbs-drivers/linux/drivers/media/video/et61x251/et61x251_core.c b/linux-tbs-drivers/linux/drivers/media/video/et61x251/et61x251_core.c
index 3146eec..c3a112e 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/et61x251/et61x251_core.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/et61x251/et61x251_core.c
@@ -1550,7 +1550,7 @@ static int et61x251_mmap(struct file* filp, struct vm_area_struct *vma)
 	}
 
 	vma->vm_flags |= VM_IO;
-	vma->vm_flags |= VM_RESERVED;
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
 
 	pos = cam->frame[i].bufmem;
 	while (size > 0) { /* size is page-aligned */
diff --git a/linux-tbs-drivers/linux/drivers/media/video/meye.c b/linux-tbs-drivers/linux/drivers/media/video/meye.c
index 2b7b1db..c80a3c5 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/meye.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/meye.c
@@ -1708,7 +1708,7 @@ static int meye_mmap(struct file *file, struct vm_area_struct *vma)
 
 	vma->vm_ops = &meye_vm_ops;
 	vma->vm_flags &= ~VM_IO;	/* not I/O memory */
-	vma->vm_flags |= VM_RESERVED;	/* avoid to swap out this VMA */
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;	/* avoid to swap out this VMA */
 	vma->vm_private_data = (void *) (offset / gbufsize);
 	meye_vm_open(vma);
 
diff --git a/linux-tbs-drivers/linux/drivers/media/video/omap/omap_vout.c b/linux-tbs-drivers/linux/drivers/media/video/omap/omap_vout.c
index b5ef362..b8119fe 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/omap/omap_vout.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/omap/omap_vout.c
@@ -896,7 +896,7 @@ static int omap_vout_mmap(struct file *file, struct vm_area_struct *vma)
 
 	q->bufs[i]->baddr = vma->vm_start;
 
-	vma->vm_flags |= VM_RESERVED;
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
 	vma->vm_page_prot = pgprot_writecombine(vma->vm_page_prot);
 	vma->vm_ops = &omap_vout_vm_ops;
 	vma->vm_private_data = (void *) vout;
diff --git a/linux-tbs-drivers/linux/drivers/media/video/sn9c102/sn9c102_core.c b/linux-tbs-drivers/linux/drivers/media/video/sn9c102/sn9c102_core.c
index 32aa3c2..49d7042 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/sn9c102/sn9c102_core.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/sn9c102/sn9c102_core.c
@@ -2130,7 +2130,7 @@ static int sn9c102_mmap(struct file* filp, struct vm_area_struct *vma)
 	}
 
 	vma->vm_flags |= VM_IO;
-	vma->vm_flags |= VM_RESERVED;
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
 
 	pos = cam->frame[i].bufmem;
 	while (size > 0) { /* size is page-aligned */
diff --git a/linux-tbs-drivers/linux/drivers/media/video/usbvision/usbvision-video.c b/linux-tbs-drivers/linux/drivers/media/video/usbvision/usbvision-video.c
index 5a74f5e..5791c67 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/usbvision/usbvision-video.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/usbvision/usbvision-video.c
@@ -1091,7 +1091,7 @@ static int usbvision_v4l2_mmap(struct file *file, struct vm_area_struct *vma)
 
 	/* VM_IO is eventually going to replace PageReserved altogether */
 	vma->vm_flags |= VM_IO;
-	vma->vm_flags |= VM_RESERVED;	/* avoid to swap out this VMA */
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;	/* avoid to swap out this VMA */
 
 	pos = usbvision->frame[i].data;
 	while (size > 0) {
diff --git a/linux-tbs-drivers/linux/drivers/media/video/videobuf-dma-sg.c b/linux-tbs-drivers/linux/drivers/media/video/videobuf-dma-sg.c
index f163a3a..844adad 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/videobuf-dma-sg.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/videobuf-dma-sg.c
@@ -586,7 +586,7 @@ static int __videobuf_mmap_mapper(struct videobuf_queue *q,
 	map->count    = 1;
 	map->q        = q;
 	vma->vm_ops   = &videobuf_vm_ops;
-	vma->vm_flags |= VM_DONTEXPAND | VM_RESERVED;
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
 	vma->vm_flags &= ~VM_IO; /* using shared anonymous pages */
 	vma->vm_private_data = map;
 	dprintk(1, "mmap %p: q=%p %08lx-%08lx pgoff %08lx bufs %d-%d\n",
diff --git a/linux-tbs-drivers/linux/drivers/media/video/videobuf-vmalloc.c b/linux-tbs-drivers/linux/drivers/media/video/videobuf-vmalloc.c
index 9f5632c..21375fa 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/videobuf-vmalloc.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/videobuf-vmalloc.c
@@ -274,7 +274,7 @@ static int __videobuf_mmap_mapper(struct videobuf_queue *q,
 	}
 
 	vma->vm_ops          = &videobuf_vm_ops;
-	vma->vm_flags       |= VM_DONTEXPAND | VM_RESERVED;
+	vma->vm_flags       |= VM_DONTEXPAND | VM_DONTDUMP;
 	vma->vm_private_data = map;
 
 	dprintk(1, "mmap %p: q=%p %08lx-%08lx (%lx) pgoff %08lx buf %d\n",
diff --git a/linux-tbs-drivers/linux/drivers/media/video/videobuf2-memops.c b/linux-tbs-drivers/linux/drivers/media/video/videobuf2-memops.c
index f542538..f330bf0 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/videobuf2-memops.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/videobuf2-memops.c
@@ -176,7 +176,7 @@ int vb2_mmap_pfn_range(struct vm_area_struct *vma, unsigned long paddr,
 		return ret;
 	}
 
-	vma->vm_flags		|= VM_DONTEXPAND | VM_RESERVED;
+	vma->vm_flags		|= VM_DONTEXPAND | VM_DONTDUMP;
 	vma->vm_private_data	= priv;
 	vma->vm_ops		= vm_ops;
 
diff --git a/linux-tbs-drivers/linux/drivers/media/video/vino.c b/linux-tbs-drivers/linux/drivers/media/video/vino.c
index 52a0a37..7a40e12 100755
--- a/linux-tbs-drivers/linux/drivers/media/video/vino.c
+++ b/linux-tbs-drivers/linux/drivers/media/video/vino.c
@@ -3950,7 +3950,7 @@ found:
 
 	fb->map_count = 1;
 
-	vma->vm_flags |= VM_DONTEXPAND | VM_RESERVED;
+	vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
 	vma->vm_flags &= ~VM_IO;
 	vma->vm_private_data = fb;
 	vma->vm_file = file;
