From: Stefan Saraev <stefan@saraev.ca>
Subject: [PATCH] don't build unused plugins

--- a/Makefile
+++ b/Makefile
@@ -16,9 +16,9 @@
 ### edit one of these lines to '1', if you don't want the addon epgsearchonly, 
 ### conflictcheckonly or quickepgsearch
 
-WITHOUT_EPGSEARCHONLY=0
-WITHOUT_CONFLICTCHECKONLY=0
-WITHOUT_QUICKSEARCH=0
+WITHOUT_EPGSEARCHONLY=1
+WITHOUT_CONFLICTCHECKONLY=1
+WITHOUT_QUICKSEARCH=1
 
 ### edit this to '0' if you don't want epgsearch to auto config itself
 AUTOCONFIG=1

