From f7fc4a0c74bd13b35fcefb913438000c220302d8 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 18 Mar 2016 11:04:06 +0100
Subject: [PATCH] don't build unused plugins

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 2c5fc47..32a605d 100644
--- a/Makefile
+++ b/Makefile
@@ -16,9 +16,9 @@
 ### edit one of these lines to '1', if you don't want the addon epgsearchonly,
 ### conflictcheckonly or quickepgsearch
 
-WITHOUT_EPGSEARCHONLY = 0
-WITHOUT_CONFLICTCHECKONLY = 0
-WITHOUT_QUICKSEARCH = 0
+WITHOUT_EPGSEARCHONLY = 1
+WITHOUT_CONFLICTCHECKONLY = 1
+WITHOUT_QUICKSEARCH = 1
 
 ### edit this to '0' if you don't want epgsearch to auto config itself
 AUTOCONFIG = 1
-- 
2.51.0

