From fc39306b438f44189b492e6c04f156627666a53c Mon Sep 17 00:00:00 2001
From: mglae <mglmail@arcor.de>
Date: Sun, 24 Jan 2021 15:57:32 +0100
Subject: [PATCH] support locale - no docs

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 32a605d..889e053 100644
--- a/Makefile
+++ b/Makefile
@@ -191,7 +191,7 @@ LIBS4    =
 
 ### The main target:
 
-all: $(ALL) i18n docs
+all: $(ALL) i18n
 
 ### Implicit rules:
 
-- 
2.51.0

