diff -Naur elisa-0.5.31/elisa/extern/coherence/inotify.py elisa-0.5.31.patch/elisa/extern/coherence/inotify.py
--- elisa-0.5.31/elisa/extern/coherence/inotify.py	2009-03-09 09:31:38.000000000 +0100
+++ elisa-0.5.31.patch/elisa/extern/coherence/inotify.py	2009-03-09 19:14:27.000000000 +0100
@@ -129,7 +129,7 @@
 
             obj = super(INotify, cls).__new__(cls, *args, **kwargs)
             try:
-                obj.libc = ctypes.CDLL("libc.so.6")
+                obj.libc = ctypes.CDLL("libc.so.0")
             except:
                 raise SystemError, "libc not found, INotify support disabled"
 
