From 61936357ef7d796c3f4fe95782309268be9e6a31 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Sat, 20 Jun 2020 19:37:42 +0200
Subject: [PATCH] Fix attribute

---
 include/iscsi_if.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/iscsi_if.h b/include/iscsi_if.h
index 20f2bc2961e9..6633bc528555 100644
--- a/include/iscsi_if.h
+++ b/include/iscsi_if.h
@@ -327,7 +327,7 @@ struct iscsi_iface_param_info {
 	uint8_t iface_type;	/* IPv4 or IPv6 */
 	uint8_t param_type;	/* iscsi_param_type */
 	uint8_t value[0];	/* length sized value follows */
-} __packed;
+} __attribute__((__packed__));
 
 /*
  * To keep the struct iscsi_uevent size the same for userspace code
-- 
2.27.0

