diff --git a/drivers/Makefile b/drivers/Makefile
index 6ea96ec16..cc83abd9a 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -9,7 +9,5 @@
 # FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 # more details.
 
-subdir-ccflags-y += -Werror
-
 obj-$(CONFIG_NVPMODEL_EMC)      += nvpmodel/
 obj-$(CONFIG_TEGRA_NVLINK)     += nvlink/
diff --git a/drivers/video/Makefile b/drivers/video/Makefile
index 200a62c22..49b37c0e4 100644
--- a/drivers/video/Makefile
+++ b/drivers/video/Makefile
@@ -1,4 +1,2 @@
-subdir-ccflags-y += -Werror
-
 obj-y += backlight/
 obj-y += tegra/
diff --git a/drivers/video/tegra/dc/Makefile b/drivers/video/tegra/dc/Makefile
index a8157a8ea..a7c93a980 100644
--- a/drivers/video/tegra/dc/Makefile
+++ b/drivers/video/tegra/dc/Makefile
@@ -1,5 +1,4 @@
 GCOV_PROFILE := y
-subdir-ccflags-y := -Werror
 ccflags-y += -I$(overlay)/drivers/video/tegra/host
 ccflags-y += -I$(overlay)/drivers/video/tegra/dc
 ccflags-y += -I$(overlay)/drivers/platform/tegra
@@ -60,7 +59,6 @@ obj-$(CONFIG_TEGRA_CEC_SUPPORT) += ../../../misc/tegra-cec/
 obj-y += crc.o
 
 GCOV_PROFILE := y
-subdir-ccflags-y := -Werror
 ccflags-y += -I$(srctree.nvidia)/drivers/video/tegra/dc
 ccflags-y += -I$(srctree.nvidia)/drivers/video/tegra/dc/nvdisp
 
diff --git a/drivers/video/tegra/dc/hdcp/hdmihdcp.c b/drivers/video/tegra/dc/hdcp/hdmihdcp.c
index 3e8656e5b..cc1cf5657 100644
--- a/drivers/video/tegra/dc/hdcp/hdmihdcp.c
+++ b/drivers/video/tegra/dc/hdcp/hdmihdcp.c
@@ -1935,7 +1935,9 @@
 	u16 rx_status = 0;
 	uint64_t *pkt = NULL;
 	int err = 0;
+#ifdef CONFIG_TRUSTED_LITTLE_KERNEL
 	unsigned char nonce[HDCP_NONCE_SIZE];
+#endif
 
 	pkt = kzalloc(PKT_SIZE, GFP_KERNEL);
 
