diff --git a/drivers/video/tegra/host/interrupt_syncpt.c b/drivers/video/tegra/host/interrupt_syncpt.c
index 3bea91c8b..aab9c54cd 100644
--- a/drivers/video/tegra/host/interrupt_syncpt.c
+++ b/drivers/video/tegra/host/interrupt_syncpt.c
@@ -31,14 +31,15 @@ EXPORT_SYMBOL(nvhost_interrupt_syncpt_get_syncpt_index);
 phys_addr_t nvhost_interrupt_syncpt_get_syncpt_addr(
     struct nvhost_interrupt_syncpt *is)
 {
-    phys_addr_t base;
-    size_t size;
-    u32 offset;
+       return 0;
+    //phys_addr_t base;
+    //size_t size;
+    //u32 offset;
 
-    nvhost_syncpt_unit_interface_get_aperture(is->host1x_pdev, &base, &size);
-    offset = nvhost_syncpt_unit_interface_get_byte_offset(is->syncpt);
+    //nvhost_syncpt_unit_interface_get_aperture(is->host1x_pdev, &base, &size);
+    //offset = nvhost_syncpt_unit_interface_get_byte_offset(is->syncpt);
 
-    return base + offset;
+    //return base + offset;
 }
 EXPORT_SYMBOL(nvhost_interrupt_syncpt_get_syncpt_addr);
 
diff --git a/sound/soc/tegra-alt/machine_drivers/tegra_machine_driver_mobile.c b/sound/soc/tegra-alt/machine_drivers/tegra_machine_driver_mobile.c
index e3428a665..b748455af 100644
--- a/sound/soc/tegra-alt/machine_drivers/tegra_machine_driver_mobile.c
+++ b/sound/soc/tegra-alt/machine_drivers/tegra_machine_driver_mobile.c
@@ -813,11 +813,11 @@
 		return err;
 	}
 
-	err = rt5659_set_jack_detect(rtd->codec, jack);
+	/*err = rt5659_set_jack_detect(rtd->codec, jack);
 	if (err) {
 		dev_err(card->dev, "Failed to set jack for RT565x: %d\n", err);
 		return err;
-	}
+	}*/
 
 	/* single button supporting play/pause */
 	snd_jack_set_key(jack->jack, SND_JACK_BTN_0, KEY_MEDIA);
