diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-03-03 12:18:48.000000000 -0800
+++ b/CMakeLists.txt	2016-04-11 12:38:06.499927119 -0700
@@ -6,8 +6,8 @@
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
 
 find_package(kodi REQUIRED)
-#find_package(platform REQUIRED)
-find_package(p8-platform REQUIRED)
+find_package(platform REQUIRED)
+#find_package(p8-platform REQUIRED)
 find_package(kodiplatform REQUIRED)
 find_package(CURL REQUIRED)
 #find_package(yajl REQUIRED) #why is it so damn difficult to include Kodi's libraries???
@@ -94,9 +94,9 @@
 endif()
 
 include_directories(${INCLUDES}
-                    ${CURL_INCLUDE_DIR}
+                    ${CURL_INCLUDE_DIR}  
                     ${KODI_INCLUDE_DIR}
-                    ${p8-platform_INCLUDE_DIRS}
+                    ${platform_INCLUDE_DIRS}
                     ${kodiplatform_INCLUDE_DIRS})
 #                    ${YAJL_INCLUDE_DIR}
 #                    ${platform_INCLUDE_DIRS}           
diff -Naur a/src/WavforHue_Kodi.h b/src/WavforHue_Kodi.h
--- a/src/WavforHue_Kodi.h	2016-03-03 12:18:48.000000000 -0800
+++ b/src/WavforHue_Kodi.h	2016-04-11 12:38:06.500927127 -0700
@@ -23,6 +23,6 @@
 // Is this worth it?
 //#include <libKODI_guilib.h>
 #include <libXBMC_addon.h>
-#include "p8-platform/util/util.h"
+#include "platform/util/util.h"
 extern ADDON::CHelper_libXBMC_addon *XBMC;
-#endif
+#endif
\ No newline at end of file
