--- cairo-1.8.8/configure.ac.orig	2009-10-31 22:32:02.000000000 -0700
+++ cairo-1.8.8/configure.ac	2009-10-31 23:34:09.000000000 -0700
@@ -160,14 +160,10 @@
   use_png=no
   AC_ARG_VAR([png_REQUIRES], [module name for libpng to search for using pkg-config])
   if test "x$png_REQUIRES" = x; then
-    # libpng13 is GnuWin32's libpng-1.2.8 :-(
-    for l in libpng12 libpng13 libpng10 ; do
-      if $PKG_CONFIG --exists $l ; then
-        png_REQUIRES=$l
-        use_png=yes
-        break
-      fi
-    done
+    if $PKG_CONFIG --exists libpng ; then
+      png_REQUIRES=libpng
+      use_png=yes
+    fi
   else
     use_png=yes
   fi
--- cairo-1.8.8/configure.orig	2009-10-31 23:43:33.000000000 -0700
+++ cairo-1.8.8/configure	2009-10-31 23:43:37.000000000 -0700
@@ -27889,14 +27889,10 @@
   use_png=no
 
   if test "x$png_REQUIRES" = x; then
-    # libpng13 is GnuWin32's libpng-1.2.8 :-(
-    for l in libpng12 libpng13 libpng10 ; do
-      if $PKG_CONFIG --exists $l ; then
-        png_REQUIRES=$l
-        use_png=yes
-        break
-      fi
-    done
+    if $PKG_CONFIG --exists libpng ; then
+      png_REQUIRES=libpng
+      use_png=yes
+    fi
   else
     use_png=yes
   fi
