diff -Naur linux-3.18.6/drivers/usb/core/otg_whitelist.h linux-3.18.6.patch/drivers/usb/core/otg_whitelist.h
--- linux-3.18.6/drivers/usb/core/otg_whitelist.h	2015-02-06 15:53:48.000000000 +0100
+++ linux-3.18.6.patch/drivers/usb/core/otg_whitelist.h	2015-02-07 17:34:40.059601833 +0100
@@ -55,11 +55,6 @@
 	     le16_to_cpu(dev->descriptor.idProduct) == 0xbadd))
 		return 0;
 
-	/* OTG PET device is always targeted (see OTG 2.0 ECN 6.4.2) */
-	if ((le16_to_cpu(dev->descriptor.idVendor) == 0x1a0a &&
-	     le16_to_cpu(dev->descriptor.idProduct) == 0x0200))
-		return 1;
-
 	/* NOTE: can't use usb_match_id() since interface caches
 	 * aren't set up yet. this is cut/paste from that code.
 	 */
diff -Naur linux-3.18.6/drivers/usb/core/quirks.c linux-3.18.6.patch/drivers/usb/core/quirks.c
--- linux-3.18.6/drivers/usb/core/quirks.c	2015-02-06 15:53:48.000000000 +0100
+++ linux-3.18.6.patch/drivers/usb/core/quirks.c	2015-02-07 17:34:40.501602833 +0100
@@ -179,10 +179,6 @@
 	{ USB_DEVICE(0x0b05, 0x17e0), .driver_info =
 			USB_QUIRK_IGNORE_REMOTE_WAKEUP },
 
-	/* Protocol and OTG Electrical Test Device */
-	{ USB_DEVICE(0x1a0a, 0x0200), .driver_info =
-			USB_QUIRK_LINEAR_UFRAME_INTR_BINTERVAL },
-
 	{ }  /* terminating entry must be last */
 };
 
