--- linux-4.4-rc7-old/sound/soc/codecs/cs4265.c	2016-01-13 20:56:05.637652775 +0000
+++ linux-4.4-rc7/sound/soc/codecs/cs4265.c	2016-01-17 11:21:16.977652775 +0000
@@ -157,7 +157,7 @@
 	SOC_SINGLE("Validity Bit Control Switch", CS4265_SPDIF_CTL2,
 				3, 1, 0),
 	SOC_ENUM("SPDIF Mono/Stereo", spdif_mono_stereo_enum),
-	SOC_SINGLE("MMTLR Data Switch", 0,
+	SOC_SINGLE("MMTLR Data Switch", CS4265_SPDIF_CTL2,
 				1, 1, 0),
 	SOC_ENUM("Mono Channel Select", spdif_mono_select_enum),
 	SND_SOC_BYTES("C Data Buffer", CS4265_C_DATA_BUFF, 24),
@@ -199,8 +199,6 @@
 			SND_SOC_NOPM, 0, 0),
 	SND_SOC_DAPM_AIF_IN("DIN2", NULL,  0,
 			SND_SOC_NOPM, 0, 0),
-	SND_SOC_DAPM_AIF_IN("TXIN", NULL,  0,
-			CS4265_SPDIF_CTL2, 5, 1),
 
 	SND_SOC_DAPM_OUTPUT("LINEOUTL"),
 	SND_SOC_DAPM_OUTPUT("LINEOUTR"),
