--- a/sound/soc/bcm/Kconfig	2016-10-24 08:46:19.034757161 +0100
+++ b/sound/soc/bcm/Kconfig	2016-10-24 08:52:36.735342726 +0100
@@ -131,6 +131,13 @@ config SND_BCM2708_SOC_ALLO_PIANO_DAC
 	help
 	  Say Y or M if you want to add support for Allo Piano DAC.
 
+config SND_BCM2708_SOC_SLICE
+	tristate "Support for Slice on-board sound"
+	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
+	select SND_SOC_CS4265
+	help
+	  Say Y or M if you want to add support for Slice on-board sound.
+
 config SND_PISOUND
 	tristate "Support for Blokas Labs pisound"
 	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 
