From a45f914870e923741f3123799bd9e126b5da4aa0 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Wed, 20 Aug 2014 22:43:28 +0300
Subject: [PATCH] dont build tools / examples

---
 Makefile.am      |    4 ++--
 dbus/Makefile.am |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index dd76c7e..78cb56f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,8 +1,8 @@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC=dbus-glib-1.pc
 
-SUBDIRS=dbus tools test doc
-DIST_SUBDIRS=dbus tools test doc m4
+SUBDIRS=dbus test doc
+DIST_SUBDIRS=dbus test doc m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index 6d885f9..8bc102d 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 INCLUDES = \
 	-I$(top_srcdir)						\
-- 
1.7.2.5

