diff -Naur kodi-14-4606179/xbmc/input/linux/LinuxInputDevices.cpp kodi-14-4606179.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-14-4606179/xbmc/input/linux/LinuxInputDevices.cpp	2015-03-06 19:47:27.000000000 +0100
+++ kodi-14-4606179.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-03-07 14:42:33.260971676 +0100
@@ -225,6 +225,7 @@
   { KEY_VOLUMEDOWN    , XBMCK_VOLUME_DOWN },
   { KEY_VOLUMEUP      , XBMCK_VOLUME_UP   },
   { KEY_POWER         , XBMCK_POWER       },
+  { KEY_COFFEE        , XBMCK_VOLUME_MUTE },
   { KEY_KPEQUAL       , XBMCK_KP_EQUALS   },
   { KEY_PAUSE         , XBMCK_PAUSE       },
   { KEY_LEFTMETA      , XBMCK_LMETA       },
@@ -242,6 +243,7 @@
   { KEY_RECORD        , XBMCK_RECORD      },
   { KEY_REWIND        , XBMCK_REWIND      },
   { KEY_PHONE         , XBMCK_PHONE       },
+  { KEY_HOMEPAGE      , XBMCK_BROWSER_HOME},
   { KEY_REFRESH       , XBMCK_SHUFFLE     },
   { KEY_SCROLLUP      , XBMCK_PAGEUP      },
   { KEY_SCROLLDOWN    , XBMCK_PAGEDOWN    },
