diff --git a/CMakeLists.txt b/CMakeLists.txt
index 485bbaa..122b812 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,19 +197,8 @@ if(DIRKSIMPLE_LIBRETRO)
         thirdparty/theoraplay
     )
     set_target_properties(dirksimple_libretro PROPERTIES PREFIX "")
-    target_include_directories(dirksimple PRIVATE
-        thirdparty/lua/src
-        thirdparty/theoraplay
-    )
     target_link_libraries(dirksimple_libretro dirksimple-theoraplay dirksimple-theora dirksimple-vorbis dirksimple-ogg dirksimple-lua)
     
-    if(USE_PTHREAD)
-        target_link_libraries(dirksimple pthread)
-    endif()
-
-    if(HAVE_LIB_M)
-        target_link_libraries(dirksimple_libretro m)
-    endif()
     if(ANDROID)  # Android builds of libretro cores need specific names.
         set_target_properties(dirksimple_libretro PROPERTIES LIBRARY_OUTPUT_NAME dirksimple_libretro_android)
     endif()
@@ -218,24 +207,6 @@ if(DIRKSIMPLE_LIBRETRO)
     endif()
 endif()
 
-add_custom_command(
-    TARGET dirksimple POST_BUILD
-    COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/games"
-)
-
-if(APPLE)
-    add_custom_command(
-        TARGET dirksimple POST_BUILD
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/games"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_BINARY_DIR}/dirksimple" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS/dirksimple"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/mac-appbundle-info.plist" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Info.plist"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/icon.icns" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/DirkSimple.icns"
-    )
-endif()
 
 # end of CMakeLists.txt ...
 
