diff --git a/arch/arm/boot/dts/bcm2711-rpi.dtsi b/arch/arm/boot/dts/bcm2711-rpi.dtsi
index 8053ef54acc8..6bcfcc5b1ac5 100644
--- a/arch/arm/boot/dts/bcm2711-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi.dtsi
@@ -179,7 +179,7 @@ &pixelvalve4 {
 };
 
 &hdmi0 {
-	dmas = <&dma (10|(1<<27)|(1<<24)|(0<<16)|(15<<20))>;
+	dmas = <&dma (10|(1<<27)|(1<<24)|(15<<16)|(15<<20))>;
 	status = "disabled";
 };
 
@@ -188,7 +188,7 @@ &ddc0 {
 };
 
 &hdmi1 {
-	dmas = <&dma (17|(1<<27)|(1<<24)|(0<<16)|(15<<20))>;
+	dmas = <&dma (17|(1<<27)|(1<<24)|(15<<16)|(15<<20))>;
 	status = "disabled";
 };
