--- a/authfile.c	2016-07-28 00:54:27.000000000 +0200
+++ b/authfile.c	2018-07-03 19:13:08.496275544 +0200
@@ -169,28 +169,6 @@
 int
 sshkey_perm_ok(int fd, const char *filename)
 {
-	struct stat st;
-
-	if (fstat(fd, &st) < 0)
-		return SSH_ERR_SYSTEM_ERROR;
-	/*
-	 * if a key owned by the user is accessed, then we check the
-	 * permissions of the file. if the key owned by a different user,
-	 * then we don't care.
-	 */
-#ifdef HAVE_CYGWIN
-	if (check_ntsec(filename))
-#endif
-	if ((st.st_uid == getuid()) && (st.st_mode & 077) != 0) {
-		error("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
-		error("@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @");
-		error("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
-		error("Permissions 0%3.3o for '%s' are too open.",
-		    (u_int)st.st_mode & 0777, filename);
-		error("It is required that your private key files are NOT accessible by others.");
-		error("This private key will be ignored.");
-		return SSH_ERR_KEY_BAD_PERMISSIONS;
-	}
 	return 0;
 }
 
