diff --git a/yabause/src/libretro/Makefile.common b/yabause/src/libretro/Makefile.common
index 59d097c18..a52326bf6 100644
--- a/yabause/src/libretro/Makefile.common
+++ b/yabause/src/libretro/Makefile.common
@@ -53,10 +53,10 @@ else
 endif
 
 ifeq ($(FORCE_GLES), 1)
-	FLAGS += -D_OGLES3_
+	FLAGS += -D_OGLES3_ -D_OGLES31_
 	LDFLAGS += -lGLESv2
 	ifeq ($(FORCE_GLEW), 0)
-		FLAGS += -DHAVE_OPENGLES -DHAVE_OPENGLES3
+		FLAGS += -DHAVE_OPENGLES -DHAVE_OPENGLES3 -DHAVE_OPENGLES_3_1
 	endif
 else
 	FLAGS += -D_OGL3_
