commit 23a2343b962777ae0dc206934abf1202c271f1e6
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 19 21:56:23 2013 +0300

    vdr 2.1.2 buildfix

diff --git a/videobuffer.c b/videobuffer.c
index 0c26807..c393827 100644
--- a/videobuffer.c
+++ b/videobuffer.c
@@ -386,7 +386,7 @@ bool cVideoBufferFile::Init()
       m_Filename = cString::sprintf("%s/Timeshift-%d.vnsi", TimeshiftBufferDir, m_ClientID);
   }
   else
-    m_Filename = cString::sprintf("%s/Timeshift-%d.vnsi", VideoDirectory, m_ClientID);
+    m_Filename = cString::sprintf("%s/Timeshift-%d.vnsi", cVideoDirectory::Name(), m_ClientID);
 
   m_Fd = open(m_Filename, O_RDWR | O_CREAT | O_TRUNC, S_IRWXU);
   if (m_Fd == -1)
diff --git a/vnsiclient.c b/vnsiclient.c
index b724ce7..ecbcd7f 100644
--- a/vnsiclient.c
+++ b/vnsiclient.c
@@ -1403,7 +1403,7 @@ bool cVNSIClient::processTIMER_Update() /* OPCODE 85 */
 bool cVNSIClient::processRECORDINGS_GetDiskSpace() /* OPCODE 100 */
 {
   int FreeMB;
-  int Percent = VideoDiskSpace(&FreeMB);
+  int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB);
   int Total   = (FreeMB / (100 - Percent)) * 100;
 
   m_resp->add_U32(Total);
diff --git a/vnsiserver.c b/vnsiserver.c
index 5748da8..568d481 100644
--- a/vnsiserver.c
+++ b/vnsiserver.c
@@ -213,7 +213,7 @@ void cVNSIServer::Action(void)
   }
   else
   {
-    cmd = cString::sprintf("rm -f %s/*.vnsi", VideoDirectory);
+    cmd = cString::sprintf("rm -f %s/*.vnsi", cVideoDirectory::Name());
   }
   int ret = system(cmd);
 
