diff --git a/tools/perf/Makefile b/tools/perf/Makefile
index b0f164b..ecb1af6 100644
--- a/tools/perf/Makefile
+++ b/tools/perf/Makefile
@@ -687,12 +687,10 @@ ifndef NO_LIBAUDIT
 endif
 
 ifndef NO_SLANG
-	FLAGS_SLANG=$(ALL_CFLAGS) $(ALL_LDFLAGS) $(EXTLIBS) -I/usr/include/slang -lslang
+	FLAGS_SLANG=$(ALL_CFLAGS) $(ALL_LDFLAGS) $(EXTLIBS) -lslang
 	ifneq ($(call try-cc,$(SOURCE_SLANG),$(FLAGS_SLANG),libslang),y)
 		msg := $(warning slang not found, disables TUI support. Please install slang-devel or libslang-dev);
 	else
-		# Fedora has /usr/include/slang/slang.h, but ubuntu /usr/include/slang.h
-		BASIC_CFLAGS += -I/usr/include/slang
 		BASIC_CFLAGS += -DSLANG_SUPPORT
 		EXTLIBS += -lslang
 		LIB_OBJS += $(OUTPUT)ui/browser.o
