diff -Naur libpng-1.4.4-old/configure.ac libpng-1.4.4-new/configure.ac
--- libpng-1.4.4-old/configure.ac	2010-09-23 05:18:36.000000000 -0700
+++ libpng-1.4.4-new/configure.ac	2010-09-23 12:20:45.000000000 -0700
@@ -18,16 +18,16 @@
 
 dnl Version number stuff here:
 
-PNGLIB_VERSION=1.4.4
-PNGLIB_MAJOR=1
-PNGLIB_MINOR=4
-PNGLIB_RELEASE=4
-
 AC_INIT([libpng], [1.4.4], [png-mng-implement@lists.sourceforge.net])
 AM_INIT_AUTOMAKE
 dnl stop configure from automagically running automake
 AM_MAINTAINER_MODE
 
+PNGLIB_VERSION=1.4.4
+PNGLIB_MAJOR=1
+PNGLIB_MINOR=4
+PNGLIB_RELEASE=4
+
 dnl End of version number stuff
 
 AC_CONFIG_SRCDIR([pngget.c])
