diff -Naur xbmc-12.2-68a881d/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp xbmc-12.2-68a881d.patch/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
--- xbmc-12.2-68a881d/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2013-09-12 19:41:16.000000000 +0200
+++ xbmc-12.2-68a881d.patch/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2013-09-13 00:12:05.538128585 +0200
@@ -1180,7 +1180,7 @@
   char *errorStr;
   if (vasprintf(&errorStr, fmt, arg) >= 0)
   {
-    CLog::Log(LOGINFO, "CAESinkALSA - ALSA: %s:%d:(%s) %s%s%s",
+    CLog::Log(LOGDEBUG, "CAESinkALSA - ALSA: %s:%d:(%s) %s%s%s",
               file, line, function, errorStr, err ? ": " : "", err ? snd_strerror(err) : "");
     free(errorStr);
   }
