From ed19aa8fbd4012ea2ccb74a2160855be01aba5e4 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sun, 5 Feb 2017 18:15:12 -0800
Subject: [PATCH 1/2] fix cross compile

---
 Makefile | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index 0ac8061..b76c4c6 100644
--- a/Makefile
+++ b/Makefile
@@ -51,13 +51,7 @@ else ifeq ($(pgo),optimize)
   flags += -fprofile-use
 endif
 
-ifeq ($(compiler),)
-  ifneq ($(CXX),)
-    compiler := $(CXX)
-  else
-    compiler := g++
-  endif
-endif
+compiler = $(CXX)
 
 # platform
 ui := target-$(target)
-- 
2.9.3


From da71b9ec79482c31c37496019792da316626f1f2 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sun, 5 Feb 2017 20:40:42 -0800
Subject: [PATCH 2/2] no -O3

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b76c4c6..9622b5a 100644
--- a/Makefile
+++ b/Makefile
@@ -32,7 +32,7 @@ sfc_lagfix := 1
 ifeq ($(DEBUG), 1)
   flags := -I. -O0 -g
 else
-  flags := -I. -O3 -fomit-frame-pointer
+  flags := -I. -fomit-frame-pointer
 endif
 
 cflags := $(CPPFLAGS) $(CFLAGS) -std=gnu99 -xc
-- 
2.9.3

