diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-01-05 20:14:14.000000000 +0100
+++ b/Makefile	2016-01-11 20:38:29.377951179 +0100
@@ -32,8 +32,8 @@
   flags := -I. -O3 -fomit-frame-pointer
 endif
 
-cflags := -std=gnu99 -xc
-cppflags := -std=gnu++0x
+cflags := $(CFLAGS) -std=gnu99 -xc
+cppflags := $(CXXFLAGS) -std=gnu++0x
 
 objects := libco
 
@@ -48,13 +48,7 @@
   flags += -fprofile-use
 endif
 
-ifeq ($(compiler),)
-  ifneq ($(CXX),)
-    compiler := $(CXX)
-  else
-    compiler := g++
-  endif
-endif
+compiler = $(CXX)
 
 # platform
 ui := target-$(target)
