From da71b9ec79482c31c37496019792da316626f1f2 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Sun, 5 Feb 2017 20:40:42 -0800
Subject: [PATCH 2/2] no -O3

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 948f5fb..04e585d 100644
--- a/Makefile
+++ b/Makefile
@@ -459,7 +459,7 @@ OBJECTS := $(SOURCES_CXX:.cpp=.o) $(SOURCES_C:.c=.o)
 all: $(TARGET)
 
 ifeq ($(DEBUG),0)
-   FLAGS += -O3 $(EXTRA_GCC_FLAGS)
+   FLAGS += $(EXTRA_GCC_FLAGS)
 else
    FLAGS += -O0 -g
 endif
-- 
2.9.3

