diff -Naur a/Config.kmk b/Config.kmk
--- a/Config.kmk	2016-04-28 06:07:46.000000000 -0700
+++ b/Config.kmk	2016-06-12 12:58:09.503823599 -0700
@@ -428,10 +428,6 @@
 if1of ($(KBUILD_TARGET), linux solaris)
  VBOX_WITH_DBUS = 1
 endif
-# Enable building PAM modules.
-if1of ($(KBUILD_TARGET), linux solaris)
- VBOX_WITH_PAM = 1
-endif
 # Disable WINMM support for Windows (not implemented).
 VBOX_WITH_WINMM =
 # Enable internal networking.
