diff -Naur a/src/VBox/Additions/common/crOpenGL/Makefile.kmk b/src/VBox/Additions/common/crOpenGL/Makefile.kmk
--- a/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2016-04-28 06:08:03.000000000 -0700
+++ b/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2016-06-10 14:13:21.014010384 -0700
@@ -214,10 +214,10 @@
 
 if1of ($(KBUILD_TARGET), linux solaris freebsd)
  VBoxOGL_LIBS += \
- 	$(PATH_STAGE_LIB)/libXcomposite.so \
- 	$(PATH_STAGE_LIB)/libXdamage.so \
- 	$(PATH_STAGE_LIB)/libXfixes.so \
- 	$(PATH_STAGE_LIB)/libXext.so
+ 	Xcomposite \
+ 	Xdamage \
+ 	Xfixes \
+ 	Xext
  ifdef VBoxOGL_FAKEDRI
   ifeq ($(KBUILD_TARGET), freebsd)
     VBoxOGL_LIBS += \
