diff -Naur a/Config.kmk b/Config.kmk
--- a/Config.kmk	2016-04-28 06:07:46.000000000 -0700
+++ b/Config.kmk	2016-06-12 14:11:25.638509165 -0700
@@ -622,8 +622,6 @@
 endif
 # Enable shared folders
 VBOX_WITH_SHARED_FOLDERS = 1
-# Enable shared clipboard
-VBOX_WITH_SHARED_CLIPBOARD = 1
 # Enable the host/guest information service (aka guest properties).
 VBOX_WITH_GUEST_PROPS = 1
 # Enable this to prevent the guest from writing guest properties.
diff -Naur a/src/VBox/Additions/x11/VBoxClient/main.cpp b/src/VBox/Additions/x11/VBoxClient/main.cpp
--- a/src/VBox/Additions/x11/VBoxClient/main.cpp	2016-04-28 06:08:08.000000000 -0700
+++ b/src/VBox/Additions/x11/VBoxClient/main.cpp	2016-06-12 14:40:00.163521913 -0700
@@ -167,7 +167,6 @@
              "--seamless|check3d [-d|--nodaemon]\n", pcszFileName);
     RTPrintf("Starts the VirtualBox X Window System guest services.\n\n");
     RTPrintf("Options:\n");
-    RTPrintf("  --clipboard        starts the shared clipboard service\n");
 #ifdef VBOX_WITH_DRAG_AND_DROP
     RTPrintf("  --draganddrop      starts the drag and drop service\n");
 #endif
@@ -228,12 +227,6 @@
         {
             fRespawn = false;
         }
-        else if (!strcmp(argv[i], "--clipboard"))
-        {
-            if (g_pService)
-                break;
-            g_pService = VBClGetClipboardService();
-        }
         else if (!strcmp(argv[i], "--display"))
         {
             if (g_pService)
diff -Naur a/src/VBox/Additions/x11/VBoxClient/Makefile.kmk b/src/VBox/Additions/x11/VBoxClient/Makefile.kmk
--- a/src/VBox/Additions/x11/VBoxClient/Makefile.kmk	2016-04-28 06:08:08.000000000 -0700
+++ b/src/VBox/Additions/x11/VBoxClient/Makefile.kmk	2016-06-12 14:39:25.451177811 -0700
@@ -31,10 +31,7 @@
 VBoxClient_DEFS.linux += _GNU_SOURCE
 VBoxClient_SOURCES = \
 	main.cpp
-VBoxClient_SOURCES += \
-  	$(PATH_ROOT)/src/VBox/GuestHost/SharedClipboard/clipboard-helper.cpp \
-  	$(PATH_ROOT)/src/VBox/GuestHost/SharedClipboard/x11-clipboard.cpp \
-	clipboard.cpp
+VBoxClient_SOURCES +=
 VBoxClient_LIBPATH = \
 	$(VBOX_LIBPATH32_X11)
 VBoxClient_LIBS.freebsd = \
