diff -Naur a/libinstaller/linuxioctl.h b/libinstaller/linuxioctl.h
--- a/libinstaller/linuxioctl.h	2011-12-09 19:28:17.000000000 +0100
+++ b/libinstaller/linuxioctl.h	2012-04-18 21:02:43.405941674 +0200
@@ -19,7 +19,7 @@
 
 #undef SECTOR_SIZE		/* Defined in msdos_fs.h for no good reason */
 #undef SECTOR_BITS
-#include <linux/ext2_fs.h>	/* EXT2_IOC_* */
+#include <ext2fs/ext2_fs.h>	/* EXT2_IOC_* */
 
 #ifndef FAT_IOCTL_GET_ATTRIBUTES
 # define FAT_IOCTL_GET_ATTRIBUTES	_IOR('r', 0x10, __u32)
