diff -Naur gtk+-2.24.14/configure.in gtk+-2.24.14.patch/configure.in
--- gtk+-2.24.14/configure.in	2012-12-06 04:46:07.000000000 +0100
+++ gtk+-2.24.14.patch/configure.in	2013-01-12 18:32:33.863798129 +0100
@@ -51,7 +51,7 @@
 cflags_set=${CFLAGS+set}
 
 AM_INIT_AUTOMAKE([no-define -Wno-portability dist-bzip2])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Support silent build rules, requires at least automake-1.11. Enable
 # by either passing --enable-silent-rules to configure or passing V=0
@@ -300,7 +300,6 @@
 
 # Checks for programs.
 AC_ISC_POSIX
-AM_PROG_CC_STDC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
