diff -Nur a/src/include/os/rt_drv.h b/src/include/os/rt_drv.h
--- a/src/include/os/rt_drv.h	2016-02-16 11:23:49.000000000 +0100
+++ b/src/include/os/rt_drv.h	2016-04-18 21:59:18.653191304 +0200
@@ -56,7 +56,7 @@
 #ifdef CONFIG_STA_SUPPORT
 
 #ifdef RTMP_MAC_USB
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2870STA/RT2870STA.dat"
+#define STA_PROFILE_PATH			"/lib/firmware/mt7601u/mt7601u_sta.dat"
 #define STA_DRIVER_VERSION			"3.0.0.3"
 #ifdef MULTIPLE_CARD_SUPPORT
 #define CARD_INFO_PATH			"/etc/Wireless/RT2870STA/RT2870STACard.dat"
diff -Nur a/src/include/os/rt_linux.h b/src/include/os/rt_linux.h
--- a/src/include/os/rt_linux.h	2016-02-16 11:23:49.000000000 +0100
+++ b/src/include/os/rt_linux.h	2016-04-18 21:59:26.406524422 +0200
@@ -127,7 +127,7 @@
 #ifdef CONFIG_STA_SUPPORT
 
 #ifdef RTMP_MAC_USB
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2870STA/RT2870STA.dat"
+#define STA_PROFILE_PATH			"/lib/firmware/mt7601u/mt7601u_sta.dat"
 #define STA_DRIVER_VERSION			"3.0.0.3"
 #ifdef MULTIPLE_CARD_SUPPORT
 #define CARD_INFO_PATH			"/etc/Wireless/RT2870STA/RT2870STACard.dat"
