--- a/src/include/rtmp_def.h	2016-02-16 11:23:49.000000000 +0100
+++ b/src/include/rtmp_def.h	2016-04-18 21:57:18.566527968 +0200
@@ -1601,8 +1601,8 @@
 #define INF_MAIN_DEV_NAME		"wlan"
 #define INF_MBSSID_DEV_NAME		"wlan"
 #else
-#define INF_MAIN_DEV_NAME		"ra"
-#define INF_MBSSID_DEV_NAME		"ra"
+#define INF_MAIN_DEV_NAME		"wlan"
+#define INF_MBSSID_DEV_NAME		"wlan"
 #endif /* ANDROID_SUPPORT */
 #define INF_WDS_DEV_NAME		"wds"
 #define INF_APCLI_DEV_NAME		"apcli"
