diff -Naur zsnes-1.51-old/src/parsegen.cpp zsnes-1.51-new/src/parsegen.cpp
--- zsnes-1.51-old/src/parsegen.cpp	2007-10-30 21:30:26.000000000 -0700
+++ zsnes-1.51-new/src/parsegen.cpp	2008-01-20 08:04:38.000000000 -0800
@@ -25,6 +25,7 @@
 
 #include <iostream>
 #include <fstream>
+#include <cstring>
 #include <string>
 #include <vector>
 #include <algorithm>
@@ -1822,7 +1823,7 @@
   }
 }
 
-int main(size_t argc, const char *const *const argv)
+int main(int argc, const char *const *const argv)
 {
   const char *cheader_file = 0;
   bool compile = false;
diff -Naur zsnes-1.51-old/src/tools/depbuild.cpp zsnes-1.51-new/src/tools/depbuild.cpp
--- zsnes-1.51-old/src/tools/depbuild.cpp	2006-12-27 03:04:05.000000000 -0800
+++ zsnes-1.51-new/src/tools/depbuild.cpp	2008-01-20 08:08:25.000000000 -0800
@@ -23,6 +23,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdlib>
 #include <cstdio>
 using namespace std;
 
@@ -183,7 +184,7 @@
   }
 }
 
-int main(size_t argc, const char *const *const argv)
+int main(int argc, const char *const *const argv)
 {
   if (argc < 5)
   {
diff -Naur zsnes-1.51-old/src/tools/strutil.h zsnes-1.51-new/src/tools/strutil.h
--- zsnes-1.51-old/src/tools/strutil.h	2006-12-27 03:04:05.000000000 -0800
+++ zsnes-1.51-new/src/tools/strutil.h	2008-01-20 08:09:12.000000000 -0800
@@ -25,6 +25,7 @@
 #include <string>
 #include <vector>
 #include <cctype>
+#include <cstring>
 
 struct ci_char_traits : public std::char_traits<char>
 {
