diff --git a/hardfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h b/hardfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
index 84ca4eb..f28cf8c 100644
--- a/hardfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
+++ b/hardfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
@@ -100,7 +100,7 @@ typedef struct VCOS_LLTHREAD_T
 #ifndef VCOS_USE_VCOS_FUTEX
 typedef pthread_mutex_t       VCOS_MUTEX_T;
 #else
-#include "vcos_futex_mutex.h"
+#include "pthreads/vcos_futex_mutex.h"
 #endif /* VCOS_USE_VCOS_FUTEX */
 
 #define VCOS_ONCE_INIT        PTHREAD_ONCE_INIT
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos.h b/hardfp/opt/vc/include/interface/vcos/vcos.h
index e37d795..a341571 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos.h
@@ -104,7 +104,7 @@ VideoCore OS Abstraction Layer - public header file
 
 #include "interface/vcos/vcos_assert.h"
 #include "vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifndef VCOS_INIT_H
 #include "interface/vcos/vcos_init.h"
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h b/hardfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
index 317abf3..5cd2bb9 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_atomic_flags.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_blockpool.h b/hardfp/opt/vc/include/interface/vcos/vcos_blockpool.h
index 2c66f09..6521b17 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_blockpool.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_blockpool.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_cfg.h b/hardfp/opt/vc/include/interface/vcos/vcos_cfg.h
index 4423051..a4e8b0d 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_cfg.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_cfg.h
@@ -20,7 +20,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 typedef struct opaque_vcos_cfg_buf_t    *VCOS_CFG_BUF_T;
 typedef struct opaque_vcos_cfg_entry_t  *VCOS_CFG_ENTRY_T;
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_dlfcn.h b/hardfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
index 0a683c0..1725ccd 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
@@ -24,7 +24,7 @@ VCOS - abstraction over dynamic library opening
 #define VCOS_DLFCN_H
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_event.h b/hardfp/opt/vc/include/interface/vcos/vcos_event.h
index 38612f9..67778b1 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_event.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_event.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** 
   * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_event_flags.h b/hardfp/opt/vc/include/interface/vcos/vcos_event_flags.h
index 9eee410..81fdfb5 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_event_flags.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_event_flags.h
@@ -29,7 +29,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #define VCOS_EVENT_FLAGS_SUSPEND    VCOS_SUSPEND
 #define VCOS_EVENT_FLAGS_NO_SUSPEND VCOS_NO_SUSPEND
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_init.h b/hardfp/opt/vc/include/interface/vcos/vcos_init.h
index 9fc5eca..38a56d5 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_init.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_init.h
@@ -22,7 +22,7 @@ VideoCore OS Abstraction Layer - initialization routines
 
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_isr.h b/hardfp/opt/vc/include/interface/vcos/vcos_isr.h
index 644996f..5ae9990 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_isr.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_isr.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
   * \file vcos_isr.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h b/hardfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
index 49ae1f9..3db722a 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_legacy_isr.h
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_logging.h b/hardfp/opt/vc/include/interface/vcos/vcos_logging.h
index 9702822..a3173bc 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_logging.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_logging.h
@@ -30,7 +30,7 @@ extern "C" {
 #include <stdarg.h>
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h b/hardfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
index 9473cec..c0b0823 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_mem.h b/hardfp/opt/vc/include/interface/vcos/vcos_mem.h
index af16208..27c4e27 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_mem.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_mem.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_mempool.h b/hardfp/opt/vc/include/interface/vcos/vcos_mempool.h
index 436cca9..fe46412 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_mempool.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_mempool.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_msgqueue.h b/hardfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
index f0ef70b..2075a60 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_mutex.h b/hardfp/opt/vc/include/interface/vcos/vcos_mutex.h
index 14387a8..9a85ee8 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_mutex.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_mutex.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h b/hardfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
index c009a09..2a76a92 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_once.h b/hardfp/opt/vc/include/interface/vcos/vcos_once.h
index 0533c10..61c28a9 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_once.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_once.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_once.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_queue.h b/hardfp/opt/vc/include/interface/vcos/vcos_queue.h
index b16bbf8..cfb0ae5 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_queue.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_queue.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_queue.h
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h b/hardfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
index 60992f1..0be4753 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_quickslow_mutex.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h b/hardfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
index c8b99c0..6d41fa9 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_semaphore.h b/hardfp/opt/vc/include/interface/vcos/vcos_semaphore.h
index 2760dd7..807dd3e 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_semaphore.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_semaphore.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_semaphore.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_string.h b/hardfp/opt/vc/include/interface/vcos/vcos_string.h
index d7d8078..0cd1a3a 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_string.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_string.h
@@ -35,7 +35,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __KERNEL__
 #include <linux/string.h>
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_thread.h b/hardfp/opt/vc/include/interface/vcos/vcos_thread.h
index ee34648..ea70bad 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_thread.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_thread.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_thread.h
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_timer.h b/hardfp/opt/vc/include/interface/vcos/vcos_timer.h
index 1612334..9120529 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_timer.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_timer.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_timer.h
   *
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_tls.h b/hardfp/opt/vc/include/interface/vcos/vcos_tls.h
index 4760f1c..ce410ea 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_tls.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_tls.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 
 /** Create a new thread local storage data key visible to all threads in
diff --git a/hardfp/opt/vc/include/interface/vcos/vcos_types.h b/hardfp/opt/vc/include/interface/vcos/vcos_types.h
index 0e6cfc5..f39ea2b 100644
--- a/hardfp/opt/vc/include/interface/vcos/vcos_types.h
+++ b/hardfp/opt/vc/include/interface/vcos/vcos_types.h
@@ -25,7 +25,7 @@ VideoCore OS Abstraction Layer - basic types
 
 #define VCOS_VERSION   1
 
-#include "vcos_platform_types.h"
+#include "pthreads/vcos_platform_types.h"
 
 #if !defined(VCOSPRE_) || !defined(VCOSPOST_)
 #error VCOSPRE_ and VCOSPOST_ not defined!
diff --git a/softfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h b/softfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
index 84ca4eb..f28cf8c 100644
--- a/softfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
+++ b/softfp/opt/vc/include/interface/vcos/pthreads/vcos_platform.h
@@ -100,7 +100,7 @@ typedef struct VCOS_LLTHREAD_T
 #ifndef VCOS_USE_VCOS_FUTEX
 typedef pthread_mutex_t       VCOS_MUTEX_T;
 #else
-#include "vcos_futex_mutex.h"
+#include "pthreads/vcos_futex_mutex.h"
 #endif /* VCOS_USE_VCOS_FUTEX */
 
 #define VCOS_ONCE_INIT        PTHREAD_ONCE_INIT
diff --git a/softfp/opt/vc/include/interface/vcos/vcos.h b/softfp/opt/vc/include/interface/vcos/vcos.h
index e37d795..a341571 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos.h
@@ -104,7 +104,7 @@ VideoCore OS Abstraction Layer - public header file
 
 #include "interface/vcos/vcos_assert.h"
 #include "vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifndef VCOS_INIT_H
 #include "interface/vcos/vcos_init.h"
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h b/softfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
index 317abf3..5cd2bb9 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_atomic_flags.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_atomic_flags.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_blockpool.h b/softfp/opt/vc/include/interface/vcos/vcos_blockpool.h
index 2c66f09..6521b17 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_blockpool.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_blockpool.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_cfg.h b/softfp/opt/vc/include/interface/vcos/vcos_cfg.h
index 4423051..a4e8b0d 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_cfg.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_cfg.h
@@ -20,7 +20,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 typedef struct opaque_vcos_cfg_buf_t    *VCOS_CFG_BUF_T;
 typedef struct opaque_vcos_cfg_entry_t  *VCOS_CFG_ENTRY_T;
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_dlfcn.h b/softfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
index 0a683c0..1725ccd 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_dlfcn.h
@@ -24,7 +24,7 @@ VCOS - abstraction over dynamic library opening
 #define VCOS_DLFCN_H
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_event.h b/softfp/opt/vc/include/interface/vcos/vcos_event.h
index 38612f9..67778b1 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_event.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_event.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** 
   * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_event_flags.h b/softfp/opt/vc/include/interface/vcos/vcos_event_flags.h
index 9eee410..81fdfb5 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_event_flags.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_event_flags.h
@@ -29,7 +29,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #define VCOS_EVENT_FLAGS_SUSPEND    VCOS_SUSPEND
 #define VCOS_EVENT_FLAGS_NO_SUSPEND VCOS_NO_SUSPEND
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_init.h b/softfp/opt/vc/include/interface/vcos/vcos_init.h
index 9fc5eca..38a56d5 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_init.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_init.h
@@ -22,7 +22,7 @@ VideoCore OS Abstraction Layer - initialization routines
 
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_isr.h b/softfp/opt/vc/include/interface/vcos/vcos_isr.h
index 644996f..5ae9990 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_isr.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_isr.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
   * \file vcos_isr.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h b/softfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
index 49ae1f9..3db722a 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_legacy_isr.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_legacy_isr.h
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_logging.h b/softfp/opt/vc/include/interface/vcos/vcos_logging.h
index 9702822..a3173bc 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_logging.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_logging.h
@@ -30,7 +30,7 @@ extern "C" {
 #include <stdarg.h>
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h b/softfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
index 9473cec..c0b0823 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_lowlevel_thread.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_mem.h b/softfp/opt/vc/include/interface/vcos/vcos_mem.h
index af16208..27c4e27 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_mem.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_mem.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_mempool.h b/softfp/opt/vc/include/interface/vcos/vcos_mempool.h
index 436cca9..fe46412 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_mempool.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_mempool.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_msgqueue.h b/softfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
index f0ef70b..2075a60 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_msgqueue.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_mutex.h b/softfp/opt/vc/include/interface/vcos/vcos_mutex.h
index 14387a8..9a85ee8 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_mutex.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_mutex.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h b/softfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
index c009a09..2a76a92 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_named_semaphore.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_once.h b/softfp/opt/vc/include/interface/vcos/vcos_once.h
index 0533c10..61c28a9 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_once.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_once.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_once.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_queue.h b/softfp/opt/vc/include/interface/vcos/vcos_queue.h
index b16bbf8..cfb0ae5 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_queue.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_queue.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_queue.h
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h b/softfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
index 60992f1..0be4753 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_quickslow_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_quickslow_mutex.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h b/softfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
index c8b99c0..6d41fa9 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_reentrant_mutex.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_semaphore.h b/softfp/opt/vc/include/interface/vcos/vcos_semaphore.h
index 2760dd7..807dd3e 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_semaphore.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_semaphore.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_semaphore.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_string.h b/softfp/opt/vc/include/interface/vcos/vcos_string.h
index d7d8078..0cd1a3a 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_string.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_string.h
@@ -35,7 +35,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 #ifdef __KERNEL__
 #include <linux/string.h>
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_thread.h b/softfp/opt/vc/include/interface/vcos/vcos_thread.h
index ee34648..ea70bad 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_thread.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_thread.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /**
  * \file vcos_thread.h
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_timer.h b/softfp/opt/vc/include/interface/vcos/vcos_timer.h
index 1612334..9120529 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_timer.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_timer.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 /** \file vcos_timer.h
   *
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_tls.h b/softfp/opt/vc/include/interface/vcos/vcos_tls.h
index 4760f1c..ce410ea 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_tls.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_tls.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "interface/vcos/vcos_types.h"
-#include "vcos_platform.h"
+#include "pthreads/vcos_platform.h"
 
 
 /** Create a new thread local storage data key visible to all threads in
diff --git a/softfp/opt/vc/include/interface/vcos/vcos_types.h b/softfp/opt/vc/include/interface/vcos/vcos_types.h
index 0e6cfc5..f39ea2b 100644
--- a/softfp/opt/vc/include/interface/vcos/vcos_types.h
+++ b/softfp/opt/vc/include/interface/vcos/vcos_types.h
@@ -25,7 +25,7 @@ VideoCore OS Abstraction Layer - basic types
 
 #define VCOS_VERSION   1
 
-#include "vcos_platform_types.h"
+#include "pthreads/vcos_platform_types.h"
 
 #if !defined(VCOSPRE_) || !defined(VCOSPOST_)
 #error VCOSPRE_ and VCOSPOST_ not defined!
