--- a/test/evtest.c	2014-03-21 19:26:36.000000000 +0000
+++ b/test/evtest.c	2022-07-20 12:35:25.235888860 +0000
@@ -241,7 +241,7 @@
 
 		for (i = 0; i < rd / (int) sizeof(struct input_event); i++)
 			printf("Event: time %ld.%06ld, type %d (%s), code %d (%s), value %d\n",
-				ev[i].time.tv_sec, ev[i].time.tv_usec, ev[i].type,
+				ev[i].input_event_sec, ev[i].input_event_usec, ev[i].type,
 				events[ev[i].type] ? events[ev[i].type] : "?",
 				ev[i].code,
 				names[ev[i].type] ? (names[ev[i].type][ev[i].code] ? names[ev[i].type][ev[i].code] : "?") : "?",
