diff --git a/buildtools/wafsamba/wafsamba.py b/buildtools/wafsamba/wafsamba.py
index 4a47dbf..1d8db7a 100644
--- a/buildtools/wafsamba/wafsamba.py
+++ b/buildtools/wafsamba/wafsamba.py
@@ -330,11 +330,6 @@ def SAMBA_LIBRARY(bld, libname, source,
     if pc_files is not None and not private_library:
         bld.PKG_CONFIG_FILES(pc_files, vnum=vnum)
 
-    if (manpages is not None and 'XSLTPROC_MANPAGES' in bld.env and
-        bld.env['XSLTPROC_MANPAGES']):
-        bld.MANPAGES(manpages, install)
-
-
 Build.BuildContext.SAMBA_LIBRARY = SAMBA_LIBRARY
 
 
@@ -437,9 +432,6 @@ def SAMBA_BINARY(bld, binname, source,
         samba_ldflags  = pie_ldflags
         )
 
-    if manpages is not None and 'XSLTPROC_MANPAGES' in bld.env and bld.env['XSLTPROC_MANPAGES']:
-        bld.MANPAGES(manpages, install)
-
 Build.BuildContext.SAMBA_BINARY = SAMBA_BINARY
 
 
diff --git a/docs-xml/wscript_build b/docs-xml/wscript_build
index 093a20d..097fd3b 100644
--- a/docs-xml/wscript_build
+++ b/docs-xml/wscript_build
@@ -127,21 +127,3 @@ bld.SAMBA_GENERATOR(parameter_all,
                     target=parameter_all,
                     rule=smbdotconf_generate_parameter_list,
                     dep_vars=bld.dynconfig_varnames())
-
-def SMBDOTCONF_MANPAGE(bld, target):
-    ''' assemble and build smb.conf.5 manual page'''
-    bld.SAMBAMANPAGES(target, parameter_all)
-
-if ('XSLTPROC_MANPAGES' in bld.env and bld.env['XSLTPROC_MANPAGES']):
-
-    SMBDOTCONF_MANPAGE(bld, 'manpages/smb.conf.5')
-    bld.SAMBAMANPAGES(manpages)
-
-    if bld.CONFIG_SET('WITH_PAM_MODULES') and bld.CONFIG_SET('HAVE_PAM_START'):
-        bld.SAMBAMANPAGES(pam_winbind_manpages)
-
-    if bld.CONFIG_SET('HAVE_KRB5_LOCATE_PLUGIN_H'):
-        bld.SAMBAMANPAGES(krb5_locator_manpages)
-
-    if bld.SAMBA3_IS_ENABLED_MODULE('vfs_zfsacl'):
-        bld.SAMBAMANPAGES('manpages/vfs_zfsacl.8')
