diff -Naur samba-3.6.23/source3/librpc/rpc/rpc_common.c samba-3.6.23.patch/source3/librpc/rpc/rpc_common.c
--- samba-3.6.23/source3/librpc/rpc/rpc_common.c	2014-03-12 16:10:57.120989895 +0100
+++ samba-3.6.23.patch/source3/librpc/rpc/rpc_common.c	2014-03-12 16:11:48.361046356 +0100
@@ -100,9 +100,11 @@
 		return false;
 	}
 #endif
+#ifdef SAMR_SUPPORT
 	if (!smb_register_ndr_interface(&ndr_table_samr)) {
 		return false;
 	}
+#endif
 #ifdef NETLOGON_SUPPORT
 	if (!smb_register_ndr_interface(&ndr_table_netlogon)) {
 		return false;
diff -Naur samba-3.6.23/source3/rpcclient/rpcclient.c samba-3.6.23.patch/source3/rpcclient/rpcclient.c
--- samba-3.6.23/source3/rpcclient/rpcclient.c	2014-03-12 16:10:57.121989948 +0100
+++ samba-3.6.23.patch/source3/rpcclient/rpcclient.c	2014-03-12 16:11:48.362046792 +0100
@@ -623,7 +623,9 @@
 	rpcclient_commands,
 	lsarpc_commands,
 	ds_commands,
+#ifdef SAMR_SUPPORT
 	samr_commands,
+#endif
 #ifdef PRINTER_SUPPORT
 	spoolss_commands,
 #endif
diff -Naur samba-3.6.23/source3/rpc_server/rpc_ep_setup.c samba-3.6.23.patch/source3/rpc_server/rpc_ep_setup.c
--- samba-3.6.23/source3/rpc_server/rpc_ep_setup.c	2014-03-12 16:10:57.120989895 +0100
+++ samba-3.6.23.patch/source3/rpc_server/rpc_ep_setup.c	2014-03-12 16:11:48.361046356 +0100
@@ -557,6 +557,7 @@
 	return true;
 }
 
+#ifdef SAMR_SUPPORT
 static bool samr_init_cb(void *ptr)
 {
 	struct dcesrv_ep_context *ep_ctx =
@@ -605,6 +606,7 @@
 
 	return true;
 }
+#endif
 
 #ifdef NETLOGON_SUPPORT
 static bool netlogon_init_cb(void *ptr)
@@ -1111,12 +1113,14 @@
 		return false;
 	}
 
+#ifdef SAMR_SUPPORT
 	samr_cb.init         = samr_init_cb;
 	samr_cb.shutdown     = NULL;
 	samr_cb.private_data = ep_ctx;
 	if (!NT_STATUS_IS_OK(rpc_samr_init(&samr_cb))) {
 		return false;
 	}
+#endif
 
 #ifdef NETLOGON_SUPPORT
 	netlogon_cb.init         = netlogon_init_cb;
diff -Naur samba-3.6.23/source3/rpc_server/rpc_handles.c samba-3.6.23.patch/source3/rpc_server/rpc_handles.c
--- samba-3.6.23/source3/rpc_server/rpc_handles.c	2014-03-11 11:17:34.000000000 +0100
+++ samba-3.6.23.patch/source3/rpc_server/rpc_handles.c	2014-03-12 16:11:48.361046356 +0100
@@ -59,8 +59,11 @@
 
 static bool is_samr_lsa_pipe(const struct ndr_syntax_id *syntax)
 {
-	return (ndr_syntax_id_equal(syntax, &ndr_table_samr.syntax_id)
-		|| ndr_syntax_id_equal(syntax, &ndr_table_lsarpc.syntax_id));
+	return
+#ifdef SAMR_SUPPORT
+		ndr_syntax_id_equal(syntax, &ndr_table_samr.syntax_id) ||
+#endif
+		ndr_syntax_id_equal(syntax, &ndr_table_lsarpc.syntax_id);
 }
 
 size_t num_pipe_handles(struct pipes_struct *p)
diff -Naur samba-3.6.23/source3/smbd/lanman.c samba-3.6.23.patch/source3/smbd/lanman.c
--- samba-3.6.23/source3/smbd/lanman.c	2014-03-12 16:10:30.535984713 +0100
+++ samba-3.6.23.patch/source3/smbd/lanman.c	2014-03-12 16:11:48.362046792 +0100
@@ -2353,6 +2353,10 @@
 	NTSTATUS status, result;
 	struct dcerpc_binding_handle *b;
 
+#ifndef SAMR_SUPPORT
+	return False;
+#endif
+
 	if (!str1 || !str2 || !p) {
 		return False;
 	}
@@ -2541,6 +2545,10 @@
 	NTSTATUS status, result;
 	struct dcerpc_binding_handle *b;
 
+#ifndef SAMR_SUPPORT
+	return False;
+#endif
+
 	if (!str1 || !str2 || !UserName || !p) {
 		return False;
 	}
@@ -2741,6 +2749,10 @@
 
 	struct dcerpc_binding_handle *b;
 
+#ifndef SAMR_SUPPORT
+	return False;
+#endif
+
 	if (!str1 || !str2 || !p) {
 		return False;
 	}
@@ -2979,6 +2991,10 @@
 	int bufsize;
 	struct dcerpc_binding_handle *b;
 
+#ifndef SAMR_SUPPORT
+	return False;
+#endif
+
 	*rparam_len = 4;
 	*rparam = smb_realloc_limit(*rparam,*rparam_len);
 	if (!*rparam) {
@@ -4020,6 +4036,10 @@
 	union samr_UserInfo *info;
 	struct dcerpc_binding_handle *b = NULL;
 
+#ifndef SAMR_SUPPORT
+	return False;
+#endif
+
 	if (!str1 || !str2 || !UserName || !p) {
 		return False;
 	}
diff -Naur samba-3.6.23/source3/smbd/server_exit.c samba-3.6.23.patch/source3/smbd/server_exit.c
--- samba-3.6.23/source3/smbd/server_exit.c	2014-03-12 16:10:57.120989895 +0100
+++ samba-3.6.23.patch/source3/smbd/server_exit.c	2014-03-12 16:11:48.361046356 +0100
@@ -159,7 +159,9 @@
 #ifdef NETLOGON_SUPPORT
 		rpc_netlogon_shutdown();
 #endif
+#ifdef SAMR_SUPPORT
 		rpc_samr_shutdown();
+#endif
 		rpc_lsarpc_shutdown();
 	}
 
