diff -Naur peripheral.joystick-21.1.22-Omega/CMakeLists.txt peripheral.joystick-21.1.22-Omega-2/CMakeLists.txt
--- peripheral.joystick-21.1.22-Omega/CMakeLists.txt	2025-11-23 22:28:45.567290418 +0100
+++ peripheral.joystick-21.1.22-Omega-2/CMakeLists.txt	2025-11-23 22:30:24.027537195 +0100
@@ -222,7 +222,7 @@
 
 set(LINUX_SELECT_LINE "\
         <setting id=\"driver_linux\" type=\"integer\" label=\"30001\">
-          <default>0</default>
+          <default>1</default>
           <constraints>
             <options>
               <option label=\"30005\">0</option>
