diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed0f308..74fc6bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ set(rsxs_dir lib/rsxs-1.0)
 include_directories(${OpenGL_INCLUDE_DIR}
                     ${PNG_INCLUDE_DIR}
                     ${KODI_INCLUDE_DIR}/.. # Hack way with "/..", need bigger Kodi cmake rework to match right include ways
-                    ${X_INCLUDE_DIRS}
+                    ${X_INCLUDE_DIR}
                     ${CMAKE_BINARY_DIR}/rsxs-prefix/src/rsxs-build
                     ${PROJECT_SOURCE_DIR}/${rsxs_dir}/src
                     ${PROJECT_SOURCE_DIR}/${rsxs_dir}/lib
@@ -67,7 +67,7 @@ externalproject_add(rsxs SOURCE_DIR ${PROJECT_SOURCE_DIR}/${rsxs_dir}
                                       --disable-skyrocket
                                       --disable-solarwinds
                                       --disable-sound
-                                      --x-includes=${X_INCLUDE_DIRS}
+                                      --x-includes=${X_INCLUDE_DIR}
                                       --x-libraries=${X_LIBRARIES}
                                       ${PNG_PATH}
                                       ${EXTRA_FLAGS}
