diff -Naur screensavers.rsxs-3241d29.orig/CMakeLists.txt screensavers.rsxs-3241d29/CMakeLists.txt
--- screensavers.rsxs-3241d29.orig/CMakeLists.txt	2015-10-20 00:21:24.000000000 -0700
+++ screensavers.rsxs-3241d29/CMakeLists.txt	2015-10-22 23:34:37.522226204 -0700
@@ -52,6 +52,7 @@
 endif()
 externalproject_add(rsxs SOURCE_DIR ${PROJECT_SOURCE_DIR}/${rsxs_dir}
                     CONFIGURE_COMMAND ${configure_start}
+                                      ac_cv_type__Bool=yes
                                       --prefix=<INSTALL_DIR>
                                       --without-xscreensaver
                                       --disable-cyclone
