diff -Naur service.openelec.settings-0.4.3/defaults.py service.openelec.settings-0.4.3.patch/defaults.py
--- service.openelec.settings-0.4.3/defaults.py	2014-09-10 11:38:24.000000000 +0200
+++ service.openelec.settings-0.4.3.patch/defaults.py	2014-10-20 03:53:04.451032921 +0200
@@ -29,7 +29,7 @@
 ################################################################################
 # Base 
 ################################################################################
-XBMC_USER_HOME = os.environ.get("XBMC_USER_HOME", "/storage/.xbmc")
+XBMC_USER_HOME = os.environ.get("XBMC_USER_HOME", "/storage/.kodi")
 CONFIG_CACHE = os.environ.get("CONFIG_CACHE", "/storage/.cache")
 USER_CONFIG = os.environ.get("USER_CONFIG", "/storage/.config")
 
diff -Naur service.openelec.settings-0.4.3/oe.py service.openelec.settings-0.4.3.patch/oe.py
--- service.openelec.settings-0.4.3/oe.py	2014-09-10 11:38:24.000000000 +0200
+++ service.openelec.settings-0.4.3.patch/oe.py	2014-10-20 03:52:36.499991207 +0200
@@ -1025,7 +1025,7 @@
 VERSION        = load_file('/etc/version')    
 BUILD          = load_file('/etc/build')
 DOWNLOAD_DIR   = "/storage/downloads"
-XBMC_USER_HOME = os.environ.get('XBMC_USER_HOME', '/storage/.xbmc')
+XBMC_USER_HOME = os.environ.get('XBMC_USER_HOME', '/storage/.kodi')
 CONFIG_CACHE   = os.environ.get('CONFIG_CACHE', '/storage/.cache')
 USER_CONFIG    = os.environ.get('USER_CONFIG', '/storage/.config')
 TEMP           = '%s/temp/' % XBMC_USER_HOME
