--- a/x86-32/src/wl/sys/wl_linux.c	2014-06-26 14:42:08.000000000 +0400
+++ b/x86-32/src/wl/sys/wl_linux.c	2014-10-23 06:28:36.987479544 +0400
@@ -215,7 +217,7 @@
 #define to_str(s) #s
 #define quote_str(s) to_str(s)
 
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 
 static char intf_name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);

--- a/x86-64/src/wl/sys/wl_linux.c      2014-06-26 14:42:08.000000000 +0400
+++ b/x86-64/src/wl/sys/wl_linux.c      2014-10-23 06:28:36.987479544 +0400
@@ -215,7 +217,7 @@
 #define to_str(s) #s
 #define quote_str(s) to_str(s)
 
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 
 static char intf_name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
