diff -Nur a/x86-64/src/wl/sys/wl_cfg80211_hybrid.c b/x86-64/src/wl/sys/wl_cfg80211_hybrid.c
--- a/x86-64/src/wl/sys/wl_cfg80211_hybrid.c	2022-08-15 14:06:51.943852581 +0000
+++ b/x86-64/src/wl/sys/wl_cfg80211_hybrid.c	2022-08-15 14:24:49.917982457 +0000
@@ -2393,8 +2393,13 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
 	{
 		struct cfg80211_roam_info roam_info = {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 0, 0)
+			.links[0].channel = &wl->conf->channel,
+			.links[0].bssid = (u8 *)&wl->bssid,
+#else
 			.channel = &wl->conf->channel,
 			.bssid = (u8 *)&wl->bssid,
+#endif
 			.req_ie = conn_info->req_ie,
 			.req_ie_len = conn_info->req_ie_len,
 			.resp_ie = conn_info->resp_ie,
