diff -Naur bcm_sta-5.100.82.38/x86-32/src/wl/sys/wl_linux.c bcm_sta-5.100.82.38.patch/x86-32/src/wl/sys/wl_linux.c
--- bcm_sta-5.100.82.38/x86-32/src/wl/sys/wl_linux.c	2010-12-15 02:58:07.000000000 +0100
+++ bcm_sta-5.100.82.38.patch/x86-32/src/wl/sys/wl_linux.c	2010-12-27 23:42:21.319475865 +0100
@@ -482,7 +482,7 @@
 #ifdef WL_ALL_PASSIVE
 		spin_lock_init(&wl->txq_lock);
 #endif 
-		init_MUTEX(&wl->sem);
+		sema_init(&wl->sem, 1);
 	}
 
 	if (!(wl->wlc = wlc_attach((void *) wl, vendor, device, unit, wl->piomode,
diff -Naur bcm_sta-5.100.82.38/x86-64/src/wl/sys/wl_linux.c bcm_sta-5.100.82.38.patch/x86-64/src/wl/sys/wl_linux.c
--- bcm_sta-5.100.82.38/x86-64/src/wl/sys/wl_linux.c	2010-12-15 03:01:09.000000000 +0100
+++ bcm_sta-5.100.82.38.patch/x86-64/src/wl/sys/wl_linux.c	2010-12-27 23:42:38.462649098 +0100
@@ -482,7 +482,7 @@
 #ifdef WL_ALL_PASSIVE
 		spin_lock_init(&wl->txq_lock);
 #endif 
-		init_MUTEX(&wl->sem);
+		sema_init(&wl->sem, 1);
 	}
 
 	if (!(wl->wlc = wlc_attach((void *) wl, vendor, device, unit, wl->piomode,
