diff -Naur bcm_sta-6.30.223.248/x86-32/src/wl/sys/wl_linux.c bcm_sta-6.30.223.248.patch/x86-32/src/wl/sys/wl_linux.c
--- bcm_sta-6.30.223.248/x86-32/src/wl/sys/wl_linux.c	2014-06-26 12:41:38.000000000 +0200
+++ bcm_sta-6.30.223.248.patch/x86-32/src/wl/sys/wl_linux.c	2015-09-04 19:58:10.496600089 +0200
@@ -2157,8 +2157,8 @@
 	wlif = WL_DEV_IF(dev);
 	wl = WL_INFO(dev);
 
+	skb->prev = NULL;
 	if (WL_ALL_PASSIVE_ENAB(wl) || (WL_RTR() && WL_CONFIG_SMP())) {
-		skb->prev = NULL;
 
 		TXQ_LOCK(wl);
 
diff -Naur bcm_sta-6.30.223.248/x86-64/src/wl/sys/wl_linux.c bcm_sta-6.30.223.248.patch/x86-64/src/wl/sys/wl_linux.c
--- bcm_sta-6.30.223.248/x86-64/src/wl/sys/wl_linux.c	2014-06-26 12:42:08.000000000 +0200
+++ bcm_sta-6.30.223.248.patch/x86-64/src/wl/sys/wl_linux.c	2015-09-04 19:56:31.957374783 +0200
@@ -2157,8 +2157,8 @@
 	wlif = WL_DEV_IF(dev);
 	wl = WL_INFO(dev);
 
+	skb->prev = NULL;
 	if (WL_ALL_PASSIVE_ENAB(wl) || (WL_RTR() && WL_CONFIG_SMP())) {
-		skb->prev = NULL;
 
 		TXQ_LOCK(wl);
 
