--- a/x86-64/src/wl/sys/wl_linux.c	2022-02-19 15:22:05.006428601 +0000
+++ b/x86-64/src/wl/sys/wl_linux.c	2022-02-19 21:55:03.290519415 +0000
@@ -3287,7 +3287,11 @@
 static ssize_t
 wl_proc_read(struct file *filp, char __user *buffer, size_t length, loff_t *offp)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
 	wl_info_t * wl = PDE_DATA(file_inode(filp));
+#else
+	wl_info_t * wl = file_inode(filp)->i_private;
+#endif
 #endif
 	int bcmerror, len;
 	int to_user = 0;
@@ -3344,7 +3348,11 @@
 static ssize_t
 wl_proc_write(struct file *filp, const char __user *buff, size_t length, loff_t *offp)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
 	wl_info_t * wl = PDE_DATA(file_inode(filp));
+#else
+	wl_info_t * wl = file_inode(filp)->i_private;
+#endif
 #endif
 	int from_user = 0;
 	int bcmerror;
