diff -Naur bcm_sta-5.100.82.112/x86-32/src/wl/sys/wl_linux.c bcm_sta-5.100.82.112.patch/x86-32/src/wl/sys/wl_linux.c
--- bcm_sta-5.100.82.112/x86-32/src/wl/sys/wl_linux.c	2011-10-22 18:55:54.000000000 +0200
+++ bcm_sta-5.100.82.112.patch/x86-32/src/wl/sys/wl_linux.c	2011-12-21 16:54:17.752384441 +0100
@@ -385,7 +385,7 @@
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
-	.ndo_set_multicast_list = wl_set_multicast_list,
+	.ndo_set_rx_mode = wl_set_multicast_list,
 	.ndo_do_ioctl = wl_ioctl
 };
 
diff -Naur bcm_sta-5.100.82.112/x86-64/src/wl/sys/wl_linux.c bcm_sta-5.100.82.112.patch/x86-64/src/wl/sys/wl_linux.c
--- bcm_sta-5.100.82.112/x86-64/src/wl/sys/wl_linux.c	2011-10-22 18:56:55.000000000 +0200
+++ bcm_sta-5.100.82.112.patch/x86-64/src/wl/sys/wl_linux.c	2011-12-21 16:54:32.486670945 +0100
@@ -385,7 +385,7 @@
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
-	.ndo_set_multicast_list = wl_set_multicast_list,
+	.ndo_set_rx_mode = wl_set_multicast_list,
 	.ndo_do_ioctl = wl_ioctl
 };
 
