diff -Naur bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_iw.h bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_iw.h
--- bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_iw.h	2010-11-22 03:01:28.267226695 +0100
+++ bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_iw.h	2010-11-22 03:00:28.308506905 +0100
@@ -62,6 +62,7 @@
 	struct timer_list *timer;
 	struct net_device *dev;
 	long sysioc_pid;
+	struct semaphore sysioc_sem;
 	struct completion sysioc_exited;
 };
 
diff -Naur bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_iw.h bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_iw.h
--- bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_iw.h	2010-11-22 03:01:49.840485699 +0100
+++ bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_iw.h	2010-11-22 03:00:28.322507073 +0100
@@ -62,6 +62,7 @@
 	struct timer_list *timer;
 	struct net_device *dev;
 	long sysioc_pid;
+	struct semaphore sysioc_sem;
 	struct completion sysioc_exited;
 };
 
