diff -Naur bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_iw.h bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_iw.h
--- bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_iw.h	2010-10-22 04:18:57.000000000 +0200
+++ bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_iw.h	2010-11-29 08:30:58.911543674 +0100
@@ -62,7 +62,6 @@
 	struct timer_list *timer;
 	struct net_device *dev;
 	long sysioc_pid;
-	struct semaphore sysioc_sem;
 	struct completion sysioc_exited;
 };
 
diff -Naur bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_iw.h bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_iw.h
--- bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_iw.h	2010-10-22 04:23:55.000000000 +0200
+++ bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_iw.h	2010-11-29 08:31:15.542760506 +0100
@@ -62,7 +62,6 @@
 	struct timer_list *timer;
 	struct net_device *dev;
 	long sysioc_pid;
-	struct semaphore sysioc_sem;
 	struct completion sysioc_exited;
 };
 
