diff -Naur bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_linux.c bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_linux.c
--- bcm_sta-5.60.246.6/x86-32/src/wl/sys/wl_linux.c	2010-10-22 04:18:57.000000000 +0200
+++ bcm_sta-5.60.246.6.patch/x86-32/src/wl/sys/wl_linux.c	2010-12-02 05:17:28.037326623 +0100
@@ -187,7 +187,7 @@
 static int nompc = 0;
 module_param(nompc, int, 0);
 
-static char name[IFNAMSIZ] = "eth%d";
+static char name[IFNAMSIZ] = "wlan%d";
 module_param_string(name, name, IFNAMSIZ, 0);
 
 #ifndef	SRCBASE
diff -Naur bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_linux.c bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_linux.c
--- bcm_sta-5.60.246.6/x86-64/src/wl/sys/wl_linux.c	2010-10-22 04:23:55.000000000 +0200
+++ bcm_sta-5.60.246.6.patch/x86-64/src/wl/sys/wl_linux.c	2010-12-02 05:17:42.554495917 +0100
@@ -187,7 +187,7 @@
 static int nompc = 0;
 module_param(nompc, int, 0);
 
-static char name[IFNAMSIZ] = "eth%d";
+static char name[IFNAMSIZ] = "wlan%d";
 module_param_string(name, name, IFNAMSIZ, 0);
 
 #ifndef	SRCBASE
