diff -Naur libdnet-1.12/acconfig.h libdnet-1.12-patch/acconfig.h
--- libdnet-1.12/acconfig.h	2018-02-01 15:55:19.346009246 -0500
+++ libdnet-1.12-patch/acconfig.h	2018-02-01 15:59:14.826008240 -0500
@@ -20,11 +20,11 @@
 #endif
 
 #ifndef HAVE_STRLCAT
-int	strlcat(char *, const char *, int);
+size_t	strlcat(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRLCPY
-int	strlcpy(char *, const char *, int);
+size_t	strlcpy(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRSEP
diff -Naur libdnet-1.12/include/config.h.in libdnet-1.12-patch/include/config.h.in
--- libdnet-1.12/include/config.h.in	2018-02-01 15:51:57.346010109 -0500
+++ libdnet-1.12-patch/include/config.h.in	2018-02-01 15:59:47.316008101 -0500
@@ -282,11 +282,11 @@
 #endif
 
 #ifndef HAVE_STRLCAT
-int	strlcat(char *, const char *, int);
+size_t	strlcat(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRLCPY
-int	strlcpy(char *, const char *, int);
+size_t	strlcpy(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRSEP
diff -Naur libdnet-1.12/trunk/acconfig.h libdnet-1.12-patch/trunk/acconfig.h
--- libdnet-1.12/trunk/acconfig.h	2018-02-01 15:51:57.356010109 -0500
+++ libdnet-1.12-patch/trunk/acconfig.h	2018-02-01 16:00:08.526008010 -0500
@@ -20,11 +20,11 @@
 #endif
 
 #ifndef HAVE_STRLCAT
-int	strlcat(char *, const char *, int);
+size_t	strlcat(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRLCPY
-int	strlcpy(char *, const char *, int);
+size_t	strlcpy(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRSEP
diff -Naur libdnet-1.12/trunk/include/config.h.in libdnet-1.12-patch/trunk/include/config.h.in
--- libdnet-1.12/trunk/include/config.h.in	2018-02-01 15:51:57.356010109 -0500
+++ libdnet-1.12-patch/trunk/include/config.h.in	2018-02-01 16:00:34.786007898 -0500
@@ -282,11 +282,11 @@
 #endif
 
 #ifndef HAVE_STRLCAT
-int	strlcat(char *, const char *, int);
+size_t	strlcat(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRLCPY
-int	strlcpy(char *, const char *, int);
+size_t	strlcpy(char *, const char *, size_t);
 #endif
 
 #ifndef HAVE_STRSEP
