--- lzma-4.57/CPP/7zip/Compress/LZMA_Alone/LzmaAlone.cpp.orig	2008-03-12 19:18:07.000000000 +0100
+++ lzma-4.57/CPP/7zip/Compress/LZMA_Alone/LzmaAlone.cpp	2008-03-12 19:20:06.000000000 +0100
@@ -26,7 +26,6 @@
 #include "../LZMA/LZMAEncoder.h"
 
 #include "LzmaBenchCon.h"
-#include "LzmaRam.h"
 
 #ifdef COMPRESS_MF_MT
 #include "../../../Windows/System.h"
@@ -36,7 +35,6 @@
 
 extern "C"
 {
-#include "LzmaRamDecode.h"
 }
 
 using namespace NCommandLineParser;
@@ -74,7 +72,6 @@
   kEOS,
   kStdIn,
   kStdOut,
-  kFilter86
 };
 }
 
@@ -94,7 +91,6 @@
   { L"EOS", NSwitchType::kSimple, false },
   { L"SI",  NSwitchType::kSimple, false },
   { L"SO",  NSwitchType::kSimple, false },
-  { L"F86",  NSwitchType::kPostChar, false, 0, 0, L"+" }
 };
 
 static const int kNumSwitches = sizeof(kSwitchForms) / sizeof(kSwitchForms[0]);
@@ -299,6 +295,7 @@
     }
   }
 
+#if 0
   if (parser[NKey::kFilter86].ThereIs)
   {
     // -f86 switch is for x86 filtered mode: BCJ + LZMA.
@@ -366,6 +363,7 @@
     MyFree(inBuffer);
     return 0;
   }
+#endif
 
 
   UInt64 fileSize;
--- lzma-4.57/CPP/7zip/Compress/LZMA_Alone/makefile.gcc.orig	2008-03-12 19:17:56.000000000 +0100
+++ lzma-4.57/CPP/7zip/Compress/LZMA_Alone/makefile.gcc	2008-03-12 19:18:25.000000000 +0100
@@ -22,7 +22,6 @@
   LzmaAlone.o \
   LzmaBench.o \
   LzmaBenchCon.o \
-  LzmaRam.o \
   LZMADecoder.o \
   LZMAEncoder.o \
   LZOutWindow.o \
@@ -41,10 +40,8 @@
   MyVector.o \
   7zCrc.o \
   Alloc.o \
-  BranchX86.o \
   MatchFinder.o \
   LzmaDecode.o \
-  LzmaRamDecode.o \
 
 
 all: $(PROG)
