From b702bd4fdb4aaebb6aadb3ce773bcc9eea8d2dc8 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 24 Apr 2014 22:05:37 +0300
Subject: [PATCH] allow newer autoconf

---
 aclocal.m4 |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index bc48586..0e3c7f4 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -2,11 +2,6 @@ dnl We require that everyone use exactly the same Autoconf version so that
 dnl the internal functions defined and used by the main configure script
 dnl match those expected by the fragments.
 m4_define([GLIBC_AUTOCONF_VERSION], [2.68])
-m4_if(m4_defn([AC_AUTOCONF_VERSION]), GLIBC_AUTOCONF_VERSION, [],
-      [m4_fatal(m4_flatten(
-Exactly version GLIBC_AUTOCONF_VERSION of Autoconf is required but you have
-m4_defn([AC_AUTOCONF_VERSION])
-), [63])])dnl
 dnl
 dnl We define the macro GLIBC_PROVIDES to do an AC_PROVIDE for each macro
 dnl which appears in configure.ac before the sysdep configure scripts are run.
-- 
1.7.2.5

