diff -Naur hts-tvheadend-c88a646/src/linuxtv_muxes.h hts-tvheadend-c88a646.patch/src/linuxtv_muxes.h
--- hts-tvheadend-c88a646/src/linuxtv_muxes.h	2011-09-26 20:18:12.000000000 +0200
+++ hts-tvheadend-c88a646.patch/src/linuxtv_muxes.h	2011-10-18 13:40:37.768853992 +0200
@@ -12001,6 +12001,39 @@
 	{ .freq = 573000000, .symrate = 5217000, .fec = 0, .constellation = 5},
 };
 
+static const struct mux muxes_DVBC_ch_GA_Weissenstein[] = {
+	{ .freq = 450000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 506000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 514000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 522000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 530000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 538000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 554000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 562000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 570000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 578000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 586000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 594000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 602000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 610000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 618000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 626000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 634000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 642000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 650000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 658000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 666000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 674000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 682000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 690000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 698000000, .symrate = 6900000, .fec = 0, .constellation = 3},
+	{ .freq = 698000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 706000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 714000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 722000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+	{ .freq = 730000000, .symrate = 6900000, .fec = 0, .constellation = 5},
+};
+
 static const struct mux muxes_DVBC_ch_Rega_Sense[] = {
 	{ .freq = 434000000, .symrate = 6900000, .fec = 0, .constellation = 3},
 	{ .freq = 714000000, .symrate = 6900000, .fec = 0, .constellation = 3},
@@ -12881,6 +12914,11 @@
 
 static const struct network networks_DVBC_ch[] = {
 	{
+		.name = "GA-Weissenstein",
+		.muxes = muxes_DVBC_ch_GA_Weissenstein,
+		.nmuxes = sizeof(muxes_DVBC_ch_GA_Weissenstein) / sizeof(struct mux),
+	},
+	{
 		.name = "Rega-Sense",
 		.muxes = muxes_DVBC_ch_Rega_Sense,
 		.nmuxes = sizeof(muxes_DVBC_ch_Rega_Sense) / sizeof(struct mux),
