From 60e5a1014c52e926fb7dbb12392f60a473e44d2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jernej=20Fija=C4=8Dko?= <jernej@jernej-ThinkCentre-M91p.(none)>
Date: Fri, 30 Sep 2011 13:23:46 +0200
Subject: [PATCH] Fix memory leak in queue size protection

---
 src/htsp.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/htsp.c b/src/htsp.c
index 8bb803b..f786a88 100644
--- a/src/htsp.c
+++ b/src/htsp.c
@@ -1508,6 +1508,9 @@ htsp_stream_deliver(htsp_subscription_t *hs, th_pkt_t *pkt)
 
     hs->hs_dropstats[pkt->pkt_frametype]++;
 
+    // destroy the already created htsmsg to avoid memory leaks
+    htsmsg_destroy(m);
+
     /* Queue size protection */
     pkt_ref_dec(pkt);
     return;
-- 
1.7.3.1

