diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-01-14 22:44:37.000000000 +0100
+++ b/Makefile	2016-01-18 20:53:35.439383675 +0100
@@ -130,27 +130,25 @@
       LIBS += -lGL
    endif
    LDFLAGS +=  $(fpic) $(SHARED)
-   REALCC   = gcc
-   NATIVECC = g++
-   NATIVECFLAGS = -std=gnu99
+   REALCC   ?= gcc
    BASELIBS +=  -lpthread
-   CC = g++
-   AR = @ar
-   LD = g++
+   CC ?= g++
+   AR ?= @ar
+   LD ?= g++
    LIBS += -lstdc++ -lpthread -ldl
-   ifeq ($(firstword $(filter x86_64,$(UNAME))),x86_64)
+   ifeq ($(firstword $(filter x86_64,$(ARCH))),x86_64)
       PTR64 = 1
    endif
-   ifeq ($(firstword $(filter amd64,$(UNAME))),amd64)
+   ifeq ($(firstword $(filter amd64,$(ARCH))),amd64)
       PTR64 = 1
    endif
-   ifeq ($(firstword $(filter ppc64,$(UNAME))),ppc64)
+   ifeq ($(firstword $(filter ppc64,$(ARCH))),ppc64)
       PTR64 = 1
    endif
-   ifneq (,$(findstring ppc,$(UNAME)))
+   ifneq (,$(findstring ppc,$(ARCH)))
       BIGENDIAN=1
    endif
-   ifneq (,$(findstring armv,$(UNAME) $(platform)))
+   ifneq (,$(findstring armv,$(ARCH) $(platform)))
       CCOMFLAGS += -mstructure-size-boundary=32
       PLATCFLAGS += -DSDLMAME_NO64BITIO -DSDLMAME_ARM -DRETRO_SETJMP_HACK -DARM
       LDFLAGS += -Wl,--fix-cortex-a8 -Wl,--no-as-needed
@@ -482,7 +478,7 @@
 # utilities
 MD = -mkdir$(EXE_EXT)
 RM = @rm -f
-OBJDUMP = @objdump
+OBJDUMP ?= @objdump
 PYTHON ?= @python2
 
 #-------------------------------------------------
