From dc70f6cdfa9980b707a958cfca9a3820d51af8f6 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sat, 23 Mar 2019 10:11:55 +0100
Subject: [PATCH] keytable: backport imon and rc-mm protocols

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 utils/keytable/keytable.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/utils/keytable/keytable.c b/utils/keytable/keytable.c
index 34a1522e..67c6f92c 100644
--- a/utils/keytable/keytable.c
+++ b/utils/keytable/keytable.c
@@ -112,6 +112,8 @@ enum sysfs_protocols {
 	SYSFS_SHARP		= (1 << 11),
 	SYSFS_XMP		= (1 << 12),
 	SYSFS_CEC		= (1 << 13),
+	SYSFS_IMON		= (1 << 14),
+	SYSFS_RCMM		= (1 << 15),
 	SYSFS_INVALID		= 0,
 };
 
@@ -145,6 +147,8 @@ const struct protocol_map_entry protocol_map[] = {
 	{ "sharp",	NULL,		SYSFS_SHARP	},
 	{ "xmp",	"/xmp_decoder",	SYSFS_XMP	},
 	{ "cec",	NULL,		SYSFS_CEC	},
+	{ "imon",	NULL,		SYSFS_IMON	},
+	{ "rc-mm",	NULL,		SYSFS_RCMM	},
 	{ NULL,		NULL,		SYSFS_INVALID	},
 };
 
-- 
2.20.1

