diff --git a/utils/keytable/rc_keymaps/cuboxi b/utils/keytable/rc_keymaps/cubox_i
new file mode 100644
index 0000000..1063253
--- /dev/null
+++ b/utils/keytable/rc_keymaps/cubox_i
@@ -0,0 +1,10 @@
+# table cubox_i, type: RC6
+0x1f020b0 KEY_MEDIA
+0x1f02078 KEY_UP
+0x1f02044 KEY_ENTER
+0x1f020f8 KEY_DOWN
+0x1f02004 KEY_LEFT
+0x1f02084 KEY_RIGHT
+0x1f020c4 KEY_BACKSPACE
+0x1f02068 KEY_PLAYPAUSE
+0x1f02064 KEY_EPG
