diff -Naur v4l-utils-1.6.0/utils/keytable/keytable.c v4l-utils-1.6.0.patch/utils/keytable/keytable.c
--- v4l-utils-1.6.0/utils/keytable/keytable.c	2014-08-22 16:59:47.000000000 +0200
+++ v4l-utils-1.6.0.patch/utils/keytable/keytable.c	2014-11-15 21:57:15.200451780 +0100
@@ -226,7 +226,7 @@
 							goto err_einval;
 						if (!strcasecmp(p,"rc5") || !strcasecmp(p,"rc-5"))
 							ch_proto |= RC_5;
-						else if (!strcasecmp(p,"rc6") || !strcasecmp(p,"rc-6"))
+						else if (!strcasecmp(p,"rc6") || !strcasecmp(p,"rc-6") || !strcasecmp(p,"rc6-mce") || !strcasecmp(p,"rc6_mce"))
 							ch_proto |= RC_6;
 						else if (!strcasecmp(p,"nec"))
 							ch_proto |= NEC;
