diff --git a/utils/keytable/keytable.c b/utils/keytable/keytable.c
index 8bcd5c4..e8e67da 100644
--- a/utils/keytable/keytable.c
+++ b/utils/keytable/keytable.c
@@ -471,6 +471,8 @@ static error_t parse_opt(int k, char *arg, struct argp_state *state)
 				ch_proto |= LIRC;
 			else if (!strcasecmp(p,"rc-5-sz"))
 				ch_proto |= RC_5_SZ;
+			else if (!strcasecmp(p,"other"))
+				ch_proto |= OTHER;
 			else
 				goto err_inval;
 			p = strtok(NULL, ",;");
