diff -Naur v4l-utils-0.8.3/utils/keytable/rc_keymaps/imon_mce v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/imon_mce
--- v4l-utils-0.8.3/utils/keytable/rc_keymaps/imon_mce	2011-02-09 23:04:27.000000000 +0100
+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/imon_mce	2011-04-05 17:55:00.641736994 +0200
@@ -16,7 +16,7 @@
 0x800ff420 KEY_LEFT
 0x800ff421 KEY_RIGHT
 0x800ff40b KEY_ENTER
-0x02000028 KEY_ENTER
+# 0x02000028 KEY_ENTER # same as KEY_OK
 0x02000028 KEY_OK
 0x800ff422 KEY_OK
 0x0200002a KEY_EXIT
@@ -67,7 +67,7 @@
 0x800ff45d KEY_YELLOW
 0x800ff45e KEY_BLUE
 0x800ff466 KEY_RED
-0x800ff425 KEY_GREEN
+# 0x800ff425 KEY_GREEN # same as KEY_TUNER
 0x800ff468 KEY_YELLOW
 0x800ff41d KEY_BLUE
 0x800ff40f KEY_INFO
@@ -75,4 +75,4 @@
 0x800ff45a KEY_SUBTITLE
 0x800ff44d KEY_TITLE
 0x800ff40c KEY_POWER
-0x800ff40d KEY_LEFTMETA
+0x800ff40d KEY_PROG1
diff -Naur v4l-utils-0.8.3/utils/keytable/rc_keymaps/imon_pad v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/imon_pad
--- v4l-utils-0.8.3/utils/keytable/rc_keymaps/imon_pad	2011-02-09 23:04:27.000000000 +0100
+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/imon_pad	2011-04-05 17:38:46.644003338 +0200
@@ -72,3 +72,15 @@
 0x2aa395b7 KEY_SCREEN
 0x299115b7 KEY_KEYBOARD
 0x299135b7 KEY_KEYBOARD
+
+0x29b195b7 KEY_EJECTCD
+0x02000065 KEY_COMPOSE
+0x02800000 KEY_CONTEXT_MENU
+0x01020000 BTN_RIGHT
+0x01010000 BTN_LEFT
+0x29b715b7 KEY_DASHBOARD
+0x2a9395b7 KEY_CYCLEWINDOWS
+0x299395b7 KEY_EJECTCLOSECD
+0x2ab195b7 KEY_PROG1
+0x2b8395b7 KEY_TIME
+0x289115b7 KEY_POWER
diff -Naur v4l-utils-0.8.3/utils/keytable/rc_keymaps/kworld_plus_tv_analog v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/kworld_plus_tv_analog
--- v4l-utils-0.8.3/utils/keytable/rc_keymaps/kworld_plus_tv_analog	2011-02-09 23:04:27.000000000 +0100
+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/kworld_plus_tv_analog	2011-04-05 17:38:46.645003319 +0200
@@ -1,5 +1,5 @@
 # table kworld_plus_tv_analog, type: UNKNOWN
-0x0c KEY_LEFTMETA
+0x0c KEY_PROG1
 0x16 KEY_CLOSECD
 0x1d KEY_POWER2
 0x00 KEY_1
diff -Naur v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce
--- v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce	2011-02-09 23:04:27.000000000 +0100
+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce	2011-04-05 17:38:46.801000225 +0200
@@ -12,7 +12,7 @@
 0x800f040a KEY_DELETE
 0x800f040b KEY_ENTER
 0x800f040c KEY_POWER
-0x800f040d KEY_LEFTMETA
+0x800f040d KEY_PROG1
 0x800f040e KEY_MUTE
 0x800f040f KEY_INFO
 0x800f0410 KEY_VOLUMEUP
