diff -Nru a/utils/keytable/rc_keymaps/rc6_mce b/utils/keytable/rc_keymaps/rc6_mce
--- a/utils/keytable/rc_keymaps/rc6_mce	2015-02-17 12:25:15.452561904 +0200
+++ b/utils/keytable/rc_keymaps/rc6_mce	2015-05-27 13:22:28.089766130 +0300
@@ -262,3 +262,15 @@
 0x80d826 KEY_INFO
 0x80d86f KEY_EPG
 0x80d86e KEY_ZOOM
+
+#CuBox-i and HummingBoard remote control
+0x1f020b0 KEY_MEDIA
+0x1f02078 KEY_UP
+0x1f02044 KEY_ENTER
+0x1f020f8 KEY_DOWN
+0x1f02004 KEY_LEFT
+0x1f02084 KEY_RIGHT
+0x1f020c4 KEY_BACKSPACE
+0x1f02068 KEY_PLAYPAUSE
+0x1f02064 KEY_EPG
+
