diff --git a/utils/keytable/rc_maps.cfg b/utils/keytable/rc_maps.cfg
index e69fd6a..c0fe122 100644
--- a/utils/keytable/rc_maps.cfg
+++ b/utils/keytable/rc_maps.cfg
@@ -109,7 +109,9 @@
 *	rc-proteus-2309          proteus_2309
 *	rc-purpletv              purpletv
 *	rc-pv951                 pv951
-*	rc-rc6-mce               rc6_mce
+# use combined multi-table on MCE receivers
+#*	rc-rc6-mce               rc6_mce
+*	rc-rc6-mce               libreelec_multi
 *	rc-real-audio-220-32-keys real_audio_220_32_keys
 *	rc-reddo                 reddo
 *	rc-snapstream-firefly    snapstream_firefly
@@ -136,6 +138,13 @@
 *	rc-videomate-tv-pvr      videomate_tv_pvr
 *	rc-winfast-usbii-deluxe  winfast_usbii_deluxe
 *	rc-winfast               winfast
+# fake tables for LibreELEC keymaps
+*	rc-cubox-i               cubox-i
+*	rc-hp-mce                hp_mce
+*	rc-samsung               samsung
+*	rc-xbox-360              xbox_360
+*	rc-xbox-one              xbox_one
+*	rc-zotac-ad10            zotac_ad10
 # *	*			 a800                 # found in a800.c
 # *	*			 af9005               # found in af9005-remote.c
 # *	*			 cinergyt2            # found in cinergyT2-core.c
