diff -Naur v4l-utils-1.6.0/utils/keytable/rc_keymaps/rc6_mce v4l-utils-1.6.0.patch/utils/keytable/rc_keymaps/rc6_mce
--- v4l-utils-1.6.0/utils/keytable/rc_keymaps/rc6_mce	2014-09-07 19:32:21.000000000 +0200
+++ v4l-utils-1.6.0.patch/utils/keytable/rc_keymaps/rc6_mce	2014-11-14 02:51:17.947769083 +0100
@@ -1,4 +1,4 @@
-# table rc6_mce, type: RC6_MCE
+# table rc6_mce, type: RC6
 0x800f0400 KEY_NUMERIC_0
 0x800f0401 KEY_NUMERIC_1
 0x800f0402 KEY_NUMERIC_2
