From 137dd2edea4d6c70b10caebebed4778716070ed2 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sun, 9 Feb 2020 00:16:29 +0100
Subject: [PATCH] don't include doxygen autotools include

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index d20dfe0f..05dc8c1a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,7 @@
 
 EXTRA_DIST = android-config.h sync-with-kernel.sh bootstrap.sh \
 	doxygen_libdvbv5.cfg include COPYING.libv4l \
 	COPYING.libdvbv5 README.libv4l README.lib-multi-threading \
 	TODO.libdvbv5 doc/libdvbv5-index.doc
 
-include $(top_srcdir)/aminclude.am
+# include $(top_srcdir)/aminclude.am
-- 
2.20.1

