diff -Naur v4l-utils-0.8.3/utils/keytable/keytable.c v4l-utils-0.8.3.patch/utils/keytable/keytable.c
--- v4l-utils-0.8.3/utils/keytable/keytable.c	2011-02-09 23:04:27.000000000 +0100
+++ v4l-utils-0.8.3.patch/utils/keytable/keytable.c	2011-03-22 22:40:16.957618675 +0100
@@ -267,7 +267,7 @@
 				perror("value");
 		}
 
-		nextkey->codes[0] = (unsigned) strtol(scancode, NULL, 0);
+		nextkey->codes[0] = (unsigned) strtoul(scancode, NULL, 0);
 		nextkey->codes[1] = (unsigned) value;
 		nextkey->next = calloc(1, sizeof(*nextkey));
 		if (!nextkey->next) {
