From dbf64abf9eed823b35a2931d4882905b6106461e Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sat, 23 Mar 2019 10:19:48 +0100
Subject: [PATCH] keytable: add xbox-dvd protocol

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 utils/keytable/keytable.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/utils/keytable/keytable.c b/utils/keytable/keytable.c
index 67c6f92c..4ee280cc 100644
--- a/utils/keytable/keytable.c
+++ b/utils/keytable/keytable.c
@@ -114,6 +114,7 @@ enum sysfs_protocols {
 	SYSFS_CEC		= (1 << 13),
 	SYSFS_IMON		= (1 << 14),
 	SYSFS_RCMM		= (1 << 15),
+	SYSFS_XBOX_DVD		= (1 << 16),
 	SYSFS_INVALID		= 0,
 };
 
@@ -149,6 +150,7 @@ const struct protocol_map_entry protocol_map[] = {
 	{ "cec",	NULL,		SYSFS_CEC	},
 	{ "imon",	NULL,		SYSFS_IMON	},
 	{ "rc-mm",	NULL,		SYSFS_RCMM	},
+	{ "xbox-dvd",	NULL,		SYSFS_XBOX_DVD	},
 	{ NULL,		NULL,		SYSFS_INVALID	},
 };
 
-- 
2.20.1

