diff --git a/utils/keytable/rc_keymaps/samsung b/utils/keytable/rc_keymaps/samsung
new file mode 100644
index 0000000..2b0c81d
--- /dev/null
+++ b/utils/keytable/rc_keymaps/samsung
@@ -0,0 +1,31 @@
+# table samsung, type: NEC
+0x43532f KEY_NUMERIC_0
+0x435341 KEY_NUMERIC_1
+0x435342 KEY_NUMERIC_2
+0x435343 KEY_NUMERIC_3
+0x435344 KEY_NUMERIC_4
+0x435345 KEY_NUMERIC_5
+0x435346 KEY_NUMERIC_6
+0x435347 KEY_NUMERIC_7
+0x43533f KEY_NUMERIC_8
+0x435337 KEY_NUMERIC_9
+0x435305 KEY_ENTER
+0x435300 KEY_TITLE
+0x435326 KEY_INFO
+0x435323 KEY_VOLUMEUP
+0x43532b KEY_VOLUMEDOWN
+0x435375 KEY_FASTFORWARD
+0x435374 KEY_REWIND
+0x43537d KEY_PLAYPAUSE
+0x43530b KEY_STOP
+0x435313 KEY_NEXT
+0x43531b KEY_PREVIOUS
+0x43530d KEY_UP
+0x435315 KEY_DOWN
+0x435325 KEY_LEFT
+0x43531d KEY_RIGHT
+0x435336 KEY_MENU
+0x43531c KEY_EXIT
+0x43530f KEY_ZOOM
+0x43530e KEY_SUBTITLE
+0x43537e KEY_BACK
diff --git a/utils/keytable/rc_maps.cfg b/utils/keytable/rc_maps.cfg
index 567b8e0..04bed89 100644
--- a/utils/keytable/rc_maps.cfg
+++ b/utils/keytable/rc_maps.cfg
@@ -49,6 +49,7 @@
 *	rc-winfast-usbii-deluxe  winfast_usbii_deluxe
 *	rc-anysee                anysee
 *	rc-rc6-mce               rc6_mce
+*	rc-samsung               samsung
 *	rc-adstech-dvb-t-pci     adstech_dvb_t_pci
 *	rc-kworld-pc150u         kworld_pc150u
 *	rc-iodata-bctv7e         iodata_bctv7e
