diff -Naur linux-3.7.10/tools/perf/Makefile linux-3.7.10.patch/tools/perf/Makefile
--- linux-3.7.10/tools/perf/Makefile	2013-02-27 18:22:04.000000000 +0100
+++ linux-3.7.10.patch/tools/perf/Makefile	2013-03-06 16:36:15.633521884 +0100
@@ -591,8 +591,6 @@
 	ifneq ($(call try-cc,$(SOURCE_NEWT),$(FLAGS_NEWT)),y)
 		msg := $(warning newt not found, disables TUI support. Please install newt-devel or libnewt-dev);
 	else
-		# Fedora has /usr/include/slang/slang.h, but ubuntu /usr/include/slang.h
-		BASIC_CFLAGS += -I/usr/include/slang
 		BASIC_CFLAGS += -DNEWT_SUPPORT
 		EXTLIBS += -lnewt -lslang
 		LIB_OBJS += $(OUTPUT)ui/setup.o
