From e275d4438c1baa34ba9d7f5880aad53d776daf2e Mon Sep 17 00:00:00 2001
From: fritsch <peter.fruehberger@gmail.com>
Date: Fri, 20 Feb 2015 08:40:05 +0100
Subject: [PATCH 14/16] IMX: Prepare for fractional modes (picked from @rabeeh)

---
 xbmc/windowing/egl/EGLNativeTypeIMX.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index cf2f706..25dbc9b 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -315,14 +315,14 @@ bool CEGLNativeTypeIMX::ModeToResolution(std::string mode, RESOLUTION_INFO *res)
   StringUtils::Trim(fromMode);
 
   CRegExp split(true);
-  split.RegComp("([0-9]+)x([0-9]+)([pi])-([0-9]+)");
+  split.RegComp("([0-9]+)x([0-9]+)([pi])-([-+]?[0-9]*\.?[0-9]*)");
   if (split.RegFind(fromMode) < 0)
     return false;
 
   int w = atoi(split.GetMatch(1).c_str());
   int h = atoi(split.GetMatch(2).c_str());
   std::string p = split.GetMatch(3);
-  int r = atoi(split.GetMatch(4).c_str());
+  float r = atof(split.GetMatch(4).c_str());
 
   res->iWidth = w;
   res->iHeight= h;
@@ -335,7 +335,7 @@ bool CEGLNativeTypeIMX::ModeToResolution(std::string mode, RESOLUTION_INFO *res)
   res->bFullScreen   = true;
   res->iSubtitles    = (int)(0.965 * res->iHeight);
   res->fPixelRatio   = 1.0f;
-  res->strMode       = StringUtils::Format("%dx%d @ %.2f%s - Full Screen", res->iScreenWidth, res->iScreenHeight, res->fRefreshRate,
+  res->strMode       = StringUtils::Format("%dx%d @ %.3f%s - Full Screen", res->iScreenWidth, res->iScreenHeight, res->fRefreshRate,
                                            res->dwFlags & D3DPRESENTFLAG_INTERLACED ? "i" : "");
   res->strId         = mode;
 
-- 
1.9.3

