From ef4b58dd279c14ea5f62fe27690217f80880dfe2 Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Mon, 5 Jan 2015 11:29:15 +0100
Subject: [PATCH 04/12] guilib: mark control dirty when setting to invisible

---
 xbmc/guilib/GUIControl.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/guilib/GUIControl.cpp b/xbmc/guilib/GUIControl.cpp
index e3d04db..ff2086b 100644
--- a/xbmc/guilib/GUIControl.cpp
+++ b/xbmc/guilib/GUIControl.cpp
@@ -511,6 +511,8 @@ void CGUIControl::SetVisible(bool bVisible, bool setVisState)
   {
     m_forceHidden = !bVisible;
     SetInvalid();
+    if (m_forceHidden)
+      MarkDirtyRegion();
   }
   if (m_forceHidden)
   { // reset any visible animations that are in process
-- 
1.9.3

