diff -Naur kodi-14-d6947be/xbmc/peripherals/devices/PeripheralCecAdapter.cpp kodi-14-d6947be.patch/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
--- kodi-14-d6947be/xbmc/peripherals/devices/PeripheralCecAdapter.cpp	2014-11-03 19:59:12.000000000 +0100
+++ kodi-14-d6947be.patch/xbmc/peripherals/devices/PeripheralCecAdapter.cpp	2014-11-04 11:47:00.790424301 +0100
@@ -1269,7 +1269,7 @@
 void CPeripheralCecAdapter::SetConfigurationFromSettings(void)
 {
   // client version matches the version of libCEC that we originally used the API from
-  m_configuration.clientVersion = CEC_CLIENT_VERSION_2_2_0;
+  m_configuration.clientVersion = CEC_CLIENT_VERSION_CURRENT;
 
   // device name 'XBMC'
   snprintf(m_configuration.strDeviceName, 13, "%s", GetSettingString("device_name").c_str());
