diff -aurNp a/xbmc/linux/imx/IMX.cpp b/xbmc/linux/imx/IMX.cpp
--- a/xbmc/linux/imx/IMX.cpp	2016-08-21 23:42:23.000000000 +0200
+++ b/xbmc/linux/imx/IMX.cpp	2016-09-02 12:51:09.516633817 +0200
@@ -198,11 +198,6 @@ bool CIMXFps::Recalc()
 
   double frameNorm = CDVDCodecUtils::NormalizeFrameduration(frameDuration, &hasMatch);
 
-  if (hasMatch && !patternLength)
-    m_patternLength = 1;
-  else
-    m_patternLength = patternLength;
-
   if (!m_hasPattern && hasMatch)
     m_frameDuration = frameNorm;
 
diff -aurNp a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-08-21 23:42:23.000000000 +0200
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-09-02 12:42:54.443140655 +0200
@@ -29,6 +29,7 @@
 #include <sys/ioctl.h>
 #ifdef HAS_IMXVPU
 #include <linux/mxcfb.h>
+#include "cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h" 
 #endif
 #include "utils/log.h"
 #include "utils/RegExp.h"
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-09-04 18:08:27.249816157 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-09-04 18:09:16.803892804 +0200
@@ -55,9 +55,9 @@
 // priorities to those subsystems can result in a very different user
 // experience. With that setting enabled we can build some statistics,
 // as numbers are always better than "feelings"
-#define IMX_PROFILE_BUFFERS
+//#define IMX_PROFILE_BUFFERS
 
-#define IMX_PROFILE
+//#define IMX_PROFILE
 //#define TRACE_FRAMES
 
 #define RENDER_USE_G2D 0
diff -aurNp a/system/peripherals.xml b/system/peripherals.xml
--- a/system/peripherals.xml	2016-09-07 16:50:55.725114234 +0200
+++ b/system/peripherals.xml	2016-09-07 16:51:13.472432981 +0200
@@ -16,7 +16,7 @@
     <setting key="send_inactive_source" type="bool" value="1" label="36025" order="5" />
     <setting key="cec_standby_screensaver" type="bool" value="0" label="36009" order="6" />
     <setting key="cec_wake_screensaver" type="bool" value="1" label="36010" order="7" />
-    <setting key="standby_pc_on_tv_standby" type="enum" value="13011" label="36029" order="8" lvalues="36028|13005|13011" />
+    <setting key="standby_pc_on_tv_standby" type="enum" value="36028" label="36029" order="8" lvalues="36028|13005|13011" />
     <setting key="standby_tv_on_pc_standby" type="bool" value="1" label="36026" order="9" />
     <setting key="use_tv_menu_language" type="bool" value="0" label="36018" order="10" />
     <setting key="pause_playback_on_deactivate" type="bool" value="1" label="36033" configurable="0" />
