disable standby

--- a/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2016-10-07 09:41:03.067310207 +0200
+++ b/xbmc/powermanagement/linux/LogindUPowerSyscall.cpp	2016-10-07 09:41:27.658371366 +0200
@@ -52,8 +52,8 @@ CLogindUPowerSyscall::CLogindUPowerSysca
 
   m_canPowerdown = LogindCheckCapability("CanPowerOff");
   m_canReboot    = LogindCheckCapability("CanReboot");
-  m_canHibernate = LogindCheckCapability("CanHibernate");
-  m_canSuspend   = LogindCheckCapability("CanSuspend");
+  m_canHibernate = false;
+  m_canSuspend   = false;
 
   InhibitDelayLock();
 
--- a/xbmc/powermanagement/linux/UPowerSyscall.cpp.orig	2016-10-07 10:01:18.236917326 +0200
+++ b/xbmc/powermanagement/linux/UPowerSyscall.cpp	2016-10-07 10:01:46.311845478 +0200
@@ -262,8 +262,8 @@ bool CUPowerSyscall::PumpPowerEvents(IPo
 
 void CUPowerSyscall::UpdateCapabilities()
 {
-  m_CanSuspend   = CDBusUtil::GetVariant("org.freedesktop.UPower", "/org/freedesktop/UPower", "org.freedesktop.UPower", "CanSuspend").asBoolean(false);
-  m_CanHibernate = CDBusUtil::GetVariant("org.freedesktop.UPower", "/org/freedesktop/UPower", "org.freedesktop.UPower", "CanHibernate").asBoolean(false);
+  m_CanSuspend   = false;
+  m_CanHibernate = false; 
 }
 
 #endif

From e0c980535b164d7c28555418e5ce2252dbada2dc Mon Sep 17 00:00:00 2001
From: Peter Vicman <peter.vicman@gmail.com>
Date: Thu, 10 Nov 2016 17:05:11 +0100
Subject: [PATCH] fix freeze on lvds display connected to udoo quad

revert 0ff257943e032e0f4b669a65e6449ab1e1699d74
[IMXRenderer] skip rendering same frame 

---
 .../cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp
index 8569c72..f21fa98 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp
@@ -132,7 +132,6 @@ bool CRendererIMX::RenderHook(int index)
 
 bool CRendererIMX::RenderUpdateVideoHook(bool clear, DWORD flags, DWORD alpha)
 {
-  static DWORD flagsPrev;
 #if 0
   static unsigned long long previous = 0;
   unsigned long long current = XbmcThreads::SystemClockMillis();
@@ -142,13 +141,6 @@ bool CRendererIMX::RenderUpdateVideoHook(bool clear, DWORD flags, DWORD alpha)
   CDVDVideoCodecIMXBuffer *buffer = static_cast<CDVDVideoCodecIMXBuffer*>(m_buffers[m_iYV12RenderBuffer].hwDec);
   if (buffer)
   {
-    if (buffer == m_bufHistory[0] && flagsPrev == flags)
-    {
-      g_IMX.WaitVsync();
-      return true;
-    }
-
-    flagsPrev = flags;
     buffer->Lock();
 
     SAFE_RELEASE(m_bufHistory[1]);
-- 
2.7.1

