diff -Naur sunxi-mali/include/GLES/gl.h sunxi-mali.patch/include/GLES/gl.h
--- sunxi-mali/include/GLES/gl.h	2014-06-19 18:03:06.078670740 +0200
+++ sunxi-mali.patch/include/GLES/gl.h	2014-06-23 13:59:30.294348458 +0200
@@ -29,6 +29,7 @@
 typedef float  GLclampf;
 typedef signed int  GLfixed;
 typedef signed int  GLclampx;
+typedef char GLchar;
 
 typedef int * GLintptr;
 typedef int *  GLsizeiptr;
diff -Naur sunxi-mali/include/GLES2/gl2.h sunxi-mali.patch/include/GLES2/gl2.h
--- sunxi-mali/include/GLES2/gl2.h	2014-06-19 18:03:06.078670740 +0200
+++ sunxi-mali.patch/include/GLES2/gl2.h	2014-06-23 13:59:48.346767319 +0200
@@ -32,6 +32,7 @@
 typedef khronos_float_t  GLfloat;
 typedef khronos_float_t  GLclampf;
 typedef khronos_int32_t  GLfixed;
+typedef char             GLchar;
 
 /* GL types for handling large vertex buffer objects */
 typedef khronos_intptr_t GLintptr;
