From 04fd6fa6e549ce0715413b9f4913356bb213683e Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:40:53 +0100
Subject: [PATCH 01/13] [keyboard.xml] add CTRL+R keymapping (Record)

---
 system/keymaps/keyboard.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index d11d765..c83e5e3 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -151,6 +151,7 @@
       <f mod="ctrl">SkipNext</f>          <!-- Skip -->
       <b mod="ctrl">SkipPrevious</b>      <!-- Replay -->
       <d mod="ctrl">Info</d>              <!-- MCE Details -->
+      <r mod="ctrl">Record</r>            <!-- Record -->
       <f10>VolumeUp</f10>                 <!-- MCE Vol up -->
       <f9>VolumeDown</f9>                 <!-- MCE Vol down -->
       <f8>Mute</f8>                       <!-- MCE mute -->

From 44e62426ae8794220e33e7ccb0da1afb14efe9a5 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:41:27 +0100
Subject: [PATCH 02/13] [keyboard.xml] add ALT+F4 keymapping (Exit)

---
 system/keymaps/keyboard.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index c83e5e3..d0bb9e0 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -152,6 +152,7 @@
       <b mod="ctrl">SkipPrevious</b>      <!-- Replay -->
       <d mod="ctrl">Info</d>              <!-- MCE Details -->
       <r mod="ctrl">Record</r>            <!-- Record -->
+      <f4 mod="alt">PreviousMenu</f4>     <!-- Exit -->
       <f10>VolumeUp</f10>                 <!-- MCE Vol up -->
       <f9>VolumeDown</f9>                 <!-- MCE Vol down -->
       <f8>Mute</f8>                       <!-- MCE mute -->

From 849842c5f62e39b882a6720af58d8b2aee09112a Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:48:17 +0100
Subject: [PATCH 03/13] [mouse.xml] add right click keymapping (Info)

---
 system/keymaps/mouse.xml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/system/keymaps/mouse.xml b/system/keymaps/mouse.xml
index d83fab0..c2b39f8 100644
--- a/system/keymaps/mouse.xml
+++ b/system/keymaps/mouse.xml
@@ -29,4 +29,11 @@
       <wheeldown>NextPicture</wheeldown>
     </mouse>
   </SlideShow>
+  <FullscreenVideo>
+    <mouse>
+      <rightclick>Info</rightclick>
+    </mouse>
+  </FullscreenVideo>
+
+
 </keymap>

From cc0dea22b08e085a31943f40a8f50cb4fa5a718b Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:50:28 +0100
Subject: [PATCH 04/13] [keyboard.xml] add KEY_ZOOM keymapping (Zoom)

---
 system/keymaps/keyboard.xml            | 2 ++
 xbmc/input/XBMC_keysym.h               | 1 +
 xbmc/input/XBMC_keytable.cpp           | 1 +
 xbmc/input/XBMC_vkeys.h                | 1 +
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 5 files changed, 6 insertions(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index d0bb9e0..b538a33 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -304,6 +304,7 @@
       <i>Info</i>
       <o>CodecInfo</o>
       <z>AspectRatio</z>
+      <zoom>AspectRatio</zoom>
       <t>ShowSubtitles</t>
       <t mod="ctrl">SubtitleAlign</t>
       <l>NextSubtitle</l>
@@ -474,6 +475,7 @@
       <i>Info</i>
       <o>CodecInfo</o>
       <z>AspectRatio</z>
+      <zoom>AspectRatio</zoom>
       <t>ShowSubtitles</t>
       <l>NextSubtitle</l>
       <a>AudioDelay</a>
diff --git a/xbmc/input/XBMC_keysym.h b/xbmc/input/XBMC_keysym.h
index 873ed8a..f14712a 100644
--- a/xbmc/input/XBMC_keysym.h
+++ b/xbmc/input/XBMC_keysym.h
@@ -224,6 +224,7 @@ typedef enum {
   XBMCK_GREEN       = 0x148,
   XBMCK_YELLOW      = 0x149,
   XBMCK_BLUE        = 0x14a,
+  XBMCK_ZOOM        = 0x14b,
 
   // Add any other keys here
 
diff --git a/xbmc/input/XBMC_keytable.cpp b/xbmc/input/XBMC_keytable.cpp
index dc46204..22942be 100644
--- a/xbmc/input/XBMC_keytable.cpp
+++ b/xbmc/input/XBMC_keytable.cpp
@@ -238,6 +238,7 @@ static const XBMCKEYTABLE XBMCKeyTable[] =
 , { XBMCK_GREEN,                  0,    0, XBMCVK_GREEN,         "green" }
 , { XBMCK_YELLOW,                 0,    0, XBMCVK_YELLOW,        "yellow" }
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
+, { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff --git a/xbmc/input/XBMC_vkeys.h b/xbmc/input/XBMC_vkeys.h
index 7fd8a86..ccec23a 100644
--- a/xbmc/input/XBMC_vkeys.h
+++ b/xbmc/input/XBMC_vkeys.h
@@ -216,6 +216,7 @@ typedef enum {
   XBMCVK_GREEN          = 0xE4,
   XBMCVK_YELLOW         = 0xE5,
   XBMCVK_BLUE           = 0xE6,
+  XBMCVK_ZOOM           = 0xE7,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 723f7ae..c999e69 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -255,6 +255,7 @@ KeyMap keyMap[] = {
   { KEY_PRINT         , XBMCK_PRINT       },
   { KEY_QUESTION      , XBMCK_HELP        },
   { KEY_BACK          , XBMCK_BACKSPACE   },
+  { KEY_ZOOM          , XBMCK_ZOOM        },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 9066630c7e3a8388b18673ef9a7ae2929479cd09 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:54:34 +0100
Subject: [PATCH 05/13] [keyboard.xml] add KEY_TEXT keymapping (Text)

---
 system/keymaps/keyboard.xml            | 6 ++++++
 xbmc/input/XBMC_keysym.h               | 1 +
 xbmc/input/XBMC_keytable.cpp           | 1 +
 xbmc/input/XBMC_vkeys.h                | 1 +
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 5 files changed, 10 insertions(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index b538a33..022177c 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -316,6 +316,7 @@
       <escape>Fullscreen</escape>
       <c>Playlist</c>
       <v>ActivateWindow(Teletext)</v>
+      <text>ActivateWindow(Teletext)</text>
       <up mod="ctrl">SubtitleShiftUp</up>
       <down mod="ctrl">SubtitleShiftDown</down>
       <pageup>SkipNext</pageup>
@@ -364,6 +365,7 @@
       <i>Info</i>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Addon.Default.OpenSettings(xbmc.player.musicviz)</v>
+      <text>Addon.Default.OpenSettings(xbmc.player.musicviz)</text>
       <n>ActivateWindow(MusicPlaylist)</n>
       <left>StepBack</left>
       <right>StepForward</right>
@@ -388,6 +390,7 @@
       <o>CodecInfo</o>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Addon.Default.OpenSettings(xbmc.player.musicviz)</v>
+      <text>Addon.Default.OpenSettings(xbmc.player.musicviz)</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </MusicOSD>
@@ -402,6 +405,7 @@
       <o>CodecInfo</o>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Back</v>
+      <text>Back</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </VisualisationSettings>
@@ -416,6 +420,7 @@
       <o>CodecInfo</o>
       <p>Back</p>
       <v>Back</v>
+      <text>Back</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </VisualisationPresetList>
@@ -565,6 +570,7 @@
   <Teletext>
     <keyboard>
       <v>Back</v>
+      <text>Back</text>
     </keyboard>
   </Teletext>
   <Favourites>
diff --git a/xbmc/input/XBMC_keysym.h b/xbmc/input/XBMC_keysym.h
index f14712a..0879943 100644
--- a/xbmc/input/XBMC_keysym.h
+++ b/xbmc/input/XBMC_keysym.h
@@ -225,6 +225,7 @@ typedef enum {
   XBMCK_YELLOW      = 0x149,
   XBMCK_BLUE        = 0x14a,
   XBMCK_ZOOM        = 0x14b,
+  XBMCK_TEXT        = 0x14c,
 
   // Add any other keys here
 
diff --git a/xbmc/input/XBMC_keytable.cpp b/xbmc/input/XBMC_keytable.cpp
index 22942be..4dea61c 100644
--- a/xbmc/input/XBMC_keytable.cpp
+++ b/xbmc/input/XBMC_keytable.cpp
@@ -239,6 +239,7 @@ static const XBMCKEYTABLE XBMCKeyTable[] =
 , { XBMCK_YELLOW,                 0,    0, XBMCVK_YELLOW,        "yellow" }
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
+, { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff --git a/xbmc/input/XBMC_vkeys.h b/xbmc/input/XBMC_vkeys.h
index ccec23a..45ab72c 100644
--- a/xbmc/input/XBMC_vkeys.h
+++ b/xbmc/input/XBMC_vkeys.h
@@ -217,6 +217,7 @@ typedef enum {
   XBMCVK_YELLOW         = 0xE5,
   XBMCVK_BLUE           = 0xE6,
   XBMCVK_ZOOM           = 0xE7,
+  XBMCVK_TEXT           = 0xE8,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index c999e69..c3ab883 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -256,6 +256,7 @@ KeyMap keyMap[] = {
   { KEY_QUESTION      , XBMCK_HELP        },
   { KEY_BACK          , XBMCK_BACKSPACE   },
   { KEY_ZOOM          , XBMCK_ZOOM        },
+  { KEY_TEXT          , XBMCK_TEXT        },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 00ed30f3f6e8a4fef5551a9312a451ed63a6a808 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 10:56:56 +0100
Subject: [PATCH 06/13] [keyboard.xml] add KEY_FAVORITES keymapping (Favorites)

---
 system/keymaps/keyboard.xml            | 1 +
 xbmc/input/XBMC_keysym.h               | 1 +
 xbmc/input/XBMC_keytable.cpp           | 1 +
 xbmc/input/XBMC_vkeys.h                | 1 +
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 5 files changed, 5 insertions(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index 022177c..05278c7 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -123,6 +123,7 @@
       <browser_stop/>
       <browser_search/>
       <browser_favorites>ActivateWindow(Favourites)</browser_favorites>
+      <favorites>ActivateWindow(Favourites)</favorites>
       <browser_home>ActivateWindow(Home)</browser_home>
       <volume_mute>Mute</volume_mute>
       <volume_down>VolumeDown</volume_down>
diff --git a/xbmc/input/XBMC_keysym.h b/xbmc/input/XBMC_keysym.h
index 0879943..0ecf3e4 100644
--- a/xbmc/input/XBMC_keysym.h
+++ b/xbmc/input/XBMC_keysym.h
@@ -226,6 +226,7 @@ typedef enum {
   XBMCK_BLUE        = 0x14a,
   XBMCK_ZOOM        = 0x14b,
   XBMCK_TEXT        = 0x14c,
+  XBMCK_FAVORITES   = 0x14d,
 
   // Add any other keys here
 
diff --git a/xbmc/input/XBMC_keytable.cpp b/xbmc/input/XBMC_keytable.cpp
index 4dea61c..13da5b1 100644
--- a/xbmc/input/XBMC_keytable.cpp
+++ b/xbmc/input/XBMC_keytable.cpp
@@ -240,6 +240,7 @@ static const XBMCKEYTABLE XBMCKeyTable[] =
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 , { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
+, { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff --git a/xbmc/input/XBMC_vkeys.h b/xbmc/input/XBMC_vkeys.h
index 45ab72c..9df9b51 100644
--- a/xbmc/input/XBMC_vkeys.h
+++ b/xbmc/input/XBMC_vkeys.h
@@ -218,6 +218,7 @@ typedef enum {
   XBMCVK_BLUE           = 0xE6,
   XBMCVK_ZOOM           = 0xE7,
   XBMCVK_TEXT           = 0xE8,
+  XBMCVK_FAVORITES      = 0xE9,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index c3ab883..523c37e 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -257,6 +257,7 @@ KeyMap keyMap[] = {
   { KEY_BACK          , XBMCK_BACKSPACE   },
   { KEY_ZOOM          , XBMCK_ZOOM        },
   { KEY_TEXT          , XBMCK_TEXT        },
+  { KEY_FAVORITES     , XBMCK_FAVORITES   },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 302b3549dcac380365fd81658794093ec6a40d3d Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:01:44 +0100
Subject: [PATCH 07/13] [keyboard.xml] add
 KEY_RED,KEY_GREEN,KEY_YELLOW,KEY_BLUE keymapping (colored buttons)

---
 system/keymaps/keyboard.xml            | 29 +++++++++++++++++++++++++++++
 xbmc/input/linux/LinuxInputDevices.cpp |  4 ++++
 2 files changed, 33 insertions(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index 05278c7..0c628cb 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -116,6 +116,11 @@
       <j>ActivateWindow(RadioChannels)</j>
       <k>ActivateWindow(TVRecordings)</k>
       <b>ActivateWindow(TVTimers)</b>
+      <!-- PVR -->
+      <red>ActivateWindow(TVChannels)</red>
+      <green>ActivateWindow(MyVideos)</green>
+      <yellow>ActivateWindow(MyMusic)</yellow>
+      <blue>ActivateWindow(MyPictures)</blue>
       <!-- Multimedia keyboard keys -->
       <browser_back>Back</browser_back>
       <browser_forward/>
@@ -214,6 +219,10 @@
       <delete>Delete</delete>
       <r>Rename</r>
       <k>PreviousMenu</k>
+      <red>Red</red>
+      <green>Green</green>
+      <yellow>Yellow</yellow>
+      <blue>Blue</blue>
     </keyboard>
   </MyTVRecordings>
   <MyTVTimers>
@@ -221,6 +230,10 @@
       <delete>Delete</delete>
       <r>Rename</r>
       <b>PreviousMenu</b>
+      <red>Red</red>
+      <green>Green</green>
+      <yellow>Yellow</yellow>
+      <blue>Blue</blue>
     </keyboard>
   </MyTVTimers>
   <MyRadioChannels>
@@ -233,12 +246,20 @@
     <keyboard>
       <delete>Delete</delete>
       <r>Rename</r>
+      <red>Red</red>
+      <green>Green</green>
+      <yellow>Yellow</yellow>
+      <blue>Blue</blue>
     </keyboard>
   </MyRadioRecordings>
   <MyRadioTimers>
     <keyboard>
       <delete>Delete</delete>
       <r>Rename</r>
+      <red>Red</red>
+      <green>Green</green>
+      <yellow>Yellow</yellow>
+      <blue>Blue</blue>
     </keyboard>
   </MyRadioTimers>
   <TVGuide>
@@ -572,6 +593,10 @@
     <keyboard>
       <v>Back</v>
       <text>Back</text>
+      <red>Red</red>
+      <green>Green</green>
+      <yellow>Yellow</yellow>
+      <blue>Blue</blue>
     </keyboard>
   </Teletext>
   <Favourites>
@@ -660,9 +685,13 @@
   <Addon>
     <keyboard>
       <f1>Red</f1>
+      <red>Red</red>
       <f2>Green</f2>
+      <green>Green</green>
       <f3>Yellow</f3>
+      <yellow>Yellow</yellow>
       <f4>Blue</f4>
+      <blue>Blue</blue>
     </keyboard>
   </Addon>
 </keymap>
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 523c37e..86dcc3c 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -258,6 +258,10 @@ KeyMap keyMap[] = {
   { KEY_ZOOM          , XBMCK_ZOOM        },
   { KEY_TEXT          , XBMCK_TEXT        },
   { KEY_FAVORITES     , XBMCK_FAVORITES   },
+  { KEY_RED           , XBMCK_RED         },
+  { KEY_GREEN         , XBMCK_GREEN       },
+  { KEY_YELLOW        , XBMCK_YELLOW      },
+  { KEY_BLUE          , XBMCK_BLUE        },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 91372b6bb35feb74ec68c72d94d4bc3c99f806b4 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:03:14 +0100
Subject: [PATCH 08/13] [keyboard.xml] add KEY_HOMEPAGE keymapping (Home)

---
 system/keymaps/keyboard.xml            | 1 +
 xbmc/input/XBMC_keysym.h               | 1 +
 xbmc/input/XBMC_keytable.cpp           | 1 +
 xbmc/input/XBMC_vkeys.h                | 1 +
 xbmc/input/linux/LinuxInputDevices.cpp | 3 +--
 5 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index 0c628cb..4eaf543 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -130,6 +130,7 @@
       <browser_favorites>ActivateWindow(Favourites)</browser_favorites>
       <favorites>ActivateWindow(Favourites)</favorites>
       <browser_home>ActivateWindow(Home)</browser_home>
+      <homepage>ActivateWindow(Home)</homepage>
       <volume_mute>Mute</volume_mute>
       <volume_down>VolumeDown</volume_down>
       <volume_up>VolumeUp</volume_up>
diff --git a/xbmc/input/XBMC_keysym.h b/xbmc/input/XBMC_keysym.h
index 0ecf3e4..169c374 100644
--- a/xbmc/input/XBMC_keysym.h
+++ b/xbmc/input/XBMC_keysym.h
@@ -227,6 +227,7 @@ typedef enum {
   XBMCK_ZOOM        = 0x14b,
   XBMCK_TEXT        = 0x14c,
   XBMCK_FAVORITES   = 0x14d,
+  XBMCK_HOMEPAGE    = 0x14e,
 
   // Add any other keys here
 
diff --git a/xbmc/input/XBMC_keytable.cpp b/xbmc/input/XBMC_keytable.cpp
index 13da5b1..7d79f23 100644
--- a/xbmc/input/XBMC_keytable.cpp
+++ b/xbmc/input/XBMC_keytable.cpp
@@ -241,6 +241,7 @@ static const XBMCKEYTABLE XBMCKeyTable[] =
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 , { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
 , { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
+, { XBMCK_HOMEPAGE ,              0,    0, XBMCVK_HOMEPAGE,      "homepage" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff --git a/xbmc/input/XBMC_vkeys.h b/xbmc/input/XBMC_vkeys.h
index 9df9b51..9c6b111 100644
--- a/xbmc/input/XBMC_vkeys.h
+++ b/xbmc/input/XBMC_vkeys.h
@@ -219,6 +219,7 @@ typedef enum {
   XBMCVK_ZOOM           = 0xE7,
   XBMCVK_TEXT           = 0xE8,
   XBMCVK_FAVORITES      = 0xE9,
+  XBMCVK_HOMEPAGE       = 0xEA,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 86dcc3c..36406dd 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -262,13 +262,12 @@ KeyMap keyMap[] = {
   { KEY_GREEN         , XBMCK_GREEN       },
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
+  { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
   { 381               , XBMCK_UP          }, // Yellow
   { 366               , XBMCK_DOWN        }, // Blue
-  // Rii i7 Home button / wetek openelec remote (code 172)
-  { KEY_HOMEPAGE      , XBMCK_HOME        },
 };
 
 typedef enum

From 39beaea722d680242660ecab28cc430fa1ac0d47 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:06:12 +0100
Subject: [PATCH 09/13] [LinuxInputDevices] add KEY_MAIL support (launch_mail)

---
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 36406dd..2a86789 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -263,6 +263,7 @@ KeyMap keyMap[] = {
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
+  { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 398d2b81b9a0b3a026e0fef25d999feb4ee27194 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:07:27 +0100
Subject: [PATCH 10/13] [LinuxInputDevices] add KEY_SEARCH support
 (browser_search)

---
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 2a86789..9ae7cb8 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -264,6 +264,7 @@ KeyMap keyMap[] = {
   { KEY_BLUE          , XBMCK_BLUE        },
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
   { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
+  { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From c9bffb68c2ff44d11527f9d3c35838490a442295 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:08:31 +0100
Subject: [PATCH 11/13] [LinuxInputDevices] add KEY_FILE support
 (launch_file_browser)

---
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 9ae7cb8..64389c4 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -265,6 +265,7 @@ KeyMap keyMap[] = {
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
   { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
+  { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 9f11b54cc4e431ba13df9544eaff8dca77220105 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:09:57 +0100
Subject: [PATCH 12/13] [LinuxInputDevices] add KEY_SELECT support (return)

---
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 64389c4..9f3e367 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -266,6 +266,7 @@ KeyMap keyMap[] = {
   { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
+  { KEY_SELECT        , XBMCK_RETURN      },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green

From 6da8abe0baa6d8fcb82b8599fa601dce7880ccfe Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 20 Nov 2015 11:11:27 +0100
Subject: [PATCH 13/13] [keyboard.xml] add KEY_CONFIG keymapping (config)

---
 system/keymaps/keyboard.xml            | 1 +
 xbmc/input/XBMC_keysym.h               | 1 +
 xbmc/input/XBMC_keytable.cpp           | 1 +
 xbmc/input/XBMC_vkeys.h                | 1 +
 xbmc/input/linux/LinuxInputDevices.cpp | 1 +
 5 files changed, 5 insertions(+)

diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index 4eaf543..7fe13d116 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -129,6 +129,7 @@
       <browser_search/>
       <browser_favorites>ActivateWindow(Favourites)</browser_favorites>
       <favorites>ActivateWindow(Favourites)</favorites>
+      <config>ActivateWindow(Settings)</config>
       <browser_home>ActivateWindow(Home)</browser_home>
       <homepage>ActivateWindow(Home)</homepage>
       <volume_mute>Mute</volume_mute>
diff --git a/xbmc/input/XBMC_keysym.h b/xbmc/input/XBMC_keysym.h
index 169c374..89c21f5 100644
--- a/xbmc/input/XBMC_keysym.h
+++ b/xbmc/input/XBMC_keysym.h
@@ -228,6 +228,7 @@ typedef enum {
   XBMCK_TEXT        = 0x14c,
   XBMCK_FAVORITES   = 0x14d,
   XBMCK_HOMEPAGE    = 0x14e,
+  XBMCK_CONFIG      = 0x14f,
 
   // Add any other keys here
 
diff --git a/xbmc/input/XBMC_keytable.cpp b/xbmc/input/XBMC_keytable.cpp
index 7d79f23..4d05464 100644
--- a/xbmc/input/XBMC_keytable.cpp
+++ b/xbmc/input/XBMC_keytable.cpp
@@ -242,6 +242,7 @@ static const XBMCKEYTABLE XBMCKeyTable[] =
 , { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
 , { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
 , { XBMCK_HOMEPAGE ,              0,    0, XBMCVK_HOMEPAGE,      "homepage" }
+, { XBMCK_CONFIG,                 0,    0, XBMCVK_CONFIG,        "config" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff --git a/xbmc/input/XBMC_vkeys.h b/xbmc/input/XBMC_vkeys.h
index 9c6b111..af114c9 100644
--- a/xbmc/input/XBMC_vkeys.h
+++ b/xbmc/input/XBMC_vkeys.h
@@ -220,6 +220,7 @@ typedef enum {
   XBMCVK_TEXT           = 0xE8,
   XBMCVK_FAVORITES      = 0xE9,
   XBMCVK_HOMEPAGE       = 0xEA,
+  XBMCVK_CONFIG         = 0xEB,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 9f3e367..1c17a13 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -267,6 +267,7 @@ KeyMap keyMap[] = {
   { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
   { KEY_SELECT        , XBMCK_RETURN      },
+  { KEY_CONFIG        , XBMCK_CONFIG      },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
