diff -Naur a/project/cmake/modules/FindPulseAudio.cmake b/project/cmake/modules/FindPulseAudio.cmake
--- a/project/cmake/modules/FindPulseAudio.cmake	2016-09-17 08:35:33.000000000 -0700
+++ b/project/cmake/modules/FindPulseAudio.cmake	2016-09-22 22:36:22.818788337 -0700
@@ -29,9 +29,6 @@
 find_library(PULSEAUDIO_LIBRARY NAMES pulse libpulse
                                 PATHS ${PC_PULSEAUDIO_LIBDIR} ${PC_PULSEAUDIO_LIBRARY_DIRS})
 
-find_library(PULSEAUDIO_MAINLOOP_LIBRARY NAMES pulse-mainloop pulse-mainloop-glib libpulse-mainloop-glib
-                                         PATHS ${PC_PULSEAUDIO_LIBDIR} ${PC_PULSEAUDIO_LIBRARY_DIRS})
-
 if(PC_PULSEAUDIO_VERSION)
   set(PULSEAUDIO_VERSION_STRING ${PC_PULSEAUDIO_VERSION})
 elseif(PULSEAUDIO_INCLUDE_DIR AND EXISTS "${PULSEAUDIO_INCLUDE_DIR}/pulse/version.h")
@@ -42,19 +39,14 @@
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(PulseAudio
-                                  REQUIRED_VARS PULSEAUDIO_LIBRARY PULSEAUDIO_MAINLOOP_LIBRARY PULSEAUDIO_INCLUDE_DIR
+                                  REQUIRED_VARS PULSEAUDIO_LIBRARY PULSEAUDIO_INCLUDE_DIR
                                   VERSION_VAR PULSEAUDIO_VERSION_STRING)
 
 if(PULSEAUDIO_FOUND)
   set(PULSEAUDIO_INCLUDE_DIRS ${PULSEAUDIO_INCLUDE_DIR})
-  set(PULSEAUDIO_LIBRARIES ${PULSEAUDIO_LIBRARY} ${PULSEAUDIO_MAINLOOP_LIBRARY})
+  set(PULSEAUDIO_LIBRARIES ${PULSEAUDIO_LIBRARY})
   set(PULSEAUDIO_DEFINITIONS -DHAVE_LIBPULSE=1)
 
-  if(NOT TARGET PulseAudio::PulseAudioMainloop)
-    add_library(PulseAudio::PulseAudioMainloop UNKNOWN IMPORTED)
-    set_target_properties(PulseAudio::PulseAudioMainloop PROPERTIES
-                                                         IMPORTED_LOCATION "${PULSEAUDIO_MAINLOOP_LIBRARY}")
-  endif()
   if(NOT TARGET PulseAudio::PulseAudio)
     add_library(PulseAudio::PulseAudio UNKNOWN IMPORTED)
     set_target_properties(PulseAudio::PulseAudio PROPERTIES
@@ -65,4 +57,4 @@
   endif()
 endif()
 
-mark_as_advanced(PULSEAUDIO_INCLUDE_DIR PULSEAUDIO_LIBRARY PULSEAUDIO_MAINLOOP_LIBRARY)
+mark_as_advanced(PULSEAUDIO_INCLUDE_DIR PULSEAUDIO_LIBRARY)
