diff -Naur a/cmake/modules/FindSmbClient.cmake b/cmake/modules/FindSmbClient.cmake
--- a/cmake/modules/FindSmbClient.cmake	2016-09-17 08:35:33.000000000 -0700
+++ b/cmake/modules/FindSmbClient.cmake	2016-09-25 13:38:38.969125892 -0700
@@ -14,7 +14,7 @@
 #
 #   SmbClient::SmbClient   - The SmbClient library
 
-if(PKGCONFIG_FOUND)
+if(PKG_CONFIG_FOUND)
   pkg_check_modules(PC_SMBCLIENT smbclient QUIET)
 endif()
 
@@ -31,7 +31,19 @@ find_package_handle_standard_args(SmbClient
                                   VERSION_VAR SMBCLIENT_VERSION)
 
 if(SMBCLIENT_FOUND)
-  set(SMBCLIENT_LIBRARIES ${SMBCLIENT_LIBRARY})
+  # check if smbclient libs are statically linked
+  set(SMBCLIENT_LIB_TYPE SHARED)
+  if(PC_SMBCLIENT_STATIC_LDFLAGS)
+    set(SMBCLIENT_LDFLAGS ${PC_SMBCLIENT_STATIC_LDFLAGS} CACHE STRING "smbclient linker flags" FORCE)
+    set(SMBCLIENT_LIB_TYPE STATIC)
+    find_library(TALLOC_LIBRARY talloc)
+    find_library(TDB_LIBRARY tdb)
+    find_library(TEVENT_LIBRARY tevent)
+    find_library(WBCLIENT_LIBRARY wbclient)
+    find_library(RESOLV_LIBRARY resolv)
+    set(SMBCLIENT_LIBRARIES ${SMBCLIENT_LIBRARY} ${TALLOC_LIBRARY} ${TDB_LIBRARY} ${TEVENT_LIBRARY} ${WBCLIENT_LIBRARY} ${RESOLV_LIBRARY})
+  endif()
+
   set(SMBCLIENT_INCLUDE_DIRS ${SMBCLIENT_INCLUDE_DIR})
   set(SMBCLIENT_DEFINITIONS -DHAVE_LIBSMBCLIENT=1)
 
