diff -Naur kodi-14-88f9fb0/xbmc/filesystem/File.cpp kodi-14-88f9fb0.patch/xbmc/filesystem/File.cpp
--- kodi-14-88f9fb0/xbmc/filesystem/File.cpp	2014-12-14 16:05:16.000000000 +0100
+++ kodi-14-88f9fb0.patch/xbmc/filesystem/File.cpp	2014-12-14 20:16:01.033786616 +0100
@@ -298,7 +298,7 @@
       return false;
     }
 
-    if (m_pFile->GetChunkSize() && !(m_flags & READ_CHUNKED))
+    if (m_pFile->GetChunkSize() > 1 && !(m_flags & READ_CHUNKED))
     {
       m_pBuffer = new CFileStreamBuffer(0);
       m_pBuffer->Attach(m_pFile);
