From 8c8803dc15ea7820f0f980bc12fcea2c7b8c4487 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Tue, 17 Dec 2019 15:51:23 +0000
Subject: [PATCH] refresh /etc/environment on install and uninstall

---
 xbmc/addons/Addon.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/xbmc/addons/Addon.cpp
+++ b/xbmc/addons/Addon.cpp
@@ -445,6 +445,9 @@ void OnPostInstall(const AddonPtr& addon
   // OE
 
   addon->OnPostInstall(update, modal);
+ 
+  if (addon->Type() == ADDON_SERVICE || addon->Type() == ADDON_SCRIPT)
+    system("/usr/bin/environment-setup");
 }
 
 void OnPreUnInstall(const AddonPtr& addon)
@@ -457,6 +460,9 @@ void OnPreUnInstall(const AddonPtr& addo
 void OnPostUnInstall(const AddonPtr& addon)
 {
   addon->OnPostUnInstall();
+
+  if (addon->Type() == ADDON_SERVICE || addon->Type() == ADDON_SCRIPT)
+    system("/usr/bin/environment-setup");
 }
 
 } /* namespace ADDON */
