--- a/cmake/modules/buildtools/FindTexturePacker.cmake
+++ b/cmake/modules/buildtools/FindTexturePacker.cmake
@@ -52,15 +52,6 @@ if(NOT TARGET TexturePacker::TexturePack
       endif()
     endif()
 
-    # Ship TexturePacker only on Linux and FreeBSD
-    if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "Linux")
-      # But skip shipping it if build architecture can be executed on host
-      # and TEXTUREPACKER_EXECUTABLE is found
-      if(NOT (HOST_CAN_EXECUTE_TARGET AND TEXTUREPACKER_EXECUTABLE))
-        set(INTERNAL_TEXTUREPACKER_INSTALLABLE TRUE CACHE BOOL "" FORCE)
-      endif()
-    endif()
-
     # Use it during build if build architecture can be executed on host
     # and TEXTUREPACKER_EXECUTABLE is not found
     if(HOST_CAN_EXECUTE_TARGET AND NOT TEXTUREPACKER_EXECUTABLE)
