diff -Naur kodi-15.0-beta1-d1a2c33/xbmc/settings/Settings.cpp kodi-15.0-beta1-d1a2c33.patch/xbmc/settings/Settings.cpp
--- kodi-15.0-beta1-d1a2c33/xbmc/settings/Settings.cpp	2015-05-03 11:53:48.000000000 +0200
+++ kodi-15.0-beta1-d1a2c33.patch/xbmc/settings/Settings.cpp	2015-05-03 16:57:27.337681882 +0200
@@ -456,6 +456,9 @@
 #elif defined(TARGET_ANDROID)
   if (CFile::Exists(SETTINGS_XML_FOLDER "android.xml") && !Initialize(SETTINGS_XML_FOLDER "android.xml"))
     CLog::Log(LOGFATAL, "Unable to load android-specific settings definitions");
+#elif defined(HAS_LIBAMCODEC)
+  if (CFile::Exists(SETTINGS_XML_FOLDER "aml.xml") && !Initialize(SETTINGS_XML_FOLDER "aml.xml"))
+    CLog::Log(LOGFATAL, "Unable to load aml-specific settings definitions");
 #elif defined(TARGET_RASPBERRY_PI)
   if (CFile::Exists(SETTINGS_XML_FOLDER "rbp.xml") && !Initialize(SETTINGS_XML_FOLDER "rbp.xml"))
     CLog::Log(LOGFATAL, "Unable to load rbp-specific settings definitions");
