--- a/cmake/modules/FindTexturePacker.cmake
+++ b/cmake/modules/FindTexturePacker.cmake
@@ -52,10 +52,6 @@ if(NOT TARGET TexturePacker::TexturePack
       endif()
     endif()
 
-    # Ship TexturePacker only on Linux and FreeBSD
-    if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "Linux")
-      set(INTERNAL_TEXTUREPACKER_INSTALLABLE TRUE)
-    endif()
 
     # Use it during build if build architecture can be executed on host
     # and TEXTUREPACKER_EXECUTABLE is not found
