diff -Naur kodi-15.2-02e7013/system/keymaps/keyboard.xml kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml
--- kodi-15.2-02e7013/system/keymaps/keyboard.xml	2015-10-23 11:58:14.088175555 +0200
+++ kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml	2015-10-23 12:05:41.104997901 +0200
@@ -111,6 +111,7 @@
       <browser_stop/>
       <browser_search/>
       <browser_favorites>ActivateWindow(Favourites)</browser_favorites>
+      <favorites>ActivateWindow(Favourites)</favorites>
       <browser_home>ActivateWindow(Home)</browser_home>
       <volume_mute>Mute</volume_mute>
       <volume_down>VolumeDown</volume_down>
diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-23 11:58:13.830175091 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-23 12:02:55.616691348 +0200
@@ -255,6 +255,7 @@
   { KEY_BACK          , XBMCK_BACKSPACE   },
   { KEY_ZOOM          , XBMCK_ZOOM        },
   { KEY_TEXT          , XBMCK_TEXT        },
+  { KEY_FAVORITES     , XBMCK_FAVORITES   },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h	2015-10-23 11:58:13.830175091 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h	2015-10-23 12:00:49.718459512 +0200
@@ -226,6 +226,7 @@
   XBMCK_BLUE        = 0x14a,
   XBMCK_ZOOM        = 0x14b,
   XBMCK_TEXT        = 0x14c,
+  XBMCK_FAVORITES   = 0x14d,
 
   // Add any other keys here
 
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp	2015-10-23 11:58:13.832175095 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp	2015-10-23 12:02:00.186589235 +0200
@@ -240,6 +240,7 @@
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 , { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
+, { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h	2015-10-23 11:58:13.831175093 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h	2015-10-23 12:04:24.224856326 +0200
@@ -218,6 +218,7 @@
   XBMCVK_BLUE           = 0xE6,
   XBMCVK_ZOOM           = 0xE7,
   XBMCVK_TEXT           = 0xE8,
+  XBMCVK_FAVORITES      = 0xE9,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
