diff -Naur kodi-15.2-02e7013/system/keymaps/keyboard.xml kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml
--- kodi-15.2-02e7013/system/keymaps/keyboard.xml	2015-10-21 23:41:22.816835383 +0200
+++ kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml	2015-10-21 23:45:30.468434393 +0200
@@ -261,6 +261,7 @@
       <i>Info</i>
       <o>CodecInfo</o>
       <z>AspectRatio</z>
+      <zoom>AspectRatio</zoom>
       <t>ShowSubtitles</t>
       <t mod="ctrl">SubtitleAlign</t>
       <l>NextSubtitle</l>
@@ -428,6 +429,7 @@
       <i>Info</i>
       <o>CodecInfo</o>
       <z>AspectRatio</z>
+      <zoom>AspectRatio</zoom>
       <t>ShowSubtitles</t>
       <l>NextSubtitle</l>
       <a>AudioDelay</a>
diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-21 23:41:22.815835358 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-21 23:43:12.144171466 +0200
@@ -253,6 +253,7 @@
   { KEY_PRINT         , XBMCK_PRINT       },
   { KEY_QUESTION      , XBMCK_HELP        },
   { KEY_BACK          , XBMCK_BACKSPACE   },
+  { KEY_ZOOM          , XBMCK_ZOOM        },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h	2015-10-20 23:51:46.000000000 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h	2015-10-22 10:28:02.144619319 +0200
@@ -224,6 +224,7 @@
   XBMCK_GREEN       = 0x148,
   XBMCK_YELLOW      = 0x149,
   XBMCK_BLUE        = 0x14a,
+  XBMCK_ZOOM        = 0x14b,
 
   // Add any other keys here
 
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp	2015-10-20 23:51:46.000000000 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp	2015-10-22 10:32:56.784811569 +0200
@@ -238,6 +238,7 @@
 , { XBMCK_GREEN,                  0,    0, XBMCVK_GREEN,         "green" }
 , { XBMCK_YELLOW,                 0,    0, XBMCVK_YELLOW,        "yellow" }
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
+, { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h	2015-10-20 23:51:46.000000000 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h	2015-10-22 10:33:50.039123956 +0200
@@ -216,6 +216,7 @@
   XBMCVK_GREEN          = 0xE4,
   XBMCVK_YELLOW         = 0xE5,
   XBMCVK_BLUE           = 0xE6,
+  XBMCVK_ZOOM           = 0xE7,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
