diff --git a/xbmc/XBApplicationEx.cpp b/xbmc/XBApplicationEx.cpp
index f6fffd8..6eeacf6 100644
--- a/xbmc/XBApplicationEx.cpp
+++ b/xbmc/XBApplicationEx.cpp
@@ -53,6 +53,10 @@ int CXBApplicationEx::Run(const CAppParamParser &params)
     KODI::MESSAGING::CApplicationMessenger::GetInstance().PostMsg(TMSG_PLAYLISTPLAYER_PLAY, -1);
   }
 
+#ifdef TARGET_RASPBERRY_PI
+  system("/sbin/fbset -g 1 1 1 1 32");
+#endif
+
   // Run xbmc
   while (!m_bStop)
   {
