diff --git a/xbmc/XBApplicationEx.cpp b/xbmc/XBApplicationEx.cpp
index f6fffd8..6eeacf6 100644
--- a/xbmc/XBApplicationEx.cpp
+++ b/xbmc/XBApplicationEx.cpp
@@ -85,6 +85,10 @@ INT CXBApplicationEx::Run()
   unsigned int frameTime = 0;
   const unsigned int noRenderFrameTime = 15;  // Simulates ~66fps
 
+#ifdef TARGET_RASPBERRY_PI
+  system("/sbin/fbset -g 1 1 1 1 32");
+#endif
+
   // Run xbmc
   while (!m_bStop)
   {
