diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-29 15:52:52.297267009 +0100
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-29 15:56:57.543790220 +0100
@@ -261,6 +261,7 @@
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
+  { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
