diff -Naur kodi-15.0-beta1-d1a2c33/xbmc/utils/AMLUtils.cpp kodi-15.0-beta1-d1a2c33.patch/xbmc/utils/AMLUtils.cpp
--- kodi-15.0-beta1-d1a2c33/xbmc/utils/AMLUtils.cpp	2015-05-03 11:53:48.000000000 +0200
+++ kodi-15.0-beta1-d1a2c33.patch/xbmc/utils/AMLUtils.cpp	2015-05-03 20:37:14.293442201 +0200
@@ -134,9 +134,9 @@
     {
       CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq");
     }
-    if (!SysfsUtils::HasRW("/sys/devices/cpu/cpu/cpu0/cpufreq/scaling_governor"))
+    if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"))
     {
-      CLog::Log(LOGERROR, "AML: no rw on /sys/devices/cpu/cpu/cpu0/cpufreq/scaling_governor");
+      CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor");
     }
 #endif
   }
