diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-29 17:41:15.031724794 +0100
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-29 17:43:27.420981312 +0100
@@ -264,6 +264,7 @@
   { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
+  { KEY_SELECT        , XBMCK_RETURN      },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
