diff -Naur a/project/cmake/scripts/linux/ExtraTargets.cmake b/project/cmake/scripts/linux/ExtraTargets.cmake
--- a/project/cmake/scripts/linux/ExtraTargets.cmake	2016-09-17 08:35:33.000000000 -0700
+++ b/project/cmake/scripts/linux/ExtraTargets.cmake	2016-09-28 00:29:12.861577741 -0700
@@ -5,8 +5,3 @@
   add_executable(${APP_NAME_LC}-xrandr ${CORE_SOURCE_DIR}/xbmc-xrandr.c)
   target_link_libraries(kodi-xrandr ${SYSTEM_LDFLAGS} ${X_LIBRARIES} m ${XRANDR_LIBRARIES})
 endif()
-
-# WiiRemote
-if(ENABLE_EVENTCLIENTS)
-  add_subdirectory(${CORE_SOURCE_DIR}/tools/EventClients/Clients/WiiRemote build/WiiRemote)
-endif()
diff -Naur a/project/cmake/scripts/linux/Install.cmake b/project/cmake/scripts/linux/Install.cmake
--- a/project/cmake/scripts/linux/Install.cmake	2016-11-19 11:55:17.791608457 -0800
+++ b/project/cmake/scripts/linux/Install.cmake	2016-11-19 11:55:44.860880383 -0800
@@ -256,11 +256,6 @@
           DESTINATION ${bindir}
           COMPONENT kodi-eventclients-ps3)
 
-  # Install kodi-eventclients-wiiremote
-  install(PROGRAMS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/WiiRemote/${APP_NAME_LC}-wiiremote
-          DESTINATION ${bindir}
-          COMPONENT kodi-eventclients-wiiremote)
-
   # Install kodi-eventclients-xbmc-send
   install(PROGRAMS "${CORE_SOURCE_DIR}/tools/EventClients/Clients/Kodi Send/kodi-send.py"
           RENAME ${APP_NAME_LC}-send
