diff -Naur kodi-14-b5dbdb5.orig/xbmc/input/linux/LinuxInputDevices.cpp kodi-14-b5dbdb5/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-14-b5dbdb5.orig/xbmc/input/linux/LinuxInputDevices.cpp	2015-02-20 22:00:07.688420179 -0800
+++ kodi-14-b5dbdb5/xbmc/input/linux/LinuxInputDevices.cpp	2015-02-20 22:01:36.495680879 -0800
@@ -255,6 +255,8 @@
   { 378               , XBMCK_RIGHT       }, // Green
   { 381               , XBMCK_UP          }, // Yellow
   { 366               , XBMCK_DOWN        }, // Blue
+  // Rii i7 Home button
+  { 172               , XBMCK_HOME        },
 };
 
 typedef enum
