From ad23989222e80d18343cca113c1d8aa3fd3ac199 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Tue, 17 Jan 2017 21:05:26 +0000
Subject: [PATCH] ADSP: Hack - disable

---
 xbmc/ServiceManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/ServiceManager.cpp b/xbmc/ServiceManager.cpp
index 091c9a3..22b1910 100644
--- a/xbmc/ServiceManager.cpp
+++ b/xbmc/ServiceManager.cpp
@@ -87,7 +87,7 @@ bool CServiceManager::Init2()
 
 bool CServiceManager::Init3()
 {
-  m_ADSPManager->Init();
+  //m_ADSPManager->Init();
   m_PVRManager->Init();
   m_contextMenuManager->Init();
   m_gameServices->Init();
