diff -Naur kodi-15.2-rc3-9efd358.orig/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp kodi-15.2-rc3-9efd358/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
--- kodi-15.2-rc3-9efd358.orig/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2015-10-13 09:55:11.000000000 -0700
+++ kodi-15.2-rc3-9efd358/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2015-10-16 13:39:27.037518528 -0700
@@ -1123,13 +1123,6 @@
 
   m_controlMonitor.Clear();
 
-  /* Always enumerate the default device.
-   * Note: If "default" is a stereo device, EnumerateDevice()
-   * will automatically add "@" instead to enable surroundXX mangling.
-   * We don't want to do that if "default" can handle multichannel
-   * itself (e.g. in case of a pulseaudio server). */
-  EnumerateDevice(list, "default", "", config);
-
   void **hints;
 
   if (snd_device_name_hint(-1, "pcm", &hints) < 0)
@@ -1554,7 +1547,8 @@
   }
 
   snd_pcm_close(pcmhandle);
-  list.push_back(info);
+  if (info.m_deviceName != "pulse")
+    list.push_back(info);
 }
 
 bool CAESinkALSA::GetELD(snd_hctl_t *hctl, int device, CAEDeviceInfo& info, bool& badHDMI)
