diff -Naur a/project/cmake/CMakeLists.txt b/project/cmake/CMakeLists.txt
--- a/project/cmake/CMakeLists.txt	2016-11-30 21:02:11.947113135 -0800
+++ b/project/cmake/CMakeLists.txt	2016-11-30 21:02:41.399396354 -0800
@@ -273,16 +273,6 @@
 copy_files_from_filelist_to_buildtree(${PROJECT_SOURCE_DIR}/installdata/common/*.txt
                                       ${PROJECT_SOURCE_DIR}/installdata/${CORE_SYSTEM_NAME}/*.txt)
 
-list(APPEND SKINS "${CORE_SOURCE_DIR}/addons/skin.estuary\;${CORE_SOURCE_DIR}")
-list(APPEND SKINS "${CORE_SOURCE_DIR}/addons/skin.estouchy\;${CORE_SOURCE_DIR}")
-
-# These are skins that are copied into place from the source tree
-foreach(skin ${SKINS})
-  list(GET skin 0 dir)
-  list(GET skin 1 relative)
-  copy_skin_to_buildtree(${dir} ${relative})
-endforeach()
-
 file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/system/players/VideoPlayer)
 
 set(LIBCEC_SONAME "${CEC_SONAME}")
diff -Naur a/project/cmake/scripts/linux/Install.cmake b/project/cmake/scripts/linux/Install.cmake
--- a/project/cmake/scripts/linux/Install.cmake	2016-11-30 21:02:11.951113174 -0800
+++ b/project/cmake/scripts/linux/Install.cmake	2016-11-30 21:02:50.984488519 -0800
@@ -307,15 +307,6 @@
         DESTINATION ${includedir}/${APP_NAME_LC}
         COMPONENT kodi-peripheral-dev)
 
-# Install XBT skin files
-foreach(texture ${XBT_FILES})
-  string(REPLACE "${CMAKE_BINARY_DIR}/" "" dir ${texture})
-  get_filename_component(dir ${dir} DIRECTORY)
-  install(FILES ${texture}
-          DESTINATION ${datarootdir}/${APP_NAME_LC}/${dir}
-          COMPONENT kodi)
-endforeach()
-
 # Install extra stuff if it exists
 if(EXISTS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/extra-installs)
   install(CODE "file(STRINGS ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/extra-installs dirs)
