--- a/xbmc/windows/GUIWindowSystemInfo.cpp	2013-02-21 22:09:04.765734381 +0400
+++ b/xbmc/windows/GUIWindowSystemInfo.cpp	2013-02-22 16:12:57.942164800 +0400
@@ -100,7 +100,6 @@
 #endif
     SetControlLabel(i++, "%s: %s", 12390, SYSTEM_UPTIME);
     SetControlLabel(i++, "%s: %s", 12394, SYSTEM_TOTALUPTIME);
-    SetControlLabel(i++, "%s: %s", 12395, SYSTEM_BATTERY_LEVEL);
   }
   else if (m_section == CONTROL_BT_STORAGE)
   {
