diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2016-02-25 15:54:54.305799646 +0000
+++ b/Makefile.in	2016-02-25 15:55:30.609990228 +0000
@@ -11,7 +11,6 @@
 
 DVDPCODECS_DIRS= \
 	lib \
-	lib/libdvd
 
 VideoPlayer_ARCHIVES=xbmc/cores/VideoPlayer/VideoPlayer.a \
                    xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecs.a \
@@ -332,7 +331,6 @@
 	$(MAKE) -C lib/addons/library.kodi.peripheral
 	$(MAKE) -C lib/addons/library.xbmc.pvr
 dvdpcodecs: dllloader
-	$(MAKE) -C lib/libdvd
 
 dvdpextcodecs:
 
