diff -Naur kodi-15.2-02e7013/system/keymaps/keyboard.xml kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml
--- kodi-15.2-02e7013/system/keymaps/keyboard.xml	2015-10-22 12:34:25.400782034 +0200
+++ kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml	2015-10-22 12:41:46.849456385 +0200
@@ -273,6 +273,7 @@
       <escape>Fullscreen</escape>
       <c>Playlist</c>
       <v>ActivateWindow(Teletext)</v>
+      <text>ActivateWindow(Teletext)</text>
       <up mod="ctrl">SubtitleShiftUp</up>
       <down mod="ctrl">SubtitleShiftDown</down>
       <pageup>SkipNext</pageup>
@@ -318,6 +319,7 @@
       <i>Info</i>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Addon.Default.OpenSettings(xbmc.player.musicviz)</v>
+      <text>Addon.Default.OpenSettings(xbmc.player.musicviz)</text>
       <n>ActivateWindow(MusicPlaylist)</n>
       <left>StepBack</left>
       <right>StepForward</right>
@@ -342,6 +344,7 @@
       <o>CodecInfo</o>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Addon.Default.OpenSettings(xbmc.player.musicviz)</v>
+      <text>Addon.Default.OpenSettings(xbmc.player.musicviz)</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </MusicOSD>
@@ -356,6 +359,7 @@
       <o>CodecInfo</o>
       <p>ActivateWindow(VisualisationPresetList)</p>
       <v>Back</v>
+      <text>Back</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </VisualisationSettings>
@@ -370,6 +374,7 @@
       <o>CodecInfo</o>
       <p>Back</p>
       <v>Back</v>
+      <text>Back</text>
       <n>ActivateWindow(MusicPlaylist)</n>
     </keyboard>
   </VisualisationPresetList>
@@ -519,6 +524,7 @@
   <Teletext>
     <keyboard>
       <v>Back</v>
+      <text>Back</text>
     </keyboard>
   </Teletext>
   <Favourites>
diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-22 12:34:25.083781548 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-22 12:38:11.522120520 +0200
@@ -254,6 +254,7 @@
   { KEY_QUESTION      , XBMCK_HELP        },
   { KEY_BACK          , XBMCK_BACKSPACE   },
   { KEY_ZOOM          , XBMCK_ZOOM        },
+  { KEY_TEXT          , XBMCK_TEXT        },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h	2015-10-22 12:34:25.082781547 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h	2015-10-22 12:37:47.986085419 +0200
@@ -225,6 +225,7 @@
   XBMCK_YELLOW      = 0x149,
   XBMCK_BLUE        = 0x14a,
   XBMCK_ZOOM        = 0x14b,
+  XBMCK_TEXT        = 0x14c,
 
   // Add any other keys here
 
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp	2015-10-22 12:34:25.085781551 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp	2015-10-22 12:38:35.907157920 +0200
@@ -239,6 +239,7 @@
 , { XBMCK_YELLOW,                 0,    0, XBMCVK_YELLOW,        "yellow" }
 , { XBMCK_BLUE,                   0,    0, XBMCVK_BLUE,          "blue" }
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
+, { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h	2015-10-22 12:34:25.083781548 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h	2015-10-22 12:39:09.033208840 +0200
@@ -217,6 +217,7 @@
   XBMCVK_YELLOW         = 0xE5,
   XBMCVK_BLUE           = 0xE6,
   XBMCVK_ZOOM           = 0xE7,
+  XBMCVK_TEXT           = 0xE8,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
