From b1a9fcaf87add539888bdb592d42262e770eda46 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Sat, 6 May 2017 04:25:16 +0100
Subject: [PATCH] libmicrohttpd: use MHD_USE_INTERNAL_POLLING_THREAD with
 version >= 0.9.54

---
 xbmc/network/WebServer.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xbmc/network/WebServer.cpp b/xbmc/network/WebServer.cpp
index 6e70d3e..c35f009 100644
--- a/xbmc/network/WebServer.cpp
+++ b/xbmc/network/WebServer.cpp
@@ -1130,6 +1130,9 @@ struct MHD_Daemon* CWebServer::StartMHD(unsigned int flags, int port)
                           // WARNING: set MHD_OPTION_CONNECTION_TIMEOUT to something higher than 1
                           // otherwise on libmicrohttpd 0.4.4-1 it spins a busy loop
                           MHD_USE_THREAD_PER_CONNECTION
+#if (MHD_VERSION >= 0x00095400)
+                          | MHD_USE_INTERNAL_POLLING_THREAD /* MHD_USE_THREAD_PER_CONNECTION must be used only with MHD_USE_INTERNAL_POLLING_THREAD since 0.9.54 */
+#endif
 #endif
 #if (MHD_VERSION >= 0x00040001)
                           | MHD_USE_DEBUG /* Print MHD error messages to log */
-- 
2.7.4

