diff -Naur kodi-14-af6d342/configure.in kodi-14-af6d342.patch/configure.in
--- kodi-14-af6d342/configure.in	2014-10-27 11:34:57.000000000 +0100
+++ kodi-14-af6d342.patch/configure.in	2014-10-31 02:34:19.322700780 +0100
@@ -1230,7 +1230,6 @@
         [AC_MSG_ERROR($missing_library)])
       ]
     )
-    AC_CHECK_LIB([SDL_image],  [main],, AC_MSG_ERROR($missing_library))
     AC_DEFINE([HAVE_SDL],[1],["Define to 1 if using sdl"])
   fi
 fi
