commit 81214c86f437863c0a8f4ac2e45de81fccc621b4
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Mon Dec 11 23:44:41 2017 -0800

    [linux] allow using alsa and pulse together

diff --git a/xbmc/windowing/X11/WinSystemX11GLContext.cpp b/xbmc/windowing/X11/WinSystemX11GLContext.cpp
index 398a311..6b3e6d8 100644
--- a/xbmc/windowing/X11/WinSystemX11GLContext.cpp
+++ b/xbmc/windowing/X11/WinSystemX11GLContext.cpp
@@ -52,31 +52,8 @@ std::unique_ptr<CWinSystemBase> CWinSystemBase::CreateWinSystem()
 
 CWinSystemX11GLContext::CWinSystemX11GLContext()
 {
-  std::string envSink;
-  if (getenv("AE_SINK"))
-    envSink = getenv("AE_SINK");
-  if (StringUtils::EqualsNoCase(envSink, "ALSA"))
-  {
-    X11::ALSARegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "PULSE"))
-  {
-    X11::PulseAudioRegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "SNDIO"))
-  {
-    X11::SndioRegister();
-  }
-  else
-  {
-    if (!X11::PulseAudioRegister())
-    {
-      if (!X11::ALSARegister())
-      {
-        X11::SndioRegister();
-      }
-    }
-  }
+  X11::ALSARegister();
+  X11::PulseAudioRegister();
 }
 
 CWinSystemX11GLContext::~CWinSystemX11GLContext()
diff --git a/xbmc/windowing/amlogic/WinSystemAmlogic.cpp b/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
index 020e501..ed0f084 100644
--- a/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
+++ b/xbmc/windowing/amlogic/WinSystemAmlogic.cpp
@@ -32,6 +32,7 @@
 // AESink Factory
 #include "cores/AudioEngine/AESinkFactory.h"
 #include "cores/AudioEngine/Sinks/AESinkALSA.h"
+#include "cores/AudioEngine/Sinks/AESinkPULSE.h"
 #include "guilib/GraphicContext.h"
 #include "guilib/Resolution.h"
 #include "powermanagement/linux/LinuxPowerSyscall.h"
@@ -79,6 +80,7 @@ CWinSystemAmlogic::CWinSystemAmlogic()
   // Register sink
   AE::CAESinkFactory::ClearSinks();
   CAESinkALSA::Register();
+  CAESinkPULSE::Register();
   CLinuxPowerSyscall::Register();
 }
 
diff --git a/xbmc/windowing/gbm/WinSystemGbm.cpp b/xbmc/windowing/gbm/WinSystemGbm.cpp
index 854040f..7ea9e75 100644
--- a/xbmc/windowing/gbm/WinSystemGbm.cpp
+++ b/xbmc/windowing/gbm/WinSystemGbm.cpp
@@ -38,31 +38,8 @@ CWinSystemGbm::CWinSystemGbm() :
   m_nativeDisplay(nullptr),
   m_nativeWindow(nullptr)
 {
-  std::string envSink;
-  if (getenv("AE_SINK"))
-    envSink = getenv("AE_SINK");
-  if (StringUtils::EqualsNoCase(envSink, "ALSA"))
-  {
-    GBM::ALSARegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "PULSE"))
-  {
-    GBM::PulseAudioRegister();
-  }
-  else if (StringUtils::EqualsNoCase(envSink, "SNDIO"))
-  {
-    GBM::SndioRegister();
-  }
-  else
-  {
-    if (!GBM::PulseAudioRegister())
-    {
-      if (!GBM::ALSARegister())
-      {
-        GBM::SndioRegister();
-      }
-    }
-  }
+  GBM::ALSARegister();
+  GBM::PulseAudioRegister();
 
   m_winEvents.reset(new CWinEventsLinux());
   CLinuxPowerSyscall::Register();
diff --git a/xbmc/windowing/rpi/WinSystemRpi.cpp b/xbmc/windowing/rpi/WinSystemRpi.cpp
index b1caf83..da1b59f 100644
--- a/xbmc/windowing/rpi/WinSystemRpi.cpp
+++ b/xbmc/windowing/rpi/WinSystemRpi.cpp
@@ -34,7 +34,9 @@
 #include "utils/log.h"
 #include "../WinEventsLinux.h"
 #include "cores/AudioEngine/AESinkFactory.h"
+#include "cores/AudioEngine/Sinks/AESinkALSA.h"
 #include "cores/AudioEngine/Sinks/AESinkPi.h"
+#include "cores/AudioEngine/Sinks/AESinkPULSE.h"
 #include "powermanagement/linux/LinuxPowerSyscall.h"
 
 #include <EGL/egl.h>
@@ -56,6 +58,8 @@ CWinSystemRpi::CWinSystemRpi()
   m_winEvents.reset(new CWinEventsLinux());
   AE::CAESinkFactory::ClearSinks();
   CAESinkPi::Register();
+  CAESinkALSA::Register();
+  CAESinkPULSE::Register();
   CLinuxPowerSyscall::Register();
 }
 
