diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-02-14 00:19:36.000000000 +0100
+++ b/configure.ac	2016-03-02 09:15:52.504860258 +0100
@@ -641,6 +641,18 @@
      use_gl=no
      USE_STATIC_FFMPEG=1
      ;;
+  aarch64*-*-linux-gnu*)
+     target_platform=target_linux
+     ARCH="aarch64"
+     use_arch="aarch64"
+     use_cpu=cortex-a53
+     use_joystick=no
+     use_neon=yes
+     use_gles=yes
+     use_gl=no
+     use_wayland=no
+     USE_STATIC_FFMPEG=1
+     ;;
   arm*-*linux-android*)
      target_platform=target_android
      use_arch="arm"
diff -Naur a/m4/xbmc_arch.m4 b/m4/xbmc_arch.m4
--- a/m4/xbmc_arch.m4	2016-02-14 00:19:42.000000000 +0100
+++ b/m4/xbmc_arch.m4	2016-03-02 09:16:33.341943374 +0100
@@ -60,7 +60,7 @@
   powerpc64-*-linux-gnu*|powerpc64-*-linux-uclibc*)
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_POWERPC64")
      ;;
-  arm*-*-linux-gnu*|arm*-*-linux-uclibc*)
+  arm*-*-linux-gnu*|arm*-*-linux-uclibc*|aarch64*-*-linux-gnu*)
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
      ;;
   mips*-*-linux-gnu*|mips*-*-linux-uclibc*)
