--- ./xbmc/platform/posix/filesystem/SMBDirectory.cpp.orig	2023-02-03 14:31:42.518877830 +0100
+++ ./xbmc/platform/posix/filesystem/SMBDirectory.cpp	2023-02-03 14:36:27.357905849 +0100
@@ -171,7 +171,7 @@
             char value[20];
             // We poll for extended attributes which symbolizes bits but split up into a string. Where 0x02 is hidden and 0x12 is hidden directory.
             // According to the libsmbclient.h it's supposed to return 0 if ok, or the length of the string. It seems always to return the length which is 4
-            if (smbc_getxattr(strFullName.c_str(), "system.dos_attr.mode", value, sizeof(value)) > 0)
+            if (smbc_getxattr(strFullName.c_str(), "system.dos_attr.mode", value, sizeof(value)) >= 0)
             {
               long longvalue = strtol(value, NULL, 16);
               if (longvalue & SMBC_DOS_MODE_HIDDEN)
