From 68428562758adbcb597cadc05ba124a6bde97291 Mon Sep 17 00:00:00 2001
From: Rainer Hochecker <fernetmenta@online.de>
Date: Fri, 13 Mar 2015 20:57:17 +0100
Subject: [PATCH] guilib: only update scrollinfo if frametime did change

---
 xbmc/guilib/GUIFont.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xbmc/guilib/GUIFont.cpp b/xbmc/guilib/GUIFont.cpp
index a7ee668..06232cb 100644
--- a/xbmc/guilib/GUIFont.cpp
+++ b/xbmc/guilib/GUIFont.cpp
@@ -52,7 +52,8 @@ float CScrollInfo::GetPixelsPerFrame()
     delta = 100; // assume a minimum of 10 fps
   m_lastFrameTime = currentTime;
   // do an exponential moving average of the frame time
-  m_averageFrameTime = m_averageFrameTime + (delta - m_averageFrameTime) * alphaEMA;
+  if (delta)
+    m_averageFrameTime = m_averageFrameTime + (delta - m_averageFrameTime) * alphaEMA;
   // and multiply by pixel speed (per ms) to get number of pixels to move this frame
   return pixelSpeed * m_averageFrameTime;
 }
-- 
1.9.1

