diff -Naur kodi-15.2-02e7013/system/keymaps/keyboard.xml kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml
--- kodi-15.2-02e7013/system/keymaps/keyboard.xml	2015-10-24 10:47:31.423447515 +0200
+++ kodi-15.2-02e7013.patch/system/keymaps/keyboard.xml	2015-10-24 10:51:32.346473050 +0200
@@ -118,6 +118,7 @@
       <browser_favorites>ActivateWindow(Favourites)</browser_favorites>
       <favorites>ActivateWindow(Favourites)</favorites>
       <browser_home>ActivateWindow(Home)</browser_home>
+      <homepage>ActivateWindow(Home)</homepage>
       <volume_mute>Mute</volume_mute>
       <volume_down>VolumeDown</volume_down>
       <volume_up>VolumeUp</volume_up>
diff -Naur kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-15.2-02e7013/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-24 10:47:31.179446989 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/linux/LinuxInputDevices.cpp	2015-10-24 10:49:34.162165486 +0200
@@ -260,13 +260,12 @@
   { KEY_GREEN         , XBMCK_GREEN       },
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
+  { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
   { 381               , XBMCK_UP          }, // Yellow
   { 366               , XBMCK_DOWN        }, // Blue
-  // Rii i7 Home button / wetek openelec remote (code 172)
-  { KEY_HOMEPAGE      , XBMCK_HOME        },
 };
 
 typedef enum
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_keysym.h	2015-10-24 10:47:31.179446989 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keysym.h	2015-10-24 10:48:22.081558452 +0200
@@ -227,6 +227,7 @@
   XBMCK_ZOOM        = 0x14b,
   XBMCK_TEXT        = 0x14c,
   XBMCK_FAVORITES   = 0x14d,
+  XBMCK_HOMEPAGE    = 0x14e,
 
   // Add any other keys here
 
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-15.2-02e7013/xbmc/input/XBMC_keytable.cpp	2015-10-24 10:47:31.181446993 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_keytable.cpp	2015-10-24 10:48:59.169880939 +0200
@@ -241,6 +241,7 @@
 , { XBMCK_ZOOM,                   0,    0, XBMCVK_ZOOM,          "zoom" }
 , { XBMCK_TEXT,                   0,    0, XBMCVK_TEXT,          "text" }
 , { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
+, { XBMCK_HOMEPAGE ,              0,    0, XBMCVK_HOMEPAGE,      "homepage" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h
--- kodi-15.2-02e7013/xbmc/input/XBMC_vkeys.h	2015-10-24 10:47:31.180446991 +0200
+++ kodi-15.2-02e7013.patch/xbmc/input/XBMC_vkeys.h	2015-10-24 10:50:25.548276428 +0200
@@ -219,6 +219,7 @@
   XBMCVK_ZOOM           = 0xE7,
   XBMCVK_TEXT           = 0xE8,
   XBMCVK_FAVORITES      = 0xE9,
+  XBMCVK_HOMEPAGE       = 0xEA,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
