diff -Naur a/project/cmake/modules/FindPython.cmake b/project/cmake/modules/FindPython.cmake
--- a/project/cmake/modules/FindPython.cmake	2016-11-30 13:39:56.000000000 -0800
+++ b/project/cmake/modules/FindPython.cmake	2016-11-30 20:55:52.581465246 -0800
@@ -16,8 +16,6 @@
 if(CMAKE_CROSSCOMPILING)
   find_library(FFI_LIBRARY ffi REQUIRED)
   find_library(EXPAT_LIBRARY expat REQUIRED)
-  find_library(INTL_LIBRARY intl REQUIRED)
-  find_library(GMP_LIBRARY gmp REQUIRED)
 
   if(NOT CORE_SYSTEM_NAME STREQUAL android)
     set(PYTHON_DEP_LIBRARIES -lpthread -ldl -lutil)
