diff -Naur kodi-16.0-beta5-19fc4fa/system/keymaps/keyboard.xml kodi-16.0-beta5-19fc4fa.patch/system/keymaps/keyboard.xml
--- kodi-16.0-beta5-19fc4fa/system/keymaps/keyboard.xml	2015-12-15 00:07:12.000000000 +0100
+++ kodi-16.0-beta5-19fc4fa.patch/system/keymaps/keyboard.xml	2016-01-05 06:41:50.857902588 +0100
@@ -113,6 +113,7 @@
       <sleep>ActivateWindow(shutdownmenu)</sleep>
       <!-- PVR windows -->
       <e>ActivateWindow(TVGuide)</e>
+      <epg>ActivateWindow(TVGuide)</epg>
       <h>ActivateWindow(TVChannels)</h>
       <j>ActivateWindow(RadioChannels)</j>
       <k>ActivateWindow(TVRecordings)</k>
@@ -268,6 +269,7 @@
   <TVGuide>
     <keyboard>
       <e>PreviousMenu</e>
+      <epg>PreviousMenu</epg>
     </keyboard>
   </TVGuide>
   <MyFiles>
diff -Naur kodi-16.0-beta5-19fc4fa/xbmc/input/linux/LinuxInputDevices.cpp kodi-16.0-beta5-19fc4fa.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-16.0-beta5-19fc4fa/xbmc/input/linux/LinuxInputDevices.cpp	2015-12-15 00:07:12.000000000 +0100
+++ kodi-16.0-beta5-19fc4fa.patch/xbmc/input/linux/LinuxInputDevices.cpp	2016-01-05 06:41:13.847846527 +0100
@@ -268,6 +268,7 @@
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
   { KEY_SELECT        , XBMCK_RETURN      },
   { KEY_CONFIG        , XBMCK_CONFIG      },
+  { KEY_EPG           , XBMCK_EPG         },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
diff -Naur kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_keysym.h kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_keysym.h
--- kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_keysym.h	2015-12-15 00:07:12.000000000 +0100
+++ kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_keysym.h	2016-01-05 06:41:13.847846527 +0100
@@ -229,6 +229,7 @@
   XBMCK_FAVORITES   = 0x14d,
   XBMCK_HOMEPAGE    = 0x14e,
   XBMCK_CONFIG      = 0x14f,
+  XBMCK_EPG         = 0x150,
 
   // Add any other keys here
 
diff -Naur kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_keytable.cpp kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_keytable.cpp	2015-12-15 00:07:12.000000000 +0100
+++ kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_keytable.cpp	2016-01-05 06:41:13.848846528 +0100
@@ -243,6 +243,7 @@
 , { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
 , { XBMCK_HOMEPAGE ,              0,    0, XBMCVK_HOMEPAGE,      "homepage" }
 , { XBMCK_CONFIG,                 0,    0, XBMCVK_CONFIG,        "config" }
+, { XBMCK_EPG   ,                 0,    0, XBMCVK_EPG,           "epg" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_vkeys.h kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_vkeys.h
--- kodi-16.0-beta5-19fc4fa/xbmc/input/XBMC_vkeys.h	2015-12-15 00:07:12.000000000 +0100
+++ kodi-16.0-beta5-19fc4fa.patch/xbmc/input/XBMC_vkeys.h	2016-01-05 06:41:13.848846528 +0100
@@ -221,6 +221,7 @@
   XBMCVK_FAVORITES      = 0xE9,
   XBMCVK_HOMEPAGE       = 0xEA,
   XBMCVK_CONFIG         = 0xEB,
+  XBMCVK_EPG            = 0xEC,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
