commit 0e78af21ce3fff4c9f4424e1ce99880a79ef4113
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sun Sep 25 15:26:18 2016 -0700

    [cmake][IMX] fix imx cmake build

diff --git a/project/cmake/CMakeLists.txt b/project/cmake/CMakeLists.txt
index 76870a4..8602a52 100644
--- a/project/cmake/CMakeLists.txt
+++ b/project/cmake/CMakeLists.txt
@@ -56,6 +56,7 @@ if(UNIX)
   if(CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd)
     option(ENABLE_X11         "Enable X11 support?" ON)
     option(ENABLE_AML         "Enable AML?" OFF)
+    option(ENABLE_IMX         "Enable IMX?" OFF)
   endif()
 endif()
 # System options
@@ -216,6 +217,7 @@ endif()
 
 if(CORE_SYSTEM_NAME STREQUAL android OR CORE_SYSTEM_NAME STREQUAL linux)
   core_optional_dep(AML)
+  core_optional_dep(IMX)
 elseif(CORE_SYSTEM_NAME STREQUAL rbpi)
   core_require_dep(MMAL)
 endif()
diff --git a/project/cmake/modules/FindIMX.cmake b/project/cmake/modules/FindIMX.cmake
new file mode 100644
index 0000000..4e1e838
--- /dev/null
+++ b/project/cmake/modules/FindIMX.cmake
@@ -0,0 +1,35 @@
+#.rst:
+# FindIMX
+# -------
+# Finds the IMX codec
+#
+# This will will define the following variables::
+#
+# IMX_FOUND - system has IMX
+# IMX_INCLUDE_DIRS - the IMX include directory
+# IMX_DEFINITIONS - the IMX definitions
+# IMX_LIBRARIES - the IMX libraries
+
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(IMX fslvpuwrap QUIET)
+endif()
+
+if(NOT IMX_FOUND)
+  find_path(IMX_INCLUDE_DIR vpu_wrapper.h
+                            PATH_SUFFIXES imx-mm/vpu)
+
+  find_library(FSLVPUWRAP_LIBRARY fslvpuwrap)
+  find_library(VPU_LIBRARY vpu)
+  find_library(G2D_LIBRARY g2d)
+
+  set(IMX_LIBRARIES ${FSLVPUWRAP_LIBRARY} ${VPU_LIBRARY} ${G2D_LIBRARY}
+      CACHE STRING "imx libraries" FORCE)
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(IMX
+                                  REQUIRED_VARS IMX_INCLUDE_DIR IMX_LIBRARIES)
+
+list(APPEND IMXVPU_DEFINITIONS -DHAS_IMXVPU=1 -DLINUX -DEGL_API_FB)
+
+mark_as_advanced(IMX_INCLUDE_DIR IMX_LIBRARIES IMX_DEFINITIONS)
diff --git a/xbmc/linux/CMakeLists.txt b/xbmc/linux/CMakeLists.txt
index 754cc1e..931c10d 100644
--- a/xbmc/linux/CMakeLists.txt
+++ b/xbmc/linux/CMakeLists.txt
@@ -40,4 +40,8 @@ if(OMXLIB_FOUND)
                       OMXCore.h)
 endif()
 
+if(IMX_FOUND)
+  add_subdirectory(imx)
+endif()
+
 core_add_library(linuxsupport)
diff --git a/xbmc/video/videosync/CMakeLists.txt b/xbmc/video/videosync/CMakeLists.txt
index 421af44..82f7169 100644
--- a/xbmc/video/videosync/CMakeLists.txt
+++ b/xbmc/video/videosync/CMakeLists.txt
@@ -35,6 +35,11 @@ if(CORE_SYSTEM_NAME STREQUAL android)
   list(APPEND HEADERS VideoSyncAndroid.h)
 endif()
 
+if(IMX_FOUND)
+  list(APPEND SOURCES VideoSyncIMX.cpp)
+  list(APPEND HEADERS VideoSyncIMX.h)
+endif()
+
 if(SOURCES AND HEADERS)
   core_add_library(video_sync)
 endif()
diff --git a/xbmc/windowing/egl/CMakeLists.txt b/xbmc/windowing/egl/CMakeLists.txt
index 3ab603e..51e13ab 100644
--- a/xbmc/windowing/egl/CMakeLists.txt
+++ b/xbmc/windowing/egl/CMakeLists.txt
@@ -1,4 +1,4 @@
-if(OPENGLES_FOUND OR AML_FOUND OR MMAL_FOUND)
+if(OPENGLES_FOUND OR AML_FOUND OR IMX_FOUND OR MMAL_FOUND)
   set(SOURCES EGLWrapper.cpp)
 
   set(HEADERS EGLNativeType.h
@@ -30,6 +30,11 @@ if(MMAL_FOUND)
   list(APPEND HEADERS EGLNativeTypeRaspberryPI.h)
 endif()
 
+if(IMX_FOUND)
+  list(APPEND SOURCES EGLNativeTypeIMX.cpp)
+  list(APPEND HEADERS EGLNativeTypeIMX.h)
+endif()
+
 if(SOURCES)
   core_add_library(windowing_egl)
 endif()
