diff --git a/Makefile b/Makefile
index 6ab27ef..17d12a0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 
 ply-image: ply-image.c ply-frame-buffer.c Makefile
-	$(CC) $(CFLAGS) `pkg-config --cflags libpng`  ply-image.c ply-frame-buffer.c -o ply-image  -lm `pkg-config --libs libpng` -lm -lz
+	$(CC) $(CFLAGS) -Wl,-Bstatic `pkg-config --cflags libpng`  ply-image.c ply-frame-buffer.c -o ply-image  -lm `pkg-config --libs libpng` -lm -lz -Wl,-Bdynamic -lc $(LDFLAGS)
 	
 clean:
 	rm -f ply-image *~ gmon.out
@@ -11,4 +11,3 @@ install: ply-image
 	cp ply-image $(DESTDIR)/usr/bin
 	cp splash.png $(DESTDIR)/usr/share/plymouth/splash.png
 	
-	
\ No newline at end of file
