diff -Naur brcmfmac_sdio-firmware-aml-0.1.orig/Makefile brcmfmac_sdio-firmware-aml-0.1/Makefile
--- brcmfmac_sdio-firmware-aml-0.1.orig/Makefile	2014-10-07 23:00:34.000000000 +0200
+++ brcmfmac_sdio-firmware-aml-0.1/Makefile	2017-11-09 11:52:47.102781882 +0100
@@ -15,8 +15,8 @@
 install:
 	mkdir -p $(DESTDIR)/usr/bin
 	cp -P $(PROG) $(DESTDIR)/usr/bin
-	mkdir -p $(DESTDIR)/lib/firmware
-	cp -PR firmware/brcm $(DESTDIR)/lib/firmware/
+	mkdir -p $(FWDIR)/lib/firmware
+	cp -PR firmware/brcm $(FWDIR)/lib/firmware/
 
 clean:
 	rm -f $(PROG)
