From 859ac87a4473dd24c6411fa2a0f77393528c904d Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Mon, 5 Jan 2026 11:29:21 +0000
Subject: [PATCH] Makefile: allow CC and CXX to be overwritten

---
 Makefile | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index f21b6be..38b9900 100644
--- a/Makefile
+++ b/Makefile
@@ -8,8 +8,10 @@
 #/******************************************************************************
 # * if you prefer verbose non-coloured build messages, remove the '@' here:
 # *****************************************************************************/
-CC  = @gcc
-CXX = @g++
+Q = @
+
+CC  ?= gcc
+CXX ?= g++
 
           
 PWD = $(shell pwd)
@@ -91,16 +93,16 @@ GN=\e[1;32m
 # *****************************************************************************/
 
 %.o: %.c
-ifeq ($(CXX),@g++)
+ifeq ($(Q),@)
 	@echo -e "${CY} CXX $@${RST}"
 endif
-	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
+	$(Q)$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
 
 %.o: %.cpp
-ifeq ($(CXX),@g++)
+ifeq ($(Q),@)
 	@echo -e "${BL} CXX $@${RST}"
 endif
-	$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
+	$(Q)$(CXX) $(CXXFLAGS) -c $(DEFINES) $(INCLUDES) -o $@ $<
 
 
 
@@ -141,10 +143,10 @@ install-i18n: $(I18Nmsgs)
 ### Targets:
 
 $(SOFILE): $(OBJS)
-ifeq ($(CXX),@g++)
+ifeq ($(Q),@)
 	@echo -e "${GN} LINK $(SOFILE)${RST}"
 endif
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LDFLAGS)
+	$(Q)$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LDFLAGS)
 
 install-lib: $(SOFILE)
 	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
-- 
2.51.0

