diff -Naur tsdecrypt-10.0/libfuncs/Makefile tsdecrypt-10.0.patch/libfuncs/Makefile
--- tsdecrypt-10.0/libfuncs/Makefile	2013-09-12 12:53:27.000000000 +0200
+++ tsdecrypt-10.0.patch/libfuncs/Makefile	2014-07-20 12:23:54.900762191 +0200
@@ -1,5 +1,4 @@
-CC = $(CROSS)$(TARGET)gcc
-LINK = $(CROSS)$(TARGET)ld -o
+LINK = $(LD) -o
 MKDEP = $(CC) -M -o $*.d $<
 
 LIBRARY_LINK_OPTS =  -L. -r
diff -Naur tsdecrypt-10.0/libtsfuncs/Makefile tsdecrypt-10.0.patch/libtsfuncs/Makefile
--- tsdecrypt-10.0/libtsfuncs/Makefile	2013-09-12 12:53:30.000000000 +0200
+++ tsdecrypt-10.0.patch/libtsfuncs/Makefile	2014-07-20 12:23:54.901762318 +0200
@@ -1,5 +1,4 @@
-CC = $(CROSS)$(TARGET)gcc
-LINK = $(CROSS)$(TARGET)ld -o
+LINK = $(LD) -o
 MKDEP = $(CC) -M -o $*.d $<
 
 LIBRARY_LINK_OPTS =  -L. -r
diff -Naur tsdecrypt-10.0/Makefile tsdecrypt-10.0.patch/Makefile
--- tsdecrypt-10.0/Makefile	2013-09-12 12:53:16.000000000 +0200
+++ tsdecrypt-10.0.patch/Makefile	2014-07-20 12:24:11.303809770 +0200
@@ -1,5 +1,3 @@
-CC = $(CROSS)$(TARGET)cc
-STRIP = $(CROSS)$(TARGET)strip
 MKDEP = $(CC) -MP -MM -o $*.d $<
 RM = rm -f
 MV = mv -f
