From 70e4aa204cf7809c91b782fa59646dfa11bad53f Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 4 May 2024 11:33:54 +0000
Subject: [PATCH] fix build with gcc-14

---
 camd-newcamd.c | 1 +
 filter.c       | 1 +
 process.c      | 1 +
 3 files changed, 3 insertions(+)

diff --git a/camd-newcamd.c b/camd-newcamd.c
index 48ad567..0d87735 100644
--- a/camd-newcamd.c
+++ b/camd-newcamd.c
@@ -18,6 +18,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <crypt.h>
 
 #include "libfuncs/libfuncs.h"
 
diff --git a/filter.c b/filter.c
index 2cac2dc..ea440e9 100644
--- a/filter.c
+++ b/filter.c
@@ -14,6 +14,7 @@
  */
 #include <ctype.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "data.h"
 #include "filter.h"
diff --git a/process.c b/process.c
index ab3be80..d3e7fb3 100644
--- a/process.c
+++ b/process.c
@@ -14,6 +14,7 @@
  */
 #include <unistd.h>
 #include <string.h>
+#include <stdlib.h>
 #include <sys/uio.h>
 
 #include "bitstream.h"
