--- a/sysdeps/unix/sysv/linux/arm/kernel-features.h	2019-01-01 17:59:37.442898104 +0100
+++ b/sysdeps/unix/sysv/linux/arm/kernel-features.h	2019-01-01 17:59:27.951409000 +0100
@@ -22,10 +22,12 @@
 /* The ARM kernel before 3.14.3 may or may not support
    futex_atomic_cmpxchg_inatomic, depending on kernel
    configuration.  */
+/*
 #if __LINUX_KERNEL_VERSION < 0x030E03
 # undef __ASSUME_REQUEUE_PI
 # undef __ASSUME_SET_ROBUST_LIST
 #endif
+*/
 
 /* Define this if your 32-bit syscall API requires 64-bit register
    pairs to start with an even-number register.  */
