diff -Naur xf86-video-vesa-2.2.0/configure xf86-video-vesa-2.2.0.patch/configure
--- xf86-video-vesa-2.2.0/configure	2009-02-18 00:03:57.000000000 +0100
+++ xf86-video-vesa-2.2.0.patch/configure	2009-02-19 08:50:04.000000000 +0100
@@ -12537,7 +12537,7 @@
 
 
 CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
-INCLUDES="$XORG_INCS -I${sdkdir} "'-I$(top_srcdir)/src -I$(prefix)/include'
+INCLUDES="$XORG_INCS -I${sdkdir} "'-I$(top_srcdir)/src'
 
 
 
diff -Naur xf86-video-vesa-2.2.0/configure.ac xf86-video-vesa-2.2.0.patch/configure.ac
--- xf86-video-vesa-2.2.0/configure.ac	2009-02-18 00:03:35.000000000 +0100
+++ xf86-video-vesa-2.2.0.patch/configure.ac	2009-02-19 08:49:53.000000000 +0100
@@ -77,7 +77,7 @@
 AM_CONDITIONAL(XSERVER_LIBPCIACCESS, test "x$XSERVER_LIBPCIACCESS" = xyes)
 
 CFLAGS="$CFLAGS $XORG_CFLAGS "' -I$(top_srcdir)/src'
-INCLUDES="$XORG_INCS -I${sdkdir} "'-I$(top_srcdir)/src -I$(prefix)/include'
+INCLUDES="$XORG_INCS -I${sdkdir} "'-I$(top_srcdir)/src'
 AC_SUBST([CFLAGS])
 AC_SUBST([INCLUDES])
 
