diff -Naur configobj-4.7.2/setup.py configobj-4.7.2.patch/setup.py
--- configobj-4.7.2/setup.py	2010-03-02 00:18:36.000000000 +0100
+++ configobj-4.7.2.patch/setup.py	2011-02-19 13:56:40.861322489 +0100
@@ -9,7 +9,7 @@
 # http://www.voidspace.org.uk/python/license.shtml
 
 import sys
-from distutils.core import setup
+from setuptools import setup
 from configobj import __version__ as VERSION
 
 NAME = 'configobj'
