diff -Naur audiodecoder.ncsf-6cbec60.orig/lib/SSEQPlayer/convert.h audiodecoder.ncsf-6cbec60/lib/SSEQPlayer/convert.h
--- audiodecoder.ncsf-6cbec60.orig/lib/SSEQPlayer/convert.h	2015-05-09 09:55:22.000000000 -0700
+++ audiodecoder.ncsf-6cbec60/lib/SSEQPlayer/convert.h	2015-10-23 00:23:22.442374128 -0700
@@ -11,12 +11,7 @@
 #include <sstream>
 #include <typeinfo>
 #include <locale>
-#if (defined(__GNUC__) || defined(__clang__)) && !defined(_LIBCPP_VERSION)
-# include "wstring_convert.h"
-# include "codecvt.h"
-#else
-# include <codecvt>
-#endif
+#include <codecvt>
 #include <vector>
 #include <cmath>
 
