From 2df5d26162b580a21489dd3e3157a9984c4c20e3 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Mon, 22 Jun 2020 19:06:04 +0200
Subject: [PATCH] fix build with gcc 10

---
 lib/SSEQPlayer/common.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/SSEQPlayer/common.h b/lib/SSEQPlayer/common.h
index 747000615dea..a447785668d0 100644
--- a/lib/SSEQPlayer/common.h
+++ b/lib/SSEQPlayer/common.h
@@ -14,6 +14,7 @@
 #include <vector>
 #include <cstring>
 #include <cstdint>
+#include <stdexcept>
 
 /*
  * Pseudo-file data structure
-- 
2.27.0

