diff --git a/mpp/hal/rkdec/h265d/hal_h265d_reg.h b/mpp/hal/rkdec/h265d/hal_h265d_reg.h
index 1bccb02..432b8db 100644
--- a/mpp/hal/rkdec/h265d/hal_h265d_reg.h
+++ b/mpp/hal/rkdec/h265d/hal_h265d_reg.h
@@ -50,7 +50,8 @@ typedef struct {
     struct swreg_int {
         RK_U32    sw_dec_e            : 1  ;
         RK_U32    sw_dec_clkgate_e    : 1  ;
-        RK_U32    reserve0            : 2  ;
+        RK_U32    reserve0            : 1  ;
+        RK_U32    sw_timeout_mode     : 1  ;
         RK_U32    sw_dec_irq_dis      : 1  ;
         RK_U32    sw_dec_timeout_e    : 1  ;
         RK_U32    sw_buf_empty_en     : 1  ;
@@ -61,8 +62,9 @@ typedef struct {
         RK_U32    sw_dec_rdy_sta      : 1  ;
         RK_U32    sw_dec_bus_sta      : 1  ;
         RK_U32    sw_dec_error_sta    : 1  ;
+        RK_U32    sw_dec_timeout_sta  : 1  ;
         RK_U32    sw_dec_empty_sta    : 1  ;
-        RK_U32    reserve4            : 4  ;
+        RK_U32    reserve3            : 3  ;
         RK_U32    sw_softrst_en_p     : 1  ;
         RK_U32    sw_force_softreset_valid: 1 ;
         RK_U32    sw_softreset_rdy    : 1  ;
