From 322efafd1f760c73accda1a7025b007f211916f7 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sat, 3 Mar 2018 10:10:01 +0100
Subject: [PATCH] [mpp_dec]: sleep when there is nothing to parse

---
 mpp/codec/mpp_dec.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mpp/codec/mpp_dec.cpp b/mpp/codec/mpp_dec.cpp
index 424604e1..dded58c6 100644
--- a/mpp/codec/mpp_dec.cpp
+++ b/mpp/codec/mpp_dec.cpp
@@ -600,8 +600,10 @@ void *mpp_dec_parser_thread(void *data)
         }
         parser->unlock();
 
-        if (try_proc_dec_task(mpp, &task))
+        if (try_proc_dec_task(mpp, &task)) {
+            msleep(1);
             continue;
+        }
 
     }
 
