--- u-boot-imx6-a06fada/include/configs/mx6_cubox-i.h.orig	2014-08-20 21:24:00.000000000 +0200
+++ u-boot-imx6-a06fada/include/configs/mx6_cubox-i.h	2014-08-21 08:12:57.282325080 +0200
@@ -155,6 +155,7 @@
 #define CONFIG_SYS_CONSOLE_IS_IN_ENV
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"zImage=zImage\0" \
         "script=boot.scr\0" \
         "bootfile=auto\0" \
         "bootenv=uEnv.txt\0" \
@@ -217,7 +218,7 @@
         "importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
                 "env import -t ${loadaddr} ${filesize};\0" \
         "autoboot=echo Booting ${boot_file}; " \
-		"if test ${boot_file} = zImage; then " \
+		"if test ${boot_file} = ${zImage}; then " \
 		    "bootz ${loadaddr} ${ramdisk_addr} ${fdt_addr}; " \
 		"else " \
 		    "bootm ${loadaddr} ${ramdisk_addr} ${fdt_addr}; " \
@@ -232,7 +233,7 @@
 		"else " \
 		    "setenv ramdisk_addr -; " \
                 "fi; " \
-                "if test ${boot_file} = zImage; then " \
+                "if test ${boot_file} = ${zImage}; then " \
                     "if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
                         "if run loadfdt; then " \
 			    "echo Loaded ${fdt_file}; " \
@@ -269,7 +270,7 @@
                     "setenv get_cmd tftp; " \
                 "fi; " \
                 "if test ${bootfile} = auto; then " \
-                     "setenv bootfile zImage; " \
+                     "setenv bootfile ${zImage}; " \
                      "if ${get_cmd} ${bootfile}; then " \
                          "run bootit; " \
                      "else " \
@@ -295,7 +296,7 @@
 			   "fi; " \
                            "if test ${bootfile} = auto; then " \
                                    "setenv origbootfile auto; " \
-                                   "setenv bootfile zImage; " \
+                                   "setenv bootfile ${zImage}; " \
                                    "if run loadbootfile; then " \
                                         "run mmcboot; " \
                                    "else " \
