--- u-boot-imx6-cuboxi-3fd848b/include/configs/mx6_cubox-i.h.orig	2014-03-02 20:44:06.000000000 +0100
+++ u-boot-imx6-cuboxi-3fd848b/include/configs/mx6_cubox-i.h	2014-03-05 08:11:31.533508170 +0100
@@ -150,6 +150,7 @@
 #define CONFIG_SYS_CONSOLE_IS_IN_ENV
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"zImage=zImage\0" \
         "script=boot.scr\0" \
         "bootfile=auto\0" \
         "bootenv=uEnv.txt\0" \
@@ -207,25 +208,25 @@
         "importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
                 "env import -t ${loadaddr} ${filesize};\0" \
         "autobootfdt=echo Booting ${boot_file}; " \
-                "if test ${boot_file} = zImage; then " \
+                "if test ${boot_file} = ${zImage}; then " \
 		    "bootz ${loadaddr} - ${fdt_addr}; " \
 		"else " \
 		    "bootm ${loadaddr} - ${fdt_addr}; " \
 		"fi;\0 " \
         "autoboot=echo Booting ${boot_file}; " \
-		"if test ${boot_file} = zImage; then " \
+		"if test ${boot_file} = ${zImage}; then " \
 		    "bootz; " \
 		"else " \
 		    "bootm; " \
 		"fi;\0 " \
 	"bootit=setenv boot_file ${bootfile}; " \
-                "if test ${boot_file} = zImage; then " \
+                "if test ${boot_file} = ${zImage}; then " \
                     "if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
                         "if run loadfdt; then " \
                             "run autobootfdt; " \
                         "else " \
                             "if test ${boot_fdt} = try; then " \
-                                  "echo WARN: Cannot load the DTB and boot file is type zImage;" \
+                                  "echo WARN: Cannot load the DTB and boot file is type ${zImage};" \
                                   "echo if you have not appended a dtb to the file it may;" \
                                   "echo hang after displaying Starting kernel...;" \
                                   "echo ;" \
@@ -255,7 +256,7 @@
                     "setenv get_cmd tftp; " \
                 "fi; " \
                 "if test ${bootfile} = auto; then " \
-                     "setenv bootfile zImage; " \
+                     "setenv bootfile ${zImage}; " \
                      "if ${get_cmd} ${bootfile}; then " \
                          "run bootit; " \
                      "else " \
@@ -278,7 +279,7 @@
 			   "fi; " \
                            "if test ${bootfile} = auto; then " \
                                    "setenv origbootfile auto; " \
-                                   "setenv bootfile zImage; " \
+                                   "setenv bootfile ${zImage}; " \
                                    "if run loadbootfile; then " \
                                         "run mmcboot; " \
                                    "else " \