Index: linux-2.6.16/scripts/gen_initramfs_list.sh
===================================================================
--- linux-2.6.16.orig/scripts/gen_initramfs_list.sh	2006-03-20 18:41:34.000000000 +0100
+++ linux-2.6.16/scripts/gen_initramfs_list.sh	2006-03-20 18:42:40.000000000 +0100
@@ -56,9 +56,7 @@
 
 parse() {
 	local location="$1"
-	local name="${location/${srcdir}//}"
-	# change '//' into '/'
-	name="${name//\/\///}"
+	local name="$(echo "$location" | sed -e 's%$srcdir%%' -e 's%//*%/%g')"
 	local mode="$2"
 	local uid="$3"
 	local gid="$4"
