diff -Naur linux-2.6.31-rc4.orig/fs/fat/inode.c linux-2.6.31-rc4/fs/fat/inode.c
--- linux-2.6.31-rc4.orig/fs/fat/inode.c	2009-07-25 12:47:41.000000000 +0200
+++ linux-2.6.31-rc4/fs/fat/inode.c	2009-07-25 13:38:18.000000000 +0200
@@ -979,7 +979,8 @@
 	}
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	opts->tz_utc = 0;
diff -Naur linux-2.6.31-rc4.orig/fs/isofs/inode.c linux-2.6.31-rc4/fs/isofs/inode.c
--- linux-2.6.31-rc4.orig/fs/isofs/inode.c	2009-07-25 12:47:41.000000000 +0200
+++ linux-2.6.31-rc4/fs/isofs/inode.c	2009-07-25 13:38:49.000000000 +0200
@@ -377,7 +377,7 @@
 	popt->gid = 0;
 	popt->uid = 0;
 	popt->iocharset = NULL;
-	popt->utf8 = 0;
+	popt->utf8 = 1;
 	popt->overriderockperm = 0;
 	popt->session=-1;
 	popt->sbsector=-1;
