diff -Naur linux-2.6.36.orig/drivers/gpu/drm/nouveau/nouveau_sgdma.c linux-2.6.36.nouveau/drivers/gpu/drm/nouveau/nouveau_sgdma.c
--- linux-2.6.36.orig/drivers/gpu/drm/nouveau/nouveau_sgdma.c	2010-10-23 16:00:10.512365461 +0200
+++ linux-2.6.36.nouveau/drivers/gpu/drm/nouveau/nouveau_sgdma.c	2010-10-23 15:57:19.079304241 +0200
@@ -224,11 +224,7 @@
 	int i, ret;
 
 	if (dev_priv->card_type < NV_50) {
-		if(dev_priv->card_type < NV_40) {
-			aper_size = (64 * 1024 * 1024);
-		} else {
-			aper_size = (512 * 1024 * 1024);
-		}
+		aper_size = (64 * 1024 * 1024);
 		obj_size  = (aper_size >> NV_CTXDMA_PAGE_SHIFT) * 4;
 		obj_size += 8; /* ctxdma header */
 	} else {
