diff -Naur linux-2.6.36/drivers/gpu/drm/nouveau/Kconfig linux-2.6.36.patch/drivers/gpu/drm/nouveau/Kconfig
--- linux-2.6.36/drivers/gpu/drm/nouveau/Kconfig	2010-10-24 04:19:00.872718004 +0200
+++ linux-2.6.36.patch/drivers/gpu/drm/nouveau/Kconfig	2010-10-24 04:19:42.287230715 +0200
@@ -10,7 +10,6 @@
 	select FB
 	select FRAMEBUFFER_CONSOLE if !EMBEDDED
 	select FB_BACKLIGHT if DRM_NOUVEAU_BACKLIGHT
-	select ACPI_VIDEO if ACPI
 	help
 	  Choose this option for open-source nVidia support.
 
