diff -Naur linux-2.6.30-rc3.orig/scripts/Makefile.lib linux-2.6.30-rc3/scripts/Makefile.lib
--- linux-2.6.30-rc3.orig/scripts/Makefile.lib	2009-04-25 12:42:08.000000000 +0200
+++ linux-2.6.30-rc3/scripts/Makefile.lib	2009-04-25 18:29:44.000000000 +0200
@@ -204,4 +204,4 @@
 # ---------------------------------------------------------------------------
 
 quiet_cmd_lzma = LZMA    $@
-cmd_lzma = (lzma -9 -c $< && $(size_append) $<) >$@ || (rm -f $@ ; false)
+cmd_lzma = (lzma e -a1 -so $< && $(size_append) $<) >$@ || (rm -f $@ ; false)
