diff -Naur linux-2.6.29.1.orig/drivers/gpu/drm/i915/i915_dma.c linux-2.6.29.1/drivers/gpu/drm/i915/i915_dma.c
--- linux-2.6.29.1.orig/drivers/gpu/drm/i915/i915_dma.c	2009-04-05 19:44:07.000000000 +0200
+++ linux-2.6.29.1/drivers/gpu/drm/i915/i915_dma.c	2009-04-05 19:45:42.000000000 +0200
@@ -1113,8 +1113,8 @@
 	/* don't enable GEM on PAE - needs agp + set_memory_* interface fixes */
 	dev_priv->has_gem = 0;
 #else
-	/* enable GEM by default */
-	dev_priv->has_gem = 1;
+	/* disable GEM by default */
+	dev_priv->has_gem = 0;
 #endif
 
 	dev->driver->get_vblank_counter = i915_get_vblank_counter;
