diff -Naur linux-2.6.30-rc4.orig/drivers/gpu/drm/i915/i915_dma.c linux-2.6.30-rc4/drivers/gpu/drm/i915/i915_dma.c
--- linux-2.6.30-rc4.orig/drivers/gpu/drm/i915/i915_dma.c	2009-05-06 21:37:41.000000000 +0200
+++ linux-2.6.30-rc4/drivers/gpu/drm/i915/i915_dma.c	2009-05-06 21:39:59.000000000 +0200
@@ -1149,7 +1149,7 @@
 	dev_priv->has_gem = 0;
 #else
 	/* enable GEM by default, except on I8xx */
-	dev_priv->has_gem = !IS_I8XX(dev) ? 1 : 0;
+	dev_priv->has_gem = !IS_I8XX(dev) ? 0 : 0;
 #endif
 
 	dev->driver->get_vblank_counter = i915_get_vblank_counter;
