* fix gcc4 compliation - "error: invalid lvalue in increment"

Index: linux-2.6.16/drivers/video/bootsplash/bootsplash.c
===================================================================
--- linux-2.6.16.orig/drivers/video/bootsplash/bootsplash.c	2006-03-20 18:46:53.000000000 +0100
+++ linux-2.6.16/drivers/video/bootsplash/bootsplash.c	2006-03-20 18:46:57.000000000 +0100
@@ -268,7 +268,8 @@
 		    i = ((CLAMP(r + add*2+1) & 0xf8) <<  8) |
 		        ((CLAMP(g + add    ) & 0xfc) <<  3) |
 		        ((CLAMP(b + add*2+1)       ) >>  3);
-		    *((unsigned short *)picp)++ = i;
+		    *((unsigned short *)picp) = i;
+		    picp = (unsigned char *)((unsigned short *)picp + 1);
 		    add ^= 3;
 		    break;
 		case 24:
