--- linux-2.6.16/drivers/media/video/saa7134/saa7134-cards.c	2006-03-20 13:53:29.000000000 +0800
+++ linux-2.6.16/drivers/media/video/saa7134/saa7134-cards.c	2006-04-13 14:07:25.000000000 +0800
@@ -2485,7 +2485,7 @@
 		/* "Cyril Lacoux (Yack)" <clacoux@ifeelgood.org> */
 		.name           = "ASUS Digimatrix TV",
 		.audio_clock    = 0x00200000,
-		.tuner_type     = TUNER_PHILIPS_FQ1216ME,
+		.tuner_type     = TUNER_LG_PAL_NEW_TAPC,
 		.tda9887_conf   = TDA9887_PRESENT,
 		.radio_type     = UNSET,
 		.tuner_addr     = ADDR_UNSET,
@@ -2498,11 +2498,11 @@
 		},{
 			.name = name_comp1,
 			.vmux = 3,
-			.amux = LINE1,
+			.amux = LINE2,
 		},{
 			.name = name_svideo,
 			.vmux = 8,
-			.amux = LINE1,
+			.amux = LINE2
 		}},
 	},
 	[SAA7134_BOARD_PHILIPS_TIGER] = {
