Index: linux-2.6.16/fs/fat/inode.c
===================================================================
--- linux-2.6.16.orig/fs/fat/inode.c	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6.16/fs/fat/inode.c	2006-03-20 18:47:09.000000000 +0100
@@ -952,7 +952,8 @@
 		opts->shortname = 0;
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->nocase = 0;
 	*debug = 0;
Index: linux-2.6.16/fs/isofs/inode.c
===================================================================
--- linux-2.6.16.orig/fs/isofs/inode.c	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6.16/fs/isofs/inode.c	2006-03-20 18:47:09.000000000 +0100
@@ -371,7 +371,7 @@
 	popt->gid = 0;
 	popt->uid = 0;
 	popt->iocharset = NULL;
-	popt->utf8 = 0;
+	popt->utf8 = 1;
 	popt->session=-1;
 	popt->sbsector=-1;
 	if (!options)
