diff -Nur linux-2.6.18.orig/drivers/net/via-rhine.c linux-2.6.18/drivers/net/via-rhine.c
--- linux-2.6.18.orig/drivers/net/via-rhine.c	2006-09-17 16:41:59.000000000 +0200
+++ linux-2.6.18/drivers/net/via-rhine.c	2006-09-17 16:43:30.000000000 +0200
@@ -46,7 +46,7 @@
 
 /* Work-around for broken BIOSes: they are unable to get the chip back out of
    power state D3 so PXE booting fails. bootparam(7): via-rhine.avoid_D3=1 */
-static int avoid_D3;
+static int avoid_D3 = 1;
 
 /*
  * In case you are looking for 'options[]' or 'full_duplex[]', they
