--- linux-2.6.22.orig/drivers/net/8139too.c	2007-07-09 01:32:17.000000000 +0200
+++ linux-2.6.22/drivers/net/8139too.c	2007-09-22 22:29:43.310538315 +0200
@@ -1395,13 +1395,6 @@
 
 	rtl_check_media (dev, 1);
 
-	if (tp->chipset >= CH_8139B) {
-		/* Disable magic packet scanning, which is enabled
-		 * when PM is enabled in Config1.  It can be reenabled
-		 * via ETHTOOL_SWOL if desired.  */
-		RTL_W8 (Config3, RTL_R8 (Config3) & ~Cfg3_Magic);
-	}
-
 	DPRINTK("init buffer addresses\n");
 
 	/* Lock Config[01234] and BMCR register writes */
