diff -Naur linux-2.6.29.1/include/linux/dvb/audio.h linux-2.6.29.1a/include/linux/dvb/audio.h
--- linux-2.6.29.1/include/linux/dvb/audio.h	2009-04-02 22:55:27.000000000 +0200
+++ linux-2.6.29.1a/include/linux/dvb/audio.h	2009-04-14 14:13:04.000000000 +0200
@@ -24,7 +24,12 @@
 #ifndef _DVBAUDIO_H_
 #define _DVBAUDIO_H_
 
+#ifdef __KERNEL__
 #include <linux/types.h>
+#else
+#include <stdint.h>
+#endif
+
 
 typedef enum {
 	AUDIO_SOURCE_DEMUX, /* Select the demux as the main source */
diff -Naur linux-2.6.29.1/include/linux/dvb/dmx.h linux-2.6.29.1a/include/linux/dvb/dmx.h
--- linux-2.6.29.1/include/linux/dvb/dmx.h	2009-04-02 22:55:27.000000000 +0200
+++ linux-2.6.29.1a/include/linux/dvb/dmx.h	2009-04-14 14:13:04.000000000 +0200
@@ -24,7 +24,7 @@
 #ifndef _DVBDMX_H_
 #define _DVBDMX_H_
 
-#include <linux/types.h>
+#include <asm/types.h>
 #ifdef __KERNEL__
 #include <linux/time.h>
 #else
diff -Naur linux-2.6.29.1/include/linux/dvb/frontend.h linux-2.6.29.1a/include/linux/dvb/frontend.h
--- linux-2.6.29.1/include/linux/dvb/frontend.h	2009-04-02 22:55:27.000000000 +0200
+++ linux-2.6.29.1a/include/linux/dvb/frontend.h	2009-04-14 14:13:04.000000000 +0200
@@ -26,7 +26,8 @@
 #ifndef _DVBFRONTEND_H_
 #define _DVBFRONTEND_H_
 
-#include <linux/types.h>
+#include <asm/types.h>
+
 
 typedef enum fe_type {
 	FE_QPSK,
diff -Naur linux-2.6.29.1/include/linux/dvb/net.h linux-2.6.29.1a/include/linux/dvb/net.h
--- linux-2.6.29.1/include/linux/dvb/net.h	2009-04-02 22:55:27.000000000 +0200
+++ linux-2.6.29.1a/include/linux/dvb/net.h	2009-04-14 14:13:04.000000000 +0200
@@ -24,7 +24,8 @@
 #ifndef _DVBNET_H_
 #define _DVBNET_H_
 
-#include <linux/types.h>
+#include <asm/types.h>
+
 
 struct dvb_net_if {
 	__u16 pid;
diff -Naur linux-2.6.29.1/include/linux/dvb/video.h linux-2.6.29.1a/include/linux/dvb/video.h
--- linux-2.6.29.1/include/linux/dvb/video.h	2009-04-02 22:55:27.000000000 +0200
+++ linux-2.6.29.1a/include/linux/dvb/video.h	2009-04-14 14:13:04.000000000 +0200
@@ -24,14 +24,17 @@
 #ifndef _DVBVIDEO_H_
 #define _DVBVIDEO_H_
 
-#include <linux/types.h>
-#ifdef __KERNEL__
 #include <linux/compiler.h>
+
+#ifdef __KERNEL__
+#include <linux/types.h>
 #else
+#include <asm/types.h>
 #include <stdint.h>
 #include <time.h>
 #endif
 
+
 typedef enum {
 	VIDEO_FORMAT_4_3,     /* Select 4:3 format */
 	VIDEO_FORMAT_16_9,    /* Select 16:9 format. */
