diff -Naur linux-2.6.29-rc3.orig/sound/usb/usbmixer_maps.c linux-2.6.29-rc3/sound/usb/usbmixer_maps.c
--- linux-2.6.29-rc3.orig/sound/usb/usbmixer_maps.c	2009-01-31 18:44:42.000000000 +0100
+++ linux-2.6.29-rc3/sound/usb/usbmixer_maps.c	2009-01-31 18:53:29.000000000 +0100
@@ -284,6 +284,11 @@
 		.id = USB_ID(0x041e, 0x3040),
 		.map = live24ext_map,
 	},
+        {
+		.id = USB_ID(0x041e, 0x3048),
+		.map = audigy2nx_map,
+		.selector_map = audigy2nx_selectors,
+	},
 	{
 		/* Hercules DJ Console (Windows Edition) */
 		.id = USB_ID(0x06f8, 0xb000),
diff -Naur linux-2.6.29-rc3.orig/sound/usb/usbquirks.h linux-2.6.29-rc3/sound/usb/usbquirks.h
--- linux-2.6.29-rc3.orig/sound/usb/usbquirks.h	2009-01-31 18:44:42.000000000 +0100
+++ linux-2.6.29-rc3/sound/usb/usbquirks.h	2009-01-31 18:54:49.000000000 +0100
@@ -39,6 +39,16 @@
 	.idProduct = prod, \
 	.bInterfaceClass = USB_CLASS_VENDOR_SPEC
 
+/* Creative/Toshiba Multimedia Center SB-0500 */
+{
+	USB_DEVICE(0x041e, 0x3048),
+	.driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
+               .vendor_name = "Toshiba",
+               .product_name = "SB-0500",
+               .ifnum = QUIRK_NO_INTERFACE
+	}
+},
+
 /* Creative/E-Mu devices */
 {
 	USB_DEVICE(0x041e, 0x3010),
