diff -Naur linux-3.0-rc7/arch/x86/pci/acpi.c linux-3.0-rc7.patch/arch/x86/pci/acpi.c
--- linux-3.0-rc7/arch/x86/pci/acpi.c	2011-07-12 01:51:52.000000000 +0200
+++ linux-3.0-rc7.patch/arch/x86/pci/acpi.c	2011-07-20 07:46:08.959060136 +0200
@@ -246,7 +246,7 @@
 
 		conflict = insert_resource_conflict(root, res);
 		if (conflict)
-			dev_err(&info->bridge->dev,
+			dev_info(&info->bridge->dev,
 				"address space collision: host bridge window %pR "
 				"conflicts with %s %pR\n",
 				res, conflict->name, conflict);
