diff -Naur linux-2.6.38-rc6/drivers/media/rc/nuvoton-cir.h linux-2.6.38-rc6.patch/drivers/media/rc/nuvoton-cir.h
--- linux-2.6.38-rc6/drivers/media/rc/nuvoton-cir.h	2011-02-22 02:25:52.000000000 +0100
+++ linux-2.6.38-rc6.patch/drivers/media/rc/nuvoton-cir.h	2011-02-28 14:26:20.385871213 +0100
@@ -306,7 +306,7 @@
 #define CIR_WAKE_IRFIFOSTS_RX_FULL	0x10
 
 /* CIR Wake FIFO buffer is 67 bytes long */
-#define CIR_WAKE_FIFO_LEN		67
+#define CIR_WAKE_FIFO_LEN		65
 /* CIR Wake byte comparison tolerance */
 #define CIR_WAKE_CMP_TOLERANCE		5
 
