diff -Naur linux-2.6.37-rc1/include/linux/fs.h linux-2.6.37-rc1.patch/include/linux/fs.h
--- linux-2.6.37-rc1/include/linux/fs.h	2010-11-01 12:54:12.000000000 +0100
+++ linux-2.6.37-rc1.patch/include/linux/fs.h	2010-11-15 19:06:11.732172159 +0100
@@ -34,9 +34,9 @@
 #define SEEK_MAX	SEEK_END
 
 struct fstrim_range {
-	uint64_t start;
-	uint64_t len;
-	uint64_t minlen;
+	__u64 start;
+	__u64 len;
+	__u64 minlen;
 };
 
 /* And dynamically-tunable limits and defaults: */
