diff -Naur linux-3.2.1/drivers/media/rc/imon.c linux-3.2.1.patch/drivers/media/rc/imon.c
--- linux-3.2.1/drivers/media/rc/imon.c	2012-01-12 20:42:45.000000000 +0100
+++ linux-3.2.1.patch/drivers/media/rc/imon.c	2012-01-22 13:13:42.713053959 +0100
@@ -1658,7 +1658,7 @@
 		return;
 
 	ictx = (struct imon_context *)urb->context;
-	if (!ictx || !ictx->dev_present_intf0)
+	if (!ictx)
 		return;
 
 	switch (urb->status) {
@@ -1690,7 +1690,7 @@
 		return;
 
 	ictx = (struct imon_context *)urb->context;
-	if (!ictx || !ictx->dev_present_intf1)
+	if (!ictx)
 		return;
 
 	switch (urb->status) {
@@ -2118,6 +2118,7 @@
 
 	ictx->dev = dev;
 	ictx->usbdev_intf0 = usb_get_dev(interface_to_usbdev(intf));
+	ictx->dev_present_intf0 = true;
 	ictx->rx_urb_intf0 = rx_urb;
 	ictx->tx_urb = tx_urb;
 	ictx->rf_device = false;
@@ -2156,8 +2157,6 @@
 		goto rdev_setup_failed;
 	}
 
-	ictx->dev_present_intf0 = true;
-
 	mutex_unlock(&ictx->lock);
 	return ictx;
 
@@ -2201,6 +2200,7 @@
 	}
 
 	ictx->usbdev_intf1 = usb_get_dev(interface_to_usbdev(intf));
+	ictx->dev_present_intf1 = true;
 	ictx->rx_urb_intf1 = rx_urb;
 
 	ret = -ENODEV;
@@ -2229,8 +2229,6 @@
 		goto urb_submit_failed;
 	}
 
-	ictx->dev_present_intf1 = true;
-
 	mutex_unlock(&ictx->lock);
 	return ictx;
 
