diff -Naur linux-2.6.37-old/drivers/usb/core/hcd.c linux-2.6.37-new/drivers/usb/core/hcd.c
--- linux-2.6.37-old/drivers/usb/core/hcd.c	2011-01-04 16:50:19.000000000 -0800
+++ linux-2.6.37-new/drivers/usb/core/hcd.c	2011-01-26 10:45:48.000000000 -0800
@@ -1993,6 +1993,8 @@
 
 	usb_lock_device(udev);
 	usb_remote_wakeup(udev);
+	if (hcd->state == HC_STATE_RUNNING)
+		clear_bit(HCD_FLAG_WAKEUP_PENDING, &hcd->flags);
 	usb_unlock_device(udev);
 }
 
