diff -Naur linux-2.6.38.2/drivers/media/rc/nuvoton-cir.c linux-2.6.38.2.patch/drivers/media/rc/nuvoton-cir.c
--- linux-2.6.38.2/drivers/media/rc/nuvoton-cir.c	2011-03-27 20:37:20.000000000 +0200
+++ linux-2.6.38.2.patch/drivers/media/rc/nuvoton-cir.c	2011-04-11 21:02:39.897028275 +0200
@@ -250,9 +250,8 @@
 
 	if (chip_major != CHIP_ID_HIGH ||
 	    (chip_minor != CHIP_ID_LOW && chip_minor != CHIP_ID_LOW2)) {
-		nvt_pr(KERN_ERR, "%s: unsupported chip, id: 0x%02x 0x%02x",
+		nvt_pr(KERN_WARNING, "warning %s: unsupported chip revision, id: 0x%02x 0x%02x, it may not work",
 		       chip_id, chip_major, chip_minor);
-		ret = -ENODEV;
 	}
 
 	nvt_efm_disable(nvt);
