diff --git a/drivers/media/rc/ite-cir.c b/drivers/media/rc/ite-cir.c
index 43908a7..1a63d75 100644
--- a/drivers/media/rc/ite-cir.c
+++ b/drivers/media/rc/ite-cir.c
@@ -1676,7 +1676,8 @@ static int ite_resume(struct pnp_dev *pdev)
 	struct ite_dev *dev = pnp_get_drvdata(pdev);
 	unsigned long flags;
 
-	ite_dbg("%s called", __func__);
+	ite_dbg("%s called (tx'ing: %s)", __func__,
+		dev->transmitting ? "true" : "false");
 
 	spin_lock_irqsave(&dev->lock, flags);
 
