---
 kernel/Kconfig.hz |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.37-ck1/kernel/Kconfig.hz
===================================================================
--- linux-2.6.37-ck1.orig/kernel/Kconfig.hz	2009-06-10 13:05:27.000000000 +1000
+++ linux-2.6.37-ck1/kernel/Kconfig.hz	2011-01-06 14:07:58.988366741 +1100
@@ -4,7 +4,7 @@
 
 choice
 	prompt "Timer frequency"
-	default HZ_250
+	default HZ_1000
 	help
 	 Allows the configuration of the timer frequency. It is customary
 	 to have the timer interrupt run at 1000 Hz but 100 Hz may be more
