diff -Naur linux-3.2.2/arch/arm/include/asm/bug.h linux-3.2.2.patch/arch/arm/include/asm/bug.h
--- linux-3.2.2/arch/arm/include/asm/bug.h	2012-01-26 01:39:32.000000000 +0100
+++ linux-3.2.2.patch/arch/arm/include/asm/bug.h	2012-02-03 02:52:23.960027219 +0100
@@ -32,7 +32,6 @@
 
 #define __BUG(__file, __line, __value)				\
 do {								\
-	BUILD_BUG_ON(sizeof(struct bug_entry) != 12);		\
 	asm volatile("1:\t" BUG_INSTR_TYPE #__value "\n"	\
 		".pushsection .rodata.str, \"aMS\", %progbits, 1\n" \
 		"2:\t.asciz " #__file "\n" 			\
