diff -Naur linux-2.6.37-old/drivers/usb/core/hcd.c linux-2.6.37-new/drivers/usb/core/hcd.c
--- linux-2.6.37-old/drivers/usb/core/hcd.c	2011-01-04 16:50:19.000000000 -0800
+++ linux-2.6.37-new/drivers/usb/core/hcd.c	2011-01-30 20:06:09.000000000 -0800
@@ -1956,7 +1956,6 @@
 
 	dev_dbg(&rhdev->dev, "usb %s%s\n",
 			(msg.event & PM_EVENT_AUTO ? "auto-" : ""), "resume");
-	clear_bit(HCD_FLAG_WAKEUP_PENDING, &hcd->flags);
 	if (!hcd->driver->bus_resume)
 		return -ENOENT;
 	if (hcd->state == HC_STATE_RUNNING)
@@ -1964,6 +1963,7 @@
 
 	hcd->state = HC_STATE_RESUMING;
 	status = hcd->driver->bus_resume(hcd);
+	clear_bit(HCD_FLAG_WAKEUP_PENDING, &hcd->flags);
 	if (status == 0) {
 		/* TRSMRCY = 10 msec */
 		msleep(10);
