diff -Naur linux-2.6.39-rc5/drivers/gpu/drm/i915/i915_drv.c linux-2.6.39-rc5.patch/drivers/gpu/drm/i915/i915_drv.c
--- linux-2.6.39-rc5/drivers/gpu/drm/i915/i915_drv.c	2011-04-27 05:48:50.000000000 +0200
+++ linux-2.6.39-rc5.patch/drivers/gpu/drm/i915/i915_drv.c	2011-04-29 02:51:28.773622809 +0200
@@ -760,7 +760,7 @@
 static int __init i915_init(void)
 {
 	if (!intel_agp_enabled) {
-		DRM_ERROR("drm/i915 can't work without intel_agp module!\n");
+		DRM_INFO("drm/i915 can't work without intel_agp module!\n");
 		return -ENODEV;
 	}
 
