diff -Naur linux-3.6.2.vc04_patch/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c linux-3.6.2.buildfix_patch/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c
--- linux-3.6.2.vc04_patch/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c	2012-10-17 18:57:47.000000000 +0200
+++ linux-3.6.2.buildfix_patch/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c	2012-10-21 15:19:56.706236450 +0200
@@ -126,7 +126,7 @@
 	}
 
 	err = request_irq(VCHIQ_DOORBELL_IRQ, vchiq_doorbell_irq,
-		IRQF_SAMPLE_RANDOM | IRQF_IRQPOLL, "VCHIQ doorbell",
+		IRQF_IRQPOLL, "VCHIQ doorbell",
 		state);
 	if (err < 0)
 	{
