---
 mm/vmscan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.37-ck1/mm/vmscan.c
===================================================================
--- linux-2.6.37-ck1.orig/mm/vmscan.c	2011-01-06 14:07:57.629285994 +1100
+++ linux-2.6.37-ck1/mm/vmscan.c	2011-01-06 14:07:57.758293657 +1100
@@ -133,7 +133,7 @@ struct scan_control {
 /*
  * From 0 .. 100.  Higher means more swappy.
  */
-int vm_swappiness = 60;
+int vm_swappiness;
 long vm_total_pages;	/* The total number of pages which the VM controls */
 
 static LIST_HEAD(shrinker_list);
