Index: linux-2.6.16/scripts/kconfig/confdata.c
===================================================================
--- linux-2.6.16.orig/scripts/kconfig/confdata.c	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6.16/scripts/kconfig/confdata.c	2006-03-20 18:47:06.000000000 +0100
@@ -340,7 +340,7 @@
 	int type, l;
 	const char *str;
 	time_t now;
-	int use_timestamp = 1;
+	int use_timestamp = 0;
 	char *env;
 
 	dirname[0] = 0;
