diff -Naur linux-2.6.39.1/drivers/media/dvb/dvb-usb/rtl2832u.h linux-2.6.39.1.patch/drivers/media/dvb/dvb-usb/rtl2832u.h
--- linux-2.6.39.1/drivers/media/dvb/dvb-usb/rtl2832u.h	2010-10-27 08:17:30.000000000 +0200
+++ linux-2.6.39.1.patch/drivers/media/dvb/dvb-usb/rtl2832u.h	2011-06-15 18:22:30.416756439 +0200
@@ -39,7 +39,8 @@
 #define	USB_PID_DEXATEK_WARM_4		       0x1104
 
 #define	USB_VID_GTEK					       0x1F4D
-#define	USB_PID_GTEK_WARM			       0x0837
+//#define	USB_PID_GTEK_WARM			       0x0837
+#define	USB_PID_GTEK_WARM			       0xC803
 #define	USB_PID_GTEK_WARM_A			       0xA803
 #define	USB_PID_GTEK_WARM_B			       0xB803
 #define	USB_PID_GTEK_WARM_C			       0xC803
