diff -Naur linux-2.6.29-rc8-old/drivers/ata/libata-core.c linux-2.6.29-rc8-new/drivers/ata/libata-core.c
--- linux-2.6.29-rc8-old/drivers/ata/libata-core.c	2009-03-15 11:44:28.000000000 -0700
+++ linux-2.6.29-rc8-new/drivers/ata/libata-core.c	2009-03-15 11:50:46.000000000 -0700
@@ -4444,6 +4444,12 @@
 			       "other device, disabling DMA\n");
 	}
 
+	if (ata_id_is_cfa(dev->id)) {
+		xfer_mask &= ~(ATA_MASK_MWDMA | ATA_MASK_UDMA);
+		ata_dev_printk(dev, KERN_WARNING,
+			       "device is a CFA, disabling DMA\n");
+	}
+
 	if (ap->flags & ATA_FLAG_NO_IORDY)
 		xfer_mask &= ata_pio_mask_no_iordy(dev);
 
diff -Naur linux-2.6.29-rc8-old/drivers/ide/ide-dma.c linux-2.6.29-rc8-new/drivers/ide/ide-dma.c
--- linux-2.6.29-rc8-old/drivers/ide/ide-dma.c	2009-03-15 11:44:28.000000000 -0700
+++ linux-2.6.29-rc8-new/drivers/ide/ide-dma.c	2009-03-15 11:50:46.000000000 -0700
@@ -217,6 +217,12 @@
 				    drive->name, (char *)&id[ATA_ID_PROD]);
 		return blacklist;
 	}
+	if (id->config == 0x848a) {
+		printk(KERN_WARNING "%s: Disabling (U)DMA for %s (CFA)\n",
+				    drive->name, id->model);
+		return 1;
+	}
+
 	return 0;
 }
 EXPORT_SYMBOL(__ide_dma_bad_drive);
