---
 Makefile |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.37-ck2/Makefile
===================================================================
--- linux-2.6.37-ck2.orig/Makefile	2011-01-06 14:04:07.000000000 +1100
+++ linux-2.6.37-ck2/Makefile	2011-02-14 10:11:20.469252000 +1100
@@ -10,6 +10,10 @@
 # Comments in this file are targeted only to the developer, do not
 # expect to learn how to build the kernel reading this file.
 
+CKVERSION = -ck2
+CKNAME = BFS Powered
+EXTRAVERSION := $(EXTRAVERSION)$(CKVERSION)
+
 # Do not:
 # o  use make's built-in rules and variables
 #    (this increases performance and avoids hard-to-debug behaviour);
