commit 15d89120d03116adbbf3226a85fbd2fff0b12576
Author: Matthew Garrett <mjg@redhat.com>
Date:   Fri Sep 17 11:09:12 2010 -0400

    Enable USB PCI autosuspend by default

diff --git a/drivers/usb/core/hcd-pci.c b/drivers/usb/core/hcd-pci.c
index c3f9854..3c020e6 100644
--- a/drivers/usb/core/hcd-pci.c
+++ b/drivers/usb/core/hcd-pci.c
@@ -248,8 +248,10 @@ int usb_hcd_pci_probe(struct pci_dev *dev, const struct pci_device_id *id)
 		goto err4;
 	set_hs_companion(dev, hcd);
 
-	if (pci_dev_run_wake(dev))
+	if (pci_dev_run_wake(dev)) {
 		pm_runtime_put_noidle(&dev->dev);
+		pm_runtime_allow(&dev->dev);
+	}
 	return retval;
 
  err4:
