---
 include/linux/swap.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.37-ck1/include/linux/swap.h
===================================================================
--- linux-2.6.37-ck1.orig/include/linux/swap.h	2011-01-06 14:04:10.493805562 +1100
+++ linux-2.6.37-ck1/include/linux/swap.h	2011-01-06 14:07:57.889301442 +1100
@@ -348,9 +348,10 @@ extern struct mm_struct *swap_token_mm;
 extern void grab_swap_token(struct mm_struct *);
 extern void __put_swap_token(struct mm_struct *);
 
+/* Only allow swap token to have effect if swap is full */
 static inline int has_swap_token(struct mm_struct *mm)
 {
-	return (mm == swap_token_mm);
+	return (mm == swap_token_mm && vm_swap_full());
 }
 
 static inline void put_swap_token(struct mm_struct *mm)
