commit 9d4c919bcfa794c054cc33155c7e3c53ac2c5684
Author: Matthew Garrett <mjg@redhat.com>
Date:   Sun Jul 19 02:24:49 2009 +0100

    Enable autosuspend on UVC by default

diff --git a/drivers/media/video/uvc/uvc_driver.c b/drivers/media/video/uvc/uvc_driver.c
index 89927b7..8de516b 100644
--- a/drivers/media/video/uvc/uvc_driver.c
+++ b/drivers/media/video/uvc/uvc_driver.c
@@ -1647,6 +1647,8 @@ static int uvc_probe(struct usb_interface *intf,
 			"supported.\n", ret);
 	}
 
+	usb_device_autosuspend_enable(udev);
+
 	uvc_trace(UVC_TRACE_PROBE, "UVC device initialized.\n");
 	return 0;
 
