diff -Naur linux-2.6.39-rc6/drivers/ata/libata-core.c linux-2.6.39-rc6.patch/drivers/ata/libata-core.c
--- linux-2.6.39-rc6/drivers/ata/libata-core.c	2011-05-04 04:59:13.000000000 +0200
+++ linux-2.6.39-rc6.patch/drivers/ata/libata-core.c	2011-05-04 19:59:38.952317088 +0200
@@ -4139,6 +4139,7 @@
 	 */
 	{ "PIONEER DVD-RW  DVRTD08",	"1.00",	ATA_HORKAGE_NOSETXFER },
 	{ "PIONEER DVD-RW  DVR-212D",	"1.28", ATA_HORKAGE_NOSETXFER },
+	{ "PIONEER DVD-RW  DVR-216D",	"1.07",	ATA_HORKAGE_NOSETXFER },
 	{ "PIONEER DVD-RW  DVR-216D",	"1.08", ATA_HORKAGE_NOSETXFER },
 
 	/* End Marker */
