diff -Naur linux-2.6.36/drivers/gpu/drm/vmwgfx/vmwgfx_fb.c linux-2.6.36.patch/drivers/gpu/drm/vmwgfx/vmwgfx_fb.c
--- linux-2.6.36/drivers/gpu/drm/vmwgfx/vmwgfx_fb.c	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36.patch/drivers/gpu/drm/vmwgfx/vmwgfx_fb.c	2010-11-09 00:40:07.000000000 +0100
@@ -615,11 +615,6 @@
 	if (unlikely(ret != 0))
 		goto err_unlock;
 
-	if (bo->mem.mem_type == TTM_PL_VRAM &&
-	    bo->mem.mm_node->start < bo->num_pages)
-		(void) ttm_bo_validate(bo, &vmw_sys_placement, false,
-				       false, false);
-
 	ret = ttm_bo_validate(bo, &ne_placement, false, false, false);
 
 	/* Could probably bug on */
