diff -Naur linux-2.6.39-rc5/drivers/media/rc/mceusb.c linux-2.6.39-rc5.patch/drivers/media/rc/mceusb.c
--- linux-2.6.39-rc5/drivers/media/rc/mceusb.c	2011-04-27 05:48:50.000000000 +0200
+++ linux-2.6.39-rc5.patch/drivers/media/rc/mceusb.c	2011-04-29 18:10:32.361073307 +0200
@@ -220,6 +220,8 @@
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x206c) },
 	/* Philips/Spinel plus IR transceiver for ASUS */
 	{ USB_DEVICE(VENDOR_PHILIPS, 0x2088) },
+	/* Philips RC6 Transceiver, Dell branded */
+	{ USB_DEVICE(VENDOR_PHILIPS, 0x2093) },
 	/* Realtek MCE IR Receiver and card reader */
 	{ USB_DEVICE(VENDOR_REALTEK, 0x0161),
 	  .driver_info = MULTIFUNCTION },
