--- vanilla-2.6.31-rc4/drivers/gpu/drm/i915/intel_lvds.c~	2009-07-31 11:23:05.000000000 -0700
+++ vanilla-2.6.31-rc4/drivers/gpu/drm/i915/intel_lvds.c	2009-07-31 11:23:05.000000000 -0700
@@ -111,19 +111,12 @@ static void intel_lvds_set_power(struct 
 	if (on) {
 		I915_WRITE(ctl_reg, I915_READ(ctl_reg) |
 			   POWER_TARGET_ON);
-		do {
-			pp_status = I915_READ(status_reg);
-		} while ((pp_status & PP_ON) == 0);
-
 		intel_lvds_set_backlight(dev, dev_priv->backlight_duty_cycle);
 	} else {
 		intel_lvds_set_backlight(dev, 0);
 
 		I915_WRITE(ctl_reg, I915_READ(ctl_reg) &
 			   ~POWER_TARGET_ON);
-		do {
-			pp_status = I915_READ(status_reg);
-		} while (pp_status & PP_ON);
 	}
 }
 
