diff -Naur linux-3.0-old/drivers/input/misc/ati_remote2.c linux-3.0-new/drivers/input/misc/ati_remote2.c
--- linux-3.0-old/drivers/input/misc/ati_remote2.c	2011-07-22 08:42:47.000000000 -0700
+++ linux-3.0-new/drivers/input/misc/ati_remote2.c	2011-07-22 08:44:20.000000000 -0700
@@ -619,7 +619,7 @@
 	idev->phys = ar2->phys;
 
 	usb_to_input_id(ar2->udev, &idev->id);
-	idev->dev.parent = &ar2->udev->dev;
+	idev->dev.parent = &ar2->intf[0]->dev;
 
 	retval = input_register_device(idev);
 	if (retval)
diff -Naur linux-3.0-old/drivers/input/misc/ati_remote.c linux-3.0-new/drivers/input/misc/ati_remote.c
--- linux-3.0-old/drivers/input/misc/ati_remote.c	2011-07-22 08:42:47.000000000 -0700
+++ linux-3.0-new/drivers/input/misc/ati_remote.c	2011-07-22 08:44:20.000000000 -0700
@@ -682,7 +682,7 @@
 	idev->phys = ati_remote->phys;
 
 	usb_to_input_id(ati_remote->udev, &idev->id);
-	idev->dev.parent = &ati_remote->udev->dev;
+	idev->dev.parent = &ati_remote->interface->dev;
 }
 
 static int ati_remote_initialize(struct ati_remote *ati_remote)
