diff -Naur linux-3.2.17/drivers/net/wireless/brcm80211/brcmsmac/main.c linux-3.2.17.patch/drivers/net/wireless/brcm80211/brcmsmac/main.c
--- linux-3.2.17/drivers/net/wireless/brcm80211/brcmsmac/main.c	2012-05-11 14:15:38.000000000 +0200
+++ linux-3.2.17.patch/drivers/net/wireless/brcm80211/brcmsmac/main.c	2012-05-15 10:39:19.751263956 +0200
@@ -7879,7 +7879,7 @@
 	if (wlc->hw->suspended_fifos) {
 		hdr = (struct ieee80211_hdr *)p->data;
 		if (ieee80211_is_beacon(hdr->frame_control))
-			brcms_b_mute(wlc->hw, false);
+			brcms_b_mute(wlc->hw, false, 0);
 	}
 
 	memcpy(IEEE80211_SKB_RXCB(p), &rx_status, sizeof(rx_status));
