diff -Naur linux-3.2.16/drivers/bluetooth/btusb.c linux-3.2.16.patch/drivers/bluetooth/btusb.c
--- linux-3.2.17/drivers/bluetooth/btusb.c	2012-04-22 18:31:32.000000000 -0400
+++ linux-3.2.17.patch/drivers/bluetooth/btusb.c	2012-04-29 15:19:24.120925633 -0400
@@ -102,6 +102,7 @@
 
 	/* Broadcom BCM20702A0 */
 	{ USB_DEVICE(0x0a5c, 0x21e3) },
+	{ USB_DEVICE(0x0a5c, 0x21e8) },
 	{ USB_DEVICE(0x413c, 0x8197) },
 
 	{ }	/* Terminating entry */
