diff -Naur linux-3.2.21/drivers/media/dvb/dvb-usb/dw2102.c linux-3.2.21.patch/drivers/media/dvb/dvb-usb/dw2102.c
--- linux-3.2.21/drivers/media/dvb/dvb-usb/dw2102.c	2012-06-20 00:18:30.000000000 +0200
+++ linux-3.2.21.patch/drivers/media/dvb/dvb-usb/dw2102.c	2012-06-28 14:08:50.721691934 +0200
@@ -1181,6 +1181,14 @@
 {
 	u8 obuf[3] = { 0xe, 0x80, 0 };
 	u8 ibuf[] = { 0 };
+	
+	if (dvb_usb_generic_rw(d->dev, obuf, 3, ibuf, 1, 0) < 0)
+		err("command 0x0e transfer failed.");
+
+	//power on su3000
+	obuf[0] = 0xe;
+	obuf[1] = 0x02;
+	obuf[2] = 1;
 
 	if (dvb_usb_generic_rw(d->dev, obuf, 3, ibuf, 1, 0) < 0)
 		err("command 0x0e transfer failed.");
@@ -1448,6 +1456,7 @@
 	{USB_DEVICE(0x3034, 0x7500)},
 	{USB_DEVICE(0x1f4d, 0x3000)},
 	{USB_DEVICE(USB_VID_TERRATEC, 0x00a8)},
+	{USB_DEVICE(USB_VID_TERRATEC, 0x00b0)},
 	{USB_DEVICE(0x9022, USB_PID_TEVII_S480_1)},
 	{USB_DEVICE(0x9022, USB_PID_TEVII_S480_2)},
 	{USB_DEVICE(0x1f4d, 0x3100)},
@@ -1839,7 +1848,7 @@
 		}},
 		}
 	},
-	.num_device_descs = 3,
+	.num_device_descs = 4,
 	.devices = {
 		{ "SU3000HD DVB-S USB2.0",
 			{ &dw2102_table[10], NULL },
@@ -1853,6 +1862,10 @@
 			{ &dw2102_table[14], NULL },
 			{ NULL },
 		},
+		{ "Terratec Cinergy S2 USB HD Rev.2",
+			{ &dw2102_table[17], NULL },
+			{ NULL },
+		},
 	}
 };
 
