--- linux-3.2.2.orig/drivers/media/rc/mceusb.c	2012-01-30 23:37:12.374473509 +0100
+++ linux-3.2.2/drivers/media/rc/mceusb.c	2012-01-30 23:40:57.989652931 +0100
@@ -350,6 +350,8 @@
 	{ USB_DEVICE(VENDOR_FORMOSA, 0xe015) },
 	/* Formosa21 / eHome Infrared Receiver */
 	{ USB_DEVICE(VENDOR_FORMOSA, 0xe016) },
+	/* Formosa21 / eHome Infrared Receiver */
+	{ USB_DEVICE(VENDOR_FORMOSA, 0xe042) },
 	/* Formosa aim / Trust MCE Infrared Receiver */
 	{ USB_DEVICE(VENDOR_FORMOSA, 0xe017),
 	  .driver_info = MCE_GEN2_NO_TX },
