From ef98446236432e102d00d841cf1a6b63ed7be41f Mon Sep 17 00:00:00 2001
From: Dom Cobley <dc4@broadcom.com>
Date: Wed, 15 Feb 2012 18:33:00 +0000
Subject: [PATCH] Fix bug where vchiq fails when L2 enabled

Signed-off-by: Dom Cobley <dc4@broadcom.com>
---
 .../interface/vchiq_arm/vchiq_2835_arm.c           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c b/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c
index ca065ff..a3d8f3c 100644
--- a/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c
+++ b/drivers/misc/vc04_services/interface/vchiq_arm/vchiq_2835_arm.c
@@ -36,7 +36,7 @@
 #define TOTAL_SLOTS (VCHIQ_SLOT_ZERO_SLOTS + 2 * 32)
 
 #define VCHIQ_DOORBELL_IRQ IRQ_ARM_DOORBELL_0
-#define VCHIQ_ARM_ADDRESS(x) __virt_to_bus(x)
+#define VCHIQ_ARM_ADDRESS(x) __virt_to_bus((unsigned)x)
 
 #include "vchiq_arm.h"
 #include "vchiq_2835.h"
-- 
1.7.5.4

