diff -Naur linux-3.7.5/drivers/input/joystick/xpad.c linux-3.7.5.patch/drivers/input/joystick/xpad.c
--- linux-3.7.5/drivers/input/joystick/xpad.c	2013-01-28 05:50:55.000000000 +0100
+++ linux-3.7.5.patch/drivers/input/joystick/xpad.c	2013-01-30 05:14:21.434759231 +0100
@@ -166,7 +166,6 @@
 	{ 0x0f0d, 0x0016, "Hori Real Arcade Pro.EX", MAP_TRIGGERS_TO_BUTTONS, XTYPE_XBOX360 },
 	{ 0x0f0d, 0x000d, "Hori Fighting Stick EX2", MAP_TRIGGERS_TO_BUTTONS, XTYPE_XBOX360 },
 	{ 0x1689, 0xfd00, "Razer Onza Tournament Edition", MAP_DPAD_TO_BUTTONS, XTYPE_XBOX360 },
-	{ 0xffff, 0xffff, "Chinese-made Xbox Controller", 0, XTYPE_XBOX },
 	{ 0x0000, 0x0000, "Generic X-Box pad", 0, XTYPE_UNKNOWN }
 };
 
