From 8628f55e551ba2d19b25d9d8ab5a62b1a42ed05d Mon Sep 17 00:00:00 2001
From: Francisco Jerez <currojerez@riseup.net>
Date: Thu, 14 Oct 2010 19:55:23 +0000
Subject: drm/nouveau: Use lazy fence waits when doing software interchannel sync.

Signed-off-by: Francisco Jerez <currojerez@riseup.net>
---
diff --git a/drivers/gpu/drm/nouveau/nouveau_fence.c b/drivers/gpu/drm/nouveau/nouveau_fence.c
index 5f9f66f..0a22955 100644
--- a/drivers/gpu/drm/nouveau/nouveau_fence.c
+++ b/drivers/gpu/drm/nouveau/nouveau_fence.c
@@ -390,7 +390,7 @@ nouveau_fence_sync(struct nouveau_fence *fence,
 	if (!sema) {
 		/* Early card or broken userspace, fall back to
 		 * software sync. */
-		return nouveau_fence_wait(fence, NULL, false, false);
+		return nouveau_fence_wait(fence, NULL, true, false);
 	}
 
 	/* try to take chan's mutex, if we can't take it right away
@@ -399,7 +399,7 @@ nouveau_fence_sync(struct nouveau_fence *fence,
 	 */
 	if (!mutex_trylock(&chan->mutex)) {
 		free_semaphore(&sema->ref);
-		return nouveau_fence_wait(fence, NULL, false, false);
+		return nouveau_fence_wait(fence, NULL, true, false);
 	}
 
 	/* Make wchan wait until it gets signalled */
--
cgit v0.8.3-6-g21f6
