diff -rupN u-boot.orig/cmd/hwrev.c u-boot/cmd/hwrev.c
--- u-boot.orig/cmd/hwrev.c	2024-02-15 14:49:02.135798042 +0000
+++ u-boot/cmd/hwrev.c	2024-02-15 14:54:37.885602790 +0000
@@ -34,6 +34,26 @@ int do_hwrev(cmd_tbl_t *cmdtp, int flag,
 		env_set("hwrev", "v10-go3");
 		env_set("dtb_name", "rk3326-odroidgo3-linux.dtb");
 	}
+        /* RG351V */
+        else if (check_range(490, 540, hwrev_adc)) {
+                env_set("hwrev", "rg351v");
+                env_set("dtb_name", "rk3326-anbernic-rg351v.dtb");
+        }
+        /* R33S */
+        else if (check_range(140, 190, hwrev_adc)) {
+                env_set("hwrev", "r33s");
+                env_set("dtb_name", "rk3326-gameconsole-r33s.dtb");
+        }
+        /* XU10 */
+        else if (check_range(1000, 1050, hwrev_adc)) {
+                env_set("hwrev", "xu10");
+                env_set("dtb_name", "rk3326-magicx-xu10.dtb");
+        }
+        /* CHI */
+        else if (check_range(460, 489, hwrev_adc)) {
+                env_set("hwrev", "chi");
+                env_set("dtb_name", "rk3326-gameforce-chi.dtb");
+        }
 	/* engineer samples */
 	else {
 		env_set("hwrev", "v00");
