diff -uNr a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
--- a/include/linux/compiler-gcc.h	2020-12-05 14:59:23.000000000 +0900
+++ b/include/linux/compiler-gcc.h	2026-02-14 23:44:21.374426877 +0900
@@ -196,7 +196,7 @@
  * this in the preprocessor, but we can live with this because they're
  * unreleased.  Really, we need to have autoconf for the kernel.
  */
-#define unreachable() __builtin_unreachable()
+/* #define unreachable() __builtin_unreachable() */
 
 /* Mark a function definition as prohibited from being cloned. */
 #define __noclone	__attribute__((__noclone__))
