diff -uNr a/board/rockchip/odroidgoa/odroidgoa.c b/board/rockchip/odroidgoa/odroidgoa.c
--- a/board/rockchip/odroidgoa/odroidgoa.c	2026-03-02 19:27:13.094218400 +0900
+++ b/board/rockchip/odroidgoa/odroidgoa.c	2026-03-02 20:36:47.477987362 +0900
@@ -207,8 +207,8 @@
 	/* show boot logo and version */
 	lcd_show_logo();
 	lcd_setfg_color("grey");
-	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
-	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
+//	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
+//	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
 
 	if (!board_check_autotest()) {
 		board_run_autotest();
