diff --git a/Makefile.include.in b/Makefile.include.in
index b70245d..0537b86 100644
--- a/Makefile.include.in
+++ b/Makefile.include.in
@@ -1,4 +1,4 @@
-AR=ar
+AR=@AR@
 ARFLAGS=crus
 RM=rm -rf
 SHELL=@SHELL@
diff --git a/configure.in b/configure.in
index 7348529..baea8e0 100755
--- a/configure.in
+++ b/configure.in
@@ -22,7 +22,7 @@ AC_DEFUN([XB_FIND_SONAME],
     AC_MSG_CHECKING([for lib$2 soname])
     $1_FILENAME=$($CC -nostdlib -o /dev/null $LDFLAGS -l$2 -Wl,-M 2>/dev/null | grep "^LOAD.*$2" | awk '{V=2; print $V}')
     if [[ ! -z $$1_FILENAME ]]; then
-      $1_SONAME=$(objdump -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
+      $1_SONAME=$($OBJDUMP -p $$1_FILENAME | grep "SONAME.*$2" | awk '{V=2; print $V}')
     fi
   else
     AC_MSG_CHECKING([for lib$2 dylib])
