diff -Naur a/xbmc/filesystem/CurlFile.cpp b/xbmc/filesystem/CurlFile.cpp
--- a/xbmc/filesystem/CurlFile.cpp	2011-03-08 02:49:14.000000000 +0100
+++ b/xbmc/filesystem/CurlFile.cpp	2011-10-17 19:21:04.180783870 +0200
@@ -803,9 +803,9 @@
 // Detect whether we are "online" or not! Very simple and dirty!
 bool CFileCurl::IsInternet(bool checkDNS /* = true */)
 {
-  CStdString strURL = "http://www.google.com";
+  CStdString strURL = "http://www.openelec.tv";
   if (!checkDNS)
-    strURL = "http://74.125.19.103"; // www.google.com ip
+    strURL = "http://212.101.13.10"; // www.openelec.tv ip
 
   bool found = Exists(strURL);
   Close();
