diff -Naur xbmc-pvr-334f9c6/configure.in xbmc-pvr-334f9c6.patch/configure.in
--- xbmc-pvr-334f9c6/configure.in	2011-04-25 02:36:33.000000000 +0200
+++ xbmc-pvr-334f9c6.patch/configure.in	2011-04-25 05:14:49.220848143 +0200
@@ -1493,6 +1493,9 @@
   AC_MSG_NOTICE($external_ffmpeg_disabled)
   USE_EXTERNAL_FFMPEG=0
   AC_DEFINE([PIX_FMT_VDPAU_MPEG4_IN_AVUTIL], [1], [Whether AVUtil defines PIX_FMT_VDPAU_MPEG4.])
+
+  # check for yasm
+  AC_CHECK_PROG(HAVE_YASM,yasm,"yes","no",)
 fi
 
 # VDPAU
@@ -2378,6 +2381,7 @@
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       `if test "$use_neon" = "yes"; then echo --enable-neon; else echo --disable-neon; fi`\
       --target-os=$ffmpeg_target_os \
+      `if test "$HAVE_YASM" = "yes"; then echo --enable-yasm; fi` \
       --disable-muxers \
       --enable-muxer=spdif \
       --enable-muxer=adts \
