diff -Naur fennec-1.0rc3.repack/config/autoconf.mk.in fennec-1.0rc3.repack.patch/config/autoconf.mk.in
--- fennec-1.0rc3.repack/config/autoconf.mk.in	2010-01-27 05:20:55.000000000 +0100
+++ fennec-1.0rc3.repack.patch/config/autoconf.mk.in	2010-02-02 16:06:29.631066553 +0100
@@ -58,14 +58,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
diff -Naur fennec-1.0rc3.repack/js/src/config/autoconf.mk.in fennec-1.0rc3.repack.patch/js/src/config/autoconf.mk.in
--- fennec-1.0rc3.repack/js/src/config/autoconf.mk.in	2010-01-27 05:21:05.000000000 +0100
+++ fennec-1.0rc3.repack.patch/js/src/config/autoconf.mk.in	2010-02-02 16:05:29.363066752 +0100
@@ -61,8 +61,8 @@
 datadir		= @datadir@
 mandir		= @mandir@
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 TOP_DIST	= @TOP_DIST@
 ifneq (,$(filter /%,$(TOP_DIST)))
diff -Naur fennec-1.0rc3.repack/toolkit/mozapps/installer/packager.mk fennec-1.0rc3.repack.patch/toolkit/mozapps/installer/packager.mk
--- fennec-1.0rc3.repack/toolkit/mozapps/installer/packager.mk	2010-01-27 05:21:32.000000000 +0100
+++ fennec-1.0rc3.repack.patch/toolkit/mozapps/installer/packager.mk	2010-02-02 16:10:37.660066944 +0100
@@ -454,7 +454,7 @@
 	  (cd $(DESTDIR)$(installdir) && tar -xf -)
 	$(NSINSTALL) -D $(DESTDIR)$(bindir)
 	$(RM) -f $(DESTDIR)$(bindir)/$(MOZ_APP_NAME)
-	ln -s $(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
+	ln -s $(DESTDIR)$(installdir)/$(MOZ_APP_NAME) $(DESTDIR)$(bindir)
 ifdef INSTALL_SDK # Here comes the hard part
 	$(NSINSTALL) -D $(DESTDIR)$(includedir)
 	(cd $(DIST)/include && tar $(TAR_CREATE_FLAGS) - .) | \
@@ -469,10 +469,10 @@
 	fi
 	(cd $(DIST)/sdk/lib && tar $(TAR_CREATE_FLAGS) - .) | (cd $(DESTDIR)$(sdkdir)/sdk/lib && tar -xf -)
 	$(RM) -f $(DESTDIR)$(sdkdir)/lib $(DESTDIR)$(sdkdir)/bin $(DESTDIR)$(sdkdir)/include $(DESTDIR)$(sdkdir)/include $(DESTDIR)$(sdkdir)/sdk/idl $(DESTDIR)$(sdkdir)/idl
-	ln -s $(sdkdir)/sdk/lib $(DESTDIR)$(sdkdir)/lib
-	ln -s $(installdir) $(DESTDIR)$(sdkdir)/bin
-	ln -s $(includedir) $(DESTDIR)$(sdkdir)/include
-	ln -s $(idldir) $(DESTDIR)$(sdkdir)/idl
+	ln -s $(DESTDIR)$(sdkdir)/sdk/lib $(DESTDIR)$(sdkdir)/lib
+	ln -s $(DESTDIR)$(installdir) $(DESTDIR)$(sdkdir)/bin
+	ln -s $(DESTDIR)$(includedir) $(DESTDIR)$(sdkdir)/include
+	ln -s $(DESTDIR)$(idldir) $(DESTDIR)$(sdkdir)/idl
 endif # INSTALL_SDK
 
 make-sdk:
