diff -Naur fennec-latest.orig/mobile/chrome/content/browser.xul fennec-latest.patch/mobile/chrome/content/browser.xul
--- fennec-latest.orig/mobile/chrome/content/browser.xul	2010-02-08 22:38:14.542981051 +0100
+++ fennec-latest.patch/mobile/chrome/content/browser.xul	2010-02-08 23:13:02.153106002 +0100
@@ -143,37 +143,29 @@
 
   <keyset id="mainKeyset">
     <!-- basic navigation -->
-    <key id="key_back" keycode="VK_LEFT" command="cmd_back" modifiers="control"/>
-    <key id="key_forward" keycode="VK_RIGHT" command="cmd_forward" modifiers="control"/>
-    <key id="key_back2" keycode="VK_BACK" command="cmd_back"/>
-    <key id="key_forward2" keycode="VK_BACK" command="cmd_forward" modifiers="shift"/>
-#ifndef MOZ_PLATFORM_HILDON
+    <key id="key_back" keycode="VK_LEFT" command="cmd_back" modifiers="alt"/>
+    <key id="key_forward" keycode="VK_RIGHT" command="cmd_forward" modifiers="alt"/>
     <key id="key_reload" keycode="VK_F5" command="cmd_reload"/>
-#else
-    <!-- F5 on maemo is "home" which is a task switching key -->
-#endif
-    <key id="key_focusURL" key="l" modifiers="accel" command="cmd_openLocation"/>
+    <key id="key_focusURL" key="l" modifiers="alt" command="cmd_openLocation"/>
 
     <!-- scrolling -->
     <key id="key_pageUp" keycode="VK_UP" modifiers="shift" command="cmd_scrollPageUp"/>
     <key id="key_pageDown" keycode="VK_DOWN" modifiers="shift" command="cmd_scrollPageDown"/>
 
     <!-- misc -->
-    <key id="key_zoomin" keycode="VK_UP" modifiers="accel" command="cmd_zoomin"/>
-    <key id="key_zoomout" keycode="VK_DOWN" modifiers="accel" command="cmd_zoomout"/>
-#ifndef WINCE
-    <key id="key_quit" key="q" modifiers="accel" command="cmd_quit"/>
-    <key id="key_menu" keycode="VK_F4" command="cmd_menu"/>
-    <key id="key_fullscreen" keycode="VK_F6" command="cmd_fullscreen"/>
-#else
-    <key id="key_quit" keycode="VK_F4" command="cmd_quit"/>
-    <key id="key_zoomin2" keycode="VK_F1" command="cmd_zoomin"/>
-    <key id="key_zoomout2" keycode="VK_F2" command="cmd_zoomout"/>
-#endif
+    <key id="key_zoomin" key="+" modifiers="alt" command="cmd_zoomin"/>
+    <key id="key_zoomout" key="-" modifiers="alt" command="cmd_zoomout"/>
+    <key id="key_quit" keycode="VK_F4" modifiers="alt" command="cmd_quit"/>
+    <key id="key_close" key="x" modifiers="alt" command="cmd_close"/>
+    <key id="key_menu" key="m" modifiers="alt" command="cmd_menu"/>
+    <key id="key_actions" key="a" modifiers="alt" command="cmd_actions"/>
+    <key id="key_go" keycode="VK_ENTER" modifiers="alt" command="cmd_go"/>
+    <key id="key_go2" keycode="VK_RETURN" modifiers="alt" command="cmd_go"/>
+    <key id="key_bookmark" key="b" modifiers="alt" command="cmd_star"/>
 
     <!-- tabs -->
-    <key id="key_newTab" key="t" modifiers="accel" command="cmd_newTab"/>
-    <key id="key_closeTab" key="w" modifiers="accel" command="cmd_closeTab"/>
+    <key id="key_newTab" key="t" modifiers="alt" command="cmd_newTab"/>
+    <key id="key_closeTab" key="w" modifiers="alt" command="cmd_closeTab"/>
   </keyset>
 
   <stack flex="1" id="stack">
