diff -Naur mozilla-1.9.2/mobile/components/Makefile.in mozilla-1.9.2.patch/mobile/components/Makefile.in
--- mozilla-1.9.2/mobile/components/Makefile.in	2010-01-27 05:28:53.000000000 +0100
+++ mozilla-1.9.2.patch/mobile/components/Makefile.in	2010-01-31 21:55:01.826947596 +0100
@@ -66,7 +66,6 @@
 	$(NULL)
 
 DIRS = protocols \
-        phone \
         $(NULL)
 
 include $(topsrcdir)/config/rules.mk
