diff -Naur fennec-latest.orig/mobile/chrome/content/browser.xul fennec-latest.patch/mobile/chrome/content/browser.xul
--- fennec-latest.orig/mobile/chrome/content/browser.xul	2010-02-06 23:19:12.000000000 +0100
+++ fennec-latest.patch/mobile/chrome/content/browser.xul	2010-02-08 22:19:50.575981168 +0100
@@ -61,8 +61,7 @@
 #elifdef MOZ_PLATFORM_HILDON
         sizemode="fullscreen"
 #else
-        width="800"
-        height="500"
+        sizemode="fullscreen"
 #endif
         onkeypress="onDebugKeyPress(event);"
         xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"
