--- libconfuse-3.3/configure.ac	2020-06-25 05:21:59.000000000 +0000
+++ libconfuse-3.3/configure.ac	2020-12-29 02:04:36.074750823 +0000
@@ -29,7 +29,8 @@
 # Unfortunately CentOS7, RHEL7 ships 0.18.2.1, so for best compat.
 # level at this point in time we set 0.18.2.
 AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION([0.18.2])
+AM_GNU_GETTEXT_REQUIRE_VERSION([0.18.2])
 
 # Checks for header files.
 AC_HEADER_STDC
