--- a/build/toolchain/linux/BUILD.gn	2016-11-09 19:49:39.000000000 -0800
+++ b/build/toolchain/linux/BUILD.gn	2016-11-10 20:33:18.200587987 -0800
@@ -102,13 +102,14 @@
 }
 
 gcc_toolchain("x64") {
-  cc = "gcc"
-  cxx = "g++"
+  toolprefix = "x86_64-libreelec-linux-gnu-"
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
 
-  readelf = "readelf"
-  nm = "nm"
-  ar = "ar"
+  ar = "${toolprefix}ar"
   ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
 
   toolchain_args = {
     current_cpu = "x64"
