diff -Naur a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
--- a/tools/gn/bootstrap/bootstrap.py	2017-08-10 17:18:53.767165155 +0200
+++ b/tools/gn/bootstrap/bootstrap.py	2017-08-10 17:23:28.668750638 +0200
@@ -657,7 +657,7 @@
         'base/allocator/allocator_shim.cc',
         'base/allocator/allocator_shim_default_dispatch_to_glibc.cc',
       ])
-      libs.extend(['-lrt', '-latomic'])
+      libs.extend(['-lrt'])
       static_libraries['libevent']['include_dirs'].extend([
           os.path.join(SRC_ROOT, 'base', 'third_party', 'libevent', 'linux')
       ])
