--- a/chrome/browser/ui/aura/chrome_browser_main_extra_parts_aura.cc	2015-12-15 12:05:07.000000000 -0800
+++ b/chrome/browser/ui/aura/chrome_browser_main_extra_parts_aura.cc	2016-01-12 16:08:45.701976368 -0800
@@ -140,7 +140,6 @@
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
   // Now that we have some minimal ui initialized, check to see if we're
   // running as root and bail if we are.
-  DetectRunningAsRoot();
 #endif
 }
 
