--- a/chrome/browser/ui/libgtkui/gtk_ui.h	2017-10-19 00:15:47.451704332 +0200
+++ b/chrome/browser/ui/libgtkui/gtk_ui.h	2017-10-18 23:53:19.328373980 +0200
@@ -14,6 +14,7 @@
 #include "base/observer_list.h"
 #include "chrome/browser/ui/libgtkui/gtk_signal.h"
 #include "chrome/browser/ui/libgtkui/libgtkui_export.h"
+#include "chrome/browser/ui/libgtkui/nav_button_layout_manager.h"
 #include "ui/gfx/color_utils.h"
 #include "ui/views/linux_ui/linux_ui.h"
 #include "ui/views/window/frame_buttons.h"
