--- chromium-59.0.3071.109/./chrome/browser/ui/views/chrome_browser_main_extra_parts_views.cc.org	2017-06-26 20:32:34.907628561 +0200
+++ chromium-59.0.3071.109/./chrome/browser/ui/views/chrome_browser_main_extra_parts_views.cc	2017-06-26 20:31:05.855206203 +0200
@@ -99,7 +99,7 @@
     IMEDriver::Register();
 #endif
 
-#if defined(OS_LINUX) && !defined(OS_CHROMEOS)
+#if 0
   // On the Linux desktop, we want to prevent the user from logging in as root,
   // so that we don't destroy the profile. Now that we have some minimal ui
   // initialized, check to see if we're running as root and bail if we are.
