--- a/tools/gn/bootstrap/bootstrap.py	2017-08-10 21:13:27.316180673 +0200
+++ b/tools/gn/bootstrap/bootstrap.py	2017-08-10 21:13:23.544160535 +0200
@@ -486,6 +486,7 @@
       'base/synchronization/lock.cc',
       'base/sys_info.cc',
       'base/task_runner.cc',
+      'base/task_scheduler/environment_config.cc',
       'base/task_scheduler/delayed_task_manager.cc',
       'base/task_scheduler/post_task.cc',
       'base/task_scheduler/priority_queue.cc',
