diff -Naur a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
--- a/tools/gn/bootstrap/bootstrap.py	2016-10-20 15:02:19.000000000 -0700
+++ b/tools/gn/bootstrap/bootstrap.py	2016-11-10 09:26:36.026611661 -0800
@@ -563,7 +563,7 @@
     }
 
   if is_linux:
-    libs.extend(['-lrt', '-latomic'])
+    libs.extend(['-lrt'])
     ldflags.extend(['-pthread'])
 
     static_libraries['xdg_user_dirs'] = {
