From 387aae00e0b88d2665a535bb1f5974bfebe57f23 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Tue, 2 May 2017 18:57:19 +0200
Subject: [PATCH 23/37] ASoC: hdmi-codec: fix I2S audio in Kodi

---
 sound/soc/codecs/hdmi-codec.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sound/soc/codecs/hdmi-codec.c b/sound/soc/codecs/hdmi-codec.c
index cb78d8971b41..9ebca57014e4 100644
--- a/sound/soc/codecs/hdmi-codec.c
+++ b/sound/soc/codecs/hdmi-codec.c
@@ -758,7 +758,6 @@ static struct snd_soc_dai_driver hdmi_i2s_dai = {
 		.channels_max = 8,
 		.rates = HDMI_RATES,
 		.formats = I2S_FORMATS,
-		.sig_bits = 24,
 	},
 	.ops = &hdmi_dai_ops,
 	.pcm_new = hdmi_codec_pcm_new,
