From ecadd1f950d8dce8a53ced1c2f106e4984bda859 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Wed, 9 Aug 2017 09:28:49 +0200
Subject: [PATCH 34/35] drm/edid: use cea modes when no edid

---
 drivers/gpu/drm/drm_edid.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/drivers/gpu/drm/drm_edid.c b/drivers/gpu/drm/drm_edid.c
index 7461348954ed..f89db040b641 100644
--- a/drivers/gpu/drm/drm_edid.c
+++ b/drivers/gpu/drm/drm_edid.c
@@ -4489,6 +4489,29 @@ int drm_add_edid_modes(struct drm_connector *connector, struct edid *edid)
 }
 EXPORT_SYMBOL(drm_add_edid_modes);
 
+static int drm_add_hdmimodes_noedid(struct drm_connector *connector)
+{
+	int i, num_modes = 0;
+	struct drm_display_mode *mode;
+	struct drm_device *dev = connector->dev;
+
+	for (i = 0; i < ARRAY_SIZE(edid_cea_modes); i++) {
+		const struct drm_display_mode *ptr = &edid_cea_modes[i];
+		if (ptr->hdisplay > 1920 ||
+		    ptr->vdisplay < 480 ||
+		    ptr->flags & DRM_MODE_FLAG_INTERLACE ||
+		    drm_mode_vrefresh(ptr) > 60 ||
+		    drm_mode_vrefresh(ptr) < 50)
+			continue;
+		mode = drm_mode_duplicate(dev, ptr);
+		if (mode) {
+			drm_mode_probed_add(connector, mode);
+			num_modes++;
+		}
+	}
+	return num_modes;
+}
+
 /**
  * drm_add_modes_noedid - add modes for the connectors without EDID
  * @connector: connector we're probing
@@ -4507,6 +4530,7 @@ int drm_add_modes_noedid(struct drm_connector *connector,
 	struct drm_display_mode *mode;
 	struct drm_device *dev = connector->dev;
 
+	return drm_add_hdmimodes_noedid(connector);
 	count = ARRAY_SIZE(drm_dmt_modes);
 	if (hdisplay < 0)
 		hdisplay = 0;
