Index: mesa/src/mesa/x86/glapi_x86.S
===================================================================
--- mesa.orig/src/mesa/x86/glapi_x86.S	2009-06-29 14:43:07.000000000 +0300
+++ mesa/src/mesa/x86/glapi_x86.S	2009-06-29 14:45:38.000000000 +0300
@@ -148,7 +148,6 @@
 
 		ALIGNTEXT16
 		GLOBL GLNAME(gl_dispatch_functions_start)
-		HIDDEN(GLNAME(gl_dispatch_functions_start))
 GLNAME(gl_dispatch_functions_start):
 
 	GL_STUB(NewList, _gloffset_NewList, NewList@8)
