diff -Nurp mesa-7.6.0~git20090817.7c422387/src/mesa/drivers/dri/r300/r300_draw.c mesa-7.6.0~git20090817.7c422387-working/src/mesa/drivers/dri/r300/r300_draw.c
--- mesa-7.6.0~git20090817.7c422387/src/mesa/drivers/dri/r300/r300_draw.c	2009-08-17 07:30:14.000000000 -0700
+++ mesa-7.6.0~git20090817.7c422387-working/src/mesa/drivers/dri/r300/r300_draw.c	2009-08-26 13:38:46.000000000 -0700
@@ -108,12 +108,13 @@ static void r300FixupIndexBuffer(GLconte
 
 #if MESA_BIG_ENDIAN
 	} else { /* if (mesa_ind_buf->type == GL_UNSIGNED_SHORT) */
+		GLuint size;
 		GLushort *in = (GLushort *)src_ptr;
 		size = sizeof(GLushort) * ((mesa_ind_buf->count + 1) & ~1);
 
 		radeonAllocDmaRegion(&r300->radeon, &r300->ind_buf.bo, &r300->ind_buf.bo_offet, size, 4);
 
-		assert(r300->ind_buf.bo->ptr != NULL)
+		assert(r300->ind_buf.bo->ptr != NULL);
 		out = (GLuint *)ADD_POINTERS(r300->ind_buf.bo->ptr, r300->ind_buf.bo_offset);
 
 		for (i = 0; i + 1 < mesa_ind_buf->count; i += 2) {
