diff --git a/src/mesa/drivers/dri/intel/intel_screen.c b/src/mesa/drivers/dri/intel/intel_screen.c
index cf09fad..572a28b 100644
--- a/src/mesa/drivers/dri/intel/intel_screen.c
+++ b/src/mesa/drivers/dri/intel/intel_screen.c
@@ -583,8 +583,8 @@ intel_init_bufmgr(intelScreenPrivate *intelScreen)
       if (gem_disable) {
 	 fprintf(stderr, "GEM disabled.  Using classic.\n");
       } else {
-	 fprintf(stderr, "Failed to initialize GEM.  "
-		 "Falling back to classic.\n");
+//	 fprintf(stderr, "Failed to initialize GEM.  "
+//		 "Falling back to classic.\n");
       }
 
       if (intelScreen->tex.size == 0) {
