diff --git a/src/amd/common/ac_surface.c b/src/amd/common/ac_surface.c
index 5ef3a0d7e84..7ee88afec76 100644
--- a/src/amd/common/ac_surface.c
+++ b/src/amd/common/ac_surface.c
@@ -2729,7 +2729,7 @@ bool ac_surface_override_offset_stride(const struct radeon_info *info, struct ra
       }
    }
 
-   if (offset & ((1 << surf->alignment_log2) - 1) ||
+   if ((surf->alignment && offset & ((1 << surf->alignment_log2) - 1)) ||
        offset >= UINT64_MAX - surf->total_size)
       return false;
 
