diff -Naur mesa-mesa-21.0.2-old/src/amd/common/ac_surface.c mesa-mesa-21.0.2-new/src/amd/common/ac_surface.c
--- mesa-mesa-21.0.2-old/src/amd/common/ac_surface.c	2021-04-07 18:35:30.000000000 +0200
+++ mesa-mesa-21.0.2-new/src/amd/common/ac_surface.c	2021-04-11 11:35:23.939998177 +0200
@@ -2878,10 +2826,6 @@
       }
    }
 
-   if (offset & (surf->alignment - 1) ||
-       offset >= UINT64_MAX - surf->total_size)
-      return false;
-
    if (surf->htile_offset)
       surf->htile_offset += offset;
    if (surf->fmask_offset)
