diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/r200/r200_cmdbuf.c Mesa-7.10.1-new/src/mesa/drivers/dri/r200/r200_cmdbuf.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/r200/r200_cmdbuf.c	2011-03-02 17:30:39.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/r200/r200_cmdbuf.c	2011-03-02 17:31:59.000000000 -0800
@@ -47,9 +47,8 @@
  */
 #define insert_at_tail_if(atom_list, atom) \
    do { \
-      struct radeon_state_atom* __atom = (atom); \
-      if (__atom->check) \
-	 insert_at_tail((atom_list), __atom); \
+      if ((atom)->check) \
+	 insert_at_tail((atom_list), (atom)); \
    } while(0)
 
 void r200SetUpAtomList( r200ContextPtr rmesa )
