diff -Naur Mesa-7.10.1-old/src/gallium/drivers/nv50/nv50_context.h Mesa-7.10.1-new/src/gallium/drivers/nv50/nv50_context.h
--- Mesa-7.10.1-old/src/gallium/drivers/nv50/nv50_context.h	2010-12-15 15:02:44.000000000 -0800
+++ Mesa-7.10.1-new/src/gallium/drivers/nv50/nv50_context.h	2011-03-02 17:31:09.000000000 -0800
@@ -15,6 +15,8 @@
 
 #include "nouveau/nouveau_winsys.h"
 #include "nouveau/nouveau_gldefs.h"
+/* Hack to silence warnings caused by nouveau/nouveau_stateobj.h using BEGIN_BIND marcro */
+#include <nouveau/nv04_pushbuf.h>
 #include "nouveau/nouveau_stateobj.h"
 #include "nv50_reg.h"
 
diff -Naur Mesa-7.10.1-old/src/gallium/drivers/nvfx/nv04_2d.c Mesa-7.10.1-new/src/gallium/drivers/nvfx/nv04_2d.c
--- Mesa-7.10.1-old/src/gallium/drivers/nvfx/nv04_2d.c	2011-01-25 15:52:45.000000000 -0800
+++ Mesa-7.10.1-new/src/gallium/drivers/nvfx/nv04_2d.c	2011-03-02 17:31:09.000000000 -0800
@@ -39,6 +39,7 @@
 #include <nouveau/nouveau_bo.h>
 #include <nouveau/nouveau_notifier.h>
 #include <nouveau/nouveau_grobj.h>
+#include <nouveau/nv04_pushbuf.h>
 #include "nv04_2d.h"
 
 #include "nouveau/nv_object.xml.h"
diff -Naur Mesa-7.10.1-old/src/gallium/drivers/nvfx/nvfx_buffer.c Mesa-7.10.1-new/src/gallium/drivers/nvfx/nvfx_buffer.c
--- Mesa-7.10.1-old/src/gallium/drivers/nvfx/nvfx_buffer.c	2010-12-15 15:02:44.000000000 -0800
+++ Mesa-7.10.1-new/src/gallium/drivers/nvfx/nvfx_buffer.c	2011-03-02 17:31:09.000000000 -0800
@@ -1,4 +1,3 @@
-
 #include "util/u_inlines.h"
 #include "util/u_memory.h"
 #include "util/u_math.h"
diff -Naur Mesa-7.10.1-old/src/gallium/drivers/nvfx/nvfx_screen.c Mesa-7.10.1-new/src/gallium/drivers/nvfx/nvfx_screen.c
--- Mesa-7.10.1-old/src/gallium/drivers/nvfx/nvfx_screen.c	2011-01-25 15:52:45.000000000 -0800
+++ Mesa-7.10.1-new/src/gallium/drivers/nvfx/nvfx_screen.c	2011-03-02 17:31:09.000000000 -0800
@@ -10,6 +10,8 @@
 #include "nvfx_resource.h"
 #include "nvfx_tex.h"
 
+#include "nouveau/nv04_pushbuf.h"
+
 #define NV30_3D_CHIPSET_3X_MASK 0x00000003
 #define NV34_3D_CHIPSET_3X_MASK 0x00000010
 #define NV35_3D_CHIPSET_3X_MASK 0x000001e0
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv04_driver.h Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv04_driver.h
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv04_driver.h	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv04_driver.h	2011-03-02 17:31:09.000000000 -0800
@@ -28,6 +28,7 @@
 #define __NV04_DRIVER_H__
 
 #include "nv04_context.h"
+#include "nv04_pushbuf.h"
 
 enum {
 	NOUVEAU_STATE_BLEND = NUM_NOUVEAU_STATE,
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_render.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_render.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_render.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_render.c	2011-03-02 17:31:09.000000000 -0800
@@ -27,6 +27,7 @@
 #include "nouveau_driver.h"
 #include "nouveau_context.h"
 #include "nv10_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 #define NUM_VERTEX_ATTRS 8
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_fb.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_fb.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_fb.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_fb.c	2011-03-02 17:31:09.000000000 -0800
@@ -30,6 +30,7 @@
 #include "nouveau_util.h"
 #include "nv_object.xml.h"
 #include "nv10_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 static inline unsigned
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_frag.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_frag.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_frag.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_frag.c	2011-03-02 17:31:09.000000000 -0800
@@ -30,6 +30,7 @@
 #include "nv10_3d.xml.h"
 #include "nouveau_util.h"
 #include "nv10_driver.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 #define RC_IN_SHIFT_A	24
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_polygon.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_polygon.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_polygon.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_polygon.c	2011-03-02 17:31:09.000000000 -0800
@@ -29,6 +29,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv10_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_raster.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_raster.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_raster.c	2011-01-25 15:52:45.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_raster.c	2011-03-02 17:31:09.000000000 -0800
@@ -29,6 +29,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv10_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_tex.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_tex.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_tex.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_tex.c	2011-03-02 17:31:09.000000000 -0800
@@ -30,6 +30,7 @@
 #include "nouveau_texture.h"
 #include "nv10_3d.xml.h"
 #include "nouveau_util.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_tnl.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_tnl.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv10_state_tnl.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv10_state_tnl.c	2011-03-02 17:31:09.000000000 -0800
@@ -29,6 +29,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv10_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_render.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_render.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_render.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_render.c	2011-03-02 17:31:09.000000000 -0800
@@ -27,6 +27,7 @@
 #include "nouveau_driver.h"
 #include "nouveau_context.h"
 #include "nv20_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 #define NUM_VERTEX_ATTRS 16
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_fb.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_fb.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_fb.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_fb.c	2011-03-02 17:31:09.000000000 -0800
@@ -30,6 +30,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv20_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 static inline unsigned
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_frag.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_frag.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_frag.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_frag.c	2011-03-02 17:31:09.000000000 -0800
@@ -28,6 +28,7 @@
 #include "nouveau_context.h"
 #include "nv20_3d.xml.h"
 #include "nv10_driver.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_polygon.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_polygon.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_polygon.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_polygon.c	2011-03-02 17:31:09.000000000 -0800
@@ -28,6 +28,7 @@
 #include "nouveau_context.h"
 #include "nouveau_gldefs.h"
 #include "nv20_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_raster.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_raster.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_raster.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_raster.c	2011-03-02 17:31:09.000000000 -0800
@@ -29,6 +29,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv20_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_tex.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_tex.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_tex.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_tex.c	2011-03-02 17:31:09.000000000 -0800
@@ -30,6 +30,7 @@
 #include "nouveau_texture.h"
 #include "nv20_3d.xml.h"
 #include "nouveau_util.h"
+#include "nv04_driver.h"
 #include "nv20_driver.h"
 
 void
diff -Naur Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_tnl.c Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_tnl.c
--- Mesa-7.10.1-old/src/mesa/drivers/dri/nouveau/nv20_state_tnl.c	2010-12-14 13:43:15.000000000 -0800
+++ Mesa-7.10.1-new/src/mesa/drivers/dri/nouveau/nv20_state_tnl.c	2011-03-02 17:31:09.000000000 -0800
@@ -29,6 +29,7 @@
 #include "nouveau_gldefs.h"
 #include "nouveau_util.h"
 #include "nv20_3d.xml.h"
+#include "nv04_driver.h"
 #include "nv10_driver.h"
 #include "nv20_driver.h"
 
