diff -Naur Mesa-7.10.2-old/src/gallium/drivers/llvmpipe/Makefile Mesa-7.10.2-new/src/gallium/drivers/llvmpipe/Makefile
--- Mesa-7.10.2-old/src/gallium/drivers/llvmpipe/Makefile	2011-01-04 09:44:17.000000000 -0800
+++ Mesa-7.10.2-new/src/gallium/drivers/llvmpipe/Makefile	2011-05-05 15:48:06.000000000 -0700
@@ -71,7 +71,7 @@
 
 LDFLAGS += $(LLVM_LDFLAGS)
 LIBS += -L../../auxiliary/ -lgallium libllvmpipe.a $(LLVM_LIBS) $(GL_LIB_DEPS)
-LD=g++
+LD = $(CXX)
 
 $(PROGS): lp_test_main.o libllvmpipe.a
 
