diff -Naur mesa-10.6.7/src/mesa/drivers/dri/i965/brw_meta_stencil_blit.c mesa-10.6.7.patch/src/mesa/drivers/dri/i965/brw_meta_stencil_blit.c
--- mesa-10.6.7/src/mesa/drivers/dri/i965/brw_meta_stencil_blit.c	2015-09-10 19:40:31.000000000 +0200
+++ mesa-10.6.7.patch/src/mesa/drivers/dri/i965/brw_meta_stencil_blit.c	2015-09-12 19:51:13.946585268 +0200
@@ -414,6 +414,12 @@
    GLenum target;
 
    _mesa_meta_fb_tex_blit_begin(ctx, &blit);
+   /* XXX: Pretend to support stencil textures so _mesa_base_tex_format()
+    * returns a valid format.  When we properly support the extension, we
+    * should remove this.
+    */
+   assert(ctx->Extensions.ARB_texture_stencil8 == false);
+   ctx->Extensions.ARB_texture_stencil8 = true;
 
    _mesa_GenFramebuffers(1, &fbo);
    /* Force the surface to be configured for level zero. */
@@ -451,6 +457,7 @@
    _mesa_DrawArrays(GL_TRIANGLE_FAN, 0, 4);
 
 error:
+   ctx->Extensions.ARB_texture_stencil8 = false;
    _mesa_meta_fb_tex_blit_end(ctx, target, &blit);
    _mesa_meta_end(ctx);
 
