diff -up Mesa-9.1.1/src/glsl/builtin_compiler/Makefile.am.cross Mesa-9.1.1/src/glsl/builtin_compiler/Makefile.am
--- Mesa-9.1.1/src/glsl/builtin_compiler/Makefile.am.cross	2013-03-28 14:56:22.990746189 +0100
+++ Mesa-9.1.1/src/glsl/builtin_compiler/Makefile.am	2013-03-28 14:56:32.483577384 +0100
@@ -64,13 +64,13 @@ AM_CXXFLAGS = $(AM_CFLAGS)
 include ../Makefile.sources
 
 noinst_PROGRAMS = builtin_compiler
-noinst_LTLIBRARIES = libglslcore.la libglcpp.la
+noinst_LIBRARIES = libglslcore.a libglcpp.a
 
-libglcpp_la_SOURCES =					\
+libglcpp_a_SOURCES =					\
 	$(LIBGLCPP_GENERATED_FILES)			\
 	$(LIBGLCPP_FILES)
 
-libglslcore_la_SOURCES =				\
+libglslcore_a_SOURCES =				\
 	$(BUILTIN_COMPILER_GENERATED_CXX_FILES)		\
 	$(LIBGLSL_FILES)
 
@@ -81,4 +81,4 @@ builtin_compiler_SOURCES = \
 	$(top_srcdir)/src/mesa/program/symbol_table.c	\
 	$(BUILTIN_COMPILER_CXX_FILES)			\
 	$(GLSL_COMPILER_CXX_FILES)
-builtin_compiler_LDADD = libglslcore.la libglcpp.la
+builtin_compiler_LDADD = libglslcore.a libglcpp.a
diff -up Mesa-9.1.1/src/glsl/Makefile.am.cross Mesa-9.1.1/src/glsl/Makefile.am
--- Mesa-9.1.1/src/glsl/Makefile.am.cross	2013-03-28 14:56:26.565682626 +0100
+++ Mesa-9.1.1/src/glsl/Makefile.am	2013-03-28 14:59:13.109723407 +0100
@@ -73,14 +73,10 @@ tests_ralloc_test_LDADD =				\
 	$(top_builddir)/src/gtest/libgtest.la		\
 	$(PTHREAD_LIBS)
 
-if CROSS_COMPILING
 libglcpp_la_SOURCES =					\
 	glcpp/glcpp-lex.c				\
 	glcpp/glcpp-parse.c				\
 	$(LIBGLCPP_FILES)
-else
-libglcpp_la_LIBADD = builtin_compiler/libglcpp.la
-endif
 
 glcpp_glcpp_SOURCES =					\
 	glcpp/glcpp.c					\
@@ -89,15 +85,10 @@ glcpp_glcpp_LDADD = libglcpp.la
 
 libglsl_la_SOURCES = builtin_function.cpp
 libglsl_la_LIBADD = libglcpp.la
-if CROSS_COMPILING
 libglsl_la_SOURCES +=					\
 	glsl_lexer.cpp					\
 	glsl_parser.cpp					\
 	$(LIBGLSL_FILES)
-else
-libglsl_la_LIBADD +=					\
-	builtin_compiler/libglslcore.la
-endif
 
 glsl_test_SOURCES = \
 	$(top_srcdir)/src/mesa/main/hash_table.c \
