diff -Naur mesa-a1b737d72f1fec1d967255f4f87b5208a857c87f/configure.ac mesa-a1b737d72f1fec1d967255f4f87b5208a857c87f.patch/configure.ac
--- mesa-a1b737d72f1fec1d967255f4f87b5208a857c87f/configure.ac	2011-11-01 13:28:47.000000000 +0100
+++ mesa-a1b737d72f1fec1d967255f4f87b5208a857c87f.patch/configure.ac	2011-11-01 15:56:10.046619433 +0100
@@ -1738,7 +1738,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags|sed 's/-DNDEBUG\>//g'`
-	LLVM_LIBS="`$LLVM_CONFIG --libs`"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	DEFINES="$DEFINES -D__STDC_CONSTANT_MACROS"
