commit d8fcfd06c417daee69335ff867d72c23b87f5fc3
Author: fritsch <peter.fruehberger@gmail.com>
Date:   Thu Oct 3 08:50:50 2013 +0200

    Fix compilation of glinterop after e9b410b54dd893568a50af661fdbba12b4573ed9

diff --git a/src/mesa/state_tracker/st_vdpau.c b/src/mesa/state_tracker/st_vdpau.c
index 406df0f..9b165ee 100644
--- a/src/mesa/state_tracker/st_vdpau.c
+++ b/src/mesa/state_tracker/st_vdpau.c
@@ -154,7 +154,7 @@ st_vdpau_map_surface(struct gl_context *ctx, GLenum target, GLenum access,
    stObj->depth0 = 1;
    stObj->surface_format = res->format;
 
-   _mesa_dirty_texobj(ctx, texObj, GL_TRUE);
+   _mesa_dirty_texobj(ctx, texObj);
 }
 
 static void
@@ -170,7 +170,7 @@ st_vdpau_unmap_surface(struct gl_context *ctx, GLenum target, GLenum access,
    pipe_sampler_view_reference(&stObj->sampler_view, NULL);
    pipe_resource_reference(&stImage->pt, NULL);
 
-   _mesa_dirty_texobj(ctx, texObj, GL_TRUE);
+   _mesa_dirty_texobj(ctx, texObj);
 }
 
 void

