diff -Naur mesa-a56a732c6991f47d63f5ccbb27a45467541c43f8/configure.ac mesa-a56a732c6991f47d63f5ccbb27a45467541c43f8.patch/configure.ac
--- mesa-a56a732c6991f47d63f5ccbb27a45467541c43f8/configure.ac	2012-01-08 05:01:59.000000000 +0100
+++ mesa-a56a732c6991f47d63f5ccbb27a45467541c43f8.patch/configure.ac	2012-01-08 15:21:49.167895590 +0100
@@ -1771,7 +1771,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version | sed 's/svn.*//g'`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags|sed -e 's/-DNDEBUG\>//g' -e 's/-pedantic//g'`
-	LLVM_LIBS="`$LLVM_CONFIG --libs`"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	DEFINES="$DEFINES -D__STDC_CONSTANT_MACROS"
