diff -Naur Mesa-7.11-rc2-old/configure.ac Mesa-7.11-rc2-new/configure.ac
--- Mesa-7.11-rc2-old/configure.ac	2011-07-19 16:39:56.000000000 -0700
+++ Mesa-7.11-rc2-new/configure.ac	2011-07-21 10:08:33.000000000 -0700
@@ -1737,7 +1737,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS llvmpipe"
