diff --git a/bin/mklib b/bin/mklib
index 9bac29e..b8fac2c 100755
--- a/bin/mklib
+++ b/bin/mklib
@@ -49,8 +49,8 @@ expand_archives() {
                     /*) ;;
                     *)  FILE="$ORIG_DIR/$FILE" ;;
                 esac
-                MEMBERS=`ar t $FILE`
-                ar x $FILE
+                MEMBERS=`$AR t $FILE`
+                $AR x $FILE
                 for MEMBER in $MEMBERS ; do
                     NEWFILES="$NEWFILES $DIR/$MEMBER"
                 done
@@ -87,7 +87,7 @@ make_ar_static_lib() {
     rm -f ${LIBNAME}
 
     # make static lib
-    ar ${OPTS} ${LIBNAME} ${OBJECTS}
+    $AR ${OPTS} ${LIBNAME} ${OBJECTS}
 
     # run ranlib
     if [ ${RANLIB} = 1 ] ; then
@@ -313,9 +313,9 @@ case $ARCH in
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
