diff -Naur Mesa-7.10.2-old/configure.ac Mesa-7.10.2-new/configure.ac
--- Mesa-7.10.2-old/configure.ac	2011-02-11 14:49:47.000000000 -0800
+++ Mesa-7.10.2-new/configure.ac	2011-05-05 16:31:18.000000000 -0700
@@ -1551,7 +1551,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs jit interpreter nativecodegen bitwriter` -lstdc++"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION -lstdc++"
 
 	if test "x$HAS_UDIS86" != xno; then
 	    LLVM_LIBS="$LLVM_LIBS -ludis86"
