diff -Naur xbmc-24229/xbmc/Settings.h xbmc-24229.patch/xbmc/Settings.h
--- xbmc-24229/xbmc/Settings.h	2009-11-03 10:50:13.106418518 +0100
+++ xbmc-24229.patch/xbmc/Settings.h	2009-11-03 12:50:04.017521297 +0100
@@ -27,7 +27,7 @@
 #define DEFAULT_VSYNC       VSYNC_DISABLED
 #define DEFAULT_THUMB_SIZE  256
 #else  // MID
-#define DEFAULT_SKIN        "Confluence"
+#define DEFAULT_SKIN        "Confluence"
 #if defined(__APPLE__) || defined(_WIN32)
 #define DEFAULT_VSYNC       VSYNC_ALWAYS
 #else
