diff -Naur xbmc/xbmc/linux/NetworkLinux.cpp xbmc.patch/xbmc/linux/NetworkLinux.cpp
--- xbmc/xbmc/linux/NetworkLinux.cpp	2009-10-21 07:57:27.000000000 +0200
+++ xbmc.patch/xbmc/linux/NetworkLinux.cpp	2009-11-05 03:20:12.898254181 +0100
@@ -549,7 +549,7 @@
    assignment = NETWORK_DISABLED;
 
 #ifndef __APPLE__
-   FILE* fp = fopen("/etc/network/interfaces", "r");
+   FILE* fp = fopen("/storage/.config/interfaces", "r");
    if (!fp)
    {
       // TODO
@@ -625,7 +625,7 @@
 void CNetworkInterfaceLinux::SetSettings(NetworkAssignment& assignment, CStdString& ipAddress, CStdString& networkMask, CStdString& defaultGateway, CStdString& essId, CStdString& key, EncMode& encryptionMode)
 {
 #ifndef __APPLE__
-   FILE* fr = fopen("/etc/network/interfaces", "r");
+   FILE* fr = fopen("/storage/.config/interfaces", "r");
    if (!fr)
    {
       // TODO
@@ -701,13 +701,13 @@
    fclose(fw);
 
    // Rename the file
-   if (rename("/tmp/interfaces.temp", "/etc/network/interfaces") < 0)
+   if (rename("/tmp/interfaces.temp", "/storage/.config/interfaces") < 0)
    {
       // TODO
       return;
    }
 
-   std::string cmd = "/sbin/ifdown " + GetName();
+   std::string cmd = "/sbin/ifdown -i /storage/.config/interfaces " + GetName();
    if (system(cmd.c_str()) != 0)
      CLog::Log(LOGERROR, "Unable to stop interface %s", GetName().c_str());
    else
@@ -715,7 +715,7 @@
 
    if (assignment != NETWORK_DISABLED)
    {
-      cmd = "/sbin/ifup " + GetName();
+      cmd = "/sbin/ifup -i /storage/.config/interfaces " + GetName();
       if (system(cmd.c_str()) != 0)
         CLog::Log(LOGERROR, "Unable to start interface %s", GetName().c_str());
       else
