diff --git a/Makefile b/Makefile
index 278af2b..996dc77 100644
--- a/Makefile
+++ b/Makefile
@@ -85,8 +85,9 @@ OBJS = CAPMT.o DeCSA.o DeCsaTSBuffer.o dll.o DVBAPI.o DVBAPISetup.o SCDeviceProb
 
 ifndef LIBDVBCSA
 # FFdeCSA
-PARALLEL   ?= PARALLEL_128_SSE2
-CSAFLAGS   ?= -fexpensive-optimizations -funroll-loops -mmmx -msse -msse2 -msse3
+CPUOPT     ?= pentium
+PARALLEL   ?= PARALLEL_32_INT
+CSAFLAGS   ?= -Wall -fPIC -g -O3 -mmmx -fomit-frame-pointer -fexpensive-optimizations -funroll-loops
 FFDECSADIR  = FFdecsa
 FFDECSA     = $(FFDECSADIR)/FFdecsa.o
 DECSALIB    = $(FFDECSA)
