From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tommi=20M=C3=A4kitalo?= <tommi@tntnet.org>
Date: Fri, 31 Jan 2021 17:20:24 +0100
Subject: [PATCH] superflous copy constuctor

---
 include/cxxtools/char.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/include/cxxtools/char.h b/include/cxxtools/char.h
index 1e637317..3a4ed62a 100644
--- a/include/cxxtools/char.h
+++ b/include/cxxtools/char.h
@@ -114,9 +112,6 @@
                 return Char(0);
             }
 
-            Char& operator=(const Char& ch)
-            { _value = ch._value; return *this; }
-
             /**
              * @brief Returns the internal value (unsigned 32 bits) of this character.
              * @return The 32-bit-value of this character.
