Index: cxxtools-2.2/src/Makefile.am
===================================================================
--- cxxtools-2.2.orig/src/Makefile.am	2013-05-05 13:56:03.225204926 +0300
+++ cxxtools-2.2/src/Makefile.am	2013-05-05 13:56:14.017219885 +0300
@@ -215,4 +215,4 @@
 endif
 
 libcxxtools_la_LIBADD = $(LIBICONV)
-libcxxtools_la_LDFLAGS = -version-info @sonumber@ @SHARED_LIB_FLAG@
+libcxxtools_la_LDFLAGS = -version-info @sonumber@
