Index: cxxtools-2.2/include/cxxtools/char.h
===================================================================
--- cxxtools-2.2.orig/include/cxxtools/char.h	2013-05-05 14:18:00.180572107 +0300
+++ cxxtools-2.2/include/cxxtools/char.h	2013-05-05 14:18:00.176571966 +0300
@@ -148,6 +148,12 @@
             friend bool operator==(char a, const Char& b)
             { return a == b.value(); }
 
+            //! @brief Returns $true$ if the a and b are the same character; $false$ otherwise.
+            //! @return $true$ if the a and b are the same character; $false$ otherwise.
+            friend bool operator==(const Char& a, unsigned int b)
+            { return a.value() == b; }
+
+
             //! @brief Returns $true$ if the a and b are not the same character; $false$ otherwise.
             //! @return $true$ if the a and b are not the same character; $false$ otherwise.
             friend bool operator!=(const Char& a, const Char& b)
