Fix error whilst initialising SDIO card.

--- a/arch/arm/boot/dts/imx6qdl-wandboard-revb1.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-wandboard-revb1.dtsi
@@ -28,9 +28,17 @@
 	};
 };
 
+/ {
+	pwrseq_usdhc2: usdhc2pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&gpio5 2 GPIO_ACTIVE_LOW>;	/* WL_RST_N */
+	};
+};
+
 &usdhc2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc2>;
+	mmc-pwrseq = <&pwrseq_usdhc2>;
 	non-removable;
 	status = "okay";
 };
