From 9f1151cd9813477dafb939f3b0234a654e80a3b0 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Tue, 28 Jan 2020 03:22:15 +0000
Subject: [PATCH] fix API change since autoconf-archive-2019.01.06

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index e69542e..19d1e31 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,7 @@ ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = COPYING.README README.android
 
-@CODE_COVERAGE_RULES@
+include $(top_srcdir)/aminclude_static.am
 
 # If git describe works, force autoconf to run in order to make sure we have the
 # current version number from git in the resulting configure script.
-- 
2.7.4

