diff -Naur ffmpeg-export-2009-06-13.orig/configure ffmpeg-export-2009-06-13/configure
--- ffmpeg-export-2009-06-13.orig/configure	2009-06-10 22:50:53.000000000 +0200
+++ ffmpeg-export-2009-06-13/configure	2009-06-21 14:51:52.000000000 +0200
@@ -2129,7 +2129,7 @@
 
 disable sdl_too_old
 disable sdl
-SDL_CONFIG="${cross_prefix}sdl-config"
+SDL_CONFIG="sdl-config"
 if "${SDL_CONFIG}" --version > /dev/null 2>&1; then
     sdl_cflags=$("${SDL_CONFIG}" --cflags)
     temp_cflags $sdl_cflags
