diff --git a/ffbuild/version.sh b/ffbuild/version.sh
index edc4dd3..d2b90a9 100755
--- a/ffbuild/version.sh
+++ b/ffbuild/version.sh
@@ -16,6 +16,9 @@ fi
 test "$revision" || revision=$(cd "$1" &&
   git log -1 --pretty=format:"git-%cd-%h" --date=short 2> /dev/null)
 
+# ignore any current revision, which is just that of the LibreELEC repository
+revision=
+
 # Snapshots from gitweb are in a directory called ffmpeg-hhhhhhh or
 # ffmpeg-HEAD-hhhhhhh.
 if [ -z "$revision" ]; then
