From 85417a4ba42360248b4999e458a6e05c1c2f9b17 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Sat, 4 Mar 2017 19:24:02 +0000
Subject: [PATCH] ffmpeg: Call get_format to fix an issue with MMAL rendering

---
 libavcodec/dvdec.c | 6 ++++++
 libavcodec/rv34.c  | 6 +++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/libavcodec/dvdec.c b/libavcodec/dvdec.c
index 89864f2edc..b5f9224d72 100644
--- a/libavcodec/dvdec.c
+++ b/libavcodec/dvdec.c
@@ -197,6 +197,12 @@ static av_cold int dvvideo_decode_init(AVCodecContext *avctx)
     s->idct_put[0] = idsp.idct_put;
     s->idct_put[1] = ff_simple_idct248_put;
 
+    static const enum AVPixelFormat pix_fmts[] = {
+       AV_PIX_FMT_YUV420P,
+       AV_PIX_FMT_NONE
+    };
+    avctx->pix_fmt = ff_get_format(avctx, pix_fmts);
+
     return ff_dvvideo_init(avctx);
 }
 
diff --git a/libavcodec/rv34.c b/libavcodec/rv34.c
index d171e6e1bd..f6f0aa1f74 100644
--- a/libavcodec/rv34.c
+++ b/libavcodec/rv34.c
@@ -1499,7 +1499,11 @@ av_cold int ff_rv34_decode_init(AVCodecContext *avctx)
     ff_mpv_decode_init(s, avctx);
     s->out_format = FMT_H263;
 
-    avctx->pix_fmt = AV_PIX_FMT_YUV420P;
+    static const enum AVPixelFormat pix_fmts[] = {
+       AV_PIX_FMT_YUV420P,
+       AV_PIX_FMT_NONE
+    };
+    avctx->pix_fmt = ff_get_format(avctx, pix_fmts);
     avctx->has_b_frames = 1;
     s->low_delay = 0;
 
-- 
2.20.1

