diff -Naur v86d-0.1.10/libs/x86emu/Makefile v86d-0.1.10.patch/libs/x86emu/Makefile
--- v86d-0.1.10/libs/x86emu/Makefile	2011-02-25 22:27:39.000000000 +0100
+++ v86d-0.1.10.patch/libs/x86emu/Makefile	2012-06-28 20:15:36.486234182 +0200
@@ -1,7 +1,7 @@
 OBJS = decode.o fpu.o ops.o ops2.o prim_ops.o sys.o
 
 libx86emu.a: $(OBJS)
-	ar rv $@ $+
+	$(AR) rv $@ $+
 
 %.o: %.c
 	$(CC) -c $(CFLAGS) -o $@ $<
