diff -Naur v86d-0.1.10/Makefile v86d-0.1.10.patch/Makefile
--- v86d-0.1.10/Makefile	2011-02-25 22:27:39.000000000 +0100
+++ v86d-0.1.10.patch/Makefile	2013-01-14 19:20:13.385465794 +0100
@@ -3,14 +3,12 @@
 .PHONY: clean install install_testvbe x86emu lrmi
 
 INSTALL = install
-KDIR   ?= /lib/modules/$(shell uname -r)/source
 
 ifeq ($(call config_opt,CONFIG_KLIBC),true)
 	export CC = klcc
 endif
 
 CFLAGS ?= -Wall -g -O2
-CFLAGS += -I$(KDIR)/include
 
 ifeq ($(call config_opt,CONFIG_X86EMU),true)
 	CFLAGS += -Ilibs/x86emu
