--- a/v4l/scripts/make_kconfig.pl
+++ b/v4l/scripts/make_kconfig.pl
@@ -626,6 +626,9 @@ ($$)
 close OUT;
 
 # These options should default to off
+disable_config('VIDEO_CX25840');
+disable_config('VIDEO_SAA7146');
+disable_config('VIDEO_SAA7146_VV');
 disable_config('MEDIA_ANALOG_TV_SUPPORT');
 disable_config('MEDIA_CAMERA_SUPPORT');
 disable_config('MEDIA_CEC_SUPPORT');
--- a/v4l/versions.txt
+++ b/v4l/versions.txt
@@ -2,6 +2,7 @@
 [9.255.255]
 VIDEO_IPU3_IMGU
 VIDEO_DW9714
+DVB_MXL5XX
 CEC_NOTIFIER
 MEDIA_CEC_RC
 MEDIA_CEC_SUPPORT
