diff --git a/lib/makefile.linux b/lib/makefile.linux
index b3b5afa..07322ad 100644
--- a/lib/makefile.linux
+++ b/lib/makefile.linux
@@ -1,10 +1,10 @@
 
 include makefile.common
 
-GCC=gcc
+GCC:=gcc
 MOC=$(strip $(shell which moc-qt4))
 ifeq ($(MOC),)
-MOC=moc
+MOC:=moc
 endif
 
 CXXOPTS=-Os -D_GNU_SOURCE -D_linux_
@@ -16,7 +16,7 @@ PREFIX=/usr
 LIBDIR=${PREFIX}/lib
 
 
-all: out/libdriveio.so.0 out/libmakemkv.so.1 out/makemkv
+all: out/libdriveio.so.0 out/libmakemkv.so.1
 	@echo "type \"sudo make -f makefile.linux install\" to install"
 
 clean:
