diff -Naur makemkv-1.6.10/lib/makefile.linux makemkv-1.6.10.patch/lib/makefile.linux
--- makemkv-1.6.10/lib/makefile.linux	2011-05-29 23:17:48.000000000 +0200
+++ makemkv-1.6.10.patch/lib/makefile.linux	2011-05-30 15:29:15.510417904 +0200
@@ -1,11 +1,11 @@
 
 include makefile.common
 
-GCC=gcc
-RCC=rcc
+GCC:=gcc
+RCC:=rcc
 MOC=$(strip $(shell which moc-qt4))
 ifeq ($(MOC),)
-MOC=moc
+MOC:=moc
 endif
 
 CXXOPTS=-Os -D_GNU_SOURCE -D_linux_
@@ -14,7 +14,7 @@
 BUILDINFO_BUILD_DATE=$(shell date)
 
 
-all: out/libdriveio.so.0 out/libmakemkv.so.1 out/makemkv
+all: out/libdriveio.so.0 out/libmakemkv.so.1
 	@echo "type \"sudo make -f makefile.linux install\" to install"
 
 clean:
