diff --git a/lib/makefile.linux b/lib/makefile.linux
index dedf8cf..5fac196 100644
--- a/lib/makefile.linux
+++ b/lib/makefile.linux
@@ -38,7 +38,7 @@ endif
 	install -D -m 644 makemkvgui/src/img/128/mkv_icon.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/128x128/apps/makemkv.png
 
 out/%: out/%.full
-	objcopy --strip-all --strip-debug --strip-unneeded --discard-all $< $@ 
+	$(OBJCOPY) --strip-all --strip-debug --strip-unneeded --discard-all $< $@ 
 
 out/libdriveio.so.0.full:
 	mkdir -p out
