diff -Naur SABnzbd-0.6.0RC2/interfaces/Classic/templates/config_general.tmpl SABnzbd-0.6.0RC2.patch/interfaces/Classic/templates/config_general.tmpl
--- SABnzbd-0.6.0RC2/interfaces/Classic/templates/config_general.tmpl	2011-04-16 23:15:34.940298239 +0200
+++ SABnzbd-0.6.0RC2.patch/interfaces/Classic/templates/config_general.tmpl	2011-04-16 23:21:11.140812740 +0200
@@ -13,16 +13,6 @@
 
 <legend>$T('webServer')</legend>
 <i>$T('restartRequired')</i><br/><br/>
-<strong>$T('opt-host'):</strong><br>
-$T('explain-host')<br>
-<input type="text" name="host" value="$host">
-<br>
-<br>
-<strong>$T('opt-port'):</strong><br>
-$T('explain-port')<br>
-<input type="text" name="port" value="$port">
-<br>
-<br>
 <strong>$T('opt-web_dir'):</strong><br>
 $T('explain-web_dir')<br>
 <select name="web_dir">
diff -Naur SABnzbd-0.6.0RC2/interfaces/Plush/templates/config_general.tmpl SABnzbd-0.6.0RC2.patch/interfaces/Plush/templates/config_general.tmpl
--- SABnzbd-0.6.0RC2/interfaces/Plush/templates/config_general.tmpl	2011-04-16 23:15:34.954297968 +0200
+++ SABnzbd-0.6.0RC2.patch/interfaces/Plush/templates/config_general.tmpl	2011-04-16 23:23:32.496084161 +0200
@@ -14,26 +14,6 @@
         </div>
         <fieldset class="component-group-list">
             <div class="field-pair">
-                <label class="nocheck clearfix" for="host">
-                    <span class="component-title">$T('opt-host')</span>
-	                <input type="text" name="host" id="host" value="$host"/>
-                </label>
-                <label class="nocheck clearfix">
-                    <span class="component-title">&nbsp;</span>
-                    <span class="component-desc">$T('explain-host')</span>
-                </label>
-            </div>
-            <div class="field-pair alt">
-                <label class="nocheck clearfix" for="port">
-                    <span class="component-title">$T('opt-port')</span>
-				    <input type="text" name="port" id="port" size=8 value="$port"/>
-                </label>
-                <label class="nocheck clearfix">
-                    <span class="component-title">&nbsp;</span>
-                    <span class="component-desc">$T('explain-port')</span>
-                </label>
-            </div>
-            <div class="field-pair">
                 <label class="nocheck clearfix" for="username">
                     <span class="component-title">$T('opt-web_username')</span>
 				    <input type="text" name="username" id="username" value="$username"/>
diff -Naur SABnzbd-0.6.0RC2/interfaces/smpl/templates/config_general.tmpl SABnzbd-0.6.0RC2.patch/interfaces/smpl/templates/config_general.tmpl
--- SABnzbd-0.6.0RC2/interfaces/smpl/templates/config_general.tmpl	2011-04-16 23:15:34.944298160 +0200
+++ SABnzbd-0.6.0RC2.patch/interfaces/smpl/templates/config_general.tmpl	2011-04-16 23:22:14.877582554 +0200
@@ -5,16 +5,6 @@
 <fieldset class="EntryFieldSet">
 <legend>$T('webServer') <i>($T('restartRequired'))</i></legend>
 <hr />
-			<label for "host" class="label">$T('opt-host'):</label>
-			<input type="text" name="host" value="$host">
-			<span class="tips">$T('explain-host')</span>
-		<br class="clear" />
-
-			<label for "port" class="label">$T('opt-port'):</label>
-			<input type="text" name="port" value="$port">
-			<span class="tips">$T('explain-port')</span>
-		<br class="clear" />
-
         <label for "userinterface" class="label">$T('opt-web_dir'):</label>
 
         <select name="web_dir" id="webdir" class="select">
