diff -Naur SABnzbd-0.6.0RC2/sabnzbd/constants.py SABnzbd-0.6.0RC2.patch/sabnzbd/constants.py
--- SABnzbd-0.6.0RC2/sabnzbd/constants.py	2011-04-10 16:06:27.000000000 +0200
+++ SABnzbd-0.6.0RC2.patch/sabnzbd/constants.py	2011-04-16 21:49:52.858788377 +0200
@@ -70,8 +70,8 @@
 DB_HISTORY_NAME = 'history%s.db' % DB_HISTORY_VERSION
 DB_QUEUE_NAME = 'queue%s.db' % DB_QUEUE_VERSION
 
-DEF_DOWNLOAD_DIR = 'Downloads/incomplete'
-DEF_COMPLETE_DIR = 'Downloads/complete'
+DEF_DOWNLOAD_DIR = '/storage/downloads/incoming/'
+DEF_COMPLETE_DIR = '/storage/downloads/'
 DEF_CACHE_DIR    = 'cache'
 DEF_ADMIN_DIR    = 'admin'
 DEF_LOG_DIR      = 'logs'
@@ -83,12 +83,12 @@
 DEF_STDINTF      = 'smpl'
 DEF_SKIN_COLORS  = {'smpl':'white', 'classic':'darkblue', 'mobile':'light', 'plush' : 'gold'}
 DEF_MAIN_TMPL    = 'templates/main.tmpl'
-DEF_INI_FILE     = 'sabnzbd.ini'
-DEF_HOST         = 'localhost'
-DEF_PORT_WIN     = 8080
-DEF_PORT_UNIX    = 8080
-DEF_PORT_WIN_SSL = 9090
-DEF_PORT_UNIX_SSL= 9090
+DEF_INI_FILE     = 'sabnzbd.conf'
+DEF_HOST         = '0.0.0.0'
+DEF_PORT_WIN     = 8081
+DEF_PORT_UNIX    = 8081
+DEF_PORT_WIN_SSL = 9081
+DEF_PORT_UNIX_SSL= 9081
 DEF_WORKDIR      = 'sabnzbd'
 DEF_LOG_FILE     = 'sabnzbd.log'
 DEF_LOG_ERRFILE  = 'sabnzbd.error.log'
