From 2f4b28a70ad73f96406a388f62bac8619a2aefe9 Mon Sep 17 00:00:00 2001
From: Kostas Missos <ctcaer@gmail.com>
Date: Tue, 31 Jul 2018 19:45:30 +0300
Subject: [PATCH] [PMIC] Turn on PCIE power on boot (fixes Wi-Fi)

---
 src/mainboard/nintendo/switch/pmic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/mainboard/nintendo/switch/pmic.c b/src/mainboard/nintendo/switch/pmic.c
index b44194b9d2..f42328b0f5 100644
--- a/src/mainboard/nintendo/switch/pmic.c
+++ b/src/mainboard/nintendo/switch/pmic.c
@@ -95,6 +95,10 @@ void pmic_init(unsigned bus)
 	/* CNFG1_L2 = 0xF2 for 3.3v, enabled */
 	pmic_write_reg_77620(bus, MAX77620_CNFG1_L2_REG, 0xf2, 1);
 
+	/* MAX77620: Turn on LDO1 to 1.05V for PEX power */
+	pmic_write_reg_77620(bus, MAX77620_CNFG1_L1_REG, 0xCA, 1);
+	pmic_write_reg_77620(bus, MAX77620_CNFG2_L1_REG, 0xCA, 1);
+
 	/* XXX MAX77621: Set VOUT_REG to 1.0V - CPU VREG */
 	pmic_write_reg_77621(bus, MAX77621_VOUT_REG, 0xB7, 1);
 
