diff -Nur a867_drv_v1.0.28/af903x-drv.c a867_drv_v1.0.28_modif//af903x-drv.c
--- a867_drv_v1.0.28/af903x-drv.c	2010-11-10 04:46:22.000000000 +0100
+++ a867_drv_v1.0.28_modif//af903x-drv.c	2011-07-24 11:41:45.162923006 +0200
@@ -1103,8 +1103,8 @@
 		PDC->StreamType = StreamType_DVBT_DATAGRAM;
 		PDC->UsbCtrlTimeOut = 1;
 
-		init_MUTEX(&PDC->powerLock);
-		init_MUTEX(&PDC->tunerLock);
+		sema_init(&PDC->powerLock,1);
+		sema_init(&PDC->tunerLock,1);
 		PDC->power_use_count = 0;
 
 		PDC->idVendor = udev->descriptor.idVendor;
@@ -1116,7 +1116,7 @@
 		PDC->fc[0].AVerFlags = 0x00;
 		PDC->fc[1].AVerFlags = 0x00;
 		
-		init_MUTEX(&PDC->regLock);
+		sema_init(&PDC->regLock,1);
 	}
 	else {
         	PDC->UsbCtrlTimeOut = 5;
diff -Nur a867_drv_v1.0.28/af903x-fe.c a867_drv_v1.0.28_modif//af903x-fe.c
--- a867_drv_v1.0.28/af903x-fe.c	2010-11-10 04:46:22.000000000 +0100
+++ a867_drv_v1.0.28_modif//af903x-fe.c	2011-07-24 11:43:58.543584444 +0200
@@ -1,5 +1,5 @@
 #include <linux/sched.h>
-#include <linux/smp_lock.h>
+//#include <linux/smp_lock.h>
 #include <linux/list.h>
 #include <linux/module.h>
 #include <linux/version.h>
@@ -678,7 +678,7 @@
 	deb_data("- Enter %s Function -\n",__FUNCTION__);
 	if( !state ) return -1;
 
-	lock_kernel();
+	//lock_kernel();
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,61)
 	daemonize();
 	sigfillset(&current->blocked);
@@ -689,7 +689,7 @@
 #endif
 	siginitsetinv(&current->blocked, sigmask(SIGKILL)|sigmask(SIGINT)|\
 			sigmask(SIGTERM));
-	unlock_kernel();
+	//unlock_kernel();
 
 	while(!state->thread_should_stop && !signal_pending(current)) {
 	
diff -Nur a867_drv_v1.0.28/af903x.h a867_drv_v1.0.28_modif//af903x.h
--- a867_drv_v1.0.28/af903x.h	2010-11-10 04:46:22.000000000 +0100
+++ a867_drv_v1.0.28_modif//af903x.h	2011-07-24 11:42:54.563267143 +0200
@@ -9,7 +9,7 @@
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
-#include <linux/smp_lock.h>
+//#include <linux/smp_lock.h>
 #include <linux/usb.h>
 #include <asm/uaccess.h>
 #include <dvb-usb.h>
diff -Nur a867_drv_v1.0.28/usb2impl.c a867_drv_v1.0.28_modif//usb2impl.c
--- a867_drv_v1.0.28/usb2impl.c	2010-11-10 10:27:27.000000000 +0100
+++ a867_drv_v1.0.28_modif//usb2impl.c	2011-07-24 11:43:09.915343276 +0200
@@ -10,7 +10,7 @@
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
-#include <linux/smp_lock.h>
+//#include <linux/smp_lock.h>
 #include <linux/usb.h>
 #include <asm/uaccess.h>
 #include <linux/device.h>
