diff -Naur udevil-0.4.4/src/device-info.c udevil-0.4.4.patch/src/device-info.c
--- udevil-0.4.4/src/device-info.c	2015-05-04 15:34:32.000000000 +0200
+++ udevil-0.4.4.patch/src/device-info.c	2015-05-23 16:25:50.545362784 +0200
@@ -4,6 +4,7 @@
 ************************************************************************** */
 
 #include "device-info.h"
+#include <sys/stat.h>
 
 static char *
 _dupv8 (const char *s)
