diff --git a/Makefile b/Makefile
index 9d62d04..5519a07 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,6 @@ VERSION = $(shell grep 'static const char \*VERSION *=' $(PLUGIN).c | awk '{ pri
 
 ### The C++ compiler and options:
 
-CXX      ?= g++
 CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -Wno-parentheses -fpermissive
 
 ### Make sure that the necessary options are included
@@ -27,7 +26,6 @@ CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -Wno-parentheses -fpermissive
 ### The directory environment:
 
 DVBDIR = ../../../../DVB
-VDRDIR = ../../..
 LIBDIR = ../../lib
 TMPDIR = /tmp
 
@@ -37,11 +35,7 @@ TMPDIR = /tmp
 
 ### The version number of VDR (taken from VDR's "config.h"):
 
-VDRVERSION = $(shell grep 'define VDRVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
 APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
-ifeq ($(strip $(APIVERSION)),)
-  APIVERSION = $(VDRVERSION)
-endif
 
 ### The name of the distribution archive:
 
