Index: vdr-plugin-rotorng-0.1.0/rotorng.c
===================================================================
--- vdr-plugin-rotorng-0.1.0.orig/rotorng.c	2012-03-14 14:18:16.610997754 +0100
+++ vdr-plugin-rotorng-0.1.0/rotorng.c	2012-03-14 14:24:59.410997997 +0100
@@ -1494,7 +1494,7 @@
                snprintf(buf, sizeof(buf),"%s %d", tr(menucaption[itemindex]), menuvalue[itemindex]);
                break;  
              case MI_SYSTEM:
-               snprintf(buf, sizeof(buf),"%s %s", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], SystemValues));
+               snprintf(buf, sizeof(buf),"%s %s", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], SystemValuesSat));
                break;
              case MI_MODULATION:
                snprintf(buf, sizeof(buf),"%s %s ", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], ModulationValues));
