diff -uNr rotorng-0.1.0-orig/rotorng.c rotorng-0.1.0/rotorng.c
--- rotorng-0.1.0-orig/rotorng.c	2012-05-06 16:14:06.000000000 +0200
+++ rotorng-0.1.0/rotorng.c	2012-05-07 10:52:18.000000000 +0200
@@ -338,7 +338,7 @@
   int last_position_shown;
   bool transfer;
 protected:
-  virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
+  virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber,bool LiveView);
 public:
   cStatusMonitor();
 };
@@ -348,7 +348,7 @@
   transfer=false;
 }
 
-void cStatusMonitor::ChannelSwitch(const cDevice *Device, int ChannelNumber)
+void cStatusMonitor::ChannelSwitch(const cDevice *Device, int ChannelNumber,bool LiveView)
 {
   //actuator_status status;
   if (ChannelNumber) {
@@ -1494,7 +1494,7 @@
                snprintf(buf, sizeof(buf),"%s %d", tr(menucaption[itemindex]), menuvalue[itemindex]);
                break;  
              case MI_SYSTEM:
-               snprintf(buf, sizeof(buf),"%s %s", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], SystemValues));
+               snprintf(buf, sizeof(buf),"%s %s", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], SystemValuesSat));
                break;
              case MI_MODULATION:
                snprintf(buf, sizeof(buf),"%s %s ", tr(menucaption[itemindex]), MapToUserString(menuvalue[itemindex], ModulationValues));
