diff --git a/dist/info/mupen64plus_next_oldgliden_libretro.info b/dist/info/mupen64plus_next_oldgliden_libretro.info
new file mode 100644
index 00000000..9b0fa81a
--- /dev/null
+++ b/dist/info/mupen64plus_next_oldgliden_libretro.info
@@ -0,0 +1,13 @@
+display_name = "Nintendo - Nintendo 64 (Mupen64Plus-Next Old GLideN64)"
+authors = "m4xw|Hacktarux|Mupen64Plus Team"
+supported_extensions = "n64|v64|z64|bin|u1|ndd"
+corename = "Mupen64Plus-Next (Old GLideN64)"
+manufacturer = "Nintendo"
+categories = "Emulator"
+systemname = "Nintendo 64"
+systemid = "nintendo_64"
+database = "Nintendo - Nintendo 64|Nintendo - Nintendo 64DD"
+license = "GPLv2"
+permissions = "dynarec_optional"
+display_version = "1.0"
+supports_no_game = "false"
