diff --git a/dist/info/pcsx_rearmed_libretro.info b/dist/info/pcsx_rearmed_libretro.info
index 151f1a2f..3cb80840 100644
--- a/dist/info/pcsx_rearmed_libretro.info
+++ b/dist/info/pcsx_rearmed_libretro.info
@@ -1,6 +1,6 @@
 display_name = "Sony - PlayStation (PCSX ReARMed)"
 authors = "PCSX Team|notaz|Exophase"
-supported_extensions = "bin|cue|img|mdf|pbp|toc|cbn|m3u|ccd"
+supported_extensions = "bin|cue|img|mdf|pbp|toc|cbn|m3u|ccd|chd"
 corename = "PCSX-ReARMed"
 manufacturer = "Sony"
 categories = "Emulator"
