diff -Naur intel-driver-1.0.18-old/src/i965_drv_video.c intel-driver-1.0.18-new/src/i965_drv_video.c
--- intel-driver-1.0.18-old/src/i965_drv_video.c	2012-08-01 22:58:37.000000000 -0700
+++ intel-driver-1.0.18-new/src/i965_drv_video.c	2012-10-10 14:34:35.000000000 -0700
@@ -2275,7 +2275,7 @@
                 unsigned int flags) /* de-interlacing flags */
 {
     struct i965_driver_data *i965 = i965_driver_data(ctx); 
-    struct dri_state *dri_state = (struct dri_state *)ctx->dri_state;
+    struct dri_state *dri_state = (struct dri_state *)ctx->drm_state;
     struct i965_render_state *render_state = &i965->render_state;
     struct dri_drawable *dri_drawable;
     union dri_buffer *buffer;
@@ -2288,7 +2288,7 @@
     int pp_flag = 0;
 
     /* Currently don't support DRI1 */
-    if (dri_state->driConnectedFlag != VA_DRI2)
+    if (dri_state->base.auth_type != VA_DRM_AUTH_DRI2)
         return VA_STATUS_ERROR_UNKNOWN;
 
     /* Some broken sources such as H.264 conformance case FM2_SVA_C
diff -Naur intel-driver-1.0.18-old/src/intel_driver.c intel-driver-1.0.18-new/src/intel_driver.c
--- intel-driver-1.0.18-old/src/intel_driver.c	2012-08-01 22:58:37.000000000 -0700
+++ intel-driver-1.0.18-new/src/intel_driver.c	2012-10-10 14:34:35.000000000 -0700
@@ -29,7 +29,7 @@
 
 #include <assert.h>
 
-#include <va/va_dricommon.h>
+#include <va/va_drmcommon.h>
 
 #include "intel_batchbuffer.h"
 #include "intel_memman.h"
@@ -50,15 +50,17 @@
 intel_driver_init(VADriverContextP ctx)
 {
     struct intel_driver_data *intel = intel_driver_data(ctx);
-    struct dri_state *dri_state = (struct dri_state *)ctx->dri_state;
+    struct drm_state * const drm_state = (struct drm_state *)ctx->drm_state;
     int has_exec2, has_bsd, has_blt;
 
-    assert(dri_state);
-    assert(dri_state->driConnectedFlag == VA_DRI2 || 
-           dri_state->driConnectedFlag == VA_DRI1);
-
-    intel->fd = dri_state->fd;
-    intel->dri2Enabled = (dri_state->driConnectedFlag == VA_DRI2);
+    assert(drm_state);
+    assert(drm_state->auth_type == VA_DRM_AUTH_DRI1 ||
+           drm_state->auth_type == VA_DRM_AUTH_DRI2 ||
+           drm_state->auth_type == VA_DRM_AUTH_CUSTOM);
+
+    intel->fd = drm_state->fd;
+    intel->dri2Enabled = (drm_state->auth_type == VA_DRM_AUTH_DRI2 ||
+                          drm_state->auth_type == VA_DRM_AUTH_CUSTOM);
 
     if (!intel->dri2Enabled) {
         return False;
