From 54cb60f3d973ce6a908ea57f5aa7481803fe653a Mon Sep 17 00:00:00 2001
From: "Xiang, Haihao" <haihao.xiang@intel.com>
Date: Mon, 30 Dec 2013 13:15:40 +0800
Subject: [PATCH] Render: Adjust the default value for contrast/saturation

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=73016
Signed-off-by: Xiang, Haihao <haihao.xiang@intel.com>
Tested-by: Mark Lee <mark@markelee.com>
---
 src/i965_drv_video.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/i965_drv_video.h b/src/i965_drv_video.h
index ccf9a59..773c8ca 100644
--- a/src/i965_drv_video.h
+++ b/src/i965_drv_video.h
@@ -60,9 +60,9 @@
 #define I965_SURFACE_FLAG_BOTTOME_FIELD_FIRST   0x00000002
 
 #define DEFAULT_BRIGHTNESS      0
-#define DEFAULT_CONTRAST        10
+#define DEFAULT_CONTRAST        50
 #define DEFAULT_HUE             0
-#define DEFAULT_SATURATION      10
+#define DEFAULT_SATURATION      50
 
 struct i965_surface
 {
-- 
1.8.3.2

