From a36f24198b46cba00a1d09f8c70dc86f36d98022 Mon Sep 17 00:00:00 2001
From: Xiang, Haihao <haihao.xiang@intel.com>
Date: Mon, 05 May 2014 04:48:27 +0000
Subject: Return error when trying to decoding an interlaced VC-1 video

https://bugs.freedesktop.org/show_bug.cgi?id=77386

Signed-off-by: Xiang, Haihao <haihao.xiang@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
diff --git a/src/i965_decoder_utils.c b/src/i965_decoder_utils.c
index 2533381..617bc15 100644
--- a/src/i965_decoder_utils.c
+++ b/src/i965_decoder_utils.c
@@ -654,7 +654,12 @@ intel_decoder_check_vc1_parameter(VADriverContextP ctx,
     VAPictureParameterBufferVC1 *pic_param = (VAPictureParameterBufferVC1 *)decode_state->pic_param->buffer;
     struct object_surface *obj_surface;	
     int i = 0;
-    
+
+    if (pic_param->sequence_fields.bits.interlace == 1 &&
+        pic_param->picture_fields.bits.frame_coding_mode != 0) { /* frame-interlace or field-interlace */
+        return VA_STATUS_ERROR_DECODING_ERROR;
+    }
+
     if (pic_param->picture_fields.bits.picture_type == 0 ||
         pic_param->picture_fields.bits.picture_type == 3) {
     } else if (pic_param->picture_fields.bits.picture_type == 1 ||
--
cgit v0.9.0.2-2-gbebe
