diff -Naur evtest-1.30-old/configure.ac evtest-1.30-new/configure.ac
--- evtest-1.30-old/configure.ac	2012-04-26 15:15:32.000000000 -0700
+++ evtest-1.30-new/configure.ac	2013-01-03 17:00:38.000000000 -0800
@@ -2,7 +2,7 @@
 AC_INIT(evtest,[1.30], [https://bugzilla.freedesktop.org/enter_bug.cgi?product=evtest], evtest)
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_PROG_CC
 AC_PROG_INSTALL
 
