diff -Naur linux-3.6.11/tools/perf/Makefile linux-3.6.11.patch/tools/perf/Makefile
--- linux-3.6.11/tools/perf/Makefile	2012-12-17 18:27:45.000000000 +0100
+++ linux-3.6.11.patch/tools/perf/Makefile	2013-03-06 16:30:43.107805512 +0100
@@ -496,8 +496,6 @@
 		msg := $(warning newt not found, disables TUI support. Please install newt-devel or libnewt-dev);
 		BASIC_CFLAGS += -DNO_NEWT_SUPPORT
 	else
-		# Fedora has /usr/include/slang/slang.h, but ubuntu /usr/include/slang.h
-		BASIC_CFLAGS += -I/usr/include/slang
 		EXTLIBS += -lnewt -lslang
 		LIB_OBJS += $(OUTPUT)ui/setup.o
 		LIB_OBJS += $(OUTPUT)ui/browser.o
