diff -Naur linux-3.6.11/drivers/media/rc/fintek-cir.c linux-3.6.11.patch/drivers/media/rc/fintek-cir.c
--- linux-3.6.11/drivers/media/rc/fintek-cir.c	2012-12-28 22:59:21.808868854 +0100
+++ linux-3.6.11.patch/drivers/media/rc/fintek-cir.c	2012-12-28 22:59:37.532767678 +0100
@@ -552,6 +552,8 @@
 	/* rx resolution is hardwired to 50us atm, 1, 25, 100 also possible */
 	rdev->rx_resolution = US_TO_NS(CIR_SAMPLE_PERIOD);
 
+	fintek->rdev = rdev;
+
 	ret = -EBUSY;
 	/* now claim resources */
 	if (!request_region(fintek->cir_addr,
@@ -567,7 +569,7 @@
 		goto exit_free_irq;
 
 	device_init_wakeup(&pdev->dev, true);
-	fintek->rdev = rdev;
+
 	fit_pr(KERN_NOTICE, "driver has been successfully loaded\n");
 	if (debug)
 		cir_dump_regs(fintek);
diff -Naur linux-3.6.11/drivers/media/rc/ite-cir.c linux-3.6.11.patch/drivers/media/rc/ite-cir.c
--- linux-3.6.11/drivers/media/rc/ite-cir.c	2012-12-28 22:59:21.808868854 +0100
+++ linux-3.6.11.patch/drivers/media/rc/ite-cir.c	2012-12-28 22:59:37.533767672 +0100
@@ -1591,6 +1591,8 @@
 	rdev->driver_name = ITE_DRIVER_NAME;
 	rdev->map_name = RC_MAP_RC6_MCE;
 
+	itdev->rdev = rdev;
+
 	ret = -EBUSY;
 	/* now claim resources */
 	if (!request_region(itdev->cir_addr,
diff -Naur linux-3.6.11/drivers/media/rc/nuvoton-cir.c linux-3.6.11.patch/drivers/media/rc/nuvoton-cir.c
--- linux-3.6.11/drivers/media/rc/nuvoton-cir.c	2012-12-28 22:59:21.807868860 +0100
+++ linux-3.6.11.patch/drivers/media/rc/nuvoton-cir.c	2012-12-28 22:59:37.533767672 +0100
@@ -1065,6 +1065,7 @@
 	/* tx bits */
 	rdev->tx_resolution = XYZ;
 #endif
+	nvt->rdev = rdev;
 
 	ret = -EBUSY;
 	/* now claim resources */
@@ -1089,7 +1090,7 @@
 		goto exit_free_wake_irq;
 
 	device_init_wakeup(&pdev->dev, true);
-	nvt->rdev = rdev;
+
 	nvt_pr(KERN_NOTICE, "driver has been successfully loaded\n");
 	if (debug) {
 		cir_dump_regs(nvt);
