diff -Naur wiiuse_v0.12/example/Makefile wiiuse_v0.12.patch/example/Makefile
--- wiiuse_v0.12/example/Makefile	2008-04-02 02:45:40.000000000 +0200
+++ wiiuse_v0.12.patch/example/Makefile	2010-09-21 21:31:12.052265717 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
diff -Naur wiiuse_v0.12/example-sdl/Makefile wiiuse_v0.12.patch/example-sdl/Makefile
--- wiiuse_v0.12/example-sdl/Makefile	2008-04-02 02:45:53.000000000 +0200
+++ wiiuse_v0.12.patch/example-sdl/Makefile	2010-09-21 21:31:19.907265540 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
diff -Naur wiiuse_v0.12/src/Makefile wiiuse_v0.12.patch/src/Makefile
--- wiiuse_v0.12/src/Makefile	2008-03-03 00:30:04.000000000 +0100
+++ wiiuse_v0.12.patch/src/Makefile	2010-09-21 21:31:27.338390537 +0200
@@ -5,7 +5,7 @@
 #
 # Change this to your GCC version.
 #
-CC = gcc
+CC ?= gcc
 
 ####################################################
 #
