diff -Naur wiiuse_v0.12/src/io_nix.c wiiuse_v0.12.patch/src/io_nix.c
--- wiiuse_v0.12/src/io_nix.c	2008-03-29 22:53:32.000000000 +0100
+++ wiiuse_v0.12.patch/src/io_nix.c	2010-09-21 22:50:08.889265428 +0200
@@ -171,6 +171,7 @@
 static int wiiuse_connect_single(struct wiimote_t* wm, char* address) {
 	struct sockaddr_l2 addr;
 
+	memset(&addr, 0, sizeof (addr)); //is missing in the line 174 of io_nix.c
 	if (!wm || WIIMOTE_IS_CONNECTED(wm))
 		return 0;
 
