diff -Naur serdisplib-1.97.9/configure serdisplib-1.97.9-b/configure
--- serdisplib-1.97.9/configure	2010-02-21 20:49:29.000000000 +0100
+++ serdisplib-1.97.9-b/configure	2015-06-10 17:49:18.780035766 +0200
@@ -4347,6 +4347,11 @@
 #    HAVE_SDL_H
 #    HAVE_SDL_SDL_H
 #  HAVE_LIBSDL
+
+has_libSDL="false"   # pre-init
+
+# SDL only if not cross compiling (until better solution has been found)
+if test "$ac_cv_build" == "$ac_cv_host" ; then
 # Extract the first word of "sdl-config", so it can be a program name with args.
 set dummy sdl-config; ac_word=$2
 echo "$as_me:$LINENO: checking for $ac_word" >&5
@@ -4387,7 +4392,6 @@
 fi
 
 
-has_libSDL="false"   # pre-init
 if test ! -z "${LIBSDL_CONFIG}"; then
   LIBSDL_CFLAGS=`${LIBSDL_CONFIG} --cflags`
   CFLAGS="${CFLAGS} ${LIBSDL_CFLAGS}"
@@ -4761,6 +4765,7 @@
       fi
   fi
 fi
+fi
 
 
 
diff -Naur serdisplib-1.97.9/configure.in serdisplib-1.97.9-b/configure.in
--- serdisplib-1.97.9/configure.in	2010-02-21 20:49:29.000000000 +0100
+++ serdisplib-1.97.9-b/configure.in	2015-06-10 17:46:33.986014606 +0200
@@ -143,9 +143,13 @@
 #    HAVE_SDL_H
 #    HAVE_SDL_SDL_H
 #  HAVE_LIBSDL
-AC_PATH_PROG(LIBSDL_CONFIG, sdl-config)
 
 has_libSDL="false"   # pre-init
+
+# SDL only if not cross compiling (until better solution has been found)
+if test "$ac_cv_build" == "$ac_cv_host" ; then
+AC_PATH_PROG(LIBSDL_CONFIG, sdl-config)
+
 if test ! -z "${LIBSDL_CONFIG}"; then
   LIBSDL_CFLAGS=`${LIBSDL_CONFIG} --cflags`
   CFLAGS="${CFLAGS} ${LIBSDL_CFLAGS}"
@@ -183,6 +187,7 @@
       fi
   fi
 fi
+fi
 AC_SUBST(has_libSDL)
 AC_SUBST(HAVE_LIBSDL)
 
