diff -Naur xbmc-master-14-dae6f76/xbmc/windowing/egl/EGLNativeTypeIMX.cpp xbmc-master-14-dae6f76.patch/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
--- xbmc-master-14-dae6f76/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2014-07-11 04:38:24.091279024 +0200
+++ xbmc-master-14-dae6f76.patch/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2014-07-11 04:48:46.630581643 +0200
@@ -211,8 +211,7 @@
 
   std::string valstr;
   get_sysfs_str("/sys/class/graphics/fb0/modes", valstr);
-  std::vector<CStdString> probe_str;
-  StringUtils::SplitString(valstr, "\n", probe_str);
+  std::vector<string> probe_str = StringUtils::Split(valstr, "\n");
 
   resolutions.clear();
   RESOLUTION_INFO res;
