diff -Naur ffmpeg-4.3.2-Matrix-19.1/configure ffmpeg-4.3.2-Matrix-19.1-2/configure
--- ffmpeg-4.3.2-Matrix-19.1/configure	2021-11-06 06:45:35.077018182 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/configure	2021-11-06 06:49:26.867662859 +0100
@@ -1846,7 +1846,7 @@
     videotoolbox
     v4l2_m2m
     xvmc
-    nvmpi
+    nvv4l2dec
 "
 
 # catchall list of things that require external libs to link
@@ -1869,6 +1869,7 @@
     opencl
     v4l2_request
     vulkan
+    nvmpi
 "
 
 DOCUMENT_LIST="
@@ -3052,12 +3053,10 @@
 transpose_npp_filter_deps="ffnvcodec libnpp"
 overlay_cuda_filter_deps="ffnvcodec"
 overlay_cuda_filter_deps_any="cuda_nvcc cuda_llvm"
-
 amf_deps_any="libdl LoadLibrary"
 nvenc_deps="ffnvcodec"
 nvenc_deps_any="libdl LoadLibrary"
 nvenc_encoder_deps="nvenc"
-
 aac_mf_encoder_deps="mediafoundation"
 ac3_mf_encoder_deps="mediafoundation"
 h263_v4l2m2m_decoder_deps="v4l2_m2m h263_v4l2_m2m"
@@ -3097,8 +3096,6 @@
 hevc_qsv_encoder_select="hevcparse qsvenc"
 hevc_rkmpp_decoder_deps="rkmpp"
 hevc_rkmpp_decoder_select="hevc_mp4toannexb_bsf"
-hevc_nvmpi_decoder_deps="nvmpi"
-hevc_nvmpi_decoder_select="hevc_mp4toannexb_bsf"
 hevc_vaapi_encoder_deps="VAEncPictureParameterBufferHEVC"
 hevc_vaapi_encoder_select="cbs_h265 vaapi_encode"
 hevc_v4l2m2m_decoder_deps="v4l2_m2m hevc_v4l2_m2m"
@@ -3115,7 +3112,6 @@
 mpeg1_v4l2m2m_decoder_deps="v4l2_m2m mpeg1_v4l2_m2m"
 mpeg2_crystalhd_decoder_select="crystalhd"
 mpeg2_cuvid_decoder_deps="cuvid"
-mpeg2_nvmpi_decoder_deps="nvmpi"
 mpeg2_mmal_decoder_deps="mmal"
 mpeg2_mediacodec_decoder_deps="mediacodec"
 mpeg2_nvv4l2dec_decoder_deps="nvv4l2dec"
@@ -3125,7 +3121,6 @@
 mpeg2_v4l2m2m_decoder_deps="v4l2_m2m mpeg2_v4l2_m2m"
 mpeg4_crystalhd_decoder_select="crystalhd"
 mpeg4_cuvid_decoder_deps="cuvid"
-mpeg4_nvmpi_decoder_deps="nvmpi"
 mpeg4_mediacodec_decoder_deps="mediacodec"
 mpeg4_mmal_decoder_deps="mmal"
 mpeg4_nvv4l2dec_decoder_deps="nvv4l2dec"
@@ -3141,7 +3136,6 @@
 vc1_qsv_decoder_select="qsvdec"
 vc1_v4l2m2m_decoder_deps="v4l2_m2m vc1_v4l2_m2m"
 vp8_cuvid_decoder_deps="cuvid"
-vp8_nvmpi_decoder_deps="nvmpi"
 vp8_mediacodec_decoder_deps="mediacodec"
 vp8_nvv4l2dec_decoder_deps="nvv4l2dec"
 vp8_qsv_decoder_select="qsvdec"
@@ -3151,7 +3145,6 @@
 vp8_v4l2m2m_decoder_deps="v4l2_m2m vp8_v4l2_m2m"
 vp8_v4l2m2m_encoder_deps="v4l2_m2m vp8_v4l2_m2m"
 vp9_cuvid_decoder_deps="cuvid"
-vp9_nvmpi_decoder_deps="nvmpi"
 vp9_mediacodec_decoder_deps="mediacodec"
 vp9_nvv4l2dec_decoder_deps="nvv4l2dec"
 vp9_qsv_decoder_select="qsvdec"
@@ -6516,7 +6509,7 @@
                              { enabled libudev ||
                                die "ERROR: v4l2-request requires --enable-libudev"; }
 enabled vapoursynth       && require_pkg_config vapoursynth "vapoursynth-script >= 42" VSScript.h vsscript_init
-enabled nvmpi     && require_pkg_config nvmpi nvmpi nvmpi.h nvmpi_create_decoder
+enabled nvmpi		  && require_pkg_config nvmpi nvmpi nvmpi.h nvmpi_create_encoder
 
 
 if enabled gcrypt; then
diff -Naur ffmpeg-4.3.2-Matrix-19.1/libavcodec/allcodecs.c ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/allcodecs.c
--- ffmpeg-4.3.2-Matrix-19.1/libavcodec/allcodecs.c	2021-11-06 06:45:35.077018182 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/allcodecs.c	2021-11-06 06:46:14.205437641 +0100
@@ -144,14 +144,12 @@
 extern AVCodec ff_h264_mmal_decoder;
 extern AVCodec ff_h264_qsv_decoder;
 extern AVCodec ff_h264_rkmpp_decoder;
-extern AVCodec ff_h264_nvmpi_decoder;
 extern AVCodec ff_h264_nvmpi_encoder;
 extern AVCodec ff_hap_encoder;
 extern AVCodec ff_hap_decoder;
 extern AVCodec ff_hevc_decoder;
 extern AVCodec ff_hevc_qsv_decoder;
 extern AVCodec ff_hevc_rkmpp_decoder;
-extern AVCodec ff_hevc_nvmpi_decoder;
 extern AVCodec ff_hevc_nvmpi_encoder;
 extern AVCodec ff_hevc_v4l2m2m_decoder;
 extern AVCodec ff_hnm4_video_decoder;
@@ -798,25 +796,21 @@
 extern AVCodec ff_mpeg1_cuvid_decoder;
 extern AVCodec ff_mpeg2_cuvid_decoder;
 extern AVCodec ff_mpeg2_nvv4l2dec_decoder;
-extern AVCodec ff_mpeg2_nvmpi_decoder;
 extern AVCodec ff_mpeg2_qsv_encoder;
 extern AVCodec ff_mpeg2_vaapi_encoder;
 extern AVCodec ff_mpeg4_cuvid_decoder;
-extern AVCodec ff_mpeg4_nvmpi_decoder;
 extern AVCodec ff_mpeg4_mediacodec_decoder;
 extern AVCodec ff_mpeg4_nvv4l2dec_decoder;
 extern AVCodec ff_mpeg4_omx_encoder;
 extern AVCodec ff_mpeg4_v4l2m2m_encoder;
 extern AVCodec ff_vc1_cuvid_decoder;
 extern AVCodec ff_vp8_cuvid_decoder;
-extern AVCodec ff_vp8_nvmpi_decoder;
 extern AVCodec ff_vp8_mediacodec_decoder;
 extern AVCodec ff_vp8_nvv4l2dec_decoder;
 extern AVCodec ff_vp8_qsv_decoder;
 extern AVCodec ff_vp8_v4l2m2m_encoder;
 extern AVCodec ff_vp8_vaapi_encoder;
 extern AVCodec ff_vp9_cuvid_decoder;
-extern AVCodec ff_vp9_nvmpi_decoder;
 extern AVCodec ff_vp9_mediacodec_decoder;
 extern AVCodec ff_vp9_nvv4l2dec_decoder;
 extern AVCodec ff_vp9_qsv_decoder;
diff -Naur ffmpeg-4.3.2-Matrix-19.1/libavcodec/allcodecs.c.orig ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/allcodecs.c.orig
--- ffmpeg-4.3.2-Matrix-19.1/libavcodec/allcodecs.c.orig	2021-11-06 06:45:35.077018182 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/allcodecs.c.orig	2021-11-06 06:36:28.677119723 +0100
@@ -144,11 +144,15 @@
 extern AVCodec ff_h264_mmal_decoder;
 extern AVCodec ff_h264_qsv_decoder;
 extern AVCodec ff_h264_rkmpp_decoder;
+extern AVCodec ff_h264_nvmpi_decoder;
+extern AVCodec ff_h264_nvmpi_encoder;
 extern AVCodec ff_hap_encoder;
 extern AVCodec ff_hap_decoder;
 extern AVCodec ff_hevc_decoder;
 extern AVCodec ff_hevc_qsv_decoder;
 extern AVCodec ff_hevc_rkmpp_decoder;
+extern AVCodec ff_hevc_nvmpi_decoder;
+extern AVCodec ff_hevc_nvmpi_encoder;
 extern AVCodec ff_hevc_v4l2m2m_decoder;
 extern AVCodec ff_hnm4_video_decoder;
 extern AVCodec ff_hq_hqa_decoder;
@@ -764,6 +768,7 @@
 extern AVCodec ff_h264_cuvid_decoder;
 extern AVCodec ff_h264_mf_encoder;
 extern AVCodec ff_h264_nvenc_encoder;
+extern AVCodec ff_h264_nvv4l2dec_decoder;
 extern AVCodec ff_h264_omx_encoder;
 extern AVCodec ff_h264_qsv_encoder;
 extern AVCodec ff_h264_v4l2m2m_encoder;
@@ -779,6 +784,7 @@
 extern AVCodec ff_hevc_mediacodec_decoder;
 extern AVCodec ff_hevc_mf_encoder;
 extern AVCodec ff_hevc_nvenc_encoder;
+extern AVCodec ff_hevc_nvv4l2dec_decoder;
 extern AVCodec ff_hevc_qsv_encoder;
 extern AVCodec ff_hevc_v4l2m2m_encoder;
 extern AVCodec ff_hevc_vaapi_encoder;
@@ -791,20 +797,28 @@
 extern AVCodec ff_mp3_mf_encoder;
 extern AVCodec ff_mpeg1_cuvid_decoder;
 extern AVCodec ff_mpeg2_cuvid_decoder;
+extern AVCodec ff_mpeg2_nvv4l2dec_decoder;
+extern AVCodec ff_mpeg2_nvmpi_decoder;
 extern AVCodec ff_mpeg2_qsv_encoder;
 extern AVCodec ff_mpeg2_vaapi_encoder;
 extern AVCodec ff_mpeg4_cuvid_decoder;
+extern AVCodec ff_mpeg4_nvmpi_decoder;
 extern AVCodec ff_mpeg4_mediacodec_decoder;
+extern AVCodec ff_mpeg4_nvv4l2dec_decoder;
 extern AVCodec ff_mpeg4_omx_encoder;
 extern AVCodec ff_mpeg4_v4l2m2m_encoder;
 extern AVCodec ff_vc1_cuvid_decoder;
 extern AVCodec ff_vp8_cuvid_decoder;
+extern AVCodec ff_vp8_nvmpi_decoder;
 extern AVCodec ff_vp8_mediacodec_decoder;
+extern AVCodec ff_vp8_nvv4l2dec_decoder;
 extern AVCodec ff_vp8_qsv_decoder;
 extern AVCodec ff_vp8_v4l2m2m_encoder;
 extern AVCodec ff_vp8_vaapi_encoder;
 extern AVCodec ff_vp9_cuvid_decoder;
+extern AVCodec ff_vp9_nvmpi_decoder;
 extern AVCodec ff_vp9_mediacodec_decoder;
+extern AVCodec ff_vp9_nvv4l2dec_decoder;
 extern AVCodec ff_vp9_qsv_decoder;
 extern AVCodec ff_vp9_vaapi_encoder;
 extern AVCodec ff_vp9_qsv_encoder;
diff -Naur ffmpeg-4.3.2-Matrix-19.1/libavcodec/Makefile ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/Makefile
--- ffmpeg-4.3.2-Matrix-19.1/libavcodec/Makefile	2021-11-06 06:45:35.077018182 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/Makefile	2021-11-06 06:46:14.205437641 +0100
@@ -368,7 +368,6 @@
 OBJS-$(CONFIG_H264_NVENC_ENCODER)      += nvenc_h264.o
 OBJS-$(CONFIG_NVENC_ENCODER)           += nvenc_h264.o
 OBJS-$(CONFIG_NVENC_H264_ENCODER)      += nvenc_h264.o
-OBJS-$(CONFIG_H264_NVMPI_DECODER)      += nvmpi_dec.o
 OBJS-$(CONFIG_H264_NVMPI_ENCODER)      += nvmpi_enc.o
 OBJS-$(CONFIG_H264_OMX_ENCODER)        += omx.o
 OBJS-$(CONFIG_H264_QSV_DECODER)        += qsvdec_h2645.o
@@ -398,7 +397,6 @@
 OBJS-$(CONFIG_HEVC_RKMPP_DECODER)      += rkmppdec.o
 OBJS-$(CONFIG_HEVC_VAAPI_ENCODER)      += vaapi_encode_h265.o h265_profile_level.o
 OBJS-$(CONFIG_HEVC_V4L2M2M_DECODER)    += v4l2_m2m_dec.o
-OBJS-$(CONFIG_HEVC_NVMPI_DECODER)      += nvmpi_dec.o
 OBJS-$(CONFIG_HEVC_NVMPI_ENCODER)      += nvmpi_enc.o
 OBJS-$(CONFIG_HEVC_V4L2M2M_ENCODER)    += v4l2_m2m_enc.o
 OBJS-$(CONFIG_HNM4_VIDEO_DECODER)      += hnm4video.o
@@ -489,14 +487,12 @@
 OBJS-$(CONFIG_MPEG2VIDEO_ENCODER)      += mpeg12enc.o mpeg12.o
 OBJS-$(CONFIG_MPEG2_CUVID_DECODER)     += cuviddec.o
 OBJS-$(CONFIG_MPEG2_NVV4L2DEC_DECODER) += nvv4l2_dec.o
-OBJS-$(CONFIG_MPEG2_NVMPI_DECODER)     += nvmpi_dec.o
 OBJS-$(CONFIG_MPEG2_MEDIACODEC_DECODER) += mediacodecdec.o
 OBJS-$(CONFIG_MPEG2_VAAPI_ENCODER)     += vaapi_encode_mpeg2.o
 OBJS-$(CONFIG_MPEG2_V4L2M2M_DECODER)   += v4l2_m2m_dec.o
 OBJS-$(CONFIG_MPEG4_DECODER)           += xvididct.o
 OBJS-$(CONFIG_MPEG4_CUVID_DECODER)     += cuviddec.o
 OBJS-$(CONFIG_MPEG4_NVV4L2DEC_DECODER) += nvv4l2_dec.o
-OBJS-$(CONFIG_MPEG4_NVMPI_DECODER)     += nvmpi_dec.o
 OBJS-$(CONFIG_MPEG4_MEDIACODEC_DECODER) += mediacodecdec.o
 OBJS-$(CONFIG_MPEG4_OMX_ENCODER)       += omx.o
 OBJS-$(CONFIG_MPEG4_V4L2M2M_DECODER)   += v4l2_m2m_dec.o
@@ -704,7 +700,6 @@
 OBJS-$(CONFIG_VP8_DECODER)             += vp8.o vp56rac.o
 OBJS-$(CONFIG_VP8_CUVID_DECODER)       += cuviddec.o
 OBJS-$(CONFIG_VP8_NVV4L2DEC_DECODER)   += nvv4l2_dec.o
-OBJS-$(CONFIG_VP8_NVMPI_DECODER)       += nvmpi_dec.o
 OBJS-$(CONFIG_VP8_MEDIACODEC_DECODER)  += mediacodecdec.o
 OBJS-$(CONFIG_VP8_QSV_DECODER)         += qsvdec_other.o
 OBJS-$(CONFIG_VP8_RKMPP_DECODER)       += rkmppdec.o
@@ -716,7 +711,6 @@
                                           vp9dsp_8bpp.o vp9dsp_10bpp.o vp9dsp_12bpp.o
 OBJS-$(CONFIG_VP9_CUVID_DECODER)       += cuviddec.o
 OBJS-$(CONFIG_VP9_NVV4L2DEC_DECODER)   += nvv4l2_dec.o
-OBJS-$(CONFIG_VP9_NVMPI_DECODER)       += nvmpi_dec.o
 OBJS-$(CONFIG_VP9_MEDIACODEC_DECODER)  += mediacodecdec.o
 OBJS-$(CONFIG_VP9_RKMPP_DECODER)       += rkmppdec.o
 OBJS-$(CONFIG_VP9_VAAPI_ENCODER)       += vaapi_encode_vp9.o
diff -Naur ffmpeg-4.3.2-Matrix-19.1/libavcodec/nvmpi_dec.c ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/nvmpi_dec.c
--- ffmpeg-4.3.2-Matrix-19.1/libavcodec/nvmpi_dec.c	2021-11-06 06:45:35.077018182 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/nvmpi_dec.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,170 +0,0 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <sys/time.h>
-
-#include <nvmpi.h>
-#include "avcodec.h"
-#include "decode.h"
-#include "internal.h"
-#include "libavutil/buffer.h"
-#include "libavutil/common.h"
-#include "libavutil/frame.h"
-#include "libavutil/hwcontext.h"
-#include "libavutil/hwcontext_drm.h"
-#include "libavutil/imgutils.h"
-#include "libavutil/log.h"
-
-
-
-
-typedef struct {
-	char eos_reached;
-	nvmpictx* ctx;
-	AVClass *av_class;
-} nvmpiDecodeContext;
-
-static nvCodingType nvmpi_get_codingtype(AVCodecContext *avctx)
-{
-	switch (avctx->codec_id) {
-		case AV_CODEC_ID_H264:          return NV_VIDEO_CodingH264;
-		case AV_CODEC_ID_HEVC:          return NV_VIDEO_CodingHEVC;
-		case AV_CODEC_ID_VP8:           return NV_VIDEO_CodingVP8;
-		case AV_CODEC_ID_VP9:           return NV_VIDEO_CodingVP9;
-		case AV_CODEC_ID_MPEG4:		return NV_VIDEO_CodingMPEG4;
-		case AV_CODEC_ID_MPEG2VIDEO:    return NV_VIDEO_CodingMPEG2;
-		default:                        return NV_VIDEO_CodingUnused;
-	}
-};
-
-
-static int nvmpi_init_decoder(AVCodecContext *avctx){
-
-	nvmpiDecodeContext *nvmpi_context = avctx->priv_data;
-	nvCodingType codectype=NV_VIDEO_CodingUnused;
-
-	codectype =nvmpi_get_codingtype(avctx);
-	if (codectype == NV_VIDEO_CodingUnused) {
-		av_log(avctx, AV_LOG_ERROR, "Unknown codec type (%d).\n", avctx->codec_id);
-		return AVERROR_UNKNOWN;
-	}
-
-	//Workaround for default pix_fmt not being set, so check if it isnt set and set it,
-   //or if it is set, but isnt set to something we can work with.
-
-	if(avctx->pix_fmt ==AV_PIX_FMT_NONE){
-		 avctx->pix_fmt=AV_PIX_FMT_YUV420P;
-	}else if(avctx-> pix_fmt != AV_PIX_FMT_YUV420P){
-		av_log(avctx, AV_LOG_ERROR, "Invalid Pix_FMT for NVMPI Only yuv420p is supported\n");
-		return AVERROR_INVALIDDATA;
-	}
-
-	nvmpi_context->ctx=nvmpi_create_decoder(codectype,NV_PIX_YUV420);
-
-	if(!nvmpi_context->ctx){
-		av_log(avctx, AV_LOG_ERROR, "Failed to nvmpi_create_decoder (code = %d).\n", AVERROR_EXTERNAL);
-		return AVERROR_EXTERNAL;
-	}
-   return 0;
-
-}
-
-
-
-static int nvmpi_close(AVCodecContext *avctx){
-
-	nvmpiDecodeContext *nvmpi_context = avctx->priv_data;
-	return nvmpi_decoder_close(nvmpi_context->ctx);
-
-}
-
-
-
-static int nvmpi_decode(AVCodecContext *avctx,void *data,int *got_frame, AVPacket *avpkt){
-
-	nvmpiDecodeContext *nvmpi_context = avctx->priv_data;
-	AVFrame *frame = data;
-	nvFrame _nvframe={0};
-	nvPacket packet;
-	uint8_t* ptrs[3];
-	int res,linesize[3];
-
-	if(avpkt->size){
-		packet.payload_size=avpkt->size;
-		packet.payload=avpkt->data;
-		packet.pts=avpkt->pts;
-
-		res=nvmpi_decoder_put_packet(nvmpi_context->ctx,&packet);
-	}
-
-	res=nvmpi_decoder_get_frame(nvmpi_context->ctx,&_nvframe,avctx->flags & AV_CODEC_FLAG_LOW_DELAY);
-
-	if(res<0)
-		return avpkt->size;
-
-	if (ff_get_buffer(avctx, frame, 0) < 0) {
-		return AVERROR(ENOMEM);
-
-	}
-
-	linesize[0]=_nvframe.linesize[0];
-	linesize[1]=_nvframe.linesize[1];
-	linesize[2]=_nvframe.linesize[2];
-
-	ptrs[0]=_nvframe.payload[0];
-	ptrs[1]=_nvframe.payload[1];
-	ptrs[2]=_nvframe.payload[2];
-
-	av_image_copy(frame->data, frame->linesize, (const uint8_t **) ptrs, linesize, avctx->pix_fmt, _nvframe.width,_nvframe.height);
-
-	frame->width=_nvframe.width;
-	frame->height=_nvframe.height;
-
-	frame->format=AV_PIX_FMT_YUV420P;
-	frame->pts=_nvframe.timestamp;
-	frame->pkt_dts = AV_NOPTS_VALUE;
-
-	avctx->coded_width=_nvframe.width;
-	avctx->coded_height=_nvframe.height;
-	avctx->width=_nvframe.width;
-	avctx->height=_nvframe.height;
-
-	*got_frame = 1;
-
-	return avpkt->size;
-}
-
-
-
-
-#define NVMPI_DEC_CLASS(NAME) \
-	static const AVClass nvmpi_##NAME##_dec_class = { \
-		.class_name = "nvmpi_" #NAME "_dec", \
-		.version    = LIBAVUTIL_VERSION_INT, \
-	};
-
-#define NVMPI_DEC(NAME, ID, BSFS) \
-	NVMPI_DEC_CLASS(NAME) \
-	AVCodec ff_##NAME##_nvmpi_decoder = { \
-		.name           = #NAME "_nvmpi", \
-		.long_name      = NULL_IF_CONFIG_SMALL(#NAME " (nvmpi)"), \
-		.type           = AVMEDIA_TYPE_VIDEO, \
-		.id             = ID, \
-		.priv_data_size = sizeof(nvmpiDecodeContext), \
-		.init           = nvmpi_init_decoder, \
-		.close          = nvmpi_close, \
-		.decode         = nvmpi_decode, \
-		.priv_class     = &nvmpi_##NAME##_dec_class, \
-		.capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_AVOID_PROBING | AV_CODEC_CAP_HARDWARE, \
-		.pix_fmts	=(const enum AVPixelFormat[]){AV_PIX_FMT_YUV420P,AV_PIX_FMT_NV12,AV_PIX_FMT_NONE},\
-		.bsfs           = BSFS, \
-		.wrapper_name   = "nvmpi", \
-	};
-
-
-
-NVMPI_DEC(h264,  AV_CODEC_ID_H264,"h264_mp4toannexb");
-NVMPI_DEC(hevc,  AV_CODEC_ID_HEVC,"hevc_mp4toannexb");
-NVMPI_DEC(mpeg2, AV_CODEC_ID_MPEG2VIDEO,NULL);
-NVMPI_DEC(mpeg4, AV_CODEC_ID_MPEG4,NULL);
-NVMPI_DEC(vp9,  AV_CODEC_ID_VP9,NULL);
-NVMPI_DEC(vp8, AV_CODEC_ID_VP8,NULL);
