From e823c8e1878c481251d97a2392c9eb168d2af9e1 Mon Sep 17 00:00:00 2001
From: CTCaer <ctcaer@gmail.com>
Date: Fri, 5 Nov 2021 21:43:52 +0000
Subject: [PATCH 4/7] formats: NVV4L2 depends on base decoders context init

---
 libavformat/utils.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/libavformat/utils.c b/libavformat/utils.c
index 1384b56771..2a02e69f06 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -211,6 +211,27 @@ static const AVCodec *find_probe_decoder(AVFormatContext *s, const AVStream *st,
     if (codec_id == AV_CODEC_ID_H264)
         return avcodec_find_decoder_by_name("h264");
 #endif
+    /* NVV4L2 decoders depend on context init from base decoders */
+#if CONFIG_HEVC_NVV4L2DEC_DECODER && CONFIG_HEVC_DECODER
+    if (codec_id == AV_CODEC_ID_HEVC)
+        return avcodec_find_decoder_by_name("hevc");
+#endif
+#if CONFIG_MPEG2_NVV4L2DEC_DECODER && CONFIG_MPEG2VIDEO_DECODER
+    if (codec_id == AV_CODEC_ID_MPEG2VIDEO)
+        return avcodec_find_decoder_by_name("mpeg2video");
+#endif
+#if CONFIG_MPEG4_NVV4L2DEC_DECODER && CONFIG_MPEG4_DECODER
+    if (codec_id == AV_CODEC_ID_MPEG4)
+        return avcodec_find_decoder_by_name("mpeg4");
+#endif
+#if CONFIG_VP8_NVV4L2DEC_DECODER && CONFIG_VP8_DECODER
+    if (codec_id == AV_CODEC_ID_VP8)
+        return avcodec_find_decoder_by_name("vp8"); 
+#endif
+#if CONFIG_VP9_NVV4L2DEC_DECODER && CONFIG_VP9_DECODER
+    if (codec_id == AV_CODEC_ID_VP9)
+        return avcodec_find_decoder_by_name("vp9");
+#endif
 
     codec = find_decoder(s, st, codec_id);
     if (!codec)
-- 
2.25.1

