From 6ac0a8b4d68701c9128aeb9b4f2f5d6ea77487f5 Mon Sep 17 00:00:00 2001
From: CTCaer <ctcaer@gmail.com>
Date: Fri, 5 Nov 2021 21:48:51 +0000
Subject: [PATCH 6/7] codecs: nvmpi: fix frame width/height

---
 libavcodec/nvmpi_enc.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/libavcodec/nvmpi_enc.c b/libavcodec/nvmpi_enc.c
index 0e1390e944..9f10894576 100644
--- a/libavcodec/nvmpi_enc.c
+++ b/libavcodec/nvmpi_enc.c
@@ -139,13 +139,12 @@ static int nvmpi_encode_frame(AVCodecContext *avctx, AVPacket *pkt,const AVFrame
 		_nvframe.payload[1]=frame->data[1];
 		_nvframe.payload[2]=frame->data[2];
 
-		_nvframe.payload_size[0]=frame->linesize[0]*frame->height;
-		_nvframe.payload_size[1]=frame->linesize[1]*frame->height/2;
-		_nvframe.payload_size[2]=frame->linesize[2]*frame->height/2;
-
 		_nvframe.linesize[0]=frame->linesize[0];
 		_nvframe.linesize[1]=frame->linesize[1];
 		_nvframe.linesize[2]=frame->linesize[2];
+		
+		_nvframe.width = frame->width;
+		_nvframe.height = frame->height;
 
 		_nvframe.timestamp=frame->pts;
 
-- 
2.25.1

