diff -Naur ffmpeg-4.3.2-Matrix-19.1/configure ffmpeg-4.3.2-Matrix-19.1-2/configure
--- ffmpeg-4.3.2-Matrix-19.1/configure	2021-11-06 09:35:11.578351617 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/configure	2021-11-06 09:36:29.391631255 +0100
@@ -6509,7 +6509,7 @@
                              { enabled libudev ||
                                die "ERROR: v4l2-request requires --enable-libudev"; }
 enabled vapoursynth       && require_pkg_config vapoursynth "vapoursynth-script >= 42" VSScript.h vsscript_init
-enabled nvmpi		  && require_pkg_config nvmpi nvmpi nvmpi.h nvmpi_create_encoder
+enabled nvmpi		  && require_headers nvmpi.h
 
 
 if enabled gcrypt; then
