diff -Naur ffmpeg-4.3.2-Matrix-19.1/libavcodec/nvv4l2_dec.c ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/nvv4l2_dec.c
--- ffmpeg-4.3.2-Matrix-19.1/libavcodec/nvv4l2_dec.c	2021-11-12 07:05:54.935143353 +0100
+++ ffmpeg-4.3.2-Matrix-19.1-2/libavcodec/nvv4l2_dec.c	2021-11-13 01:15:45.701324394 +0100
@@ -1,25 +1,22 @@
-/*
- * Copyright (c) 2020, NVIDIA CORPORATION
- * Copyright (c) 2021, CTCaer
- *
- * Permission is hereby granted, free of charge, to any person obtaining a
- * copy of this software and associated documentation files (the "Software"),
- * to deal in the Software without restriction, including without limitation
- * the rights to use, copy, modify, merge, publish, distribute, sublicense,
- * and/or sell copies of the Software, and to permit persons to whom the
- * Software is furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
- * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
- * DEALINGS IN THE SOFTWARE.
- */
+/* * Copyright (c) 2020, NVIDIA CORPORATION * Copyright (c) 
+ 2021, CTCaer * * Permission is hereby granted, free of charge, 
+ to any person obtaining a * copy of this software and 
+ associated documentation files (the "Software"), * to deal in 
+ the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, 
+ sublicense, * and/or sell copies of the Software, and to permit 
+ persons to whom the * Software is furnished to do so, subject 
+ to the following conditions: * * The above copyright notice and 
+ this permission notice shall be included in * all copies or 
+ substantial portions of the Software. * * THE SOFTWARE IS 
+ PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR * 
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
+ MERCHANTABILITY, * FITNESS FOR A PARTICULAR PURPOSE AND 
+ NONINFRINGEMENT.  IN NO EVENT SHALL * THE AUTHORS OR COPYRIGHT 
+ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER * LIABILITY, 
+ WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING * 
+ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR 
+ OTHER * DEALINGS IN THE SOFTWARE. */
 
 /**
  * Execution command:
@@ -1379,10 +1376,12 @@
         return processed_size;
     }
 
-    /* HACK: Detect seeking and drain the whole stack to avoid issues with bad players. */
-    if (abs(avpkt->pts - _nvframe.timestamp) > 5000000) {
-         while (nvv4l2dec_decoder_get_frame(avctx, nvv4l2_context->ctx, &_nvframe) != AVERROR(EAGAIN));
-         _nvframe.timestamp = avpkt->pts;
+    if (!avctx->flags & AV_CODEC_FLAG_LOW_DELAY) {
+         /* HACK: Detect seeking and drain the whole stack to avoid issues with bad players. */
+         if (abs(avpkt->pts - _nvframe.timestamp) > 5000000) {
+             while (nvv4l2dec_decoder_get_frame(avctx, nvv4l2_context->ctx, &_nvframe) != AVERROR(EAGAIN));
+             _nvframe.timestamp = avpkt->pts;
+         }
     }
 
     if (ff_get_buffer(avctx, frame, 0) < 0)
