diff -Naur ffmpeg-4.4.1-Nexus-Alpha1-2/configure ffmpeg-4.4.1-Nexus-Alpha1/configure
--- ffmpeg-4.4.1-Nexus-Alpha1-2/configure	2022-10-10 21:51:13.330387005 +0200
+++ ffmpeg-4.4.1-Nexus-Alpha1/configure	2022-10-10 22:01:51.723150253 +0200
@@ -3068,8 +3068,8 @@
 qsvvpp_select="qsv"
 vaapi_encode_deps="vaapi"
 v4l2_m2m_deps="linux_videodev2_h sem_timedwait"
-nvv4l2_deps="libv4l2 pthreads linux_videodev2_h"
-nvv4l2_extralibs="-lnvbuf_utils"
+nvv4l2_deps="pthreads"
+nvv4l2_extralibs="-lnvbuf_utils -lv4l2"
 
 hwupload_cuda_filter_deps="ffnvcodec"
 scale_npp_filter_deps="ffnvcodec libnpp"
@@ -6829,10 +6829,7 @@
     check_type "ffnvcodec/dynlink_cuda.h ffnvcodec/dynlink_cuviddec.h" "CUVIDAV1PICPARAMS"
 fi
 
-enabled nvv4l2 && add_ldflags '-L/usr/lib/aarch64-linux-gnu/tegra' \
-               && check_headers linux/videodev2.h \
-               && require_pkg_config libv4l2 libv4l2 libv4l2.h v4l2_ioctl
-
+enabled nvv4l2 && add_ldflags '-L/usr/lib/aarch64-linux-gnu/tegra'
 enabled amf &&
     check_cpp_condition amf "AMF/core/Version.h" \
         "(AMF_VERSION_MAJOR << 48 | AMF_VERSION_MINOR << 32 | AMF_VERSION_RELEASE << 16 | AMF_VERSION_BUILD_NUM) >= 0x0001000400090000"
