From 157e6cffa1464b66809d32b5d511f62882068dcd Mon Sep 17 00:00:00 2001
From: Azkali Manad <a.ffcc7@gmail.com>
Date: Wed, 25 Jan 2023 03:42:59 +0100
Subject: [PATCH 25/39] codecs: nvv4l2: add encode.h header needed for
 ff_alloc_packet

---
 libavcodec/nvv4l2_enc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/nvv4l2_enc.c b/libavcodec/nvv4l2_enc.c
index 7f1aad1b32..6c3d276f31 100644
--- a/libavcodec/nvv4l2_enc.c
+++ b/libavcodec/nvv4l2_enc.c
@@ -29,6 +29,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include "codec_internal.h"
+#include "encode.h"
 #include "libavutil/imgutils.h"
 #include "libavutil/log.h"
 #include "libavutil/opt.h"
-- 
2.25.1

