diff -Naur ffmpeg-6.0/libavcodec/nvv4l2_dec.c ffmpeg-6.0-2/libavcodec/nvv4l2_dec.c
--- ffmpeg-6.0/libavcodec/nvv4l2_dec.c	2024-01-26 23:28:12.822560258 +0100
+++ ffmpeg-6.0-2/libavcodec/nvv4l2_dec.c	2024-01-27 00:52:51.994386987 +0100
@@ -1181,11 +1181,11 @@
     if (_nvframe.pts != AV_NOPTS_VALUE) {
         avframe->pts = _nvframe.pts;
     } else {
-        avframe->pts = _nvframe.pts;
+        avframe->pts = AV_NOPTS_VALUE;
         avframe->reordered_opaque = _nvframe.user_pts;
     }
 
-    avframe->key_frame = 0;
+    //avframe->key_frame = 0;
 
     avctx->coded_width = _nvframe.width;
     avctx->coded_height = _nvframe.height;
