From c1447a426afa49ed23384f2b135ea9748c98d568 Mon Sep 17 00:00:00 2001
From: Azkali Manad <a.ffcc7@gmail.com>
Date: Wed, 25 Jan 2023 03:45:50 +0100
Subject: [PATCH 26/39] codecs: nvv4l2: fix missing comma in FFCodec struct

---
 libavcodec/nvv4l2_dec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/nvv4l2_dec.c b/libavcodec/nvv4l2_dec.c
index 26e7363823..fe54883522 100644
--- a/libavcodec/nvv4l2_dec.c
+++ b/libavcodec/nvv4l2_dec.c
@@ -1212,7 +1212,7 @@ nvv4l2dec_decode(AVCodecContext *avctx, void *data, int *got_frame,
         .priv_data_size = sizeof(nvv4l2DecodeContext),                                \
         .init           = nvv4l2dec_init,                                             \
         .close          = nvv4l2dec_close,                                            \
-	FF_CODEC_DECODE_CB(nvv4l2dec_decode)					      \
+	FF_CODEC_DECODE_CB(nvv4l2dec_decode),					      \
         .flush          = nvv4l2dec_flush,                                            \
         .p.priv_class     = &nvv4l2_##NAME##_dec_class,                                 \
         .p.capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_HARDWARE |                \
-- 
2.25.1

