From cd758dad831c7f7f115b0fcc8646e241d338c800 Mon Sep 17 00:00:00 2001
From: Azkali Manad <a.ffcc7@gmail.com>
Date: Wed, 25 Jan 2023 03:52:47 +0100
Subject: [PATCH 27/39] codecs: nvv4l2: remove second size argument as it is
 unneeded

---
 libavcodec/nvv4l2_enc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/nvv4l2_enc.c b/libavcodec/nvv4l2_enc.c
index 6c3d276f31..39a179b255 100644
--- a/libavcodec/nvv4l2_enc.c
+++ b/libavcodec/nvv4l2_enc.c
@@ -1266,7 +1266,7 @@ nvv4l2enc_encode(AVCodecContext *avctx, AVPacket *pkt,
     if (nvv4l2_encoder_get_packet(avctx, ctx, &packet))
         return 0;
 
-    ff_alloc_packet(avctx, pkt, packet.payload_size, packet.payload_size);
+    ff_alloc_packet(avctx, pkt, packet.payload_size);
 
     memcpy(pkt->data, packet.payload, packet.payload_size);
     pkt->dts = pkt->pts = packet.pts;
-- 
2.25.1

