From b4c1473d7a0db6efb31e5d5f5c7b483736ac593a Mon Sep 17 00:00:00 2001
From: theofficialgman <28281419+theofficialgman@users.noreply.github.com>
Date: Thu, 18 Jan 2024 17:48:52 -0500
Subject: [PATCH 33/39] codecs: nvv4l2: more rename AVCodec to fit API renames

---
 libavcodec/nvv4l2_dec.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libavcodec/nvv4l2_dec.c b/libavcodec/nvv4l2_dec.c
index 79116ec858..a92b191b24 100644
--- a/libavcodec/nvv4l2_dec.c
+++ b/libavcodec/nvv4l2_dec.c
@@ -39,9 +39,9 @@
 
 #include "nvv4l2.h"
 
-extern AVCodec ff_h264_decoder;
-extern AVCodec ff_hevc_decoder;
-extern AVCodec ff_vp9_decoder;
+extern const FFCodec ff_h264_decoder;
+extern const FFCodec ff_hevc_decoder;
+extern const FFCodec ff_vp9_decoder;
 
 /*
  ** Output plane format support:
-- 
2.25.1

