From 8d78a510e496ea473ea7392b828e17165bb6b6ac Mon Sep 17 00:00:00 2001
From: theofficialgman <28281419+theofficialgman@users.noreply.github.com>
Date: Sun, 21 Jan 2024 02:01:10 -0500
Subject: [PATCH 36/39] codecs: nvv4l2: update for ABI rework

---
 libavcodec/nvv4l2_dec.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/nvv4l2_dec.c b/libavcodec/nvv4l2_dec.c
index 5ccea0830b..27e06da1b5 100644
--- a/libavcodec/nvv4l2_dec.c
+++ b/libavcodec/nvv4l2_dec.c
@@ -1175,12 +1175,11 @@ static int nvv4l2dec_close(AVCodecContext *avctx)
 }
 
 static int
-nvv4l2dec_decode(AVCodecContext *avctx, void *data, int *got_frame,
+nvv4l2dec_decode(AVCodecContext *avctx, AVFrame *avframe, int *got_frame,
                  AVPacket *avpkt)
 {
     nvv4l2DecodeContext *nvv4l2_ctx = avctx->priv_data;
     nvv4l2_ctx_t *ctx = nvv4l2_ctx->ctx;
-    AVFrame *avframe = (AVFrame *)data;
     NvFrame _nvframe = { 0 };
     int processed_size = 0;
     int buf_index = -1;
-- 
2.25.1

