From a8e8a2151948790f9702802aa87b65bbf5049ac1 Mon Sep 17 00:00:00 2001
From: azkali <a.ffcc7@gmail.com>
Date: Tue, 23 Jan 2024 18:50:41 +0100
Subject: [PATCH 39/39] fftools/ffmpeg: set find_codec_or_die logctx to NULL

---
 fftools/ffmpeg_demux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fftools/ffmpeg_demux.c b/fftools/ffmpeg_demux.c
index f791c3a898..975a9f5f17 100644
--- a/fftools/ffmpeg_demux.c
+++ b/fftools/ffmpeg_demux.c
@@ -575,7 +575,7 @@ static const AVCodec *choose_decoder(const OptionsContext *o, AVFormatContext *s
     }
 #endif
 
-    const AVCodec *codec = find_codec_or_die(codec_name, st->codecpar->codec_type, 0);
+    const AVCodec *codec = find_codec_or_die(NULL, codec_name, st->codecpar->codec_type, 0);
     st->codecpar->codec_id = codec->id;
     return codec;
 }
-- 
2.25.1

