diff -Naur ffmpeg-6.0/configure ffmpeg-6.0-2/configure
--- ffmpeg-6.0/configure        2023-06-04 21:55:16.653155167 +0200
+++ ffmpeg-6.0-2/configure      2023-06-04 22:01:08.003471635 +0200
@@ -3131,8 +3131,8 @@
 qsvvpp_select="qsv"
 vaapi_encode_deps="vaapi"
 v4l2_m2m_deps="linux_videodev2_h sem_timedwait"
-nvv4l2_deps="libv4l2 pthreads linux_videodev2_h"
-nvv4l2_extralibs="-lnvbuf_utils"
+nvv4l2_deps="pthreads linux_videodev2_h"
+nvv4l2_extralibs="-lnvbuf_utils -lv4l2"
 
 bilateral_cuda_filter_deps="ffnvcodec"
 bilateral_cuda_filter_deps_any="cuda_nvcc cuda_llvm"
@@ -7069,9 +7069,7 @@
     check_type "ffnvcodec/dynlink_cuda.h ffnvcodec/dynlink_cuviddec.h" "CUVIDAV1PICPARAMS"
 fi
 
-enabled nvv4l2 && add_ldflags '-L/usr/lib/aarch64-linux-gnu/tegra' \
-               && check_headers linux/videodev2.h \
-               && require_pkg_config libv4l2 libv4l2 libv4l2.h v4l2_ioctl
+enabled nvv4l2 && add_ldflags "-L/usr/lib/aarch64-linux-gnu/tegra"
 
 enabled amf &&
     check_cpp_condition amf "AMF/core/Version.h" \
