diff -Naur pycrypto-2.3/setup.py pycrypto-2.3.patch/setup.py
--- pycrypto-2.3/setup.py	2010-08-27 06:41:55.000000000 +0200
+++ pycrypto-2.3.patch/setup.py	2011-07-20 05:46:14.760364851 +0200
@@ -37,7 +37,7 @@
 __revision__ = "$Id$"
 
 from distutils import core
-from distutils.core import Extension, Command
+from setuptools import Extension, Command
 from distutils.command.build_ext import build_ext
 import os, sys
 import struct
