diff -uNr a/Makefile b/Makefile
--- a/Makefile	2026-02-08 11:44:55.149302000 +0900
+++ b/Makefile	2026-02-08 17:13:54.691343789 +0900
@@ -367,6 +367,7 @@
    ASFLAGS = -f elf64 -d ELF_TYPE
 
 # Rockchip RK3288 e.g. Asus Tinker Board / RK3328 e.g. PINE64 Rock64 / RK3399 e.g. PINE64 RockPro64 - 32-bit userspace
+#          RK356X e.g. radxa ROCK 3A / RK3576 e.g. radxa ROCK 4D / RK3588 e.g. radxa ROCK 5A
 else ifneq (,$(findstring RK,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
@@ -380,6 +381,16 @@
          CPUFLAGS += -mtune=cortex-a53
          GLES = 1
       endif
+   else ifneq (,$(findstring RK35,$(platform)))
+      CPUFLAGS += -march=armv8-a+crc
+      GLES3 = 1
+      ifneq (,$(findstring RK356X,$(platform)))
+         CPUFLAGS += -mtune=cortex-a55
+      else ifneq (,$(findstring RK3576,$(platform)))
+         CPUFLAGS += -mtune=cortex-a72.cortex-a53
+      else ifneq (,$(findstring RK3588,$(platform)))
+         CPUFLAGS += -mtune=cortex-a76.cortex-a55
+      endif
    else ifneq (,$(findstring RK3288,$(platform)))
       CPUFLAGS += -march=armv7ve -mtune=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
       GLES3 = 1
@@ -390,8 +401,13 @@
    endif
 
    GL_LIB := -lGLESv2
-   HAVE_NEON = 1
-   WITH_DYNAREC=arm
+   ifneq (,$(findstring _64,$(platform)))
+     HAVE_NEON = 0
+     WITH_DYNAREC = aarch64
+   else
+     HAVE_NEON = 1
+     WITH_DYNAREC = arm
+   endif
    COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
 
