diff -Naur mesa-7.1.orig/src/mesa/drivers/directfb/Makefile mesa-7.1/src/mesa/drivers/directfb/Makefile
--- mesa-7.1.orig/src/mesa/drivers/directfb/Makefile	2008-08-30 17:04:52.000000000 +0200
+++ mesa-7.1/src/mesa/drivers/directfb/Makefile	2008-08-30 17:12:21.000000000 +0200
@@ -25,7 +25,7 @@
 
 DIRECTFBGL_MESA = libidirectfbgl_mesa.so
 
-LIBS = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+LIBS = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .c.o:
diff -Naur mesa-7.1.orig/src/mesa/drivers/dri/Makefile.template mesa-7.1/src/mesa/drivers/dri/Makefile.template
--- mesa-7.1.orig/src/mesa/drivers/dri/Makefile.template	2008-08-30 17:04:52.000000000 +0200
+++ mesa-7.1/src/mesa/drivers/dri/Makefile.template	2008-08-30 17:07:18.000000000 +0200
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.so
 
 COMMON_SOURCES = \
         ../common/utils.c \
diff -Naur mesa-7.1.orig/src/mesa/drivers/fbdev/Makefile mesa-7.1/src/mesa/drivers/fbdev/Makefile
--- mesa-7.1.orig/src/mesa/drivers/fbdev/Makefile	2008-08-30 17:04:52.000000000 +0200
+++ mesa-7.1/src/mesa/drivers/fbdev/Makefile	2008-08-30 17:08:06.000000000 +0200
@@ -14,7 +14,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .c.o:
diff -Naur mesa-7.1.orig/src/mesa/drivers/osmesa/Makefile mesa-7.1/src/mesa/drivers/osmesa/Makefile
--- mesa-7.1.orig/src/mesa/drivers/osmesa/Makefile	2008-08-30 17:04:52.000000000 +0200
+++ mesa-7.1/src/mesa/drivers/osmesa/Makefile	2008-08-30 17:09:15.000000000 +0200
@@ -19,7 +19,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .PHONY: osmesa8
diff -Naur mesa-7.1.orig/src/mesa/drivers/x11/Makefile mesa-7.1/src/mesa/drivers/x11/Makefile
--- mesa-7.1.orig/src/mesa/drivers/x11/Makefile	2008-08-30 17:04:52.000000000 +0200
+++ mesa-7.1/src/mesa/drivers/x11/Makefile	2008-08-30 17:10:53.000000000 +0200
@@ -43,7 +43,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 
diff -Naur mesa-7.1.orig/src/mesa/Makefile mesa-7.1/src/mesa/Makefile
--- mesa-7.1.orig/src/mesa/Makefile	2008-08-30 17:04:51.000000000 +0200
+++ mesa-7.1/src/mesa/Makefile	2008-08-30 17:11:54.000000000 +0200
@@ -20,8 +20,8 @@
 
 
 # Default: build dependencies, then asm_subdirs, then convenience
-# libs (.a) and finally the device drivers:
-default: depend asm_subdirs libmesa.a libglapi.a driver_subdirs
+# libs (.so) and finally the device drivers:
+default: depend asm_subdirs libmesa.so libglapi.so driver_subdirs
 
 
 
@@ -29,17 +29,17 @@
 # Helper libraries used by many drivers:
 
 # Make archive of core mesa object files
-libmesa.a: $(MESA_OBJECTS)
-	@ $(MKLIB) -o mesa -static $(MESA_OBJECTS)
+libmesa.so: $(MESA_OBJECTS)
+	@ $(MKLIB) -o mesa -dlopen $(MESA_OBJECTS)
 
 # Make archive of gl* API dispatcher functions only
-libglapi.a: $(GLAPI_OBJECTS)
-	@ $(MKLIB) -o glapi -static $(GLAPI_OBJECTS)
+libglapi.so: $(GLAPI_OBJECTS)
+	@ $(MKLIB) -o glapi -dlopen $(GLAPI_OBJECTS)
 
 
 ######################################################################
 # Device drivers
-driver_subdirs: libmesa.a libglapi.a
+driver_subdirs: libmesa.so libglapi.so
 	(cd drivers && $(MAKE))
 
 
@@ -127,7 +127,7 @@
 clean:
 	-rm -f */*.o
 	-rm -f */*/*.o
-	-rm -f depend depend.bak libmesa.a libglapi.a
+	-rm -f depend depend.bak libmesa.so libglapi.so
 	-rm -f drivers/*/*.o
 	-rm -f *.pc
 	-@cd drivers/dri && $(MAKE) clean
