--- configure.ac.mach64	2008-09-05 13:53:24.000000000 +1000
+++ configure.ac	2008-09-05 13:53:39.000000000 +1000
@@ -656,7 +656,7 @@
             # because there is no x86-64 system where they could *ever*
             # be used.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="i915 i965 mach64 mga r128 r200 r300 radeon \
+                DRI_DIRS="i915 i965 mga r128 r200 r300 radeon \
                     savage tdfx unichrome swrast"
             fi
             ;;
@@ -664,13 +664,13 @@
             # Build only the drivers for cards that exist on PowerPC.
             # At some point MGA will be added, but not yet.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="mach64 r128 r200 r300 radeon tdfx swrast"
+                DRI_DIRS="r128 r200 r300 radeon tdfx swrast"
             fi
             ;;
         sparc*)
             # Build only the drivers for cards that exist on sparc`
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="mach64 r128 r200 r300 radeon ffb swrast"
+                DRI_DIRS="r128 r200 r300 radeon ffb swrast"
             fi
             ;;
         esac
@@ -689,7 +689,7 @@
         # ffb and gamma are missing because they have not been converted
         # to use the new interface.
         if test "x$DRI_DIRS" = "xyes"; then
-            DRI_DIRS="i810 i915 i965 mach64 mga r128 r200 r300 radeon tdfx \
+            DRI_DIRS="i810 i915 i965 mga r128 r200 r300 radeon tdfx \
                 unichrome savage sis swrast"
         fi
         ;;
@@ -704,7 +704,7 @@
 
     # default drivers
     if test "x$DRI_DIRS" = "xyes"; then
-        DRI_DIRS="i810 i915 i965 mach64 mga r128 r200 r300 radeon s3v \
+        DRI_DIRS="i810 i915 i965 mga r128 r200 r300 radeon s3v \
             savage sis tdfx trident unichrome ffb swrast"
     fi
 
