diff -uNr pycurl-7.19.0-orig/setup.py pycurl-7.19.0-new/setup.py
--- pycurl-7.19.0-orig/setup.py	2012-12-21 19:20:27.000000000 +0100
+++ pycurl-7.19.0-new/setup.py	2012-12-21 19:31:11.000000000 +0100
@@ -104,6 +104,7 @@
             if e[2:] == 'ssl':
                 define_macros.append(('HAVE_CURL_OPENSSL', 1))
             if e[2:] == 'gnutls':
+                libraries.append("gcrypt")
                 define_macros.append(('HAVE_CURL_GNUTLS', 1))
         elif e[:2] == "-L":
             library_dirs.append(e[2:])
