--- pycurl/setup.py~	2008-04-22 17:00:45.000000000 +0300
+++ pycurl/setup.py	2008-07-03 21:53:36.000000000 +0300
@@ -97,8 +97,7 @@
         else:
             extra_compile_args.append(e)
     libs = split_quoted(
-        os.popen("'%s' --libs" % CURL_CONFIG).read()+\
-        os.popen("'%s' --static-libs" % CURL_CONFIG).read())
+        os.popen("'%s' --libs" % CURL_CONFIG).read())
     for e in libs:
         if e[:2] == "-l":
             libraries.append(e[2:])
