diff -Naur pycurl-7.19.0/setup.py pycurl-7.19.0.patch/setup.py
--- pycurl-7.19.0/setup.py	2008-09-09 19:40:34.000000000 +0200
+++ pycurl-7.19.0.patch/setup.py	2011-09-29 02:07:22.584412664 +0200
@@ -11,8 +11,8 @@
 
 import glob, os, re, sys, string
 import distutils
-from distutils.core import setup
-from distutils.extension import Extension
+from setuptools import setup
+from setuptools import Extension
 from distutils.util import split_quoted
 from distutils.version import LooseVersion
 
