diff -uNr a/gfx/drivers_context/drm_go2_ctx.c b/gfx/drivers_context/drm_go2_ctx.c
--- a/gfx/drivers_context/drm_go2_ctx.c	2026-01-20 23:02:52.000000000 +0900
+++ b/gfx/drivers_context/drm_go2_ctx.c	2026-02-13 21:27:41.270276346 +0900
@@ -43,6 +43,7 @@
 #include "../../verbosity.h"
 #include "../../frontend/frontend_driver.h"
 #include "../common/drm_common.h"
+#include "../common/gl_common.h"
 
 #include <go2/display.h>
 #include <drm/drm_fourcc.h>
