diff -uNr a/gfx/video_driver.c b/gfx/video_driver.c
--- a/gfx/video_driver.c	2026-01-20 23:02:52.000000000 +0900
+++ b/gfx/video_driver.c	2026-03-02 14:09:49.265397889 +0900
@@ -398,9 +398,10 @@
 #ifdef SWITCH
    &video_switch,
 #endif
-#ifdef HAVE_ODROIDGO2
-   &video_oga,
-#endif
+// temporary disabled for RetroArch 1.22.2 on RK3326
+// #ifdef HAVE_ODROIDGO2
+//    &video_oga,
+// #endif
 #if defined(HAVE_SDL) && !defined(HAVE_SDL_DINGUX)
    &video_sdl,
 #endif
