diff -uNr a/gfx/common/wayland_common.c b/gfx/common/wayland_common.c
--- a/gfx/common/wayland_common.c	2026-02-09 16:35:05.568964600 +0900
+++ b/gfx/common/wayland_common.c	2026-02-10 10:47:39.175279750 +0900
@@ -747,6 +747,21 @@
    wl->floating_width               = SPLASH_WINDOW_WIDTH;
    wl->floating_height              = SPLASH_WINDOW_HEIGHT;
 
+   for( int retry_cnt = 0 ; retry_cnt < 5 ; retry_cnt++ )
+   {
+      if (!wl->input.dpy)
+      {
+         RARCH_WARN("[Attempt %d/5] Unable to connect to Wayland server, retrying.\n" , retry_cnt+1);
+         sleep(1);
+         wl->input.dpy = wl_display_connect(NULL);
+         continue;
+      }
+      else
+      {
+         break;
+      }
+   }
+
    if (!wl->input.dpy)
    {
       RARCH_ERR("[Wayland] Failed to connect to Wayland server.\n");
