diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index cab64de..042d97e 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -236,6 +236,8 @@ KeyMap keyMap[] = {
   { KEY_COMPOSE       , XBMCK_LSUPER      },
   { KEY_STOP          , XBMCK_MEDIA_STOP  },
   { KEY_HELP          , XBMCK_HELP        },
+  { KEY_SLEEP         , XBMCK_SLEEP       },
+  { KEY_WAKEUP        , XBMCK_SLEEP       },
   { KEY_MENU          , XBMCK_MENU        },
   { KEY_CLOSECD       , XBMCK_EJECT       },
   { KEY_EJECTCD       , XBMCK_EJECT       },
