diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 3875138..bfc7e6e 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -228,6 +228,7 @@ KeyMap keyMap[] = {
   { KEY_VOLUMEDOWN    , XBMCK_VOLUME_DOWN },
   { KEY_VOLUMEUP      , XBMCK_VOLUME_UP   },
   { KEY_POWER         , XBMCK_POWER       },
+  { KEY_COFFEE        , XBMCK_VOLUME_MUTE },
   { KEY_KPEQUAL       , XBMCK_KP_EQUALS   },
   { KEY_PAUSE         , XBMCK_PAUSE       },
   { KEY_PAUSECD       , XBMCK_PAUSE       },
@@ -247,6 +248,7 @@ KeyMap keyMap[] = {
   { KEY_RECORD        , XBMCK_RECORD      },
   { KEY_REWIND        , XBMCK_REWIND      },
   { KEY_PHONE         , XBMCK_PHONE       },
+  { KEY_HOMEPAGE      , XBMCK_BROWSER_HOME},
   { KEY_REFRESH       , XBMCK_SHUFFLE     },
   { KEY_SCROLLUP      , XBMCK_PAGEUP      },
   { KEY_SCROLLDOWN    , XBMCK_PAGEDOWN    },
