From 110d97e5cb6000fa3e82bd6bdae4752dfe643a34 Mon Sep 17 00:00:00 2001
From: "Chris \\\"Koying\\\" Browet" <cbro@semperpax.com>
Date: Sun, 23 Aug 2015 13:09:27 +0200
Subject: [PATCH] fixup: bitstream converter aml

---
 .../DVDCodecs/Video/DVDVideoCodecAmlogic.cpp          | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
index 28a425f..1f3df3a 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
@@ -132,7 +132,7 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
       m_pFormatName = "am-h264";
       // convert h264-avcC to h264-annex-b as h264-avcC
       // under streamers can have issues when seeking.
-      if (m_hints.extradata && *(uint8_t*)m_hints.extradata == 1)
+      if (m_hints.extradata)
       {
         m_bitstream = new CBitstreamConverter;
         m_bitstream->Open(m_hints.codec, (uint8_t*)m_hints.extradata, m_hints.extrasize, true);
@@ -213,13 +213,16 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
         return false;
       }
       m_pFormatName = "am-h265";
-      m_bitstream = new CBitstreamConverter();
-      m_bitstream->Open(m_hints.codec, (uint8_t*)m_hints.extradata, m_hints.extrasize, true);
-      // make sure we do not leak the existing m_hints.extradata
-      free(m_hints.extradata);
-      m_hints.extrasize = m_bitstream->GetExtraSize();
-      m_hints.extradata = malloc(m_hints.extrasize);
-      memcpy(m_hints.extradata, m_bitstream->GetExtraData(), m_hints.extrasize);
+      if (m_hints.extradata)
+      {
+        m_bitstream = new CBitstreamConverter();
+        m_bitstream->Open(m_hints.codec, (uint8_t*)m_hints.extradata, m_hints.extrasize, true);
+        // make sure we do not leak the existing m_hints.extradata
+        free(m_hints.extradata);
+        m_hints.extrasize = m_bitstream->GetExtraSize();
+        m_hints.extradata = malloc(m_hints.extrasize);
+        memcpy(m_hints.extradata, m_bitstream->GetExtraData(), m_hints.extrasize);
+      }
       break;
     default:
       CLog::Log(LOGDEBUG, "%s: Unknown hints.codec(%d", __MODULE_NAME__, m_hints.codec);
-- 
1.8.3.1

