diff --git a/arch/arm/boot/dts/amlogic/meson8b_m201C_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m201C_512M.dtd
index 91e2227..f007d77 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m201C_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m201C_512M.dtd
@@ -132,12 +132,12 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x00800000 0x00100000>;
+		reserve-memory = <0x01800000 0x00100000>;
 		reserve-iomap = "true";
 	        vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
 		4k2k_fb = <0>;
-	        display_size_default = <1280 720 1280 1440 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
+	        display_size_default = <1920 1080 1920 3240 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
 	};
 
 //$$ DEVICE="deinterlace"
@@ -202,7 +202,6 @@ void root_func(){
 		dev_name = "ppmgr";
 		status = "okay";
 		reserve-memory = <0x01000000>;//16M
-		reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -339,8 +338,8 @@ void root_func(){
 		port_name = "uart_ao";
 		status = "okay";
 		dev_name = "uart_ao";
-		pinctrl-names = "default";
-		pinctrl-0 = <&ao_uart_pins>;
+//		pinctrl-names = "default";
+//		pinctrl-0 = <&ao_uart_pins>;
 	};
 
 //$$ DEVICE="uart_0"
@@ -420,13 +419,14 @@ void root_func(){
 //$$ DEVICE="wifi_power"
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_STR = "power_gpio"
-//	wifi_power{
-//            compatible = "amlogic,wifi_power";
-//            dev_name = "wifi_power";
-//            status = "okay";
-//            power_gpio = "GPIOAO_6";
-//            power_gpio2 = "GPIOX_11";
-//	};
+	wifi_power{
+            compatible = "amlogic,wifi_power";
+            dev_name = "wifi_power";
+            status = "okay";
+            power_gpio = "GPIOAO_6";
+            power_gpio2 = "GPIOX_11";
+	    valid = "high";
+	};
 
 /// ***************************************************************************************
 ///	-	MMC
@@ -640,17 +640,17 @@ void root_func(){
 //$$ L3 PROP_U32 = "f_max"
 //$$ L3 PROP_U32 = "max_req_size"
 //$$ L2 PROP_U32 = "card_type"
-        // sdio{
-        //     status = "okay";
-        //     port = <3>;          /**0:sdio_a, 1:sdio_b, 2:sdio_c, 3:sdhc_a, 4:sdhc_b, 5:sdhc_c */
-        //     pinname = "sdio";
-        //     ocr_avail = <0x00200080>; // 3.3:0x200000, 1.8+3.3:0x00200080
-        //     caps = "MMC_CAP_4_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE", "MMC_CAP_UHS_SDR12","MMC_CAP_UHS_SDR25","MMC_CAP_UHS_SDR50","MMC_CAP_UHS_SDR104";
-        //     f_min = <300000>;
-        //     f_max = <100000000>;
-        //     max_req_size = <0x20000>;          /**128KB*/
-        //     card_type = <3>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
-        // };
+	     sdio{
+             status = "okay";
+             port = <3>;          /**0:sdio_a, 1:sdio_b, 2:sdio_c, 3:sdhc_a, 4:sdhc_b, 5:sdhc_c */
+             pinname = "sdio";
+             ocr_avail = <0x200000>; // 3.3:0x200000, 1.8+3.3:0x00200080
+             caps = "MMC_CAP_4_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE", "MMC_CAP_UHS_SDR12","MMC_CAP_UHS_SDR25","MMC_CAP_UHS_SDR50","MMC_CAP_UHS_SDR104";
+             f_min = <300000>;
+             f_max = <50000000>;
+             max_req_size = <0x20000>;          /**128KB*/
+             card_type = <3>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
+         };
     };
 
 
@@ -1564,10 +1564,10 @@ void root_func(){
 //$$ MATCH "uart_ao_pin_match" = "&ao_uart_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_STR 2 = "amlogic,pins"
-        ao_uart_pins:ao_uart{
-			amlogic,setmask=<10 0x1800>;
-			amlogic,pins="GPIOAO_0", "GPIOAO_1";
-		};
+//        ao_uart_pins:ao_uart{
+//			amlogic,setmask=<10 0x1800>;
+//			amlogic,pins="GPIOAO_0", "GPIOAO_1";
+//		};
 
 //$$ MATCH "uart_0_pin_match" = "&a_uart_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
