diff --git a/kernel/power/main.c b/kernel/power/main.c
index 0aefd6f57e0a..0dc6bac7d7fd 100644
--- a/kernel/power/main.c
+++ b/kernel/power/main.c
@@ -95,7 +95,7 @@ int pm_notifier_call_chain(unsigned long val)
 }
 
 /* If set, devices may be suspended and resumed asynchronously. */
-int pm_async_enabled = 1;
+int pm_async_enabled = 0;
 
 static ssize_t pm_async_show(struct kobject *kobj, struct kobj_attribute *attr,
 			     char *buf)
