--- a/CMakeLists.txt	2020-12-20 15:19:33.535997406 +0100
+++ b/CMakeLists.txt	2020-12-20 15:21:56.337546477 +0100
@@ -259,7 +259,9 @@
 else()
 	find_package(ICU COMPONENTS i18n uc data REQUIRED)
 	target_link_libraries(lcf ICU::i18n ICU::uc ICU::data)
-	target_compile_definitions(lcf PRIVATE LCF_SUPPORT_ICU=1)
+	INCLUDE(FindPkgConfig)
+	pkg_get_variable(ICUDEFS icu-i18n DEFS)
+	target_compile_definitions(lcf PRIVATE LCF_SUPPORT_ICU=1 ${ICUDEFS})
 	list(APPEND LIBLCF_DEPS "icu-i18n")
 endif()
 
