From dfdfa7fb1a50c21a784a67e5f99d8714ca853c07 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Mon, 20 Aug 2018 22:55:34 +0200
Subject: [PATCH] rockchip: board: save cpuid to env

---
 arch/arm/mach-rockchip/board.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index 233f0b6f9a..6c2021e32a 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -44,6 +44,7 @@ DECLARE_GLOBAL_DATA_PTR;
 static int rockchip_set_serialno(void)
 {
 	char serialno_str[VENDOR_SN_MAX];
+	char cpuid_str[CPUID_LEN * 2 + 1];
 	int ret = 0, i;
 	u8 cpuid[CPUID_LEN] = {0};
 	u8 low[CPUID_LEN / 2], high[CPUID_LEN / 2];
@@ -89,6 +90,13 @@ static int rockchip_set_serialno(void)
 		snprintf(serialno_str, sizeof(serialno_str), "%llx", serialno);
 
 		env_set("serial#", serialno_str);
+
+		memset(cpuid_str, 0, sizeof(cpuid_str));
+		for (i = 0; i < CPUID_LEN; i++) {
+			sprintf(&cpuid_str[i * 2], "%02x", cpuid[i]);
+		}
+
+		env_set("cpuid#", cpuid_str);
 #ifdef CONFIG_ROCKCHIP_VENDOR_PARTITION
 	}
 #endif
