--- a/shlib/Makefile.in	2016-11-10 12:51:02.859118277 +0000
+++ b/shlib/Makefile.in	2016-11-10 12:52:02.392451233 +0000
@@ -174,11 +174,10 @@
 
 $(SHARED_READLINE):	$(SHARED_OBJ)
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS)
-
+	$(SHOBJ_LD) ${SHLIB_LIBS} ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ)
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so xfree.so
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so xfree.so $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHLIB_LIBS} ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so xfree.so
 
 # Since tilde.c is shared between readline and bash, make sure we compile 
 # it with the right flags when it's built as part of readline
--- a/configure	2016-11-10 13:16:57.634721586 +0000
+++ b/configure	2016-11-10 13:17:21.061388105 +0000
@@ -6405,10 +6405,10 @@
 $as_echo_n "checking configuration for building shared libraries... " >&6; }
         eval `TERMCAP_LIB=$TERMCAP_LIB ${CONFIG_SHELL-/bin/sh} ${srcdir}/support/shobj-conf -C "${CC}" -c ${host_cpu} -o ${host_os} -v ${host_vendor}`
 
-#	case "$SHLIB_LIBS" in
-#	*curses*|*termcap*|*termlib*)	;;
-#	*)			SHLIB_LIBS="$SHLIB_LIBS $TERMCAP_LIB" ;;
-#	esac
+	case "$SHLIB_LIBS" in
+	*curses*|*termcap*|*termlib*)	;;
+	*)			SHLIB_LIBS="$SHLIB_LIBS $TERMCAP_LIB" ;;
+	esac
 
 
 
