diff -Naur Sick-Beard-0f076f4/data/css/default.css Sick-Beard-0f076f4.patch/data/css/default.css
--- Sick-Beard-0f076f4/data/css/default.css	2011-04-16 17:14:02.059365807 +0200
+++ Sick-Beard-0f076f4.patch/data/css/default.css	2011-04-16 17:47:20.371757075 +0200
@@ -281,16 +281,6 @@
 background-color:#000;
 color:#fff;
 }
-#donate {
-line-height:1em;
-background: #57442B;
-float: right;
-}
-#donate a,#donate a:hover {
-background-color:#57442B;
-border:0;
-padding:4px 15px 0px;
-}
 #content {
     min-height: 100px;
     background: #fff;
@@ -361,4 +351,4 @@
     text-shadow: 1px 1px 0 rgba(255,255,255,0.3);
     font-size: 1em;
 }
-div.ui-pnotify { min-width: 340px; max-width: 550px; width: auto !important;}
\ Kein Zeilenumbruch am Dateiende.
+div.ui-pnotify { min-width: 340px; max-width: 550px; width: auto !important;}
diff -Naur Sick-Beard-0f076f4/data/interfaces/default/config_general.tmpl Sick-Beard-0f076f4.patch/data/interfaces/default/config_general.tmpl
--- Sick-Beard-0f076f4/data/interfaces/default/config_general.tmpl	2011-04-16 17:14:02.055365889 +0200
+++ Sick-Beard-0f076f4.patch/data/interfaces/default/config_general.tmpl	2011-04-16 18:04:59.879961265 +0200
@@ -40,18 +40,6 @@
                         </div>
                         
                         <div class="field-pair">
-                            <input type="checkbox" name="version_notify" id="version_notify" #if $sickbeard.VERSION_NOTIFY then "checked=\"checked\"" else ""#/>
-                            <label class="clearfix" for="version_notify">
-                                <span class="component-title">Check for Update</span>
-                                <span class="component-desc">Show notification about updates for Sick Beard if available.</span>
-                            </label>
-                            <label class="nocheck clearfix">
-                                <span class="component-title">&nbsp;</span>
-                                <span class="component-desc">Checking for updates runs on startup and every 12 hours.</span>
-                            </label>
-                        </div>
-
-                        <div class="field-pair">
                             <label class="nocheck clearfix" for="log_dir">
                                 <span class="component-title">Logging Directory</span>
                                 <input type="text" name="log_dir" id="log_dir" value="$sickbeard.LOG_DIR" size="35" />
diff -Naur Sick-Beard-0f076f4/data/interfaces/default/config.tmpl Sick-Beard-0f076f4.patch/data/interfaces/default/config.tmpl
--- Sick-Beard-0f076f4/data/interfaces/default/config.tmpl	2011-04-16 17:14:02.054365910 +0200
+++ Sick-Beard-0f076f4.patch/data/interfaces/default/config.tmpl	2011-04-16 17:21:20.366435826 +0200
@@ -16,17 +16,10 @@
     <tr><td class="infoTableHeader">SB Arguments: </td><td class="infoTableCell">$sickbeard.MY_ARGS</td></tr>
     <tr><td class="infoTableHeader">SB Web Root: </td><td class="infoTableCell">$sickbeard.WEB_ROOT</td></tr>
     <tr><td class="infoTableHeader">Python Version: </td><td class="infoTableCell">$sys.version[:120]</td></tr>
-    <tr style="border-top: 1px dotted #666666;"><td class="infoTableHeader">Homepage </td><td class="infoTableCell"><a href="http://www.sickbeard.com/">http://www.sickbeard.com/</a></td></tr>
-    <tr><td class="infoTableHeader">Forums </td><td class="infoTableCell"><a href="http://sickbeard.com/forums/">http://sickbeard.com/forums/</a></td></tr>
-    <tr><td class="infoTableHeader">Source </td><td class="infoTableCell"><a href="https://github.com/midgetspy/Sick-Beard/">https://github.com/midgetspy/Sick-Beard/</a></td></tr>
-    <tr><td class="infoTableHeader">Bug Tracker &amp;<br/> Windows Builds </td><td class="infoTableCell"><a href="http://code.google.com/p/sickbeard/">http://code.google.com/p/sickbeard/</a></td></tr>
-    <tr><td class="infoTableHeader">Internet Relay Chat </td><td class="infoTableCell"><a href="irc://irc.freenode.net/#sickbeard"><i>#sickbeard</i> on <i>irc.freenode.net</i></a></td></tr>
+    <tr style="border-top: 1px dotted #666666;"><td class="infoTableHeader">Homepage </td><td class="infoTableCell"><a href="http://www.openelec.tv">http://www.openelec.tv</a></td></tr>
+    <tr><td class="infoTableHeader">Forums </td><td class="infoTableCell"><a href="http://www.openelec.tv/forum/">http://www.openelec.tv/forum/</a></td></tr>
+    <tr><td class="infoTableHeader">Internet Relay Chat </td><td class="infoTableCell"><a href="irc://irc.freenode.net/#openelec"><i>#openelec</i> on <i>irc.freenode.net</i></a></td></tr>
 </table>
 </div>
 
-<table class="infoTable" cellspacing="1" border="0" cellpadding="0"><tr>
-    <td><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donateCC_LG.gif" alt="[donate]" /></a></td>
-    <td>Sickbeard is free, but you can contribute by giving a <b><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;">donation</a></b>.</td>
-</tr></table>
-
 #include $os.path.join($sickbeard.PROG_DIR, "data/interfaces/default/inc_bottom.tmpl")
diff -Naur Sick-Beard-0f076f4/data/interfaces/default/inc_top.tmpl Sick-Beard-0f076f4.patch/data/interfaces/default/inc_top.tmpl
--- Sick-Beard-0f076f4/data/interfaces/default/inc_top.tmpl	2011-04-16 17:14:02.055365889 +0200
+++ Sick-Beard-0f076f4.patch/data/interfaces/default/inc_top.tmpl	2011-04-16 17:51:40.232667081 +0200
@@ -147,19 +147,6 @@
 </head>
 
 <body>
-#if $sickbeard.NEWEST_VERSION_STRING:
-<div id="upgrade-notification">
-    <div>
-        <span class="notify-text">$sickbeard.NEWEST_VERSION_STRING</span>
-    </div>
-</div>
-<div id="header-fix"></div>
-#end if
-<div id="header">
-    <a name="top"></a>
-    <span id="logo"><a href="$sbRoot/home/" title="Sick Beard homepage"><img alt="Sick Beard" src="$sbRoot/images/sickbeard_small.png" width="150" /></a></span>
-    <span id="versiontext">alpha $sickbeard.version.SICKBEARD_VERSION</span>
-</div>
 <ul id="MainMenu" class="sf-menu">
             <li id="NAVsystem" class="first"><a href="#" class="navIcon"><img src="$sbRoot/images/menu/system18.png" alt="" width="18" height="18" /></a>
                 <ul>
@@ -197,7 +184,6 @@
                         <li><a href="$sbRoot/errorlogs/viewlog"><img src="$sbRoot/images/menu/viewlog16.png" alt="" width="16" height="16" />View Log</a></li>
                 </ul>
             </li>
-            <li id="donate"><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donate_LG.gif" alt="[donate]" /></a></li>
 </ul>
 #if $varExists('submenu'):
 <div id="SubMenu">
diff -Naur Sick-Beard-0f076f4/sickbeard/__init__.py Sick-Beard-0f076f4.patch/sickbeard/__init__.py
--- Sick-Beard-0f076f4/sickbeard/__init__.py	2011-04-16 17:14:02.038366239 +0200
+++ Sick-Beard-0f076f4.patch/sickbeard/__init__.py	2011-04-16 18:13:07.293414961 +0200
@@ -446,7 +446,7 @@
 
         QUALITY_DEFAULT = check_setting_int(CFG, 'General', 'quality_default', SD)
         STATUS_DEFAULT = check_setting_int(CFG, 'General', 'status_default', SKIPPED)
-        VERSION_NOTIFY = check_setting_int(CFG, 'General', 'version_notify', 1)
+        VERSION_NOTIFY = check_setting_int(CFG, 'General', 'version_notify', 0)
         SEASON_FOLDERS_FORMAT = check_setting_str(CFG, 'General', 'season_folders_format', 'Season %02d')
         SEASON_FOLDERS_DEFAULT = bool(check_setting_int(CFG, 'General', 'season_folders_default', 0))
 
