diff -Naur Sick-Beard-0f076f4/sickbeard/__init__.py Sick-Beard-0f076f4.patch/sickbeard/__init__.py
--- Sick-Beard-0f076f4/sickbeard/__init__.py	2011-04-16 22:38:48.449071968 +0200
+++ Sick-Beard-0f076f4.patch/sickbeard/__init__.py	2011-04-16 22:52:23.704291427 +0200
@@ -184,7 +184,7 @@
 SAB_PASSWORD = None
 SAB_APIKEY = None
 SAB_CATEGORY = None
-SAB_HOST = ''
+SAB_HOST = '127.0.0.1:8081'
 
 NZBGET_PASSWORD = None
 NZBGET_CATEGORY = None
@@ -522,7 +522,7 @@
         SAB_PASSWORD = check_setting_str(CFG, 'SABnzbd', 'sab_password', '')
         SAB_APIKEY = check_setting_str(CFG, 'SABnzbd', 'sab_apikey', '')
         SAB_CATEGORY = check_setting_str(CFG, 'SABnzbd', 'sab_category', 'tv')
-        SAB_HOST = check_setting_str(CFG, 'SABnzbd', 'sab_host', '')
+        SAB_HOST = check_setting_str(CFG, 'SABnzbd', 'sab_host', '127.0.0.1:8081')
 
         NZBGET_PASSWORD = check_setting_str(CFG, 'NZBget', 'nzbget_password', 'tegbzn6789')
         NZBGET_CATEGORY = check_setting_str(CFG, 'NZBget', 'nzbget_category', 'tv')
