diff -Naur Sick-Beard-0f076f4/autoProcessTV/autoProcessTV.cfg.sample Sick-Beard-0f076f4.patch/autoProcessTV/autoProcessTV.cfg.sample
--- Sick-Beard-0f076f4/autoProcessTV/autoProcessTV.cfg.sample	2011-04-16 15:57:38.000000000 +0200
+++ Sick-Beard-0f076f4.patch/autoProcessTV/autoProcessTV.cfg.sample	2011-04-16 16:52:05.332606512 +0200
@@ -1,6 +1,6 @@
 [SickBeard]
 host=localhost
-port=8081
+port=8082
 username=
 password=
-web_root=
\ Kein Zeilenumbruch am Dateiende.
+web_root=
diff -Naur Sick-Beard-0f076f4/data/interfaces/default/config_general.tmpl Sick-Beard-0f076f4.patch/data/interfaces/default/config_general.tmpl
--- Sick-Beard-0f076f4/data/interfaces/default/config_general.tmpl	2011-04-16 15:57:38.000000000 +0200
+++ Sick-Beard-0f076f4.patch/data/interfaces/default/config_general.tmpl	2011-04-16 16:53:18.753081978 +0200
@@ -95,7 +95,7 @@
                             </label>
                             <label class="nocheck clearfix">
                                 <span class="component-title">&nbsp;</span>
-                                <span class="component-desc">Web port that Sick Beard should listen on (eg. 8081)</span>
+                                <span class="component-desc">Web port that Sick Beard should listen on (eg. 8082)</span>
                             </label>
                         </div>
 
diff -Naur Sick-Beard-0f076f4/sickbeard/__init__.py Sick-Beard-0f076f4.patch/sickbeard/__init__.py
--- Sick-Beard-0f076f4/sickbeard/__init__.py	2011-04-16 15:57:38.000000000 +0200
+++ Sick-Beard-0f076f4.patch/sickbeard/__init__.py	2011-04-16 16:51:40.750117126 +0200
@@ -393,12 +393,12 @@
             logger.log(u"!!! No log folder, logging to screen only!", logger.ERROR)
 
         try:
-            WEB_PORT = check_setting_int(CFG, 'General', 'web_port', 8081)
+            WEB_PORT = check_setting_int(CFG, 'General', 'web_port', 8082)
         except:
-            WEB_PORT = 8081
+            WEB_PORT = 8082
 
         if WEB_PORT < 21 or WEB_PORT > 65535:
-            WEB_PORT = 8081
+            WEB_PORT = 8082
 
         WEB_HOST = check_setting_str(CFG, 'General', 'web_host', '0.0.0.0')
         WEB_IPV6 = bool(check_setting_int(CFG, 'General', 'web_ipv6', 0))
diff -Naur Sick-Beard-0f076f4/sickbeard/webserveInit.py Sick-Beard-0f076f4.patch/sickbeard/webserveInit.py
--- Sick-Beard-0f076f4/sickbeard/webserveInit.py	2011-04-16 15:57:38.000000000 +0200
+++ Sick-Beard-0f076f4.patch/sickbeard/webserveInit.py	2011-04-16 16:52:50.786662601 +0200
@@ -24,7 +24,7 @@
 from sickbeard.webserve import WebInterface
 
 def initWebServer(options = {}):
-        options.setdefault('port',      8081)
+        options.setdefault('port',      8082)
         options.setdefault('host',      '0.0.0.0')
         options.setdefault('log_dir',   None)
         options.setdefault('username',    '')
