commit 76f4bf49e19a856f62d0ba2bd1061ab63b45f0e4
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Sun Feb 5 17:46:16 2017 -0800

    don't hard code cc and cxx

diff --git a/Makefile b/Makefile
index 142113a..86a9fac 100644
--- a/Makefile
+++ b/Makefile
@@ -197,7 +197,7 @@ else ifneq (,$(findstring armv,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
-   CC = gcc
+   CC ?= gcc
    LDFLAGS += $(PTHREAD_FLAGS)
    FLAGS += $(PTHREAD_FLAGS) -DHAVE_MKDIR
    IS_X86 = 0
@@ -228,8 +228,8 @@ else ifeq ($(platform), emscripten)
 
 else
    TARGET := $(TARGET_NAME)_libretro.dll
-   CC = gcc
-   CXX = g++
+   CC ?= gcc
+   CXX ?= g++
    IS_X86 = 1
    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
    LDFLAGS += -static-libgcc -static-libstdc++ -lwinmm
