diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-10-18 13:25:40.000000000 -0700
+++ b/CMakeLists.txt	2016-11-17 13:02:20.763740508 -0800
@@ -4,40 +4,10 @@
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
 
 find_package(Kodi REQUIRED)
+find_package(libretro-tgbdual REQUIRED)
 
-include(ExternalProject)
-
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-  set(LIBRETRO_DEBUG DEBUG=1)
-endif()
-
-if("${CORE_SYSTEM_NAME}" STREQUAL "windows")
-  set(BUILD_COMMAND mingw32-make platform=win ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "linux")
-  set(BUILD_COMMAND make ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "darwin")
-  set(BUILD_COMMAND make platform=osx ${LIBRETRO_DEBUG})
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "ios")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "android")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "rbpi")
-  # TODO
-elseif("${CORE_SYSTEM_NAME}" STREQUAL "freebsd")
-# TODO
-endif()
-
-externalproject_add(tgbdual
-                    GIT_REPOSITORY https://github.com/libretro/tgbdual-libretro
-                    GIT_TAG ea1364c
-                    PREFIX tgbdual
-                    CONFIGURE_COMMAND ""
-                    INSTALL_COMMAND ""
-                    BUILD_COMMAND ${BUILD_COMMAND}
-                    BUILD_IN_SOURCE 1)
-
-set(TGBDUAL_CUSTOM_BINARY ${PROJECT_BINARY_DIR}/tgbdual/src/tgbdual/tgbdual_libretro${CMAKE_SHARED_LIBRARY_SUFFIX}
+set(DEPLIBS tgbdual)
+set(TGBDUAL_CUSTOM_BINARY ${TGBDUAL_LIB}
                           ${PROJECT_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
 
 build_addon(${PROJECT_NAME} TGBDUAL DEPLIBS)
-add_dependencies(${PROJECT_NAME} tgbdual)
