diff -Naur xf86-video-dove-0.3.4/configure.ac xf86-video-dove-0.3.4.patch/configure.ac
--- xf86-video-dove-0.3.4/configure.ac	2011-03-07 18:52:09.000000000 +0100
+++ xf86-video-dove-0.3.4.patch/configure.ac	2013-01-22 13:30:57.420372292 +0100
@@ -27,7 +27,7 @@
         xf86-video-mrvl)
 
 AC_CONFIG_SRCDIR([Makefile.am])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR(.)
 
 AM_INIT_AUTOMAKE([dist-bzip2])
