diff -Naur xf86-video-dove-0.3.4/src/Makefile.am xf86-video-dove-0.3.4.patch/src/Makefile.am
--- xf86-video-dove-0.3.4/src/Makefile.am	2011-03-07 18:52:09.000000000 +0100
+++ xf86-video-dove-0.3.4.patch/src/Makefile.am	2013-01-01 03:58:19.411541528 +0100
@@ -24,7 +24,7 @@
 # _ladir passes a dummy rpath to libtool so the thing will actually link
 # TODO: -nostdlib/-Bstatic/-lgcc platform magic, not installing the .a, etc.
 
-INCLUDES = -I/usr/include/HAL -I$(top_srcdir)/src/video
+INCLUDES = -I$(top_srcdir)/src/video
 
 AM_CFLAGS = @XORG_CFLAGS@
 dovefb_drv_la_LTLIBRARIES = dovefb_drv.la
