diff -Naur xbmc-audioencoder-vorbis-143d10d/CMakeLists.txt xbmc-audioencoder-vorbis-143d10d.patch/CMakeLists.txt
--- xbmc-audioencoder-vorbis-143d10d/CMakeLists.txt	2014-07-18 23:48:45.000000000 +0200
+++ xbmc-audioencoder-vorbis-143d10d.patch/CMakeLists.txt	2014-10-20 13:07:21.666556980 +0200
@@ -6,7 +6,7 @@
 
 enable_language(CXX)
 
-find_package(xbmc REQUIRED)
+find_package(kodi REQUIRED)
 find_package(Ogg REQUIRED)
 find_package(Vorbis REQUIRED)
 find_package(VorbisEnc REQUIRED)
diff -Naur xbmc-audioencoder-vorbis-143d10d/src/EncoderVorbis.cpp xbmc-audioencoder-vorbis-143d10d.patch/src/EncoderVorbis.cpp
--- xbmc-audioencoder-vorbis-143d10d/src/EncoderVorbis.cpp	2014-07-18 23:48:45.000000000 +0200
+++ xbmc-audioencoder-vorbis-143d10d.patch/src/EncoderVorbis.cpp	2014-10-20 02:54:04.983666757 +0200
@@ -20,7 +20,7 @@
 
 #include <ogg/ogg.h>
 #include <vorbis/vorbisenc.h>
-#include "xbmc/xbmc_audioenc_dll.h"
+#include "kodi/xbmc_audioenc_dll.h"
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
