diff -Naur pastebinit-1.3/pastebinit pastebinit-1.3.patch/pastebinit
--- pastebinit-1.3/pastebinit	2012-02-15 22:14:22.000000000 +0100
+++ pastebinit-1.3.patch/pastebinit	2012-05-07 03:40:49.119612858 +0200
@@ -17,7 +17,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-defaultPB = "http://pastebin.com" #Default pastebin
+defaultPB = "http://sprunge.us" #Default pastebin
 try:
     import lsb_release
     release = lsb_release.get_distro_information()['ID'].lower()
