diff -up gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.c.foo gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.c
--- gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.c.foo	2008-11-08 15:48:46.000000000 +0100
+++ gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.c	2008-12-17 10:42:17.000000000 +0100
@@ -57,6 +57,7 @@ GST_ELEMENT_DETAILS ("mpeg1 and mpeg2 vi
  */
 #define WARN_THRESHOLD (5)
 
+//#define enable_user_data
 #ifdef enable_user_data
 static GstStaticPadTemplate user_data_template_factory =
 GST_STATIC_PAD_TEMPLATE ("user_data",
@@ -78,7 +79,7 @@ GST_STATIC_PAD_TEMPLATE ("src",
     GST_PAD_SRC,
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS ("video/x-raw-yuv, "
-        "format = (fourcc) { YV12, I420, Y42B }, "
+        "format = (fourcc) { YV12, I420, Y42B, Y444 }, "
         "width = (int) [ 16, 4096 ], "
         "height = (int) [ 16, 4096 ], "
         "framerate = (fraction) [ 0/1, 2147483647/1 ]")
@@ -404,7 +405,7 @@ crop_copy_i422_buffer (GstMpeg2dec * mpe
 static gboolean
 crop_buffer (GstMpeg2dec * mpeg2dec, GstBuffer ** buf)
 {
-  gboolean result = FALSE;
+  gboolean result = TRUE;
   GstBuffer *input = *buf;
   GstBuffer *outbuf = input;
 
@@ -450,8 +451,9 @@ gst_mpeg2dec_alloc_sized_buf (GstMpeg2de
 
     ret = gst_pad_alloc_buffer_and_set_caps (mpeg2dec->srcpad,
         GST_BUFFER_OFFSET_NONE, size, GST_PAD_CAPS (mpeg2dec->srcpad), obuf);
-    if (ret != GST_FLOW_OK)
+    if (ret != GST_FLOW_OK) {
       return ret;
+    }
 
     /* libmpeg2 needs 16 byte aligned buffers... test for this here
      * and if it fails only a single time create our own buffers from
@@ -547,8 +549,10 @@ gst_mpeg2dec_negotiate_format (GstMpeg2d
     mpeg2dec->v_offs =
         I420_V_OFFSET (mpeg2dec->decoded_width, mpeg2dec->decoded_height);
 
-  } else if (sequence->width == sequence->chroma_width ||
-      sequence->height == sequence->chroma_height) {
+  } else if ((sequence->width == sequence->chroma_width &&
+          sequence->height != sequence->chroma_height) ||
+      (sequence->width != sequence->chroma_width &&
+          sequence->height == sequence->chroma_height)) {
     gint halfsize;
 
     fourcc = GST_STR_FOURCC ("Y42B");
@@ -558,8 +562,15 @@ gst_mpeg2dec_negotiate_format (GstMpeg2d
     mpeg2dec->u_offs = halfsize;
     mpeg2dec->v_offs = halfsize + (halfsize / 2);
   } else {
-    g_warning ("mpeg2dec: 4:4:4 format not yet supported");
-    return (FALSE);
+    gint size;
+
+    size = mpeg2dec->decoded_width * mpeg2dec->decoded_height;
+
+    fourcc = GST_STR_FOURCC ("Y444");
+    mpeg2dec->format = MPEG2DEC_FORMAT_Y444;
+    mpeg2dec->size = size * 3;
+    mpeg2dec->u_offs = size;
+    mpeg2dec->v_offs = size * 2;
   }
 
   if (mpeg2dec->pixel_width == 0 || mpeg2dec->pixel_height == 0) {
@@ -1122,6 +1133,11 @@ gst_mpeg2dec_chain (GstPad * pad, GstBuf
     GST_DEBUG_OBJECT (mpeg2dec, "parse state %d", state);
 
     switch (state) {
+#if MPEG2_RELEASE >= MPEG2_VERSION (0, 5, 0)
+      case STATE_SEQUENCE_MODIFIED:
+        GST_DEBUG_OBJECT (mpeg2dec, "sequence modified");
+        /* fall through */
+#endif
       case STATE_SEQUENCE:
         ret = handle_sequence (mpeg2dec, info);
         /* if there is an error handling the sequence
diff -up gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.h.foo gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.h
--- gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.h.foo	2008-11-08 15:48:46.000000000 +0100
+++ gst-plugins-ugly-0.10.10/ext/mpeg2dec/gstmpeg2dec.h	2008-12-17 10:42:40.000000000 +0100
@@ -49,7 +49,8 @@ typedef enum
   MPEG2DEC_FORMAT_NONE,
   MPEG2DEC_FORMAT_I422,
   MPEG2DEC_FORMAT_I420,
-  MPEG2DEC_FORMAT_YV12
+  MPEG2DEC_FORMAT_YV12,
+  MPEG2DEC_FORMAT_Y444
 } Mpeg2decFormat;
 
 typedef enum 
