diff -up gst-plugins-ugly-0.10.10/gst/asfdemux/asfpacket.c~ gst-plugins-ugly-0.10.10/gst/asfdemux/asfpacket.c
--- gst-plugins-ugly-0.10.10/gst/asfdemux/asfpacket.c~	2009-01-25 11:23:20.000000000 +0100
+++ gst-plugins-ugly-0.10.10/gst/asfdemux/asfpacket.c	2009-01-25 11:23:20.000000000 +0100
@@ -317,6 +317,9 @@ gst_asf_demux_parse_payload (GstASFDemux
           GST_TIME_ARGS (payload.duration));
     } else if (payload.rep_data_len != 0) {
       GST_WARNING_OBJECT (demux, "invalid replicated data length, very bad");
+      *p_data += payload_len;
+      *p_size -= payload_len;
+      return FALSE;
     }
 
     GST_LOG_OBJECT (demux, "media object offset : %u", payload.mo_offset);
