diff -Naur TwistedCore-10.1.0/twisted/python/dist.py TwistedCore-10.1.0.patch/twisted/python/dist.py
--- TwistedCore-10.1.0/twisted/python/dist.py	2009-04-14 14:03:09.000000000 +0200
+++ TwistedCore-10.1.0.patch/twisted/python/dist.py	2010-10-19 00:46:32.896878949 +0200
@@ -10,7 +10,7 @@
 from distutils.command import build_scripts, install_data, build_ext, build_py
 from distutils.errors import CompileError
 from distutils import core
-from distutils.core import Extension
+from setuptools import Extension
 
 twisted_subprojects = ["conch", "lore", "mail", "names",
                        "news", "pair", "runner", "web", "web2",
