--- /meson.build
+++ /meson.build
@@ -4,7 +4,10 @@
         meson_version: '>= 0.54.0')
 
 dir_data        = join_paths(get_option('prefix'), get_option('datadir'))
-dir_xkb_base    = join_paths(dir_data, 'X11', 'xkb')
+dir_xkb_base    = get_option('xkb-base')
+if dir_xkb_base == ''
+    dir_xkb_base    = join_paths(dir_data, 'X11', 'xkb')
+endif
 dir_xkb_rules   = join_paths(dir_xkb_base, 'rules')
 dir_man7        = join_paths(get_option('prefix'), get_option('mandir'), 'man7')
 dir_pkgconfig   = join_paths(dir_data, 'pkgconfig')

