diff -Naur xkeyboard-config-2.10.1/configure.ac xkeyboard-config-2.10.1.patch/configure.ac
--- xkeyboard-config-2.10.1/configure.ac	2013-10-03 23:28:04.000000000 +0200
+++ xkeyboard-config-2.10.1.patch/configure.ac	2013-12-11 01:08:51.903855838 +0100
@@ -1,3 +1,4 @@
+AC_CONFIG_MACRO_DIR([m4])
 AC_INIT(xkeyboard-config, 2.10.1)
 AC_CONFIG_SRCDIR(rules/base.xml.in)
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
