--- sqlite-3.3.7/configure.ac.orig	2006-08-21 00:20:50.000000000 +0200
+++ sqlite-3.3.7/configure.ac	2006-08-21 00:22:35.000000000 +0200
@@ -187,10 +187,11 @@
 default_build_cflags="-g"
 if test "$config_BUILD_CC" = ""; then
   AC_PROG_CC
-  if test "$cross_compiling" = "yes"; then
-    AC_MSG_ERROR([unable to find a compiler for building build tools])
-  fi
-  BUILD_CC=$CC
+#  if test "$cross_compiling" = "yes"; then
+#    AC_MSG_ERROR([unable to find a compiler for building build tools])
+#  fi
+#  BUILD_CC=$CC
+BUILD_CC=gcc
   default_build_cflags=$CFLAGS
 else
   BUILD_CC=$config_BUILD_CC
@@ -238,6 +239,12 @@
   TARGET_LINK=$config_TARGET_LINK
 fi
 AC_MSG_RESULT($TARGET_LINK)
+if test "$config_TARGET_LFLAGS" != ""; then
+  TARGET_LFLAGS=$config_TARGET_LFLAGS
+  else
+    TARGET_LFLAGS=$BUILD_LFLAGS
+  fi
+AC_MSG_RESULT($TARGET_LFLAGS)
 AC_MSG_CHECKING([switches on the target compiler])
 if test "$config_TARGET_TFLAGS" != ""; then
   TARGET_TFLAGS=$config_TARGET_TFLAGS
@@ -592,15 +599,7 @@
 # Figure out what C libraries are required to compile programs
 # that use "readline()" library.
 #
-if test "$config_TARGET_READLINE_LIBS" != ""; then
-  TARGET_READLINE_LIBS="$config_TARGET_READLINE_LIBS"
-else
-  CC=$TARGET_CC
-  LIBS=""
-  AC_SEARCH_LIBS(tgetent, [readline ncurses curses termcap])
-  AC_CHECK_LIB([readline], [readline])
-  TARGET_READLINE_LIBS="$LIBS"
-fi
+TARGET_READLINE_LIBS="-lreadline"
 AC_SUBST(TARGET_READLINE_LIBS)
 
 ##########
@@ -615,41 +614,8 @@
 ##########
 # Figure out where to get the READLINE header files.
 #
-AC_MSG_CHECKING([readline header files])
-found=no
-if test "$config_TARGET_READLINE_INC" != ""; then
-  TARGET_READLINE_INC=$config_TARGET_READLINE_INC
-  found=yes
-fi
-if test "$found" = "yes"; then
-  AC_MSG_RESULT($TARGET_READLINE_INC)
-else
-  AC_MSG_RESULT(not specified: still searching...)
-  AC_CHECK_HEADER(readline.h, [found=yes])
-fi
-if test "$found" = "no"; then
-  for dir in /usr /usr/local /usr/local/readline /usr/contrib /mingw; do
-    AC_CHECK_FILE($dir/include/readline.h, found=yes)
-    if test "$found" = "yes"; then
-      TARGET_READLINE_INC="-I$dir/include"
-      break
-    fi
-    AC_CHECK_FILE($dir/include/readline/readline.h, found=yes)
-    if test "$found" = "yes"; then
-      TARGET_READLINE_INC="-I$dir/include/readline"
-      break
-    fi
-  done
-fi
-if test "$found" = "yes"; then
-  if test "$TARGET_READLINE_LIBS" = ""; then
-    TARGET_HAVE_READLINE=0
-  else
-    TARGET_HAVE_READLINE=1
-  fi
-else
-  TARGET_HAVE_READLINE=0
-fi
+TARGET_READLINE_INC=""
+TARGET_HAVE_READLINE=1
 AC_SUBST(TARGET_READLINE_INC)
 AC_SUBST(TARGET_HAVE_READLINE)
 
