Index: sqlite-3.2.1/Makefile.in
===================================================================
--- sqlite-3.2.1.orig/Makefile.in	2005-03-23 17:09:39.000000000 +0100
+++ sqlite-3.2.1/Makefile.in	2005-04-25 23:11:20.000000000 +0200
@@ -15,7 +15,10 @@
 # The toplevel directory of the source tree.  This is the directory
 # that contains this "Makefile.in" and the "configure.in" script.
 #
-TOP = @srcdir@
+TOP = $(srcdir)
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+top_builddir = .
 
 # C Compiler and options for use in building executables that
 # will run on the platform that is doing the build.
@@ -96,7 +99,7 @@
 exec_prefix = @exec_prefix@
 libdir = @libdir@
 INSTALL = @INSTALL@
-LIBTOOL = ./libtool
+LIBTOOL = @LIBTOOL@
 ALLOWRELEASE = @ALLOWRELEASE@
 
 # libtool compile/link/install
