--- sqlite-3.3.7/Makefile.in.orig	2006-08-20 23:05:36.000000000 +0200
+++ sqlite-3.3.7/Makefile.in	2006-08-20 23:42:49.000000000 +0200
@@ -31,6 +31,10 @@
 #
 TCC = @TARGET_CC@ @TARGET_CFLAGS@ -I. -I${TOP}/src
 
+# OE overrides
+#
+TARGET_LFLAGS = @TARGET_LFLAGS@
+
 # Define -DNDEBUG to compile without debugging (i.e., for production usage)
 # Omitting the define will cause extra debugging code to be inserted and
 # includes extra comments when "EXPLAIN stmt" is used.
@@ -257,17 +261,17 @@
           | $(NAWK) '{print $$5,$$6}' >last_change
 
 libsqlite3.la:	$(LIBOBJ)
-	$(LTLINK) -o libsqlite3.la $(LIBOBJ) $(LIBPTHREAD) \
+	$(LTLINK) -o libsqlite3.la $(LIBOBJ) $(TARGET_LFLAGS) $(LIBPTHREAD) \
 		${ALLOWRELEASE} -rpath $(libdir) -version-info "8:6:8"
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -o libtclsqlite3.la tclsqlite.lo \
-		$(LIBOBJ) @TCL_STUB_LIB_SPEC@ $(LIBPTHREAD) \
+		$(LIBOBJ) @TCL_STUB_LIB_SPEC@ $(TARGET_LFLAGS) $(LIBPTHREAD) \
                 -rpath $(libdir)/sqlite \
 		-version-info "8:6:8"
 
 sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
-	$(LTLINK) $(READLINE_FLAGS) $(LIBPTHREAD) \
+	$(LTLINK) $(TARGET_LFLAGS) $(READLINE_FLAGS) $(LIBPTHREAD) \
 		-o $@ $(TOP)/src/shell.c libsqlite3.la \
 		$(LIBREADLINE) $(TLIBS)
 
@@ -456,12 +460,12 @@
 
 tclsqlite3:	tclsqlite-shell.lo libsqlite3.la
 	$(LTLINK) -o tclsqlite3 tclsqlite-shell.lo \
-		 libsqlite3.la $(LIBTCL)
+		 libsqlite3.la $(TARGET_LFLAGS) $(LIBTCL)
 
 testfixture$(TEXE):	$(TOP)/src/tclsqlite.c libsqlite3.la $(TESTSRC)
 	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1 -DSQLITE_CRASH_TEST=1 \
 		$(TEMP_STORE) -o testfixture $(TESTSRC) $(TOP)/src/tclsqlite.c \
-		libsqlite3.la $(LIBTCL)
+		libsqlite3.la $(TARGET_LFLAGS) $(LIBTCL)
 
 
 fulltest:	testfixture$(TEXE) sqlite3$(TEXE)
@@ -471,7 +475,7 @@
 	./testfixture $(TOP)/test/quick.test
 
 sqlite3_analyzer$(TEXE):	$(TOP)/src/tclsqlite.c libtclsqlite3.la \
-				$(TESTSRC) $(TOP)/tool/spaceanal.tcl
+				$(TARGET_LFLAGS) $(TESTSRC) $(TOP)/tool/spaceanal.tcl
 	sed \
 	  -e '/^#/d' \
 	  -e 's,\\,\\\\,g' \
@@ -481,7 +485,7 @@
 	  $(TOP)/tool/spaceanal.tcl >spaceanal_tcl.h
 	$(LTLINK) -DTCLSH=2 -DSQLITE_TEST=1 $(TEMP_STORE)\
                 -o sqlite3_analyzer$(EXE) $(TESTSRC) $(TOP)/src/tclsqlite.c \
-		libtclsqlite3.la $(LIBTCL)
+		libtclsqlite3.la $(TARGET_LFLAGS) $(LIBTCL)
 
 # Rules used to build documentation
 #
