diff -Naur sqlite-3.6.13.orig/Makefile.in sqlite-3.6.13/Makefile.in
--- sqlite-3.6.13.orig/Makefile.in.orig	2009-04-17 09:11:59.000000000 +0200
+++ sqlite-3.6.13/Makefile.in	2009-04-17 09:11:59.000000000 +0200
@@ -36,6 +36,10 @@
 # 
 TCC += -D_HAVE_SQLITE_CONFIG_H
 
+# OE overrides
+#
+TARGET_LFLAGS = $(config_TARGET_LFLAGS)
+
 # Define -DNDEBUG to compile without debugging (i.e., for production usage)
 # Omitting the define will cause extra debugging code to be inserted and
 # includes extra comments when "EXPLAIN stmt" is used.
@@ -465,18 +469,18 @@
           | $(NAWK) '{print $$5,$$6}' >last_change
 
 libsqlite3.la:	$(LIBOBJ)
-	$(LTLINK) -o $@ $(LIBOBJ) $(TLIBS) \
+	$(LTLINK) -o $@ $(LIBOBJ) $(TARGET_LFLAGS) $(TLIBS) \
 		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite.lo \
-		libsqlite3.la @TCL_STUB_LIB_SPEC@ $(TLIBS) \
+		libsqlite3.la @TCL_STUB_LIB_SPEC@ $(TARGET_LFLAGS) $(TLIBS) \
 		-rpath "$(TCLLIBDIR)" \
 		-version-info "8:6:8" \
 		-avoid-version
 
 sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
-	$(LTLINK) $(READLINE_FLAGS) \
+	$(LTLINK) $(READLINE_FLAGS) $(TARGET_LFLAGS) \
 		-o $@ $(TOP)/src/shell.c libsqlite3.la \
 		$(LIBREADLINE) $(TLIBS) -rpath "$(libdir)"
 
@@ -747,13 +751,13 @@
 
 tclsqlite3$(TEXE):	tclsqlite-shell.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite-shell.lo \
-		 libsqlite3.la $(LIBTCL)
+		 libsqlite3.la $(TARGET_LFLAGS) $(LIBTCL)
 
 testfixture$(TEXE):	$(TESTFIXTURE_SRC)
 	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1 -DSQLITE_NO_SYNC=1\
 		-DSQLITE_CRASH_TEST=1 \
                 -DSQLITE_SERVER=1 -DSQLITE_PRIVATE="" -DSQLITE_CORE $(TEMP_STORE) \
-		-o $@ $(TESTFIXTURE_SRC) $(LIBTCL) $(TLIBS)
+		-o $@ $(TESTFIXTURE_SRC) $(TARGET_LFLAGS) $(LIBTCL) $(TLIBS)
 
 
 fulltest:	testfixture$(TEXE) sqlite3$(TEXE)
@@ -772,7 +776,7 @@
 	  $(TOP)/tool/spaceanal.tcl >spaceanal_tcl.h
 	$(LTLINK) -DTCLSH=2 -DSQLITE_TEST=1 -DSQLITE_CRASH_TEST=1 \
 		-DSQLITE_SERVER=1 -DSQLITE_PRIVATE="" -DSQLITE_CORE \
-		$(TEMP_STORE) -o $@ $(TESTFIXTURE_SRC) $(LIBTCL)
+		$(TEMP_STORE) -o $@ $(TESTFIXTURE_SRC) $(TARGET_LFLAGS) $(LIBTCL)
 
 
 lib_install:	libsqlite3.la
