diff -Naur sqlite-3.6.13/Makefile.in sqlite-3.6.13.patch/Makefile.in
--- sqlite-3.6.13/Makefile.in	2009-04-13 10:57:54.000000000 +0200
+++ sqlite-3.6.13.patch/Makefile.in	2009-04-19 18:49:22.036813333 +0200
@@ -158,6 +158,11 @@
 STRIP = @STRIP@
 THREADSAFE_FLAGS = @THREADSAFE_FLAGS@
 VERSION = @VERSION@
+
+# OE overrides
+#
+TARGET_LFLAGS = $(config_TARGET_LFLAGS)
+
 ac_ct_CC = @ac_ct_CC@
 ac_ct_CXX = @ac_ct_CXX@
 ac_ct_F77 = @ac_ct_F77@
@@ -283,7 +288,7 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libsqlite3.la: $(libsqlite3_la_OBJECTS) $(libsqlite3_la_DEPENDENCIES) 
-	$(LINK) -rpath $(libdir) $(libsqlite3_la_LDFLAGS) $(libsqlite3_la_OBJECTS) $(libsqlite3_la_LIBADD) $(LIBS)
+	$(LINK) -rpath $(libdir) $(libsqlite3_la_LDFLAGS) $(libsqlite3_la_OBJECTS) $(libsqlite3_la_LIBADD) $(TARGET_LFLAGS) $(LIBS)
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
@@ -314,7 +319,7 @@
 	done
 sqlite3$(EXEEXT): $(sqlite3_OBJECTS) $(sqlite3_DEPENDENCIES) 
 	@rm -f sqlite3$(EXEEXT)
-	$(LINK) $(sqlite3_LDFLAGS) $(sqlite3_OBJECTS) $(sqlite3_LDADD) $(LIBS)
+	$(LINK) $(sqlite3_LDFLAGS) $(sqlite3_OBJECTS) $(sqlite3_LDADD) $(TARGET_LFLAGS) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
