From f1ed3a8196a202d9a933cdc0aadc4d001a85008c Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Tue, 7 Jul 2020 15:59:16 +0000
Subject: [PATCH 4/4] WIP: pcm: iec958: dw-hdmi changes

Signed-off-by: Jernej Skrabec <jernej.skrabec@siol.net>
---
 src/pcm/pcm_iec958.c | 45 +++++++++++++++++++-------------------------
 1 file changed, 19 insertions(+), 26 deletions(-)

diff --git a/src/pcm/pcm_iec958.c b/src/pcm/pcm_iec958.c
index a11a0439..dc4f5990 100644
--- a/src/pcm/pcm_iec958.c
+++ b/src/pcm/pcm_iec958.c
@@ -71,8 +71,8 @@ enum { PREAMBLE_Z, PREAMBLE_X, PREAMBLE_Y };
 #endif /* DOC_HIDDEN */
 
 /*
- * Determine parity for time slots 4 upto 30
- * to be sure that bit 4 upt 31 will carry
+ * Determine parity for time slots 0 upto 18
+ * to be sure that bit 0 upt 19 will carry
  * an even number of ones and zeros.
  */
 static unsigned int iec958_parity(unsigned int data)
@@ -80,9 +80,8 @@ static unsigned int iec958_parity(unsigned int data)
 	unsigned int parity;
 	int bit;
 
-	data >>= 4;     /* start from bit 4 */
 	parity = 0;
-	for (bit = 4; bit <= 30; bit++) {
+	for (bit = 0; bit <= 18; bit++) {
 		if (data & 1)
 			parity++;
 		data >>= 1;
@@ -94,13 +93,12 @@ static unsigned int iec958_parity(unsigned int data)
  * Compose 32bit IEC958 subframe, two sub frames
  * build one frame with two channels.
  *
- * bit 0-3  = preamble
- *     4-7  = AUX (=0)
- *     8-27 = data (12-27 for 16bit, 8-27 for 20bit, and 24bit without AUX)
- *     28   = validity (0 for valid data, else 'in error')
- *     29   = user data (0)
- *     30   = channel status (24 bytes for 192 frames)
- *     31   = parity
+ * bit 0-15 = data
+ *     16   = validity (0 for valid data, else 'in error')
+ *     17   = user data (0)
+ *     18   = channel status (24 bytes for 192 frames)
+ *     19   = parity
+ *     20   = frame start
  */
 
 static inline uint32_t iec958_subframe(snd_pcm_iec958_t *iec, uint32_t data, int channel)
@@ -108,24 +106,20 @@ static inline uint32_t iec958_subframe(snd_pcm_iec958_t *iec, uint32_t data, int
 	unsigned int byte = iec->counter >> 3;
 	unsigned int mask = 1 << (iec->counter - (byte << 3));
 
-	/* bit 4-27 */
-	data >>= 4;
-	data &= ~0xf;
+	data >>= 16;
 
 	/* set IEC status bits (up to 192 bits) */
 	if (iec->status[byte] & mask)
-		data |= 0x40000000;
+		data |= 0x40000;
 
-	if (iec958_parity(data))	/* parity bit 4-30 */
-		data |= 0x80000000;
+	if (iec958_parity(data))
+		data |= 0x80000;
 
-	/* Preamble */
-	if (channel)
-		data |= iec->preamble[PREAMBLE_Y];	/* odd sub frame, 'Y' */
-	else if (! iec->counter)
-		data |= iec->preamble[PREAMBLE_Z];	/* Block start, 'Z' */
-	else
-		data |= iec->preamble[PREAMBLE_X];	/* even sub frame, 'X' */
+	/* block start */
+	if (!iec->counter)
+		data |= 0x100000;
+
+	data <<= 11;
 
 	if (iec->byteswap)
 		data = bswap_32(data);
@@ -137,8 +131,7 @@ static inline int32_t iec958_to_s32(snd_pcm_iec958_t *iec, uint32_t data)
 {
 	if (iec->byteswap)
 		data = bswap_32(data);
-	data &= ~0xf;
-	data <<= 4;
+	data <<= 16;
 	return (int32_t)data;
 }
 
-- 
2.17.1

