From f60540a1e0c120dfb6a7452470b3e56bee91c7d7 Mon Sep 17 00:00:00 2001
From: Dirk Gerdes <dirk.gerdes@googlemail.com>
Date: Mon, 22 Dec 2014 11:07:13 +0100
Subject: [PATCH] Added support for additional model of eGalax Touchscreen

---
 plugins/galax-raw.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/plugins/galax-raw.c b/plugins/galax-raw.c
index c8689e2..6cacc56 100644
--- a/plugins/galax-raw.c
+++ b/plugins/galax-raw.c
@@ -93,8 +93,9 @@ static int ts_galax_check_fd (struct tslib_galax *i)
 
  	if ((ioctl(ts->fd, EVIOCGID, &infos) < 0)) {
  		fprintf (stderr, "tslib: warning, can not read device identifier\n");
- 	} else if (infos.bustype != 3 || infos.vendor != 0x0EEF || infos.product != 0x0001) {
- 		fprintf (stderr, "tslib: this is not an eGalax touchscreen (3,0x0EEF,1,0x0112)\n"
+ 	} else if (infos.bustype != 3 || infos.vendor != 0x0EEF 
+			|| (infos.product != 0x0001 && infos.product != 0x7200 && infos.product != 0x7201 && infos.product != 0xC000)) {
+ 		fprintf (stderr, "tslib: this is not an eGalax touchscreen (3,0x0EEF,1/7200/7201/C000,0x0112)\n"
  		"Your device: bus=%d, vendor=0x%X, product=0x%X, version=0x%X\n",infos.bustype, infos.vendor, infos.product, infos.version);
  		return -1;
  	}
