diff --git a/Makefile.libretro b/Makefile.libretro
index 793afc7..c0c7982 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -96,6 +96,19 @@ else ifeq ($(platform), libnx)
    USE_DYNAREC = 0
    STATIC_LINKING=1
 
+# Lakka Switch (arm64)
+else ifeq ($(platform), arm64)
+   TARGET := $(TARGET_NAME)_libretro.so
+   ARCH := arm64
+   BUILTIN_GPU = unai
+   fpic := -fPIC
+   CFLAGS := $(filter-out -O2, $(CFLAGS))
+   CFLAGS += -O3 -ftree-vectorize
+   #CFLAGS += -fomit-frame-pointer -ffast-math -fPIE -ftree-vectorize -ffunction-sections
+   #CFLAGS += -DNO_OS -DNO_DYLIB -DNO_SOCKET -D__arm64__ -D__ARM_NEON__ -DARM -D__aarch64__=1
+   #CFLAGS += -DHAVE_INTTYPES -DLSB_FIRST -O3
+   USE_DYNAREC = 0
+
 else ifneq (,$(findstring ios,$(platform)))
 	ARCH := arm
 	USE_DYNAREC ?= 1
