diff --git a/data/interfaces/default/config_notifications.tmpl b/data/interfaces/default/config_notifications.tmpl
index 826cf0a..02f14af 100755
--- a/data/interfaces/default/config_notifications.tmpl
+++ b/data/interfaces/default/config_notifications.tmpl
@@ -547,45 +547,6 @@
 
                 <div class="component-group clearfix">
                     <div class="component-group-desc">
-                        <img class="notifier-icon" src="$sbRoot/images/notifiers/libnotify.png" alt="" title="Libnotify" />
-                        <h3><a href="http://library.gnome.org/devel/libnotify/" onclick="window.open(this.href, '_blank'); return false;">Libnotify</a></h3>
-                        <p>The standard desktop notification API for Linux/*nix systems.  This notifier will only function if the pynotify module is installed (Ubuntu/Debian package <a href="apt:python-notify">python-notify</a>).</p>
-                    </div>
-                    <fieldset class="component-group-list">
-                        <div class="field-pair">
-                            <input type="checkbox" class="enabler" name="use_libnotify" id="use_libnotify" #if $sickbeard.USE_LIBNOTIFY then "checked=\"checked\"" else ""# /> 
-                            <label class="clearfix" for="use_libnotify">
-                                <span class="component-title">Enable</span>
-                                <span class="component-desc">Should Sick Beard send Libnotify notifications?</span>
-                            </label>
-                        </div>
-
-                        <div id="content_use_libnotify">
-                            <div class="field-pair">
-                                <input type="checkbox" name="libnotify_notify_onsnatch" id="libnotify_notify_onsnatch" #if $sickbeard.LIBNOTIFY_NOTIFY_ONSNATCH then "checked=\"checked\"" else ""# /> 
-                                <label class="clearfix" for="libnotify_notify_onsnatch">
-                                    <span class="component-title">Notify on Snatch</span>
-                                    <span class="component-desc">Send notification when we start a download?</span>
-                                </label>
-                            </div>
-                            <div class="field-pair">
-                                <input type="checkbox" name="libnotify_notify_ondownload" id="libnotify_notify_ondownload" #if $sickbeard.LIBNOTIFY_NOTIFY_ONDOWNLOAD then "checked=\"checked\"" else ""# />
-                                <label class="clearfix" for="libnotify_notify_ondownload">
-                                    <span class="component-title">Notify on Download</span>
-                                    <span class="component-desc">Send notification when we finish a download?</span>
-                                </label>
-                            </div>
-                            <div class="testNotification" id="testLibnotify-result">Click below to test.</div>
-                            <input type="button" class="btn" value="Test Libnotify" id="testLibnotify" />
-                            <input type="submit" class="btn config_submitter" value="Save Changes" />
-                        </div><!-- /content_use_libnotify //-->
-
-                    </fieldset>
-                </div><!-- /libnotify component-group //-->
-
-
-                <div class="component-group clearfix">
-                    <div class="component-group-desc">
                         <img class="notifier-icon" src="$sbRoot/images/notifiers/pushover.png" alt="" title="Pushover" />
                         <h3><a href="http://pushover.net/" onclick="window.open(this.href, '_blank'); return false;">Pushover</a></h3>
                         <p>Pushover makes it easy to send real-time notifications to your Android and iOS devices.</p>
diff --git a/data/interfaces/default/inc_top.tmpl b/data/interfaces/default/inc_top.tmpl
index fb38f17..4ead36f 100644
--- a/data/interfaces/default/inc_top.tmpl
+++ b/data/interfaces/default/inc_top.tmpl
@@ -172,8 +172,6 @@
                             <ul class="dropdown-menu">
                                 <li><a href="$sbRoot/config/"><i class="icon-question-sign"></i> Help &amp; Info</a></li>
                                 <li class="divider"></li>
-                                <li><a href="$sbRoot/config/general/"><i class="icon-cog"></i> General</a></li>
-                                <li class="divider"></li>
                                 <li><a href="$sbRoot/config/search/"><i class="icon-cog"></i> Search Settings</a></li>
                                 <li class="divider"></li>
                                 <li><a href="$sbRoot/config/providers/"><i class="icon-cog"></i> Search Providers</a></li>
diff --git a/sickbeard/webserve.py b/sickbeard/webserve.py
index 20f0dfc..b0bf6a6 100644
--- a/sickbeard/webserve.py
+++ b/sickbeard/webserve.py
@@ -609,7 +609,6 @@ class History:
 
 
 ConfigMenu = [
-    { 'title': 'General',           'path': 'config/general/'          },
     { 'title': 'Search Settings',   'path': 'config/search/'           },
     { 'title': 'Search Providers',  'path': 'config/providers/'        },
     { 'title': 'Post Processing',   'path': 'config/postProcessing/'   },
