diff -Naur SickBeard-9b43f71/data/interfaces/default/config_notifications.tmpl SickBeard-9b43f71.patch/data/interfaces/default/config_notifications.tmpl
--- SickBeard-9b43f71/data/interfaces/default/config_notifications.tmpl	2011-12-14 04:42:55.000000000 +0100
+++ SickBeard-9b43f71.patch/data/interfaces/default/config_notifications.tmpl	2011-12-14 04:57:50.375945941 +0100
@@ -485,48 +485,6 @@
 
                 </div>
 
-
-                <div id="core-component-group7" class="component-group clearfix">
-                    <div class="component-group-desc">
-                        <h3><a href="http://library.gnome.org/devel/libnotify/" onclick="window.open(this.href, '_blank'); return false;"><img src="$sbRoot/images/notifiers/libnotify.gif" alt="Libnotify" title="Libnotify" width="16" height="16" /> Libnotify</a></h3>
-                        <p>The standard desktop notification API for Linux/*nix systems.  This notifier will only function if the pynotify module is installed (Ubuntu/Debian package <a href="apt:python-notify">python-notify</a>).</p>
-                    </div>
-
-                    <fieldset class="component-group-list">
-                        <div class="field-pair">
-                            <input type="checkbox" class="enabler" name="use_libnotify" id="use_libnotify" #if $sickbeard.USE_LIBNOTIFY then "checked=\"checked\"" else ""# /> 
-                            <label class="clearfix" for="use_libnotify">
-                                <span class="component-title">Enable</span>
-                                <span class="component-desc">Should Sick Beard send Libnotify notifications?</span>
-                            </label>
-                        </div>
-
-                        <div id="content_use_libnotify">
-                        <div class="field-pair">
-                            <input type="checkbox" name="libnotify_notify_onsnatch" id="libnotify_notify_onsnatch" #if $sickbeard.LIBNOTIFY_NOTIFY_ONSNATCH then "checked=\"checked\"" else ""# /> 
-                            <label class="clearfix" for="libnotify_notify_onsnatch">
-                                <span class="component-title">Notify on Snatch</span>
-                                <span class="component-desc">Send notification when we start a download?</span>
-                            </label>
-                        </div>
-
-                        <div class="field-pair">
-                            <input type="checkbox" name="libnotify_notify_ondownload" id="libnotify_notify_ondownload" #if $sickbeard.LIBNOTIFY_NOTIFY_ONDOWNLOAD then "checked=\"checked\"" else ""# />
-                            <label class="clearfix" for="libnotify_notify_ondownload">
-                                <span class="component-title">Notify on Download</span>
-                                <span class="component-desc">Send notification when we finish a download?</span>
-                            </label>
-                        </div>
-
-                        <div class="testNotification" id="testLibnotify-result">Click below to test.</div>
-                        <input type="button" value="Test Libnotify" id="testLibnotify" />
-                        <input type="submit" class="config_submitter" value="Save Changes" />
-
-                        </div><!-- /content_use_libnotify //-->
-
-                    </fieldset>
-                </div><!-- /component-group //-->
-
                 <div id="core-component-group8" class="component-group clearfix">
 
                     <div class="component-group-desc">
diff -Naur SickBeard-9b43f71/data/interfaces/default/inc_top.tmpl SickBeard-9b43f71.patch/data/interfaces/default/inc_top.tmpl
--- SickBeard-9b43f71/data/interfaces/default/inc_top.tmpl	2011-12-14 04:57:34.798638804 +0100
+++ SickBeard-9b43f71.patch/data/interfaces/default/inc_top.tmpl	2011-12-14 04:59:41.948145818 +0100
@@ -168,7 +168,6 @@
             <li id="NAVconfig"><a href="$sbRoot/config/">Config</a>
                 <ul>
                         <li><a href="$sbRoot/config/"><img src="$sbRoot/images/menu/config_index16.png" alt="" width="16" height="16" />Help &amp; Info</a></li>
-                        <li><a href="$sbRoot/config/general/"><img src="$sbRoot/images/menu/config16.png" alt="" width="16" height="16" />General</a></li>
                         <li><a href="$sbRoot/config/search/"><img src="$sbRoot/images/menu/config16.png" alt="" width="16" height="16" />Search Settings</a></li>
                         <li><a href="$sbRoot/config/providers/"><img src="$sbRoot/images/menu/config16.png" alt="" width="16" height="16" />Search Providers</a></li>
                         <li><a href="$sbRoot/config/postProcessing/"><img src="$sbRoot/images/menu/config16.png" alt="" width="16" height="16" />Post Processing</a></li>
diff -Naur SickBeard-9b43f71/sickbeard/webserve.py SickBeard-9b43f71.patch/sickbeard/webserve.py
--- SickBeard-9b43f71/sickbeard/webserve.py	2011-12-14 04:42:55.000000000 +0100
+++ SickBeard-9b43f71.patch/sickbeard/webserve.py	2011-12-14 04:57:50.379946021 +0100
@@ -585,7 +585,6 @@
 
 
 ConfigMenu = [
-    { 'title': 'General',           'path': 'config/general/'          },
     { 'title': 'Search Settings',   'path': 'config/search/'           },
     { 'title': 'Search Providers',  'path': 'config/providers/'        },
     { 'title': 'Post Processing',   'path': 'config/postProcessing/'   },
diff -Naur SickBeard-9b43f71/sickbeard/webserve.py.orig SickBeard-9b43f71.patch/sickbeard/webserve.py.orig
