diff --git a/data/interfaces/default/config.tmpl b/data/interfaces/default/config.tmpl
index ec28766..fdf3e30 100644
--- a/data/interfaces/default/config.tmpl
+++ b/data/interfaces/default/config.tmpl
@@ -36,13 +36,4 @@
     </table>
 </div>
 
-<div class="container padding" style="width: 600px;">
-    <table class="infoTable">
-        <tr>
-            <td><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donateCC_LG.gif" alt="[donate]" /></a></td>
-            <td>Sick Beard is free, but you can contribute by giving a <b><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;">donation</a></b>.</td>
-        </tr>
-    </table>
-</div>
-
 #include $os.path.join($sickbeard.PROG_DIR, "data/interfaces/default/inc_bottom.tmpl")
diff --git a/data/interfaces/default/inc_top.tmpl b/data/interfaces/default/inc_top.tmpl
index f791be8..fb38f17 100644
--- a/data/interfaces/default/inc_top.tmpl
+++ b/data/interfaces/default/inc_top.tmpl
@@ -108,20 +108,6 @@
 
 <body>
     <header>
-#if $sickbeard.NEWEST_VERSION_STRING:
-<div id="upgrade-notification">
-    <div>
-        <span class="notify-text">$sickbeard.NEWEST_VERSION_STRING</span>
-    </div>
-</div>
-<div id="header-fix"></div>
-#end if
-<div id="header">
-    <a name="top"></a>
-    <span id="logo"><a href="$sbRoot/home/" title="Sick Beard homepage"><img alt="Sick Beard" src="$sbRoot/images/sickbeard.png" width="150" height="72" /></a></span>
-    <span id="versiontext">alpha <a href="https://github.com/midgetspy/Sick-Beard/wiki/ChangeLog" onclick="window.open(this.href, '_blank'); return false;">$sickbeard.version.SICKBEARD_VERSION</a></span>
-</div>
-
     <div class="navbar">
         <div class="navbar-inner">
             <div class="container">
@@ -208,11 +194,6 @@
                         </li>
                         <li class="divider-vertical"></li>
                     </ul>
-                    <ul class="nav pull-right">
-                        <li>
-                            <a id="navDonate" href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donate_LG.gif" alt="[donate]" height="26" width="92" /></a>
-                        </li>
-                    </ul>
                 </div><!-- /nav-collapse -->
 
             </div><!-- /container -->
