diff -Naur SickBeard-9b43f71/data/interfaces/default/config.tmpl SickBeard-9b43f71.patch/data/interfaces/default/config.tmpl
--- SickBeard-9b43f71/data/interfaces/default/config.tmpl	2011-12-14 04:42:55.000000000 +0100
+++ SickBeard-9b43f71.patch/data/interfaces/default/config.tmpl	2011-12-14 04:56:35.317466009 +0100
@@ -25,9 +25,4 @@
 </table>
 </div>
 
-<table class="infoTable" cellspacing="1" border="0" cellpadding="0"><tr>
-    <td><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donateCC_LG.gif" alt="[donate]" /></a></td>
-    <td>Sickbeard is free, but you can contribute by giving a <b><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;">donation</a></b>.</td>
-</tr></table>
-
 #include $os.path.join($sickbeard.PROG_DIR, "data/interfaces/default/inc_bottom.tmpl")
diff -Naur SickBeard-9b43f71/data/interfaces/default/inc_top.tmpl SickBeard-9b43f71.patch/data/interfaces/default/inc_top.tmpl
--- SickBeard-9b43f71/data/interfaces/default/inc_top.tmpl	2011-12-14 04:42:55.000000000 +0100
+++ SickBeard-9b43f71.patch/data/interfaces/default/inc_top.tmpl	2011-12-14 04:56:35.318466029 +0100
@@ -140,19 +140,6 @@
 </head>
 
 <body>
-#if $sickbeard.NEWEST_VERSION_STRING:
-<div id="upgrade-notification">
-    <div>
-        <span class="notify-text">$sickbeard.NEWEST_VERSION_STRING</span>
-    </div>
-</div>
-<div id="header-fix"></div>
-#end if
-<div id="header">
-    <a name="top"></a>
-    <span id="logo"><a href="$sbRoot/home/" title="Sick Beard homepage"><img alt="Sick Beard" src="$sbRoot/images/sickbeard_small.png" width="150" /></a></span>
-    <span id="versiontext">alpha $sickbeard.version.SICKBEARD_VERSION</span>
-</div>
 <ul id="MainMenu" class="sf-menu">
             <li id="NAVsystem" class="first"><a href="#" class="navIcon" onclick="return false;"><img src="$sbRoot/images/menu/system18.png" alt="" width="18" height="18" /></a>
                 <ul>
@@ -194,7 +181,6 @@
                         <li><a href="$sbRoot/errorlogs/viewlog/"><img src="$sbRoot/images/menu/viewlog16.png" alt="" width="16" height="16" />View Log</a></li>
                 </ul>
             </li>
-            <li id="donate"><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" onclick="window.open(this.href); return false;"><img src="$sbRoot/images/paypal/btn_donate_LG.gif" alt="[donate]" /></a></li>
 </ul>
 #if $varExists('submenu'):
 <div id="SubMenu">
