From c2121393b05a756266fe728f83eaddbd3755ce13 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Mon, 16 Jun 2014 12:14:33 +0300
Subject: [PATCH] dont build i18n

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index aa344a4..819b5b0 100644
--- a/Makefile
+++ b/Makefile
@@ -62,7 +62,7 @@ OBJS = $(PLUGIN).o soundex.o extpipe.o parse.o source.o import.o event.o setup.o
 
 ### The main target:
 
-all: libvdr-$(PLUGIN).so i18n
+all: libvdr-$(PLUGIN).so
 
 ### Implicit rules:
 
-- 
1.7.2.5

