commit 6d5c30323f49c7960c873a423ec948834bbe8737
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Sat Oct 19 22:24:10 2013 +0300

    vdr 2.1.2 buildfix

diff --git a/xmltv2vdr.cpp b/xmltv2vdr.cpp
index b367508..7b61c61 100644
--- a/xmltv2vdr.cpp
+++ b/xmltv2vdr.cpp
@@ -274,7 +274,7 @@ cGlobals::cGlobals()
     imgdelafter=30;
     soundex=false;
 
-    if (asprintf(&epgfile_store,"%s/epg.db",VideoDirectory)==-1) {};
+    if (asprintf(&epgfile_store,"%s/epg.db",cVideoDirectory::Name())==-1) {};
 
     if (!CheckEPGDir("/var/run/vdr"))
     {
