From c7cd9b614762103a95f3da27783e03ce115d43fb Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 20 May 2012 16:05:51 +0300
Subject: [PATCH] vdr-iptv: skip header processing

---
 protocolhttp.c |   12 +-----------
 1 files changed, 1 insertions(+), 11 deletions(-)

diff --git a/protocolhttp.c b/protocolhttp.c
index de50f64..45c5425 100644
--- a/protocolhttp.c
+++ b/protocolhttp.c
@@ -140,17 +140,7 @@ bool cIptvProtocolHttp::ProcessHeaders(void)
     memset(buf, '\0', sizeof(buf));
     if (!GetHeaderLine(buf, sizeof(buf), lineLength))
        return false;
-    if (!responseFound && sscanf(buf, fmt, &version, &response) != 2) {
-       error("Expected HTTP header not found\n");
-       continue;
-       }
-    else
-       responseFound = true;
-    // Allow only 'OK' and 'Partial Content'
-    if ((response != 200) && (response != 206)) {
-       error("Invalid HTTP response (%d): %s\n", response, buf);
-       return false;
-       }
+      responseFound = true;
     }
   return true;
 }
-- 
1.7.2.5

