commit 67d5e792aa30100b111490c3c9e1c01f9def3c86
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Thu Mar 15 00:02:50 2012 +0200

    vdr-iptv: skip header processing

diff --git a/protocolhttp.c b/protocolhttp.c
index 2d5277a..9b305aa 100644
--- a/protocolhttp.c
+++ b/protocolhttp.c
@@ -154,17 +154,7 @@ bool cIptvProtocolHttp::ProcessHeaders(void)
     memset(buf, '\0', sizeof(buf));
     if (!GetHeaderLine(buf, sizeof(buf), lineLength))
        return false;
-    if (!responseFound && sscanf(buf, "HTTP/1.%*i %i ", &response) != 1) {
-       error("Expected HTTP header not found\n");
-       continue;
-       }
-    else
-       responseFound = true;
-    // Allow only 'OK' and 'Partial Content'
-    if ((response != 200) && (response != 206)) {
-       error("Invalid HTTP response (%d): %s\n", response, buf);
-       return false;
-       }
+      responseFound = true;
     }
   return true;
 }
