diff -Naur sauerbraten/src/Makefile sauerbraten.patch/src/Makefile
--- sauerbraten/src/Makefile	2010-05-13 20:30:28.000000000 +0200
+++ sauerbraten.patch/src/Makefile	2010-11-04 01:03:27.636654907 +0100
@@ -20,8 +20,8 @@
 CLIENT_INCLUDES= $(INCLUDES) -Iinclude
 CLIENT_LIBS= -mwindows -Llib -lSDL -lSDL_image -lSDL_mixer -lzdll -lopengl32 -lenet -lws2_32 -lwinmm
 else	
-CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl-config --cflags`
-CLIENT_LIBS= -Lenet/.libs -lenet -L/usr/X11R6/lib `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
+CLIENT_INCLUDES= $(INCLUDES) `sdl-config --cflags`
+CLIENT_LIBS= -Lenet/.libs -lenet `sdl-config --libs` -lSDL_image -lSDL_mixer -lz -lGL
 endif
 ifeq ($(PLATFORM),Linux)
 CLIENT_LIBS+= -lrt
