diff -u -r libimobiledevice-1.2.0/src/idevice.c libimobiledevice-1.2.0-nossl3/src/idevice.c
--- libimobiledevice-1.2.0/src/idevice.c	2015-01-28 02:10:32.000000000 +0100
+++ libimobiledevice-1.2.0-nossl3/src/idevice.c	2016-03-03 18:33:45.912308242 +0100
@@ -678,7 +678,7 @@
 	}
 	BIO_set_fd(ssl_bio, (int)(long)connection->data, BIO_NOCLOSE);
 
-	SSL_CTX *ssl_ctx = SSL_CTX_new(SSLv3_method());
+	SSL_CTX *ssl_ctx = SSL_CTX_new(SSLv23_method());
 	if (ssl_ctx == NULL) {
 		debug_info("ERROR: Could not create SSL context.");
 		BIO_free(ssl_bio);
