From 1dddab35e61b9100b2cbcd3098596946c1c50b7a Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Sun, 21 Jun 2020 10:40:52 +0100
Subject: [PATCH] Fix link failure with gcc 10

---
 src/drmmode_display.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drmmode_display.h b/src/drmmode_display.h
index 803ac3c..9c0f25a 100644
--- a/src/drmmode_display.h
+++ b/src/drmmode_display.h
@@ -289,7 +289,7 @@ Bool drmmode_wait_vblank(xf86CrtcPtr crtc, drmVBlankSeqType type,
 			 uint64_t *ust, uint32_t *result_seq);
 
 
-miPointerSpriteFuncRec drmmode_sprite_funcs;
+extern miPointerSpriteFuncRec drmmode_sprite_funcs;
 
 
 #endif
-- 
2.20.1

