--- a/Makefile.in	2016-09-16 14:46:15.352407556 +0000
+++ b/Makefile.in	2016-09-16 14:46:31.459074122 +0000
@@ -33,7 +33,7 @@
    CC = gcc
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
-   LDFLAGS := -lz -lpthread
+   LDFLAGS := -lz -lpthread -lglib-2.0
    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T
    ifeq ($(USE_JIT), 1)
       ifeq ($(TARGET_ARCH), x86_64) # JIT => address space 32 bits
