diff -Naur glib-2.61.1.org/gio/meson.build glib-2.61.1/gio/meson.build
--- glib-2.61.1.org/gio/meson.build	2019-05-24 19:00:33.000000000 +0200
+++ glib-2.61.1/gio/meson.build	2019-08-26 09:21:14.893507207 +0200
@@ -994,8 +994,3 @@
 endif
 
 subdir('fam')
-# Don’t build the tests unless we can run them (either natively or in an exe wrapper)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper())
-if build_tests
-    subdir('tests')
-endif
\ Kein Zeilenumbruch am Dateiende.
diff -Naur glib-2.61.1.org/glib/meson.build glib-2.61.1/glib/meson.build
--- glib-2.61.1.org/glib/meson.build	2019-05-24 19:00:33.000000000 +0200
+++ glib-2.61.1/glib/meson.build	2019-08-26 09:21:22.941561248 +0200
@@ -459,8 +459,3 @@
     install : true)
 endif
 
-# Don’t build the tests unless we can run them (either natively or in an exe wrapper)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper())
-if build_tests
-  subdir('tests')
-endif
\ Kein Zeilenumbruch am Dateiende.
diff -Naur glib-2.61.1.org/gobject/meson.build glib-2.61.1/gobject/meson.build
--- glib-2.61.1.org/gobject/meson.build	2019-05-24 19:00:33.000000000 +0200
+++ glib-2.61.1/gobject/meson.build	2019-08-26 09:21:28.981601802 +0200
@@ -166,8 +166,3 @@
     install : true)
 endif
 
-# Don’t build the tests unless we can run them (either natively or in an exe wrapper)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper())
-if build_tests
-  subdir('tests')
-endif
\ Kein Zeilenumbruch am Dateiende.
diff -Naur glib-2.61.1.org/meson.build glib-2.61.1/meson.build
--- glib-2.61.1.org/meson.build	2019-05-24 19:00:33.000000000 +0200
+++ glib-2.61.1/meson.build	2019-08-26 09:21:08.029461110 +0200
@@ -2045,11 +2045,6 @@
 subdir('gmodule')
 subdir('gio')
 subdir('fuzzing')
-# Don’t build the tests unless we can run them (either natively or in an exe wrapper)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper())
-if build_tests
-  subdir('tests')
-endif
 
 # xgettext is optional (on Windows for instance)
 if find_program('xgettext', required : get_option('nls')).found()
