diff -up glib-2.16.6/configure.atomic glib-2.16.6/configure
--- glib-2.16.6/configure.atomic	2008-10-27 18:24:20.000000000 -0400
+++ glib-2.16.6/configure	2008-10-27 18:24:31.000000000 -0400
@@ -42742,11 +42742,6 @@ $as_echo_n "checking whether to use asse
 
 if test x"$GCC" = xyes; then
   case $host_cpu in
-    i386)
-      { $as_echo "$as_me:$LINENO: result: none" >&5
-$as_echo "none" >&6; }
-      glib_memory_barrier_needed=no
-      ;;
     i?86)
       { $as_echo "$as_me:$LINENO: result: i486" >&5
 $as_echo "i486" >&6; }
