From 135f1c814b81b18d7546cb5d8b7246463e7704bd Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Sun, 26 Jan 2020 05:56:34 +0000
Subject: [PATCH] tests are broken when cross-compiling so don't build them

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index cbacf0d..84d2c9e 100644
--- a/meson.build
+++ b/meson.build
@@ -77,7 +77,7 @@ installed_tests_template = files('template.test.in')
 installed_tests_template_tap = files('template-tap.test.in')
 
 # Don’t build the tests unless we can run them (either natively, in an exe wrapper, or by installing them for later use)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper()) or installed_tests_enabled
+build_tests = false
 
 add_project_arguments('-D_GNU_SOURCE', language: 'c')
 
-- 
2.7.4

