diff -Naur glib-2.23.5/gio/Makefile.am glib-2.23.5.patch/gio/Makefile.am
--- glib-2.23.5/gio/Makefile.am	2010-03-06 17:51:40.000000000 +0100
+++ glib-2.23.5.patch/gio/Makefile.am	2010-03-09 09:38:50.205664589 +0100
@@ -466,6 +466,7 @@
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	$(top_builddir)/gmodule/libgmodule-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
 	libgio-2.0.la					\
 	$(NULL)
 
diff -Naur glib-2.23.5/gio/Makefile.in glib-2.23.5.patch/gio/Makefile.in
--- glib-2.23.5/gio/Makefile.in	2010-03-08 16:35:06.000000000 +0100
+++ glib-2.23.5.patch/gio/Makefile.in	2010-03-09 09:39:16.717664516 +0100
@@ -824,6 +824,7 @@
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	$(top_builddir)/gmodule/libgmodule-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
 	libgio-2.0.la					\
 	$(NULL)
 
diff -Naur glib-2.23.5/gio/tests/Makefile.am glib-2.23.5.patch/gio/tests/Makefile.am
--- glib-2.23.5/gio/tests/Makefile.am	2010-03-06 17:51:40.000000000 +0100
+++ glib-2.23.5.patch/gio/tests/Makefile.am	2010-03-09 09:39:44.816664472 +0100
@@ -14,6 +14,8 @@
 progs_ldadd     = 					\
 	$(top_builddir)/glib/libglib-2.0.la 		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
 	$(top_builddir)/gio/libgio-2.0.la
 
 
diff -Naur glib-2.23.5/gio/tests/Makefile.in glib-2.23.5.patch/gio/tests/Makefile.in
--- glib-2.23.5/gio/tests/Makefile.in	2010-03-08 16:35:07.000000000 +0100
+++ glib-2.23.5.patch/gio/tests/Makefile.in	2010-03-09 09:40:11.250664543 +0100
@@ -448,6 +448,8 @@
 progs_ldadd = \
 	$(top_builddir)/glib/libglib-2.0.la 		\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
+	$(top_builddir)/gmodule/libgmodule-2.0.la	\
+	$(top_builddir)/gthread/libgthread-2.0.la	\
 	$(top_builddir)/gio/libgio-2.0.la
 
 SAMPLE_PROGS = \
diff -Naur glib-2.23.5/tests/Makefile.am glib-2.23.5.patch/tests/Makefile.am
--- glib-2.23.5/tests/Makefile.am	2010-01-22 03:57:55.000000000 +0100
+++ glib-2.23.5.patch/tests/Makefile.am	2010-03-09 09:36:39.805664621 +0100
@@ -86,7 +86,7 @@
 assert_msg_test_LDADD = $(libglib)
 if ENABLE_TIMELOOP
 timeloop_LDADD = $(libglib)
-timeloop_closure_LDADD = $(libglib) $(libgobject)
+timeloop_closure_LDADD = $(libglib) $(libgobject) $(libgthread)
 endif
 
 test_programs =					\
diff -Naur glib-2.23.5/tests/Makefile.in glib-2.23.5.patch/tests/Makefile.in
--- glib-2.23.5/tests/Makefile.in	2010-03-08 16:35:08.000000000 +0100
+++ glib-2.23.5.patch/tests/Makefile.in	2010-03-09 09:37:22.598664379 +0100
@@ -692,7 +692,7 @@
 errorcheck_mutex_test_LDADD = $(libglib) $(libgthread) $(G_THREAD_LIBS) 
 assert_msg_test_LDADD = $(libglib)
 @ENABLE_TIMELOOP_TRUE@timeloop_LDADD = $(libglib)
-@ENABLE_TIMELOOP_TRUE@timeloop_closure_LDADD = $(libglib) $(libgobject)
+@ENABLE_TIMELOOP_TRUE@timeloop_closure_LDADD = $(libglib) $(libgobject) $(libgthread)
 test_programs = \
 	atomic-test				\
 	base64-test				\
