diff -Naur glib-2.32.3/configure.ac glib-2.32.3.patch/configure.ac
--- glib-2.32.3/configure.ac	2012-06-01 03:46:38.181436837 +0200
+++ glib-2.32.3.patch/configure.ac	2012-06-01 03:48:42.622136804 +0200
@@ -2586,7 +2586,7 @@
                               [regenerate man pages from Docbook [default=no]])],enable_man=yes,
               enable_man=no)
 
-if test "${enable_man}" != no; then
+AS_IF([test "${enable_man}" != no], [
 dnl
 dnl Check for xsltproc
 dnl
@@ -2600,7 +2600,7 @@
      [DocBook XML DTD V4.1.2],,enable_man=no)
   JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl],
      [DocBook XSL Stylesheets],,enable_man=no)
-fi
+])
 
 AM_CONDITIONAL(ENABLE_MAN, test x$enable_man != xno)
 
