commit 51ed00db7df80caad3c3d2b136ee5578cedb7e31
Author: Lukas Rusak <lorusak@gmail.com>
Date:   Fri Jan 13 13:54:42 2017 -0800

    use short expected commit values

diff --git a/daemon/info_unix.go b/daemon/info_unix.go
index 9c41c0e..4e0e896 100644
--- a/daemon/info_unix.go
+++ b/daemon/info_unix.go
@@ -27,7 +27,7 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 	v.DefaultRuntime = daemon.configStore.GetDefaultRuntimeName()
 	v.InitBinary = daemon.configStore.GetInitPath()
 
-	v.ContainerdCommit.Expected = dockerversion.ContainerdCommitID
+	v.ContainerdCommit.Expected = dockerversion.ContainerdCommitID[0:7]
 	if sv, err := daemon.containerd.GetServerVersion(context.Background()); err == nil {
 		v.ContainerdCommit.ID = sv.Revision
 	} else {
@@ -35,7 +35,7 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 		v.ContainerdCommit.ID = "N/A"
 	}
 
-	v.RuncCommit.Expected = dockerversion.RuncCommitID
+	v.RuncCommit.Expected = dockerversion.RuncCommitID[0:7]
 	if rv, err := exec.Command(DefaultRuntimeBinary, "--version").Output(); err == nil {
 		parts := strings.Split(strings.TrimSpace(string(rv)), "\n")
 		if len(parts) == 3 {
