diff --git a/components/engine/daemon/info_unix.go b/components/engine/daemon/info_unix.go
index f43af6274f..a2c578e58b 100644
--- a/components/engine/daemon/info_unix.go
+++ b/components/engine/daemon/info_unix.go
@@ -28,7 +28,7 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 	v.DefaultRuntime = daemon.configStore.GetDefaultRuntimeName()
 	v.InitBinary = daemon.configStore.GetInitPath()
 
-	v.ContainerdCommit.Expected = dockerversion.ContainerdCommitID
+	v.ContainerdCommit.Expected = dockerversion.ContainerdCommitID[0:7]
 	if sv, err := daemon.containerd.GetServerVersion(context.Background()); err == nil {
 		v.ContainerdCommit.ID = sv.Revision
 	} else {
@@ -36,7 +36,7 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 		v.ContainerdCommit.ID = "N/A"
 	}
 
-	v.RuncCommit.Expected = dockerversion.RuncCommitID
+	v.RuncCommit.Expected = dockerversion.RuncCommitID[0:7]
 	defaultRuntimeBinary := daemon.configStore.GetRuntime(daemon.configStore.GetDefaultRuntimeName()).Path
 	if rv, err := exec.Command(defaultRuntimeBinary, "--version").Output(); err == nil {
 		parts := strings.Split(strings.TrimSpace(string(rv)), "\n")
