--- a/components/engine/vendor/github.com/containerd/containerd/runtime/v1/shim/client/client.go
+++ b/components/engine/vendor/github.com/containerd/containerd/runtime/v1/shim/client/client.go
@@ -219,7 +219,8 @@
 		if err != nil {
 			return nil, nil, err
 		}
-		client := ttrpc.NewClient(conn, ttrpc.WithOnClose(onClose))
+		client := ttrpc.NewClient(conn)
+		client.OnClose(onClose)
 		return shimapi.NewShimClient(client), conn, nil
 	}
 }

