diff -Naur xrandr-1.2.99.orig/xrandr.c xrandr-1.2.99.4/xrandr.c
--- xrandr-1.2.99.orig/xrandr.c	2009-03-15 12:29:28.000000000 +0100
+++ xrandr-1.2.99.4/xrandr.c	2009-03-15 12:31:01.000000000 +0100
@@ -1099,9 +1099,10 @@
 #endif
 	XRRPanning  *panning_info = NULL;
 
+#if 0 /* fucking crappy garbage !! */
 	if (has_1_3)
 	    panning_info = XRRGetPanning  (dpy, res, res->crtcs[c]);
-
+#endif
 	set_name_xid (&crtcs[c].crtc, res->crtcs[c]);
 	set_name_index (&crtcs[c].crtc, c);
 	if (!crtc_info) fatal ("could not get crtc 0x%x information\n", res->crtcs[c]);
