diff -Naur xrandr-1.2.3.orig/xrandr.c xrandr-1.2.3/xrandr.c
--- xrandr-1.2.3.orig/xrandr.c	2008-04-12 15:57:23.000000000 +0200
+++ xrandr-1.2.3/xrandr.c	2008-04-12 15:57:32.000000000 +0200
@@ -189,7 +189,7 @@
 
 #if HAS_RANDR_1_2
 typedef enum _policy {
-    clone, extend
+    cloned, extend
 } policy_t;
 
 typedef enum _relation {
@@ -1622,7 +1622,7 @@
     int		ret = 0;
 #if HAS_RANDR_1_2
     output_t	*output = NULL;
-    policy_t	policy = clone;
+    policy_t	policy = cloned;
     Bool    	setit_1_2 = False;
     Bool    	query_1_2 = False;
     Bool	modeit = False;
@@ -1886,7 +1886,7 @@
 	    continue;
 	}
 	if (!strcmp ("--clone", argv[i])) {
-	    policy = clone;
+	    policy = cloned;
 	    setit_1_2 = True;
 	    continue;
 	}
@@ -2319,9 +2319,9 @@
 		printf ("\tClones:    ");
 		for (j = 0; j < output_info->nclone; j++)
 		{
-		    output_t	*clone = find_output_by_xid (output_info->clones[j]);
+		    output_t	*cloned = find_output_by_xid (output_info->clones[j]);
 
-		    if (clone) printf (" %s", clone->output.string);
+		    if (cloned) printf (" %s", cloned->output.string);
 		}
 		printf ("\n");
 		if (output->crtc_info)
