diff --git a/Play/Source/gs/GSH_OpenGL/CMakeLists.txt b/Play/Source/gs/GSH_OpenGL/CMakeLists.txt
index 05148365..19e033ed 100644
--- a/Play/Source/gs/GSH_OpenGL/CMakeLists.txt
+++ b/Play/Source/gs/GSH_OpenGL/CMakeLists.txt
@@ -31,6 +31,6 @@ add_library(gsh_opengl STATIC
 target_link_libraries(gsh_opengl Boost::boost Framework_OpenGl ${PROJECT_LIBS})
 target_include_directories(gsh_opengl PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/Source/gs/GSH_OpenGL/)
 
-if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS OR TARGET_PLATFORM_UNIX_AARCH64)
+if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS)
 	target_compile_definitions(gsh_opengl PUBLIC GLES_COMPATIBILITY)
 endif()
diff --git a/Play/Source/ui_libretro/CMakeLists.txt b/Play/Source/ui_libretro/CMakeLists.txt
index 2e861305..0bc8c1be 100644
--- a/Play/Source/ui_libretro/CMakeLists.txt
+++ b/Play/Source/ui_libretro/CMakeLists.txt
@@ -26,7 +26,7 @@ if(NOT TARGET gsh_opengl)
 endif()
 list(INSERT PROJECT_LIBS 0 gsh_opengl)
 
-if(NOT TARGET_PLATFORM_UNIX_AARCH64 AND (TARGET_PLATFORM_UNIX OR TARGET_PLATFORM_WIN32))
+if(TARGET_PLATFORM_UNIX_AARCH64 AND (TARGET_PLATFORM_UNIX OR TARGET_PLATFORM_WIN32))
 	if (NOT TARGET glew_s)
 		add_subdirectory(
 			${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/glew-2.0.0
