diff --git a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
index 20ff8ba..7a701ab 100644
--- a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
+++ b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
@@ -11,7 +11,7 @@ include(Header)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS OR TARGET_PLATFORM_UNIX_AARCH64)
+if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS)
 	set(USE_GLES ON CACHE BOOL "Use GLES")
 else()
 	set(USE_GLES OFF CACHE BOOL "Use GLES")
