diff --git a/Source/ui_libretro/CMakeLists.txt b/Source/ui_libretro/CMakeLists.txt
index b84b0c40..d02a5f32 100644
--- a/Source/ui_libretro/CMakeLists.txt
+++ b/Source/ui_libretro/CMakeLists.txt
@@ -40,14 +40,6 @@ if(TARGET_PLATFORM_ANDROID)
 	list(APPEND PROJECT_LIBS android log GLESv3 EGL)
 elseif(TARGET_PLATFORM_IOS)
 	list(APPEND PROJECT_LIBS "-ObjC -lsqlite3 -framework OpenGLES")
-elseif(TARGET_PLATFORM_UNIX)
-	list(APPEND PROJECT_LIBS "-static-libgcc")
-	list(APPEND PROJECT_LIBS "-static-libstdc++")
-endif()
-
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -lstdc++fs")
-	list(APPEND PROJECT_LIBS "libstdc++fs.a")
 endif()
 
 add_library(play_libretro SHARED ${SRC})
