diff --git a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
index 610a14c..1a39543 100644
--- a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
+++ b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
@@ -9,7 +9,7 @@ set(CMAKE_MODULE_PATH
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if(UNIX OR WIN32 AND NOT ANDROID AND NOT APPLE AND NOT TARGET_PLATFORM_UNIX_AARCH64)
+if(UNIX OR WIN32 AND NOT ANDROID AND NOT APPLE)
 	if (NOT TARGET glew_s)
 		add_subdirectory(
 			${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/glew-2.0.0
diff --git a/deps/Framework/include/opengl/OpenGlDef.h b/deps/Framework/include/opengl/OpenGlDef.h
index 81f2b0a..b004e9e 100644
--- a/deps/Framework/include/opengl/OpenGlDef.h
+++ b/deps/Framework/include/opengl/OpenGlDef.h
@@ -33,7 +33,7 @@
 
 #endif
 
-#elif defined(__aarch64__) && defined(__linux__)
+#elif defined(__aarch64__) && defined(__linux__) && false
 
 #include <GLES3/gl3.h>
 #include <GL/gl.h>