diff --git a/Source/gs/GSH_OpenGL/CMakeLists.txt b/Source/gs/GSH_OpenGL/CMakeLists.txt
index 35cdc56f..1f81fdc7 100644
--- a/Source/gs/GSH_OpenGL/CMakeLists.txt
+++ b/Source/gs/GSH_OpenGL/CMakeLists.txt
@@ -26,6 +26,6 @@ add_library(gsh_opengl STATIC
 target_link_libraries(gsh_opengl Boost::boost Framework_OpenGl ${PROJECT_LIBS})
 target_include_directories(gsh_opengl PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/Source/gs/GSH_OpenGL/)
 
-if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS OR TARGET_PLATFORM_UNIX_AARCH64)
+if(TARGET_PLATFORM_ANDROID OR TARGET_PLATFORM_IOS)
 	target_compile_definitions(gsh_opengl PUBLIC GLES_COMPATIBILITY)
 endif()
