diff -Naur a/Makefile b/Makefile
--- a/Makefile	2014-10-07 14:00:34.000000000 -0700
+++ b/Makefile	2016-12-06 11:24:34.855522455 -0800
@@ -15,8 +15,8 @@
 install:
 	mkdir -p $(DESTDIR)/usr/bin
 	cp -P $(PROG) $(DESTDIR)/usr/bin
-	mkdir -p $(DESTDIR)/lib/firmware
-	cp -PR firmware/brcm $(DESTDIR)/lib/firmware/
+	mkdir -p $(DESTDIR)/usr/lib/firmware
+	cp -PR firmware/brcm $(DESTDIR)/usr/lib/firmware/
 
 clean:
 	rm -f $(PROG)
