diff -Naur fuse-exfat-0.9.7/SConstruct fuse-exfat-0.9.7.patch/SConstruct
--- fuse-exfat-0.9.7/SConstruct	2012-03-08 08:26:52.000000000 +0100
+++ fuse-exfat-0.9.7.patch/SConstruct	2012-07-17 19:59:49.650457625 +0200
@@ -33,6 +33,14 @@
 	conf.env.Replace(CC = os.environ['CC'])
 if 'CCFLAGS' in os.environ:
 	conf.env.Replace(CCFLAGS = os.environ['CCFLAGS'])
+if 'AR' in os.environ:
+	conf.env.Replace(AR = os.environ['AR'])
+if 'ARFLAGS' in os.environ:
+	conf.env.Replace(ARFLAGS = os.environ['ARFLAGS'])
+if 'RANLIB' in os.environ:
+	conf.env.Replace(RANLIB = os.environ['RANLIB'])
+if 'RANLIBFLAGS' in os.environ:
+	conf.env.Replace(RANLIBFLAGS = os.environ['RANLIBFLAGS'])
 # Set default CCFLAGS for known compilers
 if not conf.env['CCFLAGS']:
 	if conf.env['CC'] == 'gcc':
