diff -Naur crystalhd-20100118/linux_lib/libcrystalhd/Makefile crystalhd-20100118.patch/linux_lib/libcrystalhd/Makefile
--- crystalhd-20100118/linux_lib/libcrystalhd/Makefile	2010-01-07 00:08:04.000000000 +0100
+++ crystalhd-20100118.patch/linux_lib/libcrystalhd/Makefile	2010-01-18 11:16:05.835453470 +0100
@@ -15,7 +15,7 @@
 
 ROOTDIR = ../..
 
-INCLUDES = -I./ -I/usr/include -I$(ROOTDIR)/include
+INCLUDES = -I./ -I$(ROOTDIR)/include
 INCLUDES += -I$(ROOTDIR)/include/link
 
 
