diff -Naur crystalhd-171/darwin_lib/libcrystalhd/libcrystalhd_priv.cpp crystalhd-171.patch/darwin_lib/libcrystalhd/libcrystalhd_priv.cpp
--- crystalhd-171/darwin_lib/libcrystalhd/libcrystalhd_priv.cpp	2010-10-18 01:56:15.000000000 +0200
+++ crystalhd-171.patch/darwin_lib/libcrystalhd/libcrystalhd_priv.cpp	2012-01-03 22:26:10.974435389 +0100
@@ -1754,7 +1754,7 @@
 	char fwfile[MAX_PATH + 1];
 	char fwfilepath[MAX_PATH + 1];
 #ifndef __APPLE__
-	const char fwdir[] = "/lib/firmware/";
+	const char fwdir[] = "/usr/lib/firmware/";
 #else
 	const char fwdir[] = "/usr/lib/";
 #endif
diff -Naur crystalhd-171/linux_lib/libcrystalhd/libcrystalhd_priv.cpp crystalhd-171.patch/linux_lib/libcrystalhd/libcrystalhd_priv.cpp
--- crystalhd-171/linux_lib/libcrystalhd/libcrystalhd_priv.cpp	2010-10-18 01:56:14.000000000 +0200
+++ crystalhd-171.patch/linux_lib/libcrystalhd/libcrystalhd_priv.cpp	2012-01-03 22:26:36.489901837 +0100
@@ -1738,7 +1738,7 @@
 	char fwfile[MAX_PATH + 1];
 	char fwfilepath[MAX_PATH + 1];
 #ifndef __APPLE__
-	const char fwdir[] = "/lib/firmware/";
+	const char fwdir[] = "/usr/lib/firmware/";
 #else
 	const char fwdir[] = "/usr/lib/";
 #endif
