diff -Naur crystalhd-171/include/bc_dts_glob_lnx.h crystalhd-171.patch/include/bc_dts_glob_lnx.h
--- crystalhd-171/include/bc_dts_glob_lnx.h	2010-10-18 01:56:14.000000000 +0200
+++ crystalhd-171.patch/include/bc_dts_glob_lnx.h	2010-10-24 02:53:16.739741578 +0200
@@ -76,8 +76,8 @@
 	BC_LINK_MAX_OPENS	= 3,		/* Maximum simultaneous opens*/
 	BC_LINK_MAX_SGLS	= 1024,		/* Maximum SG elements 4M/4K */
 	BC_TX_LIST_CNT		= 2,		/* Max Tx DMA Rings */
-	//BC_RX_LIST_CNT	= 8,		/* Max Rx DMA Rings*/
-	BC_RX_LIST_CNT		= 16,		/* Max Rx DMA Rings*/
+	BC_RX_LIST_CNT		= 8,		/* Max Rx DMA Rings*/
+	//BC_RX_LIST_CNT	= 16,		/* Max Rx DMA Rings*/
 	BC_PROC_OUTPUT_TIMEOUT	= 2000,		/* Milliseconds */
 	BC_INFIFO_THRESHOLD	= 0x10000,
 };
