diff -Naur crystalhd-3cb6786/include/bc_dts_glob_lnx.h crystalhd-3cb6786.patch/include/bc_dts_glob_lnx.h
--- crystalhd-3cb6786/include/bc_dts_glob_lnx.h	2012-12-08 03:31:38.000000000 +0100
+++ crystalhd-3cb6786.patch/include/bc_dts_glob_lnx.h	2012-12-08 04:50:12.004056345 +0100
@@ -76,7 +76,7 @@
 	BC_LINK_MAX_OPENS	= 3,		/* Maximum simultaneous opens*/
 	BC_LINK_MAX_SGLS	= 1024,		/* Maximum SG elements 4M/4K */
 	BC_TX_LIST_CNT		= 2,		/* Max Tx DMA Rings */
-	BC_RX_LIST_CNT		= 16,		/* Max Rx DMA Rings*/
+	BC_RX_LIST_CNT		= 8,		/* Max Rx DMA Rings*/
 	BC_PROC_OUTPUT_TIMEOUT	= 2000,		/* Milliseconds */
 	BC_INFIFO_THRESHOLD	= 0x10000,
 };
