diff -Naur crystalhd-3.8.0/driver/linux/crystalhd_flea_ddr.c crystalhd-3.8.0.patch/driver/linux/crystalhd_flea_ddr.c
--- crystalhd-3.8.0/driver/linux/crystalhd_flea_ddr.c	2010-10-06 00:07:16.000000000 +0200
+++ crystalhd-3.8.0.patch/driver/linux/crystalhd_flea_ddr.c	2010-10-06 01:04:11.844215460 +0200
@@ -24,6 +24,7 @@
  * along with this driver.  If not, see <http://www.gnu.org/licenses/>.
  **********************************************************************/
 
+#include <linux/delay.h>
 #include "crystalhd_hw.h"
 #include "crystalhd_flea_ddr.h"
 
diff -Naur crystalhd-3.8.0/driver/linux/crystalhd_lnx.c crystalhd-3.8.0.patch/driver/linux/crystalhd_lnx.c
--- crystalhd-3.8.0/driver/linux/crystalhd_lnx.c	2010-10-06 00:07:16.000000000 +0200
+++ crystalhd-3.8.0.patch/driver/linux/crystalhd_lnx.c	2010-10-06 01:04:31.543484731 +0200
@@ -279,8 +279,13 @@
 }
 
 /* API interfaces */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 35)
 static int chd_dec_ioctl(struct inode *in, struct file *fd,
 			 unsigned int cmd, unsigned long ua)
+#else
+static long irctl_ioctl(struct file *file,
+			 unsigned int cmd, unsigned long ua)
+#endif
 {
 	struct crystalhd_adp *adp = chd_get_adp();
 	crystalhd_cmd_proc cproc;
@@ -365,7 +370,11 @@
 
 static const struct file_operations chd_dec_fops = {
 	.owner   = THIS_MODULE,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 35)
 	.ioctl   = chd_dec_ioctl,
+#else
+	.unlocked_ioctl = chd_dec_ioctl,
+#endif
 	.open    = chd_dec_open,
 	.release = chd_dec_close,
 };
