diff -Naur crystalhd-3cb6786/linux_lib/libcrystalhd/libcrystalhd_fwdiag_if.cpp crystalhd-3cb6786.patch/linux_lib/libcrystalhd/libcrystalhd_fwdiag_if.cpp
--- crystalhd-3cb6786/linux_lib/libcrystalhd/libcrystalhd_fwdiag_if.cpp	2012-12-08 03:31:38.000000000 +0100
+++ crystalhd-3cb6786.patch/linux_lib/libcrystalhd/libcrystalhd_fwdiag_if.cpp	2012-12-08 04:48:46.572661023 +0100
@@ -233,7 +233,7 @@
 
 		}
 		else
-			DebugLog_Trace(LDIL_DBG,"Uart Set Sucessfully\n");
+			DebugLog_Trace(LDIL_DBG,"Uart Set Successfully\n");
 
 		//START_PROCESSOR bit in DCI_CMD.
 		RegVal = 0;
