diff -Naur crystalhd-171/driver/linux/crystalhd_lnx.h crystalhd-171.patch/driver/linux/crystalhd_lnx.h
--- crystalhd-171/driver/linux/crystalhd_lnx.h	2010-10-18 01:56:15.000000000 +0200
+++ crystalhd-171.patch/driver/linux/crystalhd_lnx.h	2012-07-24 00:17:05.257202412 +0200
@@ -45,7 +45,10 @@
 #include <linux/io.h>
 #include <asm/irq.h>
 #include <asm/pgtable.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 4, 0)
 #include <asm/system.h>
+#endif
 #include <linux/uaccess.h>
 
 #include "crystalhd_cmds.h"
diff -Naur crystalhd-171/driver/linux/crystalhd_misc.h crystalhd-171.patch/driver/linux/crystalhd_misc.h
--- crystalhd-171/driver/linux/crystalhd_misc.h	2010-10-18 01:56:15.000000000 +0200
+++ crystalhd-171.patch/driver/linux/crystalhd_misc.h	2012-07-24 00:16:57.091091576 +0200
@@ -35,7 +35,10 @@
 #include <linux/ioctl.h>
 #include <linux/dma-mapping.h>
 #include <linux/sched.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 4, 0)
 #include <asm/system.h>
+#endif
 #include "bc_dts_glob_lnx.h"
 #include "crystalhd_hw.h"
 
