diff -Naur ffmpeg-0.10.7/libavcodec/h264.c ffmpeg-0.10.7.patch/libavcodec/h264.c
--- ffmpeg-0.10.7/libavcodec/h264.c	2013-04-10 03:52:33.000000000 +0200
+++ ffmpeg-0.10.7.patch/libavcodec/h264.c	2013-09-09 17:25:55.377208927 +0200
@@ -3962,8 +3962,7 @@
             continue;
         }
 
-        //FIXME do not discard SEI id
-        if(avctx->skip_frame >= AVDISCARD_NONREF && h->nal_ref_idc  == 0)
+        if (avctx->skip_frame >= AVDISCARD_NONREF && h->nal_ref_idc == 0 && h->nal_unit_type != NAL_SEI)
             continue;
 
       again:
