--- plymouth-lite-0.6.0/Makefile~	2009-07-22 11:22:32.000000000 -0700
+++ plymouth-lite-0.6.0/Makefile	2009-07-22 11:22:32.000000000 -0700
@@ -1,6 +1,6 @@
 
 ply-image: ply-image.c ply-frame-buffer.c Makefile
-	gcc -O2 -march=core2 -mtune=generic -lm `pkg-config --cflags libpng12` `pkg-config --libs libpng12` ply-image.c ply-frame-buffer.c -o ply-image
+	$(CC) $(CFLAGS) `pkg-config --cflags libpng`  ply-image.c ply-frame-buffer.c -o ply-image  -lm `pkg-config --libs libpng` -lm -lz
 	
 clean:
 	rm -f ply-image *~ gmon.out
