From ba71a5f3778df1bb19ba593f6a5550c01f5349ad Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Tue, 28 Jan 2020 03:17:14 +0000
Subject: [PATCH] fix API change since autoconf-archive-2019.01.06

---
 Makefile.am | 2 +-
 Makefile.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 723460f2e..a6a0f5f9c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -408,7 +408,7 @@ strace_LDADD += $(libiberty_LIBS)
 endif
 endif
 
-@CODE_COVERAGE_RULES@
+include $(top_srcdir)/aminclude_static.am
 CODE_COVERAGE_BRANCH_COVERAGE = 1
 CODE_COVERAGE_GENHTML_OPTIONS = $(CODE_COVERAGE_GENHTML_OPTIONS_DEFAULT) \
 	--prefix $(shell cd $(abs_top_srcdir)/.. && pwd || echo .)
diff --git a/Makefile.in b/Makefile.in
index b0ec9f686..9b4b93d8b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -10351,7 +10351,7 @@ $(top_srcdir)/xlat/xfs_dqblk_flags.h: $(top_srcdir)/xlat/xfs_dqblk_flags.in $(to
 $(top_srcdir)/xlat/xfs_quota_flags.h: $(top_srcdir)/xlat/xfs_quota_flags.in $(top_srcdir)/xlat/gen.sh
 	$(AM_V_GEN)$(top_srcdir)/xlat/gen.sh $< $@
 
-@CODE_COVERAGE_RULES@
+include $(top_srcdir)/aminclude_static.am
 
 .PHONY: check-valgrind-local
 check-valgrind-local:
-- 
2.7.4

