--- a/configure.ac
+++ b/configure.ac
@@ -31,11 +31,7 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 AC_PREREQ(2.57)
-AC_INIT([strace],
-	m4_esyscmd([./git-version-gen .tarball-version]),
-	[strace-devel@lists.sourceforge.net],
-	[strace],
-	[https://strace.io])
+AC_INIT([strace],[4.17])
 m4_define([copyright_year], m4_esyscmd([./copyright-year-gen .year]))
 AC_COPYRIGHT([Copyright (C) 1999-]copyright_year[ The strace developers.])
 AC_CONFIG_SRCDIR([strace.c])
