From 2d8dab16808ab6ceda83c33e2f662ca07271c288 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Sun, 16 Feb 2020 10:33:04 +0000
Subject: [PATCH] Disable building examples, fails due to deprecated methods

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a7d3ef..24f518d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,4 +228,4 @@ add_subdirectory( src )
 add_subdirectory( include )
 
 #install the example
-add_subdirectory( example )
+#add_subdirectory( example )
-- 
2.7.4

