diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2015-10-07 00:43:15.000000000 -0700
+++ b/Makefile.am	2016-04-22 14:16:07.656866841 -0700
@@ -11,21 +11,6 @@
 if USE_LINUX
 udevrulesdir=@UDEV@/rules.d
 hwdbdir=@UDEV@/hwdb.d
-udevrules_DATA=@UDEV_RULES@
-hwdb_DATA=69-libmtp.hwdb
-noinst_DATA=libmtp.usermap libmtp.fdi
-
-libmtp.usermap: util/mtp-hotplug
-	util/mtp-hotplug > libmtp.usermap
-
-@UDEV_RULES@: util/mtp-hotplug
-	util/mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
-
-libmtp.fdi: util/mtp-hotplug
-	util/mtp-hotplug -H > libmtp.fdi
-
-$(hwdb_DATA): util/mtp-hotplug
-	util/mtp-hotplug -w > $(hwdb_DATA)
 
 CLEANFILES = libmtp.usermap @UDEV_RULES@ libmtp.fdi libmtp.hwdb
 endif
diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2016-02-10 12:17:34.000000000 -0800
+++ b/Makefile.in	2016-04-22 14:14:22.638794537 -0700
@@ -161,7 +161,7 @@
   }
 am__installdirs = "$(DESTDIR)$(hwdbdir)" "$(DESTDIR)$(pkgconfigdir)" \
 	"$(DESTDIR)$(udevrulesdir)"
-DATA = $(hwdb_DATA) $(noinst_DATA) $(pkgconfig_DATA) $(udevrules_DATA)
+DATA = $(pkgconfig_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 am__recursive_targets = \
@@ -452,21 +452,8 @@
 
 distclean-libtool:
 	-rm -f libtool config.lt
-install-hwdbDATA: $(hwdb_DATA)
-	@$(NORMAL_INSTALL)
-	@list='$(hwdb_DATA)'; test -n "$(hwdbdir)" || list=; \
-	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(hwdbdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(hwdbdir)" || exit 1; \
-	fi; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(am__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(hwdbdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(hwdbdir)" || exit $$?; \
-	done
+install-hwdbDATA:
+	echo nope
 
 uninstall-hwdbDATA:
 	@$(NORMAL_UNINSTALL)
@@ -871,7 +858,7 @@
 
 info-am:
 
-install-data-am: install-hwdbDATA install-pkgconfigDATA \
+install-data-am: install-pkgconfigDATA \
 	install-udevrulesDATA
 
 install-dvi: install-dvi-recursive
@@ -932,7 +919,7 @@
 	distdir distuninstallcheck dvi dvi-am html html-am info \
 	info-am install install-am install-data install-data-am \
 	install-dvi install-dvi-am install-exec install-exec-am \
-	install-html install-html-am install-hwdbDATA install-info \
+	install-html install-html-am install-info \
 	install-info-am install-man install-pdf install-pdf-am \
 	install-pkgconfigDATA install-ps install-ps-am install-strip \
 	install-udevrulesDATA installcheck installcheck-am installdirs \
@@ -944,19 +931,6 @@
 
 .PRECIOUS: Makefile
 
-
-@USE_LINUX_TRUE@libmtp.usermap: util/mtp-hotplug
-@USE_LINUX_TRUE@	util/mtp-hotplug > libmtp.usermap
-
-@USE_LINUX_TRUE@@UDEV_RULES@: util/mtp-hotplug
-@USE_LINUX_TRUE@	util/mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
-
-@USE_LINUX_TRUE@libmtp.fdi: util/mtp-hotplug
-@USE_LINUX_TRUE@	util/mtp-hotplug -H > libmtp.fdi
-
-@USE_LINUX_TRUE@$(hwdb_DATA): util/mtp-hotplug
-@USE_LINUX_TRUE@	util/mtp-hotplug -w > $(hwdb_DATA)
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
