diff -Naur linux-3.9.4/tools/perf/Makefile linux-3.9.4.patch/tools/perf/Makefile
--- linux-3.9.4/tools/perf/Makefile	2013-05-24 20:45:59.000000000 +0200
+++ linux-3.9.4.patch/tools/perf/Makefile	2013-05-30 23:57:04.135031372 +0200
@@ -668,12 +668,10 @@
 endif
 
 ifndef NO_NEWT
-	FLAGS_NEWT=$(ALL_CFLAGS) $(ALL_LDFLAGS) $(EXTLIBS) -lnewt
+	FLAGS_NEWT=$(ALL_CFLAGS) $(ALL_LDFLAGS) $(EXTLIBS) -lnewt -lslang
 	ifneq ($(call try-cc,$(SOURCE_NEWT),$(FLAGS_NEWT),libnewt),y)
 		msg := $(warning newt not found, disables TUI support. Please install newt-devel or libnewt-dev);
 	else
-		# Fedora has /usr/include/slang/slang.h, but ubuntu /usr/include/slang.h
-		BASIC_CFLAGS += -I/usr/include/slang
 		BASIC_CFLAGS += -DNEWT_SUPPORT
 		EXTLIBS += -lnewt -lslang
 		LIB_OBJS += $(OUTPUT)ui/browser.o
